--- lotuswordpro/source/filter/xfilter/xffontfactory.hxx.old	2010-03-10 21:41:47.000000000 +0100
+++ lotuswordpro/source/filter/xfilter/xffontfactory.hxx	2010-03-11 14:28:02.000000000 +0100
@@ -80,8 +80,6 @@ public:
 	~XFFontFactory();
 	XFFontFactory(const XFFontFactory& other){}
 
-	XFFontFactory operator=(XFFontFactory& other){}	
-
 public:	
 	/**
 	 * @descr	Clear all fonts, this is called when load a file. 
--- lotuswordpro/source/filter/xfilter/xfstylemanager.hxx.old	2010-03-10 21:41:47.000000000 +0100
+++ lotuswordpro/source/filter/xfilter/xfstylemanager.hxx	2010-03-11 14:29:45.000000000 +0100
@@ -91,8 +91,6 @@ public:
 	~XFStyleManager();
 	XFStyleManager(XFStyleManager& other){}
 
-	XFStyleManager operator=(XFStyleManager& other){}	
-
 public:	
 	/**
 	 * @descr	clear all styles, called before load file. 
