From 04119677a711140f32208f8e1f6d6e2931c002fd Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:00:40 +0200
Subject: [PATCH 359/768] buildfix-oox-depends-on-unotools.diff

---
 oox/prj/build.lst |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git oox/prj/build.lst oox/prj/build.lst
index a1dfa58..d0e446f 100644
--- oox/prj/build.lst
+++ oox/prj/build.lst
@@ -1,4 +1,4 @@
-oox	oox : vos cppu cppuhelper comphelper sal offapi sax basegfx tools vcl BOOST:boost OPENSSL:openssl NULL
+oox	oox : vos cppu cppuhelper comphelper sal offapi sax basegfx tools vcl unotools BOOST:boost OPENSSL:openssl NULL
 oox	oox				usr1	-   all	oox_mkout NULL
 oox	oox\prj				get	-   all	oox_prj NULL
 oox	oox\source\token		nmake	-   all	oox_token NULL
-- 
1.7.0.1

