---
 oox/prj/build.lst |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git oox/prj/build.lst oox/prj/build.lst
index 345c72e..60c4b02 100644
--- oox/prj/build.lst
+++ oox/prj/build.lst
@@ -1,4 +1,4 @@
-oox	oox : cppu cppuhelper comphelper filter sal offapi sax basegfx svx xmlscript tools vcl BOOST:boost OPENSSL:openssl NULL
+oox	oox : cppu cppuhelper comphelper filter sal offapi sax basegfx svx xmlscript tools vcl unotools BOOST:boost OPENSSL:openssl NULL
 oox	oox				usr1	-   all	oox_mkout NULL
 oox	oox\prj				get	-   all	oox_prj NULL
 oox	oox\source\token		nmake	-   all	oox_token NULL
