From d87a6274bb403d9408e6671e4339178057298ce9 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:53:21 +0200
Subject: [PATCH 048/878] writerfilter-qnametostr-NOOPTFILES.diff

---
 writerfilter/source/resourcemodel/makefile.mk |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/writerfilter/source/resourcemodel/makefile.mk b/writerfilter/source/resourcemodel/makefile.mk
index f486287..fda7afd 100644
--- a/writerfilter/source/resourcemodel/makefile.mk
+++ b/writerfilter/source/resourcemodel/makefile.mk
@@ -48,6 +48,9 @@ CDEFS+=-DWRITERFILTER_DLLIMPLEMENTATION
 NOOPTFILES= \
     $(SLO)$/qnametostr.obj
 
+NOOPTFILES= \
+    $(SLO)$/qnametostr.obj
+
 SLOFILES= \
     $(SLO)$/qnametostr.obj \
     $(SLO)$/sprmcodetostr.obj \
-- 
1.7.0.1

