From 555565497f4f170bc2d826b087f7572a67cb04c4 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:00:33 +0200
Subject: [PATCH 356/768] writerfilter-source-dmapper-domainmapper-debug.diff

---
 writerfilter/source/dmapper/DomainMapper.cxx |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git writerfilter/source/dmapper/DomainMapper.cxx writerfilter/source/dmapper/DomainMapper.cxx
index df93153..643f1c5 100755
--- writerfilter/source/dmapper/DomainMapper.cxx
+++ writerfilter/source/dmapper/DomainMapper.cxx
@@ -102,6 +102,9 @@ namespace dmapper{
 #ifdef DEBUG_DOMAINMAPPER
 TagLogger::Pointer_t dmapper_logger(TagLogger::getInstance("DOMAINMAPPER"));
 #endif
+#if OSL_DEBUG_LEVEL > 0
+#include <resourcemodel/QNameToString.hxx>
+#endif
 
 /* ---- Fridrich's mess begins here ---- */
 struct _PageSz
-- 
1.7.0.1

