From aaee18bc5af161b0f73bd04d88c68336fade8aec Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:53:21 +0200
Subject: [PATCH 034/768] writerfilter-qnametostr-NOOPTFILES.diff

---
 writerfilter/source/resourcemodel/makefile.mk |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git writerfilter/source/resourcemodel/makefile.mk writerfilter/source/resourcemodel/makefile.mk
index f3869f3..4053ae0 100644
--- writerfilter/source/resourcemodel/makefile.mk
+++ writerfilter/source/resourcemodel/makefile.mk
@@ -48,6 +48,9 @@ CDEFS+=-DWRITERFILTER_DLLIMPLEMENTATION
 NOOPTFILES= \
     $(SLO)$/qnametostr.obj
 
+NOOPTFILES= \
+    $(SLO)$/qnametostr.obj
+
 SLOFILES= \
     $(SLO)$/qnametostr.obj \
     $(SLO)$/sprmcodetostr.obj \
-- 
1.7.0.1

