From f1377f0199b5f92fbc162df07a6d0c29b1b77246 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:00:33 +0200
Subject: [PATCH 413/878] writerfilter-source-dmapper-domainmapper-debug.diff

---
 writerfilter/source/dmapper/DomainMapper.cxx |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/writerfilter/source/dmapper/DomainMapper.cxx b/writerfilter/source/dmapper/DomainMapper.cxx
index f6069f2..6ad2c38 100644
--- a/writerfilter/source/dmapper/DomainMapper.cxx
+++ b/writerfilter/source/dmapper/DomainMapper.cxx
@@ -102,6 +102,9 @@ namespace dmapper{
 #ifdef DEBUG_DOMAINMAPPER
 TagLogger::Pointer_t dmapper_logger(TagLogger::getInstance("DOMAINMAPPER"));
 #endif
+#if OSL_DEBUG_LEVEL > 0
+#include <resourcemodel/QNameToString.hxx>
+#endif
 
 /* ---- Fridrich's mess begins here ---- */
 struct _PageSz
-- 
1.7.0.1

