---
 writerfilter/source/resourcemodel/makefile.mk |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git writerfilter/source/resourcemodel/makefile.mk writerfilter/source/resourcemodel/makefile.mk
index f3869f3..4053ae0 100644
--- writerfilter/source/resourcemodel/makefile.mk
+++ writerfilter/source/resourcemodel/makefile.mk
@@ -48,6 +48,9 @@ CDEFS+=-DWRITERFILTER_DLLIMPLEMENTATION
 NOOPTFILES= \
     $(SLO)$/qnametostr.obj
 
+NOOPTFILES= \
+    $(SLO)$/qnametostr.obj
+
 SLOFILES= \
     $(SLO)$/qnametostr.obj \
     $(SLO)$/sprmcodetostr.obj \
-- 
1.7.0.1

