--- sd/source/ui/view/Outliner.cxx
+++ sd/source/ui/view/Outliner.cxx
@@ -501,17 +501,19 @@ void Outliner::HandleOutsideChange (Chan
     bool bFoundNextSentence = false;
     while ( ! bFoundNextSentence)
     {
-        ESelection aCurrentSelection (GetView(0)->GetSelection());
-        if ( ! mbMatchMayExist 
-            && maStartSelection.IsLess(aCurrentSelection))
-            EndOfSearch();
-
-        // Advance to the next sentence.
         OutlinerView* pOutlinerView = GetView(0);
         if (pOutlinerView != NULL)
+        {
+            ESelection aCurrentSelection (pOutlinerView->GetSelection());
+            if ( ! mbMatchMayExist 
+                && maStartSelection.IsLess(aCurrentSelection))
+                EndOfSearch();
+            
+            // Advance to the next sentence.
             bFoundNextSentence = SpellSentence (
                 pOutlinerView->GetEditView(),
                 aResult);
+        }
 
         // When no sentence with spelling errors has been found in the
         // currently selected text shape or there is no selected text
