---
 sfx2/source/dialog/about.cxx |    5 -----
 1 files changed, 0 insertions(+), 5 deletions(-)

diff --git sfx2/source/dialog/about.cxx sfx2/source/dialog/about.cxx
index a2677e0..044ae7b 100644
--- sfx2/source/dialog/about.cxx
+++ sfx2/source/dialog/about.cxx
@@ -233,11 +233,6 @@ AboutDialog::AboutDialog( Window* pParent, const ResId& rId, const String& rVerS
     aCopyrightText.SetControlForeground( aTextColor );
     aBuildData.SetBackground( aWall );
 
-    Font aSmallFont = rSettings.GetInfoFont();
-    Size aSmaller = aNewFont.GetSize();
-    aSmaller.Width() = (long) (aSmaller.Width() * 0.75);
-    aSmaller.Height() = (long) (aSmaller.Height() * 0.75);
-    aNewFont.SetSize( aSmaller );
     aBuildData.SetFont( aNewFont );
     aBuildData.SetBackground( aWall );
     // FIXME: What is the purpose of the build data?
-- 
1.7.0.1

