From 57f0f7279efdf040764954da5e257db6cc6eb3eb Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:58:53 +0200
Subject: [PATCH 342/878] about-build-ver-readability.diff

---
 sfx2/source/dialog/about.cxx |    5 -----
 1 files changed, 0 insertions(+), 5 deletions(-)

diff --git a/sfx2/source/dialog/about.cxx b/sfx2/source/dialog/about.cxx
index bb52946..4e3762c 100644
--- a/sfx2/source/dialog/about.cxx
+++ b/sfx2/source/dialog/about.cxx
@@ -223,11 +223,6 @@ AboutDialog::AboutDialog( Window* pParent, const ResId& rId, const String& rVerS
     aCopyrightText.SetControlForeground( aTextColor );
     aBuildData.SetBackground( aWall );
 
-    Font aSmallFont = rSettings.GetInfoFont();
-    Size aSmaller = aNewFont.GetSize();
-    aSmaller.Width() = (long) (aSmaller.Width() * 0.75);
-    aSmaller.Height() = (long) (aSmaller.Height() * 0.75);
-    aNewFont.SetSize( aSmaller );
     aBuildData.SetFont( aNewFont );
     aBuildData.SetBackground( aWall );
     // FIXME: What is the purpose of the build data?
-- 
1.7.0.1

