diff --git sfx2/source/doc/guisaveas.cxx sfx2/source/doc/guisaveas.cxx
index d0b40db..43479b1 100644
--- sfx2/source/doc/guisaveas.cxx
+++ sfx2/source/doc/guisaveas.cxx
@@ -68,6 +68,7 @@
 #include <svtools/adrparse.hxx>
 #include <svtools/useroptions.hxx>
 #include <svtools/saveopt.hxx>
+#include "svtools/miscopt.hxx"
 #include <tools/debug.hxx>
 #include <tools/urlobj.hxx>
 #include <comphelper/processfactory.hxx>
@@ -627,9 +628,14 @@ sal_Int8 ModelData_Impl::CheckStateForSave()
     if ( GetMediaDescr().size() != aAcceptedArgs.size() )
         GetMediaDescr() = aAcceptedArgs;
 
-    // the document must be modified
-    if ( !GetModifiable()->isModified() && !bVersInfoNeedsStore )
-		return STATUS_NO_ACTION;
+    // the document must be modified unless the always-save flag is set.
+    SvtMiscOptions aMiscOptions;
+    sal_Bool bAlwaysAllowSave = aMiscOptions.IsSaveAlwaysAllowed();
+    if (!bAlwaysAllowSave)
+    {
+        if ( !GetModifiable()->isModified() && !bVersInfoNeedsStore )
+            return STATUS_NO_ACTION;
+    }
 
     // if the document is readonly or a new one a SaveAs operation must be used
     if ( !GetStorable()->hasLocation() || GetStorable()->isReadonly() )
diff --git sfx2/source/doc/objserv.cxx sfx2/source/doc/objserv.cxx
index 801aad7..d887e73 100644
--- sfx2/source/doc/objserv.cxx
+++ sfx2/source/doc/objserv.cxx
@@ -68,6 +68,7 @@
 #include <svtools/useroptions.hxx>
 #include <svtools/asynclink.hxx>
 #include <svtools/saveopt.hxx>
+#include "svtools/miscopt.hxx"
 #include <comphelper/documentconstants.hxx>
 
 #include <sfx2/app.hxx>
@@ -968,8 +969,11 @@ void SfxObjectShell::GetState_Impl(SfxItemSet &rSet)
                 }
             case SID_SAVEDOC:
                 {
-					BOOL bMediumRO = IsReadOnlyMedium();
-                    if ( !bMediumRO && GetMedium() && IsModified() )
+                    SvtMiscOptions aMiscOptions;
+                    bool bAlwaysAllowSave = aMiscOptions.IsSaveAlwaysAllowed();
+                    bool bAllowSave = (bAlwaysAllowSave || IsModified());
+                    bool bMediumRO = IsReadOnlyMedium();                                                  
+                    if ( !bMediumRO && GetMedium() && bAllowSave )
                         rSet.Put(SfxStringItem(
                             nWhich, String(SfxResId(STR_SAVEDOC))));
                     else
