diff --git officecfg/registry/schema/org/openoffice/Office/Common.xcs officecfg/registry/schema/org/openoffice/Office/Common.xcs
index d158081..055b62f 100644
--- officecfg/registry/schema/org/openoffice/Office/Common.xcs
+++ officecfg/registry/schema/org/openoffice/Office/Common.xcs
@@ -5765,6 +5765,13 @@ Dymamic border coloring means that when the mouse is hovered over a control, and
 				</info>
 				<value>false</value>
 			</prop>
+            <prop oor:name="AlwaysAllowSave" oor:type="xs:boolean">
+                <info>
+                    <author>kohei</author>
+                    <desc>Determines if the user can save the document even when it's not modified.</desc>
+                </info>
+                <value>false</value>
+            </prop>
 			<prop oor:name="SymbolSet" oor:type="xs:short">
 				<!-- UIHints: Tools  Options General View -->
 				<info>
