diff --git svtools/inc/svtools/miscopt.hxx svtools/inc/svtools/miscopt.hxx
index e7a9511..2146df7 100644
--- svtools/inc/svtools/miscopt.hxx
+++ svtools/inc/svtools/miscopt.hxx
@@ -149,6 +149,9 @@ class SVT_DLLPUBLIC SvtMiscOptions: public svt::detail::Options
         void        SetShowLinkWarningDialog( sal_Bool bSet );
         sal_Bool    IsShowLinkWarningDialogReadOnly() const;
 
+        void        SetSaveAlwaysAllowed( sal_Bool bSet );
+        sal_Bool    IsSaveAlwaysAllowed() const;
+
     //-------------------------------------------------------------------------------------------------------------
     //	private methods
     //-------------------------------------------------------------------------------------------------------------
diff --git svtools/source/config/miscopt.cxx svtools/source/config/miscopt.cxx
index 49e20ed..a4250bb 100644
--- svtools/source/config/miscopt.cxx
+++ svtools/source/config/miscopt.cxx
@@ -91,8 +91,10 @@ using namespace ::com::sun::star;
 #define PROPERTYHANDLE_SHOWLINKWARNINGDIALOG 7
 #define PROPERTYNAME_DISABLEUICUSTOMIZATION	ASCII_STR("DisableUICustomization")
 #define PROPERTYHANDLE_DISABLEUICUSTOMIZATION			8
+#define PROPERTYNAME_ALWAYSALLOWSAVE ASCII_STR("AlwaysAllowSave")
+#define PROPERTYHANDLE_ALWAYSALLOWSAVE 9
 
-#define PROPERTYCOUNT						9
+#define PROPERTYCOUNT                       10
 
 #define VCL_TOOLBOX_STYLE_FLAT				((USHORT)0x0004) // from <vcl/toolbox.hxx>
 
@@ -126,6 +128,7 @@ class SvtMiscOptions_Impl : public ConfigItem
     sal_Bool    m_bShowLinkWarningDialog;
     sal_Bool    m_bIsShowLinkWarningDialogRO;
     sal_Bool    m_bDisableUICustomization;
+    sal_Bool    m_bAlwaysAllowSave;
 
     //-------------------------------------------------------------------------------------------------------------
     //	public methods
@@ -206,6 +209,12 @@ class SvtMiscOptions_Impl : public ConfigItem
         inline sal_Bool DisableUICustomization() const
         { return m_bDisableUICustomization; }
 
+        inline void SetSaveAlwaysAllowed( sal_Bool bSet )
+        { m_bAlwaysAllowSave = bSet; SetModified(); }
+
+        inline sal_Bool IsSaveAlwaysAllowed() const
+        { return m_bAlwaysAllowSave; }
+
         inline sal_Bool IsPluginsEnabled() const
         { return m_bPluginsEnabled; }
 
@@ -352,6 +361,7 @@ SvtMiscOptions_Impl::SvtMiscOptions_Impl()
     , m_bIsUseSystemPrintDialogRO( sal_False )
     , m_bShowLinkWarningDialog( sal_True )
     , m_bIsShowLinkWarningDialogRO( sal_False )
+    , m_bAlwaysAllowSave( sal_False )
 
 {
     // Use our static list of configuration keys to get his values.
@@ -463,6 +473,12 @@ SvtMiscOptions_Impl::SvtMiscOptions_Impl()
                     DBG_ERROR("Wrong type of \"Misc\\DisableUICustomization\"!" );
                     break;
             }
+            case PROPERTYHANDLE_ALWAYSALLOWSAVE : 
+            {
+                if( !(seqValues[nProperty] >>= m_bAlwaysAllowSave) )
+                    DBG_ERROR("Wrong type of \"Misc\\AlwaysAllowSave\"!" );
+                    break;
+            }
         }
     }
 
@@ -583,6 +599,12 @@ void SvtMiscOptions_Impl::Load( const Sequence< OUString >& rPropertyNames )
                                                                 DBG_ERROR("Wrong type of \"Misc\\DisableUICustomization\"!" );
                                                         }
                                                     break;
+            case PROPERTYHANDLE_ALWAYSALLOWSAVE:
+            {
+                if( !(seqValues[nProperty] >>= m_bAlwaysAllowSave) )
+                    DBG_ERROR("Wrong type of \"Misc\\AlwaysAllowSave\"!" );
+            }
+            break;
         }
     }
 }
@@ -752,6 +774,11 @@ void SvtMiscOptions_Impl::Commit()
                 seqValues[nProperty] <<= m_bDisableUICustomization;
                 break;
             }
+            case PROPERTYHANDLE_ALWAYSALLOWSAVE :
+            {
+                seqValues[nProperty] <<= m_bAlwaysAllowSave;
+                break;
+            }
         }
     }
     // Set properties in configuration.
@@ -774,7 +801,8 @@ Sequence< OUString > SvtMiscOptions_Impl::GetPropertyNames()
         PROPERTYNAME_USESYSTEMPRINTDIALOG,
         PROPERTYNAME_TRYODMADIALOG,
         PROPERTYNAME_SHOWLINKWARNINGDIALOG,
-        PROPERTYNAME_DISABLEUICUSTOMIZATION
+        PROPERTYNAME_DISABLEUICUSTOMIZATION,
+        PROPERTYNAME_ALWAYSALLOWSAVE
     };
 
     // Initialize return sequence with these list ...
@@ -980,6 +1008,16 @@ sal_Bool SvtMiscOptions::IsShowLinkWarningDialogReadOnly() const
     return m_pDataContainer->IsShowLinkWarningDialogReadOnly();
 }
 
+void SvtMiscOptions::SetSaveAlwaysAllowed( sal_Bool bSet )
+{
+    m_pDataContainer->SetSaveAlwaysAllowed( bSet );
+}
+
+sal_Bool SvtMiscOptions::IsSaveAlwaysAllowed() const
+{
+    return m_pDataContainer->IsSaveAlwaysAllowed();
+}
+
 //*****************************************************************************************************************
 //	private method
 //*****************************************************************************************************************
