--- sfx2/source/doc/guisaveas.cxx.old	2010-07-26 15:06:38.000000000 +0200
+++ sfx2/source/doc/guisaveas.cxx	2010-07-30 13:34:56.000000000 +0200
@@ -67,6 +67,7 @@
 #include <svl/intitem.hxx>
 #include <unotools/useroptions.hxx>
 #include <unotools/saveopt.hxx>
+#include <svtools/miscopt.hxx>
 #include <tools/debug.hxx>
 #include <tools/urlobj.hxx>
 #include <comphelper/processfactory.hxx>
@@ -696,9 +697,14 @@ sal_Int8 ModelData_Impl::CheckStateForSa
     if ( GetMediaDescr().size() != aAcceptedArgs.size() )
         GetMediaDescr() = aAcceptedArgs;
 
-    // the document must be modified
-    if ( !GetModifiable()->isModified() && !bVersInfoNeedsStore )
-        return STATUS_NO_ACTION;
+    // the document must be modified unless the always-save flag is set.
+    SvtMiscOptions aMiscOptions;
+    sal_Bool bAlwaysAllowSave = aMiscOptions.IsSaveAlwaysAllowed();
+    if (!bAlwaysAllowSave)
+    {
+        if ( !GetModifiable()->isModified() && !bVersInfoNeedsStore )
+            return STATUS_NO_ACTION;
+    }
 
     // if the document is readonly or a new one a SaveAs operation must be used
     if ( !GetStorable()->hasLocation() || GetStorable()->isReadonly() )
--- sfx2/source/doc/objserv.cxx.old	2010-07-29 21:20:54.000000000 +0200
+++ sfx2/source/doc/objserv.cxx	2010-07-30 13:34:56.000000000 +0200
@@ -63,8 +63,9 @@
 #include <basic/sbx.hxx>
 #include <unotools/pathoptions.hxx>
 #include <unotools/useroptions.hxx>
-#include <svtools/asynclink.hxx>
 #include <unotools/saveopt.hxx>
+#include <svtools/asynclink.hxx>
+#include <svtools/miscopt.hxx>
 #include <comphelper/documentconstants.hxx>
 
 #include <sfx2/app.hxx>
@@ -936,8 +937,11 @@ void SfxObjectShell::GetState_Impl(SfxIt
                 }
             case SID_SAVEDOC:
                 {
-                    BOOL bMediumRO = IsReadOnlyMedium();
-                    if ( !bMediumRO && GetMedium() && IsModified() )
+                    SvtMiscOptions aMiscOptions;
+                    bool bAlwaysAllowSave = aMiscOptions.IsSaveAlwaysAllowed();
+                    bool bAllowSave = (bAlwaysAllowSave || IsModified());
+                    bool bMediumRO = IsReadOnlyMedium();
+                    if ( !bMediumRO && GetMedium() && bAllowSave )
                         rSet.Put(SfxStringItem(
                             nWhich, String(SfxResId(STR_SAVEDOC))));
                     else
