diff --git svx/source/cui/optgdlg.cxx svx/source/cui/optgdlg.cxx
index de0e90c..5ff4def 100644
--- svx/source/cui/optgdlg.cxx
+++ svx/source/cui/optgdlg.cxx
@@ -223,6 +223,7 @@ OfaMiscTabPage::OfaMiscTabPage(Window* pParent, const SfxItemSet& rSet ) :
     aPrintDlgCB			( this, SVX_RES( CB_PRINTDLG ) ),
     aDocStatusFL		( this, SVX_RES( FL_DOCSTATUS ) ),
     aDocStatusCB		( this, SVX_RES( CB_DOCSTATUS ) ),
+    aSaveAlwaysCB       ( this, SVX_RES( CB_SAVE_ALWAYS ) ),
     aTwoFigureFL		( this, SVX_RES( FL_TWOFIGURE ) ),
     aInterpretFT		( this, SVX_RES( FT_INTERPRET ) ),
     aYearValueField		( this, SVX_RES( NF_YEARVALUE ) ),
@@ -253,7 +254,7 @@ OfaMiscTabPage::OfaMiscTabPage(Window* pParent, const SfxItemSet& rSet ) :
 
 		Window* pWins[] =
 		{
-		    &aDocStatusFL, &aDocStatusCB, &aTwoFigureFL,
+		    &aDocStatusFL, &aDocStatusCB, &aSaveAlwaysCB, &aTwoFigureFL,
 		    &aInterpretFT, &aYearValueField, &aToYearFT
 		};
 		Window** pCurrent = pWins;
@@ -275,8 +276,8 @@ OfaMiscTabPage::OfaMiscTabPage(Window* pParent, const SfxItemSet& rSet ) :
 
         Window* pWins[] =
         {
-            &aPrintDlgFL, &aPrintDlgCB, &aDocStatusFL, &aDocStatusCB, &aTwoFigureFL,
-            &aInterpretFT, &aYearValueField, &aToYearFT
+            &aPrintDlgFL, &aPrintDlgCB, &aDocStatusFL, &aDocStatusCB, &aSaveAlwaysCB,
+            &aTwoFigureFL, &aInterpretFT, &aYearValueField, &aToYearFT
         };
         Window** pCurrent = pWins;
         const sal_Int32 nCount = sizeof( pWins ) / sizeof( pWins[ 0 ] );
@@ -301,7 +302,7 @@ OfaMiscTabPage::OfaMiscTabPage(Window* pParent, const SfxItemSet& rSet ) :
 
         Window* pWins[] =
         {
-            &aDocStatusFL, &aDocStatusCB, &aTwoFigureFL,
+            &aDocStatusFL, &aDocStatusCB, &aSaveAlwaysCB, &aTwoFigureFL,
             &aInterpretFT, &aYearValueField, &aToYearFT
         };
         Window** pCurrent = pWins;
@@ -432,6 +433,13 @@ BOOL OfaMiscTabPage::FillItemSet( SfxItemSet& rSet )
         bModified = TRUE;
     }
 
+    if ( aSaveAlwaysCB.IsChecked() != aSaveAlwaysCB.GetSavedValue() )
+    {
+        SvtMiscOptions aMiscOpt;
+        aMiscOpt.SetSaveAlwaysAllowed( aSaveAlwaysCB.IsChecked() );
+        bModified = TRUE;
+    }
+
     const SfxUInt16Item* pUInt16Item =
         PTR_CAST( SfxUInt16Item, GetOldItem( rSet, SID_ATTR_YEAR2000 ) );
     USHORT nNum = (USHORT)aYearValueField.GetText().ToInt32();
@@ -473,6 +481,8 @@ void OfaMiscTabPage::Reset( const SfxItemSet& rSet )
     aFileDlgCB.SaveValue();
     aPrintDlgCB.Check( !aMiscOpt.UseSystemPrintDialog() );
     aPrintDlgCB.SaveValue();
+    aSaveAlwaysCB.Check( aMiscOpt.IsSaveAlwaysAllowed() );
+    aSaveAlwaysCB.SaveValue();
 
     aODMADlgCB.Check( aMiscOpt.TryODMADialog() );
     aODMADlgCB.SaveValue();
diff --git svx/source/cui/optgdlg.hrc svx/source/cui/optgdlg.hrc
index bea3f72..f499400 100644
--- svx/source/cui/optgdlg.hrc
+++ svx/source/cui/optgdlg.hrc
@@ -59,7 +59,8 @@
 #define ROW9					(ROW8+RSC_CD_CHECKBOX_HEIGHT+ROWSPACE)
 #define ROW10					(ROW9+RSC_CD_FIXEDLINE_HEIGHT+ROWSPACE)
 #define ROW11					(ROW10+RSC_CD_CHECKBOX_HEIGHT+ROWSPACE)
-#define ROW12					(ROW11+RSC_CD_FIXEDLINE_HEIGHT+ROWSPACE)
+#define ROW12					(ROW11+RSC_CD_CHECKBOX_HEIGHT+ROWSPACE)
+#define ROW13					(ROW12+RSC_CD_FIXEDLINE_HEIGHT+ROWSPACE)
 
 #define DIFF(v1,v2)						(v2-v1)
 #define OFFS_TEXTBOX_FIXEDTEXT(base)	(base+(RSC_CD_TEXTBOX_HEIGHT-RSC_CD_FIXEDTEXT_HEIGHT)/2)
@@ -204,6 +205,7 @@
 #define FL_DOCSTATUS                60
 #define CB_DOCSTATUS                61
 #define CB_ODMADLG                  62
+#define CB_SAVE_ALWAYS              74
 #endif // #ifndef _SVX_OPTGDLG_HRC
 
 // ******************************************************************* EOF
diff --git svx/source/cui/optgdlg.hxx svx/source/cui/optgdlg.hxx
index 2a26918..f3d233d 100644
--- svx/source/cui/optgdlg.hxx
+++ svx/source/cui/optgdlg.hxx
@@ -66,6 +66,7 @@ private:
 
     FixedLine           aDocStatusFL;
     CheckBox            aDocStatusCB;
+    CheckBox            aSaveAlwaysCB;
 
     FixedLine           aTwoFigureFL;
     FixedText			aInterpretFT;
diff --git svx/source/cui/optgdlg.src svx/source/cui/optgdlg.src
index 8805d4b..6831324 100644
--- svx/source/cui/optgdlg.src
+++ svx/source/cui/optgdlg.src
@@ -134,22 +134,28 @@ TabPage OFA_TP_MISC
         Size = MAP_APPFONT( WHOLE_WIDTH - COL1, RSC_CD_CHECKBOX_HEIGHT );
         Text [ en-US ] = "~Printing sets \"document modified\" status";
     };
+    CheckBox CB_SAVE_ALWAYS
+    {
+        Pos = MAP_APPFONT( COL1, ROW11 );
+        Size = MAP_APPFONT( WHOLE_WIDTH - COL1, RSC_CD_CHECKBOX_HEIGHT );
+        Text [ en-US ] = "Always allow saving of document";
+    };
     FixedLine FL_TWOFIGURE
     {
-		Pos = MAP_APPFONT( COL0, ROW11 );
+        Pos = MAP_APPFONT( COL0, ROW12 );
         Size = MAP_APPFONT( WHOLE_WIDTH, RSC_CD_FIXEDLINE_HEIGHT );
         Text [ en-US ] = "Year (two digits)" ;
     };
     FixedText FT_INTERPRET
     {
-		Pos = MAP_APPFONT( COL1, OFFS_TEXTBOX_FIXEDTEXT(ROW12) );
+        Pos = MAP_APPFONT( COL1, OFFS_TEXTBOX_FIXEDTEXT(ROW13) );
         Size = MAP_APPFONT( DIFF( COL1, COL3), RSC_CD_FIXEDTEXT_HEIGHT );
         Text [ en-US ] = "Interpret as years between";
     };
     NumericField NF_YEARVALUE
     {
         Border = TRUE ;
-		Pos = MAP_APPFONT( COL4, ROW12 );
+        Pos = MAP_APPFONT( COL4, ROW13 );
         Size = MAP_APPFONT( EDIT_WIDTH, RSC_CD_TEXTBOX_HEIGHT );
         Minimum = 1583 ;
         Maximum = 9857 ;
@@ -158,7 +164,7 @@ TabPage OFA_TP_MISC
     };
     FixedText FT_TOYEAR
     {
-		Pos = MAP_APPFONT( COL6, OFFS_TEXTBOX_FIXEDTEXT(ROW12) );
+        Pos = MAP_APPFONT( COL6, OFFS_TEXTBOX_FIXEDTEXT(ROW13) );
         Size = MAP_APPFONT( WHOLE_WIDTH - COL6, RSC_CD_FIXEDTEXT_HEIGHT );
         Text [ en-US ] = "and " ;
     };
