--- vcl/source/app/settings.cxx.orig	2009-10-27 00:58:57.000000000 +0100
+++ vcl/source/app/settings.cxx	2009-10-27 00:59:38.000000000 +0100
@@ -819,11 +819,11 @@
     if( bCont )
     {
         if( rDesktopEnvironment.equalsIgnoreAsciiCaseAscii( "gnome" ) )
-            nRet = STYLE_SYMBOLS_TANGO;
+            nRet = STYLE_SYMBOLS_DEFAULT;
         else if( rDesktopEnvironment.equalsIgnoreAsciiCaseAscii( "kde" ) )
-            nRet = STYLE_SYMBOLS_CRYSTAL;
+            nRet = STYLE_SYMBOLS_DEFAULT;
 		else if( rDesktopEnvironment.equalsIgnoreAsciiCaseAscii( "kde4" ) )
-            nRet = STYLE_SYMBOLS_OXYGEN;
+            nRet = STYLE_SYMBOLS_DEFAULT;
     }
 
     // falback to any existing style
