diff -r 5c6b8c37653f apple_remote/AppleRemote.m
--- apple_remote/AppleRemote.m	Tue Dec 15 14:27:37 2009 +0100
+++ apple_remote/AppleRemote.m	Tue Dec 15 19:18:56 2009 +0100
@@ -43,6 +43,10 @@
 	#define NSAppKitVersionNumber10_4 824
 #endif
 
+#ifndef NSAppKitVersionNumber10_5
+    #define NSAppKitVersionNumber10_5 949
+#endif
+
 @implementation AppleRemote
 
 + (const char*) remoteControlDeviceName {
@@ -52,7 +56,12 @@
 - (void) setCookieMappingInDictionary: (NSMutableDictionary*) _cookieToButtonMapping	{	
     
     // TODO : avoid such magics 
-	if (floor(NSAppKitVersionNumber) <= NSAppKitVersionNumber10_4) {
+	if (floor(NSAppKitVersionNumber) <= NSAppKitVersionNumber10_4) 
+    {
+        #ifdef DEBUG
+            NSLog( @"setting 10.4 cookies" );
+        #endif
+
 		// 10.4.x Tiger
 		[_cookieToButtonMapping setObject:[NSNumber numberWithInt:kRemoteButtonPlus]		forKey:@"14_12_11_6_"];
 		[_cookieToButtonMapping setObject:[NSNumber numberWithInt:kRemoteButtonMinus]		forKey:@"14_13_11_6_"];		
@@ -65,7 +74,12 @@
 		[_cookieToButtonMapping setObject:[NSNumber numberWithInt:kRemoteButtonMenu_Hold]	forKey:@"14_6_14_6_"];
 		[_cookieToButtonMapping setObject:[NSNumber numberWithInt:kRemoteButtonPlay_Hold]	forKey:@"18_14_6_18_14_6_"];
 		[_cookieToButtonMapping setObject:[NSNumber numberWithInt:kRemoteControl_Switched]	forKey:@"19_"];			
-	} else {
+	} 
+    else if( floor(NSAppKitVersionNumber) <= NSAppKitVersionNumber10_5 )
+    {
+        #ifdef DEBUG
+            NSLog( @"setting 10.5 cookies" );
+        #endif
 		// 10.5.x Leopard
 		[_cookieToButtonMapping setObject:[NSNumber numberWithInt:kRemoteButtonPlus]		forKey:@"31_29_28_19_18_"];
 		[_cookieToButtonMapping setObject:[NSNumber numberWithInt:kRemoteButtonMinus]		forKey:@"31_30_28_19_18_"];	
@@ -78,6 +92,24 @@
 		[_cookieToButtonMapping setObject:[NSNumber numberWithInt:kRemoteButtonMenu_Hold]	forKey:@"31_19_18_31_19_18_"];
 		[_cookieToButtonMapping setObject:[NSNumber numberWithInt:kRemoteButtonPlay_Hold]	forKey:@"35_31_19_18_35_31_19_18_"];
 		[_cookieToButtonMapping setObject:[NSNumber numberWithInt:kRemoteControl_Switched]	forKey:@"19_"];			
+    }    
+    else
+    {
+        #ifdef DEBUG
+            NSLog( @"setting 10.6 cookies" );
+        #endif
+        // 10.6.x Snow Leopard
+        [_cookieToButtonMapping setObject:[NSNumber numberWithInt:kRemoteButtonPlus]		forKey:@"33_31_30_21_20_2_"];
+        [_cookieToButtonMapping setObject:[NSNumber numberWithInt:kRemoteButtonMinus]		forKey:@"33_32_30_21_20_2_"];	
+        [_cookieToButtonMapping setObject:[NSNumber numberWithInt:kRemoteButtonMenu]		forKey:@"33_22_21_20_2_33_22_21_20_2_"];
+        [_cookieToButtonMapping setObject:[NSNumber numberWithInt:kRemoteButtonPlay]		forKey:@"33_23_21_20_2_33_23_21_20_2_"];
+        [_cookieToButtonMapping setObject:[NSNumber numberWithInt:kRemoteButtonRight]		forKey:@"33_24_21_20_2_33_24_21_20_2_"];
+        [_cookieToButtonMapping setObject:[NSNumber numberWithInt:kRemoteButtonLeft]		forKey:@"33_25_21_20_2_33_25_21_20_2_"];
+        [_cookieToButtonMapping setObject:[NSNumber numberWithInt:kRemoteButtonRight_Hold]	forKey:@"33_21_20_14_12_2_"];
+        [_cookieToButtonMapping setObject:[NSNumber numberWithInt:kRemoteButtonLeft_Hold]	forKey:@"33_21_20_13_12_2_"];
+        [_cookieToButtonMapping setObject:[NSNumber numberWithInt:kRemoteButtonMenu_Hold]	forKey:@"33_21_20_2_33_21_20_2_"];
+        [_cookieToButtonMapping setObject:[NSNumber numberWithInt:kRemoteButtonPlay_Hold]	forKey:@"37_33_21_20_2_37_33_21_20_2_"];
+        [_cookieToButtonMapping setObject:[NSNumber numberWithInt:kRemoteControl_Switched]	forKey:@"19_"];
 	}
 }
 
