--- officecfg/registry/data/org/openoffice/Office/Common.xcu.ark	2007-10-19 12:01:42.000000000 +0200
+++ officecfg/registry/data/org/openoffice/Office/Common.xcu	2007-10-19 12:04:52.000000000 +0200
@@ -651,8 +651,7 @@
   </node>
   <node oor:name="ExternalMailer" install:module="unixdesktop">
     <prop oor:name="Program">
-      <value oor:external=
-          "com.sun.star.configuration.backend.DesktopBackend ExternalMailer"/>
+      <value>/usr/bin/kmail</value>
     </prop>
   </node>
   <node oor:name="Font">
