---
 automation/source/testtool/objtest.cxx |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git automation/source/testtool/objtest.cxx automation/source/testtool/objtest.cxx
index 38dfa61..665dd5a 100644
--- automation/source/testtool/objtest.cxx
+++ automation/source/testtool/objtest.cxx
@@ -2884,7 +2884,8 @@ void TestToolObj::DebugFindNoErrors( BOOL bDebugFindNoErrors )
 
 SbxVariable* TestToolObj::Find( const String& aStr, SbxClassType aType)
 {
-    if ( BasicRuntimeAccess::IsRunInit() )            // wegen Find im "Global" Befehl des Basic
+    if ( BasicRuntimeAccess::IsRunInit()
+    || ( aStr == String( RTL_CONSTASCII_USTRINGPARAM( "ThisComponent" ) ) ) )            // wegen Find im "Global" Befehl des Basic
         return NULL;
 
     SbxVariableRef Old = SbxObject::Find(aStr, aType );
-- 
1.7.0.1

