diff --git vcl/aqua/inc/salframe.h vcl/aqua/inc/salframe.h
index f853328..5eb6f43 100644
--- vcl/aqua/inc/salframe.h
+++ vcl/aqua/inc/salframe.h
@@ -154,6 +154,8 @@ public:
     virtual void				Beep( SoundType eSoundType );
     virtual const SystemEnvData*	GetSystemData() const;
     virtual SalPointerState		GetPointerState();
+    virtual SalIndicatorState   GetIndicatorState();
+    virtual void                SimulateKeyPress( USHORT nKeyCode );
     virtual void				SetParent( SalFrame* pNewParent );
     virtual bool				SetPluginParent( SystemParentData* pNewParent );
     virtual void                SetExtendedFrameStyle( SalExtStyle );
diff --git vcl/aqua/source/window/salframe.cxx vcl/aqua/source/window/salframe.cxx
index 468d7ad..ff635ff 100644
--- vcl/aqua/source/window/salframe.cxx
+++ vcl/aqua/source/window/salframe.cxx
@@ -1379,6 +1379,17 @@ SalPointerState	AquaSalFrame::GetPointerState()
     return state;
 }
 
+SalFrame::SalIndicatorState AquaSalFrame::GetIndicatorState()
+{
+    SalIndicatorState aState;
+    aState.mnState = 0;
+    return aState;
+}
+
+void AquaSalFrame::SimulateKeyPress( USHORT /*nKeyCode*/ )
+{
+}
+
 bool AquaSalFrame::SetPluginParent( SystemParentData* pNewParent )
 {
     // plugin parent may be killed unexpectedly by
diff --git vcl/inc/vcl/keycodes.hxx vcl/inc/vcl/keycodes.hxx
index 74f6deb..8721c21 100644
--- vcl/inc/vcl/keycodes.hxx
+++ vcl/inc/vcl/keycodes.hxx
@@ -161,6 +161,10 @@
 #define KEY_BRACKETLEFT ((USHORT)::com::sun::star::awt::Key::BRACKETLEFT)
 #define KEY_BRACKETRIGHT ((USHORT)::com::sun::star::awt::Key::BRACKETRIGHT)
 
+#define KEY_CAPSLOCK    ((USHORT)::com::sun::star::awt::Key::CAPSLOCK)
+#define KEY_NUMLOCK     ((USHORT)::com::sun::star::awt::Key::NUMLOCK)
+#define KEY_SCROLLLOCK  ((USHORT)::com::sun::star::awt::Key::SCROLLLOCK)
+
 #define KEY_CODE		((USHORT)0x0FFF)
 
 // Modifier-Tasten
@@ -193,4 +197,12 @@
 #define MOUSE_MIDDLE	((USHORT)0x0002)
 #define MOUSE_RIGHT 	((USHORT)0x0004)
 
+// -------------------
+// - Indicator-Types -
+// -------------------
+
+#define INDICATOR_CAPSLOCK      0x0001
+#define INDICATOR_NUMLOCK       0x0002
+#define INDICATOR_SCROLLLOCK    0x0004
+
 #endif // _SV_KEYCODES_HXX
diff --git vcl/inc/vcl/salframe.hxx vcl/inc/vcl/salframe.hxx
index 8b2f3a2..5a06d0a 100644
--- vcl/inc/vcl/salframe.hxx
+++ vcl/inc/vcl/salframe.hxx
@@ -252,6 +252,15 @@ public:
 
     virtual SalPointerState		GetPointerState() = 0;
 
+    struct SalIndicatorState
+    {
+        USHORT  mnState;
+    };
+
+    virtual SalIndicatorState   GetIndicatorState() = 0;
+
+    virtual void                SimulateKeyPress( USHORT nKeyCode ) = 0;
+
     // set new parent window
     virtual void				SetParent( SalFrame* pNewParent ) = 0;
     // reparent window to act as a plugin; implementation
diff --git vcl/inc/vcl/window.hxx vcl/inc/vcl/window.hxx
index 540906c..55f9d0b 100644
--- vcl/inc/vcl/window.hxx
+++ vcl/inc/vcl/window.hxx
@@ -1141,6 +1141,22 @@ public:
     */
     void doLazyDelete();
 
+    //-------------------------------------
+    //  Keyboard access functions
+    //-------------------------------------
+
+    /** Query the states of keyboard indicators - Caps Lock, Num Lock and
+        Scroll Lock.  Use the following mask to retrieve the state of each
+        indicator:
+        
+            INDICATOR_CAPS_LOCK
+            INDICATOR_NUM_LOCK
+            INDICATOR_SCROLL_LOCK
+      */
+    USHORT GetIndicatorState() const;
+
+    void SimulateKeyPress( USHORT nKeyCode ) const;
+
     virtual XubString GetSurroundingText() const;
     virtual Selection GetSurroundingTextSelection() const;
 };
diff --git vcl/os2/inc/salframe.h vcl/os2/inc/salframe.h
index 1887a2c..718fd23 100644
--- vcl/os2/inc/salframe.h
+++ vcl/os2/inc/salframe.h
@@ -155,6 +155,8 @@ public:
     virtual void				Beep( SoundType eSoundType );
     virtual const SystemEnvData*	GetSystemData() const;
     virtual SalPointerState		GetPointerState();
+    virtual SalIndicatorState   GetIndicatorState();
+    virtual void                SimulateKeyPress( USHORT nKeyCode );
     virtual void				SetParent( SalFrame* pNewParent );
     virtual bool				SetPluginParent( SystemParentData* pNewParent );
     virtual void                SetBackgroundBitmap( SalBitmap* );
diff --git vcl/os2/source/window/salframe.cxx vcl/os2/source/window/salframe.cxx
index fb53876..31e063d 100644
--- vcl/os2/source/window/salframe.cxx
+++ vcl/os2/source/window/salframe.cxx
@@ -2357,6 +2357,19 @@ SalFrame::SalPointerState Os2SalFrame::GetPointerState()
 
 // -----------------------------------------------------------------------
 
+SalFrame::SalIndicatorState Os2SalFrame::GetIndicatorState()
+{
+    SalIndicatorState aState;
+    aState.mnState = 0;
+    return aState;
+}
+
+void Os2SalFrame::SimulateKeyPress( USHORT /*nKeyCode*/ )
+{
+}
+
+// -----------------------------------------------------------------------
+
 void Os2SalFrame::SetBackgroundBitmap( SalBitmap* )
 {
 }
diff --git vcl/source/window/window.cxx vcl/source/window/window.cxx
index 5b3cf91..e4573f2 100644
--- vcl/source/window/window.cxx
+++ vcl/source/window/window.cxx
@@ -4770,6 +4770,16 @@ void Window::doLazyDelete()
     vcl::LazyDeletor<Window>::Delete( this );
 }
 
+USHORT Window::GetIndicatorState() const
+{
+    return mpWindowImpl->mpFrame->GetIndicatorState().mnState;
+}
+
+void Window::SimulateKeyPress( USHORT nKeyCode ) const
+{
+    mpWindowImpl->mpFrame->SimulateKeyPress(nKeyCode);
+}
+
 // -----------------------------------------------------------------------
 
 void Window::MouseMove( const MouseEvent& rMEvt )
diff --git vcl/unx/gtk/window/gtkframe.cxx vcl/unx/gtk/window/gtkframe.cxx
index 027d121..75cd135 100644
--- vcl/unx/gtk/window/gtkframe.cxx
+++ vcl/unx/gtk/window/gtkframe.cxx
@@ -2250,6 +2250,18 @@ SalFrame::SalPointerState GtkSalFrame::GetPointerState()
     return aState;
 }
 
+SalFrame::SalIndicatorState GtkSalFrame::GetIndicatorState()
+{
+    SalIndicatorState aState;
+    aState.mnState = GetX11SalData()->GetDisplay()->GetIndicatorState();
+    return aState;
+}
+
+void GtkSalFrame::SimulateKeyPress( USHORT nKeyCode )
+{
+    GetX11SalData()->GetDisplay()->SimulateKeyPress(nKeyCode);
+}
+
 void GtkSalFrame::SetInputContext( SalInputContext* pContext )
 {
     if( ! pContext )
diff --git vcl/unx/headless/svpframe.cxx vcl/unx/headless/svpframe.cxx
index a82827f..4bbaaf6 100644
--- vcl/unx/headless/svpframe.cxx
+++ vcl/unx/headless/svpframe.cxx
@@ -391,6 +391,17 @@ SalFrame::SalPointerState SvpSalFrame::GetPointerState()
     return aState;
 }
 
+SalFrame::SalIndicatorState SvpSalFrame::GetIndicatorState()
+{
+    SalIndicatorState aState;
+    aState.mnState = 0;
+    return aState;
+}
+
+void SvpSalFrame::SimulateKeyPress( USHORT /*nKeyCode*/ )
+{
+}
+
 void SvpSalFrame::SetParent( SalFrame* pNewParent )
 {
     if( m_pParent )
diff --git vcl/unx/headless/svpframe.hxx vcl/unx/headless/svpframe.hxx
index 29d3da6..9077f24 100644
--- vcl/unx/headless/svpframe.hxx
+++ vcl/unx/headless/svpframe.hxx
@@ -115,6 +115,8 @@ public:
     virtual void				Beep( SoundType eSoundType );
     virtual const SystemEnvData*	GetSystemData() const;
     virtual SalPointerState     GetPointerState();
+    virtual SalIndicatorState   GetIndicatorState();
+    virtual void                SimulateKeyPress( USHORT nKeyCode );
     virtual void				SetParent( SalFrame* pNewParent );
     virtual bool				SetPluginParent( SystemParentData* pNewParent );
     virtual void                SetBackgroundBitmap( SalBitmap* pBitmap );
diff --git vcl/unx/inc/plugins/gtk/gtkframe.hxx vcl/unx/inc/plugins/gtk/gtkframe.hxx
index f55e5ec..58f7c26 100644
--- vcl/unx/inc/plugins/gtk/gtkframe.hxx
+++ vcl/unx/inc/plugins/gtk/gtkframe.hxx
@@ -368,6 +368,10 @@ public:
     // get current modifier and button mask
     virtual SalPointerState     GetPointerState();
 
+    virtual SalIndicatorState   GetIndicatorState();
+
+    virtual void                SimulateKeyPress( USHORT nKeyCode );
+
     // set new parent window
     virtual void                SetParent( SalFrame* pNewParent );
     // reparent window to act as a plugin; implementation
diff --git vcl/unx/inc/saldisp.hxx vcl/unx/inc/saldisp.hxx
index 859f926..89fc11b 100644
--- vcl/unx/inc/saldisp.hxx
+++ vcl/unx/inc/saldisp.hxx
@@ -446,6 +446,8 @@ public:
     void			Beep() const;
 
     void			ModifierMapping();
+    void            SimulateKeyPress( USHORT nKeyCode );
+    USHORT          GetIndicatorState() const;
     String			GetKeyNameFromKeySym( KeySym keysym ) const;
     XubString		GetKeyName( USHORT nKeyCode ) const;
     USHORT			GetKeyCode( KeySym keysym, char*pcPrintable ) const;
diff --git vcl/unx/inc/salframe.h vcl/unx/inc/salframe.h
index f26e4ef..38c2897 100644
--- vcl/unx/inc/salframe.h
+++ vcl/unx/inc/salframe.h
@@ -255,6 +255,8 @@ public:
     virtual void				Beep( SoundType eSoundType );
     virtual const SystemEnvData*	GetSystemData() const;
     virtual SalPointerState     GetPointerState();
+    virtual SalIndicatorState   GetIndicatorState();
+    virtual void                SimulateKeyPress( USHORT nKeyCode );
     virtual void				SetParent( SalFrame* pNewParent );
     virtual bool				SetPluginParent( SystemParentData* pNewParent );
     virtual void                SetBackgroundBitmap( SalBitmap* pBitmap );
diff --git vcl/unx/source/app/saldisp.cxx vcl/unx/source/app/saldisp.cxx
index 451e7d5..efc9add 100644
--- vcl/unx/source/app/saldisp.cxx
+++ vcl/unx/source/app/saldisp.cxx
@@ -60,7 +60,7 @@
 #define XK_KOREAN
 #endif
 #include <X11/keysym.h>
-
+#include <X11/XKBlib.h>
 #include <X11/Xatom.h>
 
 #ifdef USE_XINERAMA
@@ -1045,6 +1045,77 @@ void SalDisplay::Beep() const
 
 // Keyboard
 // -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+
+namespace {
+
+bool InitXkb(Display* dpy)
+{
+    int nOpcode, nEvent, nError;
+    int nXkbMajor = XkbMajorVersion;
+    int nXkbMinor = XkbMinorVersion;
+
+    if (!XkbLibraryVersion(&nXkbMajor, &nXkbMinor))
+        return false;
+
+    return XkbQueryExtension(
+        dpy, &nOpcode, &nEvent, &nError, &nXkbMajor, &nXkbMinor);
+}
+
+unsigned int GetKeySymMask(Display* dpy, KeySym nKeySym)
+{
+    int nMask = 0;
+    XModifierKeymap* pXmkMap = XGetModifierMapping(dpy);
+    KeyCode nKeyCode = XKeysymToKeycode(dpy, nKeySym);
+    if (nKeyCode == NoSymbol)
+        return 0;
+
+    for (int i = 0; i < 8; ++i)
+    {
+        KeyCode nThisKeyCode = pXmkMap->modifiermap[pXmkMap->max_keypermod*i];
+        if (nThisKeyCode == nKeyCode)
+            nMask = 1 << i;
+    }
+    XFreeModifiermap(pXmkMap);
+    return nMask;
+}
+
+}
+
+void SalDisplay::SimulateKeyPress( USHORT nKeyCode )
+{
+    if (nKeyCode == KEY_CAPSLOCK)
+    {
+        Display* dpy = GetDisplay();
+        if (!InitXkb(dpy))
+            return;
+
+        unsigned int nMask = GetKeySymMask(dpy, XK_Caps_Lock);
+        XkbStateRec xkbState;
+        XkbGetState(dpy, XkbUseCoreKbd, &xkbState);
+        unsigned int nCapsLockState = xkbState.locked_mods & nMask;
+        if (nCapsLockState)
+            XkbLockModifiers (dpy, XkbUseCoreKbd, nMask, 0);
+        else
+            XkbLockModifiers (dpy, XkbUseCoreKbd, nMask, nMask);
+    }
+}
+
+USHORT SalDisplay::GetIndicatorState() const
+{
+    unsigned int _state = 0;
+    USHORT nState = 0;
+    XkbGetIndicatorState(pDisp_, XkbUseCoreKbd, &_state);
+
+    if ((_state & 0x00000001))
+        nState |= INDICATOR_CAPSLOCK;
+    if ((_state & 0x00000002))
+        nState |= INDICATOR_NUMLOCK;
+    if ((_state & 0x00000004))
+        nState |= INDICATOR_SCROLLLOCK;
+
+    return nState;
+}
+
 String SalDisplay::GetKeyNameFromKeySym( KeySym nKeySym ) const
 {
     String aRet;
diff --git vcl/unx/source/window/salframe.cxx vcl/unx/source/window/salframe.cxx
index 7e61602..30fb2f0 100644
--- vcl/unx/source/window/salframe.cxx
+++ vcl/unx/source/window/salframe.cxx
@@ -2848,6 +2848,18 @@ SalFrame::SalPointerState X11SalFrame::GetPointerState()
     return aState;
 }
 
+SalFrame::SalIndicatorState X11SalFrame::GetIndicatorState()
+{
+    SalIndicatorState aState;
+    aState.mnState = GetX11SalData()->GetDisplay()->GetIndicatorState();
+    return aState;
+}
+
+void X11SalFrame::SimulateKeyPress( USHORT nKeyCode )
+{
+    GetX11SalData()->GetDisplay()->SimulateKeyPress(nKeyCode);
+}
+
 long X11SalFrame::HandleMouseEvent( XEvent *pEvent )
 {
     SalMouseEvent		aMouseEvt;
diff --git vcl/win/inc/salframe.h vcl/win/inc/salframe.h
index 23469b4..a8576f2 100644
--- vcl/win/inc/salframe.h
+++ vcl/win/inc/salframe.h
@@ -138,6 +138,8 @@ public:
     virtual void				Beep( SoundType eSoundType );
     virtual const SystemEnvData*	GetSystemData() const;
     virtual SalPointerState		GetPointerState();
+    virtual SalIndicatorState   GetIndicatorState();
+    virtual void                SimulateKeyPress( USHORT nKeyCode );
     virtual void				SetParent( SalFrame* pNewParent );
     virtual bool				SetPluginParent( SystemParentData* pNewParent );
     virtual void                SetBackgroundBitmap( SalBitmap* );
diff --git vcl/win/source/window/salframe.cxx vcl/win/source/window/salframe.cxx
index 02152f8..0e28d75 100644
--- vcl/win/source/window/salframe.cxx
+++ vcl/win/source/window/salframe.cxx
@@ -3194,6 +3194,41 @@ SalFrame::SalPointerState WinSalFrame::GetPointerState()
 
 // -----------------------------------------------------------------------
 
+SalFrame::SalIndicatorState WinSalFrame::GetIndicatorState()
+{
+    SalIndicatorState aState;
+    aState.mnState = 0;
+    if (::GetKeyState(VK_CAPITAL))
+        aState.mnState |= INDICATOR_CAPSLOCK;
+
+    if (::GetKeyState(VK_NUMLOCK))
+        aState.mnState |= INDICATOR_NUMLOCK;
+
+    if (::GetKeyState(VK_SCROLL))
+        aState.mnState |= INDICATOR_SCROLLLOCK;
+
+    return aState;
+}
+
+void WinSalFrame::SimulateKeyPress( USHORT nKeyCode )
+{
+    BYTE nVKey = 0;
+    switch (nKeyCode)
+    {
+        case KEY_CAPSLOCK:
+            nVKey = VK_CAPITAL;
+        break;
+    }
+
+    if (nVKey > 0 && nVKey < 255)
+    {
+        ::keybd_event(nVKey, 0x45, KEYEVENTF_EXTENDEDKEY, 0);
+        ::keybd_event(nVKey, 0x45, KEYEVENTF_EXTENDEDKEY|KEYEVENTF_KEYUP, 0);
+    }
+}
+
+// -----------------------------------------------------------------------
+
 void WinSalFrame::SetBackgroundBitmap( SalBitmap* )
 {
 }
