From 490160876fae0b03ab029059f9ed01405cb53bfa Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:54:20 +0200
Subject: [PATCH 100/878] automation-testtool-oobuildfix.diff

---
 automation/source/testtool/objtest.cxx |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/automation/source/testtool/objtest.cxx b/automation/source/testtool/objtest.cxx
index c13dc41..cd605bf 100644
--- a/automation/source/testtool/objtest.cxx
+++ b/automation/source/testtool/objtest.cxx
@@ -2878,7 +2878,8 @@ void TestToolObj::DebugFindNoErrors( BOOL bDebugFindNoErrors )
 
 SbxVariable* TestToolObj::Find( const String& aStr, SbxClassType aType)
 {
-    if ( BasicRuntimeAccess::IsRunInit() )            // wegen Find im "Global" Befehl des Basic
+    if ( BasicRuntimeAccess::IsRunInit()
+    || ( aStr == String( RTL_CONSTASCII_USTRINGPARAM( "ThisComponent" ) ) ) )            // wegen Find im "Global" Befehl des Basic
         return NULL;
 
     SbxVariableRef Old = SbxObject::Find(aStr, aType );
-- 
1.7.0.1

