From 2152a859d7de2a2a042027ea5ced1ceddb90e97d Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:54:20 +0200
Subject: [PATCH 081/768] automation-testtool-oobuildfix.diff

---
 automation/source/testtool/objtest.cxx |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git automation/source/testtool/objtest.cxx automation/source/testtool/objtest.cxx
index f491fd3..48eda21 100644
--- automation/source/testtool/objtest.cxx
+++ automation/source/testtool/objtest.cxx
@@ -2882,7 +2882,8 @@ void TestToolObj::DebugFindNoErrors( BOOL bDebugFindNoErrors )
 
 SbxVariable* TestToolObj::Find( const String& aStr, SbxClassType aType)
 {
-    if ( BasicRuntimeAccess::IsRunInit() )            // wegen Find im "Global" Befehl des Basic
+    if ( BasicRuntimeAccess::IsRunInit()
+    || ( aStr == String( RTL_CONSTASCII_USTRINGPARAM( "ThisComponent" ) ) ) )            // wegen Find im "Global" Befehl des Basic
         return NULL;
 
     SbxVariableRef Old = SbxObject::Find(aStr, aType );
-- 
1.7.0.1

