---
 filter/source/xsltfilter/makefile.mk |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git filter/source/xsltfilter/makefile.mk filter/source/xsltfilter/makefile.mk
index 1a20572..46f641e 100644
--- filter/source/xsltfilter/makefile.mk
+++ filter/source/xsltfilter/makefile.mk
@@ -62,7 +62,7 @@ JAVAFILES		= $(subst,$(CLASSDIR)$/, $(subst,.class,.java $(JAVACLASSFILES)))
 CUSTOMMANIFESTFILE = Manifest
 
 JARCOMPRESS		= TRUE
-JARCLASSDIRS	= XSLTransformer*.class XSLTFilterOLEExtracter*.class
+JARCLASSDIRS	= XSLTransformer*.class XSLTFilterOLEExtracter*.class Base64*.class
 JARTARGET		= $(TARGET).jar
 
 .IF "$(SYSTEM_SAXON)" == "YES"
@@ -72,7 +72,7 @@ JARFILES += saxon9.jar
 .ENDIF
 
 # --- Files --------------------------------------------------------
-JAVACLASSFILES=$(CLASSDIR)$/XSLTransformer.class  $(CLASSDIR)$/XSLTFilterOLEExtracter.class
+JAVACLASSFILES=$(CLASSDIR)$/XSLTransformer.class  $(CLASSDIR)$/XSLTFilterOLEExtracter.class $(CLASSDIR)$/Base64.class
 .ENDIF
 
 # --- Targets ------------------------------------------------------
-- 
1.7.0.1

