From 6f7bb95e235b32bda297caf075a8349ac79f030f Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:05:38 +0200
Subject: [PATCH 681/878] base64.diff

---
 filter/source/xsltfilter/makefile.mk |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/filter/source/xsltfilter/makefile.mk b/filter/source/xsltfilter/makefile.mk
index ccb8ccb..acff87d 100644
--- a/filter/source/xsltfilter/makefile.mk
+++ b/filter/source/xsltfilter/makefile.mk
@@ -65,11 +65,11 @@ JAVAFILES		= $(subst,$(CLASSDIR)$/, $(subst,.class,.java $(JAVACLASSFILES)))
 CUSTOMMANIFESTFILE = Manifest
 
 JARCOMPRESS		= TRUE
-JARCLASSDIRS	= XSLTransformer*.class XSLTFilterOLEExtracter*.class
+JARCLASSDIRS	= XSLTransformer*.class XSLTFilterOLEExtracter*.class Base64*.class
 JARTARGET		= $(TARGET).jar
 
 # --- Files --------------------------------------------------------
-JAVACLASSFILES=$(CLASSDIR)$/XSLTransformer.class  $(CLASSDIR)$/XSLTFilterOLEExtracter.class
+JAVACLASSFILES=$(CLASSDIR)$/XSLTransformer.class  $(CLASSDIR)$/XSLTFilterOLEExtracter.class $(CLASSDIR)$/Base64.class
 .ENDIF
 
 # --- Targets ------------------------------------------------------
-- 
1.7.0.1

