From edcb333216622e97107a886765976f49e548ae61 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:06:54 +0200
Subject: [PATCH 735/878] basegfx-svgpolygon-speedup.diff

---
 basegfx/source/polygon/b2dsvgpolypolygon.cxx |   51 ++++++++++++++------------
 1 files changed, 27 insertions(+), 24 deletions(-)

diff --git a/basegfx/source/polygon/b2dsvgpolypolygon.cxx b/basegfx/source/polygon/b2dsvgpolypolygon.cxx
index 9b4d07e..c6e0912 100644
--- a/basegfx/source/polygon/b2dsvgpolypolygon.cxx
+++ b/basegfx/source/polygon/b2dsvgpolypolygon.cxx
@@ -64,10 +64,8 @@ namespace basegfx
                 }
             }
 
-            inline bool lcl_isOnNumberChar(const ::rtl::OUString& rStr, const sal_Int32 nPos, bool bSignAllowed = true)
+            inline bool lcl_isOnNumberChar(const sal_Unicode aChar, bool bSignAllowed = true)
             {
-                const sal_Unicode aChar(rStr[nPos]);
-
                 const bool bPredicate( (sal_Unicode('0') <= aChar && sal_Unicode('9') >= aChar)
                                        || (bSignAllowed && sal_Unicode('+') == aChar)
                                        || (bSignAllowed && sal_Unicode('-') == aChar) );
@@ -75,6 +73,12 @@ namespace basegfx
                 return bPredicate;
             }
 
+            inline bool lcl_isOnNumberChar(const ::rtl::OUString& rStr, const sal_Int32 nPos, bool bSignAllowed = true)
+            {
+                return lcl_isOnNumberChar(rStr[nPos],
+                                          bSignAllowed);
+            }
+
             bool lcl_getDoubleChar(double& 					o_fRetval,
                                    sal_Int32& 				io_rPos, 
                                    const ::rtl::OUString& 	rStr, 
@@ -230,16 +234,16 @@ namespace basegfx
                 lcl_skipSpacesAndCommas(io_rPos, rStr, nLen);
             }
 
-            void lcl_putNumberChar( ::rtl::OUString& rStr, 
-                                    double 		 	 fValue )
+            void lcl_putNumberChar( ::rtl::OUStringBuffer& rStr,
+                                    double 		 	       fValue )
             {
-                rStr += ::rtl::OUString::valueOf( fValue );
+                rStr.append( fValue );
             }
 
-            void lcl_putNumberCharWithSpace( ::rtl::OUString& rStr, 
-                                             double 		  fValue,
-                                             double 		  fOldValue,
-                                             bool 			  bUseRelativeCoordinates )
+            void lcl_putNumberCharWithSpace( ::rtl::OUStringBuffer& rStr,
+                                             double 		        fValue,
+                                             double 		        fOldValue,
+                                             bool 			        bUseRelativeCoordinates )
             {
                 if( bUseRelativeCoordinates )
                     fValue -= fOldValue;
@@ -247,11 +251,10 @@ namespace basegfx
                 const sal_Int32 aLen( rStr.getLength() );
                 if(aLen)
                 {
-                    if( lcl_isOnNumberChar(rStr, aLen - 1, false) && 
+                    if( lcl_isOnNumberChar(rStr.charAt(aLen - 1), false) &&
                         fValue >= 0.0 )
                     {
-                        rStr += ::rtl::OUString::valueOf( 
-                            sal_Unicode(' ') );
+                        rStr.append( sal_Unicode(' ') );
                     }
                 }
 
@@ -876,7 +879,7 @@ namespace basegfx
             bool bDetectQuadraticBeziers)
         {
             const sal_uInt32 nCount(rPolyPolygon.count());
-            ::rtl::OUString aResult;
+            ::rtl::OUStringBuffer aResult;
             B2DPoint aCurrentSVGPosition(0.0, 0.0); // SVG assumes (0,0) as the initial current point
 
             for(sal_uInt32 i(0); i < nCount; i++)
@@ -893,7 +896,7 @@ namespace basegfx
 
                     // handle polygon start point
                     B2DPoint aEdgeStart(aPolygon.getB2DPoint(0));
-                    aResult += ::rtl::OUString::valueOf(lcl_getCommand('M', 'm', bUseRelativeCoordinates));
+                    aResult.append(lcl_getCommand('M', 'm', bUseRelativeCoordinates));
                     lcl_putNumberCharWithSpace(aResult, aEdgeStart.getX(), aCurrentSVGPosition.getX(), bUseRelativeCoordinates);
                     lcl_putNumberCharWithSpace(aResult, aEdgeStart.getY(), aCurrentSVGPosition.getY(), bUseRelativeCoordinates);
                     aLastSVGCommand =  lcl_getCommand('L', 'l', bUseRelativeCoordinates);
@@ -954,7 +957,7 @@ namespace basegfx
 
                                     if(aLastSVGCommand != aCommand)
                                     {
-                                        aResult += ::rtl::OUString::valueOf(aCommand);
+                                        aResult.append(aCommand);
                                         aLastSVGCommand = aCommand;
                                     }
                                     
@@ -969,7 +972,7 @@ namespace basegfx
 
                                     if(aLastSVGCommand != aCommand)
                                     {
-                                        aResult += ::rtl::OUString::valueOf(aCommand);
+                                        aResult.append(aCommand);
                                         aLastSVGCommand = aCommand;
                                     }
                                     
@@ -990,7 +993,7 @@ namespace basegfx
 
                                     if(aLastSVGCommand != aCommand)
                                     {
-                                        aResult += ::rtl::OUString::valueOf(aCommand);
+                                        aResult.append(aCommand);
                                         aLastSVGCommand = aCommand;
                                     }
                                     
@@ -1007,7 +1010,7 @@ namespace basegfx
 
                                     if(aLastSVGCommand != aCommand)
                                     {
-                                        aResult += ::rtl::OUString::valueOf(aCommand);
+                                        aResult.append(aCommand);
                                         aLastSVGCommand = aCommand;
                                     }
                                     
@@ -1046,7 +1049,7 @@ namespace basegfx
 
                                     if(aLastSVGCommand != aCommand)
                                     {
-                                        aResult += ::rtl::OUString::valueOf(aCommand);
+                                        aResult.append(aCommand);
                                         aLastSVGCommand = aCommand;
                                     }
 
@@ -1060,7 +1063,7 @@ namespace basegfx
 
                                     if(aLastSVGCommand != aCommand)
                                     {
-                                        aResult += ::rtl::OUString::valueOf(aCommand);
+                                        aResult.append(aCommand);
                                         aLastSVGCommand = aCommand;
                                     }
 
@@ -1074,7 +1077,7 @@ namespace basegfx
 
                                     if(aLastSVGCommand != aCommand)
                                     {
-                                        aResult += ::rtl::OUString::valueOf(aCommand);
+                                        aResult.append(aCommand);
                                         aLastSVGCommand = aCommand;
                                     }
 
@@ -1092,12 +1095,12 @@ namespace basegfx
                     // close path if closed poly (Z and z are equivalent here, but looks nicer when case is matched)
                     if(aPolygon.isClosed())
                     {
-                        aResult += ::rtl::OUString::valueOf(lcl_getCommand('Z', 'z', bUseRelativeCoordinates));
+                        aResult.append(lcl_getCommand('Z', 'z', bUseRelativeCoordinates));
                     }
                 }
             }
 
-            return aResult;
+            return aResult.makeStringAndClear();
         }
     }
 }
-- 
1.7.0.1

