From be057ca8b061b016fe875c72483a1de7d8593bfe Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:06:58 +0200
Subject: [PATCH 647/768] boost-1.39-extra-header.diff

---
 boost/prj/d.lst |   17 +++++++++++++++++
 1 files changed, 17 insertions(+), 0 deletions(-)

diff --git boost/prj/d.lst boost/prj/d.lst
index 07d5a72..cef2abc 100644
--- boost/prj/d.lst
+++ boost/prj/d.lst
@@ -1,5 +1,7 @@
 mkdir: %_DEST%\inc%_EXT%\boost
 mkdir: %_DEST%\inc%_EXT%\boost\bind
+mkdir: %_DEST%\inc%_EXT%\boost\concept
+mkdir: %_DEST%\inc%_EXT%\boost\concept\detail
 mkdir: %_DEST%\inc%_EXT%\boost\config
 mkdir: %_DEST%\inc%_EXT%\boost\config\compiler
 mkdir: %_DEST%\inc%_EXT%\boost\config\no_tr1
@@ -106,6 +108,7 @@ mkdir: %_DEST%\inc%_EXT%\boost\spirit\home
 mkdir: %_DEST%\inc%_EXT%\boost\spirit\home\classic
 mkdir: %_DEST%\inc%_EXT%\boost\spirit\home\classic\actor
 mkdir: %_DEST%\inc%_EXT%\boost\spirit\home\classic\actor\impl
+mkdir: %_DEST%\inc%_EXT%\boost\spirit\home\classic\attribute
 mkdir: %_DEST%\inc%_EXT%\boost\spirit\home\classic\debug
 mkdir: %_DEST%\inc%_EXT%\boost\spirit\home\classic\debug\impl
 mkdir: %_DEST%\inc%_EXT%\boost\spirit\home\classic\meta
@@ -120,10 +123,15 @@ mkdir: %_DEST%\inc%_EXT%\boost\spirit\home\classic\core\primitives
 mkdir: %_DEST%\inc%_EXT%\boost\spirit\home\classic\core\primitives\impl
 mkdir: %_DEST%\inc%_EXT%\boost\spirit\home\classic\core\composite
 mkdir: %_DEST%\inc%_EXT%\boost\spirit\home\classic\core\composite\impl
+mkdir: %_DEST%\inc%_EXT%\boost\spirit\home\classic\dynamic
+mkdir: %_DEST%\inc%_EXT%\boost\spirit\home\classic\dynamic\impl
 mkdir: %_DEST%\inc%_EXT%\boost\spirit\home\classic\error_handling
 mkdir: %_DEST%\inc%_EXT%\boost\spirit\home\classic\error_handling\impl
 mkdir: %_DEST%\inc%_EXT%\boost\spirit\home\classic\iterator
 mkdir: %_DEST%\inc%_EXT%\boost\spirit\home\classic\iterator\impl
+mkdir: %_DEST%\inc%_EXT%\boost\spirit\home\classic\phoenix
+mkdir: %_DEST%\inc%_EXT%\boost\spirit\home\classic\symbols
+mkdir: %_DEST%\inc%_EXT%\boost\spirit\home\classic\symbols\impl
 mkdir: %_DEST%\inc%_EXT%\boost\spirit\home\classic\utility
 mkdir: %_DEST%\inc%_EXT%\boost\spirit\home\classic\utility\impl
 mkdir: %_DEST%\inc%_EXT%\boost\spirit\home\classic\utility\impl\chset
@@ -300,3 +308,12 @@ mkdir: %_DEST%\inc%_EXT%\boost\variant\detail
 ..\%__SRC%\inc\boost\utility\* %_DEST%\inc%_EXT%\boost\utility
 ..\%__SRC%\inc\boost\variant\* %_DEST%\inc%_EXT%\boost\variant
 ..\%__SRC%\inc\boost\variant\detail\* %_DEST%\inc%_EXT%\boost\variant\detail
+..\%__SRC%\misc\build\boost_1_39_0\boost\concept\* %_DEST%\inc%_EXT%\boost\concept
+..\%__SRC%\misc\build\boost_1_39_0\boost\concept\detail\* %_DEST%\inc%_EXT%\boost\concept\detail
+..\%__SRC%\misc\build\boost_1_39_0\boost\spirit\home\classic\* %_DEST%\inc%_EXT%\boost\spirit\home\classic\
+..\%__SRC%\misc\build\boost_1_39_0\boost\spirit\home\classic\attribute\* %_DEST%\inc%_EXT%\boost\spirit\home\classic\attribute
+..\%__SRC%\misc\build\boost_1_39_0\boost\spirit\home\classic\dynamic\* %_DEST%\inc%_EXT%\boost\spirit\home\classic\dynamic
+..\%__SRC%\misc\build\boost_1_39_0\boost\spirit\home\classic\dynamic\impl\* %_DEST%\inc%_EXT%\boost\spirit\home\classic\dynamic\impl
+..\%__SRC%\misc\build\boost_1_39_0\boost\spirit\home\classic\phoenix\* %_DEST%\inc%_EXT%\boost\spirit\home\classic\phoenix
+..\%__SRC%\misc\build\boost_1_39_0\boost\spirit\home\classic\symbols\* %_DEST%\inc%_EXT%\boost\spirit\home\classic\symbols
+..\%__SRC%\misc\build\boost_1_39_0\boost\spirit\home\classic\symbols\impl\* %_DEST%\inc%_EXT%\boost\spirit\home\classic\symbols\impl
-- 
1.7.0.1

