diff --git sc/source/filter/xlsx/xlsx-xestyle.cxx sc/source/filter/xlsx/xlsx-xestyle.cxx
index d882080..f60c37f 100644
--- sc/source/filter/xlsx/xlsx-xestyle.cxx
+++ sc/source/filter/xlsx/xlsx-xestyle.cxx
@@ -1557,9 +1557,26 @@ void lclGetBorderLine(
         else if( nOuterWidth > DEF_LINE_WIDTH_2 )
             rnXclLine = EXC_LINE_THICK;
         else if( nOuterWidth > DEF_LINE_WIDTH_1 )
+        {
             rnXclLine = EXC_LINE_MEDIUM;
+            if ( pLine->GetStyle( ) == EXC_LINE_DASHED )
+                rnXclLine = EXC_LINE_MEDIUMDASHED;
+        }
         else if( nOuterWidth > DEF_LINE_WIDTH_0 )
+        {
             rnXclLine = EXC_LINE_THIN;
+            switch ( pLine->GetStyle( ) )
+            {
+                case DOTTED:
+                    rnXclLine = EXC_LINE_DOTTED;
+                    break;
+                case DASHED:
+                    rnXclLine = EXC_LINE_DASHED;
+                    break;
+                default:
+                    break;
+            }
+        }
         else if( nOuterWidth > 0 )
             rnXclLine = EXC_LINE_HAIR;
         else
@@ -1725,6 +1742,9 @@ static const char* ToLineStyle( sal_uInt8 nLineStyle )
         case EXC_LINE_THICK:    return "thick";
         case EXC_LINE_DOUBLE:   return "double";
         case EXC_LINE_HAIR:     return "hair";
+        case EXC_LINE_DOTTED:   return "dotted";
+        case EXC_LINE_DASHED:   return "dashed";
+        case EXC_LINE_MEDIUMDASHED:   return "mediumdashed";
     }
     return "*unknown*";
 }
