--- /dev/null	2009-10-09 20:25:00.000000000 +0200
+++ cpputools/source/registercomponent/regcomp.sh	2009-10-09 18:21:01.000000000 +0200
@@ -0,0 +1,51 @@
+#!/bin/sh
+#*************************************************************************
+#
+# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+# 
+# Copyright 2008 by Sun Microsystems, Inc.
+#
+# OpenOffice.org - a multi-platform office productivity suite
+#
+# $RCSfile: startup.sh,v $
+#
+# $Revision: 1.4.10.1 $
+#
+# This file is part of OpenOffice.org.
+#
+# OpenOffice.org is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License version 3
+# only, as published by the Free Software Foundation.
+#
+# OpenOffice.org is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License version 3 for more details
+# (a copy is included in the LICENSE file that accompanied this code).
+#
+# You should have received a copy of the GNU Lesser General Public License
+# version 3 along with OpenOffice.org.  If not, see
+# <http://www.openoffice.org/license.html>
+# for a copy of the LGPLv3 License.
+#
+#*************************************************************************
+
+# Pass -env arguments on to javaldx; needs to be fixed:
+my_envargs=
+for my_arg in "$@" ; do
+  case ${my_arg} in
+    -env:*) my_envargs="${my_envargs} ${my_arg}" ;;
+  esac
+done
+
+# Extend the LD_LIBRARY_PATH for Java:
+epath=`dirname "$0"`
+if [ -x "${epath}/javaldx" ] ; then
+    jpath=`"${epath}/javaldx" $my_envargs`
+    if [ -n "${jpath}" ]; then
+        LD_LIBRARY_PATH=${jpath}${LD_LIBRARY_PATH+:${LD_LIBRARY_PATH}}
+        export LD_LIBRARY_PATH
+    fi
+fi
+
+exec "$0.bin" "$@"
--- cpputools/source/registercomponent/makefile.mk.old	2009-09-25 00:39:01.000000000 +0200
+++ cpputools/source/registercomponent/makefile.mk	2009-10-09 20:24:02.000000000 +0200
@@ -45,6 +45,7 @@ UNOUCRRDB=$(SOLARBINDIR)$/udkapi.rdb
 NO_OFFUH=TRUE
 CPPUMAKERFLAGS += -C
 
+UNIXTEXT= $(MISC)$/regcomp.sh
 UNOTYPES=\
              com.sun.star.uno.TypeClass \
              com.sun.star.lang.XMultiServiceFactory \
@@ -77,3 +78,10 @@ APP1STDLIBS+= \
 
 
 .INCLUDE :  target.mk
+
+.IF "$(GUI)"=="UNX"
+ALLTAR: REGCOMPSH
+
+REGCOMPSH : $(UNIXTEXT)
+	+-chmod +x $(UNIXTEXT)
+.ENDIF
--- cpputools/prj/d.lst.old	2009-09-25 00:39:01.000000000 +0200
+++ cpputools/prj/d.lst	2009-10-09 20:25:39.000000000 +0200
@@ -1,5 +1,5 @@
 ..\%__SRC%\bin\regcomp.exe %_DEST%\bin%_EXT%\regcomp.exe
-..\%__SRC%\bin\regcomp %_DEST%\bin%_EXT%\regcomp
+..\%__SRC%\misc\regcomp.sh %_DEST%\bin%_EXT%\regcomp
 ..\%__SRC%\bin\regcomp %_DEST%\bin%_EXT%\regcomp.bin
 ..\%__SRC%\bin\regcomplazy.exe %_DEST%\bin%_EXT%\regcomplazy.exe
 ..\%__SRC%\bin\regcomplazy %_DEST%\bin%_EXT%\regcomplazy
