From f62f8d1a420b65181dc5263a63e21fbef398e8c6 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:02:18 +0200
Subject: [PATCH 508/878] build-fix-layout-numfmt.diff

---
 svx/source/cui/numfmt.cxx |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/svx/source/cui/numfmt.cxx b/svx/source/cui/numfmt.cxx
index ef213e8..79e94fb 100644
--- a/svx/source/cui/numfmt.cxx
+++ b/svx/source/cui/numfmt.cxx
@@ -25,8 +25,8 @@
  *
  ************************************************************************/
 
-#if !ENABLE_LAYOUT_EXPERIMENTAL
-//#undef ENABLE_LAYOUT
+#if !ENABLE_LAYOUT_EXPERIMENTAL && defined (ENABLE_LAYOUT)
+#undef ENABLE_LAYOUT
 #endif
 
 // MARKER(update_precomp.py): autogen include statement, do not remove
-- 
1.7.0.1

