diff --git sd/source/ui/inc/res_bmp.hrc sd/source/ui/inc/res_bmp.hrc
index ec8bb6b..89841d7 100644
--- sd/source/ui/inc/res_bmp.hrc
+++ sd/source/ui/inc/res_bmp.hrc
@@ -308,8 +308,8 @@
 
 #define RID_DRAW_TABLEOBJ_INSIDE_POPUP		RID_APP_START+94
 
-#define RID_DRAW_OUTLINETEXTOBJ_POPUP		RID_APP_START+94
-#define RID_GRAPHIC_OUTLINETEXTOBJ_POPUP	RID_APP_START+95
+#define RID_DRAW_OUTLINETEXTOBJ_POPUP		RID_APP_START+95
+#define RID_GRAPHIC_OUTLINETEXTOBJ_POPUP	RID_APP_START+96
 
 /*
  * Pointer (Mauszeiger)
diff --git sc/source/ui/view/cellsh1.cxx sc/source/ui/view/cellsh1.cxx
index d6fdb75..57c4740 100644
--- sc/source/ui/view/cellsh1.cxx
+++ sc/source/ui/view/cellsh1.cxx
@@ -2205,50 +2205,6 @@ IMPL_LINK( ScCellShell, DialogClosed, AbstractScLinkedAreaDlg*, EMPTYARG )
     return 0;
 }
 
-void ScCellShell::PasteFromClipboard( ScViewData* pViewData, ScTabViewShell* pTabViewShell, BOOL bShowDialog )
-{
-    Window* pWin = pViewData->GetActiveWin();
-    ScTransferObj* pOwnClip = ScTransferObj::GetOwnClipboard( pWin );
-    ScDocument* pThisDoc = pViewData->GetDocument();
-    ScDPObject* pDPObj = pThisDoc->GetDPAtCursor( pViewData->GetCurX(),
-                         pViewData->GetCurY(), pViewData->GetTabNo() );
-    if ( pOwnClip && pDPObj )
-    {
-        // paste from Calc into DataPilot table: sort (similar to drag & drop)
-
-        ScDocument* pClipDoc = pOwnClip->GetDocument();
-        SCTAB nSourceTab = pOwnClip->GetVisibleTab();
-
-        SCCOL nClipStartX;
-        SCROW nClipStartY;
-        SCCOL nClipEndX;
-        SCROW nClipEndY;
-        pClipDoc->GetClipStart( nClipStartX, nClipStartY );
-        pClipDoc->GetClipArea( nClipEndX, nClipEndY, TRUE );
-        nClipEndX = nClipEndX + nClipStartX;
-        nClipEndY = nClipEndY + nClipStartY;   // GetClipArea returns the difference
-
-        ScRange aSource( nClipStartX, nClipStartY, nSourceTab, nClipEndX, nClipEndY, nSourceTab );
-        BOOL bDone = pTabViewShell->DataPilotMove( aSource, pViewData->GetCurPos() );
-        if ( !bDone )
-            pTabViewShell->ErrorMessage( STR_ERR_DATAPILOT_INPUT );
-    }
-    else
-    {
-        // normal paste
-        WaitObject aWait( pViewData->GetDialogParent() );
-        if (!pOwnClip)
-            pTabViewShell->PasteFromSystem();
-        else
-        {
-            pTabViewShell->PasteFromClip( IDF_ALL, pOwnClip->GetDocument(),
-                    PASTE_NOFUNC, FALSE, FALSE, FALSE, INS_NONE, IDF_NONE,
-                    bShowDialog );		// allow warning dialog
-        }
-    }
-    pTabViewShell->CellContentChanged();		// => PasteFromSystem() ???
-}
-
 void ScCellShell::PasteFromClipboard( ScViewData* pViewData, ScTabViewShell* pTabViewShell, bool bShowDialog )
 {
     Window* pWin = pViewData->GetActiveWin();
diff --git sc/source/ui/view/gridwin.cxx sc/source/ui/view/gridwin.cxx
index 07d5d14..7a13b81 100644
--- sc/source/ui/view/gridwin.cxx
+++ sc/source/ui/view/gridwin.cxx
@@ -5452,7 +5452,7 @@ void ScGridWindow::UpdateAutoFillOverlay()
                 aHandleColor = SC_MOD()->GetColorConfig().GetColorValue(svtools::CALCPAGEBREAKAUTOMATIC).nColor;
             std::vector< basegfx::B2DRange > aRanges;
             const basegfx::B2DHomMatrix aTransform(GetInverseViewTransformation());
-            basegfx::B2DRange aRB(aFillRect.Left(), aFillRect.Top(), aFillRect.Right() + 1, aFillRect.Bottom() + 1);
+            basegfx::B2DRange aRB(mpAutoFillRect->Left(), mpAutoFillRect->Top(), mpAutoFillRect->Right() + 1, mpAutoFillRect->Bottom() + 1);
             
             aRB.transform(aTransform);
             aRanges.push_back(aRB);
