From 6fe15dd938f60a54ebca86a74109282e7ef46dd5 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:54:10 +0200
Subject: [PATCH 074/768] build-ant-1.6.diff

---
 configure.in |    4 ----
 1 files changed, 0 insertions(+), 4 deletions(-)

diff --git configure.in configure.in
index 504bf63..5878fb7 100644
--- configure.in
+++ configure.in
@@ -6612,10 +6612,6 @@ else
   fi
 
   ant_minver=1.6.0
-  # update for more extensions...
-  if test "$ENABLE_MEDIAWIKI" = "YES"; then
-    ant_minver=1.7.0
-  fi
   ant_minminor1=`echo $ant_minver | cut -d"." -f2`
 
   AC_MSG_CHECKING([whether ant is >= $ant_minver])
-- 
1.7.0.1

