--- configure.in.old	2010-07-15 12:59:44.000000000 +0200
+++ configure.in	2010-07-15 13:03:35.000000000 +0200
@@ -3797,10 +3797,8 @@ if test -n "$with_system_cppunit" -o -n
         test "$with_system_cppunit" != "no"; then
     AC_MSG_RESULT([external])
     SYSTEM_CPPUNIT=YES
-    # might work for earlier, too but go sure. We didn't have
-    # a system-cppunit before the first version using a proper cppunit
-    # (and that being 1.12.1) anyway
-    PKG_CHECK_MODULES( CPPUNIT, cppunit >= 1.12.1 )
+    # might work for earlier, too but go sure
+    PKG_CHECK_MODULES( CPPUNIT, cppunit >= 1.12.0 )
     AC_MSG_CHECKING([STL compatibility])
     if test "$WITH_STLPORT" != "no"; then
 	AC_MSG_ERROR([to use system cppunit you need to use --without-stlport])
