--- l10ntools/java/jpropex/java/JPropEx.java.old	2010-11-11 16:22:47.000000000 +0100
+++ l10ntools/java/jpropex/java/JPropEx.java	2010-12-22 14:36:51.000000000 +0100
@@ -143,11 +143,14 @@ public class JPropEx
     {
         String path = makeAbs( filename ).trim();
         String myRootArg = makeAbs( rootArg ).trim();
-        myRootArg = myRootArg.replace( "\\","/");
+        // FIXME: we use .replaceAll only with Java 1.4
+        //        it is not safe because rootArg should not be treated as regular expression
+        //        it is only a helper tool, so the hack with replaceAll might be acceptable just for building on the very old system
+        myRootArg = myRootArg.replaceAll( "\\","/");
         myRootArg += "/";
-        path = path.replace("\\","/");
-        path = path.replace( myRootArg, "" );
-        path = path.replace("/","\\");
+        path = path.replaceAll("\\","/");
+        path = path.replaceAll( myRootArg, "" );
+        path = path.replaceAll("/","\\");
         // TODO: Make this static
         java.text.SimpleDateFormat dateformat = new java.text.SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
         String date = dateformat.format( new Date() );
