--- boost.m4.old	2010-12-21 21:52:42.000000000 +0100
+++ boost.m4	2010-12-22 12:06:23.000000000 +0100
@@ -1015,8 +1015,8 @@ boost_use_source=:
 # If we already have a .o, re-use it.  We change $ac_ext so that $ac_link
 # tries to link the existing object file instead of compiling from source.
 test -f conftest.$ac_objext && ac_ext=$ac_objext && boost_use_source=false &&
-  _AS_ECHO_LOG([re-using the existing conftest.$ac_objext])
-AS_IF([_AC_DO_STDERR($ac_link) && {
+  AC_MSG_NOTICE([re-using the existing conftest.$ac_objext])
+AS_IF([_AC_EVAL_STDERR($ac_link) && {
 	 test -z "$ac_[]_AC_LANG_ABBREV[]_werror_flag" ||
 	 test ! -s conftest.err
        } && test -s conftest$ac_exeext && {
