--- transex3/prj/build.lst-old	2009-09-29 00:42:03.000000000 +0200
+++ transex3/prj/build.lst	2009-09-29 00:40:57.000000000 +0200
@@ -1,4 +1,4 @@
-tr	transex3	:	tools libxslt berkeleydb lucene NULL
+tr	transex3	:	tools libxslt berkeleydb LUCENE:lucene NULL
 tr	transex3						usr1	-	all	tr_mkout NULL
 tr	transex3\inc					nmake	-	all	tr_inc NULL
 tr	transex3\source					nmake	-	all	tr_src tr_inc NULL
--- transex3/source/help/makefile.mk-old	2009-09-29 01:34:53.000000000 +0200
+++ transex3/source/help/makefile.mk	2009-09-29 01:31:23.000000000 +0200
@@ -112,7 +112,7 @@
 
 #JARFILES  = ridl.jar jurt.jar unoil.jar juh.jar
 .IF "$(SYSTEM_LUCENE)" == "YES"
-XCLASSPATH!:=$(XCLASSPATH)$(PATH_SEPERATOR)$(LUCENE_CORE_JAR)$(PATH_SEPERATOR)$(LUCENE_ANALYZERS_JAR)
+CLASSPATH!:=$(CLASSPATH)$(PATH_SEPERATOR)$(LUCENE_CORE_JAR)$(PATH_SEPERATOR)$(LUCENE_ANALYZERS_JAR)
 COMP=fix_system_lucene
 .ELSE
 JARFILES += lucene-core-2.3.jar lucene-analyzers-2.3.jar
@@ -136,4 +136,4 @@
 fix_system_lucene:
     @echo "Fix Java Class-Path entry for Lucene libraries from system."
     @$(SED) -r -e "s#^(Class-Path:).*#\1 file://$(LUCENE_CORE_JAR) file://$(LUCENE_ANALYZERS_JAR)#" \
-    -i ../../../../../$(INPATH)/class/HelpLinker/META-INF/MANIFEST.MF
+    -i ../../$(INPATH)/class/HelpLinker/META-INF/MANIFEST.MF
