diff --git a/instsetoo_native/util/makefile.mk b/instsetoo_native/util/makefile.mk
--- instsetoo_native/util/makefile.mk
+++ instsetoo_native/util/makefile.mk
@@ -201,6 +201,7 @@
 
 .ENDIF			# "$(BUILD_SPECIAL)"!=""
 
+.IF "$(PKGFORMAT)" != ""
 $(foreach,i,$(alllangiso) openoffice_$i) : $$@{$(PKGFORMAT:^".")}
 .IF "$(MAKETARGETS)"!=""
 .IF "$(MAKETARGETS:e)"=="" && "$(MAKETARGETS:s/_//)"!="$(MAKETARGETS)"
@@ -277,6 +278,7 @@
 	@echo cannot pack nothing...
 
 .ENDIF			# "$(alllangiso)"!=""
+.ENDIF
 
 .IF "$(LOCALPYFILES)"!=""
 $(foreach,i,$(alllangiso) openoffice_$i{$(PKGFORMAT:^".") .archive} openofficewithjre_$i{$(PKGFORMAT:^".")} openofficedev_$i{$(PKGFORMAT:^".")} broffice_$i{$(PKGFORMAT:^".")} brofficewithjre_$i{$(PKGFORMAT:^".")} brofficedev_$i{$(PKGFORMAT:^".")} sdkoo_$i{$(PKGFORMAT:^".")}) updatepack : $(LOCALPYFILES)
