--- sd/prj/build.lst
+++ sd/prj/build.lst
@@ -30,7 +30,6 @@
 sd	sd\source\filter						nmake	-	all	sd_filt sd_sdi sd_ppt sd_inc NULL
 sd	sd\source\filter\cgm					nmake	-	all	sd_cgm sd_inc NULL
 sd	sd\source\filter\eppt					nmake	-	all	sd_eppt sd_inc NULL
-sd	sd\source\filter\pptx					nmake	-	all	sd_pptx sd_inc NULL
 sd	sd\source\filter\grf					nmake	-	all	sd_grf sd_inc NULL
 sd	sd\source\filter\xml					nmake	-	all	sd_xml sd_inc NULL
 sd	sd\source\filter\html					nmake	-	all	sd_html sd_inc NULL
@@ -41,4 +40,4 @@
 sd	sd\source\ui\framework\module       	nmake	-	all sd_framework_module sd_inc NULL
 sd	sd\source\ui\framework\factories       	nmake	-	all sd_framework_factories sd_inc NULL
 sd	sd\source\ui\framework\tools           	nmake	-	all sd_framework_tools sd_inc NULL
-sd	sd\util									nmake	-	all	sd_util sd_app sd_cgm sd_core sd_dlg sd_docsh sd_eppt sd_pptx sd_filt sd_func sd_grf sd_unid sd_view sd_xml sd_html sd_ppt sd_accessibility sd_animations sd_toolpanel sd_toolpanel_controls sd_tools sd_slsshell sd_slsmodel sd_slsview sd_slscontroller sd_slscache sd_notes sd_table sd_slideshow sd_presenter sd_undo sd_helper sd_framework_configuration sd_framework_module sd_framework_tools sd_framework_factories NULL
+sd	sd\util									nmake	-	all	sd_util sd_app sd_cgm sd_core sd_dlg sd_docsh sd_eppt sd_filt sd_func sd_grf sd_unid sd_view sd_xml sd_html sd_ppt sd_accessibility sd_animations sd_toolpanel sd_toolpanel_controls sd_tools sd_slsshell sd_slsmodel sd_slsview sd_slscontroller sd_slscache sd_notes sd_table sd_slideshow sd_presenter sd_undo sd_helper sd_framework_configuration sd_framework_module sd_framework_tools sd_framework_factories NULL
--- scp2/source/ooo/file_library_ooo.scp
+++ scp2/source/ooo/file_library_ooo.scp
@@ -1922,6 +1922,12 @@
     Name = SCP2_URE_DL_NORMAL("msvcr80");
     Styles = (PACKED, VERSION_INDEPENDENT_COMP_ID);
 End
+File gid_File_Moz_Msvc80crt_Manifest
+    BIN_FILE_BODY;
+    Styles = (PACKED);
+    Dir = gid_Dir_Program;
+    Name = "Microsoft.VC80.CRT.manifest";
+End
 #endif
 
 #ifndef SYSTEM_REDLAND
--- scp2/source/ooo/common_brand.scp
+++ scp2/source/ooo/common_brand.scp
@@ -504,18 +504,6 @@
     Name = "unopkg.bin";
 End
 
-#if defined WNT && \
-    (defined M1310 || \
-     ((defined M1400 || defined M1500) && \
-      !(defined SYSTEM_MOZILLA || defined WITHOUT_MOZILLA)))
-File gid_Brand_File_Bin_Msvcr71
-    TXT_FILE_BODY;
-    Dir = gid_Brand_Dir_Program;
-    Name = SCP2_URE_DL_NORMAL("msvcr71" SCP2_STDLP_DEBUG);
-    Styles = (PACKED);
-End
-#endif
-
 // Due to Windows DLL search order, OOo executables might pick up a libxml2 DLL
 // from a Windows system directory instead of the one from the URE; to avoid
 // problems, as a HACK, for now a copy of the URE libxml2 DLL is placed next to
--- external/prj/d.lst
+++ external/prj/d.lst
@@ -34,8 +34,8 @@
 ..\msvcp80\msvcm80*.dll %_DEST%\bin%_EXT%
 ..\msvcp80\msvcp80*.dll %_DEST%\bin%_EXT%
 ..\msvcp80\msvcr80*.dll %_DEST%\bin%_EXT%
-..\msvcp90\Microsoft.VC80.CRT.manifest %_DEST%\bin%_EXT%\Microsoft.VC80.CRT.manifest
-..\msvcp90\Microsoft.VC80.DebugCRT.manifest %_DEST%\bin%_EXT%\Microsoft.VC80.DebugCRT.manifest
+..\msvcp80\Microsoft.VC80.CRT.manifest %_DEST%\bin%_EXT%\Microsoft.VC80.CRT.manifest
+..\msvcp80\Microsoft.VC80.DebugCRT.manifest %_DEST%\bin%_EXT%\Microsoft.VC80.DebugCRT.manifest
 
 ..\msvcp90\msvcm90*.dll %_DEST%\bin%_EXT%
 ..\msvcp90\msvcp90*.dll %_DEST%\bin%_EXT%
--- external/msvcp80/README_msvcX80.dll
+++ external/msvcp80/README_msvcX80.dll
@@ -1,2 +1,3 @@
-Put the msvcp80.dll and msvcr80.dll in this directory for Windows builds using
-a VS 2005 compiler.
+Put the msvcp80.dll, msvcr80.dll and Microsoft.VC80.CRT.manifest in this
+directory for Windows builds using VS 2005, or using prebuilt Mozilla DLLs
+that were built using VS 2005.
