diff --git sd/source/ui/inc/res_bmp.hrc sd/source/ui/inc/res_bmp.hrc
index ec8bb6b..89841d7 100644
--- sd/source/ui/inc/res_bmp.hrc
+++ sd/source/ui/inc/res_bmp.hrc
@@ -308,8 +308,8 @@
 
 #define RID_DRAW_TABLEOBJ_INSIDE_POPUP		RID_APP_START+94
 
-#define RID_DRAW_OUTLINETEXTOBJ_POPUP		RID_APP_START+94
-#define RID_GRAPHIC_OUTLINETEXTOBJ_POPUP	RID_APP_START+95
+#define RID_DRAW_OUTLINETEXTOBJ_POPUP		RID_APP_START+95
+#define RID_GRAPHIC_OUTLINETEXTOBJ_POPUP	RID_APP_START+96
 
 /*
  * Pointer (Mauszeiger)
