From 48274e0d5ee36bd996825e58bbff60585a1c8c1a Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:57:49 +0200
Subject: [PATCH 251/768] build-enable-boost-1.36.diff

---
 configure.in |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git configure.in configure.in
index 6690502..77b2ea5 100644
--- configure.in
+++ configure.in
@@ -4318,7 +4318,7 @@ if test -n "$with_system_boost" -o -n "$with_system_headers" && \
    AC_CHECK_HEADER(boost/shared_ptr.hpp, [],
        [AC_MSG_ERROR(boost/shared_ptr.hpp not found. install boost)], [])
    AC_CHECK_HEADER(boost/spirit/include/classic_core.hpp, [],
-       [AC_MSG_ERROR(boost/spirit/include/classic_core.hpp not found. install boost >= 1.38)], [])
+       [AC_MSG_ERROR(boost/spirit/include/classic_core.hpp not found. install boost >= 1.36)], [])
    AC_CHECK_HEADER(boost/function.hpp, [],
        [AC_MSG_ERROR(boost/function.hpp not found. install boost)], [])
 
-- 
1.7.0.1

