--- l10ntools/java/jpropex/java/JPropEx.java.old	2010-07-22 13:49:22.000000000 +0200
+++ l10ntools/java/jpropex/java/JPropEx.java	2010-08-06 19:27:55.000000000 +0200
@@ -139,8 +139,11 @@ public class JPropEx
     {
         String path = makeAbs( filename );
         //String path = makeAbs( inputFileArg );
-        path = path.replace( rootArg + "/" , "" );
-        path = path.replace("/","\\");
+        // FIXME: we use .replaceAll only with Java 1.4
+        //        it is not safe because rootArg should not be treated as regular expression
+        //        it is only a helper tool, so the hack with replaceAll might be acceptable just for building on the very old system
+        path = path.replaceAll( rootArg + "/" , "" );
+        path = path.replaceAll("/","\\");
         return new SdfEntity( projectArg , path , "" /* dummy1 */ , resourceType , "", "" , "" , "" , "" /* dummy2 */ , 
                               sourceLanguage , "",  "" , ""  , "" , "2002-02-02 02:02:02" );
     }
