--- toolkit/workben/layout/TEST
+++ toolkit/workben/layout/TEST
@@ -1,5 +1,5 @@
 #! /bin/sh
-for d in $(cd ../../../../.. && pwd)/ooo-svn /usr/lib/ooo3/; do
+for d in $(cd ../../../../.. && pwd)/ooo-git /usr/lib/ooo3/; do
     if test -d $d; then
         export OOO_INSTALL_PREFIX=$d
 	break
--- toolkit/workben/layout/test.cxx
+++ toolkit/workben/layout/test.cxx
@@ -199,6 +199,11 @@ void LayoutTest::ParseCommandLine()
         OUString aParam = OUString( GetCommandLineParam( i ) );
         if ( aParam.equalsAsciiL( RTL_CONSTASCII_STRINGPARAM( "-h" ) ) || aParam.equalsAsciiL( RTL_CONSTASCII_STRINGPARAM( "--help" ) ) )
             usage();
+        if ( aParam.equalsAscii( "--oustring-cstr"))
+            exit (printf ("%s: The strings: %s and %s.\n",
+                          __PRETTY_FUNCTION__,
+                          OUSTRING_CSTR (OUString::createFromAscii ("foo")),
+                          OUSTRING_CSTR (OUString::createFromAscii ("bar"))));
         if ( aParam.equalsAsciiL( RTL_CONSTASCII_STRINGPARAM( "--inst" ) ) )
         {
             if ( i >= GetCommandLineParamCount() - 1)
--- toolkit/workben/layout/tpsort.cxx
+++ toolkit/workben/layout/tpsort.cxx
@@ -442,7 +442,7 @@ void ScTabPageSortFields::FillFieldLists()
                     {
                         aFieldName	= aStrColumn;
                         aFieldName += ' ';
-                        aFieldName += ColToAlpha( col );
+                        aFieldName += ScColToAlpha( col );
                     }
                     nFieldArr[i] = col;
                     aLbSort1.InsertEntry( aFieldName, i );
