diff --git extensions/source/logging/consolehandler.cxx extensions/source/logging/consolehandler.cxx
index 2eff518..a280a9d 100644
--- extensions/source/logging/consolehandler.cxx
+++ extensions/source/logging/consolehandler.cxx
@@ -249,9 +249,9 @@ namespace logging
             return sal_False;
 
         if ( _rRecord.Level >= m_nThreshold )
-            fprintf( stderr, sEntry.getStr() );
+            fprintf( stderr, "%s", sEntry.getStr() );
         else
-            fprintf( stdout, sEntry.getStr() );
+            fprintf( stdout, "%s", sEntry.getStr() );
 
         return sal_True;
     }
diff --git fpicker/source/unx/gnome/SalGtkFilePicker.cxx fpicker/source/unx/gnome/SalGtkFilePicker.cxx
index 4b38593..3c11cc4 100644
--- fpicker/source/unx/gnome/SalGtkFilePicker.cxx
+++ fpicker/source/unx/gnome/SalGtkFilePicker.cxx
@@ -1058,7 +1058,7 @@ sal_Int16 SAL_CALL SalGtkFilePicker::execute() throw( uno::RuntimeException )
                                 GTK_DIALOG_MODAL,
                                 GTK_MESSAGE_QUESTION,
                                 GTK_BUTTONS_YES_NO,
-                                  OUStringToOString(
+                                  "\"%s\"", OUStringToOString(
                                     aResProvider.getResString( FILE_PICKER_OVERWRITE ),
                                     RTL_TEXTENCODING_UTF8 ).getStr() );
 
diff --git icc/makefile.mk icc/makefile.mk
index 87ebf3a..4494fe5 100644
--- icc/makefile.mk
+++ icc/makefile.mk
@@ -46,7 +46,7 @@ CONVERTFILES= \
     Contrib$/ICC_utils$/Stubs.h \
     Contrib$/ICC_utils$/Vetters.cpp
 
-CONFIGURE_ACTION= $(GNUCOPY) -r $(BACK_PATH)..$/source$/create_sRGB_profile Contrib$/CmdLine && unzip -o $(BACK_PATH)..$/makefiles.zip
+CONFIGURE_ACTION= $(GNUCOPY) -r $(BACK_PATH)..$/source$/create_sRGB_profile Contrib$/CmdLine && unzip -o $(BACK_PATH)..$/makefiles.zip && patch -p0 < $(BACK_PATH)..$/mdv-fmtargs-buildfix.patch
 BUILD_ACTION=dmake &&  cd Contrib$/CmdLine$/create_sRGB_profile && $(AUGMENT_LIBRARY_PATH) .$/create_sRGB_profile
 
 # --- Targets ------------------------------------------------------
diff --git icc/mdv-fmtargs-buildfix.patch icc/mdv-fmtargs-buildfix.patch
new file mode 100644
index 0000000..5774e36
--- /dev/null
+++ icc/mdv-fmtargs-buildfix.patch
@@ -0,0 +1,20 @@
+--- IccProfLib/IccProfile.cpp  2009-01-20 11:48:36.000000000 -0500
++++ IccProfile.cpp     2009-01-19 14:41:23.000000000 -0500
+@@ -1155,7 +1155,7 @@
+
+   CIccInfo Info;
+   icChar buf[128];
+-  sprintf(buf, Info.GetSigName(m_Header.deviceClass));
++  sprintf(buf, "%s", Info.GetSigName(m_Header.deviceClass));
+   if (m_Header.deviceClass!=icSigInputClass && m_Header.deviceClass!=icSigDisplayClass) {
+     if (GetTag(icSigGrayTRCTag) || GetTag(icSigRedTRCTag) || GetTag(icSigGreenTRCTag) ||
+        GetTag(icSigBlueTRCTag) || GetTag(icSigRedColorantTag) || GetTag(icSigGreenColorantTag) ||
+@@ -1230,7 +1230,7 @@
+   for (i=m_Tags->begin(); i!=m_Tags->end(); i++) {
+     tagsig = i->TagInfo.sig;
+     typesig = i->pTag->GetType();
+-    sprintf(buf, Info.GetSigName(tagsig));
++    sprintf(buf, "%s", Info.GetSigName(tagsig));
+     if (!IsTypeValid(tagsig, typesig)) {
+       sReport += icValidateNonCompliantMsg;
+       sReport += buf;
