From b7e4dcdd277444f567e880e2d256b140cfa39a14 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:02:16 +0200
Subject: [PATCH 506/878] buildfix-svx-uiconfig-layout.diff

---
 svx/uiconfig/layout/layout.mk |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/svx/uiconfig/layout/layout.mk b/svx/uiconfig/layout/layout.mk
index f6bea87..a79ae67 100644
--- a/svx/uiconfig/layout/layout.mk
+++ b/svx/uiconfig/layout/layout.mk
@@ -1,7 +1,7 @@
 # TODO: move to solenv/inc
 # copies: sw/uiconfig/layout svx/uiconfig/layout
 
-TRALAY=tralay
+TRALAY=$(AUGMENT_LIBRARY_PATH) $(SOLARBINDIR)/tralay
 XML_DEST=$(DLLDEST)
 XML_LANGS=$(alllangiso)
 
-- 
1.7.0.1

