---
 cui/source/tabpages/numfmt.cxx |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git cui/source/tabpages/numfmt.cxx cui/source/tabpages/numfmt.cxx
index 794d41f..c58c45c 100644
--- cui/source/tabpages/numfmt.cxx
+++ cui/source/tabpages/numfmt.cxx
@@ -25,8 +25,8 @@
  *
  ************************************************************************/
 
-#if !ENABLE_LAYOUT_EXPERIMENTAL
-//#undef ENABLE_LAYOUT
+#if !ENABLE_LAYOUT_EXPERIMENTAL && defined (ENABLE_LAYOUT)
+#undef ENABLE_LAYOUT
 #endif
 
 // MARKER(update_precomp.py): autogen include statement, do not remove
-- 
1.7.0.1

