--- sdext/source/pdfimport/xpdftest/makefile.mk.orig	2008-09-27 15:55:08.000000000 +0200
+++ sdext/source/pdfimport/xpdftest/makefile.mk	2008-09-27 15:55:43.000000000 +0200
@@ -39,6 +39,11 @@
 
 .INCLUDE: settings.mk
 
+.IF "$(ENABLE_PDFIMPORT)" == "NO"
+@all:
+	@echo "PDF Import extension disabled."
+.ENDIF
+
 # --- Targets ------------------------------------------------------
 
 .INCLUDE :  target.mk
--- sdext/source/pdfimport/xpdfwrapper/makefile.mk.orig	2008-09-27 15:55:00.000000000 +0200
+++ sdext/source/pdfimport/xpdfwrapper/makefile.mk	2008-09-27 15:56:03.000000000 +0200
@@ -47,6 +47,11 @@
 CFLAGS += -I$(SOLARINCDIR)$/xpdf
 .ENDIF
 
+.IF "$(ENABLE_PDFIMPORT)" == "NO"
+@all:
+	@echo "PDF Import extension disabled."
+.ENDIF
+
 # --- Files --------------------------------------------------------
 UWINAPILIB:=
 APP1TARGET=$(TARGET)
