--- accessibility/workben/org/openoffice/accessibility/awb/tree/AccessibilityNode.java.old	2010-08-25 22:34:06.000000000 +0200
+++ accessibility/workben/org/openoffice/accessibility/awb/tree/AccessibilityNode.java	2010-09-07 20:15:45.000000000 +0200
@@ -75,7 +75,7 @@ class AccessibilityNode extends DefaultM
         if (xAccessibleContext != null) {
             try {
                 XAccessibleEventBroadcaster xAccessibleEventBroadcaster = 
-                    UnoRuntime.queryInterface( XAccessibleEventBroadcaster.class, xAccessibleContext );
+                    (XAccessibleEventBroadcaster) UnoRuntime.queryInterface( XAccessibleEventBroadcaster.class, xAccessibleContext );
                 if (xAccessibleEventBroadcaster != null) {
                     if (attach) {
                         xAccessibleEventBroadcaster.addEventListener(this);
@@ -134,12 +134,12 @@ class AccessibilityNode extends DefaultM
     
     public void notifyEvent(AccessibleEventObject accessibleEventObject) {
         if (accessibleEventObject.EventId == AccessibleEventId.CHILD) {
-            XAccessible xAccessible = UnoRuntime.queryInterface( XAccessible.class, accessibleEventObject.OldValue );
+            XAccessible xAccessible = (XAccessible) UnoRuntime.queryInterface( XAccessible.class, accessibleEventObject.OldValue );
             if (xAccessible != null) {
                 handleChildRemoved(xAccessible);
             }
 
-            xAccessible = UnoRuntime.queryInterface( XAccessible.class, accessibleEventObject.NewValue );
+            xAccessible = (XAccessible) UnoRuntime.queryInterface( XAccessible.class, accessibleEventObject.NewValue );
             if (xAccessible != null) {
                 handleChildAdded(xAccessible);
             }
