--- cairo/pixman/makefile.mk
+++ cairo/pixman/makefile.mk
@@ -89,7 +89,7 @@
 .ELSE
 # ----------- Unix ---------------------------------------------------------
 .IF "$(OS)$(COM)"=="LINUXGCC" || "$(OS)$(COM)"=="FREEBSDGCC"
-LDFLAGS:=-Wl,-rpath,'$$$$ORIGIN:$$$$ORIGIN/../ure-link/lib' -Wl,-noinhibit-exec -Wl,-z,noexecstack
+LDFLAGS:=-Wl,-rpath,'$$$$ORIGIN:$$$$ORIGIN/../ure-link/lib' -Wl,-z,noexecstack
 .ENDIF                  # "$(OS)$(COM)"=="LINUXGCC"

 .IF "$(OS)$(COM)"=="SOLARISC52"
@@ -135,7 +135,7 @@
 .ENDIF
 .ELSE
 .IF "$(CPUNAME)"=="X86_64"
-OUT2LIB+=pixman$/.libs$/libpixman-1.so
+OUT2LIB+=pixman$/.libs$/libpixman-1.so*
 .ELSE
 OUT2LIB+=pixman$/.libs$/libpixman-1.a
 .ENDIF
