diff --git sc/inc/table.hxx sc/inc/table.hxx
index c6c41f1..c7836c5 100644
--- sc/inc/table.hxx
+++ sc/inc/table.hxx
@@ -638,7 +638,7 @@ public:
     void		DBShowRow(SCROW nRow, BOOL bShow);
 
     void		ShowRows(SCROW nRow1, SCROW nRow2, BOOL bShow);
-    void		DBShowRows(SCROW nRow1, SCROW nRow2, BOOL bShow);
+    void		DBShowRows(SCROW nRow1, SCROW nRow2, bool bShow, bool bDrawPageSize);
 
     void		SetColFlags( SCCOL nCol, BYTE nNewFlags );
     void		SetRowFlags( SCROW nRow, BYTE nNewFlags );
diff --git sc/source/core/data/table2.cxx sc/source/core/data/table2.cxx
index 319a7e5..f0a0815 100644
--- sc/source/core/data/table2.cxx
+++ sc/source/core/data/table2.cxx
@@ -2540,7 +2540,7 @@ void ScTable::DBShowRow(SCROW nRow, BOOL bShow)
 }
 
 
-void ScTable::DBShowRows(SCROW nRow1, SCROW nRow2, BOOL bShow)
+void ScTable::DBShowRows(SCROW nRow1, SCROW nRow2, bool bShow, bool bDrawPageSize)
 {
     SCROW nStartRow = nRow1;
     nRecalcLvl++;
@@ -2585,7 +2585,7 @@ void ScTable::DBShowRows(SCROW nRow1, SCROW nRow2, BOOL bShow)
     if (pOutlineTable)
         UpdateOutlineRow( nRow1, nRow2, bShow );
 
-    if( !--nRecalcLvl )
+    if( !--nRecalcLvl && bDrawPageSize)
         SetDrawPageSize();
 }
 
diff --git sc/source/core/data/table3.cxx sc/source/core/data/table3.cxx
index 52d7d32..7c9af91 100644
--- sc/source/core/data/table3.cxx
+++ sc/source/core/data/table3.cxx
@@ -1668,7 +1668,7 @@ SCSIZE ScTable::Query(ScQueryParam& rParamOrg, BOOL bKeepSub)
             else
             {
                 if (bStarted)
-                    DBShowRows(nOldStart,nOldEnd, bOldResult);
+                    DBShowRows(nOldStart,nOldEnd, bOldResult, false);
                 nOldStart = nOldEnd = j;
                 bOldResult = bResult;
             }
@@ -1687,10 +1687,12 @@ SCSIZE ScTable::Query(ScQueryParam& rParamOrg, BOOL bKeepSub)
     }
 
     if (aParam.bInplace && bStarted)
-        DBShowRows(nOldStart,nOldEnd, bOldResult);
+        DBShowRows(nOldStart,nOldEnd, bOldResult, false);
 
     delete[] pSpecial;
 
+    SetDrawPageSize();
+
     return nCount;
 }
 
