---
 svl/inc/PasswordHelper.hxx |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git svl/inc/PasswordHelper.hxx svl/inc/PasswordHelper.hxx
index a7b262d..5943406 100644
--- svl/inc/PasswordHelper.hxx
+++ svl/inc/PasswordHelper.hxx
@@ -36,11 +36,12 @@ class String;
 
 class SvPasswordHelper
 {
-    static void		GetHashPassword(com::sun::star::uno::Sequence <sal_Int8>& rPassHash, const sal_Char* pPass, sal_uInt32 nLen);
     static void     GetHashPasswordLittleEndian(com::sun::star::uno::Sequence<sal_Int8>& rPassHash, const String& sPass);
     static void     GetHashPasswordBigEndian(com::sun::star::uno::Sequence<sal_Int8>& rPassHash, const String& sPass);
 
 public:
+    SVL_DLLPUBLIC static void     GetHashPassword(com::sun::star::uno::Sequence <sal_Int8>& rPassHash, const sal_Char* pPass, sal_uInt32 nLen);
+
     SVL_DLLPUBLIC static void     GetHashPassword(com::sun::star::uno::Sequence<sal_Int8>& rPassHash, const String& sPass);
     /**
     Use this method to compare a given string with another given Hash value.
-- 
1.7.0.1

