diff --git sc/source/filter/excel/frmbase.cxx sc/source/filter/excel/frmbase.cxx
index 1156561..71bf8da 100644
--- sc/source/filter/excel/frmbase.cxx
+++ sc/source/filter/excel/frmbase.cxx
@@ -103,7 +103,7 @@ void _ScRangeListTabs::Append( ScSingleRefData a, SCsTAB nTab, const BOOL b )
 	if( nTab < 0)
 		nTab = a.nTab;
 
-    if( nTab >= 0 )
+    if( nTab >= 0 && nTab <= MAXTAB)
     {
         _ScRangeList*   p = ppTabLists[ nTab ];
 
@@ -168,7 +168,7 @@ void _ScRangeListTabs::Append( ScComplexRefData a, SCsTAB nTab, const BOOL b )
 	if( nTab < -1)
 		nTab = a.Ref1.nTab;
 
-    if( nTab >= 0 )
+    if( nTab >= 0 && nTab <= MAXTAB)
     {
         _ScRangeList*   p = ppTabLists[ nTab ];
 
