diff --git sc/source/core/tool/docoptio.cxx sc/source/core/tool/docoptio.cxx
index bf5fc09..0b66c22 100644
--- sc/source/core/tool/docoptio.cxx
+++ sc/source/core/tool/docoptio.cxx
@@ -128,7 +128,7 @@ void ScDocOptions::ResetDocOptions()
     bIsIter				= FALSE;
     nIterCount			= 100;
     fIterEps			= 1.0E-3;
-    nPrecStandardFormat	= 2;
+    nPrecStandardFormat = SvNumberFormatter::UNLIMITED_PRECISION;
     nDay				= 30;
     nMonth				= 12;
     nYear				= 1899;
diff --git sc/source/ui/unoobj/defltuno.cxx sc/source/ui/unoobj/defltuno.cxx
index a6a4867..bb6f411 100644
--- sc/source/ui/unoobj/defltuno.cxx
+++ sc/source/ui/unoobj/defltuno.cxx
@@ -49,6 +49,8 @@
 #include "unonames.hxx"
 #include "docoptio.hxx"
 
+#include <limits>
+
 using namespace ::com::sun::star;
 
 //------------------------------------------------------------------------
@@ -253,7 +255,12 @@ uno::Any SAL_CALL ScDocDefaultsObj::getPropertyValue( const rtl::OUString& aProp
             if (pDoc)
             {
                 const ScDocOptions& aDocOpt = pDoc->GetDocOptions();
-                aRet <<= static_cast<sal_Int16> (aDocOpt.GetStdPrecision());
+                sal_uInt16 nPrec = aDocOpt.GetStdPrecision();
+                // the max value of unsigned 16-bit integer is used as the flag
+                // value for unlimited precision, c.f.
+                // SvNumberFormatter::UNLIMITED_PRECISION.
+                if (nPrec <= ::std::numeric_limits<sal_Int16>::max())
+                    aRet <<= static_cast<sal_Int16> (nPrec);
             }
             else
                 throw uno::RuntimeException();
