diff --git svl/inc/svl/memberid.hrc svl/inc/svl/memberid.hrc
index 8ff0d91..71620ed 100644
--- svl/inc/svl/memberid.hrc
+++ svl/inc/svl/memberid.hrc
@@ -49,20 +49,21 @@
 #define MID_SEARCH_CELLTYPE             2
 #define MID_SEARCH_ROWDIRECTION         3
 #define MID_SEARCH_ALLTABLES            4
-#define MID_SEARCH_BACKWARD             5
-#define MID_SEARCH_PATTERN              6
-#define MID_SEARCH_CONTENT              7
-#define MID_SEARCH_ASIANOPTIONS         8
-#define MID_SEARCH_ALGORITHMTYPE        9
-#define MID_SEARCH_FLAGS                10
-#define MID_SEARCH_SEARCHSTRING         11
-#define MID_SEARCH_REPLACESTRING        12
-#define MID_SEARCH_LOCALE               13
-#define MID_SEARCH_CHANGEDCHARS         14
-#define MID_SEARCH_DELETEDCHARS         15
-#define MID_SEARCH_INSERTEDCHARS        16
-#define MID_SEARCH_TRANSLITERATEFLAGS   17
-#define MID_SEARCH_COMMAND          	18
+#define MID_SEARCH_SEARCHFILTERED       5
+#define MID_SEARCH_BACKWARD             6 
+#define MID_SEARCH_PATTERN              7 
+#define MID_SEARCH_CONTENT              8 
+#define MID_SEARCH_ASIANOPTIONS         9 
+#define MID_SEARCH_ALGORITHMTYPE        10
+#define MID_SEARCH_FLAGS                11
+#define MID_SEARCH_SEARCHSTRING         12
+#define MID_SEARCH_REPLACESTRING        13
+#define MID_SEARCH_LOCALE               14
+#define MID_SEARCH_CHANGEDCHARS         15
+#define MID_SEARCH_DELETEDCHARS         16
+#define MID_SEARCH_INSERTEDCHARS        17
+#define MID_SEARCH_TRANSLITERATEFLAGS   18
+#define MID_SEARCH_COMMAND              19
 
 
 #endif
diff --git svl/inc/svl/srchitem.hxx svl/inc/svl/srchitem.hxx
index be210fa..d8caf48 100644
--- svl/inc/svl/srchitem.hxx
+++ svl/inc/svl/srchitem.hxx
@@ -73,6 +73,7 @@ class SVL_DLLPUBLIC SvxSearchItem :
     sal_uInt16		nAppFlag;   		// Fuer welche Applikation ist der Dialog ueberhaupt
     sal_Bool		bRowDirection;		// Suchrichtung Zeilenweise/Spaltenweise
     sal_Bool		bAllTables;			// in alle Tabellen suchen
+    sal_Bool        bSearchFiltered;      // search filtered cells.
 
     // Writer-spezifisch
     sal_Bool		bNotes;
@@ -142,6 +143,9 @@ public:
             sal_Bool		IsAllTables() const { return bAllTables; }
             void			SetAllTables(sal_Bool bNew) { bAllTables = bNew; }
 
+            sal_Bool        IsSearchFiltered() const { return bSearchFiltered; }
+            void            SetSearchFiltered(sal_Bool b) { bSearchFiltered = b; }
+
             sal_uInt16		GetCellType() const { return nCellType; }
             void			SetCellType(sal_uInt16 nNewCellType) { nCellType = nNewCellType; }
 
diff --git svl/source/items/srchitem.cxx svl/source/items/srchitem.cxx
index 95d30aa..f3ab2a0 100644
--- svl/source/items/srchitem.cxx
+++ svl/source/items/srchitem.cxx
@@ -51,7 +51,7 @@ using namespace com::sun::star::util;
 
 #define CFG_ROOT_NODE       "Office.Common/SearchOptions"
 
-#define SRCH_PARAMS         11
+#define SRCH_PARAMS         12
 #define SRCH_PARA_OPTIONS   "Options"
 #define SRCH_PARA_FAMILY    "Family"
 #define SRCH_PARA_COMMAND   "Command"
@@ -59,6 +59,7 @@ using namespace com::sun::star::util;
 #define SRCH_PARA_APPFLAG   "AppFlag"
 #define SRCH_PARA_ROWDIR    "RowDirection"
 #define SRCH_PARA_ALLTABLES "AllTables"
+#define SRCH_PARA_SEARCHFILTERED "SearchFiltered"
 #define SRCH_PARA_BACKWARD  "Backward"
 #define SRCH_PARA_PATTERN   "Pattern"
 #define SRCH_PARA_CONTENT   "Content"
@@ -124,6 +125,7 @@ SvxSearchItem::SvxSearchItem( const sal_uInt16 nId ) :
     nAppFlag		( SVX_SEARCHAPP_WRITER ),
     bRowDirection	( sal_True ),
     bAllTables		( sal_False ),
+    bSearchFiltered   ( sal_False ),
     bNotes			( sal_False),
     bBackward		( sal_False ),
     bPattern		( sal_False ),
@@ -201,6 +203,7 @@ SvxSearchItem::SvxSearchItem( const SvxSearchItem& rItem ) :
     nAppFlag		( rItem.nAppFlag ),
     bRowDirection	( rItem.bRowDirection ),
     bAllTables		( rItem.bAllTables ),
+    bSearchFiltered   ( rItem.bSearchFiltered ),
     bNotes			( rItem.bNotes),
     bBackward		( rItem.bBackward ),
     bPattern		( rItem.bPattern ),
@@ -250,6 +253,7 @@ int SvxSearchItem::operator==( const SfxPoolItem& rItem ) const
            ( eFamily 		== rSItem.eFamily )			&&
            ( bRowDirection 	== rSItem.bRowDirection )	&&
            ( bAllTables 	== rSItem.bAllTables )		&&
+           ( bSearchFiltered  == rSItem.bSearchFiltered )   &&
            ( nCellType 		== rSItem.nCellType )		&&
            ( nAppFlag 		== rSItem.nAppFlag )		&&
            ( bAsianOptions	== rSItem.bAsianOptions )	&&
@@ -443,6 +447,8 @@ sal_Bool SvxSearchItem::QueryValue( com::sun::star::uno::Any& rVal, BYTE nMember
             aSeq[5].Value <<= bRowDirection;
             aSeq[6].Name = rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( SRCH_PARA_ALLTABLES ));
             aSeq[6].Value <<= bAllTables;
+            aSeq[6].Name = rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( SRCH_PARA_SEARCHFILTERED ));
+            aSeq[6].Value <<= bSearchFiltered;
             aSeq[7].Name = rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( SRCH_PARA_BACKWARD ));
             aSeq[7].Value <<= bBackward;
             aSeq[8].Name = rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( SRCH_PARA_PATTERN ));
@@ -464,6 +470,8 @@ sal_Bool SvxSearchItem::QueryValue( com::sun::star::uno::Any& rVal, BYTE nMember
             rVal <<= (sal_Bool) bRowDirection; break;
         case MID_SEARCH_ALLTABLES:
             rVal <<= (sal_Bool) bAllTables; break;
+        case MID_SEARCH_SEARCHFILTERED:
+            rVal <<= (sal_Bool) bSearchFiltered; break;
         case MID_SEARCH_BACKWARD:
             rVal <<= (sal_Bool) bBackward; break;
         case MID_SEARCH_PATTERN:
@@ -563,6 +571,11 @@ sal_Bool SvxSearchItem::PutValue( const com::sun::star::uno::Any& rVal, BYTE nMe
                         if ( ( aSeq[i].Value >>= bAllTables ) == sal_True )
                             ++nConvertedCount;
                     }
+                    else if ( aSeq[i].Name.equalsAscii( SRCH_PARA_SEARCHFILTERED ) )
+                    {
+                        if ( ( aSeq[i].Value >>= bSearchFiltered ) == sal_True )
+                            ++nConvertedCount;
+                    }
                     else if ( aSeq[i].Name.equalsAscii( SRCH_PARA_BACKWARD ) )
                     {
                         if ( ( aSeq[i].Value >>= bBackward ) == sal_True )
@@ -599,6 +612,8 @@ sal_Bool SvxSearchItem::PutValue( const com::sun::star::uno::Any& rVal, BYTE nMe
             bRet = (rVal >>= bRowDirection); break;
         case MID_SEARCH_ALLTABLES:
             bRet = (rVal >>= bAllTables); break;
+        case MID_SEARCH_SEARCHFILTERED:
+            bRet = (rVal >>= bSearchFiltered); break;
         case MID_SEARCH_BACKWARD:
             bRet = (rVal >>= bBackward); break;
         case MID_SEARCH_PATTERN:
