diff --git sc/source/ui/view/tabcont.cxx sc/source/ui/view/tabcont.cxx
index ec447b1..ef28033 100644
--- sc/source/ui/view/tabcont.cxx
+++ sc/source/ui/view/tabcont.cxx
@@ -57,13 +57,13 @@
 //==================================================================
 
 ScTabControl::ScTabControl( Window* pParent, ScViewData* pData ) :
-            TabBar( pParent, WinBits( WB_BORDER | WB_3DLOOK | WB_SCROLL |
+            TabBar( pParent, WinBits( WB_BORDER | WB_3DLOOK | WB_SCROLL | WB_INSERTTAB |
                                     WB_RANGESELECT | WB_MULTISELECT | WB_DRAG | WB_SIZEABLE ) ),
             DropTargetHelper( this ),
             DragSourceHelper( this ),
             pViewData( pData ),
-            nMouseClickPageId( TABBAR_PAGE_NOTFOUND ),
-            nSelPageIdByMouse( TABBAR_PAGE_NOTFOUND ),
+            nMouseClickPageId( TabBar::PAGE_NOT_FOUND ),
+            nSelPageIdByMouse( TabBar::PAGE_NOT_FOUND ),
             bErrorShown( FALSE )
 {
     ScDocument* pDoc = pViewData->GetDocument();
@@ -165,9 +165,9 @@ void ScTabControl::MouseButtonDown( const MouseEvent& rMEvt )
         Needing clean left click without modifiers (may be context menu).
         #106948# Remember clicks to all pages, to be able to move mouse pointer later. */
     if( rMEvt.IsLeft() && (rMEvt.GetModifier() == 0) )
-        nMouseClickPageId = GetPageId( rMEvt.GetPosPixel() );
+        nMouseClickPageId = GetPageId( rMEvt.GetPosPixel(), true );
     else
-        nMouseClickPageId = TABBAR_PAGE_NOTFOUND;
+        nMouseClickPageId = TabBar::PAGE_NOT_FOUND;
 
     TabBar::MouseButtonDown( rMEvt );
 }
@@ -177,13 +177,25 @@ void ScTabControl::MouseButtonUp( const MouseEvent& rMEvt )
     Point aPos = PixelToLogic( rMEvt.GetPosPixel() );
 
     // mouse button down and up on same page?
-    if( nMouseClickPageId != GetPageId( aPos ) )
-        nMouseClickPageId = TABBAR_PAGE_NOTFOUND;
+    if( nMouseClickPageId != GetPageId( aPos, true ) )
+        nMouseClickPageId = TabBar::PAGE_NOT_FOUND;
+
+    if (nMouseClickPageId == TabBar::INSERT_TAB_POS)
+    {
+        // Insert a new sheet at the right end, with default name.
+        ScDocument* pDoc = pViewData->GetDocument();
+        String aName;
+        pDoc->CreateValidTabName(aName);
+        SCTAB nTabCount = pDoc->GetTableCount();
+        pViewData->GetViewShell()->InsertTable(aName, nTabCount);
+        return;
+    }
 
     if ( rMEvt.GetClicks() == 2 && rMEvt.IsLeft() && nMouseClickPageId != 0 && nMouseClickPageId != TAB_PAGE_NOTFOUND )
     {
         SfxDispatcher* pDispatcher = pViewData->GetViewShell()->GetViewFrame()->GetDispatcher();
         pDispatcher->Execute( FID_TAB_MENU_RENAME, SFX_CALLMODE_SYNCHRON | SFX_CALLMODE_RECORD );
+        return;
     }
 
     if( nMouseClickPageId == 0 )
@@ -195,7 +207,7 @@ void ScTabControl::MouseButtonUp( const MouseEvent& rMEvt )
         SfxDispatcher* pDispatcher = pViewData->GetViewShell()->GetViewFrame()->GetDispatcher();
         pDispatcher->Execute( nSlot, SFX_CALLMODE_SYNCHRON | SFX_CALLMODE_RECORD );
         // forget page ID, to be really sure that the dialog is not called twice
-        nMouseClickPageId = TABBAR_PAGE_NOTFOUND;
+        nMouseClickPageId = TabBar::PAGE_NOT_FOUND;
     }
 
     TabBar::MouseButtonUp( rMEvt );
@@ -207,7 +219,7 @@ void ScTabControl::Select()
     nSelPageIdByMouse = nMouseClickPageId;
     /*  Reset nMouseClickPageId, so that next Select() call may invalidate
         nSelPageIdByMouse (i.e. if called from keyboard). */
-    nMouseClickPageId = TABBAR_PAGE_NOTFOUND;
+    nMouseClickPageId = TabBar::PAGE_NOT_FOUND;
 
     ScModule* pScMod = SC_MOD();
     ScDocument* pDoc = pViewData->GetDocument();
@@ -393,7 +405,7 @@ void ScTabControl::ActivateView(BOOL bActivate)
 void ScTabControl::SetSheetLayoutRTL( BOOL bSheetRTL )
 {
     SetEffectiveRTL( bSheetRTL );
-    nSelPageIdByMouse = TABBAR_PAGE_NOTFOUND;
+    nSelPageIdByMouse = TabBar::PAGE_NOT_FOUND;
 }
 
 
@@ -629,12 +641,12 @@ void ScTabControl::EndRenaming()
 void ScTabControl::Mirror()
 {
     TabBar::Mirror();
-    if( nSelPageIdByMouse != TABBAR_PAGE_NOTFOUND )
+    if( nSelPageIdByMouse != TabBar::PAGE_NOT_FOUND )
     {
         Rectangle aRect( GetPageRect( GetCurPageId() ) );
         if( !aRect.IsEmpty() )
             SetPointerPosPixel( aRect.Center() );
-        nSelPageIdByMouse = TABBAR_PAGE_NOTFOUND;  // only once after a Select()
+        nSelPageIdByMouse = TabBar::PAGE_NOT_FOUND;  // only once after a Select()
     }
 }
 
