diff --git sc/source/filter/excel/xeescher.cxx sc/source/filter/excel/xeescher.cxx
index 98eb5b7..78a0d2a 100644
--- sc/source/filter/excel/xeescher.cxx
+++ sc/source/filter/excel/xeescher.cxx
@@ -696,6 +696,8 @@ void XclExpTbxControlObj::WriteSubRecs( XclExpStream& rStrm )
             else if( mnObjType == EXC_OBJTYPE_DROPDOWN )
             {
                 rStrm << sal_uInt16( 0 ) << mnLineCount;
+                // dxMin & zero length str
+                rStrm << sal_uInt16( 0 ) << sal_uInt16( 0 );
             }
 
             rStrm.EndRecord();
