diff --git sc/inc/globstr.hrc sc/inc/globstr.hrc
index 9a50bce..d08475b 100644
--- sc/inc/globstr.hrc
+++ sc/inc/globstr.hrc
@@ -594,7 +594,10 @@
 
 #define STR_OPTIONS_WARN_SEPARATORS 451
 
-#define STR_COUNT                   452
+#define STR_UNDO_INSERT_CURRENT_DATE 452
+#define STR_UNDO_INSERT_CURRENT_TIME 453
+
+#define STR_COUNT                   454
 
 #endif
 
diff --git sc/inc/sc.hrc sc/inc/sc.hrc
index d659aa8..3d04adc 100644
--- sc/inc/sc.hrc
+++ sc/inc/sc.hrc
@@ -1678,6 +1678,9 @@
 #define SID_DATA_FORM               (SC_OOO_BUILD_START + 6) // menu (in Data menu)
 #define RID_SCDLG_DATAFORM          (SC_OOO_BUILD_START + 7) // dialog
 
+#define SID_INSERT_CURRENT_DATE     (SC_OOO_BUILD_START + 8)
+#define SID_INSERT_CURRENT_TIME     (SC_OOO_BUILD_START + 9)
+
 #endif 
 
 
diff --git sc/sdi/cellsh.sdi sc/sdi/cellsh.sdi
index 5f1c6e0..918665e 100644
--- sc/sdi/cellsh.sdi
+++ sc/sdi/cellsh.sdi
@@ -115,6 +115,8 @@ interface CellSelection
     SID_DETECTIVE_REFRESH	[ ExecMethod = ExecuteEdit; StateMethod = GetState; ]
 	SID_DETECTIVE_MARK_PRED	[ ExecMethod = ExecuteEdit; StateMethod = GetState; ]
 	SID_DETECTIVE_MARK_SUCC	[ ExecMethod = ExecuteEdit; StateMethod = GetState; ]
+    SID_INSERT_CURRENT_DATE     [ ExecMethod = ExecuteEdit; StateMethod = GetState; ]
+    SID_INSERT_CURRENT_TIME     [ ExecMethod = ExecuteEdit; StateMethod = GetState; ]
     FID_INS_ROW				[ ExecMethod = ExecuteEdit; StateMethod = GetBlockState; ]
     FID_INS_COLUMN			[ ExecMethod = ExecuteEdit; StateMethod = GetBlockState; ]
     FID_INS_CELLSDOWN		[ ExecMethod = ExecuteEdit; StateMethod = GetBlockState; ]
diff --git sc/sdi/scalc.sdi sc/sdi/scalc.sdi
index c41f4c4..af9928d 100644
--- sc/sdi/scalc.sdi
+++ sc/sdi/scalc.sdi
@@ -7952,3 +7952,54 @@ SfxVoidItem MarkDependents SID_DETECTIVE_MARK_SUCC
     ToolBoxConfig = FALSE,
     GroupId = GID_OPTIONS;
 ]
+
+//--------------------------------------------------------------------------
+SfxVoidItem InsertCurrentDate SID_INSERT_CURRENT_DATE
+()
+[
+    /* flags: */
+    AutoUpdate = FALSE,
+    Cachable = Cachable,
+    FastCall = FALSE,
+    HasCoreId = FALSE,
+    HasDialog = FALSE,
+    ReadOnlyDoc = TRUE,
+    Toggle = FALSE,
+    Container = FALSE,
+    RecordAbsolute = FALSE,
+    RecordPerSet;
+    Synchron;
+    
+    /* config: */
+    AccelConfig = TRUE,
+    MenuConfig = TRUE,
+    StatusBarConfig = FALSE,
+    ToolBoxConfig = FALSE,
+    GroupId = GID_OPTIONS;
+]
+
+//--------------------------------------------------------------------------
+SfxVoidItem InsertCurrentTime SID_INSERT_CURRENT_TIME
+()
+[
+    /* flags: */
+    AutoUpdate = FALSE,
+    Cachable = Cachable,
+    FastCall = FALSE,
+    HasCoreId = FALSE,
+    HasDialog = FALSE,
+    ReadOnlyDoc = TRUE,
+    Toggle = FALSE,
+    Container = FALSE,
+    RecordAbsolute = FALSE,
+    RecordPerSet;
+    Synchron;
+    
+    /* config: */
+    AccelConfig = TRUE,
+    MenuConfig = TRUE,
+    StatusBarConfig = FALSE,
+    ToolBoxConfig = FALSE,
+    GroupId = GID_OPTIONS;
+]
+
diff --git sc/source/ui/inc/viewfunc.hxx sc/source/ui/inc/viewfunc.hxx
index 0f71150..7686540 100644
--- sc/source/ui/inc/viewfunc.hxx
+++ sc/source/ui/inc/viewfunc.hxx
@@ -334,6 +334,8 @@ public:
     void            DetectiveMarkPred();
     void            DetectiveMarkSucc();
 
+    void            InsertCurrentTime(short nCellFmt, const ::rtl::OUString& rUndoStr);
+
     void			ShowNote( bool bShow = true );
     inline void		HideNote() { ShowNote( false ); }
     void			EditNote();
diff --git sc/source/ui/src/globstr.src sc/source/ui/src/globstr.src
index 2d0fd78..7248703 100644
--- sc/source/ui/src/globstr.src
+++ sc/source/ui/src/globstr.src
@@ -1808,5 +1808,13 @@ Resource RID_GLOBSTR
     {
         Text [ en-US ] = "Because the current formula separator settings conflict with the locale, the formula separators have been reset to their default values.";
     };
+    String STR_UNDO_INSERT_CURRENT_DATE
+    {
+        Text [ en-US ] = "Insert Current Date";
+    };
+    String STR_UNDO_INSERT_CURRENT_TIME
+    {
+        Text [ en-US ] = "Insert Current Time";
+    };
 };
 
diff --git sc/source/ui/view/cellsh1.cxx sc/source/ui/view/cellsh1.cxx
index 5a1857f..502b214 100644
--- sc/source/ui/view/cellsh1.cxx
+++ sc/source/ui/view/cellsh1.cxx
@@ -1557,6 +1557,14 @@ void ScCellShell::ExecuteEdit( SfxRequest& rReq )
         case SID_DETECTIVE_MARK_SUCC:
             pTabViewShell->DetectiveMarkSucc();
             break;
+        case SID_INSERT_CURRENT_DATE:
+            pTabViewShell->InsertCurrentTime(
+                NUMBERFORMAT_DATE, ScGlobal::GetRscString(STR_UNDO_INSERT_CURRENT_DATE));
+            break;
+        case SID_INSERT_CURRENT_TIME:
+            pTabViewShell->InsertCurrentTime(
+                NUMBERFORMAT_TIME, ScGlobal::GetRscString(STR_UNDO_INSERT_CURRENT_TIME));
+            break;
 
         case SID_SPELL_DIALOG:
 //           pTabViewShell->DoSpellingChecker();
diff --git sc/source/ui/view/viewfun6.cxx sc/source/ui/view/viewfun6.cxx
index e686cfc..e8647a9 100644
--- sc/source/ui/view/viewfun6.cxx
+++ sc/source/ui/view/viewfun6.cxx
@@ -37,6 +37,7 @@
 #include <sfx2/dispatch.hxx>
 #include <vcl/msgbox.hxx>
 #include <vcl/sound.hxx>
+#include "svtools/zforlist.hxx"
 
 #include "viewfunc.hxx"
 #include "detfunc.hxx"
@@ -52,12 +53,16 @@
 #include "fusel.hxx"
 #include "reftokenhelper.hxx"
 #include "externalrefmgr.hxx"
+#include "cell.hxx"
 
 #include <vector>
 
+using ::rtl::OUString;
 using ::rtl::OUStringBuffer;
 using ::std::vector;
 
+#define D_TIMEFACTOR              86400.0
+
 //==================================================================
 
 void ScViewFunc::DetectiveAddPred()
@@ -277,6 +282,27 @@ void ScViewFunc::DetectiveMarkSucc()
     MarkAndJumpToRanges(aDestRanges);
 }
 
+void ScViewFunc::InsertCurrentTime(short nCellFmt, const OUString& rUndoStr)
+{
+    ScViewData* pViewData = GetViewData();
+    ScAddress aCurPos = pViewData->GetCurPos();
+    ScDocShell* pDocSh = pViewData->GetDocShell();
+    ScDocument* pDoc = pDocSh->GetDocument();
+    SfxUndoManager* pUndoMgr = pDocSh->GetUndoManager();
+    SvNumberFormatter* pFormatter = pDoc->GetFormatTable();
+    Date aActDate;
+    double fDate = aActDate - *pFormatter->GetNullDate();
+    Time aActTime;
+    double fTime = 
+        aActTime.Get100Sec() / 100.0 + aActTime.GetSec() + 
+        (aActTime.GetMin() * 60.0) + (aActTime.GetHour() * 3600.0);
+    fTime /= D_TIMEFACTOR;
+    pUndoMgr->EnterListAction(rUndoStr, rUndoStr);
+    pDocSh->GetDocFunc().PutCell(aCurPos, new ScValueCell(fDate+fTime), false);
+    SetNumberFormat(nCellFmt);
+    pUndoMgr->LeaveListAction();
+}
+
 //---------------------------------------------------------------------------
 
 void ScViewFunc::ShowNote( bool bShow )
