diff --git sc/source/core/data/bcaslot.cxx sc/source/core/data/bcaslot.cxx
index 9a07109..fbf0cad 100644
--- sc/source/core/data/bcaslot.cxx
+++ sc/source/core/data/bcaslot.cxx
@@ -48,11 +48,7 @@
 // Number of slots per dimension
 // must be integer divisors of MAXCOLCOUNT respectively MAXROWCOUNT
 #define BCA_SLOTS_COL ((MAXCOLCOUNT_DEFINE) / 16)
-#if MAXROWCOUNT_DEFINE == 32000
-#define BCA_SLOTS_ROW 256
-#else
-#define BCA_SLOTS_ROW ((MAXROWCOUNT_DEFINE) / 128)
-#endif
+#define BCA_SLOTS_ROW 2048
 #define BCA_SLOT_COLS ((MAXCOLCOUNT_DEFINE) / BCA_SLOTS_COL)
 #define BCA_SLOT_ROWS ((MAXROWCOUNT_DEFINE) / BCA_SLOTS_ROW)
 // multiple?
