diff --git sc/source/filter/xlsx/xistyle.hxx sc/source/filter/xlsx/xistyle.hxx
index baebad3..138c633 100644
--- sc/source/filter/xlsx/xistyle.hxx
+++ sc/source/filter/xlsx/xistyle.hxx
@@ -411,15 +411,6 @@ public:
         @return  A read-only reference to the item set stored internally. */
     const ScPatternAttr& CreatePattern( bool bSkipPoolDefs = false );
 
-    /** Inserts all formatting attributes to the specified area in the Calc document.
-        @param nForcedNumFmt  If not set to NUMBERFORMAT_ENTRY_NOT_FOUND, it will overwrite
-        the number format of the XF. */
-    void                ApplyPattern(
-                            SCCOL nScCol1, SCROW nScRow1,
-                            SCCOL nScCol2, SCROW nScRow2,
-                            SCTAB nScTab,
-                            ULONG nForceScNumFmt = NUMBERFORMAT_ENTRY_NOT_FOUND );
-
 private:
     void                ReadXF2( XclImpStream& rStrm );
     void                ReadXF3( XclImpStream& rStrm );
@@ -516,14 +507,6 @@ public:
         @return  The pointer to the cell style sheet, or 0, if there is no style sheet. */
     ScStyleSheet*       CreateStyleSheet( sal_uInt16 nXFIndex );
 
-    /** Inserts formatting attributes from an XF to the specified area in the Calc document.
-        @param nForcedNumFmt  If not set to NUMBERFORMAT_ENTRY_NOT_FOUND, it will overwrite
-        the number format of the XF. */
-    void                ApplyPattern(
-                            SCCOL nScCol1, SCROW nScRow1,
-                            SCCOL nScCol2, SCROW nScRow2,
-                            SCTAB nScTab, const XclImpXFIndex& rXFIndex );
-
 private:
     typedef ScfDelList< XclImpStyle >               XclImpStyleList;
     typedef ::std::map< sal_uInt16, XclImpStyle* >  XclImpStyleMap;
