diff --git sc/source/ui/app/inputhdl.cxx sc/source/ui/app/inputhdl.cxx
index e5d46e5..2e450dd 100644
--- sc/source/ui/app/inputhdl.cxx
+++ sc/source/ui/app/inputhdl.cxx
@@ -2275,6 +2275,10 @@ void ScInputHandler::SetMode( ScInputMode eNewMode )
         return;
     }
 
+    if (eNewMode != SC_INPUT_NONE)
+        // Disable paste mode when edit mode starts.
+        pActiveViewSh->GetViewData()->SetPasteMode( SC_PASTE_NONE );
+
     bInOwnChange = TRUE;				// disable ModifyHdl (reset below)
 
     ScInputMode eOldMode = eMode;
