diff --git sfx2/inc/sfx2/objsh.hxx sfx2/inc/sfx2/objsh.hxx
index 488acea..d87dff2 100644
--- sfx2/inc/sfx2/objsh.hxx
+++ sfx2/inc/sfx2/objsh.hxx
@@ -354,6 +354,11 @@ public:
     virtual sal_Bool            SwitchPersistance(
                                     const ::com::sun::star::uno::Reference< ::com::sun::star::embed::XStorage >& xStorage );
     virtual void                UpdateLinks();
+    /** 
+     * Called when the Options dialog is dismissed with the OK button, to 
+     * handle potentially conflicting option settings.
+     */
+    virtual void                CheckConfigOptions();
 
     sal_Bool                    SaveChildren(BOOL bObjectsOnly=FALSE);
     sal_Bool                    SaveAsChildren( SfxMedium &rMedium );
diff --git sfx2/source/appl/appserv.cxx sfx2/source/appl/appserv.cxx
index b511170..04db6d4 100644
--- sfx2/source/appl/appserv.cxx
+++ sfx2/source/appl/appserv.cxx
@@ -855,6 +855,23 @@ namespace
     }
 }
 
+namespace {
+
+void inspectNewOptions()
+{
+    SfxViewFrame* pCurView = SfxViewFrame::Current();
+    if (!pCurView)
+        return;
+
+    SfxObjectShell* pShell = pCurView->GetObjectShell();
+    if (!pShell)
+        return;
+
+    pShell->CheckConfigOptions();
+}
+
+}
+
 void SfxApplication::OfaExec_Impl( SfxRequest& rReq )
 {
     DBG_MEMTEST();
@@ -876,7 +893,11 @@ void SfxApplication::OfaExec_Impl( SfxRequest& rReq )
             {
                 VclAbstractDialog* pDlg =
                     pFact->CreateFrameDialog( NULL, xFrame, rReq.GetSlot(), sPageURL );
-                  pDlg->Execute();
+                  if (pDlg->Execute() == RET_OK)
+                  {
+                      // Do extra sanity check for potentially clashing options.
+                      inspectNewOptions();
+                  }
                   delete pDlg;
                 SfxViewFrame* pView = SfxViewFrame::GetFirst();
                 while ( pView )
diff --git sfx2/source/doc/objstor.cxx sfx2/source/doc/objstor.cxx
index e9856fe..5930374 100644
--- sfx2/source/doc/objstor.cxx
+++ sfx2/source/doc/objstor.cxx
@@ -4001,6 +4001,11 @@ void SfxObjectShell::UpdateLinks()
 {
 }
 
+void SfxObjectShell::CheckConfigOptions()
+{
+	// not implemented.
+}
+
 sal_Bool SfxObjectShell::QuerySaveSizeExceededModules_Impl( const uno::Reference< task::XInteractionHandler >& xHandler )
 {
     if ( !HasBasic() )
