diff --git officecfg/registry/data/org/openoffice/Office/Accelerators.xcu officecfg/registry/data/org/openoffice/Office/Accelerators.xcu
index fe3b463..71118bc 100644
--- officecfg/registry/data/org/openoffice/Office/Accelerators.xcu
+++ officecfg/registry/data/org/openoffice/Office/Accelerators.xcu
@@ -779,6 +779,16 @@
       <value xml:lang="en-US">.uno:SelectRow</value>
      </prop>
     </node>
+    <node oor:name="SEMICOLON_MOD1" oor:op="replace">
+     <prop oor:name="Command"><value xml:lang="x-no-translate">I10N SHORTCUTS - NO TRANSLATE</value>
+      <value xml:lang="en-US">.uno:InsertCurrentDate</value>
+     </prop>
+    </node>
+    <node oor:name="SEMICOLON_SHIFT_MOD1" oor:op="replace">
+     <prop oor:name="Command"><value xml:lang="x-no-translate">I10N SHORTCUTS - NO TRANSLATE</value>
+      <value xml:lang="en-US">.uno:InsertCurrentTime</value>
+     </prop>
+    </node>
     <node oor:name="SPACE_SHIFT_MOD1" oor:op="replace">
      <prop oor:name="Command"><value xml:lang="x-no-translate">I10N SHORTCUTS - NO TRANSLATE</value>
       <value xml:lang="en-US">.uno:SelectColumn</value>
diff --git officecfg/registry/data/org/openoffice/Office/UI/CalcCommands.xcu officecfg/registry/data/org/openoffice/Office/UI/CalcCommands.xcu
index f60d5d0..28f7e38 100644
--- officecfg/registry/data/org/openoffice/Office/UI/CalcCommands.xcu
+++ officecfg/registry/data/org/openoffice/Office/UI/CalcCommands.xcu
@@ -1616,6 +1616,16 @@
                     <value xml:lang="en-US">Mark Dependents</value>
                 </prop>
             </node>
+            <node oor:name=".uno:InsertCurrentDate" oor:op="replace">
+                <prop oor:name="Label" oor:type="xs:string">
+                    <value xml:lang="en-US">Insert Current Date</value>
+                </prop>
+            </node>
+            <node oor:name=".uno:InsertCurrentTime" oor:op="replace">
+                <prop oor:name="Label" oor:type="xs:string">
+                    <value xml:lang="en-US">Insert Current Time</value>
+                </prop>
+            </node>
         </node>
     </node>
 </oor:component-data>
