diff --git sc/inc/address.hxx sc/inc/address.hxx
index 67cb480..6bd785e 100644
--- sc/inc/address.hxx
+++ sc/inc/address.hxx
@@ -84,7 +84,7 @@ const SCSIZE   SCSIZE_MAX   = ::std::numeric_limits<SCSIZE>::max();
 
 // The maximum values. Defines are needed for preprocessor checks in
 // bcaslot.cxx, otherwise type safe constants are preferred.
-#define MAXROWCOUNT_DEFINE 65536
+#define MAXROWCOUNT_DEFINE 1048576
 #define MAXCOLCOUNT_DEFINE 1024
 
 // Count values
