diff --git sc/source/filter/excel/impop.cxx sc/source/filter/excel/impop.cxx
index cea2a9d..16060ba 100644
--- sc/source/filter/excel/impop.cxx
+++ sc/source/filter/excel/impop.cxx
@@ -47,6 +47,7 @@
 #include <svl/zforlist.hxx>
 
 #include <sfx2/objsh.hxx>
+#include "tools/urlobj.hxx"
 #include "docuno.hxx"
 
 #include "cell.hxx"
@@ -1158,6 +1159,14 @@ void ImportExcel::NeueTabelle( void )
     if( nTab > 0 && !pD->HasTable( nTab ) )
         pD->MakeTable( nTab );
 
+    if (nTab == 0 && GetBiff() == EXC_BIFF2)
+    {
+        // For Excel 2.1 Worksheet file, we need to set the file name as the 
+        // sheet name.
+        INetURLObject aURL(GetDocUrl());
+        pD->RenameTab(0, aURL.getBase(), false);
+    }
+
     pExcRoot->pShrfmlaBuff->Clear();
 
     InitializeTable( nTab );
