diff --git sc/source/core/data/document.cxx sc/source/core/data/document.cxx
index 6490cf4..75c5266 100644
--- sc/source/core/data/document.cxx
+++ sc/source/core/data/document.cxx
@@ -3717,15 +3717,18 @@ SCROW ScDocument::GetNextDifferentChangedRow( SCTAB nTab, SCROW nStart, bool bCa
 {
     if ( ValidTab(nTab) && pTab[nTab] && pTab[nTab]->GetRowFlagsArray() && pTab[nTab]->mpRowHeights )
     {
+        // Use the original row height for comparison, which may not be zero 
+        // even for hidden rows; it's the height before the row has become
+        // hidden.
         BYTE nStartFlags = pTab[nTab]->GetRowFlags(nStart);
-        USHORT nStartHeight = pTab[nTab]->GetOriginalHeight(nStart);
+        USHORT nStartHeight = pTab[nTab]->GetRowHeight(nStart, NULL, NULL, false);
         for (SCROW nRow = nStart + 1; nRow <= MAXROW; nRow++)
         {
             size_t nIndex;          // ignored
             SCROW nFlagsEndRow;
             SCROW nHeightEndRow;
             BYTE nFlags = pTab[nTab]->GetRowFlagsArray()->GetValue( nRow, nIndex, nFlagsEndRow );
-            USHORT nHeight = pTab[nTab]->GetRowHeight(nRow, NULL, &nHeightEndRow);
+            USHORT nHeight = pTab[nTab]->GetRowHeight(nRow, NULL, &nHeightEndRow, false);
             if (((nStartFlags & CR_MANUALBREAK) != (nFlags & CR_MANUALBREAK)) ||
                 ((nStartFlags & CR_MANUALSIZE) != (nFlags & CR_MANUALSIZE)) ||
                 (bCareManualSize && (nStartFlags & CR_MANUALSIZE) && (nStartHeight != nHeight)) ||
