From 2f7c5d4a57faf8643cfe0d9fe2c0b04499924563 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:56:10 +0200
Subject: [PATCH 195/878] calc-ods-sheet-protection-svtools.diff

---
 svtools/inc/PasswordHelper.hxx |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/svtools/inc/PasswordHelper.hxx b/svtools/inc/PasswordHelper.hxx
index 779e08e..208bd7d 100644
--- a/svtools/inc/PasswordHelper.hxx
+++ b/svtools/inc/PasswordHelper.hxx
@@ -36,11 +36,12 @@ class String;
 
 class SvPasswordHelper
 {
-    static void		GetHashPassword(com::sun::star::uno::Sequence <sal_Int8>& rPassHash, const sal_Char* pPass, sal_uInt32 nLen);
     static void     GetHashPasswordLittleEndian(com::sun::star::uno::Sequence<sal_Int8>& rPassHash, const String& sPass);
     static void     GetHashPasswordBigEndian(com::sun::star::uno::Sequence<sal_Int8>& rPassHash, const String& sPass);
 
 public:
+    SVL_DLLPUBLIC static void     GetHashPassword(com::sun::star::uno::Sequence <sal_Int8>& rPassHash, const sal_Char* pPass, sal_uInt32 nLen);
+
     SVL_DLLPUBLIC static void     GetHashPassword(com::sun::star::uno::Sequence<sal_Int8>& rPassHash, const String& sPass);
     /**
     Use this method to compare a given string with another given Hash value.
-- 
1.7.0.1

