diff --git sc/source/core/data/document.cxx sc/source/core/data/document.cxx
index 59ac167..38684c3 100644
--- sc/source/core/data/document.cxx
+++ sc/source/core/data/document.cxx
@@ -2984,6 +2984,9 @@ ULONG ScDocument::GetRowHeight( SCROW nStartRow, SCROW nEndRow, SCTAB nTab ) con
 ULONG ScDocument::FastGetRowHeight( SCROW nStartRow, SCROW nEndRow,
         SCTAB nTab ) const
 {
+    if (!pTab[nTab]->pRowFlags || !pTab[nTab]->pRowHeight)
+        return GetRowHeight(nStartRow, nEndRow, nTab);
+
     return pTab[nTab]->pRowFlags->SumCoupledArrayForCondition( nStartRow,
             nEndRow, CR_HIDDEN, 0, *(pTab[nTab]->pRowHeight));
 }
