diff --git officecfg/registry/schema/org/openoffice/Office/Calc.xcs officecfg/registry/schema/org/openoffice/Office/Calc.xcs
index a6cccce..73707e4 100644
--- officecfg/registry/schema/org/openoffice/Office/Calc.xcs
+++ officecfg/registry/schema/org/openoffice/Office/Calc.xcs
@@ -1284,6 +1284,14 @@
 						</enumeration>
 					</constraints>
 				</prop>
+                <prop oor:name="EnglishFunctionName" oor:type="xs:boolean">
+                    <!-- UIHints: Tools - Options  Spreadsheet  Formula -->
+                    <info>
+                        <author>kyoshida</author>
+                        <desc>Whether to use English function names or localized function names.</desc>
+                    </info>
+                    <value>false</value>
+                </prop>
 				<prop oor:name="SeparatorArg" oor:type="xs:string">
 					<!-- UIHints: Tools - Options  Spreadsheet  Formula -->
 					<info>
