diff --git sc/source/ui/app/inputhdl.cxx sc/source/ui/app/inputhdl.cxx
index 37f485e..0b43cc3 100644
--- sc/source/ui/app/inputhdl.cxx
+++ sc/source/ui/app/inputhdl.cxx
@@ -2973,8 +2973,7 @@ BOOL ScInputHandler::KeyInput( const KeyEvent& rKEvt, BOOL bStartEdit /* = FALSE
     sal_Unicode nChar = rKEvt.GetCharCode();
 
     //	Alt-Return is accepted, everything else with ALT, or CTRL-TAB are not:
-    if (( bAlt && !bControl && nCode != KEY_RETURN ) ||
-            ( bControl && aCode.GetCode() == KEY_TAB ))
+    if ((bAlt && !bControl && nCode != KEY_RETURN) || (aCode.GetCode() == KEY_TAB))
         return FALSE;
 
     BOOL bInputLine = ( eMode==SC_INPUT_TOP );
