diff --git sc/source/filter/excel/xicontent.cxx sc/source/filter/excel/xicontent.cxx
index cfe65b4..666083d 100644
--- sc/source/filter/excel/xicontent.cxx
+++ sc/source/filter/excel/xicontent.cxx
@@ -1228,7 +1228,29 @@ void XclImpSheetProtectBuffer::ReadProtect( XclImpStream& rStrm, SCTAB nTab )
 
 void XclImpSheetProtectBuffer::ReadOptions( XclImpStream& rStrm, SCTAB nTab )
 {
-    rStrm.Ignore(19);
+    rStrm.Ignore(12);
+
+    // feature type can be either 2 or 4.  If 2, this record stores flag for 
+    // enhanced protection, whereas if 4 it stores flag for smart tag.
+    sal_uInt16 nFeatureType;
+    rStrm >> nFeatureType;
+    if (nFeatureType != 2)
+        // We currently only support import of enhanced protection data.
+        return;
+
+    rStrm.Ignore(1); // always 1
+
+    // The flag size specifies the size of bytes that follows that stores 
+    // feature data.  If -1 it depends on the feature type imported earlier.
+    // For enhanced protection data, the size is always 4.  For the most xls
+    // documents out there this value is almost always -1.
+    sal_Int32 nFlagSize;
+    rStrm >> nFlagSize;
+    if (nFlagSize != -1)
+        return;
+
+    // There are actually 4 bytes to read, but the upper 2 bytes currently 
+    // don't store any bits.
     sal_uInt16 nOptions;
     rStrm >> nOptions;
 
