diff --git svx/inc/svx/sdr/animation/scheduler.hxx svx/inc/svx/sdr/animation/scheduler.hxx
index 6ee3e41..3dd8871 100644
--- svx/inc/svx/sdr/animation/scheduler.hxx
+++ svx/inc/svx/sdr/animation/scheduler.hxx
@@ -61,7 +61,7 @@ namespace sdr
 
             // get/set time
             sal_uInt32 GetTime() const;
-            void SetTime(sal_uInt32 nNew);
+            void SVX_DLLPUBLIC SetTime(sal_uInt32 nNew);
 
             // execute event
             virtual void Trigger(sal_uInt32 nTime) = 0;
@@ -145,7 +145,7 @@ namespace sdr
             void checkTimeout();
 
             // insert/remove events, wrapper to EventList methods
-            void InsertEvent(Event* pNew);
+            void SVX_DLLPUBLIC InsertEvent(Event* pNew);
             void RemoveEvent(Event* pOld);
 
             // get/set pause
