diff --git sc/inc/rangelst.hxx sc/inc/rangelst.hxx
index ba694e2..ae1fee6 100644
--- sc/inc/rangelst.hxx
+++ sc/inc/rangelst.hxx
@@ -58,10 +58,10 @@ public:
     USHORT			Parse( const String&, ScDocument* = NULL,
                            USHORT nMask = SCA_VALID,
                            formula::FormulaGrammar::AddressConvention eConv = formula::FormulaGrammar::CONV_OOO,
-                           char cDelimiter = 0 );
+                           sal_Unicode cDelimiter = 0 );
     void 			Format( String&, USHORT nFlags = 0, ScDocument* = NULL,
                             formula::FormulaGrammar::AddressConvention eConv = formula::FormulaGrammar::CONV_OOO,
-                            char cDelimiter = 0 ) const;
+                            sal_Unicode cDelimiter = 0 ) const;
     void			Join( const ScRange&, BOOL bIsInList = FALSE );
     BOOL 			UpdateReference( UpdateRefMode, ScDocument*,
                                     const ScRange& rWhere,
diff --git sc/source/core/tool/rangelst.cxx sc/source/core/tool/rangelst.cxx
index 9c16d73..7ce1d61 100644
--- sc/source/core/tool/rangelst.cxx
+++ sc/source/core/tool/rangelst.cxx
@@ -44,7 +44,7 @@
 #include "document.hxx"
 #include "refupdat.hxx"
 #include "rechead.hxx"
-
+#include "compiler.hxx"
 
 // === ScRangeList ====================================================
 
@@ -61,32 +61,14 @@ void ScRangeList::RemoveAll()
     Clear();
 }
 
-static void defaultDelimiter( char& cDelimiter, formula::FormulaGrammar::AddressConvention eConv)
-{
-    if( cDelimiter == 0)
-    {
-        switch( eConv )
-        {
-        default :
-        case formula::FormulaGrammar::CONV_OOO :
-            cDelimiter = ';';
-            break;
-
-        case formula::FormulaGrammar::CONV_XL_A1 :
-        case formula::FormulaGrammar::CONV_XL_R1C1 :
-            cDelimiter = ',';
-            break;
-        }
-    }
-}
-
 USHORT ScRangeList::Parse( const String& rStr, ScDocument* pDoc, USHORT nMask,
                            formula::FormulaGrammar::AddressConvention eConv,
-                           char cDelimiter )
+                           sal_Unicode cDelimiter )
 {
     if ( rStr.Len() )
     {
-        defaultDelimiter( cDelimiter, eConv);
+        if (!cDelimiter)
+            cDelimiter = ScCompiler::GetNativeSymbol(ocSep).GetChar(0);
 
         nMask |= SCA_VALID;				// falls das jemand vergessen sollte
         USHORT nResult = (USHORT)~0;	// alle Bits setzen
@@ -129,11 +111,12 @@ SCA_VALID_TAB2;
 
 void ScRangeList::Format( String& rStr, USHORT nFlags, ScDocument* pDoc,
                           formula::FormulaGrammar::AddressConvention eConv,
-                          char cDelimiter ) const
+                          sal_Unicode cDelimiter ) const
 {
     rStr.Erase();
 
-    defaultDelimiter( cDelimiter, eConv);
+    if (!cDelimiter)
+        cDelimiter = ScCompiler::GetNativeSymbol(ocSep).GetChar(0);
 
     ULONG nCnt = Count();
     for ( ULONG nIdx = 0; nIdx < nCnt; nIdx++ )
diff --git sc/source/ui/drawfunc/fuins2.cxx sc/source/ui/drawfunc/fuins2.cxx
index dae97f1..2271c07 100644
--- sc/source/ui/drawfunc/fuins2.cxx
+++ sc/source/ui/drawfunc/fuins2.cxx
@@ -529,7 +529,8 @@ FuInsertChart::FuInsertChart(ScTabViewShell* pViewSh, Window* pWin, ScDrawView*
             ScRangeList aRanges;
             aMultiMark.FillRangeListWithMarks( &aRanges, FALSE );
             String aStr;
-            aRanges.Format( aStr, SCR_ABS_3D, pViewSh->GetViewData()->GetDocument() );
+            ScDocument* pDocument = pViewSh->GetViewData()->GetDocument();
+            aRanges.Format( aStr, SCR_ABS_3D, pDocument, pDocument->GetAddressConvention() );
             aRangeString = aStr;
 
             // get "total" range for positioning
