diff --git svx/source/cui/acccfg.cxx svx/source/cui/acccfg.cxx
index 43e0bb5..e7b2bd1 100644
--- svx/source/cui/acccfg.cxx
+++ svx/source/cui/acccfg.cxx
@@ -219,6 +219,9 @@ static USHORT __FAR_DATA KEYCODE_ARRAY[] =
     KEY_X         | KEY_MOD1 ,
     KEY_Y         | KEY_MOD1 ,
     KEY_Z         | KEY_MOD1 ,
+    KEY_SEMICOLON    | KEY_MOD1 ,
+    KEY_BRACKETLEFT  | KEY_MOD1 ,
+    KEY_BRACKETRIGHT | KEY_MOD1, 
 
     KEY_F1        | KEY_MOD1 ,
     KEY_F2        | KEY_MOD1 ,
@@ -288,6 +291,9 @@ static USHORT __FAR_DATA KEYCODE_ARRAY[] =
     KEY_X         | KEY_SHIFT | KEY_MOD1,
     KEY_Y         | KEY_SHIFT | KEY_MOD1,
     KEY_Z         | KEY_SHIFT | KEY_MOD1,
+    KEY_SEMICOLON    | KEY_SHIFT | KEY_MOD1 ,
+    KEY_BRACKETLEFT  | KEY_SHIFT | KEY_MOD1 ,
+    KEY_BRACKETRIGHT | KEY_SHIFT | KEY_MOD1, 
 
     KEY_F1        | KEY_SHIFT | KEY_MOD1,
     KEY_F2        | KEY_SHIFT | KEY_MOD1,
@@ -352,6 +358,9 @@ static USHORT __FAR_DATA KEYCODE_ARRAY[] =
     KEY_X         | KEY_MOD2 ,
     KEY_Y         | KEY_MOD2 ,
     KEY_Z         | KEY_MOD2 ,
+    KEY_SEMICOLON    | KEY_MOD2 ,
+    KEY_BRACKETLEFT  | KEY_MOD2 ,
+    KEY_BRACKETRIGHT | KEY_MOD2, 
 
     KEY_F1        | KEY_MOD2 ,
     KEY_F2        | KEY_MOD2 ,
@@ -416,6 +425,9 @@ static USHORT __FAR_DATA KEYCODE_ARRAY[] =
     KEY_X         | KEY_SHIFT | KEY_MOD2,
     KEY_Y         | KEY_SHIFT | KEY_MOD2,
     KEY_Z         | KEY_SHIFT | KEY_MOD2,
+    KEY_SEMICOLON    | KEY_SHIFT | KEY_MOD2 ,
+    KEY_BRACKETLEFT  | KEY_SHIFT | KEY_MOD2 ,
+    KEY_BRACKETRIGHT | KEY_SHIFT | KEY_MOD2, 
 
     KEY_F1        | KEY_SHIFT | KEY_MOD2,
     KEY_F2        | KEY_SHIFT | KEY_MOD2,
@@ -544,6 +556,9 @@ static USHORT __FAR_DATA KEYCODE_ARRAY[] =
     KEY_X         | KEY_SHIFT | KEY_MOD1 | KEY_MOD2,
     KEY_Y         | KEY_SHIFT | KEY_MOD1 | KEY_MOD2,
     KEY_Z         | KEY_SHIFT | KEY_MOD1 | KEY_MOD2,
+    KEY_SEMICOLON    | KEY_SHIFT | KEY_MOD1 | KEY_MOD2,
+    KEY_BRACKETLEFT  | KEY_SHIFT | KEY_MOD1 | KEY_MOD2,
+    KEY_BRACKETRIGHT | KEY_SHIFT | KEY_MOD1 | KEY_MOD2,
 
     KEY_F1        | KEY_SHIFT | KEY_MOD1 | KEY_MOD2,
     KEY_F2        | KEY_SHIFT | KEY_MOD1 | KEY_MOD2,
