diff --git editeng/inc/editeng/editids.hrc editeng/inc/editeng/editids.hrc
index 0d1acb4..de4f4f4 100644
--- editeng/inc/editeng/editids.hrc
+++ editeng/inc/editeng/editids.hrc
@@ -129,6 +129,8 @@
 
 // this ID didn't exist prior to the svx split
 #define SID_ATTR_CHAR_OVERLINE              			( SID_EDIT_START + 68 )
+#define SID_ATTR_ALIGN_HOR_JUSTIFY_METHOD               ( SID_EDIT_START + 69 )
+#define SID_ATTR_ALIGN_VER_JUSTIFY                      ( SID_EDIT_START + 70 )
 
 /*
 // planned, but not doable: changing value of SIDs will break code that uses them for WhichRanges
