diff --git sc/source/ui/view/dbfunc.cxx sc/source/ui/view/dbfunc.cxx
index fa05fa6..326e611 100644
--- sc/source/ui/view/dbfunc.cxx
+++ sc/source/ui/view/dbfunc.cxx
@@ -302,7 +302,7 @@ void ScDBFunc::ToggleAutoFilter()
 
     ScQueryParam	aParam;
     ScDocument*		pDoc	= GetViewData()->GetDocument();
-    ScDBData*		pDBData = GetDBData( FALSE );
+    ScDBData*		pDBData = GetDBData(false, SC_DB_MAKE, false, true);
 
     pDBData->SetByRow( TRUE );				//! Undo, vorher abfragen ??
     pDBData->GetQueryParam( aParam );
