diff --git xmloff/inc/xmlnmspe.hxx xmloff/inc/xmlnmspe.hxx
index 1d6a96e..24fd279 100644
--- xmloff/inc/xmlnmspe.hxx
+++ xmloff/inc/xmlnmspe.hxx
@@ -102,6 +102,7 @@ XML_OLD_NAMESPACE( META,	6U )
 
 // experimental namespaces
 XML_NAMESPACE( FIELD,			100U )
+XML_NAMESPACE( CSS3TEXT,        103U )      // CSS Text Level 3
 
 
 #endif	//  _XMLOFF_XMLNMSPE_HXX
diff --git xmloff/inc/xmloff/xmltoken.hxx xmloff/inc/xmloff/xmltoken.hxx
index 0788be0..f49f434 100644
--- xmloff/inc/xmloff/xmltoken.hxx
+++ xmloff/inc/xmloff/xmltoken.hxx
@@ -148,6 +148,10 @@ namespace xmloff { namespace token {
         XML_NP_TABLE_EXT,
         XML_N_TABLE_EXT,
         
+        // css text level 3
+        XML_NP_CSS3TEXT,
+        XML_N_CSS3TEXT,
+
         // units
         XML_UNIT_MM,
         XML_UNIT_M,
@@ -603,6 +607,7 @@ namespace xmloff { namespace token {
         XML_DISTANCE,
         XML_DISTANCE_AFTER_SEP,
         XML_DISTANCE_BEFORE_SEP,
+        XML_DISTRIBUTE,
         XML_DISTRIBUTE_LETTER,
         XML_DISTRIBUTE_SPACE,
         XML_DIVIDE,
@@ -1777,6 +1782,7 @@ namespace xmloff { namespace token {
         XML_TEXT_GLOBAL,
         XML_TEXT_INDENT,
         XML_TEXT_INPUT,
+        XML_TEXT_JUSTIFY,
         XML_TEXT_OUTLINE,
         XML_TEXT_POSITION,
         XML_TEXT_ROTATION_ANGLE,
@@ -1910,6 +1916,7 @@ namespace xmloff { namespace token {
         XML_VERSION_LIST,
         XML_VERTICAL,
         XML_VERTICAL_ALIGN,
+        XML_VERTICAL_JUSTIFY,
         XML_VERTICAL_LINES,
         XML_VERTICAL_POS,
         XML_VERTICAL_REL,
diff --git xmloff/source/core/xmlexp.cxx xmloff/source/core/xmlexp.cxx
index a87141a..def76f4 100644
--- xmloff/source/core/xmlexp.cxx
+++ xmloff/source/core/xmlexp.cxx
@@ -422,6 +422,12 @@ void SvXMLExport::_InitCtor()
         mpNamespaceMap->Add( GetXMLToken(XML_NP_GRDDL),
             GetXMLToken(XML_N_GRDDL), XML_NAMESPACE_GRDDL );
     }
+    // CSS Text Level 3 for distributed text justification.
+    if ( (getExportFlags() & (EXPORT_STYLES|EXPORT_AUTOSTYLES|EXPORT_MASTERSTYLES)) != 0 )
+    {
+        mpNamespaceMap->Add(
+            GetXMLToken(XML_NP_CSS3TEXT), GetXMLToken(XML_N_CSS3TEXT), XML_NAMESPACE_CSS3TEXT );
+    }
 
     mxAttrList = (xml::sax::XAttributeList*)mpAttrList;
 
diff --git xmloff/source/core/xmlimp.cxx xmloff/source/core/xmlimp.cxx
index caa2610..2ed6ed2 100644
--- xmloff/source/core/xmlimp.cxx
+++ xmloff/source/core/xmlimp.cxx
@@ -118,6 +118,7 @@ sal_Char __READONLY_DATA sXML_np__xsd[] = "_xsd";
 sal_Char __READONLY_DATA sXML_np__xsi[] = "_xsi";
 sal_Char __READONLY_DATA sXML_np__field[] = "_field";
 sal_Char __READONLY_DATA sXML_np__xhtml[] = "_xhtml";
+sal_Char __READONLY_DATA sXML_np__css3text[] = "_css3text";
 
 sal_Char __READONLY_DATA sXML_np__fo_old[] = "__fo";
 sal_Char __READONLY_DATA sXML_np__xlink_old[] = "__xlink";
@@ -319,6 +320,9 @@ void SvXMLImport::_InitCtor()
         mpNamespaceMap->Add(
             OUString( RTL_CONSTASCII_USTRINGPARAM ( sXML_np__xhtml ) ),
             GetXMLToken(XML_N_XHTML), XML_NAMESPACE_XHTML );
+        mpNamespaceMap->Add(
+            OUString( RTL_CONSTASCII_USTRINGPARAM( sXML_np__css3text ) ),
+            GetXMLToken(XML_N_CSS3TEXT), XML_NAMESPACE_CSS3TEXT );
     }
 
     msPackageProtocol = OUString( RTL_CONSTASCII_USTRINGPARAM( "vnd.sun.star.Package:" ) );
diff --git xmloff/source/core/xmltoken.cxx xmloff/source/core/xmltoken.cxx
index d23e75a..34f0353 100644
--- xmloff/source/core/xmltoken.cxx
+++ xmloff/source/core/xmltoken.cxx
@@ -156,6 +156,9 @@ namespace xmloff { namespace token {
         TOKEN( "tableooo", XML_NP_TABLE_EXT ),
         TOKEN( "http://openoffice.org/2009/table", XML_N_TABLE_EXT ),
 
+        TOKEN( "css3t",                        XML_NP_CSS3TEXT ),
+        TOKEN( "http://www.w3.org/TR/css3-text/", XML_N_CSS3TEXT ),
+
         // units
         TOKEN( "mm",                              XML_UNIT_MM ),
         TOKEN( "m",                               XML_UNIT_M ),
@@ -611,6 +614,7 @@ namespace xmloff { namespace token {
         TOKEN( "distance",                        XML_DISTANCE ),
         TOKEN( "distance-after-sep",              XML_DISTANCE_AFTER_SEP ),
         TOKEN( "distance-before-sep",             XML_DISTANCE_BEFORE_SEP ),
+        TOKEN( "distribute",                      XML_DISTRIBUTE ),
         TOKEN( "distribute-letter",               XML_DISTRIBUTE_LETTER ),
         TOKEN( "distribute-space",                XML_DISTRIBUTE_SPACE ),
         TOKEN( "divide",                          XML_DIVIDE ),
@@ -1785,6 +1789,7 @@ namespace xmloff { namespace token {
         TOKEN( "text-global",                     XML_TEXT_GLOBAL ),
         TOKEN( "text-indent",                     XML_TEXT_INDENT ),
         TOKEN( "text-input",                      XML_TEXT_INPUT ),
+        TOKEN( "text-justify",                    XML_TEXT_JUSTIFY ),
         TOKEN( "text-outline",                    XML_TEXT_OUTLINE ),
         TOKEN( "text-position",                   XML_TEXT_POSITION ),
         TOKEN( "text-rotation-angle",             XML_TEXT_ROTATION_ANGLE ),
@@ -1918,6 +1923,7 @@ namespace xmloff { namespace token {
         TOKEN( "version-list",                    XML_VERSION_LIST ),
         TOKEN( "vertical",                        XML_VERTICAL ),
         TOKEN( "vertical-align",                  XML_VERTICAL_ALIGN ),
+        TOKEN( "vertical-justify",                XML_VERTICAL_JUSTIFY ),
         TOKEN( "vertical-lines",                  XML_VERTICAL_LINES ),
         TOKEN( "vertical-pos",                    XML_VERTICAL_POS ),
         TOKEN( "vertical-rel",                    XML_VERTICAL_REL ),
