diff --git sc/source/ui/unoobj/chart2uno.cxx sc/source/ui/unoobj/chart2uno.cxx
index 4246273..cb99932 100644
--- sc/source/ui/unoobj/chart2uno.cxx
+++ sc/source/ui/unoobj/chart2uno.cxx
@@ -500,6 +500,10 @@ void Chart2Positioner::glueState()
         ScRefTokenHelper::getDoubleRefDataFromToken(aData, *itr);
         SCCOLROW n1 = aData.Ref1.nCol;
         SCCOLROW n2 = aData.Ref2.nCol;
+        if (n1 > SCCOL_MAX)
+            n1 = SCCOL_MAX;
+        if (n2 > SCCOL_MAX)
+            n2 = SCCOL_MAX;
         SCCOLROW nTmp = n2 - n1 + 1;
         if (n1 < mnStartCol)
             mnStartCol = static_cast<SCCOL>(n1);
@@ -510,14 +514,18 @@ void Chart2Positioner::glueState()
 
         n1 = aData.Ref1.nRow;
         n2 = aData.Ref2.nRow;
+        if (n1 > SCROW_MAX)
+            n1 = SCROW_MAX;
+        if (n2 > SCROW_MAX)
+            n2 = SCROW_MAX;
         nTmp = n2 - n1 + 1;
 
         if (n1 < mnStartRow)
-            mnStartRow = static_cast<SCCOL>(n1);
+            mnStartRow = static_cast<SCROW>(n1);
         if (n2 > nEndRow)
-            nEndRow = static_cast<SCCOL>(n2);
+            nEndRow = static_cast<SCROW>(n2);
         if (nTmp > nMaxRows)
-            nMaxRows = static_cast<SCCOL>(nTmp);
+            nMaxRows = static_cast<SCROW>(nTmp);
     }
 
     // total column size ?
