--- vcl/inc/vcl/keycodes.hxx.old	2009-10-08 16:17:47.000000000 +0200
+++ vcl/inc/vcl/keycodes.hxx	2010-03-03 16:59:14.000000000 +0100
@@ -158,6 +158,8 @@
 #define KEY_DECIMAL     ((USHORT)::com::sun::star::awt::Key::DECIMAL)
 #define KEY_TILDE       ((USHORT)::com::sun::star::awt::Key::TILDE)
 #define KEY_QUOTELEFT   ((USHORT)::com::sun::star::awt::Key::QUOTELEFT)
+#define KEY_BRACKETLEFT ((USHORT)::com::sun::star::awt::Key::BRACKETLEFT)
+#define KEY_BRACKETRIGHT ((USHORT)::com::sun::star::awt::Key::BRACKETRIGHT)
 
 #define KEY_CODE		((USHORT)0x0FFF)
 
--- vcl/unx/gtk/window/gtkframe.cxx.old	2010-01-21 01:00:41.000000000 +0100
+++ vcl/unx/gtk/window/gtkframe.cxx	2010-03-03 16:59:14.000000000 +0100
@@ -214,6 +214,8 @@ static USHORT GetKeyCode( guint keyval )
             case GDK_asciitilde:	nCode = KEY_TILDE;		break;
             case GDK_leftsinglequotemark:
             case GDK_quoteleft:	nCode = KEY_QUOTELEFT;		break;           
+            case GDK_bracketleft:  nCode = KEY_BRACKETLEFT;  break;
+            case GDK_bracketright: nCode = KEY_BRACKETRIGHT; break;
             // some special cases, also see saldisp.cxx
             // - - - - - - - - - - - - -  Apollo - - - - - - - - - - - - - 0x1000
             case 0x1000FF02: // apXK_Copy
--- vcl/unx/source/app/saldisp.cxx.old	2010-01-21 01:00:41.000000000 +0100
+++ vcl/unx/source/app/saldisp.cxx	2010-03-03 16:59:14.000000000 +0100
@@ -1267,6 +1267,12 @@ XubString SalDisplay::GetKeyName( USHORT
         case KEY_QUOTELEFT:
             nKeySym = XK_grave;
             break;
+        case KEY_BRACKETLEFT:
+            nKeySym = XK_bracketleft;
+            break;
+        case KEY_BRACKETRIGHT:
+            nKeySym = XK_bracketright;
+            break;
 
         default:
             nKeySym = 0;
@@ -1597,6 +1603,14 @@ USHORT SalDisplay::GetKeyCode( KeySym ke
             nKey = KEY_QUOTELEFT;
             *pcPrintable = '`';
             break;
+        case XK_bracketleft:
+            nKey = KEY_BRACKETLEFT;
+            *pcPrintable = '[';
+            break;
+         case XK_bracketright:
+             nKey = KEY_BRACKETRIGHT;
+             *pcPrintable = ']';
+             break;
 //      case XK_Linefeed:
 //          *pcPrintable = '\n';
 //          break;
--- vcl/win/source/app/salinst.cxx.old	2009-10-19 13:28:02.000000000 +0200
+++ vcl/win/source/app/salinst.cxx	2010-03-03 16:59:14.000000000 +0100
@@ -378,6 +378,8 @@ void SalData::initKeyCodeMap()
     initKey( '=', KEY_EQUAL );
     initKey( '~', KEY_TILDE );
     initKey( '`', KEY_QUOTELEFT );
+    initKey( '[', KEY_BRACKETLEFT );
+    initKey( ']', KEY_BRACKETRIGHT );
 }
 
 // =======================================================================
