From be80368f1f2940c2e092a2ce9c085ca1291b9fbc Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:56:38 +0200
Subject: [PATCH 220/878] calc-xlsx-import-default-cellstyle.diff

---
 oox/source/xls/stylesbuffer.cxx |   13 +++++++++++++
 1 files changed, 13 insertions(+), 0 deletions(-)

diff --git a/oox/source/xls/stylesbuffer.cxx b/oox/source/xls/stylesbuffer.cxx
index 6ccf171..ad61e33 100644
--- a/oox/source/xls/stylesbuffer.cxx
+++ b/oox/source/xls/stylesbuffer.cxx
@@ -2945,6 +2945,9 @@ void CellStyleBuffer::finalizeImport()
     for( CellStyleVector::iterator aIt = maBuiltinStyles.begin(), aEnd = maBuiltinStyles.end(); aIt != aEnd; ++aIt )
     {
         const CellStyleModel& rModel = (*aIt)->getModel();
+        if (rModel.isDefaultStyle())
+            continue;
+
         OUString aStyleName = lclGetBuiltinStyleName( rModel.mnBuiltinId, rModel.maName, rModel.mnLevel );
         OSL_ENSURE( bReserveAll || (aCellStyles.count( aStyleName ) == 0),
             "CellStyleBuffer::finalizeImport - multiple styles with equal built-in identifier" );
@@ -2985,6 +2988,16 @@ void CellStyleBuffer::finalizeImport()
 
     // set final names and create user-defined and modified built-in cell styles
     aCellStyles.forEachMemWithKey( &CellStyle::finalizeImport );
+
+    if (mxDefStyle)
+    {
+        Reference<XNameAccess> xNA(getStyleFamily(false), UNO_QUERY_THROW);
+        if (xNA->hasByName(CREATE_OUSTRING("Default")))
+        {
+            PropertySet aPropSet(xNA->getByName(CREATE_OUSTRING("Default")));
+            getStyles().writeStyleXfToPropertySet(aPropSet, mxDefStyle->getModel().mnXfId);
+        }
+    }
 }
 
 sal_Int32 CellStyleBuffer::getDefaultXfId() const
-- 
1.7.0.1

