diff --git sc/source/core/data/table5.cxx sc/source/core/data/table5.cxx
index 5eea2b0..97c6512 100644
--- sc/source/core/data/table5.cxx
+++ sc/source/core/data/table5.cxx
@@ -75,7 +75,7 @@ void ScTable::UpdatePageBreaks( const ScRange* pUserArea )
     if ( !pUserArea && !bPageSizeValid )
         return;
 
-    if (mbPageBreaksValid)
+    if (!pUserArea && mbPageBreaksValid)
         return;
 
     SfxStyleSheetBase* pStyle = pDocument->GetStyleSheetPool()->
