From 607e95617a5f1966f2f7d0ede67e01ad41715581 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:08:19 +0200
Subject: [PATCH 808/878] calc-insert-current-time-svx.diff

---
 svx/source/cui/acccfg.cxx |   15 +++++++++++++++
 1 files changed, 15 insertions(+), 0 deletions(-)

diff --git a/svx/source/cui/acccfg.cxx b/svx/source/cui/acccfg.cxx
index 73a3822..7a2bc4a 100644
--- a/svx/source/cui/acccfg.cxx
+++ b/svx/source/cui/acccfg.cxx
@@ -216,6 +216,9 @@ static USHORT __FAR_DATA KEYCODE_ARRAY[] =
     KEY_X         | KEY_MOD1 ,
     KEY_Y         | KEY_MOD1 ,
     KEY_Z         | KEY_MOD1 ,
+    KEY_SEMICOLON    | KEY_MOD1 ,
+    KEY_BRACKETLEFT  | KEY_MOD1 ,
+    KEY_BRACKETRIGHT | KEY_MOD1,
 
     KEY_F1        | KEY_MOD1 ,
     KEY_F2        | KEY_MOD1 ,
@@ -285,6 +288,9 @@ static USHORT __FAR_DATA KEYCODE_ARRAY[] =
     KEY_X         | KEY_SHIFT | KEY_MOD1,
     KEY_Y         | KEY_SHIFT | KEY_MOD1,
     KEY_Z         | KEY_SHIFT | KEY_MOD1,
+    KEY_SEMICOLON    | KEY_SHIFT | KEY_MOD1 ,
+    KEY_BRACKETLEFT  | KEY_SHIFT | KEY_MOD1 ,
+    KEY_BRACKETRIGHT | KEY_SHIFT | KEY_MOD1,
 
     KEY_F1        | KEY_SHIFT | KEY_MOD1,
     KEY_F2        | KEY_SHIFT | KEY_MOD1,
@@ -349,6 +355,9 @@ static USHORT __FAR_DATA KEYCODE_ARRAY[] =
     KEY_X         | KEY_MOD2 ,
     KEY_Y         | KEY_MOD2 ,
     KEY_Z         | KEY_MOD2 ,
+    KEY_SEMICOLON    | KEY_MOD2 ,
+    KEY_BRACKETLEFT  | KEY_MOD2 ,
+    KEY_BRACKETRIGHT | KEY_MOD2,
 
     KEY_F1        | KEY_MOD2 ,
     KEY_F2        | KEY_MOD2 ,
@@ -413,6 +422,9 @@ static USHORT __FAR_DATA KEYCODE_ARRAY[] =
     KEY_X         | KEY_SHIFT | KEY_MOD2,
     KEY_Y         | KEY_SHIFT | KEY_MOD2,
     KEY_Z         | KEY_SHIFT | KEY_MOD2,
+    KEY_SEMICOLON    | KEY_SHIFT | KEY_MOD2 ,
+    KEY_BRACKETLEFT  | KEY_SHIFT | KEY_MOD2 ,
+    KEY_BRACKETRIGHT | KEY_SHIFT | KEY_MOD2,
 
     KEY_F1        | KEY_SHIFT | KEY_MOD2,
     KEY_F2        | KEY_SHIFT | KEY_MOD2,
@@ -541,6 +553,9 @@ static USHORT __FAR_DATA KEYCODE_ARRAY[] =
     KEY_X         | KEY_SHIFT | KEY_MOD1 | KEY_MOD2,
     KEY_Y         | KEY_SHIFT | KEY_MOD1 | KEY_MOD2,
     KEY_Z         | KEY_SHIFT | KEY_MOD1 | KEY_MOD2,
+    KEY_SEMICOLON    | KEY_SHIFT | KEY_MOD1 | KEY_MOD2,
+    KEY_BRACKETLEFT  | KEY_SHIFT | KEY_MOD1 | KEY_MOD2,
+    KEY_BRACKETRIGHT | KEY_SHIFT | KEY_MOD1 | KEY_MOD2,
 
     KEY_F1        | KEY_SHIFT | KEY_MOD1 | KEY_MOD2,
     KEY_F2        | KEY_SHIFT | KEY_MOD1 | KEY_MOD2,
-- 
1.7.0.1

