---
 cui/source/customize/acccfg.cxx |   15 +++++++++++++++
 1 files changed, 15 insertions(+), 0 deletions(-)

diff --git cui/source/customize/acccfg.cxx cui/source/customize/acccfg.cxx
index 50e22f9..43499f9 100644
--- cui/source/customize/acccfg.cxx
+++ cui/source/customize/acccfg.cxx
@@ -218,6 +218,9 @@ static USHORT __FAR_DATA KEYCODE_ARRAY[] =
     KEY_X         | KEY_MOD1 ,
     KEY_Y         | KEY_MOD1 ,
     KEY_Z         | KEY_MOD1 ,
+    KEY_SEMICOLON    | KEY_MOD1 ,
+    KEY_BRACKETLEFT  | KEY_MOD1 ,
+    KEY_BRACKETRIGHT | KEY_MOD1,
 
     KEY_F1        | KEY_MOD1 ,
     KEY_F2        | KEY_MOD1 ,
@@ -287,6 +290,9 @@ static USHORT __FAR_DATA KEYCODE_ARRAY[] =
     KEY_X         | KEY_SHIFT | KEY_MOD1,
     KEY_Y         | KEY_SHIFT | KEY_MOD1,
     KEY_Z         | KEY_SHIFT | KEY_MOD1,
+    KEY_SEMICOLON    | KEY_SHIFT | KEY_MOD1 ,
+    KEY_BRACKETLEFT  | KEY_SHIFT | KEY_MOD1 ,
+    KEY_BRACKETRIGHT | KEY_SHIFT | KEY_MOD1,
 
     KEY_F1        | KEY_SHIFT | KEY_MOD1,
     KEY_F2        | KEY_SHIFT | KEY_MOD1,
@@ -351,6 +357,9 @@ static USHORT __FAR_DATA KEYCODE_ARRAY[] =
     KEY_X         | KEY_MOD2 ,
     KEY_Y         | KEY_MOD2 ,
     KEY_Z         | KEY_MOD2 ,
+    KEY_SEMICOLON    | KEY_MOD2 ,
+    KEY_BRACKETLEFT  | KEY_MOD2 ,
+    KEY_BRACKETRIGHT | KEY_MOD2,
 
     KEY_F1        | KEY_MOD2 ,
     KEY_F2        | KEY_MOD2 ,
@@ -415,6 +424,9 @@ static USHORT __FAR_DATA KEYCODE_ARRAY[] =
     KEY_X         | KEY_SHIFT | KEY_MOD2,
     KEY_Y         | KEY_SHIFT | KEY_MOD2,
     KEY_Z         | KEY_SHIFT | KEY_MOD2,
+    KEY_SEMICOLON    | KEY_SHIFT | KEY_MOD2 ,
+    KEY_BRACKETLEFT  | KEY_SHIFT | KEY_MOD2 ,
+    KEY_BRACKETRIGHT | KEY_SHIFT | KEY_MOD2,
 
     KEY_F1        | KEY_SHIFT | KEY_MOD2,
     KEY_F2        | KEY_SHIFT | KEY_MOD2,
@@ -543,6 +555,9 @@ static USHORT __FAR_DATA KEYCODE_ARRAY[] =
     KEY_X         | KEY_SHIFT | KEY_MOD1 | KEY_MOD2,
     KEY_Y         | KEY_SHIFT | KEY_MOD1 | KEY_MOD2,
     KEY_Z         | KEY_SHIFT | KEY_MOD1 | KEY_MOD2,
+    KEY_SEMICOLON    | KEY_SHIFT | KEY_MOD1 | KEY_MOD2,
+    KEY_BRACKETLEFT  | KEY_SHIFT | KEY_MOD1 | KEY_MOD2,
+    KEY_BRACKETRIGHT | KEY_SHIFT | KEY_MOD1 | KEY_MOD2,
 
     KEY_F1        | KEY_SHIFT | KEY_MOD1 | KEY_MOD2,
     KEY_F2        | KEY_SHIFT | KEY_MOD1 | KEY_MOD2,
-- 
1.7.0.1

