diff --git sc/source/ui/app/inputhdl.cxx sc/source/ui/app/inputhdl.cxx
index 039f6cd..6146861 100644
--- sc/source/ui/app/inputhdl.cxx
+++ sc/source/ui/app/inputhdl.cxx
@@ -2579,17 +2579,11 @@ void ScInputHandler::EnterHandler( BYTE nBlockMode )
             pObject = pEngine->CreateTextObject();
         }
 		else if (bAutoComplete)			// Gross-/Kleinschreibung anpassen
-		{
-			if (pColumnData)
-				pColumnData->GetExactMatch( aString );
-
-			//!	effizienter in der Liste suchen (ScUserList, nur einmal ToUpper)
-
-			USHORT nIndex;
-			ScUserListData* pData = ScGlobal::GetUserList()->GetData(aString);
-			if ( pData && pData->GetSubIndex( aString, nIndex ) )
-				aString = pData->GetSubStr( nIndex );
-		}
+        {
+            // Perform case-matching only when the typed text is partial.
+            if (pColumnData && aAutoSearch.Len() < aString.Len())
+                pColumnData->GetExactMatch(aString);
+        }
     }
 
     //	don't rely on ShowRefFrame switching the active view synchronously
