diff --git sc/source/core/data/table2.cxx sc/source/core/data/table2.cxx
index c3ee408..86e9e1d 100644
--- sc/source/core/data/table2.cxx
+++ sc/source/core/data/table2.cxx
@@ -2500,7 +2500,6 @@ void ScTable::DBShowRow(SCROW nRow, BOOL bShow)
 void ScTable::DBShowRows(SCROW nRow1, SCROW nRow2, BOOL bShow)
 {
     SCROW nStartRow = nRow1;
-    nRecalcLvl++;
     while (nStartRow <= nRow2)
     {
         SCROW nEndRow = -1;
@@ -2540,9 +2539,6 @@ void ScTable::DBShowRows(SCROW nRow1, SCROW nRow2, BOOL bShow)
     //	to be done here.
     if (pOutlineTable)
         UpdateOutlineRow( nRow1, nRow2, bShow );
-
-    if( !--nRecalcLvl )
-        SetDrawPageSize();
 }
 
 
diff --git sc/source/core/data/table3.cxx sc/source/core/data/table3.cxx
index 6da3d50..20b4f49 100644
--- sc/source/core/data/table3.cxx
+++ sc/source/core/data/table3.cxx
@@ -1648,6 +1648,8 @@ SCSIZE ScTable::Query(ScQueryParam& rParamOrg, BOOL bKeepSub)
 
     delete[] pSpecial;
 
+    SetDrawPageSize();
+
     return nCount;
 }
 
