---
 sw/inc/tblafmt.hxx |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git sw/inc/tblafmt.hxx sw/inc/tblafmt.hxx
index 0e74744..36864e2 100644
--- sw/inc/tblafmt.hxx
+++ sw/inc/tblafmt.hxx
@@ -61,6 +61,7 @@ JP 20.07.95:
 #include <editeng/boxitem.hxx>
 #include <editeng/brshitem.hxx>
 #include <editeng/adjitem.hxx>
+#include <editeng/justifyitem.hxx>
 #include <svx/rotmodit.hxx>
 #include <svl/intitem.hxx>
 #include <editeng/bolnitem.hxx>
-- 
1.7.0.1

