--- sc/inc/attarray.hxx.old	2010-03-03 16:59:11.000000000 +0100
+++ sc/inc/attarray.hxx	2010-03-03 16:59:18.000000000 +0100
@@ -39,6 +39,7 @@ class ScEditDataArray;
 class ScMarkArray;
 class ScPatternAttr;
 class ScStyleSheet;
+class ScFlatBoolRowSegments;
 
 class Rectangle;
 class SfxItemPoolCache;
@@ -163,7 +164,7 @@ public:
                                 BOOL bRefresh, BOOL bAttrs );
     BOOL	RemoveAreaMerge( SCROW nStartRow, SCROW nEndRow );
 
-    void	FindStyleSheet( const SfxStyleSheetBase* pStyleSheet, BOOL* pUsed, BOOL bReset );
+    void	FindStyleSheet( const SfxStyleSheetBase* pStyleSheet, ScFlatBoolRowSegments& rUsedRows, bool bReset );
     BOOL	IsStyleSheetUsed( const ScStyleSheet& rStyle, BOOL bGatherAllStyles ) const;
 
     void	DeleteAreaSafe(SCROW nStartRow, SCROW nEndRow);
--- sc/inc/column.hxx.old	2010-03-03 16:59:15.000000000 +0100
+++ sc/inc/column.hxx	2010-03-03 16:59:18.000000000 +0100
@@ -70,7 +70,7 @@ struct ScFunctionData;
 struct ScLineFlags;
 struct ScMergePatternState;
 struct ScSetStringParam;
-
+class ScFlatBoolRowSegments;
 
 #define COLUMN_DELTA	4
 
@@ -336,7 +336,7 @@ public:
     const ScStyleSheet*	GetSelectionStyle( const ScMarkData& rMark, BOOL& rFound ) const;
     const ScStyleSheet*	GetAreaStyle( BOOL& rFound, SCROW nRow1, SCROW nRow2 ) const;
 
-    void		FindStyleSheet( const SfxStyleSheetBase* pStyleSheet, BOOL* pUsed, BOOL bReset );
+    void		FindStyleSheet( const SfxStyleSheetBase* pStyleSheet, ScFlatBoolRowSegments& rUsedRows, bool bReset );
     BOOL		IsStyleSheetUsed( const ScStyleSheet& rStyle, BOOL bGatherAllStyles ) const;
 
                 /// May return -1 if not found
--- sc/source/core/data/attarray.cxx.old	2010-03-03 16:59:11.000000000 +0100
+++ sc/source/core/data/attarray.cxx	2010-03-03 16:59:18.000000000 +0100
@@ -57,6 +57,7 @@
 #include "rechead.hxx"
 #include "globstr.hrc"
 #include "cell.hxx"
+#include "segmenttree.hxx"
 
 #undef DBG_INVALIDATE
 #define DBGOUTPUT(s) \
@@ -1761,8 +1762,7 @@ SCsROW ScAttrArray::GetNextUnprotected(
     return nRet;
 }
 
-
-void ScAttrArray::FindStyleSheet( const SfxStyleSheetBase* pStyleSheet, BOOL* pUsed, BOOL bReset )
+void ScAttrArray::FindStyleSheet( const SfxStyleSheetBase* pStyleSheet, ScFlatBoolRowSegments& rUsedRows, bool bReset )
 {
     SCROW nStart = 0;
     SCSIZE nPos = 0;
@@ -1774,7 +1774,7 @@ void ScAttrArray::FindStyleSheet( const
 //			for (SCROW nRow = nStart; nRow <= nEnd; nRow++)
 //				pUsed[nRow] = TRUE;
 
-            memset( &pUsed[nStart], TRUE, nEnd-nStart+1 );
+            rUsedRows.setTrue(nStart, nEnd);
 
             if (bReset)
             {
--- sc/source/core/data/column.cxx.old	2010-03-03 16:59:13.000000000 +0100
+++ sc/source/core/data/column.cxx	2010-03-03 16:59:18.000000000 +0100
@@ -620,13 +620,11 @@ const ScStyleSheet*	ScColumn::GetAreaSty
     return bEqual ? pStyle : NULL;
 }
 
-
-void ScColumn::FindStyleSheet( const SfxStyleSheetBase* pStyleSheet, BOOL* pUsed, BOOL bReset )
+void ScColumn::FindStyleSheet( const SfxStyleSheetBase* pStyleSheet, ScFlatBoolRowSegments& rUsedRows, bool bReset )
 {
-    pAttrArray->FindStyleSheet( pStyleSheet, pUsed, bReset );
+    pAttrArray->FindStyleSheet( pStyleSheet, rUsedRows, bReset );
 }
 
-
 BOOL ScColumn::IsStyleSheetUsed( const ScStyleSheet& rStyle, BOOL bGatherAllStyles ) const
 {
     return pAttrArray->IsStyleSheetUsed( rStyle, bGatherAllStyles );
--- sc/source/core/data/table2.cxx.old	2010-03-03 16:59:17.000000000 +0100
+++ sc/source/core/data/table2.cxx	2010-03-03 16:59:18.000000000 +0100
@@ -60,6 +60,7 @@
 #include "bcaslot.hxx"
 #include "postit.hxx"
 #include "globstr.hrc"
+#include "segmenttree.hxx"
 
 #include <math.h>
 
@@ -1867,36 +1868,24 @@ void ScTable::StyleSheetChanged( const S
                                 double nPPTX, double nPPTY,
                                 const Fraction& rZoomX, const Fraction& rZoomY )
 {
-    BOOL* pUsed = new BOOL[MAXROWCOUNT];
-    memset( pUsed, 0, sizeof(BOOL) * (MAXROWCOUNT) );
+    ScFlatBoolRowSegments aUsedRows;
+    for (SCCOL i = 0; i <= MAXCOL; ++i)
+        aCol[i].FindStyleSheet(pStyleSheet, aUsedRows, bRemoved);
+
+    SCROW nRow = 0;
+    while (nRow <= MAXROW)
+    {
+        ScFlatBoolRowSegments::RangeData aData;
+        if (!aUsedRows.getRangeData(nRow, aData))
+            // search failed!
+            return;
+
+        SCROW nEndRow = aData.mnRow2;
+        if (aData.mbValue)
+            SetOptimalHeight(nRow, nEndRow, 0, pDev, nPPTX, nPPTY, rZoomX, rZoomY, FALSE);
 
-    SCCOL nCol;
-    for (nCol=0; nCol<=MAXCOL; nCol++)
-        aCol[nCol].FindStyleSheet( pStyleSheet, pUsed, bRemoved );
-
-    BOOL bFound = FALSE;
-    SCROW nStart = 0, nEnd = 0;
-    for (SCROW i=0; i<=MAXROW; i++)
-    {
-        if (pUsed[i])
-        {
-            if (!bFound)
-            {
-                nStart = i;
-                bFound = TRUE;
-            }
-            nEnd = i;
-        }
-        else if (bFound)
-        {
-            SetOptimalHeight( nStart, nEnd, 0, pDev, nPPTX, nPPTY, rZoomX, rZoomY, FALSE );
-            bFound = FALSE;
-        }
+        nRow = nEndRow + 1;
     }
-    if (bFound)
-        SetOptimalHeight( nStart, nEnd, 0, pDev, nPPTX, nPPTY, rZoomX, rZoomY, FALSE );
-
-    delete[] pUsed;
 }
 
 
