---
 svtools/inc/svtools/svtools.hrc |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git svtools/inc/svtools/svtools.hrc svtools/inc/svtools/svtools.hrc
index f2d492a..3d8010b 100644
--- svtools/inc/svtools/svtools.hrc
+++ svtools/inc/svtools/svtools.hrc
@@ -307,6 +307,7 @@
 #define BMP_HELP_AGENT_IMAGE			(RID_SVTOOLS_BITMAP_START +   1)
 #define BMP_HELP_AGENT_CLOSER			(RID_SVTOOLS_BITMAP_START +   2)
 #define BMP_PLUGIN                      (RID_SVTOOLS_BITMAP_START +   3)
+#define BMP_LIST_ADD                    (RID_SVTOOLS_BITMAP_START +   4)
 
 //.............................................................................
 // image lists
-- 
1.7.0.1

