diff --git svtools/source/control/tabbar.cxx svtools/source/control/tabbar.cxx
index 4e0e586..02da672 100644
--- svtools/source/control/tabbar.cxx
+++ svtools/source/control/tabbar.cxx
@@ -1384,6 +1384,10 @@ void TabBar::Paint( const Rectangle& )
             else
                 SetTextColor( aFaceTextColor );
 
+            // This tab is "special", and a special tab needs a blue text.
+            if (bSpecialTab)
+                SetTextColor(Color(COL_LIGHTBLUE));
+
             aDrawer.drawText(aText);
 
             if ( bCurrent )
