diff --git filter/source/pdf/impdialog.cxx filter/source/pdf/impdialog.cxx
index dff72eb..f8ac963 100644
--- filter/source/pdf/impdialog.cxx
+++ filter/source/pdf/impdialog.cxx
@@ -122,7 +122,31 @@ ImpPDFTabDialog::ImpPDFTabDialog( Window* pParent,
         {
             Reference< view::XSelectionSupplier > xView( xController, UNO_QUERY );
             if( xView.is() )
+            {
+                // In Calc, we need to treat a selection across hidden or 
+                // filtered ranges as one range, or else the export will
+                // fail because Calc doesn't support printing of multiple
+                // selections.
+                Reference< beans::XPropertySet > xPropSet(xView, UNO_QUERY);
+                if (xPropSet.is())
+                {
+                    sal_Bool bFilteredRange = sal_False; // Don't divide selection across filtered ranges.
+                    uno::Any any;
+                    any <<= bFilteredRange;
+                    try
+                    {
+                        xPropSet->setPropertyValue(
+                            rtl::OUString::createFromAscii("FilteredRangeSelection"), any);
+                    }
+                    catch (const beans::UnknownPropertyException&)
+                    {
+                        // For now, apps other than Calc fails to set this property,
+                        // hence an exception gets thrown.
+                    }
+                }
+
                 xView->getSelection() >>= maSelection;
+            }
         }
     }
     catch( RuntimeException )
