--- sc/inc/table.hxx.old	2010-09-27 14:44:56.000000000 +0200
+++ sc/inc/table.hxx	2010-09-27 14:44:49.000000000 +0200
@@ -676,6 +676,9 @@
     void		UpdatePageBreaks( const ScRange* pUserArea );
     void		RemoveManualBreaks();
     BOOL		HasManualBreaks() const;
+    void        SetRowManualBreaks( const ::std::set<SCROW>& rBreaks );
+    void        SetColManualBreaks( const ::std::set<SCCOL>& rBreaks );
+
 
     void        GetAllRowBreaks(::std::set<SCROW>& rBreaks, bool bPage, bool bManual) const;
     void        GetAllColBreaks(::std::set<SCCOL>& rBreaks, bool bPage, bool bManual) const;
--- sc/source/core/data/table2.cxx.old	2010-09-27 16:24:32.000000000 +0200
+++ sc/source/core/data/table2.cxx	2010-09-27 16:53:32.000000000 +0200
@@ -665,6 +665,7 @@
                 pDestTab->IncRecalcLevel();
 
                 if (bWidth)
+                {
                     for (SCCOL i=nCol1; i<=nCol2; i++)
                     {
                         bool bThisHidden = ColHidden(i);
@@ -680,6 +681,8 @@
 						if (bChange)
                             bFlagChange = true;
                     }
+                    pDestTab->SetColManualBreaks( maColManualBreaks );
+                }
 
                 if (bHeight)
                 {
@@ -731,6 +734,7 @@
                         pDestTab->SetRowFiltered(i, nLastRow, bFiltered);
                         i = nLastRow;
                     }
+                    pDestTab->SetRowManualBreaks( maRowManualBreaks );
                 }
                 pDestTab->DecRecalcLevel();
             }
@@ -770,9 +774,12 @@
             if (bWidth)
                 for (SCCOL i=nCol1; i<=nCol2; i++)
                     pDestTab->pColWidth[i] = pColWidth[i];
+                pDestTab->SetColManualBreaks( maColManualBreaks );
             if (bHeight)
+            {
                 pDestTab->CopyRowHeight(*this, nRow1, nRow2, 0);
-
+                pDestTab->SetRowManualBreaks( maRowManualBreaks );
+            }
             DecRecalcLevel();
         }
     }
--- sc/source/core/data/table5.cxx.old	2010-09-27 14:50:18.000000000 +0200
+++ sc/source/core/data/table5.cxx	2010-09-27 14:50:03.000000000 +0200
@@ -317,6 +317,22 @@
     return !maRowManualBreaks.empty() || !maColManualBreaks.empty();
 }
 
+void ScTable::SetRowManualBreaks( const ::std::set<SCROW>& rBreaks )
+{
+    maRowManualBreaks = rBreaks;
+    InvalidatePageBreaks();
+    if (IsStreamValid())
+        SetStreamValid(FALSE);
+}
+
+void ScTable::SetColManualBreaks( const ::std::set<SCCOL>& rBreaks )
+{
+    maColManualBreaks = rBreaks;
+    InvalidatePageBreaks();
+    if (IsStreamValid())
+        SetStreamValid(FALSE);
+}
+
 void ScTable::GetAllRowBreaks(set<SCROW>& rBreaks, bool bPage, bool bManual) const
 {
     if (bPage)

