diff --git sc/source/core/data/column2.cxx sc/source/core/data/column2.cxx
index 0ce77fb..d303e45 100644
--- sc/source/core/data/column2.cxx
+++ sc/source/core/data/column2.cxx
@@ -195,6 +195,7 @@ long ScColumn::GetNeededSize( SCROW nRow, OutputDevice* pDev,
     double nPPT = bWidth ? nPPTX : nPPTY;
     if (Search(nRow,nIndex))
     {
+        ScBaseCell* pCell = pItems[nIndex].pCell;
         const ScPatternAttr* pPattern = rOptions.pPattern;
         if (!pPattern)
             pPattern = pAttrArray->GetPattern( nRow );
@@ -235,22 +236,26 @@ long ScColumn::GetNeededSize( SCROW nRow, OutputDevice* pDev,
         else
             eHorJust = (SvxCellHorJustify)((const SvxHorJustifyItem&)
                                             pPattern->GetItem( ATTR_HOR_JUSTIFY )).GetValue();
-        BOOL bBreak;
+        bool bBreak;
         if ( eHorJust == SVX_HOR_JUSTIFY_BLOCK )
-            bBreak = TRUE;
+            bBreak = true;
         else if ( pCondSet &&
                     pCondSet->GetItemState(ATTR_LINEBREAK, TRUE, &pCondItem) == SFX_ITEM_SET)
             bBreak = ((const SfxBoolItem*)pCondItem)->GetValue();
         else
             bBreak = ((const SfxBoolItem&)pPattern->GetItem(ATTR_LINEBREAK)).GetValue();
 
+        if (pCell->HasValueData())
+            // Cell has a value.  Disable line break.
+            bBreak = false;
+
         //	get other attributes from pattern and conditional formatting
 
         SvxCellOrientation eOrient = pPattern->GetCellOrientation( pCondSet );
         BOOL bAsianVertical = ( eOrient == SVX_ORIENTATION_STACKED &&
                 ((const SfxBoolItem&)pPattern->GetItem( ATTR_VERTICAL_ASIAN, pCondSet )).GetValue() );
         if ( bAsianVertical )
-            bBreak = FALSE;
+            bBreak = false;
 
         if ( bWidth && bBreak )		// after determining bAsianVertical (bBreak may be reset)
             return 0;
@@ -302,7 +307,6 @@ long ScColumn::GetNeededSize( SCROW nRow, OutputDevice* pDev,
                 nIndent = ((const SfxUInt16Item&)pPattern->GetItem(ATTR_INDENT)).GetValue();
         }
 
-        ScBaseCell* pCell = pItems[nIndex].pCell;
         BYTE nScript = pDocument->GetScriptType( nCol, nRow, nTab, pCell );
         if (nScript == 0) nScript = ScGlobal::GetDefaultScriptType();
 
diff --git sc/source/ui/view/output2.cxx sc/source/ui/view/output2.cxx
index 8c7e649..ad3549f 100644
--- sc/source/ui/view/output2.cxx
+++ sc/source/ui/view/output2.cxx
@@ -75,6 +75,8 @@
 #include "scmod.hxx"
 #include "fillinfo.hxx"
 
+#include <boost/ptr_container/ptr_vector.hpp>
+
 #include <math.h>
 
 //!	Autofilter-Breite mit column.cxx zusammenfassen
@@ -1344,10 +1346,6 @@ void ScOutputData::DrawStrings( BOOL bPixelToLogic )
 
     ScDrawStringsVars aVars( this, bPixelToLogic );
 
-    const ScPatternAttr* pOldPattern = NULL;
-    const SfxItemSet* pOldCondSet = NULL;
-    BYTE nOldScript = 0;
-
     BOOL bProgress = FALSE;
 
     long nInitPosX = nScrX;
@@ -1370,6 +1368,13 @@ void ScOutputData::DrawStrings( BOOL bPixelToLogic )
     SvxCellHorJustify eOutHorJust = SVX_HOR_JUSTIFY_STANDARD;
     const ScPatternAttr* pPattern = NULL;
     const SfxItemSet* pCondSet = NULL;
+    const ScPatternAttr* pOldPattern = NULL;
+    const SfxItemSet* pOldCondSet = NULL;
+    BYTE nOldScript = 0;
+
+    // alternative pattern instances in case we need to modify the pattern 
+    // before processing the cell value.
+    ::boost::ptr_vector<ScPatternAttr> aAltPatterns;
 
     long nPosY = nScrY;
     for (SCSIZE nArrY=1; nArrY+1<nArrCount; nArrY++)
@@ -1507,6 +1512,18 @@ void ScOutputData::DrawStrings( BOOL bPixelToLogic )
                         pCondSet = pDoc->GetCondResult( nCellX, nCellY, nTab );
                     }
 
+                    if (pCell->HasValueData() && 
+                        static_cast<const SfxBoolItem&>(
+                            pPattern->GetItem(ATTR_LINEBREAK, pCondSet)).GetValue())
+                    {
+                        // Disable line break when the cell content is numeric.
+                        aAltPatterns.push_back(new ScPatternAttr(*pPattern));
+                        ScPatternAttr* pAltPattern = &aAltPatterns.back();
+                        SfxBoolItem aLineBreak(ATTR_LINEBREAK, false);
+                        pAltPattern->GetItemSet().Put(aLineBreak);
+                        pPattern = pAltPattern;
+                    }
+
                     BYTE nScript = GetScriptType( pDoc, pCell, pPattern, pCondSet );
                     if (nScript == 0) nScript = ScGlobal::GetDefaultScriptType();
                     if ( pPattern != pOldPattern || pCondSet != pOldCondSet ||
