From 6ebde10a24395303d885ba73c2802b3375f6ca91 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:08:22 +0200
Subject: [PATCH 715/768] calc-pdf-export-allow-filtered-range-filter.diff

---
 filter/source/pdf/impdialog.cxx |   24 ++++++++++++++++++++++++
 1 files changed, 24 insertions(+), 0 deletions(-)

diff --git filter/source/pdf/impdialog.cxx filter/source/pdf/impdialog.cxx
index 19e86cb..5de50bd 100644
--- filter/source/pdf/impdialog.cxx
+++ filter/source/pdf/impdialog.cxx
@@ -131,7 +131,31 @@ ImpPDFTabDialog::ImpPDFTabDialog( Window* pParent,
         {
             Reference< view::XSelectionSupplier > xView( xController, UNO_QUERY );
             if( xView.is() )
+            {
+                // In Calc, we need to treat a selection across hidden or
+                // filtered ranges as one range, or else the export will
+                // fail because Calc doesn't support printing of multiple
+                // selections.
+                Reference< beans::XPropertySet > xPropSet(xView, UNO_QUERY);
+                if (xPropSet.is())
+                {
+                    sal_Bool bFilteredRange = sal_False; // Don't divide selection across filtered ranges.
+                    uno::Any any;
+                    any <<= bFilteredRange;
+                    try
+                    {
+                        xPropSet->setPropertyValue(
+                            rtl::OUString::createFromAscii("FilteredRangeSelection"), any);
+                    }
+                    catch (const beans::UnknownPropertyException&)
+                    {
+                        // For now, apps other than Calc fails to set this property,
+                        // hence an exception gets thrown.
+                    }
+                }
+
                 xView->getSelection() >>= maSelection;
+            }
         }
     }
     catch( RuntimeException )
-- 
1.7.0.1

