diff --git sc/source/core/data/drwlayer.cxx sc/source/core/data/drwlayer.cxx
index 03c48f5..a0b08fd 100644
--- sc/source/core/data/drwlayer.cxx
+++ sc/source/core/data/drwlayer.cxx
@@ -547,26 +547,6 @@ void ScDrawLayer::RecalcPos( SdrObject* pObj, const ScDrawObjData& rData, bool b
     if( !pDoc )
         return;
 
-    /*  TODO CleanUp: Updating note position works just by chance currently...
-        When inserting rows/columns, this function is called after the
-        insertion, and the note is located at the new position contained in the
-        passed ScDrawObjData already. But when deleting rows/columns, this
-        function is called *before* the deletion, so the note is still at the
-        old cell position, and ScDocument::GetNote() will fail to get the note
-        or will get another note. But after the rows/columns are deleted, a
-        call to ScDrawLayer::SetPageSize() will call this function again, and
-        now the note is at the expected position in the document. */
-    if( rData.mbNote )
-    {
-        DBG_ASSERT( rData.maStart.IsValid(), "ScDrawLayer::RecalcPos - invalid position for cell note" );
-        /*  When inside an undo action, there may be pending note captions
-            where cell note is already deleted. The caption will be deleted
-            later with drawing undo. */
-        if( ScPostIt* pNote = pDoc->GetNote( rData.maStart ) )
-            pNote->UpdateCaptionPos( rData.maStart );
-        return;
-    }
-
     bool bValid1 = rData.maStart.IsValid();
     SCCOL nCol1 = rData.maStart.Col();
     SCROW nRow1 = rData.maStart.Row();
