diff --git officecfg/registry/schema/org/openoffice/Office/Calc.xcs officecfg/registry/schema/org/openoffice/Office/Calc.xcs
index 742dc7a..a6cccce 100644
--- officecfg/registry/schema/org/openoffice/Office/Calc.xcs
+++ officecfg/registry/schema/org/openoffice/Office/Calc.xcs
@@ -1209,7 +1209,7 @@
 						<desc>Specifies the number of decimals to be displayed for the Standard number format.</desc>
 						<label>Decimal places</label>
 					</info>
-					<value>2</value>
+					<value>-1</value>
 				</prop>
 				<prop oor:name="RegularExpressions" oor:type="xs:boolean">
 					<!-- UIHints: Tools - Options  Spreadsheet  Calculate  [Check box] Enable regular expressions in formulas -->
