--- sc/source/filter/excel/xistyle.cxx.old	2009-10-19 13:27:16.000000000 +0200
+++ sc/source/filter/excel/xistyle.cxx	2010-03-03 16:59:16.000000000 +0100
@@ -704,7 +704,6 @@ void XclImpCellAlign::FillToItemSet( Sfx
     sal_uInt8 nXclRot = (mnOrient == EXC_ORIENT_NONE) ? mnRotation : XclTools::GetXclRotFromOrient( mnOrient );
     bool bStacked = (nXclRot == EXC_ROT_STACKED);
     ScfTools::PutItem( rItemSet, SfxBoolItem( ATTR_STACKED, bStacked ), bSkipPoolDefs );
-    ScfTools::PutItem( rItemSet, SvxRotateModeItem( SVX_ROTATE_MODE_STANDARD, ATTR_ROTATE_MODE ), bSkipPoolDefs );
     // set an angle in the range from -90 to 90 degrees
     sal_Int32 nAngle = XclTools::GetScRotation( nXclRot, 0 );
     ScfTools::PutItem( rItemSet, SfxInt32Item( ATTR_ROTATE_VALUE, nAngle ), bSkipPoolDefs );
@@ -802,6 +801,15 @@ void XclImpCellBorder::FillFromCF8( sal_
     mbDiagUsed    = false;
 }
 
+bool XclImpCellBorder::HasAnyOuterBorder() const
+{
+    return
+        (mbLeftUsed   && (mnLeftLine != EXC_LINE_NONE)) ||
+        (mbRightUsed  && (mnRightLine != EXC_LINE_NONE)) ||
+        (mbTopUsed    && (mnTopLine != EXC_LINE_NONE)) ||
+        (mbBottomUsed && (mnBottomLine != EXC_LINE_NONE));
+}
+
 namespace {
 
 /** Converts the passed line style to a SvxBorderLine, or returns false, if style is "no line". */
@@ -1094,12 +1102,13 @@ const ScPatternAttr& XclImpXF::CreatePat
     // create new pattern attribute set
     mpPattern.reset( new ScPatternAttr( GetDoc().GetPool() ) );
     SfxItemSet& rItemSet = mpPattern->GetItemSet();
+    XclImpXF* pParentXF = IsCellXF() ? GetXFBuffer().GetXF( mnParent ) : 0;
 
     // parent cell style
     if( IsCellXF() && !mpStyleSheet )
     {
         mpStyleSheet = GetXFBuffer().CreateStyleSheet( mnParent );
-        if( XclImpXF* pParentXF = GetXFBuffer().GetXF( mnParent ) )
+        if (pParentXF)
             UpdateUsedFlags( *pParentXF );
     }
 
@@ -1140,6 +1149,20 @@ const ScPatternAttr& XclImpXF::CreatePat
             maArea.mnPattern != EXC_PATT_SOLID);
     }
 
+    /*  #i38709# Decide which rotation reference mode to use. If any outer
+        border line of the cell is set (either explicitly or via cell style),
+        and the cell contents are rotated, set rotation reference to bottom of
+        cell. This causes the borders to be painted rotated with the text. */
+    if( mbAlignUsed || mbBorderUsed )
+    {
+        SvxRotateMode eRotateMode = SVX_ROTATE_MODE_STANDARD;
+        const XclImpCellAlign* pAlign = mbAlignUsed ? &maAlignment : (pParentXF ? &pParentXF->maAlignment : 0);
+        const XclImpCellBorder* pBorder = mbBorderUsed ? &maBorder : (pParentXF ? &pParentXF->maBorder : 0);
+        if( pAlign && pBorder && (0 < pAlign->mnRotation) && (pAlign->mnRotation <= 180) && pBorder->HasAnyOuterBorder() )
+            eRotateMode = SVX_ROTATE_MODE_BOTTOM;
+        ScfTools::PutItem( rItemSet, SvxRotateModeItem( eRotateMode, ATTR_ROTATE_MODE ), bSkipPoolDefs );
+    }
+
     return *mpPattern;
 }
 
--- sc/source/filter/inc/xistyle.hxx.old	2009-10-19 13:27:16.000000000 +0200
+++ sc/source/filter/inc/xistyle.hxx	2010-03-03 16:59:16.000000000 +0100
@@ -319,6 +319,9 @@ struct XclImpCellBorder : public XclCell
     /** Fills this struct with BIFF8 CF (conditional format) record data. */
     void                FillFromCF8( sal_uInt16 nLineStyle, sal_uInt32 nLineColor, sal_uInt32 nFlags );
 
+    /** Returns true, if any of the outer border lines is visible. */
+    bool                HasAnyOuterBorder() const;
+
     /** Inserts a box item representing this border style into the item set.
         @param bSkipPoolDefs  true = Do not put items equal to pool default; false = Put all items. */
     void                FillToItemSet(
