---
 svtools/source/control/tabbar.cxx |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git svtools/source/control/tabbar.cxx svtools/source/control/tabbar.cxx
index d1578d9..79a29ff 100644
--- svtools/source/control/tabbar.cxx
+++ svtools/source/control/tabbar.cxx
@@ -1385,6 +1385,10 @@ void TabBar::Paint( const Rectangle& )
             else
                 SetTextColor( aFaceTextColor );
 
+            // This tab is "special", and a special tab needs a blue text.
+            if (bSpecialTab)
+                SetTextColor(Color(COL_LIGHTBLUE));
+
             aDrawer.drawText(aText);
 
             if ( bCurrent )
-- 
1.7.0.1

