diff --git sc/source/core/data/drwlayer.cxx sc/source/core/data/drwlayer.cxx
index ea8373e..bda0eda 100644
--- sc/source/core/data/drwlayer.cxx
+++ sc/source/core/data/drwlayer.cxx
@@ -547,17 +547,6 @@ void ScDrawLayer::RecalcPos( SdrObject* pObj, const ScDrawObjData& rData,
     if( !pDoc )
         return;
 
-    if( rData.mbNote )
-    {
-        DBG_ASSERT( rOldStart.IsValid(), "ScDrawLayer::RecalcPos - invalid position for cell note" );
-        /*  When inside an undo action, there may be pending note captions
-            where cell note is already deleted. The caption will be deleted
-            later with drawing undo. */
-        if( ScPostIt* pNote = pDoc->GetNote( rData.maStart ) )
-            pNote->UpdateCaptionPos( rData.maStart );
-        return;
-    }
-
     bool bValid1 = rData.maStart.IsValid();
     SCCOL nCol1 = rData.maStart.Col();
     SCROW nRow1 = rData.maStart.Row();
