diff --git offapi/com/sun/star/awt/Key.idl offapi/com/sun/star/awt/Key.idl
index aa72c21..b8a2c7a 100644
--- offapi/com/sun/star/awt/Key.idl
+++ offapi/com/sun/star/awt/Key.idl
@@ -481,6 +481,10 @@ published constants Key
 
     const short BRACKETRIGHT = 1316;
 
+    //------------------------------------------------------------------------- 
+
+    const short SEMICOLON = 1317;
+
     //-------------------------------------------------------------------------
     
     const short DELETE_TO_END_OF_PARAGRAPH = 1539;
