diff --git sc/source/filter/excel/xilink.cxx sc/source/filter/excel/xilink.cxx
index 9f730b2..76204a6 100644
--- sc/source/filter/excel/xilink.cxx
+++ sc/source/filter/excel/xilink.cxx
@@ -418,7 +418,7 @@ void XclImpSupbookTab::LoadCachedValues(ScExternalRefCache::TableTypeRef pCacheT
             {
                 double f = p->GetValue();
                 ScExternalRefCache::TokenRef pToken(new formula::FormulaDoubleToken(f));
-                pCacheTable->setCell(rAddr.mnCol, rAddr.mnRow, pToken);
+                pCacheTable->setCell(rAddr.mnCol, rAddr.mnRow, pToken, 0, false);
             }
             break;
             case EXC_CACHEDVAL_EMPTY:
@@ -429,7 +429,7 @@ void XclImpSupbookTab::LoadCachedValues(ScExternalRefCache::TableTypeRef pCacheT
             {
                 const String& rStr = p->GetString();
                 ScExternalRefCache::TokenRef pToken(new formula::FormulaStringToken(rStr));
-                pCacheTable->setCell(rAddr.mnCol, rAddr.mnRow, pToken);
+                pCacheTable->setCell(rAddr.mnCol, rAddr.mnRow, pToken, 0, false);
             }
             break;
             default:
