diff --git sc/source/core/data/table2.cxx sc/source/core/data/table2.cxx
index 3c35beb..f3aaf40 100644
--- sc/source/core/data/table2.cxx
+++ sc/source/core/data/table2.cxx
@@ -2380,8 +2380,6 @@ void ScTable::DBShowRow(SCROW nRow, BOOL bShow)
 void ScTable::DBShowRows(SCROW nRow1, SCROW nRow2, BOOL bShow)
 {
     SCROW nStartRow = nRow1;
-    IncRecalcLevel();
-    InitializeNoteCaptions();
     while (nStartRow <= nRow2)
     {
         BYTE nOldFlag = pRowFlags->GetValue(nStartRow) & CR_HIDDEN;
@@ -2424,8 +2422,6 @@ void ScTable::DBShowRows(SCROW nRow1, SCROW nRow2, BOOL bShow)
     //	to be done here.
     if (pOutlineTable)
         UpdateOutlineRow( nRow1, nRow2, bShow );
-
-    DecRecalcLevel();
 }
 
 
diff --git sc/source/core/data/table3.cxx sc/source/core/data/table3.cxx
index bb7786b..28727d9 100644
--- sc/source/core/data/table3.cxx
+++ sc/source/core/data/table3.cxx
@@ -1623,6 +1623,7 @@ SCSIZE ScTable::Query(ScQueryParam& rParamOrg, BOOL bKeepSub)
                             aParam.nDestCol, aParam.nDestRow, aParam.nDestTab );
     }
 
+    InitializeNoteCaptions();
     for (SCROW j=aParam.nRow1 + nHeader; j<=nEndRow; j++)
     {
         BOOL bResult;									// Filterergebnis
@@ -1700,7 +1701,7 @@ SCSIZE ScTable::Query(ScQueryParam& rParamOrg, BOOL bKeepSub)
         DBShowRows(nOldStart,nOldEnd, bOldResult);
 
     delete[] pSpecial;
-
+    SetDrawPageSize();
     return nCount;
 }
 
