diff --git sc/source/ui/unoobj/docuno.cxx sc/source/ui/unoobj/docuno.cxx
index 492a3f5..bd7a775 100644
--- sc/source/ui/unoobj/docuno.cxx
+++ sc/source/ui/unoobj/docuno.cxx
@@ -53,6 +53,7 @@
 #include <ctype.h>
 #include <float.h>	// DBL_MAX
 
+#include <com/sun/star/task/ErrorCodeIOException.hpp>
 #include <com/sun/star/util/Date.hpp>
 #include <com/sun/star/sheet/XNamedRanges.hpp>
 #include <com/sun/star/sheet/XLabelRanges.hpp>
@@ -104,6 +105,8 @@
 #include <com/sun/star/document/VbaEventId.hpp>
 using namespace com::sun::star;
 using namespace com::sun::star::document::VbaEventId;
+using ::rtl::OUString;
+using ::com::sun::star::uno::Reference;
 
 #define SC_UNO_VBADOCOBJ "ThisVBADocObj" // perhaps we want to actually make this ThisWorkbook ?
 
@@ -707,7 +710,13 @@ sal_Int32 SAL_CALL ScModelObj::getRendererCount( const uno::Any& aSelection,
     ScMarkData aMark;
     ScPrintSelectionStatus aStatus;
     if ( !FillRenderMarkData( aSelection, aMark, aStatus ) )
-        return 0;
+    {
+        // TODO: Right now this error code is not reaching where it's supposed
+        // to.  Find out why and figure out how to send correct error code to
+        // the framework code.
+        throw task::ErrorCodeIOException(
+            OUString(), Reference<XInterface>(), ERRCODE_IO_NOTSUPPORTED);
+    }
 
     //	The same ScPrintFuncCache object in pPrintFuncCache is used as long as
     //	the same selection is used (aStatus) and the document isn't changed
