diff --git sc/source/ui/view/gridwin.cxx sc/source/ui/view/gridwin.cxx
index 1079b9d..b565db1 100644
--- sc/source/ui/view/gridwin.cxx
+++ sc/source/ui/view/gridwin.cxx
@@ -3064,7 +3064,7 @@ void __EXPORT ScGridWindow::KeyInput(const KeyEvent& rKEvt)
             return;
         }
     }
-	else if( rKeyCode.GetCode() == KEY_RETURN && pViewData->IsPasteMode() )
+    else if( !SC_MOD()->IsInputMode() && rKeyCode.GetCode() == KEY_RETURN && pViewData->IsPasteMode() )
 	{
 		ScTabViewShell* pTabViewShell = pViewData->GetViewShell();
 
