diff --git oox/inc/oox/xls/stylesbuffer.hxx oox/inc/oox/xls/stylesbuffer.hxx
index e9d17db..4f288f5 100644
--- oox/inc/oox/xls/stylesbuffer.hxx
+++ oox/inc/oox/xls/stylesbuffer.hxx
@@ -552,6 +552,8 @@ public:
     /** Writes all border attributes to the passed property map. */
     void                writeToPropertyMap( PropertyMap& rPropMap ) const;
 
+    bool                hasBorder() const;
+
 private:
     /** Returns the border line struct specified by the passed XML token identifier. */
     BorderLineModel*    getBorderLine( sal_Int32 nElement );
@@ -1009,6 +1011,7 @@ public:
     /** Writes the cell formatting attributes of the specified style XF to the passed property set. */
     void                writeStyleXfToPropertySet( PropertySet& rPropSet, sal_Int32 nXfId ) const;
 
+    bool                hasBorder( sal_Int32 nBorderId ) const;
 private:
     typedef RefVector< Font >                           FontVector;
     typedef RefVector< Border >                         BorderVector; 
diff --git oox/source/xls/stylesbuffer.cxx oox/source/xls/stylesbuffer.cxx
index 5627c78..00411bb 100644
--- oox/source/xls/stylesbuffer.cxx
+++ oox/source/xls/stylesbuffer.cxx
@@ -40,6 +40,7 @@
 #include <com/sun/star/style/XStyle.hpp>
 #include <com/sun/star/text/WritingMode2.hpp>
 #include <com/sun/star/text/XText.hpp>
+#include <com/sun/star/table/CellVertJustify.hpp>
 #include <rtl/tencinfo.h>
 #include <rtl/ustrbuf.hxx>
 #include "oox/helper/attributelist.hxx"
@@ -1699,7 +1700,35 @@ void Border::writeToPropertyMap( PropertyMap& rPropMap ) const
         rPropMap[ PROP_DiagonalBLTR ] <<= maApiData.maBLtoTR;
     }
 }
 
+namespace {
+
+bool lcl_isBorder(const ::com::sun::star::table::BorderLine& rBorder)
+{
+    return (rBorder.InnerLineWidth > 0) || (rBorder.OuterLineWidth > 0);
+}
+
+}
+
+bool Border::hasBorder() const
+{
+    const ApiBorderData::ApiTableBorder& rTabBorder = maApiData.maBorder;
+
+    if (rTabBorder.IsBottomLineValid && lcl_isBorder(rTabBorder.BottomLine))
+        return true;
+
+    if (rTabBorder.IsTopLineValid && lcl_isBorder(rTabBorder.TopLine))
+        return true;
+
+    if (rTabBorder.IsLeftLineValid && lcl_isBorder(rTabBorder.LeftLine))
+        return true;
+
+    if (rTabBorder.IsRightLineValid && lcl_isBorder(rTabBorder.RightLine))
+        return true;
+
+    return false;
+}
+
 BorderLineModel* Border::getBorderLine( sal_Int32 nElement )
 {
     switch( nElement )
@@ -2325,6 +2354,11 @@ void Xf::writeToPropertyMap( PropertyMap& rPropMap ) const
         rStyles.writeBorderToPropertyMap( rPropMap, maModel.mnBorderId );
     if( maModel.mbAreaUsed )
         rStyles.writeFillToPropertyMap( rPropMap, maModel.mnFillId );
+
+    ::com::sun::star::table::CellVertJustify eRotRef = ::com::sun::star::table::CellVertJustify_STANDARD;
+    if (maModel.mbBorderUsed && rStyles.hasBorder(maModel.mnBorderId) && maAlignment.getApiData().mnRotation)
+        eRotRef = ::com::sun::star::table::CellVertJustify_BOTTOM;
+    rPropMap[ PROP_RotateReference ] <<= eRotRef;
 }
 
 void Xf::writeToPropertySet( PropertySet& rPropSet ) const
@@ -3150,6 +3184,12 @@ void StylesBuffer::writeCellXfToPropertySet( PropertySet& rPropSet, sal_Int32 nX
         pXf->writeToPropertySet( rPropSet );
 }
 
+bool StylesBuffer::hasBorder( sal_Int32 nBorderId ) const
+{
+    Border* pBorder = maBorders.get( nBorderId ).get();
+    return pBorder && pBorder->hasBorder();
+}
+
 void StylesBuffer::writeStyleXfToPropertySet( PropertySet& rPropSet, sal_Int32 nXfId ) const
 {
     if( Xf* pXf = maStyleXfs.get( nXfId ).get() )
