diff --git sc/source/ui/view/output2.cxx sc/source/ui/view/output2.cxx
index 301a973..e4a0208 100644
--- sc/source/ui/view/output2.cxx
+++ sc/source/ui/view/output2.cxx
@@ -178,6 +178,7 @@ private:
     long        GetSignWidth();
     long        GetDotWidth();
     void        TextChanged();
+    long        ConvertWidthLogicToPixel( long nWidth ) const;
 };
 
 //==================================================================
@@ -598,7 +599,12 @@ void ScDrawStringsVars::SetTextToWidthOrHash( ScBaseCell* pCell, long nWidth )
             return;
     }
 
-    if (pOutput->pFmtDevice->GetTextWidth(aString) > nWidth)
+    long nActualTextWidth = pOutput->pFmtDevice->GetTextWidth(aString);
+
+    if (bPixelToLogic)
+        nActualTextWidth = ConvertWidthLogicToPixel(nActualTextWidth);
+
+    if (nActualTextWidth > nWidth)
     {
         // Even after the decimal adjustment the text doesn't fit.  Give up.
         SetHashText();
@@ -650,6 +656,9 @@ long ScDrawStringsVars::GetMaxDigitWidth()
         long n = pOutput->pFmtDevice->GetTextWidth(String(cDigit));
         nMaxDigitWidth = ::std::max(nMaxDigitWidth, n);
     }
+
+    if (bPixelToLogic)
+        nMaxDigitWidth = ConvertWidthLogicToPixel(nMaxDigitWidth);
     return nMaxDigitWidth;
 }
 
@@ -659,6 +668,8 @@ long ScDrawStringsVars::GetSignWidth()
         return nSignWidth;
 
     nSignWidth = pOutput->pFmtDevice->GetTextWidth(String('-'));
+    if (bPixelToLogic)
+        nSignWidth = ConvertWidthLogicToPixel(nSignWidth);
     return nSignWidth;
 }
 
@@ -669,6 +680,8 @@ long ScDrawStringsVars::GetDotWidth()
 
     const ::rtl::OUString& sep = ScGlobal::GetpLocaleData()->getLocaleItem().decimalSeparator;
     nDotWidth = pOutput->pFmtDevice->GetTextWidth(sep);
+    if (bPixelToLogic)
+        nDotWidth = ConvertWidthLogicToPixel(nDotWidth);
     return nDotWidth;
 }
 
@@ -698,6 +711,13 @@ void ScDrawStringsVars::TextChanged()
         aTextSize = pRefDevice->LogicToPixel( aTextSize );
 }
 
+long ScDrawStringsVars::ConvertWidthLogicToPixel( long nWidth ) const
+{
+    Size aSize(nWidth, pOutput->pFmtDevice->GetTextHeight());
+    aSize = pOutput->pRefDevice->LogicToPixel(aSize);
+    return aSize.Width();
+}
+
 BOOL ScDrawStringsVars::HasEditCharacters() const
 {
     static const sal_Unicode pChars[] =
