diff --git sc/source/ui/view/output2.cxx sc/source/ui/view/output2.cxx
index 0d8d3f6..083576b 100644
--- sc/source/ui/view/output2.cxx
+++ sc/source/ui/view/output2.cxx
@@ -2156,12 +2156,13 @@ void EditAlignmentParam::calcPaperSize(
         rPaperSize.Height() = rAlignRect.GetHeight() - nTopM - nBottomM;
     }
 
-    if (mbAsianVertical && mbBreak)
+    if (mbAsianVertical)
     {
-        //	add some extra height (default margin value) for safety
-        //	as long as GetEditArea isn't used below
-        long nExtraHeight = static_cast<long>(20*nPPTY);
-        rPaperSize.Height() = rAlignRect.GetHeight() - nTopM - nBottomM + nExtraHeight;
+        rPaperSize.Height() = rAlignRect.GetHeight() - nTopM - nBottomM;
+        // Subtract some extra value from the height or else the text would go
+        // outside the cell area.  The value of 5 is arbitrary, and is based
+        // entirely on heuristics.
+        rPaperSize.Height() -= 5;
     }
 }
 
@@ -2944,8 +2945,8 @@ void ScOutputData::DrawEdit(BOOL bPixelToLogic)
                                         aLogicStart.Y() += nTopM;
                                 }
 
-                                if ( ( aAlignParam.meOrient==SVX_ORIENTATION_STANDARD && !aAlignParam.mbAsianVertical ) ||
-                                     aAlignParam.meOrient==SVX_ORIENTATION_STACKED || !aAlignParam.mbBreak )
+                                if (!aAlignParam.mbAsianVertical && 
+                                    (aAlignParam.meOrient == SVX_ORIENTATION_STANDARD || aAlignParam.meOrient == SVX_ORIENTATION_STACKED || !aAlignParam.mbBreak))
                                 {
                                     if (aAlignParam.meVerJust==SVX_VER_JUSTIFY_BOTTOM ||
                                         aAlignParam.meVerJust==SVX_VER_JUSTIFY_STANDARD)
