diff --git svl/inc/svl/svtools.hrc svl/inc/svl/svtools.hrc
index 3a64a3a..fddc670 100644
--- svl/inc/svl/svtools.hrc
+++ svl/inc/svl/svtools.hrc
@@ -427,6 +427,7 @@
 #define BMP_HELP_AGENT_IMAGE			(RID_SVTOOLS_BITMAP_START +   1)
 #define BMP_HELP_AGENT_CLOSER			(RID_SVTOOLS_BITMAP_START +   2)
 #define BMP_PLUGIN                      (RID_SVTOOLS_BITMAP_START +   3)
+#define BMP_LIST_ADD                    (RID_SVTOOLS_BITMAP_START +   4)
 
 //.............................................................................
 // image lists
