diff --git sc/source/ui/docshell/externalrefmgr.cxx sc/source/ui/docshell/externalrefmgr.cxx
index b831e93..749695b 100644
--- sc/source/ui/docshell/externalrefmgr.cxx
+++ sc/source/ui/docshell/externalrefmgr.cxx
@@ -991,6 +991,7 @@ void ScExternalRefLink::DataChanged(const String& /*rMimeType*/, const Any& /*rV
         // The source document has changed.
         pMgr->switchSrcFile(mnFileId, aFile, aFilter);
         maFilterName = aFilter;
+        mpDoc->GetDocumentShell()->SetModified();
     }
 }
 
