diff --git vcl/inc/vcl/keycodes.hxx vcl/inc/vcl/keycodes.hxx
index 8721c21..bd4f3d8 100644
--- vcl/inc/vcl/keycodes.hxx
+++ vcl/inc/vcl/keycodes.hxx
@@ -160,6 +160,7 @@
 #define KEY_QUOTELEFT   ((USHORT)::com::sun::star::awt::Key::QUOTELEFT)
 #define KEY_BRACKETLEFT ((USHORT)::com::sun::star::awt::Key::BRACKETLEFT)
 #define KEY_BRACKETRIGHT ((USHORT)::com::sun::star::awt::Key::BRACKETRIGHT)
+#define KEY_SEMICOLON   ((USHORT)::com::sun::star::awt::Key::SEMICOLON)
 
 #define KEY_CAPSLOCK    ((USHORT)::com::sun::star::awt::Key::CAPSLOCK)
 #define KEY_NUMLOCK     ((USHORT)::com::sun::star::awt::Key::NUMLOCK)
diff --git vcl/unx/gtk/window/gtkframe.cxx vcl/unx/gtk/window/gtkframe.cxx
index d4c9950..8f6ea8d 100644
--- vcl/unx/gtk/window/gtkframe.cxx
+++ vcl/unx/gtk/window/gtkframe.cxx
@@ -216,6 +216,7 @@ static USHORT GetKeyCode( guint keyval )
             case GDK_quoteleft:	nCode = KEY_QUOTELEFT;		break;           
             case GDK_bracketleft:  nCode = KEY_BRACKETLEFT;  break;
             case GDK_bracketright: nCode = KEY_BRACKETRIGHT; break;
+            case GDK_semicolon:    nCode = KEY_SEMICOLON;   break;
             // some special cases, also see saldisp.cxx
             // - - - - - - - - - - - - -  Apollo - - - - - - - - - - - - - 0x1000
             case 0x1000FF02: // apXK_Copy
diff --git vcl/unx/source/app/saldisp.cxx vcl/unx/source/app/saldisp.cxx
index 1b51888..8ef3762 100644
--- vcl/unx/source/app/saldisp.cxx
+++ vcl/unx/source/app/saldisp.cxx
@@ -1308,6 +1308,9 @@ XubString SalDisplay::GetKeyName( USHORT nKeyCode ) const
         case KEY_BRACKETRIGHT:
             nKeySym = XK_bracketright;
             break;
+        case KEY_SEMICOLON:
+            nKeySym = XK_semicolon;
+            break;
 
         default:
             nKeySym = 0;
@@ -1646,6 +1649,10 @@ USHORT SalDisplay::GetKeyCode( KeySym keysym, char*pcPrintable ) const
              nKey = KEY_BRACKETRIGHT;
              *pcPrintable = ']';
              break;
+        case XK_semicolon:
+            nKey = KEY_SEMICOLON;
+            *pcPrintable = ';';
+            break;
 //      case XK_Linefeed:
 //          *pcPrintable = '\n';
 //          break;
diff --git vcl/win/source/app/salinst.cxx vcl/win/source/app/salinst.cxx
index 1e085b8..7d38161 100644
--- vcl/win/source/app/salinst.cxx
+++ vcl/win/source/app/salinst.cxx
@@ -380,6 +380,7 @@ void SalData::initKeyCodeMap()
     initKey( '`', KEY_QUOTELEFT );
     initKey( '[', KEY_BRACKETLEFT );
     initKey( ']', KEY_BRACKETRIGHT );
+    initKey( ';', KEY_SEMICOLON );
 }
 
 // =======================================================================
