--- canvas/source/cairo/makefile.mk.old	2009-09-10 19:49:19.000000000 +0200
+++ canvas/source/cairo/makefile.mk	2009-10-05 21:13:42.000000000 +0200
@@ -91,7 +91,7 @@ SHL1STDLIBS= $(CPPULIB) $(TKLIB) $(SALLI
 .IF "$(SYSTEM_CAIRO)" == "YES"
 SHL1STDLIBS+= $(CAIRO_LIBS)
 .ELSE
-SHL1STDLIBS+= -lcairo
+SHL1STDLIBS+= -lcairo -lpixman-1
 .ENDIF
 
 .IF "$(GUIBASE)"=="aqua"
@@ -103,7 +103,7 @@ CFLAGSCXX+=$(OBJCXXFLAGS)
 
 # Xlib
 SLOFILES+= $(SLO)$/cairo_xlib_cairo.obj
-SHL1STDLIBS+= -lfontconfig -lX11 -lXrender -lpixman-1 $(FREETYPE_LIBS)
+SHL1STDLIBS+= -lfontconfig -lX11 -lXrender $(FREETYPE_LIBS)
 CFLAGS+=$(FREETYPE_CFLAGS)
 
 .ENDIF   # "$(GUIBASE)"=="aqua"
