From b9fdae567a92c5b71463c19dddb91509760db558 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:57:46 +0200
Subject: [PATCH 280/878] cancas-cairo-linkflags.diff

---
 canvas/source/cairo/makefile.mk |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/canvas/source/cairo/makefile.mk b/canvas/source/cairo/makefile.mk
index 45e1ca0..d4ac575 100644
--- a/canvas/source/cairo/makefile.mk
+++ b/canvas/source/cairo/makefile.mk
@@ -87,7 +87,7 @@ SHL1STDLIBS= $(CPPULIB) $(TKLIB) $(SALLIB) $(VCLLIB) $(COMPHELPERLIB) $(CPPUHELP
 .IF "$(SYSTEM_CAIRO)" == "YES"
 SHL1STDLIBS+= $(CAIRO_LIBS)
 .ELSE
-SHL1STDLIBS+= -lcairo
+SHL1STDLIBS+= -lcairo -lpixman-1
 .ENDIF
 
 .IF "$(GUIBASE)"=="aqua"
@@ -98,7 +98,7 @@ CFLAGSCXX+=$(OBJCXXFLAGS)
 
 # Xlib
 SLOFILES+= $(SLO)$/cairo_xlib_cairo.obj
-SHL1STDLIBS+= -lfontconfig -lX11 -lXrender -lpixman-1 $(FREETYPE_LIBS)
+SHL1STDLIBS+= -lfontconfig -lX11 -lXrender $(FREETYPE_LIBS)
 CFLAGS+=$(FREETYPE_CFLAGS)
 
 .ENDIF   # "$(GUIBASE)"=="aqua"
-- 
1.7.0.1

