--- canvas/source/cairo/cairo_canvashelper.cxx.old1	2010-08-30 13:45:29.000000000 +0200
+++ canvas/source/cairo/cairo_canvashelper.cxx	2010-08-30 13:45:39.000000000 +0200
@@ -1265,7 +1265,6 @@ namespace cairocanvas
                 ::rtl::math::approxEqual( aMatrix.x0, 0 ) &&
                 ::rtl::math::approxEqual( aMatrix.y0, 0 ) )
                  cairo_set_operator( mpCairo.get(), CAIRO_OPERATOR_SOURCE );
-            cairo_pattern_set_extend( cairo_get_source(mpCairo.get()), CAIRO_EXTEND_PAD );
             cairo_rectangle( mpCairo.get(), 0, 0, aBitmapSize.Width, aBitmapSize.Height );
             cairo_clip( mpCairo.get() );
             cairo_paint( mpCairo.get() );
