--- canvas/source/cairo/cairo_canvashelper.cxx.old	2010-08-19 18:28:30.000000000 +0200
+++ canvas/source/cairo/cairo_canvashelper.cxx	2010-08-19 18:17:36.000000000 +0200
@@ -1265,6 +1265,9 @@ namespace cairocanvas
                 ::rtl::math::approxEqual( aMatrix.x0, 0 ) &&
                 ::rtl::math::approxEqual( aMatrix.y0, 0 ) )
                  cairo_set_operator( mpCairo.get(), CAIRO_OPERATOR_SOURCE );
+#ifdef CAIRO_EXTEND_PAD
+            cairo_pattern_set_extend( cairo_get_source(mpCairo.get()), CAIRO_EXTEND_PAD );
+#endif
             cairo_rectangle( mpCairo.get(), 0, 0, aBitmapSize.Width, aBitmapSize.Height );
             cairo_clip( mpCairo.get() );
             cairo_paint( mpCairo.get() );
