--- officecfg/registry/data/org/openoffice/Office/Writer.xcu.orig	2010-02-12 22:41:16.000000000 +0100
+++ officecfg/registry/data/org/openoffice/Office/Writer.xcu	2010-03-18 11:51:00.908375000 +0100
@@ -56,6 +56,9 @@
   </node>
   <node oor:name="Insert">
     <node oor:name="Caption">
+      <prop oor:name="CaptionOrderNumberingFirst">
+         <value xml:lang="hu">true</value>
+      </prop>
       <node oor:name="WriterObject">
         <node oor:name="Table">
           <node oor:name="Settings">
