--- chart2/source/controller/dialogs/res_LegendPosition_tmpl.hrc.orig	2010-02-12 22:25:30.000000000 +0100
+++ chart2/source/controller/dialogs/res_LegendPosition_tmpl.hrc	2010-04-08 15:16:32.964875000 +0200
@@ -38,7 +38,7 @@
 { \
     HelpID = HID_SCH_LEGEND_SHOW ; \
     Pos = MAP_APPFONT ( xpos , ypos  ) ; \
-    Size = MAP_APPFONT ( 67 , 12 ) ; \
+    Size = MAP_APPFONT ( 83 , 12 ) ; \
     Text [ en-US ] = "~Display legend" ; \
     TabStop = TRUE ; \
 };
