diff --git sc/source/ui/view/tabview3.cxx sc/source/ui/view/tabview3.cxx
index 80e0fd7..651723f 100644
--- sc/source/ui/view/tabview3.cxx
+++ sc/source/ui/view/tabview3.cxx
@@ -80,6 +80,7 @@
 #include "rangeutl.hxx"
 #include "client.hxx"
 #include "tabprotection.hxx"
+#include "formula/FormulaCompiler.hxx"
 
 #include <com/sun/star/chart2/data/HighlightedRange.hpp>
 #include <com/sun/star/document/XVbaEventsHelper.hpp>
@@ -2178,6 +2179,7 @@ void ScTabView::DoChartSelection(
     const uno::Sequence< chart2::data::HighlightedRange > & rHilightRanges )
 {
     ClearHighlightRanges();
+    const sal_Unicode sep = ::formula::FormulaCompiler::GetNativeSymbol(ocSep).GetChar(0);
 
     for( sal_Int32 i=0; i<rHilightRanges.getLength(); ++i )
     {
@@ -2185,7 +2187,7 @@ void ScTabView::DoChartSelection(
         ScRangeList aRangeList;
         ScDocument* pDoc = aViewData.GetDocShell()->GetDocument();
         if( ScRangeStringConverter::GetRangeListFromString(
-                aRangeList, rHilightRanges[i].RangeRepresentation, pDoc, pDoc->GetAddressConvention(), ';' ))
+                aRangeList, rHilightRanges[i].RangeRepresentation, pDoc, pDoc->GetAddressConvention(), sep ))
         {
             for ( ScRangePtr p = aRangeList.First(); p; p = aRangeList.Next())
             {
