diff --git sc/source/ui/unoobj/chart2uno.cxx sc/source/ui/unoobj/chart2uno.cxx
index 62597a1..a57a500 100644
--- sc/source/ui/unoobj/chart2uno.cxx
+++ sc/source/ui/unoobj/chart2uno.cxx
@@ -44,6 +44,7 @@
 #include "hints.hxx"
 #include "unoreflist.hxx"
 #include "chartlis.hxx"
+#include "compiler.hxx"
 
 #include <sfx2/objsh.hxx>
 
@@ -1042,7 +1043,8 @@ rtl::OUString SAL_CALL ScChart2DataProvider::convertRangeToXML( const rtl::OUStr
     if( m_pDocument )
     {
         ScRangeList aList;
-        if( !ScRangeStringConverter::GetRangeListFromString(aList, sRangeRepresentation, m_pDocument, m_pDocument->GetAddressConvention(), ';'))
+        sal_Unicode sep = ScCompiler::GetNativeSymbol(ocSep).GetChar(0);
+        if( !ScRangeStringConverter::GetRangeListFromString(aList, sRangeRepresentation, m_pDocument, m_pDocument->GetAddressConvention(), sep))
             throw lang::IllegalArgumentException();
         ScRangeStringConverter::GetStringFromRangeList(sRet, &aList, m_pDocument, ScAddress::CONV_OOO);
     }
