--- configure.in.old	2010-03-03 17:00:03.000000000 +0100
+++ configure.in	2010-03-03 17:00:12.000000000 +0100
@@ -5257,6 +5257,8 @@ if test -n "$with_system_lpsolve" -o -n
     SYSTEM_LPSOLVE=YES
     AC_CHECK_HEADER(lpsolve/lp_lib.h, [],
        [ AC_MSG_ERROR(lpsolve headers not found.)], [])
+    AC_CHECK_LIB(m, floor)
+    AC_CHECK_LIB(dl, dlopen)
     AC_CHECK_LIB(lpsolve55, make_lp, ,
         [ AC_MSG_ERROR(lpsolve library not found or too old.)], [])
 else
