From 4dfdfda6cae26e5abc0d10e90368f5dfabae264c Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:07:13 +0200
Subject: [PATCH 661/768] configure-fix-lpsolve-depends.diff

---
 configure.in |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git configure.in configure.in
index c0e7d8d..0c4b188 100644
--- configure.in
+++ configure.in
@@ -5385,6 +5385,8 @@ if test -n "$with_system_lpsolve" -o -n "$with_system_libs" && \
     SYSTEM_LPSOLVE=YES
     AC_CHECK_HEADER(lpsolve/lp_lib.h, [],
        [ AC_MSG_ERROR(lpsolve headers not found.)], [])
+    AC_CHECK_LIB(m, floor)
+    AC_CHECK_LIB(dl, dlopen)
     AC_CHECK_LIB(lpsolve55, make_lp, ,
         [ AC_MSG_ERROR(lpsolve library not found or too old.)], [])
 else
-- 
1.7.0.1

