From 417c6999e81a415cf1be7f6ba765322a98c8b4ac Mon Sep 17 00:00:00 2001
From: Jan Nieuwenhuizen <janneke@gnu.org>
Date: Wed, 21 Oct 2009 08:32:16 +0200
Subject: [PATCH 3/3] configure.in: lpsolve may need libm, libdl, libcolamd.

---
 configure.in |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git configure.in configure.in
index 2c9cf6f..5aca145 100644
--- configure.in
+++ configure.in
@@ -4984,6 +4984,9 @@ AC_LANG_POP([C++])
 dnl ===================================================================
 dnl Checking for lpsolve
 dnl ===================================================================
+AC_CHECK_LIB(m, floor)
+AC_CHECK_LIB(dl, dlopen)
+AC_CHECK_LIB(colamd, colamd)
 AC_MSG_CHECKING([which lpsolve to use])
 if test -n "$with_system_lpsolve" -o -n "$with_system_libs" && \
 	test "$with_system_lpsolve" != "no"; then
-- 
1.6.3.3

