From a6a271c91cbb4ca74906edd168db5777b738ef50 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:06:24 +0200
Subject: [PATCH 710/878] configmgr-validate-removedtree.diff

---
 configmgr/source/treemgr/nodeimplobj.cxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/configmgr/source/treemgr/nodeimplobj.cxx b/configmgr/source/treemgr/nodeimplobj.cxx
index f0819f2..2d6cbc4 100644
--- a/configmgr/source/treemgr/nodeimplobj.cxx
+++ b/configmgr/source/treemgr/nodeimplobj.cxx
@@ -680,7 +680,7 @@ void DeferredSetNodeImpl::finishCommit(SubtreeChange& rChanges)
                     SetNodeImpl::removeElement(aElementName);
             }
             // handle a added or deleted node
-            if (aOriginal.isValid())
+            if (aOriginal.isValid() && aRemovedTree.is())
             {
                 OSL_ENSURE(aRemovedTree.is(), "Cannot take over the removed node");
             
-- 
1.7.0.1

