---
 configure.in |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git configure.in configure.in
index 8c2277a..15c0f6e 100644
--- configure.in
+++ configure.in
@@ -5387,6 +5387,8 @@ if test -n "$with_system_lpsolve" -o -n "$with_system_libs" && \
     SYSTEM_LPSOLVE=YES
     AC_CHECK_HEADER(lpsolve/lp_lib.h, [],
        [ AC_MSG_ERROR(lpsolve headers not found.)], [])
+    AC_CHECK_LIB(m, floor)
+    AC_CHECK_LIB(dl, dlopen)
     AC_CHECK_LIB(lpsolve55, make_lp, ,
         [ AC_MSG_ERROR(lpsolve library not found or too old.)], [])
 else
-- 
1.7.0.1

