--- connectivity/prj/d.lst
+++ connectivity/prj/d.lst
@@ -28,7 +28,7 @@
 ..\%__SRC%\inc\sqlbison.hxx %_DEST%\inc%_EXT%\connectivity\sqlbison.hxx
 ..\inc\connectivity\sdbcx\*.hxx %_DEST%\inc%_EXT%\connectivity\sdbcx\*.hxx
 #..\version.mk %_DEST%\inc%_EXT%\connectivity\version.mk
-..\%__SRC%\misc\merge\org\openoffice\Office\DataAccess\*.xcu %_DEST%\xml%_EXT%\registry\spool\DataAccess\*.xcu
+..\%__SRC%\misc\lang\*.xcu %_DEST%\xml%_EXT%\registry\spool\DataAccess\*.xcu
 ..\%COMMON_OUTDIR%\bin\fcfg_drivers_*.zip %_DEST%\pck%_EXT%\fcfg_drivers_*.zip
 ..\%__SRC%\bin\fcfg_drivers_*.zip %_DEST%\pck%_EXT%\fcfg_drivers_*.zip
 
--- connectivity/util/makefile.mk
+++ connectivity/util/makefile.mk
@@ -47,6 +47,7 @@
 DIR_LANG_SOURCE  := $(MISC)$/registry$/data
 .ENDIF
 DRIVER_MERGE_XCU := $(shell -@$(FIND) $(DIR_LANG_SOURCE)$/org$/openoffice$/Office$/DataAccess -name "*.xcu")
+DRIVER_MERGE_DEST = $(MISC)/lang/{$(DRIVER_MERGE_XCU:f)}
    
 REALFILTERPACKAGES_FILTERS_UI_LANGPACKS = \
     $(foreach,i,$(alllangiso) $(foreach,j,$(DRIVER_MERGE_XCU) $(DIR_LANGPACK)$/$i$/org$/openoffice$/Office$/DataAccess$/$(j:f)))
@@ -76,5 +77,26 @@
 
 ALLTAR: \
     $(MISC)$/$(TARGET)_delzip \
-    $(BIN)$/fcfg_drivers_{$(alllangiso)}.zip
+	$(BIN)$/fcfg_drivers_{$(alllangiso)}.zip \
+    $(MISC)/lastlang.mk
+
+.IF "$(DRIVER_MERGE_DEST)"!=""
+.INCLUDE .IGNORE : $(MISC)/lastlang.mk
+
+ALLTAR : \
+    $(DRIVER_MERGE_DEST)
+
+.IF "$(LAST_LANGS)"!="$(WITH_LANG)"
+DO_PHONY=.PHONY
+.ENDIF          # "$(LAST_LANG)"!="$(WITH_LANG)"
+
+$(MISC)/lang/%.xcu $(DO_PHONY) : $(DIR_LANG_SOURCE)$/org$/openoffice$/Office$/DataAccess/%.xcu
+    @@-$(MKDIRHIER) $(@:d:d)
+    $(COPY) $< $@
+
+.ENDIF          # "$(DRIVER_MERGE_DEST)"!=""
+
+$(MISC)/lastlang.mk $(DO_PHONY) :
+    $(RM) $@
+    echo LAST_LANGS=$(WITH_LANG) > $@
 
