--- connectivity/com/sun/star/sdbcx/comp/hsqldb/NativeLibraries.java.old	2010-02-02 12:49:41.000000000 +0100
+++ connectivity/com/sun/star/sdbcx/comp/hsqldb/NativeLibraries.java	2010-03-04 17:45:30.000000000 +0100
@@ -71,8 +71,13 @@ final class NativeLibraries {
                                      invoke(url, (java.lang.Object[])null) })).
                             getAbsolutePath());
                     } catch (Throwable t) {
-                        throw new UnsatisfiedLinkError(
-                            e.toString()+ " - " + t.toString());
+            // final fallback for Java 1.4.2
+            try {
+                            System.load(url.getPath());
+            } catch (Throwable t2) {
+                            throw new UnsatisfiedLinkError(
+                            e.toString()+ " - " + t2.toString());
+            }
                     }
                 }
             }
