--- svx/source/msfilter/msocximex.cxx.orig	2008-09-26 18:15:14.000000000 +0800
+++ svx/source/msfilter/msocximex.cxx	2008-09-26 18:18:50.000000000 +0800
@@ -1644,7 +1644,15 @@ sal_Bool OCX_OptionButton::Import(com::s
     aTmp <<= ::com::sun::star::style::VerticalAlignment_MIDDLE;
     rPropSet->setPropertyValue( WW8_ASCII2STR("VerticalAlign"), aTmp );
 
+	uno::Reference< frame::XModel > xModel ( pDocSh ? pDocSh->GetModel() : NULL );
+	lcl_ApplyListSourceAndBindableStuff( xModel, rPropSet, msCtrlSource, msRowSource );
 	aFontData.Import(rPropSet);
+
+    if ( sImageUrl.getLength() )
+    {
+        aTmp <<= sImageUrl;
+        rPropSet->setPropertyValue( WW8_ASCII2STR("ImageURL"), aTmp);
+    }
 	return sal_True;
 }
 
@@ -1897,12 +1905,6 @@ sal_Bool OCX_TextBox::Import(com::sun::s
 	}
 
 	aFontData.Import(rPropSet);
-
-    if ( sImageUrl.getLength() )
-    {
-        aTmp <<= sImageUrl;
-        rPropSet->setPropertyValue( WW8_ASCII2STR("ImageURL"), aTmp);
-    }
 	return sal_True;
 }
 
@@ -2285,6 +2287,12 @@ sal_Bool OCX_ToggleButton::Import(com::s
 	}
 
 	aFontData.Import(rPropSet);
+	
+    if ( sImageUrl.getLength() )
+    {
+        aTmp <<= sImageUrl;
+        rPropSet->setPropertyValue( WW8_ASCII2STR("ImageURL"), aTmp);
+    }
 	return sal_True;
 }
 
@@ -2489,12 +2497,6 @@ sal_Bool OCX_Label::Import(uno::Referenc
 	}
 
 	aFontData.Import(rPropSet);
-	
-    if ( sImageUrl.getLength() )
-    {
-        aTmp <<= sImageUrl;
-        rPropSet->setPropertyValue( WW8_ASCII2STR("ImageURL"), aTmp);
-    }
 	return sal_True;
 }
 
@@ -3144,8 +3146,15 @@ sal_Bool OCX_ModernControl::Read(SvStora
 	{
 		pS->Read(pPictureHeader,20);
 		*pS >> nPictureLen;
-		pPicture = new sal_uInt8[nPictureLen];
-		pS->Read(pPicture,nPictureLen);
+		long imagePos = pS->Tell();
+		mxGrfObj = lcl_readGraphicObject( pS );
+		if( mxGrfObj.is() )
+		{
+			sImageUrl = rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( GRAPHOBJ_URLPREFIX ) );
+			sImageUrl = sImageUrl + mxGrfObj->getUniqueID();
+		}
+		// make sure the stream position should be pointing after the image
+		pS->Seek( imagePos + nPictureLen );
 	}
 
 	return sal_True;
@@ -3961,15 +3970,8 @@ sal_Bool OCX_Frame::Read(SvStorageStream
 	{
 		pS->Read(pPictureHeader,20);
 		*pS >> nPictureLen;
-		long imagePos = pS->Tell();
-		mxGrfObj = lcl_readGraphicObject( pS );
-		if( mxGrfObj.is() )
-		{
-			sImageUrl = rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( GRAPHOBJ_URLPREFIX ) );
-			sImageUrl = sImageUrl + mxGrfObj->getUniqueID();
-		}
-		// make sure the stream position should be pointing after the image
-		pS->Seek( imagePos + nPictureLen );
+		pPicture = new sal_uInt8[nPictureLen];
+		pS->Read(pPicture,nPictureLen);
 	}
 
     ReadAlign( pS, pS->Tell() - nStart, 4);
