--- pyuno/source/module/makefile.mk.pristine	2008-10-29 14:25:58.000000000 +0100
+++ pyuno/source/module/makefile.mk	2008-10-29 14:25:33.000000000 +0100
@@ -57,6 +57,7 @@ EXTRA_FRAMEWORK_FLAG=-framework Python
 PYUNO_MODULE=$(DLLDEST)$/pyuno.so
 PYUNORC=pyunorc
 .ELSE
+.INCLUDEDIRS: $(PRJ)/../python
 .INCLUDE :  pyversion.mk
 PYUNORC=pyuno.ini
 .ENDIF
@@ -118,7 +119,7 @@
 	$(LB)$/lib$(TARGET).a
 
 $(LB)$/lib$(TARGET).a: $(MISC)$/$(TARGET).def
-	dlltool --dllname $(TARGET)$(DLLPOST) --input-def=$(MISC)$/$(TARGET).def --kill-at --output-lib=$(LB)$/lib$(TARGET).a
+    $(DLLTOOL) --dllname $(TARGET)$(DLLPOST) --input-def=$(MISC)$/$(TARGET).def --kill-at --output-lib=$(LB)$/lib$(TARGET).a
 .ELSE
 ALLTAR : \
 	$(DLLDEST)$/uno.py

--- pyuno/source/loader/makefile.mk.orig	2008-04-11 12:31:24.000000000 +0200
+++ pyuno/source/loader/makefile.mk	2009-07-14 11:36:44.000000000 +0200
@@ -117,7 +117,7 @@
 	cd $(DLLDEST) && sh -c "DYLD_LIBRARY_PATH=$(DYLD_LIBRARY_PATH):$(OUT)$/lib;export DYLD_LIBRARY_PATH;regcomp -register -r pyuno_services.tmp $(foreach,i,$(COMPONENTS) -c $(i))"
 .ELSE
 .IF "$(GUI)$(COM)"=="WNTGCC"
-	cd $(DLLDEST) && sh -c "export PATH='$(PATH):$(OUT)$/bin'; regcomp -register -r pyuno_services.tmp $(foreach,i,$(COMPONENTS) -c $(i))"
+    cd $(DLLDEST) && sh -c "export PATH='$(PATH):$(OUT)$/bin'; wine $(SOLARBINDIR)/regcomp -register -r pyuno_services.tmp $(foreach,i,$(COMPONENTS) -c $(i))"
 .ELSE
 	cd $(DLLDEST) && regcomp -register -r pyuno_services.tmp $(foreach,i,$(COMPONENTS) -c $(i))
 .ENDIF    # "$(GUI)$(COM)"=="WNTGCC" 
