--- solenv/inc/shlinfo.rc.pristine	2008-10-29 09:33:25.000000000 +0100
+++ solenv/inc/shlinfo.rc	2008-10-29 09:33:33.000000000 +0100
@@ -29,8 +29,9 @@
  ************************************************************************/
 
 #define VERSION			3
-#define SUBVERSION		1
-#define VERVARIANT		0
+#if !defined(VERVARIANT)
+  #define VERVARIANT 		0
+#endif
 
 // .0 + VER_CONCEPT
 // .100 + VER_ALPHA
@@ -90,16 +91,16 @@
 // 				version information
 // -----------------------------------------------------------------------
 
-VS_VERSION_INFO versioninfo
+VS_VERSION_INFO VERSIONINFO
 #ifndef SUBVERSION
-	fileversion    VERSION, 0, 0, VERVARIANT
-	productversion VERSION, 0, 0, VERVARIANT
+    FILEVERSION    VERSION, 0, 0, VERVARIANT
+    PRODUCTVERSION VERSION, 0, 0, VERVARIANT
 #else
-	fileversion    VERSION, VERVARIANT, VER_COUNT
-	productversion VERSION, VERVARIANT, VER_COUNT
+    FILEVERSION    VERSION, VERVARIANT, VER_COUNT
+    PRODUCTVERSION VERSION, VERVARIANT, VER_COUNT
 #endif
-	fileflagsmask 0x3F
-	fileflags
+    FILEFLAGSMASK 0x3F
+    FILEFLAGS
 #if defined(DEBUG)
 	VS_FF_DEBUG |
 #endif
@@ -108,36 +109,36 @@
 #endif
 	0
 #ifndef WIN32
-	fileos VOS_DOS_WINDOWS16
+    FILEOS VOS_DOS_WINDOWS16
 #else
-	fileos VOS_NT_WINDOWS32
+    FILEOS VOS_NT_WINDOWS32
 #endif
-	filetype VFT_APP
+    FILETYPE VFT_APP
 	{
-		block "StringFileInfo"
+        BLOCK "StringFileInfo"
 		{
 #ifdef LG_D
-			block "040704E4"
+            BLOCK "040704E4"
 			{
 				// German StringTable
-				value "CompanyName",		"Sun Microsystems, Inc.\0"
-				value "FileVersion",		PPS(VER_LEVEL) "\0"
-				value "OriginalFilename",	PPS(ORG_NAME) "\0"
-				value "InternalName", 		PPS(INTERNAL_NAME) "\0"
-				value "LegalCopyright",		S_CRIGHT " Sun Microsystems, Inc.\0"
+                VALUE "CompanyName",		"Sun Microsystems, Inc.\0"
+                VALUE "FileVersion",		PPS(VER_LEVEL) "\0"
+                VALUE "OriginalFilename",	PPS(ORG_NAME) "\0"
+                VALUE "InternalName", 		PPS(INTERNAL_NAME) "\0"
+                VALUE "LegalCopyright",		S_CRIGHT " Sun Microsystems, Inc.\0"
 				ADDITIONAL_VERINFO1
 				ADDITIONAL_VERINFO2
 				ADDITIONAL_VERINFO3
 			}
 #else
-			block "040904E4"
+            BLOCK "040904E4"
 			{
 				// International StringTable
-				value "CompanyName", 		"Sun Microsystems, Inc.\0"
-				value "FileVersion",		PPS(VER_LEVEL) "\0"
-				value "OriginalFilename",	PPS(ORG_NAME) "\0"
-				value "InternalName", 		PPS(INTERNAL_NAME) "\0"
-				value "LegalCopyright", 	S_CRIGHT " Sun Microsystems, Inc.\0"
+                VALUE "CompanyName", 		"Sun Microsystems, Inc.\0"
+                VALUE "FileVersion",		PPS(VER_LEVEL) "\0"
+                VALUE "OriginalFilename",	PPS(ORG_NAME) "\0"
+                VALUE "InternalName", 		PPS(INTERNAL_NAME) "\0"
+                VALUE "LegalCopyright", 	S_CRIGHT " Sun Microsystems, Inc.\0"
 				ADDITIONAL_VERINFO1
 				ADDITIONAL_VERINFO2
 				ADDITIONAL_VERINFO3
@@ -145,12 +146,12 @@
 #endif
 		}
 
-		block "VarFileInfo"
+        BLOCK "VarFileInfo"
 		{
 #ifdef LG_D
-			value "Translation", 0x0407, 1252
+            VALUE "Translation", 0x0407, 1252
 #else
-			value "Translation", 0x0409, 1252
+            VALUE "Translation", 0x0409, 1252
 #endif
 		}
 	}
@@ -160,6 +161,6 @@
 {
 	0xF0, "sw", 0x0F, VER_YEAR, VER_MONTH, VER_DAY,
 	VERSION, VERVARIANT, VER_COUNT
-};
+}
 
 #endif // OS2
