--- desktop/win32/source/setup/setup_w.cxx.~1.5.~	2008-04-11 06:50:30.000000000 +0200
+++ desktop/win32/source/setup/setup_w.cxx	2008-11-05 15:53:43.000000000 +0100
@@ -34,5 +34,9 @@
 #define  UNICODE    1
 #define _UNICODE    1
 
+#ifdef __MINGW32__
+#define _ttempnam(a, b) (TCHAR*)tempnam ((char*)a, (char*)b)
+#endif
+
 #include "setup.cpp"
 
--- desktop/win32/source/setup/setup_a.cxx.~1.5.~	2008-04-11 06:49:31.000000000 +0200
+++ desktop/win32/source/setup/setup_a.cxx	2008-11-05 15:52:36.000000000 +0100
@@ -31,4 +31,8 @@
 // MARKER(update_precomp.py): autogen include statement, do not remove
 #include "precompiled_desktop.hxx"
 
+#ifdef __MINGW32__
+#define _ttempnam tempnam
+#endif
+
 #include "setup.cpp"
--- desktop/win32/source/setup/rcheader.txt.~1.6.~	2008-01-15 14:33:41.000000000 +0100
+++ desktop/win32/source/setup/rcheader.txt	2008-11-05 15:16:00.000000000 +0100
@@ -7,7 +7,7 @@
 #define CB_ADDSTRING	(WM_USER+3)
 #define IDC_STATIC		(-1)
 #endif
-#include "resource.h"
+#include "Resource.h"
 
 /////////////////////////////////////////////////////////////////////////////
 //
--- desktop/win32/source/setup/setup.cpp.~1.15.12.1.~	2008-09-03 15:00:32.000000000 +0200
+++ desktop/win32/source/setup/setup.cpp	2008-11-05 15:52:16.000000000 +0100
@@ -49,7 +49,7 @@
 
 #include "setup.hxx"
 
-#include "resource.h"
+#include "Resource.h"
 
 //--------------------------------------------------------------------------
 
--- desktop/win32/source/setup/makefile.mk.~1.12.~	2008-04-11 06:48:48.000000000 +0200
+++ desktop/win32/source/setup/makefile.mk	2008-11-05 15:19:13.000000000 +0100
@@ -91,5 +91,9 @@
 .INCLUDE :  target.mk
 
 $(RCFILES) : $(ULFDIR)$/setup.ulf makefile.mk rcfooter.txt rcheader.txt rctmpl.txt
+.IF "$(CROSS_COMPILING)" == "yes"
+    $(WRAPCMD) wine $(SOLARBINDIR)/lngconvex.exe -ulf $(ULFDIR)$/setup.ulf -rc $(RCFILES) -rct rctmpl.txt -rch rcheader.txt -rcf rcfooter.txt
+.ELSE
     $(LNGCONVEX) -ulf $(ULFDIR)$/setup.ulf -rc $(RCFILES) -rct rctmpl.txt -rch rcheader.txt -rcf rcfooter.txt
+.ENDIF

--- /dev/null	2009-04-14 12:16:08.000000000 +0200
+++ desktop/win32/source/setup/strsafe.h	2009-07-15 16:53:22.000000000 +0200
@@ -0,0 +1,45 @@
+//http://baseutils.googlecode.com/svn/trunk/str_strsafe.h
+
+/* Written by Krzysztof Kowalczyk (http://blog.kowalczyk.info)
+   The author disclaims copyright to this source code. */
+#ifndef __STR_STRSAFE_H
+#define __STR_STRSAFE_H
+
+/* When using MSVC, use <strsafe.h>, emulate it on other compiler (e.g. mingw) */
+
+#define DISABLE_STRSAFE
+#ifndef DISABLE_STRSAFE
+  #include <strsafe.h>
+#else
+  #include <stdio.h>
+  #include <string.h>
+  #include <windows.h>
+  #define	STRSAFE_E_INSUFFICIENT_BUFFER   -1
+  #define	_vsnprintf_s(p,s,z,f,a)		vsnprintf(p,s,f,a)
+
+#ifndef min
+  #define min(a,b) ((a)<(b)?(a):(b))
+#endif
+
+  /* WARNING: the return values of these two pairs of functions aren't
+  compatible. The strsafe functions return an error code and the standard C
+  functions return a character count */
+  #define	StringCchVPrintfA			vsnprintf
+  #define	StringCchVPrintfW			vsnprintf
+  #define	StringCchVPrintf			vsnprintf
+  
+  #define StringCchPrintfA(str, n, format, ...) snprintf ((char*)str, n, (char const*)format, __VA_ARGS__)
+  #define StringCchPrintfW(str, n, format, ...) snprintf ((char*)str, n, (char const*)format, __VA_ARGS__)
+  #define StringCchPrintf(str, n, format, ...) snprintf ((char*)str, n, (char const*)format, __VA_ARGS__)
+  
+  #define StringCchCopy(dest, n, src) strncpy ((char*)dest, (char const*)src, n)
+  #define StringCchCopyN(dest, n1, src, n2) strncpy ((char*)dest, (char const*)src, min (n1, n2))
+  #define StringCchCat(dest, n, src) strncat ((char*)dest, (char const*)src, n)
+
+  #define STRSAFE_MAX_CCH 0x7FFFFFFF
+
+  #define	_stricmp					strcasecmp
+  #define	_strnicmp					strncasecmp
+#endif
+
+#endif

--- desktop/source/pkgchk/unopkg/makefile.mk.~1.17.~	2008-07-22 19:16:08.000000000 +0200
+++ desktop/source/pkgchk/unopkg/makefile.mk	2008-11-05 14:58:45.000000000 +0100
@@ -55,7 +55,7 @@ APP1STDLIBS = $(SALLIB) $(UNOPKGAPPLIB)
 APP1DEPN = $(SHL1TARGETN)
 APP1NOSAL = TRUE
 APP1RPATH = BRAND
-.IF "$(OS)" == "WNT"
+.IF "$(TARGET_OS)" == "WNT"
 APP1ICON = $(SOLARRESDIR)$/icons/so9_main_app.ico
 APP1LINKRES = $(MISC)$/$(TARGET)1.res
 .ENDIF
@@ -66,7 +66,7 @@ APP2STDLIBS = $(SALLIB) $(UNOPKGAPPLIB)
 APP2DEPN = $(SHL1TARGETN)
 APP2NOSAL = TRUE
 APP2RPATH = BRAND
-.IF "$(OS)" == "WNT"
+.IF "$(TARGET_OS)" == "WNT"
 APP2ICON = $(SOLARRESDIR)$/icons/ooo3_main_app.ico
 APP2LINKRES = $(MISC)$/$(TARGET)2.res
 .ENDIF
--- desktop/source/deployment/misc/dp_misc.cxx.pristine	2009-04-06 12:16:26.807994125 +0200
+++ desktop/source/deployment/misc/dp_misc.cxx	2009-04-06 12:16:33.664089092 +0200
@@ -55,6 +55,6 @@
 #define UNICODE
 #define _UNICODE
 #define WIN32_LEAN_AND_MEAN
-#include <Windows.h>
+#include <windows.h>
 //#include "tools/postwin.h"
 #endif
