--- external/mingwheaders/makefile.mk	2009-12-03 15:56:23.000000000 +0100
+++ external/mingwheaders/makefile.mk	2009-12-03 15:58:17.000000000 +0100
@@ -35,17 +35,22 @@
 
 .INCLUDE :  settings.mk
 
-.IF "$(OS)$(COM)" == "WNTGCC"
+.IF "$(OS)$(COM)" == "WNTGCC" && "$(CROSS_COMPILING)" != "yes"
 
 # Cygwin and MinGW use different directories for the W32API headers
 .IF "$(USE_MINGW)" == "cygwin"
 MINGW_INCLUDE_DIR=$/usr$/include/mingw/
 MINGW_W32API_INCLUDE_DIR=$/usr$/include/w32api/
 MINGW_W32API_LIB_DIR=$/usr$/include/w32api/
-.ELSE
+.ELIF "$(CROSS_COMPILING)" == "no"
 MINGW_INCLUDE_DIR=$(COMPATH)$/include/
 MINGW_W32API_INCLUDE_DIR=$(COMPATH)$/include/
 MINGW_W32API_LIB_DIR=$(COMPATH)$/lib/
+.ELSE # "$(CROSS_COMPILING)" == "yes"
+MINGW_INCLUDE_DIR=$(COMPATH)/../../include/
+MINGW_W32API_INCLUDE_DIR=$(COMPATH)/../../include/
+MINGW_W32API_LIB_DIR=$(COMPATH)/../../lib/
+PSDK_HOME=$(COMPATH)/../..
 .ENDIF
 
 SYS_INCLUDE_DIR=$(MINGW_INCLUDE_DIR)$/sys/
@@ -174,8 +179,10 @@
     $(COPY) $(FILES_TO_COPY_FROM_MINGW:^$(MINGW_INCLUDE_DIR)) $(MISC)$/mingw$/include
     $(COPY) $(FILES_TO_COPY_FROM_W32API:^$(MINGW_W32API_INCLUDE_DIR)) $(MISC)$/mingw$/include
     $(COPY) $(FILES_TO_COPY_FROM_SYS:^$(SYS_INCLUDE_DIR)) $(MISC)$/mingw$/include$/sys
-    $(COPY) $(FILES_TO_COPY_FROM_PSDK:^$(PSDK_INCLUDE_DIR)) $(MISC)$/mingw$/include
+    $(COPY) $(FILES_TO_COPY_FROM_PSDK:^$(PSDK_INCLUDE_DIR)) $(MISC)$/mingw$/include || :
+.IF "$(DIRECTXSDK_HOME)" != ""
     $(COPY) $(FILES_TO_COPY_FROM_DIRECTXSDK:^$(DIRECTXSDK_INCLUDE_DIR)) $(MISC)$/mingw$/include
+.ENDIF # "$(DIRECTXSDK_HOME)" != ""
     $(TOUCH) $(MISC)$/mingwheader_copy
 
 .IF "$(DISABLE_ATL)"==""
@@ -191,16 +198,16 @@
 .ELSE
     $(COPY) $(COMPATH)$/lib/libmsvcrt.a $(LB)$/libmsvcrt_orig.a
 .ENDIF
-    cd $(LB) && ar -x libmsvcrt_orig.a ofmt_stub.o
-    ar -d $(LB)$/libmsvcrt_orig.a ofmt_stub.o
-    nm -g --defined-only $(LB)$/libmsvcrt_orig.a > $(MISC)$/msvcrt.nm
+    cd $(LB) && $(AR) -x libmsvcrt_orig.a ofmt_stub.o
+    $(AR) -d $(LB)$/libmsvcrt_orig.a ofmt_stub.o
+    $(NM) -g --defined-only $(LB)$/libmsvcrt_orig.a > $(MISC)$/msvcrt.nm
     echo EXPORTS > $(MISC)$/msvcrt.def
     sed -ne 's/.* T _//p' $(MISC)$/msvcrt.nm | sort >> $(MISC)$/msvcrt.def
     -sed -ne 's/.* I __imp__//p' $(MISC)$/msvcrt.nm | sort | diff - $(MISC)$/msvcrt.def | \
         sed -ne 's/^< \(.*\)$/\1 DATA/p' > $(MISC)$/msvcrtdata.def
     cat $(MISC)$/msvcrtdata.def >> $(MISC)$/msvcrt.def
-    dlltool --dllname msvcrt.dll --input-def=$(MISC)$/msvcrt.def --kill-at --output-lib=$(LB)$/libmsvcrt.a
-    cd $(LB) && ar -rs libmsvcrt.a ofmt_stub.o
+    $(DLLTOOL) --dllname msvcrt.dll --input-def=$(MISC)$/msvcrt.def --kill-at --output-lib=$(LB)$/libmsvcrt.a
+    cd $(LB) && $(AR) -rs libmsvcrt.a ofmt_stub.o
 
 clean:
     -$(RM) $(MISC)$/mingw$/include$/*.h
