--- sd/source/ui/dlg/vectdlg.src.orig	2008-03-23 15:10:03.984375000 +0100
+++ sd/source/ui/dlg/vectdlg.src	2008-03-23 15:19:09.062500000 +0100
@@ -136,7 +136,7 @@
 		Last = 32 ;
 		SpinSize = 1 ;
 		Unit = FUNIT_CUSTOM ;
-		CustomUnitText = " Pixel" ;
+        CustomUnitText [ en-US ] = " Pixel" ;
 	};
 
 	CheckBox CB_FILLHOLES
@@ -169,7 +169,7 @@
 		Last = 128;
 		SpinSize = 1;
 		Unit = FUNIT_CUSTOM ;
-		CustomUnitText = " Pixel" ;
+        CustomUnitText [ en-US ] = " Pixel" ;
 	};
 
 	FixedText FT_ORIGINAL
--- svx/source/cui/grfflt.src
+++ svx/source/cui/grfflt.src
@@ -83,7 +83,7 @@
         Maximum = 999 ;
         StrictFormat = TRUE ;
         Unit = FUNIT_CUSTOM ;
-        CustomUnitText = " Pixel" ;
+        CustomUnitText [ en-US ] = " Pixel" ;
         First = 1 ;
         Last = 999 ;
         SpinSize = 1 ;
@@ -108,7 +108,7 @@
         Maximum = 999 ;
         StrictFormat = TRUE ;
         Unit = FUNIT_CUSTOM ;
-        CustomUnitText = " Pixel" ;
+        CustomUnitText [ en-US ] = " Pixel" ;
         First = 1 ;
         Last = 999 ;
         SpinSize = 1 ;
--- svx/source/cui/textanim.src
+++ svx/source/cui/textanim.src
@@ -232,7 +232,7 @@
 		Spin = TRUE ;
 		StrictFormat = TRUE ;
 		Unit = FUNIT_CUSTOM ;
-		CustomUnitText = " Pixel" ;
+        CustomUnitText [ en-US ] = " Pixel" ;
 		Minimum = 1 ;
 		First = 1 ;
 		Maximum = 10000 ;
