diff --git accessibility/source/helper/acc_factory.cxx accessibility/source/helper/acc_factory.cxx
index aea02be..84c702a 100644
--- accessibility/source/helper/acc_factory.cxx
+++ accessibility/source/helper/acc_factory.cxx
@@ -72,6 +72,7 @@
 #include <accessibility/extended/accessibleeditbrowseboxcell.hxx>
 #include <vcl/lstbox.hxx>
 #include <vcl/combobox.hxx>
+#include "vcl/popupmenuwindow.hxx"
 
 #include <floatingwindowaccessible.hxx>
 
@@ -376,7 +377,17 @@ inline bool hasFloatingChild(Window *pWindow)
             }
             else if ( nType == WINDOW_BORDERWINDOW && hasFloatingChild( pWindow ) ) 
             {
-                xContext = new FloatingWindowAccessible( _pXWindow );
+                PopupMenuFloatingWindow* pChild = dynamic_cast<PopupMenuFloatingWindow*>(
+                    pWindow->GetAccessibleChildWindow(0));
+                if ( pChild && pChild->IsPopupMenu() )
+                {
+                    // Get the accessible context from the child window.
+                    Reference<XAccessible> xAccessible = pChild->CreateAccessible();
+                    if (xAccessible.is())
+                        xContext = xAccessible->getAccessibleContext();
+                }
+                else
+                    xContext = new FloatingWindowAccessible( _pXWindow );
             }            
             else if ( nType == WINDOW_HELPTEXTWINDOW )
             {
