diff --git a/sc/inc/column.hxx b/sc/inc/column.hxx
index a664cfd..a734f42 100644
--- sc/inc/column.hxx
+++ sc/inc/column.hxx
@@ -36,6 +36,7 @@
 #include "compressedarray.hxx"
 #include "address.hxx"
 #include "rangenam.hxx"
+#include "dociter.hxx"
 #include <tools/solar.h>
 
 #include <set>
@@ -118,6 +119,7 @@ friend class ScDocument;					// fuer FillInfo
 friend class ScDocumentIterator;
 friend class ScValueIterator;
 friend class ScDBQueryDataIterator;
+friend class ScDBQueryDataIterator::DataAccessInternal;
 friend class ScColumnIterator;
 friend class ScQueryCellIterator;
 friend class ScMarkedDataIter;
diff --git a/sc/inc/dociter.hxx b/sc/inc/dociter.hxx
index 22cc806..60b57a6 100644
--- sc/inc/dociter.hxx
+++ sc/inc/dociter.hxx
@@ -161,6 +161,7 @@ private:
         const ScDBQueryDataIterator* mpParent;
     };
 
+public:
     class DataAccessInternal : public DataAccess
     {
     public:
@@ -185,6 +186,7 @@ private:
         bool                bCalcAsShown;
     };
 
+private:
     class DataAccessMatrix : public DataAccess
     {
     public:
diff --git a/sc/inc/document.hxx b/sc/inc/document.hxx
index 4d94701..43642e8 100644
--- sc/inc/document.hxx
+++ sc/inc/document.hxx
@@ -239,6 +239,7 @@ class ScDocument
 friend class ScDocumentIterator;
 friend class ScValueIterator;
 friend class ScDBQueryDataIterator;
+friend class ScDBQueryDataIterator::DataAccessInternal;
 friend class ScCellIterator;
 friend class ScQueryCellIterator;
 friend class ScHorizontalCellIterator;
diff --git a/sc/inc/table.hxx b/sc/inc/table.hxx
index 9606b98..73bdb6a 100644
--- sc/inc/table.hxx
+++ sc/inc/table.hxx
@@ -188,6 +188,7 @@ friend class ScDocument;					// fuer FillInfo
 friend class ScDocumentIterator;
 friend class ScValueIterator;
 friend class ScDBQueryDataIterator;
+friend class ScDBQueryDataIterator::DataAccessInternal;
 friend class ScCellIterator;
 friend class ScQueryCellIterator;
 friend class ScHorizontalCellIterator;
