diff --git sc/inc/scabstdlg.hxx sc/inc/scabstdlg.hxx
index 97f6513..5506af1 100644
--- sc/inc/scabstdlg.hxx
+++ sc/inc/scabstdlg.hxx
@@ -68,6 +68,7 @@ class AbstractScImportAsciiDlg : public VclAbstractDialog  //add for ScImportAsc
 public:
     virtual void                        GetOptions( ScAsciiOptions& rOpt ) = 0;
     virtual void                        SetTextToColumnsMode() = 0;
+    virtual void                        SaveParameters() = 0;
 };
 
 
diff --git sc/source/ui/attrdlg/scdlgfact.cxx sc/source/ui/attrdlg/scdlgfact.cxx
index 53f52ab..801b2ca 100644
--- sc/source/ui/attrdlg/scdlgfact.cxx
+++ sc/source/ui/attrdlg/scdlgfact.cxx
@@ -242,6 +242,11 @@ void AbstractScImportAsciiDlg_Impl::SetTextToColumnsMode()
     pDlg->SetTextToColumnsMode();
 }
 
+void AbstractScImportAsciiDlg_Impl::SaveParameters()
+{
+    pDlg->SaveParameters();
+}
+
 // AbstractScImportAsciiDlg_Impl end
 
 //AbstractScAutoFormatDlg_Impl begin
diff --git sc/source/ui/attrdlg/scdlgfact.hxx sc/source/ui/attrdlg/scdlgfact.hxx
index 68245f2..117f478 100644
--- sc/source/ui/attrdlg/scdlgfact.hxx
+++ sc/source/ui/attrdlg/scdlgfact.hxx
@@ -122,6 +122,7 @@ class AbstractScImportAsciiDlg_Impl : public AbstractScImportAsciiDlg  //add for
     DECL_ABSTDLG_BASE(AbstractScImportAsciiDlg_Impl, ScImportAsciiDlg)
     virtual void                        GetOptions( ScAsciiOptions& rOpt );
     virtual void                        SetTextToColumnsMode();
+    virtual void                        SaveParameters();
 };
 
 class AbstractScAutoFormatDlg_Impl : public AbstractScAutoFormatDlg  //add for ScAutoFormatDlg
diff --git sc/source/ui/dbgui/scuiasciiopt.cxx sc/source/ui/dbgui/scuiasciiopt.cxx
index e18b420..8c075ab 100644
--- sc/source/ui/dbgui/scuiasciiopt.cxx
+++ sc/source/ui/dbgui/scuiasciiopt.cxx
@@ -422,11 +422,6 @@ ScImportAsciiDlg::ScImportAsciiDlg( Window* pParent,String aDatName,
 
 ScImportAsciiDlg::~ScImportAsciiDlg()
 {
-    if (mbFileImport)
-        save_Separators( maFieldSeparators, aCbTextSep.GetText(), aCkbAsOnce.IsChecked(), 
-                         aCkbQuotedAsText.IsChecked(), aCkbDetectNumber.IsChecked(), 
-                         aRbFixed.IsChecked(), aNfRow.GetValue(), aLbCharSet.GetSelectEntryPos(),
-                         static_cast<sal_Int32>(aLbCustomLang.GetSelectLanguage()) );
     delete[] mpRowPosArray;
 }
 
@@ -540,6 +535,18 @@ void ScImportAsciiDlg::SetTextToColumnsMode()
     aCkbDetectNumber.Disable();
 }
 
+void ScImportAsciiDlg::SaveParameters()
+{
+    if (!mbFileImport)
+        // We save parameters only for file import.
+        return;
+
+    save_Separators( maFieldSeparators, aCbTextSep.GetText(), aCkbAsOnce.IsChecked(), 
+                     aCkbQuotedAsText.IsChecked(), aCkbDetectNumber.IsChecked(), 
+                     aRbFixed.IsChecked(), aNfRow.GetValue(), aLbCharSet.GetSelectEntryPos(),
+                     static_cast<sal_Int32>(aLbCustomLang.GetSelectLanguage()) );
+}
+
 void ScImportAsciiDlg::SetSelectedCharSet()
 {
     meCharSet = aLbCharSet.GetSelectTextEncoding();
diff --git sc/source/ui/inc/scuiasciiopt.hxx sc/source/ui/inc/scuiasciiopt.hxx
index 4515a86..be01df5 100644
--- sc/source/ui/inc/scuiasciiopt.hxx
+++ sc/source/ui/inc/scuiasciiopt.hxx
@@ -107,6 +107,7 @@ public:
 
     void                        GetOptions( ScAsciiOptions& rOpt );
     void                        SetTextToColumnsMode();
+    void                        SaveParameters();
 
 private:
     /** Sets the selected char set data to meCharSet and mbCharSetSystem. */
diff --git sc/source/ui/unoobj/filtuno.cxx sc/source/ui/unoobj/filtuno.cxx
index 02d2f9d..54066a3 100644
--- sc/source/ui/unoobj/filtuno.cxx
+++ sc/source/ui/unoobj/filtuno.cxx
@@ -179,6 +179,7 @@ sal_Int16 SAL_CALL ScFilterOptionsObj::execute() throw(uno::RuntimeException)
         {
             ScAsciiOptions aOptions;
             pDlg->GetOptions( aOptions );
+            pDlg->SaveParameters();
             aFilterOptions = aOptions.WriteToString();
             nRet = ui::dialogs::ExecutableDialogResults::OK;
         }
