diff --git xmloff/inc/xmlnmspe.hxx xmloff/inc/xmlnmspe.hxx
index c2447dd..2a1c9f4 100644
--- xmloff/inc/xmlnmspe.hxx
+++ xmloff/inc/xmlnmspe.hxx
@@ -89,8 +89,9 @@ const sal_uInt16 XML_NAMESPACE_##prefix##_EXT	    = key; \
 const sal_uInt16 XML_NAMESPACE_##prefix##_EXT_IDX	= key;
 
 XML_NAMESPACE_EXT( OFFICE,      37U )
+XML_NAMESPACE_EXT( TABLE,       38U )
 
-#define _XML_OLD_NAMESPACE_BASE 38U
+#define _XML_OLD_NAMESPACE_BASE 39U
 
 // namespaces used in the technical preview (SO 5.2)
 XML_OLD_NAMESPACE( FO,		0U )
diff --git xmloff/inc/xmloff/xmltoken.hxx xmloff/inc/xmloff/xmltoken.hxx
index daca654..7381057 100644
--- xmloff/inc/xmloff/xmltoken.hxx
+++ xmloff/inc/xmloff/xmltoken.hxx
@@ -147,6 +147,9 @@ namespace xmloff { namespace token {
         // ODF Enhanced namespaces
         XML_NP_OFFICE_EXT,
         XML_N_OFFICE_EXT,
+
+        XML_NP_TABLE_EXT,
+        XML_N_TABLE_EXT,
         
         // units
         XML_UNIT_MM,
@@ -520,6 +523,7 @@ namespace xmloff { namespace token {
         XML_DATA_LABEL_SYMBOL,
         XML_DATA_LABEL_TEXT,
         XML_DATA_PILOT_FIELD,
+        XML_DATA_PILOT_GRAND_TOTAL,
         XML_DATA_PILOT_LEVEL,
         XML_DATA_PILOT_MEMBER,
         XML_DATA_PILOT_MEMBERS,
diff --git xmloff/source/core/xmlexp.cxx xmloff/source/core/xmlexp.cxx
index 37e74b4..842ea86 100644
--- xmloff/source/core/xmlexp.cxx
+++ xmloff/source/core/xmlexp.cxx
@@ -322,6 +322,12 @@ void SvXMLExport::_InitCtor()
         mpNamespaceMap->Add( GetXMLToken(XML_NP_OOOW),  GetXMLToken(XML_N_OOOW),    XML_NAMESPACE_OOOW );
         mpNamespaceMap->Add( GetXMLToken(XML_NP_OOOC),  GetXMLToken(XML_N_OOOC),    XML_NAMESPACE_OOOC );
         mpNamespaceMap->Add( GetXMLToken(XML_NP_OF),    GetXMLToken(XML_N_OF),      XML_NAMESPACE_OF );
+
+        if (getDefaultVersion() == SvtSaveOptions::ODFVER_LATEST)
+        {
+            mpNamespaceMap->Add(
+                GetXMLToken(XML_NP_TABLE_EXT), GetXMLToken(XML_N_TABLE_EXT), XML_NAMESPACE_TABLE_EXT);
+        }
     }
     if( (getExportFlags() & (EXPORT_MASTERSTYLES|EXPORT_CONTENT) ) != 0 )
     {
diff --git xmloff/source/core/xmlimp.cxx xmloff/source/core/xmlimp.cxx
index 9807e79..3ecc7f2 100644
--- xmloff/source/core/xmlimp.cxx
+++ xmloff/source/core/xmlimp.cxx
@@ -101,6 +101,7 @@ sal_Char __READONLY_DATA sXML_np__of[] = "_of";
 sal_Char __READONLY_DATA sXML_np__style[] = "_style";
 sal_Char __READONLY_DATA sXML_np__text[] = "_text";
 sal_Char __READONLY_DATA sXML_np__table[] = "_table";
+sal_Char __READONLY_DATA sXML_np__table_ext[] = "_table_ooo";
 sal_Char __READONLY_DATA sXML_np__draw[] = "_draw";
 sal_Char __READONLY_DATA sXML_np__dr3d[] = "_dr3d";
 sal_Char __READONLY_DATA sXML_np__fo[] = "_fo";
@@ -259,6 +260,9 @@ void SvXMLImport::_InitCtor()
         mpNamespaceMap->Add( OUString( RTL_CONSTASCII_USTRINGPARAM ( sXML_np__table ) ),
                             GetXMLToken(XML_N_TABLE),
                             XML_NAMESPACE_TABLE );
+        mpNamespaceMap->Add( OUString( RTL_CONSTASCII_USTRINGPARAM ( sXML_np__table_ext ) ),
+                            GetXMLToken(XML_N_TABLE_EXT),
+                            XML_NAMESPACE_TABLE_EXT );
         mpNamespaceMap->Add( OUString( RTL_CONSTASCII_USTRINGPARAM ( sXML_np__draw ) ),
                             GetXMLToken(XML_N_DRAW),
                             XML_NAMESPACE_DRAW );
diff --git xmloff/source/core/xmltoken.cxx xmloff/source/core/xmltoken.cxx
index 0a2ccd5..7a50942 100644
--- xmloff/source/core/xmltoken.cxx
+++ xmloff/source/core/xmltoken.cxx
@@ -156,6 +156,9 @@ namespace xmloff { namespace token {
         TOKEN( "officeooo", XML_NP_OFFICE_EXT ),
         TOKEN( "http://openoffice.org/2009/office", XML_N_OFFICE_EXT ),
 
+        TOKEN( "tableooo", XML_NP_TABLE_EXT ),
+        TOKEN( "http://openoffice.org/2009/table", XML_N_TABLE_EXT ),
+
         // units
         TOKEN( "mm",                              XML_UNIT_MM ),
         TOKEN( "m",                               XML_UNIT_M ),
@@ -528,6 +531,7 @@ namespace xmloff { namespace token {
         TOKEN( "data-label-symbol",               XML_DATA_LABEL_SYMBOL ),
         TOKEN( "data-label-text",                 XML_DATA_LABEL_TEXT ),
         TOKEN( "data-pilot-field",                XML_DATA_PILOT_FIELD ),
+        TOKEN( "data-pilot-grand-total",          XML_DATA_PILOT_GRAND_TOTAL ),
         TOKEN( "data-pilot-level",                XML_DATA_PILOT_LEVEL ),
         TOKEN( "data-pilot-member",               XML_DATA_PILOT_MEMBER ),
         TOKEN( "data-pilot-members",              XML_DATA_PILOT_MEMBERS ),
