diff --git a/configure.in b/configure.in
--- configure.in
+++ configure.in
@@ -5134,6 +5134,21 @@
    fi
 fi
 
+if test "$_os" != "WNT"; then
+  AC_CHECK_FUNCS(getopt, HAVE_GETOPT=YES, [HAVE_GETOPT=NO])
+  AC_CHECK_FUNCS(readdir_r, HAVE_READDIR_R=YES, [HAVE_READDIR_R=NO])
+  if test "$HAVE_GETOPT" = "YES" -a "$HAVE_READDIR_R" = "YES"; then
+	SYSTEM_LIBC=YES
+  fi
+else
+  HAVE_GETOPT=NO
+  HAVE_READDIR_R=NO
+  SYSTEM_LIBC=YES
+fi
+AC_SUBST(HAVE_GETOPT)
+AC_SUBST(HAVE_READDIR_R)
+AC_SUBST(SYSTEM_LIBC)
+
 dnl =========================================
 dnl Check for the Microsoft Platform SDK.
 dnl =========================================
diff --git a/external/glibc-2.1.3.patch b/external/glibc-2.1.3.patch
--- external/glibc-2.1.3.patch
+++ external/glibc-2.1.3.patch
@@ -76,7 +76,7 @@
  extern int getopt ();
 --- misc/glibc-2.1.3/posix/makefile.mk	Mon Mar 31 09:43:50 2008
 +++ misc/build/glibc-2.1.3/posix/makefile.mk	Mon Mar 31 09:43:38 2008
-@@ -1 +1,63 @@
+@@ -1 +1,70 @@
 -dummy
 +#*************************************************************************
 +#
@@ -123,14 +123,16 @@
 +
 +# --- Files --------------------------------------------------------
 +
++.IF "$(HAVE_GETOPT)" != "YES"
 +OBJFILES=	$(OBJ)$/getopt.obj		\
 +			$(OBJ)$/getopt1.obj
 +
 +LIB1TARGET=$(SLB)$/$(TARGET).lib
 +LIB1ARCHIV=$(LB)$/lib$(TARGET).a
 +LIB1OBJFILES=$(OBJFILES)
++.ENDIF
 +
-+.IF "$(OS)"=="NETBSD"
++.IF "$(HAVE_READDIR_R)" != "YES" && "$(OS)" != "WNT"
 +TARGET2=gnu_readdir_r
 +OBJFILES+= $(OBJ)$/readdir_r.obj
 +LIB2TARGET=$(SLB)$/$(TARGET2).lib
@@ -138,6 +140,11 @@
 +LIB2OBJFILES= $(OBJ)$/readdir_r.obj
 +.ENDIF
 +
++.IF "$(HAVE_READDIR_R)" == "YES" && "$(HAVE_GETOPT)" == "YES"
++@all:
++	@echo "Nothing to do here.
++.ENDIF
++
 +# --- Targets ------------------------------------------------------
 +
 +.INCLUDE :  target.mk
diff --git a/external/glibc/makefile.mk b/external/glibc/makefile.mk
--- external/glibc/makefile.mk
+++ external/glibc/makefile.mk
@@ -36,10 +36,14 @@
 
 # --- Files --------------------------------------------------------
 
+.IF "$(HAVE_GETOPT)" != "YES" || "$(HAVE_READDIR_R)" != "YES"
 TARFILE_NAME=glibc-2.1.3-stub
 TARFILE_MD5=4a660ce8466c9df01f19036435425c3a
 TARFILE_ROOTDIR=glibc-2.1.3
-ADDITIONAL_FILES=posix$/makefile.mk posix$/config.h posix$/readdir_r.c
+ADDITIONAL_FILES=posix$/makefile.mk posix$/config.h
+.IF "$(HAVE_READDIR_R)" != "YES"
+ADDITIONAL_FILES += posix$/readdir_r.c
+.ENDIF
 
 PATCH_FILES=$(PRJ)$/glibc-2.1.3.patch
 
@@ -49,6 +53,15 @@
 BUILD_DIR=posix
 BUILD_ACTION=dmake $(MFLAGS) $(CALLMACROS)
 
+OUT2INC= \
+	posix/getopt.h \
+	posix/config.h
+
+.ELSE
+@all:
+	@echo "Nothing to do here."
+.ENDIF
+
 # --- Targets ------------------------------------------------------
 
 .INCLUDE : set_ext.mk
diff --git a/external/prj/d.lst b/external/prj/d.lst
--- external/prj/d.lst
+++ external/prj/d.lst
@@ -12,6 +12,8 @@
 ..\%__SRC%\lib\libgnu_getopt.a %_DEST%\lib%_EXT%\libgnu_getopt.a
 ..\%__SRC%\lib\libgnu_getopt_static.a %_DEST%\lib%_EXT%\libgnu_getopt_static.a
 
+..\%__SRC%\inc\*.h %_DEST%\inc%_EXT%\external\glibc\*.h
+
 ..\%__SRC%\misc\mingw\include\*.h %_DEST%\inc%_EXT%\external\mingw\include\*
 ..\%__SRC%\misc\mingw\include\atl\*.h %_DEST%\inc%_EXT%\external\mingw\include\atl\*
 ..\%__SRC%\misc\mingw\include\sys\*.h %_DEST%\inc%_EXT%\external\mingw\include\sys\*
diff --git a/idlc/source/preproc/getopt.c b/idlc/source/preproc/getopt.c
deleted file mode 100644
--- idlc/source/preproc/getopt.c
+++ /dev/null
@@ -1,94 +0,0 @@
-/*************************************************************************
- *
- * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
- * 
- * Copyright 2000, 2010 Oracle and/or its affiliates.
- *
- * OpenOffice.org - a multi-platform office productivity suite
- *
- * This file is part of OpenOffice.org.
- *
- * OpenOffice.org is free software: you can redistribute it and/or modify
- * it under the terms of the GNU Lesser General Public License version 3
- * only, as published by the Free Software Foundation.
- *
- * OpenOffice.org is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU Lesser General Public License version 3 for more details
- * (a copy is included in the LICENSE file that accompanied this code).
- *
- * You should have received a copy of the GNU Lesser General Public License
- * version 3 along with OpenOffice.org.  If not, see
- * <http://www.openoffice.org/license.html>
- * for a copy of the LGPLv3 License.
- *
- ************************************************************************/
-
-#include	<stdio.h>
-#include    <string.h>
-
-#define EPR                 fprintf(stderr,
-#define ERR(str, chr)       if(opterr) { EPR "%s%c\n", str, chr); }
-
-int opterr = 1;
-int optind = 1;
-int optopt;
-char *optarg;
-
-int
-    cppgetopt(int argc, char *const argv[], const char *opts)
-{
-    static int sp = 1;
-    register int c;
-    register char *cp;
-
-    if (sp == 1)
-    {
-        if (optind >= argc ||
-            argv[optind][0] != '-' || argv[optind][1] == '\0')
-            return -1;
-        else
-            if (strcmp(argv[optind], "--") == 0)
-            {
-                optind++;
-                return -1;
-            }
-    }
-    optopt = c = argv[optind][sp];
-    if (c == ':' || (cp = strchr(opts, c)) == 0)
-    {
-        ERR(": illegal option -- ", c);
-        if (argv[optind][++sp] == '\0')
-        {
-            optind++;
-            sp = 1;
-        }
-        return '?';
-    }
-    if (*++cp == ':')
-    {
-        if (argv[optind][sp + 1] != '\0')
-            optarg = &argv[optind++][sp + 1];
-        else
-            if (++optind >= argc)
-            {
-                ERR(": option requires an argument -- ", c);
-                sp = 1;
-                return '?';
-            }
-            else
-                optarg = argv[optind++];
-        sp = 1;
-    }
-    else
-    {
-        if (argv[optind][++sp] == '\0')
-        {
-            sp = 1;
-            optind++;
-        }
-        optarg = 0;
-    }
-    return c;
-}
diff --git a/idlc/source/preproc/makefile.mk b/idlc/source/preproc/makefile.mk
--- idlc/source/preproc/makefile.mk
+++ idlc/source/preproc/makefile.mk
@@ -43,7 +43,6 @@
 OBJFILES=   \
 			$(OBJ)$/cpp.obj	\
 			$(OBJ)$/eval.obj	\
-			$(OBJ)$/getopt.obj	\
 			$(OBJ)$/include.obj	\
 			$(OBJ)$/lex.obj	\
 			$(OBJ)$/macro.obj	\
@@ -64,6 +63,12 @@
 
 APP1LIBS= $(LB)$/idlcpp.lib
 
+.IF "$(HAVE_GETOPT)" != "YES"
+.IF "$(GUI)" == "WNT"
+APP1STDLIBS=gnu_getopt.lib
+.ENDIF
+.ENDIF
+
 # --- Targets ------------------------------------------------------
 
 .INCLUDE :  target.mk
diff --git a/idlc/source/preproc/nlist.c b/idlc/source/preproc/nlist.c
--- idlc/source/preproc/nlist.c
+++ idlc/source/preproc/nlist.c
@@ -29,9 +29,6 @@
 #include <string.h>
 #include "cpp.h"
 
-extern int cppgetopt(int, char *const *, const char *);
-extern char *optarg;
-extern int optind;
 extern int Cplusplus;
 Nlist *kwdefined;
 char wd[128];
diff --git a/idlc/source/preproc/unix.c b/idlc/source/preproc/unix.c
--- idlc/source/preproc/unix.c
+++ idlc/source/preproc/unix.c
@@ -33,15 +33,14 @@
 #if (defined(_WIN32) || defined(_MSDOS) || defined(__IBMC__))
 #include <io.h>
 #include <sys/stat.h>
+#include <external/glibc/getopt.h>
 #else
 #include <unistd.h>
 #endif
 
 #include "cpp.h"
 
-extern int cppgetopt(int, char *const *, const char *);
-extern char *optarg, rcsid[];
-extern int optind;
+extern char rcsid[];
 
 int Pflag = 0;                          /* print no line information */
 int Iflag = 0;							/* print includes */
@@ -62,7 +61,7 @@
     Tokenrow tr;
 
     setup_kwtab();
-    while ((c = cppgetopt(argc, argv, "NOPV:I:D:U:F:A:X:u:l:+")) != -1)
+    while ((c = getopt(argc, argv, "NOPV:I:D:U:F:A:X:u:l:+")) != -1)
         switch (c)
         {
             case 'N':
diff --git a/set_soenv.in b/set_soenv.in
--- set_soenv.in
+++ set_soenv.in
@@ -1875,6 +1875,9 @@
 ToFile( "HUNSPELL_CFLAGS",   "@HUNSPELL_CFLAGS@",  "e" );
 ToFile( "HUNSPELL_LIBS",     "@HUNSPELL_LIBS@",    "e" );
 ToFile( "SYSTEM_LPSOLVE",    "@SYSTEM_LPSOLVE@",   "e" );
+ToFile( "HAVE_GETOPT",       "@HAVE_GETOPT@",      "e" );
+ToFile( "HAVE_READDIR_R",    "@HAVE_READDIR_R@",   "e" );
+ToFile( "SYSTEM_LIBC",       "@SYSTEM_LIBC@",      "e" );
 ToFile( "SYSTEM_HYPH",       "@SYSTEM_HYPH@",      "e" );
 ToFile( "HYPHEN_LIB",        "@HYPHEN_LIB@",       "e" );
 ToFile( "SYSTEM_MYTHES",     "@SYSTEM_MYTHES@",    "e" );
diff --git a/soltools/cpp/_nlist.c b/soltools/cpp/_nlist.c
--- soltools/cpp/_nlist.c
+++ soltools/cpp/_nlist.c
@@ -3,9 +3,6 @@
 #include <string.h>
 #include "cpp.h"
 
-extern int stgetopt(int, char *const *, const char *);
-extern char *optarg;
-extern int optind;
 extern int Cplusplus;
 Nlist *kwdefined;
 char wd[128];
diff --git a/soltools/cpp/_unix.c b/soltools/cpp/_unix.c
--- soltools/cpp/_unix.c
+++ soltools/cpp/_unix.c
@@ -12,9 +12,15 @@
 
 #include "cpp.h"
 
+#if defined MACOSX || !defined HAVE_GETOPT
 extern int stgetopt(int, char *const *, const char *);
-extern char *optarg, rcsid[];
+extern char *optarg;
 extern int optind;
+#else
+#include <getopt.h>
+#endif
+
+extern char rcsid[];
 
 int Pflag = 0;                          /* print no line information */
 int Iflag = 0;							/* print includes */
@@ -36,7 +42,11 @@
     Tokenrow tr;
 
     setup_kwtab();
+#if defined MACOSX || !defined HAVE_GETOPT
     while ((c = stgetopt(argc, argv, "NOPV:I:D:U:F:A:X:u:l:+")) != -1)
+#else
+    while ((c = getopt(argc, argv, "NOPV:I:D:U:F:A:X:u:l:+")) != -1)
+#endif
         switch (c)
         {
             case 'N':
diff --git a/soltools/cpp/makefile.mk b/soltools/cpp/makefile.mk
--- soltools/cpp/makefile.mk
+++ soltools/cpp/makefile.mk
@@ -45,7 +45,6 @@
 OBJFILES= \
 	$(OBJ)$/_cpp.obj		\
 	$(OBJ)$/_eval.obj 	\
-	$(OBJ)$/_getopt.obj	\
 	$(OBJ)$/_include.obj \
 	$(OBJ)$/_lex.obj 	\
 	$(OBJ)$/_macro.obj 	\
@@ -54,6 +53,15 @@
 	$(OBJ)$/_tokens.obj 	\
 	$(OBJ)$/_unix.obj
 
+# nonstandard cpp options needed for Mac (-isysroot),
+# needs the custom stgetopt defined here :/
+.IF "$(OS)" == "MACOSX" || "$(HAVE_GETOPT)" == "NO"
+OBJFILES += $(OBJ)$/_getopt.obj
+.ENDIF
+.IF "$(HAVE_GETOPT)" == "YES"
+CDEFS += -DHAVE_GETOPT
+.ENDIF
+
 APP1TARGET	=	$(TARGET)
 .IF "$(GUI)" != "OS2"
 APP1STACK	=	1000000
diff --git a/soltools/javadep/javadep.c b/soltools/javadep/javadep.c
--- soltools/javadep/javadep.c
+++ soltools/javadep/javadep.c
@@ -138,12 +138,14 @@
 void    err_quit(const char *, ...);
 void    silent_quit(void);
 
+#ifdef WNT
 /* poor man's getopt() */
 int     simple_getopt(char *pargv[], const char *poptstring);
 char    *optarg = NULL;
 int     optind  = 1;
 int     optopt  = 0;
 int     opterr  = 0;
+#endif
 
 uint8
 read_uint8(const file_t *pfile)
@@ -748,7 +750,7 @@
             char *popt;
             int c = parg[1];
             if ( (popt = strchr(poptstring, c)) == NULL ) {
-                optopt = c;
+               optopt = c;
                 if ( opterr )
                     fprintf(stderr, "Unknown option character `\\x%x'.\n", optopt);
                 return '?';
@@ -769,7 +771,6 @@
     return -1;
 }
 
-
 int CDECL
 main(int argc, char *argv[])
 {
@@ -817,7 +818,11 @@
     opterr = 0;
     pincs = allocate_growable();
 
-    while( (c = simple_getopt(pall_argv, ":i:I:s:S:o:OhHvV")) != -1 ) {
+#ifdef WNT
+	while( (c = simple_getopt(pall_argv, ":i:I:s:S:o:OhHvV")) != -1 ) {
+#else
+    while( (c = getopt(nall_argc, pall_argv, ":i:I:s:S:o:OhHvV")) != -1 ) {
+#endif
         switch(c) {
             case 'i':
             case 'I':
diff --git a/tools/bootstrp/makefile.mk b/tools/bootstrp/makefile.mk
--- tools/bootstrp/makefile.mk
+++ tools/bootstrp/makefile.mk
@@ -38,6 +38,10 @@
 
 CDEFS+=-D_TOOLS_STRINGLIST
 
+.IF "$(HAVE_GETOPT)" == "YES"
+CDEFS += -DHAVE_GETOPT
+.ENDIF
+
 # --- Files --------------------------------------------------------
 
 OBJFILES= \
@@ -83,6 +87,11 @@
 APP2OBJS=	$(OBJ)$/rscdep.obj
 APP2LIBS=   $(LB)$/$(TARGET).lib $(LB)$/$(TARGET1).lib
 APP2STDLIBS= $(SALLIB) $(VOSLIB) $(TOOLSLIB) $(BASEGFXLIB) $(UCBHELPERLIB) $(CPPULIB) $(COMPHELPERLIB) $(I18NISOLANGLIB) $(CPPUHELPERLIB) $(SALHELPERLIB)
+.IF "$(HAVE_GETOPT)" != "YES"
+.IF "$(OS)"=="WNT"
+APP2STDLIBS+=gnu_getopt.lib
+.ENDIF
+.ENDIF
 APP2RPATH=  NONE
 APP2RPATH=  NONE
 APP2RPATH=  NONE
diff --git a/tools/bootstrp/rscdep.cxx b/tools/bootstrp/rscdep.cxx
--- tools/bootstrp/rscdep.cxx
+++ tools/bootstrp/rscdep.cxx
@@ -45,6 +45,16 @@
 
 #include "cppdep.hxx"
 
+#if defined WNT 
+#if !defined HAVE_GETOPT
+#define __STDC__ 1
+#define __GNU_LIBRARY__
+#include <external/glibc/getopt.h>
+#else
+#include <getopt.h>
+#endif
+#endif
+
 class RscHrcDep : public CppDep
 {
 public:
@@ -70,15 +80,6 @@
 
 //static String aDelim;
 
-/* poor man's getopt() */
-int     simple_getopt(char *argv[], const char *optstring);
-#if defined(WNT) || defined(OS2)
-static char	*optarg = NULL;
-static int	optind = 1; 
-static int	optopt = 0; 
-static int	opterr = 0; 
-#endif
-
 SAL_IMPLEMENT_MAIN_WITH_ARGS( argc, argv )
 {
 	int c;
@@ -179,7 +180,7 @@
 
 	while( 1 )
 	{
-		c = simple_getopt( argv,
+		c = getopt( argc, argv,
 		"_abcdefghi:jklmnopqrstuvwxyzABCDEFGHI:JKLMNOPQRSTUVWXYZ1234567890/-+=.\\()\"");
 		if ( c == -1 )
 			break;
@@ -296,41 +297,3 @@
 	return 0;
 }
 
-/* my very simple minded implementation of getopt()
- * it's too sad that getopt() is not available everywhere
- * note: this is not a full POSIX conforming getopt()
- */
-int simple_getopt(char *argv[], const char *optstring)
-{
-    char *arg = argv[optind];
-
-    /* skip all response file arguments */
-    if ( arg ) {
-		while ( *arg == '@' )
-        	arg = argv[++optind];
-        
-    	if ( arg[0] == '-' && arg[1] != '\0' ) {
-        	const char *popt;
-        	int c = arg[1];
-        	if ( (popt = strchr(optstring, c)) == NULL ) {
-            	optopt = c;
-            	if ( opterr )
-                	fprintf(stderr, "Unknown option character `\\x%x'.\n", optopt);
-            	return '?';
-        	}
-        	if ( *(++popt) == ':') {
-             	if ( arg[2] != '\0' ) {
-                 	optarg = ++arg;
-             	} else {
-                 	optarg = argv[++optind];
-             	}
-         	} else {
-             	optarg = NULL;
-         	}
-         	++optind;
-         	return c;
-    	}
-	}
-    return -1;
-}
-
