From 317136fc9a5d44f6cab2cc2bb8c58130782c55ac Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:52:57 +0200
Subject: [PATCH 027/878] cws-koheiextref01-oox.diff

---
 oox/source/xls/externallinkbuffer.cxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/oox/source/xls/externallinkbuffer.cxx b/oox/source/xls/externallinkbuffer.cxx
index a4c0731..04fcd2f 100644
--- a/oox/source/xls/externallinkbuffer.cxx
+++ b/oox/source/xls/externallinkbuffer.cxx
@@ -909,7 +909,7 @@ void ExternalLink::insertExternalSheet( const OUString& rSheetName )
     OSL_ENSURE( rSheetName.getLength() > 0, "ExternalLink::insertExternalSheet - empty sheet name" );
     if( mxDocLink.is() )
     {
-        Reference< XExternalSheetCache > xSheetCache = mxDocLink->addSheetCache( rSheetName );
+        Reference< XExternalSheetCache > xSheetCache = mxDocLink->addSheetCache( rSheetName, false );
         sal_Int32 nCacheIdx = xSheetCache.is() ? xSheetCache->getTokenIndex() : -1;
         maSheetCaches.push_back( nCacheIdx );
     }
-- 
1.7.0.1

