diff --git oox/source/xls/externallinkbuffer.cxx oox/source/xls/externallinkbuffer.cxx
index c756252..5006204 100644
--- oox/source/xls/externallinkbuffer.cxx
+++ oox/source/xls/externallinkbuffer.cxx
@@ -912,7 +912,7 @@ void ExternalLink::insertExternalSheet( const OUString& rSheetName )
     OSL_ENSURE( rSheetName.getLength() > 0, "ExternalLink::insertExternalSheet - empty sheet name" );
     if( mxDocLink.is() )
     {
-        Reference< XExternalSheetCache > xSheetCache = mxDocLink->addSheetCache( rSheetName );
+        Reference< XExternalSheetCache > xSheetCache = mxDocLink->addSheetCache( rSheetName, false );
         sal_Int32 nCacheIdx = xSheetCache.is() ? xSheetCache->getTokenIndex() : -1;
         maSheetCaches.push_back( nCacheIdx );
     }
