diff --git officecfg/registry/data/org/openoffice/Office/UI/CalcCommands.xcu officecfg/registry/data/org/openoffice/Office/UI/CalcCommands.xcu
index 991ea9a..3ae5ca8 100644
--- officecfg/registry/data/org/openoffice/Office/UI/CalcCommands.xcu
+++ officecfg/registry/data/org/openoffice/Office/UI/CalcCommands.xcu
@@ -1243,12 +1243,25 @@
             </node>
             <node oor:name=".uno:Name" oor:op="replace">
                 <prop oor:name="Label" oor:type="xs:string">
-                    <value xml:lang="en-US">Rename Sheet</value>
+                    <value xml:lang="en-US">Rename Sheet</value>
+                </prop>
+            </node>
+            <node oor:name=".uno:SetTabBgColor" oor:op="replace">
+                <prop oor:name="Label" oor:type="xs:string">
+                    <value xml:lang="en-US">~Tab Color...</value>
+                </prop>
+                <prop oor:name="ContextLabel" oor:type="xs:string">
+                    <value xml:lang="en-US">~Tab Color...</value>
                 </prop>
             </node>
-            <node oor:name=".uno:Move" oor:op="replace">
+            <node oor:name=".uno:TabBgColor" oor:op="replace">
                 <prop oor:name="Label" oor:type="xs:string">
-                    <value xml:lang="en-US">~Move/Copy Sheet...</value>
+                    <value xml:lang="en-US">Tab Color</value>
+                </prop>
+            </node>
+            <node oor:name=".uno:Move" oor:op="replace">
+                <prop oor:name="Label" oor:type="xs:string">
+                    <value xml:lang="en-US">~Move/Copy Sheet...</value>
                 </prop>
                 <prop oor:name="ContextLabel" oor:type="xs:string">
                     <value xml:lang="en-US">~Move/Copy...</value>
