diff --git testautomation/global/win/tab_a_d.win testautomation/global/win/tab_a_d.win
index af6d933..a3135d8 100755
--- testautomation/global/win/tab_a_d.win
+++ testautomation/global/win/tab_a_d.win
@@ -151,6 +151,7 @@ GenauigkeitWieAngezeigt sc:CheckBox:RID_SCPAGE_CALC:BTN_CALC
 Suchkriterien sc:CheckBox:RID_SCPAGE_CALC:BTN_MATCH
 RegulaererAusdruck sc:CheckBox:RID_SCPAGE_CALC:BTN_REGEX
 SpaltenZeilenbeschriftung sc:CheckBox:RID_SCPAGE_CALC:BTN_LOOKUP
+LimitDecimals sc:CheckBox:RID_SCPAGE_CALC:BTN_GENERAL_PREC
 
 *TabBereiche HID_INSERT_SECTION_PAGE
 Bereichsliste sw:ComboBox:TP_INSERT_SECTION:ED_RNAME
diff --git testautomation/spreadsheet/optional/includes/arrayconstants/c_arrayconstants.inc testautomation/spreadsheet/optional/includes/arrayconstants/c_arrayconstants.inc
index cdb4691..20fd979 100644
--- testautomation/spreadsheet/optional/includes/arrayconstants/c_arrayconstants.inc
+++ testautomation/spreadsheet/optional/includes/arrayconstants/c_arrayconstants.inc
@@ -92,17 +92,17 @@ testcase tArrayconstants01
     kontext "DocumentCalc"
     DocumentCalc.TypeKeys ("=" & sFunctionSIN & "({1;2;3}) <MOD1 SHIFT RETURN>")
     '///Check that the cellvalue  of A4 is "0.84"
-    printlog "Check that the cellvalue  of A4 is ""0.84"""
+    printlog "Check that the cellvalue  of A4 is ""0.841470984807896"""
     call fCalcSelectRange ("A5")
     kontext "DocumentCalc"
     DocumentCalc.TypeKeys "=A4<TAB>=B4<TAB>=C4<RETURN>" 'because a part of a matrix is protected we need a helpcell with only the values
-    call fCalcCompareCellValue ("A5","0" & sDecimalseperator & "84")
+    call fCalcCompareCellValue ("A5","0" & sDecimalseperator & "841470984807896")
     '///Check that the cell contents of B4 is "0.91"
-    printlog "Check that the cell contents of B4 is ""0.91"""
-    call fCalcCompareCellValue ("B5","0" & sDecimalseperator & "91")
+    printlog "Check that the cell contents of B4 is ""0.909297426825682"""
+    call fCalcCompareCellValue ("B5","0" & sDecimalseperator & "909297426825682")
     '///Check that the cell contents of C4 is "0.14"
-    printlog "Check that the cell contents of C4 is ""0.14"""
-    call fCalcCompareCellValue ("C5","0" & sDecimalseperator & "14")
+    printlog "Check that the cell contents of C4 is ""0.141120008059867"""
+    call fCalcCompareCellValue ("C5","0" & sDecimalseperator & "141120008059867")
 
     '///In cell A6 enter "={1;2|4;5;6}"
     printlog "In cell A6 enter ""={1;2|4;5;6}"""
diff --git testautomation/spreadsheet/optional/includes/solver/c_solver.inc testautomation/spreadsheet/optional/includes/solver/c_solver.inc
index 623b43e..b7c69cb 100644
--- testautomation/spreadsheet/optional/includes/solver/c_solver.inc
+++ testautomation/spreadsheet/optional/includes/solver/c_solver.inc
@@ -99,8 +99,8 @@ testcase tExampleCalculation
 	call fCalcCompareCellValue ("C15","4" & sDecimalseperator & "0000 mg")
 	call fCalcCompareCellValue ("D15","18" & sDecimalseperator & "0000 mg")
 	call fCalcCompareCellValue ("E15","22" & sDecimalseperator & "0000 ct")
-	call fCalcCompareCellValue ("G13","120")
-	call fCalcCompareCellValue ("G14","180")
+	call fCalcCompareCellValue ("G13","120" & sDecimalseperator & "00")
+	call fCalcCompareCellValue ("G14","180" & sDecimalseperator & "00")
 	call fCalcCompareCellValue ("G15","300")
 	printlog " Tools - Solver"
 	ToolsSolver
@@ -116,8 +116,8 @@ testcase tExampleCalculation
 	call fCalcCompareCellValue ("C15","4" & sDecimalseperator & "0000 mg")
 	call fCalcCompareCellValue ("D15","51" & sDecimalseperator & "3333 mg")
 	call fCalcCompareCellValue ("E15","68" & sDecimalseperator & "6667 ct")
-	call fCalcCompareCellValue ("G13","520")
-	call fCalcCompareCellValue ("G14","-20")
+	call fCalcCompareCellValue ("G13","520" & sDecimalseperator & "00")
+	call fCalcCompareCellValue ("G14","-20" & sDecimalseperator & "00")
 	call fCalcCompareCellValue ("G15","500")
 	printlog " So the solution found is correct but not usefull. Let's limit the lemonade to positive values by adding a new limitation."
 	printlog " Tools - Solver"
