Index: desktop/prj/d.lst
===================================================================
RCS file: /cvs/framework/desktop/prj/d.lst,v
retrieving revision 1.67.12.2
retrieving revision 1.67.12.2.2.1
diff -u -p -u -p -b -w -B -r1.67.12.2 -r1.67.12.2.2.1
--- desktop/prj/d.lst	11 Dec 2008 15:22:37 -0000	1.67.12.2
+++ desktop/prj/d.lst	7 Jan 2009 15:06:38 -0000	1.67.12.2.2.1
@@ -12,11 +12,11 @@ mkdir: %_DEST%\bin%_EXT%\odf4ms
 ..\%__SRC%\bin\soffice.bin %_DEST%\bin%_EXT%\soffice.bin
 ..\%__SRC%\bin\officeloader.exe %_DEST%\bin%_EXT%\soffice.exe
 ..\%__SRC%\bin\soffice %_DEST%\bin%_EXT%\soffice.bin
-..\%__SRC%\bin\soffice %_DEST%\bin%_EXT%\soffice
+..\%__SRC%\bin\soffice_mac %_DEST%\bin%_EXT%\soffice
 ..\%__SRC%\bin\so\soffice.bin %_DEST%\bin%_EXT%\so\soffice.bin
 ..\%__SRC%\bin\so\officeloader.exe %_DEST%\bin%_EXT%\so\soffice.exe
 ..\%__SRC%\bin\so\soffice %_DEST%\bin%_EXT%\so\soffice.bin
-..\%__SRC%\bin\so\soffice %_DEST%\bin%_EXT%\so\soffice
+..\%__SRC%\bin\so\soffice_mac %_DEST%\bin%_EXT%\so\soffice
 ..\%__SRC%\bin\sofficeapp.dll %_DEST%\bin%_EXT%\sofficeapp.dll
 ..\%__SRC%\lib\isofficeapp.lib %_DEST%\lib%_EXT%\isofficeapp.lib
 ..\%__SRC%\lib\libsofficeapp.dylib %_DEST%\lib%_EXT%\libsofficeapp.dylib
Index: desktop/util/makefile.mk
===================================================================
RCS file: /cvs/framework/desktop/util/makefile.mk,v
retrieving revision 1.82.12.1
retrieving revision 1.82.12.1.20.1
diff -u -p -u -p -b -w -B -r1.82.12.1 -r1.82.12.1.20.1
--- desktop/util/makefile.mk	27 Aug 2008 14:32:53 -0000	1.82.12.1
+++ desktop/util/makefile.mk	7 Jan 2009 15:06:38 -0000	1.82.12.1.20.1
@@ -210,6 +210,16 @@ ALLTAR : $(BIN)$/so$/soffice_so$(EXECPOS
 
 .ENDIF
 
+.IF "$(OS)" == "MACOSX"
+$(BIN)$/so$/soffice_mac$(EXECPOST) : $(APP1TARGETN)
+	$(COPY) $< $@
+	
+$(BIN)$/soffice_mac$(EXECPOST) : $(APP5TARGETN)
+	$(COPY) $< $@
+
+ALLTAR : $(BIN)$/so$/soffice_mac$(EXECPOST) $(BIN)$/soffice_mac$(EXECPOST)
+
+.ENDIF
 
 .IF "$(GUI)" == "WNT"
 
