diff --git oox/source/token/properties.txt oox/source/token/properties.txt
index 4d20b8f..4569d8c 100644
--- oox/source/token/properties.txt
+++ oox/source/token/properties.txt
@@ -355,6 +355,7 @@ Suffix
 SwapXAndYAxis
 Symbol
 SymbolColor
+TabColor
 TableBorder
 TableLayout
 TableSelected
diff --git oox/source/xls/worksheethelper.cxx oox/source/xls/worksheethelper.cxx
index b92050e..cb1c7bf 100644
--- oox/source/xls/worksheethelper.cxx
+++ oox/source/xls/worksheethelper.cxx
@@ -997,6 +997,8 @@ void WorksheetData::finalizeWorksheetImport()
     maCondFormats.finalizeImport();
     maPageSett.finalizeImport();
     maSheetViewSett.finalizeImport();
+    maSheetSett.finalizeImport();
+
     lclUpdateProgressBar( mxFinalProgress, 0.5 );
     convertColumns();
     convertRows();
diff --git oox/source/xls/worksheetsettings.cxx oox/source/xls/worksheetsettings.cxx
index b95bada..0a9885b 100644
--- oox/source/xls/worksheetsettings.cxx
+++ oox/source/xls/worksheetsettings.cxx
@@ -26,14 +26,18 @@
  ************************************************************************/
 
 #include "oox/xls/worksheetsettings.hxx"
-#include <com/sun/star/util/XProtectable.hpp>
 #include "oox/helper/attributelist.hxx"
 #include "oox/helper/recordinputstream.hxx"
 #include "oox/xls/biffinputstream.hxx"
 #include "oox/xls/pagesettings.hxx"
 #include "oox/xls/workbooksettings.hxx"
+#include "properties.hxx"
+
+#include <com/sun/star/util/XProtectable.hpp>
 
 using ::rtl::OUString;
+using ::com::sun::star::beans::XPropertySet;
+using ::com::sun::star::uno::Any;
 using ::com::sun::star::uno::Exception;
 using ::com::sun::star::uno::Reference;
 using ::com::sun::star::uno::UNO_QUERY_THROW;
@@ -308,6 +312,13 @@ void WorksheetSettings::finalizeImport()
     catch( Exception& )
     {
     }
+
+    if (!maSheetSettings.maTabColor.isAuto())
+    {
+        sal_Int32 nColor = maSheetSettings.maTabColor.getColor(getBaseFilter());
+        PropertySet aSheetProp(getSheet());
+        aSheetProp.setProperty(PROP_TabColor, nColor);
+    }
 }
 
 // ============================================================================
