diff --git xmloff/inc/xmloff/xmltoken.hxx xmloff/inc/xmloff/xmltoken.hxx
index 44c5f36..ec9101f 100644
--- xmloff/inc/xmloff/xmltoken.hxx
+++ xmloff/inc/xmloff/xmltoken.hxx
@@ -1705,6 +1705,7 @@ namespace xmloff { namespace token {
         XML_SYMBOL_IMAGE_NAME,
         XML_SYMBOL_WIDTH,
         XML_SYSTEM,
+        XML_TAB_COLOR,
         XML_TAB_STOP,
         XML_TAB_STOP_DISTANCE,
         XML_TAB_STOPS,
diff --git xmloff/source/core/xmltoken.cxx xmloff/source/core/xmltoken.cxx
index 7913bde..f802d89 100644
--- xmloff/source/core/xmltoken.cxx
+++ xmloff/source/core/xmltoken.cxx
@@ -1713,6 +1713,7 @@ namespace xmloff { namespace token {
         TOKEN( "symbol-image-name",               XML_SYMBOL_IMAGE_NAME ),
         TOKEN( "symbol-width",                    XML_SYMBOL_WIDTH ),
         TOKEN( "system",                          XML_SYSTEM ),
+        TOKEN( "tab-color",                       XML_TAB_COLOR ),
         TOKEN( "tab-stop",                        XML_TAB_STOP ),
         TOKEN( "tab-stop-distance",               XML_TAB_STOP_DISTANCE ),
         TOKEN( "tab-stops",                       XML_TAB_STOPS ),
