Index: comphelper/source/misc/mediadescriptor.cxx
===================================================================
RCS file: /cvs/util/comphelper/source/misc/mediadescriptor.cxx,v
retrieving revision 1.20
retrieving revision 1.20.36.1
diff -u -p -u -p -b -w -B -r1.20 -r1.20.36.1
--- comphelper/source/misc/mediadescriptor.cxx	6 Jun 2008 14:36:10 -0000	1.20
+++ comphelper/source/misc/mediadescriptor.cxx	28 Oct 2008 20:23:19 -0000	1.20.36.1
@@ -524,18 +524,15 @@ sal_Bool MediaDescriptor::addInputStream
 			return impl_openStreamWithPostData( xPostData );
 		}
 
-        // b) is there a reference to a file which is just being salvaged?
-        ::rtl::OUString sFileURL = getUnpackedValueOrDefault(MediaDescriptor::PROP_SALVAGEDFILE(), ::rtl::OUString());
-		// c) finally, the last resort is the URL property
-        if ( !sFileURL.getLength() )
-            sFileURL = getUnpackedValueOrDefault(MediaDescriptor::PROP_URL(), ::rtl::OUString());
-		if (!sFileURL.getLength())
+		// b) ... or we must get it from the given URL
+		::rtl::OUString sURL = getUnpackedValueOrDefault(MediaDescriptor::PROP_URL(), ::rtl::OUString());
+		if (!sURL.getLength())
 			throw css::uno::Exception(
 					::rtl::OUString::createFromAscii("Found no URL."),
 					css::uno::Reference< css::uno::XInterface >());
 
         // Parse URL! Only the main part has to be used further. E.g. a jumpmark can make trouble
-        ::rtl::OUString sNormalizedURL = impl_normalizeURL( sFileURL );
+        ::rtl::OUString sNormalizedURL = impl_normalizeURL( sURL );
 		return impl_openStreamWithURL( sNormalizedURL, bLockFile );
 	}
 #if OSL_DEBUG_LEVEL>0
Index: scp2/source/ooo/file_library_ooo.scp
===================================================================
RCS file: /cvs/installation/scp2/source/ooo/file_library_ooo.scp,v
retrieving revision 1.303.2.3
retrieving revision 1.303.2.3.20.1
diff -u -p -u -p -b -w -B -r1.303.2.3 -r1.303.2.3.20.1
--- scp2/source/ooo/file_library_ooo.scp	21 Oct 2008 13:18:37 -0000	1.303.2.3
+++ scp2/source/ooo/file_library_ooo.scp	3 Nov 2008 13:59:35 -0000	1.303.2.3.20.1
@@ -200,7 +200,7 @@ File gid_File_Lib_Comphelper2
     Name = STRING(CONCAT3(comphelp4,COMID,.dll));
   #endif
     Dir = gid_Dir_Program;
-    Styles = (PACKED,UNO_COMPONENT);
+    Styles = (PACKED,UNO_COMPONENT,PATCH);
     RegistryID = gid_Starregistry_Services_Rdb;
 End
 
