From 940f15515d57efea5c27d8e2a838541b448ff1d5 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:57:51 +0200
Subject: [PATCH 284/878] db4.8.diff

---
 configure.in |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/configure.in b/configure.in
index e79b1be..eed386e 100644
--- a/configure.in
+++ b/configure.in
@@ -3983,7 +3983,7 @@ if test -n "$with_system_db" -o -n "$with_system_libs" && \
         ], []
     )
     AC_MSG_CHECKING([whether db is at least 4.1])
-    for v in `seq 1 7`; do
+    for v in `seq 1 8`; do
        AC_TRY_RUN([
 #include <db.h>
 
-- 
1.7.0.1

