From 83d239235dcf368b5b0623c6875a33283227b025 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:53:08 +0200
Subject: [PATCH 023/768] dbaccess-default-varchar-lenght.diff

---
 .../source/ui/tabledesign/FieldDescriptions.cxx    |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git dbaccess/source/ui/tabledesign/FieldDescriptions.cxx dbaccess/source/ui/tabledesign/FieldDescriptions.cxx
index 1b004d8..81fde57 100644
--- dbaccess/source/ui/tabledesign/FieldDescriptions.cxx
+++ dbaccess/source/ui/tabledesign/FieldDescriptions.cxx
@@ -58,7 +58,7 @@
 #include <com/sun/star/util/NumberFormat.hpp>
 #endif
 
-#define DEFAULT_VARCHAR_PRECSION	50
+#define DEFAULT_VARCHAR_PRECSION	100
 #define DEFAULT_OTHER_PRECSION		16
 #define DEFAULT_NUMERIC_PRECSION	5
 #define DEFAULT_NUMERIC_SCALE		0
-- 
1.7.0.1

