---
 .../source/ui/tabledesign/FieldDescriptions.cxx    |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git dbaccess/source/ui/tabledesign/FieldDescriptions.cxx dbaccess/source/ui/tabledesign/FieldDescriptions.cxx
index 77d7ccf..785769f 100644
--- dbaccess/source/ui/tabledesign/FieldDescriptions.cxx
+++ dbaccess/source/ui/tabledesign/FieldDescriptions.cxx
@@ -58,7 +58,7 @@
 #include <com/sun/star/util/NumberFormat.hpp>
 #endif
 
-#define DEFAULT_VARCHAR_PRECSION	50
+#define DEFAULT_VARCHAR_PRECSION	100
 #define DEFAULT_OTHER_PRECSION		16
 #define DEFAULT_NUMERIC_PRECSION	5
 #define DEFAULT_NUMERIC_SCALE		0
-- 
1.7.0.1

