diff --git cui/source/dialogs/linkdlg.cxx cui/source/dialogs/linkdlg.cxx
index 965aefe..a689d85 100644
--- cui/source/dialogs/linkdlg.cxx
+++ cui/source/dialogs/linkdlg.cxx
@@ -330,6 +330,8 @@ IMPL_LINK( SvBaseLinksDlg, UpdateNowClickHdl, PushButton *, EMPTYARG )
             rListBox.Select( pE );
             rListBox.MakeVisible( pE );
         }
+
+        pNewMgr->CloseCachedComps();
     }
     return 0;
 }
