diff --git sfx2/inc/sfx2/objsh.hxx sfx2/inc/sfx2/objsh.hxx
index a1396a3..4faccea 100644
--- sfx2/inc/sfx2/objsh.hxx
+++ sfx2/inc/sfx2/objsh.hxx
@@ -528,6 +528,9 @@ public:
                                             const String& rMimeType,
                                 const ::com::sun::star::uno::Any & rValue );
     virtual ::sfx2::SvLinkSource* 		DdeCreateLinkSource( const String& rItem );
+    virtual void                ReconnectDdeLink(SfxObjectShell& rServer);
+
+    static void                 ReconnectDdeLinks(SfxObjectShell& rServer);
 
     // Contents
     virtual SfxStyleSheetBasePool*	GetStyleSheetPool();
diff --git sfx2/source/appl/appdde.cxx sfx2/source/appl/appdde.cxx
index 3cc38c9..ce4fe23 100644
--- sfx2/source/appl/appdde.cxx
+++ sfx2/source/appl/appdde.cxx
@@ -365,6 +365,23 @@ long SfxObjectShell::DdeSetData
     return 0;
 }
 
+void SfxObjectShell::ReconnectDdeLink(SfxObjectShell& /*rServer*/)
+{
+}
+
+void SfxObjectShell::ReconnectDdeLinks(SfxObjectShell& rServer)
+{
+    TypeId aType = TYPE(SfxObjectShell);
+    SfxObjectShell* p = GetFirst(&aType, false);
+    while (p)
+    {
+        if (&rServer != p)
+            p->ReconnectDdeLink(rServer);
+
+        p = GetNext(*p, &aType, false);
+    }
+}
+
 //========================================================================
 
 long SfxViewFrame::DdeExecute
diff --git sfx2/source/doc/objstor.cxx sfx2/source/doc/objstor.cxx
index 89e1618..4f4c540 100644
--- sfx2/source/doc/objstor.cxx
+++ sfx2/source/doc/objstor.cxx
@@ -1011,6 +1011,8 @@ sal_Bool SfxObjectShell::DoLoad( SfxMedium *pMed )
             }
         }
 
+        ReconnectDdeLinks(*this);
+
 #if 0
         if ( pMedium->HasStorage_Impl() )
         {
