diff --git sw/inc/docsh.hxx sw/inc/docsh.hxx
index 6c2fd87..7a47dc2 100644
--- sw/inc/docsh.hxx
+++ sw/inc/docsh.hxx
@@ -238,6 +238,8 @@ public:
     virtual long DdeSetData( const String& rItem, const String& rMimeType,
                                 const ::com::sun::star::uno::Any & rValue );
     virtual ::sfx2::SvLinkSource* DdeCreateLinkSource( const String& rItem );
+    virtual void ReconnectDdeLink(SfxObjectShell& rServer);
+
     virtual void FillClass( SvGlobalName * pClassName,
                                    sal_uInt32 * pClipFormat,
                                    String * pAppName,
diff --git sw/source/ui/app/docsh2.cxx sw/source/ui/app/docsh2.cxx
index aa59379..3022ba2 100644
--- sw/source/ui/app/docsh2.cxx
+++ sw/source/ui/app/docsh2.cxx
@@ -64,6 +64,7 @@
 #include <editeng/svxacorr.hxx>
 #include <editeng/langitem.hxx>
 #include <svx/fmshell.hxx>
+#include <sfx2/linkmgr.hxx>
 
 #include <svtools/htmlcfg.hxx>
 #include <svx/ofaitem.hxx>
@@ -1564,6 +1565,12 @@ long SwDocShell::DdeSetData( const String& rItem, const String& rMimeType,
     return pDoc->CreateLinkSource( rItem );
 }
 
+void SwDocShell::ReconnectDdeLink(SfxObjectShell& rServer)
+{
+    ::sfx2::LinkManager& rLinkManager = pDoc->GetLinkManager();
+    rLinkManager.ReconnectDdeLink(rServer);
+}
+
 /*--------------------------------------------------------------------
     Beschreibung:
  --------------------------------------------------------------------*/
