diff --git sc/source/core/data/documen8.cxx sc/source/core/data/documen8.cxx
index 1daa520..b210025 100644
--- sc/source/core/data/documen8.cxx
+++ sc/source/core/data/documen8.cxx
@@ -1097,6 +1097,7 @@ void ScDocument::UpdateDdeLinks()
             if (pBase->ISA(ScDdeLink))
                 ((ScDdeLink*)pBase)->TryUpdate();       // bei DDE-Links TryUpdate statt Update
         }
+        pLinkManager->CloseCachedComps();
     }
 }
 
@@ -1126,6 +1127,7 @@ BOOL ScDocument::UpdateDdeLink( const String& rAppl, const String& rTopic, const
                 }
             }
         }
+        pLinkManager->CloseCachedComps();
     }
     return bFound;
 }
diff --git sc/source/core/tool/interpr2.cxx sc/source/core/tool/interpr2.cxx
index 09b6408..b560f93 100644
--- sc/source/core/tool/interpr2.cxx
+++ sc/source/core/tool/interpr2.cxx
@@ -2218,6 +2218,7 @@ void ScInterpreter::ScDde()
             PushNA();
 
         pDok->DisableIdle( bOldDis );
+        pLinkMgr->CloseCachedComps();
     }
 }
 
diff --git sc/source/ui/docshell/docsh6.cxx sc/source/ui/docshell/docsh6.cxx
index bb7b82f..8d9a9f0 100644
--- sc/source/ui/docshell/docsh6.cxx
+++ sc/source/ui/docshell/docsh6.cxx
@@ -393,6 +393,15 @@ BOOL __EXPORT ScDocShell::Insert( SfxObjectShell &rSource,
     return bRet;
 }
 
+void ScDocShell::ReconnectDdeLink(SfxObjectShell& rServer)
+{
+    ::sfx2::LinkManager* pLinkManager = aDocument.GetLinkManager();
+    if (!pLinkManager)
+        return;
+
+    pLinkManager->ReconnectDdeLink(rServer);
+}
+
 void ScDocShell::UpdateLinks()
 {
     sfx2::LinkManager* pLinkManager = aDocument.GetLinkManager();
diff --git sc/source/ui/docshell/servobj.cxx sc/source/ui/docshell/servobj.cxx
index 1d69e26..4790d68 100644
--- sc/source/ui/docshell/servobj.cxx
+++ sc/source/ui/docshell/servobj.cxx
@@ -41,6 +41,8 @@
 #include "rangenam.hxx"
 #include "sc.hrc"				// SC_HINT_AREAS_CHANGED
 
+using namespace formula;
+
 // -----------------------------------------------------------------------
 
 BOOL lcl_FillRangeFromName( ScRange& rRange, ScDocShell* pDocSh, const String& rName )
@@ -97,11 +99,13 @@ ScServerObject::ScServerObject( ScDocShell* pShell, const String& rItem ) :
         SCTAB nTab = pDocSh->GetCurTab();
         aRange.aStart.SetTab( nTab );
 
-        if ( aRange.Parse( rItem, pDoc ) & SCA_VALID )
+        // For DDE link, we always must parse references using OOO A1 convention.
+
+        if ( aRange.Parse( rItem, pDoc, FormulaGrammar::CONV_OOO ) & SCA_VALID )
         {
             // area reference
         }
-        else if ( aRange.aStart.Parse( rItem, pDoc, pDoc->GetAddressConvention() ) & SCA_VALID )
+        else if ( aRange.aStart.Parse( rItem, pDoc, FormulaGrammar::CONV_OOO ) & SCA_VALID )
         {
             // cell reference
             aRange.aEnd = aRange.aStart;
diff --git sc/source/ui/inc/docsh.hxx sc/source/ui/inc/docsh.hxx
index d9d3629..7df6e9f 100755
--- sc/source/ui/inc/docsh.hxx
+++ sc/source/ui/inc/docsh.hxx
@@ -324,6 +324,7 @@ public:
     ScDBData*       GetOldAutoDBRange();    // has to be deleted by caller!
     void            CancelAutoDBRange();    // called when dialog is cancelled
 
+    virtual void    ReconnectDdeLink(SfxObjectShell& rServer);
     void			UpdateLinks();			// Link-Eintraege aktuallisieren
     BOOL			ReloadTabLinks();		// Links ausfuehren (Inhalt aktualisieren)
 
