diff --git sw/inc/docsh.hxx sw/inc/docsh.hxx
index eb116ef..f64d7ef 100644
--- sw/inc/docsh.hxx
+++ sw/inc/docsh.hxx
@@ -235,6 +235,8 @@ public:
     virtual long DdeSetData( const String& rItem, const String& rMimeType,
                                 const ::com::sun::star::uno::Any & rValue );
     virtual ::sfx2::SvLinkSource* DdeCreateLinkSource( const String& rItem );
+    virtual void ReconnectDdeLink(SfxObjectShell& rServer);
+
     virtual void FillClass( SvGlobalName * pClassName,
                                    sal_uInt32 * pClipFormat,
                                    String * pAppName,
diff --git sw/source/ui/app/docsh2.cxx sw/source/ui/app/docsh2.cxx
index 8ff7f79..7fd0d25 100644
--- sw/source/ui/app/docsh2.cxx
+++ sw/source/ui/app/docsh2.cxx
@@ -75,6 +75,7 @@
 #include <svx/svxacorr.hxx>
 #include <svx/langitem.hxx>
 #include <svx/fmshell.hxx>
+#include <svx/linkmgr.hxx>
 
 #include <svx/htmlcfg.hxx>
 #include <svx/ofaitem.hxx>
@@ -1607,6 +1608,12 @@ long SwDocShell::DdeSetData( const String& rItem, const String& rMimeType,
     return pDoc->CreateLinkSource( rItem );
 }
 
+void SwDocShell::ReconnectDdeLink(SfxObjectShell& rServer)
+{
+    SvxLinkManager& rLinkManager = pDoc->GetLinkManager();
+    rLinkManager.ReconnectDdeLink(rServer);
+}
+
 /*--------------------------------------------------------------------
     Beschreibung:
  --------------------------------------------------------------------*/
diff --git sw/source/ui/uno/unotxdoc.cxx sw/source/ui/uno/unotxdoc.cxx
index 61c78bf..e4b71bf 100644
--- sw/source/ui/uno/unotxdoc.cxx
+++ sw/source/ui/uno/unotxdoc.cxx
@@ -2407,6 +2407,7 @@ void SwXTextDocument::updateLinks(  ) throw(RuntimeException)
     {
         UnoActionContext aAction(pDoc);
         rLnkMan.UpdateAllLinks( FALSE, FALSE, TRUE );
+        rLnkMan.CloseCachedComps();
     }
 }
 //XPropertyState
