Index: mysqlc/source/makefile.mk
===================================================================
--- mysqlc/source/makefile.mk	(Revision 275542)
+++ mysqlc/source/makefile.mk	(Arbeitskopie)
@@ -113,10 +113,18 @@
 
 CDEFS+=-DCPPDBC_EXPORTS -DCPPCONN_LIB_BUILD
 .IF "$(SYSTEM_MYSQL_CPPCONN)" != "YES"
+.IF "$(USE_SYSTEM_STL)" != "YES"
+CDEFS += -DCPPCONN_LIB=\"$(DLLPRE)mysqlcppconn-stlport$(DLLPOST)\"
+.ELSE
 CDEFS += -DCPPCONN_LIB=\"$(DLLPRE)mysqlcppconn$(DLLPOST)\"
+.ENDIF
 .ELSE
+.IF "$(USE_SYSTEM_STL)" != "YES"
+CDEFS += -DCPPCONN_LIB=\"$(shell readlink /usr/lib/libmysqlcppconn-stlport.so)\"
+.ELSE
 CDEFS += -DCPPCONN_LIB=\"$(shell readlink /usr/lib/libmysqlcppconn.so)\"
 .ENDIF
+.ENDIF
 
 # --------------- MySQL settings ------------------
 .IF "$(GUI)"=="WNT"
@@ -145,7 +153,11 @@
     .ENDIF
     MYSQL_CPPCONNFILE=$(SOLARLIBDIR)$/$(DLLPRE)mysqlcppconn$(DLLPOST)
   .ELSE
-    MYSQL_CPPCONN_LIB+=-lmysqlcppconn
+    .IF "$(USE_SYSTEM_STL)" != "YES"
+      MYSQL_CPPCONN_LIB+=-lmysqlcppconn-stlport
+    .ELSE
+      MYSQL_CPPCONN_LIB+=-lmysqlcppconn
+    .ENDIF
   .ENDIF
 .ENDIF
 
Index: configure.in
===================================================================
--- configure.in	(Revision 275540)
+++ configure.in	(Arbeitskopie)
@@ -3802,8 +3802,13 @@
         AC_LANG_PUSH([C++])
 	AC_CHECK_HEADER(mysql_driver.h, [],
 					[AC_MSG_ERROR(mysql_driver.h not found. install MySQL C++ Connectivity)], [])
+if test "$USE_SYSTEM_STL" != "YES"; then
+	AC_CHECK_LIB(mysqlcppconn-stlport, main, [],
+					[AC_MSG_ERROR(MySQL C++ Connectivity lib not found or functional)], [])
+else
 	AC_CHECK_LIB(mysqlcppconn, main, [],
 					[AC_MSG_ERROR(MySQL C++ Connectivity lib not found or functional)], [])
+fi
    AC_MSG_CHECKING([version])
       AC_TRY_RUN([
 #include <mysql_driver.h>
@@ -3821,13 +3826,6 @@
       ], [AC_MSG_RESULT(OK)], [AC_MSG_ERROR([not suitable, we need >= 1.0.6])])
         AC_LANG_POP([C++])
 
-	AC_MSG_CHECKING([STL compatibility])
-	if test "$WITH_STLPORT" != "no"; then
-		AC_MSG_ERROR([to use system mysqlcppconn you need to use --without-stlport])
-	else
-		AC_MSG_RESULT([OK])	
-	fi
-
 else
 	AC_MSG_RESULT([internal])
         AC_MSG_CHECKING([for mysqlcppconn module])
