Index: unxlngs.mk
===================================================================
--- solenv/inc/unxlngs.mk	(Revision 276754)
+++ solenv/inc/unxlngs.mk	(Arbeitskopie)
@@ -33,7 +33,7 @@
 
 ASM=$(CC)
 AFLAGS=-Wa,-K,PIC -c $(CDEFS)
-CDEFAULTOPT=-Os
+CDEFAULTOPT=-O2
 PICSWITCH:=-fPIC
 .INCLUDE : unxlng.mk
 CDEFS+=-DSPARC
Index: unxlngi.mk
===================================================================
--- solenv/inc/unxlngi.mk	(Revision 276754)
+++ solenv/inc/unxlngi.mk	(Arbeitskopie)
@@ -30,7 +30,7 @@
 #*************************************************************************
 
 # mk file for Unix Linux Intel (X86) using GCC, please make generic modifications to unxlng.mk
-CDEFAULTOPT=-Os
+CDEFAULTOPT=-O2
 ARCH_FLAGS*=-mtune=pentiumpro
 .INCLUDE : unxlng.mk
 CDEFS+=-DX86
--- solenv/inc/unxlngppc.mk	(Revision 277719)
+++ solenv/inc/unxlngppc.mk	(Arbeitskopie)
@@ -31,7 +31,7 @@
 
 # mk file for Unix Linux PowerPC using GCC, please make generic modifications to unxlng.mk
 
-DEFAULTOPT=-Os
+DEFAULTOPT=-O2
 PICSWITCH:=-fPIC
 .INCLUDE : unxlng.mk
 CDEFS+=-DPOWERPC -DPPC
