From cd06aed38a1374fae1aa4f5fa54cbb33b82c198e Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:53:48 +0200
Subject: [PATCH 071/878] debug-bits.diff

---
 tools/source/debug/debug.cxx |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/tools/source/debug/debug.cxx b/tools/source/debug/debug.cxx
index b2ae91c..77f43e5 100644
--- a/tools/source/debug/debug.cxx
+++ b/tools/source/debug/debug.cxx
@@ -178,7 +178,11 @@ struct DebugData
         aDbgData.bOverwrite = TRUE;
         aDbgData.nTraceOut = DBG_OUT_NULL;
         aDbgData.nWarningOut = DBG_OUT_NULL;
+#ifdef UNX
+        aDbgData.nErrorOut = DBG_OUT_SHELL;
+#else
         aDbgData.nErrorOut = DBG_OUT_MSGBOX;
+#endif
         aDbgData.bMemInit = 0x77;
         aDbgData.bMemBound = 0x55;
         aDbgData.bMemFree = 0x33;
-- 
1.7.0.1

