---
 sfx2/source/appl/appreg.cxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git sfx2/source/appl/appreg.cxx sfx2/source/appl/appreg.cxx
index a80b014..6a854b9 100644
--- sfx2/source/appl/appreg.cxx
+++ sfx2/source/appl/appreg.cxx
@@ -67,7 +67,7 @@ void SfxApplication::Registrations_Impl()
     SfxRecordingFloatWrapper_Impl::RegisterChildWindow();
     SfxNavigatorWrapper::RegisterChildWindow( FALSE, NULL, SFX_CHILDWIN_NEVERHIDE );
     SfxPartChildWnd_Impl::RegisterChildWindow();
-    SfxTemplateDialogWrapper::RegisterChildWindow(TRUE);
+    SfxTemplateDialogWrapper::RegisterChildWindow();
     SfxDockingWrapper::RegisterChildWindow();
 
     // Controller
-- 
1.7.0.1

