From 147424004191358fcccf2c4ed0bee62906b7fde2 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:58:01 +0200
Subject: [PATCH 259/768] default-no-style-dialog.diff

---
 sfx2/source/appl/appreg.cxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git sfx2/source/appl/appreg.cxx sfx2/source/appl/appreg.cxx
index 37ca1f5..af4cb5b 100644
--- sfx2/source/appl/appreg.cxx
+++ sfx2/source/appl/appreg.cxx
@@ -66,7 +66,7 @@ void SfxApplication::Registrations_Impl()
     SfxRecordingFloatWrapper_Impl::RegisterChildWindow();
     SfxNavigatorWrapper::RegisterChildWindow( FALSE, NULL, SFX_CHILDWIN_NEVERHIDE );
     SfxPartChildWnd_Impl::RegisterChildWindow();
-    SfxTemplateDialogWrapper::RegisterChildWindow(TRUE);
+    SfxTemplateDialogWrapper::RegisterChildWindow();
     SfxDockingWrapper::RegisterChildWindow();
 
     // Controller
-- 
1.7.0.1

