---
 openssl/makefile.mk |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git openssl/makefile.mk openssl/makefile.mk
index 0816807..53a5239 100644
--- openssl/makefile.mk
+++ openssl/makefile.mk
@@ -53,7 +53,7 @@ TARFILE_MD5=05a0ece1372392a2cf310ebb96333025
 
 CONFIGURE_DIR=.
 CONFIGURE_ACTION=config
-CONFIGURE_FLAGS=-I$(SYSBASE)$/usr$/include -L$(SYSBASE)$/usr$/lib shared 
+CONFIGURE_FLAGS=-I$(SYSBASE)$/usr$/include -L$(SYSBASE)$/usr$/lib shared no-idea
 
 BUILD_DIR=.
 BUILD_ACTION=make CC='$(CC)'
@@ -156,7 +156,7 @@ OUT2BIN += out/libeay32.dll
 
         #CONFIGURE_ACTION=cmd /c $(PERL:s!\!/!) configure
         CONFIGURE_ACTION=$(PERL) configure
-        CONFIGURE_FLAGS=VC-WIN32
+        CONFIGURE_FLAGS=VC-WIN32 no-idea
         BUILD_ACTION=cmd /c "ms$(EMQ)\do_ms.bat $(subst,/,\ $(normpath,1 $(PERL)))" && nmake -f ms/ntdll.mak
 
         OUT2LIB = out32dll$/ssleay32.lib
-- 
1.7.0.1

