From c914def20729249e00d32ca24ae65f2357f1388c Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:58:10 +0200
Subject: [PATCH 266/768] default-font-size.diff

---
 vcl/source/app/settings.cxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git vcl/source/app/settings.cxx vcl/source/app/settings.cxx
index fb2f0d3..987abbe 100644
--- vcl/source/app/settings.cxx
+++ vcl/source/app/settings.cxx
@@ -547,7 +547,7 @@ ImplStyleData::ImplStyleData( const ImplStyleData& rData ) :
 
 void ImplStyleData::SetStandardStyles()
 {
-    Font aStdFont( FAMILY_SWISS, Size( 0, 8 ) );
+    Font aStdFont( FAMILY_SWISS, Size( 0, 10 ) );
     aStdFont.SetCharSet( gsl_getSystemTextEncoding() );
     aStdFont.SetWeight( WEIGHT_NORMAL );
     aStdFont.SetName( utl::DefaultFontConfiguration::get()->getUserInterfaceFont(com::sun::star::lang::Locale( rtl::OUString( RTL_CONSTASCII_USTRINGPARAM("en") ), rtl::OUString(), rtl::OUString() ) ) );
-- 
1.7.0.1

