From eb75f4c78d76d4190567ee1d83ef9985d619795f Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:58:16 +0200
Subject: [PATCH 307/878] default-openssl-no-idea-alg.diff

---
 openssl/makefile.mk |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/openssl/makefile.mk b/openssl/makefile.mk
index b883e06..375999d 100644
--- a/openssl/makefile.mk
+++ b/openssl/makefile.mk
@@ -52,7 +52,7 @@ TARFILE_NAME=$(OPENSSL_NAME)
 
 CONFIGURE_DIR=.
 CONFIGURE_ACTION=config
-CONFIGURE_FLAGS=-I$(SYSBASE)$/usr$/include -L$(SYSBASE)$/usr$/lib shared 
+CONFIGURE_FLAGS=-I$(SYSBASE)$/usr$/include -L$(SYSBASE)$/usr$/lib shared no-idea
 
 BUILD_DIR=.
 BUILD_ACTION=make CC='$(CC)'
@@ -141,7 +141,7 @@ OUT2BIN += out/libeay32.dll
 
         #CONFIGURE_ACTION=cmd /c $(PERL:s!\!/!) configure
         CONFIGURE_ACTION=$(PERL) configure
-        CONFIGURE_FLAGS=VC-WIN32
+        CONFIGURE_FLAGS=VC-WIN32 no-idea
         BUILD_ACTION=cmd /c "ms$(EMQ)\do_ms.bat $(subst,/,\ $(normpath,1 $(PERL)))" && nmake -f ms/ntdll.mak
 
         OUT2LIB = out32dll$/ssleay32.lib
-- 
1.7.0.1

