--- scp2/source/ooo/file_font_ooo.scp.orig	2008-01-08 16:14:16.000000000 +0100
+++ scp2/source/ooo/file_font_ooo.scp	2008-04-05 09:44:51.000000000 +0200
@@ -134,78 +134,54 @@
 
 // Liberation Mono
 #ifndef WITHOUT_FONTS 
-#ifndef WNT 
 STD_FONT_FILE( gid_File_Fnt_LiberationMono_Regular, LiberationMono-Regular.ttf, Liberation Mono Regular) 
 #endif
-#endif
 
 #ifndef WITHOUT_FONTS 
-#ifndef WNT 
 STD_FONT_FILE( gid_File_Fnt_LiberationMono_Bold, LiberationMono-Bold.ttf, Liberation Mono Bold) 
 #endif
-#endif
 
 #ifndef WITHOUT_FONTS 
-#ifndef WNT 
 STD_FONT_FILE( gid_File_Fnt_LiberationMono_Italic, LiberationMono-Italic.ttf, Liberation Mono Italic) 
 #endif
-#endif
 
 #ifndef WITHOUT_FONTS 
-#ifndef WNT 
 STD_FONT_FILE( gid_File_Fnt_LiberationMono_BoldItalic, LiberationMono-BoldItalic.ttf, Liberation Mono Bold Italic) 
 #endif
-#endif
 
 // Liberation Sans
 #ifndef WITHOUT_FONTS 
-#ifndef WNT 
 STD_FONT_FILE( gid_File_Fnt_LiberationSans_Regular, LiberationSans-Regular.ttf, Liberation Sans Regular) 
 #endif
-#endif
 
 #ifndef WITHOUT_FONTS 
-#ifndef WNT 
 STD_FONT_FILE( gid_File_Fnt_LiberationSans_Italic, LiberationSans-Italic.ttf, Liberation Sans Italic) 
 #endif
-#endif
 
 #ifndef WITHOUT_FONTS 
-#ifndef WNT 
 STD_FONT_FILE( gid_File_Fnt_LiberationSans_Bold, LiberationSans-Bold.ttf, Liberation Serif Bold) 
 #endif
-#endif
 
 #ifndef WITHOUT_FONTS 
-#ifndef WNT 
 STD_FONT_FILE( gid_File_Fnt_LiberationSans_BoldItalic, LiberationSans-BoldItalic.ttf, Liberation Sans Bold Italic) 
 #endif
-#endif
 
 // Liberation Serif
 #ifndef WITHOUT_FONTS 
-#ifndef WNT 
 STD_FONT_FILE( gid_File_Fnt_LiberationSerif_Regular, LiberationSerif-Regular.ttf, Liberation Serif Regular) 
 #endif
-#endif
 
 #ifndef WITHOUT_FONTS 
-#ifndef WNT 
 STD_FONT_FILE( gid_File_Fnt_LiberationSerif_Bold, LiberationSerif-Bold.ttf, Liberation Serif Bold) 
 #endif
-#endif
 
 #ifndef WITHOUT_FONTS 
-#ifndef WNT 
 STD_FONT_FILE( gid_File_Fnt_LiberationSerif_Italic, LiberationSerif-Italic.ttf, Liberation Serif Italic) 
 #endif
-#endif
 
 #ifndef WITHOUT_FONTS 
-#ifndef WNT 
 STD_FONT_FILE( gid_File_Fnt_LiberationSerif_BoldItalic, LiberationSerif-BoldItalic.ttf, Liberation Serif Bold Italic) 
 #endif
-#endif
 
 // Gentium fonts
 
