Index: math.desktop
===================================================================
--- sysui/desktop/menus/math.desktop	(Revision 277909)
+++ sysui/desktop/menus/math.desktop	(Arbeitskopie)
@@ -9,3 +9,4 @@
 Name=%PRODUCTNAME Math
 GenericName=Formula Editor
 Comment=Create and edit scientific formulas and equations by using Math.
+InitialPreference=5
Index: base.desktop
===================================================================
--- sysui/desktop/menus/base.desktop	(Revision 277909)
+++ sysui/desktop/menus/base.desktop	(Arbeitskopie)
@@ -9,3 +9,4 @@
 Name=%PRODUCTNAME Base
 GenericName=Database Development
 Comment=Manage databases, create queries and reports to track and manage your information by using Base.
+InitialPreference=5
Index: writer.desktop
===================================================================
--- sysui/desktop/menus/writer.desktop	(Revision 277909)
+++ sysui/desktop/menus/writer.desktop	(Arbeitskopie)
@@ -9,3 +9,4 @@
 Name=%PRODUCTNAME Writer
 GenericName=Word Processor
 Comment=Create and edit text and graphics in letters, reports, documents and Web pages by using Writer.
+InitialPreference=5
Index: draw.desktop
===================================================================
--- sysui/desktop/menus/draw.desktop	(Revision 277909)
+++ sysui/desktop/menus/draw.desktop	(Arbeitskopie)
@@ -9,3 +9,4 @@
 Name=%PRODUCTNAME Draw
 GenericName=Drawing Program
 Comment=Create and edit drawings, flow charts, and logos by using Draw.
+InitialPreference=5
Index: calc.desktop
===================================================================
--- sysui/desktop/menus/calc.desktop	(Revision 277909)
+++ sysui/desktop/menus/calc.desktop	(Arbeitskopie)
@@ -9,3 +9,4 @@
 Name=%PRODUCTNAME Calc
 GenericName=Spreadsheet
 Comment=Perform calculation, analyze information and manage lists in spreadsheets by using Calc.
+InitialPreference=5
Index: impress.desktop
===================================================================
--- sysui/desktop/menus/impress.desktop	(Revision 277909)
+++ sysui/desktop/menus/impress.desktop	(Arbeitskopie)
@@ -9,3 +9,4 @@
 Name=%PRODUCTNAME Impress
 GenericName=Presentation
 Comment=Create and edit presentations for slideshows, meeting and Web pages by using Impress.
+InitialPreference=5
