--- desktop/source/app/app.cxx.old	2010-03-03 16:59:08.000000000 +0100
+++ desktop/source/app/app.cxx	2010-03-03 16:59:21.000000000 +0100
@@ -47,6 +47,7 @@
 #include "desktopcontext.hxx"
 #include "exithelper.hxx"
 #include "../migration/pages.hxx"
+#include "../migration/migration.hxx"
 
 #include <svtools/javacontext.hxx>
 #include <com/sun/star/frame/XSessionManagerListener.hpp>
@@ -1424,6 +1425,7 @@ void Desktop::Main()
         if ( bAbort )
             return;
 
+#if 0 // ooo-build doesn't use First Start Wizard at all.
         // First Start Wizard allowed ?
         if ( ! pCmdLineArgs->IsNoFirstStartWizard())
         {
@@ -1455,6 +1457,7 @@ void Desktop::Main()
             
             RTL_LOGFILE_CONTEXT_TRACE( aLog, "} FirstStartWizard" );
         }
+#endif
 
         // keep a language options instance...
         pLanguageOptions.reset( new SvtLanguageOptions(sal_True));
