diff -r 3c24aa9a7810 desktop/source/app/cmdlineargs.cxx
--- desktop/source/app/cmdlineargs.cxx	Sat Jan 23 16:31:03 2010 +0000
+++ desktop/source/app/cmdlineargs.cxx	Tue Feb 02 13:59:39 2010 +0000
@@ -387,7 +387,7 @@
 	}
 	else if ( aArg.equalsAsciiL( RTL_CONSTASCII_STRINGPARAM( "-quickstart" )) == sal_True )
 	{
-#if defined(WNT) || defined(OS2) || defined(QUARTZ)
+#if defined(ENABLE_QUICKSTART_APPLET)
 		SetBoolParam_Impl( CMD_BOOLPARAM_QUICKSTART, sal_True );
 #endif
 		return sal_True;
diff -r 3c24aa9a7810 desktop/source/app/makefile.mk
--- desktop/source/app/makefile.mk	Sat Jan 23 16:31:03 2010 +0000
+++ desktop/source/app/makefile.mk	Tue Feb 02 13:59:39 2010 +0000
@@ -44,6 +44,10 @@
 CFLAGS+=-DGNOME_VFS_ENABLED
 .ENDIF
 
+.IF "$(GUI)"=="WNT" || "$(GUI)"=="OS2" || "$(GUIBASE)"=="aqua" || "$(ENABLE_SYSTRAY_GTK)"=="TRUE"
+CFLAGS+=-DENABLE_QUICKSTART_APPLET
+.ENDIF
+
 SHL1TARGET = sofficeapp
 SHL1OBJS = \
     $(SLO)$/app.obj \
