---
 desktop/source/app/cmdlinehelp.cxx |    9 +++++----
 1 files changed, 5 insertions(+), 4 deletions(-)

diff --git desktop/source/app/cmdlinehelp.cxx desktop/source/app/cmdlinehelp.cxx
index 8af72e6..3032315 100644
--- desktop/source/app/cmdlinehelp.cxx
+++ desktop/source/app/cmdlinehelp.cxx
@@ -135,7 +135,8 @@ namespace desktop
         aHelpMessage_head.SearchAndReplaceAscii( "%CMDNAME", String( "soffice", RTL_TEXTENCODING_ASCII_US) );
 #ifdef UNX
         // on unix use console for output
-        fprintf(stderr, "%s\n", ByteString(aHelpMessage_head,
+        FILE* fs = stdout;
+        fprintf(fs, "%s\n", ByteString(aHelpMessage_head,
                     RTL_TEXTENCODING_ASCII_US).GetBuffer());
         // merge left and right column
         int n = aHelpMessage_left.GetTokenCount ('\n');
@@ -143,10 +144,10 @@ namespace desktop
         ByteString bsRight(aHelpMessage_right, RTL_TEXTENCODING_ASCII_US);
         for ( int i = 0; i < n; i++ )
         {
-            fprintf(stderr, "%s", bsLeft.GetToken(i, '\n').GetBuffer());
-            fprintf(stderr, "%s\n", bsRight.GetToken(i, '\n').GetBuffer());
+            fprintf(fs, "%s", bsLeft.GetToken(i, '\n').GetBuffer());
+            fprintf(fs, "%s\n", bsRight.GetToken(i, '\n').GetBuffer());
         }
-        fprintf(stderr, "%s", ByteString(aHelpMessage_bottom,
+        fprintf(fs, "%s", ByteString(aHelpMessage_bottom,
                     RTL_TEXTENCODING_ASCII_US).GetBuffer());
 #else
         // rest gets a dialog box
-- 
1.7.0.1

