---
 desktop/source/app/userinstall.cxx |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git desktop/source/app/userinstall.cxx desktop/source/app/userinstall.cxx
index f72d138..d2ac260 100644
--- desktop/source/app/userinstall.cxx
+++ desktop/source/app/userinstall.cxx
@@ -246,6 +246,11 @@ namespace desktop {
         FileBase::RC rc = Directory::createPath(aUserPath);
         if ((rc != FileBase::E_None) && (rc != FileBase::E_EXIST)) return UserInstall::E_Creation;
 
+#ifdef UNIX
+    // set safer permissions for the user directory by default
+    File::setAttributes(aUserPath, Attribute_OwnWrite| Attribute_OwnRead| Attribute_OwnExe);
+#endif
+
             // copy data from shared data directory of base installation
         for (sal_Int32 i=0; pszSrcList[i]!=NULL && pszDstList[i]!=NULL; i++)
         {
-- 
1.7.0.1

