--- basegfx/prj/build.lst.orig	2010-01-04 22:05:14.000000000 +0100
+++ basegfx/prj/build.lst	2010-01-04 22:07:03.000000000 +0100
@@ -1,4 +1,4 @@
-fx	basegfx	:	o3tl sal offuh cppuhelper cppu testshl2 NULL
+fx	basegfx	:	o3tl sal offuh cppuhelper cppu QADEVOOO:testshl2 NULL
 fx	basegfx									usr1	-	all	fx_mkout NULL
 fx	basegfx\inc								nmake	-	all	fx_inc NULL
 fx	basegfx\prj								get		-	all	fx_prj NULL
--- o3tl/prj/build.lst.orig	2010-01-04 22:05:24.000000000 +0100
+++ o3tl/prj/build.lst	2010-01-04 22:07:03.000000000 +0100
@@ -1,4 +1,4 @@
-ot	o3tl	:	sal testshl2 BOOST:boost NULL
+ot	o3tl	:	sal QADEVOOO:testshl2 BOOST:boost NULL
 ot	o3tl									usr1	-	all	ot_mkout NULL
 ot	o3tl\inc								get		-	all	ot_inc NULL
 ot	o3tl\prj								get		-	all	ot_prj NULL
--- postprocess/prj/build.lst.orig	2010-01-04 22:05:36.000000000 +0100
+++ postprocess/prj/build.lst	2010-01-04 22:07:03.000000000 +0100
@@ -1,4 +1,4 @@
-po	postprocess     ::      accessibility automation basctl bean fondu BINFILTER:binfilter chart2 configmgr CRASHREP:crashrep dbaccess desktop dtrans embeddedobj embedserv EPM:epm eventattacher extensions extras fileaccess filter forms fpicker helpcontent2 hwpfilter io JAVAINSTALLER2:javainstaller2 lingucomponent MATHMLDTD:MathMLDTD ODK:odk officecfg package padmin psprint_config remotebridges sc scaddins sccomp scp2 scripting sd setup_native slideshow starmath sw sysui testshl2 testtools ucb UnoControls unoxml ure wizards xmerge xmlsecurity BITSTREAM_VERA_FONTS:bitstream_vera_fonts AGFA_MONOTYPE_FONTS:agfa_monotype_fonts DICTIONARIES:dictionaries PYUNO:pyuno OOo:readlicense_oo SO:top unodevtools JFREEREPORT:jfreereport CT2N:ct2n NLPSOLVER:nlpsolver LANGUAGETOOL:languagetool REPORTBUILDER:reportbuilder reportdesign SDEXT:sdext SWEXT:swext writerfilter oox MYSQLC:mysqlc NULL
+po     postprocess     ::      accessibility automation basctl bean fondu BINFILTER:binfilter chart2 configmgr CRASHREP:crashrep dbaccess desktop dtrans embeddedobj embedserv EPM:epm eventattacher extensions extras fileaccess filter forms fpicker helpcontent2 hwpfilter io JAVAINSTALLER2:javainstaller2 lingucomponent MATHMLDTD:MathMLDTD ODK:odk officecfg package padmin psprint_config remotebridges sc scaddins sccomp scp2 scripting sd setup_native slideshow starmath sw sysui QADEVOOO:testshl2 testtools ucb UnoControls unoxml ure wizards xmerge xmlsecurity BITSTREAM_VERA_FONTS:bitstream_vera_fonts AGFA_MONOTYPE_FONTS:agfa_monotype_fonts DICTIONARIES:dictionaries PYUNO:pyuno OOo:readlicense_oo SO:top unodevtools JFREEREPORT:jfreereport CT2N:ct2n NLPSOLVER:nlpsolver LANGUAGETOOL:languagetool REPORTBUILDER:reportbuilder reportdesign SDEXT:sdext SWEXT:swext writerfilter oox MYSQLC:mysqlc NULL
 po	postprocess			    	usr1	-	all	po_mkout NULL
 po	postprocess\checkxml		nmake	-	all	po_checkxml NULL
 po	postprocess\checkdeliver	nmake	-	all	po_checkdlv NULL
--- salhelper/prj/build.lst.orig	2010-01-04 22:05:45.000000000 +0100
+++ salhelper/prj/build.lst	2010-01-04 22:07:03.000000000 +0100
@@ -1,3 +1,3 @@
-shp	salhelper	:	sal testshl2 NULL
+shp	salhelper	:	sal QADEVOOO:testshl2 NULL
 shp	salhelper		 	usr1	-	all	shp_mkout NULL
 shp	salhelper\source 	nmake	-	all shp_source NULL
