From 65a6b5f012fa716c187a263e45e17fc03cd5ceaa Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:00:39 +0200
Subject: [PATCH 358/768] docx-ugly-fix.diff

---
 writerfilter/source/dmapper/PropertyMap.cxx |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git writerfilter/source/dmapper/PropertyMap.cxx writerfilter/source/dmapper/PropertyMap.cxx
index e35ec3d..16b8260 100644
--- writerfilter/source/dmapper/PropertyMap.cxx
+++ writerfilter/source/dmapper/PropertyMap.cxx
@@ -719,10 +719,12 @@ void SectionPropertyMap::CloseSectionGroup( DomainMapper_Impl& rDM_Impl )
 //        ::rtl::OUString sSelection = m_xStartingRange->getString();
 //        sSelection.getLength();
         //-->debug
+#if 0
         uno::Reference< beans::XPropertySet > xSection =
                                     rDM_Impl.appendTextSectionAfter( m_xStartingRange );
         if( m_nColumnCount > 0 && xSection.is())
             ApplyColumnProperties( xSection );
+#endif
     }
     else
     {
-- 
1.7.0.1

