---
 writerfilter/source/dmapper/PropertyMap.cxx |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git writerfilter/source/dmapper/PropertyMap.cxx writerfilter/source/dmapper/PropertyMap.cxx
index 53275e3..50cd8b3 100644
--- writerfilter/source/dmapper/PropertyMap.cxx
+++ writerfilter/source/dmapper/PropertyMap.cxx
@@ -812,10 +812,12 @@ void SectionPropertyMap::CloseSectionGroup( DomainMapper_Impl& rDM_Impl )
 //        ::rtl::OUString sSelection = m_xStartingRange->getString();
 //        sSelection.getLength();
         //-->debug
+#if 0
         uno::Reference< beans::XPropertySet > xSection =
                                     rDM_Impl.appendTextSectionAfter( m_xStartingRange );
         if( m_nColumnCount > 0 && xSection.is())
             ApplyColumnProperties( xSection );
+#endif
     }
     else
     {
-- 
1.7.0.1

