From 08d3460a67ea31126594b4a976f24f3960dbc767 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:00:39 +0200
Subject: [PATCH 417/878] docx-ugly-fix.diff

---
 writerfilter/source/dmapper/PropertyMap.cxx |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/writerfilter/source/dmapper/PropertyMap.cxx b/writerfilter/source/dmapper/PropertyMap.cxx
index faf4237..7ca6860 100644
--- a/writerfilter/source/dmapper/PropertyMap.cxx
+++ b/writerfilter/source/dmapper/PropertyMap.cxx
@@ -764,10 +764,12 @@ void SectionPropertyMap::CloseSectionGroup( DomainMapper_Impl& rDM_Impl )
 //        ::rtl::OUString sSelection = m_xStartingRange->getString();
 //        sSelection.getLength();
         //-->debug
+#if 0
         uno::Reference< beans::XPropertySet > xSection =
                                     rDM_Impl.appendTextSectionAfter( m_xStartingRange );
         if( m_nColumnCount > 0 && xSection.is())
             ApplyColumnProperties( xSection );
+#endif
     }
     else
     {
-- 
1.7.0.1

