diff --git writerfilter/source/dmapper/PropertyMap.cxx writerfilter/source/dmapper/PropertyMap.cxx
index fda15ed..a0248b0 100644
--- writerfilter/source/dmapper/PropertyMap.cxx
+++ writerfilter/source/dmapper/PropertyMap.cxx
@@ -768,10 +768,12 @@ void SectionPropertyMap::CloseSectionGroup( DomainMapper_Impl& rDM_Impl )
 //        ::rtl::OUString sSelection = m_xStartingRange->getString();
 //        sSelection.getLength();
         //-->debug
+#if 0
         uno::Reference< beans::XPropertySet > xSection =
                                     rDM_Impl.appendTextSectionAfter( m_xStartingRange );
         if( m_nColumnCount > 0 && xSection.is())
             ApplyColumnProperties( xSection );
+#endif
     }
     else
     {
