diff --git sd/source/ui/view/drviews5.cxx sd/source/ui/view/drviews5.cxx
index 01e1c54..5ab8cf9 100644
--- sd/source/ui/view/drviews5.cxx
+++ sd/source/ui/view/drviews5.cxx
@@ -638,7 +638,7 @@ void DrawViewShell::ReadUserDataSequence ( const ::com::sun::star::uno::Sequence
     }
 
     ChangeEditMode (meEditMode, ! IsLayerModeActive());
-    ChangeEditMode (meEditMode, ! IsLayerModeActive());
+    ResetActualLayer();
 }
 
 void DrawViewShell::VisAreaChanged(const Rectangle& rRect)
