--- dtrans/source/aqua/DropTarget.cxx~	2009-07-27 11:20:36.000000000 +0200
+++ dtrans/source/aqua/DropTarget.cxx	2009-08-20 23:47:30.000000000 +0200
@@ -297,6 +297,9 @@
 
       fire_dragOver(dtde);
 
+      // drag over callbacks likely have rendered something 
+      [mView setNeedsDisplay: TRUE];
+
       dragOp = OfficeToSystemDragActions(mSelectedDropAction);
 
       //NSLog(@"Drag update: Source actions: %x proposed action %x selected action %x", mDragSourceSupportedActions, currentAction, mSelectedDropAction);
