From 072eca2a543164eeb0242c184217f4ac55277efe Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:06:29 +0200
Subject: [PATCH 714/878] edit-links-decode-unicode-uri.diff

---
 svx/source/cui/linkdlg.cxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/svx/source/cui/linkdlg.cxx b/svx/source/cui/linkdlg.cxx
index 456bc71..e7375aa 100644
--- a/svx/source/cui/linkdlg.cxx
+++ b/svx/source/cui/linkdlg.cxx
@@ -221,7 +221,7 @@ IMPL_LINK( SvBaseLinksDlg, LinksSelectHdl, SvTabListBox *, pSvTabListBox )
 
         String aFileName;
         pLinkMgr->GetDisplayNames( pLink, &sType, &aFileName, pLinkNm, pFilter );
-
+        aFileName = INetURLObject::decode(aFileName, INET_HEX_ESCAPE, INetURLObject::DECODE_UNAMBIGUOUS);
         FileName().SetText( aFileName );
         SourceName().SetText( sLink );
         TypeName().SetText( sType );
-- 
1.7.0.1

