From 01171f67b243f5c34d787cc782f4077d2491ae04 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:55:32 +0200
Subject: [PATCH 163/878] enable-sheet-protection-options.diff

---
 sc/inc/tabprotection.hxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/sc/inc/tabprotection.hxx b/sc/inc/tabprotection.hxx
index 8e40590..2671df3 100644
--- a/sc/inc/tabprotection.hxx
+++ b/sc/inc/tabprotection.hxx
@@ -35,7 +35,7 @@
 #include <vector>
 #include <boost/shared_ptr.hpp>
 
-#define ENABLE_SHEET_PROTECTION 0
+#define ENABLE_SHEET_PROTECTION 1
 
 class ScDocument;
 class ScTableProtectionImpl;
-- 
1.7.0.1

