---
 sc/inc/tabprotection.hxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git sc/inc/tabprotection.hxx sc/inc/tabprotection.hxx
index 99294a6..965ea30 100644
--- sc/inc/tabprotection.hxx
+++ sc/inc/tabprotection.hxx
@@ -35,7 +35,7 @@
 #include <vector>
 #include <boost/shared_ptr.hpp>
 
-#define ENABLE_SHEET_PROTECTION 0
+#define ENABLE_SHEET_PROTECTION 1
 
 class ScDocument;
 class ScTableProtectionImpl;
-- 
1.7.0.1

