diff --git a/l10n/source/es/localize.sdf b/l10n/source/es/localize.sdf
--- l10n/source/es/localize.sdf
+++ l10n/source/es/localize.sdf
@@ -21545,7 +21545,7 @@
 svx	source\cui\numfmt.src	0	stringlist	RID_SVXPAGE_NUMBERFORMAT.LB_CATEGORY	7			0	es	Tiempo				2002-02-02 02:02:02
 svx	source\cui\numfmt.src	0	stringlist	RID_SVXPAGE_NUMBERFORMAT.LB_CATEGORY	8			0	es	Científico				2002-02-02 02:02:02
 svx	source\cui\numfmt.src	0	stringlist	RID_SVXPAGE_NUMBERFORMAT.LB_CATEGORY	9			0	es	Fracción				2002-02-02 02:02:02
-svx	source\cui\numfmt.src	0	stringlist	RID_SVXPAGE_NUMBERFORMAT.LB_CATEGORY	10			0	es	Valor buleano				2002-02-02 02:02:02
+svx	source\cui\numfmt.src	0	stringlist	RID_SVXPAGE_NUMBERFORMAT.LB_CATEGORY	10			0	es	Valor booleano				2002-02-02 02:02:02
 svx	source\cui\numfmt.src	0	stringlist	RID_SVXPAGE_NUMBERFORMAT.LB_CATEGORY	11			0	es	Texto				2002-02-02 02:02:02
 svx	source\cui\numfmt.src	0	fixedtext	RID_SVXPAGE_NUMBERFORMAT	FT_EDFORMAT			248	es	~Formato de código				2002-02-02 02:02:02
 svx	source\cui\numfmt.src	0	fixedtext	RID_SVXPAGE_NUMBERFORMAT	FT_FORMAT			90	es	F~ormato				2002-02-02 02:02:02
@@ -23370,7 +23370,7 @@
 svx	source\cui\numfmt.src	0	stringlist	RID_SVXPAGE_NUMBERFORMAT.LB_CATEGORY	7			0	es	Tiempo				2002-02-02 02:02:02
 svx	source\cui\numfmt.src	0	stringlist	RID_SVXPAGE_NUMBERFORMAT.LB_CATEGORY	8			0	es	Científico				2002-02-02 02:02:02
 svx	source\cui\numfmt.src	0	stringlist	RID_SVXPAGE_NUMBERFORMAT.LB_CATEGORY	9			0	es	Fracción				2002-02-02 02:02:02
-svx	source\cui\numfmt.src	0	stringlist	RID_SVXPAGE_NUMBERFORMAT.LB_CATEGORY	10			0	es	Valor buleano				2002-02-02 02:02:02
+svx	source\cui\numfmt.src	0	stringlist	RID_SVXPAGE_NUMBERFORMAT.LB_CATEGORY	10			0	es	Valor booleano				2002-02-02 02:02:02
 svx	source\cui\numfmt.src	0	stringlist	RID_SVXPAGE_NUMBERFORMAT.LB_CATEGORY	11			0	es	Texto				2002-02-02 02:02:02
 svx	source\cui\numfmt.src	0	fixedtext	RID_SVXPAGE_NUMBERFORMAT	FT_EDFORMAT			248	es	~Formato de código				2002-02-02 02:02:02
 svx	source\cui\numfmt.src	0	fixedtext	RID_SVXPAGE_NUMBERFORMAT	FT_FORMAT			90	es	F~ormato				2002-02-02 02:02:02
