--- extensions/source/propctrlr/standardcontrol.cxx	2010-04-27 14:06:02.000000000 +0100
+++ extensions/source/propctrlr/standardcontrol.cxx	2010-06-04 11:54:00.000000000 +0100
@@ -593,7 +593,7 @@ namespace pcr
             // everything which survived the checks above should result in a factor of 1, i.e.,
             // it should have a direct counterpart as FieldUnit
             throw RuntimeException();
-        getTypedControlWindow()->SetUnit( eFieldUnit );
+        getTypedControlWindow()->MetricFormatter::SetUnit( eFieldUnit );
     }
 
     //--------------------------------------------------------------------
