--- configure.in.old	2009-10-26 16:02:41.000000000 +0100
+++ configure.in	2009-10-27 16:57:35.000000000 +0100
@@ -74,6 +74,13 @@ AC_ARG_ENABLE(mozilla,
                           binary for your platform, to build without this
                           version, use this option.
 ],,enable_mozilla="yes")
+
+AC_ARG_WITH(extension-integration,
+[
+  --with-extension-integration      It will integrate the builded extensions
+			  to the installer of the product.],
+,)
+
 AC_ARG_WITH(fonts,
 [  --without-fonts         OOo includes some third-party fonts to provide a reliable
                           basis for help content, templates, samples, etc.
@@ -829,6 +836,31 @@ AC_ARG_WITH(alloc,
 [  --with-alloc            Define which allocator to build with
 			                 (choices are oo, system, tcmalloc)
 ],,)
+AC_ARG_WITH(sun-templates,
+[
+  --with-sun-templates      Integrate Sun template packages.],
+,)
+
+AC_ARG_WITH(lightproof,
+[
+  --with-lightproof      Integrate Lightproof extension],
+,)
+
+AC_ARG_WITH(google-docs,
+[
+  --with-google-docs      Download and integrate Google Documents extension.],
+,)
+
+AC_ARG_WITH(nlpsolver,
+[
+  --with-nlpsolver      Download and integrate NLPSolver extension.],
+,)
+
+AC_ARG_WITH(oooblogger,
+[
+  --with-oooblogger      Download and integrate oooblogger extension.],
+,)
+
 AC_ARG_ENABLE(verbose,
 [  --enable-verbose        Increase build verbosity.
   --disable-verbose       Decrease build verbosity.
@@ -5929,6 +5961,7 @@ if test -n "$enable_minimizer" -a "$enab
 else
    AC_MSG_RESULT([no])
    ENABLE_MINIMIZER=NO
+   SCPDEFS="$SCPDEFS -DWITHOUT_EXTENSION_MINIMIZER"
 fi
 AC_SUBST(ENABLE_MINIMIZER)
 
@@ -5939,6 +5972,7 @@ if test -n "$enable_presenter_console" -
 else
    AC_MSG_RESULT([no])
    ENABLE_PRESENTER_SCREEN=NO
+   SCPDEFS="$SCPDEFS -DWITHOUT_EXTENSION_PRESENTER_SCREEN"
 fi
 AC_SUBST(ENABLE_PRESENTER_SCREEN)
 
@@ -5970,6 +6004,7 @@ if test -n "$enable_pdfimport" -a "$enab
 else
    AC_MSG_RESULT([no])
    ENABLE_PDFIMPORT=NO
+   SCPDEFS="$SCPDEFS -DWITHOUT_EXTENSION_PDFIMPORT"
 fi
 AC_SUBST(ENABLE_PDFIMPORT)
 AC_SUBST(SYSTEM_POPPLER)
@@ -6000,6 +6035,7 @@ if test -n "$enable_wiki_publisher" -a "
 else
   AC_MSG_RESULT([no])
   ENABLE_MEDIAWIKI=NO
+  SCPDEFS="$SCPDEFS -DWITHOUT_EXTENSION_MEDIAWIKI"
 fi
 AC_SUBST(ENABLE_MEDIAWIKI)
 
@@ -6208,6 +6244,7 @@ if test -n "$enable_report_builder" -a "
 else
   AC_MSG_RESULT([no])
   ENABLE_REPORTBUILDER=NO
+  SCPDEFS="$SCPDEFS -DWITHOUT_EXTENSION_REPORTDESIGN"
   SYSTEM_JFREEREPORT=NO
 fi
 AC_SUBST(ENABLE_REPORTBUILDER)
@@ -6609,6 +6646,98 @@ else
 fi
 
 dnl ===================================================================
+dnl Test whether to integrate extensions into the product's installer
+dnl ===================================================================
+AC_MSG_CHECKING([for extensions intergration])
+if test "z$with_extension_integration" = "z" -o "z$with_extension_integration" = "zno" ; then
+  WITH_EXTENSION_INTEGRATION=NO
+  AC_MSG_RESULT([no integration])
+else
+  WITH_EXTENSION_INTEGRATION=YES
+  SCPDEFS="$SCPDEFS -DWITH_EXTENSION_INTEGRATION"
+  AC_MSG_RESULT([integration])
+fi
+AC_SUBST(WITH_EXTENSION_INTEGRATION)
+
+dnl ===================================================================
+dnl Test whether to include Lightproof extension
+dnl ===================================================================
+AC_MSG_CHECKING([for Lightproof extension integration])
+if test "z$with_lightproof" = "z" -o "z$with_lightproof" = "zno" ; then
+  AC_MSG_RESULT([no])
+  SCPDEFS="$SCPDEFS -DWITHOUT_EXTENSION_LIGHTPROOF"
+else
+  AC_MSG_RESULT([yes])
+fi
+AC_SUBST(WITH_LIGHTPROOF_EXTENSION)
+
+dnl ===================================================================
+dnl Test whether to include Google Docs extension
+dnl ===================================================================
+AC_MSG_CHECKING([for Google Docs extension integration])
+if test "z$with_google_docs" = "z" -o "z$with_google_docs" = "zno" ; then
+  AC_MSG_RESULT([no])
+else
+  AC_MSG_RESULT([yes])
+  SCPDEFS="$SCPDEFS -DWITH_EXTENSION_GOOGLE_DOCS"
+fi
+
+dnl ===================================================================
+dnl Test whether to include NLPSolver extension
+dnl ===================================================================
+AC_MSG_CHECKING([for NLPSolver extension integration])
+if test "z$with_nlpsolver" = "z" -o "z$with_nlpsolver" = "zno" ; then
+  AC_MSG_RESULT([no])
+else
+  AC_MSG_RESULT([yes])
+  BUILD_TYPE="$BUILD_TYPE NLPSOLVER"
+  SCPDEFS="$SCPDEFS -DWITH_EXTENSION_NLPSOLVER"
+fi
+
+dnl ===================================================================
+dnl Test whether to include oooblogger extension
+dnl ===================================================================
+AC_MSG_CHECKING([for NLPSolver extension integration])
+if test "z$with_oooblogger" = "z" -o "z$with_oooblogger" = "zno" ; then
+  AC_MSG_RESULT([no])
+else
+  AC_MSG_RESULT([yes])
+  SCPDEFS="$SCPDEFS -DWITH_EXTENSION_OOOBLOGGER"
+fi
+
+dnl ===================================================================
+dnl Test whether to include Sun Professinal Template Pack
+dnl ===================================================================
+AC_MSG_CHECKING([for Sun Professinal Template Pack integration (only supported languages displayed)])
+if test "z$with_sun_templates" = "z" -o "z$with_sun_templates" = "zno" ; then
+    SUNTEMPLATES_LANG=""
+    AC_MSG_RESULT([no integration])
+else
+    SCPDEFS="$SCPDEFS -DWITH_EXTENSION_SUN_TEMPLATE_PACK"
+    sun_supported_langs="en-US de it fr es hu"
+    if test "z$with_sun_templates" = "zyes" ; then
+	wanted_sun_templates="$sun_supported_langs"
+    else
+	# check whether the langs are supported by Sun
+	wanted_sun_templates=
+	for lang in $with_sun_templates ; do
+	    if test -n "`echo $sun_supported_langs | grep "$lang"`" ; then
+		wanted_sun_templates="$wanted_sun_templates $lang"
+	    fi
+	done
+    fi
+    SUNTEMPLATES_LANG=
+    # check whether the langs are requested at all
+    for lang in $wanted_sun_templates ; do
+	if test -n "`echo $with_lang | grep "$lang"`" ; then
+		SUNTEMPLATES_LANG="$SUNTEMPLATES_LANG $lang"
+	fi
+    done
+    AC_MSG_RESULT([$SUNTEMPLATES_LANG])
+fi
+AC_SUBST(SUNTEMPLATES_LANG)
+
+dnl ===================================================================
 dnl Test whether to include fonts
 dnl ===================================================================
 AC_MSG_CHECKING([whether to include Bitstream Vera fonts])
