---
 unotools/source/config/fltrcfg.cxx |    3 +--
 1 files changed, 1 insertions(+), 2 deletions(-)

diff --git unotools/source/config/fltrcfg.cxx unotools/source/config/fltrcfg.cxx
index 4707b4a..35f13c0 100644
--- unotools/source/config/fltrcfg.cxx
+++ unotools/source/config/fltrcfg.cxx
@@ -530,8 +530,7 @@ void SvtFilterOptions::SetWriter2WinWord( BOOL bFlag )
 
 BOOL SvtFilterOptions::IsUseEnhancedFields() const
 {
-    return false; // disable for now;
-//    return pImp->IsFlag( FILTERCFG_USE_ENHANCED_FIELDS );
+    return pImp->IsFlag( FILTERCFG_USE_ENHANCED_FIELDS );
 }
 
 void SvtFilterOptions::SetUseEnhancedFields( BOOL bFlag )
-- 
1.7.0.1

