diff --git sfx2/source/doc/docfile.cxx sfx2/source/doc/docfile.cxx
index 55ac8bb..f960140 100644
--- sfx2/source/doc/docfile.cxx
+++ sfx2/source/doc/docfile.cxx
@@ -1177,6 +1177,12 @@ sal_Bool SfxMedium::LockOrigFileOnDemand( sal_Bool bLoading, sal_Bool bNoUI )
             catch( uno::Exception )
             {}
 
+#if 0
+            // TODO: This extra ACL check was added as a fix to i#102464, but
+            // this appears to be unnecessary.  In fact, having this block
+            // surpresses the "Document in Use" dialog when someone tries to
+            // open a document already opened by someone, and the document is
+            // located in a samba share.
             if ( !bContentReadonly )
             {
                 // the file is not readonly, check the ACL
@@ -1185,6 +1191,7 @@ sal_Bool SfxMedium::LockOrigFileOnDemand( sal_Bool bLoading, sal_Bool bNoUI )
                 if ( ::utl::LocalFileHelper::ConvertURLToPhysicalName( GetURLObject().GetMainURL( INetURLObject::NO_DECODE ), aPhysPath ) )
                     bContentReadonly = IsReadonlyAccordingACL( aPhysPath.GetBuffer() );
             }
+#endif
         }
 
         // do further checks only if the file not readonly in fs
