diff -rup filter-orig/source/config/tools/merge/pyAltFCFGMerge filter/source/config/tools/merge/pyAltFCFGMerge
--- filter-orig/source/config/tools/merge/pyAltFCFGMerge	2010-05-25 11:47:25.000000000 +0200
+++ filter/source/config/tools/merge/pyAltFCFGMerge	2010-05-25 11:50:06.000000000 +0200
@@ -11,7 +11,7 @@
 
 import sys, string, os.path
 
-CFGFILE = "../../../../transex3/source/filter/merge/FCFGMerge.cfg"
+CFGFILE = "../../../source/config/tools/merge/FCFGMerge.cfg"
 
 PROP_XMLVERSION      = "xmlversion"               # // <= global cfg file
 PROP_XMLENCODING     = "xmlencoding"              # // <= global cfg file
