diff -rupw filter-orig/source/config/fragments/types/writer_PocketWord_File.xcu filter/source/config/fragments/types/writer_PocketWord_File.xcu
--- filter-orig/source/config/fragments/types/writer_PocketWord_File.xcu	2010-12-20 16:52:33.000000000 +0100
+++ filter/source/config/fragments/types/writer_PocketWord_File.xcu	2010-12-20 18:04:06.000000000 +0100
@@ -8,5 +8,4 @@
         <prop oor:name="UIName">
             <value>Pocket Word</value>
         </prop>
-        <prop oor:name="ClipboardFormat"><value>doctype:pwi</value></prop>
     </node>
