--- sd/inc/app.hrc.old	2009-10-08 16:17:24.000000000 +0200
+++ sd/inc/app.hrc	2010-03-03 16:59:59.000000000 +0100
@@ -459,11 +459,12 @@
 #define SID_DISPLAY_MASTER_OBJECTS          (SID_SD_START+437)
 #define SID_INSERTPAGE_LAYOUT_MENU          (SID_SD_START+438)
 #define SID_TP_EDIT_MASTER					(SID_SD_START+439)
+#define SID_OUTLINE_TEXT_AUTOFIT            (SID_SD_START+440)
 
 // Add companion for the SID_HIDE_SLIDE (that is defined in svx)
-#define SID_SHOW_SLIDE                      (SID_SD_START+440)
+#define SID_SHOW_SLIDE                      (SID_SD_START+441)
 
-#define SID_ADD_MOTION_PATH					(SID_SD_START+441)
-#define SID_TABLE_TOOLBOX					(SID_SD_START+442)
+#define SID_ADD_MOTION_PATH					(SID_SD_START+442)
+#define SID_TABLE_TOOLBOX					(SID_SD_START+443)
 
 #endif
--- sd/sdi/_drvwsh.sdi.old	2010-03-03 16:59:59.000000000 +0100
+++ sd/sdi/_drvwsh.sdi	2010-03-03 16:59:59.000000000 +0100
@@ -665,6 +665,11 @@ interface DrawView
         ExecMethod = FuPermanent ;
         StateMethod = GetMenuState ;
     ]
+    SID_OUTLINE_TEXT_AUTOFIT // ole : no, status : ?
+    [
+        ExecMethod = FuTemporary ;
+        StateMethod = GetMenuState ;
+    ]
     SID_TEXT_FITTOSIZE_VERTICAL // ole : no, status : ?
     [
         ExecMethod = FuPermanent ;
--- sd/sdi/sdraw.sdi.old	2009-10-08 16:17:24.000000000 +0200
+++ sd/sdi/sdraw.sdi	2010-03-03 16:59:59.000000000 +0100
@@ -6351,6 +6351,33 @@ SfxBoolItem TextFitToSizeTool SID_TEXT_F
 ]
 
 //--------------------------------------------------------------------------
+SfxBoolItem TextAutoFitToSize SID_OUTLINE_TEXT_AUTOFIT
+
+[
+    /* flags: */  
+    AutoUpdate = TRUE, 
+    Cachable = Cachable, 
+    FastCall = FALSE, 
+    HasCoreId = FALSE, 
+    HasDialog = FALSE, 
+    ReadOnlyDoc = FALSE, 
+    Toggle = FALSE, 
+    Container = FALSE, 
+    RecordAbsolute = FALSE, 
+    RecordPerSet;
+    Synchron;
+
+    Readonly = FALSE, 
+
+    /* config: */ 
+    AccelConfig = TRUE, 
+    MenuConfig = TRUE, 
+    StatusBarConfig = FALSE, 
+    ToolBoxConfig = TRUE, 
+    GroupId = GID_FORMAT;
+]
+
+//--------------------------------------------------------------------------
 TbxImageItem TextToolbox SID_DRAWTBX_TEXT
 
 [
--- sd/source/ui/app/menuids_tmpl.src.old	2009-10-08 16:17:24.000000000 +0200
+++ sd/source/ui/app/menuids_tmpl.src	2010-03-03 16:59:59.000000000 +0100
@@ -307,6 +307,14 @@
         Text [ en-US ] = "~Text..." ; \
     };
 
+#define MN_OUTLINE_TEXT_AUTOFIT \
+    MenuItem\
+    {\
+        Identifier = SID_OUTLINE_TEXT_AUTOFIT ; \
+        HelpID = SID_OUTLINE_TEXT_AUTOFIT ; \
+        Text [ en-US ] = "~Autofit Text" ; \
+    };
+
 #define MN_CONNECTION \
     MenuItem\
     {\
--- sd/source/ui/app/popup2_tmpl.src.old	2009-10-08 16:17:24.000000000 +0200
+++ sd/source/ui/app/popup2_tmpl.src	2010-03-03 16:59:59.000000000 +0100
@@ -82,6 +82,49 @@
 };
 
 //
+// OUTLINETEXTOBJ_POPUP
+//
+#if SD_POPUP == RID_DRAW_TEXTOBJ_POPUP
+    Menu RID_DRAW_OUTLINETEXTOBJ_POPUP
+#elif  SD_POPUP == RID_GRAPHIC_TEXTOBJ_POPUP
+    Menu RID_GRAPHIC_OUTLINETEXTOBJ_POPUP
+#endif
+{
+    ItemList =
+    {
+        MN_SET_DEFAULT
+        SEPARATOR
+        MN_LINE
+        MN_AREA
+        MN_TEXTATTR
+        MN_OUTLINE_TEXT_AUTOFIT
+        MN_TRANSFORM
+        MN_FONTWORK
+        SEPARATOR
+        MN_CHAR_PARAGRAPH
+        SEPARATOR
+        MN_POSITION
+        MN_OBJECT_ALIGN
+        MN_DISTRIBUTE
+        MNSUB_CONVERT
+
+        //-#i68101#----------------------
+        SEPARATOR
+        MN_OBJECT_TITLE_DESCRIPTION
+        MN_NAME_GROUP
+        //-#i68101#----------------------
+
+#if SD_POPUP == RID_DRAW_TEXTOBJ_POPUP
+        SEPARATOR
+        MN_EFFECT
+        MN_ACTION
+#endif
+        SEPARATOR
+        MN_STYLEDT
+    };
+};
+
+//
 // GEOMOBJ_POPUP
 //
 #if SD_POPUP == RID_DRAW_TEXTOBJ_POPUP
--- sd/source/ui/inc/res_bmp.hrc.old	2009-10-08 16:17:24.000000000 +0200
+++ sd/source/ui/inc/res_bmp.hrc	2010-03-03 16:59:59.000000000 +0100
@@ -311,6 +311,9 @@
 
 #define RID_DRAW_TABLEOBJ_INSIDE_POPUP		RID_APP_START+94
 
+#define RID_DRAW_OUTLINETEXTOBJ_POPUP		RID_APP_START+95
+#define RID_GRAPHIC_OUTLINETEXTOBJ_POPUP	RID_APP_START+96
+
 /*
  * Pointer (Mauszeiger)
  */
--- sd/source/ui/view/drviews2.cxx.old	2009-10-08 16:17:24.000000000 +0200
+++ sd/source/ui/view/drviews2.cxx	2010-03-03 16:59:59.000000000 +0100
@@ -175,6 +175,31 @@ void DrawViewShell::FuTemporary(SfxReque
 
     switch ( nSId )
     {
+        case SID_OUTLINE_TEXT_AUTOFIT:
+        {
+            SfxUndoManager* pUndoManager = GetDocSh()->GetUndoManager();
+            SdrObject* pObj = NULL;
+            const SdrMarkList& rMarkList = mpDrawView->GetMarkedObjectList();
+            if( rMarkList.GetMarkCount() == 1 )
+            {
+                pUndoManager->EnterListAction( String(), String() );
+                mpDrawView->BegUndo();
+
+                pObj = rMarkList.GetMark(0)->GetMarkedSdrObj();
+                bool bSet = ((const SdrTextFitToSizeTypeItem*)pObj->GetMergedItemSet().GetItem(SDRATTR_TEXT_FITTOSIZE))->GetValue() != SDRTEXTFIT_NONE;
+
+                mpDrawView->AddUndo(GetDoc()->GetSdrUndoFactory().CreateUndoAttrObject(*pObj));
+
+                pObj->SetMergedItem(SdrTextFitToSizeTypeItem(bSet ? SDRTEXTFIT_NONE : SDRTEXTFIT_AUTOFIT));
+                
+                mpDrawView->EndUndo();
+                pUndoManager->LeaveListAction();
+            }
+            Cancel();
+            rReq.Done();
+        }
+        break;
+
         // Flaechen und Linien-Attribute:
         // Sollten (wie StateMethode) eine eigene
         // Execute-Methode besitzen
--- sd/source/ui/view/drviews4.cxx.old	2009-10-08 16:17:24.000000000 +0200
+++ sd/source/ui/view/drviews4.cxx	2010-03-03 16:59:59.000000000 +0100
@@ -682,9 +682,13 @@ void DrawViewShell::Command(const Comman
                             {
                                 switch ( nId )
                                 {
+                                    case OBJ_OUTLINETEXT:
+                                        nSdResId = bGraphicShell ? RID_GRAPHIC_OUTLINETEXTOBJ_POPUP :
+                                                    				RID_DRAW_OUTLINETEXTOBJ_POPUP;
+                                        break;
+
                                     case OBJ_CAPTION:
                                     case OBJ_TITLETEXT:
-                                    case OBJ_OUTLINETEXT:
                                     case OBJ_TEXT:
                                         nSdResId = bGraphicShell ? RID_GRAPHIC_TEXTOBJ_POPUP :
                                                                     RID_DRAW_TEXTOBJ_POPUP;
--- sd/source/ui/view/drviewsj.cxx.old	2009-10-08 16:17:24.000000000 +0200
+++ sd/source/ui/view/drviewsj.cxx	2010-03-03 16:59:59.000000000 +0100
@@ -211,9 +211,6 @@ void DrawViewShell::GetMenuStateSel( Sfx
                 rSet.DisableItem( SID_CHANGEPOLYGON );
             }
 
-            if(nInv == SdrInventor && (nId == OBJ_TITLETEXT || nId == OBJ_OUTLINETEXT))
-                rSet.DisableItem( SID_TEXTATTR_DLG );
-
             if(nInv == SdrInventor && nId == OBJ_TABLE )
             {
                 rSet.DisableItem( SID_TEXTATTR_DLG );
@@ -306,6 +303,17 @@ void DrawViewShell::GetMenuStateSel( Sfx
             else
                 rSet.DisableItem( SID_MODIFY_FIELD );
         }
+        if( SFX_ITEM_AVAILABLE == rSet.GetItemState( SID_OUTLINE_TEXT_AUTOFIT ) )
+        {
+            bool bSet = false;
+            const SdrMarkList& rMarkList = mpDrawView->GetMarkedObjectList();
+            if( rMarkList.GetMarkCount() == 1 )
+            {
+                SdrObject* pObj = rMarkList.GetMark(0)->GetMarkedSdrObj();
+                bSet = ((const SdrTextFitToSizeTypeItem*)pObj->GetMergedItemSet().GetItem(SDRATTR_TEXT_FITTOSIZE))->GetValue() != SDRTEXTFIT_NONE;
+            }
+            rSet.Put(SfxBoolItem(SID_OUTLINE_TEXT_AUTOFIT, bSet));
+        }
 
         rSet.DisableItem( SID_GROUP );
         rSet.DisableItem( SID_COMBINE );
@@ -355,7 +363,6 @@ void DrawViewShell::GetMenuStateSel( Sfx
             BOOL bGraf = FALSE;
             BOOL bDrawObj = FALSE;
             BOOL b3dObj = FALSE;
-            BOOL bTitOutText = FALSE;
             bool bTable = false;
             BOOL bMeasureObj = FALSE;
             BOOL bEdgeObj = FALSE; // Connector
@@ -395,8 +402,6 @@ void DrawViewShell::GetMenuStateSel( Sfx
 
                         case OBJ_GRAF: bGraf = TRUE; break;
 
-                        case OBJ_TITLETEXT:
-                        case OBJ_OUTLINETEXT: bTitOutText = TRUE; break;
                         case OBJ_TABLE: bTable = true; break;
                     }
                 }
@@ -444,7 +449,7 @@ void DrawViewShell::GetMenuStateSel( Sfx
             {
                 rSet.DisableItem( SID_UNGROUP );
             }
-            if( bTitOutText || bTable )
+            if( bTable )
                 rSet.DisableItem( SID_TEXTATTR_DLG );
 
             if( !bMeasureObj )
