diff --git svx/source/svdraw/svdotextdecomposition.cxx svx/source/svdraw/svdotextdecomposition.cxx
index 2cc2385..2905cab 100644
--- svx/source/svdraw/svdotextdecomposition.cxx
+++ svx/source/svdraw/svdotextdecomposition.cxx
@@ -1179,9 +1179,11 @@ bool SdrTextObj::impDecomposeAutoFitTextPrimitive(
 	aNewTransformB.rotate(fRotate);
 	aNewTransformB.translate(aTranslate.getX(), aTranslate.getY());
 
+    basegfx::B2DRange aClipRange;
+
 	// now break up text primitives.
 	impTextBreakupHandler aConverter(rOutliner);
-	aConverter.decomposeBlockTextPrimitive(aNewTransformA, aNewTransformB);
+	aConverter.decomposeBlockTextPrimitive(aNewTransformA, aNewTransformB, aClipRange);
 
 	// cleanup outliner
 	rOutliner.Clear();
