--- vcl/unx/source/plugadapt/salplug.cxx
+++ vcl/unx/source/plugadapt/salplug.cxx
@@ -205,14 +205,16 @@
 
     static const char* pUsePlugin = getenv( "SAL_USE_VCLPLUGIN" );
 
-    if( !(pUsePlugin && *pUsePlugin) )
+    if( pUsePlugin && *pUsePlugin )
+        pInst = tryInstance( OUString::createFromAscii( pUsePlugin ) );
+    else
         pInst = check_headless_plugin();
     
     if( ! pInst && !(pUsePlugin && *pUsePlugin) )
         pInst = autodetect_plugin();
     
     // fallback, try everything
-    const char* pPlugin[] = { "gtk", "kde", "gen", 0 };
+    const char* pPlugin[] = { "gtk", "kde4", "kde", "gen", 0 };
 
     for ( int i = 0; !pInst && pPlugin[ i ]; ++i )
         pInst = tryInstance( OUString::createFromAscii( pPlugin[ i ] ) );
