diff --git sfx2/source/doc/sfxbasemodel.cxx sfx2/source/doc/sfxbasemodel.cxx
index 28355da..1ebd5ae 100644
--- sfx2/source/doc/sfxbasemodel.cxx
+++ sfx2/source/doc/sfxbasemodel.cxx
@@ -3624,7 +3624,12 @@ void SAL_CALL SfxBaseModel::switchToStorage( const uno::Reference< XSTORAGE >& x
                                             uno::Reference< uno::XInterface >(),
                                             nError ? nError : ERRCODE_IO_GENERAL );
     }
-
+    // UICfgMgr has a reference to the old storage, update it
+    uno::Reference< ui::XUIConfigurationStorage > xUICfgMgrStorage( getUIConfigurationManager(), uno::UNO_QUERY );
+    if ( xUICfgMgrStorage.is() )
+    {
+        xUICfgMgrStorage->setStorage( xStorage ); 
+    } 
     m_pData->m_pObjectShell->Get_Impl()->bOwnsStorage = FALSE;
 }
 
