--- framework/source/uiconfiguration/uiconfigurationmanagerimpl.cxx	2009-09-15 21:10:20.000000000 +0100
+++ framework.patched/source/uiconfiguration/uiconfigurationmanagerimpl.cxx	2009-09-17 16:23:59.000000000 +0100
@@ -1281,7 +1281,7 @@ throw ( ElementExistException, IllegalAr
                 bInsertData   = true;
             }
         }
-        if ( !pDataSettings )
+        if ( !pDataSettings || bInsertData )
         {
             aUIElementData.bDefault     = false;
             if ( !m_bUseDefault )
