--- officecfg/registry/data/org/openoffice/Office/Accelerators.xcu	2009-11-25 11:07:30.000000000 +0000
+++ officecfg/registry/data/org/openoffice/Office/Accelerators.xcu	2009-11-25 11:22:09.000000000 +0000
@@ -196,6 +196,7 @@
     <node oor:name="A_MOD1" oor:op="replace">
      <prop oor:name="Command"><value xml:lang="x-no-translate">I10N SHORTCUTS - NO TRANSLATE</value>
       <value xml:lang="en-US">.uno:SelectAll</value>
+      <value xml:lang="es">.uno:Open</value>
      </prop>
     </node>
     <node oor:name="C_MOD1" oor:op="replace">
@@ -397,6 +398,7 @@
      <prop oor:name="Command"><value xml:lang="x-no-translate">I10N SHORTCUTS - NO TRANSLATE</value>
       <value xml:lang="de">.uno:AlignBlock</value>
       <value xml:lang="en-US">.uno:Bold</value>
+      <value xml:lang="es">.uno:SearchDialog</value>
      </prop>
     </node>
     <node oor:name="B_SHIFT_MOD1" oor:op="replace">
@@ -462,6 +464,7 @@
     <node oor:name="E_MOD1" oor:op="replace">
      <prop oor:name="Command"><value xml:lang="x-no-translate">I10N SHORTCUTS - NO TRANSLATE</value>
       <value xml:lang="en-US">.uno:AlignHorizontalCenter</value>
+      <value xml:lang="es">.uno:SelectAll</value>
      </prop>
     </node>
     <node oor:name="F11" oor:op="replace">
@@ -831,6 +834,7 @@
     <node oor:name="U_MOD1" oor:op="replace">
      <prop oor:name="Command"><value xml:lang="x-no-translate">I10N SHORTCUTS - NO TRANSLATE</value>
       <value xml:lang="en-US">.uno:Underline</value>
+      <value xml:lang="es">.uno:AddDirect</value>
      </prop>
     </node>
     <node oor:name="U_SHIFT_MOD1" oor:op="replace">
@@ -991,6 +995,7 @@
      <prop oor:name="Command"><value xml:lang="x-no-translate">I10N SHORTCUTS - NO TRANSLATE</value>
       <value xml:lang="de">.uno:JustifyPara</value>
       <value xml:lang="en-US">.uno:Bold</value>
+      <value xml:lang="es">.uno:SearchDialog</value>
      </prop>
     </node>
     <node oor:name="B_SHIFT_MOD1" oor:op="replace">
@@ -1026,6 +1031,7 @@
     <node oor:name="E_MOD1" oor:op="replace">
      <prop oor:name="Command"><value xml:lang="x-no-translate">I10N SHORTCUTS - NO TRANSLATE</value>
       <value xml:lang="en-US">.uno:CenterPara</value>
+      <value xml:lang="es">.uno:SelectAll</value>
      </prop>
     </node>
     <node oor:name="F11" oor:op="replace">
@@ -1231,6 +1237,7 @@
     <node oor:name="U_MOD1" oor:op="replace">
      <prop oor:name="Command"><value xml:lang="x-no-translate">I10N SHORTCUTS - NO TRANSLATE</value>
       <value xml:lang="en-US">.uno:Underline</value>
+      <value xml:lang="es">.uno:AddDirect</value>
      </prop>
     </node>
     <node oor:name="U_SHIFT_MOD1" oor:op="replace">
@@ -1339,6 +1346,7 @@
      <prop oor:name="Command"><value xml:lang="x-no-translate">I10N SHORTCUTS - NO TRANSLATE</value>
       <value xml:lang="de">.uno:JustifyPara</value>
       <value xml:lang="en-US">.uno:Bold</value>
+      <value xml:lang="es">.uno:SearchDialog</value>
      </prop>
     </node>
     <node oor:name="B_SHIFT_MOD1" oor:op="replace">
@@ -1424,6 +1432,7 @@
     <node oor:name="E_MOD1" oor:op="replace">
      <prop oor:name="Command"><value xml:lang="x-no-translate">I10N SHORTCUTS - NO TRANSLATE</value>
       <value xml:lang="en-US">.uno:CenterPara</value>
+      <value xml:lang="es">.uno:SelectAll</value>
      </prop>
     </node>
     <node oor:name="F10_MOD1" oor:op="replace">
@@ -1838,6 +1847,7 @@
     <node oor:name="U_MOD1" oor:op="replace">
      <prop oor:name="Command"><value xml:lang="x-no-translate">I10N SHORTCUTS - NO TRANSLATE</value>
       <value xml:lang="en-US">.uno:Underline</value>
+      <value xml:lang="es">.uno:AddDirect</value>
      </prop>
     </node>
     <node oor:name="U_SHIFT_MOD1" oor:op="replace">
@@ -1928,6 +1938,7 @@
      <prop oor:name="Command"><value xml:lang="x-no-translate">I10N SHORTCUTS - NO TRANSLATE</value>
       <value xml:lang="de">.uno:JustifyPara</value>
       <value xml:lang="en-US">.uno:Bold</value>
+      <value xml:lang="es">.uno:SearchDialog</value>
      </prop>
     </node>
     <node oor:name="B_SHIFT_MOD1" oor:op="replace">
@@ -1968,6 +1979,7 @@
     <node oor:name="E_MOD1" oor:op="replace">
      <prop oor:name="Command"><value xml:lang="x-no-translate">I10N SHORTCUTS - NO TRANSLATE</value>
       <value xml:lang="en-US">.uno:CenterPara</value>
+      <value xml:lang="es">.uno:SelectAll</value>
      </prop>
     </node>
     <node oor:name="F11" oor:op="replace">
@@ -2198,6 +2210,7 @@
     <node oor:name="U_MOD1" oor:op="replace">
      <prop oor:name="Command"><value xml:lang="x-no-translate">I10N SHORTCUTS - NO TRANSLATE</value>
       <value xml:lang="en-US">.uno:Underline</value>
+      <value xml:lang="es">.uno:AddDirect</value>
      </prop>
     </node>
     <node oor:name="U_SHIFT_MOD1" oor:op="replace">
@@ -2240,8 +2253,14 @@
     <node oor:name="A_MOD1" oor:op="replace">
      <prop oor:name="Command"><value xml:lang="x-no-translate">I10N SHORTCUTS - NO TRANSLATE</value>
       <value xml:lang="en-US">.uno:Select</value>
+      <value xml:lang="es">.uno:Open</value>
      </prop>
     </node>
+     <node oor:name="E_MOD1" oor:op="replace">
+      <prop oor:name="Command"><value xml:lang="x-no-translate">I10N SHORTCUTS - NO TRANSLATE</value>
+       <value xml:lang="es">.uno:Select</value>
+      </prop>
+     </node>
     <node oor:name="F3" oor:op="replace">
      <prop oor:name="Command"><value xml:lang="x-no-translate">I10N SHORTCUTS - NO TRANSLATE</value>
       <value xml:lang="en-US">.uno:NextError</value>
@@ -2358,6 +2377,7 @@
      <prop oor:name="Command"><value xml:lang="x-no-translate">I10N SHORTCUTS - NO TRANSLATE</value>
       <value xml:lang="de">.uno:JustifyPara</value>
       <value xml:lang="en-US">.uno:Bold</value>
+      <value xml:lang="es">.uno:SearchDialog</value>
      </prop>
     </node>
     <node oor:name="B_SHIFT_MOD1" oor:op="replace">
@@ -2443,6 +2463,7 @@
     <node oor:name="E_MOD1" oor:op="replace">
      <prop oor:name="Command"><value xml:lang="x-no-translate">I10N SHORTCUTS - NO TRANSLATE</value>
       <value xml:lang="en-US">.uno:CenterPara</value>
+      <value xml:lang="es">.uno:SelectAll</value>
      </prop>
     </node>
     <node oor:name="F10_MOD1" oor:op="replace">
@@ -2857,6 +2878,7 @@
     <node oor:name="U_MOD1" oor:op="replace">
      <prop oor:name="Command"><value xml:lang="x-no-translate">I10N SHORTCUTS - NO TRANSLATE</value>
       <value xml:lang="en-US">.uno:Underline</value>
+      <value xml:lang="es">.uno:AddDirect</value>
      </prop>
     </node>
     <node oor:name="U_SHIFT_MOD1" oor:op="replace">
@@ -2970,6 +2992,7 @@
      <prop oor:name="Command"><value xml:lang="x-no-translate">I10N SHORTCUTS - NO TRANSLATE</value>
       <value xml:lang="de">.uno:JustifyPara</value>
       <value xml:lang="en-US">.uno:Bold</value>
+      <value xml:lang="es">.uno:SearchDialog</value>
      </prop>
     </node>
     <node oor:name="B_SHIFT_MOD1" oor:op="replace">
@@ -3060,6 +3083,7 @@
     <node oor:name="E_MOD1" oor:op="replace">
      <prop oor:name="Command"><value xml:lang="x-no-translate">I10N SHORTCUTS - NO TRANSLATE</value>
       <value xml:lang="en-US">.uno:CenterPara</value>
+      <value xml:lang="es">.uno:SelectAll</value>
      </prop>
     </node>
     <node oor:name="F10_MOD1" oor:op="replace">
@@ -3494,6 +3518,7 @@
     <node oor:name="U_MOD1" oor:op="replace">
      <prop oor:name="Command"><value xml:lang="x-no-translate">I10N SHORTCUTS - NO TRANSLATE</value>
       <value xml:lang="en-US">.uno:Underline</value>
+      <value xml:lang="es">.uno:AddDirect</value>
      </prop>
     </node>
     <node oor:name="U_SHIFT_MOD1" oor:op="replace">
@@ -3592,6 +3617,7 @@
      <prop oor:name="Command"><value xml:lang="x-no-translate">I10N SHORTCUTS - NO TRANSLATE</value>
       <value xml:lang="de">.uno:JustifyPara</value>
       <value xml:lang="en-US">.uno:Bold</value>
+      <value xml:lang="es">.uno:SearchDialog</value>
      </prop>
     </node>
     <node oor:name="B_SHIFT_MOD1" oor:op="replace">
@@ -3677,6 +3703,7 @@
     <node oor:name="E_MOD1" oor:op="replace">
      <prop oor:name="Command"><value xml:lang="x-no-translate">I10N SHORTCUTS - NO TRANSLATE</value>
       <value xml:lang="en-US">.uno:CenterPara</value>
+      <value xml:lang="es">.uno:SelectAll</value>
      </prop>
     </node>
     <node oor:name="F10_MOD1" oor:op="replace">
@@ -4091,6 +4118,7 @@
     <node oor:name="U_MOD1" oor:op="replace">
      <prop oor:name="Command"><value xml:lang="x-no-translate">I10N SHORTCUTS - NO TRANSLATE</value>
       <value xml:lang="en-US">.uno:Underline</value>
+      <value xml:lang="es">.uno:AddDirect</value>
      </prop>
     </node>
     <node oor:name="U_SHIFT_MOD1" oor:op="replace">
@@ -4194,6 +4222,7 @@
      <prop oor:name="Command"><value xml:lang="x-no-translate">I10N SHORTCUTS - NO TRANSLATE</value>
       <value xml:lang="de">.uno:JustifyPara</value>
       <value xml:lang="en-US">.uno:Bold</value>
+      <value xml:lang="es">.uno:SearchDialog</value>
      </prop>
     </node>
     <node oor:name="B_SHIFT_MOD1" oor:op="replace">
@@ -4279,6 +4308,7 @@
     <node oor:name="E_MOD1" oor:op="replace">
      <prop oor:name="Command"><value xml:lang="x-no-translate">I10N SHORTCUTS - NO TRANSLATE</value>
       <value xml:lang="en-US">.uno:CenterPara</value>
+      <value xml:lang="es">.uno:SelectAll</value>
      </prop>
     </node>
     <node oor:name="F10_MOD1" oor:op="replace">
@@ -4693,6 +4723,7 @@
     <node oor:name="U_MOD1" oor:op="replace">
      <prop oor:name="Command"><value xml:lang="x-no-translate">I10N SHORTCUTS - NO TRANSLATE</value>
       <value xml:lang="en-US">.uno:Underline</value>
+      <value xml:lang="es">.uno:AddDirect</value>
      </prop>
     </node>
     <node oor:name="U_SHIFT_MOD1" oor:op="replace">
@@ -4801,6 +4832,7 @@
      <prop oor:name="Command"><value xml:lang="x-no-translate">I10N SHORTCUTS - NO TRANSLATE</value>
       <value xml:lang="de">.uno:JustifyPara</value>
       <value xml:lang="en-US">.uno:Bold</value>
+      <value xml:lang="es">.uno:SearchDialog</value>
      </prop>
     </node>
     <node oor:name="B_SHIFT_MOD1" oor:op="replace">
@@ -4886,6 +4918,7 @@
     <node oor:name="E_MOD1" oor:op="replace">
      <prop oor:name="Command"><value xml:lang="x-no-translate">I10N SHORTCUTS - NO TRANSLATE</value>
       <value xml:lang="en-US">.uno:CenterPara</value>
+      <value xml:lang="es">.uno:SelectAll</value>
      </prop>
     </node>
     <node oor:name="F10_MOD1" oor:op="replace">
@@ -5300,6 +5333,7 @@
     <node oor:name="U_MOD1" oor:op="replace">
      <prop oor:name="Command"><value xml:lang="x-no-translate">I10N SHORTCUTS - NO TRANSLATE</value>
       <value xml:lang="en-US">.uno:Underline</value>
+      <value xml:lang="es">.uno:AddDirect</value>
      </prop>
     </node>
     <node oor:name="U_SHIFT_MOD1" oor:op="replace">
