--- vcl/source/glyphs/graphite_layout.cxx	2010-07-30 10:08:56.000000000 +0100
+++ vcl/source/glyphs/graphite_layout.cxx	2010-07-30 10:10:10.000000000 +0100
@@ -659,7 +659,7 @@
         return hash;
     };
 protected:
-    virtual void UniqueCacheInfo(std::wstring & stuFace, bool & fBold, bool & fItalic)
+    virtual void UniqueCacheInfo(ext_std::wstring & stuFace, bool & fBold, bool & fItalic)
     {
 #ifdef WIN32
         dynamic_cast<GraphiteWinFont&>(mrRealFont).UniqueCacheInfo(stuFace, fBold, fItalic);
