---
 svtools/inc/svtools/svtdata.hxx |    7 +++++--
 1 files changed, 5 insertions(+), 2 deletions(-)

diff --git svtools/inc/svtools/svtdata.hxx svtools/inc/svtools/svtdata.hxx
index 8f21b61..d4fb00c 100644
--- svtools/inc/svtools/svtdata.hxx
+++ svtools/inc/svtools/svtdata.hxx
@@ -28,6 +28,8 @@
 #ifndef _SVTOOLS_SVTDATA_HXX
 #define _SVTOOLS_SVTDATA_HXX
 
+#include "svtools/svtdllapi.h"
+
 #include <tools/resid.hxx>
 #include <com/sun/star/lang/Locale.hpp>
 
@@ -51,11 +53,12 @@ private:
 
 public:
     ResMgr * GetResMgr(const ::com::sun::star::lang::Locale aLocale);
-    ResMgr * GetResMgr(); // VCL dependant, only available in SVT, not in SVL!
+    SVT_DLLPUBLIC ResMgr * GetResMgr(); // VCL dependant, only available in SVT, not in SVL!
+
     ResMgr * GetPatchResMgr();
     ResMgr * GetPatchResMgr(const ::com::sun::star::lang::Locale& aLocale);
 
-    static ImpSvtData & GetSvtData();
+    SVT_DLLPUBLIC static ImpSvtData & GetSvtData();
 };
 
 //============================================================================
-- 
1.7.0.1

