---
 framework/source/uielement/menubarmerger.cxx |    9 ++++++++-
 1 files changed, 8 insertions(+), 1 deletions(-)

diff --git framework/source/uielement/menubarmerger.cxx framework/source/uielement/menubarmerger.cxx
index 51f87a4..852e456 100644
--- framework/source/uielement/menubarmerger.cxx
+++ framework/source/uielement/menubarmerger.cxx
@@ -101,6 +101,14 @@ ReferencePathInfo MenuBarMerger::FindReferencePath(
 {
     sal_uInt32       i( 0 );
     const sal_uInt32 nCount( rReferencePath.size() );
+
+    ReferencePathInfo aResult;
+    if ( !nCount )
+    {
+        aResult.eResult = RP_MENUITEM_NOT_FOUND;
+        return aResult;
+    }
+
     Menu*            pCurrMenu( pMenu );
     RPResultInfo     eResult( RP_OK );
 
@@ -142,7 +150,6 @@ ReferencePathInfo MenuBarMerger::FindReferencePath(
     }
     while (( pCurrMenu != 0 ) && ( i < nCount ) && ( eResult == RP_OK ));
 
-    ReferencePathInfo aResult;
     aResult.pPopupMenu = pCurrMenu;
     aResult.nPos       = nPos;
     aResult.nLevel     = nLevel;
-- 
1.7.0.1

