diff -rup framework-orig/source/layoutmanager/layoutmanager.cxx framework/source/layoutmanager/layoutmanager.cxx
--- framework-orig/source/layoutmanager/layoutmanager.cxx	2010-02-11 10:35:54.000000000 +0100
+++ framework/source/layoutmanager/layoutmanager.cxx	2010-02-11 10:41:54.000000000 +0100
@@ -6074,8 +6074,7 @@ void LayoutManager::implts_updateMenuBar
             if ( pMenuBar )
             {
                 // TODO remove link on FALSE ?!
-	      // TODO: override only when we are in small screen mode to not modify old behaviour
-	      if ( TRUE /*bShowCloser*/ )
+            if ( pWindow->GetSmallScreenMode() || bShowCloser )
                 {
                     pMenuBar->ShowCloser( TRUE );
                     pMenuBar->SetCloserHdl( LINK( this, LayoutManager, MenuBarClose ));
