From 6b5b9e44660ac69e8b13ddba4d4a76223085b3bc Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:08:46 +0200
Subject: [PATCH 833/878] framework-netbook-use-small-screen-mode.diff

---
 framework/source/layoutmanager/layoutmanager.cxx |    3 +--
 1 files changed, 1 insertions(+), 2 deletions(-)

diff --git a/framework/source/layoutmanager/layoutmanager.cxx b/framework/source/layoutmanager/layoutmanager.cxx
index c27a7bc..07a23ce 100644
--- a/framework/source/layoutmanager/layoutmanager.cxx
+++ b/framework/source/layoutmanager/layoutmanager.cxx
@@ -6071,8 +6071,7 @@ void LayoutManager::implts_updateMenuBarClose()
             if ( pMenuBar )
             {
                 // TODO remove link on FALSE ?!
-          // TODO: override only when we are in small screen mode to not modify old behaviour
-          if ( TRUE /*bShowCloser*/ )
+            if ( pWindow->GetSmallScreenMode() || bShowCloser )
                 {
                     pMenuBar->ShowCloser( TRUE );
                     pMenuBar->SetCloserHdl( LINK( this, LayoutManager, MenuBarClose ));
-- 
1.7.0.1

