--- framework/source/layoutmanager/layoutmanager.cxx.old	2011-04-01 13:21:18.000000000 +0200
+++ framework/source/layoutmanager/layoutmanager.cxx	2011-04-15 15:19:27.000000000 +0200
@@ -2600,7 +2600,7 @@ void LayoutManager::implts_updateMenuBar
             if ( pMenuBar )
             {
                 // TODO remove link on sal_False ?!
-                pMenuBar->ShowCloser( bShowCloser );
+                pMenuBar->ShowCloser( pSysWindow->GetSmallScreenMode() || bShowCloser );
                 pMenuBar->SetCloserHdl( LINK( this, LayoutManager, MenuBarClose ));
             }
         }
