From db8abeed4d55f8c3e7377f2354d84c995f217043 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:08:39 +0200
Subject: [PATCH 827/878] framework-menubar-fix.diff

---
 framework/source/uielement/menubarmerger.cxx |    9 ++++++++-
 1 files changed, 8 insertions(+), 1 deletions(-)

diff --git a/framework/source/uielement/menubarmerger.cxx b/framework/source/uielement/menubarmerger.cxx
index 12fd910..e8731cd 100644
--- a/framework/source/uielement/menubarmerger.cxx
+++ b/framework/source/uielement/menubarmerger.cxx
@@ -101,6 +101,14 @@ ReferencePathInfo MenuBarMerger::FindReferencePath(
 {
     sal_uInt32       i( 0 );
     const sal_uInt32 nCount( rReferencePath.size() );
+
+    ReferencePathInfo aResult;
+    if ( !nCount )
+    {
+        aResult.eResult = RP_MENUITEM_NOT_FOUND;
+        return aResult;
+    }
+
     Menu*            pCurrMenu( pMenu );
     RPResultInfo     eResult( RP_OK );
 
@@ -142,7 +150,6 @@ ReferencePathInfo MenuBarMerger::FindReferencePath(
     }
     while (( pCurrMenu != 0 ) && ( i < nCount ) && ( eResult == RP_OK ));
 
-    ReferencePathInfo aResult;
     aResult.pPopupMenu = pCurrMenu;
     aResult.nPos       = nPos;
     aResult.nLevel     = nLevel;
-- 
1.7.0.1

