diff --git sdext/source/presenter/registry/data/org/openoffice/Office/Jobs.xcu sdext/source/presenter/registry/data/org/openoffice/Office/Jobs.xcu
index ba74332..1241a72 100644
--- sdext/source/presenter/registry/data/org/openoffice/Office/Jobs.xcu
+++ sdext/source/presenter/registry/data/org/openoffice/Office/Jobs.xcu
@@ -5,6 +5,9 @@
             <prop oor:name="Service">
                 <value>com.sun.star.comp.Draw.framework.PresenterScreenJob</value>
             </prop>
+            <prop oor:name="Context">
+                <value>com.sun.star.presentation.PresentationDocument</value>
+            </prop>
         </node>
     </node>
     <node oor:name="Events">
