---
 framework/source/layoutmanager/layoutmanager.cxx |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git framework/source/layoutmanager/layoutmanager.cxx framework/source/layoutmanager/layoutmanager.cxx
index 1b0cd57..6889d13 100644
--- framework/source/layoutmanager/layoutmanager.cxx
+++ framework/source/layoutmanager/layoutmanager.cxx
@@ -6141,7 +6141,8 @@ void LayoutManager::implts_updateMenuBarClose()
             if ( pMenuBar )
             {
                 // TODO remove link on FALSE ?!
-                if ( bShowCloser )
+          // TODO: override only when we are in small screen mode to not modify old behaviour
+          if ( TRUE /*bShowCloser*/ )
                 {
                     pMenuBar->ShowCloser( TRUE );
                     pMenuBar->SetCloserHdl( LINK( this, LayoutManager, MenuBarClose ));
-- 
1.7.0.1

