From c7973b18d57eed16a9e409b006f058c9f519dcae Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:08:43 +0200
Subject: [PATCH 733/768] framework-netbook.diff

---
 framework/source/layoutmanager/layoutmanager.cxx |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git framework/source/layoutmanager/layoutmanager.cxx framework/source/layoutmanager/layoutmanager.cxx
index 2d57da8..f5047d3 100644
--- framework/source/layoutmanager/layoutmanager.cxx
+++ framework/source/layoutmanager/layoutmanager.cxx
@@ -6133,7 +6133,8 @@ void LayoutManager::implts_updateMenuBarClose()
             if ( pMenuBar )
             {
                 // TODO remove link on FALSE ?!
-                if ( bShowCloser )
+          // TODO: override only when we are in small screen mode to not modify old behaviour
+          if ( TRUE /*bShowCloser*/ )
                 {
                     pMenuBar->ShowCloser( TRUE );
                     pMenuBar->SetCloserHdl( LINK( this, LayoutManager, MenuBarClose ));
-- 
1.7.0.1

