diff -rup ../../build-orig/ooo320-m8/framework//source/layoutmanager/layoutmanager.cxx framework//source/layoutmanager/layoutmanager.cxx
--- ../../build-orig/ooo320-m8/framework//source/layoutmanager/layoutmanager.cxx	2009-12-18 13:41:49.000000000 +0100
+++ framework//source/layoutmanager/layoutmanager.cxx	2010-02-01 16:16:08.000000000 +0100
@@ -6071,7 +6071,8 @@ void LayoutManager::implts_updateMenuBar
             if ( pMenuBar )
             {
                 // TODO remove link on FALSE ?!
-                if ( bShowCloser )
+          // TODO: override only when we are in small screen mode to not modify old behaviour
+          if ( TRUE /*bShowCloser*/ )
                 {
                     pMenuBar->ShowCloser( TRUE );
                     pMenuBar->SetCloserHdl( LINK( this, LayoutManager, MenuBarClose ));
