--- l10n/source/hu/localize.sdf	2010-04-23 21:13:50.939000000 +0200
+++ l10n/source/hu/localize.sdf	2010-04-23 10:30:26.681000000 +0200
@@ -49237,7 +49237,7 @@
 instsetoo_native	inc_openoffice\windows\msi_languages\Control.ulf	0	LngText	OOO_CONTROL_121				0	hu	Egy szinttel feljebb|				2002-02-02 02:02:02
 instsetoo_native	inc_openoffice\windows\msi_languages\Control.ulf	0	LngText	OOO_CONTROL_122				0	hu	< &Vissza				2002-02-02 02:02:02
 instsetoo_native	inc_openoffice\windows\msi_languages\Control.ulf	0	LngText	OOO_CONTROL_123				0	hu	Mégse				2002-02-02 02:02:02
-instsetoo_native	inc_openoffice\windows\msi_languages\Control.ulf	0	LngText	OOO_CONTROL_124				0	hu	A buildet a közösséggel együttműködésben az FSFhu készítette. A stáblista itt található: http://www.openoffice.org/welcome/credits.html				2002-02-02 02:02:02
+instsetoo_native	inc_openoffice\windows\msi_languages\Control.ulf	0	LngText	OOO_CONTROL_124				0	hu	A buildet a közösséggel együttműködésben az FSF.hu Alapítvány készítette. A stáblista itt található: http://www.openoffice.org/welcome/credits.html				2002-02-02 02:02:02
 instsetoo_native	inc_openoffice\windows\msi_languages\Control.ulf	0	LngText	OOO_CONTROL_125				0	hu	&Tovább >				2002-02-02 02:02:02
 instsetoo_native	inc_openoffice\windows\msi_languages\Control.ulf	0	LngText	OOO_CONTROL_126				0	hu	{&TahomaBold10}Üdvözli az [ProductName] Telepítője				2002-02-02 02:02:02
 instsetoo_native	inc_openoffice\windows\msi_languages\Control.ulf	0	LngText	OOO_CONTROL_127				0	hu	A Telepítő telepíteni fogja az [ProductName] alkalmazást a számítógépére. A folytatáshoz kattintson a Tovább gombra.				2002-02-02 02:02:02
