--- instsetoo_native/util/makefile.mk.old	2006-01-16 15:10:22.000000000 +0200
+++ instsetoo_native/util/makefile.mk	2006-01-17 12:58:26.000000000 +0200
@@ -408,9 +408,9 @@
     $(RM) $(MSILANGPACKTEMPLATEDIR)$/Binary$/Image.bmp
     $(RM) $(MSIURETEMPLATEDIR)$/Binary$/Image.bmp
     $(RM) $(MSISDKOOTEMPLATEDIR)$/Binary$/Image.bmp
-    $(COPY) $(PRJ)$/res$/nologoinstall.bmp $(MSIOFFICETEMPLATEDIR)$/Binary$/Image.bmp
-    $(COPY) $(PRJ)$/res$/nologoinstall.bmp $(MSILANGPACKTEMPLATEDIR)$/Binary$/Image.bmp
-    $(COPY) $(PRJ)$/res$/nologoinstall.bmp $(MSIURETEMPLATEDIR)$/Binary$/Image.bmp
-    $(COPY) $(PRJ)$/res$/nologoinstall.bmp $(MSISDKOOTEMPLATEDIR)$/Binary$/Image.bmp
+    $(COPY) $(PRJ)$/res$/nologoinstall_fsfhu.bmp $(MSIOFFICETEMPLATEDIR)$/Binary$/Image.bmp
+    $(COPY) $(PRJ)$/res$/nologoinstall_fsfhu.bmp $(MSILANGPACKTEMPLATEDIR)$/Binary$/Image.bmp
+    $(COPY) $(PRJ)$/res$/nologoinstall_fsfhu.bmp $(MSISDKOOTEMPLATEDIR)$/Binary$/Image.bmp
+    $(COPY) $(PRJ)$/res$/nologoinstall_fsfhu.bmp $(MSIURETEMPLATEDIR)$/Binary$/Image.bmp
 
 

--- instsetoo_native/util/openoffice.lst.old	2010-05-13 07:59:38.161250000 +0200
+++ instsetoo_native/util/openoffice.lst	2010-05-16 14:27:06.231375000 +0200
@@ -91,7 +91,7 @@
 			LIBRARYVERSION 9.3.0
 			PATCHCODEFILE ooo_patchcodes.txt
 			DOWNLOADBANNER	ooobanner.bmp
-			DOWNLOADBITMAP	ooobitmap.bmp
+			DOWNLOADBITMAP	ooobitmap_fsfhu.bmp
 			DOWNLOADSETUPICO	ooosetup.ico
 			RELATIVE_PATHES_IN_DDF 1
             STARTCENTER_ADDFEATURE_URL http://extensions.go-oo.org/getmore?cid=920794
@@ -157,7 +157,7 @@
 			PATCHCODEFILE ooo_patchcodes.txt
 			JAVAPRODUCT 1
 			DOWNLOADBANNER	ooobanner.bmp
-			DOWNLOADBITMAP	ooobitmap.bmp
+			DOWNLOADBITMAP	ooobitmap_fsfhu.bmp
 			DOWNLOADSETUPICO	ooosetup.ico
 			RELATIVE_PATHES_IN_DDF 1
             STARTCENTER_ADDFEATURE_URL http://extensions.go-oo.org/getmore?cid=920794
@@ -228,7 +228,7 @@
 			PATCHCODEFILE ooodev_patchcodes.txt
 			CODEFILENAME codes_ooodev.txt
 			DOWNLOADBANNER	ooobanner.bmp
-			DOWNLOADBITMAP	ooobitmap.bmp
+			DOWNLOADBITMAP	ooobitmap_fsfhu.bmp
 			DOWNLOADSETUPICO	ooosetup.ico
 			LOCALUSERDIR $ORIGIN/..
 			RELATIVE_PATHES_IN_DDF 1
