--- sw/source/core/text/inftxt.cxx.old	2010-07-22 13:25:45.000000000 +0200
+++ sw/source/core/text/inftxt.cxx	2010-07-29 16:36:16.000000000 +0200
@@ -84,6 +84,10 @@
 
 #include <unomid.h>
 
+#ifdef DEBUG
+#include <stdio.h>
+#endif
+
 using namespace ::com::sun::star;
 using namespace ::com::sun::star::linguistic2;
 using namespace ::com::sun::star::uno;
--- sw/source/filter/ww8/WW8Sttbf.cxx.old	2010-07-22 13:25:45.000000000 +0200
+++ sw/source/filter/ww8/WW8Sttbf.cxx	2010-07-29 16:36:16.000000000 +0200
@@ -31,6 +31,10 @@
 #include "WW8Sttbf.hxx"
 #include <cstdio>
 
+#ifdef DEBUG
+#include <stdio.h>
+#endif
+
 namespace ww8
 {
     WW8Struct::WW8Struct(SvStream& rSt, sal_uInt32 nPos, sal_uInt32 nSize)
--- sw/source/filter/ww8/wrtww8gr.cxx.old	2010-07-22 13:25:45.000000000 +0200
+++ sw/source/filter/ww8/wrtww8gr.cxx	2010-07-29 16:36:16.000000000 +0200
@@ -79,6 +79,10 @@
 #include "docsh.hxx"
 #include <cstdio>
 
+#if OSL_DEBUG_LEVEL > 0
+#include <stdio.h>
+#endif
+
 using namespace ::com::sun::star;
 using namespace nsFieldFlags;
 
--- writerfilter/source/dmapper/DomainMapperTableHandler.cxx.old	2010-07-22 13:12:44.000000000 +0200
+++ writerfilter/source/dmapper/DomainMapperTableHandler.cxx	2010-07-29 16:36:16.000000000 +0200
@@ -36,6 +36,10 @@
 #include <PropertyMapHelper.hxx>
 #endif
 
+#if OSL_DEBUG_LEVEL > 1
+#include <stdio.h>
+#endif
+
 namespace writerfilter {
 namespace dmapper {
 
--- writerfilter/source/dmapper/DomainMapper_Impl.cxx.old	2010-07-22 13:12:44.000000000 +0200
+++ writerfilter/source/dmapper/DomainMapper_Impl.cxx	2010-07-29 16:36:16.000000000 +0200
@@ -79,6 +79,7 @@
 #include <ooxml/OOXMLFastTokens.hxx>
 
 #if DEBUG
+#include <stdio.h>
 #include <com/sun/star/lang/XServiceInfo.hpp>
 #include <com/sun/star/style/TabStop.hpp>
 #endif
