--- sw/source/core/doc/dbgoutsw.cxx.old	2009-10-21 11:49:24.000000000 +0200
+++ sw/source/core/doc/dbgoutsw.cxx	2010-03-03 16:59:22.000000000 +0100
@@ -54,7 +54,7 @@
 #include <SwNodeNum.hxx>
 #include <dbgoutsw.hxx>
 #include <SwRewriter.hxx>
-#include <iostream>
+#include <stdio.h>
 
 using namespace std;
 
--- sw/source/core/text/inftxt.cxx.old	2009-10-08 16:17:18.000000000 +0200
+++ sw/source/core/text/inftxt.cxx	2010-03-03 16:59:22.000000000 +0100
@@ -87,6 +87,10 @@
 
 #include <unomid.h>
 
+#ifdef DEBUG
+#include <stdio.h>
+#endif
+
 using namespace ::com::sun::star;
 using namespace ::com::sun::star::linguistic2;
 using namespace ::com::sun::star::uno;
--- sw/source/filter/ww8/WW8Sttbf.cxx.old	2009-10-21 11:49:24.000000000 +0200
+++ sw/source/filter/ww8/WW8Sttbf.cxx	2010-03-03 16:59:22.000000000 +0100
@@ -33,6 +33,10 @@
 #include <dbgoutsw.hxx>
 #include "WW8Sttbf.hxx"
 
+#ifdef DEBUG
+#include <stdio.h>
+#endif
+
 namespace ww8
 {
     WW8Struct::WW8Struct(SvStream& rSt, sal_uInt32 nPos, sal_uInt32 nSize)
--- sw/source/filter/ww8/wrtw8nds.cxx.old	2010-03-03 16:44:12.000000000 +0100
+++ sw/source/filter/ww8/wrtw8nds.cxx	2010-03-03 16:59:22.000000000 +0100
@@ -106,6 +106,10 @@
 #include <ndgrf.hxx>
 #include <ndole.hxx>
 
+#if OSL_DEBUG_LEVEL > 0
+#include <stdio.h>
+#endif
+
 using namespace ::com::sun::star;
 using namespace ::com::sun::star::i18n;
 using namespace sw::util;
--- sw/source/filter/ww8/wrtww8gr.cxx.old	2009-10-08 16:17:18.000000000 +0200
+++ sw/source/filter/ww8/wrtww8gr.cxx	2010-03-03 16:59:22.000000000 +0100
@@ -77,6 +77,10 @@
 
 #include "docsh.hxx"
 
+#if OSL_DEBUG_LEVEL > 0
+#include <stdio.h>
+#endif
+
 using namespace ::com::sun::star;
 using namespace nsFieldFlags;
 
--- writerfilter/source/dmapper/DomainMapperTableHandler.cxx.old	2009-11-20 11:38:33.000000000 +0100
+++ writerfilter/source/dmapper/DomainMapperTableHandler.cxx	2010-03-03 16:59:22.000000000 +0100
@@ -34,7 +34,7 @@
 #include <com/sun/star/table/TableBorder.hpp>
 #include <com/sun/star/text/HoriOrientation.hpp>
 #if OSL_DEBUG_LEVEL > 1
-#include <iostream>
+#include <stdio.h>
 #endif
 
 namespace writerfilter {
--- writerfilter/source/dmapper/DomainMapper_Impl.cxx.old	2009-11-20 11:38:33.000000000 +0100
+++ writerfilter/source/dmapper/DomainMapper_Impl.cxx	2010-03-03 16:59:22.000000000 +0100
@@ -81,6 +81,7 @@
 #include <ooxml/OOXMLFastTokens.hxx>
 
 #if DEBUG
+#include <stdio.h>
 #include <com/sun/star/lang/XServiceInfo.hpp>
 #endif
 
--- writerfilter/source/dmapper/SettingsTable.cxx.old	2009-11-20 11:38:33.000000000 +0100
+++ writerfilter/source/dmapper/SettingsTable.cxx	2010-03-03 16:59:22.000000000 +0100
@@ -6,7 +6,7 @@
 #include <com/sun/star/text/XTextDocument.hpp>
 
 #if DEBUG
-#include <iostream>
+#include <stdio.h>
 #endif
 
 using namespace com::sun::star;
