From 7758a4e758cbc254c954421fc10cab46ea5c74bc Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:54:49 +0200
Subject: [PATCH 125/878] gnome-vfs-locking-config_office.diff

---
 configure.in |   12 ++++++++++++
 set_soenv.in |    1 +
 2 files changed, 13 insertions(+), 0 deletions(-)

diff --git a/configure.in b/configure.in
index 40f87b3..8d5e7e8 100644
--- a/configure.in
+++ b/configure.in
@@ -5673,6 +5673,7 @@ dnl Gnome VFS check
 dnl ===================================================================
 
 ENABLE_GNOMEVFS=""
+GNOMEVFS_HAS_LOCKING=""
 AC_MSG_CHECKING([whether to enable GNOME VFS support])
 if test "$_os" != "WINNT" -a "$_os" != "Darwin" -a "$enable_gnome_vfs" = "yes"; then
     ENABLE_GNOMEVFS="TRUE"
@@ -5681,10 +5682,21 @@ if test "$_os" != "WINNT" -a "$_os" != "Darwin" -a "$enable_gnome_vfs" = "yes";
     if test "$ENABLE_GCONF" != "TRUE"; then
         PKG_CHECK_MODULES( GCONF, gconf-2.0 )
     fi
+
+    AC_LANG_SAVE
+    AC_LANG_CPLUSPLUS
+    save_CXXFLAGS=$CXXFLAGS
+    CXXFLAGS="$CXXFLAGS $GNOMEVFS_CFLAGS"
+    AC_CHECK_DECL(GNOME_VFS_OPEN_LOCKED,
+                  GNOMEVFS_HAS_LOCKING="YES",,
+                  [#include <libgnomevfs/gnome-vfs-handle.h>])
+    CXXFLAGS=$save_CXXFLAGS
+    AC_LANG_RESTORE
 else
     AC_MSG_RESULT([no])
 fi
 AC_SUBST(ENABLE_GNOMEVFS)
+AC_SUBST(GNOMEVFS_HAS_LOCKING)
 
 dnl ===================================================================
 dnl Check whether the gtk 2.0 libraries are available.
diff --git a/set_soenv.in b/set_soenv.in
index b1881ca..ebf5784 100644
--- a/set_soenv.in
+++ b/set_soenv.in
@@ -1680,6 +1680,7 @@ ToFile( "SERVLETAPI_JAR",    "@SERVLETAPI_JAR@",   "e" );
 ToFile( "ENABLE_DBUS",       "@ENABLE_DBUS@",      "e" );
 ToFile( "ENABLE_GCONF",      "@ENABLE_GCONF@",     "e" );
 ToFile( "ENABLE_GNOMEVFS",   "@ENABLE_GNOMEVFS@",  "e" );
+ToFile( "GNOMEVFS_HAS_LOCKING","@GNOMEVFS_HAS_LOCKING@",  "e" );
 ToFile( "ENABLE_GIO",        "@ENABLE_GIO@",       "e" );
 ToFile( "ENABLE_KDE",        "@ENABLE_KDE@",       "e" );
 ToFile( "KDE_CFLAGS",        "@KDE_CFLAGS@",       "e" );
-- 
1.7.0.1

