--- ucb/source/ucp/gvfs/content.cxx.old	2009-12-07 17:53:02.000000000 +0200
+++ ucb/source/ucp/gvfs/content.cxx	2009-12-11 18:21:42.000000000 +0200
@@ -1719,9 +1719,17 @@
                     ucbhelper::InteractionSupplyAuthentication > & xSupp
                     = xRequest->getAuthenticationSupplier();
 
-                aUserName = xSupp->getUserName();
-                aDomain   = xSupp->getRealm();
-                aPassword = xSupp->getPassword();
+        ::rtl::OUString aNewDomain, aNewUserName, aNewPassword;
+
+                aNewUserName = xSupp->getUserName();
+        if (aNewUserName.getLength())
+            aUserName = aNewUserName;
+                aNewDomain   = xSupp->getRealm();
+        if (aNewDomain.getLength())
+            aDomain = aNewDomain;
+                aNewPassword = xSupp->getPassword();
+        if (aNewPassword.getLength())
+            aPassword = aNewPassword;
 
                 {
                     osl::Guard< osl::Mutex > aGuard( osl::Mutex::getGlobalMutex() );
