--- instsetoo_native/util/openoffice.lst.old	2010-03-03 17:50:21.000000000 +0100
+++ instsetoo_native/util/openoffice.lst	2010-03-03 17:50:36.000000000 +0100
@@ -25,10 +25,10 @@ Globals
             POOLPRODUCT 1
             REGISTRATION_HOST https://registration2.services.openoffice.org/RegistrationWeb
 			REGISTRATIONURL http://survey.services.openoffice.org/user/index.php
-			PROGRESSBARCOLOR 4,68,155
-			PROGRESSSIZE 412,7
-			PROGRESSPOSITION 14,230
-			PROGRESSFRAMECOLOR 58,159,232
+            PROGRESSBARCOLOR 186,189,182
+            PROGRESSSIZE 186,5
+            PROGRESSPOSITION 126,183
+            PROGRESSFRAMECOLOR 255,255,255
 			REGISTRYLAYERNAME Layers
             SERVICEPACK 1
             UPDATE_DATABASE 1
