--- instsetoo_native/inc_openoffice/windows/msi_languages/Control.ulf.old	2010-10-07 23:40:27.000000000 +0200
+++ instsetoo_native/inc_openoffice/windows/msi_languages/Control.ulf	2010-10-15 22:58:29.227734400 +0200
@@ -284,7 +284,7 @@ en-US = "< &Back"
 en-US = "Cancel"
 
 [OOO_CONTROL_124]
-en-US = "Build contributed in collaboration with the community by The Document Foundation. For credits, see: http://www.documentfoundation.org"
+en-US = "Build contributed in collaboration with the community by Go-oo Project. For credits, see: http://www.documentfoundation.org"
 
 [OOO_CONTROL_125]
 en-US = "&Next >"
--- instsetoo_native/util/makefile.mk.old	2010-10-15 21:12:29.412769600 +0200
+++ instsetoo_native/util/makefile.mk	2010-10-15 22:58:29.237748800 +0200
@@ -319,10 +319,14 @@ hack_msitemplates .PHONY:
     $(RM) $(MSILANGPACKTEMPLATEDIR)$/Binary$/Image.bmp
     $(RM) $(MSIURETEMPLATEDIR)$/Binary$/Image.bmp
     $(RM) $(MSISDKOOTEMPLATEDIR)$/Binary$/Image.bmp
-    $(COPY) $(PRJ)$/res$/nologoinstall.bmp $(MSIOFFICETEMPLATEDIR)$/Binary$/Image.bmp
-    $(COPY) $(PRJ)$/res$/nologoinstall.bmp $(MSILANGPACKTEMPLATEDIR)$/Binary$/Image.bmp
-    $(COPY) $(PRJ)$/res$/nologoinstall.bmp $(MSIURETEMPLATEDIR)$/Binary$/Image.bmp
-    $(COPY) $(PRJ)$/res$/nologoinstall.bmp $(MSISDKOOTEMPLATEDIR)$/Binary$/Image.bmp
+    $(COPY) $(PRJ)$/res$/nologoinstall_go-oo.bmp $(MSIOFFICETEMPLATEDIR)$/Binary$/Image.bmp
+    $(COPY) $(PRJ)$/res$/nologoinstall_go-oo.bmp $(MSILANGPACKTEMPLATEDIR)$/Binary$/Image.bmp
+    $(COPY) $(PRJ)$/res$/nologoinstall_go-oo.bmp $(MSIURETEMPLATEDIR)$/Binary$/Image.bmp
+    $(COPY) $(PRJ)$/res$/nologoinstall_go-oo.bmp $(MSISDKOOTEMPLATEDIR)$/Binary$/Image.bmp
+    $(COPY) $(PRJ)$/res$/banner_go-oo.bmp $(MSIOFFICETEMPLATEDIR)$/Binary$/Banner.bmp
+    $(COPY) $(PRJ)$/res$/banner_go-oo.bmp $(MSILANGPACKTEMPLATEDIR)$/Binary$/Banner.bmp
+    $(COPY) $(PRJ)$/res$/banner_go-oo.bmp $(MSISDKOOTEMPLATEDIR)$/Binary$/Banner.bmp
+    $(COPY) $(PRJ)$/res$/banner_go-oo.bmp $(MSIURETEMPLATEDIR)$/Binary$/Banner.bmp
 
 .IF "$(OS)" == "WNT"
 
--- instsetoo_native/util/openoffice.lst.old	2010-10-15 21:13:31.922654400 +0200
+++ instsetoo_native/util/openoffice.lst	2010-10-15 22:58:29.247763200 +0200
@@ -90,9 +90,9 @@ LibreOffice
 			USE_FILEVERSION 1
 			LIBRARYVERSION 9.3.0
 			PATCHCODEFILE ooo_patchcodes.txt
-			DOWNLOADBANNER	ooobanner.bmp
-			DOWNLOADBITMAP	ooobitmap.bmp
-			DOWNLOADSETUPICO	ooosetup.ico
+            DOWNLOADBANNER	ooobanner_go-oo.bmp
+            DOWNLOADBITMAP	ooobitmap_go-oo.bmp
+            DOWNLOADSETUPICO	ooosetup_go-oo.ico
 			RELATIVE_PATHES_IN_DDF 1
             STARTCENTER_ADDFEATURE_URL http://extensions.documentfoundation.org/
             STARTCENTER_INFO_URL http://www.documentfoundation.org/
@@ -156,9 +156,9 @@ LibreOffice_wJRE
 			LIBRARYVERSION 9.3.0
 			PATCHCODEFILE ooo_patchcodes.txt
 			JAVAPRODUCT 1
-			DOWNLOADBANNER	ooobanner.bmp
-			DOWNLOADBITMAP	ooobitmap.bmp
-			DOWNLOADSETUPICO	ooosetup.ico
+            DOWNLOADBANNER	ooobanner_go-oo.bmp
+            DOWNLOADBITMAP	ooobitmap_go-oo.bmp
+            DOWNLOADSETUPICO	ooosetup_go-oo.ico
 			RELATIVE_PATHES_IN_DDF 1
             STARTCENTER_ADDFEATURE_URL http://extensions.documentfoundation.org/
             STARTCENTER_INFO_URL http://www.documentfoundation.org/
@@ -227,9 +227,9 @@ LibreOffice_Dev
 			LIBRARYVERSION 9.3.0
 			PATCHCODEFILE ooodev_patchcodes.txt
 			CODEFILENAME codes_ooodev.txt
-			DOWNLOADBANNER	ooobanner.bmp
-			DOWNLOADBITMAP	ooobitmap.bmp
-			DOWNLOADSETUPICO	ooosetup.ico
+            DOWNLOADBANNER	ooobanner_go-oo.bmp
+            DOWNLOADBITMAP	ooobitmap_go-oo.bmp
+            DOWNLOADSETUPICO	ooosetup_go-oo.ico
 			LOCALUSERDIR $ORIGIN/..
 			RELATIVE_PATHES_IN_DDF 1
             STARTCENTER_ADDFEATURE_URL http://extensions.documentfoundation.org/
@@ -270,9 +270,9 @@ URE
             PCPFILENAME ure.pcp
             POOLPRODUCT 0
             GLOBALFILEGID gid_File_Dl_Cppu
-            DOWNLOADBANNER urebanner.bmp
-            DOWNLOADBITMAP urebitmap.bmp
-            DOWNLOADSETUPICO ooosetup.ico
+            DOWNLOADBANNER ooobanner_go-oo.bmp
+            DOWNLOADBITMAP ooobitmap_go-oo.bmp
+            DOWNLOADSETUPICO ooosetup_go-oo.ico
             DONTUSESTARTMENUFOLDER 1
             SERVICESPROJEKT 1
             RELATIVE_PATHES_IN_DDF 1
@@ -318,9 +318,9 @@ LibreOffice_SDK
             NOVERSIONINDIRNAME 0
             NOSPACEINDIRECTORYNAME 1
             CHANGETARGETDIR 1
-            DOWNLOADBANNER ooosdkbanner.bmp
-            DOWNLOADBITMAP ooosdkbitmap.bmp
-            DOWNLOADSETUPICO ooosetup.ico
+            DOWNLOADBANNER ooobanner_go-oo.bmp
+            DOWNLOADBITMAP ooobitmap_go-oo.bmp
+            DOWNLOADSETUPICO ooosetup_go-oo.ico
             DONTUSESTARTMENUFOLDER 1
             RELATIVE_PATHES_IN_DDF 1
             STARTCENTER_ADDFEATURE_URL http://extensions.documentfoundation.org/
