--- sw/source/ui/dochdl/dochdl.src.old	2009-10-08 16:17:19.000000000 +0200
+++ sw/source/ui/dochdl/dochdl.src	2010-03-03 16:59:21.000000000 +0100
@@ -86,3 +86,13 @@ String STR_SW_TEAM_NAMES
     Text = "Back row: Aidan Butler, Hans-Peter Burow, Caolan McNamara, Michael Brauer, Martin Maher, Gunnar Timm\nFront row: Thomas Lange, Oliver-Rainer Duesterhoeft, Henning Brinkmann, Andreas Martens, Oliver Specht, Frank Meies, Daniel Vogelheim";
 };
 
+Bitmap BMP_GO_OO_TEAM_MUGSHOT
+{
+	File ="go-oo-team.bmp";
+};
+
+String STR_GO_OO_TEAM_NAMES
+{
+    TEXT = "From top left: Andreas, Bernhard, David, Dhananjay, Eric, Federico, Fong\nFridrich, Giuseppe, Hubert, Jakub, Jan, Jan, Jiao\nJody, Kai, Kalman, Kohei, Martin, Matthias, Michael\nMike, Miklos, Noel, Petr, Radek, Rene, Silvan\nThorsten, Tor, Volker, Yin, ZhangYun.\n";
+};
+
--- sw/source/ui/dochdl/gloshdl.cxx.old	2009-10-08 16:17:19.000000000 +0200
+++ sw/source/ui/dochdl/gloshdl.cxx	2010-03-03 16:59:21.000000000 +0100
@@ -645,6 +645,41 @@ BOOL SwGlossaryHdl::Expand( const String
                 pWrtShell->EndAllAction();
                 pWrtShell->EndUndo( UNDO_INSGLOSSARY );
             }
+				else if ( aShortName.EqualsAscii ( "GoOOTeam", 0, 14 ) )
+				{
+					String sGraphicName ( RTL_CONSTASCII_USTRINGPARAM ( "Go-oo Team photo" ) );
+					String sTeamCredits ( RTL_CONSTASCII_USTRINGPARAM ( "go-oo team credits" ) );
+					pWrtShell->StartUndo ( UNDO_INSGLOSSARY );
+					pWrtShell->StartAllAction();
+					if(pWrtShell->HasSelection())
+						pWrtShell->DelLeft();
+					Bitmap aBitmap ( SW_RES ( BMP_GO_OO_TEAM_MUGSHOT ) );
+					pWrtShell->Insert ( aEmptyStr, aEmptyStr, aBitmap);
+					pWrtShell->SetFlyName ( sGraphicName );
+					SwTxtFmtColl* pColl = pWrtShell->GetTxtCollFromPool ( RES_POOLCOLL_LABEL_ABB );
+					const IDocumentFieldsAccess* pIDFA = pWrtShell->getIDocumentFieldsAccess();
+					SwFieldType* pType = pIDFA->GetFldType( RES_SETEXPFLD, pColl->GetName(), false );
+					sal_uInt16 nId = pIDFA->GetFldTypes()->GetPos( pType );
+					pWrtShell->InsertLabel( LTYPE_OBJECT, aEmptyStr, aEmptyStr, aEmptyStr, FALSE, nId, aEmptyStr );
+					pWrtShell->SwFEShell::SetFlyName( sTeamCredits );
+					pWrtShell->SwFEShell::SelectObj ( Point ( ULONG_MAX, ULONG_MAX ) );
+					pWrtShell->EnterStdMode();
+					pWrtShell->EndPara ( TRUE );
+					String aTmp ( SW_RES ( STR_GO_OO_TEAM_NAMES ) );
+					pWrtShell->Insert ( aTmp );
+					SvxAdjustItem aAdjustItem( SVX_ADJUST_CENTER, RES_PARATR_ADJUST );
+					pWrtShell->SetAttr( aAdjustItem );
+					pWrtShell->SttPara ();
+					pWrtShell->SplitNode();
+					pWrtShell->Left(CRSR_SKIP_CHARS, FALSE, 1, FALSE );
+					SvxWeightItem aWeightItem ( WEIGHT_BOLD, RES_CHRATR_WEIGHT );
+					pWrtShell->Insert ( String ( RTL_CONSTASCII_USTRINGPARAM ( "The go-oo.org team!" ) ) );
+					pWrtShell->SttPara ( TRUE );
+					pWrtShell->SetAttr( aWeightItem);
+					pWrtShell->GotoFly ( sTeamCredits);
+					pWrtShell->EndAllAction();
+					pWrtShell->EndUndo( UNDO_INSGLOSSARY );
+				}
             else
             {
                 String aTmp( SW_RES(STR_NOGLOS));
--- sw/inc/dochdl.hrc.old	2009-10-08 16:17:18.000000000 +0200
+++ sw/inc/dochdl.hrc	2010-03-03 16:59:21.000000000 +0100
@@ -49,7 +49,10 @@
 #define BMP_SW_TEAM_MUGSHOT			(RC_DOCHDL_BEGIN + 12)
 #define STR_SW_TEAM_NAMES			(RC_DOCHDL_BEGIN + 13)
 
-#define DOCHDL_ACT_END              STR_SW_TEAM_NAMES
+#define BMP_GO_OO_TEAM_MUGSHOT			(RC_DOCHDL_BEGIN + 14)
+#define STR_GO_OO_TEAM_NAMES			(RC_DOCHDL_BEGIN + 25)
+
+#define DOCHDL_ACT_END              STR_GO_OO_TEAM_NAMES
 
 #if DOCHDL_ACT_END > RC_DOCHDL_END
 #error Resource-Id Ueberlauf in #file, #line
