diff --git a/l10n/source/bg/localize.sdf b/l10n/source/bg/localize.sdf
--- l10n/source/bg/localize.sdf
+++ l10n/source/bg/localize.sdf
@@ -27751,7 +27751,6 @@
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id3154342	5			0	bg	Клетките в масива – резултат са защитени срещу промени. Можете обаче да редактирате или копирате формулата за масив, като изберете цялата област от клетки.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	hd_id8834803				0	bg	Масиви с константи, вмъкнати във формули				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id985747				0	bg	Calc поддържа вмъкване на матрични константи във формули. Вмъкнатите масиви се ограждат с фигурни скоби – „{“ и „}“. Елементите могат да са числа, включително отрицателни, логически константи (TRUE и FALSE) или низове. Не се разрешават изрази, които не са константи. Масивите могат да съдържат един или няколко реда и една или няколко колони. Всички редове трябва да са с еднакъв брой елементи помежду си, всички колони – също.				2002-02-02 02:02:02
-helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id936613				0	bg	Разделителят на колони (който отделя елементите в един ред) е точка и запетая – „;“. Разделителят на редове е знакът вертикална черта – „|“. Разделителите не зависят от езика и локала.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id1877498				0	bg	Не се допуска влагане на масиви.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id4262520				0	bg	\<emph\>Примери:\</emph\>				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id9387493				0	bg	={1;2;3}				2002-02-02 02:02:02
diff --git a/l10n/source/br/localize.sdf b/l10n/source/br/localize.sdf
--- l10n/source/br/localize.sdf
+++ l10n/source/br/localize.sdf
@@ -11764,7 +11764,6 @@
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id7781914				0	br	=SIN({1;2;3})				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id8207037				0	br	An array with one row consisting of the three numbers 1, 2, and 3.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id8868068				0	br	={1;2;3|4;5;6}				2002-02-02 02:02:02
-helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id936613				0	br	The column separator (separating elements in one row) is the ';' semicolon. The row separator is a '|' pipe symbol. The separators are not language and locale dependent. 				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id9387493				0	br	={1;2;3}				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id985747				0	br	Calc supports inline matrix/array constants in formulas. An inline array is surrounded by curly braces '{' and '}'. Elements can be each a number (including negatives), a logical constant (TRUE, FALSE), or a literal string. Non-constant expressions are not allowed. Arrays can be entered with one or more rows, and one or more columns. All rows must consist of the same number of elements, all columns must consist of the same number of elements. 				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_idN10B75				0	br	B (result)				2002-02-02 02:02:02
diff --git a/l10n/source/ca/localize.sdf b/l10n/source/ca/localize.sdf
--- l10n/source/ca/localize.sdf
+++ l10n/source/ca/localize.sdf
@@ -37022,7 +37022,6 @@
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id3154342	5			0	ca	Les cel·les d'una matriu de resultats es protegeixen automàticament dels canvis. No obstant això, podeu editar o copiar la fórmula de matrius en seleccionar la matriu sencera de l'interval de cel·les.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	hd_id8834803				0	ca	Ús de constants de matriu en línia en fórmules				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id985747				0	ca	El Calc admet constants de matriu en línia en les fórmules. Una matriu en línia es troba entre claudàtors '{' i '}'. Els elements poden ser nombres (inclosos els negatius), una constant lògica (CERT, FALS) o una cadena literal. No es permeten les expressions no constants. Es poden introduir les matrius en una fila o en més d'una, i en una columna o en més d'una. Totes les files han de constar del mateix nombre d'elements i totes les columnes han d'estar formades pel mateix nombre d'elements. 				2002-02-02 02:02:02
-helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id936613				0	ca	El separador de columna (que separa els elements d'una fila) és el punt i coma ';'. El separador de fila és un símbol de barra vertical '|'. Els separadors no depenen de la llengua ni de la configuració regional. 				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id1877498				0	ca	No es poden incrustar les matrius.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id4262520				0	ca	\<emph\>Exemples:\</emph\>				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id9387493				0	ca	={1;2;3}				2002-02-02 02:02:02
diff --git a/l10n/source/cs/localize.sdf b/l10n/source/cs/localize.sdf
--- l10n/source/cs/localize.sdf
+++ l10n/source/cs/localize.sdf
@@ -13150,7 +13150,6 @@
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id3154342	5			0	cs	Buňky v matici výsledků jsou automaticky chráněny před změnami. Maticový vzorec je ovšem možné upravit nebo zkopírovat tak, že vyberete celý rozsah buněk matice.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	hd_id8834803				0	cs	Použití vložených matic ve vzorcích				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id985747				0	cs	Calc podporuje ve vzorcích vložené matice. Vložená matice je uzavřena ve složených závorkách '{' a '}'. Prvky matice mohou být čísla (i záporná), logické konstanty (TRUE, FALSE) nebo řetězec. Nejsou dovoleny nekonstantové výrazy. Matici lze zadat s jedním či více sloupci a jedním či více řádky. Každý řádek musí obsahovat stejný počet prvků, každý sloupec musí obsahovat stejný počet prvků.				2002-02-02 02:02:02
-helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id936613				0	cs	Oddělovačem sloupců (rozdělení prvků jednoho řádku) je středník ';'. Oddělovačem řádků je symbol roury '|'. Oddělovače nezávisejí na jazykovém nastavení.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id1877498				0	cs	Matice nemohou být vnořené.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id4262520				0	cs	\<emph\>Příklady:\</emph\>				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id9387493				0	cs	={1;2;3}				2002-02-02 02:02:02
diff --git a/l10n/source/da/localize.sdf b/l10n/source/da/localize.sdf
--- l10n/source/da/localize.sdf
+++ l10n/source/da/localize.sdf
@@ -12226,7 +12226,6 @@
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id3154342	5			0	da	Cellerne i en resultattabel er automatisk beskyttet imod ændringer. Dog kan du redigere eller kopiere matrixformlen ved at vælge hele tabelcelleområdet.				20050602 23:18:50
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	hd_id8834803				0	da	Brug af indlejrede områdekonstanter i formler				20071229 13:42:26
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id985747				0	da	Calc understøtter indlejrede konstanter af typen matrix/område i formler. Et indlejret område er omgivet af krølleparenteser '{' og '}'. Elementerne kan hver især være et tal (inklusiv negative), en logisk konstant (SAND, FALSK), eller en bogstavelig streng. Ikke konstante udtryk er ikke tilladt. Områder kan indtastes med en eller flere rækker, og en eller flere kolonner. Alle rækker skal indeholde det samme antal elementer, alle kolonner skal indeholde det samme antal elementer.				20080101 09:03:22
-helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id936613				0	da	Kolonneadskilleren (opdeler elementerne i en række) er ';' semikolon. Rækkeadskilleren er '|' dataledningssymbolet. Skilletegnene er ikke sprog- eller lokalitetsafhængige. 				20080213 11:34:12
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id1877498				0	da	Matrixer kan ikke være indlejret.				20071218 05:49:31
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id4262520				0	da	\<emph\>Eksempler:\</emph\>				20071218 01:53:16
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id9387493				0	da	={1;2;3}				20071220 13:51:20
diff --git a/l10n/source/de/localize.sdf b/l10n/source/de/localize.sdf
--- l10n/source/de/localize.sdf
+++ l10n/source/de/localize.sdf
@@ -40381,7 +40381,6 @@
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id3154342	5			0	de	Die Zellen in Ergebnismatrizen werden automatisch vor Änderungen geschützt. Sie können aber die Matrixformel selbst bearbeiten oder kopieren, wenn Sie den gesamten Zellbereich der Matrix auswählen.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	hd_id8834803				0	de	Benutzung von inline Matrizen Konstanten in Fomeln				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id985747				0	de	Calc unterstützt inline Matrizen/Vektoren in Formeln. Eine inline Matrix wird in geschwungenen Klammern '{' und '}' eingeschlossen. Elemente können entweder eine Zahl (einschließlich negativer), eine logische Konstante (WAHR, FALSCH), oder eine Zeichenkette sein. Variable Ausdrucke sind nicht erlaubt . Matrizen können eine oder mehrere Zeilen und eine oder mehrere Spalten enthalten. Alle Zeilen müssen dieselbe Anzahl von Elementen enthalten, alle Spalten dieselbe Anzahl von Elementen. 				2002-02-02 02:02:02
-helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id936613				0	de	Der Spaltentrenner (trennt die Elemente in einer Zeile) ist das ';' Semikolon. Der Zeilentrenner ist ein '|' Pipe Symbol. Die Trenner sind sprachunabhängig. 				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id1877498				0	de	Matrizen dürfen nicht verschachtelt sein.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id4262520				0	de	\<emph\>Beispiele:\</emph\>				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id9387493				0	de	={1;2;3}				2002-02-02 02:02:02
diff --git a/l10n/source/el/localize.sdf b/l10n/source/el/localize.sdf
--- l10n/source/el/localize.sdf
+++ l10n/source/el/localize.sdf
@@ -29879,7 +29879,6 @@
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id3154342	5			0	el	Τα κελιά στον πίνακα αποτελεσμάτων προστατεύονται αυτόματα από αλλαγές. Πάντως, μπορείτε να επεξεργασθείτε ή να αντιγράψετε τον τύπο πίνακα επιλέγοντας ολόκληρη την περιοχή κελιών του πίνακα.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	hd_id8834803				0	el	Χρήση σταθερών πίνακα σε τύπους				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id985747				0	el	Το Calc υποστηρίζει σταθερές τιμές από πίνακες/μήτρες δεδομένων σε τύπους. Ένας συνδεδεμένος πίνακας περιβάλλεται από άγκιστρα '{' και '}'. Τα στοιχεία μπορούν να είναι το καθένα ένας αριθμός (που περιέχει και αρνητικές τιμές), μία λογική σταθερά (TRUE, FALSE), ή ένα κυριολεκτικό αλφαριθμητικό. Οι μη σταθερές εκφράσεις δεν επιτρέπονται. Οι πίνακες μπορούν να εισαχθούν με μία ή περισσότερες γραμμές, και μία ή περισσότερες στήλες. Όλες οι γραμμές πρέπει να αποτελούνται από τον ίδιο αριθμό στοιχείων, το ίδιο και όλες οι στήλες.				2002-02-02 02:02:02
-helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id936613				0	el	Το διαχωριστικό στηλών (διαχωρίζει τα στοιχεία μιας γραμμής) είναι το ελληνικό ερωτηματικό ';'. Τα διαχωριστικά γραμμών είναι η κατακόρυφη κάθετος '|'. Τα διαχωριστικά είναι ανεξάρτητα γλώσσας και τοποθεσίας.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id1877498				0	el	Οι πίνακες δεν μπορούν να εμφωλευθούν.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id4262520				0	el	\<emph\>Παραδείγματα:\</emph\>				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id9387493				0	el	={1;2;3}				2002-02-02 02:02:02
diff --git a/l10n/source/en-GB/localize.sdf b/l10n/source/en-GB/localize.sdf
--- l10n/source/en-GB/localize.sdf
+++ l10n/source/en-GB/localize.sdf
@@ -13234,7 +13234,6 @@
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id3154342	5			0	en-GB	The cells in a results array are automatically protected against changes. However, you can edit or copy the array formula by selecting the entire array cell range.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	hd_id8834803				0	en-GB	Using Inline Array Constants in Formulae				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id985747				0	en-GB	Calc supports inline matrix/array constants in formulae. An inline array is enclosed with braces '{' and '}'. Elements can be each a number (including negatives), a logical constant (TRUE, FALSE), or a literal string. Non-constant expressions are not allowed. Arrays can be entered with one or more rows, and one or more columns. All rows must consist of the same number of elements, all columns must consist of the same number of elements.				2002-02-02 02:02:02
-helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id936613				0	en-GB	The column separator (separating elements in one row) is the ';' semicolon. The row separator is a '|' pipe symbol. The separators are not language and locale dependent.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id1877498				0	en-GB	Arrays can not be nested.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id4262520				0	en-GB	\<emph\>Examples:\</emph\>				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id9387493				0	en-GB	={1;2;3}				2002-02-02 02:02:02
diff --git a/l10n/source/es/localize.sdf b/l10n/source/es/localize.sdf
--- l10n/source/es/localize.sdf
+++ l10n/source/es/localize.sdf
@@ -40323,7 +40323,6 @@
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id3154342	5			0	es	Las celdas de una matriz de resultados están automáticamente protegidas contra modificaciones. No obstante, puede editar o copiar la fórmula de matriz si selecciona toda el área de celdas de la matriz.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	hd_id8834803				0	es	Utilizando Constantes de Matrices En línea en Formulas				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id985747				0	es	Calc apoya constantes de matrices en línea en formulas.  Un matriz en línea es rodeado por llaves, '{' y '}'. Los elementos puede ser cada una un número (incluso negativos), una constante lógica (TRUE, FALSE), o una cadena literal. Expresiones no-constantes no están permitido. Los matrices puede ser entrada en una ó más filas, y una ó más columnas. Todas las filas tiene que tener igual número de elementos, todos columnas debe consista de la misma número de elementos.				2002-02-02 02:02:02
-helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id936613				0	es	El separador de columnas (separando elementos en una fila) es el ";" punto y coma. El separador de filas es la "|" barra. Los separadores no están dependiente de idioma ni local.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id1877498				0	es	Los matrices no pueden ser anidados.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id4262520				0	es	\<emph\>Ejemplo:\</emph\>				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id9387493				0	es	={1;2;3}				2002-02-02 02:02:02
diff --git a/l10n/source/et/localize.sdf b/l10n/source/et/localize.sdf
--- l10n/source/et/localize.sdf
+++ l10n/source/et/localize.sdf
@@ -16727,7 +16727,6 @@
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id3154342	5			0	et	Tulemuste massiivi lahtrid on automaatselt kaitstud muudatuste vastu. Massiivi valemit saab redigeerida ja kopeerida, kui valida kogu massiivi lahtrite vahemik.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	hd_id8834803				0	et	Sisestatud konstantsete massiivide kasutamine valemites.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id985747				0	et	Calc toetab valemites põimitud maatriksi- ja massiivikonstante. Põimitud massiiv ümbritsetakse looksulgudega '{' ja '}'. Elementideks võivad olla arvud (sealhulgas negatiivsed), tõeväärtused (TÕENE, VÄÄR) ja tekstistringid. Mittekonstantsed avaldised pole lubatud. Massiive võib sisestada ühe või mitme rea või veeruna. Kõik read peavad sisaldama sama arvu elemente, samuti ka kõik veerud.				2002-02-02 02:02:02
-helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id936613				0	et	Veergude eraldaja (ühes reas asuvate elementide eraldaja) on semikoolon ';'. Ridade eraldajaks on püstkriipsu märk '|'. Need eraldajad ei sõltu keelest ega lokaadist.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id1877498				0	et	Massiive ei saa põimida.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id4262520				0	et	\<emph\>Näited:\</emph\>				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id9387493				0	et	={1;2;3}				2002-02-02 02:02:02
diff --git a/l10n/source/fi/localize.sdf b/l10n/source/fi/localize.sdf
--- l10n/source/fi/localize.sdf
+++ l10n/source/fi/localize.sdf
@@ -67404,7 +67404,6 @@
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id3154342	5			0	fi	Matriisikaavan solut muuntuvat muutoksilta suojatuiksi. Kuitenkin matriisikaavan voi muokata tai kopioida, kun valitaan koko matriisisolualue.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	hd_id8834803				0	fi	Sisäisten taulukkovakioiden käyttö lausekkeissa				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id985747				0	fi	Calc tukee sisäisiä matriisi/taulukkovakioita kaavoissa. Sisäinen matriisi on rajattu aaltosulkeilla '{' ja '}'. Osatekijät voivat olla lukuja (myös negatiivisia) loogisia vakioita (TOSI, EPÄTOSI) tai merkkijonoja. Lausekkeet, jotka eivät ole vakioita, eivät ole sallittuja. Taulukot voidaan syöttää yksi- tai monirivisinä tai -sarakkeisina. Kaikkien rivien pitää olla yhtä monesta osatekijästä koostuvia ja kaikissa sarakkeissa pitää olla sama lukumäärä vakioelementtejä.				2002-02-02 02:02:02
-helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id936613				0	fi	Sarake-erottimena (yhden rivin arvojen erottelemiseksi) käytetään puolipistettä ';'. Rivierottimena on pystyviiva '|'. Erottimet eivät ole kieli- tai aluekohtaisia.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id1877498				0	fi	Taulukot eivät voi olla sisäkkäin.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id4262520				0	fi	\<emph\>Esimerkkejä\</emph\>				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id9387493				0	fi	={1;2;3}				2002-02-02 02:02:02
diff --git a/l10n/source/fr/localize.sdf b/l10n/source/fr/localize.sdf
--- l10n/source/fr/localize.sdf
+++ l10n/source/fr/localize.sdf
@@ -38498,7 +38498,6 @@
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id3154342	5			0	fr	Les cellules composant une matrice de résultat sont automatiquement protégées contre les modifications. Vous pouvez cependant éditer ou copier la formule en sélectionnant la totalité de la plage de cellules de la matrice.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	hd_id8834803				0	fr	Utilisation des constantes de matrice en ligne dans les formules				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id985747				0	fr	Calc supporte les constantes de matrice en ligne dans les formules. Une matrice en ligne est entourée par des accolades '{' et '}'. Les éléments peuvent être soit un nombre (y compris les nombres négatifs), une constante logique (VRAIE, FAUX) ou une chaîne de texte. Les expressions non constantes ne sont pas autorisées. La matrice peut être saisie avec une ou plusieurs lignes et une ou plusieurs colonnes. Toutes les lignes doivent avoir le même nombre d'éléments, toutes les colonnes doivent avoir le même nombre d'éléments. 				2002-02-02 02:02:02
-helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id936613				0	fr	Le séparateur de colonne (séparant les éléments dans une ligne) est le ';' point-virgule. Le séparateur de ligne est '|' la barre verticale. Les séparateurs ne sont pas dépendants de la langue ou de l'environnement localisé. 				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id1877498				0	fr	Les matrices ne peuvent être imbriquées.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id4262520				0	fr	\<emph\>Exemples:\</emph\>				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id9387493				0	fr	={1;2;3}				2002-02-02 02:02:02
@@ -46846,7 +46845,6 @@
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id3154342	5			0	fr	Les cellules composant une matrice de résultat sont automatiquement protégées contre les modifications. Vous pouvez cependant éditer ou copier la formule en sélectionnant la totalité de la plage de cellules de la matrice.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	hd_id8834803				0	fr	Utilisation des constantes de matrice en ligne dans les formules				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id985747				0	fr	Calc supporte les constantes de matrice en ligne dans les formules. Une matrice en ligne est entourée par des accolades '{' et '}'. Les éléments peuvent être soit un nombre (y compris les nombres négatifs), une constante logique (VRAIE, FAUX) ou une chaîne de texte. Les expressions non constantes ne sont pas autorisées. La matrice peut être saisie avec une ou plusieurs lignes et une ou plusieurs colonnes. Toutes les lignes doivent avoir le même nombre d'éléments, toutes les colonnes doivent avoir le même nombre d'éléments. 				2002-02-02 02:02:02
-helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id936613				0	fr	Le séparateur de colonne (séparant les éléments dans une ligne) est le ';' point-virgule. Le séparateur de ligne est '|' la barre verticale. Les séparateurs ne sont pas dépendants de la langue ou de l'environnement localisé. 				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id1877498				0	fr	Les matrices ne peuvent être imbriquées.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id4262520				0	fr	\<emph\>Exemples:\</emph\>				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id9387493				0	fr	={1;2;3}				2002-02-02 02:02:02
diff --git a/l10n/source/ga/localize.sdf b/l10n/source/ga/localize.sdf
--- l10n/source/ga/localize.sdf
+++ l10n/source/ga/localize.sdf
@@ -20703,7 +20703,6 @@
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id3154342	5			0	ga	The cells in a results array are automatically protected against changes. However, you can edit or copy the array formula by selecting the entire array cell range.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	hd_id8834803				0	ga	Using Inline Array Constants in Formulas				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id985747				0	ga	Calc supports inline matrix/array constants in formulas. An inline array is surrounded by curly braces '{' and '}'. Elements can be each a number (including negatives), a logical constant (TRUE, FALSE), or a literal string. Non-constant expressions are not allowed. Arrays can be entered with one or more rows, and one or more columns. All rows must consist of the same number of elements, all columns must consist of the same number of elements. 				2002-02-02 02:02:02
-helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id936613				0	ga	The column separator (separating elements in one row) is the ';' semicolon. The row separator is a '|' pipe symbol. The separators are not language and locale dependent. 				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id1877498				0	ga	Arrays can not be nested.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id4262520				0	ga	\<emph\>Examples:\</emph\>				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id9387493				0	ga	={1;2;3}				2002-02-02 02:02:02
diff --git a/l10n/source/hu/localize.sdf b/l10n/source/hu/localize.sdf
--- l10n/source/hu/localize.sdf
+++ l10n/source/hu/localize.sdf
@@ -37590,7 +37590,6 @@
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id3154342	5			0	hu	Az eredménytömb cellái rendszerint módosítás elleni védelmet élveznek. Ennek ellenére a tömbképletet szerkesztheti, illetve másolhatja a tömb teljes cellatartományának kijelölésével.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	hd_id8834803				0	hu	Helyi tömbkonstansok használata képletekben				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id985747				0	hu	A Calc támogatja a helyi mátrixokat/tömbkonstansokat a képletekben. Egy helyi tömb „{” és „}” jel közé van zárva. Az elemek mindegyike lehet szám (beleértve a negatív számokat is), logikai konstans (IGAZ, HAMIS) vagy karakterlánc. A nem konstans kifejezések nem használhatók. Megadhatók egy vagy több sorból, illetve egy vagy több oszlopból álló tömbök. Minden sornak azonos számú elemből kell állnia, valamint minden oszlopnak is azonos számú elemből kell állnia. 				2002-02-02 02:02:02
-helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id936613				0	hu	Az oszlopelválasztó (amely elválasztja a soron belül az elemeket) a pontosvessző (;). A sorelválasztó a függőleges vonal (|). Az elválasztók nem függnek a nyelvtől és a területi beállításoktól. 				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id1877498				0	hu	A tömbök nem ágyazhatók egymásba.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id4262520				0	hu	\<emph\>Példák:\</emph\>				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id9387493				0	hu	={1;2;3}				2002-02-02 02:02:02
diff --git a/l10n/source/it/localize.sdf b/l10n/source/it/localize.sdf
--- l10n/source/it/localize.sdf
+++ l10n/source/it/localize.sdf
@@ -37576,7 +37576,6 @@
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id3154342	5			0	it	Le celle della matrice dei risultati sono automaticamente protette dalle modifiche. Potete tuttavia modificare o copiare la formula della matrice selezionando l'intera area di celle della matrice.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	hd_id8834803				0	it	Utilizzo di Vettori di costanti inline nelle formule				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id985747				0	it	Calc supporta matrici/vettori di costanti inline. Un vettore inline è circondato da parentesi graffe '{' e '}'. Gli elementi posso essere un numero (incluso numeri negativi), una costante logica (VERO, FALSO), o una stringa. Non sono consentite espressioni non costanti. I vettori possono essere inseriti con una o più righe, e una o più colonne. Tutte le righe devono avere lo stesso numero di elementi e così le colonne.				2002-02-02 02:02:02
-helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id936613				0	it	Il separatore di colonne (che separa gli elementi appartententi ad una stessa riga) è il punto e virgola ';'. Il separatore di riga è il simbolo '|'. I separatori non dipendono dal linguaggio o dalla localizzazione.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id1877498				0	it	I vettori non possono essere annidati.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id4262520				0	it	\<emph\>Esempi:\</emph\>				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id9387493				0	it	={1;2;3}				2002-02-02 02:02:02
diff --git a/l10n/source/ja/localize.sdf b/l10n/source/ja/localize.sdf
--- l10n/source/ja/localize.sdf
+++ l10n/source/ja/localize.sdf
@@ -37658,7 +37658,6 @@
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id3154342	5			0	ja	結果行列のセルは、変更されないよう自動的に保護されます。ただし、行列のセル範囲全体を選択すると、行列式を編集またはコピーできます。				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	hd_id8834803				0	ja	数式でインライン行列定数を使用する				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id985747				0	ja	Calc は数式でのインライン行列定数をサポートしています。 インライン行列は、中括弧「{」および「}」で囲まれています。 要素は、数字 (負の数を含む)、論理定数 (TRUE、FALSE)、文字列のいずれかにできます。 定数以外の式は許可されません。 行列は 1 行以上の行、1 列以上の列で入力できます。 すべての行は同一数の要素で構成される必要があり、すべての列は同一数の要素で構成される必要があります。 				2002-02-02 02:02:02
-helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id936613				0	ja	列区切り記号 (1 列の要素を区切るもの) は セミコロン (;) です。 行区切り記号はパイプ記号 (|) です。 区切り記号は言語ではなく、ロケールにも依存しません。 				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id1877498				0	ja	行列を入れ子にすることはできません。				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id4262520				0	ja	\<emph\>例:\</emph\>				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id9387493				0	ja	={1;2;3}				2002-02-02 02:02:02
diff --git a/l10n/source/kid/localize.sdf b/l10n/source/kid/localize.sdf
--- l10n/source/kid/localize.sdf
+++ l10n/source/kid/localize.sdf
@@ -11761,7 +11761,6 @@
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id3154342	5			0	kid	y-8-0r‖The cells in a results array are automatically protected against changes. However, you can edit or copy the array formula by selecting the entire array cell range.				20100401 13:55:47
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	hd_id8834803				0	kid	y-ya7a‖Using Inline Array Constants in Formulas				20100401 13:55:47
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id985747				0	kid	+xv242‖Calc supports inline matrix/array constants in formulas. An inline array is surrounded by curly braces '{' and '}'. Elements can be each a number (including negatives), a logical constant (TRUE, FALSE), or a literal string. Non-constant expressions are not allowed. Arrays can be entered with one or more rows, and one or more columns. All rows must consist of the same number of elements, all columns must consist of the same number of elements. 				20100401 13:55:47
-helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id936613				0	kid	4epc7f‖The column separator (separating elements in one row) is the ';' semicolon. The row separator is a '|' pipe symbol. The separators are not language and locale dependent. 				20100401 13:55:47
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id1877498				0	kid	p=jt8g‖Arrays can not be nested.				20100401 13:55:47
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id4262520				0	kid	5vyume‖\<emph\>Examples:\</emph\>				20100401 13:55:47
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id9387493				0	kid	6kr>zs‖={1;2;3}				20100401 13:55:47
diff --git a/l10n/source/km/localize.sdf b/l10n/source/km/localize.sdf
--- l10n/source/km/localize.sdf
+++ l10n/source/km/localize.sdf
@@ -19005,7 +19005,6 @@
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id3154342	5			0	km	ក្រឡា​ក្នុង​លទ្ធផល​អារេ ត្រូវ​បាន​ការពារ​ដោយ​ស្វ័យ​ប្រវត្តិ​ទល់​នឹង​ភាព​ផ្លាស់ប្តូរ ។ ទោះ​យ៉ាង​ណា អ្នក​អាច​កែសម្រួល ឬ​ចម្លង​រូបមន្ត​អារេ​ដោយ​ជ្រើស​ជួរ​ក្រឡា​អារេ​ទាំងមូល ។				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	hd_id8834803				0	km	ប្រើ​ចំនួន​អារេ​ថេរ​ក្នុង​តួរ ក្នុង​រូបមន្ត				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id985747				0	km	Calc គាំទ្រ​ចំនួន​ម៉ាទ្រីស/​អារេ​ថេរ​ក្នុង​តួ ក្នុង​រូបមន្ត ។ អារេ​ក្នុង​តួ គឺត្រូវបាន​ដាក់​នៅ​ក្នុង​ដង្កៀប '{' និង '}' ។ ធាតុ អាច​ជា​លេខ​នីមួយៗ (រួមទាំង អវិជ្ជមាន) ចំនួន​ថេរ​តក្កវិជ្ជា (TRUE, FALSE) ឬ​ខ្សែអក្សរ​តម្លៃ​ត្រង់ ។ កន្សោម​មិនថេរ គឺ​មិនអនុញ្ញាត​ទេ ។ អារេ អាច​ត្រូវបាន​បញ្ចូល​ជាមួយ​ជួរដេក​មួយ ឬ​ច្រើន និង​ជួរឈរ​មួយ ឬ​ច្រើន ។ ជួរ​ដេក​ទាំងអស់​ត្រូវតែមាន​ចំនួន​ធាតុ​ដូចគ្នា ហើយ​ជួរ​ឈរទាំងអស់​ត្រូវតែ​មាន​ចំនួន​ធាតុ​ដូចគ្នា ។				2002-02-02 02:02:02
-helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id936613				0	km	អ្នកបំបែក​ជួរ​ឈរ (បំបែក​ធាតុ​ជា​មួយ​ជួរដេក) គឺជា​សញ្ញា​ចំណុច​ក្បៀស ';' ។ អ្នក​បំបែក​ជួរ​ដេក​គឺ​ជា​និមិត្តសញ្ញា​​បំពង់ '|' ។ អ្នក​បំបែក គឺ​មិនមែន​ជា​ភាសា និង​ការ​ដាក់​មូលដ្ឋានទេ ។				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id1877498				0	km	មិនអាច​ផ្ដុំ​អារេ​បានទេ ។				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id4262520				0	km	\<emph\>ឧទាហរណ៍ ៖\</emph\>				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id9387493				0	km	={1;2;3}				2002-02-02 02:02:02
diff --git a/l10n/source/ko/localize.sdf b/l10n/source/ko/localize.sdf
--- l10n/source/ko/localize.sdf
+++ l10n/source/ko/localize.sdf
@@ -37858,7 +37858,6 @@
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id3154342	5			0	ko	결과 배열의 셀은 변경할 수 없도록 자동으로 보호됩니다. 그러나 전체 배열 셀 범위를 선택하여 배열 수식을 편집 또는 복사할 수 있습니다.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	hd_id8834803				0	ko	수식에서 인라인 배열 상수 사용				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id985747				0	ko	Calc는 수식 안에서의 인라인 행렬/배열 상수를 지원합니다. 인라인 배열은 중괄호 { }로 묶습니다. 요소는 숫자 (음수 포함), 논리 상수 (TRUE, FALSE), 또는 문자열이 될 수 있습니다. 상수가 아닌 표현식은 허용되지 않습니다. 배열은 하나 이상의 행과 열로 입력될 수 있습니다. 모든 행과 열은 반드시 같은 수의 요소로 구성되어야 합니다. 				2002-02-02 02:02:02
-helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id936613				0	ko	(한 행에서 요소를 분리하는) 열 구분자는 ';' 세미콜론입니다. 행 구분자는 '|' 파이프 기호입니다. 구분자는 언어 및 로케일에 독립적입니다. 				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id1877498				0	ko	배열은 중첩될 수 없습니다.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id4262520				0	ko	\<emph\>예:\</emph\>				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id9387493				0	ko	={1;2;3}				2002-02-02 02:02:02
diff --git a/l10n/source/nb/localize.sdf b/l10n/source/nb/localize.sdf
--- l10n/source/nb/localize.sdf
+++ l10n/source/nb/localize.sdf
@@ -36110,7 +36110,6 @@
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id3154342	5			0	nb	Cellene i en resultattabell er automatisk beskyttet mot endringer. Men du kan redigere eller kopiere matriseformelen ved å velge hele tabellcelleområdet.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	hd_id8834803				0	nb	Bruk av innebygde områdekonstanter i formler				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id985747				0	nb	Calc støtter innebygde konstanter av typen matrise/område i formler. Et nøstet område er omgitt av krøllparenteser '{' og '}'. Elementene kan hver for seg være et tall (også negative), en logisk konstant (SANN, USANN), eller en bokstavstreng. Ikke-konstante uttrykk er ikke tillatt. Områder kan skrives inn med én eller flere rader, og en eller flere kolonner. Alle rader må inneholde det samme antallet elementer, alle kolonner må inneholde det samme antallet elementer.				2002-02-02 02:02:02
-helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id936613				0	nb	Kolonneskillet (deler opp elementene i en rad) er ';' semikolon. Radskillet er et '|'-rørsymbol. Skilletegnene er ikke språk- eller regionavhengige. 				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id1877498				0	nb	Matriser kan ikke være nøstet.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id4262520				0	nb	\<emph\>Eksempler:\</emph\>				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id9387493				0	nb	={1;2;3}				2002-02-02 02:02:02
diff --git a/l10n/source/nl/localize.sdf b/l10n/source/nl/localize.sdf
--- l10n/source/nl/localize.sdf
+++ l10n/source/nl/localize.sdf
@@ -44094,7 +44094,6 @@
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id3154342	5			0	nl	De cellen in een resultatenmatrix worden automatisch tegen wijzigingen beschermd. U kunt de matrixformule echter bewerken of kopiëren door het hele celbereik van de matrix te selecteren.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	hd_id8834803				0	nl	Het gebruik van in-regelige matrixconstanten in formules				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id985747				0	nl	Calc ondersteunt in-regelige matrix-/arrayconstanten in formules. Een in-regelige matrix wordt omsloten door accolades '{' en '}'. Elementen kunnen zowel een getal (inclusief negatieve), een logische constante (WAAR, ONWAAR) of een letterlijke tekenreeks zijn. Niet-constante expressies zijn niet geldig. Matrices kunnen worden ingevoerd met één of meer rijen en één of meer kolommen. Alle rijen moeten bestaan uit hetzelfde aantal elementen en alle kolommen moeten bestaan uit hetzelfde aantal elementen. 				2002-02-02 02:02:02
-helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id936613				0	nl	Het kolomscheidingsteken (dat de elementen in één rij scheidt) is de puntkomma ';'. Het rijscheidingsteken is het 'pipe'-teken '|'. De scheidingstekens zijn niet afhankelijk van taal en locale. 				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id1877498				0	nl	Matrices kunnen niet worden genest.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id4262520				0	nl	\<emph\>Voorbeelden:\</emph\>				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id9387493				0	nl	={1;2;3}				2002-02-02 02:02:02
diff --git a/l10n/source/om/localize.sdf b/l10n/source/om/localize.sdf
--- l10n/source/om/localize.sdf
+++ l10n/source/om/localize.sdf
@@ -37693,7 +37693,6 @@
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id3154342	5			0	om	Man'eewwan waraantoo firiwwanii ofumaan jijjiirraa irra dhorkamu.Haata'u malee, formulaa waraantoo man'ee hangii waraantoo mara filuudhan gulaaluu fi garagalchuu ni dandeessa. 				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	hd_id8834803				0	om	Waraantoo Sarkee Dhaabbataa Formuulaa keessati Fayyadamun				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id985747				0	om	Sarkee maatiksii/waraantoo dhabataa formulaa keessaa shallaguuf gargaara. sarkeen waraantoo curlii cuuftuudhan '{' '}' deeggarama. Miseensonni tokkoon tokkoon lakkoofsa (lakkoofsa negatiivii) ta'uu danda'a, yaayaa dhabbataa (TRUE, FLASE), ykn jecha jargifaa. Himannoon dhabbataa hin tannee hin hayyamamu. waraantoon tarree ykn tarja tokkoo ykn isaa oliin galfamuu danda'a. tarreewwan hundinu lakkoofsa miseensotaa wal fakkaata qabaachuu qabu, tarjaawwan hundinuu lakkoofsa miseensotaa walfakkataa qabaachuu qabu.    				2002-02-02 02:02:02
-helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id936613				0	om	Tarja adda baasaa(tarree tokkoo keessati addabaasuu)n buufata xiqqaaa ';'. tarree adda basaa '|' mallattoo ujumoo. Adda bastoonnii afaanii fi gitoo biyyaa miti.    				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id1877498				0	om	Waraantonni dilgataa'uu hin danda'an				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id4262520				0	om	\<emph\>Fakkeenyota:\</emph\>				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id9387493				0	om	={1;2;3}				2002-02-02 02:02:02
diff --git a/l10n/source/pl/localize.sdf b/l10n/source/pl/localize.sdf
--- l10n/source/pl/localize.sdf
+++ l10n/source/pl/localize.sdf
@@ -38498,7 +38498,6 @@
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id3154342	5			0	pl	Komórki w macierzy wynikowej są automatycznie chronione przed zmianami, Możliwa jest jednak edycja i kopiowanie formuły macierzowej przez zaznaczenie całego zakresu komórek.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	hd_id8834803				0	pl	Korzystanie z wewnętrznych stałych tablicowych w formułach				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id985747				0	pl	Program Calc obsługuje wewnętrzne stałe macierzowe/tablicowe w formułach. Wewnętrzna tablica znajduje się w nawiasach klamrowych „{” i „}”. Elementy mogą być liczbami (w tym liczbami ujemnymi), stałymi logicznymi (PRAWDA, FAŁSZ) lub ciągami literałów. Wyrażenia nie będące stałymi nie są dozwolone. Tablice mogą być wprowadzane w jednym lub kilku wierszach oraz w jednej lub kilku kolumnach. Wszystkie wiersze muszą składać się z tej samej liczby elementów, podobnie wszystkie kolumny muszą składać się z tej samej liczby elementów. 				2002-02-02 02:02:02
-helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id936613				0	pl	Separator kolumn (oddzielający elementy w jednym wierszu) to średnik (;). Separator wierszy to symbol potoku „|”. Separatory są niezależne od języka i ustawień regionalnych. 				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id1877498				0	pl	Tablice nie mogą być zagnieżdżone.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id4262520				0	pl	\<emph\>Przykłady:\</emph\>				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id9387493				0	pl	={1;2;3}				2002-02-02 02:02:02
diff --git a/l10n/source/pt-BR/localize.sdf b/l10n/source/pt-BR/localize.sdf
--- l10n/source/pt-BR/localize.sdf
+++ l10n/source/pt-BR/localize.sdf
@@ -38498,7 +38498,6 @@
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id3154342	5			0	pt-BR	As células de uma matriz de resultados estão automaticamente protegidas contra alterações. Entretanto, você pode editar ou copiar a fórmula de matriz selecionando todo o intervalo de células da matriz.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	hd_id8834803				0	pt-BR	Inserir matrizes de constantes em fórmulas				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id985747				0	pt-BR	O Calc permite inserir matrizes/vetores de constantes em fórmulas. Uma matriz inserida na fórmula é envolta por chaves '{' e '}'. Os elementos da matriz podem ser números (incluindo negativos), constantes lógicas (VERDADEIRO, FALSO), ou texto literal. Expressões não constantes não são permitidas. As matrizes podem ser inseridas com uma ou mais linhas e uma ou mais colunas. Todas as linhas devem conter o mesmo número de elementos, todas as colunas devem conter o mesmo número de elementos. 				2002-02-02 02:02:02
-helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id936613				0	pt-BR	O separador de coluna (separando elementos em uma linha) é o ponto-e-vírgula ';'. O separador de linhas é o símbolo '|'. Os separadores não dependem do idioma ou da locale. 				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id1877498				0	pt-BR	Matrizes não podem ser aninhadas.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id4262520				0	pt-BR	\<emph\>Exemplos:\</emph\>				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id9387493				0	pt-BR	={1;2;3} 				2002-02-02 02:02:02
diff --git a/l10n/source/pt/localize.sdf b/l10n/source/pt/localize.sdf
--- l10n/source/pt/localize.sdf
+++ l10n/source/pt/localize.sdf
@@ -38541,7 +38541,6 @@
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id3154342	5			0	pt	As células numa matriz de resultados ficam automaticamente protegidas contra modificações. No entanto, é possível editar ou copiar uma fórmula matriz seleccionando completamente o intervalo de células que constitui a matriz.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	hd_id8834803				0	pt	Utilizar constantes de matrizes inline em fórmulas				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id985747				0	pt	O Calc suporta matrizes inline em fórmulas. Uma matriz inline está delimitada por chavetas '{' e '}'. Os elementos podem ser um número (incluindo números negativos), uma constante lógica (TRUE, FALSE) ou uma cadeia de caracteres. Não são permitidas expressões não-constantes. As matrizes podem ser introduzidas com uma ou mais linhas e com uma ou mais colunas. Todas as linhas e colunas terão de ser compostas pelo mesmo número de elementos. 				2002-02-02 02:02:02
-helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id936613				0	pt	O separador de colunas (que separa elementos numa linha) é o ponto e vírgula ';'. O separador de linhas é o símbolo '|'. Os separadores não estão dependentes de idioma ou configuração regional. 				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id1877498				0	pt	Não é possível imbricar matrizes.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id4262520				0	pt	\<emph\>Exemplos:\</emph\>				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id9387493				0	pt	={1;2;3}				2002-02-02 02:02:02
diff --git a/l10n/source/ru/localize.sdf b/l10n/source/ru/localize.sdf
--- l10n/source/ru/localize.sdf
+++ l10n/source/ru/localize.sdf
@@ -37958,7 +37958,6 @@
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id3154342	5			0	ru	Ячейки в массиве результатов автоматически становятся недоступными для изменений. Однако формулу массива можно редактировать или копировать путем выделения всего массива.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	hd_id8834803				0	ru	Использование строчных констант-массивов в формулах				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id985747				0	ru	Calc поддерживает в формулах строчные матрицы/массивы констант. Строчный массив заключается в фигурные скобки ''{" и "}". Элементы могут быть как числовыми (в т.ч. отрицательными), так и логическими (TRUE, FALSE) или строковыми литералами. Выражения, не являющиеся константами, не допускаются. Возможен ввод массивов с одной или несколькими строками, а также с одним или несколькими столбцами. Все строки должны состоять из одинакового числа элементов; все столбцы также должны состоять из одинакового числа элементов. 				2002-02-02 02:02:02
-helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id936613				0	ru	Разделитель столбцов (разделяющий элементы в одной строке) - ";" (точка с запятой). Разделитель строк - "|" (символ конвейеризации). Разделители не зависят от языка и локали. 				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id1877498				0	ru	Вложенность массивов не допускается.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id4262520				0	ru	\<emph\>Примеры:\</emph\>				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id9387493				0	ru	={1;2;3}				2002-02-02 02:02:02
diff --git a/l10n/source/sl/localize.sdf b/l10n/source/sl/localize.sdf
--- l10n/source/sl/localize.sdf
+++ l10n/source/sl/localize.sdf
@@ -14329,7 +14329,6 @@
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id3154342	5			0	sl	Celice v rezultatu matrike so samodejno zavarovane pred spremembami. Lahko pa urejate ali kopirate matrično formulo s tem, da izberete ves obseg celic matrike, torej celo matriko.				20100323 18:28:05
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	hd_id8834803				0	sl	Uporaba matričnih konstant v formulah				20100323 18:28:05
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id985747				0	sl	Calc podpira matrične konstante v formulah. Vdelana matrika je zaobjeta z zavitim oklepajem in zaklepajem ('{' in '}'). Element je lahko poljubna številka (tudi negativna), logična konstanta (TRUE, FALSE) ali niz besedila. Ne-konstantni izrazi niso dovoljeni. Vnesena matrika ima lahko eno ali več vrstic ter enega ali več stolpcev. Vse vrstice morajo imeti enako število elementov, pa tudi vsi stolpci morajo imeti enako število elementov. 				20100323 18:28:05
-helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id936613				0	sl	Ločilo stolpcev (ki ločuje elemente ene vrstice) je podpičje (';'). Ločilo vrstic je simbol '|'. Ločila niso odvisna od jezikovnih in krajevnih nastavitev. 				20100323 18:28:05
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id1877498				0	sl	Matrik ni mogoče gnezditi.				20100323 18:28:05
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id4262520				0	sl	\<emph\>Primeri:\</emph\>				20100323 18:28:05
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id9387493				0	sl	={1;2;3}				20100323 18:28:05
diff --git a/l10n/source/sv/localize.sdf b/l10n/source/sv/localize.sdf
--- l10n/source/sv/localize.sdf
+++ l10n/source/sv/localize.sdf
@@ -38158,7 +38158,6 @@
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id3154342	5			0	sv	Cellerna i resultatmatrisen är automatiskt skyddade mot ändringar. Du kan däremot redigera eller kopiera matrisformeln genom att markera hela matrisområdet.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	hd_id8834803				0	sv	Använda infogade matriskonstanter i formler				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id985747				0	sv	Calc har stöd för infogade matriskonstanter i formler. En infogad matris är omgiven av klammerparenteserna '{' och '}'. Elementen kan antingen vara tal (även negativa tal), en logisk konstant (SANT och FALSKT) eller en literal sträng. Icke-konstanta uttryck tillåts ej. Matriserna kan ha en eller flera rader och en eller flera kolumner. Alla rader måste bestå av samma antal element och alla kolumnerna måste bestå av samma antal element. 				2002-02-02 02:02:02
-helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id936613				0	sv	Semikolon ';' används som kolumnavgränsare (avgränsar elementen på en rad). Som radavgränsare används symbolen '|'. Avgränsarna är inte språk- eller språkvariantberoende. 				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id1877498				0	sv	Du kan inte kapsla matriser.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id4262520				0	sv	\<emph\>Exempel:\</emph\>				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id9387493				0	sv	={1;2;3}				2002-02-02 02:02:02
diff --git a/l10n/source/tr/localize.sdf b/l10n/source/tr/localize.sdf
--- l10n/source/tr/localize.sdf
+++ l10n/source/tr/localize.sdf
@@ -35030,7 +35030,6 @@
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id3154342	5			0	tr	Sonuç dizisinde bulunan hücreler değişikliklere karşı otomatik korunurlar. Buna rağmen bir dizi formülünü, tüm hücre aralığının seçiminden sonra düzenleyebilir veya kopyalayabilirsiniz.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	hd_id8834803				0	tr	Formüllerde Satır İçi Dizi Sabitlerinin Kullanılması				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id985747				0	tr	Calc formüllerde satır içi matris/dizi sabitlerini destekler. Bir satır içi dizisi süslü ayraçlar ile '{' ve '}' ile çevrelenir. Elemanların her biri sayı (negatif sayılarda bulunur) veya mantıksal sabitler (DOĞRU,YANLIŞ) veya bir harf dizisi olabilir. Sabit olmayan deyimlere izin verilmez. Diziler bir veya daha fazla satırlı ve bir veya daha fazla sütunlu girilebilir. Bütün satırlarda aynı sayıda eleman olmalıdır, bütün sütunlar aynı miktarda elemandan oluşmalıdır.				2002-02-02 02:02:02
-helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id936613				0	tr	Sütun ayracı ( bir satırdaki elemanları ayıran) ';' noktalı virgül karakteridir. Satır ayracı ise '|' boru sembolüdür. Ayraçlar yerel veya dil bağımlı değildir.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id1877498				0	tr	Diziler iç içe geçirilemezler.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id4262520				0	tr	\<emph\>Örnekler:\</emph\>				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id9387493				0	tr	={1;2;3}				2002-02-02 02:02:02
diff --git a/l10n/source/uz/localize.sdf b/l10n/source/uz/localize.sdf
--- l10n/source/uz/localize.sdf
+++ l10n/source/uz/localize.sdf
@@ -31185,7 +31185,6 @@
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id3154342	5			0	uz	The cells in a results array are automatically protected against changes. However, you can edit or copy the array formula by selecting the entire array cell range.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	hd_id8834803				0	uz	Using Inline Array Constants in Formulas				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id985747				0	uz	Calc supports inline matrix/array constants in formulas. An inline array is surrounded by curly braces '{' and '}'. Elements can be each a number (including negatives), a logical constant (TRUE, FALSE), or a literal string. Non-constant expressions are not allowed. Arrays can be entered with one or more rows, and one or more columns. All rows must consist of the same number of elements, all columns must consist of the same number of elements. 				2002-02-02 02:02:02
-helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id936613				0	uz	The column separator (separating elements in one row) is the ';' semicolon. The row separator is a '|' pipe symbol. The separators are not language and locale dependent. 				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id1877498				0	uz	Arrays can not be nested.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id4262520				0	uz	\<emph\>Examples:\</emph\>				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id9387493				0	uz	={1;2;3}				2002-02-02 02:02:02
diff --git a/l10n/source/vi/localize.sdf b/l10n/source/vi/localize.sdf
--- l10n/source/vi/localize.sdf
+++ l10n/source/vi/localize.sdf
@@ -34583,7 +34583,6 @@
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id3154342	5			0	vi	Những ô trong một phạm vi kết quả được tự động bảo vệ khỏi thay đổi. Tuy nhiên, bạn vẫn còn có thể chỉnh sửa công thức mảng bằng cách lựa chọn toàn bộ phạm vi ô mảng.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	hd_id8834803				0	vi	Sử dụng Hằng số Mảng Trực tiếp trong Công thức				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id985747				0	vi	Trình Calc hỗ trợ hằng số kiểu ma trận/mảng trực tiếp trong công thức. Một mảng trực tiếp {được bao quanh bởi dấu ngoặc móc}. Mỗi phần tử có thể là một con số (gồm có số âm), một hằng số lôgic (TRUE (đúng), FALSE (sai)), hoặc một chuỗi nghĩa chữ. Không cho phép dùng biểu thức không phải hằng số. Có thể nhập mảng có một hay nhiều hàng/cột. Mọi hàng đều phải có cùng một số phần tử ; mọi cột đều phải chứa cùng một số phần tử. 				2002-02-02 02:02:02
-helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id936613				0	vi	Dấu tách cột (phân cách các phần tử theo một hàng) là dấu chấm phẩy « ; ». Dấu tách hàng là ký hiệu ống dẫn « | ». Hai dấu tách không phụ thuộc vào ngôn ngữ hay miền địa phương. 				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id1877498				0	vi	Không thể lồng nhau các mảng.				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id4262520				0	vi	\<emph\>Ví dụ :\</emph\>				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id9387493				0	vi	={1;2;3}				2002-02-02 02:02:02
diff --git a/l10n/source/zh-CN/localize.sdf b/l10n/source/zh-CN/localize.sdf
--- l10n/source/zh-CN/localize.sdf
+++ l10n/source/zh-CN/localize.sdf
@@ -39208,7 +39208,6 @@
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id3154342	5			0	zh-CN	结果矩阵的单元格将自动保护，无法对其进行修改。但是，您可以选择整个矩阵单元格区域后，对其进行编辑或复制矩阵公式。				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	hd_id8834803				0	zh-CN	在公式中使用内联数组常量				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id985747				0	zh-CN	Calc 支持在函数中使用内联矩阵/数组常量。一个内联数组用一对花括号括起来。数组元素可以是数字（包括负数）、逻辑常量（TRUE、FALSE）或字符串，但不可以是非常量表达式。一个数组可以输入一行或多行，也可以输入一列或多列。所有行必须含有同样数目的元素， 所有列必须含有同样数目的元素。				2002-02-02 02:02:02
-helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id936613				0	zh-CN	列分隔符（用于分隔行中的元素）是分号 ';'。行分隔符是竖线符号 '|'。这些分隔符与语言和语言环境无关。 				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id1877498				0	zh-CN	数组不可以嵌套。				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id4262520				0	zh-CN	\<emph\>示例:\</emph\>				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id9387493				0	zh-CN	={1;2;3}				2002-02-02 02:02:02
diff --git a/l10n/source/zh-TW/localize.sdf b/l10n/source/zh-TW/localize.sdf
--- l10n/source/zh-TW/localize.sdf
+++ l10n/source/zh-TW/localize.sdf
@@ -38484,7 +38484,6 @@
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id3154342	5			0	zh-TW	結果矩陣中的儲存格自動受保護，無法變更。但是，您可以透過選取整個矩陣儲存格範圍來編輯或複製矩陣公式。				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	hd_id8834803				0	zh-TW	在公式中使用內嵌陣列常數				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id985747				0	zh-TW	Calc 支援在公式中使用內嵌矩陣/陣列常數。內嵌陣列會以大括號「{」和「}」括住。元素可以是數字 (包含負數)、邏輯常數 (TRUE、FALSE) 或文字字串。不允許非常數表示式。輸入的陣列可包含一或多列以及一或多欄。所有列和所有欄皆必須包含相同的元素數目。 				2002-02-02 02:02:02
-helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id936613				0	zh-TW	欄分隔符 (分隔同一列中的元素) 為分號「;」。列分隔符為直線符號「|」。分隔符與語言和語言環境無關。 				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id1877498				0	zh-TW	您無法嵌套陣列。				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id4262520				0	zh-TW	\<emph\>範例：\</emph\>				2002-02-02 02:02:02
 helpcontent2	source\text\scalc\01\04060107.xhp	0	help	par_id9387493				0	zh-TW	={1;2;3}				2002-02-02 02:02:02
