From 0a44e948ab94e3c0ce6feff8cd955d8d77213d6a Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:54:23 +0200
Subject: [PATCH 103/878] helpcontent2-auxiliary-sharedtree.diff

---
 helpcontent2/source/auxiliary/shared.tree |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/helpcontent2/source/auxiliary/shared.tree b/helpcontent2/source/auxiliary/shared.tree
index 857fcab..c0fb09b 100755
--- a/helpcontent2/source/auxiliary/shared.tree
+++ b/helpcontent2/source/auxiliary/shared.tree
@@ -166,7 +166,6 @@
 <topic id="shared/text/shared/guide/space_hyphen.xhp">Inserting Protected Spaces, Hyphens and Conditional Separators</topic>
 <topic id="shared/text/shared/guide/insert_specialchar.xhp">Inserting Special Characters</topic>
 <topic id="shared/text/shared/guide/tabs.xhp">Inserting and Editing Tab Stops</topic>
-<topic id="shared/text/shared/guide/table_cellmerge.xhp">Merging and Splitting Cells</topic>
 <topic id="shared/text/shared/guide/protection.xhp">Protecting Contents in %PRODUCTNAME</topic>
 <topic id="shared/text/shared/guide/redlining_protect.xhp">Protecting Records</topic>
 <topic id="shared/text/shared/guide/pageformat_max.xhp">Selecting the Maximum Printable Area on a Page</topic>
-- 
1.7.0.1

