--- binfilter/bf_svtools/source/config/svt_miscopt.cxx	2010-03-25 20:30:12 +0000
+++ binfilter/bf_svtools/source/config/svt_miscopt.cxx	2010-03-25 21:02:18 +0000
@@ -268,6 +268,7 @@
 		case STYLE_SYMBOLS_INDUSTRIAL: return SFX_SYMBOLS_STYLE_INDUSTRIAL;
 		case STYLE_SYMBOLS_CRYSTAL:    return SFX_SYMBOLS_STYLE_CRYSTAL;
 		case STYLE_SYMBOLS_TANGO:      return SFX_SYMBOLS_STYLE_TANGO;
+		case STYLE_SYMBOLS_HUMAN:      return SFX_SYMBOLS_STYLE_HUMAN;
 	}
 
 	return SFX_SYMBOLS_STYLE_AUTO;
@@ -283,6 +284,7 @@
 		case SFX_SYMBOLS_STYLE_INDUSTRIAL: return STYLE_SYMBOLS_INDUSTRIAL;
 		case SFX_SYMBOLS_STYLE_CRYSTAL:    return STYLE_SYMBOLS_CRYSTAL;
 		case SFX_SYMBOLS_STYLE_TANGO:      return STYLE_SYMBOLS_TANGO;
+		case SFX_SYMBOLS_STYLE_HUMAN:      return STYLE_SYMBOLS_HUMAN;
 	}
 
 	return STYLE_SYMBOLS_AUTO;

--- binfilter/inc/bf_svtools/imgdef.hxx	2010-03-25 20:30:12 +0000
+++ binfilter/inc/bf_svtools/imgdef.hxx	2010-03-25 21:03:12 +0000
@@ -48,7 +48,8 @@
     SFX_SYMBOLS_STYLE_HICONTRAST,
     SFX_SYMBOLS_STYLE_INDUSTRIAL,
     SFX_SYMBOLS_STYLE_CRYSTAL,
-    SFX_SYMBOLS_STYLE_TANGO
+    SFX_SYMBOLS_STYLE_TANGO,
+    SFX_SYMBOLS_STYLE_HUMAN
 };
 
 #define SFX_TOOLBOX_CHANGESYMBOLSET		0x0001

--- officecfg/registry/schema/org/openoffice/Office/Common.xcs	2010-03-25 20:30:12 +0000
+++ officecfg/registry/schema/org/openoffice/Office/Common.xcs	2010-03-25 20:58:36 +0000
@@ -5839,6 +5839,11 @@
 							<desc>Oxygen - the KDE4 default theme</desc>
 						</info>
 					</enumeration>
+					<enumeration oor:value="human">
+						<info>
+							<desc>Human - the Gnome default theme for Ubuntu</desc>
+						</info>
+					</enumeration>
 				</constraints>
 				<value>auto</value>
 			</prop>

--- packimages/pack/makefile.mk	2010-03-25 20:30:12 +0000
+++ packimages/pack/makefile.mk	2010-03-25 21:09:53 +0000
@@ -41,7 +41,7 @@
 IMAGES := $(COMMONBIN)$/images.zip
 SORTED_LIST=$(RES)$/img$/sorted.lst
 # Custom sets, at 24x24 & 16x16 fall-back to industrial preferentially
-CUSTOM_IMAGE_SETS=hicontrast industrial crystal tango classic oxygen
+CUSTOM_IMAGE_SETS=hicontrast industrial crystal tango classic oxygen human
 CUSTOM_IMAGES+=$(foreach,i,$(CUSTOM_IMAGE_SETS) images_$i)
 CUSTOM_PREFERRED_FALLBACK_1*=-c $(SOLARSRC)$/ooo_custom_images$/tango
 CUSTOM_PREFERRED_FALLBACK_2*=-c $(SOLARSRC)$/ooo_custom_images$/industrial

--- svtools/inc/imgdef.hxx	2010-03-25 20:30:12 +0000
+++ svtools/inc/imgdef.hxx	2010-03-25 20:59:24 +0000
@@ -47,7 +47,8 @@
     SFX_SYMBOLS_STYLE_CRYSTAL,
     SFX_SYMBOLS_STYLE_TANGO,
     SFX_SYMBOLS_STYLE_CLASSIC,
-    SFX_SYMBOLS_STYLE_OXYGEN
+    SFX_SYMBOLS_STYLE_OXYGEN,
+    SFX_SYMBOLS_STYLE_HUMAN
 };
 
 #define SFX_TOOLBOX_CHANGESYMBOLSET		0x0001

--- svtools/source/config/miscopt.cxx	2010-03-25 20:30:12 +0000
+++ svtools/source/config/miscopt.cxx	2010-03-25 21:00:18 +0000
@@ -317,6 +317,7 @@
 		case STYLE_SYMBOLS_TANGO:      return SFX_SYMBOLS_STYLE_TANGO;
 		case STYLE_SYMBOLS_CLASSIC:    return SFX_SYMBOLS_STYLE_CLASSIC;
 		case STYLE_SYMBOLS_OXYGEN:     return SFX_SYMBOLS_STYLE_OXYGEN;
+		case STYLE_SYMBOLS_HUMAN:      return SFX_SYMBOLS_STYLE_HUMAN;
 	}
 
 	return SFX_SYMBOLS_STYLE_AUTO;
@@ -334,6 +335,7 @@
 		case SFX_SYMBOLS_STYLE_TANGO:      return STYLE_SYMBOLS_TANGO;
 		case SFX_SYMBOLS_STYLE_CLASSIC:    return STYLE_SYMBOLS_CLASSIC;
 		case SFX_SYMBOLS_STYLE_OXYGEN:     return STYLE_SYMBOLS_OXYGEN;
+		case SFX_SYMBOLS_STYLE_HUMAN:      return STYLE_SYMBOLS_HUMAN;
 	}
 
 	return STYLE_SYMBOLS_AUTO;

--- svx/source/cui/optgdlg.src	2010-03-25 20:30:12 +0000
+++ svx/source/cui/optgdlg.src	2010-03-25 21:14:27 +0000
@@ -265,6 +265,7 @@
             < "Tango" ; > ;
             < "Classic" ; > ;
 			< "Oxygen" ; > ;
+            < "Human" ; > ;
         };
     };
 	CheckBox CB_SYSTEM_FONT

--- vcl/inc/vcl/settings.hxx	2010-03-25 20:30:12 +0000
+++ vcl/inc/vcl/settings.hxx	2010-03-25 21:00:57 +0000
@@ -525,7 +525,8 @@
 #define STYLE_SYMBOLS_TANGO		((ULONG)5)
 #define STYLE_SYMBOLS_CLASSIC		((ULONG)6)
 #define STYLE_SYMBOLS_OXYGEN		((ULONG)7)
-#define STYLE_SYMBOLS_THEMES_MAX	((ULONG)8)
+#define STYLE_SYMBOLS_HUMAN		((ULONG)8)
+#define STYLE_SYMBOLS_THEMES_MAX	((ULONG)9)
 
 #define STYLE_CURSOR_NOBLINKTIME    ((ULONG)0xFFFFFFFF)
 

--- vcl/source/app/settings.cxx	2010-03-25 20:30:12 +0000
+++ vcl/source/app/settings.cxx	2010-03-25 21:25:20 +0000
@@ -712,6 +712,7 @@
 		case STYLE_SYMBOLS_TANGO:      return ::rtl::OUString::createFromAscii( "tango" );
 		case STYLE_SYMBOLS_CLASSIC:    return ::rtl::OUString::createFromAscii( "classic" );
 		case STYLE_SYMBOLS_OXYGEN:     return ::rtl::OUString::createFromAscii( "oxygen" );
+		case STYLE_SYMBOLS_HUMAN:      return ::rtl::OUString::createFromAscii( "human" );
 	}
 
 	return ::rtl::OUString::createFromAscii( "auto" );
@@ -735,6 +736,8 @@
 		return STYLE_SYMBOLS_CLASSIC;
 	else if ( rName == ::rtl::OUString::createFromAscii( "oxygen" ) )
 		return STYLE_SYMBOLS_OXYGEN;
+	else if ( rName == ::rtl::OUString::createFromAscii( "human" ) )
+		return STYLE_SYMBOLS_HUMAN;
 
 	return STYLE_SYMBOLS_AUTO;
 }
@@ -830,7 +833,7 @@
     {
         if( rDesktopEnvironment.equalsIgnoreAsciiCaseAscii( "gnome" ) ||
 	     rDesktopEnvironment.equalsIgnoreAsciiCaseAscii( "windows" ) )
-            nRet = STYLE_SYMBOLS_TANGO;
+            nRet = STYLE_SYMBOLS_HUMAN;
         else if( rDesktopEnvironment.equalsIgnoreAsciiCaseAscii( "kde" ) )
             nRet = STYLE_SYMBOLS_CRYSTAL;
 		else if( rDesktopEnvironment.equalsIgnoreAsciiCaseAscii( "kde4" ) )

