=== modified file 'officecfg/registry/schema/org/openoffice/Office/Common.xcs'
--- officecfg/registry/schema/org/openoffice/Office/Common.xcs	2009-10-11 20:36:10 +0000
+++ officecfg/registry/schema/org/openoffice/Office/Common.xcs	2009-10-11 21:08:29 +0000
@@ -5696,6 +5696,11 @@
 							<desc>Oxygen - the KDE4 default theme</desc>
 						</info>
 					</enumeration>
+					<enumeration oor:value="human">
+						<info>
+							<desc>Human - the Gnome default theme for Ubuntu</desc>
+						</info>
+					</enumeration>
 				</constraints>
 				<value>auto</value>
 			</prop>

=== modified file 'svtools/inc/imgdef.hxx'
--- svtools/inc/imgdef.hxx	2009-10-11 20:36:10 +0000
+++ svtools/inc/imgdef.hxx	2009-10-11 21:11:14 +0000
@@ -47,7 +47,8 @@
     SFX_SYMBOLS_STYLE_CRYSTAL,
     SFX_SYMBOLS_STYLE_TANGO,
     SFX_SYMBOLS_STYLE_CLASSIC,
-    SFX_SYMBOLS_STYLE_OXYGEN
+    SFX_SYMBOLS_STYLE_OXYGEN,
+    SFX_SYMBOLS_STYLE_HUMAN
 };
 
 #define SFX_TOOLBOX_CHANGESYMBOLSET		0x0001

=== modified file 'svtools/source/config/miscopt.cxx'
--- svtools/source/config/miscopt.cxx	2009-10-11 20:36:10 +0000
+++ svtools/source/config/miscopt.cxx	2009-10-11 21:13:30 +0000
@@ -276,6 +276,7 @@
 		case STYLE_SYMBOLS_TANGO:      return SFX_SYMBOLS_STYLE_TANGO;
 		case STYLE_SYMBOLS_CLASSIC:    return SFX_SYMBOLS_STYLE_CLASSIC;
 		case STYLE_SYMBOLS_OXYGEN:     return SFX_SYMBOLS_STYLE_OXYGEN;
+		case STYLE_SYMBOLS_HUMAN:      return SFX_SYMBOLS_STYLE_HUMAN;
 	}
 
 	return SFX_SYMBOLS_STYLE_AUTO;
@@ -293,6 +294,7 @@
 		case SFX_SYMBOLS_STYLE_TANGO:      return STYLE_SYMBOLS_TANGO;
 		case SFX_SYMBOLS_STYLE_CLASSIC:    return STYLE_SYMBOLS_CLASSIC;
 		case SFX_SYMBOLS_STYLE_OXYGEN:     return STYLE_SYMBOLS_OXYGEN;
+		case SFX_SYMBOLS_STYLE_HUMAN:      return STYLE_SYMBOLS_HUMAN;
 	}
 
 	return STYLE_SYMBOLS_AUTO;

=== modified file 'svx/source/dialog/optgdlg.src'
--- svx/source/dialog/optgdlg.src	2009-10-11 20:36:10 +0000
+++ svx/source/dialog/optgdlg.src	2009-10-11 21:20:23 +0000
@@ -252,6 +252,7 @@
             < "Tango" ; > ;
             < "Classic" ; > ;
 			< "Oxygen" ; > ;
+            < "Human" ; > ;
         };
     };
 	CheckBox CB_SYSTEM_FONT

=== modified file 'vcl/inc/vcl/settings.hxx'
--- vcl/inc/vcl/settings.hxx	2009-10-11 20:36:10 +0000
+++ vcl/inc/vcl/settings.hxx	2009-10-11 21:24:08 +0000
@@ -520,7 +520,8 @@
 #define STYLE_SYMBOLS_TANGO		((ULONG)5)
 #define STYLE_SYMBOLS_CLASSIC		((ULONG)6)
 #define STYLE_SYMBOLS_OXYGEN		((ULONG)7)
-#define STYLE_SYMBOLS_THEMES_MAX	((ULONG)8)
+#define STYLE_SYMBOLS_HUMAN		((ULONG)8)
+#define STYLE_SYMBOLS_THEMES_MAX	((ULONG)9)
 
 #define STYLE_CURSOR_NOBLINKTIME    ((ULONG)0xFFFFFFFF)
 

=== modified file 'vcl/source/app/settings.cxx'
--- vcl/source/app/settings.cxx	2009-10-11 20:36:10 +0000
+++ vcl/source/app/settings.cxx	2009-10-11 21:55:35 +0000
@@ -710,6 +710,7 @@
 		case STYLE_SYMBOLS_TANGO:      return ::rtl::OUString::createFromAscii( "tango" );
 		case STYLE_SYMBOLS_CLASSIC:    return ::rtl::OUString::createFromAscii( "classic" );
 		case STYLE_SYMBOLS_OXYGEN:     return ::rtl::OUString::createFromAscii( "oxygen" );
+		case STYLE_SYMBOLS_HUMAN:      return ::rtl::OUString::createFromAscii( "human" );
 	}
 
 	return ::rtl::OUString::createFromAscii( "auto" );
@@ -733,6 +734,8 @@
 		return STYLE_SYMBOLS_CLASSIC;
 	else if ( rName == ::rtl::OUString::createFromAscii( "oxygen" ) )
 		return STYLE_SYMBOLS_OXYGEN;
+	else if ( rName == ::rtl::OUString::createFromAscii( "human" ) )
+		return STYLE_SYMBOLS_HUMAN;
 
 	return STYLE_SYMBOLS_AUTO;
 }
@@ -782,7 +785,7 @@
 
 			// use a hardcoded desktop-specific fallback if no preferred style has been detected
 			static bool sbFallbackDesktopChecked = false;
-			static ULONG snFallbackDesktopStyle = STYLE_SYMBOLS_DEFAULT;
+			static ULONG snFallbackDesktopStyle = STYLE_SYMBOLS_HUMAN;
 
 			if ( !sbFallbackDesktopChecked )
 			{
@@ -807,7 +810,7 @@
 ULONG StyleSettings::GetAutoSymbolsStyle() const
 {
     const ::rtl::OUString&      rDesktopEnvironment = Application::GetDesktopEnvironment();
-    ULONG                       nRet = STYLE_SYMBOLS_DEFAULT;
+    ULONG                       nRet = STYLE_SYMBOLS_HUMAN;
     bool                        bCont = true;
 
     try
@@ -828,7 +831,7 @@
     {
         if( rDesktopEnvironment.equalsIgnoreAsciiCaseAscii( "gnome" ) ||
 	     rDesktopEnvironment.equalsIgnoreAsciiCaseAscii( "windows" ) )
-            nRet = STYLE_SYMBOLS_TANGO;
+            nRet = STYLE_SYMBOLS_HUMAN;
         else if( rDesktopEnvironment.equalsIgnoreAsciiCaseAscii( "kde" ) )
             nRet = STYLE_SYMBOLS_CRYSTAL;
 		else if( rDesktopEnvironment.equalsIgnoreAsciiCaseAscii( "kde4" ) )

