From 0abe7da142ff9c37bb114e21f5a71f692782024e Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:58:23 +0200
Subject: [PATCH 314/878] i18nutil-glibc-braindamage-workaround.diff

---
 i18nutil/source/utility/unicode.cxx |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/i18nutil/source/utility/unicode.cxx b/i18nutil/source/utility/unicode.cxx
index 9bbd8cb..08f1af7 100644
--- a/i18nutil/source/utility/unicode.cxx
+++ b/i18nutil/source/utility/unicode.cxx
@@ -30,6 +30,11 @@
 #include <i18nutil/unicode.hxx>
 #include "unicode_data.h"
 
+// Workaround for glibc braindamage:
+// glibc 2.4's langinfo.h does "#define CURRENCY_SYMBOL __CURRENCY_SYMBOL"
+// which (obviously) breaks UnicodeType::CURRENCY_SYMBOL
+#undef CURRENCY_SYMBOL
+
 using namespace ::com::sun::star::i18n;
 
 static ScriptTypeList defaultTypeList[] = {
-- 
1.7.0.1

