---
 i18nutil/source/utility/unicode.cxx |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git i18nutil/source/utility/unicode.cxx i18nutil/source/utility/unicode.cxx
index 6e14509..6772da7 100644
--- i18nutil/source/utility/unicode.cxx
+++ i18nutil/source/utility/unicode.cxx
@@ -30,6 +30,11 @@
 #include <i18nutil/unicode.hxx>
 #include "unicode_data.h"
 
+// Workaround for glibc braindamage:
+// glibc 2.4's langinfo.h does "#define CURRENCY_SYMBOL __CURRENCY_SYMBOL"
+// which (obviously) breaks UnicodeType::CURRENCY_SYMBOL
+#undef CURRENCY_SYMBOL
+
 using namespace ::com::sun::star::i18n;
 
 static ScriptTypeList defaultTypeList[] = {
-- 
1.7.0.1

