Index: solenv/inc/unxfbsdx.mk
===================================================================
--- solenv/inc/unxfbsdx.mk	(revision 276718)
+++ solenv/inc/unxfbsdx.mk	(working copy)
@@ -50,3 +50,4 @@
 
 # platform specific identifier for shared libs
 DLLPOSTFIX=fx
+BUILD64=1
Index: solenv/inc/unxlnga.mk
===================================================================
--- solenv/inc/unxlnga.mk	(revision 276792)
+++ solenv/inc/unxlnga.mk	(working copy)
@@ -34,3 +34,4 @@
 .INCLUDE : unxlng.mk
 CFLAGS+=-DIA64
 DLLPOSTFIX=la
+BUILD64=1
Index: nss/makefile.mk
===================================================================
--- nss/makefile.mk	(revision 276718)
+++ nss/makefile.mk	(working copy)
@@ -58,7 +58,7 @@
 
 .IF "$(GUI)"=="UNX"
 .IF "$(OS)$(COM)"=="LINUXGCC"
-.IF "$(CPU)"=="X"
+.IF "$(BUILD64)"=="1"
 # force 64-bit buildmode
 USE_64:=1
 .EXPORT : USE_64
