--- solenv/inc/unxfbsdx.mk.old	2009-10-07 15:06:50.000000000 +0200
+++ solenv/inc/unxfbsdx.mk	2010-03-03 16:59:23.000000000 +0100
@@ -50,3 +50,4 @@ CFLAGSOPT=   					# no optimizing for no
 
 # platform specific identifier for shared libs
 DLLPOSTFIX=fx
+BUILD64=1
--- solenv/inc/unxlnga.mk.old	2009-10-19 13:27:59.000000000 +0200
+++ solenv/inc/unxlnga.mk	2010-03-03 16:59:23.000000000 +0100
@@ -35,3 +35,4 @@
 CFLAGS+=-DIA64
 CFLAGSCXX+=-fno-use-cxa-atexit
 DLLPOSTFIX=la
+BUILD64=1
--- nss/makefile.mk.old	2010-01-21 01:01:04.000000000 +0100
+++ nss/makefile.mk	2010-03-03 16:59:23.000000000 +0100
@@ -60,7 +60,7 @@ BUILD_OPT=1
 
 .IF "$(GUI)"=="UNX"
 .IF "$(OS)$(COM)"=="LINUXGCC"
-.IF "$(CPU)"=="X"
+.IF "$(BUILD64)"=="1"
 # force 64-bit buildmode
 USE_64:=1
 .EXPORT : USE_64
