--- icu/makefile.mk~	2009-09-12 12:57:47.000000000 +0200
+++ icu/makefile.mk	2009-09-12 13:13:18.000000000 +0200
@@ -84,8 +84,16 @@
 .ENDIF
 
 icu_CFLAGS+=-O $(ARCH_FLAGS) $(EXTRA_CDEFS)
+.IF "$(CPUNAME)" == "ARM" && "$(OS)" == "LINUX"
+icu_CFLAGS+=-Os
+.ENDIF
+icu_CFLAGS+=$(EXTRA_CDEFS)
 icu_LDFLAGS+=$(EXTRA_LINKFLAGS)
-icu_CXXFLAGS+=-O $(ARCH_FLAGS) $(EXTRA_CDEFS)
+icu_CXXFLAGS+=-O $(ARCH_FLAGS)
+.IF "$(CPUNAME)" == "ARM" && "$(OS)" == "LINUX"
+icu_CXXFLAGS+=-Os
+.ENDIF
+icu_CXXFLAGS+=$(EXTRA_CDEFS)
 
 # remove conversion and transliteration data to reduce binary size.
 CONFIGURE_ACTION=rm data/mappings/ucm*.mk data/translit/trn*.mk ;
