--- icu/makefile.mk~	2009-09-12 12:57:47.000000000 +0200
+++ icu/makefile.mk	2009-09-12 13:13:18.000000000 +0200
@@ -85,8 +85,8 @@
 icu_LDFLAGS+=-Wl,-z,noexecstack
 .ENDIF
 
-icu_CFLAGS+=-O $(ARCH_FLAGS) $(EXTRA_CDEFS)
-icu_CXXFLAGS+=-O $(ARCH_FLAGS) $(EXTRA_CDEFS)
+icu_CFLAGS+=-O $(ARCH_FLAGS) $(ICU_ARCH_FLAGS) $(EXTRA_CDEFS)
+icu_CXXFLAGS+=-O $(ARCH_FLAGS) $(ICU_ARCH_FLAGS) $(EXTRA_CDEFS)
 
 BUILD_ACTION_SEP=;
 # remove conversion and transliteration data to reduce binary size.
