--- instsetoo_native/util/makefile.mk
+++ instsetoo_native/util/makefile.mk
@@ -85,9 +85,9 @@ ALLTAR  : $(LOCALPYFILES)
 .ELSE			# "$(GUI)"!="WNT" && "$(EPM)"=="NO" && "$(USE_PACKAGER)"==""
 .IF "$(UPDATER)"=="" || "$(USE_PACKAGER)"==""
 .IF "$(BUILD_TYPE)"=="$(BUILD_TYPE:s/ODK//)"
-ALLTAR : openoffice_$(defaultlangiso) ooolanguagepack ooohelppack
+ALLTAR : openofficeall ooohelppack
 .ELSE
-ALLTAR : openoffice_$(defaultlangiso) ooolanguagepack ooohelppack sdkoo_en-US ure_en-US
+ALLTAR : openofficeall sdkooall ooohelppack
 .ENDIF
 .ELSE			# "$(UPDATER)"=="" || "$(USE_PACKAGER)"==""
 ALLTAR : updatepack
@@ -129,7 +129,13 @@
 
 ooodevlanguagepack: $(foreach,i,$(alllangiso) ooodevlanguagepack_$i)
 
-ooohelppack : $(foreach,i,$(alllangiso) ooohelppack_$i)
+goodhelplangs := $(alllangiso)
+
+.IF "$(WITH_POOR_HELP_LOCALIZATIONS)"!=""
+$(foreach,i,$(WITH_POOR_HELP_LOCALIZATIONS) dog_vomit_slime_mold_$i : $(assign goodhelplangs := $(goodhelplangs:s/$i//)))
+.ENDIF
+
+ooohelppack : $(foreach,i,$(goodhelplangs) ooohelppack_$i)
 
 ooodevhelppack: $(foreach,i,$(alllangiso) ooodevhelppack_$i)
 
