From d14e0471cee57f14e6dfb7889c9e825876b3d013 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:04:47 +0200
Subject: [PATCH 560/768] internal-mesa-headers-config_office.diff

---
 configure.in |   17 +++++++++++++++++
 set_soenv.in |    1 +
 2 files changed, 18 insertions(+), 0 deletions(-)

diff --git configure.in configure.in
index 45f7202..cee6af5 100644
--- configure.in
+++ configure.in
@@ -529,6 +529,9 @@ AC_ARG_WITH(system-sane,
 AC_ARG_WITH(system-xrender,
 [  --with-system-xrender-headers  Use XRender headers already on system
 ],,)
+AC_ARG_WITH(system-mesa-headers,
+[  --with-system-mesa-headers     Use Mesa headers already on system
+],,)
 AC_ARG_WITH(system-curl,
 [  --with-system-curl      Use curl already on system
 ],,)
@@ -5131,6 +5134,20 @@ AC_SUBST(XRANDR_LIBS)
 AC_SUBST(ENABLE_RANDR)
 
 dnl ===================================================================
+dnl Check for system Mesa
+dnl ===================================================================
+AC_MSG_CHECKING([which Mesa headers to use])
+if test "$with_system_mesa_headers" == "no"; then
+   AC_MSG_RESULT([internal])
+   SYSTEM_MESA_HEADERS=NO
+   BUILD_TYPE="$BUILD_TYPE MESA"
+else
+   AC_MSG_RESULT([external])
+   SYSTEM_MESA_HEADERS=YES
+fi
+AC_SUBST(SYSTEM_MESA_HEADERS)
+
+dnl ===================================================================
 dnl Check for building neon
 dnl ===================================================================
 AC_MSG_CHECKING([whether to use neon])
diff --git set_soenv.in set_soenv.in
index ad90636..5e1ceb3 100644
--- set_soenv.in
+++ set_soenv.in
@@ -1875,6 +1875,7 @@ ToFile( "LIBXSLT_CFLAGS",    "@LIBXSLT_CFLAGS@",   "e" );
 ToFile( "LIBXSLT_LIBS",      "@LIBXSLT_LIBS@",     "e" );
 ToFile( "SYSTEM_SANE_HEADER","@SYSTEM_SANE_HEADER@","e" );
 ToFile( "SYSTEM_XRENDER_HEADERS","@SYSTEM_XRENDER_HEADERS@","e" );
+ToFile( "SYSTEM_MESA_HEADERS","@SYSTEM_MESA_HEADERS@","e" );
 ToFile( "SYSTEM_CURL",       "@SYSTEM_CURL@",      "e" );
 ToFile( "CURL_CFLAGS",       "@CURL_CFLAGS@",      "e" );
 ToFile( "CURL_LIBS",         "@CURL_LIBS@",        "e" );
-- 
1.7.0.1

