---
 configure.in |   17 +++++++++++++++++
 set_soenv.in |    1 +
 2 files changed, 18 insertions(+), 0 deletions(-)

diff --git configure.in configure.in
index 6bcf3f0..411e922 100644
--- configure.in
+++ configure.in
@@ -525,6 +525,9 @@ AC_ARG_WITH(system-sane,
 AC_ARG_WITH(system-xrender,
 [  --with-system-xrender-headers  Use XRender headers already on system
 ],,)
+AC_ARG_WITH(system-mesa-headers,
+[  --with-system-mesa-headers     Use Mesa headers already on system
+],,)
 AC_ARG_WITH(system-curl,
 [  --with-system-curl      Use curl already on system
 ],,)
@@ -5133,6 +5136,20 @@ AC_SUBST(XRANDR_LIBS)
 AC_SUBST(ENABLE_RANDR)
 
 dnl ===================================================================
+dnl Check for system Mesa
+dnl ===================================================================
+AC_MSG_CHECKING([which Mesa headers to use])
+if test "$with_system_mesa_headers" == "no"; then
+   AC_MSG_RESULT([internal])
+   SYSTEM_MESA_HEADERS=NO
+   BUILD_TYPE="$BUILD_TYPE MESA"
+else
+   AC_MSG_RESULT([external])
+   SYSTEM_MESA_HEADERS=YES
+fi
+AC_SUBST(SYSTEM_MESA_HEADERS)
+
+dnl ===================================================================
 dnl Check for building neon
 dnl ===================================================================
 AC_MSG_CHECKING([whether to use neon])
diff --git set_soenv.in set_soenv.in
index 6790259..8824c8b 100644
--- set_soenv.in
+++ set_soenv.in
@@ -1916,6 +1916,7 @@ ToFile( "LIBXSLT_CFLAGS",    "@LIBXSLT_CFLAGS@",   "e" );
 ToFile( "LIBXSLT_LIBS",      "@LIBXSLT_LIBS@",     "e" );
 ToFile( "SYSTEM_SANE_HEADER","@SYSTEM_SANE_HEADER@","e" );
 ToFile( "SYSTEM_XRENDER_HEADERS","@SYSTEM_XRENDER_HEADERS@","e" );
+ToFile( "SYSTEM_MESA_HEADERS","@SYSTEM_MESA_HEADERS@","e" );
 ToFile( "SYSTEM_CURL",       "@SYSTEM_CURL@",      "e" );
 ToFile( "CURL_CFLAGS",       "@CURL_CFLAGS@",      "e" );
 ToFile( "CURL_LIBS",         "@CURL_LIBS@",        "e" );
-- 
1.7.0.1

