From 27bb1c46ed61f021c747ac6a4526d8371858c373 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:04:48 +0200
Subject: [PATCH 638/878] internal-mesa-headers-slideshow.diff

---
 slideshow/prj/build.lst |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/slideshow/prj/build.lst b/slideshow/prj/build.lst
index ec0cf8c..7b51243 100644
--- a/slideshow/prj/build.lst
+++ b/slideshow/prj/build.lst
@@ -1,4 +1,4 @@
-pe	slideshow	:	tools sal cppu vcl comphelper cppuhelper offuh cppcanvas basegfx goodies unotools toolkit animations avmedia NULL
+pe	slideshow	:	tools sal cppu vcl comphelper cppuhelper offuh cppcanvas basegfx goodies unotools toolkit animations avmedia MESA:Mesa NULL
 pe	slideshow								usr1	-	all	pe_mkout NULL
 pe	slideshow\inc							nmake	-	all	pe_inc NULL
 pe	slideshow\source\engine					nmake	-	all	pe_engine pe_inc NULL
-- 
1.7.0.1

