--- configure.in.orig	2008-03-18 19:27:49.000000000 +0100
+++ configure.in	2008-03-18 19:29:49.000000000 +0100
@@ -745,7 +745,7 @@
 ],with_build_version=$withval)
 AC_ARG_WITH(alloc,
 [  --with-alloc            Define which allocator to build with
-			                 (choices are oo, system, tcmalloc)
+                           (choices are oo, system, tcmalloc, jemalloc)
 ],,)
 AC_ARG_WITH(sun-templates,
 [
@@ -2499,6 +2499,14 @@
         [AC_MSG_ERROR(tcmalloc not found or functional. Install the Google Profiling Tools)], [])
     ALLOC="TCMALLOC";
 fi
+if test "$with_alloc" = "jemalloc"; then
+    save_CFLAGS=$CFLAGS
+    CFLAGS="$CFLAGS -pthread"
+    AC_CHECK_LIB(jemalloc, malloc, [],
+        [AC_MSG_ERROR(jemalloc not found or functional. Install the jemalloc allocator.)], [])
+    ALLOC="JEMALLOC";
+    CFLAGS=$save_CFLAGS
+fi
 if test "$with_alloc" = "internal" -o -z "$with_alloc"; then
    AC_MSG_RESULT([internal])
 fi
Index: unxlng.mk
===================================================================
--- solenv/inc/unxlng.mk	(Revision 276754)
+++ solenv/inc/unxlng.mk	(Arbeitskopie)
@@ -201,6 +201,14 @@
 STDSHLGUIMT+=-ltcmalloc
 STDSHLCUIMT+=-ltcmalloc
 .ENDIF
+
+.IF "$(ALLOC)" == "JEMALLOC"
+STDLIBGUIMT+=-ljemalloc
+STDLIBCUIMT+=-ljemalloc
+STDSHLGUIMT+=-ljemalloc
+STDSHLCUIMT+=-ljemalloc
+.ENDIF
+
 .IF "$(HAVE_LD_HASH_STYLE)"  == "TRUE"
 LINKFLAGS += -Wl,--hash-style=$(WITH_LINKER_HASH_STYLE)
 .ELSE
--- sal/rtl/source/makefile.mk.orig	2008-03-18 19:41:00.000000000 +0100
+++ sal/rtl/source/makefile.mk	2008-03-18 19:42:42.000000000 +0100
@@ -54,7 +54,7 @@
 
 .INCLUDE :  settings.mk
 
-.IF "$(ALLOC)" == "SYS_ALLOC" || "$(ALLOC)" == "TCMALLOC"
+.IF "$(ALLOC)" == "SYS_ALLOC" || "$(ALLOC)" == "TCMALLOC" || "$(ALLOC)" == "JEMALLOC"
 CDEFS+= -DFORCE_SYSALLOC
 .ENDIF
 
