From b8c987cab4122b200087f5664f505b5d1a501da9 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:05:05 +0200
Subject: [PATCH 654/878] jurt-jnilib-deliver.diff

---
 jurt/prj/d.lst |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/jurt/prj/d.lst b/jurt/prj/d.lst
index 5223fe2..f07cd29 100644
--- a/jurt/prj/d.lst
+++ b/jurt/prj/d.lst
@@ -1,6 +1,6 @@
 ..\%__SRC%\class\jurt.jar %_DEST%\bin%_EXT%\jurt.jar
 ..\%__SRC%\bin\jpipe*.dll %_DEST%\bin%_EXT%\jpipe*.dll
 ..\%__SRC%\lib\libjpipe*.so %_DEST%\lib%_EXT%\libjpipe*.so
-..\%__SRC%\lib\libjpipe*.dylib %_DEST%\lib%_EXT%\libjpipe*.dylib
+..\%__SRC%\lib\libjpipe.jnilib %_DEST%\lib%_EXT%\libjpipe.jnilib
 
 ..\%__SRC%\bin\jurt_src.zip %COMMON_DEST%\bin%_EXT%\jurt_src.zip
-- 
1.7.0.1

