---
 jurt/prj/d.lst |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git jurt/prj/d.lst jurt/prj/d.lst
index 848435f..a55c9fa 100644
--- jurt/prj/d.lst
+++ jurt/prj/d.lst
@@ -2,6 +2,6 @@
 ..\%__SRC%\bin\jpipe.dll %_DEST%\bin%_EXT%\jpipe.dll
 ..\%__SRC%\bin\jpipx.dll %_DEST%\bin%_EXT%\jpipx.dll
 ..\%__SRC%\lib\libjpipe*.so %_DEST%\lib%_EXT%\libjpipe*.so
-..\%__SRC%\lib\libjpipe*.dylib %_DEST%\lib%_EXT%\libjpipe*.dylib
+..\%__SRC%\lib\libjpipe.jnilib %_DEST%\lib%_EXT%\libjpipe.jnilib
 
 ..\%__SRC%\bin\jurt_src.zip %COMMON_DEST%\bin%_EXT%\jurt_src.zip
-- 
1.7.0.1

