diff --git jvmfwk/plugins/sunmajor/pluginlib/sunjavaplugin.cxx jvmfwk/plugins/sunmajor/pluginlib/sunjavaplugin.cxx
index a4c7f4f..2e7ea13 100644
--- jvmfwk/plugins/sunmajor/pluginlib/sunjavaplugin.cxx
+++ jvmfwk/plugins/sunmajor/pluginlib/sunjavaplugin.cxx
@@ -516,6 +516,7 @@ javaPluginError jfw_plugin_startJavaVirtualMachine(
     // handler
 #ifdef UNX
     struct sigaction act;
+    sigemptyset(&act.sa_mask);
     act.sa_handler=SIG_DFL;
     act.sa_flags= 0;
     sigaction( SIGSEGV, &act, NULL);
