diff --git fpicker/source/unx/kde4/KDE4FilePicker.cxx fpicker/source/unx/kde4/KDE4FilePicker.cxx
index d7dfbc1..a5418d3 100644
--- fpicker/source/unx/kde4/KDE4FilePicker.cxx
+++ fpicker/source/unx/kde4/KDE4FilePicker.cxx
@@ -163,8 +163,6 @@ KDE4FilePicker::KDE4FilePicker( const uno::Reference<lang::XMultiServiceFactory>
     
 	_dialog = new KFileDialog(KUrl("~"), QString(""), 0, _extraControls);
     _dialog->setMode(KFile::File | KFile::LocalOnly);
-    
-	_dialog->setStyleSheet("color: black;");
 }
 
 KDE4FilePicker::~KDE4FilePicker()
@@ -209,6 +207,7 @@ sal_Int16 SAL_CALL KDE4FilePicker::execute()
     
 	_dialog->clearFilter();
     _dialog->setFilter(_filter);
+	_dialog->filterWidget()->setEditable(false);
 	
     //block and wait for user input
 	if (_dialog->exec() == KFileDialog::Accepted)
@@ -260,7 +259,7 @@ uno::Sequence< ::rtl::OUString > SAL_CALL KDE4FilePicker::getFiles()
 		QCheckBox *cb = dynamic_cast<QCheckBox*> (
 			_customWidgets[ExtendedFilePickerElementIds::CHECKBOX_AUTOEXTENSION ]);
 			
-		if (cb->isChecked())
+		if (cb && cb->isChecked())
 		{
 			extension = _dialog->currentFilter(); // assuming filter value is like this *.ext
 			extension.replace("*","");
@@ -655,27 +654,27 @@ void SAL_CALL KDE4FilePicker::initialize( const uno::Sequence<uno::Any> &args )
             
         case FILESAVE_AUTOEXTENSION:
             operationMode = KFileDialog::Saving;
-            addCustomControl( ExtendedFilePickerElementIds::CHECKBOX_AUTOEXTENSION );
+            //addCustomControl( ExtendedFilePickerElementIds::CHECKBOX_AUTOEXTENSION );
             break;
 
         case FILESAVE_AUTOEXTENSION_PASSWORD:
         {
             operationMode = KFileDialog::Saving;
-            addCustomControl( ExtendedFilePickerElementIds::CHECKBOX_AUTOEXTENSION );
+            //addCustomControl( ExtendedFilePickerElementIds::CHECKBOX_AUTOEXTENSION );
             addCustomControl( ExtendedFilePickerElementIds::CHECKBOX_PASSWORD );
             break;
         }
         case FILESAVE_AUTOEXTENSION_PASSWORD_FILTEROPTIONS:
         {
             operationMode = KFileDialog::Saving;
-            addCustomControl( ExtendedFilePickerElementIds::CHECKBOX_AUTOEXTENSION );
+            //addCustomControl( ExtendedFilePickerElementIds::CHECKBOX_AUTOEXTENSION );
             addCustomControl( ExtendedFilePickerElementIds::CHECKBOX_PASSWORD );
             addCustomControl( ExtendedFilePickerElementIds::CHECKBOX_FILTEROPTIONS );
             break;
         }
         case FILESAVE_AUTOEXTENSION_SELECTION:
             operationMode = KFileDialog::Saving;
-            addCustomControl( ExtendedFilePickerElementIds::CHECKBOX_AUTOEXTENSION );
+            //addCustomControl( ExtendedFilePickerElementIds::CHECKBOX_AUTOEXTENSION );
             addCustomControl( ExtendedFilePickerElementIds::CHECKBOX_SELECTION );
             break;
 
@@ -712,6 +711,9 @@ void SAL_CALL KDE4FilePicker::initialize( const uno::Sequence<uno::Any> &args )
     }
     
     _dialog->setOperationMode(operationMode);
+#if KDE_IS_VERSION(4,2,0) // version 4.2 and above
+	_dialog->setConfirmOverwrite(true);
+endif
 }
 
 void SAL_CALL KDE4FilePicker::cancel()
