diff --git a/vcl/unx/kde4/KDESalGraphics.cxx b/vcl/unx/kde4/KDESalGraphics.cxx
index 3be4baa..7e69722 100644
--- vcl/unx/kde4/KDESalGraphics.cxx
+++ vcl/unx/kde4/KDESalGraphics.cxx
@@ -671,7 +671,7 @@ BOOL KDESalGraphics::getNativeControlRegion( ControlType type, ControlPart part,
         {
             if( part == PART_BORDER )
             {
-                int size = kapp->style()->pixelMetric(QStyle::PM_LayoutLeftMargin);
+                int size = kapp->style()->pixelMetric(QStyle::PM_DefaultFrameWidth);
                 USHORT nStyle = val.getNumericVal();
                 if( nStyle & FRAME_DRAW_NODRAW )
                 {
