From a6d68aa60b47257bccb9cda14ec44fba2e961850 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:01:09 +0200
Subject: [PATCH 379/768] kde4-border-frame.diff

---
 vcl/unx/kde4/KDESalGraphics.cxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git vcl/unx/kde4/KDESalGraphics.cxx vcl/unx/kde4/KDESalGraphics.cxx
index 45feda1..7ed346f 100644
--- vcl/unx/kde4/KDESalGraphics.cxx
+++ vcl/unx/kde4/KDESalGraphics.cxx
@@ -744,7 +744,7 @@ BOOL KDESalGraphics::getNativeControlRegion( ControlType type, ControlPart part,
         {
             if( part == PART_BORDER )
             {
-                int size = kapp->style()->pixelMetric(QStyle::PM_LayoutLeftMargin);
+                int size = kapp->style()->pixelMetric(QStyle::PM_DefaultFrameWidth);
                 USHORT nStyle = val.getNumericVal();
                 if( nStyle & FRAME_DRAW_NODRAW )
                 {
-- 
1.7.0.1

