Index: source/unx/kde4/KDE4FilePicker.cxx
===================================================================
--- fpicker/source/unx/kde4/KDE4FilePicker.cxx	(Revision 275397)
+++ fpicker/source/unx/kde4/KDE4FilePicker.cxx	(Arbeitskopie)
@@ -136,7 +136,7 @@
 
 rtl::OUString toOUString(const QString& s)
 {
-	return rtl::OUString(s.toUtf8().data(), s.length(), RTL_TEXTENCODING_UTF8);
+	return rtl::OUString(s.toUtf8().data(), s.toUtf8().length(), RTL_TEXTENCODING_UTF8);
 }
 
 QString toQString(const rtl::OUString& s)
