From 6bb8e24cfda130dc840bc721ddd13d2d5015a66e Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:01:09 +0200
Subject: [PATCH 445/878] kde4-border-frame.diff

---
 vcl/unx/kde4/KDESalGraphics.cxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/vcl/unx/kde4/KDESalGraphics.cxx b/vcl/unx/kde4/KDESalGraphics.cxx
index 0609e11..86c9eb3 100644
--- a/vcl/unx/kde4/KDESalGraphics.cxx
+++ b/vcl/unx/kde4/KDESalGraphics.cxx
@@ -671,7 +671,7 @@ BOOL KDESalGraphics::getNativeControlRegion( ControlType type, ControlPart part,
         {
             if( part == PART_BORDER )
             {
-                int size = kapp->style()->pixelMetric(QStyle::PM_LayoutLeftMargin);
+                int size = kapp->style()->pixelMetric(QStyle::PM_DefaultFrameWidth);
                 USHORT nStyle = val.getNumericVal();
                 if( nStyle & FRAME_DRAW_NODRAW )
                 {
-- 
1.7.0.1

