--- officecfg/registry/schema/org/openoffice/Office/Common.xcs.old	2010-03-03 16:59:29.000000000 +0100
+++ officecfg/registry/schema/org/openoffice/Office/Common.xcs	2010-03-03 16:59:44.000000000 +0100
@@ -5784,6 +5784,11 @@ Dymamic border coloring means that when
 							<desc>Classic - the OpenOffice.org 2.0 default theme</desc>
 						</info>
 					</enumeration>
+                    <enumeration oor:value="oxygen">
+                        <info>
+                            <desc>Oxygen - the KDE4 default theme</desc>
+                        </info>
+                    </enumeration>
 				</constraints>
 				<value>auto</value>
 			</prop>
--- packimages/pack/makefile.mk.old	2009-10-07 15:07:27.000000000 +0200
+++ packimages/pack/makefile.mk	2010-03-03 16:59:44.000000000 +0100
@@ -41,13 +41,14 @@ RSCCUSTOMIMG*=$(PRJ)
 IMAGES := $(COMMONBIN)$/images.zip
 SORTED_LIST=$(RES)$/img$/sorted.lst
 # Custom sets, at 24x24 & 16x16 fall-back to industrial preferentially
-CUSTOM_IMAGE_SETS=hicontrast industrial crystal tango classic
+CUSTOM_IMAGE_SETS=hicontrast industrial crystal tango classic oxygen
 CUSTOM_IMAGES+=$(foreach,i,$(CUSTOM_IMAGE_SETS) images_$i)
 CUSTOM_PREFERRED_FALLBACK_1*=-c $(SOLARSRC)$/ooo_custom_images$/tango
 CUSTOM_PREFERRED_FALLBACK_2*=-c $(SOLARSRC)$/ooo_custom_images$/industrial
 
 CRYSTAL_TARBALL=$(SOLARSRC)$/external_images$/ooo_crystal_images-1.tar.gz
 CLASSIC_TARBALL=$(SOLARSRC)$/ooo_custom_images$/classic/classic_images.tar.gz
+OXYGEN_TARBALL=$(SOLARSRC)$/external_images$/ooo_oxygen_images.tar.gz
 
 ALLTAR : $(IMAGES) $(CUSTOM_IMAGES) $(COMMONBIN)$/images_brand.zip
 
@@ -88,6 +89,14 @@ $(MISC)$/crystal.flag : $(CRYSTAL_TARBAL
     chmod -R g+w $(MISC)$/crystal
 .ENDIF
     @$(TYPE) $@ || echo "ERROR: unpacking $(CRYSTAL_TARBALL) failed"
+    
+# unpack the Oxygen icon set
+$(MISC)$/oxygen.flag : $(OXYGEN_TARBALL)
+    cd $(MISC) && gzip -d -c $(OXYGEN_TARBALL) | ( tar -xf - ) && $(TOUCH) $(@:f)
+.IF "$(GUI)"=="UNX"
+    chmod -R g+w $(MISC)$/oxygen
+.ENDIF
+    @$(TYPE) $@ || echo "ERROR: unpacking $(OXYGEN_TARBALL) failed"
 
 # unpack the classic icon set
 $(MISC)$/classic.flag : $(CLASSIC_TARBALL)
@@ -104,3 +113,5 @@ images_crystal : $(MISC)$/crystal.flag $
 
 images_classic : $(MISC)$/classic.flag $(RES)$/img$/commandimagelist.ilst
 
+images_oxygen : $(MISC)$/oxygen.flag $(RES)$/img$/commandimagelist.ilst
+
--- scp2/source/ooo/file_ooo.scp.old	2010-01-21 01:00:35.000000000 +0100
+++ scp2/source/ooo/file_ooo.scp	2010-03-03 16:59:44.000000000 +0100
@@ -2903,6 +2903,13 @@ File gid_File_ImagesCrystal_Zip
     Styles = (PACKED);
 End
 
+File gid_File_ImagesOxygen_Zip
+    TXT_FILE_BODY;
+    Dir = gid_Dir_Share_Config;
+    Name = "images_oxygen.zip";
+    Styles = (PACKED);
+End
+
 File gid_File_ImagesHicontrast_Zip
     TXT_FILE_BODY;
     Dir = gid_Dir_Share_Config;
--- scp2/source/ooo/module_hidden_ooo.scp.old	2010-01-21 01:00:35.000000000 +0100
+++ scp2/source/ooo/module_hidden_ooo.scp	2010-03-03 16:59:44.000000000 +0100
@@ -867,6 +867,7 @@ Module gid_Module_Root_Files_Images
 	Styles = (HIDDEN_ROOT);
 	Files = (gid_File_Images_Zip,
              gid_File_ImagesCrystal_Zip,
+             gid_File_ImagesOxygen_Zip,
              gid_File_ImagesHicontrast_Zip,
              gid_File_ImagesIndustrial_Zip,
              gid_File_ImagesTango_Zip,
--- svtools/inc/imgdef.hxx.old	2009-10-07 15:06:52.000000000 +0200
+++ svtools/inc/imgdef.hxx	2010-03-03 16:59:44.000000000 +0100
@@ -46,7 +46,8 @@ enum SfxSymbolsStyle
     SFX_SYMBOLS_STYLE_INDUSTRIAL,
     SFX_SYMBOLS_STYLE_CRYSTAL,
     SFX_SYMBOLS_STYLE_TANGO,
-    SFX_SYMBOLS_STYLE_CLASSIC
+    SFX_SYMBOLS_STYLE_CLASSIC,
+    SFX_SYMBOLS_STYLE_OXYGEN
 };
 
 #define SFX_TOOLBOX_CHANGESYMBOLSET		0x0001
--- svtools/source/config/miscopt.cxx.old	2009-10-07 15:06:52.000000000 +0200
+++ svtools/source/config/miscopt.cxx	2010-03-03 16:59:44.000000000 +0100
@@ -275,6 +275,7 @@ static sal_Int16 implSymbolsStyleFromVCL
         case STYLE_SYMBOLS_CRYSTAL:    return SFX_SYMBOLS_STYLE_CRYSTAL;
         case STYLE_SYMBOLS_TANGO:      return SFX_SYMBOLS_STYLE_TANGO;
         case STYLE_SYMBOLS_CLASSIC:    return SFX_SYMBOLS_STYLE_CLASSIC;
+        case STYLE_SYMBOLS_OXYGEN:     return SFX_SYMBOLS_STYLE_OXYGEN;
     }
 
     return SFX_SYMBOLS_STYLE_AUTO;
@@ -291,6 +292,7 @@ static ULONG implSymbolsStyleToVCL( sal_
         case SFX_SYMBOLS_STYLE_CRYSTAL:    return STYLE_SYMBOLS_CRYSTAL;
         case SFX_SYMBOLS_STYLE_TANGO:      return STYLE_SYMBOLS_TANGO;
         case SFX_SYMBOLS_STYLE_CLASSIC:    return STYLE_SYMBOLS_CLASSIC;
+        case SFX_SYMBOLS_STYLE_OXYGEN:     return STYLE_SYMBOLS_OXYGEN;
     }
 
     return STYLE_SYMBOLS_AUTO;
--- svx/source/cui/optgdlg.cxx.old	2009-10-08 16:17:55.000000000 +0200
+++ svx/source/cui/optgdlg.cxx	2010-03-03 16:59:44.000000000 +0100
@@ -796,6 +796,7 @@ OfaViewTabPage::OfaViewTabPage(Window* p
             case STYLE_SYMBOLS_TANGO:       aAutoStr += aIconStyleLB.GetEntry( 4 ); break;
             case STYLE_SYMBOLS_CLASSIC:     aAutoStr += aIconStyleLB.GetEntry( 5 ); break;
             case STYLE_SYMBOLS_HICONTRAST:  aAutoStr += aIconStyleLB.GetEntry( 6 ); break;
+            case STYLE_SYMBOLS_OXYGEN:      aAutoStr += aIconStyleLB.GetEntry( 7 ); break;
         }
 
         aIconStyleLB.RemoveEntry( 0 );
@@ -889,6 +890,7 @@ BOOL OfaViewTabPage::FillItemSet( SfxIte
             case 4: eSet = SFX_SYMBOLS_STYLE_CRYSTAL;    break;
             case 5: eSet = SFX_SYMBOLS_STYLE_TANGO;      break;
             case 6: eSet = SFX_SYMBOLS_STYLE_CLASSIC;    break;
+            case 7: eSet = SFX_SYMBOLS_STYLE_OXYGEN;     break;
             default:
                 DBG_ERROR( "OfaViewTabPage::FillItemSet(): This state of aIconStyleLB should not be possible!" );
         }
@@ -1073,6 +1075,7 @@ void OfaViewTabPage::Reset( const SfxIte
             case SFX_SYMBOLS_STYLE_CRYSTAL:    nStyleLB_InitialSelection = 4; break;
             case SFX_SYMBOLS_STYLE_TANGO:      nStyleLB_InitialSelection = 5; break;
             case SFX_SYMBOLS_STYLE_CLASSIC:    nStyleLB_InitialSelection = 6; break;
+            case SFX_SYMBOLS_STYLE_OXYGEN:     nStyleLB_InitialSelection = 7; break;
             default:                           nStyleLB_InitialSelection = 0; break;
         }
     }
--- svx/source/cui/optgdlg.src.old	2009-10-08 16:17:55.000000000 +0200
+++ svx/source/cui/optgdlg.src	2010-03-03 16:59:44.000000000 +0100
@@ -251,6 +251,7 @@ TabPage OFA_TP_VIEW
             < "Crystal" ; > ;
             < "Tango" ; > ;
             < "Classic" ; > ;
+            < "Oxygen" ; > ;
         };
     };
     CheckBox CB_SYSTEM_FONT
--- vcl/inc/vcl/settings.hxx.old	2009-10-08 16:17:47.000000000 +0200
+++ vcl/inc/vcl/settings.hxx	2010-03-03 16:59:44.000000000 +0100
@@ -523,7 +523,8 @@ private:
 #define STYLE_SYMBOLS_CRYSTAL		((ULONG)4)
 #define STYLE_SYMBOLS_TANGO		((ULONG)5)
 #define STYLE_SYMBOLS_CLASSIC		((ULONG)6)
-#define STYLE_SYMBOLS_THEMES_MAX	((ULONG)6)
+#define STYLE_SYMBOLS_OXYGEN		((ULONG)7)
+#define STYLE_SYMBOLS_THEMES_MAX	((ULONG)7)
 
 #define STYLE_CURSOR_NOBLINKTIME    ((ULONG)0xFFFFFFFF)
 
--- vcl/source/app/settings.cxx.old	2010-03-03 16:59:29.000000000 +0100
+++ vcl/source/app/settings.cxx	2010-03-03 16:59:44.000000000 +0100
@@ -707,6 +707,7 @@ void StyleSettings::Set3DColors( const C
         case STYLE_SYMBOLS_CRYSTAL:    return ::rtl::OUString::createFromAscii( "crystal" );
         case STYLE_SYMBOLS_TANGO:      return ::rtl::OUString::createFromAscii( "tango" );
         case STYLE_SYMBOLS_CLASSIC:    return ::rtl::OUString::createFromAscii( "classic" );
+        case STYLE_SYMBOLS_OXYGEN:     return ::rtl::OUString::createFromAscii( "oxygen" );
     }
 
     return ::rtl::OUString::createFromAscii( "auto" );
@@ -728,6 +729,8 @@ ULONG StyleSettings::ImplNameToSymbolsSt
         return STYLE_SYMBOLS_TANGO;
     else if ( rName == ::rtl::OUString::createFromAscii( "classic" ) )
         return STYLE_SYMBOLS_CLASSIC;
+    else if ( rName == ::rtl::OUString::createFromAscii( "oxygen" ) )
+        return STYLE_SYMBOLS_OXYGEN;
 
     return STYLE_SYMBOLS_AUTO;
 }
@@ -823,6 +826,8 @@ ULONG StyleSettings::GetAutoSymbolsStyle
             nRet = STYLE_SYMBOLS_TANGO;
         else if( rDesktopEnvironment.equalsIgnoreAsciiCaseAscii( "kde" ) )
             nRet = STYLE_SYMBOLS_CRYSTAL;
+        else if( rDesktopEnvironment.equalsIgnoreAsciiCaseAscii( "kde4" ) )
+            nRet = STYLE_SYMBOLS_OXYGEN;
     }
 
     return nRet;
