--- l10n/source/makefile.mk.old1	2009-10-19 17:01:03.000000000 +0200
+++ l10n/source/makefile.mk	2009-10-20 16:16:15.000000000 +0200
@@ -67,14 +67,22 @@ TARGET=l10n_merge
 # --- Targets ------------------------------------------------------
 .INCLUDE : settings.mk
 
-.IF "$(WITH_LANG)" == ""
+.IF "$(WITH_LANG)" == "ALL"
+with_lang_list=$(alllangiso)
+.ELSE
+with_lang_list=$(WITH_LANG)
+.ENDIF
+
+
+.IF "$(with_lang_list)" == ""
 
 @all:
     @echo "Nothing to do - en-US only build."
+
 .ELSE
 
 .IF "$(USE_SHELL)"!="4nt"
-all_sdfs:=$(shell cd $(PRJ)$/source && ls -1 *$/localize.sdf)
+all_sdfs:=$(shell ls $(foreach,lang,$(with_lang_list) $(lang)/localize.sdf) 2>/dev/null || true)
 .ELSE          # "$(USE_SHELL)"!="4nt"
 all_sdfs:=$(shell $(CDD) $(PRJ)$/source && find * -name localize.sdf)
 all_sdfs!:=$(subst,/,\ $(all_sdfs))
@@ -84,7 +92,14 @@ all_sdfs!:=$(subst,/,\ $(all_sdfs))
 
 .INCLUDE : target.mk
 
-ALLTAR : $(COMMONMISC)$/merge.done
+
+ALLTAR : check_merged $(COMMONMISC)$/merge.done
+
+check_merged .PHONY :
+    @+echo Checking merged sdf files...
+    @+echo "$(all_sdfs)" >$(COMMONMISC)$/merge.done.check
+    @test -f $(COMMONMISC)$/merge.done && diff -q $(COMMONMISC)$/merge.done $(COMMONMISC)$/merge.done.check || rm -f $(COMMONMISC)$/merge.done
+    @+rm $(COMMONMISC)$/merge.done.check
 
 $(COMMONMISC)$/merge.done : $(all_sdfs)
 .IF "$(L10N_LOCK)" != "YES"
@@ -93,6 +93,6 @@ $(COMMONMISC)$/merge.done : $(all_sdfs)
     -rm -rf $(COMMONMISC)$/sdf$(INPATH)_begone
     -$(MKDIRHIER) $(COMMONMISC)$/sdf
 .ENDIF			# "$(L10n_LOCK)" != "YES"
-    $(PERL) $(SOLARVER)$/$(INPATH)$/bin$(UPDMINOREXT)$/fast_merge.pl -sdf_files $(mktmp $<) -merge_dir $(COMMONMISC)$/sdf && $(TOUCH) $@
+    $(PERL) $(SOLARVER)$/$(INPATH)$/bin$(UPDMINOREXT)$/fast_merge.pl -sdf_files $(mktmp $<) -merge_dir $(COMMONMISC)$/sdf && echo $< >$@
 
 .ENDIF
