From c4837706db377d1d6fc8bfc4a02b8d893a14e33e Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:02:21 +0200
Subject: [PATCH 511/878] layout-crash-fix.diff

---
 toolkit/inc/layout/layout.hxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/toolkit/inc/layout/layout.hxx b/toolkit/inc/layout/layout.hxx
index 78c7ea2..e24e8f8 100644
--- a/toolkit/inc/layout/layout.hxx
+++ b/toolkit/inc/layout/layout.hxx
@@ -110,7 +110,7 @@ public:
         inline t##Impl &getImpl() const
 
 #define DECL_GET_WINDOW( cls ) ::cls* Get##cls() const
-#define IMPL_GET_WINDOW( cls ) ::cls* cls::Get##cls() const { return dynamic_cast< ::cls*>( GetWindow() ); }
+#define IMPL_GET_WINDOW( cls ) ::cls* cls::Get##cls() const { return static_cast< ::cls*>( GetWindow() ); }
 
 #define DECL_GET_VCLXWINDOW( cls ) ::VCLX##cls* GetVCLX##cls() const
 #define IMPL_GET_VCLXWINDOW( cls ) ::VCLX##cls* cls::Get##VCLX##cls() const { return dynamic_cast< ::VCLX##cls*>( GetVCLXWindow() ); }
-- 
1.7.0.1

