From 39f916c184f75c8b502a9ff15a6bc9474665fcd7 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:02:11 +0200
Subject: [PATCH 502/878] layout-default-enable-config_office.diff

---
 configure.in |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.in b/configure.in
index c5e3d72..cdd82bf 100644
--- a/configure.in
+++ b/configure.in
@@ -241,7 +241,7 @@ AC_ARG_ENABLE(static-gtk,
                           (enables -Bstatic linker flag for gtk libraries)
 ],,)
 AC_ARG_ENABLE(layout,
-[  --enable-layout         Enable the compilation and use of layout dialogs
+[  --disable-layout        Disable the compilation and use of layout dialogs
 ],,)
 AC_ARG_ENABLE(build-mozilla,
 [  --disable-build-mozilla Use this option if you do not want to build the
@@ -7073,7 +7073,7 @@ fi
 AC_SUBST(ENABLE_STATIC_GTK)
 
 AC_MSG_CHECKING([whether to use layout dialogs])
-if test -n "$enable_layout" && test "$enable_layout" != "no"; then
+if test -z "$enable_layout" || test "$enable_layout" != "no"; then
    ENABLE_LAYOUT="TRUE"
    AC_MSG_RESULT([yes])
 else
-- 
1.7.0.1

