---
 vcl/source/control/lstbox.cxx |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git vcl/source/control/lstbox.cxx vcl/source/control/lstbox.cxx
index a78ef20..7756d03 100644
--- vcl/source/control/lstbox.cxx
+++ vcl/source/control/lstbox.cxx
@@ -98,6 +98,7 @@ ListBox::~ListBox()
 
 void ListBox::ImplInitListBoxData()
 {
+    mpImplLB        = NULL;
     mpFloatWin		= NULL;
     mpImplWin		= NULL;
     mpBtn			= NULL;
-- 
1.7.0.1

