--- toolkit/source/layout/vcl/wfield.cxx
+++ toolkit/source/layout/vcl/wfield.cxx
@@ -420,6 +420,8 @@ public:
 #ifndef __SUNPRO_CC
     OSL_TRACE ("%s: deleting ComboBox for window: %p", __FUNCTION__, GetWindow ());
 #endif
+    SetClickHdl (Link ());
+    SetSelectHdl (Link ());
 }
 
 ComboBoxImpl::~ComboBoxImpl ()
@@ -630,6 +632,8 @@ public:
 
 ListBox::~ListBox ()
 {
+    SetClickHdl (Link ());
+    SetSelectHdl (Link ());
 }
 
 sal_uInt16 ListBox::InsertEntry (String const& rStr, sal_uInt16 nPos)
diff --git toolkit/source/layout/vcl/wrapper.cxx toolkit/source/layout/vcl/wrapper.cxx
index 31a1fcb..f9c0a4c 100644
--- toolkit/source/layout/vcl/wrapper.cxx
+++ toolkit/source/layout/vcl/wrapper.cxx
@@ -178,7 +178,10 @@ WindowImpl::WindowImpl (Context *context, const PeerHandle &peer, Window *window
 WindowImpl::~WindowImpl ()
 {
     if (mpWindow)
+    {
+        mpWindow->GetWindow ()->SetAccessible (0);
         mpWindow->mpImpl = 0;
+    }
     if (mvclWindow)
     {
         VCLXWindow *v = mvclWindow->GetWindowPeer ();
@@ -674,10 +677,6 @@ ControlImpl::ControlImpl (Context *context, const PeerHandle &peer, Window *wind
 
 ControlImpl::~ControlImpl ()
 {
-    if ((!!mGetFocusHdl || !!mLoseFocusHdl) && mxWindow.is ())
-        /* Disposing will be done @ VCLXWindow::dispose () maFocusListeners.disposeAndClear()
-           don't do it twice */
-        mxWindow.clear ();
 }
 
 void ControlImpl::SetGetFocusHdl (Link const& link)
@@ -701,7 +700,7 @@ void ControlImpl::SetLoseFocusHdl (Link const& link)
 
 Link& ControlImpl::GetLoseFocusHdl ()
 {
-    return mGetFocusHdl;
+    return mLoseFocusHdl;
 }
 
 void ControlImpl::UpdateListening (Link const& link)
@@ -713,10 +712,10 @@ void ControlImpl::UpdateListening (Link const& link)
         mxWindow->addFocusListener (this);
 }
 
-void SAL_CALL ControlImpl::disposing (lang::EventObject const&)
+void SAL_CALL ControlImpl::disposing (lang::EventObject const& e)
     throw (uno::RuntimeException)
 {
-///    mxWindow.clear ();
+    WindowImpl::disposing (e);
 }
 
 void SAL_CALL ControlImpl::focusGained (awt::FocusEvent const&)
