From 4e43ea0214573762789bf1fa4f8ceadf97a6e221 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:02:22 +0200
Subject: [PATCH 512/878] layout-listbox-minsize.diff

---
 vcl/source/control/lstbox.cxx |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/vcl/source/control/lstbox.cxx b/vcl/source/control/lstbox.cxx
index f3199af..39d683d 100644
--- a/vcl/source/control/lstbox.cxx
+++ b/vcl/source/control/lstbox.cxx
@@ -98,6 +98,7 @@ ListBox::~ListBox()
 
 void ListBox::ImplInitListBoxData()
 {
+    mpImplLB        = NULL;
     mpFloatWin		= NULL;
     mpImplWin		= NULL;
     mpBtn			= NULL;
@@ -105,6 +106,7 @@ void ListBox::ImplInitListBoxData()
     mnDDHeight		= 0;
     mbDDAutoSize	= TRUE;
     mnSaveValue 	= LISTBOX_ENTRY_NOTFOUND;
+    mnLineCount     = 0;
 }
 
 // -----------------------------------------------------------------------
-- 
1.7.0.1

