From b05b3fbfc5d34d0ef064b405d0c102cf4a50219d Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:02:25 +0200
Subject: [PATCH 515/878] layout-disabled-layout-build-fix.diff

---
 sfx2/inc/sfx2/layout.hxx |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/sfx2/inc/sfx2/layout.hxx b/sfx2/inc/sfx2/layout.hxx
index d92ed2a..665f639 100644
--- a/sfx2/inc/sfx2/layout.hxx
+++ b/sfx2/inc/sfx2/layout.hxx
@@ -61,6 +61,7 @@ public:
 };
 #endif
 
+#if ENABLE_LAYOUT
 class SFX2_DLLPUBLIC SfxTabPage
     : public ::SfxTabPage
     , public InPlug
@@ -73,6 +74,7 @@ public:
     Size GetOptimalSize (WindowSizeType eType) const;
     void FreeResource ();
 };
+#endif
 } // end namespace layout
 
 #endif /* _SFX2_LAYOUT_HXX */
-- 
1.7.0.1

