--- svx/uiconfig/layout/makefile.mk.old	2010-11-10 20:50:39.000000000 +0100
+++ svx/uiconfig/layout/makefile.mk	2010-11-23 20:47:22.000000000 +0100
@@ -47,9 +47,9 @@ all .PHONY:
 
 .INCLUDE :  target.mk
 
-localize.sdf: $(COMMONMISC)$/$(PRJNAME)$/source$/dialog$/localize.sdf
+localize.sdf: $(COMMONMISC)$/$(PRJNAME)$/source$/dialog$/localize.sdf $(COMMONMISC)$/$(PRJNAME)$/source$/cui$/localize.sdf
     grep srchdlg.src $(COMMONMISC)$/$(PRJNAME)$/source$/dialog$/localize.sdf | awk -F'\t' '{{printf "%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n", "layout", "svx\\uiconfig\\layout\\find-and-replace.xml", $$3, "layout", $$5 $$6 "_label", "", "", $$8, "0", $$10, $$11, $$12, "", $$14, $$15}}' | sed -e 's/\(\(FL\|STR\)_[^\t]*\)_label/\1_text/' -e 's/\tRID_SVXDLG_SEARCH/\t/' -e 's/\t_label/\tRID_SVXDLG_SEARCH_title/' > find-and-replace-$@
-    grep zoom.src $(COMMONMISC)$/$(PRJNAME)$/source$/dialog$/localize.sdf | awk -F'\t' '{{printf "%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n", "layout", "svx\\uiconfig\\layout\\zoom.xml", $$3, "layout", $$5 $$6 "_label", "", "", $$8, "0", $$10, $$11, $$12, "", $$14, $$15}}' | sed -e 's/\(\(FL\|STR\)_[^\t]*\)_label/\1_text/' -e 's/\tRID_SVXDLG_ZOOM/\t/' -e 's/\t_label/\tRID_SVXDLG_ZOOM_title/' > zoom-$@
+    grep zoom.src $(COMMONMISC)$/$(PRJNAME)$/source$/cui$/localize.sdf | awk -F'\t' '{{printf "%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n", "layout", "svx\\uiconfig\\layout\\zoom.xml", $$3, "layout", $$5 $$6 "_label", "", "", $$8, "0", $$10, $$11, $$12, "", $$14, $$15}}' | sed -e 's/\(\(FL\|STR\)_[^\t]*\)_label/\1_text/' -e 's/\tRID_SVXDLG_ZOOM/\t/' -e 's/\t_label/\tRID_SVXDLG_ZOOM_title/' > zoom-$@
     echo '#empty' | cat - find-and-replace-$@ zoom-$@ > $@
     rm -f *-$@
 
@@ -61,10 +61,19 @@ $(COMMONMISC)$/$(PRJNAME)$/source$/dialo
     -unzip -o -d $(COMMONMISC)$/$(PRJNAME) $(SOLARCOMMONSDFDIR)$/$(PRJNAME).zip $(subst,$(COMMONMISC)$/$(PRJNAME)$/, $@)
     $(TOUCH) $@
 
+$(COMMONMISC)$/$(PRJNAME)$/source$/cui$/localize.sdf : $(SOLARCOMMONSDFDIR)$/$(PRJNAME).zip 
+    @@-$(MKDIRHIER) $(@:d) 
+    -unzip -o -d $(COMMONMISC)$/$(PRJNAME) $(SOLARCOMMONSDFDIR)$/$(PRJNAME).zip $(subst,$(COMMONMISC)$/$(PRJNAME)$/, $@) 
+    $(TOUCH) $@ 
+
 .ELSE
 
 $(COMMONMISC)$/$(PRJNAME)$/source$/dialog$/localize.sdf :
     @@-$(MKDIRHIER) $(@:d)
     $(TOUCH) $@
 
+$(COMMONMISC)$/$(PRJNAME)$/source$/cui$/localize.sdf : 
+    @@-$(MKDIRHIER) $(@:d) 
+    $(TOUCH) $@ 
+
 .ENDIF # "$(WITH_LANG)"!=""
