From 4957af0c7d47888ed55e32e43a789d38713ff2b1 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:02:21 +0200
Subject: [PATCH 441/768] layout-crash-fix.diff

---
 toolkit/inc/layout/layout.hxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git toolkit/inc/layout/layout.hxx toolkit/inc/layout/layout.hxx
index d0e0b2e..a9cd3bb 100644
--- toolkit/inc/layout/layout.hxx
+++ toolkit/inc/layout/layout.hxx
@@ -110,7 +110,7 @@ public:
         inline t##Impl &getImpl() const
 
 #define DECL_GET_WINDOW( cls ) ::cls* Get##cls() const
-#define IMPL_GET_WINDOW( cls ) ::cls* cls::Get##cls() const { return dynamic_cast< ::cls*>( GetWindow() ); }
+#define IMPL_GET_WINDOW( cls ) ::cls* cls::Get##cls() const { return static_cast< ::cls*>( GetWindow() ); }
 
 #define DECL_GET_VCLXWINDOW( cls ) ::VCLX##cls* GetVCLX##cls() const
 #define IMPL_GET_VCLXWINDOW( cls ) ::VCLX##cls* cls::Get##VCLX##cls() const { return dynamic_cast< ::VCLX##cls*>( GetVCLXWindow() ); }
-- 
1.7.0.1

