---
 configure.in |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git configure.in configure.in
index 1cc6fbd..6017fc9 100644
--- configure.in
+++ configure.in
@@ -235,7 +235,7 @@ AC_ARG_ENABLE(static-gtk,
                           (enables -Bstatic linker flag for gtk libraries)
 ],,)
 AC_ARG_ENABLE(layout,
-[  --enable-layout         Enable the compilation and use of layout dialogs
+[  --disable-layout        Disable the compilation and use of layout dialogs
 ],,)
 AC_ARG_ENABLE(build-mozilla,
 [  --disable-build-mozilla Use this option if you do not want to build the
@@ -7134,7 +7134,7 @@ fi
 AC_SUBST(ENABLE_STATIC_GTK)
 
 AC_MSG_CHECKING([whether to use layout dialogs])
-if test -n "$enable_layout" && test "$enable_layout" != "no"; then
+if test -z "$enable_layout" || test "$enable_layout" != "no"; then
    ENABLE_LAYOUT="TRUE"
    AC_MSG_RESULT([yes])
 else
-- 
1.7.0.1

