--- configure.in
+++ configure.in
@@ -248,6 +248,10 @@ AC_ARG_ENABLE(static-gtk,
 AC_ARG_ENABLE(layout,
 [  --disable-layout        Disable the compilation and use of layout dialogs
 ],,)
+AC_ARG_ENABLE(layout_experimental,
+[  --enable-layout-experimental  Enable the compilation and use of layout dialogs
+                               tagged experimental.
+],,)
 AC_ARG_ENABLE(build-mozilla,
 [  --disable-build-mozilla Use this option if you do not want to build the
                           mozilla components from the mozilla source code but
@@ -6650,6 +6654,16 @@ else
 fi
 AC_SUBST(ENABLE_LAYOUT)
 
+AC_MSG_CHECKING([whether to use experimental layout dialogs])
+if test -n "$enable_layout_experimental" && test "$enable_layout_experimental" == "yes"; then
+   ENABLE_LAYOUT_EXPERIMENTAL="TRUE"
+   AC_MSG_RESULT([yes])
+else
+   ENABLE_LAYOUT_EXPERIMENTAL="FALSE"
+   AC_MSG_RESULT([no])
+fi
+AC_SUBST(ENABLE_LAYOUT_EXPERIMENTAL)
+
 # ===================================================================
 # De- or increase default verbosity of build process
 # ===================================================================
--- set_soenv.in
+++ set_soenv.in
@@ -1906,6 +1906,7 @@
 ToFile( "ENABLE_FONTCONFIG", "@ENABLE_FONTCONFIG@", "e" );
 ToFile( "ENABLE_DIRECTX",    "@ENABLE_DIRECTX@",    "e" );
 ToFile( "ENABLE_LAYOUT",     "@ENABLE_LAYOUT@",     "e" );
+ToFile( "ENABLE_LAYOUT_EXPERIMENTAL", "@ENABLE_LAYOUT_EXPERIMENTAL@", "e" );
 ToFile( "ENABLE_PCH",        "@ENABLE_PCH@",       "e" );
 ToFile( "NO_HIDS",           "@NO_HIDS@",       "e" );
 ToFile( "VC_STANDARD",       "@VC_STANDARD@",      "e" );
