From 3abe29f0e523aae4668fe933bf86f4a679e9ea79 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:02:09 +0200
Subject: [PATCH 500/878] layout-localize-fix-sc.diff

---
 sc/uiconfig/layout/makefile.mk |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/sc/uiconfig/layout/makefile.mk b/sc/uiconfig/layout/makefile.mk
index c12e8f5..e340ae6 100644
--- a/sc/uiconfig/layout/makefile.mk
+++ b/sc/uiconfig/layout/makefile.mk
@@ -57,3 +57,9 @@ localize.sdf: $(PRJ)/source/ui/miscdlgs/localize.sdf $(PRJ)/source/ui/src/locali
     grep sortdlg.src $(PRJ)/source/ui/src/localize.sdf | awk -F'\t' '{{printf "%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n", "layout", "sc\\uiconfig\\layout\\sort-options.xml", $$3, "layout", $$5 $$6 "_label", "", "", $$8, "0", $$10, $$11, $$12, "", $$14, $$15}}' | sed -e 's/\(\(FL\|STR\)_[^\t]*\)_label/\1_text/' -e 's/\tRID_SCDLG_SORT/\t/' -e 's/\t_label/\tRID_SCDLG_SORT_title/' -e 's/\tRID_SCPAGE_SORT_OPTIONS/\t/' -e 's/\tRID_SCPAGE_SORT_FIELDS/\t/' > sort-options-$@
     echo '#empty' | cat - insert-sheet-$@ move-copy-sheet-$@ sort-options-$@ > $@
     rm -f *-$@
+
+$(PRJ)/source/ui/miscdlgs/localize.sdf:
+    touch $@
+
+$(PRJ)/source/ui/src/localize.sdf:
+    touch $@
-- 
1.7.0.1

