diff --git toolkit/source/vclcompat/wfield.cxx toolkit/source/vclcompat/wfield.cxx
index 8bd4be4..93130d6 100644
--- toolkit/source/vclcompat/wfield.cxx
+++ toolkit/source/vclcompat/wfield.cxx
@@ -326,7 +326,8 @@ public:
     {
         uno::Sequence< rtl::OUString> aItems( mxComboBox->getItems() );
         rtl::OUString rKey( rStr );
-        for( unsigned int i = 0; aItems.getLength(); i++ )
+        sal_Int32 n = aItems.getLength();
+        for( unsigned int i = 0; i < n; i++ )
         {
             if ( aItems[ i ] == rKey )
                 return sal::static_int_cast< sal_uInt16 >( i );
@@ -477,7 +478,8 @@ public:
     {
         uno::Sequence< rtl::OUString> aItems( mxListBox->getItems() );
         rtl::OUString rKey( rStr );
-        for( unsigned int i = 0; aItems.getLength(); i++ )
+        sal_Int32 n = aItems.getLength();
+        for( unsigned int i = 0; i < n; i++ )
         {
             if ( aItems[ i ] == rKey )
                 return sal::static_int_cast< sal_uInt16 >( i );
