---
 sfx2/inc/sfx2/layout.hxx |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git sfx2/inc/sfx2/layout.hxx sfx2/inc/sfx2/layout.hxx
index 836a762..b4e145c 100644
--- sfx2/inc/sfx2/layout.hxx
+++ sfx2/inc/sfx2/layout.hxx
@@ -49,6 +49,7 @@
     SfxDialog (::Window* parent, char const* xml_file, char const* id, SfxBindings* bindings=0, SfxChildWindow* child=0);
 };

+#if ENABLE_LAYOUT
 class SFX2_DLLPUBLIC SfxTabPage
     : public ::SfxTabPage
     , public InPlug
@@ -61,6 +62,7 @@
     Size GetOptimalSize (WindowSizeType eType) const;
     void FreeResource ();
 };
+#endif
 } // end namespace layout

 #endif /* _SFX2_LAYOUT_HXX */
-- 
1.7.0.1

