From d0e4070d187e8282b12d35d964843b24a4ec5abd Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:02:19 +0200
Subject: [PATCH 509/878] layout-parallel-build.diff

---
 solenv/inc/layout.mk |   23 ++++++++++-------------
 1 files changed, 10 insertions(+), 13 deletions(-)

diff --git a/solenv/inc/layout.mk b/solenv/inc/layout.mk
index 97bdc9e..227da15 100644
--- a/solenv/inc/layout.mk
+++ b/solenv/inc/layout.mk
@@ -1,32 +1,29 @@
-TRALAY=$(AUGMENT_LIBRARY_PATH) tralay
+TRALAY=$(AUGMENT_LIBRARY_PATH) $(SOLARBINDIR)$/tralay
 XML_DEST=$(DLLDEST)
 XML_LANGS=$(alllangiso)
 
-ALL_XMLS=$(foreach,i,$(XML_FILES) "$(XML_DEST)/$i") $(foreach,i,$(XML_LANGS) $(foreach,j,$(XML_FILES) "$(XML_DEST)/$i/$j"))
+ALL_XMLS=$(foreach,i,$(XML_FILES) $(XML_DEST)$/$i) $(foreach,i,$(XML_LANGS) $(foreach,j,$(XML_FILES) $(XML_DEST)$/$i$/$j))
+XML_DEPS=$(foreach,i,$(XML_FILES) $(MISC)$/$i.dep)
 
 # Must remove the -j (no duplicate base file names) flag
 ZIPUPDATE=-u
 XML_ZIP = $(PRJNAME)-layout
 
-ALLTAR: localize.sdf $(XML_ZIP)
+ALLTAR .SEQUENTIAL: localize.sdf ALL_XML_DEPS
 
-$(XML_ZIP): $(ALL_XMLS)
+ALL_XML_DEPS: $(XML_DEPS)
 
 ZIP1DIR=$(XML_DEST)
 ZIP1TARGET=$(XML_ZIP)
 ZIP1LIST=$(ALL_XMLS:s@$(XML_DEST)/@@)
+ZIP1DEPS=localize.sdf $(XML_DEPS)
 
-$(foreach,i,$(XML_LANGS) $(XML_DEST)/$i/%.xml): %.xml
-    -$(MKDIR) $(@:d)
-    @echo $(foreach,i,$(XML_LANGS) $(XML_DEST)/$i/%.xml): %.xml
+$(MISC)$/%.xml.dep: %.xml
+    -$(MKDIR) "$(XML_DEST)"
+    cp "$<" "$(XML_DEST)"
     $(TRALAY) -m localize.sdf -o "$(XML_DEST)" -l $(XML_LANGS:f:t" -l ") "$<"
-
-$(XML_DEST)/%.xml: %.xml
-    -$(MKDIR) $(@:d)
-    $(COPY) "$<" "$@"
+    $(TOUCH) "$@"
 
 # Don't want to overwrite filled localize.sdf with empty template
 template.sdf:
     $(foreach,i,$(XML_FILES) $(TRALAY) -l en-US "$i") > $@
-
-$(ZIP1LIST): localize.sdf
-- 
1.7.0.1

