--- sfx2/inc/sfx2/layout.hxx.old	2009-12-20 13:35:29.000000000 +0000
+++ sfx2/inc/sfx2/layout.hxx	2009-12-20 13:02:22.000000000 +0000
@@ -63,6 +63,7 @@ public:
 };
 #endif
 
+#if ENABLE_LAYOUT
 class SFX2_DLLPUBLIC SfxTabPage
     : public ::SfxTabPage
     , public InPlug
@@ -75,6 +76,7 @@ public:
     Size GetOptimalSize (WindowSizeType eType) const;
     void FreeResource ();
 };
+#endif
 } // end namespace layout
 
 #endif /* _SFX2_LAYOUT_HXX */
