From 74148cfdfcecd4e8bbf4da4875b24a3f8587cf78 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:02:25 +0200
Subject: [PATCH 445/768] layout-disabled-layout-build-fix.diff

---
 sfx2/inc/sfx2/layout.hxx |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git sfx2/inc/sfx2/layout.hxx sfx2/inc/sfx2/layout.hxx
index 836a762..b4e145c 100644
--- sfx2/inc/sfx2/layout.hxx
+++ sfx2/inc/sfx2/layout.hxx
@@ -61,6 +61,7 @@ public:
 };
 #endif
 
+#if ENABLE_LAYOUT
 class SFX2_DLLPUBLIC SfxTabPage
     : public ::SfxTabPage
     , public InPlug
@@ -73,6 +74,7 @@ public:
     Size GetOptimalSize (WindowSizeType eType) const;
     void FreeResource ();
 };
+#endif
 } // end namespace layout
 
 #endif /* _SFX2_LAYOUT_HXX */
-- 
1.7.0.1

