From 497d56bbeb0e35e2d65cbdbf8543d8728548cf88 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:02:12 +0200
Subject: [PATCH 503/878] layout-disable-experimental.diff

---
 configure.in |   14 ++++++++++++++
 set_soenv.in |    1 +
 2 files changed, 15 insertions(+), 0 deletions(-)

diff --git a/configure.in b/configure.in
index cdd82bf..624b30a 100644
--- a/configure.in
+++ b/configure.in
@@ -243,6 +243,10 @@ AC_ARG_ENABLE(static-gtk,
 AC_ARG_ENABLE(layout,
 [  --disable-layout        Disable the compilation and use of layout dialogs
 ],,)
+AC_ARG_ENABLE(layout_experimental,
+[  --enable-layout-experimental  Enable the compilation and use of layout dialogs
+                               tagged experimental.
+],,)
 AC_ARG_ENABLE(build-mozilla,
 [  --disable-build-mozilla Use this option if you do not want to build the
                           mozilla components from the mozilla source code but
@@ -7082,6 +7086,16 @@ else
 fi
 AC_SUBST(ENABLE_LAYOUT)
 
+AC_MSG_CHECKING([whether to use experimental layout dialogs])
+if test -n "$enable_layout_experimental" && test "$enable_layout_experimental" == "yes"; then
+   ENABLE_LAYOUT_EXPERIMENTAL="TRUE"
+   AC_MSG_RESULT([yes])
+else
+   ENABLE_LAYOUT_EXPERIMENTAL="FALSE"
+   AC_MSG_RESULT([no])
+fi
+AC_SUBST(ENABLE_LAYOUT_EXPERIMENTAL)
+
 # ===================================================================
 # De- or increase default verbosity of build process
 # ===================================================================
diff --git a/set_soenv.in b/set_soenv.in
index c760fbe..cda1217 100644
--- a/set_soenv.in
+++ b/set_soenv.in
@@ -1830,6 +1830,7 @@ ToFile( "ENABLE_RANDR",      "@ENABLE_RANDR@",      "e" );
 ToFile( "ENABLE_FONTCONFIG", "@ENABLE_FONTCONFIG@", "e" );
 ToFile( "ENABLE_DIRECTX",    "@ENABLE_DIRECTX@",    "e" );
 ToFile( "ENABLE_LAYOUT",     "@ENABLE_LAYOUT@",     "e" );
+ToFile( "ENABLE_LAYOUT_EXPERIMENTAL", "@ENABLE_LAYOUT_EXPERIMENTAL@", "e" );
 ToFile( "ENABLE_PCH",        "@ENABLE_PCH@",       "e" );
 ToFile( "NO_HIDS",           "@NO_HIDS@",       "e" );
 ToFile( "ENABLE_GRAPHITE",   "@ENABLE_GRAPHITE@",   "e");
-- 
1.7.0.1

