From 20110da47287a4447e5cb5337a86ac6fc2c7ca7b Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:53:54 +0200
Subject: [PATCH 061/768] libwpd.diff

---
 libwpd/makefile.mk |    5 ++---
 1 files changed, 2 insertions(+), 3 deletions(-)

diff --git libwpd/makefile.mk libwpd/makefile.mk
index abca6ff..9beba21 100644
--- libwpd/makefile.mk
+++ libwpd/makefile.mk
@@ -41,9 +41,8 @@ TARGET=wpd
     @echo "Using system libwpd..."
 .ENDIF
 
-TARFILE_NAME=libwpd-0.8.8
-TARFILE_MD5=cd5997284f4ba1e8dde5d1e5869fc342
-PATCH_FILES=$(TARFILE_NAME).diff
+TARFILE_NAME=libwpd-0.8.14
+#PATCH_FILES=$(TARFILE_NAME).diff
 BUILD_ACTION=dmake $(MFLAGS) $(CALLMACROS)
 BUILD_DIR=src$/lib
 
-- 
1.7.0.1

