---
 libwpd/makefile.mk |    5 ++---
 1 files changed, 2 insertions(+), 3 deletions(-)

diff --git libwpd/makefile.mk libwpd/makefile.mk
index abca6ff..104d79f 100644
--- libwpd/makefile.mk
+++ libwpd/makefile.mk
@@ -41,9 +41,8 @@ TARGET=wpd
     @echo "Using system libwpd..."
 .ENDIF
 
-TARFILE_NAME=libwpd-0.8.8
-TARFILE_MD5=cd5997284f4ba1e8dde5d1e5869fc342
-PATCH_FILES=$(TARFILE_NAME).diff
+TARFILE_NAME=libwpd-0.8.14
+TARFILE_MD5=64d66018897d759358f454010b6e75d2
 BUILD_ACTION=dmake $(MFLAGS) $(CALLMACROS)
 BUILD_DIR=src$/lib
 
-- 
1.7.0.1

