From 46960f97a2bddc942c55f6162589bd72096a4ebd Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:53:54 +0200
Subject: [PATCH 077/878] libwpd.diff

---
 libwpd/makefile.mk |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/libwpd/makefile.mk b/libwpd/makefile.mk
index 6fa4cb1..3d58c51 100644
--- a/libwpd/makefile.mk
+++ b/libwpd/makefile.mk
@@ -41,8 +41,8 @@ TARGET=wpd
     @echo "Using system libwpd..."
 .ENDIF
 
-TARFILE_NAME=libwpd-0.8.8
-PATCH_FILES=$(TARFILE_NAME).diff
+TARFILE_NAME=libwpd-0.8.14
+#PATCH_FILES=$(TARFILE_NAME).diff
 BUILD_ACTION=dmake $(MFLAGS) $(CALLMACROS)
 BUILD_DIR=src$/lib
 
-- 
1.7.0.1

