--- configure.in.old	2010-07-29 16:48:44.000000000 +0200
+++ configure.in	2010-07-29 16:50:30.000000000 +0200
@@ -1650,6 +1650,26 @@ AC_SUBST(HAVE_LD_HASH_STYLE)
 AC_SUBST(WITH_LINKER_HASH_STYLE)
 
 dnl ===================================================================
+dnl  Check for -Wl,--as-needed linker support
+dnl ===================================================================
+HAVE_WORKING_LD_AS_NEEDED=
+if test \( "$_os" != "WINNT" -o "$WITH_MINGWIN" = "yes" \) -a "$GCC" = "yes"; then
+   AC_MSG_CHECKING([for working -Wl,--as-needed linker support])
+   _ld_version=`$CC -Wl,--version 2>&1 | head -n 1 | $SED -e "s|(.*)||" -e "s|^[[^0-9]]*||"`
+   _ld_major=`echo $_ld_version | cut -d"." -f 1`
+   _ld_minor=`echo $_ld_version | cut -d"." -f 2`
+   if test "$_ld_major" -gt 2 || \
+      test "$_ld_major" -eq 2 -a "$_ld_minor" -gt 17 ; then
+      # there are problems with ld <= 2.17, see also http://www.gentoo.org/proj/en/qa/asneeded.xml
+      HAVE_WORKING_LD_AS_NEEDED=TRUE
+      AC_MSG_RESULT([yes])
+   else
+      AC_MSG_RESULT([no])
+   fi
+fi
+AC_SUBST(HAVE_WORKING_LD_AS_NEEDED)
+
+dnl ===================================================================
 dnl  Test the Compaq compiler for OSF1
 dnl ===================================================================
 if test "$_os" = "OSF1"; then
--- cui/util/makefile.mk.old	2010-07-22 13:28:05.000000000 +0200
+++ cui/util/makefile.mk	2010-07-29 16:50:30.000000000 +0200
@@ -73,7 +73,6 @@ SHL1STDLIBS= \
             $(I18NISOLANGLIB) \
             $(COMPHELPERLIB) \
             $(UCBHELPERLIB) \
-            $(CPPUHELPERLIB)        \
             $(CPPULIB) \
             $(SALLIB) \
             $(JVMFWKLIB) \
@@ -82,8 +81,9 @@ SHL1STDLIBS= \
 
 .IF "$(GUI)"=="WNT"
 SHL1STDLIBS+= \
-             $(SHLWAPILIB) \
-             $(ADVAPI32LIB)
+            $(SHLWAPILIB) \
+            $(CPPUHELPERLIB)        \
+            $(ADVAPI32LIB)
 .ENDIF # WNT
 
 # ------------------------------------------------------------------------------
--- set_soenv.in.old	2010-07-29 16:48:44.000000000 +0200
+++ set_soenv.in	2010-07-29 16:50:30.000000000 +0200
@@ -1779,6 +1779,8 @@ ToFile( "HAVE_LD_HASH_STYLE","@HAVE_LD_H
 ToFile( "WITH_LINKER_HASH_STYLE","@WITH_LINKER_HASH_STYLE@","e" );
 ToFile( "HAVE_LD_BSYMBOLIC_FUNCTIONS",
 		"@HAVE_LD_BSYMBOLIC_FUNCTIONS@","e" );
+ToFile( "HAVE_WORKING_LD_AS_NEEDED",
+        "@HAVE_WORKING_LD_AS_NEEDED@","e" );
 ToFile( "CXX",               $CXX,               "e" );
 ToFile( "USE_CCACHE",        "@USE_CCACHE@",     "e" );
 ToFile( "MINGWCXX",          "@MINGWCXX@",       "e" );
--- solenv/inc/unxbsdi.mk.old	2010-07-23 17:00:36.000000000 +0200
+++ solenv/inc/unxbsdi.mk	2010-07-29 16:50:30.000000000 +0200
@@ -104,6 +104,9 @@ DYNAMIC		= -Wl,-Bdynamic
 LINK*=gcc
 # default linker flags
 LINKFLAGS=
+.IF "$(HAVE_WORKING_LD_AS_NEEDED)" == "TRUE"
+LINKFLAGS += -Wl,--as-needed
+.ENDIF
 
 # linker flags for linking applications
 LINKFLAGSAPPGUI= -Wl,-export-dynamic 
--- solenv/inc/unxbsdi2.mk.old	2010-07-23 17:00:36.000000000 +0200
+++ solenv/inc/unxbsdi2.mk	2010-07-29 16:50:30.000000000 +0200
@@ -124,6 +124,9 @@ LINKFLAGSRUNPATH_BRAND=-Wl,-rpath,\''$$O
 LINKFLAGSRUNPATH_OXT=
 LINKFLAGSRUNPATH_NONE=
 LINKFLAGS=-z combreloc $(LINKFLAGSDEFS)
+.IF "$(HAVE_WORKING_LD_AS_NEEDED)" == "TRUE"
+LINKFLAGS += -Wl,--as-needed
+.ENDIF
 
 # linker flags for linking applications
 LINKFLAGSAPPGUI= -Wl,-export-dynamic 
--- solenv/inc/unxfbsd.mk.old	2010-07-23 17:00:36.000000000 +0200
+++ solenv/inc/unxfbsd.mk	2010-07-29 16:50:30.000000000 +0200
@@ -146,6 +146,9 @@ LINKFLAGSRUNPATH_BRAND=-Wl,-rpath,\''$$O
 LINKFLAGSRUNPATH_OXT=
 LINKFLAGSRUNPATH_NONE=
 LINKFLAGS=-Wl,-z,combreloc $(LINKFLAGSDEFS)
+.IF "$(HAVE_WORKING_LD_AS_NEEDED)" == "TRUE"
+LINKFLAGS += -Wl,--as-needed
+.ENDIF
 
 # linker flags for linking applications
 LINKFLAGSAPPGUI= -Wl,-export-dynamic -Wl,--noinhibit-exec
--- solenv/inc/unxgcc.mk
+++ solenv/inc/unxgcc.mk
@@ -158,6 +158,9 @@ LINKFLAGS=-Wl,-z,noexecstack -Wl,-z,comb
 .IF "$(HAVE_LD_BSYMBOLIC_FUNCTIONS)"  == "TRUE"
 LINKFLAGS += -Wl,-Bsymbolic-functions -Wl,--dynamic-list-cpp-new -Wl,--dynamic-list-cpp-typeinfo
 .ENDIF
+.IF "$(HAVE_WORKING_LD_AS_NEEDED)" == "TRUE"
+LINKFLAGS += -Wl,--as-needed
+.ENDIF
 
 # linker flags for linking applications
 LINKFLAGSAPPGUI= -Wl,-export-dynamic -Wl,--noinhibit-exec \
--- solenv/inc/unxmacx.mk.old	2010-07-23 17:00:36.000000000 +0200
+++ solenv/inc/unxmacx.mk	2010-07-29 16:50:30.000000000 +0200
@@ -210,6 +210,10 @@ LINKFLAGSRUNPATH_BOXT=
 LINKFLAGSRUNPATH_NONE=-install_name '@__________________________________________________NONE/$(@:f)'
 LINKFLAGS=$(LINKFLAGSDEFS)
 
+.IF "$(HAVE_WORKING_LD_AS_NEEDED)" == "TRUE"
+LINKFLAGS += -Wl,--as-needed
+.ENDIF
+
 # [ed] 5/14/02 If we're building for aqua, add in the objc runtime library into our link line
 .IF "$(GUIBASE)" == "aqua"
     LINKFLAGS+=-lobjc
--- solenv/inc/unxsogi.mk.old	2010-07-23 17:00:36.000000000 +0200
+++ solenv/inc/unxsogi.mk	2010-07-29 16:50:30.000000000 +0200
@@ -78,6 +78,10 @@ LINK*=$(CXX)
 LINKC*=$(CC)
 
 LINKFLAGS=
+.IF "$(HAVE_WORKING_LD_AS_NEEDED)" == "TRUE"
+LINKFLAGS += -Wl,--as-needed
+.ENDIF
+
 LINKFLAGSAPPGUI=-Wl,-export-dynamic
 LINKFLAGSSHLGUI=-shared
 LINKFLAGSAPPCUI=-Wl,-export-dynamic
--- solenv/inc/unxsoli4.mk.old	2010-07-23 17:00:36.000000000 +0200
+++ solenv/inc/unxsoli4.mk	2010-07-29 16:50:30.000000000 +0200
@@ -140,6 +140,9 @@ LINKFLAGSRUNPATH_OXT=
 LINKFLAGSRUNPATH_BOXT=-R\''$$ORIGIN/../../../basis-link/program'\'
 LINKFLAGSRUNPATH_NONE=
 LINKFLAGS=-w -mt -z combreloc -PIC -temp=/tmp -norunpath -library=no%Cstd
+.IF "$(HAVE_WORKING_LD_AS_NEEDED)" == "TRUE"
+LINKFLAGS += -Wl,--as-needed
+.ENDIF
 LINKCFLAGS=-w -mt -z combreloc -norunpath
 
 # -z text force fatal error if non PIC code is linked into shared library. Such code
--- solenv/inc/unxsols4.mk.old	2010-07-23 17:00:36.000000000 +0200
+++ solenv/inc/unxsols4.mk	2010-07-29 16:50:30.000000000 +0200
@@ -146,6 +146,9 @@ LINKFLAGSRUNPATH_OXT=
 LINKFLAGSRUNPATH_BOXT=-R\''$$ORIGIN/../../../basis-link/program'\'
 LINKFLAGSRUNPATH_NONE=
 LINKFLAGS=-w -mt -z combreloc -PIC -temp=/tmp -norunpath -library=no%Cstd
+.IF "$(HAVE_WORKING_LD_AS_NEEDED)" == "TRUE"
+LINKFLAGS += -Wl,--as-needed
+.ENDIF
 LINKCFLAGS=-w -mt -z combreloc -norunpath
 
 # -z text force fatal error if non PIC code is linked into shared library. Such code
--- solenv/inc/unxsolu4.mk.old	2010-07-23 17:00:36.000000000 +0200
+++ solenv/inc/unxsolu4.mk	2010-07-29 16:50:30.000000000 +0200
@@ -141,6 +141,9 @@ LINKFLAGSRUNPATH_BOXT=-R\''$$ORIGIN/../.
 LINKFLAGSRUNPATH_NONE=
 LINKFLAGS=-m64 -w -mt -z combreloc -PIC -temp=/tmp -norunpath -library=stlport4
 LINKCFLAGS=-m64 -w -mt -z combreloc -norunpath
+.IF "$(HAVE_WORKING_LD_AS_NEEDED)" == "TRUE"
+LINKFLAGS += -Wl,--as-needed
+.ENDIF
 
 # -z text force fatal error if non PIC code is linked into shared library. Such code
 #    would be expensive on startup
--- unodevtools/source/skeletonmaker/makefile.mk.old	2010-06-21 14:49:59.000000000 +0200
+++ unodevtools/source/skeletonmaker/makefile.mk	2010-07-29 16:50:30.000000000 +0200
@@ -49,8 +49,9 @@ APP1OBJS = $(OBJ)$/skeletonmaker.obj \
 
 APP1DEPN= $(OUT)$/lib$/$(UNODEVTOOLSLIBDEPN) $(SOLARLIBDIR)$/$(CODEMAKERLIBDEPN) \
     $(SOLARLIBDIR)$/$(COMMONCPPLIBDEPN) $(SOLARLIBDIR)$/$(COMMONJAVALIBDEPN)
-APP1STDLIBS = $(REGLIB) $(SALLIB) $(SALHELPERLIB) $(CPPULIB) $(CPPUHELPERLIB) \
-    $(UNODEVTOOLSLIBST) $(CODEMAKERLIBST) $(COMMONCPPLIBST) $(COMMONJAVALIBST)
+APP1STDLIBS =\
+    $(UNODEVTOOLSLIBST) $(CODEMAKERLIBST) $(COMMONCPPLIBST) $(COMMONJAVALIBST)\
+    $(REGLIB) $(CPPULIB) $(CPPUHELPERLIB) $(SALHELPERLIB) $(SALLIB)
 
 OBJFILES = $(APP1OBJS)
 
