From de137f2ebfbd7530216af20f4f72447dda4f6c6a Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:53:09 +0200
Subject: [PATCH 038/878] linkoo-type.diff

---
 solenv/bin/linkoo |    9 +++++++++
 1 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/solenv/bin/linkoo b/solenv/bin/linkoo
index 1bc06ed..b6f62e3 100755
--- a/solenv/bin/linkoo
+++ b/solenv/bin/linkoo
@@ -272,6 +272,14 @@ sub link_iso_res()
 }
 
 # Hack for (renamed) types.rdb (types.db)
+sub link_oovbaapi_rdb()
+{
+    print "oovbaapi.rdb case:";
+    my $src  = "$OOO_BUILD/oovbaapi/$TARGET/ucr";
+    my $dest = "$OOO_INSTALL/program";
+    do_link ($src, $dest, 'oovbaapi.db', 'oovbaapi.rdb');
+    print "\n";
+}
 sub link_types_rdb()
 {
     print "Types.rdb case:";
@@ -349,6 +357,7 @@ my $installed_files = build_installed_list ($OOO_INSTALL);
 scan_and_link_files ($OOO_BUILD, $installed_files, $TARGET);
 link_iso_res();
 link_types_rdb();
+link_oovbaapi_rdb();
 link_soffice_bin_files();
 
 if (!-f "$OOO_INSTALL/" . $program_dir . "/ooenv") {
-- 
1.7.0.1

