--- officecfg/registry/schema/org/openoffice/Office/Common.xcs
+++ officecfg/registry/schema/org/openoffice/Office/Common.xcs
@@ -5590,6 +5590,15 @@ Dymamic border coloring means that when the mouse is hovered over a control, and
 				</info>
 				<value>true</value>
 			</prop>
+            <prop oor:name="ShowLinkWarningDialog" oor:type="xs:boolean">
+                <info>
+                    <author>THB</author>
+                    <desc>Determines if a warning dialog should be
+                    displayed when a user checks the link checkbox in
+                    the insert graphics dialog.</desc>
+                </info>
+                <value>true</value>
+            </prop>
 			<prop oor:name="UseSystemPrintDialog" oor:type="xs:boolean">
 				<info>
 					<author>PL</author>
