From af6f6a6b01150263e8455b73c2ac0390138ed5e9 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:04:18 +0200
Subject: [PATCH 541/768] linkwarn-sfx2-disable-cb-persistency.diff

---
 sfx2/source/dialog/filedlghelper.cxx |   12 ------------
 1 files changed, 0 insertions(+), 12 deletions(-)

diff --git sfx2/source/dialog/filedlghelper.cxx sfx2/source/dialog/filedlghelper.cxx
index 5ca70ef..f6ba2b4 100644
--- sfx2/source/dialog/filedlghelper.cxx
+++ sfx2/source/dialog/filedlghelper.cxx
@@ -2012,11 +2012,7 @@ void FileDialogHelper_Impl::saveConfig()
 
         try
         {
-            aValue = xDlg->getValue( ExtendedFilePickerElementIds::CHECKBOX_LINK, 0 );
             sal_Bool bValue = sal_False;
-            aValue >>= bValue;
-            aUserData.SetToken( 0, ' ', String::CreateFromInt32( (sal_Int32) bValue ) );
-
             aValue = xDlg->getValue( ExtendedFilePickerElementIds::CHECKBOX_PREVIEW, 0 );
             bValue = sal_False;
             aValue >>= bValue;
@@ -2141,14 +2137,6 @@ void FileDialogHelper_Impl::loadConfig()
         {
             try
             {
-                // respect the last "insert as link" state
-                sal_Bool bLink = (sal_Bool) aUserData.GetToken( 0, ' ' ).ToInt32();
-                if ( !xDlg->getValue( ExtendedFilePickerElementIds::CHECKBOX_LINK, 0 ).hasValue() )
-                {
-                    aValue <<= bLink;
-                    xDlg->setValue( ExtendedFilePickerElementIds::CHECKBOX_LINK, 0, aValue );
-                }
-
                 // respect the last "show preview" state
                 sal_Bool bShowPreview = (sal_Bool) aUserData.GetToken( 1, ' ' ).ToInt32();
                 if  ( !xDlg->getValue( ExtendedFilePickerElementIds::CHECKBOX_PREVIEW, 0 ).hasValue() )
-- 
1.7.0.1

