--- transex3/scripts/localize.pl.old	2010-01-29 18:34:30.000000000 +0100
+++ transex3/scripts/localize.pl	2010-01-30 14:02:06.558804800 +0100
@@ -401,12 +401,7 @@
     my $command = '';
     my ( $TMPHANDLE , $tmpfile ) = File::Temp::tempfile();
     close ( $TMPHANDLE );
-
-    if( $ENV{WRAPCMD} ){
-        $command = "$ENV{WRAPCMD} gsicheck";
-    }else{
-        $command = "gsicheck";
-    }
+    $command = "$ENV{SOLARVER}/$ENV{INPATH}/bin/gsicheck";
     my $errfile = $sdffile.".err";
     $command .= " -k -c -wcf $tmpfile -wef ".fix_cygwin_path($errfile)." -l \"\" ".fix_cygwin_path($sdffile);
     #my $rc = system( $command );
