From d05e2f1f96af09469ca2a1b0d9b1c9b62abcf436 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:54:39 +0200
Subject: [PATCH 117/878] localize-pl-cygwin-paths.diff

---
 transex3/scripts/localize.pl |   20 +++++++++++++++++---
 1 files changed, 17 insertions(+), 3 deletions(-)

diff --git a/transex3/scripts/localize.pl b/transex3/scripts/localize.pl
index 122e443..66e0e06 100755
--- l10ntools/scripts/localize.pl
+++ l10ntools/scripts/localize.pl
@@ -407,8 +407,9 @@ sub merge_gsicheck{
         $command = "gsicheck";
     }
     my $errfile = $sdffile.".err";
-    $command .= " -k -c -wcf $tmpfile -wef $errfile -l \"\" $sdffile";
+    $command .= " -k -c -wcf $tmpfile -wef ".fix_cygwin_path($errfile)." -l \"\" ".fix_cygwin_path($sdffile);
     #my $rc = system( $command );
+    if ($bVerbose) { print STDOUT "localize.pl running $command\n"; }
     my $output = `$command`;
     my $rc = $? << 8;
     if ( $output ne "" ){
@@ -513,7 +514,7 @@ sub collectfiles{
         # if ( -x $command ){
         if( $command ){
             if( !$bVerbose  ){ $args .= " -QQ "; }
-            $args .= " -e -f $localizeSDF -l ";
+            $args .= " -e -f ".fix_cygwin_path($localizeSDF)." -l ";
             my $bFlag="";
             if( $bAll ) {$args .= " en-US";}
             else{
@@ -1173,6 +1174,19 @@ sub usage{
     print STDERR "\nlocalize -m -l cs -f my.sdf\n( Merge cs translation into the sourcecode ) \n";
 }
 
+sub fix_cygwin_path
+{
+    my ( $path ) = @_;
+
+    if ( $^O eq 'cygwin' )
+    {
+    $path = qx{cygpath -m "$path"};
+    chomp($path);
+    }
+
+    return $path;
+}
+
 #            my $line           = defined $_ ? $_ : '';
 #            my $leftpart       = defined $2 ? $2 : '';
 #            my $prj            = defined $3 ? $3 : '';
