---
 l10ntools/scripts/localize.pl |   18 ++++++++++++++++--
 1 files changed, 16 insertions(+), 2 deletions(-)

diff --git l10ntools/scripts/localize.pl l10ntools/scripts/localize.pl
index 8310435..78b6b6e 100644
--- l10ntools/scripts/localize.pl
+++ l10ntools/scripts/localize.pl
@@ -386,8 +386,9 @@ sub merge_gsicheck{
     $command .= "$ENV{SOLARVER}/$ENV{INPATH}/bin/gsicheck";
 
     my $errfile = $sdffile.".err";
-    $command .= " -k -c -wcf $tmpfile -wef $errfile -l \"\" $sdffile";
+    $command .= " -k -c -wcf $tmpfile -wef ".fix_cygwin_path($errfile)." -l \"\" ".fix_cygwin_path($sdffile);
     #my $rc = system( $command );
+    if ($bVerbose) { print STDOUT "localize.pl running $command\n"; }
     my $output = `$command`;
     my $rc = $? << 8;
     if ( $output ne "" ){
@@ -492,7 +493,7 @@ sub collectfiles{
         # if ( -x $command ){
         if( $command ){
             if( !$bVerbose  ){ $args .= " "; }
-            $args .= " -e -f $localizeSDF -l ";
+            $args .= " -e -f ".fix_cygwin_path($localizeSDF)." -l ";
             my $bFlag="";
             if( $bAll ) {$args .= " en-US";}
             else{
@@ -1154,6 +1155,19 @@ sub usage{
     print STDERR "\nlocalize -m -l cs -f my.sdf\n( Merge cs translation into the sourcecode ) \n";
 }
 
+sub fix_cygwin_path
+{
+    my ( $path ) = @_;
+
+    if ( $^O eq 'cygwin' )
+    {
+    $path = qx{cygpath -m "$path"};
+    chomp($path);
+    }
+
+    return $path;
+}
+
 #            my $line           = defined $_ ? $_ : '';
 #            my $leftpart       = defined $2 ? $2 : '';
 #            my $prj            = defined $3 ? $3 : '';
-- 
1.7.0.1

