diff -Naur bitstream_vera_fonts.orig/prj/build.lst bitstream_vera_fonts/prj/build.lst
--- bitstream_vera_fonts/prj/build.lst	2010-02-12 22:44:46.000000000 +0100
+++ bitstream_vera_fonts/prj/build.lst	2010-03-19 12:25:39.452375000 +0100
@@ -2,4 +2,4 @@
 bs	bitstream_vera_fonts\fonts\ttf_dejavu		nmake	-	all	deja_fonts NULL
 bs	bitstream_vera_fonts\fonts\ttf_liberation	nmake	-	all	libe_fonts NULL
 bs	bitstream_vera_fonts\fonts\ttf_gentium		nmake	-	all	gent_fonts NULL
-
+bs	bitstream_vera_fonts\fonts\ttf_magyarlinuxlibertine		nmake	-	all	mlli_fonts NULL
diff -Naur bitstream_vera_fonts.orig/prj/d.lst bitstream_vera_fonts/prj/d.lst
--- bitstream_vera_fonts/prj/d.lst	2010-02-12 22:44:46.000000000 +0100
+++ bitstream_vera_fonts/prj/d.lst	2010-03-19 15:30:42.268250000 +0100
@@ -5,4 +5,5 @@
 ..\%__SRC%\misc\build\liberation-fonts-1.05\License          %COMMON_DEST%\pck%_EXT%\LICENSE_liberation
 ..\%__SRC%\misc\build\gentiumbasic-fonts-1.10\*.ttf          %COMMON_DEST%\pck%_EXT%\*.ttf
 ..\%__SRC%\misc\build\gentiumbasic-fonts-1.10\OFL.txt        %COMMON_DEST%\pck%_EXT%\LICENSE_gentium
-
+..\%__SRC%\misc\build\magyarlinuxlibertine\*.ttf             %COMMON_DEST%\pck%_EXT%\*.ttf
+..\%__SRC%\misc\build\magyarlinuxlibertine\LICENCE.txt       %COMMON_DEST%\pck%_EXT%\LICENSE_magyarlinuxlibertine
diff -Naur bitstream_vera_fonts.orig/fonts/ttf_magyarlinuxlibertine/makefile.mk bitstream_vera_fonts/fonts/ttf_magyarlinuxlibertine/makefile.mk
--- bitstream_vera_fonts/fonts/ttf_magyarlinuxlibertine/makefile.mk	1970-01-01 01:00:00.000000000 +0100
+++ bitstream_vera_fonts/fonts/ttf_magyarlinuxlibertine/makefile.mk	2010-03-19 15:14:13.265625000 +0100
@@ -0,0 +1,60 @@
+#*************************************************************************
+#
+# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+# 
+# Copyright 2008 by Sun Microsystems, Inc.
+#
+# OpenOffice.org - a multi-platform office productivity suite
+#
+# This file is part of OpenOffice.org.
+#
+# OpenOffice.org is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License version 3
+# only, as published by the Free Software Foundation.
+#
+# OpenOffice.org is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License version 3 for more details
+# (a copy is included in the LICENSE file that accompanied this code).
+#
+# You should have received a copy of the GNU Lesser General Public License
+# version 3 along with OpenOffice.org.  If not, see
+# <http://www.openoffice.org/license.html>
+# for a copy of the LGPLv3 License.
+#
+#*************************************************************************
+
+PRJ=..$/..
+
+PRJNAME=bitstream_vera_fonts
+TARGET=ttf_magyarlinuxlibertine
+
+# --- Settings -----------------------------------------------------
+
+.INCLUDE :	settings.mk
+
+# --- Files --------------------------------------------------------
+
+TARFILE_NAME=MagyarLinLibertine
+TARFILE_ROOTDIR=magyarlinuxlibertine
+
+PATCH_FILES=
+
+
+CONFIGURE_DIR=
+CONFIGURE_ACTION=
+
+BUILD_DIR=
+BUILD_ACTION=
+BUILD_FLAGS=
+
+# --- Targets ------------------------------------------------------
+
+.INCLUDE : set_ext.mk
+.INCLUDE : target.mk
+
+.IF "$(WITH_FONTS)"!="NO"
+.INCLUDE : tg_ext.mk
+.ENDIF
+
--- configure.in	2010-03-16 12:08:56.323375000 +0100
+++ configure.in	2010-03-19 13:14:09.766500000 +0100
@@ -102,6 +102,9 @@
                                 (even if present) in the openoffice.org installation
                                 set.
 ],,with_agfa_monotype_fonts="yes")
+AC_ARG_WITH(magyarlinuxlibertine-fonts,
+[  --with-magyarlinuxlibertine-fonts    Include Magyar Linux Libertine font set
+],,)
 AC_ARG_ENABLE(epm,
 [  --disable-epm           OO.o includes self-packaging code, that requires
                           epm, however epm is useless for large scale
@@ -6861,6 +6864,19 @@
 fi
 AC_SUBST(WITH_AGFA_MONOTYPE_FONTS)
 
+AC_MSG_CHECKING([whether to include Magyar Linux Libertine fonts])
+if test "$with_magyarlinuxlibertine_fonts" != "no" ; then
+  AC_MSG_RESULT([yes])
+  WITH_MAGYARLINUXLIBERTINE_FONTS=YES
+  BUILD_TYPE="$BUILD_TYPE MAGYARLINUXLIBERTINE_FONTS"
+  SCPDEFS="$SCPDEFS -DWITH_MAGYARLINUXLIBERTINE_FONTS"
+else
+  AC_MSG_RESULT([no])
+  WITH_MAGYARLINUXLIBERTINE_FONTS=NO
+fi
+AC_SUBST(WITH_MAGYARLINUXLIBERTINE_FONTS)
+
+
 dnl ===================================================================
 dnl Test whether to include ppds
 dnl ===================================================================
diff -Naur scp2.orig/source/ooo/file_font_ooo.scp scp2/source/ooo/file_font_ooo.scp
--- scp2/source/ooo/file_font_ooo.scp	2010-03-16 12:04:42.306375000 +0100
+++ scp2/source/ooo/file_font_ooo.scp	2010-03-19 13:02:02.527125000 +0100
@@ -212,3 +212,22 @@
 STD_FONTWIN_FILE( gid_File_Fnt_thowr, thowr___.ttf, ThorndaleAMT)
 #endif
 #endif
+
+#ifndef WITHOUT_FONTS 
+#ifdef WITH_MAGYARLINUXLIBERTINE_FONTS
+STD_FONT_FILE( gid_File_Fnt_LinuxLibertineI, LinLibertineI.ttf, Linux Libertine I)
+STD_FONT_FILE( gid_File_Fnt_LinuxLibertineN, LinLibertineN.ttf, Linux Libertine N)
+STD_FONT_FILE( gid_File_Fnt_MagyarLinLibertine, MagyarLinLibertine.ttf, Magyar Linux Libertine)
+STD_FONT_FILE( gid_File_Fnt_MagyarLinLibertineB, MagyarLinLibertineB.ttf, Magyar Linux Libertine Bold)
+STD_FONT_FILE( gid_File_Fnt_MagyarLinLibertineBI, MagyarLinLibertineBI.ttf, Magyar Linux Libertine Bold Italic)
+STD_FONT_FILE( gid_File_Fnt_MagyarLinLibertineI, MagyarLinLibertineI.ttf, Magyar Linux Libertine Italic)
+STD_FONT_FILE( gid_File_Fnt_MagyarLinLibertineC, MagyarLinLibertineC_Re.ttf, Magyar Linux Libertine Capital)
+STD_FONT_FILE( gid_File_Fnt_MagyarLinBiolinum, MagyarLinBiolinum.ttf, Magyar Linux Biolinum)
+STD_FONT_FILE( gid_File_Fnt_MagyarLinBiolinumB, MagyarLinBiolinumB.ttf, Magyar Linux Biolinum Bold)
+STD_FONT_FILE( gid_File_Fnt_MagyarLinLibertineN, MagyarLinLibertineN.ttf, Magyar Linux N Libertine)
+STD_FONT_FILE( gid_File_Fnt_MagyarLinLibertineNB, MagyarLinLibertineNB.ttf, Magyar Linux Libertine N Bold)
+STD_FONT_FILE( gid_File_Fnt_MagyarLinLibertineNBI, MagyarLinLibertineNBI.ttf, Magyar Linux Libertine N Bold Italic)
+STD_FONT_FILE( gid_File_Fnt_MagyarLinLibertineNI, MagyarLinLibertineNI.ttf, Magyar Linux Libertine N Italic)
+STD_FONT_FILE( gid_File_Fnt_MagyarLinLibertineNC, MagyarLinLibertineNC_Re.ttf, Magyar Linux Libertine N Capital)
+#endif 
+#endif
