--- sfx2/source/dialog/mailmodel.cxx.original
+++ sfx2/source/dialog/mailmodel.cxx
@@ -851,7 +851,13 @@
 
 					Sequence< OUString > aAttachmentSeq(&(maAttachedDocuments[0]),maAttachedDocuments.size());
  
-				    xSimpleMailMessage->setSubject( maSubject );
+                    if ( xSimpleMailMessage->getSubject().getLength() == 0 ) {
+                        OUString baseName( maAttachedDocuments[0].copy( maAttachedDocuments[0].lastIndexOf( '/' ) + 1 ) );
+                        OUString subject( baseName );
+                        if ( maAttachedDocuments.size() > 1 )
+                            subject += OUString::createFromAscii( ", ..." );
+                        xSimpleMailMessage->setSubject( subject );
+                    }
 				    xSimpleMailMessage->setAttachement( aAttachmentSeq );
 
 	                sal_Bool bSend( sal_False );
--- shell/source/win32/simplemail/senddoc.cxx
+++ shell/source/win32/simplemail/senddoc.cxx
@@ -34,9 +34,10 @@
 #include "precompiled_shell.hxx"
 #include <osl/diagnose.h>
 
-#ifndef _RTL_STRING_HXX_
 #include <rtl/string.hxx>
+#include <rtl/ustring.hxx>
+#include <rtl/uri.hxx>
+#include <osl/thread.hxx>
-#endif
 #include "simplemapi.hxx"
 
 #define WIN32_LEAN_AND_MEAN
@@ -175,7 +175,17 @@
 {
     ZeroMemory(pMapiMessage, sizeof(MapiMessage));
              
-    pMapiMessage->lpszSubject = const_cast<char*>(gSubject.c_str());   
+    try {
+         rtl_uString *subject = NULL;
+         rtl_uString_newFromAscii(&subject, const_cast<char*>(gSubject.c_str()));
+         rtl_uString *decoded_subject = NULL;
+         rtl_uriDecode(subject, rtl_UriDecodeWithCharset, RTL_TEXTENCODING_UTF8, &decoded_subject);
+         rtl::OUString ou_subject(decoded_subject);
+         pMapiMessage->lpszSubject = strdup(OUStringToOString(ou_subject, osl_getThreadTextEncoding(), RTL_UNICODETOTEXT_FLAGS_UNDEFINED_QUESTIONMARK).getStr());
+    }
+    catch (...) {
+    pMapiMessage->lpszSubject = const_cast<char*>(gSubject.c_str());
+    }
     pMapiMessage->lpszNoteText = (gBody.length() ? const_cast<char*>(gBody.c_str()) : NULL);
     pMapiMessage->lpOriginator = aMapiOriginator;       
     pMapiMessage->lpRecips = &aMapiRecipientList[0];
