---
 makefile.rc |   10 ++++++++++
 1 files changed, 10 insertions(+), 0 deletions(-)

diff --git makefile.rc makefile.rc
index ebe4104..80a9b2a 100644
--- makefile.rc
+++ makefile.rc
@@ -40,6 +40,16 @@ check_modules .PHONY :
 	@+echo Checking module list
 	@+perl $(SOLARENV)$/bin$/build.pl --checkmodules
 
+idl_offapi .SETDIR=offapi .PHONY :
+    @+perl $(SOLARENV)$/bin$/build.pl
+    @+perl $(SOLARENV)$/bin$/deliver.pl
+
+idl_offuh .SETDIR=offuh .PHONY :
+    @+perl $(SOLARENV)$/bin$/build.pl
+    @+perl $(SOLARENV)$/bin$/deliver.pl
+
+idl  : idl_offapi idl_offuh
+
 # Only build when all modules available
 build_instsetoo_native .SETDIR=instsetoo_native/prj : check_modules
 	@+perl $(SOLARENV)$/bin$/build.pl --all $(PROFULLSWITCH)
-- 
1.7.0.1

