diff --git sc/prj/build.lst sc/prj/build.lst
index e7da6af..09cf47e 100644
--- sc/prj/build.lst
+++ sc/prj/build.lst
@@ -1,4 +1,4 @@
-sc     sc      :       l10n vbahelper oovbaapi svx stoc uui BOOST:boost formula oox NULL
+sc     sc      :       l10n vbahelper oovbaapi svx stoc uui BOOST:boost formula oox MDDS:mdds NULL
 sc	sc							usr1	-	all	sc_mkout NULL
 sc	sc\inc							nmake	-	all	sc_inc NULL
 sc      sc\uiconfig\layout                                      nmake   -       all     sc_layout NULL
