From ff8087f1339caa4ff41e4c3f361740291f5593bd Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:08:36 +0200
Subject: [PATCH 824/878] mdds-build-dependency-sc.diff

---
 sc/prj/build.lst |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/sc/prj/build.lst b/sc/prj/build.lst
index e7da6af..e49d272 100644
--- a/sc/prj/build.lst
+++ b/sc/prj/build.lst
@@ -1,4 +1,4 @@
-sc     sc      :       l10n vbahelper oovbaapi svx stoc uui BOOST:boost formula oox NULL
+sc     sc      :       l10n vbahelper oovbaapi svx stoc uui BOOST:boost formula oox MDDS:mdds NULL
 sc	sc							usr1	-	all	sc_mkout NULL
 sc	sc\inc							nmake	-	all	sc_inc NULL
 sc      sc\uiconfig\layout                                      nmake   -       all     sc_layout NULL
-- 
1.7.0.1

