From 76c7f400ab89ce477f86893bcdf68e90ccc76582 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:08:36 +0200
Subject: [PATCH 728/768] mdds-build-dependency-sc.diff

---
 sc/prj/build.lst |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git sc/prj/build.lst sc/prj/build.lst
index 5b23458..296ab3a 100755
--- sc/prj/build.lst
+++ sc/prj/build.lst
@@ -1,4 +1,4 @@
-sc      sc      :       filter l10n vbahelper oovbaapi svx uui stoc BOOST:boost formula oox NULL
+sc      sc      :       filter l10n vbahelper oovbaapi svx uui stoc BOOST:boost formula oox MDDS:mdds NULL
 sc	sc							usr1	-	all	sc_mkout NULL
 sc	sc\inc							nmake	-	all	sc_inc NULL
 sc      sc\uiconfig\layout                                      nmake   -       all     sc_layout NULL
-- 
1.7.0.1

