diff -p -up officecfg/registry/data/org/openoffice/Office/Common.xcu.orig.2 officecfg/registry/data/org/openoffice/Office/Common.xcu
--- officecfg/registry/data/org/openoffice/Office/Common.xcu.orig.2	2008-11-17 07:03:40.000000000 -0500
+++ officecfg/registry/data/org/openoffice/Office/Common.xcu	2008-11-19 08:53:56.000000000 -0500
@@ -576,4 +576,10 @@
       </prop>
     </node>
   </node>
+  <!-- Freedesktop as default mailer -->
+  <node oor:name="ExternalMailer">
+   <prop oor:name="Program" oor:type="xs:string">
+      <value>xdg-mail</value>
+   </prop>
+ </node>
 </oor:component-data>
