---
 sfx2/source/dialog/passwd.cxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git sfx2/source/dialog/passwd.cxx sfx2/source/dialog/passwd.cxx
index d454c8c..d71d7e9 100644
--- sfx2/source/dialog/passwd.cxx
+++ sfx2/source/dialog/passwd.cxx
@@ -110,7 +110,7 @@ SfxPasswordDialog::SfxPasswordDialog( Window* pParent, const String* pGroupText
     maHelpBtn		( this, SfxResId( BTN_PASSWD_HELP ) ),
     maConfirmStr	( 		SfxResId( STR_PASSWD_CONFIRM ) ),
 
-    mnMinLen		( 5 ),
+    mnMinLen		( 1 ),
     maMinLenPwdStr	( SfxResId( STR_PASSWD_MIN_LEN ) ),
     maEmptyPwdStr	( SfxResId( STR_PASSWD_EMPTY ) ),
     maMainPwdStr    ( ),
-- 
1.7.0.1

