From 0c27aa1895c05237024d1fe88b1a87e3b0f463eb Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:57:19 +0200
Subject: [PATCH 227/768] min-password-length-sfx2.diff

---
 sfx2/source/dialog/passwd.cxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git sfx2/source/dialog/passwd.cxx sfx2/source/dialog/passwd.cxx
index d454c8c..d71d7e9 100644
--- sfx2/source/dialog/passwd.cxx
+++ sfx2/source/dialog/passwd.cxx
@@ -110,7 +110,7 @@ SfxPasswordDialog::SfxPasswordDialog( Window* pParent, const String* pGroupText
     maHelpBtn		( this, SfxResId( BTN_PASSWD_HELP ) ),
     maConfirmStr	( 		SfxResId( STR_PASSWD_CONFIRM ) ),
 
-    mnMinLen		( 5 ),
+    mnMinLen		( 1 ),
     maMinLenPwdStr	( SfxResId( STR_PASSWD_MIN_LEN ) ),
     maEmptyPwdStr	( SfxResId( STR_PASSWD_EMPTY ) ),
     maMainPwdStr    ( ),
-- 
1.7.0.1

