From 3fa9f2541865bb94df06c20f047ee790164878d9 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:57:19 +0200
Subject: [PATCH 256/878] min-password-length-sfx2.diff

---
 sfx2/source/dialog/passwd.cxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/sfx2/source/dialog/passwd.cxx b/sfx2/source/dialog/passwd.cxx
index 21af5c8..8c9af6b 100644
--- a/sfx2/source/dialog/passwd.cxx
+++ b/sfx2/source/dialog/passwd.cxx
@@ -85,7 +85,7 @@ SfxPasswordDialog::SfxPasswordDialog( Window* pParent, const String* pGroupText
     maHelpBtn		( this, SfxResId( BTN_PASSWD_HELP ) ),
     maConfirmStr	( 		SfxResId( STR_PASSWD_CONFIRM ) ),
 
-    mnMinLen		( 5 ),
+    mnMinLen		( 1 ),
     maMinLenPwdStr	( SfxResId( STR_PASSWD_MIN_LEN ) ),
     maEmptyPwdStr	( SfxResId( STR_PASSWD_EMPTY ) ),
     maMainPwdStr    ( ),
-- 
1.7.0.1

