--- helpcontent2/source/text/shared/guide/protection.xhp.old	2010-06-18 10:31:55.000000000 +0200
+++ helpcontent2/source/text/shared/guide/protection.xhp	2010-07-07 18:04:28.865750000 +0200
@@ -84,7 +84,7 @@
                <paragraph xml-lang="en-US" id="par_id3154684" role="paragraph" l10n="U" oldref="12">Turning on protection</paragraph>
             </tablecell>
             <tablecell>
-               <paragraph xml-lang="en-US" id="par_id3153104" role="paragraph" l10n="U" oldref="13">Choose <emph>Edit - Changes - Protect Records</emph>. Enter and confirm a password of at least 5 characters.</paragraph>
+               <paragraph xml-lang="en-US" id="par_id3153104" role="paragraph" l10n="U" oldref="13">Choose <emph>Edit - Changes - Protect Records</emph>. Enter and confirm a password.</paragraph>
             </tablecell>
          </tablerow>
          <tablerow>
--- helpcontent2/source/text/shared/guide/redlining_protect.xhp.old	2010-06-18 10:31:55.000000000 +0200
+++ helpcontent2/source/text/shared/guide/redlining_protect.xhp	2010-07-07 18:04:51.662625000 +0200
@@ -58,7 +58,7 @@
     <paragraph xml-lang="en-US" id="par_id3147088" role="listitem" l10n="U" oldref="3">Choose <emph>Protect Records</emph>. This opens the <link href="text/shared/01/password_dlg.xhp" name="Password"><emph>Password</emph></link> dialog.</paragraph>
    </listitem>
    <listitem>
-    <paragraph xml-lang="en-US" id="par_id3153345" role="listitem" l10n="U" oldref="4">Enter a password consisting of at least 5 characters and confirm it. Click <emph>OK</emph>.</paragraph>
+    <paragraph xml-lang="en-US" id="par_id3153345" role="listitem" l10n="U" oldref="4">Enter a password and confirm it. Click <emph>OK</emph>.</paragraph>
    </listitem>
   </list>
   <embed href="text/shared/00/00000004.xhp#related"/>
