diff --git a/mysqlc/source/makefile.mk b/mysqlc/source/makefile.mk
--- mysqlc/source/makefile.mk
+++ mysqlc/source/makefile.mk
@@ -115,7 +115,7 @@
 .IF "$(SYSTEM_MYSQL_CPPCONN)" != "YES"
 CDEFS += -DCPPCONN_LIB=\"$(DLLPRE)mysqlcppconn$(DLLPOST)\"
 .ELSE
-CDEFS += -DCPPCONN_LIB=\"libmysqlcppconn.so.2\"
+CDEFS += -DCPPCONN_LIB=\"$(shell readlink /usr/lib/libmysqlcppconn.so)\"
 .ENDIF
 
 # --------------- MySQL settings ------------------
diff --git a/mysqlcppconn/mysql_config.pmk b/mysqlcppconn/mysql_config.pmk
--- mysqlcppconn/mysql_config.pmk
+++ mysqlcppconn/mysql_config.pmk
@@ -35,13 +35,7 @@
   .ENDIF
 .ENDIF
 
-.IF "$(GUI)"!="WNT"
-
-.IF "$(SYSTEM_MYSQL)" == "YES"
-    MYSQL_LIB+=-lmysqlcppconn
-.ENDIF
-
-.ELSE   # "$(GUI)"!="WNT"
+.IF "$(GUI)"=="WNT"
 
 CFLAGS+=-DCPPDBC_WIN32 -D_CRT_SECURE_NO_WARNINGS -D_SCL_SECURE_NO_WARNINGS
 
