diff -purN nlpsolver.old/makefile.mk nlpsolver/makefile.mk
--- nlpsolver.old/makefile.mk	1970-01-01 01:00:00.000000000 +0100
+++ nlpsolver/makefile.mk	2009-10-26 22:34:07.000000000 +0100
@@ -0,0 +1,72 @@
+#*************************************************************************
+#
+# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+# 
+# Copyright 2008 by Sun Microsystems, Inc.
+#
+# OpenOffice.org - a multi-platform office productivity suite
+#
+# $RCSfile: makefile.mk,v $
+#
+# $Revision: 1.10 $
+#
+# This file is part of OpenOffice.org.
+#
+# OpenOffice.org is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License version 3
+# only, as published by the Free Software Foundation.
+#
+# OpenOffice.org is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License version 3 for more details
+# (a copy is included in the LICENSE file that accompanied this code).
+#
+# You should have received a copy of the GNU Lesser General Public License
+# version 3 along with OpenOffice.org.  If not, see
+# <http://www.openoffice.org/license.html>
+# for a copy of the LGPLv3 License.
+#
+#*************************************************************************
+
+PRJ=.
+
+PRJNAME=nlpsolver
+TARGET=nlpsolver
+
+.IF "$(SOLAR_JAVA)"!=""
+# --- Settings -----------------------------------------------------
+
+.INCLUDE :	settings.mk
+.INCLUDE :  antsettings.mk
+
+# --- Files --------------------------------------------------------
+
+TARFILE_NAME=nlpsolver-0.9
+TARFILE_ROOTDIR=nlpsolver-0.9
+
+#ADDITIONAL_FILES= \
+#    toolsrc/org/mozilla/javascript/tools/debugger/OfficeScriptInfo.java
+
+PATCH_FILES=nlpsolver-0.9.patch
+
+ANT_FLAGS+=-Dplatforms.JDK_1.5.home=$(JAVA_HOME) -Doffice.program.dir=$(SOLARBINDIR)
+
+.IF "$(JAVACISGCJ)"=="yes"
+JAVA_HOME=
+.EXPORT : JAVA_HOME
+BUILD_ACTION=$(ANT) $(ANT_FLAGS) -Dbuild.label="build-$(RSCREVISION)" -Dbuild.compiler=gcj uno-package
+.ELSE
+BUILD_ACTION=$(ANT) $(ANT_FLAGS) -Dbuild.label="build-$(RSCREVISION)" -Dplatforms.JDK_1.5.home=$(JAVA_HOME) -Dant.build.javac.source=$(JAVA_SOURCE_VER) -Dant.build.javac.target=$(JAVA_TARGET_VER) uno-package
+.ENDIF
+
+# --- Targets ------------------------------------------------------
+
+.INCLUDE : set_ext.mk
+.INCLUDE : target.mk
+.INCLUDE : tg_ext.mk
+
+.ELSE
+all:
+        @echo java disabled
+.ENDIF
diff -purN nlpsolver.old/nlpsolver-0.9.patch nlpsolver/nlpsolver-0.9.patch
--- nlpsolver.old/nlpsolver-0.9.patch	1970-01-01 01:00:00.000000000 +0100
+++ nlpsolver/nlpsolver-0.9.patch	2009-10-26 22:32:49.000000000 +0100
@@ -0,0 +1,51 @@
+--- misc/nlpsolver-0.9/nbproject/build-uno-impl.xml.old	2009-03-30 18:32:00.000000000 +0200
++++ misc/build/nlpsolver-0.9/nbproject/build-uno-impl.xml	2009-10-28 18:22:40.000000000 +0100
+@@ -197,7 +197,7 @@
+     <!-- enter the class path with external jars, if external jars exit and the user did not enter his own -->
+     <target name="-do-class-path" depends="-external-jars,-check-class-path" unless="class.path.exists">
+         <manifest file="${build.dir}/MANIFEST.MF" mode="update">
+-            <attribute name="Class-Path" value="${external.jars}"/> 
++            <attribute name="Class-Path" value="${external.jar.dir}/EvolutionarySolver.jar"/> 
+         </manifest>            
+     </target>
+     <target name="-check-class-path">
+--- misc/nlpsolver-0.9/nbproject/project.properties.old	2009-03-30 18:32:00.000000000 +0200
++++ misc/build/nlpsolver-0.9/nbproject/project.properties	2009-10-28 18:09:01.000000000 +0100
+@@ -27,8 +27,10 @@ file.reference.EvolutionarySolver.jar=Th
+ includes=**
+ jar.compress=false
+ javac.classpath=\
+-    ${libs.StarOffice 9.classpath}:\
+-    ${libs.OpenOffice.org 3.0.classpath}:\
++    ${solar.jar}/juh.jar:\
++    ${solar.jar}/jurt.jar:\
++    ${solar.jar}/ridl.jar:\
++    ${solar.jar}/unoil.jar:\
+     ${reference.EvolutionarySolver.jar}
+ # Space-separated list of extra javac options
+ javac.compilerargs=
+--- misc/nlpsolver-0.9/nbproject/build-impl.xml.old	2009-03-30 18:32:00.000000000 +0200
++++ misc/build/nlpsolver-0.9/nbproject/build-impl.xml	2009-10-28 18:19:53.000000000 +0100
+@@ -360,6 +360,8 @@ is divided into following sections:
+             -->
+     <target depends="init" name="deps-jar" unless="no.deps">
+         <ant antfile="${project.EvolutionarySolver}/build.xml" inheritall="false" target="jar"/>
++	<mkdir dir="${dist.dir}/${external.jar.dir}"/>
++	<copy file="${reference.EvolutionarySolver.jar}" todir="${dist.dir}/${external.jar.dir}"/>
+     </target>
+     <target depends="init,-check-automatic-build,-clean-after-automatic-build" name="-verify-automatic-build"/>
+     <target depends="init" name="-check-automatic-build">
+--- misc/nlpsolver-0.9/src/description.xml.old	2009-03-30 16:56:42.000000000 +0200
++++ misc/build/nlpsolver-0.9/src/description.xml	2009-10-28 18:29:13.000000000 +0100
+@@ -12,11 +12,6 @@
+     <extension-description>
+         <src lang="en" xlink:href="description/extensiondescription.txt"/>
+     </extension-description>
+-    <registration>
+-        <simple-license accept-by="admin" default-license-id="en">
+-            <license-text lang="en" license-id="en" xlink:href="licenses/lgpl-3.0.txt"/>
+-        </simple-license>
+-    </registration>
+     <dependencies>
+         <OpenOffice.org-minimal-version xmlns:d="http://openoffice.org/extensions/description/2006" d:name="OpenOffice.org 3.0" value="3.0"/>
+     </dependencies>
diff -purN nlpsolver.old/prj/build.lst nlpsolver/prj/build.lst
--- nlpsolver.old/prj/build.lst	1970-01-01 01:00:00.000000000 +0100
+++ nlpsolver/prj/build.lst	2009-10-26 20:50:12.000000000 +0100
@@ -0,0 +1,3 @@
+nlpsolver	nlpsolver	:	javaunohelper unoil jurt NULL NULL
+nlpsolver	nlpsolver		usr1	-	all	nlpsolver_mkout NULL
+nlpsolver	nlpsolver      nmake	-	all	nlpsolver_nlpsolver NULL
diff -purN nlpsolver.old/prj/d.lst nlpsolver/prj/d.lst
--- nlpsolver.old/prj/d.lst	1970-01-01 01:00:00.000000000 +0100
+++ nlpsolver/prj/d.lst	2009-10-26 22:12:02.000000000 +0100
@@ -0,0 +1 @@
+..\%__SRC%\misc\build\nlpsolver-0.9/dist\NLPSolver.oxt %_DEST%\bin%_EXT%\NLPSolver.oxt
