Index: svx/source/dialog/treeopt.cxx
===================================================================
--- svx/source/cui/treeopt.cxx	(Revision 277995)
+++ svx/source/cui/treeopt.cxx	(Arbeitskopie)
@@ -2204,11 +2204,9 @@
             if ( nPageId == RID_SVXPAGE_INET_MAIL )
                 continue;
 #endif
-#if defined MACOSX
-            // Disable Mozilla Plug-in tab-page on Mac
+            // Disable Mozilla Plug-in tab-page
             if ( nPageId == RID_SVXPAGE_INET_MOZPLUGIN )
                 continue;
-#endif
             AddTabPage( nPageId, rInetArray.GetString(i), nGroup );
         }
     }
