From 44b96c4448bb5e33107bf4ab6e65d863eee74650 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:57:33 +0200
Subject: [PATCH 268/878] no-mozilla-plug-in-option.diff

---
 svx/source/cui/treeopt.cxx |    4 +---
 1 files changed, 1 insertions(+), 3 deletions(-)

diff --git a/svx/source/cui/treeopt.cxx b/svx/source/cui/treeopt.cxx
index e096027..458d28c 100644
--- a/svx/source/cui/treeopt.cxx
+++ b/svx/source/cui/treeopt.cxx
@@ -2202,11 +2202,9 @@ void OfaTreeOptionsDialog::Initialize( const Reference< XFrame >& _xFrame )
             if ( nPageId == RID_SVXPAGE_INET_MAIL )
                 continue;
 #endif
-#if defined MACOSX
-            // Disable Mozilla Plug-in tab-page on Mac
+            // Disable Mozilla Plug-in tab-page
             if ( nPageId == RID_SVXPAGE_INET_MOZPLUGIN )
                 continue;
-#endif
             AddTabPage( nPageId, rInetArray.GetString(i), nGroup );
         }
     }
-- 
1.7.0.1

