From 3911afe0ed7b1c63e231ab85b9ace48732144c68 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:58:43 +0200
Subject: [PATCH 332/878] no-splash-takeover.diff

---
 vcl/source/window/window.cxx |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

diff --git a/vcl/source/window/window.cxx b/vcl/source/window/window.cxx
index dbc6eb5..3979624 100644
--- a/vcl/source/window/window.cxx
+++ b/vcl/source/window/window.cxx
@@ -802,8 +802,6 @@ void Window::ImplInit( Window* pParent, WinBits nStyle, SystemParentData* pSyste
         else if( mpWindowImpl->mbFloatWin )
             nFrameStyle |= SAL_FRAME_STYLE_TOOLWINDOW;
 
-        if( nStyle & WB_INTROWIN )
-            nFrameStyle |= SAL_FRAME_STYLE_INTRO;
         if( nStyle & WB_TOOLTIPWIN )
             nFrameStyle |= SAL_FRAME_STYLE_TOOLTIP;
 
-- 
1.7.0.1

