---
 vcl/source/window/window.cxx |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

diff --git vcl/source/window/window.cxx vcl/source/window/window.cxx
index fcad734..a231e12 100644
--- vcl/source/window/window.cxx
+++ vcl/source/window/window.cxx
@@ -804,8 +804,6 @@ void Window::ImplInit( Window* pParent, WinBits nStyle, SystemParentData* pSyste
         else if( mpWindowImpl->mbFloatWin )
             nFrameStyle |= SAL_FRAME_STYLE_TOOLWINDOW;
 
-        if( nStyle & WB_INTROWIN )
-            nFrameStyle |= SAL_FRAME_STYLE_INTRO;
         if( nStyle & WB_TOOLTIPWIN )
             nFrameStyle |= SAL_FRAME_STYLE_TOOLTIP;
 
-- 
1.7.0.1

