--- instsetoo_native/inc_openoffice/windows/msi_languages/Control.ulf 
+++ instsetoo_native/inc_openoffice/windows/msi_languages/Control.ulf
@@ -379,7 +379,7 @@
 en-US = "Cancel"
 
 [OOO_CONTROL_124]
-en-US = "Build contributed in collaboration with the community by Oracle. For credits, see: http://www.openoffice.org/welcome/credits.html"
+en-US = "Build contributed in collaboration with the community by Novell, Inc. For credits, see: http://www.openoffice.org/welcome/credits.html"
 
 [OOO_CONTROL_125]
 en-US = "&Next >"
--- instsetoo_native/util/makefile.mk
+++ instsetoo_native/util/makefile.mk
@@ -372,9 +372,15 @@
 	$(RM) $(MSILANGPACKTEMPLATEDIR)$/Binary$/Image.bmp
 	$(RM) $(MSIURETEMPLATEDIR)$/Binary$/Image.bmp
 	$(RM) $(MSISDKOOTEMPLATEDIR)$/Binary$/Image.bmp
-	$(COPY) $(PRJ)$/res$/nologoinstall.bmp $(MSIOFFICETEMPLATEDIR)$/Binary$/Image.bmp
-	$(COPY) $(PRJ)$/res$/nologoinstall.bmp $(MSILANGPACKTEMPLATEDIR)$/Binary$/Image.bmp
-	$(COPY) $(PRJ)$/res$/nologoinstall.bmp $(MSIURETEMPLATEDIR)$/Binary$/Image.bmp
-	$(COPY) $(PRJ)$/res$/nologoinstall.bmp $(MSISDKOOTEMPLATEDIR)$/Binary$/Image.bmp
+    $(COPY) $(PRJ)$/res$/nologoinstall_nld.bmp $(MSIOFFICETEMPLATEDIR)$/Binary$/Image.bmp
+    $(COPY) $(PRJ)$/res$/nologoinstall_nld.bmp $(MSILANGPACKTEMPLATEDIR)$/Binary$/Image.bmp
+    $(COPY) $(PRJ)$/res$/nologoinstall_nld.bmp $(MSISDKOOTEMPLATEDIR)$/Binary$/Image.bmp
+    $(RM) $(MSIOFFICETEMPLATEDIR)$/Binary$/Banner.bmp
+    $(RM) $(MSILANGPACKTEMPLATEDIR)$/Binary$/Banner.bmp
+    $(RM) $(MSISDKOOTEMPLATEDIR)$/Binary$/Banner.bmp
+    $(COPY) $(PRJ)$/res$/banner_nld.bmp $(MSIOFFICETEMPLATEDIR)$/Binary$/Banner.bmp
+    $(COPY) $(PRJ)$/res$/banner_nld.bmp $(MSILANGPACKTEMPLATEDIR)$/Binary$/Banner.bmp
+    $(COPY) $(PRJ)$/res$/banner_nld.bmp $(MSISDKOOTEMPLATEDIR)$/Binary$/Banner.bmp
+    $(COPY) $(PRJ)$/res$/banner_nld.bmp $(MSIURETEMPLATEDIR)$/Binary$/Banner.bmp
 
 
--- instsetoo_native/util/openoffice.lst
+++ instsetoo_native/util/openoffice.lst
@@ -29,9 +29,9 @@
 			USE_FILEVERSION 1
 			LIBRARYVERSION 9.3.0
 			PATCHCODEFILE ooo_patchcodes.txt
-			DOWNLOADBANNER	ooobanner.bmp
-			DOWNLOADBITMAP	ooobitmap.bmp
-			DOWNLOADSETUPICO	ooosetup.ico
+            DOWNLOADBANNER	ooobanner_nld.bmp
+            DOWNLOADBITMAP	ooobitmap_nld.bmp
+            DOWNLOADSETUPICO	ooosetup_nld.ico
 			RELATIVE_PATHES_IN_DDF 1
             STARTCENTER_ADDFEATURE_URL http://extensions.go-oo.org/getmore?cid=920794
             STARTCENTER_INFO_URL http://www.go-oo.org
@@ -76,9 +76,9 @@
 			LIBRARYVERSION 9.3.0
 			PATCHCODEFILE ooo_patchcodes.txt
 			JAVAPRODUCT 1
-			DOWNLOADBANNER	ooobanner.bmp
-			DOWNLOADBITMAP	ooobitmap.bmp
-			DOWNLOADSETUPICO	ooosetup.ico
+            DOWNLOADBANNER	ooobanner_nld.bmp
+            DOWNLOADBITMAP	ooobitmap_nld.bmp
+            DOWNLOADSETUPICO	ooosetup_nld.ico
 			RELATIVE_PATHES_IN_DDF 1
             STARTCENTER_ADDFEATURE_URL http://extensions.go-oo.org/getmore?cid=920794
             STARTCENTER_INFO_URL http://www.go-oo.org
@@ -122,9 +122,9 @@
 			LIBRARYVERSION 9.3.0
 			PATCHCODEFILE ooodev_patchcodes.txt
 			CODEFILENAME codes_ooodev.txt
-			DOWNLOADBANNER	ooobanner.bmp
-			DOWNLOADBITMAP	ooobitmap.bmp
-			DOWNLOADSETUPICO	ooosetup.ico
+            DOWNLOADBANNER	ooobanner_nld.bmp
+            DOWNLOADBITMAP	ooobitmap_nld.bmp
+            DOWNLOADSETUPICO	ooosetup_nld.ico
 			LOCALUSERDIR $ORIGIN/..
 			RELATIVE_PATHES_IN_DDF 1
             STARTCENTER_ADDFEATURE_URL http://extensions.go-oo.org/getmore?cid=920794
@@ -150,9 +150,9 @@
             PCPFILENAME ure.pcp
             POOLPRODUCT 0
             GLOBALFILEGID gid_File_Dl_Cppu
-            DOWNLOADBANNER urebanner.bmp
-            DOWNLOADBITMAP urebitmap.bmp
-            DOWNLOADSETUPICO ooosetup.ico
+            DOWNLOADBANNER ooobanner_nld.bmp
+            DOWNLOADBITMAP ooobitmap_nld.bmp
+            DOWNLOADSETUPICO ooosetup_nld.ico
             DONTUSESTARTMENUFOLDER 1
             SERVICESPROJEKT 1
             RELATIVE_PATHES_IN_DDF 1
@@ -92,9 +92,9 @@
             NOVERSIONINDIRNAME 0
             NOSPACEINDIRECTORYNAME 1
             CHANGETARGETDIR 1
-            DOWNLOADBANNER ooosdkbanner.bmp
-            DOWNLOADBITMAP ooosdkbitmap.bmp
-            DOWNLOADSETUPICO ooosetup.ico
+            DOWNLOADBANNER ooobanner_nld.bmp
+            DOWNLOADBITMAP ooobitmap_nld.bmp
+            DOWNLOADSETUPICO ooosetup_nld.ico
             DONTUSESTARTMENUFOLDER 1
             RELATIVE_PATHES_IN_DDF 1
             STARTCENTER_ADDFEATURE_URL http://extensions.go-oo.org/getmore?cid=920794
