From 807f0ec8fd3f8f34724160b94b425de4f4806016 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:57:26 +0200
Subject: [PATCH 233/768] novell-palette.diff

---
 extras/source/palettes/standard.soc |   12 ++++++++++++
 1 files changed, 12 insertions(+), 0 deletions(-)

diff --git extras/source/palettes/standard.soc extras/source/palettes/standard.soc
index b275e95..6209fd9 100644
--- extras/source/palettes/standard.soc
+++ extras/source/palettes/standard.soc
@@ -88,6 +88,18 @@
  <draw:color draw:name="Dark violet" draw:color="#660066"/>
  <draw:color draw:name="Salmon" draw:color="#ff8080"/>
  <draw:color draw:name="Sea blue" draw:color="#0066cc"/>
+ <draw:color draw:name="openSUSE light green" draw:color="#bddd8d"/>
+ <draw:color draw:name="openSUSE medium green" draw:color="#198533"/>
+ <draw:color draw:name="openSUSE dark green" draw:color="#0c5248"/>
+ <draw:color draw:name="Novell red" draw:color="#e00000"/>
+ <draw:color draw:name="Novell orange" draw:color="#e67814"/>
+ <draw:color draw:name="Novell yellow" draw:color="#ffdd00"/>
+ <draw:color draw:name="Novell green" draw:color="#629e1f"/>
+ <draw:color draw:name="Novell teal" draw:color="#6ea19b"/>
+ <draw:color draw:name="Novell blue" draw:color="#1c82b9"/>
+ <draw:color draw:name="Novell light gray" draw:color="#cccccd"/>
+ <draw:color draw:name="Novell medium gray" draw:color="#5a5a64"/>
+ <draw:color draw:name="Novell dark gray" draw:color="#3c3c41"/>
  <draw:color draw:name="Chart 1" draw:color="#004586"/>
  <draw:color draw:name="Chart 2" draw:color="#ff420e"/>
  <draw:color draw:name="Chart 3" draw:color="#ffd320"/>
-- 
1.7.0.1

