From eb3b38873788ef3c03e8e2776d5ca1af674d43cd Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:07:29 +0200
Subject: [PATCH 766/878] odf-export-invalid-memory-write-fix.diff

---
 sax/source/expatwrap/saxwriter.cxx |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/sax/source/expatwrap/saxwriter.cxx b/sax/source/expatwrap/saxwriter.cxx
index d4c7e31..3f5a6f1 100644
--- a/sax/source/expatwrap/saxwriter.cxx
+++ b/sax/source/expatwrap/saxwriter.cxx
@@ -700,6 +700,7 @@ inline void SaxWriterHelper::clearBuffer() throw( SAXException )
         m_Sequence.realloc(nCurrentPos);
         nCurrentPos = writeSequence();
         m_Sequence.realloc(SEQUENCESIZE);
+        mp_Sequence = m_Sequence.getArray();
     }
 }
 
-- 
1.7.0.1

