diff --git sax/source/expatwrap/saxwriter.cxx sax/source/expatwrap/saxwriter.cxx
index f208ee6..11311ef 100644
--- sax/source/expatwrap/saxwriter.cxx
+++ sax/source/expatwrap/saxwriter.cxx
@@ -703,6 +703,7 @@ inline void SaxWriterHelper::clearBuffer() throw( SAXException )
         m_Sequence.realloc(nCurrentPos);
         nCurrentPos = writeSequence();
         m_Sequence.realloc(SEQUENCESIZE);
+        mp_Sequence = m_Sequence.getArray();
     }
 }
 
