Index: pack/gendocu/makefile.mk
===================================================================
--- odk/pack/gendocu/makefile.mk	(revision 276718)
+++ odk/pack/gendocu/makefile.mk	(working copy)
@@ -52,7 +52,6 @@
         com.sun.star.comp.helper \
         com.sun.star.lib.uno.helper \
         com.sun.star.lib.unoloader \
-        com.sun.star.tools.uno \
         com.sun.star.uno \
         $(JAVA_SRC_DIR)$/com$/sun$/star$/lib$/util$/UrlToFileMapper.java
 #        com.sun.star.beans \
