--- odk/source/unowinreg/win/makefile.mk-old	2009-12-21 08:20:26.000000000 -0700
+++ odk/source/unowinreg/win/makefile.mk	2009-12-21 08:16:54.000000000 -0700
@@ -46,7 +46,6 @@
 JAVA_INCLUDES:= -I$(JAVA_HOME)/include 
 
 # values taken from set_soenv.in
-.IF "$(JDK)" != "gcj"
 .IF "$(OS)" == "LINUX"
 JAVA_INCLUDES+= -I$(JAVA_HOME)/include/linux
 .ELIF "$(OS)" == "FREEBSD"
@@ -58,7 +57,6 @@
 .ELIF "$(OS)" == "IRIX"
 JAVA_INCLUDES+= -I$(JAVA_HOME)/include/solaris
 .ENDIF
-.ENDIF
 
 .IF "$(SOLAR_JAVA)"==""
 nojava:
