Index: localize.sdf
===================================================================
RCS file: /cvs/util/officecfg/registry/data/org/openoffice/Office/UI/localize.sdf,v
retrieving revision 1.45
retrieving revision 1.46
diff -u -r1.45 -r1.46
--- officecfg/registry/data/org/openoffice/Office/UI/localize.sdf	11 Apr 2008 13:44:48 -0000	1.45
+++ officecfg/registry/data/org/openoffice/Office/UI/localize.sdf	16 Apr 2008 13:29:11 -0000	1.46
@@ -143041,7 +144325,7 @@
 officecfg	registry\data\org\openoffice\Office\UI\GenericCommands.xcu	0	value	..GenericCommands.UserInterface.Popups..uno:EditMenu	Label			0	da	~Rediger				2002-02-02 02:02:02
 officecfg	registry\data\org\openoffice\Office\UI\GenericCommands.xcu	0	value	..GenericCommands.UserInterface.Popups..uno:EditMenu	Label			0	de	~Bearbeiten				2002-02-02 02:02:02
 officecfg	registry\data\org\openoffice\Office\UI\GenericCommands.xcu	0	value	..GenericCommands.UserInterface.Popups..uno:EditMenu	Label			0	dz	ཞུན་དག(~E)				2002-02-02 02:02:02
-officecfg	registry\data\org\openoffice\Office\UI\GenericCommands.xcu	0	value	..GenericCommands.UserInterface.Popups..uno:EditMenu	Label			0	el	Ε~πεξεργασία				2002-02-02 02:02:02
+officecfg	registry\data\org\openoffice\Office\UI\GenericCommands.xcu	0	value	..GenericCommands.UserInterface.Popups..uno:EditMenu	Label			0	el	~				2002-02-02 02:02:02
 officecfg	registry\data\org\openoffice\Office\UI\GenericCommands.xcu	0	value	..GenericCommands.UserInterface.Popups..uno:EditMenu	Label			0	en-GB	~Edit				2002-02-02 02:02:02
 officecfg	registry\data\org\openoffice\Office\UI\GenericCommands.xcu	0	value	..GenericCommands.UserInterface.Popups..uno:EditMenu	Label			0	en-ZA	~Edit				2002-02-02 02:02:02
 officecfg	registry\data\org\openoffice\Office\UI\GenericCommands.xcu	0	value	..GenericCommands.UserInterface.Popups..uno:EditMenu	Label			0	eo	Redakti				2002-02-02 02:02:02
