From 5696e58208a9dd1eb1d2e835c0a4ee9c6a0d9b96 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:53:02 +0200
Subject: [PATCH 031/878] ok-cancel-btn-add-accel.diff

---
 vcl/source/src/btntext.src |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/vcl/source/src/btntext.src b/vcl/source/src/btntext.src
index ba7376e..0e342e1 100644
--- a/vcl/source/src/btntext.src
+++ b/vcl/source/src/btntext.src
@@ -33,12 +33,12 @@
 
 String SV_BUTTONTEXT_OK
 {
-    Text [ en-US ] = "OK";
+    Text [ en-US ] = "~OK";
 };
 
 String SV_BUTTONTEXT_CANCEL
 {
-    Text [ en-US ] = "Cancel";
+    Text [ en-US ] = "~Cancel";
 };
 
 String SV_BUTTONTEXT_YES
-- 
1.7.0.1

