From b827edde5c2f5ffc6121703dfef9ac3b6431a632 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:53:02 +0200
Subject: [PATCH 017/768] ok-cancel-btn-add-accel.diff

---
 vcl/source/src/btntext.src |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git vcl/source/src/btntext.src vcl/source/src/btntext.src
index 99579f0..1d07a15 100644
--- vcl/source/src/btntext.src
+++ vcl/source/src/btntext.src
@@ -33,12 +33,12 @@
 
 String SV_BUTTONTEXT_OK
 {
-    Text [ en-US ] = "OK";
+    Text [ en-US ] = "~OK";
 };
 
 String SV_BUTTONTEXT_CANCEL
 {
-    Text [ en-US ] = "Cancel";
+    Text [ en-US ] = "~Cancel";
 };
 
 String SV_BUTTONTEXT_YES
-- 
1.7.0.1

