From a684289ce59111fef2a24a7eee3b777df6fcf384 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:52:19 +0200
Subject: [PATCH 001/768] ooo-build-resources-sc-sc-hrc.diff

---
 sc/inc/sc.hrc |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git sc/inc/sc.hrc sc/inc/sc.hrc
index d26a508..c9708c3 100644
--- sc/inc/sc.hrc
+++ sc/inc/sc.hrc
@@ -1650,6 +1650,11 @@
 #define MID_3   3
 #define MID_4   4
 
+// ----------------------------------------------------------------------------
+// ooo-build specific resources
+
+#define SC_OOO_BUILD_START          (SC_DIALOGS_END)
+
 #endif
 
 
-- 
1.7.0.1

