---
 sc/inc/sc.hrc |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git sc/inc/sc.hrc sc/inc/sc.hrc
index 6141ba2..2c7afaa 100644
--- sc/inc/sc.hrc
+++ sc/inc/sc.hrc
@@ -1646,6 +1646,11 @@
 #define MID_3   3
 #define MID_4   4
 
+// ----------------------------------------------------------------------------
+// ooo-build specific resources
+
+#define SC_OOO_BUILD_START          (SC_DIALOGS_END)
+
 #endif
 
 
-- 
1.7.0.1

