From 80fba47ae9ad39a2fc63cb1cd3200cff2abab372 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:52:23 +0200
Subject: [PATCH 002/878] ooo-build-resources-svx-dialogs-hrc.diff

---
 svx/inc/svx/dialogs.hrc |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/svx/inc/svx/dialogs.hrc b/svx/inc/svx/dialogs.hrc
index 7a7a77a..f72f199 100644
--- a/svx/inc/svx/dialogs.hrc
+++ b/svx/inc/svx/dialogs.hrc
@@ -1571,5 +1571,10 @@
 #define RID_SCH_TransformTabDLG_SVXPAGE_ANGLE			(RID_SVX_START + 998)
 #define RID_SCH_TransformTabDLG_SVXPAGE_SLANT			(RID_SVX_START + 999)
 
+// ----------------------------------------------------------------------------
+// ooo-build specific resources
+
+#define SVX_OOO_BUILD_START             (RID_SVX_START + 1200)
+
 #endif
 
-- 
1.7.0.1

