From cabd09e8b5789dba929a346a5a4f57a943dfdba8 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:52:19 +0200
Subject: [PATCH 001/878] ooo-build-resources-sc-sc-hrc.diff

---
 sc/inc/sc.hrc |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/sc/inc/sc.hrc b/sc/inc/sc.hrc
index 313ba56..8f4196c 100644
--- a/sc/inc/sc.hrc
+++ b/sc/inc/sc.hrc
@@ -1640,6 +1640,11 @@
 #define MID_3   3
 #define MID_4   4
 
+// ----------------------------------------------------------------------------
+// ooo-build specific resources
+
+#define SC_OOO_BUILD_START          (SC_DIALOGS_END)
+
 #endif
 
 
-- 
1.7.0.1

