---
 svx/inc/svx/dialogs.hrc |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git svx/inc/svx/dialogs.hrc svx/inc/svx/dialogs.hrc
index eb304d0..eb1d9c5 100755
--- svx/inc/svx/dialogs.hrc
+++ svx/inc/svx/dialogs.hrc
@@ -1051,5 +1051,10 @@
 #define OFA_TP_TABPROPERTIES_TEXT	    (RID_OFA_START + 245)
 #define OFA_TP_HELPERPROG               (RID_OFA_START + 249)
 
+// ----------------------------------------------------------------------------
+// ooo-build specific resources
+
+#define SVX_OOO_BUILD_START             (RID_SVX_START + 1200)
+
 #endif
 
-- 
1.7.0.1

