From fbce7b4f52f0ac02492f9acea3fbf5d1f30704e4 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:01:16 +0200
Subject: [PATCH 452/878] ooo67658.sfx2.reloadcrash.diff

---
 sfx2/source/doc/objxtor.cxx |   11 ++++++++++-
 1 files changed, 10 insertions(+), 1 deletions(-)

diff --git a/sfx2/source/doc/objxtor.cxx b/sfx2/source/doc/objxtor.cxx
index e6f402e..5005d7b 100644
--- a/sfx2/source/doc/objxtor.cxx
+++ b/sfx2/source/doc/objxtor.cxx
@@ -411,7 +411,16 @@ SfxObjectShell::~SfxObjectShell()
     }
 
     if ( pImp->bOwnsStorage && pImp->m_xDocStorage.is() )
-        pImp->m_xDocStorage->dispose();
+    {
+    try
+    {
+            pImp->m_xDocStorage->dispose();
+    }
+    catch( Exception& )
+    {
+            /* what to do ? */
+    }
+    }
 
     if ( pMedium )
     {
-- 
1.7.0.1

