--- vcl/unx/gtk/gdi/salnativewidgets-gtk.cxx.old	2009-10-29 15:25:08.000000000 +0100
+++ vcl/unx/gtk/gdi/salnativewidgets-gtk.cxx	2009-10-29 15:24:44.000000000 +0100
@@ -3449,10 +3449,11 @@ void GtkSalGraphics::updateSettings( All
     //  FIXME: need some way of fetching toolbar icon size.
 //	aStyleSet.SetToolbarIconSize( STYLE_TOOLBAR_ICONSIZE_SMALL );
 
-    const cairo_font_options_t *pNewOptions = 0;
-    if (GdkScreen* pScreen = gdk_display_get_screen( gdk_display_get_default(), m_nScreen ))
-        pNewOptions = gdk_screen_get_font_options(pScreen);
-    aStyleSet.SetCairoFontOptions( pNewOptions );
+// FIXME: gdk_screen_get_font_options is not supported by gdk2-2.8 on SLED10
+//    const cairo_font_options_t *pNewOptions = 0;
+//    if (GdkScreen* pScreen = gdk_display_get_screen( gdk_display_get_default(), m_nScreen ))
+//        pNewOptions = gdk_screen_get_font_options(pScreen);
+//    aStyleSet.SetCairoFontOptions( pNewOptions );
 
     // finally update the collected settings
     rSettings.SetStyleSettings( aStyleSet );
