---
 desktop/scripts/unopkg.sh |   29 +++++++++++++++++++++++++++--
 1 files changed, 27 insertions(+), 2 deletions(-)

diff --git desktop/scripts/unopkg.sh desktop/scripts/unopkg.sh
index 5a8ee26..df3856a 100644
--- desktop/scripts/unopkg.sh
+++ desktop/scripts/unopkg.sh
@@ -43,6 +43,29 @@ fi
     ;;
 esac
 
+isshared=0
+for arg in $@
+do
+if [ "$arg" = "--shared" ]; then
+    isshared=1
+fi
+done
+if [ $isshared -eq 1 ]; then
+    echo $@ | grep -q env:JFW_PLUGIN_DO_NOT_CHECK_ACCESSIBILITY
+    if [ $? -ne 0 ]; then
+        set -- $@ '-env:JFW_PLUGIN_DO_NOT_CHECK_ACCESSIBILITY=1'
+    fi
+    echo $@ | grep -q env:UNO_JAVA_JFW_INSTALL_DATA
+    if [ $? -ne 0 -a -w $sd_prog/../share/config/javasettingsunopkginstall.xml ]; then
+        set -- $@ '-env:UNO_JAVA_JFW_INSTALL_DATA=$$ORIGIN/../share/config/javasettingsunopkginstall.xml'
+    fi
+    echo $@ | grep -q env:UserInstallation
+    if [ $? -ne 0 ]; then
+        INSTDIR=`mktemp -d -t unoinstall.XXXXXX`
+        set -- $@ '-env:UserInstallation=file://'$INSTDIR
+    fi
+fi
+
 #collect all bootstrap variables specified on the command line
 #so that they can be passed as arguments to javaldx later on
 for arg in $@
@@ -71,6 +94,8 @@ unset XENVIRONMENT
 # SAL_NO_XINITTHREADS=true; export SAL_NO_XINITTHREADS
 
 # execute binary
-exec "$sd_prog/unopkg.bin" "$@" \
+"$sd_prog/unopkg.bin" "$@" \
     "-env:INIFILENAME=vnd.sun.star.pathname:$sd_prog/redirectrc"
-
+if [ -n "$INSTDIR" ]; then
+   rm -rf $INSTDIR
+fi
-- 
1.7.0.1

