From 63affdd19f9d02f7ea4bada970cef2c3ddc88c23 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:01:20 +0200
Subject: [PATCH 455/878] ooo86080.unopkg.bodge.diff

---
 desktop/scripts/unopkg.sh |   29 +++++++++++++++++++++++++++--
 1 files changed, 27 insertions(+), 2 deletions(-)

diff --git a/desktop/scripts/unopkg.sh b/desktop/scripts/unopkg.sh
index 4d5dd87..926244f 100644
--- a/desktop/scripts/unopkg.sh
+++ b/desktop/scripts/unopkg.sh
@@ -43,6 +43,29 @@ fi
 sd_prog=`pwd`
 cd "$sd_cwd"
 
+isshared=0
+for arg in $@
+do
+if [ "$arg" = "--shared" ]; then
+    isshared=1
+fi
+done
+if [ $isshared -eq 1 ]; then
+    echo $@ | grep -q env:JFW_PLUGIN_DO_NOT_CHECK_ACCESSIBILITY
+    if [ $? -ne 0 ]; then
+        set -- $@ '-env:JFW_PLUGIN_DO_NOT_CHECK_ACCESSIBILITY=1'
+    fi
+    echo $@ | grep -q env:UNO_JAVA_JFW_INSTALL_DATA
+    if [ $? -ne 0 -a -w $sd_prog/../share/config/javasettingsunopkginstall.xml ]; then
+        set -- $@ '-env:UNO_JAVA_JFW_INSTALL_DATA=$$ORIGIN/../share/config/javasettingsunopkginstall.xml'
+    fi
+    echo $@ | grep -q env:UserInstallation
+    if [ $? -ne 0 ]; then
+        INSTDIR=`mktemp -d -t unoinstall.XXXXXX`
+        set -- $@ '-env:UserInstallation=file://'$INSTDIR
+    fi
+fi
+
 #collect all bootstrap variables specified on the command line
 #so that they can be passed as arguments to javaldx later on
 for arg in $@
@@ -75,6 +98,8 @@ PATH=$sd_prog${PATH+:$PATH}
 export PATH
 
 # execute binary
-exec "$sd_prog/unopkg.bin" "$@" \
+"$sd_prog/unopkg.bin" "$@" \
     "-env:INIFILENAME=vnd.sun.star.pathname:$sd_prog/redirectrc"
-
+if [ -n "$INSTDIR" ]; then
+   rm -rf $INSTDIR
+fi
-- 
1.7.0.1

