--- sw/source/ui/uno/unotxdoc.cxx.old	2008-11-12 17:44:39.000000000 +0000
+++ sw/source/ui/uno/unotxdoc.cxx	2008-12-18 12:32:42.000000000 +0000
@@ -2808,12 +2816,15 @@
         // <--
 
         pVwSh->SetPDFExportOption( sal_False );
+        // #i96167# haggai: delete pViewOptionsAdjust here because it makes use
+        // of the shell, which might get destroyed in lcl_DisposeView!
+        delete pViewOptionAdjust;
+
         if( bLastPage && m_pHiddenViewFrame)
         {
             lcl_DisposeView( m_pHiddenViewFrame, pDocShell );
             m_pHiddenViewFrame = 0;
         }
-        delete pViewOptionAdjust;
     }
 }
 /* -----------------------------03.10.04 -------------------------------------
