--- psprint/source/fontmanager/fontconfig.cxx	10 Sep 2008 13:53:12 -0000	1.30.20.1
+++ psprint/source/fontmanager/fontconfig.cxx	19 Sep 2008 11:32:56 -0000
@@ -91,7 +91,6 @@
 class FontCfgWrapper
 {
     oslModule		m_pLib;
-    FcConfig*		m_pDefConfig;
     FcFontSet*      m_pOutlineSet;
 
     FcBool          (*m_pFcInit)();
@@ -140,7 +139,6 @@
     bool isValid() const 
     { return m_pLib != NULL;}
 
-    FcConfig* getDefConfig() { return m_pDefConfig; }
     FcFontSet* getFontSet();
     
     FcBool FcInit()
@@ -252,7 +250,6 @@
 
 FontCfgWrapper::FontCfgWrapper()
         : m_pLib( NULL ),
-          m_pDefConfig( NULL ),
           m_pOutlineSet( NULL )
 {
      OUString aLib( RTL_CONSTASCII_USTRINGPARAM( "libfontconfig.so.1" ) );
@@ -379,8 +376,7 @@
 
 
     FcInit();
-    m_pDefConfig = FcConfigGetCurrent();
-    if( ! m_pDefConfig )
+    if( ! FcConfigGetCurrent() )
     {
         osl_unloadModule( (oslModule)m_pLib );
         m_pLib = NULL;
@@ -394,7 +390,7 @@
       add only acceptable outlined fonts to our config, 
       for future fontconfig use
     */
-    FcFontSet* pOrig = FcConfigGetFonts( getDefConfig(), eSetName );
+    FcFontSet* pOrig = FcConfigGetFonts( FcConfigGetCurrent(), eSetName );
     if( !pOrig )
         return;
 
@@ -794,7 +790,7 @@
     if( nVersion <= 20400 )
         return false;
     const char* pDirName = (const char*)rDirName.getStr();
-    bool bRet = (rWrapper.FcConfigAppFontAddDir( rWrapper.getDefConfig(), (FcChar8*)pDirName ) == FcTrue);
+    bool bRet = (rWrapper.FcConfigAppFontAddDir( rWrapper.FcConfigGetCurrent(), (FcChar8*)pDirName ) == FcTrue);
 
 #if OSL_DEBUG_LEVEL > 1
     fprintf( stderr, "FcConfigAppFontAddDir( \"%s\") => %d\n", pDirName, bRet );
@@ -914,13 +910,13 @@
     addtopattern(rWrapper, pPattern, eItalic, eWeight, eWidth, ePitch);
 
     // query fontconfig for a substitute
-    rWrapper.FcConfigSubstitute( rWrapper.getDefConfig(), pPattern, FcMatchPattern );
+    rWrapper.FcConfigSubstitute( rWrapper.FcConfigGetCurrent(), pPattern, FcMatchPattern );
     rWrapper.FcDefaultSubstitute( pPattern );
 
     // process the result of the fontconfig query
     FcResult eResult = FcResultNoMatch;
     FcFontSet* pFontSet = rWrapper.getFontSet();
-    FcPattern* pResult = rWrapper.FcFontSetMatch( rWrapper.getDefConfig(), &pFontSet, 1, pPattern, &eResult );
+    FcPattern* pResult = rWrapper.FcFontSetMatch( rWrapper.FcConfigGetCurrent(), &pFontSet, 1, pPattern, &eResult );
     rWrapper.FcPatternDestroy( pPattern );
 
     FcFontSet*  pSet = NULL;
@@ -982,7 +978,7 @@
     if( ! rWrapper.isValid() )
         return false;
 
-    FcConfig* pConfig = rWrapper.getDefConfig();
+    FcConfig* pConfig = rWrapper.FcConfigGetCurrent();
     FcPattern* pPattern = rWrapper.FcPatternCreate();
 
     OString aLangAttrib;
