--- instsetoo_native/inc_openoffice/windows/msi_languages/Control.ulf
+++ instsetoo_native/inc_openoffice/windows/msi_languages/Control.ulf
@@ -284,7 +284,7 @@
 en-US = "Cancel"
 
 [OOO_CONTROL_124]
-en-US = "Build contributed in collaboration with the community by The Document Foundation. For credits, see: http://www.documentfoundation.org"
+en-US = "Build contributed in collaboration with the community by OxygenOffice Professional Team. For credits, see: http://www.documentfoundation.org"
 
 [OOO_CONTROL_125]
 en-US = "&Next >"
