---
 vcl/unx/gtk/a11y/atkutil.cxx |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git vcl/unx/gtk/a11y/atkutil.cxx vcl/unx/gtk/a11y/atkutil.cxx
index 1f0eca8..0d54d8e 100644
--- vcl/unx/gtk/a11y/atkutil.cxx
+++ vcl/unx/gtk/a11y/atkutil.cxx
@@ -624,6 +624,7 @@ static void handle_menu_highlighted(::VclMenuEvent const * pEvent)
 
 long WindowEventHandler(void *, ::VclSimpleEvent const * pEvent)
 {
+    try {
     switch (pEvent->GetId())
     {
     case VCLEVENT_WINDOW_SHOW:
@@ -708,6 +709,10 @@ long WindowEventHandler(void *, ::VclSimpleEvent const * pEvent)
 //        OSL_TRACE("got event %d \n", pEvent->GetId());
         break;
     }
+    } catch(lang::IndexOutOfBoundsException e)
+    {
+    g_warning("Focused object has invalid index in parent");
+    }
     return 0;
 }
 
-- 
1.7.0.1

