From ee40cb24852d2ecec942a560afd188b233d148e8 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:01:17 +0200
Subject: [PATCH 385/768] oooXXXXX.vcl.x86_64.impressatk.diff

---
 vcl/unx/gtk/a11y/atkutil.cxx |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git vcl/unx/gtk/a11y/atkutil.cxx vcl/unx/gtk/a11y/atkutil.cxx
index 1f0eca8..0d54d8e 100644
--- vcl/unx/gtk/a11y/atkutil.cxx
+++ vcl/unx/gtk/a11y/atkutil.cxx
@@ -624,6 +624,7 @@ static void handle_menu_highlighted(::VclMenuEvent const * pEvent)
 
 long WindowEventHandler(void *, ::VclSimpleEvent const * pEvent)
 {
+    try {
     switch (pEvent->GetId())
     {
     case VCLEVENT_WINDOW_SHOW:
@@ -708,6 +709,10 @@ long WindowEventHandler(void *, ::VclSimpleEvent const * pEvent)
 //        OSL_TRACE("got event %d \n", pEvent->GetId());
         break;
     }
+    } catch(lang::IndexOutOfBoundsException e)
+    {
+    g_warning("Focused object has invalid index in parent");
+    }
     return 0;
 }
 
-- 
1.7.0.1

