From 162d07130704b349dc3a3c74c69a55a2a4251dcd Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:01:21 +0200
Subject: [PATCH 388/768] oooXXXX.partial-revert-ooo95118.diff

---
 scripting/source/pyprov/pythonscript.py |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

diff --git scripting/source/pyprov/pythonscript.py scripting/source/pyprov/pythonscript.py
index 88e0a9e..f9f0408 100644
--- scripting/source/pyprov/pythonscript.py
+++ scripting/source/pyprov/pythonscript.py
@@ -817,8 +817,6 @@ def expandUri(  uri ):
         uri = uri.replace( "vnd.sun.star.expand:", "",1)
         uri = uno.getComponentContext().getByName(
                     "/singletons/com.sun.star.util.theMacroExpander" ).expandMacros( uri )
-    if uri.startswith( "file:" ):
-        uri = uno.absolutize("",uri)   # necessary to get rid of .. in uri
     return uri
     
 #--------------------------------------------------------------
-- 
1.7.0.1

