---
 oowintool |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git oowintool oowintool
index 0b4be43..8d2e475 100755
--- oowintool
+++ oowintool
@@ -265,6 +265,7 @@ sub print_jdk_dir()
 {
     my $dir =
 	  reg_get_value ("HKEY_LOCAL_MACHINE/SOFTWARE/JavaSoft/Java\ Development\ Kit/1.5/JavaHome") ||
+      reg_get_value ("HKEY_LOCAL_MACHINE/SOFTWARE/JavaSoft/Java\ Development\ Kit/1.6/JavaHome") ||
 	  reg_get_value ("HKEY_LOCAL_MACHINE/SOFTWARE/JavaSoft/Java\ Development\ Kit/1.4/JavaHome") ||
 	  reg_get_value ("HKEY_LOCAL_MACHINE/SOFTWARE/JavaSoft/Java\ Development\ Kit/1.3/JavaHome");
     print cygpath($dir, 'w', $output_format); 
-- 
1.7.0.1

