From e1d4dc046e90811eca5d0b8cd4bd1ba956b301bb Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:57:56 +0200
Subject: [PATCH 289/878] oox-tokenmap.string.h.diff

---
 oox/source/token/tokenmap.cxx |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/oox/source/token/tokenmap.cxx b/oox/source/token/tokenmap.cxx
index 339ff5c..cff214a 100644
--- a/oox/source/token/tokenmap.cxx
+++ b/oox/source/token/tokenmap.cxx
@@ -25,6 +25,7 @@
  *
  ************************************************************************/
 
+#include <string.h>
 #include "oox/token/tokenmap.hxx"
 #include <string.h>
 #include <rtl/strbuf.hxx>
-- 
1.7.0.1

