---
 oox/source/ppt/pptshape.cxx |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git oox/source/ppt/pptshape.cxx oox/source/ppt/pptshape.cxx
index 3b32443..48810e2 100644
--- oox/source/ppt/pptshape.cxx
+++ oox/source/ppt/pptshape.cxx
@@ -186,8 +186,9 @@ void PPTShape::addShape(
 
             if ( sServiceName.getLength() )
             {
+            // use style from master slide for placeholders only, otherwise use slide's style, which might be the default style from presentation
                 if ( !aMasterTextListStyle.get() )
-                    aMasterTextListStyle = rSlidePersist.getMasterPersist().get() ? rSlidePersist.getMasterPersist()->getOtherTextStyle() : rSlidePersist.getOtherTextStyle();
+                    aMasterTextListStyle = ( mnSubType && rSlidePersist.getMasterPersist().get() ) ? rSlidePersist.getMasterPersist()->getOtherTextStyle() : rSlidePersist.getOtherTextStyle();
                 setMasterTextListStyle( aMasterTextListStyle );
  
                 Reference< XShape > xShape( createAndInsert( rFilterBase, sServiceName, pTheme, rxShapes, pShapeRect, bClearText ) );
-- 
1.7.0.1

