From ed913ba50beed5f59df438ad3b453e61300ceb4e Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:00:42 +0200
Subject: [PATCH 361/768] oox-fix-list-style-apply.diff

---
 oox/source/ppt/pptshape.cxx |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git oox/source/ppt/pptshape.cxx oox/source/ppt/pptshape.cxx
index 8d31195..61678d1 100644
--- oox/source/ppt/pptshape.cxx
+++ oox/source/ppt/pptshape.cxx
@@ -164,8 +164,9 @@ void PPTShape::addShape(
 
             if ( sServiceName.getLength() )
             {
+            // use style from master slide for placeholders only, otherwise use slide's style, which might be the default style from presentation
                 if ( !aMasterTextListStyle.get() )
-                    aMasterTextListStyle = rSlidePersist.getMasterPersist().get() ? rSlidePersist.getMasterPersist()->getOtherTextStyle() : rSlidePersist.getOtherTextStyle();
+                    aMasterTextListStyle = ( mnSubType && rSlidePersist.getMasterPersist().get() ) ? rSlidePersist.getMasterPersist()->getOtherTextStyle() : rSlidePersist.getOtherTextStyle();
                 setMasterTextListStyle( aMasterTextListStyle );
  
                 Reference< XShape > xShape( createAndInsert( rFilterBase, sServiceName, pTheme, rxShapes, pShapeRect, bClearText ) );
-- 
1.7.0.1

