--- sc/source/filter/xlsx/xlsx-xestream.cxx.old	2010-03-03 17:00:07.000000000 +0100
+++ sc/source/filter/xlsx/xlsx-xestream.cxx	2010-03-03 17:00:08.000000000 +0100
@@ -58,6 +58,10 @@
 #include <formula/grammar.hxx>
 #include <oox/export/drawingml.hxx>
 
+#include <sfx2/docfile.hxx>
+#include <sfx2/objsh.hxx>
+#include <sfx2/app.hxx>
+
 #define DEBUG_XL_ENCRYPTION 0
 
 using ::com::sun::star::beans::PropertyValue;
@@ -1124,7 +1128,9 @@ bool XclExpXmlStream::exportDocument() t
 {
     ScDocShell* pShell = getDocShell();
     ScDocument* pDoc = pShell->GetDocument();
-    SotStorageRef rStorage = dynamic_cast <SotStorage*>( Reference<XStorage>( pShell->GetStorage() ).get() );
+    SfxMedium* pMedium = pShell->GetMedium();
+    SvStream* pMediumStream = pMedium->GetOutStream();
+    SotStorageRef rStorage = new SotStorage( pMediumStream, false );
 
     XclExpRootData aData( EXC_BIFF8, *pShell->GetMedium (), rStorage, *pDoc, RTL_TEXTENCODING_DONTKNOW );
     aData.meOutput = EXC_OUTPUT_XML_2007;
