diff -rup ../../build-bak/ooo310-m17/oox/source/drawingml/chart/datasourceconverter.cxx oox/source/drawingml/chart/datasourceconverter.cxx
--- ../../build-bak/ooo310-m17/oox/source/drawingml/chart/datasourceconverter.cxx	2009-08-24 16:45:12.000000000 +0200
+++ oox/source/drawingml/chart/datasourceconverter.cxx	2009-08-24 16:58:03.000000000 +0200
@@ -59,7 +59,7 @@ Reference< XDataSequence > DataSequenceC
     Reference< XDataSequence > xDataSeq;
     if( getChartConverter() )
     {
-        getChartConverter()->createDataSequence( getChartDocument()->getDataProvider(), mrModel );
+        xDataSeq = getChartConverter()->createDataSequence( getChartDocument()->getDataProvider(), mrModel );
 
         // set sequence role
         PropertySet aSeqProp( xDataSeq );
