diff -rup ../../build-orig/ooo-build-3.2.98.0/oox/inc/oox/drawingml/textbody.hxx oox/inc/oox/drawingml/textbody.hxx
--- ../../build-orig/ooo-build-3.2.98.0/oox/inc/oox/drawingml/textbody.hxx	2010-08-24 17:01:37.000000000 +0200
+++ oox/inc/oox/drawingml/textbody.hxx	2010-09-10 18:01:39.000000000 +0200
@@ -29,6 +29,7 @@
 #define OOX_DRAWINGML_TEXTBODY_HXX
 
 #include "oox/helper/containerhelper.hxx"
+#include "oox/drawingml/drawingmltypes.hxx"
 #include "oox/drawingml/textbodyproperties.hxx"
 #include "oox/drawingml/textliststyle.hxx"
 
@@ -48,6 +49,7 @@ class TextBody
 {
 public:
     TextBody();
+    TextBody( TextBodyPtr pBody );
     ~TextBody();
 
     inline const TextParagraphVector&   getParagraphs() const { return maParagraphs; }
diff -rup ../../build-orig/ooo-build-3.2.98.0/oox/source/drawingml/textbody.cxx oox/source/drawingml/textbody.cxx
--- ../../build-orig/ooo-build-3.2.98.0/oox/source/drawingml/textbody.cxx	2010-08-24 17:01:37.000000000 +0200
+++ oox/source/drawingml/textbody.cxx	2010-09-10 18:02:46.000000000 +0200
@@ -42,6 +42,14 @@ TextBody::TextBody()
 {
 }
 
+TextBody::TextBody( TextBodyPtr pBody )
+{
+    if( pBody.get() ) {
+        maTextProperties = pBody->maTextProperties;
+        maTextListStyle = pBody->maTextListStyle;
+    }
+}
+
 TextBody::~TextBody()
 {
 }
diff -rup ../../build-orig/ooo-build-3.2.98.0/oox/source/ppt/pptshapecontext.cxx oox/source/ppt/pptshapecontext.cxx
--- ../../build-orig/ooo-build-3.2.98.0/oox/source/ppt/pptshapecontext.cxx	2010-09-10 18:24:39.000000000 +0200
+++ oox/source/ppt/pptshapecontext.cxx	2010-09-10 18:03:15.000000000 +0200
@@ -203,7 +203,7 @@ Reference< XFastContextHandler > PPTShap
 
         case NMSP_PPT|XML_txBody:
         {
-            oox::drawingml::TextBodyPtr xTextBody( new oox::drawingml::TextBody );
+            oox::drawingml::TextBodyPtr xTextBody( new oox::drawingml::TextBody( mpShapePtr->getTextBody() ) );
             xTextBody->getTextProperties().maPropertyMap[ PROP_FontIndependentLineSpacing ] <<= static_cast< sal_Bool >( sal_True );
             mpShapePtr->setTextBody( xTextBody );
             xRet = new oox::drawingml::TextBodyContext( *this, *xTextBody );
