diff --git a/oox/source/drawingml/diagram/diagramfragmenthandler.cxx b/oox/source/drawingml/diagram/diagramfragmenthandler.cxx
index fca0382..9c7dce1 100644
--- oox/source/drawingml/diagram/diagramfragmenthandler.cxx
+++ oox/source/drawingml/diagram/diagramfragmenthandler.cxx
@@ -239,17 +239,17 @@ ColorFragmentHandler::ColorFragmentHandler( ::oox::core::XmlFilterBase& rFilter,
         // definitions in it, after all it's called list. but
         // apparently colorChoiceContext doesn't handle that anyway...
         case NMSP_DIAGRAM|XML_fillClrLst:
-            return new ColorContext( *this, maColorEntry.maFillColor );
+            return new ColorValueContext( *this, maColorEntry.maFillColor );
         case NMSP_DIAGRAM|XML_linClrLst:
-            return new ColorContext( *this, maColorEntry.maLineColor );
+            return new ColorValueContext( *this, maColorEntry.maLineColor );
         case NMSP_DIAGRAM|XML_effectClrLst:
-            return new ColorContext( *this, maColorEntry.maEffectColor );
+            return new ColorValueContext( *this, maColorEntry.maEffectColor );
         case NMSP_DIAGRAM|XML_txFillClrLst:
-            return new ColorContext( *this, maColorEntry.maTextFillColor );
+            return new ColorValueContext( *this, maColorEntry.maTextFillColor );
         case NMSP_DIAGRAM|XML_txLinClrLst:
-            return new ColorContext( *this, maColorEntry.maTextLineColor );
+            return new ColorValueContext( *this, maColorEntry.maTextLineColor );
         case NMSP_DIAGRAM|XML_txEffectClrLst:
-            return new ColorContext( *this, maColorEntry.maTextEffectColor );
+            return new ColorValueContext( *this, maColorEntry.maTextEffectColor );
     }
 
     return 0;
