From e66ac1bd86d27fc17985dac508bbee199848c922 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:09:36 +0200
Subject: [PATCH 766/768] oox-drawingml-fix-apply-shape-reference-crash.diff

---
 oox/source/drawingml/shape.cxx |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git oox/source/drawingml/shape.cxx oox/source/drawingml/shape.cxx
index 0f725d1..c80b357 100644
--- oox/source/drawingml/shape.cxx
+++ oox/source/drawingml/shape.cxx
@@ -168,6 +168,7 @@ void Shape::addShape(
 
 void Shape::applyShapeReference( const Shape& rReferencedShape )
 {
+    if( rReferencedShape.mpTextBody.get() )
     mpTextBody = TextBodyPtr( new TextBody( *rReferencedShape.mpTextBody.get() ) );
     maShapeProperties = rReferencedShape.maShapeProperties;
     mpLinePropertiesPtr = LinePropertiesPtr( new LineProperties( *rReferencedShape.mpLinePropertiesPtr.get() ) );
-- 
1.7.0.1

