diff -rup ../ooo320-m15-orig/oox-orig-2/source/drawingml/shape.cxx oox/source/drawingml/shape.cxx
--- ../ooo320-m15-orig/oox-orig-2/source/drawingml/shape.cxx	2010-04-22 11:06:57.000000000 +0200
+++ oox/source/drawingml/shape.cxx	2010-04-22 11:10:44.000000000 +0200
@@ -496,7 +496,6 @@ Reference< XShape > Shape::createAndInse
         PropertyMap aShapeProperties;
         PropertyMap::const_iterator	aShapePropIter;
 
-        aShapeProperties.insert( getShapeProperties().begin(), getShapeProperties().end() );
         if( mxCreateCallback.get() )
         {
             for ( aShapePropIter = mxCreateCallback->getShapeProperties().begin();
@@ -512,6 +511,7 @@ Reference< XShape > Shape::createAndInse
                 aShapeProperties[ (*aShapePropIter).first ] = (*aShapePropIter).second;		
         }
 
+        aShapeProperties.insert( getShapeProperties().begin(), getShapeProperties().end() );
         // applying properties
         PropertySet aPropSet( xSet );
         if ( aServiceName == OUString::createFromAscii( "com.sun.star.drawing.GraphicObjectShape" ) )
diff -rup ../ooo320-m15-orig/oox-orig-2/source/drawingml/textbodypropertiescontext.cxx oox/source/drawingml/textbodypropertiescontext.cxx
--- ../ooo320-m15-orig/oox-orig-2/source/drawingml/textbodypropertiescontext.cxx	2010-04-22 11:06:57.000000000 +0200
+++ oox/source/drawingml/textbodypropertiescontext.cxx	2010-04-22 11:07:59.000000000 +0200
@@ -66,21 +66,25 @@ TextBodyPropertiesContext::TextBodyPrope
     // ST_Coordinate
     OUString sValue;
     sValue = xAttributes->getOptionalValue( XML_lIns );
-    sal_Int32 nLeftInset = ( sValue.getLength() != 0 ? GetCoordinate(  sValue ) : 91440 / 360 );
-    mrTextBodyProp.maPropertyMap[ PROP_TextLeftDistance ]  <<= static_cast< sal_Int32 >( nLeftInset );
-
+    if( sValue.getLength() ) {
+	sal_Int32 nLeftInset = ( sValue.getLength() != 0 ? GetCoordinate(  sValue ) : 91440 / 360 );
+	mrTextBodyProp.maPropertyMap[ PROP_TextLeftDistance ]  <<= static_cast< sal_Int32 >( nLeftInset );
+    }
     sValue = xAttributes->getOptionalValue( XML_tIns );
-    sal_Int32 nTopInset  = ( sValue.getLength() != 0 ? GetCoordinate(  sValue ) : 91440 / 360 );
-    mrTextBodyProp.maPropertyMap[ PROP_TextUpperDistance ] <<= static_cast< sal_Int32 >( nTopInset );
-
+    if( sValue.getLength() ) {
+	sal_Int32 nTopInset  = ( sValue.getLength() != 0 ? GetCoordinate(  sValue ) : 91440 / 360 );
+	mrTextBodyProp.maPropertyMap[ PROP_TextUpperDistance ] <<= static_cast< sal_Int32 >( nTopInset );
+    }
     sValue = xAttributes->getOptionalValue( XML_rIns );
-    sal_Int32 nRightInset  = ( sValue.getLength() != 0 ? GetCoordinate(  sValue ) : 91440 / 360 );
-    mrTextBodyProp.maPropertyMap[ PROP_TextRightDistance ] <<= static_cast< sal_Int32 >( nRightInset );
-
+    if( sValue.getLength() ) {
+	sal_Int32 nRightInset  = ( sValue.getLength() != 0 ? GetCoordinate(  sValue ) : 91440 / 360 );
+	mrTextBodyProp.maPropertyMap[ PROP_TextRightDistance ] <<= static_cast< sal_Int32 >( nRightInset );
+    }
     sValue = xAttributes->getOptionalValue( XML_bIns );
-    sal_Int32 nBottonInset = ( sValue.getLength() != 0 ? GetCoordinate(  sValue ) : 45720 / 360 );
-    mrTextBodyProp.maPropertyMap[ PROP_TextLowerDistance ] <<= static_cast< sal_Int32 >( nBottonInset );
-
+    if( sValue.getLength() ) {
+	sal_Int32 nBottonInset = ( sValue.getLength() != 0 ? GetCoordinate(  sValue ) : 45720 / 360 );
+	mrTextBodyProp.maPropertyMap[ PROP_TextLowerDistance ] <<= static_cast< sal_Int32 >( nBottonInset );
+    }
 
     // ST_TextAnchoringType
     mrTextBodyProp.maPropertyMap[ PROP_TextVerticalAdjust ] <<= GetTextVerticalAdjust( xAttributes->getOptionalValueToken( XML_anchor, XML_t ) );
