diff -wrup oox-orig-2/source/ppt/pptshape.cxx oox/source/ppt/pptshape.cxx
--- oox-orig-2/source/ppt/pptshape.cxx	2010-04-21 15:27:28.000000000 +0200
+++ oox/source/ppt/pptshape.cxx	2010-04-21 15:34:33.000000000 +0200
@@ -99,6 +99,11 @@ void PPTShape::addShape(
                     {
                         if ( ( meShapeLocation == Master ) || ( meShapeLocation == Layout ) )
                             sServiceName = rtl::OUString();
+            else {
+                const rtl::OUString sTitleShapeService( RTL_CONSTASCII_USTRINGPARAM( "com.sun.star.presentation.SubtitleShape" ) );
+                sServiceName = sTitleShapeService;
+                aMasterTextListStyle = rSlidePersist.getMasterPersist().get() ? rSlidePersist.getMasterPersist()->getTitleTextStyle() : rSlidePersist.getTitleTextStyle();
+            }
                     }
                     break;
                     case XML_obj :
