--- sc/source/filter/excel/xestream.cxx.old	2010-03-03 16:59:16.000000000 +0100
+++ sc/source/filter/excel/xestream.cxx	2010-03-03 17:00:08.000000000 +0100
@@ -1011,10 +1011,9 @@ const oox::drawingml::table::TableStyleL
     return oox::drawingml::table::TableStyleListPtr();
 }
 
-oox::drawingml::chart::ChartConverter& XclExpXmlStream::getChartConverter()
+oox::drawingml::chart::ChartConverter* XclExpXmlStream::getChartConverter()
 {
-    // DO NOT CALL
-    return * (oox::drawingml::chart::ChartConverter*) NULL;
+    return NULL;
 }
 
 bool XclExpXmlStream::exportDocument() throw()
