diff -rup oox-orig/inc/oox/drawingml/drawingmltypes.hxx oox/inc/oox/drawingml/drawingmltypes.hxx
--- oox-orig/inc/oox/drawingml/drawingmltypes.hxx	2009-10-12 14:39:27.000000000 +0200
+++ oox/inc/oox/drawingml/drawingmltypes.hxx	2009-10-12 14:39:38.000000000 +0200
@@ -121,6 +121,7 @@ float GetTextSize( const ::rtl::OUString
 
 /** converts the ST_TextSpacingPoint to 1/100mm */
 sal_Int32 GetTextSpacingPoint(  const ::rtl::OUString& sValue );
+sal_Int32 GetTextSpacingPoint(  const sal_Int32 nValue );
 
 /** */
 ::com::sun::star::style::TabAlign GetTabAlign( ::sal_Int32 aToken );
diff -rup oox-orig/inc/oox/drawingml/textcharacterproperties.hxx oox/inc/oox/drawingml/textcharacterproperties.hxx
--- oox-orig/inc/oox/drawingml/textcharacterproperties.hxx	2009-10-12 14:39:27.000000000 +0200
+++ oox/inc/oox/drawingml/textcharacterproperties.hxx	2009-10-12 14:39:38.000000000 +0200
@@ -55,6 +55,7 @@ struct TextCharacterProperties
     Color               maHighlightColor;
     OptValue< ::rtl::OUString > moLang;
     OptValue< sal_Int32 > moHeight;
+    OptValue< sal_Int32 > moSpacing;
     OptValue< sal_Int32 > moUnderline;
     OptValue< sal_Int32 > moStrikeout;
     OptValue< sal_Int32 > moCaseMap;
diff -rup oox-orig/source/drawingml/drawingmltypes.cxx oox/source/drawingml/drawingmltypes.cxx
--- oox-orig/source/drawingml/drawingmltypes.cxx	2009-10-12 14:39:27.000000000 +0200
+++ oox/source/drawingml/drawingmltypes.cxx	2009-10-12 14:39:38.000000000 +0200
@@ -128,10 +128,14 @@ sal_Int32 GetTextSpacingPoint( const OUS
 {
     sal_Int32 nRet;
     if( ::sax::Converter::convertNumber( nRet, sValue ) )
-        nRet = ( nRet * 254 + 360 ) / 720;
+        nRet = GetTextSpacingPoint( nRet );
     return nRet;
 }
 
+sal_Int32 GetTextSpacingPoint( const sal_Int32 nValue )
+{
+    return ( nValue * 254 + 360 ) / 720;
+}
 
 float GetFontHeight( sal_Int32 nHeight )
 {
diff -rup oox-orig/source/drawingml/textcharacterpropertiescontext.cxx oox/source/drawingml/textcharacterpropertiescontext.cxx
--- oox-orig/source/drawingml/textcharacterpropertiescontext.cxx	2009-10-12 14:39:27.000000000 +0200
+++ oox/source/drawingml/textcharacterpropertiescontext.cxx	2009-10-12 14:39:38.000000000 +0200
@@ -61,6 +61,8 @@ TextCharacterPropertiesContext::TextChar
     mrTextCharacterProperties.moLang        = aAttribs.getString( XML_lang );
     if ( aAttribs.hasAttribute( XML_sz ) )
     mrTextCharacterProperties.moHeight      = aAttribs.getInteger( XML_sz );
+    if ( aAttribs.hasAttribute( XML_spc ) )
+    mrTextCharacterProperties.moSpacing     = aAttribs.getInteger( XML_spc );
     if ( aAttribs.hasAttribute( XML_u ) )
     mrTextCharacterProperties.moUnderline   = aAttribs.getToken( XML_u );
     if ( aAttribs.hasAttribute( XML_strike ) )
diff -rup oox-orig/source/drawingml/textcharacterproperties.cxx oox/source/drawingml/textcharacterproperties.cxx
--- oox-orig/source/drawingml/textcharacterproperties.cxx	2009-10-12 14:39:27.000000000 +0200
+++ oox/source/drawingml/textcharacterproperties.cxx	2009-10-12 14:40:54.000000000 +0200
@@ -62,6 +62,7 @@ void TextCharacterProperties::assignUsed
     maHighlightColor.assignIfUsed( rSourceProps.maHighlightColor );
     maUnderlineColor.assignIfUsed( rSourceProps.maUnderlineColor );
     moHeight.assignIfUsed( rSourceProps.moHeight );
+    moSpacing.assignIfUsed( rSourceProps.moSpacing );
     moUnderline.assignIfUsed( rSourceProps.moUnderline );
     moStrikeout.assignIfUsed( rSourceProps.moStrikeout );
     moCaseMap.assignIfUsed( rSourceProps.moCaseMap );
@@ -129,6 +130,8 @@ void TextCharacterProperties::pushToProp
         rPropMap[ PROP_CharHeightComplex ] <<= fHeight;
     }
 
+    rPropMap[ PROP_CharKerning ] <<= (sal_Int16) GetTextSpacingPoint( moSpacing.get( 0 ) );
+
     rPropMap[ PROP_CharUnderline ] <<= GetFontUnderline( moUnderline.get( XML_none ) );
     rPropMap[ PROP_CharStrikeout ] <<= GetFontStrikeout( moStrikeout.get( XML_noStrike ) );
     rPropMap[ PROP_CharCaseMap ] <<= GetCaseMap( moCaseMap.get( XML_none ) );
diff -rup oox-orig/source/token/properties.txt oox/source/token/properties.txt
--- oox-orig/source/token/properties.txt	2009-10-12 14:39:27.000000000 +0200
+++ oox/source/token/properties.txt	2009-10-12 14:41:22.000000000 +0200
@@ -58,6 +58,7 @@ CharFontPitchComplex
 CharHeight
 CharHeightAsian
 CharHeightComplex
+CharKerning
 CharLocale
 CharLocaleAsian
 CharLocaleComplex
