diff -rup oox/source/ppt-orig/pptshape.cxx oox/source/ppt/pptshape.cxx
--- oox/source/ppt-orig/pptshape.cxx	2010-06-16 15:04:08.000000000 +0200
+++ oox/source/ppt/pptshape.cxx	2010-06-16 15:04:33.000000000 +0200
@@ -84,7 +84,8 @@ void PPTShape::addShape(
             Reference< lang::XMultiServiceFactory > xServiceFact( rFilterBase.getModel(), UNO_QUERY_THROW );
             sal_Bool bClearText = sal_False;
 
-            if ( sServiceName != OUString::createFromAscii( "com.sun.star.drawing.GraphicObjectShape" ) )
+            if ( sServiceName != OUString::createFromAscii( "com.sun.star.drawing.GraphicObjectShape" ) &&
+                 sServiceName != OUString::createFromAscii( "com.sun.star.drawing.OLE2Shape" ) )
             {
                 switch( mnSubType )
                 {
