diff -rup oox-orig/source/drawingml/shape.cxx oox/source/drawingml/shape.cxx
--- oox-orig/source/drawingml/shape.cxx	2009-10-23 14:52:57.000000000 +0200
+++ oox/source/drawingml/shape.cxx	2009-10-23 15:05:01.000000000 +0200
@@ -442,7 +442,6 @@ Reference< XShape > Shape::createAndInse
         aFillProperties.assignUsed( getFillProperties() );
 
         PropertyMap aShapeProperties;
-        aShapeProperties.insert( getShapeProperties().begin(), getShapeProperties().end() );
         if( mxCreateCallback.get() )
             aShapeProperties.insert( mxCreateCallback->getShapeProperties().begin(), mxCreateCallback->getShapeProperties().end() );
 
@@ -450,6 +449,7 @@ Reference< XShape > Shape::createAndInse
         if( mpTextBody.get() )
             aShapeProperties.insert( mpTextBody->getTextProperties().maPropertyMap.begin(), mpTextBody->getTextProperties().maPropertyMap.end() );
 
+        aShapeProperties.insert( getShapeProperties().begin(), getShapeProperties().end() );
         // applying properties
         PropertySet aPropSet( xSet );
         if ( aServiceName == OUString::createFromAscii( "com.sun.star.drawing.GraphicObjectShape" ) )
diff -rup oox-orig/source/drawingml/textbodypropertiescontext.cxx oox/source/drawingml/textbodypropertiescontext.cxx
--- oox-orig/source/drawingml/textbodypropertiescontext.cxx	2009-10-23 14:52:57.000000000 +0200
+++ oox/source/drawingml/textbodypropertiescontext.cxx	2009-10-23 14:58:58.000000000 +0200
@@ -67,21 +67,25 @@ TextBodyPropertiesContext::TextBodyPrope
     // ST_Coordinate
     OUString sValue;
     sValue = xAttributes->getOptionalValue( XML_lIns );
-    sal_Int32 nLeftInset = ( sValue.getLength() != 0 ? GetCoordinate(  sValue ) : 91440 / 360 );
-    mrTextBodyProp.maPropertyMap[ PROP_TextLeftDistance ]  <<= static_cast< sal_Int32 >( nLeftInset );
-
+    if( sValue.getLength() ) {
+	sal_Int32 nLeftInset = ( sValue.getLength() != 0 ? GetCoordinate(  sValue ) : 91440 / 360 );
+	mrTextBodyProp.maPropertyMap[ PROP_TextLeftDistance ]  <<= static_cast< sal_Int32 >( nLeftInset );
+    }
     sValue = xAttributes->getOptionalValue( XML_tIns );
-    sal_Int32 nTopInset  = ( sValue.getLength() != 0 ? GetCoordinate(  sValue ) : 91440 / 360 );
-    mrTextBodyProp.maPropertyMap[ PROP_TextUpperDistance ] <<= static_cast< sal_Int32 >( nTopInset );
-
+    if( sValue.getLength() ) {
+	sal_Int32 nTopInset  = ( sValue.getLength() != 0 ? GetCoordinate(  sValue ) : 91440 / 360 );
+	mrTextBodyProp.maPropertyMap[ PROP_TextUpperDistance ] <<= static_cast< sal_Int32 >( nTopInset );
+    }
     sValue = xAttributes->getOptionalValue( XML_rIns );
-    sal_Int32 nRightInset  = ( sValue.getLength() != 0 ? GetCoordinate(  sValue ) : 91440 / 360 );
-    mrTextBodyProp.maPropertyMap[ PROP_TextRightDistance ] <<= static_cast< sal_Int32 >( nRightInset );
-
+    if( sValue.getLength() ) {
+	sal_Int32 nRightInset  = ( sValue.getLength() != 0 ? GetCoordinate(  sValue ) : 91440 / 360 );
+	mrTextBodyProp.maPropertyMap[ PROP_TextRightDistance ] <<= static_cast< sal_Int32 >( nRightInset );
+    }
     sValue = xAttributes->getOptionalValue( XML_bIns );
-    sal_Int32 nBottonInset = ( sValue.getLength() != 0 ? GetCoordinate(  sValue ) : 45720 / 360 );
-    mrTextBodyProp.maPropertyMap[ PROP_TextLowerDistance ] <<= static_cast< sal_Int32 >( nBottonInset );
-
+    if( sValue.getLength() ) {
+	sal_Int32 nBottonInset = ( sValue.getLength() != 0 ? GetCoordinate(  sValue ) : 45720 / 360 );
+	mrTextBodyProp.maPropertyMap[ PROP_TextLowerDistance ] <<= static_cast< sal_Int32 >( nBottonInset );
+    }
 
     // ST_TextAnchoringType
     mrTextBodyProp.maPropertyMap[ PROP_TextVerticalAdjust ] <<= GetTextVerticalAdjust( xAttributes->getOptionalValueToken( XML_anchor, XML_t ) );
