From 26b2f5c39ef341babd31def6c3a29b04ecdcff99 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:09:33 +0200
Subject: [PATCH 763/768] oox-pptx-import-fix-text-body-vert-2.diff

---
 oox/source/drawingml/textbodypropertiescontext.cxx |   16 +++++++++-------
 1 files changed, 9 insertions(+), 7 deletions(-)

diff --git oox/source/drawingml/textbodypropertiescontext.cxx oox/source/drawingml/textbodypropertiescontext.cxx
index 0d28569..a557145 100644
--- oox/source/drawingml/textbodypropertiescontext.cxx
+++ oox/source/drawingml/textbodypropertiescontext.cxx
@@ -132,13 +132,15 @@ TextBodyPropertiesContext::TextBodyPropertiesContext( ContextHandler& rParent,
     bool bRtl = aAttribs.getBool( XML_rtl, false );
     sal_Int32 tVert = mrTextBodyProp.moVert.get( XML_horz );
     if( tVert == XML_vert || tVert == XML_eaVert || tVert == XML_vert270 || tVert == XML_mongolianVert ) {
-    mrTextBodyProp.maPropertyMap[ PROP_TextWritingMode ]
-        <<= ( bRtl ? WritingMode_RL_TB : WritingMode_LR_TB );
-    // workaround for TB_LR as using WritingMode2 doesn't work
-    if( !bAnchorCenter )
-        mrTextBodyProp.maPropertyMap[ PROP_TextHorizontalAdjust ] <<=
-        TextHorizontalAdjust_LEFT;
-    }
+      mrTextBodyProp.maPropertyMap[ PROP_TextWritingMode ]
+    <<= WritingMode_TB_RL;
+      // workaround for TB_LR as using WritingMode2 doesn't work
+        if( !bAnchorCenter )
+            mrTextBodyProp.maPropertyMap[ PROP_TextHorizontalAdjust ] <<=
+            TextHorizontalAdjust_LEFT;
+    } else
+      mrTextBodyProp.maPropertyMap[ PROP_TextWritingMode ]
+    <<= ( bRtl ? WritingMode_RL_TB : WritingMode_LR_TB );
 }
 
 // --------------------------------------------------------------------
-- 
1.7.0.1

