---
 .../drawingml/textcharacterpropertiescontext.cxx   |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git oox/source/drawingml/textcharacterpropertiescontext.cxx oox/source/drawingml/textcharacterpropertiescontext.cxx
index c74dba5..9fa4851 100644
--- oox/source/drawingml/textcharacterpropertiescontext.cxx
+++ oox/source/drawingml/textcharacterpropertiescontext.cxx
@@ -60,6 +60,8 @@ TextCharacterPropertiesContext::TextCharacterPropertiesContext(
         mrTextCharacterProperties.moLang = aAttribs.getString( XML_lang );
     if ( aAttribs.hasAttribute( XML_sz ) )
         mrTextCharacterProperties.moHeight = aAttribs.getInteger( XML_sz );
+    if ( aAttribs.hasAttribute( XML_spc ) )
+        mrTextCharacterProperties.moSpacing = aAttribs.getInteger( XML_spc );
     if ( aAttribs.hasAttribute( XML_u ) )
         mrTextCharacterProperties.moUnderline = aAttribs.getToken( XML_u );
     if ( aAttribs.hasAttribute( XML_strike ) )
-- 
1.7.0.1

