diff -rup sc-orig/source/filter/xcl97/xcl97rec.cxx sc/source/filter/xcl97/xcl97rec.cxx
--- sc-orig/source/filter/xlsx/xlsx-xcl97rec.cxx	2009-07-14 17:32:13.000000000 +0200
+++ sc/source/filter/xlsx/xlsx-xcl97rec.cxx	2009-07-14 17:35:50.000000000 +0200
@@ -1162,7 +1162,7 @@ void XclObjAny::SaveXml( XclExpXmlStream
 
     sax_fastparser::FSHelperPtr pDrawing = rStrm.GetCurrentStream();
 
-    ShapeExport aDML( XML_xdr, pDrawing, &rStrm, DrawingML::DOCUMENT_XLSX );
+    ShapeExport aDML( XML_xdr, pDrawing, NULL, &rStrm, DrawingML::DOCUMENT_XLSX );
 
     pDrawing->startElement( FSNS( XML_xdr, XML_twoCellAnchor ), // OOXTODO: oneCellAnchor, absoluteAnchor
             XML_editAs, GetEditAs( *this ),
