diff -rup oox-orig/inc/oox/ppt/slidetransition.hxx oox/inc/oox/ppt/slidetransition.hxx
--- oox-orig/inc/oox/ppt/slidetransition.hxx	2009-10-05 17:12:59.000000000 +0200
+++ oox/inc/oox/ppt/slidetransition.hxx	2009-10-05 17:13:29.000000000 +0200
@@ -54,6 +54,8 @@ namespace oox { namespace ppt {
             { mnFadeColor = nColor; }
         void setMode( sal_Bool bMode )
             { mbMode = bMode; }
+	        void setOoxAdvanceTime( sal_Int32 nAdvanceTime )
+		        { mnAdvanceTime = nAdvanceTime; }
 
     static sal_Int16 ooxToOdpDirection( ::sal_Int32 nOoxType );
     static sal_Int16 ooxToOdpEightDirections( ::sal_Int32 nOoxType );
@@ -71,6 +73,7 @@ namespace oox { namespace ppt {
         ::sal_Int16 mnAnimationSpeed;
         ::sal_Int32 mnFadeColor;
         ::sal_Bool  mbMode; /**< http://api.openoffice.org/docs/common/ref/com/sun/star/animations/XTransitionFilter.html Mode property */
+		::sal_Int32 mnAdvanceTime;
     };
 
 } }
diff -rup oox-orig/source/ppt/slidetransitioncontext.cxx oox/source/ppt/slidetransitioncontext.cxx
--- oox-orig/source/ppt/slidetransitioncontext.cxx	2009-10-05 17:13:00.000000000 +0200
+++ oox/source/ppt/slidetransitioncontext.cxx	2009-10-05 17:13:29.000000000 +0200
@@ -70,12 +70,9 @@ SlideTransitionContext::SlideTransitionC
     attribs.getBool( XML_advClick, true );
 
     // careful. if missing, no auto advance... 0 looks like a valid value
-  // for auto advance
+	// for auto advance
     if(attribs.hasAttribute( XML_advTm ))
-	{
-		// TODO
-		xAttribs->getOptionalValue( XML_advTm );
-	}
+	    maTransition.setOoxAdvanceTime( attribs.getInteger( XML_advTm, -1 ) );
 }
 
 SlideTransitionContext::~SlideTransitionContext() throw()
diff -rup oox-orig/source/ppt/slidetransition.cxx oox/source/ppt/slidetransition.cxx
--- oox-orig/source/ppt/slidetransition.cxx	2009-10-05 17:13:00.000000000 +0200
+++ oox/source/ppt/slidetransition.cxx	2009-10-05 17:17:56.000000000 +0200
@@ -59,6 +59,7 @@ namespace oox { namespace ppt {
         , mnAnimationSpeed( AnimationSpeed_FAST )
         , mnFadeColor( 0 )
         , mbMode( true )
+		, mnAdvanceTime( -1 )
     {
 
     }
@@ -71,6 +72,7 @@ namespace oox { namespace ppt {
         , mnAnimationSpeed( AnimationSpeed_FAST )
         , mnFadeColor( 0 )
         , mbMode( true )
+		, mnAdvanceTime( -1 )
     {
         const transition *p = transition::find( sFilterName );
         if( p )
@@ -91,6 +93,10 @@ namespace oox { namespace ppt {
             aProps[ PROP_TransitionDirection ] <<= mbTransitionDirectionNormal;
             aProps[ PROP_Speed ] <<= mnAnimationSpeed;
             aProps[ PROP_TransitionFadeColor ] <<= mnFadeColor;
+	    if( mnAdvanceTime != -1 ) {
+		aProps[ PROP_Duration ] <<= mnAdvanceTime/1000;
+		aProps[ PROP_Change ] <<= static_cast<sal_Int32>(1);
+	    }
         }
         catch( Exception& )
         {
@@ -141,8 +147,6 @@ namespace oox { namespace ppt {
         }
     }
 
-
-
     sal_Int16 SlideTransition::ooxToOdpEightDirections( ::sal_Int32 nOoxType )
     {
     sal_Int16 nOdpDirection;
diff -rup oox-orig/source/token/properties.txt oox/source/token/properties.txt
--- oox-orig/source/token/properties.txt	2009-10-05 17:13:00.000000000 +0200
+++ oox/source/token/properties.txt	2009-10-05 17:16:55.000000000 +0200
@@ -37,6 +37,7 @@ CellProtection
 CellStyle
 CenterHorizontally
 CenterVertically
+Change
 CharCaseMap
 CharColor
 CharContoured
@@ -106,6 +107,7 @@ DiagonalTLBR
 DisplayLabels
 DrillDownOnDoubleClick
 Dropdown
+Duration
 EchoChar
 Enabled
 EndPosition
