diff -rup oox-orig/source/ppt/slidefragmenthandler.cxx oox/source/ppt/slidefragmenthandler.cxx
--- oox-orig/source/ppt/slidefragmenthandler.cxx	2009-10-15 11:54:42.000000000 +0200
+++ oox/source/ppt/slidefragmenthandler.cxx	2009-10-15 12:20:25.000000000 +0200
@@ -35,6 +35,7 @@
 #include <com/sun/star/container/XNamed.hpp>
 
 #include "tokens.hxx"
+#include "properties.hxx"
 #include "oox/helper/propertyset.hxx"
 #include "oox/core/namespaces.hxx"
 #include "oox/core/xmlfilterbase.hxx"
@@ -89,6 +90,18 @@ Reference< XFastContextHandler > SlideFr
     case NMSP_PPT|XML_sldMaster:		// CT_SlideMaster
     case NMSP_PPT|XML_handoutMaster:	// CT_HandoutMaster
     case NMSP_PPT|XML_sld:				// CT_CommonSlideData
+	{
+	    AttributeList attribs( xAttribs );
+
+	    Reference< XDrawPage > xSlide( mpSlidePersistPtr->getPage() );
+	    PropertyMap aPropMap;
+	    PropertySet aSlideProp( xSlide );
+
+	    aPropMap[ PROP_Visible ] = Any( attribs.getBool( XML_show, sal_True ) );
+	    aSlideProp.setProperties( aPropMap );
+
+	    break;
+	}
     case NMSP_PPT|XML_notes:			// CT_NotesSlide
     case NMSP_PPT|XML_notesMaster:		// CT_NotesMaster
         break;
diff -rup oox-orig/source/token/properties.txt oox/source/token/properties.txt
--- oox-orig/source/token/properties.txt	2009-10-15 11:54:42.000000000 +0200
+++ oox/source/token/properties.txt	2009-10-15 12:19:51.000000000 +0200
@@ -400,6 +400,7 @@ VertJustify
 VerticalAlign
 VerticalSplitMode
 VerticalSplitPositionTwips
+Visible
 VisibleSize
 VisualArea
 VisualEffect
