---
 oox/source/token/tokenmap.cxx |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git oox/source/token/tokenmap.cxx oox/source/token/tokenmap.cxx
index 17979d7..a3dac5c 100644
--- oox/source/token/tokenmap.cxx
+++ oox/source/token/tokenmap.cxx
@@ -25,6 +25,7 @@
  *
  ************************************************************************/
 
+#include <string.h>
 #include "oox/token/tokenmap.hxx"
 #include <string.h>
 #include <rtl/strbuf.hxx>
-- 
1.7.0.1

