diff -rup oox/source/export-orig/drawingml.cxx oox/source/export/drawingml.cxx
--- oox/source/export-orig/drawingml.cxx	2009-07-29 14:40:11.000000000 +0200
+++ oox/source/export/drawingml.cxx	2009-07-29 15:47:21.000000000 +0200
@@ -567,13 +567,25 @@ OUString DrawingML::WriteImage( const Gr
     return sRelId;
 }
 
-OUString DrawingML::WriteBlip( OUString& rURL )
+OUString DrawingML::WriteBlip( Reference< XPropertySet > rXPropSet, OUString& rURL )
 {
         OUString sRelId = WriteImage( rURL );
+    sal_Int16 nBright = 0;
+    sal_Int32 nContrast = 0;
 
-        mpFS->singleElementNS( XML_a, XML_blip,
-                               FSNS( XML_r, XML_embed), OUStringToOString( sRelId, RTL_TEXTENCODING_UTF8 ).getStr(),
-                               FSEND );
+    GET( nBright, AdjustLuminance );
+    GET( nContrast, AdjustContrast );
+
+        mpFS->startElementNS( XML_a, XML_blip,
+                  FSNS( XML_r, XML_embed), OUStringToOString( sRelId, RTL_TEXTENCODING_UTF8 ).getStr(),
+                  FSEND );
+    if( nBright || nContrast )
+        mpFS->singleElementNS( XML_a, XML_lum,
+                   XML_bright, nBright ? I32S( nBright*1000 ) : NULL,
+                   XML_contrast, nContrast ? I32S( nContrast*1000 ) : NULL,
+                   FSEND );
+        
+        mpFS->endElementNS( XML_a, XML_blip );
 
         return sRelId;
 }
@@ -613,7 +625,7 @@ void DrawingML::WriteBlipFill( Reference
 
         mpFS->startElementNS( nXmlNamespace , XML_blipFill, FSEND );
 
-        WriteBlip( aURL );
+        WriteBlip( rXPropSet, aURL );
 
         if( sURLPropName == S( "FillBitmapURL" ) )
             WriteBlipMode( rXPropSet );
diff -rup oox/source/export-orig/shapes.cxx oox/source/export/shapes.cxx
--- oox/source/export-orig/shapes.cxx	2009-07-29 14:40:12.000000000 +0200
+++ oox/source/export/shapes.cxx	2009-07-29 14:59:08.000000000 +0200
@@ -652,7 +652,7 @@ ShapeExport& ShapeExport::WriteGraphicOb
 
     pFS->startElementNS( mnXmlNamespace, XML_blipFill, FSEND );
     
-    WriteBlip( sGraphicURL );
+    WriteBlip( xShapeProps, sGraphicURL );
 
     bool bStretch = false;
     if( ( xShapeProps->getPropertyValue( S( "FillBitmapStretch" ) ) >>= bStretch ) && bStretch )
diff -rup oox/inc/oox/export-orig/drawingml.hxx oox/inc/oox/export/drawingml.hxx
--- oox/inc/oox/export-orig/drawingml.hxx	2009-07-29 14:56:39.000000000 +0200
+++ oox/inc/oox/export/drawingml.hxx	2009-07-29 14:57:36.000000000 +0200
@@ -81,7 +81,7 @@ public:
     void WriteStretch();
     void WriteLinespacing( ::com::sun::star::style::LineSpacing& rLineSpacing );
 
-    ::rtl::OUString WriteBlip( ::rtl::OUString& rURL );
+    ::rtl::OUString WriteBlip( ::com::sun::star::uno::Reference< ::com::sun::star::beans::XPropertySet > rXPropSet, ::rtl::OUString& rURL );
     void WriteBlipMode( ::com::sun::star::uno::Reference< ::com::sun::star::beans::XPropertySet > rXPropSet );
 
     void WriteShapeTransformation( ::com::sun::star::uno::Reference< ::com::sun::star::drawing::XShape > rXShape,
