diff -rup oox-orig/inc/oox/drawingml/customshapeproperties.hxx oox/inc/oox/drawingml/customshapeproperties.hxx
--- oox-orig/inc/oox/drawingml/customshapeproperties.hxx	2010-05-25 16:55:39.000000000 +0200
+++ oox/inc/oox/drawingml/customshapeproperties.hxx	2010-05-25 16:56:11.000000000 +0200
@@ -74,6 +74,7 @@ public:
             const ::com::sun::star::uno::Reference < ::com::sun::star::beans::XPropertySet > & xPropSet,
                         const ::com::sun::star::uno::Reference < ::com::sun::star::drawing::XShape > & xShape) const;
 
+    const rtl::OUString getShapePresetType() const { return maShapePresetType; }
     void setShapePresetType( const rtl::OUString& rShapePresetType ){ maShapePresetType = rShapePresetType; };
     std::vector< CustomShapeGuide >& getAdjustmentValues(){ return maAdjustmentValues; };
 
diff -rup oox-orig/inc/oox/drawingml/shape.hxx oox/inc/oox/drawingml/shape.hxx
--- oox-orig/inc/oox/drawingml/shape.hxx	2010-05-25 16:55:39.000000000 +0200
+++ oox/inc/oox/drawingml/shape.hxx	2010-05-25 16:56:11.000000000 +0200
@@ -160,6 +160,11 @@ public:
 
     virtual void        applyShapeReference( const Shape& rReferencedShape );
 
+    const ::com::sun::star::awt::Size&
+                        getSize() const { return maSize; }
+    const ::com::sun::star::awt::Point&
+                        getPosition() const { return maPosition; }
+
 protected:
 
     ::com::sun::star::uno::Reference< ::com::sun::star::drawing::XShape >
diff -rup oox-orig/inc/oox/export/drawingml.hxx oox/inc/oox/export/drawingml.hxx
--- oox-orig/inc/oox/export/drawingml.hxx	2010-05-25 16:55:39.000000000 +0200
+++ oox/inc/oox/export/drawingml.hxx	2010-05-25 16:56:11.000000000 +0200
@@ -7,7 +7,7 @@
 #include <com/sun/star/awt/FontDescriptor.hpp>
 #include <com/sun/star/uno/XReference.hpp>
 #include <tools/poly.hxx>
-#include <svx/escherex.hxx>
+#include <filter/msfilter/escherex.hxx>
 
 class Graphic;
 class String;
diff -rup oox-orig/inc/oox/export/vmlexport.hxx oox/inc/oox/export/vmlexport.hxx
--- oox-orig/inc/oox/export/vmlexport.hxx	2010-05-25 16:55:39.000000000 +0200
+++ oox/inc/oox/export/vmlexport.hxx	2010-05-25 16:56:11.000000000 +0200
@@ -27,7 +27,7 @@
 
 #include <oox/dllapi.h>
 #include <sax/fshelper.hxx>
-#include <svx/escherex.hxx>
+#include <filter/msfilter/escherex.hxx>
 
 namespace rtl {
     class OString;
diff -rup oox-orig/prj/build.lst oox/prj/build.lst
--- oox-orig/prj/build.lst	2010-05-25 16:55:39.000000000 +0200
+++ oox/prj/build.lst	2010-05-25 16:56:11.000000000 +0200
@@ -1,4 +1,4 @@
-oox	oox : vos cppu cppuhelper comphelper sal offapi sax basegfx svx tools vcl unotools BOOST:boost OPENSSL:openssl NULL
+oox	oox : vos cppu cppuhelper comphelper sal offapi sax basegfx svx tools vcl filter unotools BOOST:boost OPENSSL:openssl NULL
 oox	oox				usr1	-   all	oox_mkout NULL
 oox	oox\prj				get	-   all	oox_prj NULL
 oox	oox\source\token		nmake	-   all	oox_token NULL
diff -rup oox-orig/source/export/drawingml.cxx oox/source/export/drawingml.cxx
--- oox-orig/source/export/drawingml.cxx	2010-05-25 16:55:39.000000000 +0200
+++ oox/source/export/drawingml.cxx	2010-05-25 16:56:50.000000000 +0200
@@ -65,13 +65,13 @@
 #include <tools/string.hxx>
 #include <vcl/cvtgrf.hxx>
 #include <unotools/fontcvt.hxx>
+#include <unotools/fontdefs.hxx>
 #include <vcl/graph.hxx>
 #include <svtools/grfmgr.hxx>
 #include <rtl/strbuf.hxx>
 #include <sfx2/app.hxx>
 #include <svl/languageoptions.hxx>
-#include <svx/escherex.hxx>
-#include <svx/svxenum.hxx>
+#include <editeng/svxenum.hxx>
 
 using namespace ::com::sun::star;
 using namespace ::com::sun::star::beans;
@@ -520,7 +520,7 @@ OUString DrawingML::WriteImage( const Gr
             }
 
             aData = aStream.GetData();
-            nDataSize = aStream.GetSize();
+            nDataSize = aStream.GetEndOfData();
             break;
             }
     }
diff -rup oox-orig/source/export/preset-definitions-to-shape-types.pl oox/source/export/preset-definitions-to-shape-types.pl
--- oox-orig/source/export/preset-definitions-to-shape-types.pl	2010-05-25 16:55:39.000000000 +0200
+++ oox/source/export/preset-definitions-to-shape-types.pl	2010-05-25 16:56:11.000000000 +0200
@@ -1204,7 +1204,7 @@ print <<EOF;
 //   '$src_text'
 // which are part of the OOXML documentation
 
-#include <svx/escherex.hxx>
+#include <filter/msfilter/escherex.hxx>
 
 const char* pShapeTypes[ ESCHER_ShpInst_COUNT ] =
 {
diff -rup oox-orig/source/export/shapes.cxx oox/source/export/shapes.cxx
--- oox-orig/source/export/shapes.cxx	2010-05-25 16:55:39.000000000 +0200
+++ oox/source/export/shapes.cxx	2010-05-25 16:56:11.000000000 +0200
@@ -66,9 +66,9 @@
 #include <rtl/strbuf.hxx>
 #include <sfx2/app.hxx>
 #include <svl/languageoptions.hxx>
-#include <svx/escherex.hxx>
+#include <filter/msfilter/escherex.hxx>
 #include <svx/svdoashp.hxx>
-#include <svx/svxenum.hxx>
+#include <editeng/svxenum.hxx>
 #include <svx/unoapi.hxx>
 
 using namespace ::com::sun::star;
diff -rup oox-orig/source/export/vmlexport.cxx oox/source/export/vmlexport.cxx
--- oox-orig/source/export/vmlexport.cxx	2010-05-25 16:55:39.000000000 +0200
+++ oox/source/export/vmlexport.cxx	2010-05-25 16:56:11.000000000 +0200
@@ -64,7 +64,7 @@ public:
 };
 
 VMLExport::VMLExport( ::sax_fastparser::FSHelperPtr pSerializer )
-    : EscherEx( *( new SvNullStream ), 0 ),
+    : EscherEx( EscherExGlobalRef(new EscherExGlobal(0)), *( new SvNullStream ) ),
       m_pSerializer( pSerializer ),
       m_pShapeAttrList( NULL ),
       m_nShapeType( ESCHER_ShpInst_Nil ),
@@ -128,6 +128,9 @@ void VMLExport::CloseContainer()
 
 UINT32 VMLExport::EnterGroup( const String& rShapeName, const Rectangle* pRect )
 {
+#if 1 // FIXME dev300-m77 please fix this.
+    return 0;
+#else
     UINT32 nShapeId = GetShapeID();
 
     OStringBuffer aStyle( 200 );
@@ -163,6 +166,7 @@ UINT32 VMLExport::EnterGroup( const Stri
 
     mnGroupLevel++;
     return nShapeId;
+#endif
 }
 
 void VMLExport::LeaveGroup()
diff -rup oox-orig/util/makefile.mk oox/util/makefile.mk
--- oox-orig/util/makefile.mk	2010-05-25 16:55:39.000000000 +0200
+++ oox/util/makefile.mk	2010-05-25 16:56:11.000000000 +0200
@@ -74,7 +74,9 @@ SHL1STDLIBS= \
         $(GOODIESLIB)		\
         $(SVTOOLLIB)		\
         $(SVXCORELIB)		\
-        $(SVXMSFILTERLIB)	\
+        $(SVLLIB)		\
+        $(UNOTOOLSLIB)		\
+        $(MSFILTERLIB)	\
         $(TOOLSLIB)
 
 # link openssl, copied this bit from ucb/source/ucp/webdav/makefile.mk
