From a83a9a3723baf2dd5ed347c9540b69bb59d71f39 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:09:35 +0200
Subject: [PATCH 875/878] oox-pptx-import-fix-placeholder-text-style-1.diff

---
 oox/source/ppt/pptshape.cxx |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/oox/source/ppt/pptshape.cxx b/oox/source/ppt/pptshape.cxx
index 0586639..1134a93 100644
--- a/oox/source/ppt/pptshape.cxx
+++ b/oox/source/ppt/pptshape.cxx
@@ -175,7 +175,11 @@ void PPTShape::addShape(
             if( mnSubType && getSubTypeIndex() && rSlidePersist.getMasterPersist().get() ) {
             oox::drawingml::ShapePtr pPlaceholder = PPTShape::findPlaceholderByIndex( getSubTypeIndex(), rSlidePersist.getMasterPersist()->getShapes()->getChildren() );
             if( pPlaceholder.get() && pPlaceholder->getTextBody() ) {
-                TextListStylePtr pNewTextListStyle (new TextListStyle());
+            TextListStylePtr pNewTextListStyle ( new TextListStyle() );
+
+            pNewTextListStyle->apply( pPlaceholder->getTextBody()->getTextListStyle() );
+            if( pPlaceholder->getMasterTextListStyle().get() )
+                pNewTextListStyle->apply( *pPlaceholder->getMasterTextListStyle() );
 
             pNewTextListStyle->apply( pPlaceholder->getTextBody()->getTextListStyle() );
             aMasterTextListStyle = pNewTextListStyle;
-- 
1.7.0.1

