diff -rup oox-orig-1/source/drawingml/shape.cxx oox/source/drawingml/shape.cxx
--- oox-orig-1/source/drawingml/shape.cxx	2010-05-07 12:18:01.000000000 +0200
+++ oox/source/drawingml/shape.cxx	2010-05-07 12:18:20.000000000 +0200
@@ -197,7 +197,8 @@ void Shape::addShape(
 
 void Shape::applyShapeReference( const Shape& rReferencedShape )
 {
-    mpTextBody = TextBodyPtr( new TextBody( *rReferencedShape.mpTextBody.get() ) );
+    if( rReferencedShape.mpTextBody.get() )
+	mpTextBody = TextBodyPtr( new TextBody( *rReferencedShape.mpTextBody.get() ) );
     maShapeProperties = rReferencedShape.maShapeProperties;
     mpLinePropertiesPtr = LinePropertiesPtr( new LineProperties( *rReferencedShape.mpLinePropertiesPtr.get() ) );
     mpFillPropertiesPtr = FillPropertiesPtr( new FillProperties( *rReferencedShape.mpFillPropertiesPtr.get() ) );
