From 5ece641359215247a81bb16ad7fd0f8629a5267b Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:06:37 +0200
Subject: [PATCH 722/878] oox-pptx-export-fix-action-button-presets.diff

---
 oox/source/export/drawingml.cxx |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/oox/source/export/drawingml.cxx b/oox/source/export/drawingml.cxx
index b075a28..6bc93d1 100644
--- a/oox/source/export/drawingml.cxx
+++ b/oox/source/export/drawingml.cxx
@@ -1328,7 +1328,10 @@ void DrawingML::WritePresetShape( const char* pShape, MSO_SPT eShapeType, sal_Bo
     mpFS->startElementNS( XML_a, XML_avLst, FSEND );
 
     Sequence< drawing::EnhancedCustomShapeAdjustmentValue > aAdjustmentSeq;
-    if ( rProp.Value >>= aAdjustmentSeq ) {
+    if ( ( rProp.Value >>= aAdjustmentSeq )
+     && eShapeType != mso_sptActionButtonForwardNext  // we have adjustments values for these type of shape, but MSO doesn't like them
+     && eShapeType != mso_sptActionButtonBackPrevious // so they are now disabled
+    ) {
         DBG(printf("adj seq len: %d\n", int( aAdjustmentSeq.getLength() )));
         if ( bPredefinedHandlesUsed )
             EscherPropertyContainer::LookForPolarHandles( eShapeType, nAdjustmentsWhichNeedsToBeConverted );
-- 
1.7.0.1

