From 5d96c273e7bdd8493685904893fc57b1ac881e91 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:00:52 +0200
Subject: [PATCH 429/878] oox-pptx-import-fix-text-body-vert.diff

---
 oox/source/drawingml/textbodypropertiescontext.cxx |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/oox/source/drawingml/textbodypropertiescontext.cxx b/oox/source/drawingml/textbodypropertiescontext.cxx
index 67a68e9..40583fc 100644
--- a/oox/source/drawingml/textbodypropertiescontext.cxx
+++ b/oox/source/drawingml/textbodypropertiescontext.cxx
@@ -126,7 +126,8 @@ TextBodyPropertiesContext::TextBodyPropertiesContext( ContextHandler& rParent,
     // ST_TextVerticalType
     mrTextBodyProp.moVert = aAttribs.getToken( XML_vert );
     bool bRtl = aAttribs.getBool( XML_rtl, false );
-    if( mrTextBodyProp.moVert.get( XML_horz ) == XML_vert ) {
+    sal_Int32 tVert = mrTextBodyProp.moVert.get( XML_horz );
+    if( tVert == XML_vert || tVert == XML_eaVert || tVert == XML_vert270 || tVert == XML_mongolianVert ) {
     mrTextBodyProp.maPropertyMap[ PROP_TextWritingMode ]
         <<= ( bRtl ? WritingMode_RL_TB : WritingMode_LR_TB );
     // workaround for TB_LR as using WritingMode2 doesn't work
-- 
1.7.0.1

