From 31ef1c4404fd4f75b15215c842e0ccf390f0f25a Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:00:51 +0200
Subject: [PATCH 428/878] oox-pptx-import-fix-subtitle-placeholder.diff

---
 oox/source/ppt/pptshape.cxx |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/oox/source/ppt/pptshape.cxx b/oox/source/ppt/pptshape.cxx
index 39dc495..1102995 100644
--- a/oox/source/ppt/pptshape.cxx
+++ b/oox/source/ppt/pptshape.cxx
@@ -99,6 +99,11 @@ void PPTShape::addShape(
                     {
                         if ( ( meShapeLocation == Master ) || ( meShapeLocation == Layout ) )
                             sServiceName = rtl::OUString();
+            else {
+                const rtl::OUString sTitleShapeService( RTL_CONSTASCII_USTRINGPARAM( "com.sun.star.presentation.SubtitleShape" ) );
+                sServiceName = sTitleShapeService;
+                aMasterTextListStyle = rSlidePersist.getMasterPersist().get() ? rSlidePersist.getMasterPersist()->getTitleTextStyle() : rSlidePersist.getTitleTextStyle();
+            }
                     }
                     break;
                     case XML_obj :
-- 
1.7.0.1

