From 8b7a6e42f9fa91ae7d6f991118609fd3d9d77187 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:09:35 +0200
Subject: [PATCH 765/768] oox-pptx-import-fix-placeholder-text-style-1.diff

---
 oox/source/ppt/pptshape.cxx |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git oox/source/ppt/pptshape.cxx oox/source/ppt/pptshape.cxx
index cdf0fee..f6cfc66 100644
--- oox/source/ppt/pptshape.cxx
+++ oox/source/ppt/pptshape.cxx
@@ -175,7 +175,11 @@ void PPTShape::addShape(
             if( mnSubType && getSubTypeIndex() && rSlidePersist.getMasterPersist().get() ) {
             oox::drawingml::ShapePtr pPlaceholder = PPTShape::findPlaceholderByIndex( getSubTypeIndex(), rSlidePersist.getMasterPersist()->getShapes()->getChildren() );
             if( pPlaceholder.get() && pPlaceholder->getTextBody() ) {
-                TextListStylePtr pNewTextListStyle (new TextListStyle());
+            TextListStylePtr pNewTextListStyle ( new TextListStyle() );
+
+            pNewTextListStyle->apply( pPlaceholder->getTextBody()->getTextListStyle() );
+            if( pPlaceholder->getMasterTextListStyle().get() )
+                pNewTextListStyle->apply( *pPlaceholder->getMasterTextListStyle() );
 
             pNewTextListStyle->apply( pPlaceholder->getTextBody()->getTextListStyle() );
             aMasterTextListStyle = pNewTextListStyle;
-- 
1.7.0.1

