From f1aa91a65e4a021b5459bebca276f1f1abb42dfa Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:57:56 +0200
Subject: [PATCH 254/768] oox-tokenmap.string.h.diff

---
 oox/source/token/tokenmap.cxx |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git oox/source/token/tokenmap.cxx oox/source/token/tokenmap.cxx
index 17979d7..a3dac5c 100644
--- oox/source/token/tokenmap.cxx
+++ oox/source/token/tokenmap.cxx
@@ -25,6 +25,7 @@
  *
  ************************************************************************/
 
+#include <string.h>
 #include "oox/token/tokenmap.hxx"
 #include <string.h>
 #include <rtl/strbuf.hxx>
-- 
1.7.0.1

