---
 oox/source/ppt/pptshape.cxx |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git oox/source/ppt/pptshape.cxx oox/source/ppt/pptshape.cxx
index 48810e2..c24478e 100644
--- oox/source/ppt/pptshape.cxx
+++ oox/source/ppt/pptshape.cxx
@@ -83,7 +83,8 @@ void PPTShape::addShape(
             Reference< lang::XMultiServiceFactory > xServiceFact( rFilterBase.getModel(), UNO_QUERY_THROW );
             sal_Bool bClearText = sal_False;
 
-            if ( sServiceName != OUString::createFromAscii( "com.sun.star.drawing.GraphicObjectShape" ) )
+            if ( sServiceName != OUString::createFromAscii( "com.sun.star.drawing.GraphicObjectShape" ) &&
+                 sServiceName != OUString::createFromAscii( "com.sun.star.drawing.OLE2Shape" ) )
             {
                 switch( mnSubType )
                 {
-- 
1.7.0.1

