--- oox/source/ppt/pptshape.cxx.old	2010-07-22 13:12:44.000000000 +0200
+++ oox/source/ppt/pptshape.cxx	2010-07-29 21:06:00.000000000 +0200
@@ -186,8 +186,9 @@ void PPTShape::addShape(
 */
             if ( sServiceName.getLength() )
             {
+            // use style from master slide for placeholders only, otherwise use slide's style, which might be the default style from presentation
                 if ( !aMasterTextListStyle.get() )
-                    aMasterTextListStyle = rSlidePersist.getMasterPersist().get() ? rSlidePersist.getMasterPersist()->getOtherTextStyle() : rSlidePersist.getOtherTextStyle();
+                    aMasterTextListStyle = ( mnSubType && rSlidePersist.getMasterPersist().get() ) ? rSlidePersist.getMasterPersist()->getOtherTextStyle() : rSlidePersist.getOtherTextStyle();
                 setMasterTextListStyle( aMasterTextListStyle );
  
                 Reference< XShape > xShape( createAndInsert( rFilterBase, sServiceName, pTheme, rxShapes, pShapeRect, bClearText ) );
