---
 oox/source/drawingml/shape.cxx |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git oox/source/drawingml/shape.cxx oox/source/drawingml/shape.cxx
index 765a559..bd251d1 100644
--- oox/source/drawingml/shape.cxx
+++ oox/source/drawingml/shape.cxx
@@ -172,6 +172,7 @@ void Shape::addShape(
 
 void Shape::applyShapeReference( const Shape& rReferencedShape )
 {
+    if( rReferencedShape.mpTextBody.get() )
     mpTextBody = TextBodyPtr( new TextBody( *rReferencedShape.mpTextBody.get() ) );
     maShapeProperties = rReferencedShape.maShapeProperties;
     mpLinePropertiesPtr = LinePropertiesPtr( new LineProperties( *rReferencedShape.mpLinePropertiesPtr.get() ) );
-- 
1.7.0.1

