--- oox/source/ppt/pptshape.cxx.old	2009-10-08 16:17:33.000000000 +0200
+++ oox/source/ppt/pptshape.cxx	2010-03-03 16:59:43.000000000 +0100
@@ -158,10 +158,12 @@ void PPTShape::addShape(
                     break;
                 }
             }
+
             if ( sServiceName.getLength() )
             {
+            // use style from master slide for placeholders only, otherwise use slide's style, which might be the default style from presentation
             if ( !aMasterTextListStyle.get() )
-                aMasterTextListStyle = rSlidePersist.getMasterPersist().get() ? rSlidePersist.getMasterPersist()->getOtherTextStyle() : rSlidePersist.getOtherTextStyle();
+                    aMasterTextListStyle = ( mnSubType && rSlidePersist.getMasterPersist().get() ) ? rSlidePersist.getMasterPersist()->getOtherTextStyle() : rSlidePersist.getOtherTextStyle();
             setMasterTextListStyle( aMasterTextListStyle );
 
             Reference< XShape > xShape( createAndInsert( rFilterBase, sServiceName, rxTheme, rxShapes, pShapeRect, bClearText ) );
