---
 oox/source/ppt/pptshapecontext.cxx |   27 ++++++++++++++++++++-------
 1 files changed, 20 insertions(+), 7 deletions(-)

diff --git oox/source/ppt/pptshapecontext.cxx oox/source/ppt/pptshapecontext.cxx
index ef096c4..3c3f339 100644
--- oox/source/ppt/pptshapecontext.cxx
+++ oox/source/ppt/pptshapecontext.cxx
@@ -86,6 +86,26 @@ oox::drawingml::ShapePtr findPlaceholder( const sal_Int32 nMasterPlaceholder, sa
     return aShapePtr;
 }
 
+oox::drawingml::ShapePtr findPlaceholderByIndex( const sal_Int32 nIdx, std::vector< oox::drawingml::ShapePtr >& rShapes )
+{
+    oox::drawingml::ShapePtr aShapePtr;
+    std::vector< oox::drawingml::ShapePtr >::reverse_iterator aRevIter( rShapes.rbegin() );
+    while( aRevIter != rShapes.rend() )
+    {
+        if ( (*aRevIter)->getSubTypeIndex() == nIdx )
+        {
+            aShapePtr = *aRevIter;
+            break;
+        }
+        std::vector< oox::drawingml::ShapePtr >& rChildren = (*aRevIter)->getChildren();
+        aShapePtr = findPlaceholderByIndex( nIdx, rChildren );
+        if ( aShapePtr.get() )
+            break;
+        aRevIter++;
+    }
+    return aShapePtr;
+}
+
 // if nFirstPlaceholder can't be found, it will be searched for nSecondPlaceholder
 oox::drawingml::ShapePtr findPlaceholder( sal_Int32 nFirstPlaceholder, sal_Int32 nSecondPlaceholder,
     sal_Int32 nSubTypeIndex, std::vector< oox::drawingml::ShapePtr >& rShapes )
@@ -179,13 +199,6 @@ Reference< XFastContextHandler > PPTShapeContext::createFastChildContext( sal_In
                                 pPlaceholder = findPlaceholder( nFirstPlaceholder, nSecondPlaceholder,
                                     pPPTShapePtr->getSubTypeIndex(), pMasterPersist->getShapes()->getChildren() );
                         }
-                        if ( pPlaceholder.get() )
-                        {
-                            mpShapePtr->applyShapeReference( *pPlaceholder.get() );
-                            PPTShape* pPPTShape = dynamic_cast< PPTShape* >( pPlaceholder.get() );
-                            if ( pPPTShape )
-                                pPPTShape->setReferenced( sal_True );
-                        }
                     }
                 }
                 if ( pPlaceholder.get() )
-- 
1.7.0.1

