diff -r 1444aeef22df oox/inc/oox/drawingml/textliststyle.hxx
--- oox/inc/oox/drawingml/textliststyle.hxx	Wed Apr 14 14:29:10 2010 +0300
+++ oox/inc/oox/drawingml/textliststyle.hxx	Fri Apr 30 14:45:50 2010 +0200
@@ -50,6 +50,10 @@
 	inline const TextParagraphPropertiesVector& getAggregationListStyle() const { return maAggregationListStyle; };
     inline TextParagraphPropertiesVector&       getAggregationListStyle() { return maAggregationListStyle; };
 
+#if OSL_DEBUG_LEVEL > 0
+    void dump( int nLevels=9 );
+#endif
+
 protected:
 
     TextParagraphPropertiesVector maListStyle;
diff -r 1444aeef22df oox/inc/oox/drawingml/textparagraphproperties.hxx
--- oox/inc/oox/drawingml/textparagraphproperties.hxx	Wed Apr 14 14:29:10 2010 +0300
+++ oox/inc/oox/drawingml/textparagraphproperties.hxx	Fri Apr 30 14:45:50 2010 +0200
@@ -112,6 +112,10 @@
         size can be zero and the default value is returned. */
     float                               getCharHeightPoints( float fDefault ) const;
 
+#if OSL_DEBUG_LEVEL > 0
+    void dump() { maTextParagraphPropertyMap.dump(); OSL_TRACE("character height: %f", maTextCharacterProperties.getCharHeightPoints(-1)); }
+#endif
+
 protected:
 
     TextCharacterProperties         maTextCharacterProperties;
diff -r 1444aeef22df oox/inc/oox/helper/propertymap.hxx
--- oox/inc/oox/helper/propertymap.hxx	Wed Apr 14 14:29:10 2010 +0300
+++ oox/inc/oox/helper/propertymap.hxx	Fri Apr 30 14:45:50 2010 +0200
@@ -90,6 +90,7 @@
                         makePropertySet() const;
 
 #if OSL_DEBUG_LEVEL > 0
+  static void dump( ::com::sun::star::uno::Reference< ::com::sun::star::beans::XPropertySet > rXPropSet);
   void dump();
 #endif
 
diff -r 1444aeef22df oox/inc/oox/helper/propertyset.hxx
--- oox/inc/oox/helper/propertyset.hxx	Wed Apr 14 14:29:10 2010 +0300
+++ oox/inc/oox/helper/propertyset.hxx	Fri Apr 30 14:45:50 2010 +0200
@@ -123,6 +123,10 @@
         @param rPropertyMap  The property map. */
     void                setProperties( const PropertyMap& rPropertyMap );
 
+#if OSL_DEBUG_LEVEL > 0
+    void dump();
+#endif
+
     // ------------------------------------------------------------------------
 private:
     /** Gets the specified property from the property set.
diff -r 1444aeef22df oox/source/drawingml/textliststyle.cxx
--- oox/source/drawingml/textliststyle.cxx	Wed Apr 14 14:29:10 2010 +0300
+++ oox/source/drawingml/textliststyle.cxx	Fri Apr 30 14:45:50 2010 +0200
@@ -65,4 +65,15 @@
     applyStyleList( rTextListStyle.getListStyle(), getListStyle() );
 }
 
+#if OSL_DEBUG_LEVEL > 0
+void TextListStyle::dump( int nLevels )
+{
+    for ( int i = 0; i < nLevels; i++ )
+    {
+	OSL_TRACE("level: %d", i);
+        maListStyle[ i ]->dump();
+    }
+}
+#endif
+
 } }
diff -r 1444aeef22df oox/source/helper/propertymap.cxx
--- oox/source/helper/propertymap.cxx	Wed Apr 14 14:29:10 2010 +0300
+++ oox/source/helper/propertymap.cxx	Fri Apr 30 14:45:50 2010 +0200
@@ -242,13 +242,13 @@
 }
 
 #if OSL_DEBUG_LEVEL > 0
-void PropertyMap::dump()
+void PropertyMap::dump( Reference< XPropertySet > rXPropSet )
 {
-  Reference< XPropertySet > rXPropSet( makePropertySet(), UNO_QUERY );
-
     Reference< XPropertySetInfo > info = rXPropSet->getPropertySetInfo ();
     Sequence< beans::Property > props = info->getProperties ();
 
+    OSL_TRACE("dump props, len: %d", props.getLength ());
+
     for (int i=0; i < props.getLength (); i++) {
         OString name = OUStringToOString( props [i].Name, RTL_TEXTENCODING_UTF8);
         fprintf (stderr,"%30s = ", name.getStr() );
@@ -294,6 +294,11 @@
 	}
     }
 }
+
+void PropertyMap::dump()
+{
+    dump( Reference< XPropertySet >( makePropertySet(), UNO_QUERY ) );
+}
 #endif
 
 // ============================================================================
diff -r 1444aeef22df oox/source/helper/propertyset.cxx
--- oox/source/helper/propertyset.cxx	Wed Apr 14 14:29:10 2010 +0300
+++ oox/source/helper/propertyset.cxx	Fri Apr 30 14:45:50 2010 +0200
@@ -168,6 +168,13 @@
     }
 }
 
+#if OSL_DEBUG_LEVEL > 0
+void PropertySet::dump()
+{
+    PropertyMap::dump( Reference< XPropertySet >( getXPropertySet(), UNO_QUERY ) );
+}
+#endif
+
 // ============================================================================
 
 } // namespace oox
