diff -rup oox-orig/source/export/drawingml.cxx oox/source/export/drawingml.cxx
--- oox-orig/source/export/drawingml.cxx	2009-07-09 12:24:09.000000000 +0200
+++ oox/source/export/drawingml.cxx	2009-07-13 17:07:04.000000000 +0200
@@ -1329,7 +1213,10 @@ void DrawingML::WritePresetShape( const
     mpFS->startElementNS( XML_a, XML_avLst, FSEND );
 
     Sequence< drawing::EnhancedCustomShapeAdjustmentValue > aAdjustmentSeq;
-    if ( rProp.Value >>= aAdjustmentSeq ) {
+    if ( ( rProp.Value >>= aAdjustmentSeq )
+	 && eShapeType != mso_sptActionButtonForwardNext  // we have adjustments values for these type of shape, but MSO doesn't like them
+	 && eShapeType != mso_sptActionButtonBackPrevious // so they are now disabled
+	) {
         DBG(printf("adj seq len: %d\n", int( aAdjustmentSeq.getLength() )));
         if ( bPredefinedHandlesUsed )
             EscherPropertyContainer::LookForPolarHandles( eShapeType, nAdjustmentsWhichNeedsToBeConverted );
