From 5b38a8ec82a4f611af9e155bae809b286ea65371 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:09:36 +0200
Subject: [PATCH 876/878] oox-drawingml-fix-apply-shape-reference-crash.diff

---
 oox/source/drawingml/shape.cxx |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/oox/source/drawingml/shape.cxx b/oox/source/drawingml/shape.cxx
index da6901e..2422be2 100644
--- a/oox/source/drawingml/shape.cxx
+++ b/oox/source/drawingml/shape.cxx
@@ -168,6 +168,7 @@ void Shape::addShape(
 
 void Shape::applyShapeReference( const Shape& rReferencedShape )
 {
+    if( rReferencedShape.mpTextBody.get() )
     mpTextBody = TextBodyPtr( new TextBody( *rReferencedShape.mpTextBody.get() ) );
     maShapeProperties = rReferencedShape.maShapeProperties;
     mpLinePropertiesPtr = LinePropertiesPtr( new LineProperties( *rReferencedShape.mpLinePropertiesPtr.get() ) );
-- 
1.7.0.1

