--- oox/source/xls/worksheetfragment.cxx	2010-07-16 21:21:27.000000000 +0530
+++ oox/source/xls/worksheetfragment.cxx	2010-07-19 14:46:13.000000000 +0530
@@ -570,8 +570,8 @@ void OoxWorksheetFragment::importBrk( co
 {
     PageBreakModel aModel;
     aModel.mnColRow = rAttribs.getInteger( XML_id, 0 );
-    aModel.mnMin    = rAttribs.getInteger( XML_id, 0 );
-    aModel.mnMax    = rAttribs.getInteger( XML_id, 0 );
+    aModel.mnMin    = rAttribs.getInteger( XML_min, aModel.mnColRow );
+    aModel.mnMax    = rAttribs.getInteger( XML_max, aModel.mnColRow );
     aModel.mbManual = rAttribs.getBool( XML_man, false );
     setPageBreak( aModel, bRowBreak );
 }
