--- filter/source/config/fragments/fcfg_calc.mk	2010-04-09 16:13:42.000000000 +0530
+++ filter/source/config/fragments/fcfg_calc.mk	2010-05-07 11:19:25.000000000 +0530
@@ -27,7 +27,9 @@ T4_CALC = \
     calc_MS_Excel_2003_XML \
     MS_Excel_2007_XML \
     MS_Excel_2007_XML_Template \
-    MS_Excel_2007_Binary
+    MS_Excel_2007_Binary \
+    calc_OOXML \
+    calc_OOXML_Template
 
 # -----------------------------------------------
 # count = 28
@@ -57,7 +59,9 @@ F4_CALC = \
     MS_Excel_2003_XML \
     calc_MS_Excel_2007_XML \
     calc_MS_Excel_2007_XML_Template \
-    calc_MS_Excel_2007_Binary
+    calc_MS_Excel_2007_Binary \
+    calc_OOXML \
+    calc_OOXML_Template
 
 # -----------------------------------------------
 # count = 12
@@ -76,7 +80,9 @@ F4_UI_CALC = \
     MS_Excel_2003_XML_ui \
     calc_MS_Excel_2007_XML_ui \
     calc_MS_Excel_2007_XML_Template_ui \
-    calc_MS_Excel_2007_Binary_ui
+    calc_MS_Excel_2007_Binary_ui \
+    calc_OOXML_ui \
+    calc_OOXML_Template_ui
 
 # -----------------------------------------------
 # count = 0
--- filter/source/config/fragments/filters/calc_OOXML_Template_ui.xcu	1970-01-01 05:30:00.000000000 +0530
+++ filter/source/config/fragments/filters/calc_OOXML_Template_ui.xcu	2010-05-07 10:42:42.000000000 +0530
@@ -0,0 +1,6 @@
+<node oor:name="Calc Office Open XML Template">
+    <prop oor:name="UIName">
+        <value xml:lang="en-US">Office Open XML Spreadsheet Template</value>
+        <value xml:lang="de">Office Open XML Spreadsheet Vorlage</value>
+    </prop>
+</node>
--- filter/source/config/fragments/filters/calc_OOXML_Template.xcu	1970-01-01 05:30:00.000000000 +0530
+++ filter/source/config/fragments/filters/calc_OOXML_Template.xcu	2010-05-07 10:18:37.000000000 +0530
@@ -0,0 +1,10 @@
+<node oor:name="Calc Office Open XML Template" oor:op="replace">
+    <prop oor:name="Flags"><value>IMPORT ALIEN 3RDPARTYFILTER TEMPLATE TEMPLATEPATH</value></prop>
+    <prop oor:name="UIComponent"/>
+    <prop oor:name="FilterService"><value>com.sun.star.comp.oox.ExcelFilter</value></prop>
+    <prop oor:name="UserData"><value>OOXML</value></prop>
+    <prop oor:name="FileFormatVersion"/>
+    <prop oor:name="Type"><value>Office Open XML Spreadsheet Template</value></prop>
+    <prop oor:name="TemplateName"/>
+    <prop oor:name="DocumentService"><value>com.sun.star.sheet.SpreadsheetDocument</value></prop>
+</node>
--- filter/source/config/fragments/filters/calc_OOXML_ui.xcu	1970-01-01 05:30:00.000000000 +0530
+++ filter/source/config/fragments/filters/calc_OOXML_ui.xcu	2010-05-07 10:37:16.000000000 +0530
@@ -0,0 +1,6 @@
+<node oor:name="Calc Office Open XML">
+    <prop oor:name="UIName">
+        <value xml:lang="en-US">Office Open XML Spreadsheet</value>
+        <value xml:lang="de">Office Open XML Spreadsheet</value>
+    </prop>
+</node>
--- filter/source/config/fragments/filters/calc_OOXML.xcu	1970-01-01 05:30:00.000000000 +0530
+++ filter/source/config/fragments/filters/calc_OOXML.xcu	2010-05-07 10:16:41.000000000 +0530
@@ -0,0 +1,10 @@
+<node oor:name="Calc Office Open XML" oor:op="replace">
+	<prop oor:name="Flags"><value>IMPORT EXPORT ALIEN 3RDPARTYFILTER PREFERRED</value></prop>
+    <prop oor:name="UIComponent"/>
+    <prop oor:name="FilterService"><value>com.sun.star.comp.oox.ExcelFilter</value></prop>
+    <prop oor:name="UserData"><value>OOXML</value></prop>
+    <prop oor:name="FileFormatVersion"/>
+    <prop oor:name="Type"><value>Office Open XML Spreadsheet</value></prop>
+    <prop oor:name="TemplateName"/>
+    <prop oor:name="DocumentService"><value>com.sun.star.sheet.SpreadsheetDocument</value></prop>
+</node>
--- filter/source/config/fragments/types/calc_OOXML_Template.xcu	1970-01-01 05:30:00.000000000 +0530
+++ filter/source/config/fragments/types/calc_OOXML_Template.xcu	2010-05-07 10:50:30.000000000 +0530
@@ -0,0 +1,10 @@
+<node oor:name="Office Open XML Spreadsheet Template" oor:op="replace" >
+    <prop oor:name="DetectService"><value>com.sun.star.comp.oox.FormatDetector</value></prop>
+    <prop oor:name="URLPattern"/>
+    <prop oor:name="Extensions"><value>xltx xltm</value></prop>
+    <prop oor:name="MediaType"/>
+    <prop oor:name="Preferred"><value>false</value></prop>
+    <prop oor:name="PreferredFilter"><value>Calc Office Open XML Template</value></prop>
+    <prop oor:name="UIName"><value xml:lang="x-default">Office Open XML Spreadsheet Template</value></prop>
+    <prop oor:name="ClipboardFormat"/>
+</node>
--- filter/source/config/fragments/types/calc_OOXML.xcu	1970-01-01 05:30:00.000000000 +0530
+++ filter/source/config/fragments/types/calc_OOXML.xcu	2010-05-07 10:47:38.000000000 +0530
@@ -0,0 +1,10 @@
+<node oor:name="Office Open XML Spreadsheet" oor:op="replace" >
+    <prop oor:name="DetectService"><value>com.sun.star.comp.oox.FormatDetector</value></prop>
+    <prop oor:name="URLPattern"/>
+	<prop oor:name="Extensions"><value>xlsx xlsm</value></prop>
+    <prop oor:name="MediaType"/>
+    <prop oor:name="Preferred"><value>false</value></prop>
+    <prop oor:name="PreferredFilter"><value>Calc Office Open XML</value></prop>
+    <prop oor:name="UIName"><value xml:lang="x-default">Office Open XML Spreadsheet</value></prop>
+    <prop oor:name="ClipboardFormat"/>
+</node>
