--- sc/source/filter/excel/xestream.cxx	2010-06-03 15:44:46.000000000 +0530
+++ sc/source/filter/excel/xestream.cxx	2010-07-06 19:10:49.000000000 +0530
@@ -47,6 +47,7 @@
 #include "compiler.hxx"
 
 #include <../../ui/inc/docsh.hxx>
+#include <../../ui/inc/viewdata.hxx>
 #include <excdoc.hxx>
 
 #include <oox/core/tokens.hxx>
@@ -1146,6 +1148,10 @@ bool XclExpXmlStream::exportDocument() t
                 pShell->SetError( nError, ::rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( OSL_LOG_PREFIX ) ) );
             }
 #endif
+    // Get the viewsettings before processing
+    if( pShell->GetViewData() )
+        pShell->GetViewData()->WriteExtOptions( mpRoot->GetExtDocOptions() );
+ 
     OUString const workbook = CREATE_OUSTRING( "xl/workbook.xml" );
     PushStream( CreateOutputStream( workbook, workbook,
                                     Reference <XOutputStream>(),
