diff --git filter/source/config/fragments/fcfg_impress.mk filter/source/config/fragments/fcfg_impress.mk
index 43447cb..b5625ef 100644
--- filter/source/config/fragments/fcfg_impress.mk
+++ filter/source/config/fragments/fcfg_impress.mk
@@ -18,7 +18,8 @@ T4_IMPRESS = \
     MS_PowerPoint_2007_XML_AutoPlay\
     MS_PowerPoint_2007_XML_Template\
     impress_OOXML_Presentation\
-    impress_OOXML_Presentation_Template
+    impress_OOXML_Presentation_Template\
+    impress_OOXML_AutoPlay
 
 # -----------------------------------------------
 # count = 20
@@ -38,7 +39,8 @@ F4_IMPRESS = \
     impress_MS_PowerPoint_2007_XML_AutoPlay\
     impress_MS_PowerPoint_2007_XML_Template\
     impress_OOXML\
-    impress_OOXML_Template
+    impress_OOXML_Template\
+    impress_OOXML_AutoPlay
 
 # -----------------------------------------------
 # count = 12
--- /dev/null
+++ filter/source/config/fragments/filters/impress_OOXML_AutoPlay.xcu
@@ -0,0 +1,11 @@
+<node oor:name="Impress Office Open XML AutoPlay" oor:op="replace">
+	<prop oor:name="Flags"><value>IMPORT EXPORT ALIEN 3RDPARTYFILTER PREFERRED STARTPRESENTATION</value></prop>
+	<prop oor:name="UIComponent"/>
+	<prop oor:name="FilterService"><value>com.sun.star.comp.Impress.oox.PowerPointImport</value></prop>
+	<prop oor:name="UserData"/>
+	<prop oor:name="UIName"><value xml:lang="x-default">Office Open XML Presentation AutoPlay</value></prop>
+	<prop oor:name="FileFormatVersion"/>
+	<prop oor:name="Type"><value>Office Open XML Presentation AutoPlay</value></prop>
+	<prop oor:name="TemplateName"/>
+	<prop oor:name="DocumentService"><value>com.sun.star.presentation.PresentationDocument</value></prop>
+</node>
--- /dev/null
+++ filter/source/config/fragments/types/impress_OOXML_Presentation_AutoPlay.xcu
@@ -0,0 +1,10 @@
+<node oor:name="Office Open XML Presentation AutoPlay" oor:op="replace" >
+	<prop oor:name="DetectService"><value>com.sun.star.comp.oox.FormatDetector</value></prop>
+	<prop oor:name="URLPattern"/>
+	<prop oor:name="Extensions"><value>ppsx</value></prop>
+	<prop oor:name="MediaType"/>
+	<prop oor:name="Preferred"><value>true</value></prop>
+	<prop oor:name="PreferredFilter"><value>Impress Office Open XML AutoPlay</value></prop>
+	<prop oor:name="UIName"><value xml:lang="x-default">Office Open XML Presentation AutoPlay</value></prop>
+	<prop oor:name="ClipboardFormat"/>
+</node>
