Index: open-url.sh
===================================================================
--- open-url.sh	(Revision 277618)
+++ open-url.sh	(Arbeitskopie)
@@ -52,7 +52,7 @@
     exit 0
   else
     # mozilla derivates may need -remote semantics
-    for i in thunderbird mozilla netscape; do
+    for i in icedove thunderbird iceape mozilla netscape; do
       mailer=`which $i`
       if [ ! -z "$mailer" ]; then
         run_mozilla "$mailer" "$1"
@@ -69,7 +69,7 @@
     exit 0
   else
     # mozilla derivates may need -remote semantics
-    for i in firefox mozilla netscape; do
+    for i in iceweasel firefox iceape mozilla netscape; do
       browser=`which $i`
       if [ ! -z "$browser" ]; then
         run_mozilla "$browser" "$1"
