diff --git a/officecfg/registry/schema/org/openoffice/Office/Calc.xcs b/officecfg/registry/schema/org/openoffice/Office/Calc.xcs
index fbec7ab..7cc3864 100644
--- officecfg/registry/schema/org/openoffice/Office/Calc.xcs
+++ officecfg/registry/schema/org/openoffice/Office/Calc.xcs
@@ -329,7 +329,7 @@
 						<desc>Specifies whether guides are displayed when moving objects.</desc>
 						<label>Guides</label>
 					</info>
-					<value>false</value>
+					<value>true</value>
 				</prop>
 				<prop oor:name="SimpleControlPoint" oor:type="xs:boolean">
 					<!-- OldPath: Calc/Layout/Lines -->
@@ -351,7 +351,7 @@
 						<desc>Specifies whether control points are displayed as larger than the default size.</desc>
 						<label>Large control points</label>
 					</info>
-					<value>false</value>
+					<value>true</value>
 				</prop>
 			</group>
 			<group oor:name="Window">
@@ -1428,7 +1428,7 @@
 							<desc>Indicates whether the import of Lotus wk3 files is enabled.</desc>
 							<label>Enable WK3 Filter</label>
 						</info>
-						<value>false</value>
+						<value>true</value>
 					</prop>
 				</group>
 				<group oor:name="MS_Excel">
diff --git a/officecfg/registry/schema/org/openoffice/Office/Common.xcs b/officecfg/registry/schema/org/openoffice/Office/Common.xcs
index c191089..27fe87f 100644
--- officecfg/registry/schema/org/openoffice/Office/Common.xcs
+++ officecfg/registry/schema/org/openoffice/Office/Common.xcs
@@ -592,7 +592,7 @@
 						</info>
 					</maxInclusive>
 				</constraints>
-				<value>10000</value>
+				<value>50000</value>
 			</prop>
 			<prop oor:name="LogicalHeight" oor:type="xs:int">
 				<info>
@@ -610,7 +610,7 @@
 						</info>
 					</maxInclusive>
 				</constraints>
-				<value>10000</value>
+				<value>50000</value>
 			</prop>
 		</group>
  		<group oor:name="ObjectNames">
@@ -949,7 +949,7 @@
 					<desc>Specifies if 3D graphics from Draw and Impress will be displayed on the user&apos;s system using an OpenGL capable hardware. If the system does not have an OpenGL capable hardware, this setting will be ignored. The 3D display is always done per software.</desc>
 					<label>Use OpenGL</label>
 				</info>
-				<value>false</value>
+				<value>true</value>
 			</prop>
 			<prop oor:name="OpenGL_Faster" oor:type="xs:boolean">
 				<!-- OldPath: 3D-Engine/OpenGL_Faster -->
@@ -971,7 +971,7 @@
 					<desc>Specifies interaction with full display. If a 3-D object is rotated or moved, the full display is rotated or moved and not a grid frame.</desc>
 					<label>Object refresh during interaction</label>
 				</info>
-				<value>false</value>
+				<value>true</value>
 			</prop>
 		</group>
     <group oor:name="Dictionaries">
@@ -1086,7 +1086,7 @@
 					<desc>Specifies the maximum allowed Paper Width for page definitions in cm. Default is 3m, i.e. 300 cm. When this is changed to higher values, it is done on own risk.</desc>
 					<label>Maximum allowed Paper Width for page definitions</label>
 				</info>
-				<value>300</value>
+				<value>1000</value>
 			</prop>
 			<prop oor:name="MaximumPaperHeight" oor:type="xs:int">
 				<info>
@@ -1094,7 +1094,7 @@
 					<desc>Specifies the maximum allowed Paper Height for page definitions in cm. Default is 3m, i.e. 300 cm. When this is changed to higher values, it is done on own risk.</desc>
 					<label>Maximum allowed Paper Height for page definitions</label>
 				</info>
-				<value>300</value>
+				<value>1000</value>
 			</prop>
 			<prop oor:name="MaximumPaperLeftMargin" oor:type="xs:int">
 				<info>
@@ -1240,7 +1240,7 @@
           </desc>
           <label>Specifies the percent of transparence to use if transparent selection is used.</label>
         </info>
-        <value>75</value>
+        <value>60</value>
       </prop>
       <prop oor:name="SelectionMaximumLuminancePercent" oor:type="xs:short">
         <info>
@@ -1357,7 +1357,7 @@
 					<desc>Specifies if multiple spaces should be combined into one.</desc>
 					<label>Ignore double spaces</label>
 				</info>
-				<value>false</value>
+				<value>true</value>
 			</prop>
             <prop oor:name="CorrectAccidentalCapsLock" oor:type="xs:boolean">
                 <!-- UIHints: Tools  AutoCorrect/AutoFormat  Options  Correct accidental use of cAPS LOCK key -->
@@ -1476,7 +1476,7 @@
 						<author>AF</author>
 						<desc>Determines the maximum number of OLE objects that can be held in RAM for drawings, presentations and inserted drawing objects. The fewer OLE objects contained in RAM, the more space is available for other functions. The more OLE objects in RAM, the faster you can page through the objects since they do not always need to be loaded from the hard drive.</desc>
 					</info>
-					<value>20</value>
+					<value>64</value>
 				</prop>
 			</group>
 			<group oor:name="Writer">
@@ -1492,7 +1492,7 @@
 						<author>AF</author>
 						<desc>Determines the maximum number of OLE objects that can be held in RAM for the writer. The fewer OLE objects contained in RAM, the more space is available for other functions. The more OLE objects in RAM, the faster you can page through the objects since they do not always need to be loaded from the hard drive.</desc>
 					</info>
-					<value>20</value>
+					<value>64</value>
 				</prop>
 			</group>
 			<group oor:name="GraphicManager">
@@ -1506,7 +1506,7 @@
 						<desc>Specifies the maximum cache size for all graphical display objects.</desc>
 						<label>Total Graphic Cache Size</label>
 					</info>
-					<value>22000000</value>
+					<value>256000000</value>
 				</prop>
 				<prop oor:name="ObjectCacheSize" oor:type="xs:int">
 					<info>
@@ -1514,7 +1514,7 @@
 						<desc>Specifies the maximum cache size for a single graphic display object.</desc>
 						<label>Graphic Object Cache Size</label>
 					</info>
-					<value>5500000</value>
+					<value>10000000</value>
 				</prop>
 				<prop oor:name="ObjectReleaseTime" oor:type="xs:int">
 					<info>
@@ -2068,7 +2068,7 @@
 						</info>
 					</maxInclusive>
 				</constraints>
-				<value>10</value>
+				<value>20</value>
 			</prop>
 			<set oor:name="HelpBookmarks" oor:node-type="HistoryType">
 				<info>
@@ -2156,7 +2156,7 @@
 					<desc>Specifies if the all open windows and documents should be saved. If set to true, the URLs of all open documents and all view properties of all open views are saved when terminating the application.</desc>
 					<label>Open documents</label>
 				</info>
-				<value>false</value>
+				<value>true</value>
 			</prop>
 			<group oor:name="Document">
 				<info>
@@ -2191,7 +2191,7 @@
 						<desc>Specifies whether all modified documents are automatically saved in a time interval.</desc>
 						<label>AutoSave every</label>
 					</info>
-					<value>false</value>
+					<value>true</value>
 				</prop>
 				<prop oor:name="CreateBackup" oor:type="xs:boolean">
 					<!-- OldPath: General/Save/Documents -->
@@ -2202,7 +2202,7 @@
 						<desc>Specifies whether to create a backup copy when a modified document is saved.</desc>
 						<label>Always create backup copy</label>
 					</info>
-					<value>false</value>
+					<value>true</value>
 				</prop>
 				<prop oor:name="EditProperty" oor:type="xs:boolean">
 					<!-- OldPath: General/Save/Documents -->
@@ -2247,7 +2247,7 @@
 							</info>
 						</maxInclusive>
 					</constraints>
-					<value>15</value>
+					<value>10</value>
 				</prop>
 				<prop oor:name="ViewInfo" oor:type="xs:boolean">
 					<!-- OldPath: General/Restore -->
@@ -2501,7 +2501,7 @@
 						<author>GT</author>
 						<desc>Specifies wether to warn when saving or sending documents with personal/hidden data.</desc>
 					</info>
-					<value>false</value>
+					<value>true</value>
 				</prop>
 				<prop oor:name="WarnSignDoc" oor:type="xs:boolean">
 					<info>
@@ -2522,7 +2522,7 @@
 						<author>GT</author>
 						<desc>Specifies wether to warn when creating PDF documents with personal/hidden data.</desc>
 					</info>
-					<value>false</value>
+					<value>true</value>
 				</prop>
 				<prop oor:name="RemovePersonalInfoOnSaving" oor:type="xs:boolean">
 					<info>
@@ -2652,7 +2652,7 @@
 						<author>CD</author>
 						<desc>Every new document window will be forced to front and grabs the focus.</desc>
 					</info>
-					<value>true</value>
+					<value>false</value>
 				</prop>
 			</group>
 			<group oor:name="AppWindow">
@@ -3121,7 +3121,7 @@
 						</info>
 					</maxInclusive>
 				</constraints>
-				<value>100</value>
+				<value>256</value>
 			</prop>
 		</group>
 		<group oor:name="Setup">
@@ -3681,7 +3681,7 @@
 						</info>
 					</maxInclusive>
 				</constraints>
-				<value>8</value>
+				<value>32</value>
 			</prop>
 			<prop oor:name="PointReduce" oor:type="xs:short">
 				<!-- OldPath: Draw_Impress/Vectorize -->
@@ -4542,7 +4542,7 @@
 									</info>
 								</enumeration>
 							</constraints>
-							<value>0</value>
+							<value>3</value>
 						</prop>
 						<prop oor:name="Version" oor:type="xs:int">
 							<info>
@@ -4580,7 +4580,7 @@
 									</info>
 								</enumeration>
 							</constraints>
-							<value>2</value>
+							<value>1</value>
 						</prop>
 						<prop oor:name="CompressionMode" oor:type="xs:int">
 							<info>
@@ -4599,7 +4599,7 @@
 									</info>
 								</enumeration>
 							</constraints>
-							<value>2</value>
+							<value>1</value>
 						</prop>
 						<prop oor:name="TextMode" oor:type="xs:int">
 							<info>
@@ -5803,7 +5803,7 @@
                     <author>mmeeks</author>
                     <desc>Determins if various experimental, and potentially unstable features should be enabled in the user interface.</desc>
                 </info>
-                <value>false</value>
+                <value>true</value>
             </prop>
 			<prop oor:name="SymbolSet" oor:type="xs:short">
 				<!-- UIHints: Tools  Options General View -->
@@ -5823,7 +5823,7 @@
 						</info>
 					</enumeration>
 				</constraints>
-				<value>2</value>
+				<value>0</value>
 			</prop>
 			<prop oor:name="SymbolStyle" oor:type="xs:string">
 				<!-- UIHints: Tools  Options General View -->
@@ -5873,7 +5873,7 @@
 						</info>
 					</enumeration>
 				</constraints>
-				<value>auto</value>
+				<value>default</value>
 			</prop>
 			<prop oor:name="ToolboxStyle" oor:type="xs:short">
 				<!-- UIHints: Tools  Options General View -->
@@ -5977,7 +5977,7 @@
                         <author>FS</author>
                         <desc>Enables or disables a help section at the bottom of the property browser, which shows the help of the currently active property.</desc>
                     </info>
-                    <value>false</value>
+                    <value>true</value>
                 </prop>
             </group>
 		</group>
@@ -6118,7 +6118,7 @@
                		<author>mod</author>
                 	<desc>Specifies if search includes comments(SwPostItFields)</desc>
                 </info>
-		<value>false</value>
+		<value>true</value>
             </prop>
             <group oor:name="Japanese">
 				<info>
diff --git a/officecfg/registry/schema/org/openoffice/Office/Draw.xcs b/officecfg/registry/schema/org/openoffice/Office/Draw.xcs
index a48598e..214d4ea 100644
--- officecfg/registry/schema/org/openoffice/Office/Draw.xcs
+++ officecfg/registry/schema/org/openoffice/Office/Draw.xcs
@@ -71,7 +71,7 @@
 						<desc>Indicates whether to show help lines when moving an object.</desc>
 						<label>Guides when moving</label>
 					</info>
-					<value>false</value>
+					<value>true</value>
 				</prop>
 				<prop oor:name="Bezier" oor:type="xs:boolean">
 					<!-- OldPath: Draw/Layout/Display -->
@@ -82,7 +82,7 @@
 						<desc>Indicates whether to show all control points in the Bezier curve editor.</desc>
 						<label>All control points in Bezier editor</label>
 					</info>
-					<value>false</value>
+					<value>true</value>
 				</prop>
 				<prop oor:name="Contour" oor:type="xs:boolean">
 					<!-- OldPath: Draw/Layout/Display -->
@@ -282,7 +282,7 @@
 					<desc>Indicates whether moving while holding the Control key makes a copy of the moved object.</desc>
 					<label>Copy while moving</label>
 				</info>
-				<value>false</value>
+				<value>true</value>
 			</prop>
 			<prop oor:name="ObjectMoveable" oor:type="xs:boolean">
 				<!-- OldPath: Draw/Other -->
@@ -373,7 +373,7 @@
           <desc>Indicates whether to show simple (true) or 3D-handles (false).</desc>
           <label>Simple Handles</label>
         </info>
-        <value>true</value>
+        <value>false</value>
       </prop>
       <prop oor:name="BigHandles" oor:type="xs:boolean">
 				<!-- OldPath: Draw/Other -->
diff --git a/officecfg/registry/schema/org/openoffice/Office/Impress.xcs b/officecfg/registry/schema/org/openoffice/Office/Impress.xcs
index 947a26c..5377f26 100644
--- officecfg/registry/schema/org/openoffice/Office/Impress.xcs
+++ officecfg/registry/schema/org/openoffice/Office/Impress.xcs
@@ -108,7 +108,7 @@
             <desc>Indicates whether to show the rulers at the left and at the top of the working area.</desc>
             <label>Rulers visible</label>
           </info>
-          <value>false</value>
+          <value>true</value>
         </prop>
         <prop oor:name="Helpline" oor:type="xs:boolean">
           <!-- OldPath: Impress/Layout/Display -->
@@ -128,7 +128,7 @@
             <desc>Indicates whether to show help-lines when moving an object.</desc>
             <label>Guides when moving</label>
           </info>
-          <value>false</value>
+          <value>true</value>
         </prop>
         <prop oor:name="Bezier" oor:type="xs:boolean">
           <!-- OldPath: Impress/Layout/Display -->
@@ -138,7 +138,7 @@
             <desc>Indicates whether to show all control points in the Bezier curve editor.</desc>
             <label>All control points in Bezier editor</label>
           </info>
-          <value>false</value>
+          <value>true</value>
         </prop>
         <prop oor:name="Contour" oor:type="xs:boolean">
           <!-- OldPath: Impress/Layout/Display -->
@@ -331,7 +331,7 @@
           <desc>Indicates whether moving while holding the Control key makes a copy of the moved object.</desc>
           <label>Copy while moving</label>
         </info>
-        <value>false</value>
+        <value>true</value>
       </prop>
       <prop oor:name="ObjectMoveable" oor:type="xs:boolean">
         <!-- OldPath: Impress/Other -->
diff --git a/officecfg/registry/schema/org/openoffice/Office/Recovery.xcs b/officecfg/registry/schema/org/openoffice/Office/Recovery.xcs
index f424180..52ddc96 100644
--- officecfg/registry/schema/org/openoffice/Office/Recovery.xcs
+++ officecfg/registry/schema/org/openoffice/Office/Recovery.xcs
@@ -93,7 +93,7 @@
                     <author>MAV</author>
                     <desc>Determines if the office is allowed to show request to store modified documents on session shutdown.</desc>
                 </info>
-                <value>false</value>
+                <value>true</value>
             </prop>
         </group>
         <group oor:name="RecoveryInfo">
@@ -144,7 +144,7 @@
                 <info>
                     <desc> </desc>
                 </info>
-                <value>15</value>
+                <value>10</value>
             </prop>
             <prop oor:name="MinSpaceDocSave" oor:type="xs:int">
                 <info>
diff --git a/officecfg/registry/schema/org/openoffice/Office/Scripting.xcs b/officecfg/registry/schema/org/openoffice/Office/Scripting.xcs
index b77cb00..0bf5ba0 100644
--- officecfg/registry/schema/org/openoffice/Office/Scripting.xcs
+++ officecfg/registry/schema/org/openoffice/Office/Scripting.xcs
@@ -56,7 +56,7 @@
 				<info>
 					<desc>Show Basic scripts in assignment dialogs</desc>
 				</info>
-				<value>false</value>
+				<value>true</value>
 			</prop>
 			<prop oor:name="ShowSF" oor:type="xs:boolean">
 				<info>
diff --git a/officecfg/registry/schema/org/openoffice/Office/UI.xcs b/officecfg/registry/schema/org/openoffice/Office/UI.xcs
index 48d018d..dad5871 100644
--- officecfg/registry/schema/org/openoffice/Office/UI.xcs
+++ officecfg/registry/schema/org/openoffice/Office/UI.xcs
@@ -661,7 +661,7 @@
 				<info>
 					<desc>Specifies whether the dialog should fill its view asynchronously. Asnychronous filling allows the user to cancel blocking operations, which may be usefull for instance for mounted, but dead NFS devices.</desc>
 				</info>
-				<value>false</value>
+				<value>true</value>
 			</prop>
 			<group oor:name="Timeout">
 				<info>
@@ -692,7 +692,7 @@ These settings are not evaluated if the FillAsynchronously property is false.</d
 							</info>
 						</minInclusive>
 					</constraints>
-					<value>30000</value>
+					<value>20000</value>
 				</prop>
 			</group>
 		</group>
diff --git a/officecfg/registry/schema/org/openoffice/Office/Writer.xcs b/officecfg/registry/schema/org/openoffice/Office/Writer.xcs
index 3c3c5db..4be07b6 100644
--- officecfg/registry/schema/org/openoffice/Office/Writer.xcs
+++ officecfg/registry/schema/org/openoffice/Office/Writer.xcs
@@ -1131,7 +1131,7 @@
 						<desc>Determines whether certain settings are applied (display of tab stops, spaces, breaks)</desc>
 						<label>Meta characters</label>
 					</info>
-					<value>false</value>
+					<value>true</value>
 				</prop>
 				<prop oor:name="ParagraphEnd" oor:type="xs:boolean">
 					<!-- OldPath: Writer/Contents/Nonprinting_characters -->
@@ -1142,7 +1142,7 @@
 						<desc>Specifies if paragraph ends are shown on the screen.</desc>
 						<label>Paragraph end</label>
 					</info>
-					<value>false</value>
+					<value>true</value>
 				</prop>
 				<prop oor:name="OptionalHyphen" oor:type="xs:boolean">
 					<!-- OldPath: Writer/Contents/Nonprinting_characters -->
@@ -1186,7 +1186,7 @@
 						<desc>Specifies if breaks are shown on the screen.</desc>
 						<label>Tabs</label>
 					</info>
-					<value>false</value>
+					<value>true</value>
 				</prop>
 				<prop oor:name="Break" oor:type="xs:boolean">
 					<!-- OldPath: Writer/Contents/Nonprinting_characters -->
@@ -1197,7 +1197,7 @@
 						<desc>Specifies if paragraph ends are shown on the screen.</desc>
 						<label>Breaks</label>
 					</info>
-					<value>false</value>
+					<value>true</value>
 				</prop>
 				<prop oor:name="HiddenText" oor:type="xs:boolean">
 					<!-- OldPath: Writer/Contents/Nonprinting_characters -->
@@ -1208,7 +1208,7 @@
                         <desc>Specifies if hidden text fields are displayed on the screen.</desc>
                         <label>Fields: Hidden text</label>
 					</info>
-					<value>false</value>
+					<value>true</value>
 				</prop>
 				<prop oor:name="HiddenParagraph" oor:type="xs:boolean">
 					<!-- OldPath: Writer/Contents/Nonprinting_characters -->
@@ -1228,7 +1228,7 @@
                         <desc>Specifies if hidden paragraphs are displayed on the screen.</desc>
                         <label>Hidden text</label>
                     </info>
-                    <value>false</value>
+                    <value>true</value>
                 </prop>
             </group>
 			<group oor:name="Update">
@@ -1304,7 +1304,7 @@
 						<desc>Displays guides around the frames when frames are moved.</desc>
 						<label>Guides</label>
 					</info>
-					<value>false</value>
+					<value>true</value>
 				</prop>
 				<prop oor:name="SimpleControlPoint" oor:type="xs:boolean">
 					<!-- OldPath: Writer/Layout/Lines -->
@@ -1326,7 +1326,7 @@
 						<desc>Displays larger control points.</desc>
 						<label>Large control points</label>
 					</info>
-					<value>false</value>
+					<value>true</value>
 				</prop>
 			</group>
 			<group oor:name="Window">
@@ -3085,7 +3085,7 @@
 									<desc>Specifies the character that appears directly after the number of the heading or chapter level.</desc>
 									<label>Separator</label>
 								</info>
-								<value>.</value>
+								<value>. </value>
 							</prop>
                             <prop oor:name="CharacterStyle" oor:type="xs:string">
                                 <info>
@@ -3198,7 +3198,7 @@
 									<desc>Specifies the character that appears directly after the number of the heading or chapter level.</desc>
 									<label>Separator</label>
 								</info>
-								<value>.</value>
+								<value>. </value>
 							</prop>
                             <prop oor:name="Position" oor:type="xs:int">
                                 <info>
@@ -3360,7 +3360,7 @@
 									<desc>Specifies the character that appears directly after the number of the heading or chapter level.</desc>
 									<label>Separator</label>
 								</info>
-								<value>.</value>
+								<value>. </value>
 							</prop>
                             <prop oor:name="CharacterStyle" oor:type="xs:string">
                                 <info>
@@ -3498,7 +3498,7 @@
 									<desc>Specifies the character that appears directly after the number of the heading or chapter level.</desc>
 									<label>Separator</label>
 								</info>
-								<value>.</value>
+								<value>. </value>
 							</prop>
                             <prop oor:name="CharacterStyle" oor:type="xs:string">
                                 <info>
@@ -3636,7 +3636,7 @@
 									<desc>Specifies the character that appears directly after the number of the heading or chapter level.</desc>
 									<label>Separator</label>
 								</info>
-								<value>.</value>
+								<value>. </value>
 							</prop>
                             <prop oor:name="CharacterStyle" oor:type="xs:string">
                                 <info>
@@ -3774,7 +3774,7 @@
 									<desc>Specifies the character that appears directly after the number of the heading or chapter level.</desc>
 									<label>Separator</label>
 								</info>
-								<value>.</value>
+								<value>. </value>
 							</prop>
                             <prop oor:name="CharacterStyle" oor:type="xs:string">
                                 <info>
@@ -3912,7 +3912,7 @@
 									<desc>Specifies the character that appears directly after the number of the heading or chapter level.</desc>
 									<label>Separator</label>
 								</info>
-								<value>.</value>
+								<value>. </value>
 							</prop>
                             <prop oor:name="CharacterStyle" oor:type="xs:string">
                                 <info>
@@ -4050,7 +4050,7 @@
 									<desc>Specifies the character that appears directly after the number of the heading or chapter level.</desc>
 									<label>Separator</label>
 								</info>
-								<value>.</value>
+								<value>. </value>
 							</prop>
                             <prop oor:name="CharacterStyle" oor:type="xs:string">
                                 <info>
@@ -4188,7 +4188,7 @@
 									<desc>Specifies the character that appears directly after the number of the heading or chapter level.</desc>
 									<label>Separator</label>
 								</info>
-								<value>.</value>
+								<value>. </value>
 							</prop>
                             <prop oor:name="CharacterStyle" oor:type="xs:string">
                                 <info>
@@ -4219,14 +4219,14 @@
 						<desc>Specifies if the table has a header.</desc>
 						<label>Header</label>
 					</info>
-                                        <value>false</value>
+                                        <value>true</value>
 				</prop>
 				<prop oor:name="RepeatHeader" oor:type="xs:boolean">
 					<info>
 						<desc>Specifies if the header in the new page is adopted when inserting a page break. This option is only available if Header = True.</desc>
 						<label>Repeat header</label>
 					</info>
-                                        <value>false</value>
+                                        <value>true</value>
 				</prop>
 				<prop oor:name="Split" oor:type="xs:boolean">
 					<info>
@@ -4261,7 +4261,7 @@
 						<desc>Specifies if spacing between paragraphs and tables are added to ensure the compatibility to MS Word documents.</desc>
 						<label>Add spacing between paragraphs and tables</label>
 					</info>
-					<value>false</value>
+					<value>true</value>
 				</prop>
 				<prop oor:name="AddToStart" oor:type="xs:boolean">
 					<!-- OldPath: Writer/Other/Compatibility -->
@@ -4272,7 +4272,7 @@
 						<desc>Specifies if paragraph and table spacing are added to the start of pages to ensure compatibility with MS Word documents.</desc>
 						<label>Add paragraph and table spacing to start of pages</label>
 					</info>
-					<value>false</value>
+					<value>true</value>
 				</prop>
 			</group>
 		</group>
@@ -4315,7 +4315,7 @@
 						<desc>Specifies if a preview of the AutoText is displayed.</desc>
 						<label>Show Preview</label>
 					</info>
-					<value>false</value>
+					<value>true</value>
 				</prop>
 				<prop oor:name="ShowToolTip" oor:type="xs:boolean">
 					<!-- OldPath: Writer/Dialog/AutoText -->
@@ -4326,7 +4326,7 @@
 						<desc>Specifies if the Tip help of the AutoText is displayed.</desc>
 						<label>Display AutoText name while typing</label>
 					</info>
-					<value>false</value>
+					<value>true</value>
 				</prop>
                                 <prop oor:name="SearchInAllCategories" oor:type="xs:boolean">
                                         <!-- #b6633427# -->
@@ -4335,7 +4335,7 @@
                                                 <desc>Specifies if the search for an AutoText shortcut will always be done in all AutoText categories.</desc>
                                                 <label>Search for short cut in all categories</label>
                                         </info>
-                                        <value>false</value>
+                                        <value>true</value>
                                 </prop>
 			</group>
 			<group oor:name="Format">
@@ -4763,7 +4763,7 @@
 						<desc>Sets the maximum number of words to be recalled.</desc>
 						<label>Max. entries</label>
 					</info>
-					<value>500</value>
+					<value>2500</value>
 				</prop>
 				<prop oor:name="CollectWords" oor:type="xs:boolean">
 					<!-- OldPath: Writer/AutoComplete/Words -->
@@ -4796,7 +4796,7 @@
 						<desc>Specifies if a blank is added after the automatic word completion.</desc>
 						<label>Append space</label>
 					</info>
-					<value>false</value>
+					<value>true</value>
 				</prop>
 				<prop oor:name="ShowAsTip" oor:type="xs:boolean">
 					<!-- OldPath: Writer/AutoComplete/Words -->
@@ -4807,7 +4807,7 @@
 						<desc>Specifies if the suggestions for the automatic word completion function should be displayed as Tip helps.</desc>
 						<label>Show as tip</label>
 					</info>
-					<value>false</value>
+					<value>true</value>
 				</prop>
 				<prop oor:name="AcceptKey" oor:type="xs:int">
 					<!-- OldPath: Writer/AutoComplete/Words -->
@@ -5676,7 +5676,7 @@
                         	<desc>configures if comment anchor should be shown or hidden when display is off</desc>
                         	<label>Comments Ankor on/off</label>
                 	</info>
-                	<value>false</value>
+                	<value>true</value>
                 	</prop>
                 </group>
 		<group oor:name="Navigator">
diff --git a/officecfg/registry/schema/org/openoffice/Office/WriterWeb.xcs b/officecfg/registry/schema/org/openoffice/Office/WriterWeb.xcs
index 8738d9d..9389fca 100644
--- officecfg/registry/schema/org/openoffice/Office/WriterWeb.xcs
+++ officecfg/registry/schema/org/openoffice/Office/WriterWeb.xcs
@@ -116,7 +116,7 @@
 						<desc>Determines whether certain settings are applied or not (display of tab stops, spaces, breaks).</desc>
 						<label>Meta characters</label>
 					</info>
-					<value>false</value>
+					<value>true</value>
 				</prop>
 				<prop oor:name="ParagraphEnd" oor:type="xs:boolean">
 					<!-- OldPath: HTML_Editor/Content/Nonprinting_characters -->
@@ -127,7 +127,7 @@
 						<desc>Specifies if paragraph ends are shown on the screen.</desc>
 						<label>Paragraph end</label>
 					</info>
-					<value>false</value>
+					<value>true</value>
 				</prop>
 				<prop oor:name="OptionalHyphen" oor:type="xs:boolean">
 					<!-- OldPath: HTML_Editor/Content/Nonprinting_characters -->
@@ -171,7 +171,7 @@
 						<desc>Specifies if breaks are shown on the screen.</desc>
 						<label>Breaks</label>
 					</info>
-					<value>false</value>
+					<value>true</value>
 				</prop>
 			</group>
 			<group oor:name="Highlighting">
@@ -208,7 +208,7 @@
 						<desc>Displays guides around the frames when frames are moved.</desc>
 						<label>Guides</label>
 					</info>
-					<value>false</value>
+					<value>true</value>
 				</prop>
 				<prop oor:name="SimpleControlPoint" oor:type="xs:boolean">
 					<!-- OldPath: HTML_Editor/Layout/Lines -->
@@ -230,7 +230,7 @@
 						<desc>Displays larger control points.</desc>
 						<label>Large control points</label>
 					</info>
-					<value>false</value>
+					<value>true</value>
 				</prop>
 			</group>
 			<group oor:name="Window">
@@ -344,7 +344,7 @@
 						<desc>Specifies if the vertical ruler is displayed.</desc>
 						<label>Vertical Ruler</label>
 					</info>
-					<value>false</value>
+					<value>true</value>
 				</prop>
 				<prop oor:name="VerticalRulerUnit" oor:type="xs:int">
 					<!-- OldPath: Writer/Layout -->
@@ -943,7 +943,7 @@
 						<desc>Specifies if the table has a header</desc>
 						<label>Header</label>
 					</info>
-                                        <value>false</value>
+                                        <value>true</value>
 				</prop>
 				<prop oor:name="RepeatHeader" oor:type="xs:boolean">
 					<!-- OldPath: HTML_Editor/Insert/Tables -->
@@ -954,7 +954,7 @@
 						<desc>Specifies if the header in the new page is adopted when inserting a page break. This option is only available if Header=True.</desc>
 						<label>Repeat header</label>
 					</info>
-                                        <value>false</value>
+                                        <value>true</value>
 				</prop>
 				<prop oor:name="Border" oor:type="xs:boolean">
 					<!-- OldPath: HTML_Editor/Insert/Tables -->
