--- solenv/bin/packregistry.xslt
+++ solenv/bin/packregistry.xslt
@@ -29,9 +29,6 @@
 <xsl:stylesheet version="1.0" xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
    xmlns:oor="http://openoffice.org/2001/registry">
   <xsl:param name="prefix"/>
-  <xsl:strip-space elements="*"/>
-  <xsl:preserve-space elements="value it"/>
-    <!-- TODO: strip space from "value" elements that have "it" children -->
   <xsl:template match="/">
     <oor:data xmlns:xs="http://www.w3.org/2001/XMLSchema"
        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
