--- scp2/source/ooo/file_library_ooo.scp	2009-09-17 10:28:37.000000000 +0200
+++ scp2/source/ooo/file_library_ooo.scp	2009-09-17 10:27:52.000000000 +0200
@@ -698,6 +698,21 @@
   #endif
 End
 
+File gid_File_Lib_Pixman
+    TXT_FILE_BODY;
+    Styles = (PACKED);
+    Dir = gid_Dir_Program;
+  #ifdef UNX
+    #ifdef MACOSX
+       Name = STRING(CONCAT3(libpixman-1,.0,UNXSUFFIX));
+    #else
+       Name = STRING(CONCAT3(libpixman-1,UNXSUFFIX,.0));
+    #endif
+  #elif defined(WNT)
+    Name = "libpixman-1-0.dll";
+  #endif
+End
+
 #endif
 
 STD_UNO_LIB_FILE( gid_File_Lib_Frm , frm)
