--- officecfg/registry/data/org/openoffice/Office/Common.xcu.ark	2006-03-12 03:53:26.000000000 +0000
+++ officecfg/registry/data/org/openoffice/Office/Common.xcu	2006-03-12 03:59:16.000000000 +0000
@@ -651,8 +651,7 @@
   </node>
   <node oor:name="ExternalMailer" install:module="unixdesktop">
     <prop oor:name="Program">
-      <value oor:external=
-          "com.sun.star.configuration.backend.DesktopBackend ExternalMailer"/>
+      <value>/usr/kde/3.5/bin/kmail</value>
     </prop>
   </node>
   <node oor:name="Font">
