diff -urEbwB --exclude=wntmsci12.pro reportbuilder/prj/d.lst reportbuilder/prj/d.lst
--- reportbuilder/prj/d.lst	2010-04-26 12:28:00.843750000 +0200
+++ reportbuilder/prj/d.lst	2010-04-26 12:28:59.796875000 +0200
@@ -1,6 +1,6 @@
 # Extensions
 
-..\%__SRC%\bin\*.oxt %_DEST%\bin%_EXT%\*.oxt
+..\%__SRC%\bin\oracle-report-builder.oxt %_DEST%\bin%_EXT%\sun-report-builder.oxt
 
 # License
 mkdir: %_DEST%\bin%_EXT%\reportbuilder
