diff -urEbwB --exclude=wntmsci12.pro sdext/prj/d.lst sdext/prj/d.lst
--- sdext/prj/d.lst	2010-04-26 12:19:16.062500000 +0200
+++ sdext/prj/d.lst	2010-04-26 12:20:40.187500000 +0200
@@ -1,11 +1,11 @@
 mkdir: %_DEST%\bin%_EXT%\minimizer
 mkdir: %_DEST%\bin%_EXT%\presenter
 mkdir: %_DEST%\bin%_EXT%\pdfimport
-..\%__SRC%\bin\presentation-minimizer.oxt %_DEST%\bin%_EXT%\presentation-minimizer.oxt
+..\%__SRC%\bin\presentation-minimizer.oxt %_DEST%\bin%_EXT%\sun-presentation-minimizer.oxt
 ..\%__SRC%\bin\presenter-screen.oxt %_DEST%\bin%_EXT%\presenter-screen.oxt
 ..\%__SRC%\bin\pdfimport.oxt %_DEST%\bin%_EXT%\pdfimport.oxt
 ..\%__SRC%\bin\pdf2xml.* %_DEST%\bin%_EXT%\pdf2xml.*
 ..\%__SRC%\bin\pdfunzip.* %_DEST%\bin%_EXT%\pdfunzip.*
 ..\%__SRC%\bin\pdfimport.oxt %_DEST%\bin%_EXT%\pdfimport\pdfimport.oxt
-..\%__SRC%\bin\presentation-minimizer.oxt %_DEST%\bin%_EXT%\minimizer\presentation-minimizer.oxt
+..\%__SRC%\bin\presentation-minimizer.oxt %_DEST%\bin%_EXT%\minimizer\sun-presentation-minimizer.oxt
 ..\%__SRC%\bin\presenter-screen.oxt %_DEST%\bin%_EXT%\presenter\presenter-screen.oxt
