diff -urEbwB --exclude=wntmsci12.pro scp2/source/extensions/file_extensions.scp scp2/source/extensions/file_extensions.scp
--- scp2/source/extensions/file_extensions.scp	2010-04-26 12:24:00.468750000 +0200
+++ scp2/source/extensions/file_extensions.scp	2010-04-26 13:06:42.625000000 +0200
@@ -41,11 +41,11 @@
 
 #ifndef WITHOUT_EXTENSION_MINIMIZER
 
-File gid_File_Oxt_Presentation_Minimizer
+File gid_File_Oxt_Sun_Presentation_Minimizer
     TXT_FILE_BODY;
     Styles = (PACKED);
     Dir = gid_Brand_Dir_Share_Extension_Install;
-    Name = "presentation-minimizer.oxt";
+    Name = "sun-presentation-minimizer.oxt";
 End
 
 #endif
@@ -54,11 +54,11 @@
 
 #ifndef WITHOUT_EXTENSION_REPORTDESIGN
 
-File gid_File_Oxt_Oracle_Report_Builder
+File gid_File_Oxt_Sun_Report_Builder
     TXT_FILE_BODY;
     Styles = (PACKED);
     Dir = gid_Brand_Dir_Share_Extension_Install;
-    Name = "oracle-report-builder.oxt";
+    Name = "sun-report-builder.oxt";
 End
 
 #endif
@@ -67,7 +67,7 @@
 
 #ifndef WITHOUT_EXTENSION_MEDIAWIKI
 
-File gid_File_Oxt_MediaWiki
+File gid_File_Oxt_Sun_MediaWiki
     TXT_FILE_BODY;
     Styles = (PACKED);
     Dir = gid_Brand_Dir_Share_Extension_Install;
@@ -80,7 +80,7 @@
 
 #ifndef WITHOUT_EXTENSION_PRESENTER_SCREEN
 
-File gid_File_Oxt_Presenter_Screen
+File gid_File_Oxt_Sun_Presenter_Screen
     TXT_FILE_BODY;
     Styles = (PACKED);
     Dir = gid_Brand_Dir_Share_Extension_Install;
@@ -93,7 +93,7 @@
 
 #ifndef WITHOUT_EXTENSION_PDFIMPORT
 
-File gid_File_Oxt_PDF_Import
+File gid_File_Oxt_Sun_PDF_Import
     TXT_FILE_BODY;
     Styles = (PACKED);
     Dir = gid_Brand_Dir_Share_Extension_Install;
