--- sc/util/createExtPackage.pl.orig	2009-11-26 22:26:31.000000000 +0100
+++ sc/util/createExtPackage.pl	2009-11-26 22:26:45.000000000 +0100
@@ -66,6 +66,6 @@
 $zip->addFile( $rdbName, $rdb );
 $zip->addFile( $libName, $lib );
 $zip->addDirectory( "META-INF" );
-$zip->addFile( "META-INF/manifest.xml", "manifest.xml" );
+#$zip->addFile( "META-INF/manifest.xml", "manifest.xml" );
 $zip->addString( $content2, "META-INF/manifest.xml" );
 exit( $zip->writeToFileNamed($zipName) );
