--- saxon/makefile.mk.old	2008-10-15 11:43:33.000000000 +0200
+++ saxon/makefile.mk	2009-10-16 16:16:27.000000000 +0200
@@ -54,7 +54,7 @@ TARFILE_ROOTDIR=src
 
 PATCH_FILE_NAME=saxon-9.0.0.7-bj.patch
 
-BUILD_ACTION=$(ANT) $(ANT_FLAGS) -Dsolarbindir=$(SOLARBINDIR) jar-bj 
+BUILD_ACTION=$(ANT) $(ANT_FLAGS) -Dsolar.bin=$(SOLARPIECE)$/$(INPATH)$/bin jar-bj
 
 OUT2CLASS= saxon-build$/9.0.0.7$/bj$/saxon9.jar
 
