--- mysqlc/source/makefile.mk.old	2010-04-28 11:35:17.000000000 +0200
+++ mysqlc/source/makefile.mk	2010-05-11 10:30:58.000000000 +0200
@@ -264,7 +264,7 @@ EXTENSION_PACKDEPS+=$(COMPONENT_MYSQL_CP
 .INCLUDE : target.mk
 .INCLUDE : extension_post.mk
 
-$(COMPONENT_IMAGES) : $(SOLARSRC)$/$(RSCDEFIMG)$/desktop$/res$/$$(@:f)
+$(COMPONENT_IMAGES) : $(SOLARVER)$/$(RSCDEFIMG)$/desktop$/res$/$$(@:f)
     @@-$(MKDIRHIER) $(@:d)
     $(COPY) $< $@
 
