--- l10n/prj/build.lst.old	2009-10-08 08:11:52.000000000 +0200
+++ l10n/prj/build.lst	2009-10-16 19:29:55.000000000 +0200
@@ -1,3 +1,3 @@
-l10n l10n  : transex3 NULL
+l10n l10n  : solenv NULL
 l10n l10n\source     nmake - all l10n_source NULL
 l10n l10n\util          nmake - all l10n_util l10n_source NULL
--- l10n/source/makefile.mk.old	2009-10-08 08:11:52.000000000 +0200
+++ l10n/source/makefile.mk	2009-10-19 16:57:24.000000000 +0200
@@ -93,6 +93,6 @@ $(COMMONMISC)$/merge.done : $(all_sdfs)
     -rm -rf $(COMMONMISC)$/sdf$(INPATH)_begone
     -$(MKDIRHIER) $(COMMONMISC)$/sdf
 .ENDIF			# "$(L10n_LOCK)" != "YES"
-    $(PERL) $(SOLARVER)$/$(INPATH)$/bin$(UPDMINOREXT)$/fast_merge.pl -sdf_files $(mktmp $<) -merge_dir $(COMMONMISC)$/sdf && echo $< >$@
+    $(PERL) fast_merge.pl -sdf_files $(mktmp $<) -merge_dir $(COMMONMISC)$/sdf && echo $< >$@
 
 .ENDIF
--- solenv/inc/target.mk.old	2009-10-20 16:25:25.000000000 +0200
+++ solenv/inc/target.mk	2009-10-22 11:55:59.000000000 +0200
@@ -1394,12 +1394,20 @@ $(UNIXTEXT) : $(UNIXTEXT:f)
 .IF "$(WITH_LANG)"!=""
 .IF "$(LOCALIZATION_FOUND)"==""
 .IF "$(LOCALIZESDF)"!=""
+
+# check whether zip file with sdf files exists for this project
+TRYSDFZIP=$(SOLARCOMMONSDFDIR)/$(PRJNAME).zip
+HAVE_LOCALIZESDFZIP:=$(strip $(shell @+$(IFEXIST) $(TRYSDFZIP) $(THEN) echo 'TRUE' $(FI)))
+
+.IF "$(HAVE_LOCALIZESDFZIP)"=="TRUE"
 "$(LOCALIZESDF)" : $(SOLARCOMMONSDFDIR)/$(PRJNAME).zip
     @@-$(MKDIRHIER) $(@:d)
     @@-$(MKDIRHIER) $(COMMONMISC)/$(PRJNAME)_$(TARGET)
     @@$(IFNOTEXIST) $(LOCALIZESDF) $(THEN) unzip -o -d $(COMMONMISC)/$(PRJNAME)_$(TARGET) $(SOLARCOMMONSDFDIR)/$(PRJNAME).zip $(FI)
     @@-cp -r $(COMMONMISC)/$(PRJNAME)_$(TARGET)/* $(COMMONMISC)/$(PRJNAME)
     @@-$(RM) -rf $(COMMONMISC)/$(PRJNAME)_$(TARGET)
+.ENDIF			# "$(HAVE_LOCALIZESDFZIP)"=="YES"
+
 .ENDIF			# "$(LOCALIZESDF)"!=""
 .ENDIF			# "$(LOCALIZATION_FOUND)"==""
 .ENDIF			# "$(WITH_LANG)"!=""
