--- reportbuilder/registry/data/org/openoffice/Office/UI/makefile.mk.old	2010-02-02 12:49:41.000000000 +0100
+++ reportbuilder/registry/data/org/openoffice/Office/UI/makefile.mk	2010-03-12 17:11:54.000000000 +0100
@@ -49,4 +49,8 @@ MODULEFILES=
 
 .INCLUDE : target.mk
 
-   
+ALLTAR: "$(PWD)$/$(MISC)$/$(EXTNAME)$/registry$/schema$/$(PACKAGEDIR)$/WindowState.xcs"
+
+"$(PWD)$/$(MISC)$/$(EXTNAME)$/registry$/schema$/$(PACKAGEDIR)$/WindowState.xcs" : $(SPLITINSTALLDIR)$/..$/share$/registry$/schema$/$(PACKAGEDIR)$/WindowState.xcs
+    @@-$(MKDIRHIER) $(@:d)
+    $(COPY) $< $@
--- reportbuilder/registry/data/org/openoffice/Office/makefile.mk.old	2010-02-02 12:49:41.000000000 +0100
+++ reportbuilder/registry/data/org/openoffice/Office/makefile.mk	2010-03-12 17:11:54.000000000 +0100
@@ -54,19 +54,19 @@ XCUFILES= \
 
 ALLTAR: "$(PWD)$/$(MISC)$/$(EXTNAME)$/registry$/schema$/$(PACKAGEDIR)$/Embedding.xcs" "$(PWD)$/$(MISC)$/$(EXTNAME)$/registry$/schema$/$(PACKAGEDIR)$/ExtendedColorScheme.xcs" "$(PWD)$/$(MISC)$/$(EXTNAME)$/registry$/schema$/$(PACKAGEDIR)$/Paths.xcs"
 
-"$(PWD)$/$(MISC)$/$(EXTNAME)$/registry$/schema$/$(PACKAGEDIR)$/Embedding.xcs" : $(SOLARXMLDIR)$/registry$/schema$/$(PACKAGEDIR)$/Embedding.xcs
+"$(PWD)$/$(MISC)$/$(EXTNAME)$/registry$/schema$/$(PACKAGEDIR)$/Embedding.xcs" : $(SPLITINSTALLDIR)$/..$/share$/registry$/schema$/$(PACKAGEDIR)$/Embedding.xcs
     @@-$(MKDIRHIER) $(@:d)
     $(COPY) $< $@
 
-"$(PWD)$/$(MISC)$/$(EXTNAME)$/registry$/schema$/$(PACKAGEDIR)$/ExtendedColorScheme.xcs" : $(SOLARXMLDIR)$/registry$/schema$/$(PACKAGEDIR)$/ExtendedColorScheme.xcs
+"$(PWD)$/$(MISC)$/$(EXTNAME)$/registry$/schema$/$(PACKAGEDIR)$/ExtendedColorScheme.xcs" : $(SPLITINSTALLDIR)$/..$/share$/registry$/schema$/$(PACKAGEDIR)$/ExtendedColorScheme.xcs
     @@-$(MKDIRHIER) $(@:d)
     $(COPY) $< $@
 
-"$(PWD)$/$(MISC)$/$(EXTNAME)$/registry$/schema$/$(PACKAGEDIR)$/Paths.xcs" : $(SOLARXMLDIR)$/registry$/schema$/$(PACKAGEDIR)$/Paths.xcs
+"$(PWD)$/$(MISC)$/$(EXTNAME)$/registry$/schema$/$(PACKAGEDIR)$/Paths.xcs" : $(SPLITINSTALLDIR)$/..$/share$/registry$/schema$/$(PACKAGEDIR)$/Paths.xcs
     @@-$(MKDIRHIER) $(@:d)
     $(COPY) $< $@
 
-"$(PWD)$/$(MISC)$/$(EXTNAME)$/registry$/schema$/$(PACKAGEDIR)$/Accelerators.xcs" : $(SOLARXMLDIR)$/registry$/schema$/$(PACKAGEDIR)$/Accelerators.xcs
+"$(PWD)$/$(MISC)$/$(EXTNAME)$/registry$/schema$/$(PACKAGEDIR)$/Accelerators.xcs" : $(SPLITINSTALLDIR)$/..$/share$/registry$/schema$/$(PACKAGEDIR)$/Accelerators.xcs
     @@-$(MKDIRHIER) $(@:d)
     $(COPY) $< $@
 
--- reportbuilder/registry/data/org/openoffice/TypeDetection/makefile.mk.old	2010-02-02 12:49:41.000000000 +0100
+++ reportbuilder/registry/data/org/openoffice/TypeDetection/makefile.mk	2010-03-12 17:11:54.000000000 +0100
@@ -50,10 +50,10 @@ XCUFILES= \
 
 ALLTAR: "$(PWD)$/$(MISC)$/$(EXTNAME)$/registry$/schema$/$(PACKAGEDIR)$/Filter.xcs" "$(PWD)$/$(MISC)$/$(EXTNAME)$/registry$/schema$/$(PACKAGEDIR)$/Types.xcs"
 
-"$(PWD)$/$(MISC)$/$(EXTNAME)$/registry$/schema$/$(PACKAGEDIR)$/Filter.xcs" : $(SOLARXMLDIR)$/registry$/schema$/$(PACKAGEDIR)$/Filter.xcs
+"$(PWD)$/$(MISC)$/$(EXTNAME)$/registry$/schema$/$(PACKAGEDIR)$/Filter.xcs" : $(SPLITINSTALLDIR)$/..$/share$/registry$/schema$/$(PACKAGEDIR)$/Filter.xcs
     @@-$(MKDIRHIER) $(@:d)
     $(COPY) $< $@
 
-"$(PWD)$/$(MISC)$/$(EXTNAME)$/registry$/schema$/$(PACKAGEDIR)$/Types.xcs" : $(SOLARXMLDIR)$/registry$/schema$/$(PACKAGEDIR)$/Types.xcs
+"$(PWD)$/$(MISC)$/$(EXTNAME)$/registry$/schema$/$(PACKAGEDIR)$/Types.xcs" : $(SPLITINSTALLDIR)$/..$/share$/registry$/schema$/$(PACKAGEDIR)$/Types.xcs
     @@-$(MKDIRHIER) $(@:d)
     $(COPY) $< $@
--- reportbuilder/registry/data/org/openoffice/makefile.mk.old	2010-02-02 12:49:41.000000000 +0100
+++ reportbuilder/registry/data/org/openoffice/makefile.mk	2010-03-12 17:11:54.000000000 +0100
@@ -49,6 +49,6 @@ XCUFILES= \
 
 ALLTAR: "$(PWD)$/$(MISC)$/$(EXTNAME)$/registry$/schema$/$(PACKAGEDIR)$/Setup.xcs"
 
-"$(PWD)$/$(MISC)$/$(EXTNAME)$/registry$/schema$/$(PACKAGEDIR)$/Setup.xcs": $(SOLARXMLDIR)$/registry$/schema$/$(PACKAGEDIR)$/Setup.xcs
+"$(PWD)$/$(MISC)$/$(EXTNAME)$/registry$/schema$/$(PACKAGEDIR)$/Setup.xcs": $(SPLITINSTALLDIR)$/..$/share$/registry$/schema$/$(PACKAGEDIR)$/Setup.xcs
     @@-$(MKDIRHIER) $(@:d)
     $(COPY) $< $@
--- reportbuilder/registry/schema/org/openoffice/Office/UI/makefile.mk.old	2010-02-02 12:49:41.000000000 +0100
+++ reportbuilder/registry/schema/org/openoffice/Office/UI/makefile.mk	2010-03-12 17:11:54.000000000 +0100
@@ -42,21 +42,22 @@ XCSFILES= \
     ReportCommands.xcs \
     DbReportWindowState.xcs
 
+XSLTPROC+= --stringparam root $(SPLITINSTALLDIR)$/..$/share$/registry$/schema
+
 MODULEFILES=
 
 .INCLUDE : target.mk
 
 ALLTAR: "$(PWD)$/$(MISC)$/$(EXTNAME)$/registry$/schema$/$(PACKAGEDIR)$/Commands.xcs" "$(PWD)$/$(MISC)$/$(EXTNAME)$/registry$/schema$/$(PACKAGEDIR)$/WindowState.xcs" "$(PWD)$/$(MISC)$/$(EXTNAME)$/registry$/schema$/$(PACKAGEDIR)$/Controller.xcs"
 
-"$(PWD)$/$(MISC)$/$(EXTNAME)$/registry$/schema$/$(PACKAGEDIR)$/Commands.xcs": $(SOLARXMLDIR)$/registry$/schema$/$(PACKAGEDIR)$/Commands.xcs
+"$(PWD)$/$(MISC)$/$(EXTNAME)$/registry$/schema$/$(PACKAGEDIR)$/Commands.xcs": $(SPLITINSTALLDIR)$/..$/share$/registry$/schema$/$(PACKAGEDIR)$/Commands.xcs
     @@-$(MKDIRHIER) $(@:d)
     $(COPY) $< $@
     
-"$(PWD)$/$(MISC)$/$(EXTNAME)$/registry$/schema$/$(PACKAGEDIR)$/WindowState.xcs" : $(SOLARXMLDIR)$/registry$/schema$/$(PACKAGEDIR)$/WindowState.xcs
+"$(PWD)$/$(MISC)$/$(EXTNAME)$/registry$/schema$/$(PACKAGEDIR)$/WindowState.xcs" : $(SPLITINSTALLDIR)$/..$/share$/registry$/schema$/$(PACKAGEDIR)$/WindowState.xcs
     @@-$(MKDIRHIER) $(@:d)
     $(COPY) $< $@
     
-"$(PWD)$/$(MISC)$/$(EXTNAME)$/registry$/schema$/$(PACKAGEDIR)$/Controller.xcs" : $(SOLARXMLDIR)$/registry$/schema$/$(PACKAGEDIR)$/Controller.xcs
+"$(PWD)$/$(MISC)$/$(EXTNAME)$/registry$/schema$/$(PACKAGEDIR)$/Controller.xcs" : $(SPLITINSTALLDIR)$/..$/share$/registry$/schema$/$(PACKAGEDIR)$/Controller.xcs
     @@-$(MKDIRHIER) $(@:d)
     $(COPY) $< $@
-
--- reportbuilder/registry/schema/org/openoffice/Office/makefile.mk.old	2010-02-02 12:49:41.000000000 +0100
+++ reportbuilder/registry/schema/org/openoffice/Office/makefile.mk	2010-03-12 17:11:54.000000000 +0100
@@ -45,6 +45,6 @@ XCSFILES= \
 
 ALLTAR: "$(PWD)$/$(MISC)$/$(EXTNAME)$/registry$/schema$/$(PACKAGEDIR)$/DataAccess.xcs"
 
-"$(PWD)$/$(MISC)$/$(EXTNAME)$/registry$/schema$/$(PACKAGEDIR)$/DataAccess.xcs" : $(SOLARXMLDIR)$/registry$/schema$/$(PACKAGEDIR)$/DataAccess.xcs
+"$(PWD)$/$(MISC)$/$(EXTNAME)$/registry$/schema$/$(PACKAGEDIR)$/DataAccess.xcs" : $(SPLITINSTALLDIR)$/..$/share$/registry$/schema$/$(PACKAGEDIR)$/DataAccess.xcs
     @@-$(MKDIRHIER) $(@:d)
     $(COPY) $< $@
--- reportbuilder/util/makefile.pmk.old	2010-02-02 12:49:41.000000000 +0100
+++ reportbuilder/util/makefile.pmk	2010-03-12 17:11:54.000000000 +0100
@@ -35,7 +35,7 @@ ABSXCSROOT=$(PWD)$/$(MISC)$/$(EXTNAME)
 XCSROOT=..
 # XCSROOT=$(SOLARXMLDIR)
 # ABSXCSROOT=$(SOLARXMLDIR)
-DTDDIR=$(SOLARXMLDIR)
+DTDDIR=$(SOLARVER)/xml$(EXT_UPDMINOR)
 PROCESSOUT=$(MISC)$/$(EXTNAME)
 PROCESSORDIR=$(SOLARBINDIR)
 
--- reportbuilder/util/makefile.mk.old	2010-04-28 11:35:17.000000000 +0200
+++ reportbuilder/util/makefile.mk	2010-05-11 10:43:42.000000000 +0200
@@ -129,11 +129,15 @@ $(EXTENSIONDIR)$/%.jar : $(SOLARBINDIR)$
     @@-$(MKDIRHIER) $(@:d)
     $(COPY) $< $@
 
+$(EXTENSIONDIR)$/%.jar : $(SOLARPIECE)$/$(INPATH)$/bin$/%.jar
+    @@-$(MKDIRHIER) $(@:d)
+    $(COPY) $< $@
+
 $(EXTENSIONDIR)$/readme_en-US.% : $(PRJ)$/license$/readme_en-US.%
     @@-$(MKDIRHIER) $(@:d)
     $(COPY) $< $@
 
-$(COMPONENT_IMAGES) : $(SOLARSRC)$/$(RSCDEFIMG)$/desktop$/res$/$$(@:f)
+$(COMPONENT_IMAGES) : $(SOLARVER)$/$(RSCDEFIMG)$/desktop$/res$/$$(@:f)
     @@-$(MKDIRHIER) $(@:d)
     $(COPY) $< $@
 
