--- odk/pack/gendocu/makefile.mk.old	2009-04-02 12:54:49.000000000 +0200
+++ odk/pack/gendocu/makefile.mk	2009-04-14 17:28:52.000000000 +0200
@@ -81,7 +81,7 @@ AUTODOCPARAMS= -lg c++ \
         -p salhelper $(INCOUT) -t salhelper \
         -p bridges $(INCOUT) -t bridges
 
-JAVADOCPARAMS= -use -splitindex -windowtitle "Java UNO Runtime Reference" -header $(JAVADOCREFNAME) -d $(DESTDIRGENJAVAREF) -sourcepath $(JAVA_SRC_DIR) -classpath $(SOLARBINDIR)$/ridl.jar -linkoffline ../../common/ref ./uno -linkoffline http://java.sun.com/j2se/1.4.1/docs/api ./java $(JAVA_PACKAGES)
+JAVADOCPARAMS= -use -splitindex -windowtitle "Java UNO Runtime Reference" -header $(JAVADOCREFNAME) -d $(DESTDIRGENJAVAREF) -sourcepath $(JAVA_SRC_DIR) -classpath $(SOLARPIECE)$/$(INPATH)$/bin$/ridl.jar -linkoffline ../../common/ref ./uno -linkoffline http://java.sun.com/j2se/1.4.1/docs/api ./java $(JAVA_PACKAGES)
 
 JAVADOCLOG = $(MISC)$/javadoc_log.txt
 
--- odk/util/makefile.pmk.old	2009-04-02 12:54:49.000000000 +0200
+++ odk/util/makefile.pmk	2009-04-14 17:35:01.000000000 +0200
@@ -68,7 +68,7 @@ CONVERTTAGSCRIPT=$(SOLARENV)/bin/convert
 CONVERTTAGFLAGNAME=converttags_html_files.flag
 CONVERTTAGFLAG = $(MISC)/$(CONVERTTAGFLAGNAME)
 
-MY_AUTODOC=$(AUGMENT_LIBRARY_PATH) $(WRAPCMD) $(SOLARBINDIR)/autodoc
+MY_AUTODOC=$(AUGMENT_LIBRARY_PATH) $(WRAPCMD) $(SOLARPIECE)/$(INPATH)/bin/autodoc
 
 .IF "$(OS)"=="FREEBSD" || "$(OS)"=="NETBSD"
 MY_COPY=$(GNUCOPY)
--- odk/util/odk_rules.pmk.old	2009-04-02 12:54:49.000000000 +0200
+++ odk/util/odk_rules.pmk	2009-04-15 16:52:42.000000000 +0200
@@ -54,13 +54,13 @@ $(DESTDIR)/configure.pl : $(PRJ)/configu
     @@-rm -f $@
     $(MY_TEXTCOPY) $(PRJ)/configure.pl $@
 
-$(DESTDIR)/config.guess : $(SRC_ROOT)/config.guess
+$(DESTDIR)/config.guess : $(SOLARVER)/bin/config.guess
     @@-rm -f $@
-    $(MY_TEXTCOPY) $(SRC_ROOT)/config.guess $@
+    $(MY_TEXTCOPY) $(SOLARVER)/bin/config.guess $@
 
-$(DESTDIR)/config.sub : $(SRC_ROOT)/config.sub
+$(DESTDIR)/config.sub : $(SOLARVER)/bin/config.sub
     @@-rm -f $@
-    $(MY_TEXTCOPY) $(SRC_ROOT)/config.sub $@
+    $(MY_TEXTCOPY) $(SOLARVER)/bin/config.sub $@
 
 $(DESTDIR)/setsdkenv_unix : $(PRJ)/setsdkenv_unix
     @@-rm -f $@
@@ -105,6 +105,10 @@ $(DESTDIRBIN)/%$(EXEPOSTFIX) : $(BINOUT)
     @@-rm -f $@
     $(GNUCOPY) $(BINOUT)/$(@:f) $@
 
+$(DESTDIRBIN)/%$(EXEPOSTFIX) : $(SOLARPIECE)/$(INPATH)/bin/%$(EXEPOSTFIX)
+    @@-rm -f $@
+    $(GNUCOPY) $(SOLARPIECE)/$(INPATH)/bin/$(@:f) $@
+
 $(DESTDIRBIN)/%.bin : $(BINOUT)/%.bin
     @@-rm -f $@
     $(GNUCOPY) $(BINOUT)/$(@:f) $@
