--- scp2/prj/build.lst.old	2010-03-05 19:59:27.000000000 +0100
+++ scp2/prj/build.lst	2010-03-05 20:25:14.000000000 +0100
@@ -1,4 +1,4 @@
-cp    scp2    :    l10n i18npool setup_native l10ntools PYTHON:python ICU:icu REDLAND:redland LIBXSLT:libxslt NULL
+cp    scp2    :    soltools PYTHON:python ICU:icu NULL
 cp    scp2                        usr1     -    all    cp_mkout NULL
 cp    scp2\macros                 nmake    -    all    cp_langmacros NULL
 cp    scp2\source\templates       nmake    -    all    cp_langtemplates NULL
--- scp2/source/ooo/makefile.mk.old	2010-03-05 19:59:20.000000000 +0100
+++ scp2/source/ooo/makefile.mk	2010-03-05 20:25:14.000000000 +0100
@@ -39,7 +39,11 @@ TARGETTYPE=CUI
 # --- Settings -----------------------------------------------------
 
 .INCLUDE :  settings.mk
-.INCLUDE :  i18npool/version.mk
+ISOLANG_TARGET=i18nisolang
+ISOLANG_MAJOR=1
+ISOLANG_MINOR=0
+ISOLANG_MICRO=0
+ISOLANG_CPP=0
 
 .IF "$(COM)" == "MSC"
 SCPDEFS+=-D_MSC
@@ -174,7 +178,17 @@ SCPDEFS+=-DSYSTEM_HUNSPELL
 .IF "$(SYSTEM_REDLAND)" == "YES"
 SCPDEFS+=-DSYSTEM_REDLAND
 .ELSE
-.INCLUDE :  redlandversion.mk
+RAPTOR_MAJOR=1
+RAPTOR_MINOR=1
+RAPTOR_MICRO=0
+# rasqal has no stable abi
+RASQAL_MAJOR=1
+RASQAL_MINOR=0
+RASQAL_MICRO=0
+# redland has no stable abi
+REDLAND_MAJOR=0
+REDLAND_MINOR=0
+REDLAND_MICRO=0
 SCPDEFS+=\
     -DRAPTOR_MAJOR=$(RAPTOR_MAJOR) \
     -DRASQAL_MAJOR=$(RASQAL_MAJOR) \
@@ -209,7 +223,9 @@ SCPDEFS+=-DSYSTEM_LUCENE
 .IF "$(SYSTEM_ICU)" == "YES"
 SCPDEFS+=-DSYSTEM_ICU
 .ELSE
-.INCLUDE :  icuversion.mk
+ICU_MAJOR=4
+ICU_MINOR=0
+ICU_MICRO=1
 SCPDEFS+=\
     -DICU_MAJOR=$(ICU_MAJOR) \
     -DICU_MINOR=$(ICU_MINOR) \
@@ -323,8 +339,8 @@ ULFFILES=                          \
 .INCLUDE :  target.mk
 
 .IF "$(SYSTEM_ICU)" != "YES"
-$(ICUVERSION_DEPENDENT_FILES) : $(SOLARINCDIR)$/icuversion.mk
+#$(ICUVERSION_DEPENDENT_FILES) : $(SOLARINCDIR)$/icuversion.mk
 .ENDIF
 
-$(ISOLANGVERSION_DEPENDENT_FILES) : $(SOLARINCDIR)$/i18npool/version.mk
+# $(ISOLANGVERSION_DEPENDENT_FILES) : $(SOLARINCDIR)$/i18npool/version.mk
 
