--- offapi/util/makefile.mk.orig	2009-10-16 20:01:41.000000000 +0200
+++ offapi/util/makefile.mk	2009-10-19 14:59:28.000000000 +0200
@@ -154,7 +154,7 @@ REGISTRYCHECKFLAG=$(MISC)$/registrycheck
 
 UNOTYPE_STATISTICS=$(MISC)$/unotype_statistics.txt
 
-REGVIEWTOOL=$(SOLARBINDIR)$/regview$(EXECPOST)
+REGVIEWTOOL=$(SOLARPIECE)/$(INPATH)/bin/regview$(EXECPOST)
 
 # --- Targets ------------------------------------------------------
 
@@ -163,15 +163,16 @@ ALLTAR : $(UCR)$/types.db \
        $(REGISTRYCHECKFLAG) \
        $(UNOTYPE_STATISTICS)
 
-$(UCR)$/types.db : $(UCR)$/offapi.db $(SOLARBINDIR)$/udkapi.rdb
+UDKRDB=$(SOLARPIECE)$/$(INPATH)$/bin
+$(UCR)$/types.db : $(UCR)$/offapi.db $(UDKRDB)$/udkapi.rdb
     @-$(RM) $(REGISTRYCHECKFLAG)
     @$(GNUCOPY) -f $(UCR)$/offapi.db $@
-    $(COMMAND_ECHO)$(REGMERGE) $@ / $(SOLARBINDIR)$/udkapi.rdb
+    $(COMMAND_ECHO)$(REGMERGE) $@ / $(UDKRDB)$/udkapi.rdb
 
-$(OUT)$/ucrdoc$/types_doc.db : $(OUT)$/ucrdoc$/offapi_doc.db $(SOLARBINDIR)$/udkapi_doc.rdb
+$(OUT)$/ucrdoc$/types_doc.db : $(OUT)$/ucrdoc$/offapi_doc.db $(UDKRDB)$/udkapi_doc.rdb
     @-$(RM) $(REGISTRYCHECKFLAG)
     @$(GNUCOPY) -f $(OUT)$/ucrdoc$/offapi_doc.db $@
-    $(COMMAND_ECHO)$(REGMERGE) $@ / $(SOLARBINDIR)$/udkapi_doc.rdb
+    $(COMMAND_ECHO)$(REGMERGE) $@ / $(UDKRDB)$/udkapi_doc.rdb
 
 #JSC: The type library has changed, all temporary not checked types are removed
 #     and will be check from now on.
