--- sdext/source/presenter/makefile.mk.old	2010-04-28 11:35:17.000000000 +0200
+++ sdext/source/presenter/makefile.mk	2010-05-11 10:49:19.000000000 +0200
@@ -293,7 +293,7 @@ $(COMPONENT_BITMAPS) : bitmaps$/$$(@:f)
     @-$(MKDIRHIER) $(@:d)
     +$(COPY) $< $@
 
-$(COMPONENT_IMAGES) : $(SOLARSRC)$/$(RSCDEFIMG)$/desktop$/res$/$$(@:f)
+$(COMPONENT_IMAGES) : $(SOLARVER)$/$(RSCDEFIMG)$/desktop$/res$/$$(@:f)
     @@-$(MKDIRHIER) $(@:d)
     $(COPY) $< $@
 
--- sdext/source/pdfimport/makefile.mk.old	2010-04-28 11:35:17.000000000 +0200
+++ sdext/source/pdfimport/makefile.mk	2010-05-11 10:49:19.000000000 +0200
@@ -148,7 +148,7 @@ $(COMPONENT_HELP) : help$/$$(@:f)
     @@-$(MKDIRHIER) $(@:d)
     $(COPY) $< $@
 
-$(COMPONENT_IMAGES) :  $(SOLARSRC)$/$(RSCDEFIMG)$/desktop$/res$/$$(@:f)
+$(COMPONENT_IMAGES) :  $(SOLARVER)$/$(RSCDEFIMG)$/desktop$/res$/$$(@:f)
     @@-$(MKDIRHIER) $(@:d)
     $(COPY) $< $@
 .ENDIF # L10N_framework
--- sdext/source/minimizer/makefile.mk.old	2010-04-28 11:35:17.000000000 +0200
+++ sdext/source/minimizer/makefile.mk	2010-05-11 10:49:19.000000000 +0200
@@ -125,11 +125,11 @@ EXTENSION_PACKDEPS=$(COMPONENT_BITMAPS)
 .IF "$(L10N_framework)"==""
 .INCLUDE : extension_post.mk
 
-$(COMPONENT_BITMAPS) : $(SOLARSRC)$/$(RSCDEFIMG)$/minimizer$/$$(@:f)
+$(COMPONENT_BITMAPS) : $(SOLARVER)$/$(RSCDEFIMG)$/minimizer$/$$(@:f)
     @@-$(MKDIRHIER) $(@:d)
     $(COPY) $< $@
 
-$(COMPONENT_IMAGES) : $(SOLARSRC)$/$(RSCDEFIMG)$/desktop$/res$/$$(@:f)
+$(COMPONENT_IMAGES) : $(SOLARVER)$/$(RSCDEFIMG)$/desktop$/res$/$$(@:f)
     @@-$(MKDIRHIER) $(@:d)
     $(COPY) $< $@
 
