--- sw/source/ui/vba/makefile.mk.old	2009-03-02 18:24:23.000000000 +0100
+++ sw/source/ui/vba/makefile.mk	2009-03-02 18:43:15.000000000 +0100
@@ -80,7 +80,7 @@ ALLTAR : \
 
 $(SLOFILES) : $(MISC)$/$(TARGET).don
 
-$(MISC)$/$(TARGET).don : $(SOLARBINDIR)$/oovbaapi.rdb
-        +$(CPPUMAKER) -O$(INCCOM)$/$(TARGET) -BUCR $(SOLARBINDIR)$/oovbaapi.rdb -X$(SOLARBINDIR)$/types.rdb && echo > $@
+$(MISC)$/$(TARGET).don : $(SPLITINSTALLDIR)$/oovbaapi.rdb
+        +$(CPPUMAKER) -O$(INCCOM)$/$(TARGET) -BUCR $(SPLITINSTALLDIR)$/oovbaapi.rdb -X$(SOLARBINDIR)$/types.rdb && echo > $@
         echo $@
 
--- sw/uiconfig/layout/layout.mk.old1	2009-10-23 21:23:59.000000000 +0200
+++ sw/uiconfig/layout/layout.mk	2009-10-23 21:24:26.000000000 +0200
@@ -1,7 +1,7 @@
 # TODO: move to solenv/inc
 # copies: sw/uiconfig/layout svx/uiconfig/layout
 
-TRALAY=$(AUGMENT_LIBRARY_PATH) $(SOLARBINDIR)/tralay
+TRALAY=$(AUGMENT_LIBRARY_PATH) tralay
 XML_DEST=$(DLLDEST)
 XML_LANGS=$(alllangiso)
 
