--- cpputools/source/registercomponent/makefile.mk.old	2010-02-02 12:49:41.000000000 +0100
+++ cpputools/source/registercomponent/makefile.mk	2010-03-05 20:11:19.000000000 +0100
@@ -39,8 +39,8 @@ ENABLE_EXCEPTIONS=TRUE
 # --- Settings -----------------------------------------------------
 .INCLUDE :  settings.mk
 
-UNOUCRDEP=$(SOLARBINDIR)$/udkapi.rdb 
-UNOUCRRDB=$(SOLARBINDIR)$/udkapi.rdb
+UNOUCRDEP=$(SOLARPIECE)$/$(INPATH)$/bin$/udkapi.rdb 
+UNOUCRRDB=$(UNOUCRDEP)
 
 NO_OFFUH=TRUE
 CPPUMAKERFLAGS += -C
--- cpputools/source/regsingleton/makefile.mk.old	2010-02-02 12:49:41.000000000 +0100
+++ cpputools/source/regsingleton/makefile.mk	2010-03-05 20:11:19.000000000 +0100
@@ -39,8 +39,8 @@ ENABLE_EXCEPTIONS=TRUE
 
 .INCLUDE :  settings.mk
 
-UNOUCRDEP=$(SOLARBINDIR)$/udkapi.rdb 
-UNOUCRRDB=$(SOLARBINDIR)$/udkapi.rdb
+UNOUCRDEP=$(SOLARPIECE)$/$(INPATH)$/bin$/udkapi.rdb 
+UNOUCRRDB=$(UNOUCRDEP)
 
 NO_OFFUH=TRUE
 CPPUMAKERFLAGS+= -C
--- cpputools/source/unoexe/makefile.mk.old	2010-02-02 12:49:41.000000000 +0100
+++ cpputools/source/unoexe/makefile.mk	2010-03-05 20:11:19.000000000 +0100
@@ -39,8 +39,8 @@ ENABLE_EXCEPTIONS=TRUE
 
 .INCLUDE :  settings.mk
 
-UNOUCRDEP=$(SOLARBINDIR)$/udkapi.rdb 
-UNOUCRRDB=$(SOLARBINDIR)$/udkapi.rdb
+UNOUCRDEP=$(SOLARPIECE)$/$(INPATH)$/bin$/udkapi.rdb 
+UNOUCRRDB=$(UNOUCRDEP)
 
 NO_OFFUH=TRUE
 CPPUMAKERFLAGS+= -C
