--- rsc/source/prj/start.cxx.old	2009-10-19 19:11:30.000000000 +0200
+++ rsc/source/prj/start.cxx	2009-10-19 20:15:55.000000000 +0200
@@ -338,10 +338,9 @@ int cdecl main ( int argc, char ** argv)
     BOOL			bHelp	 = FALSE;
     BOOL			bError	 = FALSE;
     BOOL			bResponse = FALSE;
-    ByteString		aSolarbin(getenv("SOLARBINDIR"));
     ByteString		aDelim("/");
-    ByteString		aPrePro; //( aSolarbin + aDelim + ByteString("rscpp"));
-    ByteString		aRsc2Name; //(  aSolarbin + aDelim + ByteString("rsc2"));
+    ByteString		aPrePro = ByteString("rscpp");
+    ByteString		aRsc2Name = ByteString("rsc2");
     ByteString		aSrsName;
     ByteString		aResName;
     RscStrList		aInputList;
@@ -352,14 +351,6 @@ int cdecl main ( int argc, char ** argv)
     sal_uInt32		i;
     ByteString* 	pString;
 
-    aPrePro = aSolarbin;
-    aPrePro += aDelim;
-    aPrePro += ByteString("rscpp");
-
-    aRsc2Name = aSolarbin;
-    aRsc2Name += aDelim;
-    aRsc2Name += ByteString("rsc2");
-
     pStr = ::ResponseFile( &aCmdLine, argv, argc );
     if( pStr )
     {
