--- solenv/inc/extension_post.mk.old	2010-07-19 11:41:49.000000000 +0200
+++ solenv/inc/extension_post.mk	2010-07-19 11:43:56.000000000 +0200
@@ -30,6 +30,8 @@ EXTENSION_TARGET:=$(ZIP9TARGETN)
 .SOURCE.xcu : $(MISC)/$(EXTNAME)/merge $(MISC)/$(EXTNAME)/registry/data $(MISC)/$(COMPONENT_SHARED_CONFIG)_in/merge $(MISC)/$(COMPONENT_SHARED_CONFIG)_in/registry/data $(COMPONENT_CONFIGDIR) .
 .SOURCE.xcs : $(MISC)/$(EXTNAME)/registry $(MISC)/$(EXTNAME)/registry/schema .
 
+SOLARLICENSEDIR*=$(SOLARBINDIR)/osl
+
 fixme=$(MISC)/$(EXTNAME)/merge/$(MISC)
 fixme2=$(fixme:n)
 
@@ -145,9 +147,9 @@ $(DESCRIPTION) $(PHONYDESC) : $(DESCRIPT
 # may not fit...
 .IF "$(CUSTOM_LICENSE)"==""
 .IF "$(GUI)" == "WNT"
-PACKLICDEPS=$(SOLARBINDIR)/osl/license$$(@:b:s/_/./:e:s/./_/)$$(@:e)
+PACKLICDEPS=$(SOLARLICENSEDIR)/license$$(@:b:s/_/./:e:s/./_/)$$(@:e)
 .ELSE			# "$(GUI)" == "WNT"
-PACKLICDEPS=$(SOLARBINDIR)/osl/LICENSE$$(@:b:s/_/./:e:s/./_/)$$(@:e)
+PACKLICDEPS=$(SOLARLICENSEDIR)/LICENSE$$(@:b:s/_/./:e:s/./_/)$$(@:e)
 .ENDIF			# "$(GUI)" == "WNT"
 .ELSE			# "$(CUSTOM_LICENSE)" == ""
 PACKLICDEPS=$(CUSTOM_LICENSE)
--- sfx2/workben/custompanel/makefile.mk.old	2010-06-18 10:32:12.000000000 +0200
+++ sfx2/workben/custompanel/makefile.mk	2010-07-19 11:52:10.000000000 +0200
@@ -33,6 +33,8 @@ ENABLE_EXCEPTIONS=TRUE
 LIBTARGET=NO
 EXTENSIONNAME:=custom-tool-panel
 
+SOLARLICENSEDIR=$(SOLARPIECE)$/$(INPATH)$/bin$/osl
+
 # --- Settings -----------------------------------------------------
 
 .INCLUDE :  settings.mk
@@ -67,7 +69,7 @@ SHL1STDLIBS= \
         $(SALHELPERLIB)     \
         $(CPPUHELPERLIB)
 
-SHL1VERSIONMAP=$(SOLARSRC)/solenv/src/component.map
+SHL1VERSIONMAP=$(SOLARENV)/src/component.map
 SHL1LIBS=	$(LIB1TARGET)
 SHL1DEF=	$(MISC)/$(SHL1TARGET).def
 SHL1RPATH=  OXT
