--- scripting/source/dlgprov/makefile.mk.old	2010-03-23 14:25:13.000000000 +0100
+++ scripting/source/dlgprov/makefile.mk	2010-03-23 15:10:03.000000000 +0100
@@ -77,6 +77,7 @@ ALLTAR : \
 
 $(SLOFILES) : $(MISC)$/$(TARGET).don
 
-$(MISC)$/$(TARGET).don : $(SOLARBINDIR)$/oovbaapi.rdb
-        +$(CPPUMAKER) -O$(INCCOM)$/$(TARGET) -BUCR $(SOLARBINDIR)$/oovbaapi.rdb -X$(SOLARBINDIR)$/types.rdb && echo > $@
+OOVBADIR=$(SOLARPIECE)$/$(INPATH)$/bin
+$(MISC)$/$(TARGET).don : $(OOVBADIR)$/oovbaapi.rdb
+        +$(CPPUMAKER) -O$(INCCOM)$/$(TARGET) -BUCR $(OOVBADIR)$/oovbaapi.rdb -X$(SOLARBINDIR)$/types.rdb && echo > $@
         echo $@
--- scripting/source/vbaevents/makefile.mk.old	2010-03-23 13:43:40.000000000 +0100
+++ scripting/source/vbaevents/makefile.mk	2010-03-23 14:06:33.000000000 +0100
@@ -57,6 +57,7 @@ ALLTAR : \
 
 $(SLOFILES) : $(MISC)$/$(TARGET).don
 
-$(MISC)$/$(TARGET).don : $(SOLARBINDIR)$/oovbaapi.rdb
-        +$(CPPUMAKER) -O$(INCCOM)$/$(TARGET) -BUCR $(SOLARBINDIR)$/oovbaapi.rdb -X$(SOLARBINDIR)$/types.rdb && echo > $@
+OOVBADIR=$(SOLARPIECE)$/$(INPATH)$/bin
+$(MISC)$/$(TARGET).don : $(OOVBADIR)$/oovbaapi.rdb
+        +$(CPPUMAKER) -O$(INCCOM)$/$(TARGET) -BUCR $(OOVBADIR)$/oovbaapi.rdb -X$(SOLARBINDIR)$/types.rdb && echo > $@
         echo $@
