--- solenv/bin/localize.old	2004-08-30 19:29:46.000000000 +0200
+++ solenv/bin/localize	2008-11-14 17:51:04.000000000 +0100
@@ -13,8 +13,13 @@ if [ x${SOLARENV}x = xx ]; then
 fi
 
 if [ x${SOLARVER}x = xx -o x${UPDMINOREXT}x = xx  ]; then
-    exec perl -w $SOLARVERSION/$INPATH/bin/localize.pl "$@"
+    bindir=bin
 else
-    exec perl -w $SOLARVERSION/$INPATH/bin$UPDMINOREXT/localize.pl "$@"
+    bindir=bin$UPDMINOREXT
 fi
 
+if test -f $SOLARVERSION/$INPATH/$bindir/localize.pl ; then
+    exec perl -w $SOLARVERSION/$INPATH/$bindir/localize.pl "$@"
+else
+    exec perl -w $SOLARPIECE/$INPATH/$bindir/localize.pl "$@"
+fi
--- l10ntools/scripts/localize.pl.old1	2010-02-17 18:05:32.000000000 +0100
+++ l10ntools/scripts/localize.pl	2010-02-17 18:06:53.000000000 +0100
@@ -403,7 +403,7 @@ sub merge_gsicheck{
     close ( $TMPHANDLE );
     
     $command = "$ENV{WRAPCMD} " if( $ENV{WRAPCMD} );
-    $command .= "$ENV{SOLARVER}/$ENV{INPATH}/bin/gsicheck";
+    $command .= "gsicheck";
 
     my $errfile = $sdffile.".err";
     $command .= " -k -c -wcf $tmpfile -wef ".fix_cygwin_path($errfile)." -l \"\" ".fix_cygwin_path($sdffile);
