--- solenv/bin/deliver.pl.old	2010-05-19 19:57:28.000000000 +0200
+++ solenv/bin/deliver.pl	2010-06-22 16:09:23.000000000 +0200
@@ -443,6 +443,13 @@ sub parse_options 
     $opt_force = 1 if $opt_delete;
 }
 
+# return the base directory to install to
+sub get_dest_base()
+{
+    return $ENV{'SOLARPIECEVERSION'} if (defined $ENV{'SOLARPIECEVERSION'});
+    return $ENV{'SOLARVERSION'};
+}
+
 sub init_globals 
 {
     my $ext;
@@ -461,11 +468,11 @@ sub init_globals 
     my $build_sosl    = $ENV{'BUILD_SOSL'};
     my $common_outdir = $ENV{'COMMON_OUTDIR'};
     my $inpath        = $ENV{'INPATH'};
-    my $solarversion  = $ENV{'SOLARVERSION'};
     my $updater       = $ENV{'UPDATER'};
     my $updminor      = $ENV{'UPDMINOR'};
     my $updminorext   = $ENV{'UPDMINOREXT'};
     my $work_stamp    = $ENV{'WORK_STAMP'};
+    my $solarversion  = get_dest_base();
     
     # special security check for release engineers
     if ( defined($updater) && !defined($build_sosl) && !$opt_force) {
@@ -1165,8 +1172,8 @@ sub push_on_loglist
         warn "Neither common nor platform tree?";
         return;
     }
-    # make 'to' relative to SOLARVERSION
-    my $solarversion  = $ENV{'SOLARVERSION'};
+    # make 'to' relative to the base path
+    my $solarversion  = get_dest_base();
     $solarversion =~ s#\\#/#g;
     $entry[2] =~ s/^\Q$solarversion\E\///;
     # strip minor from 'to'
