--- reportbuilder/util/makefile.mk.old	2010-04-28 11:35:17.000000000 +0200
+++ reportbuilder/util/makefile.mk	2010-05-11 10:43:42.000000000 +0200
@@ -129,11 +129,15 @@ $(EXTENSIONDIR)$/%.jar : $(SOLARBINDIR)$
     @@-$(MKDIRHIER) $(@:d)
     $(COPY) $< $@
 
+$(EXTENSIONDIR)$/%.jar : $(SOLARPIECE)$/$(INPATH)$/bin$/%.jar
+    @@-$(MKDIRHIER) $(@:d)
+    $(COPY) $< $@
+
 $(EXTENSIONDIR)$/readme_en-US.% : $(PRJ)$/license$/readme_en-US.%
     @@-$(MKDIRHIER) $(@:d)
     $(COPY) $< $@
 
-$(COMPONENT_IMAGES) : $(SOLARSRC)$/$(RSCDEFIMG)$/desktop$/res$/$$(@:f)
+$(COMPONENT_IMAGES) : $(SOLARVER)$/$(RSCDEFIMG)$/desktop$/res$/$$(@:f)
     @@-$(MKDIRHIER) $(@:d)
     $(COPY) $< $@
 
