--- saxon/makefile.mk.old	2010-02-02 12:49:41.000000000 +0100
+++ saxon/makefile.mk	2010-03-05 20:24:05.000000000 +0100
@@ -54,7 +54,7 @@ TARFILE_ROOTDIR=src
 
 PATCH_FILES=saxon-9.0.0.7-bj.patch
 
-BUILD_ACTION=$(ANT) $(ANT_FLAGS) -Dsolarbindir=$(SOLARBINDIR) jar-bj 
+BUILD_ACTION=$(ANT) $(ANT_FLAGS) -Dsolar.bin=$(SOLARPIECE)$/$(INPATH)$/bin jar-bj
 
 OUT2CLASS= saxon-build$/9.0.0.7$/bj$/saxon9.jar
 
