--- sdext/source/minimizer/makefile.mk.old	2010-02-02 12:49:41.000000000 +0100
+++ sdext/source/minimizer/makefile.mk	2010-03-05 20:10:38.000000000 +0100
@@ -129,7 +129,7 @@ EXTENSION_PACKDEPS=$(COMPONENT_BITMAPS)
 .IF "$(L10N_framework)"==""
 .INCLUDE : extension_post.mk
 
-$(COMPONENT_BITMAPS) : $(SOLARSRC)$/$(RSCDEFIMG)$/minimizer$/$$(@:f)
+$(COMPONENT_BITMAPS) : $(SOLARVER)$/$(RSCDEFIMG)$/minimizer$/$$(@:f)
     @@-$(MKDIRHIER) $(@:d)
     $(COPY) $< $@
 
--- sdext/util/makefile.pmk.old	2010-02-02 12:49:41.000000000 +0100
+++ sdext/util/makefile.pmk	2010-03-05 20:10:38.000000000 +0100
@@ -31,7 +31,7 @@
 
 XSLDIR=$(SOLARXMLDIR)$/processing
 XCSROOT=$(SOLARXMLDIR)
-ABSXCSROOT=$(SOLARXMLDIR)
+ABSXCSROOT=$(SPLITINSTALLDIR)$/..$/share
 DTDDIR=$(SOLARXMLDIR)
 PROCESSOUT=$(MISC)$/$(EXTNAME)
 PROCESSORDIR=$(SOLARBINDIR)
