--- scripting/source/vbaevents/eventhelper.cxx.old	2010-07-19 15:16:09.000000000 +0200
+++ scripting/source/vbaevents/eventhelper.cxx	2010-07-19 16:11:49.000000000 +0200
@@ -78,7 +78,7 @@
 #include <basic/sbmeth.hxx>
 #include <basic/sbmod.hxx>
 #include <basic/sbx.hxx>
-#include <filter/msfilter/msvbahelper.hxx>
+//#include <filter/msfilter/msvbahelper.hxx>
 
 
 
@@ -1097,7 +1097,7 @@ EventListener::firing_Impl(const ScriptE
             
             OSL_TRACE("*** trying to invoke %s ",
                 rtl::OUStringToOString( sToResolve, RTL_TEXTENCODING_UTF8 ).getStr() );
-            ooo::vba::VBAMacroResolvedInfo aMacroResolvedInfo = ooo::vba::resolveVBAMacro( mpShell, sToResolve ); 
+/*            ooo::vba::VBAMacroResolvedInfo aMacroResolvedInfo = ooo::vba::resolveVBAMacro( mpShell, sToResolve ); 
             if ( aMacroResolvedInfo.IsResolved() )
             {
                 //liuchen 2009-6-8
@@ -1139,6 +1139,7 @@ EventListener::firing_Impl(const ScriptE
                     }
                } 
            }
+*/
        }
     }
 }
--- scripting/source/vbaevents/makefile.mk.old	2010-07-19 15:16:20.000000000 +0200
+++ scripting/source/vbaevents/makefile.mk	2010-07-19 16:12:53.000000000 +0200
@@ -69,7 +69,6 @@ SHL1STDLIBS= \
         $(BASICLIB) \
         $(COMPHELPERLIB) \
         $(SFXLIB) \
-		$(MSFILTERLIB) \
                 $(CPPULIB) \
                 $(TOOLSLIB) \
                 $(SALLIB)
--- scripting/source/dlgprov/dlgevtatt.cxx.old	2010-07-19 17:22:49.000000000 +0200
+++ scripting/source/dlgprov/dlgevtatt.cxx	2010-07-19 17:24:06.000000000 +0200
@@ -53,7 +53,7 @@
 
 #include <ooo/vba/XVBAToOOEventDescGen.hpp>
 #include <com/sun/star/lang/XUnoTunnel.hpp>
-#include <vbahelper/vbaaccesshelper.hxx>
+//#include <vbahelper/vbaaccesshelper.hxx>
 
 using namespace ::com::sun::star;
 using namespace ::com::sun::star::awt;
@@ -186,9 +186,9 @@ namespace dlgprov
             uno::Reference< lang::XUnoTunnel >  xObjShellTunnel( rxModel, uno::UNO_QUERY );
             if ( xObjShellTunnel.is() )
             {
-                pFoundShell = reinterpret_cast<SfxObjectShell*>( xObjShellTunnel->getSomething(SfxObjectShell::getUnoTunnelId()));
-                if ( pFoundShell )
-                    mbUseFakeVBAEvents = ooo::vba::isAlienExcelDoc( *pFoundShell ) || ooo::vba::isAlienWordDoc( *pFoundShell ) ;
+//                pFoundShell = reinterpret_cast<SfxObjectShell*>( xObjShellTunnel->getSomething(SfxObjectShell::getUnoTunnelId()));
+//                if ( pFoundShell )
+//                    mbUseFakeVBAEvents = ooo::vba::isAlienExcelDoc( *pFoundShell ) || ooo::vba::isAlienWordDoc( *pFoundShell ) ;
             }
         }
         if ( mbUseFakeVBAEvents )
