--- remotebridges/source/bridge/makefile.mk.old	2010-02-02 12:49:41.000000000 +0100
+++ remotebridges/source/bridge/makefile.mk	2010-03-05 20:24:42.000000000 +0100
@@ -35,6 +35,7 @@ PRJNAME=remotebridges
 TARGET = remotebridge.uno
 ENABLE_EXCEPTIONS=TRUE
 COMP1TYPELIST = remotebridge
+UNOUCRRDB=$(SOLARPIECE)$/$(INPATH)$/bin$/types.rdb 
 
 # --- Settings -----------------------------------------------------
 .INCLUDE :  settings.mk
--- remotebridges/source/dynamicloader/makefile.mk.old	2010-02-02 12:49:41.000000000 +0100
+++ remotebridges/source/dynamicloader/makefile.mk	2010-03-05 20:24:42.000000000 +0100
@@ -35,6 +35,7 @@ PRJNAME=remotebridges
 TARGET = dynamicloader.uno
 ENABLE_EXCEPTIONS=TRUE
 COMP1TYPELIST = dynamicloader
+UNOUCRRDB=$(SOLARPIECE)$/$(INPATH)$/bin$/types.rdb 
 
 # --- Settings -----------------------------------------------------
 
--- remotebridges/source/factory/makefile.mk.old	2010-02-02 12:49:41.000000000 +0100
+++ remotebridges/source/factory/makefile.mk	2010-03-05 20:24:42.000000000 +0100
@@ -35,6 +35,7 @@ PRJNAME=remotebridges
 TARGET = bridgefac.uno
 ENABLE_EXCEPTIONS=TRUE
 COMP1TYPELIST = brdgfctr
+UNOUCRRDB=$(SOLARPIECE)$/$(INPATH)$/bin$/types.rdb 
 
 # --- Settings -----------------------------------------------------
 .INCLUDE :  settings.mk
--- remotebridges/source/unourl_resolver/makefile.mk.old	2010-02-02 12:49:41.000000000 +0100
+++ remotebridges/source/unourl_resolver/makefile.mk	2010-03-05 20:24:42.000000000 +0100
@@ -34,6 +34,7 @@ PRJNAME=remotebridges
 TARGET = uuresolver.uno
 ENABLE_EXCEPTIONS=TRUE
 COMP1TYPELIST = uuresolver
+UNOUCRRDB=$(SOLARPIECE)$/$(INPATH)$/bin$/types.rdb 
 
 # --- Settings -----------------------------------------------------
 
