--- connectivity/source/drivers/hsqldb/makefile.mk.old	2008-04-10 11:21:52.000000000 +0200
+++ connectivity/source/drivers/hsqldb/makefile.mk	2008-10-08 15:52:20.000000000 +0200
@@ -59,7 +59,7 @@ RES1FILELIST=\
 	$(SRS)$/$(TARGET).srs
 
 RESLIB1NAME=$(TARGET)
-RESLIB1IMAGES=$(SOLARDEFIMG)$/database
+RESLIB1IMAGES=$(IMGSRC)$/$(RSCDEFIMG)$/database
 RESLIB1SRSFILES=$(RES1FILELIST)
 
 # Note that the resource file built here is currently *not* included
--- connectivity/makefile.pmk.old	2009-10-21 14:39:24.000000000 +0200
+++ connectivity/makefile.pmk	2009-10-21 14:42:46.000000000 +0200
@@ -29,13 +29,13 @@
 #
 #*************************************************************************
 
-XSLDIR=$(SOLARXMLDIR)$/processing
+XSLDIR=$(SOLARPIECE)$/$(INPATH)$/xml$(EXT_UPDMINOR)$/processing
 ABSXCSROOT=$(PWD)$/$(MISC)
 # ABSXCSROOT=$(MISC)
 XCSROOT=..
 # XCSROOT=$(SOLARXMLDIR)
 # ABSXCSROOT=$(SOLARXMLDIR)
-DTDDIR=$(SOLARXMLDIR)
+DTDDIR=$(SOLARPIECE)$/$(INPATH)$/xml$(EXT_UPDMINOR)
 PROCESSOUT=$(MISC)
 PROCESSORDIR=$(SOLARBINDIR)
 
--- connectivity/target.pmk.old	2009-10-08 08:11:52.000000000 +0200
+++ connectivity/target.pmk	2009-10-21 14:41:41.000000000 +0200
@@ -34,7 +34,7 @@
 COMPONENT_CONFIG_SCHEMA*=$(TARGET).xcs
 
 ALLTAR: "$(PWD)$/$(MISC)$/registry$/schema$/$(PACKAGEDIR)$/$(COMPONENT_CONFIG_SCHEMA)"
-"$(PWD)$/$(MISC)$/registry$/schema$/$(PACKAGEDIR)$/$(COMPONENT_CONFIG_SCHEMA)" : $(SOLARXMLDIR)$/registry$/schema$/$(PACKAGEDIR)$/Drivers.xcs
+"$(PWD)$/$(MISC)$/registry$/schema$/$(PACKAGEDIR)$/$(COMPONENT_CONFIG_SCHEMA)" : $(SOLARPIECE)$/$(INPATH)$/xml$(EXT_UPDMINOR)$/registry$/schema$/$(PACKAGEDIR)$/Drivers.xcs
     @@-$(MKDIRHIER) $(@:d)
     $(COPY) $< $@
 
