--- mysqlc/source/makefile.mk.old	2010-04-15 18:37:58.000000000 +0200
+++ mysqlc/source/makefile.mk	2010-04-15 18:56:14.000000000 +0200
@@ -143,14 +143,14 @@ CDEFS += -DCPPCONN_LIB=\"$(shell readlin
     .ELSE
       MYSQL_LIB+=-rdynamic -lz -lcrypt -lnsl -lm
     .ENDIF
-    MYSQL_CPPCONNFILE=$(SOLARLIBDIR)$/$(DLLPRE)mysqlcppconn$(DLLPOST)
+    MYSQL_CPPCONNFILE=$(SOLARPIECE)$/$(INPATH)$/lib$/$(DLLPRE)mysqlcppconn$(DLLPOST)
   .ELSE
     MYSQL_CPPCONN_LIB+=-lmysqlcppconn
   .ENDIF
 .ENDIF
 
 .IF "$(SYSTEM_MYSQL_CPPCONN)" != "YES"
-  MYSQL_INC+=-I$(SOLARINCDIR)$/mysqlcppconn -I$(SOLARINCDIR)$/mysqlcppconn/cppconn
+  MYSQL_INC+=-I$(SOLARPIECE)$/$(INPATH)$/inc$/mysqlcppconn -I$(SOLARPIECE)$/$(INPATH)$/inc$/mysqlcppconn/cppconn
 .ENDIF
 
 CFLAGS+=-I..$/..$/inc $(MYSQL_INC) \
