--- sc/source/ui/vba/makefile.mk.old	2010-03-12 13:56:03.000000000 +0100
+++ sc/source/ui/vba/makefile.mk	2010-03-12 14:08:23.000000000 +0100
@@ -128,8 +128,8 @@ ALLTAR : \
 
 $(SLOFILES) : $(MISC)$/$(TARGET).don
 
-$(MISC)$/$(TARGET).don : $(SOLARBINDIR)$/oovbaapi.rdb
-        +$(CPPUMAKER) -O$(INCCOM)$/$(TARGET) -BUCR $(SOLARBINDIR)$/oovbaapi.rdb -X$(SOLARBINDIR)$/types.rdb && echo > $@
+$(MISC)$/$(TARGET).don : $(SPLITINSTALLDIR)$/oovbaapi.rdb
+        +$(CPPUMAKER) -O$(INCCOM)$/$(TARGET) -BUCR $(SPLITINSTALLDIR)$/oovbaapi.rdb -X$(SOLARBINDIR)$/types.rdb && echo > $@
         echo $@
 
 .ENDIF
