--- l10n/source/makefile.mk.old	2009-12-01 17:12:34.000000000 +0100
+++ l10n/source/makefile.mk	2009-12-01 17:12:01.000000000 +0100
@@ -107,6 +107,6 @@ $(COMMONMISC)$/merge.done : $(all_sdfs)
     -rm -rf $(COMMONMISC)$/sdf$(INPATH)_begone
     -$(MKDIRHIER) $(COMMONMISC)$/sdf
 .ENDIF			# "$(L10n_LOCK)" != "YES"
-    $(PERL) $(SOLARVER)$/$(INPATH)$/bin$(UPDMINOREXT)$/fast_merge.pl -sdf_files $(mktmp $<) -merge_dir $(COMMONMISC)$/sdf && echo $< >$@
+    $(PERL) $(SOLARPIECE)$/$(INPATH)$/bin$(UPDMINOREXT)$/fast_merge.pl -sdf_files $(mktmp $<) -merge_dir $(COMMONMISC)$/sdf && echo $< >$@
 
 .ENDIF
