--- offapi/util/makefile.mk.old	2009-10-14 15:37:57.000000000 +0200
+++ offapi/util/makefile.mk	2009-10-14 16:42:17.000000000 +0200
@@ -163,15 +163,16 @@ ALLTAR : $(UCR)$/types.db \
        $(REGISTRYCHECKFLAG) \
        $(UNOTYPE_STATISTICS)
 
-$(UCR)$/types.db : $(UCR)$/offapi.db $(SOLARBINDIR)$/udkapi.rdb
+UDKRDB=$(SOLARPIECE)$/$(INPATH)$/bin
+$(UCR)$/types.db : $(UCR)$/offapi.db $(UDKRDB)$/udkapi.rdb
     @-$(RM) $(REGISTRYCHECKFLAG)
     @$(GNUCOPY) -f $(UCR)$/offapi.db $@
-    $(COMMAND_ECHO)$(REGMERGE) $@ / $(SOLARBINDIR)$/udkapi.rdb
+    $(COMMAND_ECHO)$(REGMERGE) $@ / $(UDKRDB)$/udkapi.rdb
 
-$(OUT)$/ucrdoc$/types_doc.db : $(OUT)$/ucrdoc$/offapi_doc.db $(SOLARBINDIR)$/udkapi_doc.rdb
+$(OUT)$/ucrdoc$/types_doc.db : $(OUT)$/ucrdoc$/offapi_doc.db $(UDKRDB)$/udkapi_doc.rdb
     @-$(RM) $(REGISTRYCHECKFLAG)
     @$(GNUCOPY) -f $(OUT)$/ucrdoc$/offapi_doc.db $@
-    $(COMMAND_ECHO)$(REGMERGE) $@ / $(SOLARBINDIR)$/udkapi_doc.rdb
+    $(COMMAND_ECHO)$(REGMERGE) $@ / $(UDKRDB)$/udkapi_doc.rdb
 
 #JSC: The type library has changed, all temporary not checked types are removed
 #     and will be check from now on.
