--- basic/source/classes/makefile.mk.old	2010-03-05 19:59:21.000000000 +0100
+++ basic/source/classes/makefile.mk	2010-03-05 20:08:02.000000000 +0100
@@ -45,8 +45,9 @@ ALLTAR .SEQUENTIAL : \
         $(MISC)$/$(TARGET).don \
         $(MISC)$/$(TARGET).slo
 
-$(MISC)$/$(TARGET).don : $(SOLARBINDIR)$/oovbaapi.rdb
-        +$(CPPUMAKER) -O$(OUT)$/inc -BUCR $(SOLARBINDIR)$/oovbaapi.rdb -X$(SOLARBINDIR)$/types.rdb && echo > $@
+OOVBADIR=$(SOLARPIECE)$/$(INPATH)$/bin
+$(MISC)$/$(TARGET).don : $(OOVBADIR)$/oovbaapi.rdb
+        +$(CPPUMAKER) -O$(OUT)$/inc -BUCR $(OOVBADIR)$/oovbaapi.rdb -X$(SOLARBINDIR)$/types.rdb && echo > $@
         echo $@
 
 $(MISC)$/$(TARGET).slo : $(SLOTARGET)
