--- oox/source/export/drawingml.cxx.old	2010-03-03 17:00:04.000000000 +0100
+++ oox/source/export/drawingml.cxx	2010-03-03 17:00:05.000000000 +0100
@@ -1331,14 +1331,16 @@ void DrawingML::WritePolyPolygon( const
 
 void DrawingML::WriteConnectorConnections( EscherConnectorListEntry& rConnectorEntry, sal_Int32 nStartID, sal_Int32 nEndID )
 {
-    mpFS->singleElementNS( XML_a, XML_stCxn,
-                           XML_id, I32S( nStartID ),
-                           XML_idx, I64S( rConnectorEntry.GetConnectorRule( TRUE ) ),
-                           FSEND );
-    mpFS->singleElementNS( XML_a, XML_endCxn,
-                           XML_id, I32S( nEndID ),
-                           XML_idx, I64S( rConnectorEntry.GetConnectorRule( FALSE ) ),
-                           FSEND );
+    if( nStartID != -1 )
+        mpFS->singleElementNS( XML_a, XML_stCxn,
+                               XML_id, I32S( nStartID ),
+                               XML_idx, I64S( rConnectorEntry.GetConnectorRule( TRUE ) ),
+                               FSEND );
+    if( nEndID != -1 )
+        mpFS->singleElementNS( XML_a, XML_endCxn,
+                               XML_id, I32S( nEndID ),
+                               XML_idx, I64S( rConnectorEntry.GetConnectorRule( FALSE ) ),
+                               FSEND );
 }
 
 // from sw/source/filter/ww8/wrtw8num.cxx for default bullets to export to MS intact
--- oox/source/export/shapes.cxx.old	2010-03-03 17:00:05.000000000 +0100
+++ oox/source/export/shapes.cxx	2010-03-03 17:00:05.000000000 +0100
@@ -984,6 +984,9 @@ sal_Int32 ShapeExport::GetNewShapeID( co
 
 sal_Int32 ShapeExport::GetNewShapeID( const Reference< XShape > rXShape, XmlFilterBase* pFB )
 {
+    if( !rXShape.is() )
+        return -1;
+
     sal_Int32 nID = pFB->GetUniqueId();
 
     saShapeMap[ rXShape ] = nID;
@@ -993,6 +996,9 @@ sal_Int32 ShapeExport::GetNewShapeID( co
 
 sal_Int32 ShapeExport::GetShapeID( const Reference< XShape > rXShape )
 {
+    if( !rXShape.is() )
+        return -1;
+
     ShapeHashMap::const_iterator aIter = saShapeMap.find( rXShape );
 
     if( aIter == saShapeMap.end() )
