From dcae40e328a7019a055cfbf5d31a6a1bef79232b Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:01:08 +0200
Subject: [PATCH 378/768] qt-moc-detect.diff

---
 configure.in |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git configure.in configure.in
index 3dce2a7..82b8989 100644
--- configure.in
+++ configure.in
@@ -6464,7 +6464,7 @@ your Qt installation by exporting QTDIR before running "configure".])
     fi
 
     dnl Check for Meta Object Compiler
-    AC_PATH_PROG( MOC, moc, no, [$QTDIR/bin:$PATH] )
+    AC_PATH_PROG( MOC, moc, no, [$QTDIR/bin:$PATH])
     if test "$MOC" = "no" ; then
         AC_MSG_ERROR([Qt Meta Object Compiler not found.  Please specify
 the root of your Qt installation by exporting QTDIR before running "configure".])
@@ -6571,7 +6571,7 @@ if test "$test_kde4" = "yes" -a "$ENABLE_KDE4" = "TRUE" ; then
    fi
 
    dnl Check for Meta Object Compiler
-   AC_PATH_PROG( MOC4, moc, no, [$PATH:$QT4DIR/bin] )
+   AC_PATH_PROG( MOC4, moc, no, [$QT4DIR/bin:$PATH])
    if test "$MOC4" = "no" ; then
       AC_MSG_ERROR([Qt Meta Object Compiler not found.  Please specify
 the root of your Qt installation by exporting QT4DIR before running "configure".])
-- 
1.7.0.1

