From 8cbfdaa5f963d4e229ee432c37169c7a5665872f Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:01:08 +0200
Subject: [PATCH 444/878] qt-moc-detect.diff

---
 configure.in |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.in b/configure.in
index 1fdf875..02d34da 100644
--- a/configure.in
+++ b/configure.in
@@ -6448,7 +6448,7 @@ your Qt installation by exporting QTDIR before running "configure".])
     fi
 
     dnl Check for Meta Object Compiler
-    AC_PATH_PROG( MOC, moc, no, [$QTDIR/bin:$PATH] )
+    AC_PATH_PROG( MOC, moc, no, [$QTDIR/bin:$PATH])
     if test "$MOC" = "no" ; then
         AC_MSG_ERROR([Qt Meta Object Compiler not found.  Please specify
 the root of your Qt installation by exporting QTDIR before running "configure".])
@@ -6555,7 +6555,7 @@ if test "$test_kde4" = "yes" -a "$ENABLE_KDE4" = "TRUE" ; then
    fi
 
    dnl Check for Meta Object Compiler
-   AC_PATH_PROG( MOC4, moc, no, [$PATH:$QT4DIR/bin] )
+   AC_PATH_PROG( MOC4, moc, no, [$QT4DIR/bin:$PATH])
    if test "$MOC4" = "no" ; then
       AC_MSG_ERROR([Qt Meta Object Compiler not found.  Please specify
 the root of your Qt installation by exporting QT4DIR before running "configure".])
-- 
1.7.0.1

