--- reportbuilder/registry/schema/org/openoffice/Office/ReportDesign.xcs.old	2010-07-22 13:08:24.000000000 +0200
+++ reportbuilder/registry/schema/org/openoffice/Office/ReportDesign.xcs	2010-07-29 14:58:55.000000000 +0200
@@ -52,7 +52,7 @@
                     <author>LLA</author>
                     <desc>Direct download URL to the Oracle(tm) Report Builder extension.</desc>
                 </info>
-                <value>http://extensions.services.openoffice.org</value>
+                <value>http://extensions.go-oo.org</value>
             </prop>
             <prop oor:name="Name" oor:type="xs:string">
                 <info>
