--- registry/source/keyimpl.cxx~	2009-11-23 17:07:31.000000000 +0100
+++ registry/source/keyimpl.cxx	2009-11-23 21:47:24.718463289 +0100
@@ -365,7 +365,6 @@
         return REG_SET_VALUE_FAILED;
     }
 
-    rValue.flush();
     rtl_freeMemory(pBuffer);
     return REG_NO_ERROR;
 }
@@ -424,7 +423,6 @@
         return REG_SET_VALUE_FAILED;
     }
 
-    rValue.flush();
     rtl_freeMemory(pBuffer);
     return REG_NO_ERROR;
 }
@@ -492,7 +490,6 @@
         return REG_SET_VALUE_FAILED;
     }
 
-    rValue.flush();
     rtl_freeMemory(pBuffer);
     return REG_NO_ERROR;
 }
@@ -560,7 +557,6 @@
         return REG_SET_VALUE_FAILED;
     }
 
-    rValue.flush();
     rtl_freeMemory(pBuffer);
     return REG_NO_ERROR;
 }
--- registry/source/regimpl.cxx~	2009-11-23 21:07:23.629214375 +0100
+++ registry/source/regimpl.cxx	2009-11-23 21:58:43.982214209 +0100
@@ -851,7 +851,7 @@
     {
         return REG_DELETE_KEY_FAILED;
     }
-    sFile.flush();
+    //sFile.flush();
 
     // set flag deleted !!!
     ((ORegKey*)hOldKey)->setDeleted(sal_True);
@@ -897,7 +897,7 @@
             {
                 return REG_DELETE_VALUE_FAILED;
             }
-        ((OStoreFile&)pKey->getStoreFile()).flush();
+        //((OStoreFile&)pKey->getStoreFile()).flush();
         }
 
         _err = rStoreDir.next(iter);
@@ -1068,7 +1068,7 @@
     {
         return REG_VALUE_NOT_EXISTS;
     }
-    pSourceKey->getStoreFile().flush();
+    //pSourceKey->getStoreFile().flush();
 
     pBuffer = (sal_uInt8*)rtl_allocateMemory(VALUE_HEADERSIZE);
 
@@ -1140,7 +1140,7 @@
         rtl_freeMemory(pBuffer);
         return REG_INVALID_VALUE;
     }
-    rTargetFile.flush();
+    //rTargetFile.flush();
 
     if (rwBytes != nSize)
     {
