--- dbaccess/source/ui/misc/dbumiscres.src-old	2011-05-24 19:52:49.000000000 +0200
+++ dbaccess/source/ui/misc/dbumiscres.src	2011-05-24 19:53:55.000000000 +0200
@@ -96,5 +96,5 @@
 String RID_STR_EXTENSION_NOT_PRESENT
 {
     // #i96130# use hard coded name
-    Text [ en-US ] = "The report, \"$file$\", requires the extension Oracle Report Builder.";
+    Text [ en-US ] = "The report, \"$file$\", requires the extension Oracle Report Builder. For this, install the package 'libreoffice-report-builder'.";
 };
