Index: dbaccess/source/ui/dlg/ExtensionNotPresent.src
===================================================================
--- dbaccess/source/ui/dlg/ExtensionNotPresent.src	(Revision 274294)
+++ dbaccess/source/ui/dlg/ExtensionNotPresent.src	(Arbeitskopie)
@@ -37,8 +37,8 @@
 String RID_STR_EXTENSION_NOT_PRESENT
 {
 	// #i96130# use hard coded name
-	Text [ en-US ] = "To open a report you require the extension Sun™ Report Builder.\n\nClick 'Download...' to download and install the extension.";
-	// OLD: Text [ en-US ] = "To open a report you require the extension %RPT_EXTENSION_NAME.\n\nClick 'Download...' to download and install the extension.";
+    Text [ en-US ] = "To open a report you require the extension Sun™ Report Builder.\n\nor this, install the package 'openoffice.org-report-builder'.";
+    // OLD: Text [ en-US ] = "To open a report you require the extension %RPT_EXTENSION_NAME.\n\nor this, install the package 'openoffice.org-report-builder'.";
 };
 // To open a report you require the extension Sun Report Designer weiss der Geier Hauptsache extra langer Name
 // String RID_STR_EXTENSION_NAME
@@ -73,15 +73,6 @@
         // Text will set outside from RID_STR_EXTENSION_NOT_PRESENT
 	};
 
-	PushButton PB_DOWNLOAD
-	{
-		Pos = MAP_APPFONT ( DLG_WIDTH / 2 - (CELL_PADDING/2) - BUTTON_WIDTH, ACTION_LINE_START ) ;
-		Size = MAP_APPFONT ( BUTTON_WIDTH , BUTTON_HEIGHT ) ;
-		DefButton = TRUE ;
-		TabStop = TRUE ;
-		Text [ en-US ] = "~Download..." ;
-	};
-
 	CancelButton PB_CANCEL
 	{
 		Pos = MAP_APPFONT ( DLG_WIDTH / 2 + (CELL_PADDING/2), ACTION_LINE_START) ;
