Index: l10n/source/uk/localize.sdf
===================================================================
--- l10n/source/uk/localize.sdf	(Revision 274294)
+++ l10n/source/uk/localize.sdf	(Arbeitskopie)
@@ -1368,7 +1368,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	uk	Джерело даних Adabas D не знайдено у вашій системі.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	uk	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	uk	~Завантажити...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	uk	Для редагування звіту потрібне розширення %RPT_EXTENSION_NAME.\n\nНатисніть  "Завантажити..." для завантаження та встановлення розширення.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	uk	Параметри видалення				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	uk	Параметри оновлення				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	uk	Включені поля				2002-02-02 02:02:02
Index: l10n/source/bg/localize.sdf
===================================================================
--- l10n/source/bg/localize.sdf	(Revision 274294)
+++ l10n/source/bg/localize.sdf	(Arbeitskopie)
@@ -1538,7 +1538,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	bg	На вашата система не бяха открити източници на данни на Adabas D.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	bg	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	bg	Из~тегляне...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	bg	За да отворите справка, ви е необходимо разширението Sun™ Report Builder.\n\nНатиснете "Изтегляне...", за да изтеглите и инсталирате разширението.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	bg	Действие при изтриване				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	bg	Действие при обновяване				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	bg	Участващи полета				2002-02-02 02:02:02
Index: l10n/source/ja/localize.sdf
===================================================================
--- l10n/source/ja/localize.sdf	(Revision 274294)
+++ l10n/source/ja/localize.sdf	(Arbeitskopie)
@@ -1578,7 +1578,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	ja	システム上に Adabas D データソースがありません。				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	ja	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	ja	ダウンロード(~D)...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	ja	レポートを開くには、拡張機能 Sun™ Report Builder が必要です。\n\n「ダウンロード...」をクリックすると、拡張機能のダウンロードとインストールができます。				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	ja	削除オプション				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	ja	更新オプション				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	ja	含まれるフィールド				2002-02-02 02:02:02
Index: l10n/source/fi/localize.sdf
===================================================================
--- l10n/source/fi/localize.sdf	(Revision 274294)
+++ l10n/source/fi/localize.sdf	(Arbeitskopie)
@@ -1579,7 +1579,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	fi	Adabas D -tietolähteitä ei löydetty järjestelmästäsi.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	fi	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	fi	Lataa...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				1	fi	Avataksesi raportin tarvitset Sun™ Report Builder -lisäosan.\n\nNapsauta 'Lataa...' ladataksesi ja asentaaksesi sen.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	fi	Poista asetukset				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	fi	Päivitä asetukset				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	fi	Sisällytetyt kentät				2002-02-02 02:02:02
Index: l10n/source/bn/localize.sdf
===================================================================
--- l10n/source/bn/localize.sdf	(Revision 274294)
+++ l10n/source/bn/localize.sdf	(Arbeitskopie)
@@ -1579,7 +1579,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	bn	আপনার পদ্ধতিতে কোনো Adabas D উপাত্ত উৎস খুঁজে পাওয়া যায়নি।				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	bn	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	bn	ডাউনলোড... (~D)				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	bn	একটি প্রতিবেদন খুলতে আপনার সমপ্রসারন Sun™ প্রতিবেদন প্রস্তুতকারকের প্রয়োজন।\n\nসমপ্রসারন ডাউনলোড এবং সংস্থাপন করতে 'ডাউনলোড...' এ ক্লিক করুন।				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	bn	মুছে ফেলার পছন্দসমূহ				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	bn	হালনাগাদ পছন্দসমূহ				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	bn	ক্ষেত্রসমূহ সংশ্লিষ্ট				2002-02-02 02:02:02
Index: l10n/source/nb/localize.sdf
===================================================================
--- l10n/source/nb/localize.sdf	(Revision 274294)
+++ l10n/source/nb/localize.sdf	(Arbeitskopie)
@@ -1378,7 +1378,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	nb	Fant ingen Adabas D-datakilder på systemet.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	nb	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	nb	~Last ned				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	nb	Du trenger utvidelsen %RPT_EXTENSION_NAME for å åpne rapporter.\n\nTrykk på «Last ned» for å laste ned og installere utvidelsen.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	nb	Slett valg				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	nb	Oppdateringsvalg				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	nb	Involverte felter				2002-02-02 02:02:02
Index: l10n/source/ne/localize.sdf
===================================================================
--- l10n/source/ne/localize.sdf	(Revision 274294)
+++ l10n/source/ne/localize.sdf	(Arbeitskopie)
@@ -1226,7 +1226,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	ne	तपाईँको प्रणालीमा Adabas D डेटा स्रोतहरू फेला पार्न सकिएन ।				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	ne	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	ne	डाउनलोड...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	ne	प्रतिवेदन खोल्न तपाईँलाई विस्तार %RPT_EXTENSION_NAME आवश्यक पर्दछ ।\n\nविस्तार डाउनलोड र स्थापना गर्न 'डाउनलोड...' क्लिक गर्नुहोस् ।				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	ne	विकल्प मेट्नुहोस्				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	ne	अद्यावधिक विकल्प				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	ne	संलग्न फाँटहरू				2002-02-02 02:02:02
Index: l10n/source/br/localize.sdf
===================================================================
--- l10n/source/br/localize.sdf	(Revision 274294)
+++ l10n/source/br/localize.sdf	(Arbeitskopie)
@@ -1268,7 +1268,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	br	Kavet ez eus bet tarzh roadennoù Adabas D ebet war ar reizhiad.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	br	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	br	~Pellgargañ...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	br	Evit digeriñ ur marilh ho po ezhomm eus an askouezhad %RPT_EXTENSION_NAME.\n\nKlikañ war 'Pellgargañ...' evit pellgargañ ha staliañ an askouezhad.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	br	Dibarzhioù dilemel				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	br	Dibarzhioù hizivaat				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	br	Maeziennoù empleget				2002-02-02 02:02:02
Index: l10n/source/as-IN/localize.sdf
===================================================================
--- l10n/source/as-IN/localize.sdf	(Revision 274294)
+++ l10n/source/as-IN/localize.sdf	(Arbeitskopie)
@@ -1579,7 +1579,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	as-IN	কোনো এডাবেছ D ডাটা উত্স আপোনাৰ ছিষ্টেমত পোৱা নগ'ল.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	as-IN	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	as-IN	ডাউন্‌লোড কৰক (~D)...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	as-IN	এটা প্ৰতিবেদন খুলিবলৈ আপোনাক Sun™ Report Builder সম্প্ৰসাৰণৰ প্ৰয়োজন ।\n\nসম্প্ৰসাৰণ ডাউন্‌লোড কৰি তাক সংস্থাপন কৰিবলৈ 'ডাউন্‌লোড কৰক...' টিপক ।				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	as-IN	বিকল্পবোৰ ডিলিট কৰক				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	as-IN	বিকল্পবোৰ আপডেট কৰক				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	as-IN	ফিল্ডবোৰ জড়িত				2002-02-02 02:02:02
Index: l10n/source/fr/localize.sdf
===================================================================
--- l10n/source/fr/localize.sdf	(Revision 274294)
+++ l10n/source/fr/localize.sdf	(Arbeitskopie)
@@ -1578,7 +1578,7 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	fr	Aucune source de données Adabas D n'a été trouvée sur le système.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	fr	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	fr	~Télécharger...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	fr	L'ouverture d'un rapport requiert l'extension Sun™ Report Builder.\n\nCliquez sur 'Télécharger...' pour télécharger et installer l'extension.				2002-02-02 02:02:02
+dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	fr	L'ouverture d'un rapport requiert l'extension %RPT_EXTENSION_NAME.\n\nPour cela, installez le package 'openoffice.org-report-builder'.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	fr	Options de suppression				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	fr	Options de mise à jour				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	fr	Champs impliqués				2002-02-02 02:02:02
Index: l10n/source/nl/localize.sdf
===================================================================
--- l10n/source/nl/localize.sdf	(Revision 274294)
+++ l10n/source/nl/localize.sdf	(Arbeitskopie)
@@ -1578,7 +1578,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	nl	Er zijn geen Adabas D-gegevensbronnen gevonden op uw systeem.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	nl	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	nl	~Downloaden...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	nl	U hebt de extensie Sun™ Report Builder nodig om een rapport te kunnen openen.\n\nKlik op 'Downloaden...' om de extensie te downloaden en te installeren.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	nl	Wisopties				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	nl	Bijwerkopties				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	nl	betrokken velden				2002-02-02 02:02:02
Index: l10n/source/nn/localize.sdf
===================================================================
--- l10n/source/nn/localize.sdf	(Revision 274294)
+++ l10n/source/nn/localize.sdf	(Arbeitskopie)
@@ -1402,7 +1402,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	nn	Fann ingen Adabas D-datakjelder på systemet.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	nn	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	nn	~Last ned				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	nn	Du treng utvidinga %RPT_EXTENSION_NAME for å opna rapportar.\n\nTrykk på «Last ned» for å lasta ned og installera utvidinga.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	nn	Innstillingar for sletting				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	nn	Oppdateringsval				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	nn	Involverte felt				2002-02-02 02:02:02
Index: l10n/source/vi/localize.sdf
===================================================================
--- l10n/source/vi/localize.sdf	(Revision 274294)
+++ l10n/source/vi/localize.sdf	(Arbeitskopie)
@@ -1578,7 +1578,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	vi	Không tìm thấy nguồn dữ liệu Adabas D trên hệ thống của bạn.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	vi	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	vi	Tải ~về...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	vi	Để mở một báo cáo bạn cần phần mở rộng Sun™ Report Builder.\n\nNhấn vào « Tải về... » để tải về và cài đặt phần mở rộng này.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	vi	Xoá tùy chọn				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	vi	Cập nhật tùy chọn				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	vi	Trường liên quan				2002-02-02 02:02:02
Index: l10n/source/ca/localize.sdf
===================================================================
--- l10n/source/ca/localize.sdf	(Revision 274294)
+++ l10n/source/ca/localize.sdf	(Arbeitskopie)
@@ -1578,7 +1578,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	ca	No s'han trobat fonts de dades Adabas D al sistema.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	ca	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	ca	~Baixa...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	ca	Per a obrir un informe necessiteu l'extensió Constructor d'informes de Sun™.\n\nFeu clic a "Baixa..." per a baixar i instal·lar l'extensió.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	ca	Opcions de supressió				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	ca	Opcions d'actualització				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	ca	Camps implicats				2002-02-02 02:02:02
Index: l10n/source/pt-BR/localize.sdf
===================================================================
--- l10n/source/pt-BR/localize.sdf	(Revision 274294)
+++ l10n/source/pt-BR/localize.sdf	(Arbeitskopie)
@@ -1578,7 +1578,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	pt-BR	Não foi encontrada nenhuma fonte de dados do Adabas D no sistema.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	pt-BR	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	pt-BR	~Download...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	pt-BR	Para abrir um relatório, é necessário a extensão Sun™ Report Builder.\n\nClique em 'Download...' para fazer o download e instalar a extensão.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	pt-BR	Excluir opções				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	pt-BR	Atualizar opções				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	pt-BR	Campos envolvidos				2002-02-02 02:02:02
Index: l10n/source/ru/localize.sdf
===================================================================
--- l10n/source/ru/localize.sdf	(Revision 274294)
+++ l10n/source/ru/localize.sdf	(Arbeitskopie)
@@ -1578,7 +1578,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	ru	Источник данных Adabas D был найден в вашей системе.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	ru	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	ru	Загрузить...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	ru	Для редактирования отчета требуется расширение Sun™ Report Builder.\n\nЩелкните  "Загрузить..." для загрузки и установки расширения.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	ru	Параметры удаления				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	ru	Параметры обновления				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	ru	Включенные поля				2002-02-02 02:02:02
Index: l10n/source/rw/localize.sdf
===================================================================
--- l10n/source/rw/localize.sdf	(Revision 274294)
+++ l10n/source/rw/localize.sdf	(Arbeitskopie)
@@ -1208,7 +1208,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	rw	Nta masoko y'ibyatanzwe Adabas D yabonetse muri sisitemu yawe.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	rw	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	rw	Gutangiza...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	rw	Kugira ngo ufungure raporo usaba umugereka %RPT_EXTENSION_NAME.\n\nKanda 'Gukurura...' kugira ngo ukurure uninjizemo umugereka.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	rw	Amahitamo yo gusiba				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	rw	Amahitamo y'Ihuzagihe				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	rw	Imyanya ikoreshwa				2002-02-02 02:02:02
Index: l10n/source/gl/localize.sdf
===================================================================
--- l10n/source/gl/localize.sdf	(Revision 274294)
+++ l10n/source/gl/localize.sdf	(Arbeitskopie)
@@ -1579,7 +1579,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	gl	Non se encontraron fontes de bases de datos Adabas D no seu sistema.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	gl	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	gl	~Descargar...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	gl	Para abrir un informe necesita a extensión Sun™ Report Builder.\n\nPrema 'Descargar...' para descargar e instalar a extensión.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	gl	Opcións de eliminación				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	gl	Actualizar opcións				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	gl	Campos implicados				2002-02-02 02:02:02
Index: l10n/source/cs/localize.sdf
===================================================================
--- l10n/source/cs/localize.sdf	(Revision 274294)
+++ l10n/source/cs/localize.sdf	(Arbeitskopie)
@@ -1579,7 +1579,7 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	cs	Žádné datové zdroje Adabas D nebyly ve vašem systému nalezeny.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	cs	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	cs	Stáhnout...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	cs	Pro otevření sestavy je potřeba rozšíření Sun™ Report Builder.\n\nKlepněte na 'Stáhnout...', pokud chcete rozšíření stáhnout a nainstalovat.				2002-02-02 02:02:02
+dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	cs	Prootevření sestavy je potřeba rozšíření Sun™ Report Builder.\n\nProsím nainstalujte si balík 'openoffice.org-report-builder'.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	cs	Nastavení smazání				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	cs	Nastavení aktualizace				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	cs	Použitá pole				2002-02-02 02:02:02
Index: l10n/source/km/localize.sdf
===================================================================
--- l10n/source/km/localize.sdf	(Revision 274294)
+++ l10n/source/km/localize.sdf	(Arbeitskopie)
@@ -1579,7 +1579,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	km	គ្មាន​ប្រភព​ទិន្នន័យ​ Adabas D ត្រូវ​បាន​រកឃើញលើ​ប្រព័ន្ធ​របស់​អ្នក ។				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	km	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	km	ទាញយក... 				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	km	ដើម្បី​បើក​របាយការណ៍ គឺ​អ្នក​ត្រូវការ​ផ្នែក​បន្ថែម Sun™ Report Builder ។\n\nចុច 'ទាញយក...' ដើម្បី​ទាញយក និង​ដំឡើង​ផ្នែក​បន្ថែម ។				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	km	លុប​​ជម្រើស				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	km	ជម្រើស​ធ្វើ​ឲ្យ​ទាន់​សម័យ				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	km	វាល​ដែល​​ពាក់​ព័ន្ធ				2002-02-02 02:02:02
Index: l10n/source/ko/localize.sdf
===================================================================
--- l10n/source/ko/localize.sdf	(Revision 274294)
+++ l10n/source/ko/localize.sdf	(Arbeitskopie)
@@ -1578,7 +1578,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	ko	시스템에 Adabas D 데이터 원본이 없습니다.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	ko	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	ko	다운로드(~D)...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	ko	보고서를 열려면 Sun™ Report Builder 확장이 필요합니다.\n\n'다운로드...'를 클릭하고 확장 기능을 다운로드하여 설치합니다.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	ko	삭제 옵션				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	ko	업데이트 옵션				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	ko	관련 필드				2002-02-02 02:02:02
Index: l10n/source/sh/localize.sdf
===================================================================
--- l10n/source/sh/localize.sdf	(Revision 274294)
+++ l10n/source/sh/localize.sdf	(Arbeitskopie)
@@ -1579,7 +1579,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	sh	Nije pronađen Adabas D izvor podataka na vašem sistemu.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	sh	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	sh	~Preuzmi...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	sh	Za otvaranje izveštaja potreban je dodatak Sun-ov konstruktor izveštaja.\n\nKliknite „Preuzmi...“ kako biste preuzeli i instalirali dodatak.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	sh	Obriši opcije				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	sh	Opcije osvežavanja				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	sh	Uključena polja				2002-02-02 02:02:02
Index: l10n/source/gu/localize.sdf
===================================================================
--- l10n/source/gu/localize.sdf	(Revision 274294)
+++ l10n/source/gu/localize.sdf	(Arbeitskopie)
@@ -1534,7 +1534,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	gu	કોઈ Adabas D માહિતી સ્રોતો તમારી સિસ્ટમ પર મળ્યા ન હતા.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	gu	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	gu	ડાઉનલોડ (~D)...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	gu	અહેવાલ ખોલવા માટે તમને એક્સટેન્સન Sun™ અહેવાલ બિલ્ડરની જરૂર છે.\n\nએક્સટેન્સન ડાઉનલોડ કરવા અને સ્થાપિત કરવા માટે 'ડાઉનલોડ...' ક્લિક કરો.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	gu	વિકલ્પોને રદ કરો				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	gu	વિકલ્પો બદલવા				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	gu	સમાવિષ્ટ કરાયેલ ક્ષેત્રો				2002-02-02 02:02:02
Index: l10n/source/kn-IN/localize.sdf
===================================================================
--- l10n/source/kn-IN/localize.sdf	(Revision 274294)
+++ l10n/source/kn-IN/localize.sdf	(Arbeitskopie)
@@ -1269,7 +1269,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	kn-IN	ನಿಮ್ಮ ಗಣಕದಲ್ಲಿ ಯಾವುದೆ Adabas D ದತ್ತ ಆಕರಗಳು ಕಂಡುಬಂದಿಲ್ಲ.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	kn-IN	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	kn-IN	ಡೌನ್‌ಲೋಡ್ ಮಾಡು(~D)...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	kn-IN	ಒಂದು ವರದಿಯನ್ನು ತೆರೆಯಲು ನಿಮ್ಮಲ್ಲಿ Sun™ ರಿಪೋರ್ಟ್ ಬಿಲ್ಡರ್ ಎನ್ನುವ ಒಂದು ವಿಸ್ತರಣೆಯು ಇರಬೇಕು.\n\nಈ ವಿಸ್ತರಣೆಯನ್ನು ಡೌನ್‌ಲೋಡ್‌ ಮಾಡಿ ನಂತರ ಅನುಸ್ಥಾಪಿಸಲು 'ಡೌನ್‌ಲೋಡ್‌ ಮಾಡು...' ಅನ್ನು ಕ್ಲಿಕ್ಕಿಸಿ.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	kn-IN	ಅಳಿಸುವ ಆಯ್ಕೆಗಳು				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	kn-IN	ಅಪ್‌ಡೇಟ್‌ ಆಯ್ಕೆಗಳು				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	kn-IN	ಒಳಗೊಂಡಿರುವ ಕ್ಷೇತ್ರಗಳು				2002-02-02 02:02:02
Index: l10n/source/sk/localize.sdf
===================================================================
--- l10n/source/sk/localize.sdf	(Revision 274294)
+++ l10n/source/sk/localize.sdf	(Arbeitskopie)
@@ -1537,7 +1537,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	sk	Databázový zdroj Adabas D  nebol na tomto systéme nájdený				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	sk	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	sk	Stia~hnuť...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				1	sk	Pre otvorenie správy potrebujete rozšírenie Sun™ Report Builder.\n\nKliknite na 'Stiahnuť...' pre stahnutie a inštaláciu rozšírenia.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	sk	Zmazať možnosti				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	sk	Aktualizovať možnosti				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	sk	Zahrnuté polia				2002-02-02 02:02:02
Index: l10n/source/sl/localize.sdf
===================================================================
--- l10n/source/sl/localize.sdf	(Revision 274294)
+++ l10n/source/sl/localize.sdf	(Arbeitskopie)
@@ -1579,7 +1579,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	sl	V sistemu ni mogoče najti virov podatkov vrste Adabas D.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	sl	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	sl	~Prenesi ...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	sl	Če želite odpreti poročilo, potrebujete razširitev Oblikovalec poročil podjetja Sun™.\n\nKliknite 'Prenesi ...', če želite prenesti in namestiti razširitev.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	sl	Možnosti brisanja				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	sl	Možnosti posodobitve				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	sl	Vključena so polja				2002-02-02 02:02:02
Index: l10n/source/zh-CN/localize.sdf
===================================================================
--- l10n/source/zh-CN/localize.sdf	(Revision 274294)
+++ l10n/source/zh-CN/localize.sdf	(Arbeitskopie)
@@ -1578,7 +1578,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	zh-CN	未在系统中找到任何 Adabas D 数据源。				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	zh-CN	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	zh-CN	下载(~D)...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	zh-CN	为打开报表您需要 Sun™ 报表构建器扩展。\n\n请单击“下载...”以下载和安装该扩展。				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	zh-CN	删除的选项				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	zh-CN	更新选项				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	zh-CN	关联字段				2002-02-02 02:02:02
Index: l10n/source/mr-IN/localize.sdf
===================================================================
--- l10n/source/mr-IN/localize.sdf	(Revision 274294)
+++ l10n/source/mr-IN/localize.sdf	(Arbeitskopie)
@@ -1544,7 +1544,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	mr-IN	आपल्या पद्धतीवर कुठलाही Adabas D डाटा स्त्रोत सापडत नाही.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	mr-IN	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	mr-IN	डाऊनलोड (~D)...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	mr-IN	अहवाल उघडण्यासाठी आपणास एक्सटेन्शन  %RPT_EXTENSION_NAME ची आवश्यकता आहे.\n\nएक्सटेन्शला उतरवून घेण्यासाठी  'उतरवून घ्या...' वर क्लिक करा आणि एक्सटेन्शला स्थापित करा.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	mr-IN	पर्याय नष्ट करा				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	mr-IN	अद्ययावत करण्याचे पर्याय				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	mr-IN	समाविष्ट असलेली क्षेत्रे				2002-02-02 02:02:02
Index: l10n/source/da/localize.sdf
===================================================================
--- l10n/source/da/localize.sdf	(Revision 274294)
+++ l10n/source/da/localize.sdf	(Arbeitskopie)
@@ -1579,7 +1579,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	da	Ingen Adabas D-datakilder blev fundet på dit system.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	da	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	da	~Hent...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				1	da	For at åbne en rapport kræves udvidelsen Sun™ Report Builder.\n\nKlik på 'Hent...' for at hente og installere udvidelsen.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	da	Sletningsindstillinger				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	da	Opdateringsindstillinger				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	da	Involverede felter				2002-02-02 02:02:02
Index: l10n/source/sr/localize.sdf
===================================================================
--- l10n/source/sr/localize.sdf	(Revision 274294)
+++ l10n/source/sr/localize.sdf	(Arbeitskopie)
@@ -1579,7 +1579,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	sr	Није пронађен Adabas D извор података на вашем систему.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	sr	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	sr	~Преузми...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	sr	За отварање извештаја потребан је додатак Sun-ов конструктор извештаја.\n\nКликните „Преузми...“ како бисте преузели и инсталирали додатак.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	sr	Обриши опције				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	sr	Опције освежавања				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	sr	Укључена поља				2002-02-02 02:02:02
Index: l10n/source/de/localize.sdf
===================================================================
--- l10n/source/de/localize.sdf	(Revision 274294)
+++ l10n/source/de/localize.sdf	(Arbeitskopie)
@@ -1578,7 +1578,7 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	de	Auf Ihrem System wurden keine Adabas D Datenquellen gefunden.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	de	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	de	~Download...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	de	Zum Öffnen eines Berichts wird die Erweiterung Sun™ Report Builder benötigt.\n\nKlicken Sie auf 'Download...', um die Erweiterung herunterzuladen und zu installieren.				2002-02-02 02:02:02
+dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	de	Zum Öffnen eines Berichts wird die Erweiterung Sun™ Report Builder benötigt.\n\nInstallieren Sie dafür das Paket 'openoffice.org-report-builder'.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	de	Löschoptionen				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	de	Update Optionen				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	de	beteiligte Felder				2002-02-02 02:02:02
Index: l10n/source/sv/localize.sdf
===================================================================
--- l10n/source/sv/localize.sdf	(Revision 274294)
+++ l10n/source/sv/localize.sdf	(Arbeitskopie)
@@ -1578,7 +1578,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	sv	Inga Adabas D-datakällor kunde hittas i systemet.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	sv	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	sv	~Ladda ner...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	sv	För att öppna en rapport behöver du tillägget Sunâ„¢ Rapportdesign.\n\nKlicka 'Hämta...' för att hämta och installera tillägget.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	sv	Raderingsalternativ				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	sv	Uppdateringsalternativ				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	sv	Inkluderade fält				2002-02-02 02:02:02
Index: l10n/source/hi-IN/localize.sdf
===================================================================
--- l10n/source/hi-IN/localize.sdf	(Revision 274294)
+++ l10n/source/hi-IN/localize.sdf	(Arbeitskopie)
@@ -1579,7 +1579,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	hi-IN	कोई Adabas D data स्रोत आपके तंत्र पर नहीं पाया गया.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	hi-IN	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	hi-IN	डॉउनलोड करें (~D)...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	hi-IN	एक रिपोर्ट खोलने के लिए आपको Sun™ Report Builder विस्तार की जरूरत है.\n\n'डाउनलोड करें...' को विस्तार को डाउनलोड और संस्थापित करने के लिए क्लिक करें.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	hi-IN	विकल्प मिटाएँ				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	hi-IN	विकल्प अद्यतन करें				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	hi-IN	क्षेत्र समाहित				2002-02-02 02:02:02
Index: l10n/source/pa-IN/localize.sdf
===================================================================
--- l10n/source/pa-IN/localize.sdf	(Revision 274294)
+++ l10n/source/pa-IN/localize.sdf	(Arbeitskopie)
@@ -1526,7 +1526,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	pa-IN	ਤੁਹਾਡੇ ਸਿਸਟਮ ਉੱਪਰ ਕੋਈ Adabas D ਡਾਟਾ ਸਰੋਤ ਨਹੀਂ ਲੱਭਿਆ।				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	pa-IN	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	pa-IN	ਡਾਊਨਲੋਡ(~d)...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	pa-IN	ਇੱਕ ਰਿਪੋਰਟ ਖੋਲ੍ਹਣ ਵਾਸਤੇ ਇਕਸਟੈਨਸ਼ਨ %RPT_EXTENSION_NAME ਲੋੜੀਦੀ ਹੈ।\n\nਇਕਸਟੈਨਸ਼ਨ ਡਾਊਨਲੋਡ ਅਤੇ ਇੰਸਟਾਲ ਕਰਨ ਵਾਸਤੇ 'ਡਾਊਨਲੋਡ...' ਉੱਤੇ ਕਲਿੱਕ ਕਰੋ।				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	pa-IN	ਚੋਣ ਹਟਾਓ				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	pa-IN	ਅੱਪਡੇਟ ਚੋਣ				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	pa-IN	ਸ਼ਾਮਿਲ ਖੇਤਰ				2002-02-02 02:02:02
Index: l10n/source/ta/localize.sdf
===================================================================
--- l10n/source/ta/localize.sdf	(Revision 274294)
+++ l10n/source/ta/localize.sdf	(Arbeitskopie)
@@ -1534,7 +1534,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	ta	Adabas D தரவு மூலங்கள் உங்கள் முறைமையில் காணப்படவில்லை.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	ta	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	ta	~பதிவிறக்கு...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	ta	ஒரு அறிக்கையைத் திறக்க Sun™ Report Builder நீட்சி உங்களுக்குத் தேவை.\n\nநீட்சியை பதிவிறக்கி நிறுவ, 'பதிவிறக்கு...'ஐச் சொடுக்கவும்.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	ta	நீக்கத் தேர்வுகள்				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	ta	தேர்வுகளைப் புதுப்பி				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	ta	ஈடுபட்டுள்ள புலமங்கள்				2002-02-02 02:02:02
Index: l10n/source/sa-IN/localize.sdf
===================================================================
--- l10n/source/sa-IN/localize.sdf	(Revision 274294)
+++ l10n/source/sa-IN/localize.sdf	(Arbeitskopie)
@@ -1378,7 +1378,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	sa-IN	किमपि Adabas D लेखास्रोतः भवतः संगणके न प्राप्तम् ।				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	sa-IN	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	sa-IN	~डौन्लोड्...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				1	sa-IN	रिपोर्ट् उद्घाटितुं तुभ्यं %RPT_EXTENSION_NAME विस्तारणम् आवश्यकम्।\n\nविस्तारणं संस्थापितिम् अथवा डौन्लोड् कर्तुं 'डोन्लोड्...' क्लिक् कुरु।				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	sa-IN	लोपनविकल्पाः				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	sa-IN	अद्यतनीकरणविकल्पाः				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	sa-IN	क्षेत्राणि सम्मिलितानि				2002-02-02 02:02:02
Index: l10n/source/hr/localize.sdf
===================================================================
--- l10n/source/hr/localize.sdf	(Revision 274294)
+++ l10n/source/hr/localize.sdf	(Arbeitskopie)
@@ -1225,7 +1225,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	hr	Adabas D izvori podataka nisu pronađeni na vašem sistemu.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	hr	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	hr	~Preuzmi...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	hr	Za otvaranje izvješća morate imati %RPT_EXTENSION_NAME proširenje.\n\nKliknite na  'Preuzmi...' kako bi preuzeli i instalirali proširenje.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	hr	Brisanje odrednica				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	hr	Ažuriranje odrednica				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	hr	Uključena polja				2002-02-02 02:02:02
Index: l10n/source/pl/localize.sdf
===================================================================
--- l10n/source/pl/localize.sdf	(Revision 274294)
+++ l10n/source/pl/localize.sdf	(Arbeitskopie)
@@ -1578,7 +1578,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	pl	W systemie nie znaleziono źródeł danych Adabas D.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	pl	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	pl	~Pobierz...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	pl	Aby otworzyć raport, musisz skorzystać z rozszerzenia Sun™ Report Builder.\n\nKliknij przycisk "Pobierz...", aby pobrać i zainstalować rozszerzenie.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	pl	Opcje usuwania				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	pl	Opcje aktualizacji				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	pl	Złączone pola				2002-02-02 02:02:02
Index: l10n/source/hu/localize.sdf
===================================================================
--- l10n/source/hu/localize.sdf	(Revision 274294)
+++ l10n/source/hu/localize.sdf	(Arbeitskopie)
@@ -1579,7 +1579,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	hu	A rendszeren nem található Adabas D-adatforrás.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	hu	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	hu	~Letöltés...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	hu	Jelentés megnyitásához a Sun™ Report Builder kiterjesztésre van szükség.\n\nKattintson a „Letöltés...” lehetőségre a kiterjesztés letöltéséhez és telepítéséhez.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	hu	Beállítások törlése				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	hu	Beállítások frissítése				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	hu	Érintett mezők				2002-02-02 02:02:02
Index: l10n/source/te-IN/localize.sdf
===================================================================
--- l10n/source/te-IN/localize.sdf	(Revision 274294)
+++ l10n/source/te-IN/localize.sdf	(Arbeitskopie)
@@ -1579,7 +1579,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	te-IN	మీ వ్యవస్థలో అడాబాస్ దత్తాంశం మూలాధారం దోరకలేదు				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	te-IN	%PRODUCTNAME %PRODUCTVERSIONము				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	te-IN	సమాచార దిగుమతి (~D)				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	te-IN	ఒక నివేదికను తెరువుటకు మీకు Sun™ రిపోర్టు బల్డర్ పొడిగింపు అవసరము.\n\nపొడిగింపును డౌన్‌లోడ్ చేసి సంస్థాపించుటకు 'డౌన్‌లోడ్...' నొక్కుము.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	te-IN	తోలగించు ఇచ్ఛాపూర్వకాలు				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	te-IN	తాజాపరుచు ఇచ్ఛాపూర్వకాలు				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	te-IN	 నిహితమైన క్షేత్రములు				2002-02-02 02:02:02
Index: l10n/source/lt/localize.sdf
===================================================================
--- l10n/source/lt/localize.sdf	(Revision 274294)
+++ l10n/source/lt/localize.sdf	(Arbeitskopie)
@@ -1488,7 +1488,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	lt	„Adabas D“ duomenų šaltinių kompiuteryje nerasta.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	lt	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	lt	~Atsisiųsti...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	lt	Ataskaitai atverti reikalingas plėtinys „Sun™ Report Builder“.\n\nSpustelėkite „Atsisiųsti...“, kad atsisiųstumėte ir įdiegtumėte plėtinį.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	lt	Šalinti parinktis				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	lt	Atnaujinti parinktis				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	lt	Esami laukai				2002-02-02 02:02:02
Index: l10n/source/pt/localize.sdf
===================================================================
--- l10n/source/pt/localize.sdf	(Revision 274294)
+++ l10n/source/pt/localize.sdf	(Arbeitskopie)
@@ -1578,7 +1578,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	pt	Não foram encontradas origens de dados Adabas D no sistema.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	pt	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	pt	~Download...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	pt	Para abrir um relatório é necessária a extensão do Sun™ Report Builder.\n\nFaça clique em 'Transferir de...' para transferir e instalar a extensão.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	pt	Opções de eliminação				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	pt	Opções de actualização				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	pt	Campos considerados				2002-02-02 02:02:02
Index: l10n/source/tr/localize.sdf
===================================================================
--- l10n/source/tr/localize.sdf	(Revision 274294)
+++ l10n/source/tr/localize.sdf	(Arbeitskopie)
@@ -1578,7 +1578,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	tr	Adabas D veri kaynakları sisteminizde bulunamadı.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	tr	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	tr	İn~dir...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	tr	Raporu açmak için Sun™ Rapor Oluşturucu gereklidir.\n\nEklentiyi indirip yüklemek için "İndir..." düğmesine tıklayın.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	tr	Silme seçenekleri				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	tr	Güncelleme seçenekleri				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	tr	Kapsayan alanlar				2002-02-02 02:02:02
Index: l10n/source/en-GB/localize.sdf
===================================================================
--- l10n/source/en-GB/localize.sdf	(Revision 274294)
+++ l10n/source/en-GB/localize.sdf	(Arbeitskopie)
@@ -1579,7 +1579,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	en-GB	No Adabas D data sources were found on your system.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	en-GB	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	en-GB	~Download...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	en-GB	To open a report you require the extension Sun™ Report Builder.\n\nClick 'Download...' to download and install the extension.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	en-GB	Delete options				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	en-GB	Update options				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	en-GB	Fields involved				2002-02-02 02:02:02
Index: l10n/source/id/localize.sdf
===================================================================
--- l10n/source/id/localize.sdf	(Revision 274294)
+++ l10n/source/id/localize.sdf	(Arbeitskopie)
@@ -1158,7 +1158,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	id	Tidak ada sumber data Adabas D yang ditemukan pada sistem Anda.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	id	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	id	~Download...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	id	Untuk membuka laporan, Anda memerlukan ekstensi Sun™ Report Builder.\n\nKlik 'Download...' untuk mendapatkan dan memasang ekstensi.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	id	Opsi-opsi penghapusan				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	id	Opsi-opsi update				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	id	Bagian-bagian yang terlibat				2002-02-02 02:02:02
Index: l10n/source/be-BY/localize.sdf
===================================================================
--- l10n/source/be-BY/localize.sdf	(Revision 274294)
+++ l10n/source/be-BY/localize.sdf	(Arbeitskopie)
@@ -1374,7 +1374,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	be-BY	У вашай сістэме не знойдзена крыніц даных Adabas D.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	be-BY	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	be-BY	Атрымаць...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	be-BY	Каб адкрыць паведамленне, патрэбнае пашырэнне %RPT_EXTENSION_NAME.\n\nНацісніце 'Атрымаць...' каб атрымаць і ўстанавіць пашырэнне.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	be-BY	Настаўленні сцірання				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	be-BY	Настаўленні абнаўлення				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	be-BY	Палі, якія ўдзельнічаюць				2002-02-02 02:02:02
Index: l10n/source/el/localize.sdf
===================================================================
--- l10n/source/el/localize.sdf	(Revision 274294)
+++ l10n/source/el/localize.sdf	(Arbeitskopie)
@@ -1470,7 +1470,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	el	Δεν βρέθηκε καμία προέλευση δεδομένων του Adabas D στο σύστημα σας.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	el	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	el	~Λήψη...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	el	Για να ανοίξετε μια αναφορά χρειάζεστε την επέκταση Sun™ Report Builder.\n\nΠατήστε 'Κατέβασμα...' για να κατεβάσετε και να εγκαταστήσετε την επέκταση.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	el	Επιλογές διαγραφής				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	el	Επιλογές ενημέρωσης				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	el	Εμπλεκόμενα πεδία				2002-02-02 02:02:02
Index: l10n/source/ar/localize.sdf
===================================================================
--- l10n/source/ar/localize.sdf	(Revision 274294)
+++ l10n/source/ar/localize.sdf	(Arbeitskopie)
@@ -1578,7 +1578,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	ar	لم يتم العثور على مصادر بيانات Adabas D على نظامك.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	ar	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	ar	~تنزيل...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	ar	لفتح تقرير، تحتاج إلى الامتداد Sun™ Report Builder.\n\nانقر فوق "تنزيل..." لتنزيل الامتداد وتثبيته.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	ar	خيارات الحذف				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	ar	خيارات التحديث				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	ar	الحقول المعنية				2002-02-02 02:02:02
Index: l10n/source/eo/localize.sdf
===================================================================
--- l10n/source/eo/localize.sdf	(Revision 274294)
+++ l10n/source/eo/localize.sdf	(Arbeitskopie)
@@ -1206,7 +1206,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	eo	Neniu datumfonto Adabas D estis trovita en via sistemo.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	eo	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	eo	Elsxuti...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				1	eo	Por malfermi raporton vi bezonas la etendajxon %RPT_EXTENSION_NAME.\n\nAlklaku 'Elsxuti...' por elsxuti kaj instali la etendajxon.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	eo	Forigaj agordajxoj				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	eo	Gxisdatigi agordajxojn				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	eo	Kampoj koncernataj!				2002-02-02 02:02:02
Index: l10n/source/mk/localize.sdf
===================================================================
--- l10n/source/mk/localize.sdf	(Revision 274294)
+++ l10n/source/mk/localize.sdf	(Arbeitskopie)
@@ -1579,7 +1579,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	mk	На Вашиот систем не беа пронајдени Adabas D извори на податоци.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	mk	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	mk	~Симни...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	mk	За да го отворите извештајот потребна ви е екстензијата Sun™ Report Builder.\n\nКликнете на 'Симни...' за да ја симнете и инсталирате екстензијата.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	mk	Опции при бришење				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	mk	Опции при ажурирање				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	mk	Вклучени полиња				2002-02-02 02:02:02
Index: l10n/source/es/localize.sdf
===================================================================
--- l10n/source/es/localize.sdf	(Revision 274294)
+++ l10n/source/es/localize.sdf	(Arbeitskopie)
@@ -1578,7 +1578,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	es	No se han encontrado orígenes de datos Adabas D en el sistema.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	es	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	es	~Descargar...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	es	Para abrir un informe se requiere la extensión Sun™ Report Builder.\n\nHaga clic en 'Descargar...' para descargar e instalar la extensión.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	es	Opciones de eliminación				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	es	Opciones de actualización				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	es	Campos implicados				2002-02-02 02:02:02
Index: l10n/source/et/localize.sdf
===================================================================
--- l10n/source/et/localize.sdf	(Revision 274294)
+++ l10n/source/et/localize.sdf	(Arbeitskopie)
@@ -1579,7 +1579,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	et	Sinu süsteemist ei leitud Adabas D andmeallikaid.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	et	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	et	Laadi alla...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	et	Aruande avamiseks on vaja laiendust Sun™-i aruandekoostaja.\n\nLaienduse allalaadimiseks ja paigaldamiseks klõpsa 'Laadi alla...'.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	et	Kustutamise sätted				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	et	Värskendamise sätted				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	et	Asjassepuutuvad väljad				2002-02-02 02:02:02
Index: l10n/source/eu/localize.sdf
===================================================================
--- l10n/source/eu/localize.sdf	(Revision 274294)
+++ l10n/source/eu/localize.sdf	(Arbeitskopie)
@@ -1535,7 +1535,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	eu	Ez da Adabas D datu-iturbururik aurkitu zure sisteman.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	eu	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	eu	~Deskargatu...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	eu	Txosten bat irekitzeko 'Sun™ txostenen diseinatzailea'\n(Sun™ Report Builder) hedapena behar duzu.\n\nEgin klik "Deskargatu..." hedapena deskargatu eta instalatzeko.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	eu	Ezabatzeko aukerak				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	eu	Eguneratzeko aukerak				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	eu	Hartzen diren eremuak				2002-02-02 02:02:02
Index: l10n/source/zh-TW/localize.sdf
===================================================================
--- l10n/source/zh-TW/localize.sdf	(Revision 274294)
+++ l10n/source/zh-TW/localize.sdf	(Arbeitskopie)
@@ -1578,7 +1578,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	zh-TW	系統中找不到 Adabas D 資料來源。				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	zh-TW	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	zh-TW	下載(~D)...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	zh-TW	您需要有 Sun™ Report Builder 擴充軟體才能開啟報告。\n\n請按一下 [下載...] 以下載及安裝此擴充軟體。				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	zh-TW	刪除的選項				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	zh-TW	更新的選項				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	zh-TW	相關的欄位				2002-02-02 02:02:02
Index: l10n/source/mn/localize.sdf
===================================================================
--- l10n/source/mn/localize.sdf	(Revision 274294)
+++ l10n/source/mn/localize.sdf	(Arbeitskopie)
@@ -1397,7 +1397,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	mn	Таны системээс Adabas D өгөгдлийн эх сурвалж олдсонгүй.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	mn	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	mn	~Татах...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	mn	Тайлан нээхийн тулд танд %RPT_EXTENSION_NAME өргөтгөл шаардлагатай.\n\nТатаж авах болон өргөтгөлийг суулгахдаа 'Татах...' товшино уу.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	mn	 Сонголтуудыг устгах				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	mn	Шинэчлэх тохируулга				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	mn	Оролцсон талбарууд				2002-02-02 02:02:02
Index: l10n/source/it/localize.sdf
===================================================================
--- l10n/source/it/localize.sdf	(Revision 274294)
+++ l10n/source/it/localize.sdf	(Arbeitskopie)
@@ -1578,7 +1578,6 @@
 dbaccess	source\ui\dlg\ConnectionPage.src	0	string	STR_NO_ADABASE_DATASOURCES				0	it	Non è stata trovata nessuna sorgente dati Adabas D sul sistema.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	modaldialog	RID_EXTENSION_NOT_PRESENT_DLG		HID_EXTENSION_NOT_PRESENT_DLG		0	it	%PRODUCTNAME %PRODUCTVERSION				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	it	~Scarica...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	it	Per aprire un rapporto è necessaria l'estensione Sun™ Report Builder.\n\nFate clic su 'Scarica...' per scaricare e installare l'estensione.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	it	Opzioni di eliminazione				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_UPD			84	it	Opzioni di aggiornamento				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES.WND_CONTROL	FL_INVOLVED_FIELDS			174	it	Campi considerati				2002-02-02 02:02:02
