--- sal/rtl/source/bootstrap.cxx.old	2010-04-01 16:42:46.000000000 +0200
+++ sal/rtl/source/bootstrap.cxx	2010-04-14 17:07:25.000000000 +0200
@@ -281,9 +281,9 @@ static OUString & getIniFileName_Impl()
 			fileName += OUString(RTL_CONSTASCII_USTRINGPARAM(SAL_CONFIGFILE("")));
 		}
 
-		OUString workDir;
-		osl_getProcessWorkingDir(&workDir.pData);
-		osl::FileBase::getAbsoluteFileURL(workDir, fileName, fileName);
+        // do not use getAbsoluteFileURL because sofficerc might be linked from /usr/share (noarch branding package)
+        // the resolved symlink breaks other stuff because it searches other files in /usr/share then
+        osl::FileBase::getFileURLFromSystemPath(fileName, fileName);
 
 		static OUString theFileName;
 		if(fileName.getLength())
@@ -653,11 +653,10 @@ rtlBootstrapHandle SAL_CALL rtl_bootstra
 	rtl_uString * pIniName
 ) SAL_THROW_EXTERN_C()
 {
-	OUString workDir;
 	OUString iniName( pIniName );
-
-	osl_getProcessWorkingDir( &workDir.pData );
-	osl::FileBase::getAbsoluteFileURL( workDir, iniName, iniName );
+    // do not use getAbsoluteFileURL because sofficerc might be linked from /usr/share (noarch branding package)
+    // the resolved symlink breaks other stuff because it searches other files in /usr/share then
+    osl::FileBase::getFileURLFromSystemPath( iniName, iniName );
 
     // normalize path
     FileStatus status( FileStatusMask_FileURL );
