From 40e8835cc31c82cdfe3cf75fc23c5018ae35a3b5 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:53:14 +0200
Subject: [PATCH 028/768] sal-strintern-speed-fix.diff

---
 sal/rtl/source/hash.cxx |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git sal/rtl/source/hash.cxx sal/rtl/source/hash.cxx
index 60ba180..8e49e10 100644
--- sal/rtl/source/hash.cxx
+++ sal/rtl/source/hash.cxx
@@ -291,6 +291,9 @@ rtl_str_hash_remove (StringHashTable   *pHash,
     pHash->pData[n++] = NULL;
     pHash->nEntries--;
 
+    if (n >= pHash->nSize)
+        n = 0;
+
     while ((pHashStr = pHash->pData[n]) != NULL) {
         pHash->pData[n] = NULL;
         // FIXME: rather unsophisticated and N^2 in chain-length, but robust.
-- 
1.7.0.1

