From 5dadaf7e340124ba938d60f9835fa206c0a6e02d Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:53:50 +0200
Subject: [PATCH 057/768] sal-doublefree.diff

---
 sal/rtl/source/alloc_cache.c |   13 +++++++++++++
 1 files changed, 13 insertions(+), 0 deletions(-)

diff --git sal/rtl/source/alloc_cache.c sal/rtl/source/alloc_cache.c
index 51de21f..f7796f8 100644
--- sal/rtl/source/alloc_cache.c
+++ sal/rtl/source/alloc_cache.c
@@ -1268,6 +1268,19 @@ SAL_CALL rtl_cache_free (
             curr = cache->m_cpu_curr;
             if ((curr != 0) && (curr->m_mag_used < curr->m_mag_size))
             {
+#if OSL_DEBUG_LEVEL != 0
+                int i;
+                for (i = 0; i < curr->m_mag_used; ++i)
+                {
+                    OSL_ENSURE(curr->m_objects[i] != obj, "DOUBLE FREE!");
+                    if (curr->m_objects[i] == obj)
+                    {
+                        RTL_MEMORY_LOCK_RELEASE(&(cache->m_depot_lock));
+                        return;
+                    }
+
+                }
+#endif
                 curr->m_objects[curr->m_mag_used++] = obj;
                 cache->m_cpu_stats.m_free += 1;
                 RTL_MEMORY_LOCK_RELEASE(&(cache->m_depot_lock));
-- 
1.7.0.1

