From 6357a1d5b291335089222e17d59b7d73763bd357 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:57:58 +0200
Subject: [PATCH 291/878] sal-cpprt-gcc-4.5-workaround.diff

---
 sal/cpprt/makefile.mk |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/sal/cpprt/makefile.mk b/sal/cpprt/makefile.mk
index 3d7d672..8a98bbe 100644
--- a/sal/cpprt/makefile.mk
+++ b/sal/cpprt/makefile.mk
@@ -39,6 +39,11 @@ LIBTARGET=NO
 CFLAGS+= $(LFS_CFLAGS)
 CXXFLAGS+= $(LFS_CFLAGS)
 
+.IF "$(CCNUMVER)" >= "000400050000" && "$(CCNUMVER)" < "000400060000"
+CFLAGS+= -fno-ipa-sra
+CXXFLAGS+= -fno-ipa-sra
+.ENDIF
+
 # --- Files --------------------------------------------------------
 
 SLOFILES =	\
-- 
1.7.0.1

