From bfb546e61647f9f647ade14f5ce346c0402ee1f8 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:05:11 +0200
Subject: [PATCH 659/878] sal-disable-backtrace.diff

---
 sal/osl/unx/diagnose.c |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/sal/osl/unx/diagnose.c b/sal/osl/unx/diagnose.c
index bb8cbca..c7dde90 100644
--- a/sal/osl/unx/diagnose.c
+++ b/sal/osl/unx/diagnose.c
@@ -248,8 +248,9 @@ sal_Bool SAL_CALL osl_assertFailedLine (
     /* output message buffer */
     OSL_DIAGNOSE_OUTPUTMESSAGE(f, szMessage);
 
-    /* output backtrace */
-    osl_diagnose_backtrace_Impl(f);
+    /* if SAL backtrace for assertions is disabled, skip here */
+    if ( !getenv("DISABLE_SAL_BACKTRACE") )
+        osl_diagnose_backtrace_Impl(f);
 
     /* release lock and leave, w/o calling osl_breakDebug() */
     pthread_mutex_unlock(&g_mutex);
-- 
1.7.0.1

