From 9fac4a5c9f35e7ed6635b4db2231809cabea8e6c Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:57:18 +0200
Subject: [PATCH 226/768] sal-bootstrap-sofficerc-symlink.diff

---
 sal/rtl/source/bootstrap.cxx |   13 ++++++-------
 1 files changed, 6 insertions(+), 7 deletions(-)

diff --git sal/rtl/source/bootstrap.cxx sal/rtl/source/bootstrap.cxx
index a0bd0cc..39967ea 100644
--- sal/rtl/source/bootstrap.cxx
+++ sal/rtl/source/bootstrap.cxx
@@ -281,9 +281,9 @@ static OUString & getIniFileName_Impl()
             fileName += OUString(RTL_CONSTASCII_USTRINGPARAM(SAL_CONFIGFILE("")));
         }
 
-        OUString workDir;
-        osl_getProcessWorkingDir(&workDir.pData);
-        osl::FileBase::getAbsoluteFileURL(workDir, fileName, fileName);
+        // do not use getAbsoluteFileURL because sofficerc might be linked from /usr/share (noarch branding package)
+        // the resolved symlink breaks other stuff because it searches other files in /usr/share then
+        osl::FileBase::getFileURLFromSystemPath(fileName, fileName);
 
         static OUString theFileName;
         if(fileName.getLength())
@@ -653,11 +653,10 @@ rtlBootstrapHandle SAL_CALL rtl_bootstrap_args_open (
     rtl_uString * pIniName
 ) SAL_THROW_EXTERN_C()
 {
-    OUString workDir;
     OUString iniName( pIniName );
-
-    osl_getProcessWorkingDir( &workDir.pData );
-    osl::FileBase::getAbsoluteFileURL( workDir, iniName, iniName );
+    // do not use getAbsoluteFileURL because sofficerc might be linked from /usr/share (noarch branding package)
+    // the resolved symlink breaks other stuff because it searches other files in /usr/share then
+    osl::FileBase::getFileURLFromSystemPath( iniName, iniName );
 
     // normalize path
     FileStatus status( FileStatusMask_FileURL );
-- 
1.7.0.1

