---
 sal/osl/unx/diagnose.c |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git sal/osl/unx/diagnose.c sal/osl/unx/diagnose.c
index e98910d..13d977c 100644
--- sal/osl/unx/diagnose.c
+++ sal/osl/unx/diagnose.c
@@ -248,8 +248,9 @@ sal_Bool SAL_CALL osl_assertFailedLine (
     /* output message buffer */
     OSL_DIAGNOSE_OUTPUTMESSAGE(f, szMessage);
 
-    /* output backtrace */
-    osl_diagnose_backtrace_Impl(f);
+    /* if SAL backtrace for assertions is disabled, skip here */
+    if ( !getenv("DISABLE_SAL_BACKTRACE") )
+        osl_diagnose_backtrace_Impl(f);
 
     /* release lock and leave, w/o calling osl_breakDebug() */
     pthread_mutex_unlock(&g_mutex);
-- 
1.7.0.1

