Index: localize.sdf
===================================================================
--- l10n/source/es/localize.sdf	(Revision 275843)
+++ l10n/source/es/localize.sdf	(Arbeitskopie)
@@ -12663,7 +12663,7 @@
 sc	source\ui\src\scstring.src	0	string	SCSTR_NOTEMPTY				0	es	- no vacío -				2002-02-02 02:02:02
 sc	source\ui\src\scstring.src	0	string	SCSTR_NEWTABLE				0	es	- nueva hoja -				2002-02-02 02:02:02
 sc	source\ui\src\scstring.src	0	string	SCSTR_ALL				0	es	- todo -				2002-02-02 02:02:02
-sc	source\ui\src\scstring.src	0	string	SCSTR_STDFILTER				0	es	Filtro ~predeterminado...				2002-02-02 02:02:02
+sc	source\ui\src\scstring.src	0	string	SCSTR_STDFILTER				0	es	Filtro predeterminado...				2002-02-02 02:02:02
 sc	source\ui\src\scstring.src	0	string	SCSTR_TOP10FILTER				0	es	Los 10 primeros				2002-02-02 02:02:02
 sc	source\ui\src\scstring.src	0	string	SCSTR_NONAME				0	es	Sin nombre				2002-02-02 02:02:02
 sc	source\ui\src\scstring.src	0	string	SCSTR_COLUMN				0	es	Columna				2002-02-02 02:02:02
