Index: localize.sdf
===================================================================
--- l10n/source/es/localize.sdf	(Revision 275843)
+++ l10n/source/es/localize.sdf	(Arbeitskopie)
@@ -57074,7 +57074,7 @@
 sc	source\ui\src\scstring.src	0	string	SCSTR_SHORT_SCDOC_NAME				0	es	Hoja de cálculo				2002-02-02 02:02:02
 sc	source\ui\src\scstring.src	0	string	SCSTR_SHOWTABLE				0	es	Mostrar hoja				2002-02-02 02:02:02
 sc	source\ui\src\scstring.src	0	string	SCSTR_STAT_PRINT				0	es	Imprimiendo...				2002-02-02 02:02:02
-sc	source\ui\src\scstring.src	0	string	SCSTR_STDFILTER				0	es	Filtro ~predeterminado...				2002-02-02 02:02:02
+sc	source\ui\src\scstring.src	0	string	SCSTR_STDFILTER				0	es	Filtro predeterminado...				2002-02-02 02:02:02
 sc	source\ui\src\scstring.src	0	string	SCSTR_TABLE				0	es	Hoja				2002-02-02 02:02:02
 sc	source\ui\src\scstring.src	0	string	SCSTR_TABVIEWSHELL				0	es	Celdas				2002-02-02 02:02:02
 sc	source\ui\src\scstring.src	0	string	SCSTR_TOP10FILTER				0	es	Los 10 primeros				2002-02-02 02:02:02
