--- sc/source/ui/src/localize.sdf.old	2009-07-23 15:31:19.000000000 +0200
+++ sc/source/ui/src/localize.sdf	2009-07-27 16:11:21.000000000 +0200
@@ -238580,7 +238580,7 @@ sc	source\ui\src\scstring.src	0	string	S
 sc	source\ui\src\scstring.src	0	string	SCSTR_STDFILTER				13691	en-GB	Standard Filter...				2002-02-02 02:02:02
 sc	source\ui\src\scstring.src	0	string	SCSTR_STDFILTER				0	en-ZA	- Standard -				2002-02-02 02:02:02
 sc	source\ui\src\scstring.src	0	string	SCSTR_STDFILTER				0	eo	- Normala -				2002-02-02 02:02:02
-sc	source\ui\src\scstring.src	0	string	SCSTR_STDFILTER				0	es	Filtro ~predeterminado...				2002-02-02 02:02:02
+sc	source\ui\src\scstring.src	0	string	SCSTR_STDFILTER				0	es	Filtro predeterminado...				2002-02-02 02:02:02
 sc	source\ui\src\scstring.src	0	string	SCSTR_STDFILTER				13691	et	Standardfilter...				2002-02-02 02:02:02
 sc	source\ui\src\scstring.src	0	string	SCSTR_STDFILTER				13691	eu	Iragazki estandarra...				2002-02-02 02:02:02
 sc	source\ui\src\scstring.src	0	string	SCSTR_STDFILTER				0	fa	‐ استاندارد ‐				2002-02-02 02:02:02
