From 165782300ac432c967e4eb38cb74a2dc67c1d05b Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:55:23 +0200
Subject: [PATCH 155/878] sc-date-fix.diff

---
 svtools/source/numbers/zforfind.cxx |   18 ++++++++++++++++--
 1 files changed, 16 insertions(+), 2 deletions(-)

diff --git a/svtools/source/numbers/zforfind.cxx b/svtools/source/numbers/zforfind.cxx
index 11f845b..12188ff 100644
--- a/svtools/source/numbers/zforfind.cxx
+++ b/svtools/source/numbers/zforfind.cxx
@@ -1109,9 +1109,16 @@ input for the following reasons:
                         switch (DateFmt)
                         {
                             case MDY:
-                            case YMD:
-                                pCal->setValue( CalendarFieldIndex::DAY_OF_MONTH, ImplGetDay(0) );
+                            case YMD: {
+                                USHORT nDay = ImplGetDay(0);
+                                USHORT nYear = ImplGetYear(0);
+                                if (nDay == 0 || nDay > 32) {
+                                    pCal->setValue( CalendarFieldIndex::YEAR, nYear);
+                                }
+                                else
+                                    pCal->setValue( CalendarFieldIndex::DAY_OF_MONTH, ImplGetDay(0) );
                                 break;
+                            }
                             case DMY:
                                 pCal->setValue( CalendarFieldIndex::YEAR, ImplGetYear(0) );
                                 break;
@@ -1550,6 +1557,13 @@ BOOL ImpSvNumberInputScan::ScanStartString( const String& rString,
         }
     }
 
+    // skip any trailing '-' or '/' chars
+    if (nPos < rString.Len())
+    {
+        while (SkipChar ('-', rString, nPos) || SkipChar ('/', rString, nPos)) {
+            // do nothing
+        }
+    }
     if (nPos < rString.Len())                       // not everything consumed
     {
         // Does input StartString equal StartString of format?
-- 
1.7.0.1

