--- sc/source/ui/view/gridwin.cxx	2010-08-11 16:29:06.000000000 +0530
+++ sc/source/ui/view/gridwin.cxx	2010-08-11 16:41:56.000000000 +0530
@@ -2529,11 +2529,8 @@ long ScGridWindow::PreNotify( NotifyEven
                 nMouseStatus = SC_GM_IGNORE;
             }
         }
-
-        return 1;
     }
-    else
-        return Window::PreNotify( rNEvt );
+    return Window::PreNotify( rNEvt );
 }
 
 void ScGridWindow::Tracking( const TrackingEvent& rTEvt )
