--- sc/source/ui/inc/preview.hxx.old	2009-10-07 15:06:14.000000000 +0200
+++ sc/source/ui/inc/preview.hxx	2010-03-03 16:59:11.000000000 +0100
@@ -66,6 +66,8 @@ private:
     ScPreviewLocationData* pLocationData;	// stores table layout for accessibility API
     FmFormView*		pDrawView;
 
+    SCTAB           nCurTab;
+
                                         // intern:
     BOOL			bInPaint;
     BOOL			bInGetState;
--- sc/source/ui/view/preview.cxx.old	2009-10-08 16:17:09.000000000 +0200
+++ sc/source/ui/view/preview.cxx	2010-03-03 16:59:11.000000000 +0100
@@ -112,6 +112,7 @@ ScPreview::ScPreview( Window* pParent, S
     bLocationValid( FALSE ),
     pLocationData( NULL ),
     pDrawView( NULL ),
+    nCurTab ( ScDocShell::GetCurTab() ),
     bInPaint( FALSE ),
     bInGetState( FALSE ),
     pDocShell( pDocSh ),
@@ -252,8 +253,17 @@ void ScPreview::CalcPages( SCTAB /*nToWh
     //	but always all sheets are used (there is no selected sheet)
     ScPrintOptions aOptions = SC_MOD()->GetPrintOptions();
 
+	ScMarkData aMarkData;
+	aMarkData.SelectTable( nCurTab, TRUE );
+
     for (SCTAB i=nStart; i<nAnz; i++)
     {
+		if (!aOptions.GetAllSheets() && !aMarkData.GetTableSelect( i )) {
+			nPages[i] = 0;
+			nFirstAttr[i] = 0;
+			continue;
+		}
+
         long nAttrPage = i > 0 ? nFirstAttr[i-1] : 1;
 
         long nThisStart = nTotalPages;
--- sc/source/ui/view/tabvwsh4.cxx.old	2009-10-08 16:17:09.000000000 +0200
+++ sc/source/ui/view/tabvwsh4.cxx	2010-03-03 16:59:11.000000000 +0100
@@ -1194,8 +1194,13 @@ PrintDialog* __EXPORT ScTabViewShell::Cr
     // instead of a separate progress for each sheet from ScPrintFunc
     pDocShell->UpdatePendingRowHeights( MAXTAB, true );
 
+	ScMarkData aMarkData;
+	aMarkData.SelectTable( GetViewData()->GetTabNo(), TRUE );
+
     for ( SCTAB i=0; i<nTabCount; i++ )
     {
+		if ( !bAllTabs && !aMarkData.GetTableSelect( i ) )
+			continue;
         ScPrintFunc aPrintFunc( pDocShell, pPrinter, i );
         nDocPageMax += aPrintFunc.GetTotalPages();
     }
