diff --git sc/source/ui/view/gridwin.cxx sc/source/ui/view/gridwin.cxx
index a8a01e6..2500129 100644
--- sc/source/ui/view/gridwin.cxx
+++ sc/source/ui/view/gridwin.cxx
@@ -439,6 +439,10 @@ ScGridWindow::ScGridWindow( Window* pParent, ScViewData* pData, ScSplitPos eWhic
             bPagebreakDrawn( FALSE ),
             nPageScript( 0 ),
             bDragRect( FALSE ),
+            nDragStartX( -1 ),
+            nDragStartY( -1 ),
+            nDragEndX( -1 ),
+            nDragEndY( -1 ),
             meDragInsertMode( INS_NONE ),
             nCurrentPointer( 0 ),
             bIsInScroll( FALSE ),
