diff --git sc/source/filter/excel/excform.cxx sc/source/filter/excel/excform.cxx
index b1e05d0..f30eb80 100644
--- sc/source/filter/excel/excform.cxx
+++ sc/source/filter/excel/excform.cxx
@@ -904,7 +904,8 @@ ConvErr ExcelToSc::Convert( const ScTokenArray*& pErgebnis, XclImpStream& aIn, s
 
 
 // stream seeks to first byte after <nFormulaLen>
-ConvErr ExcelToSc::Convert( _ScRangeListTabs& rRangeList, XclImpStream& aIn, sal_Size nFormulaLen, const FORMULA_TYPE eFT )
+ConvErr ExcelToSc::Convert( _ScRangeListTabs& rRangeList, XclImpStream& aIn, sal_Size nFormulaLen,
+                            SCsTAB nTab, const FORMULA_TYPE eFT )
 {
     RootData&       rR = GetOldRoot();
 	BYTE			nOp, nLen;
@@ -1072,7 +1073,7 @@ ConvErr ExcelToSc::Convert( _ScRangeListTabs& rRangeList, XclImpStream& aIn, sal
 
 				ExcRelToScRel( nUINT16, nByte, aSRD, bRangeName );
 
-				rRangeList.Append( aSRD );
+                rRangeList.Append( aSRD, nTab );
 				break;
 			case 0x45:
 			case 0x65:
@@ -1099,7 +1100,7 @@ ConvErr ExcelToSc::Convert( _ScRangeListTabs& rRangeList, XclImpStream& aIn, sal
 				else if( IsComplRowRange( nRowFirst, nRowLast ) )
 					SetComplRow( aCRD );
 
-				rRangeList.Append( aCRD );
+                rRangeList.Append( aCRD, nTab );
 			}
 				break;
 			case 0x46:
@@ -1136,7 +1137,7 @@ ConvErr ExcelToSc::Convert( _ScRangeListTabs& rRangeList, XclImpStream& aIn, sal
 
 				ExcRelToScRel( nUINT16, nByte, aSRD, bRNorSF );
 
-				rRangeList.Append( aSRD );
+                rRangeList.Append( aSRD, nTab );
 			}
 				break;
 			case 0x4D:
@@ -1162,7 +1163,7 @@ ConvErr ExcelToSc::Convert( _ScRangeListTabs& rRangeList, XclImpStream& aIn, sal
 				else if( IsComplRowRange( nRowFirst, nRowLast ) )
 					SetComplRow( aCRD );
 
-				rRangeList.Append( aCRD );
+                rRangeList.Append( aCRD, nTab );
 			}
 				break;
             case 0x49:
@@ -1232,10 +1233,10 @@ ConvErr ExcelToSc::Convert( _ScRangeListTabs& rRangeList, XclImpStream& aIn, sal
 						b3D = ( static_cast<SCTAB>(nTabLast) != aEingPos.Tab() );
 						aCRD.Ref2.SetFlag3D( b3D );
 						aCRD.Ref2.SetTabRel( FALSE );
-						rRangeList.Append( aCRD );
+                        rRangeList.Append( aCRD, nTab );
 					}
 					else
-						rRangeList.Append( aSRD );
+                        rRangeList.Append( aSRD, nTab );
 				}
 			}
 
@@ -1290,7 +1291,7 @@ ConvErr ExcelToSc::Convert( _ScRangeListTabs& rRangeList, XclImpStream& aIn, sal
 					else if( IsComplRowRange( nRowFirst, nRowLast ) )
 						SetComplRow( aCRD );
 
-					rRangeList.Append( aCRD );
+                    rRangeList.Append( aCRD, nTab );
 				}//ENDE in aktuellem Workbook
 			}
 				break;
diff --git sc/source/filter/excel/excform8.cxx sc/source/filter/excel/excform8.cxx
index 1bf13ef..22da392 100644
--- sc/source/filter/excel/excform8.cxx
+++ sc/source/filter/excel/excform8.cxx
@@ -877,7 +877,8 @@ ConvErr ExcelToSc8::Convert( const ScTokenArray*& rpTokArray, XclImpStream& aIn,
 
 
 // stream seeks to first byte after <nFormulaLen>
-ConvErr ExcelToSc8::Convert( _ScRangeListTabs& rRangeList, XclImpStream& aIn, sal_Size nFormulaLen, const FORMULA_TYPE eFT )
+ConvErr ExcelToSc8::Convert( _ScRangeListTabs& rRangeList, XclImpStream& aIn, sal_Size nFormulaLen,
+                              SCsTAB nTab, const FORMULA_TYPE eFT )
 {
 	BYTE					nOp, nLen;//, nByte;
 	BOOL					bError = FALSE;
@@ -1008,7 +1009,7 @@ ConvErr ExcelToSc8::Convert( _ScRangeListTabs& rRangeList, XclImpStream& aIn, sa
 
                 ExcRelToScRel8( nRow, nCol, aSRD, bRangeName );
 
-				rRangeList.Append( aSRD );
+                rRangeList.Append( aSRD, nTab );
 			}
 				break;
 			case 0x45:
@@ -1036,7 +1037,7 @@ ConvErr ExcelToSc8::Convert( _ScRangeListTabs& rRangeList, XclImpStream& aIn, sa
 				else if( IsComplRowRange( nRowFirst, nRowLast ) )
 					SetComplRow( aCRD );
 
-				rRangeList.Append( aCRD );
+                rRangeList.Append( aCRD, nTab );
 			}
 				break;
 			case 0x46:
@@ -1080,7 +1081,7 @@ ConvErr ExcelToSc8::Convert( _ScRangeListTabs& rRangeList, XclImpStream& aIn, sa
 
                 ExcRelToScRel8( nRow, nCol, aSRD, bRNorSF );
 
-				rRangeList.Append( aSRD );
+                rRangeList.Append( aSRD, nTab );
 			}
 				break;
 			case 0x4D:
@@ -1106,7 +1107,7 @@ ConvErr ExcelToSc8::Convert( _ScRangeListTabs& rRangeList, XclImpStream& aIn, sa
 				else if( IsComplRowRange( nRowFirst, nRowLast ) )
 					SetComplRow( aCRD );
 
-				rRangeList.Append( aCRD );
+                rRangeList.Append( aCRD, nTab );
 			}
 				break;
 			case 0x4E:
@@ -1148,10 +1149,10 @@ ConvErr ExcelToSc8::Convert( _ScRangeListTabs& rRangeList, XclImpStream& aIn, sa
 						aCRD.Ref2.nCol = aSRD.nCol;
 						aCRD.Ref2.nRow = aSRD.nRow;
                         aCRD.Ref2.nTab = nLastScTab;
-						rRangeList.Append( aCRD );
+                        rRangeList.Append( aCRD, nTab );
 					}
 					else
-						rRangeList.Append( aSRD );
+                        rRangeList.Append( aSRD, nTab );
 				}
 			}
 				break;
@@ -1184,7 +1185,7 @@ ConvErr ExcelToSc8::Convert( _ScRangeListTabs& rRangeList, XclImpStream& aIn, sa
 					else if( IsComplRowRange( nRw1, nRw2 ) )
 						SetComplRow( aCRD );
 
-					rRangeList.Append( aCRD );
+                    rRangeList.Append( aCRD, nTab );
 				}
 			}
 				break;
diff --git sc/source/filter/excel/frmbase.cxx sc/source/filter/excel/frmbase.cxx
index ad4bf6b..aef072f 100644
--- sc/source/filter/excel/frmbase.cxx
+++ sc/source/filter/excel/frmbase.cxx
@@ -81,7 +81,7 @@ _ScRangeListTabs::~_ScRangeListTabs()
 }
 
 
-void _ScRangeListTabs::Append( ScSingleRefData a, const BOOL b )
+void _ScRangeListTabs::Append( ScSingleRefData a, SCsTAB nTab, const BOOL b )
 {
 	if( b )
 	{
@@ -101,19 +101,24 @@ void _ScRangeListTabs::Append( ScSingleRefData a, const BOOL b )
 
 	bHasRanges = TRUE;
 
-    if( a.nTab >= 0 )
+    if( nTab == SCTAB_MAX)
+        return;
+    if( nTab < 0)
+        nTab = a.nTab;
+
+    if( nTab >= 0 )
     {
-        _ScRangeList*   p = ppTabLists[ a.nTab ];
+        _ScRangeList*   p = ppTabLists[ nTab ];
 
         if( !p )
-            p = ppTabLists[ a.nTab ] = new _ScRangeList;
-
+            p = ppTabLists[ nTab ] = new _ScRangeList;
+  
         p->Append( a );
     }
 }
 
 
-void _ScRangeListTabs::Append( ScComplexRefData a, const BOOL b )
+void _ScRangeListTabs::Append( ScComplexRefData a, SCsTAB nTab, const BOOL b )
 {
 	if( b )
 	{
@@ -161,12 +166,17 @@ void _ScRangeListTabs::Append( ScComplexRefData a, const BOOL b )
 
 	bHasRanges = TRUE;
 
-    if( a.Ref1.nTab >= 0 )
+    if( nTab == SCTAB_MAX)
+        return;
+    if( nTab < -1)
+        nTab = a.Ref1.nTab;
+
+    if( nTab >= 0 )
     {
-        _ScRangeList*   p = ppTabLists[ a.Ref1.nTab ];
+        _ScRangeList*   p = ppTabLists[ nTab ];
 
         if( !p )
-            p = ppTabLists[ a.Ref1.nTab ] = new _ScRangeList;
+            p = ppTabLists[ nTab ] = new _ScRangeList;
 
         p->Append( a );
     }
diff --git sc/source/filter/excel/xiname.cxx sc/source/filter/excel/xiname.cxx
index 3647a51..72f4701 100644
--- sc/source/filter/excel/xiname.cxx
+++ sc/source/filter/excel/xiname.cxx
@@ -160,16 +160,18 @@ XclImpName::XclImpName( XclImpStream& rStrm, sal_uInt16 nXclNameIdx ) :
     }
     else if( bBuiltIn )
     {
+        SCsTAB const nLocalTab = (nXclTab == EXC_NAME_GLOBAL) ? SCTAB_MAX : (nXclTab - 1);
+
         // --- print ranges or title ranges ---
         rStrm.PushPosition();
         switch( mcBuiltIn )
         {
             case EXC_BUILTIN_PRINTAREA:
-                if( rFmlaConv.Convert( GetPrintAreaBuffer(), rStrm, nFmlaSize, FT_RangeName ) == ConvOK )
+                if( rFmlaConv.Convert( GetPrintAreaBuffer(), rStrm, nFmlaSize, nLocalTab, FT_RangeName ) == ConvOK )
                     nNameType |= RT_PRINTAREA;
             break;
             case EXC_BUILTIN_PRINTTITLES:
-                if( rFmlaConv.Convert( GetTitleAreaBuffer(), rStrm, nFmlaSize, FT_RangeName ) == ConvOK )
+                if( rFmlaConv.Convert( GetTitleAreaBuffer(), rStrm, nFmlaSize, nLocalTab, FT_RangeName ) == ConvOK )
                     nNameType |= RT_COLHEADER | RT_ROWHEADER;
             break;
         }
diff --git sc/source/filter/inc/excform.hxx sc/source/filter/inc/excform.hxx
index c587569..a067f9d 100644
--- sc/source/filter/inc/excform.hxx
+++ sc/source/filter/inc/excform.hxx
@@ -64,7 +64,7 @@ public:
     virtual ConvErr     Convert( const ScTokenArray*&, XclImpStream& rStrm, sal_Size nFormulaLen,
                                  bool bAllowArrays, const FORMULA_TYPE eFT = FT_CellFormula );
 
-    virtual ConvErr     Convert( _ScRangeListTabs&, XclImpStream& rStrm, sal_Size nFormulaLen, const FORMULA_TYPE eFT = FT_CellFormula );
+    virtual ConvErr     Convert( _ScRangeListTabs&, XclImpStream& rStrm, sal_Size nFormulaLen, SCsTAB nTab, const FORMULA_TYPE eFT = FT_CellFormula );
 
     virtual ConvErr     ConvertExternName( const ScTokenArray*& rpArray, XclImpStream& rStrm, sal_Size nFormulaLen,
                                            const String& rUrl, const ::std::vector<String>& rTabNames );
@@ -140,7 +140,7 @@ public:
 
     virtual ConvErr     Convert( const ScTokenArray*& rpTokArray, XclImpStream& rStrm, sal_Size nFormulaLen, bool bAllowArrays, const FORMULA_TYPE eFT = FT_CellFormula );
 
-    virtual ConvErr     Convert( _ScRangeListTabs&, XclImpStream& rStrm, sal_Size nFormulaLen, const FORMULA_TYPE eFT = FT_CellFormula );
+    virtual ConvErr     Convert( _ScRangeListTabs&, XclImpStream& rStrm, sal_Size nFormulaLen, SCsTAB nTab, const FORMULA_TYPE eFT = FT_CellFormula );
 
     virtual ConvErr     ConvertExternName( const ScTokenArray*& rpArray, XclImpStream& rStrm, sal_Size nFormulaLen,
                                            const String& rUrl, const ::std::vector<String>& rTabNames );
diff --git sc/source/filter/inc/formel.hxx sc/source/filter/inc/formel.hxx
index 9b5d524..c025393 100644
--- sc/source/filter/inc/formel.hxx
+++ sc/source/filter/inc/formel.hxx
@@ -154,8 +154,8 @@ public:
 								_ScRangeListTabs( void );
 	virtual						~_ScRangeListTabs();
 
-	void						Append( ScSingleRefData aSRD, const BOOL bLimit = TRUE );
-	void						Append( ScComplexRefData aCRD, const BOOL bLimit = TRUE );
+    void						Append( ScSingleRefData aSRD, SCsTAB nTab, const BOOL bLimit = TRUE );
+    void						Append( ScComplexRefData aCRD, SCsTAB nTab, const BOOL bLimit = TRUE );
 
 	inline BOOL					HasRanges( void ) const;
 
@@ -216,7 +216,7 @@ public:
 
     virtual ConvErr     Convert( const ScTokenArray*& rpErg, XclImpStream& rStrm, sal_Size nFormulaLen,
                                  bool bAllowArrays, const FORMULA_TYPE eFT = FT_CellFormula ) = 0;
-    virtual ConvErr     Convert( _ScRangeListTabs&, XclImpStream& rStrm, sal_Size nFormulaLen,
+    virtual ConvErr     Convert( _ScRangeListTabs&, XclImpStream& rStrm, sal_Size nFormulaLen, SCsTAB nTab,
 									const FORMULA_TYPE eFT = FT_CellFormula ) = 0;
 };
 
