--- scp2/source/extensions/file_extensions.scp.old	2010-08-31 21:22:37.000000000 +0200
+++ scp2/source/extensions/file_extensions.scp	2010-09-03 12:08:09.000000000 +0200
@@ -186,7 +186,14 @@ End
 #ifdef WITH_EXTENSION_GOOGLE_DOCS
 
 File gid_File_Oxt_Google_Docs
-    TXT_FILE_BODY;
+/*  FIXME: gdocs_2.1.0_modified.oxt pack twice two files;
+    it breaks installation with non-root user rights because
+    the file can't be overwriten with unix rights 444
+    => temporary using USER_FILE_BODY instead of TXT_FILE_BODY
+    to set 644 user rights; can be removed if we update
+    gdocs_*_modified.oxt
+*/
+    USER_FILE_BODY;
     Styles = (PACKED, ARCHIVE);
     Dir = gid_Brand_Dir_Share_Extensions_Google_Docs;
     Name = "gdocs.oxt";
