--- scp2/util/makefile.mk.old	2009-10-08 16:17:46.000000000 +0200
+++ scp2/util/makefile.mk	2010-03-03 18:13:36.000000000 +0100
@@ -77,6 +77,9 @@ SCP1FILES  = installation_ooo.par
              module_systemint.par          \
              module_graphicfilter.par      \
              file_graphicfilter.par        \
+             module_extensions.par         \
+             module_extensions_sun_templates.par         \
+             file_extensions.par           \
              module_testtool.par           \
              file_testtool.par             \
              module_xsltfilter.par         \
@@ -235,6 +238,9 @@ SCP2FILES  = installation_ooo.par
              module_systemint.par          \
              module_graphicfilter.par      \
              file_graphicfilter.par        \
+             module_extensions.par         \
+             module_extensions_sun_templates.par         \
+             file_extensions.par           \
              module_testtool.par           \
              file_testtool.par             \
              module_xsltfilter.par         \
