--- scp2/source/templates/makefile.mk
+++ scp2/source/templates/makefile.mk
@@ -44,13 +44,38 @@
 COMPLETELANGISO_VAR:=$(uniq $(completelangiso) $(alllangiso))
 .EXPORT : COMPLETELANGISO_VAR
 
-ALLTAR : $(INCCOM)$/alllangmodules.inc $(INCCOM)$/alllangmodules_root.inc $(INCCOM)$/alllangmodules_base.inc $(INCCOM)$/alllangmodules_calc.inc $(INCCOM)$/alllangmodules_draw.inc $(INCCOM)$/alllangmodules_impress.inc $(INCCOM)$/alllangmodules_math.inc $(INCCOM)$/alllangmodules_writer.inc $(INCCOM)$/alllangmodules_binfilter.inc
+ALLTAR : $(INCCOM)$/alllangmodules.inc $(INCCOM)$/alllangmodules_root.inc $(INCCOM)$/alllangmodules_base.inc $(INCCOM)$/alllangmodules_calc.inc $(INCCOM)$/alllangmodules_draw.inc $(INCCOM)$/alllangmodules_impress.inc $(INCCOM)$/alllangmodules_math.inc $(INCCOM)$/alllangmodules_writer.inc $(INCCOM)$/alllangmodules_binfilter.inc $(INCCOM)$/alllangmodules_accessories_templates_root.inc $(INCCOM)$/alllangmodules_accessories_samples_root.inc $(INCCOM)$/alllangmodules_accessories_templates.inc $(INCCOM)$/alllangmodules_accessories_samples.inc $(INCCOM)$/alllangmodules_extensions_root.inc $(INCCOM)$/alllangmodules_extensions_templates.inc
 
 .INCLUDE .IGNORE : $(MISC)$/$(TARGET)_lang_track.mk
 .IF "$(LAST_COMPLETELANGISO_VAR)"!="$(COMPLETELANGISO_VAR)"
 PHONYTEMPL=.PHONY
 .ENDIF			# "$(LAST_COMPLETELANGISO_VAR)"!="$(COMPLETELANGISO_VAR)"
 $(INCCOM)$/alllangmodules%.inc $(PHONYTEMPL) : module_langpack%.sct
 	@@-$(RENAME) $@ $@.tmp
 	$(PERL) -w modules.pl -i $< -o $@.tmp && $(RENAME:s/+//) $@.tmp $@
 	@echo LAST_COMPLETELANGISO_VAR=$(COMPLETELANGISO_VAR) > $(MISC)$/$(TARGET)_lang_track.mk
+
+.PHONY $(INCCOM)$/alllangmodules_accessories_templates_root.inc:
+	@@-$(RENAME) $@ $@.tmp
+	$(PERL) -w modules.pl -i $(PRJ)$/source$/templates$/module_langpack_accessories_templates_root.sct -o $@.tmp && $(RENAME:s/+//) $@.tmp $@
+
+.PHONY $(INCCOM)$/alllangmodules_accessories_samples_root.inc:
+	@@-$(RENAME) $@ $@.tmp
+	$(PERL) -w modules.pl -i $(PRJ)$/source$/templates$/module_langpack_accessories_samples_root.sct -o $@.tmp && $(RENAME:s/+//) $@.tmp $@
+
+.PHONY $(INCCOM)$/alllangmodules_accessories_templates.inc:
+	@@-$(RENAME) $@ $@.tmp
+	$(PERL) -w modules.pl -i $(PRJ)$/source$/templates$/module_langpack_accessories_templates.sct -o $@.tmp && $(RENAME:s/+//) $@.tmp $@
+
+.PHONY $(INCCOM)$/alllangmodules_accessories_samples.inc:
+	@@-$(RENAME) $@ $@.tmp
+	$(PERL) -w modules.pl -i $(PRJ)$/source$/templates$/module_langpack_accessories_samples.sct -o $@.tmp && $(RENAME:s/+//) $@.tmp $@
+
+.PHONY $(INCCOM)$/alllangmodules_extensions_root.inc:
+	@@-$(RENAME) $@ $@.tmp
+	$(PERL) -w modules.pl -i $(PRJ)$/source$/templates$/module_langpack_extensions_root.sct -o $@.tmp && $(RENAME:s/+//) $@.tmp $@
+
+.PHONY $(INCCOM)$/alllangmodules_extensions_templates.inc:
+	@@-$(RENAME) $@ $@.tmp
+	$(PERL) -w modules.pl -i $(PRJ)$/source$/templates$/module_langpack_extensions_templates.sct -o $@.tmp && $(RENAME:s/+//) $@.tmp $@
+
