diff --git a/sd/source/filter/ppt/pptatom.cpp b/sd/source/filter/ppt/pptatom.cpp
index d519898..3374b1c 100644
--- sd/source/filter/ppt/pptatom.cpp
+++ sd/source/filter/ppt/pptatom.cpp
@@ -52,7 +52,15 @@ Atom::Atom( const DffRecordHeader& rRecordHeader, SvStream& rStream )
 
             Atom* pLastAtom = NULL;
 		
-            while( (mrStream.GetError() == 0 ) && ( mrStream.Tell() < maRecordHeader.GetRecEndFilePos() ) )
+            // retrieve file size (to allow sanity checks) 
+            const sal_Size nStreamPos = mrStream.Tell();
+            mrStream.Seek( STREAM_SEEK_TO_END );
+            const sal_Size nStreamSize = mrStream.Tell();
+            mrStream.Seek( nStreamPos );
+
+            while( (mrStream.GetError() == 0 ) 
+                   && ( mrStream.Tell() < nStreamSize )
+                   && ( mrStream.Tell() < maRecordHeader.GetRecEndFilePos() ) )
             {
                 mrStream >> aChildHeader;
 
