commit 89fe9346a5292ddb1ac19ba734034b61e4b628b2
Author: Michael Meeks <michael.meeks@novell.com>
Date:   Tue Feb 22 11:23:46 2011 +0000

    fix fdo#34533 - occasional blank first slide in presentation

diff --git a/sd/source/ui/slideshow/slideshowviewimpl.cxx b/sd/source/ui/slideshow/slideshowviewimpl.cxx
index 2000ae0..fa293b2 100644
--- sd/source/ui/slideshow/slideshowviewimpl.cxx
+++ sd/source/ui/slideshow/slideshowviewimpl.cxx
@@ -500,7 +500,17 @@ void SlideShowView::updateimpl( ::osl::ClearableMutexGuard& rGuard, SlideshowImp
     if( pSlideShow )
     {
         ::rtl::Reference< SlideshowImpl > aSLGuard( pSlideShow );
-        rGuard.clear();
+
+        if( mbFirstPaint )
+        {
+            mbFirstPaint = false;
+            SlideshowImpl* pSlideShow = mpSlideShow;
+            rGuard.clear();
+            if( pSlideShow )
+                pSlideShow->onFirstPaint();
+        } else
+            rGuard.clear();
+
         pSlideShow->startUpdateTimer();
     }
 }
