---
 .../ui/animations/CustomAnimationCreateDialog.cxx  |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git sd/source/ui/animations/CustomAnimationCreateDialog.cxx sd/source/ui/animations/CustomAnimationCreateDialog.cxx
index 6d0d1d4..8774ca9 100755
--- sd/source/ui/animations/CustomAnimationCreateDialog.cxx
+++ sd/source/ui/animations/CustomAnimationCreateDialog.cxx
@@ -525,7 +525,9 @@ CustomAnimationCreateDialog::CustomAnimationCreateDialog( Window* pParent, Custo
 {
     mpTabControl = new TabControl( this, SdResId( 1 ) );
     mpOKButton = new OKButton(this, SdResId( 1 ) ) ;
+    mpOKButton->SetStyle(WB_DEFBUTTON);
     mpCancelButton = new CancelButton(this, SdResId( 1 ) );
+    mpCancelButton->SetStyle(WB_DEFBUTTON);
     mpHelpButton = new HelpButton(this, SdResId( 1 ) );
 
     FreeResource();
-- 
1.7.0.1

