--- ../ooo310-m19-orig/sd/source/filter/ppt/propread.cxx	2009-10-20 12:13:06.000000000 +0200
+++ sd/source/filter/ppt/propread.cxx	2009-10-20 12:04:46.000000000 +0200
@@ -471,11 +471,13 @@ void Section::Read( SvStorageStream *pSt
 						nPropSize += ( nTemp + 4 );
 					break;
 
-					case VT_LPWSTR :
+				    case VT_LPWSTR : {
 						*pStrm >> nTemp;
-						nPropSize += ( nTemp << 1 ) + 4;
+						// looks like these are aligned to 4 bytes
+						sal_uInt32 nLength = nPropOfs + nSecOfs + nPropSize + ( nTemp << 1 ) + 4;
+						nPropSize += ( nTemp << 1 ) + 4 + (nLength % 4);
 					break;
-
+				    }
 					case VT_BLOB_OBJECT :
 					case VT_BLOB :
 					case VT_CF :
@@ -504,6 +506,9 @@ void Section::Read( SvStorageStream *pSt
 			if ( nPropSize )
 			{
 				pStrm->Seek( nPropOfs + nSecOfs );
+				// make sure we don't overflow the section size
+				if( nPropSize > nSecSize - nSecOfs )
+				    nPropSize = nSecSize - nSecOfs;
 				sal_uInt8* pBuf = new sal_uInt8[ nPropSize ];
 				pStrm->Read( pBuf, nPropSize );
 				AddProperty( nPropId, pBuf, nPropSize );
