--- sd/source/filter/ppt/pptin.cxx.1	2010-03-24 17:27:21.000000000 +0100
+++ sd/source/filter/ppt/pptin.cxx	2010-03-24 17:27:32.000000000 +0100
@@ -1833,7 +1833,8 @@ void ImplSdPPTImport::ImportPageEffect(
                                 {	// Dia mit Soundeffekt
                                     pPage->SetSound( TRUE );
                                     sdr::media::MediaLinkRef aSoundFile( ReadSound( nSoundRef ) );
-                                    pPage->SetSoundFile( aSoundFile->getMediaURL() );
+				    if ( aSoundFile.is() )
+				        pPage->SetSoundFile( aSoundFile->getMediaURL() );
                                 }
                                 if ( nBuildFlags & ( 1 << 6 ) ) 	// Loop until next sound
                                     pPage->SetLoopSound( sal_True );
