diff -rup sd/source/ui/view-orig//PrintManager.cxx sd/source/ui/view/PrintManager.cxx
--- sd/source/ui/view-orig//PrintManager.cxx	2009-09-17 15:45:07.000000000 +0200
+++ sd/source/ui/view/PrintManager.cxx	2009-09-17 17:43:47.000000000 +0200
@@ -1610,15 +1610,18 @@ void PrintManager::PrintStdOrNotes (
                             }
                             else
                             {
+				Point aOrigin;
+
+				// if the printable area is bigger than page size, center the page
+				if( aPageSize.Width() < aPrintWidth )
+				    aOrigin.X() = ( aPrintWidth - aPageSize.Width() )/2;
+				if( aPageSize.Height() < aPrintHeight )
+				    aOrigin.Y() = ( aPrintHeight - aPageSize.Height() )/2;
+
                                 // Handle 3.  Print parts of the page in the
                                 // size of the printable area until the
                                 // whole page is covered.
 
-                                // keep the page content at its
-                                // position if it fits, otherwise move
-                                // it to the printable area
-                                Point aOrigin (aPageWidth < aPrintWidth ? -aPageOfs.X() : 0, aPageHeight < aPrintHeight ? -aPageOfs.Y() : 0);
-
                                 for (aPageOrigin = aOrigin;
                                      -aPageOrigin.Y()<aPageHeight;
                                          aPageOrigin.Y() -= aPrintHeight)
