diff -rup sd-orig/source/filter/pptx/pptx-eppt.cxx sd/source/filter/pptx/pptx-eppt.cxx
--- sd-orig/source/filter/pptx/pptx-eppt.cxx	2010-05-25 18:10:07.000000000 +0200
+++ sd/source/filter/pptx/pptx-eppt.cxx	2010-05-25 21:12:26.000000000 +0200
@@ -121,7 +121,7 @@ void PPTWriter::exportPPTPre()
     if ( !mpPicStrm )
         mpPicStrm = mrStg->OpenSotStream( String( RTL_CONSTASCII_USTRINGPARAM( "Pictures" ) ) );
 
-    mpPptEscherEx = new PptEscherEx( *mpStrm, mnDrawings );
+    mpPptEscherEx = new PptEscherEx( *mpStrm );
 };
 
 void PPTWriter::exportPPTPost()
diff -rup sd-orig/util/makefile.mk sd/util/makefile.mk
--- sd-orig/util/makefile.mk	2010-05-25 17:38:01.000000000 +0200
+++ sd/util/makefile.mk	2010-05-25 21:45:48.000000000 +0200
@@ -282,6 +282,7 @@ SHL6STDLIBS = $(ISDLIB) \
               $(SOTLIB)           \
               $(GOODIESLIB)       \
               $(VCLLIB)           \
+              $(MSFILTERLIB)      \
               $(SVLLIB)           \
               $(SOTLIB)           \
               $(UNOTOOLSLIB)      \
diff -rup sd-orig/util/pptx.map sd/util/pptx.map
--- sd-orig/util/pptx.map	2010-05-25 17:38:01.000000000 +0200
+++ sd/util/pptx.map	2010-05-25 21:41:41.000000000 +0200
@@ -1,4 +1,4 @@
-PPTX_1_0 {
+UDK_3_0_0 {
         global:
                 component_getImplementationEnvironment;
                 component_writeInfo;
