--- sdext/source/presenter/PresenterExtensionIdentifier.txx.old	2010-02-02 12:49:41.000000000 +0100
+++ sdext/source/presenter/PresenterExtensionIdentifier.txx	2010-03-09 14:05:36.000000000 +0100
@@ -38,7 +38,7 @@ namespace sdext { namespace presenter {
     makefile.mk with the actual, platform specific, extension identifier.
 */
 const ::rtl::OUString PresenterComponent::gsExtensionIdentifier(
-    ::rtl::OUString::createFromAscii("com.sun.PresenterScreen-UPDATED_PLATFORM"));
+    ::rtl::OUString::createFromAscii("com.sun.PresenterScreen"));
 
 } }
 
--- sdext/source/presenter/description.xml.old	2010-03-09 11:58:47.000000000 +0100
+++ sdext/source/presenter/description.xml	2010-03-09 14:05:36.000000000 +0100
@@ -5,7 +5,7 @@
   xmlns:dep="http://openoffice.org/extensions/description/2006"
   xmlns:xlink="http://www.w3.org/1999/xlink">
 
-  <identifier value="com.sun.PresenterScreen-UPDATED_PLATFORM" />
+  <identifier value="com.sun.PresenterScreen" />
 
   <dependencies>
     <OpenOffice.org-minimal-version value="3.0" dep:name="OpenOffice.org 3.0"/>
--- sdext/source/pdfimport/pdfisettings.pmk.old	2010-02-02 12:49:41.000000000 +0100
+++ sdext/source/pdfimport/pdfisettings.pmk	2010-03-09 14:05:36.000000000 +0100
@@ -29,4 +29,4 @@
 #
 #*************************************************************************
 
-IMPLEMENTATION_IDENTIFIER="com.sun.star.PDFImport-$(PLATFORMID)"
+IMPLEMENTATION_IDENTIFIER="com.sun.star.PDFImport"
