--- sdext/source/presenter/makefile.mk.old	2010-01-18 17:59:14.000000000 +0100
+++ sdext/source/presenter/makefile.mk	2010-01-20 13:21:06.000000000 +0100
@@ -41,14 +41,15 @@ ENABLE_EXCEPTIONS=TRUE
 
 .INCLUDE : rtlbootstrap.mk
 .INCLUDE : settings.mk
-.IF "$(L10N_framework)"==""
-.INCLUDE :  $(PRJ)$/util$/makefile.pmk
 
 .IF "$(ENABLE_PRESENTER_SCREEN)" == "NO"
 @all:
     @echo "Presenter Screen build disabled."
 .ELSE
 
+.IF "$(L10N_framework)"==""
+.INCLUDE :  $(PRJ)$/util$/makefile.pmk
+
 DLLPRE=
 common_build_zip=
 
@@ -349,9 +350,9 @@ $(DESCRIPTION) $(PHONYDESC) : $$(@:f)
     @@-$(RM) $(DESCRIPTION_TMP)
 
 
-.ENDIF # "$(ENABLE_PRESENTER_SCREEN)" != "NO"
 .ELSE
 ivo:
     $(ECHO)
 .ENDIF # L10N_framework
+.ENDIF # "$(ENABLE_PRESENTER_SCREEN)" != "NO"
 
