From c4be0a0a869e40b1e176d579b794ed4e381f3217 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:53:11 +0200
Subject: [PATCH 039/878] selection-crash-svx-svdmrkv.diff

---
 svx/source/svdraw/svdmrkv.cxx |   13 +++++++++++++
 1 files changed, 13 insertions(+), 0 deletions(-)

diff --git a/svx/source/svdraw/svdmrkv.cxx b/svx/source/svdraw/svdmrkv.cxx
index 0e0badb..c140a2e 100644
--- a/svx/source/svdraw/svdmrkv.cxx
+++ b/svx/source/svdraw/svdmrkv.cxx
@@ -227,6 +227,19 @@ void SdrMarkView::ModelHasChanged()
     GetMarkedObjectListWriteAccess().SetUnsorted();
     SortMarkedObjects();
     bMrkPntDirty=TRUE;
+    //remove empty maked SdrObject
+    ULONG nMarkAnz=GetMarkedObjectCount();
+    for (ULONG nMarkNum=nMarkAnz; nMarkNum>0;)
+    {
+        nMarkNum--;
+        SdrMark* pM=GetSdrMarkByIndex(nMarkNum);
+        const SdrObject* pObj=pM->GetMarkedSdrObj();
+        if( pObj == NULL )
+        {
+            GetMarkedObjectListWriteAccess().DeleteMark(nMarkNum);
+        }
+    }
+
     UndirtyMrkPnt();
     SdrView* pV=(SdrView*)this;
     if (pV!=NULL && !pV->IsDragObj() && !pV->IsInsObjPoint()) { // an dieser Stelle habe ich ein ziemliches Problem !!!
-- 
1.7.0.1

