---
 svx/source/svdraw/svdmrkv.cxx |   13 +++++++++++++
 1 files changed, 13 insertions(+), 0 deletions(-)

diff --git svx/source/svdraw/svdmrkv.cxx svx/source/svdraw/svdmrkv.cxx
index 3d72ef2..e2008cf 100644
--- svx/source/svdraw/svdmrkv.cxx
+++ svx/source/svdraw/svdmrkv.cxx
@@ -227,6 +227,19 @@ void SdrMarkView::ModelHasChanged()
     GetMarkedObjectListWriteAccess().SetUnsorted();
     SortMarkedObjects();
     bMrkPntDirty=TRUE;
+    //remove empty maked SdrObject
+    ULONG nMarkAnz=GetMarkedObjectCount();
+    for (ULONG nMarkNum=nMarkAnz; nMarkNum>0;)
+    {
+        nMarkNum--;
+        SdrMark* pM=GetSdrMarkByIndex(nMarkNum);
+        const SdrObject* pObj=pM->GetMarkedSdrObj();
+        if( pObj == NULL )
+        {
+            GetMarkedObjectListWriteAccess().DeleteMark(nMarkNum);
+        }
+    }
+
     UndirtyMrkPnt();
     SdrView* pV=(SdrView*)this;
     if (pV!=NULL && !pV->IsDragObj() && !pV->IsInsObjPoint()) { // an dieser Stelle habe ich ein ziemliches Problem !!!
-- 
1.7.0.1

