diff -rup sfx2-orig/inc/sfx2/objsh.hxx sfx2/inc/sfx2/objsh.hxx
--- sfx2-orig/inc/sfx2/objsh.hxx	2010-03-04 11:39:35.000000000 +0100
+++ sfx2/inc/sfx2/objsh.hxx	2010-03-04 15:27:38.000000000 +0100
@@ -257,6 +257,9 @@ protected:
     /// template method, called by FlushDocInfo; this implementation is empty
     virtual void                DoFlushDocInfo();
 
+    // helper method
+    void AddToRecentlyUsedList();
+
 public:
                                 TYPEINFO();
                                 SFX_DECL_INTERFACE(SFX_INTERFACE_SFXDOCSH)
diff -rup sfx2-orig/source/doc/objstor.cxx sfx2/source/doc/objstor.cxx
--- sfx2-orig/source/doc/objstor.cxx	2010-03-04 11:39:31.000000000 +0100
+++ sfx2/source/doc/objstor.cxx	2010-03-04 15:30:43.000000000 +0100
@@ -1010,15 +1010,7 @@ sal_Bool SfxObjectShell::DoLoad( SfxMedi
             && !( pPreviewItem && pPreviewItem->GetValue() )
             && !( pHiddenItem && pHiddenItem->GetValue() ) )
             {
-                INetURLObject aUrl( pMedium->GetOrigURL() );
-
-                if ( aUrl.GetProtocol() == INET_PROT_FILE )
-                {
-                    const SfxFilter* pOrgFilter = pMedium->GetOrigFilter();
-                    Application::AddToRecentDocumentList(
-                        aUrl.GetURLNoPass( INetURLObject::NO_DECODE ),
-                        (pOrgFilter) ? pOrgFilter->GetMimeType() : String() );
-                }
+	        AddToRecentlyUsedList();
             }
         }
 
@@ -2413,9 +2405,23 @@ sal_Bool SfxObjectShell::DoSaveCompleted
     pMedium->ClearBackup_Impl();
     pMedium->LockOrigFileOnDemand( sal_True, sal_False );
 
+    AddToRecentlyUsedList();
+
     return bOk;
 }
 
+void SfxObjectShell::AddToRecentlyUsedList()
+{
+    INetURLObject aUrl( pMedium->GetOrigURL() );
+
+    if ( aUrl.GetProtocol() == INET_PROT_FILE )
+    {
+	const SfxFilter* pOrgFilter = pMedium->GetOrigFilter();
+	Application::AddToRecentDocumentList( aUrl.GetURLNoPass( INetURLObject::NO_DECODE ),
+					      (pOrgFilter) ? pOrgFilter->GetMimeType() : String() );
+    }
+}
+
 //-------------------------------------------------------------------------
 
 sal_Bool SfxObjectShell::ConvertFrom
