---
 sfx2/source/doc/objstor.cxx |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git sfx2/source/doc/objstor.cxx sfx2/source/doc/objstor.cxx
index 902793a..a2c1047 100644
--- sfx2/source/doc/objstor.cxx
+++ sfx2/source/doc/objstor.cxx
@@ -789,6 +789,7 @@ sal_Bool SfxObjectShell::DoLoad( SfxMedium *pMed )
             }
         }
 
+#if 0
         if ( pMedium->HasStorage_Impl() )
         {
             uno::Reference< XInteractionHandler > xHandler( pMedium->GetInteractionHandler() );
@@ -842,6 +843,7 @@ sal_Bool SfxObjectShell::DoLoad( SfxMedium *pMed )
                 }
             }
         }
+#endif
     }
     else
         GetpApp()->HideStatusText();
-- 
1.7.0.1

