From 8a99d16c71def6587e383c311c3b6785398a8e02 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:57:21 +0200
Subject: [PATCH 229/768] sfx2-remove-check-update-on-fileload.diff

---
 sfx2/source/doc/objstor.cxx |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git sfx2/source/doc/objstor.cxx sfx2/source/doc/objstor.cxx
index 902793a..a2c1047 100644
--- sfx2/source/doc/objstor.cxx
+++ sfx2/source/doc/objstor.cxx
@@ -789,6 +789,7 @@ sal_Bool SfxObjectShell::DoLoad( SfxMedium *pMed )
             }
         }
 
+#if 0
         if ( pMedium->HasStorage_Impl() )
         {
             uno::Reference< XInteractionHandler > xHandler( pMedium->GetInteractionHandler() );
@@ -842,6 +843,7 @@ sal_Bool SfxObjectShell::DoLoad( SfxMedium *pMed )
                 }
             }
         }
+#endif
     }
     else
         GetpApp()->HideStatusText();
-- 
1.7.0.1

