From f9e2dd3475cb642fe099000870885ac65f1c7a82 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:57:21 +0200
Subject: [PATCH 258/878] sfx2-remove-check-update-on-fileload.diff

---
 sfx2/source/doc/objstor.cxx |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/sfx2/source/doc/objstor.cxx b/sfx2/source/doc/objstor.cxx
index 0508df6..9950a8d 100644
--- a/sfx2/source/doc/objstor.cxx
+++ b/sfx2/source/doc/objstor.cxx
@@ -834,6 +834,7 @@ sal_Bool SfxObjectShell::DoLoad( SfxMedium *pMed )
             }
         }
 
+#if 0
         if ( pMedium->HasStorage_Impl() )
         {
             uno::Reference< XInteractionHandler > xHandler( pMedium->GetInteractionHandler() );
@@ -887,6 +888,7 @@ sal_Bool SfxObjectShell::DoLoad( SfxMedium *pMed )
                 }
             }
         }
+#endif
     }
     else
         GetpApp()->HideStatusText();
-- 
1.7.0.1

