From 6d622a7731181a94c46cb2b47e483be5d6fd6531 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:05:28 +0200
Subject: [PATCH 593/768] sfx2-privateprotectedpublic.diff

---
 sfx2/inc/sfx2/tabdlg.hxx |    8 ++++++--
 1 files changed, 6 insertions(+), 2 deletions(-)

diff --git sfx2/inc/sfx2/tabdlg.hxx sfx2/inc/sfx2/tabdlg.hxx
index 6f84e0c..55d0199 100644
--- sfx2/inc/sfx2/tabdlg.hxx
+++ sfx2/inc/sfx2/tabdlg.hxx
@@ -92,8 +92,9 @@ friend class SfxTabPage;
 friend class SfxTabDialogController;
 
     SfxViewFrame*	pFrame;
-
+public:
     TabControl		aTabCtrl;
+private:
     OKButton		aOKBtn;
     PushButton*		pUserBtn;
     CancelButton	aCancelBtn;
@@ -101,7 +102,9 @@ friend class SfxTabDialogController;
     PushButton		aResetBtn;
     PushButton		aBaseFmtBtn;
 
+public:
     const SfxItemSet*	pSet;
+private:
     SfxItemSet*			pOutSet;
     TabDlg_Impl*		pImpl;
     USHORT*				pRanges;
@@ -121,9 +124,10 @@ friend class SfxTabDialogController;
     SAL_DLLPRIVATE void Init_Impl(BOOL, const String *);
 //#endif
 
-protected:
+public:
     virtual short				Ok();
     // wird im Sfx gel"oscht!
+protected:
     virtual SfxItemSet*			CreateInputItemSet( USHORT nId );
     // wird *nicht* im Sfx gel"oscht!
     virtual const SfxItemSet*	GetRefreshedSet();
-- 
1.7.0.1

