From d3132cec18e67d075fdbe31d0a919af35a68b0e1 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:54:31 +0200
Subject: [PATCH 109/878] sfx2-check-existing-template-dirs.diff

---
 sfx2/source/doc/doctemplates.cxx |   12 +++++++++---
 1 files changed, 9 insertions(+), 3 deletions(-)

diff --git a/sfx2/source/doc/doctemplates.cxx b/sfx2/source/doc/doctemplates.cxx
index 317d2a4..71e465d 100644
--- a/sfx2/source/doc/doctemplates.cxx
+++ b/sfx2/source/doc/doctemplates.cxx
@@ -34,6 +34,7 @@
 #include <tools/urlobj.hxx>
 #include <rtl/ustring.hxx>
 #include <rtl/ustrbuf.hxx>
+#include <osl/file.hxx>
 #ifndef _SV_RESARY_HXX
 #include <tools/resary.hxx>
 #endif
@@ -1226,11 +1227,16 @@ void SfxDocTplService_Impl::doUpdate()
     while ( nCountDir )
     {
         nCountDir--;
-        if ( Content::create( pDirs[ nCountDir ], aQuietEnv, aDirContent ) )
+        osl::Directory aDirectory (pDirs[ nCountDir ]);
+        osl::FileBase::RC nError = aDirectory.open();
+        if (nError == osl::FileBase::E_None || bWriteableDirectory)
         {
-            createFromContent( aGroupList, aDirContent, sal_False, bWriteableDirectory );
+            if ( Content::create( pDirs[ nCountDir ], aQuietEnv, aDirContent ) )
+            {
+                createFromContent( aGroupList, aDirContent, sal_False, bWriteableDirectory );
+            }
+            aDirectory.close();
         }
-
         bWriteableDirectory = sal_False;
     }
 
-- 
1.7.0.1

