diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sfx2.vpj uui/source/newerverwarn.cxx uui/source/newerverwarn.cxx
--- uui/source/newerverwarn.cxx	2008-05-06 23:56:41.000000000 -0400
+++ uui/source/newerverwarn.cxx	2008-05-07 00:00:36.000000000 -0400
@@ -58,6 +58,7 @@ namespace lang  = ::com::sun::star::lang
 
 #define DEFINE_CONST_UNICODE( CONSTASCII )  ::rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( CONSTASCII ) )
 
+#if 0
 namespace uui
 {
 
@@ -160,4 +161,5 @@ void NewerVersionWarningDialog::InitButt
 }
 
 } // end of namespace uui
+#endif
 
diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sfx2.vpj sfx2.clean/source/doc/objstor.cxx sfx2/source/doc/objstor.cxx
--- sfx2.clean/source/doc/objstor.cxx	2008-05-06 23:56:40.000000000 -0400
+++ sfx2/source/doc/objstor.cxx	2008-05-06 23:59:24.000000000 -0400
@@ -1082,6 +1082,7 @@ sal_Bool SfxObjectShell::DoLoad( SfxMedi
             }
         }
 
+#if 0
         uno::Reference< XInteractionHandler > xHandler( pMedium->GetInteractionHandler() );
         if ( xHandler.is() && !SFX_APP()->Get_Impl()->bODFVersionWarningLater )
         {
@@ -1125,6 +1126,7 @@ sal_Bool SfxObjectShell::DoLoad( SfxMedi
                 }
             }
         }
+#endif
     }
     else
         GetpApp()->HideStatusText();
diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sfx2.vpj uui/source/newerverwarn.hxx uui/source/newerverwarn.hxx
--- uui/source/newerverwarn.hxx	2008-05-06 23:56:41.000000000 -0400
+++ uui/source/newerverwarn.hxx	2008-05-06 23:59:58.000000000 -0400
@@ -39,6 +39,7 @@
 
 #define RET_ASK_LATER   short( 100 )
 
+#if 0
 namespace uui
 {
     class NewerVersionWarningDialog : public SfxModalDialog
@@ -60,6 +61,7 @@ namespace sfx2
         ~NewerVersionWarningDialog();
     };
 } // namespace uui
+#endif
 
 #endif // #ifndef _NEWERVERSIONWARNING_HXX
 
