From 2f8b02cd46c3e3b5963d5b9494a6270ea7515491 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:58:19 +0200
Subject: [PATCH 310/878] sfx2-template-category-labels.diff

---
 sfx2/source/doc/doctempl.src |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/sfx2/source/doc/doctempl.src b/sfx2/source/doc/doctempl.src
index 2c91358..e71e77f 100644
--- a/sfx2/source/doc/doctempl.src
+++ b/sfx2/source/doc/doctempl.src
@@ -43,6 +43,7 @@ StringArray TEMPLATE_SHORT_NAMES_ARY
         < "layout" ; > ;
         < "presnt" ; > ;
         < "misc" ; > ;
+        < "labels" ; > ;
     };
 };
 
@@ -60,6 +61,7 @@ StringArray TEMPLATE_LONG_NAMES_ARY
         < "Presentation Backgrounds" ; > ;
         < "Presentations" ; > ;
         < "Miscellaneous" ; > ;
+        < "Labels" ; > ;
     };
 
     };
-- 
1.7.0.1

