--- sfx2/inc/sfx2/tabdlg.hxx
+++ sfx2/inc/sfx2/tabdlg.hxx
@@ -92,8 +92,9 @@ friend class SfxTabPage;
 friend class SfxTabDialogController;
 
     SfxViewFrame*	pFrame;
-
+public:
     TabControl		aTabCtrl;
+private:
     OKButton		aOKBtn;
     PushButton*		pUserBtn;
     CancelButton	aCancelBtn;
@@ -101,7 +102,9 @@ friend class SfxTabDialogController;
     PushButton		aResetBtn;
     PushButton		aBaseFmtBtn;
 
+public:
     const SfxItemSet*	pSet;
+private:
     SfxItemSet*			pOutSet;
     TabDlg_Impl*		pImpl;
     sal_uInt16*				pRanges;
@@ -119,9 +119,10 @@
     DECL_DLLPRIVATE_LINK( CancelHdl, Button * );
     SAL_DLLPRIVATE void Init_Impl(sal_Bool, const String *);
 
-protected:
+public:
     virtual short				Ok();
     // wird im Sfx gel"oscht!
+protected:
     virtual SfxItemSet*			CreateInputItemSet( sal_uInt16 nId );
     // wird *nicht* im Sfx gel"oscht!
     virtual const SfxItemSet*	GetRefreshedSet();
