--- instsetoo_native/util/openoffice.lst.old	2010-04-13 17:48:03.000000000 +0200
+++ instsetoo_native/util/openoffice.lst	2010-04-14 17:29:11.000000000 +0200
@@ -25,10 +25,10 @@ Globals
 			POOLPRODUCT 1
 			REGISTRATION_HOST https://registration2.services.openoffice.org/RegistrationWeb
 			REGISTRATIONURL http://survey.services.openoffice.org/user/index.php
-			PROGRESSBARCOLOR 14,133,205
-			PROGRESSSIZE 320,7
-			PROGRESSPOSITION 185,206
-			PROGRESSFRAMECOLOR 207,208,211
+            PROGRESSBARCOLOR 4,68,155
+            PROGRESSSIZE 412,7
+            PROGRESSPOSITION 14,230
+            PROGRESSFRAMECOLOR 58,159,232
 			NATIVEPROGRESS true
 			REGISTRYLAYERNAME Layers
 			SERVICEPACK 1
