--- sdext/source/presenter/PresenterController.cxx.old	2010-01-12 11:14:53.000000000 +0100
+++ sdext/source/presenter/PresenterController.cxx	2010-03-03 18:23:48.000000000 +0100
@@ -1002,14 +1002,23 @@ void SAL_CALL PresenterController::keyRe
             }
             break;
 
-        case awt::Key::LEFT:
         case awt::Key::PAGEUP:
+            if (mxSlideShowController.is())
+            {
+                if (rEvent.Modifiers == awt::KeyModifier::MOD2)
+                    mxSlideShowController->gotoPreviousSlide();
+                else
+                    mxSlideShowController->gotoPreviousEffect();
+            }
+            break;
+
+        case awt::Key::LEFT:
         case awt::Key::UP:
         case awt::Key::P:
         case awt::Key::BACKSPACE:
             if (mxSlideShowController.is())
             {
-                mxSlideShowController->gotoPreviousSlide();
+                mxSlideShowController->gotoPreviousEffect();
             }
             break;
 
--- sdext/source/presenter/PresenterCurrentSlideObserver.cxx.old	2009-10-07 15:07:29.000000000 +0200
+++ sdext/source/presenter/PresenterCurrentSlideObserver.cxx	2010-03-03 18:23:48.000000000 +0100
@@ -109,13 +109,13 @@ void SAL_CALL PresenterCurrentSlideObser
 {
 }
 
-void SAL_CALL PresenterCurrentSlideObserver::slideEnded (void)
+void SAL_CALL PresenterCurrentSlideObserver::slideEnded (sal_Bool bReverse)
     throw (css::uno::RuntimeException)
 {
     // Determine whether the new current slide (the one after the one that
     // just ended) is the slide past the last slide in the presentation,
     // i.e. the one that says something like "click to end presentation...".
-    if (mxSlideShowController.is())
+    if (mxSlideShowController.is() && !bReverse)
         if (mxSlideShowController->getNextSlideIndex() < 0)
             if( mpPresenterController.is() )
                 mpPresenterController->UpdateCurrentSlide(+1);
--- sdext/source/presenter/PresenterCurrentSlideObserver.hxx.old	2009-10-07 15:07:29.000000000 +0200
+++ sdext/source/presenter/PresenterCurrentSlideObserver.hxx	2010-03-03 18:23:48.000000000 +0100
@@ -76,7 +76,7 @@ public:
     virtual void SAL_CALL slideTransitionStarted(  ) throw (::com::sun::star::uno::RuntimeException);
     virtual void SAL_CALL slideTransitionEnded(  ) throw (::com::sun::star::uno::RuntimeException);
     virtual void SAL_CALL slideAnimationsEnded(  ) throw (::com::sun::star::uno::RuntimeException);
-    virtual void SAL_CALL slideEnded(  ) throw (::com::sun::star::uno::RuntimeException);
+    virtual void SAL_CALL slideEnded(sal_Bool bReverse) throw (::com::sun::star::uno::RuntimeException);
     virtual void SAL_CALL hyperLinkClicked( const ::rtl::OUString& hyperLink ) throw (::com::sun::star::uno::RuntimeException);
 
     // XAnimationListener
--- sdext/source/presenter/description.xml.old	2010-03-03 18:22:50.000000000 +0100
+++ sdext/source/presenter/description.xml	2010-03-03 18:23:48.000000000 +0100
@@ -8,10 +8,10 @@
   <identifier value="com.sun.PresenterScreen-UPDATED_PLATFORM" />
 
   <dependencies>
-    <OpenOffice.org-minimal-version value="3.0" dep:name="OpenOffice.org 3.0"/>
+    <OpenOffice.org-minimal-version value="3.1" dep:name="OpenOffice.org 3.2"/>
   </dependencies>
 
-  <version value="1.0.2" />
+  <version value="1.1.0" />
 
   <platform value="UPDATED_PLATFORM" />
 
