---
 sdext/source/presenter/PresenterController.cxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git sdext/source/presenter/PresenterController.cxx sdext/source/presenter/PresenterController.cxx
index cf3ec87..db7d0f9 100644
--- sdext/source/presenter/PresenterController.cxx
+++ sdext/source/presenter/PresenterController.cxx
@@ -998,7 +998,6 @@ void SAL_CALL PresenterController::keyReleased (const awt::KeyEvent& rEvent)
             }
             break;
 
-        case awt::Key::LEFT:
         case awt::Key::PAGEUP:
             if (mxSlideShowController.is())
             {
@@ -1009,6 +1008,7 @@ void SAL_CALL PresenterController::keyReleased (const awt::KeyEvent& rEvent)
             }
             break;
 
+        case awt::Key::LEFT:
         case awt::Key::UP:
         case awt::Key::P:
         case awt::Key::BACKSPACE:
-- 
1.7.0.1

