From 284e94e2eb3aa132a0460d188ffa94545e5e2a86 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:54:07 +0200
Subject: [PATCH 072/768] slideshow-effect-rewind-sdext.diff

---
 sdext/source/presenter/PresenterController.cxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git sdext/source/presenter/PresenterController.cxx sdext/source/presenter/PresenterController.cxx
index cf3ec87..db7d0f9 100644
--- sdext/source/presenter/PresenterController.cxx
+++ sdext/source/presenter/PresenterController.cxx
@@ -998,7 +998,6 @@ void SAL_CALL PresenterController::keyReleased (const awt::KeyEvent& rEvent)
             }
             break;
 
-        case awt::Key::LEFT:
         case awt::Key::PAGEUP:
             if (mxSlideShowController.is())
             {
@@ -1009,6 +1008,7 @@ void SAL_CALL PresenterController::keyReleased (const awt::KeyEvent& rEvent)
             }
             break;
 
+        case awt::Key::LEFT:
         case awt::Key::UP:
         case awt::Key::P:
         case awt::Key::BACKSPACE:
-- 
1.7.0.1

