diff --git test/source/cpp/officeconnection.cxx test/source/cpp/officeconnection.cxx
index 96f735a..9222c3a 100644
--- test/source/cpp/officeconnection.cxx
+++ test/source/cpp/officeconnection.cxx
@@ -108,6 +108,10 @@ void OfficeConnection::setUp() {
                     argSoffice.copy(RTL_CONSTASCII_LENGTH("path:"))).pData,
                 args, sizeof args / sizeof args[0], 0, 0, 0, envs,
                 envs == 0 ? 0 : 1, &process_));
+        // ooo-build soffice.bin doesn't block
+        // desc string is already set up so below will try to connect
+        // using the existing pipe
+        process_ = 0; // don't join
     } else if (argSoffice.matchAsciiL(RTL_CONSTASCII_STRINGPARAM("connect:"))) {
         desc = argSoffice.copy(RTL_CONSTASCII_LENGTH("connect:"));
     } else {
