--- solenv/inc/settings.mk.old	2010-07-21 19:58:19.000000000 +0200
+++ solenv/inc/settings.mk	2010-07-23 16:03:29.000000000 +0200
@@ -835,6 +835,9 @@ LOCALIZESDF:=$(strip $(shell @+$(IFEXIST
 # dummy target to keep the build happy.
 .IF  "$(LOCALIZESDF)"==""
 LOCALIZESDF:=$(COMMONMISC)$/$(PRJNAME)$/dummy$/localize.sdf
+DUMMYLOCALIZESDF:="YES"
+.ELSE
+DUMMYLOCALIZESDF:="NO"
 .ENDIF			# "$(LOCALIZESDF)"==""
 .ENDIF			# "$(WITH_LANG)"!=""
 
--- solenv/inc/target.mk.old	2010-07-23 16:06:29.000000000 +0200
+++ solenv/inc/target.mk	2010-07-23 16:06:04.000000000 +0200
@@ -1390,7 +1390,7 @@ $(UNIXTEXT) : $(UNIXTEXT:f)
     $(TOUCH) $@
 
 .IF "$(LOCALIZATION_FOUND)"==""
-.IF "$(LOCALSDFFILE)"!=""
+.IF "$(LOCALSDFFILE)"!="" && "$(DUMMYLOCALIZESDF)"!="YES"
 "$(LOCALIZESDF)" : $(SOLARCOMMONSDFDIR)/$(PRJNAME).zip
     @@-$(MKDIRHIER) $(@:d)
     @@-$(MKDIRHIER) $(COMMONMISC)/$(PRJNAME)_$(TARGET)
