From 3b268bcecf76bad5925df320ba784c9294dc03e4 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:57:27 +0200
Subject: [PATCH 263/878] solenv-really-use-unsinged-char-on-ppc.diff

---
 solenv/inc/unxlngppc.mk |    3 ---
 1 files changed, 0 insertions(+), 3 deletions(-)

diff --git a/solenv/inc/unxlngppc.mk b/solenv/inc/unxlngppc.mk
index fa582e8..0f0fd3e 100644
--- a/solenv/inc/unxlngppc.mk
+++ b/solenv/inc/unxlngppc.mk
@@ -31,7 +31,4 @@ DEFAULTOPT=-Os
 PICSWITCH:=-fPIC
 .INCLUDE : unxlng.mk
 CDEFS+=-DPOWERPC -DPPC
-CFLAGS+=-fsigned-char
-CFLAGSCC+=-fsigned-char
-CFLAGSCXX+=-fsigned-char
 DLLPOSTFIX=lp
-- 
1.7.0.1

