From c6517f3635e21efd0a7bf48c6eae49640973556b Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:07:31 +0200
Subject: [PATCH 674/768] solenv-unpack-oxt.diff

---
 solenv/inc/tg_ext.mk |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git solenv/inc/tg_ext.mk solenv/inc/tg_ext.mk
index e19c712..b8f7ac2 100644
--- solenv/inc/tg_ext.mk
+++ solenv/inc/tg_ext.mk
@@ -151,6 +151,12 @@ $(MISC)/%.unpack : $(TARFILE_LOCATION2)/%.zip
     @$(TYPE) $(mktmp $(UNPACKCMD)) > $@.$(INPATH)
     @$(RENAME) $@.$(INPATH) $@
 
+$(MISC)/%.unpack : $(PRJ)/download/%.oxt
+    @-$(RM) $@
+    noop $(assign UNPACKCMD := unzip $(unzip_quiet_switch)  -o $(BACK_PATH)$(fake_back)download/$(TARFILE_NAME).oxt)
+    @$(TYPE) $(mktmp $(UNPACKCMD)) > $@.$(INPATH)
+    @$(RENAME) $@.$(INPATH) $@
+
 $(MISC)/%.unpack : $(TARFILE_LOCATION2)/%.jar
     @-$(RM) $@
 .IF "$(OS)"=="SOLARIS"
-- 
1.7.0.1

