From f98ae926a447195d992e8b22baff2fd609841291 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:07:31 +0200
Subject: [PATCH 768/878] solenv-unpack-oxt.diff

---
 solenv/inc/tg_ext.mk |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/solenv/inc/tg_ext.mk b/solenv/inc/tg_ext.mk
index 4c1b9b4..df1cdc3 100644
--- a/solenv/inc/tg_ext.mk
+++ b/solenv/inc/tg_ext.mk
@@ -152,6 +152,12 @@ $(MISC)/%.unpack : $(PRJ)/download/%.zip
     @$(TYPE) $(mktmp $(UNPACKCMD)) > $@.$(INPATH)
     @$(RENAME) $@.$(INPATH) $@
 
+$(MISC)/%.unpack : $(PRJ)/download/%.oxt
+    @-$(RM) $@
+    noop $(assign UNPACKCMD := unzip $(unzip_quiet_switch)  -o $(BACK_PATH)$(fake_back)download/$(TARFILE_NAME).oxt)
+    @$(TYPE) $(mktmp $(UNPACKCMD)) > $@.$(INPATH)
+    @$(RENAME) $@.$(INPATH) $@
+
 $(MISC)/%.unpack : $(PRJ)/download/%.jar
     @-$(RM) $@
 .IF "$(OS)"=="SOLARIS"
-- 
1.7.0.1

