--- solenv/bin/modules/installer/worker.pm.old	2009-10-08 16:17:46.000000000 +0200
+++ solenv/bin/modules/installer/worker.pm	2010-03-03 16:59:22.000000000 +0100
@@ -776,9 +776,16 @@ sub install_simple ($$$$$$)
     
         if ((!($dir =~ /\bPREDEFINED_/ )) || ( $dir =~ /\bPREDEFINED_PROGDIR\b/ ))
         {
-            # printf "mkdir $destdir$onedir->{'HostName'}\n";
-            mkdir $destdir . $onedir->{'HostName'};
-            push @lines, "%dir " . $onedir->{'HostName'} . "\n";
+			my $hostname = $onedir->{'HostName'};
+
+			# ignore '.' subdirectories
+			next if ( $hostname =~ m/\.$/ );
+			# remove './' from the path
+			$hostname =~ s/\.\///g;
+
+			# printf "mkdir $destdir$hostname\n";
+			mkdir $destdir . $hostname;
+			push @lines, "%dir " . $hostname . "\n";
         }
     }
 
@@ -795,6 +802,10 @@ sub install_simple ($$$$$$)
         $destination =~ s/\$\$/\$/;
         $sourcepath =~ s/\$\$/\$/;
 
+		# remove './' from the path
+		$sourcepath =~ s/\.\///g;
+		$destination =~ s/\.\///g;
+
         push @lines, "$destination\n";
         # printf "cp $sourcepath $destdir$destination\n";
         copy ("$sourcepath", "$destdir$destination") || die "Can't copy file: $sourcepath -> $destdir$destination $!";
