---
 solenv/inc/unxlngppc.mk |    3 ---
 1 files changed, 0 insertions(+), 3 deletions(-)

diff --git solenv/inc/unxlngppc.mk solenv/inc/unxlngppc.mk
index 4082335..89b2501 100644
--- solenv/inc/unxlngppc.mk
+++ solenv/inc/unxlngppc.mk
@@ -31,7 +31,4 @@ DEFAULTOPT=-Os
 PICSWITCH:=-fPIC
 .INCLUDE : unxlng.mk
 CDEFS+=-DPOWERPC -DPPC
-CFLAGS+=-fsigned-char
-CFLAGSCC+=-fsigned-char
-CFLAGSCXX+=-fsigned-char
 DLLPOSTFIX=lp
-- 
1.7.0.1

