--- solenv/inc/tg_ext.mk.old	2010-03-05 17:38:54.000000000 +0100
+++ solenv/inc/tg_ext.mk	2010-03-08 18:53:36.000000000 +0100
@@ -156,6 +156,12 @@ $(MISC)/%.unpack : $(PRJ)/download/%.zip
     @$(TYPE) $(mktmp $(UNPACKCMD)) > $@.$(INPATH)
     @$(RENAME) $@.$(INPATH) $@
 
+$(MISC)/%.unpack : $(PRJ)/download/%.oxt
+    @-$(RM) $@
+    noop $(assign UNPACKCMD := unzip $(unzip_quiet_switch)  -o $(BACK_PATH)$(fake_back)download/$(TARFILE_NAME).oxt)
+    @$(TYPE) $(mktmp $(UNPACKCMD)) > $@.$(INPATH)
+    @$(RENAME) $@.$(INPATH) $@
+
 $(MISC)/%.unpack : $(PRJ)/download/%.jar
     @-$(RM) $@
 .IF "$(OS)"=="SOLARIS"
