--- configure.in.old	2010-07-29 16:40:48.000000000 +0200
+++ configure.in	2010-07-29 16:48:44.000000000 +0200
@@ -566,6 +566,10 @@ AC_ARG_WITH(system-mozilla,
                           mozilla
 ], WITH_SYSTEM_MOZILLA=$withval, WITH_SYSTEM_MOZILLA=no)
 
+AC_ARG_WITH(linker-hash-style,
+[  --with-linker-hash-style
+], WITH_LINKER_HASH_STYLE=$withval, WITH_LINKER_HASH_STYLE=gnu)
+
 AC_ARG_WITH(stlport,
 [  --with-stlport         The location that STLport is installed in. The STL
                           header files are assumed to be in
@@ -1628,9 +1631,9 @@ if test "$_os" = "SunOS"; then
 fi
 
 if test "$GCC" = "yes"; then
-   AC_MSG_CHECKING( for --hash-style=both linker support )
+   AC_MSG_CHECKING( for --hash-style=$WITH_LINKER_HASH_STYLE linker support )
    hash_style_ldflags_save=$LDFLAGS
-   LDFLAGS="$LDFLAGS -Wl,--hash-style=both"
+   LDFLAGS="$LDFLAGS -Wl,--hash-style=$WITH_LINKER_HASH_STYLE"
    AC_LINK_IFELSE([AC_LANG_PROGRAM([
    #include <stdio.h>
    ],[
@@ -1644,6 +1647,7 @@ if test "$GCC" = "yes"; then
    LDFLAGS=$hash_style_ldflags_save
 fi
 AC_SUBST(HAVE_LD_HASH_STYLE)
+AC_SUBST(WITH_LINKER_HASH_STYLE)
 
 dnl ===================================================================
 dnl  Test the Compaq compiler for OSF1
--- set_soenv.in.old	2010-07-29 16:35:25.000000000 +0200
+++ set_soenv.in	2010-07-29 16:48:44.000000000 +0200
@@ -1776,6 +1776,7 @@ ToFile( "HAVE_GCC_VISIBILITY_FEATURE",
 ToFile( "HAVE_GCC_VISIBILITY_BROKEN",
         "@HAVE_GCC_VISIBILITY_BROKEN@", "e" );
 ToFile( "HAVE_LD_HASH_STYLE","@HAVE_LD_HASH_STYLE@","e" );
+ToFile( "WITH_LINKER_HASH_STYLE","@WITH_LINKER_HASH_STYLE@","e" );
 ToFile( "HAVE_LD_BSYMBOLIC_FUNCTIONS",
 		"@HAVE_LD_BSYMBOLIC_FUNCTIONS@","e" );
 ToFile( "CXX",               $CXX,               "e" );
--- solenv/inc/unxgcc.mk
+++ solenv/inc/unxgcc.mk
@@ -207,7 +207,7 @@ STDSHLCUIMT+=-ljemalloc
 .ENDIF
 
 .IF "$(HAVE_LD_HASH_STYLE)"  == "TRUE"
-LINKFLAGS += -Wl,--hash-style=both
+LINKFLAGS += -Wl,--hash-style=$(WITH_LINKER_HASH_STYLE)
 .ELSE
 LINKFLAGS += -Wl,-zdynsort
 .ENDIF
--- stlport/makefile.mk.old	2010-07-23 17:00:36.000000000 +0200
+++ stlport/makefile.mk	2010-07-29 16:48:44.000000000 +0200
@@ -144,7 +144,7 @@ BUILD_FLAGS=-f vc7.mak EXFLAGS="/EHa /Zc
     BUILD_FLAGS+= -j$(MAXPROCESS)
 .ENDIF
 .IF "$(HAVE_LD_HASH_STYLE)"  == "TRUE"
-CXX+= -Wl,--hash-style=both
+CXX+= -Wl,--hash-style=$(WITH_LINKER_HASH_STYLE)
 .ENDIF
 
 .IF "$(HAVE_LD_BSYMBOLIC_FUNCTIONS)"  == "TRUE"
