---
 offuh/prj/d.lst |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git offuh/prj/d.lst offuh/prj/d.lst
index ba73430..acb5869 100644
--- offuh/prj/d.lst
+++ offuh/prj/d.lst
@@ -353,6 +353,9 @@ mkdir: %_DEST%\inc%_EXT%\offuh\com\sun\star\rdf
 
 ..\%__SRC%\inc\offuh\com\sun\star\script\browse\*.hpp %_DEST%\inc%_EXT%\offuh\com\sun\star\script\browse\*.hpp
 ..\%__SRC%\inc\offuh\com\sun\star\script\browse\*.hdl %_DEST%\inc%_EXT%\offuh\com\sun\star\script\browse\*.hdl
+
+..\%__SRC%\inc\offuh\linking_catch.hxx %_DEST%\inc%_EXT%\offuh\linking_catch.hxx
+
 ..\%__SRC%\inc\offuh\com\sun\star\script\provider\*.hpp %_DEST%\inc%_EXT%\offuh\com\sun\star\script\provider\*.hpp
 ..\%__SRC%\inc\offuh\com\sun\star\script\provider\*.hdl %_DEST%\inc%_EXT%\offuh\com\sun\star\script\provider\*.hdl
 
-- 
1.7.0.1

