--- desktop/scripts/makefile.mk.orig	2009-11-05 16:00:58.000000000 +0100
+++ desktop/scripts/makefile.mk	2009-11-13 15:16:40.000000000 +0100
@@ -53,8 +53,16 @@ UNIXTEXT= \
     $(MISC)$/unoinfo.sh \
     $(MISC)$/unopkg.sh
 
-.IF "$(OS)" != "MACOSX"
-UNIXTEXT+= $(MISC)$/soffice.sh
+.IF "$(OS)" == "MACOSX"
+all: $(UNIXTEXT)
+.ELSE
+all: $(UNIXTEXT) $(MISC)$/soffice.sh
 .ENDIF
 
 .INCLUDE :  target.mk
+
+$(MISC)$/soffice.sh : soffice.sh
+    @echo "Building wrapper: " $@
+    @@-$(RM) -f $@ $@.tmp
+    @tr -d "\015" < $(@:f) > $@.tmp
+    $(TYPE) $@.tmp | $(SED) "s/%%OS%%/$(OS)/" | $(SED) "s/%%CPUNAME%%/$(CPUNAME)/" > $@
diff -u -r /opt/OpenOffice/HEAD/build/ooo320-m2/desktop/scripts/soffice.sh desktop/scripts/soffice.sh
--- desktop/scripts/soffice.sh
+++ desktop/scripts/soffice.sh
@@ -44,7 +44,8 @@
 # SAL_NOOPENGL=true; export SAL_NOOPENGL
 
 # the following test is needed on Linux PPC with IBM j2sdk142
-if [ "`uname -s`" = "Linux" -a "`uname -m`" = "ppc" ] ; then
+# These variables are substituted during the build process
+if [ "%%OS%%" = "LINUX" -a "%%CPUNAME%%" = "POWERPC" ] ; then
     JITC_PROCESSOR_TYPE=6
     export JITC_PROCESSOR_TYPE
 fi
