From 63a431b1bc49189d26fcb30f13f29be051fa0be7 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:53:04 +0200
Subject: [PATCH 019/768] speed-pagein.diff

---
 desktop/source/pagein/makefile.mk |   36 +++++++++++++++++++++++++++++++++++-
 1 files changed, 35 insertions(+), 1 deletions(-)

diff --git desktop/source/pagein/makefile.mk desktop/source/pagein/makefile.mk
index 01452a8..637f290 100644
--- desktop/source/pagein/makefile.mk
+++ desktop/source/pagein/makefile.mk
@@ -160,6 +160,40 @@ $(MISC)$/$(TARGET)-common : makefile.mk
     @-echo $(DLLPRE)svl$(DFTDLLPOST)   >> $@
     @-echo $(DLLPRE)vcl$(DFTDLLPOST)   >> $@
     @-echo $(DLLPRE)tk$(DFTDLLPOST)    >> $@
-    @-echo $(DLLPRE)svt$(DFTDLLPOST)   >> $@
+    @-echo types.rdb                   >> $@
+    @-echo services.rdb                >> $@
+# urgh ...
+    @-echo oovbaapi.rdb                                         >> $@
+    @-echo deployment$$(DLLPOSTFIX)$(UNODLLPOST)           >> $@
+    @-echo $(DLLPRE)deploymentmisc$(DLLPOSTFIX)$(DLLPOST) >> $@
+# omitted pieces (from writer startup ...) hopefully common
+    @-echo $(DLLPRE)ucb1$(DLLPOST)               >> $@
+    @-echo $(DLLPRE)xstor$(DLLPOST)              >> $@
+    @-echo $(DLLPRE)package2$(DLLPOST)           >> $@
+    @-echo $(DLLPRE)filterconfig1$(DLLPOST)      >> $@
+    @-echo $(DLLPRE)uui$(DFTDLLPOST)             >> $@
+    @-echo $(DLLPRE)lng$(DFTDLLPOST)             >> $@
+    @-echo $(DLLPRE)svt$(DFTDLLPOST)             >> $@
+    @-echo $(DLLPRE)spl$(DFTDLLPOST)             >> $@
+    @-echo $(DLLPRE)basegfx$(DFTDLLPOST)         >> $@
+    @-echo $(DLLPRE)avmedia$(DFTDLLPOST)         >> $@
+    @-echo $(DLLPRE)dtransX11$(DFTDLLPOST)       >> $@
+    @-echo $(DLLPRE)helplinker$(DFTDLLPOST)      >> $@
+    @-echo $(DLLPRE)vclplug_gen$(DFTDLLPOST)     >> $@
+    @-echo $(DLLPRE)icule$(ICUDLLPOST)           >> $@
+    @-echo $(DLLPRE)icudata$(ICUDLLPOST)         >> $@
+    @-echo sax$(UNODLLPOST)                      >> $@
+    @-echo gconfbe1$(UNODLLPOST)                 >> $@
+    @-echo fsstorage$(UNODLLPOST)                >> $@
+    @-echo desktopbe$(UNODLLPOST)                >> $@
+    @-echo localebe1$(UNODLLPOST)                >> $@
+    @-echo ucpexpand1$(UNODLLPOST)               >> $@
+# uno bits
+    @-echo $(DLLPRE)reg$(DLLPOST)                >> $@
+    @-echo $(DLLPRE)store$(DLLPOST)              >> $@
+# stoc bits
+    @-echo bootstrap$(UNODLLPOST)                >> $@
+    @-echo reflection$(UNODLLPOST)               >> $@
+    @-echo stocservices$(UNODLLPOST)             >> $@
     @-echo $(DLLPRE)sfx$(DFTDLLPOST)   >> $@
     @-echo $(DLLPRE)sofficeapp$(DLLPOST) >> $@
-- 
1.7.0.1

