diff -u -r /opt/OpenOffice/HEAD/build/ooo320-m2/desktop/scripts/makefile.mk desktop/scripts/makefile.mk
--- /opt/OpenOffice/HEAD/build/ooo320-m2/desktop/scripts/makefile.mk	2009-08-20 11:47:59.000000000 +0200
+++ desktop/scripts/makefile.mk	2009-11-07 20:07:43.000000000 +0100
@@ -54,7 +54,12 @@
     $(MISC)$/unopkg.sh
 
 .IF "$(OS)" != "MACOSX"
-UNIXTEXT+= $(MISC)$/soffice.sh
+$(MISC)$/soffice.sh : soffice.sh
+	@echo "Building wrapper: " $@
+	@@-$(RM) -f $@ $@.tmp
+	@tr -d "\015" < $(@:f) > $@.tmp
+	$(TYPE) $@.tmp | $(SED) "s/%%OS%%/$(OS)/" | $(SED) "s/%%CPUNAME%%/$(CPUNAME)/" > $@
 .ENDIF
 
 .INCLUDE :  target.mk
+
Only in desktop/scripts: makefile.mk~
diff -u -r /opt/OpenOffice/HEAD/build/ooo320-m2/desktop/scripts/soffice.sh desktop/scripts/soffice.sh
--- /opt/OpenOffice/HEAD/build/ooo320-m2/desktop/scripts/soffice.sh	2009-10-19 17:08:33.000000000 +0200
+++ desktop/scripts/soffice.sh	2009-11-07 20:08:40.000000000 +0100
@@ -44,7 +44,8 @@
 # SAL_NOOPENGL=true; export SAL_NOOPENGL
 
 # the following test is needed on Linux PPC with IBM j2sdk142
-if [ "`uname -s`" = "Linux" -a "`uname -m`" = "ppc" ] ; then
+# These variables are substituted during the build process
+if [ "%%OS%%" = "LINUX" -a "%%CPUNAME%%" = "POWERPC" ] ; then
     JITC_PROCESSOR_TYPE=6
     export JITC_PROCESSOR_TYPE
 fi
Only in desktop/scripts: soffice.sh~
