Index: lingucomponent/source/thesaurus/libnth/makefile.mk
===================================================================
RCS file: /cvs/whiteboard/lingucomponent/source/thesaurus/libnth/makefile.mk,v
retrieving revision 1.4
diff -u -u -r1.4 makefile.mk
--- lingucomponent/source/thesaurus/libnth/makefile.mk	17 Jun 2005 09:22:43 -0000	1.4
+++ lingucomponent/source/thesaurus/libnth/makefile.mk	8 Aug 2005 20:42:23 -0000
@@ -39,7 +39,11 @@
 
 .IF "$(MYTHESLIB)"==""
 .IF "$(GUI)"=="UNX"
+.IF "$(SYSTEM_MYTHES)" == "YES"
+MYTHESLIB=-lmythes_pic
+.ELSE
 MYTHESLIB=-lmythes
+.ENDIF
 .ENDIF # unx
 .IF "$(GUI)"=="OS2"
 MYTHESLIB=$(SLB)\libmythes.lib
--- solenv/inc/libs.mk.old 2005-12-09 08:25:13.000000000 +0100
+++ solenv/inc/libs.mk     2005-12-09 08:25:28.000000000 +0100
@@ -151,7 +151,7 @@
 FORUILIB=-lforui$(DLLPOSTFIX)
 
 .IF "$(SYSTEM_AGG)" == "YES"
-AGGLIB=-lagg
+AGGLIB=-lagg_pic
 .ELSE
 AGGLIB=-lagg$(DLLPOSTFIX)
 .ENDIF
@@ -338,7 +338,7 @@
 .ELSE
 HUNSPELLLIB=-lhunspell-1.2
 .ENDIF
-MYTHESLIB=-lmythes
+MYTHESLIB*=-lmythes
 PYUNOLIB=-lpyuno
 LPSOLVELIB=-llpsolve55
 SOFFICELIB=-lsofficeapp
