--- helpcontent2/source/text/shared/02/20060000.xhp.old	2009-09-03 10:39:54.000000000 +0200
+++ helpcontent2/source/text/shared/02/20060000.xhp	2009-09-03 10:40:12.000000000 +0200
@@ -49,7 +49,7 @@
 <bookmark xml-lang="en-US" branch="hid/SID_MODIFIED" id="bm_id1803964" localize="false"/>
 <bookmark xml-lang="en-US" branch="hid/.uno:Modified" id="bm_id3156183" localize="false"/>
 <bookmark xml-lang="en-US" branch="hid/SID_MODIFYSTATUS" id="bm_id3154682" localize="false"/><paragraph role="heading" id="hd_id3147477" xml-lang="en-US" level="1" l10n="U" oldref="1"><link href="text/shared/02/20060000.xhp" name="Document Modification">Document Modification</link></paragraph>
-<paragraph role="paragraph" id="par_id3148731" xml-lang="en-US" l10n="CHG" oldref="2"><ahelp hid="SID_MODIFYSTATUS">If changes to the document have not yet been saved, a "*" is displayed in this field on the Status Bar. This also applies to new, not yet saved documents.</ahelp></paragraph>
+<paragraph role="paragraph" id="par_id3148731" xml-lang="en-US" l10n="CHG" oldref="2"><ahelp hid="SID_MODIFYSTATUS">If changes to the document have not yet been saved, a "!" is displayed in this field on the Status Bar. This also applies to new, not yet saved documents.</ahelp></paragraph>
 </section>
 </body>
 </helpdocument>
