From c7f8fb626d39d215058d3d5c144516abcfcead8e Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:07:10 +0200
Subject: [PATCH 658/768] store-crash-fix.diff

---
 store/source/storbase.hxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git store/source/storbase.hxx store/source/storbase.hxx
index e2d23d8..334c967 100644
--- store/source/storbase.hxx
+++ store/source/storbase.hxx
@@ -680,7 +680,7 @@ class PageHolderObject
     template< class U >
     static bool isA (PageData const * p)
     {
-        return ((p != 0) && (p->type() == U::theTypeId));
+        return p != 0;
     }
 
     template< class U >
-- 
1.7.0.1

