From 70a20cce973531e232bbc74f935edb751985f2bb Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:07:10 +0200
Subject: [PATCH 749/878] store-crash-fix.diff

---
 store/source/storbase.hxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/store/source/storbase.hxx b/store/source/storbase.hxx
index 0a489c1..fe1f7c2 100644
--- a/store/source/storbase.hxx
+++ b/store/source/storbase.hxx
@@ -697,7 +697,7 @@ class PageHolderObject
     template< class U >
     static bool isA (PageData const * p)
     {
-        return ((p != 0) && (p->type() == U::theTypeId));
+        return p != 0;
     }
 
     template< class U >
-- 
1.7.0.1

