From 6c84c1eee584d8922cc941818410b83b26de4a7d Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:02:04 +0200
Subject: [PATCH 425/768] svg-import-empty-gradient-fix.diff

---
 filter/source/svg/svgreader.cxx |   18 ++++++++++--------
 1 files changed, 10 insertions(+), 8 deletions(-)

diff --git filter/source/svg/svgreader.cxx filter/source/svg/svgreader.cxx
index 4fe718b..b674c0c 100644
--- filter/source/svg/svgreader.cxx
+++ filter/source/svg/svgreader.cxx
@@ -308,10 +308,11 @@ struct AnnotatingVisitor
     bool hasGradientOpacity( const Gradient& rGradient )
     {
         return
-            maGradientStopVector[
-                rGradient.maStops[0]].maStopColor.a != 1.0 ||
-            maGradientStopVector[
-                rGradient.maStops[1]].maStopColor.a != 1.0;
+            !rGradient.maStops.empty() &&
+            (maGradientStopVector[
+                 rGradient.maStops[0]].maStopColor.a != 1.0 ||
+             maGradientStopVector[
+                 rGradient.maStops[1]].maStopColor.a != 1.0);
     }
 
     struct StopSorter
@@ -2464,10 +2465,11 @@ struct ShapeRenderingVisitor
     bool hasGradientOpacity( const Gradient& rGradient )
     {
         return
-            mrGradientStopVector[
-                rGradient.maStops[0]].maStopColor.a != 1.0 ||
-            mrGradientStopVector[
-                rGradient.maStops[1]].maStopColor.a != 1.0;
+            !rGradient.maStops.empty() &&
+            (mrGradientStopVector[
+                 rGradient.maStops[0]].maStopColor.a != 1.0 ||
+             mrGradientStopVector[
+                 rGradient.maStops[1]].maStopColor.a != 1.0);
     }
 
     sal_Int8 toByteColor( double val )
-- 
1.7.0.1

