diff -ru filter.orig/source/svg/svgfilter.cxx filter/source/svg/svgfilter.cxx
--- filter.orig/source/svg/svgfilter.cxx	2010-02-16 20:20:25.000000000 +0000
+++ filter/source/svg/svgfilter.cxx	2010-02-17 14:58:24.000000000 +0000
@@ -211,7 +211,14 @@
 class FilterConfigItem;
 extern "C" SAL_DLLPUBLIC_EXPORT BOOL __LOADONCALLAPI GraphicImport(SvStream & rStream, Graphic & rGraphic, FilterConfigItem*, BOOL )
 {
-	return importSvg( rStream, rGraphic );
+    BOOL bRet = FALSE;
+    try
+    {
+        bRet = importSvg( rStream, rGraphic );
+    }
+    catch (const uno::Exception&) {
+    }
+    return bRet;
 }
 
 // -----------------------------------------------------------------------------
diff -ru filter.orig/source/svg/svgreader.cxx filter/source/svg/svgreader.cxx
--- filter.orig/source/svg/svgreader.cxx	2010-02-16 20:20:25.000000000 +0000
+++ filter/source/svg/svgreader.cxx	2010-02-17 14:50:39.000000000 +0000
@@ -1796,7 +1796,7 @@
 {
 	uno::Reference<xml::dom::XDocumentBuilder> xDomBuilder( 
         m_xServiceFactory->createInstance( 
-            rtl::OUString::createFromAscii("com.sun.star.xml.dom.DocumentBuilder")), uno::UNO_QUERY );
+            rtl::OUString::createFromAscii("com.sun.star.xml.dom.DocumentBuilder")), uno::UNO_QUERY_THROW );
 
     uno::Reference<xml::dom::XDocument> xDom(
         xDomBuilder->parse(m_xInputStream),
@@ -2702,7 +2702,7 @@
 	uno::Reference<xml::dom::XDocumentBuilder> xDomBuilder( 
         xServiceFactory->createInstance( 
             rtl::OUString::createFromAscii("com.sun.star.xml.dom.DocumentBuilder")), 
-        uno::UNO_QUERY );
+        uno::UNO_QUERY_THROW );
 
     uno::Reference<io::XInputStream> xStream(
         new utl::OInputStreamWrapper(rStream) );
