diff --git svtools/source/numbers/zforfind.cxx svtools/source/numbers/zforfind.cxx
index 9fba39e..872ada2 100644
--- svtools/source/numbers/zforfind.cxx
+++ svtools/source/numbers/zforfind.cxx
@@ -138,6 +138,7 @@ void ImpSvNumberInputScan::Reset()
     nAnzNums     = 0;
     nThousand    = 0;
     eScannedType = NUMBERFORMAT_UNDEFINED;
+    eSetType     = NUMBERFORMAT_UNDEFINED;
     nAmPm        = 0;
     nPosThousandString = 0;
     nLogical     = 0;
