From 01a9d3ee36af470c17d0818325d817c41740eb7c Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:05:17 +0200
Subject: [PATCH 663/878] svtools-svrtf-token-ignore-case.diff

---
 svtools/source/svrtf/rtfkeywd.cxx |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/svtools/source/svrtf/rtfkeywd.cxx b/svtools/source/svrtf/rtfkeywd.cxx
index b6fa332..9709efb 100644
--- a/svtools/source/svrtf/rtfkeywd.cxx
+++ b/svtools/source/svrtf/rtfkeywd.cxx
@@ -1201,13 +1201,13 @@ _Optlink
             nRet = ((RTF_TokenEntry*)pFirst)->pUToken->CompareTo(
                             *((RTF_TokenEntry*)pSecond)->pUToken );
         else
-            nRet = ((RTF_TokenEntry*)pFirst)->pUToken->CompareToAscii(
+            nRet = ((RTF_TokenEntry*)pFirst)->pUToken->CompareIgnoreCaseToAscii(
                             ((RTF_TokenEntry*)pSecond)->sToken );
     }
     else
     {
         if( -1 == ((RTF_TokenEntry*)pSecond)->nToken )
-            nRet = -1 * ((RTF_TokenEntry*)pSecond)->pUToken->CompareToAscii(
+            nRet = -1 * ((RTF_TokenEntry*)pSecond)->pUToken->CompareIgnoreCaseToAscii(
                             ((RTF_TokenEntry*)pFirst)->sToken );
         else
             nRet = strcmp( ((RTF_TokenEntry*)pFirst)->sToken,
-- 
1.7.0.1

