diff -rup svtools/source/filter.vcl/wmf-orig/enhwmf.cxx svtools/source/filter.vcl/wmf/enhwmf.cxx
--- svtools/source/filter.vcl/wmf-orig/enhwmf.cxx	2011-01-26 14:48:00.000000000 +0100
+++ svtools/source/filter.vcl/wmf/enhwmf.cxx	2011-01-26 14:53:30.000000000 +0100
@@ -344,10 +344,17 @@ BOOL EnhWMFReader::ReadEnhWMF()
     sal_Bool    bHaveDC = false;
 
 #ifdef UNX
-    static sal_Bool bEnableEMFPlus = ( getenv( "EMF_PLUS_DISABLE" ) == NULL );
+    static sal_Bool bEnableEMFPlus = sal_False; // disable it for now as
+                                                // drawing layer changed the
+                                                // way the editing mode is
+                                                // rendered and emf+ support
+                                                // doesn't work anymore
+                                                // ( getenv( "EMF_PLUS_DISABLE" ) == NULL );
 #else
     // TODO: make it possible to disable emf+ on windows
-    static sal_Bool bEnableEMFPlus = sal_True;
+    // static sal_Bool bEnableEMFPlus = sal_True;
+    // disable it on windows as well (see above)
+    static sal_Bool bEnableEMFPlus = sal_False;
 #endif
 
     while( bStatus && nRecordCount-- )
Only in svtools/source/filter.vcl/wmf: enhwmf.cxx~
