From 5b9b0a475411627cdd5010e7fa3513516b74f9b1 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:05:03 +0200
Subject: [PATCH 652/878] svtools-update-ole.diff

---
 svtools/source/misc/embedhlp.cxx |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/svtools/source/misc/embedhlp.cxx b/svtools/source/misc/embedhlp.cxx
index eb36541..217a918 100644
--- a/svtools/source/misc/embedhlp.cxx
+++ b/svtools/source/misc/embedhlp.cxx
@@ -180,7 +180,9 @@ void SAL_CALL EmbedEventListener_Impl::modified( const lang::EventObject& ) thro
             else
                 pObject->UpdateReplacement();
         }
-        else if ( nState == embed::EmbedStates::UI_ACTIVE || nState == embed::EmbedStates::INPLACE_ACTIVE )
+        else if ( nState == embed::EmbedStates::ACTIVE ||
+                  nState == embed::EmbedStates::UI_ACTIVE ||
+                  nState == embed::EmbedStates::INPLACE_ACTIVE )
         {
             // in case the object is inplace or UI active the replacement image should be updated on demand
             pObject->UpdateReplacementOnDemand();
-- 
1.7.0.1

