---
 svtools/source/misc/embedhlp.cxx |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git svtools/source/misc/embedhlp.cxx svtools/source/misc/embedhlp.cxx
index d94c798..762e8af 100644
--- svtools/source/misc/embedhlp.cxx
+++ svtools/source/misc/embedhlp.cxx
@@ -181,7 +181,9 @@ void SAL_CALL EmbedEventListener_Impl::modified( const lang::EventObject& ) thro
             else
                 pObject->UpdateReplacement();
         }
-        else if ( nState == embed::EmbedStates::UI_ACTIVE || nState == embed::EmbedStates::INPLACE_ACTIVE )
+        else if ( nState == embed::EmbedStates::ACTIVE ||
+                  nState == embed::EmbedStates::UI_ACTIVE ||
+                  nState == embed::EmbedStates::INPLACE_ACTIVE )
         {
             // in case the object is inplace or UI active the replacement image should be updated on demand
             pObject->UpdateReplacementOnDemand();
-- 
1.7.0.1

