From 985c3628c25453e18ae9e454f859d55a688dd9c6 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:05:03 +0200
Subject: [PATCH 572/768] svtools-update-ole.diff

---
 svtools/source/misc/embedhlp.cxx |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git svtools/source/misc/embedhlp.cxx svtools/source/misc/embedhlp.cxx
index b877b7a..04e6ab5 100644
--- svtools/source/misc/embedhlp.cxx
+++ svtools/source/misc/embedhlp.cxx
@@ -181,7 +181,9 @@ void SAL_CALL EmbedEventListener_Impl::modified( const lang::EventObject& ) thro
             else
                 pObject->UpdateReplacement();
         }
-        else if ( nState == embed::EmbedStates::UI_ACTIVE || nState == embed::EmbedStates::INPLACE_ACTIVE )
+        else if ( nState == embed::EmbedStates::ACTIVE ||
+                  nState == embed::EmbedStates::UI_ACTIVE ||
+                  nState == embed::EmbedStates::INPLACE_ACTIVE )
         {
             // in case the object is inplace or UI active the replacement image should be updated on demand
             pObject->UpdateReplacementOnDemand();
-- 
1.7.0.1

