From ae973821477f41e4198551bf4846075a1e2b3980 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:57:11 +0200
Subject: [PATCH 220/768] svtools-default-new-document.diff

---
 svtools/source/contnr/templwin.cxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git svtools/source/contnr/templwin.cxx svtools/source/contnr/templwin.cxx
index 7f5797f..f76b220 100644
--- svtools/source/contnr/templwin.cxx
+++ svtools/source/contnr/templwin.cxx
@@ -1551,7 +1551,7 @@ long SvtTemplateWindow::CalcHeight() const
 void SvtTemplateWindow::ReadViewSettings()
 {
     // defaults
-    sal_Int32 nSelectedGroup	=	ICON_POS_TEMPLATES;
+    sal_Int32 nSelectedGroup	=	ICON_POS_NEWDOC;
     sal_Int32 nSelectedView		=	TI_DOCTEMPLATE_DOCINFO;
     double nSplitRatio			=	0.5;
     ::rtl::OUString sLastFolder;
-- 
1.7.0.1

