diff -rup svtools-orig/source/filter.vcl/wmf/enhwmf.cxx svtools/source/filter.vcl/wmf/enhwmf.cxx
--- svtools-orig/source/filter.vcl/wmf/enhwmf.cxx	2009-12-08 17:34:50.000000000 +0100
+++ svtools/source/filter.vcl/wmf/enhwmf.cxx	2009-12-08 17:59:03.000000000 +0100
@@ -265,7 +265,7 @@ void EnhWMFReader::ReadEMFPlusComment(sa
 
     void *buffer = malloc( length );
 
-    int count = 0, next, pos = pWMF->Tell();
+    int pos = pWMF->Tell();
     pOut->PassEMFPlus( buffer, pWMF->Read( buffer, length ) );
     pWMF->Seek( pos );
 
@@ -321,14 +321,10 @@ void EnhWMFReader::ReadGDIComment()
         break;
     }
     case 3: {
-        sal_uInt32 x, y, w, h;
-
         EMFP_DEBUG(printf ("\t\tENDGROUP\n"));
         break;
     }
     case 0x40000004: {
-        sal_uInt32 x, y, w, h;
-
         EMFP_DEBUG(printf ("\t\tMULTIFORMATS\n"));
         break;
     }
@@ -400,10 +396,11 @@ BOOL EnhWMFReader::ReadEnhWMF()
                 if( id == 0x2B464D45 && nRecSize >= 12 )
                     ReadEMFPlusComment( length, bHaveDC );
                 // GDIC comment, doesn't do anything useful yet => enabled only for debug
-                else if( id == 0x43494447 && nRecSize >= 12 )
+                else if( id == 0x43494447 && nRecSize >= 12 ) {
                     EMFP_DEBUG(ReadGDIComment());
-                else
+                } else {
                     EMFP_DEBUG(printf ("\t\tunknown id: 0x%x\n", id));
+        }
             }
         } else if( !bEMFPlus || bHaveDC || nRecType == EMR_EOF )
 
@@ -1342,7 +1339,6 @@ BOOL EnhWMFReader::ReadEnhWMF()
 
             case EMR_CREATEDIBPATTERNBRUSHPT :
             {
-                static int count = 0;
 				UINT32	nStart = pWMF->Tell() - 8;
                 Bitmap aBitmap;
 
diff -rup svtools-orig/source/filter.vcl/wmf/winmtf.hxx svtools/source/filter.vcl/wmf/winmtf.hxx
--- svtools-orig/source/filter.vcl/wmf/winmtf.hxx	2009-12-08 17:34:50.000000000 +0100
+++ svtools/source/filter.vcl/wmf/winmtf.hxx	2009-12-08 17:57:54.000000000 +0100
@@ -393,8 +393,8 @@ struct WinMtfFillStyle
 	};
 
 	WinMtfFillStyle( Bitmap& rBmp ) :
-		aBmp ( rBmp ),
-        aType( FillStylePattern )
+        aType( FillStylePattern ),
+        aBmp ( rBmp )
     {
     };
 
