From 037f60bd4b401478488f9aad88ed7766fe0f87c3 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:56:40 +0200
Subject: [PATCH 222/878] svtools-uninitialized-member-fix.diff

---
 svtools/source/numbers/zforfind.cxx |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/svtools/source/numbers/zforfind.cxx b/svtools/source/numbers/zforfind.cxx
index 12188ff..d0c4c43 100644
--- a/svtools/source/numbers/zforfind.cxx
+++ b/svtools/source/numbers/zforfind.cxx
@@ -84,7 +84,9 @@ ImpSvNumberInputScan::ImpSvNumberInputScan( SvNumberFormatter* pFormatterP )
         pUpperMonthText( NULL ),
         pUpperAbbrevMonthText( NULL ),
         pUpperDayText( NULL ),
-        pUpperAbbrevDayText( NULL )
+        pUpperAbbrevDayText( NULL ),
+        eScannedType( NUMBERFORMAT_UNDEFINED ),
+        eSetType( NUMBERFORMAT_UNDEFINED )
 {
     pFormatter = pFormatterP;
     pNullDate = new Date(30,12,1899);
-- 
1.7.0.1

