From 6ec17b9a645148606befa55b8622706595b92a24 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:56:40 +0200
Subject: [PATCH 194/768] svtools-uninitialized-member-fix.diff

---
 svl/source/numbers/zforfind.cxx |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git svl/source/numbers/zforfind.cxx svl/source/numbers/zforfind.cxx
index a049895..70ea935 100644
--- svl/source/numbers/zforfind.cxx
+++ svl/source/numbers/zforfind.cxx
@@ -84,7 +84,9 @@ ImpSvNumberInputScan::ImpSvNumberInputScan( SvNumberFormatter* pFormatterP )
         pUpperMonthText( NULL ),
         pUpperAbbrevMonthText( NULL ),
         pUpperDayText( NULL ),
-        pUpperAbbrevDayText( NULL )
+        pUpperAbbrevDayText( NULL ),
+        eScannedType( NUMBERFORMAT_UNDEFINED ),
+        eSetType( NUMBERFORMAT_UNDEFINED )
 {
     pFormatter = pFormatterP;
     pNullDate = new Date(30,12,1899);
-- 
1.7.0.1

