---
 svx/source/svdraw/svdoashp.cxx |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git svx/source/svdraw/svdoashp.cxx svx/source/svdraw/svdoashp.cxx
index 6beb3db..82066f9 100644
--- svx/source/svdraw/svdoashp.cxx
+++ svx/source/svdraw/svdoashp.cxx
@@ -1669,6 +1669,11 @@ void SdrObjCustomShape::TakeObjInfo(SdrObjTransformInfoRec& rInfo) const
                 {
                     rInfo.bCanConvToContour = aInfo.bCanConvToContour;
                 }
+
+                if( !rInfo.bShearAllowed )
+                    rInfo.bShearAllowed=FALSE;
+                if( !aInfo.bEdgeRadiusAllowed )
+                    rInfo.bEdgeRadiusAllowed=FALSE;
             }
         }
     }
-- 
1.7.0.1

