--- svx/inc/svx/sdtfchim.hxx
+++ svx/inc/svx/sdtfchim.hxx
@@ -32,28 +32,28 @@
 
 #include <svx/svddef.hxx>
 
-class SdrTextFixedCellHeightItem : public SfxBoolItem
+class SVX_DLLPUBLIC SdrTextFixedCellHeightItem : public SfxBoolItem
 {
 public:
 
 	TYPEINFO();
 	SdrTextFixedCellHeightItem( BOOL bUseFixedCellHeight = FALSE );
-	SdrTextFixedCellHeightItem( SvStream & rStream, sal_uInt16 nVersion );
+	SVX_DLLPRIVATE SdrTextFixedCellHeightItem( SvStream & rStream, sal_uInt16 nVersion );
 
-	virtual SfxItemPresentation GetPresentation(SfxItemPresentation ePresentation,
+	SVX_DLLPRIVATE virtual SfxItemPresentation GetPresentation(SfxItemPresentation ePresentation,
 									SfxMapUnit eCoreMetric, SfxMapUnit ePresentationMetric,
                                         String &rText, const IntlWrapper * = 0) const;
 
-	virtual SfxPoolItem*		Create( SvStream&, sal_uInt16 nItem ) const;
-	virtual SvStream&			Store( SvStream&, sal_uInt16 nVersion ) const;
-	virtual SfxPoolItem*		Clone( SfxItemPool* pPool = NULL ) const;
-	virtual	sal_uInt16			GetVersion( sal_uInt16 nFileFormatVersion ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*		Create( SvStream&, sal_uInt16 nItem ) const;
+	SVX_DLLPRIVATE virtual SvStream&			Store( SvStream&, sal_uInt16 nVersion ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*		Clone( SfxItemPool* pPool = NULL ) const;
+	SVX_DLLPRIVATE virtual	sal_uInt16			GetVersion( sal_uInt16 nFileFormatVersion ) const;
 
-	virtual	sal_Bool			QueryValue( com::sun::star::uno::Any& rVal, BYTE nMemberId = 0 ) const;
-	virtual	sal_Bool			PutValue( const com::sun::star::uno::Any& rVal, BYTE nMemberId = 0 );
+	SVX_DLLPRIVATE virtual	sal_Bool			QueryValue( com::sun::star::uno::Any& rVal, BYTE nMemberId = 0 ) const;
+	SVX_DLLPRIVATE virtual	sal_Bool			PutValue( const com::sun::star::uno::Any& rVal, BYTE nMemberId = 0 );
 
 #ifdef SDR_ISPOOLABLE
-	virtual int IsPoolable() const;
+	SVX_DLLPRIVATE virtual int IsPoolable() const;
 #endif
 };
 
--- svx/inc/svx/xflbckit.hxx
+++ svx/inc/svx/xflbckit.hxx
@@ -41,13 +41,13 @@
 							TYPEINFO();
 
 							XFillBackgroundItem( BOOL bFill = FALSE );
-							XFillBackgroundItem( SvStream& rIn );
+							SVX_DLLPRIVATE XFillBackgroundItem( SvStream& rIn );
 
 
-	virtual SfxPoolItem*    Clone( SfxItemPool* pPool = NULL ) const;
-	virtual SfxPoolItem*    Create( SvStream& rIn, USHORT nVer ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Clone( SfxItemPool* pPool = NULL ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Create( SvStream& rIn, USHORT nVer ) const;
 
-	virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
+	SVX_DLLPRIVATE virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
 									SfxMapUnit eCoreMetric,
 									SfxMapUnit ePresMetric,
                                     String &rText, const IntlWrapper * = 0 ) const;
--- svx/inc/svx/xflbmpit.hxx
+++ svx/inc/svx/xflbmpit.hxx
@@ -39,18 +39,18 @@
 public:
 							TYPEINFO();
 							XFillBmpPosItem( RECT_POINT eRP = RP_MM );
-							XFillBmpPosItem( SvStream& rIn );
+							SVX_DLLPRIVATE XFillBmpPosItem( SvStream& rIn );
 
-	virtual SfxPoolItem*    Clone( SfxItemPool* pPool = 0 ) const;
-	virtual SfxPoolItem*    Create( SvStream& rIn, USHORT nVer ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Clone( SfxItemPool* pPool = 0 ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Create( SvStream& rIn, USHORT nVer ) const;
 
-	virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
+	SVX_DLLPRIVATE virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
 									SfxMapUnit eCoreMetric,
 									SfxMapUnit ePresMetric,
                                     String &rText, const IntlWrapper * = 0 ) const;
 
-	virtual USHORT			GetValueCount() const;
-	RECT_POINT 				GetValue() const { return (RECT_POINT) SfxEnumItem::GetValue(); }
+	SVX_DLLPRIVATE virtual USHORT			GetValueCount() const;
+	SVX_DLLPRIVATE RECT_POINT 				GetValue() const { return (RECT_POINT) SfxEnumItem::GetValue(); }
 };
 
 
--- svx/inc/svx/xflbmsli.hxx
+++ svx/inc/svx/xflbmsli.hxx
@@ -39,12 +39,12 @@
 public:
 							TYPEINFO();
 							XFillBmpSizeLogItem( BOOL bLog = TRUE );
-							XFillBmpSizeLogItem( SvStream& rIn );
+							SVX_DLLPRIVATE XFillBmpSizeLogItem( SvStream& rIn );
 
-	virtual SfxPoolItem*    Clone( SfxItemPool* pPool = 0 ) const;
-	virtual SfxPoolItem*    Create( SvStream& rIn, USHORT nVer ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Clone( SfxItemPool* pPool = 0 ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Create( SvStream& rIn, USHORT nVer ) const;
 
-	virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
+	SVX_DLLPRIVATE virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
 									SfxMapUnit eCoreMetric,
 									SfxMapUnit ePresMetric,
                                     String &rText, const IntlWrapper * = 0 ) const;
--- svx/inc/svx/xflbmsxy.hxx
+++ svx/inc/svx/xflbmsxy.hxx
@@ -39,12 +39,12 @@
 public:
 							TYPEINFO();
 							XFillBmpSizeXItem( long nSizeX = 0 );
-							XFillBmpSizeXItem( SvStream& rIn );
+							SVX_DLLPRIVATE XFillBmpSizeXItem( SvStream& rIn );
 
-	virtual SfxPoolItem*    Clone( SfxItemPool* pPool = 0 ) const;
-	virtual SfxPoolItem*    Create( SvStream& rIn, USHORT nVer ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Clone( SfxItemPool* pPool = 0 ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Create( SvStream& rIn, USHORT nVer ) const;
 
-	virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
+	SVX_DLLPRIVATE virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
 									SfxMapUnit eCoreMetric,
 									SfxMapUnit ePresMetric,
                                     String &rText, const IntlWrapper * = 0 ) const;
@@ -57,7 +57,7 @@
 	// Dieses Item nimmt sowohl metrische als auch
 	// Prozentwerte ( sind negativ ) auf; das muessen
 	// wir hier beruecksichtigen
-	virtual FASTBOOL		HasMetrics() const;
+	SVX_DLLPRIVATE virtual FASTBOOL		HasMetrics() const;
 };
 
 /*************************************************************************
@@ -66,12 +66,12 @@
 public:
 							TYPEINFO();
 							XFillBmpSizeYItem( long nSizeY = 0 );
-							XFillBmpSizeYItem( SvStream& rIn );
+							SVX_DLLPRIVATE XFillBmpSizeYItem( SvStream& rIn );
 
-	virtual SfxPoolItem*    Clone( SfxItemPool* pPool = 0 ) const;
-	virtual SfxPoolItem*    Create( SvStream& rIn, USHORT nVer ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Clone( SfxItemPool* pPool = 0 ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Create( SvStream& rIn, USHORT nVer ) const;
 
-	virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
+	SVX_DLLPRIVATE virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
 									SfxMapUnit eCoreMetric,
 									SfxMapUnit ePresMetric,
                                     String &rText, const IntlWrapper * = 0 ) const;
@@ -84,7 +84,7 @@
 	// Dieses Item nimmt sowohl metrische als auch
 	// Prozentwerte ( sind negativ ) auf; das muessen
 	// wir hier beruecksichtigen
-	virtual FASTBOOL		HasMetrics() const;
+	SVX_DLLPRIVATE virtual FASTBOOL		HasMetrics() const;
 };
 
 #endif
--- svx/inc/svx/xflboxy.hxx
+++ svx/inc/svx/xflboxy.hxx
@@ -39,12 +39,12 @@
 public:
 							TYPEINFO();
 							XFillBmpPosOffsetXItem( USHORT nOffPosX = 0 );
-							XFillBmpPosOffsetXItem( SvStream& rIn );
+							SVX_DLLPRIVATE XFillBmpPosOffsetXItem( SvStream& rIn );
 
-	virtual SfxPoolItem*    Clone( SfxItemPool* pPool = 0 ) const;
-	virtual SfxPoolItem*    Create( SvStream& rIn, USHORT nVer ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Clone( SfxItemPool* pPool = 0 ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Create( SvStream& rIn, USHORT nVer ) const;
 
-	virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
+	SVX_DLLPRIVATE virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
 									SfxMapUnit eCoreMetric,
 									SfxMapUnit ePresMetric,
                                     String &rText, const IntlWrapper * = 0 ) const;
@@ -62,12 +62,12 @@
 public:
 							TYPEINFO();
 							XFillBmpPosOffsetYItem( USHORT nOffPosY = 0 );
-							XFillBmpPosOffsetYItem( SvStream& rIn );
+							SVX_DLLPRIVATE XFillBmpPosOffsetYItem( SvStream& rIn );
 
-	virtual SfxPoolItem*    Clone( SfxItemPool* pPool = 0 ) const;
-	virtual SfxPoolItem*    Create( SvStream& rIn, USHORT nVer ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Clone( SfxItemPool* pPool = 0 ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Create( SvStream& rIn, USHORT nVer ) const;
 
-	virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
+	SVX_DLLPRIVATE virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
 									SfxMapUnit eCoreMetric,
 									SfxMapUnit ePresMetric,
                                     String &rText, const IntlWrapper * = 0 ) const;
--- svx/inc/svx/xflbtoxy.hxx
+++ svx/inc/svx/xflbtoxy.hxx
@@ -39,12 +39,12 @@
 public:
 							TYPEINFO();
 							XFillBmpTileOffsetXItem( USHORT nOffX = 0 );
-							XFillBmpTileOffsetXItem( SvStream& rIn );
+							SVX_DLLPRIVATE XFillBmpTileOffsetXItem( SvStream& rIn );
 
-	virtual SfxPoolItem*    Clone( SfxItemPool* pPool = 0 ) const;
-	virtual SfxPoolItem*    Create( SvStream& rIn, USHORT nVer ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Clone( SfxItemPool* pPool = 0 ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Create( SvStream& rIn, USHORT nVer ) const;
 
-	virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
+	SVX_DLLPRIVATE virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
 									SfxMapUnit eCoreMetric,
 									SfxMapUnit ePresMetric,
                                     String &rText, const IntlWrapper * = 0 ) const;
@@ -62,12 +62,12 @@
 public:
 							TYPEINFO();
 							XFillBmpTileOffsetYItem( USHORT nOffX = 0 );
-							XFillBmpTileOffsetYItem( SvStream& rIn );
+							SVX_DLLPRIVATE XFillBmpTileOffsetYItem( SvStream& rIn );
 
-	virtual SfxPoolItem*    Clone( SfxItemPool* pPool = 0 ) const;
-	virtual SfxPoolItem*    Create( SvStream& rIn, USHORT nVer ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Clone( SfxItemPool* pPool = 0 ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Create( SvStream& rIn, USHORT nVer ) const;
 
-	virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
+	SVX_DLLPRIVATE virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
 									SfxMapUnit eCoreMetric,
 									SfxMapUnit ePresMetric,
                                     String &rText, const IntlWrapper * = 0 ) const;
--- svx/inc/svx/xsflclit.hxx
+++ svx/inc/svx/xsflclit.hxx
@@ -36,24 +36,24 @@
 //------------------------------
 // class XSecondaryFillColorItem
 //------------------------------
-class XSecondaryFillColorItem : public XColorItem
+class SVX_DLLPUBLIC XSecondaryFillColorItem : public XColorItem
 {
 public:
 			TYPEINFO();
-			XSecondaryFillColorItem() {}
-			XSecondaryFillColorItem(INT32 nIndex, const Color& rTheColor);
+			SVX_DLLPRIVATE XSecondaryFillColorItem() {}
+			SVX_DLLPRIVATE XSecondaryFillColorItem(INT32 nIndex, const Color& rTheColor);
 			XSecondaryFillColorItem(const String& rName, const Color& rTheColor);
-			XSecondaryFillColorItem(SvStream& rIn);
+			SVX_DLLPRIVATE XSecondaryFillColorItem(SvStream& rIn);
 
-	virtual SfxPoolItem*    Clone(SfxItemPool* pPool = 0) const;
-	virtual SfxPoolItem*    Create(SvStream& rIn, USHORT nVer) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Clone(SfxItemPool* pPool = 0) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Create(SvStream& rIn, USHORT nVer) const;
 
-	virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
+	SVX_DLLPRIVATE virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
 									SfxMapUnit eCoreMetric,
 									SfxMapUnit ePresMetric,
                                     String &rText, const IntlWrapper * = 0 ) const;
 
-	virtual USHORT GetVersion( USHORT nFileFormatVersion ) const;
+	SVX_DLLPRIVATE virtual USHORT GetVersion( USHORT nFileFormatVersion ) const;
 
 };
 
