diff --git filter/source/msfilter/msdffimp.cxx filter/source/msfilter/msdffimp.cxx
index b72b0fc..f1b45ca 100644
--- filter/source/msfilter/msdffimp.cxx
+++ filter/source/msfilter/msdffimp.cxx
@@ -5373,8 +5373,9 @@ SdrObject* SvxMSDffManager::ImportShape( const DffRecordHeader& rHd, SvStream& r
                         aSet.Put( SdrEdgeNode2VertDistItem( n2VertDist ) );
 
                         ((SdrEdgeObj*)pRet)->SetEdgeTrackPath( aPoly );
+
+                        pRet->SetMergedItemSet( aSet );
                     }
-                    pRet->SetMergedItemSet( aSet );
                 }
             }
 
