--- svx/source/cui/optgenrl.cxx.orig	2010-02-12 22:41:38.000000000 +0100
+++ svx/source/cui/optgenrl.cxx	2010-03-21 18:30:59.244875000 +0100
@@ -192,6 +192,7 @@
     }
     else if ( LANGUAGE_JAPANESE == eLang ||
           LANGUAGE_KOREAN == eLang ||
+          LANGUAGE_HUNGARIAN == eLang ||
           LANGUAGE_CHINESE_TRADITIONAL == eLang ||
           LANGUAGE_CHINESE_SIMPLIFIED == eLang)
     {
