From c4ddcfeb1c29db07e899b9f0fbeda40b003d6efb Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:08:58 +0200
Subject: [PATCH 843/878] svx-validate-paradepth.diff

---
 svx/source/svdraw/svdfppt.cxx |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/svx/source/svdraw/svdfppt.cxx b/svx/source/svdraw/svdfppt.cxx
index 6b62f71..382640d 100644
--- a/svx/source/svdraw/svdfppt.cxx
+++ b/svx/source/svdraw/svdfppt.cxx
@@ -5096,6 +5096,10 @@ void PPTStyleTextPropReader::ReadParaProps( SvStream& rIn, SdrPowerPointImport&
             rIn >> nCharCount
                 >> aParaPropSet.pParaSet->mnDepth;	// Einruecktiefe
 
+            aParaPropSet.pParaSet->mnDepth =
+                std::min(sal_uInt16(9),
+                         aParaPropSet.pParaSet->mnDepth);
+
             nCharCount--;
 
             rIn >> nMask;
-- 
1.7.0.1

