From 6ddb0d206e945b33cb7750a51074bbf99e1bbc88 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:08:58 +0200
Subject: [PATCH 741/768] svx-validate-paradepth.diff

---
 filter/source/msfilter/svdfppt.cxx |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git filter/source/msfilter/svdfppt.cxx filter/source/msfilter/svdfppt.cxx
index 4191dd4..6a57772 100644
--- filter/source/msfilter/svdfppt.cxx
+++ filter/source/msfilter/svdfppt.cxx
@@ -5087,6 +5087,10 @@ void PPTStyleTextPropReader::ReadParaProps( SvStream& rIn, SdrPowerPointImport&
             rIn >> nCharCount
                 >> aParaPropSet.pParaSet->mnDepth;	// Einruecktiefe
 
+            aParaPropSet.pParaSet->mnDepth =
+                std::min(sal_uInt16(9),
+                         aParaPropSet.pParaSet->mnDepth);
+
             nCharCount--;
 
             rIn >> nMask;
-- 
1.7.0.1

