---
 svx/source/svdraw/svdedxv.cxx |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git svx/source/svdraw/svdedxv.cxx svx/source/svdraw/svdedxv.cxx
index f277e35..ab0148b 100644
--- svx/source/svdraw/svdedxv.cxx
+++ svx/source/svdraw/svdedxv.cxx
@@ -455,8 +455,9 @@ OutlinerView* SdrObjEditView::ImpMakeOutlinerView(Window* pWin, BOOL /*bNoPaint*
         pOutlView->SetAnchorMode((EVAnchorMode)(pText->GetOutlinerViewAnchorMode()));
         pTextEditOutliner->SetFixedCellHeight(((const SdrTextFixedCellHeightItem&)pText->GetMergedItem(SDRATTR_TEXT_USEFIXEDCELLHEIGHT)).GetValue());
     }
-    pOutlView->SetOutputArea(aTextEditArea);
+    // do update before setting output area so that aTextEditArea can be recalculated
     pTextEditOutliner->SetUpdateMode(TRUE);
+    pOutlView->SetOutputArea(aTextEditArea);
     ImpInvalidateOutlinerView(*pOutlView);
     return pOutlView;
 }
-- 
1.7.0.1

