From d8cdb5e8c9f15b64568ef74ee7b67541d467280b Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:09:00 +0200
Subject: [PATCH 845/878] svx-textpropreader-limit-fix.diff

---
 svx/source/svdraw/svdfppt.cxx |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/svx/source/svdraw/svdfppt.cxx b/svx/source/svdraw/svdfppt.cxx
index 382640d..bf18489 100644
--- a/svx/source/svdraw/svdfppt.cxx
+++ b/svx/source/svdraw/svdfppt.cxx
@@ -5333,8 +5333,8 @@ void PPTStyleTextPropReader::Init( SvStream& rIn, SdrPowerPointImport& rMan, con
     DffRecordHeader aTextHd;
     rIn >> aTextHd;
     sal_uInt32 nMaxLen = aTextHd.nRecLen;
-    if ( nMaxLen > 0xFFFF )
-        nMaxLen = 0xFFFF;
+    if ( nMaxLen >= 0xFFFF )
+        nMaxLen = 0xFFFE;
 
     if( aTextHd.nRecType == PPT_PST_TextCharsAtom )
     {
-- 
1.7.0.1

