From 3aa6a040844d603878388cad7f49254293f0064f Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:54:09 +0200
Subject: [PATCH 073/768] svx-ppt-tablerow-height-fix.diff

---
 svx/source/table/cell.cxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git svx/source/table/cell.cxx svx/source/table/cell.cxx
index 3d75c03..e8ea7c7 100644
--- svx/source/table/cell.cxx
+++ svx/source/table/cell.cxx
@@ -653,7 +653,7 @@ sal_Int32 Cell::getMinimumHeight()
         pEditOutliner->SetMaxAutoPaperSize(aSize);
         nMinimumHeight = pEditOutliner->GetTextHeight()+1;
     }
-    else /*if ( hasText() )*/
+    else if( hasText() )
     {
         Outliner& rOutliner=rTableObj.ImpGetDrawOutliner();
         rOutliner.SetPaperSize(aSize);
-- 
1.7.0.1

