From e71e99603e53efc86d974f54154ea78bb97c9681 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:54:36 +0200
Subject: [PATCH 114/878] svx-text-rotation-fix.diff

---
 svx/source/msfilter/escherex.cxx |   23 +++++++++++++++++++++++
 1 files changed, 23 insertions(+), 0 deletions(-)

diff --git a/svx/source/msfilter/escherex.cxx b/svx/source/msfilter/escherex.cxx
index fc9e605..2f23b55 100644
--- a/svx/source/msfilter/escherex.cxx
+++ b/svx/source/msfilter/escherex.cxx
@@ -721,6 +721,29 @@ void EscherPropertyContainer::CreateTextProperties(
 
     if ( nTextId )
         AddOpt( ESCHER_Prop_lTxid, nTextId );
+
+    // n#404221: In case of rotation we need to write the txtflTextFlow
+    // attribute too.
+    if (bIsTextFrame) {
+        sal_uInt16 nAngle = EscherPropertyValueHelper::GetPropertyValue(
+        aAny,
+        rXPropSet,
+        String( RTL_CONSTASCII_USTRINGPARAM( "RotateAngle" ) ),
+        sal_True )
+        ? (sal_uInt16)( ( *((sal_Int32*)aAny.getValue() ) ) + 5 ) / 10 : 0;
+        if (nAngle==900) {
+        AddOpt( ESCHER_Prop_txflTextFlow, 1 );
+        bSuppressRotation=true;
+        }
+        if (nAngle==1800) {
+        AddOpt( ESCHER_Prop_txflTextFlow, 2 );
+        bSuppressRotation=true;
+        }
+        if (nAngle==2700) {
+        AddOpt( ESCHER_Prop_txflTextFlow, 3 );
+        bSuppressRotation=true;
+        }
+    }
 }
 
 sal_Bool EscherPropertyContainer::GetLineArrow( const sal_Bool bLineStart,
-- 
1.7.0.1

