diff --git svx/source/msfilter/msdffimp.cxx svx/source/msfilter/msdffimp.cxx
index dae1295..645c71c 100644
--- svx/source/msfilter/msdffimp.cxx
+++ svx/source/msfilter/msdffimp.cxx
@@ -5427,6 +5427,8 @@ SdrObject* SvxMSDffManager::ImportShape( const DffRecordHeader& rHd, SvStream& r
                         aSet.Put( SdrEdgeNode2VertDistItem( n2VertDist ) );
 
                         ((SdrEdgeObj*)pRet)->SetEdgeTrackPath( aPoly );
+
+                        pRet->SetMergedItemSet( aSet );
                     }
                 }
             }
