--- svx/source/svdraw/svdoashp.cxx
+++ svx/source/svdraw/svdoashp.cxx
@@ -1669,6 +1669,11 @@ void SdrObjCustomShape::TakeObjInfo(SdrObjTransformInfoRec& rInfo) const
                 {
                     rInfo.bCanConvToContour = aInfo.bCanConvToContour;
                 }
+
+                if( !rInfo.bShearAllowed )
+                    rInfo.bShearAllowed=sal_False;
+                if( !aInfo.bEdgeRadiusAllowed )
+                    rInfo.bEdgeRadiusAllowed=sal_False;
             }
         }
     }
