---
 .../customshapes/EnhancedCustomShapeFontWork.cxx   |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git svx/source/customshapes/EnhancedCustomShapeFontWork.cxx svx/source/customshapes/EnhancedCustomShapeFontWork.cxx
index 33d9bef..bd24198 100644
--- svx/source/customshapes/EnhancedCustomShapeFontWork.cxx
+++ svx/source/customshapes/EnhancedCustomShapeFontWork.cxx
@@ -636,7 +636,7 @@ void InsertMissingOutlinePoints( const Polygon& /*rOutlinePoly*/, const std::vec
 void GetPoint( const Polygon& rPoly, const std::vector< double >& rDistances, const double& fX, double& fx1, double& fy1 )
 {
     fy1 = fx1 = 0.0;
-    if ( rPoly.GetSize() )
+    if ( rPoly.GetSize() > 1 )
     {
         std::vector< double >::const_iterator aIter = std::lower_bound( rDistances.begin(), rDistances.end(), fX );
         sal_uInt16 nIdx = sal::static_int_cast<sal_uInt16>( std::distance( rDistances.begin(), aIter ) );
-- 
1.7.0.1

