---
 editeng/source/editeng/impedit3.cxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git editeng/source/editeng/impedit3.cxx editeng/source/editeng/impedit3.cxx
index 0511385..8ed92e7 100644
--- editeng/source/editeng/impedit3.cxx
+++ editeng/source/editeng/impedit3.cxx
@@ -946,7 +946,7 @@ sal_Bool ImpEditEngine::CreateLines( USHORT nPara, sal_uInt32 nStartPosY )
 //						nCurPos -= rLRItem.GetTxtLeft();	// Tabs relativ zu LI
                         // Skalierung rausrechnen
                         if ( aStatus.DoStretch() && ( nStretchX != 100 ) )
-                            nCurPos = nCurPos*100/nStretchX;
+                            nCurPos = nCurPos*100/std::max(static_cast<sal_Int32>(nStretchX), static_cast<sal_Int32>(1));
 
                         short nAllSpaceBeforeText = static_cast< short >(rLRItem.GetTxtLeft()/* + rLRItem.GetTxtLeft()*/ + nSpaceBeforeAndMinLabelWidth);
                         aCurrentTab.aTabStop = pNode->GetContentAttribs().FindTabStop( nCurPos - nAllSpaceBeforeText /*rLRItem.GetTxtLeft()*/, aEditDoc.GetDefTab() );
-- 
1.7.0.1

