From 0e30d556e4f36105775f1c7bd8ca30437f24b171 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:05:10 +0200
Subject: [PATCH 658/878] svx-debug-sdrolestreams.diff

---
 svx/source/msfilter/msdffimp.cxx |   12 ++++++++++++
 1 files changed, 12 insertions(+), 0 deletions(-)

diff --git a/svx/source/msfilter/msdffimp.cxx b/svx/source/msfilter/msdffimp.cxx
index 1275a70..a3a834e 100644
--- a/svx/source/msfilter/msdffimp.cxx
+++ b/svx/source/msfilter/msdffimp.cxx
@@ -7735,6 +7735,18 @@ com::sun::star::uno::Reference < com::sun::star::embed::XEmbeddedObject >  SvxMS
                 pFilter = aMatch.GetFilter4EA( aType );
         }
 
+//#define DBG_EXTRACTOLESTREAMS
+#ifdef DBG_EXTRACTOLESTREAMS
+        static sal_Int32 nCount(0);
+        String aTmpName(String::CreateFromAscii(RTL_CONSTASCII_STRINGPARAM("/tmp/embedded_stream_")));
+        aTmpName += String::CreateFromInt32(nCount++);
+        aTmpName += String::CreateFromAscii(RTL_CONSTASCII_STRINGPARAM(".bin"));
+        SvFileStream aTmpStream(aTmpName,STREAM_READ|STREAM_WRITE|STREAM_TRUNC);
+        pStream->Seek(0);
+        *pStream >> aTmpStream;
+        aTmpStream.Close();
+#endif
+
         if ( pName || pFilter )
         {
             //Reuse current ole name
-- 
1.7.0.1

