--- svx/source/customshapes/makefile.mk.old	2009-10-08 16:17:55.000000000 +0200
+++ svx/source/customshapes/makefile.mk	2010-03-03 16:59:28.000000000 +0100
@@ -48,6 +48,10 @@ ENABLE_EXCEPTIONS=TRUE
 .IF ("$(OS)$(CPU)"=="SOLARISS" && "$(COM)"!="GCC") || "$(OS)"=="MACOSX" || ("$(OS)"=="LINUX" && "$(CPU)"=="P") 
 NOOPTFILES= $(SLO)$/EnhancedCustomShapeFunctionParser.obj
 .ENDIF
+# http://gcc.gnu.org/PR22392
+.IF "$(OS)"=="LINUX" && ("$(CPU)"=="P" || "$(CPU)"=="X")
+NOOPTFILES= $(SLO)$/EnhancedCustomShapeFunctionParser.obj
+.ENDIF
 
 ENVCFLAGS += -DBOOST_SPIRIT_USE_OLD_NAMESPACE
 
