diff --git svx/source/msfilter/msdffimp.cxx svx/source/msfilter/msdffimp.cxx
index 3f023b7..35b6dd0 100644
--- svx/source/msfilter/msdffimp.cxx
+++ svx/source/msfilter/msdffimp.cxx
@@ -5109,7 +5109,6 @@ SdrObject* SvxMSDffManager::ImportShape( const DffRecordHeader& rHd, SvStream& r
                         if ( GetPropertyValue( DFF_Prop_gtextFStrikethrough, 0 ) & 0x1000 )	// SJ: Font Kerning On ?
                             aSet.Put( SvxKerningItem( 1, EE_CHAR_KERNING ) );
                     }
-                    pRet->SetMergedItemSet( aSet );
 
                     // sj: taking care of rtl, ltr. In case of fontwork mso. seems not to be able to set
                     // proper text directions, instead the text default is depending to the string.
@@ -5428,6 +5427,7 @@ SdrObject* SvxMSDffManager::ImportShape( const DffRecordHeader& rHd, SvStream& r
 
                         ((SdrEdgeObj*)pRet)->SetEdgeTrackPath( aPoly );
                     }
+                    pRet->SetMergedItemSet( aSet );
                 }
             }
 
