From 926a9bbe92cf2072361dc43894adebc7c5e88ff2 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:09:00 +0200
Subject: [PATCH 743/768] svx-textpropreader-limit-fix.diff

---
 filter/source/msfilter/svdfppt.cxx |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git filter/source/msfilter/svdfppt.cxx filter/source/msfilter/svdfppt.cxx
index 6a57772..f9b0114 100644
--- filter/source/msfilter/svdfppt.cxx
+++ filter/source/msfilter/svdfppt.cxx
@@ -5324,8 +5324,8 @@ void PPTStyleTextPropReader::Init( SvStream& rIn, SdrPowerPointImport& rMan, con
     DffRecordHeader aTextHd;
     rIn >> aTextHd;
     sal_uInt32 nMaxLen = aTextHd.nRecLen;
-    if ( nMaxLen > 0xFFFF )
-        nMaxLen = 0xFFFF;
+    if ( nMaxLen >= 0xFFFF )
+        nMaxLen = 0xFFFE;
 
     if( aTextHd.nRecType == PPT_PST_TextCharsAtom )
     {
-- 
1.7.0.1

