From 92cc773724b49ef9937e734dbe0d582296aaf5ba Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:06:57 +0200
Subject: [PATCH 737/878] svx-msdff-import-connector-style-fix.diff

---
 svx/source/msfilter/msdffimp.cxx |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/svx/source/msfilter/msdffimp.cxx b/svx/source/msfilter/msdffimp.cxx
index a3a834e..7ad2a4b 100644
--- a/svx/source/msfilter/msdffimp.cxx
+++ b/svx/source/msfilter/msdffimp.cxx
@@ -5424,6 +5424,8 @@ SdrObject* SvxMSDffManager::ImportShape( const DffRecordHeader& rHd, SvStream& r
                         aSet.Put( SdrEdgeNode2VertDistItem( n2VertDist ) );
 
                         ((SdrEdgeObj*)pRet)->SetEdgeTrackPath( aPoly );
+
+                        pRet->SetMergedItemSet( aSet );
                     }
                 }
             }
-- 
1.7.0.1

