From e55414e16288af84188414b39c00635c3bbf61f6 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:54:16 +0200
Subject: [PATCH 096/878] svx-honour-customshape-capabilities.diff

---
 svx/source/svdraw/svdoashp.cxx |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/svx/source/svdraw/svdoashp.cxx b/svx/source/svdraw/svdoashp.cxx
index 9be605e..5656eb3 100644
--- a/svx/source/svdraw/svdoashp.cxx
+++ b/svx/source/svdraw/svdoashp.cxx
@@ -1665,6 +1665,11 @@ void SdrObjCustomShape::TakeObjInfo(SdrObjTransformInfoRec& rInfo) const
                 {
                     rInfo.bCanConvToContour = aInfo.bCanConvToContour;
                 }
+
+                if( !rInfo.bShearAllowed )
+                    rInfo.bShearAllowed=FALSE;
+                if( !aInfo.bEdgeRadiusAllowed )
+                    rInfo.bEdgeRadiusAllowed=FALSE;
             }
         }
     }
-- 
1.7.0.1

