From 80d2b377627ddb9c54861723ef1704ef0463d227 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:09:12 +0200
Subject: [PATCH 749/768] svx-hu-name-order.diff

---
 cui/source/options/optgenrl.cxx |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git cui/source/options/optgenrl.cxx cui/source/options/optgenrl.cxx
index c7310bd..d6681cc 100644
--- cui/source/options/optgenrl.cxx
+++ cui/source/options/optgenrl.cxx
@@ -186,6 +186,7 @@ SvxGeneralTabPage::SvxGeneralTabPage( Window* pParent, const SfxItemSet& rCoreSe
     }
     else if ( LANGUAGE_JAPANESE == eLang ||
           LANGUAGE_KOREAN == eLang ||
+          LANGUAGE_HUNGARIAN == eLang ||
           LANGUAGE_CHINESE_TRADITIONAL == eLang ||
           LANGUAGE_CHINESE_SIMPLIFIED == eLang)
     {
-- 
1.7.0.1

