From 1f2e22f35393311bcddaa0a0f0b8d08c844067ed Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:09:12 +0200
Subject: [PATCH 856/878] svx-hu-name-order.diff

---
 svx/source/cui/optgenrl.cxx |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/svx/source/cui/optgenrl.cxx b/svx/source/cui/optgenrl.cxx
index 8730958..3328ed8 100644
--- a/svx/source/cui/optgenrl.cxx
+++ b/svx/source/cui/optgenrl.cxx
@@ -189,6 +189,7 @@ SvxGeneralTabPage::SvxGeneralTabPage( Window* pParent, const SfxItemSet& rCoreSe
     }
     else if ( LANGUAGE_JAPANESE == eLang ||
           LANGUAGE_KOREAN == eLang ||
+          LANGUAGE_HUNGARIAN == eLang ||
           LANGUAGE_CHINESE_TRADITIONAL == eLang ||
           LANGUAGE_CHINESE_SIMPLIFIED == eLang)
     {
-- 
1.7.0.1

