diff --git a/svx/source/svdraw/svdfppt.cxx b/svx/source/svdraw/svdfppt.cxx
index c452472..c25164a 100644
--- svx/source/svdraw/svdfppt.cxx
+++ svx/source/svdraw/svdfppt.cxx
@@ -5337,7 +5337,7 @@ void PPTStyleTextPropReader::Init( SvStream& rIn, SdrPowerPointImport& rMan, con
     rIn >> aTextHd;
     sal_uInt32 nMaxLen = aTextHd.nRecLen;
     if ( nMaxLen > 0xFFFF )
-        nMaxLen = 0xFFFF;
+        nMaxLen = 0xFFFE;
 
     if( aTextHd.nRecType == PPT_PST_TextCharsAtom )
     {
