From ccea70a1ed2d752852ef19f348b9124b553dacb0 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:54:09 +0200
Subject: [PATCH 090/878] svx-ppt-tablerow-height-fix.diff

---
 svx/source/table/cell.cxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/svx/source/table/cell.cxx b/svx/source/table/cell.cxx
index 7ce2dd8..fe90e3a 100644
--- a/svx/source/table/cell.cxx
+++ b/svx/source/table/cell.cxx
@@ -655,7 +655,7 @@ sal_Int32 Cell::getMinimumHeight()
         pEditOutliner->SetMaxAutoPaperSize(aSize);
         nMinimumHeight = pEditOutliner->GetTextHeight()+1;
     }
-    else /*if ( hasText() )*/
+    else if( hasText() )
     {
         Outliner& rOutliner=rTableObj.ImpGetDrawOutliner();
         rOutliner.SetPaperSize(aSize);
-- 
1.7.0.1

