From 4018cdc2f4650ebf688228fdbe13e7bf597cde90 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:03:58 +0200
Subject: [PATCH 524/768] svx-sdrobjeditview-update-edit-area.diff

---
 svx/source/svdraw/svdedxv.cxx |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git svx/source/svdraw/svdedxv.cxx svx/source/svdraw/svdedxv.cxx
index 054a9e8..f03c226 100644
--- svx/source/svdraw/svdedxv.cxx
+++ svx/source/svdraw/svdedxv.cxx
@@ -455,8 +455,9 @@ OutlinerView* SdrObjEditView::ImpMakeOutlinerView(Window* pWin, BOOL /*bNoPaint*
         pOutlView->SetAnchorMode((EVAnchorMode)(pText->GetOutlinerViewAnchorMode()));
         pTextEditOutliner->SetFixedCellHeight(((const SdrTextFixedCellHeightItem&)pText->GetMergedItem(SDRATTR_TEXT_USEFIXEDCELLHEIGHT)).GetValue());
     }
-    pOutlView->SetOutputArea(aTextEditArea);
+    // do update before setting output area so that aTextEditArea can be recalculated
     pTextEditOutliner->SetUpdateMode(TRUE);
+    pOutlView->SetOutputArea(aTextEditArea);
     ImpInvalidateOutlinerView(*pOutlView);
     return pOutlView;
 }
-- 
1.7.0.1

