From 85992a799c871d01fb438cd4fcf7f42060764e7b Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:54:16 +0200
Subject: [PATCH 078/768] svx-honour-customshape-capabilities.diff

---
 svx/source/svdraw/svdoashp.cxx |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git svx/source/svdraw/svdoashp.cxx svx/source/svdraw/svdoashp.cxx
index 03a7d30..4876ff5 100644
--- svx/source/svdraw/svdoashp.cxx
+++ svx/source/svdraw/svdoashp.cxx
@@ -1666,6 +1666,11 @@ void SdrObjCustomShape::TakeObjInfo(SdrObjTransformInfoRec& rInfo) const
                 {
                     rInfo.bCanConvToContour = aInfo.bCanConvToContour;
                 }
+
+                if( !rInfo.bShearAllowed )
+                    rInfo.bShearAllowed=FALSE;
+                if( !aInfo.bEdgeRadiusAllowed )
+                    rInfo.bEdgeRadiusAllowed=FALSE;
             }
         }
     }
-- 
1.7.0.1

