diff -rup svx-orig/source/svdraw/svdfppt.cxx svx/source/svdraw/svdfppt.cxx
--- svx-orig/source/svdraw/svdfppt.cxx	2009-09-23 11:36:41.000000000 +0200
+++ svx/source/svdraw/svdfppt.cxx	2009-09-23 11:36:51.000000000 +0200
@@ -5021,7 +5021,7 @@ void PPTStyleTextPropReader::ReadParaPro
             {
                 rIn >> aSet.mpArry[ PPT_ParaAttr_BulletHeight ];
                 if ( ! ( ( nMask & ( 1 << PPT_ParaAttr_BuHardHeight ) )
-                         && ( nBulFlg && ( 1 << PPT_ParaAttr_BuHardHeight ) ) ) )
+                         && ( nBulFlg & ( 1 << PPT_ParaAttr_BuHardHeight ) ) ) )
                     aSet.mnAttrSet ^= 0x40;
             }
             if ( nMask & 0x0020 )	// buColor
