diff --git svx/source/msfilter/msdffimp.cxx svx/source/msfilter/msdffimp.cxx
index 3f023b7..3bc067e 100644
--- svx/source/msfilter/msdffimp.cxx
+++ svx/source/msfilter/msdffimp.cxx
@@ -5428,6 +5428,7 @@ SdrObject* SvxMSDffManager::ImportShape( const DffRecordHeader& rHd, SvStream& r
 
                         ((SdrEdgeObj*)pRet)->SetEdgeTrackPath( aPoly );
                     }
+                    pRet->SetMergedItemSet( aSet );
                 }
             }
 
