From 47a91c5833d8926b25850d966804b31d857546d0 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:06:57 +0200
Subject: [PATCH 646/768] svx-msdff-import-connector-style-fix.diff

---
 filter/source/msfilter/msdffimp.cxx |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git filter/source/msfilter/msdffimp.cxx filter/source/msfilter/msdffimp.cxx
index 0859abb..374ea81 100644
--- filter/source/msfilter/msdffimp.cxx
+++ filter/source/msfilter/msdffimp.cxx
@@ -5373,6 +5373,8 @@ SdrObject* SvxMSDffManager::ImportShape( const DffRecordHeader& rHd, SvStream& r
                         aSet.Put( SdrEdgeNode2VertDistItem( n2VertDist ) );
 
                         ((SdrEdgeObj*)pRet)->SetEdgeTrackPath( aPoly );
+
+                        pRet->SetMergedItemSet( aSet );
                     }
                     pRet->SetMergedItemSet( aSet );
                 }
-- 
1.7.0.1

