From e3bd748aa99a3d95310b9230db5b671b2e5dd508 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:58:31 +0200
Subject: [PATCH 284/768] svx-customshapes-NOOPTFILES.diff

---
 svx/source/customshapes/makefile.mk |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git svx/source/customshapes/makefile.mk svx/source/customshapes/makefile.mk
index 3feca8f..ece0edb 100644
--- svx/source/customshapes/makefile.mk
+++ svx/source/customshapes/makefile.mk
@@ -44,6 +44,10 @@ ENABLE_EXCEPTIONS=TRUE
 .IF ("$(OS)$(CPU)"=="SOLARISS" && "$(COM)"!="GCC") || "$(OS)"=="MACOSX" || ("$(OS)"=="LINUX" && "$(CPU)"=="P") 
 NOOPTFILES= $(SLO)$/EnhancedCustomShapeFunctionParser.obj
 .ENDIF
+# http://gcc.gnu.org/PR22392
+.IF "$(OS)"=="LINUX" && ("$(CPU)"=="P" || "$(CPU)"=="X")
+NOOPTFILES= $(SLO)$/EnhancedCustomShapeFunctionParser.obj
+.ENDIF
 
 ENVCFLAGS += -DBOOST_SPIRIT_USE_OLD_NAMESPACE
 
-- 
1.7.0.1

