From 7af612f97809d1149ae0b55cff7a375466717948 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:06:25 +0200
Subject: [PATCH 626/768] svx-fix-fit-to-frame-crash.diff

---
 editeng/source/editeng/impedit3.cxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git editeng/source/editeng/impedit3.cxx editeng/source/editeng/impedit3.cxx
index 363ea2d..2c2be37 100644
--- editeng/source/editeng/impedit3.cxx
+++ editeng/source/editeng/impedit3.cxx
@@ -946,7 +946,7 @@ sal_Bool ImpEditEngine::CreateLines( USHORT nPara, sal_uInt32 nStartPosY )
 //						nCurPos -= rLRItem.GetTxtLeft();	// Tabs relativ zu LI
                         // Skalierung rausrechnen
                         if ( aStatus.DoStretch() && ( nStretchX != 100 ) )
-                            nCurPos = nCurPos*100/nStretchX;
+                            nCurPos = nCurPos*100/std::max(static_cast<sal_Int32>(nStretchX), static_cast<sal_Int32>(1));
 
                         short nAllSpaceBeforeText = static_cast< short >(rLRItem.GetTxtLeft()/* + rLRItem.GetTxtLeft()*/ + nSpaceBeforeAndMinLabelWidth);
                         aCurrentTab.aTabStop = pNode->GetContentAttribs().FindTabStop( nCurPos - nAllSpaceBeforeText /*rLRItem.GetTxtLeft()*/, aEditDoc.GetDefTab() );
-- 
1.7.0.1

