From b112a2f1ca0b8d65e043f08ae4731caabd02ec82 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:03:58 +0200
Subject: [PATCH 597/878] svx-sdrobjeditview-update-edit-area.diff

---
 svx/source/svdraw/svdedxv.cxx |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/svx/source/svdraw/svdedxv.cxx b/svx/source/svdraw/svdedxv.cxx
index 53936c6..c665a35 100644
--- a/svx/source/svdraw/svdedxv.cxx
+++ b/svx/source/svdraw/svdedxv.cxx
@@ -451,8 +451,9 @@ OutlinerView* SdrObjEditView::ImpMakeOutlinerView(Window* pWin, BOOL /*bNoPaint*
         pOutlView->SetAnchorMode((EVAnchorMode)(pText->GetOutlinerViewAnchorMode()));
         pTextEditOutliner->SetFixedCellHeight(((const SdrTextFixedCellHeightItem&)pText->GetMergedItem(SDRATTR_TEXT_USEFIXEDCELLHEIGHT)).GetValue());
     }
-    pOutlView->SetOutputArea(aTextEditArea);
+    // do update before setting output area so that aTextEditArea can be recalculated
     pTextEditOutliner->SetUpdateMode(TRUE);
+    pOutlView->SetOutputArea(aTextEditArea);
     ImpInvalidateOutlinerView(*pOutlView);
     return pOutlView;
 }
-- 
1.7.0.1

