diff -r 03068afc23c6 sw/source/core/unocore/unosett.cxx
--- sw/source/core/unocore/unosett.cxx	Sun Apr 25 15:43:24 2010 +0100
+++ sw/source/core/unocore/unosett.cxx	Tue Apr 27 16:55:25 2010 +0100
@@ -1659,9 +1659,8 @@
 				aPropertyValues.Insert(pData, aPropertyValues.Count());
 			}
 		 	Size aSize = rFmt.GetGraphicSize();
-			aSize.Width() = TWIP_TO_MM100( aSize.Width() );
-			aSize.Height() = TWIP_TO_MM100( aSize.Height() );
-			pData = new PropValData((void*)&aSize, SW_PROP_NAME_STR(UNO_NAME_GRAPHIC_SIZE), ::getCppuType((const awt::Size*)0));
+		 	awt::Size aAwtSize(TWIP_TO_MM100(aSize.Width()), TWIP_TO_MM100(aSize.Height()));
+			pData = new PropValData((void*)&aAwtSize, SW_PROP_NAME_STR(UNO_NAME_GRAPHIC_SIZE), ::getCppuType((const awt::Size*)0));
 			aPropertyValues.Insert(pData, aPropertyValues.Count());
 
 			const SwFmtVertOrient* pOrient = rFmt.GetGraphicOrientation();
