diff -wrup ../ooo320-m12-orig/sw/source/ui/inc/view.hxx sw/source/ui/inc/view.hxx
--- ../ooo320-m12-orig/sw/source/ui/inc/view.hxx	2010-03-18 16:50:37.000000000 +0100
+++ sw/source/ui/inc/view.hxx	2010-03-17 11:35:22.000000000 +0100
@@ -330,6 +330,7 @@ class SW_DLLPUBLIC SwView: public SfxVie
     SW_DLLPRIVATE int               _CreateScrollbar( BOOL bHori );
     SW_DLLPRIVATE DECL_LINK( ScrollHdl, SwScrollbar * );
     SW_DLLPRIVATE DECL_LINK( EndScrollHdl, SwScrollbar * );
+    SW_DLLPRIVATE DECL_LINK( WindowChildEventListener, VclSimpleEvent* );
     SW_DLLPRIVATE BOOL			UpdateScrollbars();
     SW_DLLPRIVATE void			CalcVisArea( const Size &rPixelSz );
 
diff -wrup ../ooo320-m12-orig/sw/source/ui/uiview/viewmdi.cxx sw/source/ui/uiview/viewmdi.cxx
--- ../ooo320-m12-orig/sw/source/ui/uiview/viewmdi.cxx	2009-12-18 13:40:10.000000000 +0100
+++ sw/source/ui/uiview/viewmdi.cxx	2010-03-18 19:00:49.000000000 +0100
@@ -312,6 +312,35 @@ void SwView::SetViewLayout( USHORT nColu
  * Scrollbar - Handler
  */
 
+IMPL_LINK( SwView, WindowChildEventListener, VclSimpleEvent*, pEvent )
+{
+    DBG_ASSERT( pEvent && pEvent->ISA( VclWindowEvent ), "Unknown WindowEvent!" );
+    if ( pEvent && pEvent->ISA( VclWindowEvent ) )
+    {
+        VclWindowEvent *pVclEvent = static_cast< VclWindowEvent * >( pEvent );
+        DBG_ASSERT( pVclEvent->GetWindow(), "Window???" );
+    Window* pChildWin = static_cast< Window* >( pVclEvent->GetData() );
+
+        switch ( pVclEvent->GetId() )
+        {
+        case VCLEVENT_WINDOW_HIDE:
+      if( pChildWin == pHScrollbar )
+        ShowHScrollbar( FALSE );
+      else if( pChildWin == pVScrollbar )
+        ShowVScrollbar( FALSE );
+      break;
+        case VCLEVENT_WINDOW_SHOW:
+      if( pChildWin == pHScrollbar )
+        ShowHScrollbar( TRUE );
+      else if( pChildWin == pVScrollbar )
+        ShowVScrollbar( TRUE );
+      break;
+    }
+    }
+
+    return 0;
+}
+
 int SwView::_CreateScrollbar( BOOL bHori )
 {
     Window *pMDI = &GetViewFrame()->GetWindow();
@@ -332,6 +361,10 @@ int SwView::_CreateScrollbar( BOOL bHori
 
     (*ppScrollbar)->EnableDrag( TRUE );
 
+    (*ppScrollbar)->SetAuto( TRUE );
+
+    pMDI->AddChildEventListener( LINK( this, SwView, WindowChildEventListener ));
+
     if(GetWindow())
         InvalidateBorder();
 
diff -wrup ../ooo320-m12-orig/sw/source/ui/uiview/viewport.cxx sw/source/ui/uiview/viewport.cxx
--- ../ooo320-m12-orig/sw/source/ui/uiview/viewport.cxx	2010-03-18 16:49:58.000000000 +0100
+++ sw/source/ui/uiview/viewport.cxx	2010-03-18 13:06:43.000000000 +0100
@@ -1139,6 +1139,7 @@ void SwView::OuterResizePixel( const Poi
             {
                 bShowH = pVOpt->IsViewHScrollBar();
                 bShowV = pVOpt->IsViewVScrollBar();
+        bAuto = bHAuto = TRUE;
                 break;
             }
         }
@@ -1160,10 +1161,10 @@ void SwView::OuterResizePixel( const Poi
     {
         bShowH = bShowV = bHAuto = bAuto = FALSE;
     }
-    if(pHScrollbar->IsVisible(FALSE) != bShowH)
+    if(pHScrollbar->IsVisible(FALSE) != bShowH && !bHAuto)
         ShowHScrollbar(bShowH);
     pHScrollbar->SetAuto( bHAuto );
-    if(pVScrollbar->IsVisible(FALSE) != bShowV)
+    if(pVScrollbar->IsVisible(FALSE) != bShowV && !bAuto)
         ShowVScrollbar(bShowV);
     pVScrollbar->SetAuto(bAuto);
 
