diff -rup sw-M/source/ui/uiview/view.cxx sw/source/ui/uiview/view.cxx
--- sw-M/source/ui/uiview/view.cxx	2010-03-18 23:13:14.000000000 +0100
+++ sw/source/ui/uiview/view.cxx	2010-03-22 18:41:07.000000000 +0100
@@ -1129,6 +1129,10 @@ SwView::SwView( SfxViewFrame *_pFrame, S
         pDocSh->EnableSetModified( sal_True );
     InvalidateBorder();
 
+    if( !pHScrollbar->IsVisible( TRUE ) )
+        ShowHScrollbar( FALSE );
+    if( !pVScrollbar->IsVisible( TRUE ) )
+        ShowVScrollbar( FALSE );
 }
 
 /*--------------------------------------------------------------------
