diff -rup sw/source/ui/uiview-orig/view.cxx sw/source/ui/uiview/view.cxx
--- sw/source/ui/uiview-orig/view.cxx	2010-03-25 16:38:12.000000000 +0100
+++ sw/source/ui/uiview/view.cxx	2010-03-25 16:49:08.000000000 +0100
@@ -1133,6 +1133,8 @@ SwView::SwView( SfxViewFrame *_pFrame, S
       	    ShowHScrollbar( FALSE );
     if( !pVScrollbar->IsVisible( TRUE ) )
       	    ShowVScrollbar( FALSE );
+
+    GetViewFrame()->GetWindow().AddChildEventListener( LINK( this, SwView, WindowChildEventListener ) );
 }
 
 /*--------------------------------------------------------------------
@@ -1142,6 +1144,7 @@ SwView::SwView( SfxViewFrame *_pFrame, S
 
 SwView::~SwView()
 {
+    GetViewFrame()->GetWindow().RemoveChildEventListener( LINK( this, SwView, WindowChildEventListener ) );
     delete mpPostItMgr;
     // --> OD 2009-03-10 #i100035#
     mpPostItMgr = 0;
diff -rup sw/source/ui/uiview-orig/viewmdi.cxx sw/source/ui/uiview/viewmdi.cxx
--- sw/source/ui/uiview-orig/viewmdi.cxx	2010-03-25 16:38:12.000000000 +0100
+++ sw/source/ui/uiview/viewmdi.cxx	2010-03-25 16:39:31.000000000 +0100
@@ -363,8 +363,6 @@ int SwView::_CreateScrollbar( BOOL bHori
 
     (*ppScrollbar)->SetAuto( TRUE );
 
-    pMDI->AddChildEventListener( LINK( this, SwView, WindowChildEventListener ));
-
     if(GetWindow())
         InvalidateBorder();
 
