--- sw/source/core/doc/doclay.cxx.orig	2010-02-12 22:29:13.000000000 +0100
+++ sw/source/core/doc/doclay.cxx	2010-03-20 16:03:33.169875000 +0100
@@ -1474,7 +1474,10 @@
                 aTxt += ' ';
         }
         xub_StrLen nIdx = aTxt.Len();
-        aTxt += rSeparator;
+        if ( rTxt.Len() > 0 )
+        {
+            aTxt += rSeparator;
+        }
         xub_StrLen nSepIdx = aTxt.Len();
         aTxt += rTxt;
 
--- sw/source/ui/frmdlg/cption.cxx.orig	2010-03-11 22:05:24.000000000 +0100
+++ sw/source/ui/frmdlg/cption.cxx	2010-03-29 22:33:03.494932817 +0200
@@ -385,6 +385,7 @@
 void SwCaptionDialog::DrawSample()
 {
     String aStr;
+    String sCaption = aTextEdit.GetText();
 
     // Nummer
     String sFldTypeName = aCategoryBox.GetText();
@@ -439,9 +440,12 @@
             }
 
         }
-        aStr += aSepEdit.GetText();
+        if( sCaption.Len() > 0 )
+        {
+            aStr += aSepEdit.GetText();
+        }
     }
-    aStr += aTextEdit.GetText();
+    aStr += sCaption;
     // do preview!
     aPrevWin.SetPreviewText( aStr );
 }
