--- sw/source/ui/dochdl/swdtflvr.cxx	2010-09-16 08:28:23.000000000 +0200
+++ sw/source/ui/dochdl/swdtflvr.cxx	2010-09-16 08:28:18.000000000 +0200
@@ -2222,7 +2222,7 @@ int SwTransferable::_PasteDDE( Transfera
                 }
 
                 rWrtShell.InsertDDETable(
-                    SwInsertTableOptions( tabopts::HEADLINE_NO_BORDER, 1 ), // TODO MULTIHEADER
+                    SwInsertTableOptions( tabopts::SPLIT_LAYOUT, 1 ), // TODO MULTIHEADER
                     pDDETyp, nRows, nCols );
             }
             else if( 1 < aExpand.GetTokenCount( '\n' ) )
