--- sw/source/ui/table/convert.src.orig	2009-01-16 22:22:15.000000000 +0100
+++ sw/source/ui/table/convert.src	2009-01-16 22:27:09.000000000 +0100
@@ -36,7 +36,7 @@
 {
 	HelpID = FN_CONVERT_TEXT_TABLE ;
 	OUTPUTSIZE = TRUE ;
-    Size = MAP_APPFONT ( 203 , 169 ) ;
+    Size = MAP_APPFONT ( 223 , 169 ) ;
 	Text [ en-US ] = "Convert Table to Text" ;
 	MOVEABLE = TRUE ;
 	SVLOOK = TRUE ;
@@ -158,26 +158,26 @@
 	OkButton BT_OK
 	{
         Pos = MAP_APPFONT ( 147 , 6 ) ;
-		Size = MAP_APPFONT ( 50 , 14 ) ;
+		Size = MAP_APPFONT ( 70 , 14 ) ;
 		TABSTOP = TRUE ;
 		DefButton = TRUE ;
 	};
 	CancelButton BT_CANCEL
 	{
         Pos = MAP_APPFONT ( 147 , 23 ) ;
-		Size = MAP_APPFONT ( 50 , 14 ) ;
+		Size = MAP_APPFONT ( 70 , 14 ) ;
 		TABSTOP = TRUE ;
 	};
 	HelpButton BT_HELP
 	{
         Pos = MAP_APPFONT ( 147 , 43 ) ;
-		Size = MAP_APPFONT ( 50 , 14 ) ;
+		Size = MAP_APPFONT ( 70 , 14 ) ;
 		TABSTOP = TRUE ;
 	};
 	PushButton BT_AUTOFORMAT
 	{
         Pos = MAP_APPFONT ( 12 , 152 ) ;
-		Size = MAP_APPFONT ( 66 , 14 ) ;
+		Size = MAP_APPFONT ( 70 , 14 ) ;
 		TabStop = TRUE ;
 		Hide = TRUE ;
 		Text [ en-US ] = "Auto~Format..." ;
--- sw/source/ui/table/colwd.src
+++ sw/source/ui/table/colwd.src
@@ -63,14 +63,14 @@
     FixedText FT_WIDTH
     {
         Pos = MAP_APPFONT ( 12 , 31 ) ;
-        Size = MAP_APPFONT ( 35 , 8 ) ;
+        Size = MAP_APPFONT ( 41 , 8 ) ;
         Text [ en-US ] = "~Width" ;
         Left = TRUE ;
     };
     MetricField ED_WIDTH
     {
         Border = TRUE ;
-        Pos = MAP_APPFONT ( 51 , 30 ) ;
+        Pos = MAP_APPFONT ( 54 , 30 ) ;
         Size = MAP_APPFONT ( 38 , 12 ) ;
         TabStop = TRUE ;
         Left = TRUE ;
@@ -101,7 +101,7 @@
     NumericField ED_COL
     {
         Border = TRUE ;
-        Pos = MAP_APPFONT ( 51 , 14 ) ;
+        Pos = MAP_APPFONT ( 54 , 14 ) ;
         Size = MAP_APPFONT ( 38 , 12 ) ;
         TabStop = TRUE ;
         Left = TRUE ;
--- sw/source/ui/config/optdlg.src
+++ sw/source/ui/config/optdlg.src
@@ -60,7 +60,7 @@
     CheckBox CB_CROSS
     {
         Pos = MAP_APPFONT ( 12 , 14 ) ;
-        Size = MAP_APPFONT ( 118 , 10 ) ;
+        Size = MAP_APPFONT ( 126 , 10 ) ;
         Text [ en-US ] = "Guides ~while moving";
     };
     CheckBox CB_HANDLE
