diff --git sw/source/ui/shells/textsh1.cxx sw/source/ui/shells/textsh1.cxx
index 1d2fe1b..6457c36 100644
--- sw/source/ui/shells/textsh1.cxx
+++ sw/source/ui/shells/textsh1.cxx
@@ -1100,27 +1100,12 @@ void SwTextShell::Execute(SfxRequest &rReq)
                 SvxColorItem aItem(aSet, RES_CHRATR_COLOR);
 
                 // besteht eine Selektion, wird sie gleich gefaerbt
-                if(!pApply && rWrtSh.HasSelection())
-                {
-                    rWrtSh.SetAttr(SvxColorItem (aSet, RES_CHRATR_COLOR));
-                }
-                else if(!pApply || pApply->nColor != SID_ATTR_CHAR_COLOR_EXT)
+                if(!pApply || pApply->nColor != SID_ATTR_CHAR_COLOR_EXT)
                 {
-                    GetView().GetViewFrame()->GetDispatcher()->Execute(SID_ATTR_CHAR_COLOR_EXT);
+                    rWrtSh.SetAttr(aItem);
                 }
 
                 rReq.Done();
-/*		 		OS 22.02.97 18:40 Das alte Verhalten ist unerwuenscht
-                SwEditWin& rEdtWin = GetView().GetEditWin();
-
-                SwApplyTemplate* pApply = rEdtWin.GetApplyTemplate();
-                SvxColorItem aItem(aSet, RES_CHRATR_COLOR);
-
-                if(!pApply || pApply->nColor != SID_ATTR_CHAR_COLOR_EXT)
-                {
-                    GetShell().SetAttr(aItem);
-                }
-*/
             }
         }
         break;
