diff --git sw/source/filter/rtf/rtfatr.cxx sw/source/filter/rtf/rtfatr.cxx
index bbe9030..7f191ad 100644
--- sw/source/filter/rtf/rtfatr.cxx
+++ sw/source/filter/rtf/rtfatr.cxx
@@ -1223,7 +1223,7 @@ static Writer& OutRTF_SwTxtNode( Writer& rWrt, SwCntntNode& rNode )
     }
 
     // gibt es harte Attributierung ?
-    if( bNewFmts && pNd->HasSwAttrSet())
+    if( pNd->HasSwAttrSet() )
     {
         rRTFWrt.pFlyFmt = 0;
 
@@ -1246,7 +1246,7 @@ static Writer& OutRTF_SwTxtNode( Writer& rWrt, SwCntntNode& rNode )
             SfxItemSet aSet( rNdSet );
             SvxLRSpaceItem aLR( (SvxLRSpaceItem&)rNdSet.Get( RES_LR_SPACE ) );
 
-            aLR.SetTxtLeft( aLR.GetTxtLeft() + pFmt->GetAbsLSpace() );
+            aLR.SetTxtLeft( aLR.GetTxtLeft() + pFmt->GetIndentAt() );
             if( MAXLEVEL > pNd->GetActualListLevel() )
                 aLR.SetTxtFirstLineOfst( pFmt->GetFirstLineOffset() );
             else
