diff --git sw/source/filter/basflt/shellio.cxx sw/source/filter/basflt/shellio.cxx
index 3d547ea..15d6877 100644
--- sw/source/filter/basflt/shellio.cxx
+++ sw/source/filter/basflt/shellio.cxx
@@ -206,6 +206,16 @@ ULONG SwReader::Read( const Reader& rOptions )
             else
                 nEndCntnt = nLen - nEndCntnt;
             pPam->GetPoint()->nContent.Assign( pCNd, nEndCntnt );
+
+            const SwStartNode* pTblBoxStart = pCNd->FindTableBoxStartNode();
+            if ( pTblBoxStart )
+            {
+                SwTableBox* pBox = pTblBoxStart->GetTblBox();
+                if ( pBox )
+                {
+                    pDoc->ChkBoxNumFmt( *pBox, TRUE );
+                }
+            }
         }
 
         if( pCrsr )
