--- sw/source/core/layout/tabfrm.cxx.old	2009-10-08 16:17:18.000000000 +0200
+++ sw/source/core/layout/tabfrm.cxx	2010-03-03 17:00:02.000000000 +0100
@@ -2135,6 +2135,11 @@ void SwTabFrm::MakeAll()
                            pPre->GetAttrSet()->GetKeep().GetValue()) )
             {
                 bCalcLowers = TRUE;
+                // --> OD 2009-03-06 #i99267#
+                // reset <bSplit> after forward move to assure that follows
+                // can be joined, if further space is available.
+                bSplit = FALSE;
+                // <--
             }
         }
 
