diff --git sw/source/core/unocore/unostyle.cxx sw/source/core/unocore/unostyle.cxx
index 6033541..a935395 100644
--- sw/source/core/unocore/unostyle.cxx
+++ sw/source/core/unocore/unostyle.cxx
@@ -1266,6 +1266,7 @@ sal_Bool SwStyleProperties_Impl::GetProperty(const ::rtl::OUString& rName, uno::
             bRet = sal_True;
             break;
         }
+        ++nPos;
         ++aIt;
     }
     
