--- /home/cbosdonnat/go-oo/ooo-build/build/ooo310-m16/sw/source/filter/ww8/ww8par5.cxx	2009-08-04 15:54:05.000000000 +0200
+++ ww8par5.cxx	2009-08-14 10:56:23.000000000 +0200
@@ -994,9 +994,12 @@
         pStrm->Seek( nOldPos );
 
         //#124725# field codes which contain '/' or '.' are not displayed in WinWord
-        if (!aStr.EqualsAscii(" ADDIN", 0, 6) &&
-            (aStr.Search('.') != STRING_NOTFOUND ||
-             aStr.Search('/') != STRING_NOTFOUND))
+        xub_StrLen nSpacePos = aStr.Search( ' ', 1 );
+        if ( STRING_NOTFOUND == nSpacePos )
+            nSpacePos = aStr.Len( );
+        xub_StrLen nSearchPos = STRING_NOTFOUND;
+        if ( ( ( nSearchPos = aStr.Search('.') ) != STRING_NOTFOUND && nSearchPos < nSpacePos ) ||
+             ( ( nSearchPos = aStr.Search('/') ) != STRING_NOTFOUND && nSearchPos < nSpacePos ) )
             return aF.nLen;
         else
             return aF.nLen - aF.nLRes - 1;  // so viele ueberlesen, das Resultfeld
