diff --git sw/source/filter/ww8/ww8scan.cxx sw/source/filter/ww8/ww8scan.cxx
index f4707a5..250fdbd 100644
--- sw/source/filter/ww8/ww8scan.cxx
+++ sw/source/filter/ww8/ww8scan.cxx
@@ -35,6 +35,7 @@
 
 
 #include <functional>
+#include <algorithm>
 
 #include <string.h>         // memset()
 #include <rtl/tencinfo.h>
@@ -6402,8 +6403,10 @@ WW8Fonts::WW8Fonts( SvStream& rSt, WW8Fib& rFib )
 #ifdef __WW8_NEEDS_COPY
                 {
                     BYTE nLen = 0x28;
+                    BYTE nLength = sizeof( pVer8->szFfn ) / sizeof( SVBT16 );
+                    nLength = std::min( nLength, BYTE( pVer8->cbFfnM1+1 ) );
                     for( UINT16* pTmp = pVer8->szFfn;
-                        nLen < pVer8->cbFfnM1 + 1 ; ++pTmp, nLen+=2 )
+                        nLen < nLength; ++pTmp, nLen+=2 )
                     {
                         *pTmp = SVBT16ToShort( *(SVBT16*)pTmp );
                     }
