diff --git sw/source/filter/ww8/wrtw8nds.cxx sw/source/filter/ww8/wrtw8nds.cxx
index f207dd8..57e4920 100644
--- sw/source/filter/ww8/wrtw8nds.cxx
+++ sw/source/filter/ww8/wrtw8nds.cxx
@@ -36,6 +36,7 @@
 #include <functional>
 #include <iostream>
 #include <hintids.hxx>
+#include <comphelper/processfactory.hxx>
 #include <tools/urlobj.hxx>
 #include <editeng/boxitem.hxx>
 #include <editeng/cmapitem.hxx>
@@ -50,8 +51,10 @@
 #include <editeng/brkitem.hxx>
 #include <editeng/frmdiritem.hxx>
 #include <editeng/tstpitem.hxx>
+#include <editeng/unolingu.hxx>
 #include "svl/urihelper.hxx"
 #include <svl/whiter.hxx>
+#include <unotools/localedatawrapper.hxx>
 #include <fmtpdsc.hxx>
 #include <fmtfsize.hxx>
 #include <fmtornt.hxx>
@@ -109,6 +112,7 @@
 
 using namespace ::com::sun::star;
 using namespace ::com::sun::star::i18n;
+using namespace ::com::sun::star::lang;
 using namespace sw::util;
 using namespace sw::types;
 using namespace sw::mark;
@@ -739,7 +743,17 @@ void WW8AttributeOutput::StartRuby( const SwTxtNode& rNode, xub_StrLen /*nPos*/,
     aStr += String::CreateFromInt32(nHeight);
     aStr += '(';
     aStr += rRuby.GetText();
-    aStr.APPEND_CONST_ASC( ");" );
+    aStr.APPEND_CONST_ASC( ")" );
+
+    // The parameter separator depends on the FIB.lid
+    Locale aTempLocale;
+    SvxLanguageToLocale( aTempLocale, m_rWW8Export.pFib->lid );
+    LocaleDataWrapper aLocaleWrapper( ::comphelper::getProcessServiceFactory(), aTempLocale );
+    if ( aLocaleWrapper.getNumDecimalSep().EqualsAscii( "." ) )
+        aStr.APPEND_CONST_ASC( "," );
+    else
+        aStr.APPEND_CONST_ASC( ";" );
+
     m_rWW8Export.OutputField( 0, ww::eEQ, aStr,
             WRITEFIELD_START | WRITEFIELD_CMD_START );
 }
