diff --git sw/source/filter/ww8/writerwordglue.cxx sw/source/filter/ww8/writerwordglue.cxx
index 55e6868..3a77426 100644
--- sw/source/filter/ww8/writerwordglue.cxx
+++ sw/source/filter/ww8/writerwordglue.cxx
@@ -162,7 +162,7 @@ namespace myImplHelpers
             RES_NONE, RES_NONE, RES_NONE, RES_POOLCOLL_LISTS_BEGIN,
             RES_NONE, RES_NONE, RES_NONE, RES_NONE, RES_NONE, RES_NONE,
             RES_NONE, RES_NONE, RES_NONE, RES_NONE, RES_NONE, RES_NONE,
-            RES_NONE, RES_NONE, RES_POOLCOLL_DOC_TITEL, RES_NONE,
+            RES_NONE, RES_NONE, RES_POOLCOLL_HEADLINE_BASE, RES_NONE,
             RES_POOLCOLL_SIGNATURE, RES_NONE, RES_POOLCOLL_TEXT,
             RES_POOLCOLL_TEXT_MOVE, RES_NONE, RES_NONE, RES_NONE, RES_NONE,
             RES_NONE, RES_NONE, RES_POOLCOLL_DOC_SUBTITEL
