diff --git sw/source/filter/ww8/ww8graf.cxx sw/source/filter/ww8/ww8graf.cxx
index 2853bcc..add9cd2 100644
--- sw/source/filter/ww8/ww8graf.cxx
+++ sw/source/filter/ww8/ww8graf.cxx
@@ -1906,6 +1906,7 @@ void SwWW8ImplReader::MatchSdrItemsIntoFlySet( SdrObject* pSdrObj,
                 }
             break;
             case XFILL_SOLID:
+            case XFILL_GRADIENT:
                 {
                     const Color aColor = static_cast< XFillColorItem const & >(
                         rOldSet.Get(XATTR_FILLCOLOR)).GetColorValue();
@@ -1917,8 +1918,8 @@ void SwWW8ImplReader::MatchSdrItemsIntoFlySet( SdrObject* pSdrObj,
                     bBrushItemOk = true;
                 }
             break;
-            case XFILL_GRADIENT:
-            break;
+            //case XFILL_GRADIENT:
+            //break;
             case XFILL_HATCH:
             break;
             case XFILL_BITMAP:
