diff --git extras/source/truetype/symbol/OpenSymbol.sfd extras/source/truetype/symbol/OpenSymbol.sfd
index e04c03f..d6ecc95 100644
--- extras/source/truetype/symbol/OpenSymbol.sfd
+++ extras/source/truetype/symbol/OpenSymbol.sfd
@@ -13,13 +13,14 @@ Descent: 410
 LayerCount: 2
 Layer: 0 1 "Back"  1
 Layer: 1 1 "Fore"  0
+NeedsXUIDChange: 1
 XUID: [1021 161 2043615882 15846768]
 FSType: 8
 OS2Version: 0
 OS2_WeightWidthSlopeOnly: 0
 OS2_UseTypoMetrics: 1
 CreationTime: 1144938807
-ModificationTime: 1238168413
+ModificationTime: 1281980608
 PfmFamily: 81
 TTFWeight: 400
 TTFWidth: 5
@@ -591,8 +592,8 @@ NameList: Adobe Glyph List
 DisplaySize: -24
 AntiAlias: 1
 FitToEm: 1
-WinInfo: 0 45 18
-BeginChars: 65539 912
+WinInfo: 135 45 18
+BeginChars: 65539 914
 
 StartChar: .notdef
 Encoding: 65536 -1 0
@@ -100162,5 +100163,88 @@ SplineSet
 EndSplineSet
 Validated: 1
 EndChar
+
+StartChar: uni27FC
+Encoding: 10236 10236 912
+Width: 2048
+VWidth: 0
+Flags: W
+LayerCount: 2
+Fore
+SplineSet
+120.605 893.365 m 0,0,1
+ 130.745 894.649 130.745 894.649 139.122 886.971 c 128,-1,2
+ 147.499 879.293 147.499 879.293 147.099 869.08 c 2,3,-1
+ 147.099 657.135 l 1,4,-1
+ 1753.62 657.135 l 1,5,6
+ 1721.36 678.613 1721.36 678.613 1692.21 706.222 c 128,-1,7
+ 1663.06 733.83 1663.06 733.83 1644.5 757.892 c 128,-1,8
+ 1625.94 781.953 1625.94 781.953 1611.99 802.947 c 128,-1,9
+ 1598.04 823.94 1598.04 823.94 1591.94 836.207 c 2,10,-1
+ 1585.83 848.475 l 1,11,12
+ 1580.32 857.07 1580.32 857.07 1583.64 867.92 c 128,-1,13
+ 1586.96 878.77 1586.96 878.77 1596.33 882.81 c 128,-1,14
+ 1605.71 886.85 1605.71 886.85 1615.88 881.814 c 128,-1,15
+ 1626.05 876.778 1626.05 876.778 1628.51 866.872 c 1,16,17
+ 1631.31 860.546 1631.31 860.546 1636.84 849.729 c 128,-1,18
+ 1642.37 838.911 1642.37 838.911 1662.49 809.124 c 128,-1,19
+ 1682.61 779.338 1682.61 779.338 1706.09 753.681 c 128,-1,20
+ 1729.56 728.024 1729.56 728.024 1768.64 699.899 c 128,-1,21
+ 1807.71 671.774 1807.71 671.774 1850.76 656.4 c 1,22,23
+ 1865.45 654.098 1865.45 654.098 1865.48 633.585 c 0,24,25
+ 1865.49 624.275 1865.49 624.275 1861.81 618.755 c 128,-1,26
+ 1858.13 613.236 1858.13 613.236 1854.45 612.371 c 2,27,-1
+ 1850.76 611.507 l 1,28,29
+ 1807.71 596.133 1807.71 596.133 1768.99 568.798 c 128,-1,30
+ 1730.26 541.464 1730.26 541.464 1705.74 513.434 c 128,-1,31
+ 1681.21 485.405 1681.21 485.405 1662.84 459.572 c 128,-1,32
+ 1644.47 433.739 1644.47 433.739 1636.49 417.386 c 2,33,-1
+ 1628.51 401.033 l 1,34,35
+ 1626.05 391.127 1626.05 391.127 1615.88 386.09 c 128,-1,36
+ 1605.71 381.053 1605.71 381.053 1596.33 385.093 c 128,-1,37
+ 1586.96 389.134 1586.96 389.134 1583.64 399.985 c 128,-1,38
+ 1580.32 410.837 1580.32 410.837 1585.83 419.432 c 0,39,40
+ 1587.95 424.066 1587.95 424.066 1592.14 432.201 c 128,-1,41
+ 1596.33 440.337 1596.33 440.337 1611.48 464.117 c 128,-1,42
+ 1626.63 487.897 1626.63 487.897 1644.23 510.113 c 128,-1,43
+ 1661.82 532.328 1661.82 532.328 1690.99 560.441 c 128,-1,44
+ 1720.15 588.555 1720.15 588.555 1752.15 610.037 c 1,45,-1
+ 147.099 610.037 l 1,46,-1
+ 147.099 398.091 l 2,47,48
+ 147.231 388.76 147.231 388.76 140.056 381.483 c 128,-1,49
+ 132.882 374.206 132.882 374.206 123.55 374.206 c 128,-1,50
+ 114.217 374.206 114.217 374.206 107.042 381.483 c 128,-1,51
+ 99.8674 388.76 99.8674 388.76 100 398.091 c 2,52,-1
+ 100 629.171 l 2,53,54
+ 99.1577 633.586 99.1577 633.586 100 638.002 c 2,55,-1
+ 100 869.082 l 2,56,57
+ 99.6602 877.819 99.6602 877.819 105.795 885.049 c 128,-1,58
+ 111.93 892.279 111.93 892.279 120.605 893.365 c 0,0,1
+EndSplineSet
+EndChar
+
+StartChar: uni2312
+Encoding: 8978 8978 913
+Width: 760
+VWidth: 0
+Flags: W
+LayerCount: 2
+Fore
+SplineSet
+30.002 1123.25 m 1,0,1
+ 30.002 1218.06 30.002 1218.06 76.9414 1298.72 c 128,-1,2
+ 123.88 1379.38 123.88 1379.38 204.536 1426.31 c 128,-1,3
+ 285.192 1473.25 285.192 1473.25 380.001 1473.25 c 128,-1,4
+ 474.809 1473.25 474.809 1473.25 555.465 1426.31 c 128,-1,5
+ 636.121 1379.38 636.121 1379.38 683.061 1298.72 c 128,-1,6
+ 730 1218.06 730 1218.06 730 1123.25 c 1,7,-1
+ 643.486 1123.25 l 1,8,9
+ 643.486 1232.78 643.486 1232.78 566.506 1309.76 c 128,-1,10
+ 489.525 1386.74 489.525 1386.74 380.001 1386.74 c 128,-1,11
+ 270.477 1386.74 270.477 1386.74 193.496 1309.76 c 128,-1,12
+ 116.515 1232.78 116.515 1232.78 116.515 1123.25 c 1,13,-1
+ 30.002 1123.25 l 1,0,1
+EndSplineSet
+EndChar
 EndChars
 EndSplineFont
diff --git extras/source/truetype/symbol/opens___.ttf extras/source/truetype/symbol/opens___.ttf
index b4f169f..b7fb394 100644
Binary files extras/source/truetype/symbol/opens___.ttf and extras/source/truetype/symbol/opens___.ttf differ
diff --git officecfg/registry/data/org/openoffice/Office/Math.xcu officecfg/registry/data/org/openoffice/Office/Math.xcu
index cf11e57..b185902 100644
--- officecfg/registry/data/org/openoffice/Office/Math.xcu
+++ officecfg/registry/data/org/openoffice/Office/Math.xcu
@@ -994,5 +994,19 @@
         <value>Id1</value>
       </prop>
     </node>
+    <node oor:name="mapsto" oor:op="replace">
+      <prop oor:name="Char">
+        <value>10236</value>
+      </prop>
+      <prop oor:name="Set">
+        <value>Special</value>
+      </prop>
+      <prop oor:name="Predefined">
+        <value>true</value>
+      </prop>
+      <prop oor:name="FontFormatId">
+        <value>Id1</value>
+      </prop>
+    </node>
   </node>
 </oor:component-data>
diff --git starmath/inc/parse.hxx starmath/inc/parse.hxx
index 0211d97..b462abb 100644
--- starmath/inc/parse.hxx
+++ starmath/inc/parse.hxx
@@ -106,7 +106,7 @@ enum SmTokenType
     TLEFTARROW,		TRIGHTARROW,	TUPARROW,		TDOWNARROW,		TDIVIDES,
     TNDIBVIDES,		TSETN,			TSETZ,			TSETQ,			TSETR,
     TSETC,			TWIDEVEC,		TWIDETILDE,		TWIDEHAT,		TWIDESLASH,
-    TWIDEBACKSLASH, TLDBRACKET,		TRDBRACKET,
+    TWIDEBACKSLASH, TLDBRACKET,		TRDBRACKET,     TBOW,
     TUNKNOWN,		TDEBUG
 };
 
diff --git starmath/inc/types.hxx starmath/inc/types.hxx
index b8cb4d0..4596877 100644
--- starmath/inc/types.hxx
+++ starmath/inc/types.hxx
@@ -165,6 +165,7 @@ enum MathSymbol
     MS_FORALL       = (sal_Unicode) 0x2200,
 
     MS_HAT          = (sal_Unicode) 0xE091,
+    MS_BOW          = (sal_Unicode) 0x2312,
     MS_CHECK        = (sal_Unicode) 0xE092,
     MS_BREVE        = (sal_Unicode) 0xE093,
     MS_ACUTE        = (sal_Unicode) 0xE094,
diff --git starmath/source/parse.cxx starmath/source/parse.cxx
index aab3502..3fc2912 100644
--- starmath/source/parse.cxx
+++ starmath/source/parse.cxx
@@ -202,7 +202,7 @@ static const SmTokenTableEntry aTokenTable[] =
     { "infinity" , TINFINITY, MS_INFINITY, TGSTANDALONE, 5},
     { "infty" , TINFINITY, MS_INFINITY, TGSTANDALONE, 5},
     { "int", TINT, MS_INT, TGOPER, 5},
-    { "intersection", TINTERSECT, MS_INTERSECT, TGPRODUCT, 0},
+    { "intersection", TINTERSECT, MS_INTERSECT, TGPRODUCT | TGOPER, 0},
     { "ital", TITALIC, '\0', TGFONTATTR, 5},
     { "italic", TITALIC, '\0', TGFONTATTR, 5},
     { "lambdabar" , TLAMBDABAR, MS_LAMBDABAR, TGSTANDALONE, 5},
@@ -312,13 +312,14 @@ static const SmTokenTableEntry aTokenTable[] =
     { "transr", TTRANSR, MS_TRANSR, TGRELATION, 0},
     { "underbrace", TUNDERBRACE, MS_UNDERBRACE, TGPRODUCT, 5},
     { "underline", TUNDERLINE, '\0', TGATTRIBUT, 5},
-    { "union", TUNION, MS_UNION, TGSUM, 0},
+    { "union", TUNION, MS_UNION, TGSUM | TGOPER, 0},
     { "uoper", TUOPER, '\0', TGUNOPER, 5},
     { "uparrow" , TUPARROW, MS_UPARROW, TGSTANDALONE, 5},
     { "vec", TVEC, MS_VEC, TGATTRIBUT, 5},
     { "white", TWHITE, '\0', TGCOLOR, 0},
     { "widebslash", TWIDEBACKSLASH, MS_BACKSLASH, TGPRODUCT, 0 },
     { "widehat", TWIDEHAT, MS_HAT, TGATTRIBUT, 5},
+    { "bow", TBOW, MS_BOW, TGATTRIBUT, 5},
     { "widetilde", TWIDETILDE, MS_TILDE, TGATTRIBUT, 5},
     { "wideslash", TWIDESLASH, MS_SLASH, TGPRODUCT, 0 },
     { "widevec", TWIDEVEC, MS_VEC, TGATTRIBUT, 5},
@@ -1624,6 +1625,8 @@ void SmParser::Oper()
 
     switch (eType)
     {
+        case TINTERSECT:
+        case TUNION :
         case TSUM :
         case TPROD :
         case TCOPROD :
@@ -1781,6 +1784,7 @@ void SmParser::Attribut()
 
         case TWIDEVEC :
         case TWIDEHAT :
+        case TBOW :
         case TWIDETILDE :
             pAttr = new SmMathSymbolNode(CurToken);
             eScaleMode = SCALE_WIDTH;
diff --git starmath/source/symbol.src starmath/source/symbol.src
index c4cab66..ed0fb9b 100644
--- starmath/source/symbol.src
+++ starmath/source/symbol.src
@@ -268,6 +268,7 @@ Resource RID_LOCALIZED_NAMES
             < "perthousand" ; > ;
             < "and" ; > ;
             < "or" ; > ;
+            < "mapsto" ; > ;
         };
     };
     StringArray RID_UI_SYMBOL_NAMES
@@ -340,6 +341,7 @@ Resource RID_LOCALIZED_NAMES
         < "perthousand" ; > ;
         < "and" ; > ;
         < "or" ; > ;
+        < "mapsto" ; > ;
     };
     };
 };
