diff --git a/configure.in b/configure.in
--- configure.in
+++ configure.in
@@ -517,6 +517,9 @@
 AC_ARG_WITH(system-boost,
 [  --with-system-boost     Use boost already on system
 ],,)
+AC_ARG_WITH(system-mdds,
+[  --with-system-mdds      Use mdds already on system
+],,)
 AC_ARG_WITH(system-vigra,
 [  --with-system-vigra     Use vigra already on system
 ],,)
@@ -4215,6 +4218,25 @@
 AC_SUBST(CURL_LIBS)
 
 dnl ===================================================================
+dnl Check for system mdds
+dnl ===================================================================
+AC_MSG_CHECKING([which mdds to use])
+if test -n "$with_system_mdds" -o -n "$with_system_headers" && \
+	test "$with_system_mdds" != "no"; then
+    AC_MSG_RESULT([external])
+    SYSTEM_MDDS=YES
+    AC_LANG_PUSH([C++])
+    AC_CHECK_HEADER(mdds/flat_segment_tree.hpp, [],
+        [AC_MSG_ERROR(mdds/flat_segment_tree.hpp not found. install mdds)], [])
+    AC_LANG_POP([C++])
+else
+   AC_MSG_RESULT([internal])
+   BUILD_TYPE="$BUILD_TYPE MDDS"
+   SYSTEM_MDDS=NO
+fi
+AC_SUBST(SYSTEM_MDDS)
+
+dnl ===================================================================
 dnl Check for system boost
 dnl ===================================================================
 AC_MSG_CHECKING([which boost to use])
diff --git a/set_soenv.in b/set_soenv.in
--- set_soenv.in
+++ set_soenv.in
@@ -1920,6 +1920,7 @@
 ToFile( "CURL_CFLAGS",       "@CURL_CFLAGS@",      "e" );
 ToFile( "CURL_LIBS",         "@CURL_LIBS@",        "e" );
 ToFile( "SYSTEM_BOOST",      "@SYSTEM_BOOST@",     "e" );
+ToFile( "SYSTEM_MDDS",       "@SYSTEM_MDDS@",      "e" );
 ToFile( "SYSTEM_VIGRA",      "@SYSTEM_VIGRA@",     "e" );
 ToFile( "SYSTEM_NEON",       "@SYSTEM_NEON@",      "e" );
 ToFile( "NEON_VERSION",      "@NEON_VERSION@",     "e" );
diff --git a/solenv/config/sooo320.ini b/solenv/config/sooo320.ini
--- solenv/config/sooo320.ini
+++ solenv/config/sooo320.ini
@@ -13,7 +13,7 @@
 			BMP_WRITES_FLAG TRUE
 			BTARGET 1
 			BUILD_SPECIAL TRUE
-			BUILD_TYPE SO OOo EXT BINFILTER BITSTREAM_VERA_FONTS BSH CURL DICTIONARIES HSQLDB HUNSPELL JPEG LIBXML2 LIBXMLSEC LPSOLVE MOZ NEON TWAIN PYTHON ZLIB SANE UNIXODBC X11_EXTENSIONS LIBWPD EPM QADEVOOO ODK MSFONTEXTRACT MATHMLDTD BOOST EXPAT CRASHREP BERKELEYDB LIBXSLT SUN AGG GTK ICU SYSTRAY_GTK JAVAINSTALLER2 VIGRA OPENSSL JFREEREPORT APACHE_COMMONS TOMCAT REPORTBUILDER SDEXT SWEXT XPDF LUCENE REDLAND SAXON WRITER2LATEX NSS L10N GRAPHITE MYSQLCPPCONN MYSQLC
+			BUILD_TYPE SO OOo EXT BINFILTER BITSTREAM_VERA_FONTS BSH CURL DICTIONARIES HSQLDB HUNSPELL JPEG LIBXML2 LIBXMLSEC LPSOLVE MOZ NEON TWAIN PYTHON ZLIB SANE UNIXODBC X11_EXTENSIONS LIBWPD EPM QADEVOOO ODK MSFONTEXTRACT MATHMLDTD BOOST MDDS EXPAT CRASHREP BERKELEYDB LIBXSLT SUN AGG GTK ICU SYSTRAY_GTK JAVAINSTALLER2 VIGRA OPENSSL JFREEREPORT APACHE_COMMONS TOMCAT REPORTBUILDER SDEXT SWEXT XPDF LUCENE REDLAND SAXON WRITER2LATEX NSS L10N GRAPHITE MYSQLCPPCONN MYSQLC
 			common_build TRUE
 			COMMON_OUTDIR common
 			CONFIG_PROJECT config_office
