diff --git a/configure.in b/configure.in
--- configure.in
+++ configure.in
@@ -5185,6 +5185,7 @@
     # some systems need this. Like Ubuntu....
     AC_CHECK_LIB(m, floor)
     AC_CHECK_LIB(dl, dlopen)
+    AC_CHECK_LIB(colamd, colamd)
     AC_CHECK_LIB(lpsolve55, make_lp, ,
         [ AC_MSG_ERROR(lpsolve library not found or too old.)], [])
 else
diff --git a/sccomp/source/solver/makefile.mk b/sccomp/source/solver/makefile.mk
--- sccomp/source/solver/makefile.mk
+++ sccomp/source/solver/makefile.mk
@@ -58,6 +58,12 @@
                 $(TOOLSLIB)         \
                 $(LPSOLVELIB)
 
+# DebianBaseOnly; we build lp_solve without colamd included but with system-colamd
+# (which is not linked into the .a...) so we need -lcolamd here, too
+.IF "$(SYSTEM_LPSOLVE)" == "YES"
+SHL1STDLIBS += -lcolamd
+.ENDIF
+
 SHL1DEPN=       makefile.mk
 SHL1DEF=        $(MISC)$/$(SHL1TARGET).def
 SHL1VERSIONMAP= $(SOLARENV)/src/component.map
