--- wizards/com/sun/star/wizards/table/ScenarioSelector.java.orig	2009-08-09 09:21:24.000000000 +0400
+++ wizards/com/sun/star/wizards/table/ScenarioSelector.java	2009-08-09 09:23:37.000000000 +0400
@@ -67,6 +67,9 @@
 public class ScenarioSelector extends FieldSelection implements XItemListener, XFieldSelectionListener
 {
 
+    final static int PRIVATE = 0;
+    final static int BUSINESS = 1;
+    
     XFixedText lblExplanation;
     XFixedText lblCategories;
     XRadioButton optBusiness;
@@ -171,18 +174,18 @@
         {
             e.printStackTrace(System.out);
         }
-        initializeCategory(0);
+        initializeCategory(BUSINESS);
     }
 
     public int getCategory()
     {
         if (optBusiness.getState())
         {
-            curcategory = 0;
+            curcategory = BUSINESS;
         }
         else
         {
-            curcategory = 1;
+            curcategory = PRIVATE;
         }
         return curcategory;
     }
@@ -191,11 +194,11 @@
     {
         if (optBusiness.getState())
         {
-            initializeCategory(0);
+            initializeCategory(BUSINESS);
         }
         else
         {
-            initializeCategory(1);
+            initializeCategory(PRIVATE);
         }
     }
 
