From 076c44a28a2279611141a2067433c1560b7081ca Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:58:03 +0200
Subject: [PATCH 261/768] tango-prefer-over-industrial.diff

---
 vcl/source/app/settings.cxx |   10 +++++++++-
 1 files changed, 9 insertions(+), 1 deletions(-)

diff --git vcl/source/app/settings.cxx vcl/source/app/settings.cxx
index 4bd3cca..fb2f0d3 100644
--- vcl/source/app/settings.cxx
+++ vcl/source/app/settings.cxx
@@ -752,7 +752,15 @@ void StyleSettings::SetPreferredSymbolsStyleName( const ::rtl::OUString &rName )
 
         for( sal_uInt32 n = 0; n <= STYLE_SYMBOLS_THEMES_MAX; n++ )
             if ( rNameLowCase.indexOf( ImplSymbolsStyleToName( n ) ) != -1 )
-                SetPreferredSymbolsStyle( n );
+            {
+                if (n == STYLE_SYMBOLS_INDUSTRIAL)
+                    // The Industrial OOo icon theme is too old.
+                    // The OOo Tango icon theme is closer to the current GNOME Industrial theme rules.
+                    SetPreferredSymbolsStyle( STYLE_SYMBOLS_TANGO );
+                else
+                    SetPreferredSymbolsStyle( n );
+                return;
+            }
     }
 }
 
-- 
1.7.0.1

