---
 vcl/source/app/settings.cxx |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git vcl/source/app/settings.cxx vcl/source/app/settings.cxx
index 22ca7dc..467fdd1 100755
--- vcl/source/app/settings.cxx
+++ vcl/source/app/settings.cxx
@@ -825,7 +825,8 @@ ULONG StyleSettings::GetAutoSymbolsStyle() const
 
     if( bCont )
     {
-        if( rDesktopEnvironment.equalsIgnoreAsciiCaseAscii( "gnome" ) )
+        if( rDesktopEnvironment.equalsIgnoreAsciiCaseAscii( "gnome" ) ||
+         rDesktopEnvironment.equalsIgnoreAsciiCaseAscii( "windows" ) )
             nRet = STYLE_SYMBOLS_TANGO;
         else if( rDesktopEnvironment.equalsIgnoreAsciiCaseAscii( "kde" ) )
             nRet = STYLE_SYMBOLS_CRYSTAL;
-- 
1.7.0.1

