From 5fa12a017f860e9f43b7f9de6880e4530a32439d Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:58:52 +0200
Subject: [PATCH 341/878] tango-icons-default-w32.diff

---
 vcl/source/app/settings.cxx |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/vcl/source/app/settings.cxx b/vcl/source/app/settings.cxx
index 6d0636a..92fb0cc 100644
--- a/vcl/source/app/settings.cxx
+++ b/vcl/source/app/settings.cxx
@@ -815,7 +815,8 @@ ULONG StyleSettings::GetAutoSymbolsStyle() const
 
     if( bCont )
     {
-        if( rDesktopEnvironment.equalsIgnoreAsciiCaseAscii( "gnome" ) )
+        if( rDesktopEnvironment.equalsIgnoreAsciiCaseAscii( "gnome" ) ||
+         rDesktopEnvironment.equalsIgnoreAsciiCaseAscii( "windows" ) )
             nRet = STYLE_SYMBOLS_TANGO;
         else if( rDesktopEnvironment.equalsIgnoreAsciiCaseAscii( "kde" ) )
             nRet = STYLE_SYMBOLS_CRYSTAL;
-- 
1.7.0.1

