From 40eeb58fca6d8a046c4a27de7c5c7b75bc218e4b Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:58:52 +0200
Subject: [PATCH 304/768] tango-icons-default-w32.diff

---
 vcl/source/app/settings.cxx |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git vcl/source/app/settings.cxx vcl/source/app/settings.cxx
index 987abbe..7e8a4c1 100644
--- vcl/source/app/settings.cxx
+++ vcl/source/app/settings.cxx
@@ -825,7 +825,8 @@ ULONG StyleSettings::GetAutoSymbolsStyle() const
 
     if( bCont )
     {
-        if( rDesktopEnvironment.equalsIgnoreAsciiCaseAscii( "gnome" ) )
+        if( rDesktopEnvironment.equalsIgnoreAsciiCaseAscii( "gnome" ) ||
+         rDesktopEnvironment.equalsIgnoreAsciiCaseAscii( "windows" ) )
             nRet = STYLE_SYMBOLS_TANGO;
         else if( rDesktopEnvironment.equalsIgnoreAsciiCaseAscii( "kde" ) )
             nRet = STYLE_SYMBOLS_CRYSTAL;
-- 
1.7.0.1

