diff --git vcl/inc/vcl/brdwin.hxx vcl/inc/vcl/brdwin.hxx
index b6fcb85..3de3671 100644
--- vcl/inc/vcl/brdwin.hxx
+++ vcl/inc/vcl/brdwin.hxx
@@ -84,7 +84,8 @@ class ImplBorderWindowView;
 #define BORDERWINDOW_TITLE_NORMAL			((USHORT)0x0001)
 #define BORDERWINDOW_TITLE_SMALL			((USHORT)0x0002)
 #define BORDERWINDOW_TITLE_TEAROFF			((USHORT)0x0004)
-#define BORDERWINDOW_TITLE_NONE 			((USHORT)0x0008)
+#define BORDERWINDOW_TITLE_POPUP			((USHORT)0x0008)
+#define BORDERWINDOW_TITLE_NONE 			((USHORT)0x0010)
 
 // --------------------
 // - ImplBorderWindow -
diff --git vcl/inc/vcl/floatwin.hxx vcl/inc/vcl/floatwin.hxx
index e6bb97c..b315910 100644
--- vcl/inc/vcl/floatwin.hxx
+++ vcl/inc/vcl/floatwin.hxx
@@ -65,7 +65,8 @@ class PopupModeEvent;
 
 #define FLOATWIN_TITLE_NORMAL					((USHORT)0x0001)
 #define FLOATWIN_TITLE_TEAROFF					((USHORT)0x0002)
-#define FLOATWIN_TITLE_NONE 					((USHORT)0x0004)
+#define FLOATWIN_TITLE_POPUP					((USHORT)0x0004)
+#define FLOATWIN_TITLE_NONE 					((USHORT)0x0008)
 
 // ------------------
 // - FloatingWindow -
diff --git vcl/inc/vcl/wintypes.hxx vcl/inc/vcl/wintypes.hxx
index 6abe87d..02f4a4a 100644
--- vcl/inc/vcl/wintypes.hxx
+++ vcl/inc/vcl/wintypes.hxx
@@ -190,9 +190,10 @@ typedef sal_Int64 WinBits;
 #define WB_INTROWIN             ((WinBits)SAL_CONST_INT64(0x200000000))
 #define WB_NOSHADOW             ((WinBits)SAL_CONST_INT64(0x400000000))
 #define WB_TOOLTIPWIN           ((WinBits)SAL_CONST_INT64(0x800000000))
-#define WB_OWNERDRAWDECORATION  ((WinBits)SAL_CONST_INT64(0x2000000000))
+#define WB_OWNERDRAWDECORATION  ((WinBits)SAL_CONST_INT64(0x10000000))
 #define WB_DEFAULTWIN           ((WinBits)SAL_CONST_INT64(0x4000000000))
 #define WB_NEEDSFOCUS           ((WinBits)SAL_CONST_INT64(0x1000000000))
+#define WB_POPUP                ((WinBits)SAL_CONST_INT64(0x20000000))
 
 #define WB_HSCROLL				WB_HORZ
 #define WB_VSCROLL				WB_VERT
@@ -271,6 +272,7 @@ typedef sal_Int64 WinBits;
 #define WB_STDMODAL 			(WB_STDDIALOG)
 #define WB_STDTABDIALOG 		(WB_STDDIALOG)
 #define WB_STDTABCONTROL		0
+#define WB_STDPOPUP				(WB_BORDER | WB_POPUP | WB_SYSTEMWINDOW | WB_OWNERDRAWDECORATION | WB_3DLOOK | WB_DIALOGCONTROL)
 
 // For TreeListBox
 #define WB_HASBUTTONS			((WinBits)0x00800000)
diff --git vcl/source/app/settings.cxx vcl/source/app/settings.cxx
index 6424fe8..04004a1 100644
--- vcl/source/app/settings.cxx
+++ vcl/source/app/settings.cxx
@@ -562,8 +562,8 @@ void ImplStyleData::SetStandardStyles()
     maPushButtonFont            = aStdFont;
     maFieldFont                 = aStdFont;
     maIconFont                  = aStdFont;
-    maFloatTitleFont            = aStdFont;
     aStdFont.SetWeight( WEIGHT_BOLD );
+    maFloatTitleFont            = aStdFont;
     maTitleFont                 = aStdFont;
 
     maFaceColor                 = Color( COL_LIGHTGRAY );
diff --git vcl/source/window/brdwin.cxx vcl/source/window/brdwin.cxx
index 4bebcd9..9b0d610 100644
--- vcl/source/window/brdwin.cxx
+++ vcl/source/window/brdwin.cxx
@@ -211,7 +211,7 @@ void ImplBorderWindowView::ImplInitTitle( ImplBorderFrameData* pData )
 {
     ImplBorderWindow* pBorderWindow = pData->mpBorderWindow;
 
-    if ( !(pBorderWindow->GetStyle() & WB_MOVEABLE) ||
+    if ( !(pBorderWindow->GetStyle() & (WB_MOVEABLE | WB_POPUP)) ||
           (pData->mnTitleType == BORDERWINDOW_TITLE_NONE) )
     {
         pData->mnTitleType	 = BORDERWINDOW_TITLE_NONE;
@@ -1495,7 +1495,7 @@ void ImplStdBorderWindowView::Init( OutputDevice* pDev, long nWidth, long nHeigh
     pData->mnTitleType		= pBorderWindow->mnTitleType;
     pData->mbFloatWindow	= pBorderWindow->mbFloatWindow;
 
-    if ( !(pBorderWindow->GetStyle() & WB_MOVEABLE) || (pData->mnTitleType == BORDERWINDOW_TITLE_NONE) )
+    if ( !(pBorderWindow->GetStyle() & (WB_MOVEABLE | WB_POPUP)) || (pData->mnTitleType == BORDERWINDOW_TITLE_NONE) )
         pData->mnBorderSize = 0;
     else if ( pData->mnTitleType == BORDERWINDOW_TITLE_TEAROFF )
         pData->mnBorderSize = 0;
@@ -1520,7 +1520,7 @@ void ImplStdBorderWindowView::Init( OutputDevice* pDev, long nWidth, long nHeigh
 
         // set a proper background for drawing
         // highlighted buttons in the title
-        pBorderWindow->SetBackground( rStyleSettings.GetWindowColor() );
+        pBorderWindow->SetBackground( rStyleSettings.GetFaceColor() );
 
         pData->maTitleRect.Left()	 = pData->mnLeftBorder;
         pData->maTitleRect.Right()	 = nWidth-pData->mnRightBorder-1;
@@ -1670,12 +1670,35 @@ void ImplStdBorderWindowView::DrawWindow( USHORT nDrawFlags, OutputDevice* pOutD
     // Draw Frame
     if ( nDrawFlags & BORDERWINDOW_DRAW_FRAME )
     {
+        Region oldClipRgn( pDev->GetClipRegion( ) );
+
+        // for popups, don't draw part of the frame
+        if ( pData->mnTitleType == BORDERWINDOW_TITLE_POPUP )
+        {
+            FloatingWindow *pWin = dynamic_cast< FloatingWindow* >( pData->mpBorderWindow->GetWindow( WINDOW_CLIENT ) );
+            if ( pWin )
+            {
+                Region aClipRgn( aInRect );
+                Rectangle aItemClipRect( pWin->ImplGetItemEdgeClipRect() );
+                if( !aItemClipRect.IsEmpty() )
+                {
+                    aItemClipRect.SetPos( pData->mpBorderWindow->AbsoluteScreenToOutputPixel( aItemClipRect.TopLeft() ) );
+                    aClipRgn.Exclude( aItemClipRect );
+                    pDev->SetClipRegion( aClipRgn );
+                }
+            }
+        }
+
         // single line frame
         pDev->SetLineColor( aFrameColor );
         pDev->SetFillColor();
         pDev->DrawRect( aInRect );
         aInRect.nLeft++; aInRect.nRight--;
         aInRect.nTop++; aInRect.nBottom--;
+
+        // restore
+        if ( pData->mnTitleType == BORDERWINDOW_TITLE_POPUP )
+            pDev->SetClipRegion( oldClipRgn );
     }
     else
         aInRect = aDecoView.DrawFrame( aInRect, FRAME_DRAW_DOUBLEOUT | FRAME_DRAW_NODRAW);
@@ -1707,8 +1730,9 @@ void ImplStdBorderWindowView::DrawWindow( USHORT nDrawFlags, OutputDevice* pOutD
         Rectangle aTitleRect( pData->maTitleRect );
         if( pOffset )
             aTitleRect.Move( pOffset->X(), pOffset->Y() );
-        pDev->DrawRect( aTitleRect );
 
+        if ( pData->mnTitleType != BORDERWINDOW_TITLE_POPUP )
+            pDev->DrawRect( aTitleRect );
 
         if ( pData->mnTitleType != BORDERWINDOW_TITLE_TEAROFF )
         {
@@ -1837,7 +1861,7 @@ void ImplBorderWindow::ImplInit( Window* pParent,
 {
     // Alle WindowBits entfernen, die wir nicht haben wollen
     WinBits nOrgStyle = nStyle;
-    WinBits nTestStyle = (WB_MOVEABLE | WB_SIZEABLE | WB_ROLLABLE | WB_PINABLE | WB_CLOSEABLE | WB_STANDALONE | WB_DIALOGCONTROL | WB_NODIALOGCONTROL | WB_SYSTEMFLOATWIN | WB_INTROWIN | WB_DEFAULTWIN | WB_TOOLTIPWIN | WB_NOSHADOW | WB_OWNERDRAWDECORATION | WB_SYSTEMCHILDWINDOW  | WB_NEEDSFOCUS);
+    WinBits nTestStyle = (WB_MOVEABLE | WB_SIZEABLE | WB_ROLLABLE | WB_PINABLE | WB_CLOSEABLE | WB_STANDALONE | WB_DIALOGCONTROL | WB_NODIALOGCONTROL | WB_SYSTEMFLOATWIN | WB_INTROWIN | WB_DEFAULTWIN | WB_TOOLTIPWIN | WB_NOSHADOW | WB_OWNERDRAWDECORATION | WB_SYSTEMCHILDWINDOW  | WB_NEEDSFOCUS | WB_POPUP);
     if ( nTypeStyle & BORDERWINDOW_STYLE_APP )
         nTestStyle |= WB_APP;
     nStyle &= nTestStyle;
diff --git vcl/source/window/floatwin.cxx vcl/source/window/floatwin.cxx
index 0d39900..117a9fd 100644
--- vcl/source/window/floatwin.cxx
+++ vcl/source/window/floatwin.cxx
@@ -138,7 +138,7 @@ void FloatingWindow::ImplInit( Window* pParent, WinBits nStyle )
     mpNextFloat             = NULL;
     mpFirstPopupModeWin     = NULL;
     mnPostId                = 0;
-    mnTitle                 = (nStyle & WB_MOVEABLE) ? FLOATWIN_TITLE_NORMAL : FLOATWIN_TITLE_NONE;
+    mnTitle                 = (nStyle & (WB_MOVEABLE | WB_POPUP)) ? FLOATWIN_TITLE_NORMAL : FLOATWIN_TITLE_NONE;
     mnOldTitle              = mnTitle;
     mnPopupModeFlags        = 0;
     mbInPopupMode           = FALSE;
@@ -652,6 +652,8 @@ void FloatingWindow::SetTitleType( USHORT nTitle )
             nTitleStyle = BORDERWINDOW_TITLE_SMALL;
         else if ( nTitle == FLOATWIN_TITLE_TEAROFF )
             nTitleStyle = BORDERWINDOW_TITLE_TEAROFF;
+        else if ( nTitle == FLOATWIN_TITLE_POPUP )
+            nTitleStyle = BORDERWINDOW_TITLE_POPUP;
         else // nTitle == FLOATWIN_TITLE_NONE
             nTitleStyle = BORDERWINDOW_TITLE_NONE;
         ((ImplBorderWindow*)mpWindowImpl->mpBorderWindow)->SetTitleType( nTitleStyle, aOutSize );
@@ -673,7 +675,12 @@ void FloatingWindow::StartPopupMode( const Rectangle& rRect, ULONG nFlags )
     // remove title 
     mnOldTitle = mnTitle;
     if ( nFlags & FLOATWIN_POPUPMODE_ALLOWTEAROFF )
-        SetTitleType( FLOATWIN_TITLE_TEAROFF );
+    {
+        if ( mpWindowImpl->mnStyle & WB_POPUP )
+            SetTitleType( FLOATWIN_TITLE_POPUP );
+        else
+            SetTitleType( FLOATWIN_TITLE_TEAROFF );
+    }
     else
         SetTitleType( FLOATWIN_TITLE_NONE );
 
diff --git vcl/unx/gtk/gdi/salnativewidgets-gtk.cxx vcl/unx/gtk/gdi/salnativewidgets-gtk.cxx
index 1d163d9..3f1afb3 100644
--- vcl/unx/gtk/gdi/salnativewidgets-gtk.cxx
+++ vcl/unx/gtk/gdi/salnativewidgets-gtk.cxx
@@ -3407,8 +3407,6 @@ void GtkSalGraphics::updateSettings( AllSettings& rSettings )
 
     aStyleSet.SetAppFont( aFont );
     aStyleSet.SetHelpFont( aFont );
-    aStyleSet.SetTitleFont( aFont );
-    aStyleSet.SetFloatTitleFont( aFont );
     aStyleSet.SetMenuFont( aFont );
     aStyleSet.SetToolFont( aFont );
     aStyleSet.SetLabelFont( aFont );
@@ -3419,6 +3417,10 @@ void GtkSalGraphics::updateSettings( AllSettings& rSettings )
     aStyleSet.SetIconFont( aFont );
     aStyleSet.SetGroupFont( aFont );
 
+    aFont.SetWeight( WEIGHT_BOLD );
+    aStyleSet.SetTitleFont( aFont );
+    aStyleSet.SetFloatTitleFont( aFont );
+
     // get cursor blink time
     GtkSettings *pSettings = gtk_widget_get_settings( gWidgetData[m_nScreen].gEditBoxWidget );
     gboolean blink = false;
diff --git vcl/unx/kde/salnativewidgets-kde.cxx vcl/unx/kde/salnativewidgets-kde.cxx
index 2e3d71d..9da0fde 100644
--- vcl/unx/kde/salnativewidgets-kde.cxx
+++ vcl/unx/kde/salnativewidgets-kde.cxx
@@ -1954,9 +1954,6 @@ void KDESalFrame::UpdateSettings( AllSettings& rSettings )
 
     aStyleSettings.SetAppFont( aFont );
     aStyleSettings.SetHelpFont( aFont );
-    if( !bSetTitleFont )
-        aStyleSettings.SetTitleFont( aFont );
-    aStyleSettings.SetFloatTitleFont( aFont );
     aStyleSettings.SetMenuFont( aFont ); // will be changed according to pMenuBar
     aStyleSettings.SetToolFont( aFont ); // will be changed according to pToolBar
     aStyleSettings.SetLabelFont( aFont );
@@ -1966,6 +1963,12 @@ void KDESalFrame::UpdateSettings( AllSettings& rSettings )
     aStyleSettings.SetFieldFont( aFont );
     aStyleSettings.SetIconFont( aFont );
     aStyleSettings.SetGroupFont( aFont );
+
+    aFont.SetWeight( WEIGHT_BOLD );
+    if( !bSetTitleFont )
+        aStyleSettings.SetTitleFont( aFont );
+    aStyleSettings.SetFloatTitleFont( aFont );
+
     int flash_time = QApplication::cursorFlashTime();
     aStyleSettings.SetCursorBlinkTime( flash_time != 0 ? flash_time/2 : STYLE_CURSOR_NOBLINKTIME );
 
diff --git vcl/unx/kde4/KDESalFrame.cxx vcl/unx/kde4/KDESalFrame.cxx
index 95ca526..b4417c9 100644
--- vcl/unx/kde4/KDESalFrame.cxx
+++ vcl/unx/kde4/KDESalFrame.cxx
@@ -288,12 +288,6 @@ void KDESalFrame::UpdateSettings( AllSettings& rSettings )
     style.SetAppFont( aFont );
     style.SetHelpFont( aFont );
     
-    if( !bSetTitleFont )
-    {
-        style.SetTitleFont( aFont );
-    }
-    
-    style.SetFloatTitleFont( aFont );
     style.SetMenuFont( aFont ); // will be changed according to pMenuBar
     //style.SetToolFont( aFont ); //already set above
     style.SetLabelFont( aFont );
@@ -303,6 +297,13 @@ void KDESalFrame::UpdateSettings( AllSettings& rSettings )
     style.SetFieldFont( aFont );
     style.SetIconFont( aFont );
     style.SetGroupFont( aFont );
+
+    aFont.SetWeight( WEIGHT_BOLD );
+    if( !bSetTitleFont )
+    {
+        style.SetTitleFont( aFont );
+    }
+    style.SetFloatTitleFont( aFont );
     
     int flash_time = QApplication::cursorFlashTime();
     style.SetCursorBlinkTime( flash_time != 0 ? flash_time/2 : STYLE_CURSOR_NOBLINKTIME );
@@ -407,4 +408,4 @@ SalGraphics* KDESalFrame::GetGraphics()
     }
     
     return NULL;
-}
\ No newline at end of file
+}
