diff --git svx/source/tbxctrls/extrusioncontrols.src svx/source/tbxctrls/extrusioncontrols.src
index 96c4439..f6d1461 100644
--- svx/source/tbxctrls/extrusioncontrols.src
+++ svx/source/tbxctrls/extrusioncontrols.src
@@ -32,13 +32,8 @@
 
 FloatingWindow RID_SVXFLOAT_EXTRUSION_DIRECTION
 {
-    Border = TRUE ;
+    StdPopup = TRUE ;
     Hide = TRUE ;
-    SVLook = TRUE ;
-    Sizeable = FALSE ;
-    Moveable = TRUE ;
-    Closeable = TRUE ;
-    Zoomable = TRUE ;
 
     Text [ en-US ] = "Extrusion Direction" ;
 
@@ -202,13 +197,8 @@ FloatingWindow RID_SVXFLOAT_EXTRUSION_DIRECTION
 
 FloatingWindow RID_SVXFLOAT_EXTRUSION_DEPTH
 {
-    Border = TRUE ;
+    StdPopup = TRUE ;
     Hide = TRUE ;
-    SVLook = TRUE ;
-    Sizeable = FALSE ;
-    Moveable = TRUE ;
-    Closeable = TRUE ;
-    Zoomable = TRUE ;
 
     Text [ en-US ] = "Extrusion Depth" ;
     
@@ -291,13 +281,8 @@ FloatingWindow RID_SVXFLOAT_EXTRUSION_DEPTH
 
 FloatingWindow RID_SVXFLOAT_EXTRUSION_LIGHTING
 {
-    Border = TRUE ;
+    StdPopup = TRUE ;
     Hide = TRUE ;
-    SVLook = TRUE ;
-    Sizeable = FALSE ;
-    Moveable = TRUE ;
-    Closeable = TRUE ;
-    Zoomable = TRUE ;
 
     Text [ en-US ] = "Extrusion Lighting" ;
     
@@ -607,13 +592,8 @@ FloatingWindow RID_SVXFLOAT_EXTRUSION_LIGHTING
 
 FloatingWindow RID_SVXFLOAT_EXTRUSION_SURFACE
 {
-    Border = TRUE ;
+    StdPopup = TRUE ;
     Hide = TRUE ;
-    SVLook = TRUE ;
-    Sizeable = FALSE ;
-    Moveable = TRUE ;
-    Closeable = TRUE ;
-    Zoomable = TRUE ;
 
     Text [ en-US ] = "Extrusion Surface" ;
     
diff --git svx/source/tbxctrls/fontworkgallery.src svx/source/tbxctrls/fontworkgallery.src
index e2e6e47..2b960db 100644
--- svx/source/tbxctrls/fontworkgallery.src
+++ svx/source/tbxctrls/fontworkgallery.src
@@ -94,13 +94,8 @@ ModalDialog RID_SVX_MDLG_FONTWORK_GALLERY
 
 FloatingWindow RID_SVXFLOAT_FONTWORK_ALIGNMENT
 {
-    Border = TRUE ;
+    StdPopup = TRUE ;
     Hide = TRUE ;
-    SVLook = TRUE ;
-    Sizeable = FALSE ;
-    Moveable = TRUE ;
-    Closeable = TRUE ;
-    Zoomable = TRUE ;
 
     Text [ en-US ] = "Fontwork Alignment" ;
     
@@ -229,13 +224,8 @@ FloatingWindow RID_SVXFLOAT_FONTWORK_ALIGNMENT
 
 FloatingWindow RID_SVXFLOAT_FONTWORK_CHARSPACING
 {
-    Border = TRUE ;
+    StdPopup = TRUE ;
     Hide = TRUE ;
-    SVLook = TRUE ;
-    Sizeable = FALSE ;
-    Moveable = TRUE ;
-    Closeable = TRUE ;
-    Zoomable = TRUE ;
 
     Text [ en-US ] = "Fontwork Character Spacing" ;
     
diff --git svx/source/tbxctrls/linectrl.cxx svx/source/tbxctrls/linectrl.cxx
index 2a69a5f..df3378c 100644
--- svx/source/tbxctrls/linectrl.cxx
+++ svx/source/tbxctrls/linectrl.cxx
@@ -382,7 +382,7 @@ SvxLineEndWindow::SvxLineEndWindow(
     const String& rWndTitle ) :
     SfxPopupWindow( nSlotId,
                     rFrame,
-                    WinBits( WB_BORDER | WB_STDFLOATWIN | WB_SIZEABLE | WB_3DLOOK ) ),
+                    WinBits( WB_STDPOPUP ) ),
     pLineEndList	( NULL ),
     aLineEndSet		( this, WinBits( WB_ITEMBORDER | WB_3DLOOK | WB_NO_DIRECTSELECT ) ),
     nCols			( 2 ),
@@ -404,7 +404,7 @@ SvxLineEndWindow::SvxLineEndWindow(
     SfxPopupWindow( nSlotId,
                     rFrame,
                     pParentWindow,
-                    WinBits( WB_BORDER | WB_STDFLOATWIN | WB_SIZEABLE | WB_3DLOOK ) ),
+                    WinBits( WB_STDPOPUP ) ),
     pLineEndList	( NULL ),
     aLineEndSet		( this, WinBits( WB_ITEMBORDER | WB_3DLOOK | WB_NO_DIRECTSELECT ) ),
     nCols			( 2 ),
diff --git svx/source/tbxctrls/tbcontrl.cxx svx/source/tbxctrls/tbcontrl.cxx
index be75081..131636d 100644
--- svx/source/tbxctrls/tbcontrl.cxx
+++ svx/source/tbxctrls/tbcontrl.cxx
@@ -831,7 +831,7 @@ SvxColorWindow_Impl::SvxColorWindow_Impl( const OUString&            rCommand,
                                           const String&              rWndTitle,
                                           Window*                    pParentWindow ) :
 
-    SfxPopupWindow( nSlotId, rFrame, pParentWindow, WinBits( WB_BORDER | WB_STDFLOATWIN | WB_3DLOOK|WB_DIALOGCONTROL ) ),
+    SfxPopupWindow( nSlotId, rFrame, pParentWindow, WinBits( WB_STDPOPUP ) ),
 
     theSlotId( nSlotId ),
     aColorSet( this, WinBits( WB_ITEMBORDER | WB_NAMEFIELD | WB_3DLOOK | WB_NO_DIRECTSELECT) ),
@@ -1057,7 +1057,7 @@ void SvxColorWindow_Impl::StateChanged( USHORT nSID, SfxItemState eState, const
 
 SvxFrameWindow_Impl::SvxFrameWindow_Impl( USHORT nId, const Reference< XFrame >& rFrame, Window* pParentWindow ) :
 
-    SfxPopupWindow( nId, rFrame, pParentWindow, WinBits( WB_BORDER | WB_STDFLOATWIN | WB_3DLOOK | WB_DIALOGCONTROL ) ),
+    SfxPopupWindow( nId, rFrame, pParentWindow, WinBits( WB_STDPOPUP ) ),
     aFrameSet   ( this, WinBits( WB_ITEMBORDER | WB_DOUBLEBORDER | WB_3DLOOK | WB_NO_DIRECTSELECT ) ),
     bParagraphMode(sal_False)
 
@@ -1331,7 +1331,7 @@ BOOL SvxFrameWindow_Impl::Close()
 
 SvxLineWindow_Impl::SvxLineWindow_Impl( USHORT nId, const Reference< XFrame >& rFrame, Window* pParentWindow ) :
 
-    SfxPopupWindow( nId, rFrame, pParentWindow, WinBits( WB_BORDER | WB_STDFLOATWIN | WB_3DLOOK | WB_DIALOGCONTROL ) ),
+    SfxPopupWindow( nId, rFrame, pParentWindow, WinBits( WB_STDPOPUP ) ),
 
     aLineSet( this, WinBits( WB_3DLOOK | WB_ITEMBORDER | WB_DOUBLEBORDER | WB_NAMEFIELD | WB_NONEFIELD | WB_NO_DIRECTSELECT ) )
 {
