--- transex3/scripts/localize.pl.old	2010-02-04 14:55:42.000000000 +0100
+++ transex3/scripts/localize.pl	2010-02-17 18:05:32.000000000 +0100
@@ -402,11 +402,9 @@ sub merge_gsicheck{
     my ( $TMPHANDLE , $tmpfile ) = File::Temp::tempfile();
     close ( $TMPHANDLE );
     
-    if( $ENV{WRAPCMD} ){
-        $command = "$ENV{WRAPCMD} gsicheck";
-    }else{
-        $command = "gsicheck";
-    }
+    $command = "$ENV{WRAPCMD} " if( $ENV{WRAPCMD} );
+    $command .= fix_cygwin_path("$ENV{SOLARVER}/$ENV{INPATH}/bin/gsicheck");
+
     my $errfile = $sdffile.".err";
     $command .= " -k -c -wcf $tmpfile -wef ".fix_cygwin_path($errfile)." -l \"\" ".fix_cygwin_path($sdffile);
     #my $rc = system( $command );
