From defe8b414566cfc9d5d8c0aa9f91677bd202f6ce Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:57:52 +0200
Subject: [PATCH 285/878] transex3-localize-gsicheck-path.diff

---
 transex3/scripts/localize.pl |    8 +++-----
 1 files changed, 3 insertions(+), 5 deletions(-)

diff --git a/transex3/scripts/localize.pl b/transex3/scripts/localize.pl
index 66e0e06..9da77eb 100755
--- a/transex3/scripts/localize.pl
+++ b/transex3/scripts/localize.pl
@@ -401,11 +401,9 @@ sub merge_gsicheck{
     my ( $TMPHANDLE , $tmpfile ) = File::Temp::tempfile();
     close ( $TMPHANDLE );
     
-    if( $ENV{WRAPCMD} ){
-        $command = "$ENV{WRAPCMD} gsicheck";
-    }else{
-        $command = "gsicheck";
-    }
+    $command = "$ENV{WRAPCMD} " if( $ENV{WRAPCMD} );
+    $command .= fix_cygwin_path("$ENV{SOLARVER}/$ENV{INPATH}/bin/gsicheck");
+
     my $errfile = $sdffile.".err";
     $command .= " -k -c -wcf $tmpfile -wef ".fix_cygwin_path($errfile)." -l \"\" ".fix_cygwin_path($sdffile);
     #my $rc = system( $command );
-- 
1.7.0.1

