--- tools/source/datetime/makefile.mk.orig	2008-04-11 13:39:17.000000000 +0000
+++ tools/source/datetime/makefile.mk	2010-01-12 12:11:44.000000000 +0000
@@ -39,6 +39,10 @@
 .INCLUDE :  settings.mk
 .INCLUDE :      $(PRJ)$/util$/makefile.pmk
 
+.IF "$(CPUNAME)"=="ARM"
+CFLAGSOPT += -O2
+.ENDIF
+
 # --- Files --------------------------------------------------------
 
 SLOFILES=   $(SLO)$/tdate.obj     \
--- solenv/inc/unxlngr.mk.orig	2010-01-12 08:15:59.000000000 +0000
+++ solenv/inc/unxlngr.mk	2010-01-12 13:29:27.000000000 +0000
@@ -113,7 +113,7 @@
 CFLAGSDBGUTIL=
 # Compiler flags for enabling optimizations
 .IF "$(PRODUCT)"!=""
-CFLAGSOPT=-Os -fno-strict-aliasing             # optimizing for products
+CFLAGSOPT=-O2 -fno-strict-aliasing             # optimizing for products
 CFLAGSOPT+=-Wuninitialized                             # not supported without optimization
 .ELSE  # "$(PRODUCT)"!=""
 CFLAGSOPT=                                                     # no optimizing for non products
