--- vcl/source/window/makefile.mk.orig	2010-11-11 16:22:48.000000000 +0100
+++ vcl/source/window/makefile.mk	2010-12-29 23:44:40.518213478 +0100
@@ -38,6 +38,10 @@
 .INCLUDE :	settings.mk
 .INCLUDE :  $(PRJ)$/util$/makefile2.pmk
 
+.IF "$(CPUNAME)"=="ARM"
+CFLAGS += -fno-schedule-insns
+.ENDIF
+
 # --- Files --------------------------------------------------------
 
 SLOFILES= \
--- vcl/source/gdi/makefile.mk.orig	2010-11-11 16:22:48.000000000 +0100
+++ vcl/source/gdi/makefile.mk	2010-12-29 23:44:40.518213478 +0100
@@ -45,6 +45,10 @@
 CDEFS+=-DENABLE_GRAPHITE
 .ENDIF
 
+.IF "$(CPUNAME)"=="ARM"
+CFLAGS += -fno-schedule-insns
+.ENDIF
+
 # --- Files --------------------------------------------------------
 
 EXCEPTIONSFILES=	$(SLO)$/salmisc.obj 	\
--- forms/source/richtext/makefile.mk.orig	2010-11-11 20:08:57.000000000 +0100
+++ forms/source/richtext/makefile.mk	2010-12-29 23:44:40.518213478 +0100
@@ -35,6 +35,10 @@
 .INCLUDE : settings.mk
 .INCLUDE: $(PRJ)$/makefile.pmk
 
+.IF "$(CPUNAME)"=="ARM"
+CFLAGS += -fno-schedule-insns
+.ENDIF
+
 # --- Files -------------------------------------
 
 EXCEPTIONSFILES=\
--- tools/source/datetime/makefile.mk.orig	2010-11-11 16:22:48.000000000 +0100
+++ tools/source/datetime/makefile.mk	2010-12-29 23:44:40.518213478 +0100
@@ -35,6 +35,10 @@
 .INCLUDE :  settings.mk
 .INCLUDE :      $(PRJ)$/util$/makefile.pmk
 
+.IF "$(CPUNAME)"=="ARM"
+CFLAGS += -fno-schedule-insns
+.ENDIF
+
 # --- Files --------------------------------------------------------
 
 SLOFILES=   $(SLO)$/tdate.obj     \
--- extensions/source/propctrlr/makefile.mk.orig	2010-11-11 16:12:41.000000000 +0100
+++ extensions/source/propctrlr/makefile.mk	2010-12-29 23:44:40.518213478 +0100
@@ -38,6 +38,10 @@
 
 .INCLUDE :	$(PRJ)$/util$/makefile.pmk
 
+.IF "$(CPUNAME)"=="ARM"
+CFLAGS += -fno-schedule-insns
+.ENDIF
+
 # --- Files --------------------------------------------------------
 
 SLOFILES=           $(SLO)$/controlfontdialog.obj \
--- svtools/source/contnr/makefile.mk.orig	2010-11-11 16:22:47.000000000 +0100
+++ svtools/source/contnr/makefile.mk	2010-12-29 23:44:40.518213478 +0100
@@ -39,6 +39,10 @@
 .INCLUDE :  settings.mk
 .INCLUDE :  $(PRJ)$/util$/svt.pmk
 
+.IF "$(CPUNAME)"=="ARM"
+CFLAGS += -fno-schedule-insns
+.ENDIF
+
 # --- Files --------------------------------------------------------
 
 EXCEPTIONSFILES=\
--- svtools/source/control/makefile.mk.orig	2010-11-11 16:22:47.000000000 +0100
+++ svtools/source/control/makefile.mk	2010-12-29 23:44:40.518213478 +0100
@@ -35,6 +35,10 @@
 .INCLUDE :	settings.mk
 .INCLUDE :  $(PRJ)$/util$/svt.pmk
 
+.IF "$(CPUNAME)"=="ARM"
+CFLAGS += -fno-schedule-insns
+.ENDIF
+
 # --- Files --------------------------------------------------------
 
 SRS1NAME=$(TARGET)
--- oox/source/xls/makefile.mk.orig	2010-11-11 16:18:11.000000000 +0100
+++ oox/source/xls/makefile.mk	2010-12-29 23:44:40.518213478 +0100
@@ -38,6 +38,10 @@
 .INCLUDE :  settings.mk
 .INCLUDE: $(PRJ)$/util$/makefile.pmk
 
+.IF "$(CPUNAME)"=="ARM"
+CFLAGS += -fno-schedule-insns
+.ENDIF
+
 # --- Files --------------------------------------------------------
 
 SLOFILES =										\
--- basic/source/app/makefile.mk.orig	2010-11-11 16:30:07.000000000 +0100
+++ basic/source/app/makefile.mk	2010-12-29 23:44:40.518213478 +0100
@@ -36,6 +36,10 @@
 
 .INCLUDE :  settings.mk
 
+.IF "$(CPUNAME)"=="ARM"
+CFLAGS += -fno-schedule-insns
+.ENDIF
+
 # --- Common ------------------------------------------------------------
 
 EXCEPTIONSFILES = \
