=== modified file 'solenv/inc/settings.mk'
--- solenv/inc/settings.mk	2010-06-05 03:12:38 +0000
+++ solenv/inc/settings.mk	2010-06-05 03:21:48 +0000
@@ -1325,6 +1325,9 @@
 INCPRE+=-I$(UNOUCROUT)
 UNOUCRRDB*=$(SOLARBINDIR)/types.rdb
 
+CFLAGS+= -fno-stack-protector
+CXXFLAGS+= -fno-stack-protector
+
 # location of trustedinfo.manifest for Windows Vista
 .IF "$(COM)"=="MSC"
 TRUSTED_MANIFEST_LOCATION*=$(SOLARENV)/inc

