--- extras/source/palettes/standard.soc.orig	2010-11-11 16:14:35.000000000 +0100
+++ extras/source/palettes/standard.soc	2010-12-24 19:36:52.708227152 +0100
@@ -87,6 +87,9 @@
   <draw:color draw:name="Dark violet" draw:color="#660066"/>
   <draw:color draw:name="Salmon" draw:color="#ff8080"/>
   <draw:color draw:name="Sea blue" draw:color="#0066cc"/>
+  <draw:color draw:name="Ubuntu orange" draw:color="#ff6309"/>
+  <draw:color draw:name="Ubuntu yellow" draw:color="#ffb515"/>
+  <draw:color draw:name="Ubuntu red" draw:color="#c90016"/>
   <draw:color draw:name="Chart 1" draw:color="#004586"/>
   <draw:color draw:name="Chart 2" draw:color="#ff420e"/>
   <draw:color draw:name="Chart 3" draw:color="#ffd320"/>
