=== modified file 'extras/source/palettes/standard.soc'
--- extras/source/palettes/standard.soc	2010-06-05 03:12:38 +0000
+++ extras/source/palettes/standard.soc	2010-06-05 03:19:58 +0000
@@ -88,6 +88,9 @@
  <draw:color draw:name="Dark violet" draw:color="#660066"/>
  <draw:color draw:name="Salmon" draw:color="#ff8080"/>
  <draw:color draw:name="Sea blue" draw:color="#0066cc"/>
+ <draw:color draw:name="Ubuntu orange" draw:color="#ff6309"/>
+ <draw:color draw:name="Ubuntu yellow" draw:color="#ffb515"/>
+ <draw:color draw:name="Ubuntu red" draw:color="#c90016"/>
  <draw:color draw:name="Chart 1" draw:color="#004586"/>
  <draw:color draw:name="Chart 2" draw:color="#ff420e"/>
  <draw:color draw:name="Chart 3" draw:color="#ffd320"/>
@@ -100,4 +103,4 @@
  <draw:color draw:name="Chart 10" draw:color="#ff950e"/>
  <draw:color draw:name="Chart 11" draw:color="#c5000b"/>
  <draw:color draw:name="Chart 12" draw:color="#0084d1"/>
-</office:color-table>
\ No newline at end of file
+</office:color-table>

