--- idlc/source/idlcmain.cxx.old	2010-03-25 18:52:17.000000000 +0100
+++ idlc/source/idlcmain.cxx	2010-03-29 15:50:45.000000000 +0200
@@ -33,12 +33,29 @@
 #include <idlc/idlc.hxx>
 #include "sal/main.h"
 
+extern "C" {
+#include <stdio.h>
+#include "sched.h"
+}
+
+void set_affinity(void)
+{
+    cpu_set_t mask;
+
+    CPU_ZERO(&mask);
+    CPU_SET(0, &mask);
+    if (sched_setaffinity(0, sizeof(cpu_set_t), &mask))
+        perror("sched_setaffinity");
+}
+
 using namespace ::rtl;
 
 SAL_IMPLEMENT_MAIN_WITH_ARGS(argc, argv)
 {
     Options options;
 
+    set_affinity();
+
     try 
     {
         if (!options.initOptions(argc, argv))
