--- solenv/inc/unxlngr.mk.orig	2010-01-15 21:15:20.000000000 -0600
+++ solenv/inc/unxlngr.mk	2010-01-20 14:02:29.000000000 -0600
@@ -31,7 +31,7 @@
 
 # mk file for Unix Linux ARM using GCC, please make generic modifications to unxlng.mk
 
-CDEFAULTOPT=-Os
+CDEFAULTOPT=-O2
 .INCLUDE : unxlng.mk
 CDEFS+=-DARM32
 CFLAGS+=-fno-omit-frame-pointer
--- tools/source/datetime/makefile.mk.orig	2010-01-15 21:15:20.000000000 -0600
+++ tools/source/datetime/makefile.mk	2010-01-20 13:56:49.000000000 -0600
@@ -39,6 +39,10 @@
 .INCLUDE :  settings.mk
 .INCLUDE :      $(PRJ)$/util$/makefile.pmk
 
+.IF "$(CPUNAME)"=="ARM"
+CFLAGSOPT += -O2
+.ENDIF
+
 # --- Files --------------------------------------------------------
 
 SLOFILES=   $(SLO)$/tdate.obj     \
