--- instsetoo_native/util/openoffice.lst~	2010-08-19 06:51:44.000000000 +0100
+++ instsetoo_native/util/openoffice.lst	2010-08-19 06:53:51.000000000 +0100
@@ -25,10 +25,10 @@
 			POOLPRODUCT 1
 			REGISTRATION_HOST https://registration2.services.openoffice.org/RegistrationWeb
 			REGISTRATIONURL http://survey.services.openoffice.org/user/index.php
-			PROGRESSBARCOLOR 14,133,205
-			PROGRESSSIZE 320,7
-			PROGRESSPOSITION 185,206
-			PROGRESSFRAMECOLOR 207,208,211
+			PROGRESSBARCOLOR 171,117,61
+			PROGRESSSIZE 405,7
+			PROGRESSPOSITION 15,220
+			PROGRESSFRAMECOLOR 207,208,212
 			NATIVEPROGRESS true
 			REGISTRYLAYERNAME Layers
 			SERVICEPACK 1
