From e6c67040f258c8d2b9722c95a1fc5e9852183ca4 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:07:23 +0200
Subject: [PATCH 761/878] unotools-tempfile-nonlocking.diff

---
 unotools/source/ucbhelper/tempfile.cxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/unotools/source/ucbhelper/tempfile.cxx b/unotools/source/ucbhelper/tempfile.cxx
index 7dcd450..5394bb1 100644
--- a/unotools/source/ucbhelper/tempfile.cxx
+++ b/unotools/source/ucbhelper/tempfile.cxx
@@ -306,7 +306,7 @@ void lcl_createName(TempFile_Impl& _rImpl,const String& rLeadingChars,sal_Bool _
 #ifdef UNX
 umask(old_mode);
 #endif
-            if ( err == FileBase::E_None )
+            if ( err == FileBase::E_None || err == FileBase::E_NOLCK )
             {
                 _rImpl.aName = aTmp;
                 aFile.close();
-- 
1.7.0.1

