---
 unotools/source/ucbhelper/tempfile.cxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git unotools/source/ucbhelper/tempfile.cxx unotools/source/ucbhelper/tempfile.cxx
index 530892a..b8ae1e7 100644
--- unotools/source/ucbhelper/tempfile.cxx
+++ unotools/source/ucbhelper/tempfile.cxx
@@ -306,7 +306,7 @@ void lcl_createName(TempFile_Impl& _rImpl,const String& rLeadingChars,sal_Bool _
 #ifdef UNX
 umask(old_mode);
 #endif
-            if ( err == FileBase::E_None )
+            if ( err == FileBase::E_None || err == FileBase::E_NOLCK )
             {
                 _rImpl.aName = aTmp;
                 aFile.close();
-- 
1.7.0.1

