diff -r 21777680017e unoxml/source/rdf/librdf_repository.cxx
--- unoxml/source/rdf/librdf_repository.cxx	Thu Mar 18 17:15:13 2010 +0000
+++ unoxml/source/rdf/librdf_repository.cxx	Thu Apr 01 16:23:56 2010 +0100
@@ -52,6 +52,7 @@
 #include <rtl/ustring.hxx>
 
 #include <librdf.h>
+#include <libxslt/security.h>
 
 #include <boost/utility.hpp>
 #include <boost/shared_ptr.hpp>
@@ -1887,7 +1888,11 @@
             m_rRep);
     }
     //FIXME logger, digest, features?
+    xsltSecurityPrefsPtr origprefs = xsltGetDefaultSecurityPrefs();
     librdf_world_open(pWorld);
+    xsltSecurityPrefsPtr newprefs = xsltGetDefaultSecurityPrefs();
+    if (newprefs != origprefs)
+        xsltSetDefaultSecurityPrefs(origprefs);
     return pWorld;
 }
 
diff -r 21777680017e unoxml/source/rdf/makefile.mk
--- unoxml/source/rdf/makefile.mk	Thu Mar 18 17:15:13 2010 +0000
+++ unoxml/source/rdf/makefile.mk	Thu Apr 01 16:23:56 2010 +0100
@@ -41,6 +41,13 @@
 CFLAGS+=-DSYSTEM_REDLAND $(REDLAND_CFLAGS)
 .ENDIF
 
+.IF "$(SYSTEM_LIBXSLT)" == "YES"
+CFLAGS+= $(LIBXSLT_CFLAGS)
+.ELSE
+LIBXSLTINCDIR=external$/libxslt
+CFLAGS+= -I$(SOLARINCDIR)$/$(LIBXSLTINCDIR)
+.ENDIF
+
 # --- Files --------------------------------------------------------
 .IF "$(L10N_framework)"==""
 
@@ -64,6 +71,7 @@
 
 SHL1STDLIBS= \
     $(REDLANDLIB) \
+    $(XSLTLIB) \
 	$(CPPUHELPERLIB) \
 	$(CPPULIB)	\
 	$(SALLIB) \
