From b55699b9fffb480207109ae652c2210f1b459331 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:07:12 +0200
Subject: [PATCH 660/768] unoxml-fix-empty-xmlns.diff

---
 unoxml/source/dom/node.cxx |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git unoxml/source/dom/node.cxx unoxml/source/dom/node.cxx
index f44e56d..57779db 100644
--- unoxml/source/dom/node.cxx
+++ unoxml/source/dom/node.cxx
@@ -68,8 +68,9 @@ namespace DOM
         // add node's namespaces to current context
         for (xmlNsPtr pNs = pNode->nsDef; pNs != 0; pNs = pNs->next) {
             const xmlChar *pPrefix = pNs->prefix;
+        // prefix can be NULL when xmlns attribute is empty (xmlns="")
             OString prefix(reinterpret_cast<const sal_Char*>(pPrefix),
-                           strlen(reinterpret_cast<const char*>(pPrefix)));
+                           pPrefix ? strlen(reinterpret_cast<const char*>(pPrefix)) : 0);
             const xmlChar *pHref = pNs->href;
             OUString val(reinterpret_cast<const sal_Char*>(pHref),
                 strlen(reinterpret_cast<const char*>(pHref)),
-- 
1.7.0.1

