Index: unxlngi.mk
===================================================================
--- solenv/inc/unxlngi.mk	(Revision 276754)
+++ solenv/inc/unxlngi.mk	(Arbeitskopie)
@@ -31,7 +31,7 @@
 
 # mk file for Unix Linux Intel (X86) using GCC, please make generic modifications to unxlng.mk
 CDEFAULTOPT=-Os
-ARCH_FLAGS*=-mtune=pentiumpro
+#ARCH_FLAGS*=-mtune=pentiumpro
 .INCLUDE : unxlng.mk
 CDEFS+=-DX86
 DLLPOST=.so
