---
 desktop/unx/source/makefile.mk |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git desktop/unx/source/makefile.mk desktop/unx/source/makefile.mk
index f10ad70..66d1bac 100644
--- desktop/unx/source/makefile.mk
+++ desktop/unx/source/makefile.mk
@@ -13,6 +13,7 @@ OBJFILES= \
     $(OBJ)$/start.obj
 
 APP1TARGET = $(TARGET)
+APP1RPATH  = BRAND
 APP1OBJS   = $(OBJFILES)
 APP1LIBSALCPPRT=
 APP1CODETYPE = C
-- 
1.7.0.1

