From 06be201a36c297472de32788484074cfbcad6cfa Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:55:00 +0200
Subject: [PATCH 134/878] unxsplash-rpath.diff

---
 desktop/unx/source/makefile.mk |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/desktop/unx/source/makefile.mk b/desktop/unx/source/makefile.mk
index f10ad70..66d1bac 100644
--- a/desktop/unx/source/makefile.mk
+++ b/desktop/unx/source/makefile.mk
@@ -13,6 +13,7 @@ OBJFILES= \
     $(OBJ)$/start.obj
 
 APP1TARGET = $(TARGET)
+APP1RPATH  = BRAND
 APP1OBJS   = $(OBJFILES)
 APP1LIBSALCPPRT=
 APP1CODETYPE = C
-- 
1.7.0.1

