From a5a017db0b41b932e31cd32012f333e600c43a73 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:54:57 +0200
Subject: [PATCH 108/768] unxsplash-new-osl-pipe-slot.diff

---
 desktop/unx/source/start.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git desktop/unx/source/start.c desktop/unx/source/start.c
index c48cfff..42c7bc3 100644
--- desktop/unx/source/start.c
+++ desktop/unx/source/start.c
@@ -331,7 +331,8 @@ send_args( int fd, rtl_uString *pCwdPath )
 
         // this is not a param, we have to prepend filenames with file://
         // FIXME: improve the check
-        if ( pTmp->buffer[0] != (sal_Unicode)'-' )
+        if ( ( pTmp->buffer[0] != (sal_Unicode)'-' ) &&
+             ( rtl_ustr_indexOfAscii_WithLength( pTmp->buffer, pTmp->length, "slot:", 5 /* length */ ) ) )
         {
             sal_Int32 nFirstColon = rtl_ustr_indexOfChar_WithLength( pTmp->buffer, pTmp->length, ':' );
             sal_Int32 nFirstSlash = rtl_ustr_indexOfChar_WithLength( pTmp->buffer, pTmp->length, '/' );
-- 
1.7.0.1

