From a2787d12635b3910988fe41b719956bf418c4545 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:55:03 +0200
Subject: [PATCH 113/768] unxsplash-install-bmp-intro.diff

---
 desktop/zipintro/makefile.mk |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git desktop/zipintro/makefile.mk desktop/zipintro/makefile.mk
index fef106d..488e956 100644
--- desktop/zipintro/makefile.mk
+++ desktop/zipintro/makefile.mk
@@ -45,6 +45,7 @@ ZIP3LIST= \
     $(null,$(INTRO_BITMAPS) $(MISC)$/ooo_custom_images$/nologo$/introabout$/intro.png $(INTRO_BITMAPS)) \
     $(null,$(ABOUT_BITMAPS) $(MISC)$/$(RSCDEFIMG)$/introabout$/about.png $(ABOUT_BITMAPS))
 ZIP4LIST= \
+    $(null,$(INTRO_BITMAPS) $(MISC)$/$(RSCDEFIMG)$/introabout$/intro.bmp $(INTRO_BITMAPS)) \
     $(null,$(INTRO_BITMAPS) $(MISC)$/$(RSCDEFIMG)$/introabout$/intro.png $(INTRO_BITMAPS)) \
     $(null,$(ABOUT_BITMAPS) $(MISC)$/$(RSCDEFIMG)$/introabout$/about.png $(ABOUT_BITMAPS))
 ZIP5LIST= \
-- 
1.7.0.1

