--- solenv/inc/libs.mk.old	2010-03-03 16:59:55.000000000 +0100
+++ solenv/inc/libs.mk	2010-03-03 17:00:18.000000000 +0100
@@ -190,7 +190,9 @@ SAXLIB=-lsax$(DLLPOSTFIX)
 MAILLIB=-lmail
 DOCMGRLIB=-ldmg$(DLLPOSTFIX)
 BASICLIB=-lsb$(DLLPOSTFIX)
+.IF "$(ENABLE_VBA)"=="YES"
 VBAHELPERLIB=-lvbahelper$(DLLPOSTFIX)
+.ENDIF
 DBTOOLSLIB=-ldbtools$(DLLPOSTFIX)
 HM2LIBSH=-lhmwrpdll
 HM2LIBST=-lhmwrap
--- sw/util/makefile.mk.old	2010-03-03 17:00:04.000000000 +0100
+++ sw/util/makefile.mk	2010-03-03 17:00:18.000000000 +0100
@@ -344,6 +344,7 @@ SHL4STDLIBS= \
     $(BASICLIB)     \
     $(I18NUTILLIB)
 
+.IF "$(ENABLE_VBA)" == "YES"
 #target vba
 TARGET_VBA=vbaswobj
 SHL5TARGET=$(TARGET_VBA)$(DLLPOSTFIX).uno
@@ -380,5 +381,6 @@ SHL5STDLIBS+=$(SHELLLIB)
 
 SHL5DEPN=$(SHL1TARGETN)
 SHL5LIBS=$(SLB)$/$(TARGET_VBA).lib
+.ENDIF # .IF "$(ENABLE_VBA)" == "YES"
 
 .INCLUDE :  target.mk
--- vbahelper/source/msforms/makefile.mk.old	2010-03-03 16:59:35.000000000 +0100
+++ vbahelper/source/msforms/makefile.mk	2010-03-03 17:00:18.000000000 +0100
@@ -38,6 +38,11 @@ VISIBILITY_HIDDEN=TRUE
 
 # --- Settings -----------------------------------------------------
 
+.IF "$(ENABLE_VBA)" == "NO"
+dummy:
+	@echo "Nothing to build"
+.ENDIF
+
 .INCLUDE :  settings.mk
 
 CDEFS+=-DVBA_OOBUILD_HACK
--- vbahelper/source/vbahelper/makefile.mk.old	2010-03-03 16:59:35.000000000 +0100
+++ vbahelper/source/vbahelper/makefile.mk	2010-03-03 17:00:18.000000000 +0100
@@ -37,6 +37,11 @@ ENABLE_EXCEPTIONS := TRUE
 VISIBILITY_HIDDEN=TRUE
 # --- Settings -----------------------------------------------------
 
+.IF "$(ENABLE_VBA)" == "NO"
+dummy:
+	@echo "Nothing to build"
+.ENDIF
+
 .INCLUDE :  settings.mk
 
 SLOFILES=\
--- vbahelper/util/makefile.mk.old	2010-03-03 16:59:35.000000000 +0100
+++ vbahelper/util/makefile.mk	2010-03-03 17:00:18.000000000 +0100
@@ -38,6 +38,10 @@ TARGET=vbahelper
 
 .INCLUDE :  settings.mk
 
+.IF "$(ENABLE_VBA)" == "NO"
+dummy:
+	@echo "Nothing to build"
+.ENDIF
 
 TARGET_HELPER=vbahelper
 
