---
 vcl/source/gdi/outdev3.cxx |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git vcl/source/gdi/outdev3.cxx vcl/source/gdi/outdev3.cxx
index 92235e7..4c9fb8c 100644
--- vcl/source/gdi/outdev3.cxx
+++ vcl/source/gdi/outdev3.cxx
@@ -5924,6 +5924,11 @@ SalLayout* OutputDevice::ImplLayout( const String& rOrigStr,
     if( mbInitFont )
         ImplInitFont();
 
+    // we need a graphics
+    if( !mpGraphics )
+        if( !ImplGetGraphics() )
+            return NULL;
+
     // check string index and length
     if( (unsigned)nMinIndex + nLen > rOrigStr.Len() )
     {
-- 
1.7.0.1

