From c1b1dd674431598d126d0b6d9b7a15782d86749a Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:05:32 +0200
Subject: [PATCH 595/768] vcl-permit-global-shortcuts.diff

---
 vcl/source/window/dlgctrl.cxx |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git vcl/source/window/dlgctrl.cxx vcl/source/window/dlgctrl.cxx
index 306a5ea..ce7808a 100644
--- vcl/source/window/dlgctrl.cxx
+++ vcl/source/window/dlgctrl.cxx
@@ -845,7 +845,9 @@ BOOL Window::ImplDlgCtrl( const KeyEvent& rKEvt, BOOL bKeyInput )
                 pWindow = pWindow->GetWindow( WINDOW_NEXT );
             }
         }
-        else
+        else if ( !aKeyCode.IsMod1() ) // no ctrl pressed - those are
+                                       // application-wide shortcuts
+                                       // we want to let pass
         {
             xub_Unicode c = rKEvt.GetCharCode();
             if ( c )
-- 
1.7.0.1

