diff --git a/vcl/source/gdi/pngread.cxx b/vcl/source/gdi/pngread.cxx
index fe559f5..de2ddf7 100644
--- vcl/source/gdi/pngread.cxx
+++ vcl/source/gdi/pngread.cxx
@@ -376,6 +376,9 @@ BitmapEx PNGReaderImpl::GetBitmapEx( const Size& rPreviewSizeHint )
     // parse the chunks
     while( mbStatus && !mbIDAT && ReadNextChunk() )
     {
+        if( !mbStatus )
+            break;
+
         switch( mnChunkType )
         {
             case PNGCHUNK_IHDR :
@@ -706,7 +709,7 @@ void PNGReaderImpl::ImplGetGrayPalette( sal_uInt16 nBitDepth )
         nBitDepth = 8;
 
     sal_uInt16  nPaletteEntryCount = 1 << nBitDepth;
-    sal_uInt32  nAdd = 256 / (nPaletteEntryCount - 1);
+    sal_uInt32  nAdd = nBitDepth ? 256 / (nPaletteEntryCount - 1) : 0;
 
     // no bitdepth==2 available
     // but bitdepth==4 with two unused bits is close enough
