#i107763# don't crash with BadMatch on 1bpp Pixmaps

diff --git a/vcl/source/gdi/outdev6.cxx b/vcl/source/gdi/outdev6.cxx
index cf20a29..db34542 100644
--- vcl/source/gdi/outdev6.cxx
+++ vcl/source/gdi/outdev6.cxx
@@ -269,6 +269,7 @@ void OutputDevice::DrawTransparent( const PolyPolygon& rPolyPoly,
     // try hard to draw it directly, because the emulation layers are slower
     if( !pDisableNative
         && mpGraphics->supportsOperation( OutDevSupport_B2DDraw ) 
+        && GetBitCount() > 8
 #ifdef WIN32
         // workaround bad dithering on remote displaying when using GDI+ with toolbar buttoin hilighting
         && !rPolyPoly.IsRect()
