From 1c02978690ed50e25bfea96ca26475cb19acddd4 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:09:15 +0200
Subject: [PATCH 750/768] vcl-default-wheel-behaviour.diff

---
 vcl/source/app/settings.cxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git vcl/source/app/settings.cxx vcl/source/app/settings.cxx
index 7e8a4c1..467fdd1 100644
--- vcl/source/app/settings.cxx
+++ vcl/source/app/settings.cxx
@@ -188,7 +188,7 @@ ImplMouseData::ImplMouseData()
     mnActionDelay               = 250;
     mnMenuDelay                 = 150;
     mnFollow                    = MOUSE_FOLLOW_MENU | MOUSE_FOLLOW_DDLIST;
-    mnWheelBehavior             = MOUSE_WHEEL_FOCUS_ONLY;
+    mnWheelBehavior             = MOUSE_WHEEL_ALWAYS;
 }
 
 // -----------------------------------------------------------------------
-- 
1.7.0.1

