diff -rup vcl-orig-1/unx/gtk/app/gtkdata.cxx vcl/unx/gtk/app/gtkdata.cxx
--- vcl-orig-1/unx/gtk/app/gtkdata.cxx	2010-03-01 21:34:40.000000000 +0100
+++ vcl/unx/gtk/app/gtkdata.cxx	2010-03-02 11:17:00.000000000 +0100
@@ -554,6 +554,7 @@ GtkXLib::~GtkXLib()
 #if OSL_DEBUG_LEVEL > 1
     fprintf( stderr, "GtkXLib::~GtkXLib()\n" );
 #endif
+    Yield( true, true );
     StopTimer();
      // sanity check: at this point nobody should be yielding, but wake them
      // up anyway before the condition they're waiting on gets destroyed.
