From ce71a0beb1f4829f1c7eab2995195d2593201894 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:08:59 +0200
Subject: [PATCH 742/768] vcl-pngread-greypalette-fix.diff

---
 vcl/source/gdi/pngread.cxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git vcl/source/gdi/pngread.cxx vcl/source/gdi/pngread.cxx
index ac52fd0..c7c50ac 100644
--- vcl/source/gdi/pngread.cxx
+++ vcl/source/gdi/pngread.cxx
@@ -703,7 +703,7 @@ void PNGReaderImpl::ImplGetGrayPalette( sal_uInt16 nBitDepth )
         nBitDepth = 8;
 
     sal_uInt16  nPaletteEntryCount = 1 << nBitDepth;
-    sal_uInt32  nAdd = 256 / (nPaletteEntryCount - 1);
+    sal_uInt32  nAdd = nBitDepth ? 256 / (nPaletteEntryCount - 1) : 0;
 
     // no bitdepth==2 available
     // but bitdepth==4 with two unused bits is close enough
-- 
1.7.0.1

