From 330a62810d84a8b421e8f54886b3a320a5b21af8 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:07:28 +0200
Subject: [PATCH 765/878] vcl-uninitialized-val-fix.diff

---
 vcl/source/window/status.cxx |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/vcl/source/window/status.cxx b/vcl/source/window/status.cxx
index 9fc2d32..0f16042 100644
--- a/vcl/source/window/status.cxx
+++ b/vcl/source/window/status.cxx
@@ -153,6 +153,7 @@ void StatusBar::ImplInit( Window* pParent, WinBits nStyle )
     mbProgressMode	= FALSE;
     mbInUserDraw	= FALSE;
     mbBottomBorder	= FALSE;
+    mnItemsWidth    = STATUSBAR_OFFSET_X;
     mnDX			= 0;
     mnDY			= 0;
     mnCalcHeight	= 0;
-- 
1.7.0.1

