diff -rup vcl-orig/inc/vcl/salframe.hxx vcl/inc/vcl/salframe.hxx
--- vcl-orig/inc/vcl/salframe.hxx	2010-02-11 10:05:50.000000000 +0100
+++ vcl/inc/vcl/salframe.hxx	2010-02-11 10:05:53.000000000 +0100
@@ -295,6 +295,8 @@ public:
     // (e.g. input methods, printer update handlers).
     long						CallCallback( USHORT nEvent, const void* pEvent ) const
     { return m_pProc ? m_pProc( m_pWindow, const_cast<SalFrame*>(this), nEvent, pEvent ) : 0; }
+
+    virtual BOOL GetSmallScreenMode() { return FALSE; }
 };
 
 
diff -rup vcl-orig/inc/vcl/window.hxx vcl/inc/vcl/window.hxx
--- vcl-orig/inc/vcl/window.hxx	2010-02-11 10:05:50.000000000 +0100
+++ vcl/inc/vcl/window.hxx	2010-02-11 10:39:08.000000000 +0100
@@ -1159,6 +1159,8 @@ public:
 
     virtual XubString GetSurroundingText() const;
     virtual Selection GetSurroundingTextSelection() const;
+
+    BOOL GetSmallScreenMode();
 };
 
 
diff -rup vcl-orig/source/window/window3.cxx vcl/source/window/window3.cxx
--- vcl-orig/source/window/window3.cxx	2010-02-11 10:05:44.000000000 +0100
+++ vcl/source/window/window3.cxx	2010-02-11 10:41:33.000000000 +0100
@@ -37,6 +37,7 @@
 #include "vcl/window.h"
 #include "vcl/window.hxx"
 #include "vcl/waitobj.hxx"
+#include "vcl/salframe.hxx"
 #include "vcl/salgdi.hxx"
 #include "vcl/salnativewidgets.hxx"
 #include "vcl/salctrlhandle.hxx"
@@ -336,3 +337,8 @@ void Window::ImplAdjustNWFSizes()
         break;
     }
 }
+
+BOOL Window::GetSmallScreenMode()
+{
+    return ImplGetFrame()->GetSmallScreenMode();
+}
diff -rup vcl-orig/unx/gtk/window/gtkframe.cxx vcl/unx/gtk/window/gtkframe.cxx
--- vcl-orig/unx/gtk/window/gtkframe.cxx	2010-02-11 10:05:41.000000000 +0100
+++ vcl/unx/gtk/window/gtkframe.cxx	2010-02-11 10:08:14.000000000 +0100
@@ -842,7 +842,7 @@ void GtkSalFrame::Init( SalFrame* pParen
             eType = GDK_WINDOW_TYPE_HINT_UTILITY;
         }
         
-	if( getDisplay()->getWMAdaptor()->getSmallScreen() && eWinType == GTK_WINDOW_TOPLEVEL && !(nStyle & SAL_FRAME_STYLE_DIALOG ) )
+    if( GetSmallScreenMode() && eWinType == GTK_WINDOW_TOPLEVEL && !(nStyle & SAL_FRAME_STYLE_DIALOG ) )
 	    bNoDecor = true;
 
         gtk_window_set_type_hint( GTK_WINDOW(m_pWindow), eType );
@@ -1617,7 +1617,7 @@ void GtkSalFrame::SetWindowState( const
         SAL_FRAMESTATE_MASK_MAXIMIZED_X | SAL_FRAMESTATE_MASK_MAXIMIZED_Y |
         SAL_FRAMESTATE_MASK_MAXIMIZED_WIDTH | SAL_FRAMESTATE_MASK_MAXIMIZED_HEIGHT;
     
-    if( getDisplay()->getWMAdaptor()->getSmallScreen() && !(m_nStyle & SAL_FRAME_STYLE_DIALOG) )
+    if( GetSmallScreenMode() && !(m_nStyle & SAL_FRAME_STYLE_DIALOG) )
       gtk_window_maximize( GTK_WINDOW(m_pWindow) );
     else {
     if( (pState->mnMask & SAL_FRAMESTATE_MASK_STATE) &&
@@ -3732,3 +3732,8 @@ gboolean GtkSalFrame::IMHandler::signalI
 
     return FALSE;
 }
+
+BOOL GtkSalFrame::GetSmallScreenMode()
+{
+    return getDisplay()->getWMAdaptor()->getSmallScreen();  
+}
diff -rup vcl-orig/unx/inc/plugins/gtk/gtkframe.hxx vcl/unx/inc/plugins/gtk/gtkframe.hxx
--- vcl-orig/unx/inc/plugins/gtk/gtkframe.hxx	2010-02-11 10:05:42.000000000 +0100
+++ vcl/unx/inc/plugins/gtk/gtkframe.hxx	2010-02-11 10:07:02.000000000 +0100
@@ -394,6 +394,8 @@ public:
     virtual void					EndSetClipRegion();
 
     static GtkSalFrame         *getFromWindow( GtkWindow *pWindow );
+
+    virtual BOOL GetSmallScreenMode();
 };
 
 
