--- vcl/source/gdi/outdev3.cxx.old	2010-01-27 21:32:30.000000000 +0100
+++ vcl/source/gdi/outdev3.cxx	2010-02-02 17:27:52.000000000 +0100
@@ -6444,6 +6444,11 @@ SalLayout* OutputDevice::ImplLayout( con
     if( mbInitFont )
         ImplInitFont();
 
+    // we need a graphics
+    if( !mpGraphics )
+        if( !ImplGetGraphics() )
+            return NULL;
+
     // check string index and length
     String aStr = rOrigStr;
     if( (ULONG)nMinIndex + nLen >= aStr.Len() )
