From 9a31556f0e2ee5ff5224a86630206f65d5ab13f3 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:54:44 +0200
Subject: [PATCH 121/878] vcl-outputdevice-lost-mpGraphics.diff

---
 vcl/source/gdi/outdev3.cxx |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/vcl/source/gdi/outdev3.cxx b/vcl/source/gdi/outdev3.cxx
index 6084874..a37d086 100644
--- a/vcl/source/gdi/outdev3.cxx
+++ b/vcl/source/gdi/outdev3.cxx
@@ -6448,6 +6448,11 @@ SalLayout* OutputDevice::ImplLayout( const String& rOrigStr,
     if( mbInitFont )
         ImplInitFont();
 
+    // we need a graphics
+    if( !mpGraphics )
+        if( !ImplGetGraphics() )
+            return NULL;
+
     // check string index and length
     String aStr = rOrigStr;
     if( (ULONG)nMinIndex + nLen >= aStr.Len() )
-- 
1.7.0.1

