--- vcl/source/app/settings.cxx~	2010-04-06 17:44:46.000000000 +0200
+++ vcl/source/app/settings.cxx	2010-04-06 23:39:12.000000000 +0200
@@ -191,7 +191,7 @@
     mnActionDelay               = 250;
     mnMenuDelay                 = 150;
     mnFollow                    = MOUSE_FOLLOW_MENU | MOUSE_FOLLOW_DDLIST;
-    mnWheelBehavior             = MOUSE_WHEEL_FOCUS_ONLY;
+    mnWheelBehavior             = MOUSE_WHEEL_ALWAYS;
 }
 
 // -----------------------------------------------------------------------
