diff -rup vcl-orig/source/helper/canvastools.cxx vcl/source/helper/canvastools.cxx
--- vcl-orig/source/helper/canvastools.cxx	2010-11-29 12:14:14.000000000 +0100
+++ vcl/source/helper/canvastools.cxx	2010-11-29 12:16:29.000000000 +0100
@@ -819,13 +819,13 @@ namespace vcl
             const uno::Sequence< double >                   rColor,
             const uno::Reference< rendering::XColorSpace >& xColorSpace )
         {
-            const rendering::ARGBColor& rARGBColor(
+            const rendering::ARGBColor aARGBColor(
                 xColorSpace->convertToARGB(rColor)[0]);
 
-            return Color( 255-toByteColor(rARGBColor.Alpha),
-                          toByteColor(rARGBColor.Red),
-                          toByteColor(rARGBColor.Green),
-                          toByteColor(rARGBColor.Blue) );
+            return Color( 255-toByteColor(aARGBColor.Alpha),
+                          toByteColor(aARGBColor.Red),
+                          toByteColor(aARGBColor.Green),
+                          toByteColor(aARGBColor.Blue) );
         }
 
         //---------------------------------------------------------------------------------------
