From 6c51698481ce5c1a0bf298edd18f31d9f470a116 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:09:15 +0200
Subject: [PATCH 858/878] vcl-default-wheel-behaviour.diff

---
 vcl/source/app/settings.cxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/vcl/source/app/settings.cxx b/vcl/source/app/settings.cxx
index cfa4d89..4fd884f 100644
--- a/vcl/source/app/settings.cxx
+++ b/vcl/source/app/settings.cxx
@@ -188,7 +188,7 @@ ImplMouseData::ImplMouseData()
     mnActionDelay               = 250;
     mnMenuDelay                 = 150;
     mnFollow                    = MOUSE_FOLLOW_MENU | MOUSE_FOLLOW_DDLIST;
-    mnWheelBehavior             = MOUSE_WHEEL_FOCUS_ONLY;
+    mnWheelBehavior             = MOUSE_WHEEL_ALWAYS;
 }
 
 // -----------------------------------------------------------------------
-- 
1.7.0.1

