diff --git vcl/source/window/status.cxx vcl/source/window/status.cxx
index 28967d2..49b16a2 100644
--- vcl/source/window/status.cxx
+++ vcl/source/window/status.cxx
@@ -156,6 +156,7 @@ void StatusBar::ImplInit( Window* pParent, WinBits nStyle )
     mbProgressMode	= FALSE;
     mbInUserDraw	= FALSE;
     mbBottomBorder	= FALSE;
+    mnItemsWidth    = STATUSBAR_OFFSET_X;
     mnDX			= 0;
     mnDY			= 0;
     mnCalcHeight	= 0;
