From 599db3d98ba88b8020675f7c00cf522ca67c4722 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:54:44 +0200
Subject: [PATCH 100/768] vcl-outputdevice-lost-mpGraphics.diff

---
 vcl/source/gdi/outdev3.cxx |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git vcl/source/gdi/outdev3.cxx vcl/source/gdi/outdev3.cxx
index 5760f7e..4cb6683 100644
--- vcl/source/gdi/outdev3.cxx
+++ vcl/source/gdi/outdev3.cxx
@@ -5911,6 +5911,11 @@ SalLayout* OutputDevice::ImplLayout( const String& rOrigStr,
     if( mbInitFont )
         ImplInitFont();
 
+    // we need a graphics
+    if( !mpGraphics )
+        if( !ImplGetGraphics() )
+            return NULL;
+
     // check string index and length
     if( (unsigned)nMinIndex + nLen > rOrigStr.Len() )
     {
-- 
1.7.0.1

