---
 desktop/source/app/appinit.cxx |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git desktop/source/app/appinit.cxx desktop/source/app/appinit.cxx
index 25798d9..2f77133 100644
--- desktop/source/app/appinit.cxx
+++ desktop/source/app/appinit.cxx
@@ -146,7 +146,8 @@ static bool configureUcb(bool bServer, rtl::OUString const & rPortalConnect)
                 );
                 rtl::OUString aDesktopEnvironment;
                 if ((aValue >>= aDesktopEnvironment)
-                    && aDesktopEnvironment.equalsAscii("GNOME"))
+                    && (aDesktopEnvironment.equalsAscii("GNOME") ||
+                        aDesktopEnvironment.equalsAscii("KDE")))
                 {
                     Reference<XContentProviderManager> xCPM =
                         cb->getContentProviderManagerInterface();
-- 
1.7.0.1

