From 4a3d99eee0bda3b9e9444d1a20e6f17a09f4e610 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:58:13 +0200
Subject: [PATCH 269/768] vfs-kde-too.diff

---
 desktop/source/app/appinit.cxx |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git desktop/source/app/appinit.cxx desktop/source/app/appinit.cxx
index 8a48325..912e9df 100644
--- desktop/source/app/appinit.cxx
+++ desktop/source/app/appinit.cxx
@@ -146,7 +146,8 @@ static bool configureUcb(bool bServer, rtl::OUString const & rPortalConnect)
                 );
                 rtl::OUString aDesktopEnvironment;
                 if ((aValue >>= aDesktopEnvironment)
-                    && aDesktopEnvironment.equalsAscii("GNOME"))
+                    && (aDesktopEnvironment.equalsAscii("GNOME") ||
+                        aDesktopEnvironment.equalsAscii("KDE")))
                 {
                     Reference<XContentProviderManager> xCPM =
                         cb->getContentProviderManagerInterface();
-- 
1.7.0.1

