From 3a9dfd0b9083f8feb8b4b3452a4128f67ef45118 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:58:13 +0200
Subject: [PATCH 305/878] vfs-kde-too.diff

---
 desktop/source/app/appinit.cxx |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/desktop/source/app/appinit.cxx b/desktop/source/app/appinit.cxx
index 9b55744..b1a7769 100644
--- a/desktop/source/app/appinit.cxx
+++ b/desktop/source/app/appinit.cxx
@@ -146,7 +146,8 @@ static bool configureUcb(bool bServer, rtl::OUString const & rPortalConnect)
                 );
                 rtl::OUString aDesktopEnvironment;
                 if ((aValue >>= aDesktopEnvironment)
-                    && aDesktopEnvironment.equalsAscii("GNOME"))
+                    && (aDesktopEnvironment.equalsAscii("GNOME") ||
+                        aDesktopEnvironment.equalsAscii("KDE")))
                 {
                     Reference<XContentProviderManager> xCPM =
                         cb->getContentProviderManagerInterface();
-- 
1.7.0.1

