From d6ed94e4949b322bc4829d3860c6db77ed317fee Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:58:34 +0200
Subject: [PATCH 324/878] warning-uninitialized-variables-tools-2.diff

---
 tools/source/inet/inetmime.cxx |    8 ++++++--
 1 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/tools/source/inet/inetmime.cxx b/tools/source/inet/inetmime.cxx
index 8ab2cd4..0a2a6fa 100644
--- a/tools/source/inet/inetmime.cxx
+++ b/tools/source/inet/inetmime.cxx
@@ -3065,7 +3065,9 @@ ByteString INetMIME::decodeUTF8(const ByteString & rText,
     ByteString sDecoded;
     while (p != pEnd)
     {
-        sal_uInt32 nCharacter;
+        // the value is initialized just to avoid warning: ‘nCharacter’ is used uninitialized in this function
+        // there seems to be a bug in gcc-4.1 because this value is used only if it is initialized
+        sal_uInt32 nCharacter=0;
         if (translateUTF8Char(p, pEnd, eEncoding, nCharacter))
             sDecoded += sal_Char(nCharacter);
         else
@@ -3498,7 +3500,9 @@ UniString INetMIME::decodeHeaderFieldBody(HeaderFieldType eType,
             {
                 const sal_Char * pUTF8Begin = p - 1;
                 const sal_Char * pUTF8End = pUTF8Begin;
-                sal_uInt32 nCharacter;
+                // the value is initialized just to avoid warning: ‘nCharacter’ is used uninitialized in this function
+                // there seems to be a bug in gcc-4.1 because this value is used only if it is initialized
+                sal_uInt32 nCharacter=0;
                 if (translateUTF8Char(pUTF8End, pEnd, RTL_TEXTENCODING_UCS4,
                                       nCharacter))
                 {
-- 
1.7.0.1

