--- sdext/source/pdfimport/pdfiadaptor.cxx
+++ sdext/source/pdfimport/pdfiadaptor.cxx
@@ -254,7 +254,7 @@
 
     bool bSuccess=false;
 
-    if( xInput.is() && (!rURL.getLength() || rURL.compareToAscii( "file:", 5 ) != 0) )
+    if( xInput.is() )
         bSuccess = xpdf_ImportFromStream( xInput, pSink, xIHdl, rPwd, m_xContext );
     else
         bSuccess = xpdf_ImportFromFile( rURL, pSink, xIHdl, rPwd, m_xContext );
--- sdext/source/pdfimport/wrapper/wrapper.cxx
+++ sdext/source/pdfimport/wrapper/wrapper.cxx
@@ -1041,7 +1041,11 @@
 
     osl_closeFile( aFile );
 
-    return bSuccess && xpdf_ImportFromFile( aURL, rSink, xIHdl, rPwd, xContext );
+    if ( bSuccess )
+        bSuccess = xpdf_ImportFromFile( aURL, rSink, xIHdl, rPwd, xContext );
+    osl_removeFile( aURL.pData );
+
+    return bSuccess;
 }
 
 }
