--- scp2/source/ooo/windowscustomaction_ooo.scp
+++ scp2/source/ooo/windowscustomaction_ooo.scp
@@ -502,15 +502,6 @@
 	Assignment2 = ("AdminExecuteSequence", "Not REMOVE=\"ALL\"", "end");
 End
 
-WindowsCustomAction gid_Customaction_ShowSurvey
-	Name = "ShowSurveyAfterUninstall";
-	Typ = "65";
-	Source = "relnotes.dll";
-	Target = "ShowSurveyAfter";
-	Inbinarytable = 1;
-	Assignment1 = ("InstallExecuteSequence", "REMOVE=\"ALL\" and Not UPGRADINGPRODUCTCODE", "end");
-End
-
 WindowsCustomAction gid_Customaction_SelectLanguage
 	Name = "SelectLanguage";
 	Typ = "321";
