diff --git a/canvas/source/cairo/cairo_win32_cairo.cxx b/canvas/source/cairo/cairo_win32_cairo.cxx
--- canvas/source/cairo/cairo_win32_cairo.cxx
+++ canvas/source/cairo/cairo_win32_cairo.cxx
@@ -28,6 +28,8 @@
  *
  ************************************************************************/
 
+#define _WIN32_WINNT 0x0500
+
 // MARKER(update_precomp.py): autogen include statement, do not remove
 #include "precompiled_canvas.hxx"
 
diff --git a/connectivity/source/drivers/ado/ADriver.cxx b/connectivity/source/drivers/ado/ADriver.cxx
--- connectivity/source/drivers/ado/ADriver.cxx
+++ connectivity/source/drivers/ado/ADriver.cxx
@@ -28,6 +28,8 @@
  *
  ************************************************************************/
 
+#define _WIN32_WINNT 0x0501
+
 // MARKER(update_precomp.py): autogen include statement, do not remove
 #include "precompiled_connectivity.hxx"
 #include "ado/ADriver.hxx"
diff --git a/desktop/source/deployment/gui/dp_gui_extensioncmdqueue.cxx b/desktop/source/deployment/gui/dp_gui_extensioncmdqueue.cxx
--- desktop/source/deployment/gui/dp_gui_extensioncmdqueue.cxx
+++ desktop/source/deployment/gui/dp_gui_extensioncmdqueue.cxx
@@ -29,6 +29,8 @@
  *
  ************************************************************************/
 
+#define _WIN32_WINNT 0x0500
+
 // MARKER(update_precomp.py): autogen include statement, do not remove
 #include "precompiled_desktop.hxx"
 
diff --git a/desktop/win32/source/setup/setup.cpp b/desktop/win32/source/setup/setup.cpp
--- desktop/win32/source/setup/setup.cpp
+++ desktop/win32/source/setup/setup.cpp
@@ -29,6 +29,7 @@
  ************************************************************************/
 
 #define WIN // scope W32 API
+#define _WIN32_WINNT 0x0501
 
 #if defined _MSC_VER
 #pragma warning(push, 1)
diff --git a/fpicker/source/win32/filepicker/FileOpenDlg.cxx b/fpicker/source/win32/filepicker/FileOpenDlg.cxx
--- fpicker/source/win32/filepicker/FileOpenDlg.cxx
+++ fpicker/source/win32/filepicker/FileOpenDlg.cxx
@@ -28,6 +28,8 @@
  *
  ************************************************************************/
 
+#define _WIN32_WINNT 0x0500
+
 // MARKER(update_precomp.py): autogen include statement, do not remove
 #include "precompiled_fpicker.hxx"
 
diff --git a/i18npool/source/isolang/inwnt.cxx b/i18npool/source/isolang/inwnt.cxx
--- i18npool/source/isolang/inwnt.cxx
+++ i18npool/source/isolang/inwnt.cxx
@@ -32,6 +32,8 @@
 
 #include <sal/config.h>
 
+#define WINVER 0x0500
+
 #ifdef _MSC_VER
 #pragma warning(push,1) // disable warnings within system headers
 #endif
diff --git a/sal/osl/w32/file_dirvol.cxx b/sal/osl/w32/file_dirvol.cxx
--- sal/osl/w32/file_dirvol.cxx
+++ sal/osl/w32/file_dirvol.cxx
@@ -30,7 +30,7 @@
 
 #define UNICODE
 #define _UNICODE
-#define _WIN32_WINNT_0x0500
+#define _WIN32_WINNT 0x0500
 #include "systools/win32/uwinapi.h"
 
 #include "osl/file.h"
diff --git a/sal/osl/w32/file_error.c b/sal/osl/w32/file_error.c
--- sal/osl/w32/file_error.c
+++ sal/osl/w32/file_error.c
@@ -30,7 +30,7 @@
 
 #define UNICODE
 #define _UNICODE
-#define _WIN32_WINNT_0x0500
+#define _WIN32_WINNT 0x0500
 #include "systools/win32/uwinapi.h"
 
 #include "file_error.h"
diff --git a/sal/osl/w32/file_url.cxx b/sal/osl/w32/file_url.cxx
--- sal/osl/w32/file_url.cxx
+++ sal/osl/w32/file_url.cxx
@@ -30,7 +30,7 @@
 
 #define UNICODE
 #define _UNICODE
-#define _WIN32_WINNT_0x0500
+#define _WIN32_WINNT 0x0500
 #include "systools/win32/uwinapi.h"
 
 #include "file_url.h"
diff --git a/sal/osl/w32/tempfile.cxx b/sal/osl/w32/tempfile.cxx
--- sal/osl/w32/tempfile.cxx
+++ sal/osl/w32/tempfile.cxx
@@ -30,7 +30,7 @@
 
 #define UNICODE
 #define _UNICODE
-#define _WIN32_WINNT_0x0500
+#define _WIN32_WINNT 0x0500
 #include "systools/win32/uwinapi.h"
 
 #include "osl/file.h"
diff --git a/vcl/win/source/gdi/salprn.cxx b/vcl/win/source/gdi/salprn.cxx
--- vcl/win/source/gdi/salprn.cxx
+++ vcl/win/source/gdi/salprn.cxx
@@ -31,6 +31,8 @@
 // MARKER(update_precomp.py): autogen include statement, do not remove
 #include "precompiled_vcl.hxx"
 
+#define WINVER 0x0500
+
 #include <string.h>
 #include <tools/svwin.h>
 
