--- canvas/source/directx/dx_winstuff.hxx
+++ canvas/source/directx/dx_winstuff.hxx
@@ -89,9 +89,14 @@
 
 #else
 
+    #include <dxsdkver.h>
     #include <d3d9.h>
     #include <d3dx9.h>
+    #if _DXSDK_BUILD_MAJOR < 1734 /* Earlier than the August 2009 DXSDK */
-    #include <dxerr9.h>
+        #include <dxerr9.h>
+    #else
+        #include <dxerr.h>
+    #endif
 
     typedef IDirect3DSurface9 surface_type;
 
