From 9780f810716f6d4e8669cb07c0f8e876109b8381 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:05:25 +0200
Subject: [PATCH 670/878] win32-sdk7.diff

---
 configure.in |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/configure.in b/configure.in
index 7452ab8..d67b7c8 100644
--- a/configure.in
+++ b/configure.in
@@ -5493,6 +5493,9 @@ dnl Check if this is the right SDK.
     elif echo $PSDK_HOME | grep "v6.0" >/dev/null 2>/dev/null; then
         AC_MSG_RESULT([Found Windows SDK 6.0 ($PSDK_HOME)])
         WINDOWS_VISTA_PSDK=TRUE
+    elif echo $PSDK_HOME | grep "v7" >/dev/null 2>/dev/null; then
+        AC_MSG_RESULT([Found Windows SDK 7 ($PSDK_HOME)])
+        WINDOWS_VISTA_PSDK=TRUE
     else
         AC_MSG_RESULT([Found Legacy Windows Platform SDK ($PSDK_HOME)])
     fi
-- 
1.7.0.1

