--- sal/osl/w32/dllentry.c
+++ sal/osl/w32/dllentry.c
@@ -197,8 +197,10 @@
                     osl_isSingleCPU = 1;
                 }
 #endif
+#if OSL_DEBUG_LEVEL < 2
                 /* Suppress file error messages from system like "Floppy A: not inserted" */
                 SetErrorMode( SEM_NOOPENFILEERRORBOX | SEM_FAILCRITICALERRORS );
+#endif
 
                 /* initialize global mutex */
                 g_Mutex = osl_createMutex();
--- sal/osl/w32/module.c
+++ sal/osl/w32/module.c
@@ -48,7 +48,9 @@
 oslModule SAL_CALL osl_loadModule(rtl_uString *strModuleName, sal_Int32 nRtldMode )
 {
     HINSTANCE hInstance;
+#if OSL_DEBUG_LEVEL < 2
     UINT errorMode = SetErrorMode(SEM_NOOPENFILEERRORBOX | SEM_FAILCRITICALERRORS);
+#endif
     rtl_uString* Module = NULL;
     oslModule ret = 0;
     oslFileError	nError;
@@ -74,7 +74,9 @@
 
     ret = (oslModule) hInstance;
     rtl_uString_release(Module);
+#if OSL_DEBUG_LEVEL < 2
     SetErrorMode(errorMode);
+#endif
 
     RTL_LOGFILE_TRACE1( "} osl_loadModule end: %S", (LPTSTR)&strModuleName->buffer );
 
