--- sal/osl/w32/file_url.h
+++ sal/osl/w32/file_url.h
@@ -175,6 +175,7 @@
 #define VALIDATEPATH_ALLOW_ELLIPSE			0x0002
 #define VALIDATEPATH_ALLOW_RELATIVE			0x0004
 #define VALIDATEPATH_ALLOW_UNC				0x0008
+#define VALIDATEPATH_ALLOW_INVALID_SPACE_AND_PERIOD 0x0010
 
 DWORD IsValidFilePath (
     rtl_uString *  path,
--- sal/osl/w32/file_url.cxx
+++ sal/osl/w32/file_url.cxx
@@ -1108,9 +1109,10 @@
 			    case '.':
 				    if ( dwFlags & VALIDATEPATH_ALLOW_ELLIPSE )
 				    {
-					    if ( 1 == lpCurrent - lpComponent )
+                        if ( (dwFlags & VALIDATEPATH_ALLOW_INVALID_SPACE_AND_PERIOD) ||
+                             1 == lpCurrent - lpComponent )
 					    {
-						    /* Current directory is O.K. */
+                            /* Either do allow periods anywhere, or current directory */
 						    lpComponentEnd = lpCurrent;
 						    break;
 					    }
@@ -1123,8 +1125,13 @@ namespace /* private */
 				    }
 			    case 0:
 			    case ' ':
-				    lpComponentEnd = lpCurrent - 1;
-				    fValid = FALSE;
+                    if ( dwFlags & VALIDATEPATH_ALLOW_INVALID_SPACE_AND_PERIOD )
+                        lpComponentEnd = lpCurrent;
+                    else
+                    {
+                        lpComponentEnd = lpCurrent - 1;
+                        fValid = FALSE;
+                    }
 				    break;
 			    default:
 				    lpComponentEnd = lpCurrent;
@@ -1365,7 +1372,7 @@ namespace /* private */
                 lpComponent = lpszPath + i;
             }
 
-		    fValid = IsValidFilePathComponent( lpComponent, &lpComponent, dwFlags );
+            fValid = IsValidFilePathComponent( lpComponent, &lpComponent, dwFlags | VALIDATEPATH_ALLOW_INVALID_SPACE_AND_PERIOD);
 
 		    if ( fValid && lpComponent )
 		    {
