--- sdext/source/pdfimport/wrapper/wrapper.cxx	2010-01-13 10:39:42.000000000 +0100
+++ sdext/source/pdfimport/wrapper/wrapper.cxx	2010-01-13 10:41:01.000000000 +0100
@@ -827,6 +827,9 @@
                              bool&                                              o_rIsEncrypted
                              )
 {
+    // Ugly temporary workaround to some file-locking
+    return true;
+
     bool bSuccess = false;
     rtl::OString aPDFFile;
     aPDFFile = rtl::OUStringToOString( i_rPath, osl_getThreadTextEncoding() );
