--- solenv/bin/deliver.pl
+++ solenv/bin/deliver.pl
@@ -113,9 +113,11 @@
 
 if ($^O ne 'cygwin') {              # iz59477 - cygwin needes a dot "." at the end of filenames to disable
   $maybedot     = '';               # some .exe transformation magic.
-} else {
+} elsif ( `uname -r` lt '1.7') {    # but not in cygwin 1.7 it seems
   $maybedot     = '.';
-}
+} else {
+  $maybedot     = '';
+}
 
 ($gui		= lc($ENV{GUI})) 		|| die "Can't determine 'GUI'. Please set environment.\n";
 $tempcounter        = 0;


--- packimages/pack/makefile.mk
+++ packimages/pack/makefile.mk
@@ -91,7 +91,7 @@
 
 # unpack the classic icon set
 $(MISC)$/classic.flag : $(CLASSIC_TARBALL)
-    cd $(MISC) && gunzip -c $(CLASSIC_TARBALL) | ( tar -xf - ) && $(TOUCH) $(@:f)
+    cd $(MISC) && gunzip -d -c $(CLASSIC_TARBALL) | ( tar -xf - ) && $(TOUCH) $(@:f)
 .IF "$(GUI)"=="UNX"
     chmod -R g+w $(MISC)$/classic
 .ENDIF
