From 4cb6d64f3a5257d5693a3fe878d137a2de6be328 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:05:27 +0200
Subject: [PATCH 672/878] win32-interix.diff

---
 config.guess |    2 +-
 configure.in |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/config.guess b/config.guess
index 8229471..d0d5d63 100755
--- a/config.guess
+++ b/config.guess
@@ -789,7 +789,7 @@ EOF
     i*:PW*:*)
 	echo ${UNAME_MACHINE}-pc-pw32
 	exit 0 ;;
-    x86:Interix*:[34]*)
+    x86:Interix*:[3-6]*)
 	echo i586-pc-interix${UNAME_RELEASE}|sed -e 's/\..*//'
 	exit 0 ;;
     [345]86:Windows_95:* | [345]86:Windows_98:* | [345]86:Windows_NT:*)
diff --git a/configure.in b/configure.in
index d67b7c8..d5e95a1 100644
--- a/configure.in
+++ b/configure.in
@@ -1004,7 +1004,7 @@ case "$build_os" in
 		test_cups=no
 		_os=GNU
 		;;
-	cygwin*) # Windows
+    cygwin*|interix*) # Windows
 		test_cups=no
 		test_cairo=yes
 		test_freetype=no
-- 
1.7.0.1

