--- solenv/bin/mhids.pl
+++ solenv/bin/mhids.pl
@@ -151,7 +151,11 @@
 $filebase =~ s/.*[\\\/]//;
 $filebase =~ s/\..*?$//;
 $workfile = "$tmpdir/${filebase}_".$$;
-#$workfile =~ s/setup/set_up/;
+# Avoid UAC prompts on Windows
+$workfile =~ s/setup/set_up/;
+$workfile =~ s/patch/pat_ch/;
+$workfile =~ s/update/up_date/;
+$workfile =~ s/install/in_stall/;
 
 # now get $workfile ready for shell usage...
 $shell_workfile = $workfile;
