--- scp2/source/base/folderitem_base.ulf
+++ scp2/source/base/folderitem_base.ulf
@@ -35,5 +35,5 @@
  ************************************************************************/
 
 [STR_FI_TOOLTIP_BASE]
-en-US = "Manage databases, create queries and reports to track and manage your information by using Base."
+en-US = "Manage databases. Create queries and reports against databases."
 
--- scp2/source/calc/folderitem_calc.ulf
+++ scp2/source/calc/folderitem_calc.ulf
@@ -35,4 +35,4 @@
 en-US = "Spreadsheet"
 
 [STR_FI_TOOLTIP_CALC]
-en-US = "Perform calculation, analyze information and manage lists in spreadsheets by using Calc."
+en-US = "Spreadsheet calculator."
--- scp2/source/impress/folderitem_impress.ulf
+++ scp2/source/impress/folderitem_impress.ulf
@@ -35,6 +35,6 @@
 en-US = "Presentation"
 
 [STR_FI_TOOLTIP_IMPRESS]
-en-US = "Create and edit presentations for slideshows, meeting and Web pages by using Impress."
+en-US = "Edit presentations."
 
 
--- scp2/source/math/folderitem_math.ulf
+++ scp2/source/math/folderitem_math.ulf
@@ -35,6 +35,6 @@
  ************************************************************************/
 
 [STR_FI_TOOLTIP_MATH]
-en-US = "Create and edit scientific formulas and equations by using Math."
+en-US = "Edit mathematical formulas."
 
 
--- scp2/source/draw/folderitem_draw.ulf
+++ scp2/source/draw/folderitem_draw.ulf
@@ -35,6 +35,6 @@
 en-US = "Drawing"
 
 [STR_FI_TOOLTIP_DRAW]
-en-US = "Create and edit drawings, flow charts, and logos by using Draw."
+en-US = "Edit drawings."
 
 
--- scp2/source/ooo/folderitem_ooo.ulf
+++ scp2/source/ooo/folderitem_ooo.ulf
@@ -35,4 +35,4 @@
 en-US = "Open Document"
 
 [STR_FI_TOOLTIP_SOFFICE]
-en-US = "The office productivity suite compatible to the open and standardized ODF document format. Supported by Sun Microsystems."
+en-US = "Office productivity suite supporting many document formats."
--- scp2/source/writer/folderitem_writer.ulf
+++ scp2/source/writer/folderitem_writer.ulf
@@ -35,4 +35,4 @@
 en-US = "Text Document"
 
 [STR_FI_TOOLTIP_WRITER]
-en-US = "Create and edit text and graphics in letters, reports, documents and Web pages by using Writer."
+en-US = "Edit documents containing mostly text."
