From d61e5273e2e8a716cd015921b5e4fbca07724681 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:54:19 +0200
Subject: [PATCH 099/878] wizards-source-importwizard-filesmodul-xba.diff

---
 wizards/source/importwizard/FilesModul.xba |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/wizards/source/importwizard/FilesModul.xba b/wizards/source/importwizard/FilesModul.xba
index cb99925..e35e0b3 100644
--- a/wizards/source/importwizard/FilesModul.xba
+++ b/wizards/source/importwizard/FilesModul.xba
@@ -41,6 +41,7 @@ Dim NewContentList() as String
 Dim XMLTemplateContentString as String
 Dim ApplIndex as Integer
 Dim bAssignFileName as Boolean
+Dim bInterruptSearch as Boolean
 	bInterruptSearch = False
 	For i = 0 To MaxCollectIndex
 		SearchDir = PathCollection(i,0)
-- 
1.7.0.1

