---
 wizards/source/importwizard/FilesModul.xba |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git wizards/source/importwizard/FilesModul.xba wizards/source/importwizard/FilesModul.xba
index cb99925..e35e0b3 100644
--- wizards/source/importwizard/FilesModul.xba
+++ wizards/source/importwizard/FilesModul.xba
@@ -41,6 +41,7 @@ Dim NewContentList() as String
 Dim XMLTemplateContentString as String
 Dim ApplIndex as Integer
 Dim bAssignFileName as Boolean
+Dim bInterruptSearch as Boolean
 	bInterruptSearch = False
 	For i = 0 To MaxCollectIndex
 		SearchDir = PathCollection(i,0)
-- 
1.7.0.1

