diff -rup svtools-orig/source/filter.vcl/wmf/winmtf.cxx svtools/source/filter.vcl/wmf/winmtf.cxx
--- svtools-orig/source/filter.vcl/wmf/winmtf.cxx	2009-08-28 12:49:41.000000000 +0200
+++ svtools/source/filter.vcl/wmf/winmtf.cxx	2009-08-31 17:52:27.000000000 +0200
@@ -422,10 +422,15 @@ Point WinMtfOutput::ImplMap( const Point
 			        case MM_TEXT:
  					fX2 -= mnWinOrgX;
  					fY2 -= mnWinOrgY;
-					fX2 *= 2540.0/mnUnitsPerInch;
-					fY2 *= 2540.0/mnUnitsPerInch;
+					if( mnDevWidth != 1 || mnDevHeight != 1 ) {
+					    fX2 *= 2540.0/mnUnitsPerInch;
+					    fY2 *= 2540.0/mnUnitsPerInch;
+					}
  					fX2 += mnDevOrgX;
  					fY2 += mnDevOrgY;
+					fX2 *= (double)mnMillX * 100.0 / (double)mnPixX;
+					fY2 *= (double)mnMillY * 100.0 / (double)mnPixY;
+
 					break;
                 case MM_LOENGLISH :
                 {
@@ -503,8 +508,13 @@ Size WinMtfOutput::ImplMap( const Size&
             switch( mnMapMode )
             {
 			    case MM_TEXT:
+				if( mnDevWidth != 1 && mnDevHeight != 1 ) {
 					fWidth *= 2540.0/mnUnitsPerInch;
 					fHeight*= 2540.0/mnUnitsPerInch;
+				} else {
+				    fWidth *= (double)mnMillX * 100 / (double)mnPixX;
+				    fHeight *= (double)mnMillY * 100 / (double)mnPixY;
+				}
 				break;
                 case MM_LOENGLISH :
                 {
