--- svtools/source/filter.vcl/wmf/winmtf.cxx.old	2010-03-03 16:59:57.000000000 +0100
+++ svtools/source/filter.vcl/wmf/winmtf.cxx	2010-03-03 17:00:09.000000000 +0100
@@ -419,6 +419,14 @@ Point WinMtfOutput::ImplMap( const Point
         {
             switch( mnMapMode )
             {
+			        case MM_TEXT:
+ 					fX2 -= mnWinOrgX;
+ 					fY2 -= mnWinOrgY;
+					fX2 *= 2540.0/mnUnitsPerInch;
+					fY2 *= 2540.0/mnUnitsPerInch;
+ 					fX2 += mnDevOrgX;
+ 					fY2 += mnDevOrgY;
+					break;
                 case MM_LOENGLISH :
                 {
                     fX2 -= mnWinOrgX;
@@ -494,6 +502,10 @@ Size WinMtfOutput::ImplMap( const Size&
         {
             switch( mnMapMode )
             {
+			    case MM_TEXT:
+					fWidth *= 2540.0/mnUnitsPerInch;
+					fHeight*= 2540.0/mnUnitsPerInch;
+				break;
                 case MM_LOENGLISH :
                 {
                     fWidth *= 25.40;
@@ -930,6 +942,7 @@ WinMtfOutput::WinMtfOutput( GDIMetaFile&
     mbFillStyleSelected	( sal_False ),
     mnGfxMode			( GM_COMPATIBLE ),
     mnMapMode           ( MM_TEXT ),
+    mnUnitsPerInch ( 96 ),
     mnDevOrgX			( 0 ),
     mnDevOrgY			( 0 ),
     mnDevWidth			( 1 ),
@@ -1693,7 +1706,7 @@ void WinMtfOutput::ImplDrawBitmap( const
     if ( aBmpEx.IsTransparent() )
         mpGDIMetaFile->AddAction( new MetaBmpExScaleAction( rPos, rSize, aBmpEx ) );
     else
-        mpGDIMetaFile->AddAction( new MetaBmpScaleAction( rPos, rSize, aBmpEx.GetBitmap() ) );
+	    mpGDIMetaFile->AddAction( new MetaBmpScaleAction( rPos, rSize, aBmpEx.GetBitmap() ) );
 }
 
 //-----------------------------------------------------------------------------------
@@ -2055,6 +2068,14 @@ void WinMtfOutput::SetMapMode( sal_uInt3
 }
 
 //-----------------------------------------------------------------------------------
+
+void WinMtfOutput::SetUnitsPerInch( UINT16 nUnitsPerInch )
+{
+    if( nUnitsPerInch != 0 )
+	mnUnitsPerInch = nUnitsPerInch;
+}
+
+//-----------------------------------------------------------------------------------
 
 void WinMtfOutput::SetWorldTransform( const XForm& rXForm )
 {
--- svtools/source/filter.vcl/wmf/winmtf.hxx.old	2010-03-03 17:00:09.000000000 +0100
+++ svtools/source/filter.vcl/wmf/winmtf.hxx	2010-03-03 17:00:09.000000000 +0100
@@ -590,6 +590,8 @@ class WinMtfOutput
 
         sal_uInt32			mnGfxMode;
         sal_uInt32          mnMapMode;
+	UINT16			mnUnitsPerInch;
+
         XForm				maXForm;
         sal_Int32			mnDevOrgX, mnDevOrgY;
         sal_Int32			mnDevWidth, mnDevHeight;
@@ -636,6 +638,7 @@ class WinMtfOutput
 
         sal_uInt32			GetMapMode() const { return mnMapMode; };
         void                SetMapMode( sal_uInt32 mnMapMode );
+    void SetUnitsPerInch( UINT16 nUnitsPerInch );
         void				SetWorldTransform( const XForm& rXForm );
         void				ModifyWorldTransform( const XForm& rXForm, UINT32 nMode );
 
--- svtools/source/filter.vcl/wmf/winwmf.cxx.old	2010-03-03 17:00:09.000000000 +0100
+++ svtools/source/filter.vcl/wmf/winwmf.cxx	2010-03-03 17:00:09.000000000 +0100
@@ -990,6 +990,7 @@ BOOL WMFReader::ReadHeader(WMF_APMFILEHE
 	  }
 	}	
 
+	pOut->SetUnitsPerInch( nUnitsPerInch );
     pOut->SetWinOrg( aPlaceableBound.TopLeft() );
     aWMFSize = Size( labs( aPlaceableBound.GetWidth() ), labs( aPlaceableBound.GetHeight() ) );
     pOut->SetWinExt( aWMFSize );
