---
 configure.in |   21 +++++++++++++++++++++
 set_soenv.in |    3 +++
 2 files changed, 24 insertions(+), 0 deletions(-)

diff --git configure.in configure.in
index 0f658c8..a062c42 100644
--- configure.in
+++ configure.in
@@ -343,6 +343,9 @@ AC_ARG_WITH(system-expat,
 AC_ARG_WITH(system-libwpd,
 [  --with-system-libwpd    Use libwpd already on system
 ],,)
+AC_ARG_WITH(system-libwps,
+[  --with-system-libwps    Use libwps already on system
+],,)
 AC_ARG_WITH(system-libxml,
 [  --with-system-libxml    Use libxml already on system
 ],,)
@@ -3644,6 +3647,24 @@ AC_SUBST(FREETYPE_CFLAGS)
 AC_SUBST(FREETYPE_LIBS)
 
 dnl ===================================================================
+dnl Check for system libwps
+dnl ===================================================================
+AC_MSG_CHECKING([which libwps to use])
+if test -n "$with_system_libwps" -o -n "$with_system_libs" && \
+        test "$with_system_libwps" != "no"; then
+    AC_MSG_RESULT([external])
+    SYSTEM_LIBWPS=YES
+    PKG_CHECK_MODULES( LIBWPS, libwps-0.1 )
+else
+    AC_MSG_RESULT([internal])
+    SYSTEM_LIBWPS=NO
+    BUILD_TYPE="$BUILD_TYPE LIBWPS"
+fi
+AC_SUBST(SYSTEM_LIBWPS)
+AC_SUBST(LIBWPS_CFLAGS)
+AC_SUBST(LIBWPS_LIBS)
+
+dnl ===================================================================
 dnl Check whether freetype2 supports emboldening
 dnl ===================================================================
 if test  "$test_freetype" = "yes"; then
diff --git set_soenv.in set_soenv.in
index 7983933..94a3cd9 100644
--- set_soenv.in
+++ set_soenv.in
@@ -1859,6 +1859,9 @@ ToFile( "XRANDR_DLOPEN",     "@XRANDR_DLOPEN@",    "e" );
 ToFile( "SYSTEM_LIBWPD",     "@SYSTEM_LIBWPD@",    "e" );
 ToFile( "LIBWPD_CFLAGS",     "@LIBWPD_CFLAGS@",    "e" );
 ToFile( "LIBWPD_LIBS",       "@LIBWPD_LIBS@",      "e" );
+ToFile( "SYSTEM_LIBWPS",     "@SYSTEM_LIBWPS@",    "e" );
+ToFile( "LIBWPS_CFLAGS",     "@LIBWPS_CFLAGS@",    "e" );
+ToFile( "LIBWPS_LIBS",       "@LIBWPS_LIBS@",      "e" );
 ToFile( "SYSTEM_CPPUNIT",    "@SYSTEM_CPPUNIT@",   "e" );
 ToFile( "CPPUNIT_CFLAGS",    "@CPPUNIT_CFLAGS@",   "e" );
 ToFile( "CPPUNIT_LIBS",      "@CPPUNIT_LIBS@",     "e" );
-- 
1.7.0.1

