---
 writerperfect/source/filter/TextRunStyle.cxx |    7 ++++++-
 1 files changed, 6 insertions(+), 1 deletions(-)

diff --git writerperfect/source/filter/TextRunStyle.cxx writerperfect/source/filter/TextRunStyle.cxx
index a11e4a7..26fa6cc 100644
--- writerperfect/source/filter/TextRunStyle.cxx
+++ writerperfect/source/filter/TextRunStyle.cxx
@@ -77,7 +77,12 @@ void ParagraphStyle::write(DocumentHandler *pHandler) const
         if (strcmp(k.key(), "fo:margin-top") == 0)
             propList.insert("fo:margin-top", k()->getStr());
         if (strcmp(k.key(), "fo:margin-bottom") == 0)
-            propList.insert("fo:margin-bottom", k()->getStr());
+        {
+            if (k()->getFloat() > 0.0f)
+                propList.insert("fo:margin-bottom", k()->getStr());
+            else
+                propList.insert("fo:margin-bottom", 0.0f);
+        }
         if (strcmp(k.key(), "fo:line-height") == 0)
             propList.insert("fo:line-height", k()->getStr());
         if (strcmp(k.key(), "fo:break-before") == 0) 
-- 
1.7.0.1

