diff --git sc/source/filter/inc/XclExpChangeTrack.hxx sc/source/filter/inc/XclExpChangeTrack.hxx
index a9ece3d..0d9407b 100644
--- sc/source/filter/inc/XclExpChangeTrack.hxx
+++ sc/source/filter/inc/XclExpChangeTrack.hxx
@@ -31,6 +31,10 @@
 #ifndef SC_XCLEXPCHANGETRACK_HXX
 #define SC_XCLEXPCHANGETRACK_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include <tools/debug.hxx>
 #include <tools/datetime.hxx>
 #include <rtl/uuid.h>
diff --git sc/source/filter/inc/excdoc.hxx sc/source/filter/inc/excdoc.hxx
index 3330335..6d6858e 100644
--- sc/source/filter/inc/excdoc.hxx
+++ sc/source/filter/inc/excdoc.hxx
@@ -31,6 +31,10 @@
 #ifndef SC_EXCDOC_HXX
 #define SC_EXCDOC_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include <tools/solar.h>
 #include "excrecds.hxx"
 #include "xeroot.hxx"
diff --git sc/source/filter/inc/excrecds.hxx sc/source/filter/inc/excrecds.hxx
index ba8a573..c05dd6b 100644
--- sc/source/filter/inc/excrecds.hxx
+++ sc/source/filter/inc/excrecds.hxx
@@ -31,6 +31,10 @@
 #ifndef SC_EXCRECDS_HXX
 #define SC_EXCRECDS_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include <tools/solar.h>
 #include <svtools/zforlist.hxx>
 #include <tools/string.hxx>
diff --git sc/source/filter/inc/exp_op.hxx sc/source/filter/inc/exp_op.hxx
index 0e05e00..0a306d3 100644
--- sc/source/filter/inc/exp_op.hxx
+++ sc/source/filter/inc/exp_op.hxx
@@ -31,6 +31,10 @@
 #ifndef SC_EXP_OP_HXX
 #define SC_EXP_OP_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include "filter.hxx"
 #include "root.hxx"
 #include "xeroot.hxx"
diff --git sc/source/filter/inc/xcl97esc.hxx sc/source/filter/inc/xcl97esc.hxx
index 67a6e3f..e9c6f5c 100644
--- sc/source/filter/inc/xcl97esc.hxx
+++ sc/source/filter/inc/xcl97esc.hxx
@@ -31,6 +31,10 @@
 #ifndef SC_XCL97ESC_HXX
 #define SC_XCL97ESC_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include <svx/escherex.hxx>
 #include <tools/table.hxx>
 #include <tools/stack.hxx>
diff --git sc/source/filter/inc/xcl97rec.hxx sc/source/filter/inc/xcl97rec.hxx
index 30b6e8a..54032e0 100644
--- sc/source/filter/inc/xcl97rec.hxx
+++ sc/source/filter/inc/xcl97rec.hxx
@@ -31,6 +31,10 @@
 #ifndef SC_XCL97REC_HXX
 #define SC_XCL97REC_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include "excrecds.hxx"
 #include "xcl97esc.hxx"
 #include "xlstyle.hxx"
diff --git sc/source/filter/inc/xechart.hxx sc/source/filter/inc/xechart.hxx
index b5964ea..fbda089 100644
--- sc/source/filter/inc/xechart.hxx
+++ sc/source/filter/inc/xechart.hxx
@@ -31,6 +31,10 @@
 #ifndef SC_XECHART_HXX
 #define SC_XECHART_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include "xerecord.hxx"
 #include "xlchart.hxx"
 #include "xlformula.hxx"
diff --git sc/source/filter/inc/xecontent.hxx sc/source/filter/inc/xecontent.hxx
index aeb6c13..85e6543 100644
--- sc/source/filter/inc/xecontent.hxx
+++ sc/source/filter/inc/xecontent.hxx
@@ -31,6 +31,10 @@
 #ifndef SC_XECONTENT_HXX
 #define SC_XECONTENT_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include "rangelst.hxx"
 #include "xlcontent.hxx"
 #include "xladdress.hxx"
diff --git sc/source/filter/inc/xeescher.hxx sc/source/filter/inc/xeescher.hxx
index 8b3f9a6..0a4e29f 100644
--- sc/source/filter/inc/xeescher.hxx
+++ sc/source/filter/inc/xeescher.hxx
@@ -31,6 +31,10 @@
 #ifndef SC_XEESCHER_HXX
 #define SC_XEESCHER_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include <vcl/graph.hxx>
 #include "xcl97rec.hxx"
 #include "xlescher.hxx"
diff --git sc/source/filter/inc/xelink.hxx sc/source/filter/inc/xelink.hxx
index 7da521f..737bc72 100644
--- sc/source/filter/inc/xelink.hxx
+++ sc/source/filter/inc/xelink.hxx
@@ -31,6 +31,10 @@
 #ifndef SC_XELINK_HXX
 #define SC_XELINK_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include "markdata.hxx"
 #include "xllink.hxx"
 #include "xerecord.hxx"
diff --git sc/source/filter/inc/xename.hxx sc/source/filter/inc/xename.hxx
index 1c6f5e3..d847d1d 100644
--- sc/source/filter/inc/xename.hxx
+++ sc/source/filter/inc/xename.hxx
@@ -31,6 +31,10 @@
 #ifndef SC_XENAME_HXX
 #define SC_XENAME_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include "xerecord.hxx"
 #include "xlname.hxx"
 #include "xlformula.hxx"
diff --git sc/source/filter/inc/xepage.hxx sc/source/filter/inc/xepage.hxx
index 3e119d3..5e9f88d 100644
--- sc/source/filter/inc/xepage.hxx
+++ sc/source/filter/inc/xepage.hxx
@@ -31,6 +31,10 @@
 #ifndef SC_XEPAGE_HXX
 #define SC_XEPAGE_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include "xerecord.hxx"
 #include "xlpage.hxx"
 #include "xeroot.hxx"
diff --git sc/source/filter/inc/xepivot.hxx sc/source/filter/inc/xepivot.hxx
index de9d284..c5ff185 100644
--- sc/source/filter/inc/xepivot.hxx
+++ sc/source/filter/inc/xepivot.hxx
@@ -31,6 +31,10 @@
 #ifndef SC_XEPIVOT_HXX
 #define SC_XEPIVOT_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include <map>
 #include "xerecord.hxx"
 #include "xlpivot.hxx"
diff --git sc/source/filter/inc/xerecord.hxx sc/source/filter/inc/xerecord.hxx
index 8abd4d0..c592e8d 100644
--- sc/source/filter/inc/xerecord.hxx
+++ sc/source/filter/inc/xerecord.hxx
@@ -31,6 +31,10 @@
 #ifndef SC_XERECORD_HXX
 #define SC_XERECORD_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include "xlconst.hxx"
 #include "xestream.hxx"
 
diff --git sc/source/filter/inc/xestream.hxx sc/source/filter/inc/xestream.hxx
index 49bdf1d..d191b05 100644
--- sc/source/filter/inc/xestream.hxx
+++ sc/source/filter/inc/xestream.hxx
@@ -33,6 +33,10 @@
 #ifndef SC_XESTREAM_HXX
 #define SC_XESTREAM_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include <map>
 #include <stack>
 #include <string>
diff --git sc/source/filter/inc/xestyle.hxx sc/source/filter/inc/xestyle.hxx
index 9508c8b..b5db56c 100644
--- sc/source/filter/inc/xestyle.hxx
+++ sc/source/filter/inc/xestyle.hxx
@@ -31,6 +31,10 @@
 #ifndef SC_XESTYLE_HXX
 #define SC_XESTYLE_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include <map>
 #include <tools/mempool.hxx>
 #include <tools/string.hxx>
diff --git sc/source/filter/inc/xetable.hxx sc/source/filter/inc/xetable.hxx
index 2beb423..09e900c 100644
--- sc/source/filter/inc/xetable.hxx
+++ sc/source/filter/inc/xetable.hxx
@@ -31,6 +31,10 @@
 #ifndef SC_XETABLE_HXX
 #define SC_XETABLE_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include "xltable.hxx"
 
 #include <deque>
diff --git sc/source/filter/inc/xeview.hxx sc/source/filter/inc/xeview.hxx
index 5c54ecd..c2ce621 100644
--- sc/source/filter/inc/xeview.hxx
+++ sc/source/filter/inc/xeview.hxx
@@ -31,6 +31,10 @@
 #ifndef SC_XEVIEW_HXX
 #define SC_XEVIEW_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include "xerecord.hxx"
 #include "xlview.hxx"
 #include "xeroot.hxx"
diff --git sc/source/filter/xlsx/makefile.mk sc/source/filter/xlsx/makefile.mk
index 656ecf3..fdaa385 100644
--- sc/source/filter/xlsx/makefile.mk
+++ sc/source/filter/xlsx/makefile.mk
@@ -48,6 +48,8 @@ PROJECTPCHSOURCE=..\pch\filt_pch
 
 # --- Files --------------------------------------------------------
 
+CDEFS+=-DXLSX_GUARD
+
 SLOFILES =	\
 		$(SLO)$/xlsx-colrowst.obj				\
 		$(SLO)$/xlsx-excdoc.obj					\
