diff --git sc/source/filter/xlsx/xlsx-xeformula.cxx sc/source/filter/xlsx/xlsx-xeformula.cxx
index 31af0f9..c1989b8 100644
--- sc/source/filter/xlsx/xlsx-xeformula.cxx
+++ sc/source/filter/xlsx/xlsx-xeformula.cxx
@@ -619,7 +619,7 @@ void XclExpFmlaCompImpl::Init( XclFormulaType eType, const ScTokenArray& rScTokA
             DBG_ASSERT( mbOk, "XclExpFmlaCompImpl::Init - missing cell address" );
             // clone the passed token array, convert references relative to current cell position
             mxOwnScTokArr.reset( rScTokArr.Clone() );
-            ScCompiler::MoveRelWrap( *mxOwnScTokArr, GetDocPtr(), *pScBasePos );
+            ScCompiler::MoveRelWrap( *mxOwnScTokArr, GetDocPtr(), *pScBasePos, MAXCOL, MAXROW );
             // don't remember pScBasePos in mpScBasePos, shared formulas use real relative refs
         break;
         default:;
diff --git sc/source/filter/xlsx/xlsx-xiescher.cxx sc/source/filter/xlsx/xlsx-xiescher.cxx
index 1637ab1..48ebcb6 100644
--- sc/source/filter/xlsx/xlsx-xiescher.cxx
+++ sc/source/filter/xlsx/xlsx-xiescher.cxx
@@ -2791,6 +2791,12 @@ SdrObject* XclImpPictureObj::DoCreateSdrObj( const Rectangle& rAnchorRect, ScfPr
     return xSdrObj.release();
 }
 
+String XclImpPictureObj::GetObjName() const
+{
+    // dummy function definition.  This is never called.
+    return String();
+}
+
 void XclImpPictureObj::DoProcessSdrObj( SdrObject& rSdrObj ) const
 {
     if( IsOcxControl() )
