--- sc/source/filter/inc/XclExpChangeTrack.hxx.old	2009-10-08 16:17:08.000000000 +0200
+++ sc/source/filter/inc/XclExpChangeTrack.hxx	2010-03-03 17:00:06.000000000 +0100
@@ -31,6 +31,10 @@
 #ifndef SC_XCLEXPCHANGETRACK_HXX
 #define SC_XCLEXPCHANGETRACK_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include <tools/debug.hxx>
 #include <tools/datetime.hxx>
 #include <rtl/uuid.h>
--- sc/source/filter/inc/excdoc.hxx.old	2009-10-07 15:06:14.000000000 +0200
+++ sc/source/filter/inc/excdoc.hxx	2010-03-03 17:00:06.000000000 +0100
@@ -31,6 +31,10 @@
 #ifndef SC_EXCDOC_HXX
 #define SC_EXCDOC_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include <tools/solar.h>
 #include "excrecds.hxx"
 #include "xeroot.hxx"
--- sc/source/filter/inc/excrecds.hxx.old	2009-10-19 13:27:16.000000000 +0200
+++ sc/source/filter/inc/excrecds.hxx	2010-03-03 17:00:06.000000000 +0100
@@ -31,6 +31,10 @@
 #ifndef SC_EXCRECDS_HXX
 #define SC_EXCRECDS_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include <tools/solar.h>
 #include <svtools/zforlist.hxx>
 #include <tools/string.hxx>
--- sc/source/filter/inc/exp_op.hxx.old	2009-10-07 15:06:14.000000000 +0200
+++ sc/source/filter/inc/exp_op.hxx	2010-03-03 17:00:06.000000000 +0100
@@ -31,6 +31,10 @@
 #ifndef SC_EXP_OP_HXX
 #define SC_EXP_OP_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include "filter.hxx"
 #include "root.hxx"
 #include "xeroot.hxx"
--- sc/source/filter/inc/xcl97esc.hxx.old	2010-03-03 16:59:35.000000000 +0100
+++ sc/source/filter/inc/xcl97esc.hxx	2010-03-03 17:00:06.000000000 +0100
@@ -31,6 +31,10 @@
 #ifndef SC_XCL97ESC_HXX
 #define SC_XCL97ESC_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include <svx/escherex.hxx>
 #include <tools/table.hxx>
 #include <tools/stack.hxx>
--- sc/source/filter/inc/xcl97rec.hxx.old	2010-03-03 16:59:35.000000000 +0100
+++ sc/source/filter/inc/xcl97rec.hxx	2010-03-03 17:00:06.000000000 +0100
@@ -31,6 +31,10 @@
 #ifndef SC_XCL97REC_HXX
 #define SC_XCL97REC_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include "excrecds.hxx"
 #include "xcl97esc.hxx"
 #include "xlstyle.hxx"
--- sc/source/filter/inc/xechart.hxx.old	2010-03-03 16:59:13.000000000 +0100
+++ sc/source/filter/inc/xechart.hxx	2010-03-03 17:00:06.000000000 +0100
@@ -31,6 +31,10 @@
 #ifndef SC_XECHART_HXX
 #define SC_XECHART_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include "xerecord.hxx"
 #include "xlchart.hxx"
 #include "xlformula.hxx"
--- sc/source/filter/inc/xecontent.hxx.old	2010-03-03 16:59:35.000000000 +0100
+++ sc/source/filter/inc/xecontent.hxx	2010-03-03 17:00:06.000000000 +0100
@@ -31,6 +31,10 @@
 #ifndef SC_XECONTENT_HXX
 #define SC_XECONTENT_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include "rangelst.hxx"
 #include "xlcontent.hxx"
 #include "xladdress.hxx"
--- sc/source/filter/inc/xeescher.hxx.old	2010-03-03 16:59:35.000000000 +0100
+++ sc/source/filter/inc/xeescher.hxx	2010-03-03 17:00:06.000000000 +0100
@@ -31,6 +31,10 @@
 #ifndef SC_XEESCHER_HXX
 #define SC_XEESCHER_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include <vcl/graph.hxx>
 #include "xcl97rec.hxx"
 #include "xlescher.hxx"
--- sc/source/filter/inc/xelink.hxx.old	2009-10-08 16:17:08.000000000 +0200
+++ sc/source/filter/inc/xelink.hxx	2010-03-03 17:00:06.000000000 +0100
@@ -31,6 +31,10 @@
 #ifndef SC_XELINK_HXX
 #define SC_XELINK_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include "markdata.hxx"
 #include "xllink.hxx"
 #include "xerecord.hxx"
--- sc/source/filter/inc/xename.hxx.old	2009-10-08 16:17:08.000000000 +0200
+++ sc/source/filter/inc/xename.hxx	2010-03-03 17:00:06.000000000 +0100
@@ -31,6 +31,10 @@
 #ifndef SC_XENAME_HXX
 #define SC_XENAME_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include "xerecord.hxx"
 #include "xlname.hxx"
 #include "xlformula.hxx"
--- sc/source/filter/inc/xepage.hxx.old	2009-10-07 15:06:14.000000000 +0200
+++ sc/source/filter/inc/xepage.hxx	2010-03-03 17:00:06.000000000 +0100
@@ -31,6 +31,10 @@
 #ifndef SC_XEPAGE_HXX
 #define SC_XEPAGE_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include "xerecord.hxx"
 #include "xlpage.hxx"
 #include "xeroot.hxx"
--- sc/source/filter/inc/xepivot.hxx.old	2010-03-03 16:44:01.000000000 +0100
+++ sc/source/filter/inc/xepivot.hxx	2010-03-03 17:00:06.000000000 +0100
@@ -31,6 +31,10 @@
 #ifndef SC_XEPIVOT_HXX
 #define SC_XEPIVOT_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include <map>
 #include "xerecord.hxx"
 #include "xlpivot.hxx"
--- sc/source/filter/inc/xerecord.hxx.old	2009-10-08 16:17:08.000000000 +0200
+++ sc/source/filter/inc/xerecord.hxx	2010-03-03 17:00:06.000000000 +0100
@@ -31,6 +31,10 @@
 #ifndef SC_XERECORD_HXX
 #define SC_XERECORD_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include "xlconst.hxx"
 #include "xestream.hxx"
 
--- sc/source/filter/inc/xestream.hxx.old	2010-03-03 16:59:16.000000000 +0100
+++ sc/source/filter/inc/xestream.hxx	2010-03-03 17:00:06.000000000 +0100
@@ -33,6 +33,10 @@
 #ifndef SC_XESTREAM_HXX
 #define SC_XESTREAM_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include <map>
 #include <stack>
 #include <string>
--- sc/source/filter/inc/xestyle.hxx.old	2010-03-03 16:44:04.000000000 +0100
+++ sc/source/filter/inc/xestyle.hxx	2010-03-03 17:00:06.000000000 +0100
@@ -31,6 +31,10 @@
 #ifndef SC_XESTYLE_HXX
 #define SC_XESTYLE_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include <map>
 #include <tools/mempool.hxx>
 #include <tools/string.hxx>
--- sc/source/filter/inc/xetable.hxx.old	2009-10-08 16:17:08.000000000 +0200
+++ sc/source/filter/inc/xetable.hxx	2010-03-03 17:00:06.000000000 +0100
@@ -31,6 +31,10 @@
 #ifndef SC_XETABLE_HXX
 #define SC_XETABLE_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include "xltable.hxx"
 
 #include <deque>
--- sc/source/filter/inc/xeview.hxx.old	2010-03-03 16:44:04.000000000 +0100
+++ sc/source/filter/inc/xeview.hxx	2010-03-03 17:00:06.000000000 +0100
@@ -31,6 +31,10 @@
 #ifndef SC_XEVIEW_HXX
 #define SC_XEVIEW_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include "xerecord.hxx"
 #include "xlview.hxx"
 #include "xeroot.hxx"
--- sc/source/filter/xlsx/makefile.mk.old	2010-03-03 17:00:05.000000000 +0100
+++ sc/source/filter/xlsx/makefile.mk	2010-03-03 17:00:06.000000000 +0100
@@ -48,6 +48,8 @@ PROJECTPCHSOURCE=..\pch\filt_pch
 
 # --- Files --------------------------------------------------------
 
+CDEFS+=-DXLSX_GUARD
+
 SLOFILES =	\
 		$(SLO)$/xlsx-excdoc.obj					\
 		$(SLO)$/xlsx-excrecds.obj				\
--- sc/source/filter/inc/imp_op.hxx.old	2009-10-19 13:27:16.000000000 +0200
+++ sc/source/filter/inc/imp_op.hxx	2010-03-03 17:00:06.000000000 +0100
@@ -31,6 +31,10 @@
 #ifndef SC_IMP_OP_HXX
 #define SC_IMP_OP_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include <tools/gen.hxx>
 #include "xiroot.hxx"
 #include "xistream.hxx"
--- sc/source/filter/inc/xehelper.hxx.old	2009-10-07 15:06:14.000000000 +0200
+++ sc/source/filter/inc/xehelper.hxx	2010-03-03 17:00:06.000000000 +0100
@@ -31,6 +31,10 @@
 #ifndef SC_XEHELPER_HXX
 #define SC_XEHELPER_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include "xladdress.hxx"
 #include "xeroot.hxx"
 #include "xestring.hxx"
--- sc/source/filter/inc/xeroot.hxx.old	2009-10-08 16:17:08.000000000 +0200
+++ sc/source/filter/inc/xeroot.hxx	2010-03-03 17:00:06.000000000 +0100
@@ -31,6 +31,10 @@
 #ifndef SC_XEROOT_HXX
 #define SC_XEROOT_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include "xlroot.hxx"
 
 // Forward declarations of objects in public use ==============================
--- sc/source/filter/inc/xistyle.hxx.old	2010-03-03 16:59:35.000000000 +0100
+++ sc/source/filter/inc/xistyle.hxx	2010-03-03 17:00:06.000000000 +0100
@@ -31,6 +31,10 @@
 #ifndef SC_XISTYLE_HXX
 #define SC_XISTYLE_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include <list>
 #include <tools/mempool.hxx>
 #include "rangelst.hxx"
--- sc/source/filter/inc/xladdress.hxx.old	2009-10-07 15:06:14.000000000 +0200
+++ sc/source/filter/inc/xladdress.hxx	2010-03-03 17:00:06.000000000 +0100
@@ -31,6 +31,10 @@
 #ifndef SC_XLADDRESS_HXX
 #define SC_XLADDRESS_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include <vector>
 #include "address.hxx"
 
--- sc/source/filter/inc/xlescher.hxx.old	2010-03-03 16:59:35.000000000 +0100
+++ sc/source/filter/inc/xlescher.hxx	2010-03-03 17:00:06.000000000 +0100
@@ -31,6 +31,10 @@
 #ifndef SC_XLESCHER_HXX
 #define SC_XLESCHER_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include <vcl/mapunit.hxx>
 #include "fapihelper.hxx"
 #include "xladdress.hxx"
--- sc/source/filter/inc/xlpivot.hxx.old	2010-03-03 16:44:01.000000000 +0100
+++ sc/source/filter/inc/xlpivot.hxx	2010-03-03 17:00:06.000000000 +0100
@@ -31,6 +31,10 @@
 #ifndef SC_XLPIVOT_HXX
 #define SC_XLPIVOT_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include <com/sun/star/sheet/GeneralFunction.hpp>
 #include <com/sun/star/sheet/DataPilotFieldOrientation.hpp>
 #include <com/sun/star/sheet/DataPilotFieldSortMode.hpp>
--- sc/source/filter/inc/xlview.hxx.old	2010-03-03 16:44:04.000000000 +0100
+++ sc/source/filter/inc/xlview.hxx	2010-03-03 17:00:06.000000000 +0100
@@ -31,6 +31,10 @@
 #ifndef SC_XLVIEW_HXX
 #define SC_XLVIEW_HXX
 
+#ifdef XLSX_GUARD
+#error Should not include this from sc/source/filter/xlsx code
+#endif
+
 #include <map>
 #include <tools/color.hxx>
 #include "ftools.hxx"
