From 9e6917b365f346edc6297c0e20a0f7c4de4e4276 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:05:59 +0200
Subject: [PATCH 695/878] xlsx-export-notes-avoid-dereferencing-0-svx.diff

---
 svx/source/msfilter/eschesdo.cxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/svx/source/msfilter/eschesdo.cxx b/svx/source/msfilter/eschesdo.cxx
index 55ea6bd..80f7071 100644
--- a/svx/source/msfilter/eschesdo.cxx
+++ b/svx/source/msfilter/eschesdo.cxx
@@ -248,7 +248,7 @@ UINT32 ImplEESdrWriter::ImplWriteShape( ImplEESdrObject& rObj,
         // #i51348# shape name
         if( aShapeName.Len() > 0 )
             aPropOpt.AddOpt( ESCHER_Prop_wzName, aShapeName );
-        if ( InteractionInfo* pInteraction = mpHostAppData->GetInteractionInfo() )
+        if ( InteractionInfo* pInteraction = mpHostAppData ? mpHostAppData->GetInteractionInfo() : 0)
         {
             const std::auto_ptr< SvMemoryStream >& pMemStrm = pInteraction->getHyperlinkRecord();
             if ( pMemStrm.get() )
-- 
1.7.0.1

