diff -rup xmlhelp-orig/source/com/sun/star/help/makefile.mk xmlhelp/source/com/sun/star/help/makefile.mk
--- xmlhelp-orig/source/com/sun/star/help/makefile.mk	2009-10-12 14:21:40.000000000 +0200
+++ xmlhelp/source/com/sun/star/help/makefile.mk	2009-10-12 14:25:53.000000000 +0200
@@ -40,6 +40,8 @@ TARGETTYPE=CUI
 
 .INCLUDE : settings.mk
 .INCLUDE : helplinker.pmk
+
+.IF "$(SOLAR_JAVA)"=="TRUE"
  
 .IF "$(SYSTEM_LIBXSLT)" == "YES"
 CFLAGS+= $(LIBXSLT_CFLAGS)
@@ -109,3 +111,9 @@ fix_system_lucene:
     @echo "Fix Java Class-Path entry for Lucene libraries from system."
     @$(SED) -r -e "s#^(Class-Path:).*#\1 file://$(LUCENE_CORE_JAR) file://$(LUCENE_ANALYZERS_JAR)#" \
     -i ../../../../../$(INPATH)/class/HelpLinker/META-INF/MANIFEST.MF
+
+.ELSE
+
+.INCLUDE :  target.mk
+
+.ENDIF
