diff -Nrup xmloff-m16/source/draw/sdxmlexp.cxx xmloff/source/draw/sdxmlexp.cxx
--- xmloff-m16/source/draw/sdxmlexp.cxx	2009-07-27 00:38:42.000000000 -0700
+++ xmloff/source/draw/sdxmlexp.cxx	2009-07-29 15:37:06.000000000 -0700
@@ -47,6 +47,10 @@
 #include <com/sun/star/drawing/XMasterPagesSupplier.hpp>
 #include <com/sun/star/presentation/XHandoutMasterSupplier.hpp>
 #include <com/sun/star/container/XIndexContainer.hpp>
+#include <com/sun/star/presentation/XLayoutList.hpp>
+#include <com/sun/star/presentation/AutoLayoutBuiltin.hpp>
+#include <com/sun/star/awt/Rectangle.hpp>
+#include <com/sun/star/awt/Size.hpp>
 #include <com/sun/star/view/PaperOrientation.hpp>
 #include <com/sun/star/style/XStyleFamiliesSupplier.hpp>
 #include <com/sun/star/style/XStyle.hpp>
@@ -221,7 +225,7 @@ class ImpXMLAutoLayoutInfo
 	sal_Int32					mnGapY;
 
 public:
-	ImpXMLAutoLayoutInfo(sal_uInt16 nTyp, ImpXMLEXPPageMasterInfo* pInf);
+	ImpXMLAutoLayoutInfo(sal_uInt16 nTyp, ImpXMLEXPPageMasterInfo* pInf, Reference< XLayoutList > rLList);
 
 	BOOL operator==(const ImpXMLAutoLayoutInfo& rInfo) const;
 
@@ -253,7 +257,7 @@ BOOL ImpXMLAutoLayoutInfo::operator==(co
 		&& mpPageMasterInfo == rInfo.mpPageMasterInfo));
 }
 
-ImpXMLAutoLayoutInfo::ImpXMLAutoLayoutInfo(sal_uInt16 nTyp, ImpXMLEXPPageMasterInfo* pInf)
+ImpXMLAutoLayoutInfo::ImpXMLAutoLayoutInfo(sal_uInt16 nTyp, ImpXMLEXPPageMasterInfo* pInf, Reference< XLayoutList > rLList)
 :	mnType(nTyp),
 	mpPageMasterInfo(pInf)
 {
@@ -261,6 +265,25 @@ ImpXMLAutoLayoutInfo::ImpXMLAutoLayoutIn
 	Point aPagePos(0,0);
 	Size aPageSize(28000, 21000);
 	Size aPageInnerSize(28000, 21000);
+    
+    if(rLList.is())
+    {
+      ::uno::Sequence< ::awt::Rectangle >  rectlist;
+      ::awt::Rectangle  awtrect;
+      ::awt::Size  awtsize;
+    
+      awtrect.X = 0;
+      awtrect.Y = 0;
+      awtrect.Width = 28000;
+      awtrect.Height = 2100;
+      awtsize.Width = 28000;
+      awtsize.Height = 21000;
+      
+      rectlist = rLList->getAutoLayoutRectangles((sal_uInt32) nTyp, awtrect, awtrect, awtsize, false);
+        
+      sal_uInt32 r = (sal_uInt32) nTyp;
+        
+    }
 
 	if(mpPageMasterInfo)
 	{
@@ -886,9 +909,21 @@ BOOL SdXMLExport::ImpPrepAutoLayoutInfo(
 						}
 					}
 				}
+                
+                Reference< XLayoutList > llist;
+                
+                Reference< beans::XPropertySetInfo > xPropsInfo( xPropSet->getPropertySetInfo() );
+                if( xPropsInfo.is() )
+		        {
+                    if(xPropsInfo->hasPropertyByName(OUString(RTL_CONSTASCII_USTRINGPARAM("AutoLayoutList") )))
+                    {                
+                       aAny = xPropSet->getPropertyValue(OUString(RTL_CONSTASCII_USTRINGPARAM("AutoLayoutList")));
+                       aAny >>= llist;
+                    }
+                }    
 
 				// create entry and look for existance
-				ImpXMLAutoLayoutInfo* pNew = new ImpXMLAutoLayoutInfo(nType, pInfo);
+				ImpXMLAutoLayoutInfo* pNew = new ImpXMLAutoLayoutInfo(nType, pInfo, llist);
 				BOOL bDidExist(FALSE);
 
 				for(sal_uInt32 nCnt = 0L; !bDidExist && nCnt < mpAutoLayoutInfoList->Count(); nCnt++)
