From 1c386ea3a20d840020fca2967ca54cccf7f8cc9e Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:54:28 +0200
Subject: [PATCH 107/878] xmlsecurity-verify-usages.diff

---
 .../xmlsec/nss/securityenvironment_nssimpl.cxx     |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/xmlsecurity/source/xmlsec/nss/securityenvironment_nssimpl.cxx b/xmlsecurity/source/xmlsec/nss/securityenvironment_nssimpl.cxx
index 5e6efdd..78a254e 100644
--- a/xmlsecurity/source/xmlsec/nss/securityenvironment_nssimpl.cxx
+++ b/xmlsecurity/source/xmlsec/nss/securityenvironment_nssimpl.cxx
@@ -873,6 +873,8 @@ verifyCertificate( const Reference< csss::XCertificate >& aCert,
                 || usage & certificateUsageSSLCA
                 || usage & certificateUsageSSLServer
                 || usage & certificateUsageSSLClient 
+                || usage & certificateUsageProtectedObjectSigner
+                || usage & certificateUsageObjectSigner
                 // || usage & certificateUsageUserCertImport
                 // || usage & certificateUsageVerifyCA
                 || usage & certificateUsageStatusResponder )
-- 
1.7.0.1

