From 4c110dfb8e9d5807546a0ac2d644f8149051a72a Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:01:56 +0200
Subject: [PATCH 487/878] xulrunner-1.9.1.diff

---
 np_sdk/mozsrc/npapi.h |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/np_sdk/mozsrc/npapi.h b/np_sdk/mozsrc/npapi.h
index af64042..4a14df5 100644
--- a/np_sdk/mozsrc/npapi.h
+++ b/np_sdk/mozsrc/npapi.h
@@ -42,8 +42,9 @@
  *  Netscape client plug-in API spec
  */
 
-#ifndef _NPAPI_H_
+#if !defined(_NPAPI_H_) && !defined(npapi_h_)
 #define _NPAPI_H_
+#define npapi_h_
 
 #ifdef __OS2__
 #pragma pack(1)
-- 
1.7.0.1

