---
 np_sdk/mozsrc/npapi.h |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git np_sdk/mozsrc/npapi.h np_sdk/mozsrc/npapi.h
index af64042..4a14df5 100644
--- np_sdk/mozsrc/npapi.h
+++ np_sdk/mozsrc/npapi.h
@@ -42,8 +42,9 @@
  *  Netscape client plug-in API spec
  */
 
-#ifndef _NPAPI_H_
+#if !defined(_NPAPI_H_) && !defined(npapi_h_)
 #define _NPAPI_H_
+#define npapi_h_
 
 #ifdef __OS2__
 #pragma pack(1)
-- 
1.7.0.1

