--- sal/osl/unx/process.c.ark	2004-11-30 07:20:21.000000000 +0000
+++ sal/osl/unx/process.c	2004-11-30 07:22:51.000000000 +0000
@@ -1017,9 +1017,11 @@
 			for (i = 0; data.m_pszEnv[i] != NULL; i++)
 	     		putenv(data.m_pszEnv[i]);
 
-#if defined(LINUX)
+#if 0 && defined(LINUX)
             /* mfe: linux likes to have just one thread when the exec family is called */
             /*      this np function has this purpose ...                              */
+            /* bero: This is no longer true with NPTL. Unfortunately there's no define */
+            /*       with which we can detect NPTL.                                    */
             pthread_kill_other_threads_np();
 #endif
             OSL_TRACE("ChildStatusProc : starting '%s'",data.m_pszArgs[0]);
