--- officecfg/registry/data/org/openoffice/Office/TypeDetection.xcu
+++ officecfg/registry/data/org/openoffice/Office/TypeDetection.xcu
@@ -1685,6 +1685,14 @@
         <value>0,,,,doc,20002,</value>
       </prop>
     </node>
+    <node oor:name="writer_WordPerfect_Document" oor:op="replace">
+      <prop oor:name="Data" oor:type="xs:string">
+        <value>1,application/wpwin,WordPerfect%20Document,,wp;wpd,0</value>
+      </prop>
+      <prop oor:name="UIName" oor:type="xs:string">
+        <value xml:lang="en-US">WordPerfect Document</value>
+      </prop>
+    </node>
     <node oor:name="writer_WordPerfect_41_W4W" oor:op="replace">
       <prop oor:name="UIName">
         <value xml:lang="en-US">WordPerfect 4.1</value>
@@ -3456,6 +3464,17 @@
         <value>0,writer_WiziWord_30_W4W,com.sun.star.text.TextDocument,,65,W4W47_0,0,,</value>
       </prop>
     </node>
+    <node oor:name="WordPerfect Document" oor:op="replace">
+      <prop oor:name="Data" oor:type="xs:string">
+        <value>0,writer_WordPerfect_Document,com.sun.star.text.TextDocument,com.sun.star.comp.Writer.WordPerfectImportFilter,268959809,WPD,0,,</value>
+      </prop>
+      <prop oor:name="Installed" oor:type="xs:boolean">
+        <value>true</value>
+      </prop>
+      <prop oor:name="UIName" oor:type="xs:string">
+        <value xml:lang="en-US">WordPerfect Document</value>
+      </prop>
+    </node>
     <node oor:name="WordPerfect 4.1 (W4W)" oor:op="replace">
       <prop oor:name="Installed">
         <value>false</value>
@@ -8344,6 +8363,11 @@
 		<value>writer_DocBook_File writer_Flat_XML_File writer_XHTML_File writer_MS_Word_2003 calc_MS_Excel_2003</value>
 	</prop>
     </node>
+    <node oor:name="com.sun.star.comp.Writer.WordPerfectImportFilter" oor:op="replace">
+        <prop oor:name="Types">
+                <value>writer_WordPerfect_Document</value>
+        </prop>
+    </node>
   </node>
   <node oor:name="FrameLoaders">
     <node oor:name="com.sun.star.comp.sfx2.SfxMacroLoader" oor:op="replace">
