---
 avmedia/source/viewer/mediawindow.cxx |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git avmedia/source/viewer/mediawindow.cxx avmedia/source/viewer/mediawindow.cxx
index edc9ee5..e9ade6d 100644
--- avmedia/source/viewer/mediawindow.cxx
+++ avmedia/source/viewer/mediawindow.cxx
@@ -368,6 +368,7 @@ void MediaWindow::getMediaFilters( FilterNameVector& rFilterNameVector )
                                         "AU Audio", "au",
                                         "AVI", "avi",
                                         "CD Audio", "cda",
+                                        "FLAC Audio", "flac",
                                         "MIDI Audio", "mid;midi",
                                         "MPEG Audio", "mp2;mp3;mpa",
                                         "MPEG Video", "mpg;mpeg;mpv;mp4",
-- 
1.7.0.1

