From b2bd915d6dc6b9eac8fa9399d3c1fd190afe06eb Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:01:30 +0200
Subject: [PATCH 464/878] gstreamer-solenv.diff

---
 solenv/inc/settings.mk |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/solenv/inc/settings.mk b/solenv/inc/settings.mk
index d21f3a8..ec40c32 100644
--- a/solenv/inc/settings.mk
+++ b/solenv/inc/settings.mk
@@ -1166,6 +1166,10 @@ CDEFS+=-DENABLE_LAYOUT_EXPERIMENTAL=1
 CFLAGS+=-DENABLE_LAYOUT_EXPERIMENTAL=0
 .ENDIF # ENABLE_LAYOUT_EXPERIMENTAL != TRUE
 
+.IF "$(ENABLE_GSTREAMER)" == "TRUE"
+CDEFS+=-DGSTREAMER
+.ENDIF
+
 # compose flags and defines for GUI
 .IF "$(TARGETTYPE)"=="GUI"
 CDEFS+= $(CDEFSGUI)
-- 
1.7.0.1

