--- configure.in.old	2010-03-03 16:59:45.000000000 +0100
+++ configure.in	2010-03-03 16:59:46.000000000 +0100
@@ -304,6 +304,9 @@ AC_ARG_ENABLE(randr-link,
 [  --disable-randr-link    disable linking with libXrandr, instead dynamically 
                            open it at runtime
 ],,enable_randr_link=yes)
+AC_ARG_ENABLE(gstreamer,
+[  --disable-gstreamer     disable building the gstreamer avmedia backend
+],,enable_gstreamer=yes)
 AC_ARG_WITH(myspell-dicts,
 [  --without-myspell-dicts Removes myspell dictionaries from openoffice.org
                           installation set, for people building for specific
@@ -930,6 +933,7 @@ case "$build_os" in
 	solaris*)
 		test_gtk=yes
 		test_cairo=yes
+		build_gstreamer=yes
 		test_kde=yes
 		test_cups=yes
         test_randr=yes
@@ -963,6 +967,7 @@ case "$build_os" in
 	linux-gnu*)
 		test_gtk=yes
 		test_cairo=yes
+		build_gstreamer=yes
 		test_kde=yes
 		test_kde4=yes
 		test_cups=yes
@@ -1007,6 +1012,7 @@ case "$build_os" in
 	freebsd*)
 		test_gtk=yes
 		test_cairo=yes
+		build_gstreamer=yes
 		test_kde=yes
 		test_kde4=yes
 		test_cups=yes
@@ -1041,6 +1047,7 @@ case "$build_os" in
 	netbsd)
 		test_gtk=yes
 		test_cairo=yes
+		build_gstreamer=yes
 		test_kde=yes
 		test_kde4=yes
 		test_cups=no
@@ -5857,6 +5864,26 @@ AC_SUBST(CAIRO_CFLAGS)
 AC_SUBST(CAIRO_LIBS)
 
 dnl ===================================================================
+dnl Check whether the GStreamer libraries are available.
+dnl ===================================================================
+
+ENABLE_GSTREAMER=""
+
+if test  "$build_gstreamer" = "yes"; then
+
+    AC_MSG_CHECKING([whether to enable the GStreamer avmedia backend])
+    if test "x$enable_gstreamer" != "xno" ; then
+        ENABLE_GSTREAMER="TRUE"
+	AC_MSG_RESULT([yes])
+	PKG_CHECK_MODULES( GSTREAMER, gstreamer-0.10 gstreamer-plugins-base-0.10 )
+    else
+	     AC_MSG_RESULT([no])
+    fi
+fi
+
+AC_SUBST(ENABLE_GSTREAMER)
+
+dnl ===================================================================
 dnl Check whether the OpenGL libraries are available
 dnl ===================================================================
 
--- set_soenv.in.old	2010-03-03 16:59:44.000000000 +0100
+++ set_soenv.in	2010-03-03 16:59:46.000000000 +0100
@@ -1681,6 +1681,7 @@ ToFile( "ENABLE_DBUS",       "@ENABLE_DB
 ToFile( "ENABLE_GCONF",      "@ENABLE_GCONF@",     "e" );
 ToFile( "ENABLE_GNOMEVFS",   "@ENABLE_GNOMEVFS@",  "e" );
 ToFile( "GNOMEVFS_HAS_LOCKING","@GNOMEVFS_HAS_LOCKING@",  "e" );
+ToFile( "ENABLE_GSTREAMER",  "@ENABLE_GSTREAMER@", "e" );
 ToFile( "ENABLE_GIO",        "@ENABLE_GIO@",       "e" );
 ToFile( "ENABLE_KDE",        "@ENABLE_KDE@",       "e" );
 ToFile( "KDE_CFLAGS",        "@KDE_CFLAGS@",       "e" );
