---
 solenv/inc/settings.mk |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git solenv/inc/settings.mk solenv/inc/settings.mk
index 03cc24f..76f1203 100644
--- solenv/inc/settings.mk
+++ solenv/inc/settings.mk
@@ -1170,6 +1170,10 @@ CDEFS+=-DENABLE_LAYOUT_EXPERIMENTAL=1
 CFLAGS+=-DENABLE_LAYOUT_EXPERIMENTAL=0
 .ENDIF # ENABLE_LAYOUT_EXPERIMENTAL != TRUE
 
+.IF "$(ENABLE_GSTREAMER)" == "TRUE"
+CDEFS+=-DGSTREAMER
+.ENDIF
+
 # compose flags and defines for GUI
 .IF "$(TARGETTYPE)"=="GUI"
 CDEFS+= $(CDEFSGUI)
-- 
1.7.0.1

