From 76a40a72c01440345c7903fc356a86a6122dc72d Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:01:30 +0200
Subject: [PATCH 395/768] gstreamer-solenv.diff

---
 solenv/inc/settings.mk |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git solenv/inc/settings.mk solenv/inc/settings.mk
index b50ff86..ae1fc8e 100644
--- solenv/inc/settings.mk
+++ solenv/inc/settings.mk
@@ -1170,6 +1170,10 @@ CDEFS+=-DENABLE_LAYOUT_EXPERIMENTAL=1
 CFLAGS+=-DENABLE_LAYOUT_EXPERIMENTAL=0
 .ENDIF # ENABLE_LAYOUT_EXPERIMENTAL != TRUE
 
+.IF "$(ENABLE_GSTREAMER)" == "TRUE"
+CDEFS+=-DGSTREAMER
+.ENDIF
+
 # compose flags and defines for GUI
 .IF "$(TARGETTYPE)"=="GUI"
 CDEFS+= $(CDEFSGUI)
-- 
1.7.0.1

