--- testtools/source/bridgetest/idl/makefile.mk.old	2009-10-08 16:17:37.000000000 +0200
+++ testtools/source/bridgetest/idl/makefile.mk	2010-03-03 16:59:32.000000000 +0100
@@ -59,6 +59,8 @@ CLIMAKERFLAGS =
 CLIMAKERFLAGS += --verbose
 .ENDIF
 
+
+.IF "$(BUILD_FOR_CLI)" == "true"
 ALLTAR: $(MISC)$/$(TARGET).cppumaker.done \
     $(MISC)$/$(TARGET).javamaker.done \
     $(BIN)$/cli_types_bridgetest.dll 
@@ -67,6 +69,10 @@ $(BIN)$/cli_types_bridgetest.dll: $(BIN)
     $(CLIMAKER) $(CLIMAKERFLAGS) --out $@ -r $(SOLARBINDIR)$/cli_uretypes.dll \
         -X $(SOLARBINDIR)$/types.rdb $< 
     $(TOUCH) $@
+.ELSE
+ALLTAR: $(MISC)$/$(TARGET).cppumaker.done \
+    $(MISC)$/$(TARGET).javamaker.done
+.ENDIF
 
 .ENDIF
 .ENDIF # GUI, WNT
--- testtools/util/makefile.pmk.old	2009-10-07 15:06:44.000000000 +0200
+++ testtools/util/makefile.pmk	2010-03-03 16:59:32.000000000 +0100
@@ -31,7 +31,7 @@
 
 # no translation here anyway
 .IF "$(L10N_framework)"==""
-.IF "$(COM)" == "MSC"
+.IF "$(COM)" == "MSC" || "$(ENABLE_MONO)" == "YES"
 BUILD_FOR_CLI = "true"
 CFLAGSENABLESYMBOLS:=-Z7
 .ENDIF
