---
 cli_ure/prj/build.lst |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git cli_ure/prj/build.lst cli_ure/prj/build.lst
index 00f7f53..23761c0 100644
--- cli_ure/prj/build.lst
+++ cli_ure/prj/build.lst
@@ -4,10 +4,10 @@ ure	cli_ure\inc					nmake   -   all   ure_inc NULL
 ure	cli_ure\version				nmake   -   all   ure_source_version ure_inc NULL
 ure	cli_ure\source					nmake   -		all		ure_source_source ure_inc NULL
 ure	cli_ure\source\basetypes        nmake   -       all     ure_source_basetypes ure_source_version ure_source_source ure_inc NULL
-ure	cli_ure\source\bootstrap        nmake   -       u       ure_source_bootstrap ure_source_source ure_source_basetypes ure_inc NULL
+ure	cli_ure\source\bootstrap        nmake   -       u       ure_source_bootstrap ure_source_climaker ure_source_source ure_source_basetypes ure_inc ure_unotypes NULL
 ure	cli_ure\source\climaker         nmake   -       all     ure_source_climaker ure_source_basetypes ure_inc NULL
 ure	cli_ure\unotypes				nmake   -       all     ure_unotypes ure_source_version ure_source_source ure_source_climaker ure_source_basetypes ure_inc NULL
-ure	cli_ure\source\ure				nmake   -       all     ure_source_ure ure_source_version ure_source_source ure_source_basetypes ure_unotypes ure_inc NULL
+ure	cli_ure\source\ure                      nmake   -       all     ure_source_ure ure_source_bootstrap.u ure_source_version ure_source_source ure_source_basetypes ure_unotypes ure_inc NULL
 ure	cli_ure\source\uno_bridge		nmake	-		w,vc7	ure_source_uno_bridge ure_source_basetypes ure_unotypes ure_source_ure ure_inc NULL
 ure	cli_ure\source\mono_bridge      nmake   -       u       ure_source_mono_bridge ure_unotypes ure_source_ure ure_inc NULL
 ure	cli_ure\source\native           nmake   -       w,vc7   ure_source_native ure_source_version ure_source_source ure_source_ure ure_unotypes ure_source_uno_bridge ure_inc NULL
-- 
1.7.0.1

