---
 configure.in |   16 ++++++++++++++++
 set_soenv.in |    1 +
 2 files changed, 17 insertions(+), 0 deletions(-)

diff --git configure.in configure.in
index 9c8aa00..cc0f65b 100644
--- configure.in
+++ configure.in
@@ -1972,6 +1972,21 @@ if test "$_os" = "Linux"; then
                         "`echo $MCS_VERSION | cut -d"." -f3`" -ge "8"; then
                 ENABLE_MONO=YES
                 CSC=$MCS
+                AC_PATH_PROG(GMCS, gmcs, no)
+                AC_PATH_PROG(MKBUNDLE2, mkbundle2, no)
+                GMCS_VERSION=`$GMCS --version | cut -d" " -f5`
+                if test "`echo $GMCS_VERSION | cut -d"." -f1`" -gt "1" || \
+                   test "`echo $GMCS_VERSION | cut -d"." -f1`" = "1" -a \
+                        "`echo $GMCS_VERSION | cut -d"." -f2`" -ge "3" || \
+                   test "`echo $GMCS_VERSION | cut -d"." -f1`" = "1" -a \
+                    "`echo $GMCS_VERSION | cut -d"." -f2`" = "2" -a \
+                    "`echo $GMCS_VERSION | cut -d"." -f3`" -ge "3"; then
+                    # mkbundle2 does not work on ppc, http://bugzilla.ximian.com/show_bug.cgi?id=81525
+                    if test "`uname -m`" != "ppc" -a "`uname -m`" != "ppc64" ; then
+                    ENABLE_MONO_CLIMAKER=YES
+                    AC_MSG_NOTICE([mono is up-to-date enough - building mono climaker])
+                    fi
+                fi
             else
                 if test -n "$enable_mono" ; then
                     AC_MSG_ERROR([no, mcs >= 1.1.8 is needed.])
@@ -1993,6 +2008,7 @@ AC_SUBST(AL)
 AC_SUBST(ENABLE_MONO)
 AC_SUBST(MONO_CFLAGS)
 AC_SUBST(MONO_LIBS)
+AC_SUBST(ENABLE_MONO_CLIMAKER)
 
 dnl ===================================================================
 dnl Check if stdc headers are available excluding windows.
diff --git set_soenv.in set_soenv.in
index 6bf714f..d36115a 100644
--- set_soenv.in
+++ set_soenv.in
@@ -1671,6 +1671,7 @@ ToFile( "DO_FETCH_TARBALLS", "@DO_FETCH_TARBALLS@",  "e" );
 ToFile( "Platform dependent constant values.", $empty, "c" );
 ToFile( "SOLAR_JAVA",        $SOLAR_JAVA,        "e" );
 ToFile( "ENABLE_MONO",       "@ENABLE_MONO@",    "e" );
+ToFile( "ENABLE_MONO_CLIMAKER",       "@ENABLE_MONO_CLIMAKER@",    "e" );
 ToFile( "CSC",               "@CSC@",            "e" );
 ToFile( "AL",                "@AL@",             "e" );
 ToFile( "CSC_PATH",          "@CSC_PATH@",       "e" );
-- 
1.7.0.1

