From e3fad9bf7cdb27be814a6134b5ed1ad179d50b11 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:00:58 +0200
Subject: [PATCH 435/878] sdbc-postgresql-config_office.diff

---
 configure.in |   17 +++++++++++++++++
 set_soenv.in |    1 +
 2 files changed, 18 insertions(+), 0 deletions(-)

diff --git a/configure.in b/configure.in
index bcfd8f4..1fdf875 100644
--- a/configure.in
+++ b/configure.in
@@ -4329,6 +4329,23 @@ AC_SUBST(SYSTEM_HSQLDB)
 AC_SUBST(HSQLDB_JAR)
 
 dnl ===================================================================
+dnl Check for PostgreSQL stuff
+dnl ===================================================================
+AC_PATH_PROG(PGCONFIG, pg_config)
+if test -z "$PGCONFIG"; then
+    AC_MSG_ERROR([pg_config needed])
+fi
+save_CFLAGS=$CFLAGS
+save_LIBS=$LIBS
+CFLAGS="-I`pg_config --includedir`"
+AC_CHECK_HEADER([libpq-fe.h], [], [AC_MSG_ERROR([libpq-fe.h is needed])], [])
+AC_HAVE_LIBRARY(pq, [], [AC_MSG_ERROR(libpq is needed)], [])
+CFLAGS=$save_CFLAGS
+LIBS=$save_LIBS
+SYSTEM_POSTGRESQL=YES # only option atm :-)
+AC_SUBST(SYSTEM_POSTGRESQL)
+
+dnl ===================================================================
 dnl Check for system beanshell
 dnl ===================================================================
 AC_MSG_CHECKING([which beanshell to use])
diff --git a/set_soenv.in b/set_soenv.in
index 45739a5..ca2f1e4 100644
--- a/set_soenv.in
+++ b/set_soenv.in
@@ -1959,6 +1959,7 @@ ToFile( "USE_XINERAMA",      "@USE_XINERAMA@",     "e" );
 ToFile( "XINERAMA_LINK",     "@XINERAMA_LINK@",    "e" );
 ToFile( "SYSTEM_REDLAND",    "@SYSTEM_REDLAND@",   "e" );
 ToFile( "REDLAND_LIBS",      "@REDLAND_LIBS@",     "e" );
+ToFile( "SYSTEM_POSTGRESQL",  "@SYSTEM_POSTGRESQL@", "e" );
 ToFile( "SYSTEM_HUNSPELL",   "@SYSTEM_HUNSPELL@",  "e" );
 ToFile( "HUNSPELL_CFLAGS",   "@HUNSPELL_CFLAGS@",  "e" );
 ToFile( "HUNSPELL_LIBS",     "@HUNSPELL_LIBS@",    "e" );
-- 
1.7.0.1

