Index: registry/data/org/openoffice/Office/Common.xcu
===================================================================
RCS file: /cvs/util/officecfg/registry/data/org/openoffice/Office/Common.xcu,v
retrieving revision 1.13.34.2
retrieving revision 1.13.34.2.22.1
diff -u -r1.13.34.2 -r1.13.34.2.22.1
--- officecfg/registry/data/org/openoffice/Office/Common.xcu	6 Jul 2004 10:50:06 -0000	1.13.34.2
+++ officecfg/registry/data/org/openoffice/Office/Common.xcu	14 Jun 2006 09:22:13 -0000	1.13.34.2.22.1
@@ -103,6 +103,13 @@
         <value xml:lang="zh-TW">13</value>
         <value xml:lang="tr">13</value>
         <value xml:lang="ar">13</value>
+      </prop>
+    </node>
+  </node>
+  <node oor:name="Java">
+    <node oor:name="Applet">
+      <prop oor:name="Enable" oor:type="xs:boolean">
+        <value>false</value>
       </prop>
     </node>
   </node>
Index: registry/schema/org/openoffice/Office/Common.xcs
===================================================================
RCS file: /cvs/util/officecfg/registry/schema/org/openoffice/Office/Common.xcs,v
retrieving revision 1.84.8.6
retrieving revision 1.84.8.6.16.1
diff -u -r1.84.8.6 -r1.84.8.6.16.1
--- officecfg/registry/schema/org/openoffice/Office/Common.xcs	27 Aug 2004 12:19:35 -0000	1.84.8.6
+++ officecfg/registry/schema/org/openoffice/Office/Common.xcs	13 Jun 2006 15:52:29 -0000	1.84.8.6.16.1
@@ -23447,7 +23447,7 @@
 						<label xml:lang="kn-IN">Enable Applets</label>
 						<label xml:lang="gl">Enable Applets</label>
 					</info>
-					<value>true</value>
+					<value>false</value>
 				</prop>
 			</group>
 		</group>
