Index: Common.xcu
===================================================================
RCS file: /cvs/util/officecfg/registry/data/org/openoffice/Office/Common.xcu,v
retrieving revision 1.13.34.2
diff -u -u -r1.13.34.2 Common.xcu
--- officecfg/registry/data/org/openoffice/Office/Common.xcu	6 Jul 2004 10:50:06 -0000	1.13.34.2
+++ officecfg/registry/data/org/openoffice/Office/Common.xcu	19 Mar 2005 15:27:54 -0000
@@ -437,4 +437,17 @@
       </node>
     </node>
   </node>
+ <node oor:name="Save">
+  <node oor:name="Document">
+   <prop oor:name="AutoSave" oor:type="xs:boolean">
+    <value>true</value>
+   </prop>
+   <prop oor:name="AutoSavePrompt" oor:type="xs:boolean">
+    <value>true</value>
+   </prop>
+   <prop oor:name="AutoSaveTimeIntervall" oor:type="xs:int">
+    <value>15</value>
+   </prop>
+  </node>
+ </node>
 </oor:component-data>
