Index: methods1.cxx
===================================================================
RCS file: /cvs/script/basic/source/runtime/methods1.cxx,v
retrieving revision 1.14.32.3
retrieving revision 1.14.32.3.2.1
diff -u -p -u -r1.14.32.3 -r1.14.32.3.2.1
--- methods1.cxx	12 Oct 2004 11:01:42 -0000	1.14.32.3
+++ methods1.cxx	21 Apr 2005 11:38:37 -0000	1.14.32.3.2.1
@@ -625,6 +625,8 @@ RTLFUNC(Array)
 	// Option Base zunaechst ignorieren (kennt leider nur der Compiler)
 	if( nArraySize )
 		pArray->AddDim( 0, nArraySize-1 );
+	else
+		pArray->unoAddDim( 0, -1 );
 
 	// Parameter ins Array uebernehmen
 	for( short i = 0 ; i < nArraySize ; i++ )
@@ -664,6 +666,9 @@ RTLFUNC(DimArray)
 			pArray->AddDim32( 0, ub );
 		}
 	}
+	else
+		pArray->unoAddDim( 0, -1 );
+
 	// Array zurueckliefern
 	SbxVariableRef refVar = rPar.Get(0);
 	USHORT nFlags = refVar->GetFlags();
