Index: svx/source/svdraw/svdpagv.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/svdraw/svdpagv.cxx,v
retrieving revision 1.34
diff -u -p -u -r1.34 svdpagv.cxx
--- svx/source/svdraw/svdpagv.cxx	21 Jul 2003 10:57:08 -0000	1.34
+++ svx/source/svdraw/svdpagv.cxx	15 Jun 2005 13:48:40 -0000
@@ -1128,8 +1128,10 @@ FASTBOOL SdrPageView::DoCachedMasterPain
 	// Seitennummer der MasterPage muss gleich sein
 	if (!bCreate)
 		bCreate=pBmp->nMasterPageNum!=pPg->GetPageNum();
+	Fraction aUnity(1, 1);
 	if (!bCreate) // Scaling checken
-		bCreate=rMap.GetScaleX()!=pBmp->aMapX || rMap.GetScaleY()!=pBmp->aMapY;
+		bCreate=rMap.GetScaleX()!=pBmp->aMapX || rMap.GetScaleY()!=pBmp->aMapY ||
+				(!pWin->IsMapModeEnabled() && (pBmp->aMapX != aUnity || pBmp->aMapY != aUnity));
 
 	// Sichtbare Layer muessen gleich sein
 	if (!bCreate)
@@ -1199,6 +1201,8 @@ FASTBOOL SdrPageView::DoCachedMasterPain
 			aMapOrgGridTmp-=aOfs; // Position der PageView abziehen fuer Border, Grid, ...
 			aNewMap.SetOrigin(aMapOrgGridTmp);
 			pBmp->aVD.SetMapMode(aNewMap);
+			pBmp->aVD.EnableMapMode(pWin->IsMapModeEnabled());
+
 			rXOut.SetOutDev(&pBmp->aVD);
 
             // Paper, Border etc. painten
