--- officecfg/registry/schema/org/openoffice/Office/Common.xcs.old	2005-03-22 20:05:58.000000000 +0100
+++ officecfg/registry/schema/org/openoffice/Office/Common.xcs	2005-03-23 12:20:22.000000000 +0100
@@ -9098,7 +9098,7 @@
 						<label xml:lang="kn-IN">Total Graphic Cache Size</label>
 						<label xml:lang="gl">Total Graphic Cache Size</label>
 					</info>
-					<value>10240000</value>
+					<value>20480000</value>
 				</prop>
 				<prop oor:name="ObjectCacheSize" oor:type="xs:int">
 					<info>
@@ -9184,7 +9184,7 @@
 						<label xml:lang="kn-IN">Graphic Object Cache Size</label>
 						<label xml:lang="gl">Graphic Object Cache Size</label>
 					</info>
-					<value>2560000</value>
+					<value>5120000</value>
 				</prop>
 				<prop oor:name="ObjectReleaseTime" oor:type="xs:int">
 					<info>
