Index: desktop/source/app/dispatchwatcher.cxx
===================================================================
RCS file: /cvs/framework/desktop/source/app/dispatchwatcher.cxx,v
retrieving revision 1.15.82.1
diff -u -r1.15.82.1 dispatchwatcher.cxx
--- desktop/source/app/dispatchwatcher.cxx	9 Jan 2004 16:43:26 -0000	1.15.82.1
+++ desktop/source/app/dispatchwatcher.cxx	2 Oct 2004 16:36:24 -0000
@@ -353,14 +353,14 @@
                 OUString aMsg = OUString::createFromAscii(
                     "Dispatchwatcher IllegalArgumentException while calling loadComponentFromURL: ") 
                     + iae.Message;
-                OSL_ENSURE( sal_False, aMsg);
+                OSL_ENSURE( sal_False, OUStringToOString(aMsg, RTL_TEXTENCODING_ASCII_US).getStr());
             }
             catch (com::sun::star::io::IOException& ioe)
             {
                 OUString aMsg = OUString::createFromAscii(
                     "Dispatchwatcher IOException while calling loadComponentFromURL: ") 
                     + ioe.Message;
-                OSL_ENSURE( sal_False, aMsg);
+                OSL_ENSURE( sal_False, OUStringToOString(aMsg, RTL_TEXTENCODING_ASCII_US).getStr());
             }
 			if ( aDispatchRequest.aRequestType == REQUEST_OPEN ||
 				 aDispatchRequest.aRequestType == REQUEST_VIEW ||
