Index: configure.in
===================================================================
RCS file: /cvs/tools/config_office/configure.in,v
retrieving revision 1.55.6.10
diff -u -u -r1.55.6.10 configure.in
--- config_office/configure.in	27 Aug 2004 10:03:41 -0000	1.55.6.10
+++ config_office/configure.in	17 Jan 2005 23:45:18 -0000
@@ -1401,7 +1457,7 @@
     # different
     # This test _should_ work because the one accepts -V (ours) and one
     # (the other) not...
-    dmake -V 2>/dev/null | grep 'dmake .* Version 4.*, PL ' >/dev/null
+    dmake -V 2>/dev/null | grep 'dmake .* Version .*' >/dev/null
     if test $? -eq 0; then 
        BUILD_DMAKE=NO
        AC_MSG_RESULT([yes])
