--- unotools/source/ucbhelper/tempfile.cxx	2004-10-12 10:47:47.000000000 +0530
+++ unotools/source/ucbhelper/tempfile.cxx	2004-10-19 16:20:49.663737320 +0530
@@ -324,7 +324,8 @@
     for ( sal_Int32 i=0;; i++ )
     {
         String aTmp( aName );
-       	aTmp += String::CreateFromInt32( i );
+	if (i)
+        	aTmp += String::CreateFromInt32( i );
         if ( pExtension )
             aTmp += *pExtension;
         else
--- sfx2/source/dialog/mailmodel.cxx	2004-10-19 16:52:56.703782704 +0530
+++ sfx2/source/dialog/mailmodel.cxx	2004-10-11 20:48:22.000000000 +0530
@@ -341,12 +341,12 @@
 				pExt = new String( String::CreateFromAscii( "." ) + aFileObj.getExtension() );
 				aFileObj.removeExtension();
 				aLeadingStr = aFileObj.getName( INetURLObject::LAST_SEGMENT, true, INetURLObject::DECODE_WITH_CHARSET );
-				 aLeadingStr += String::CreateFromAscii( "_" );
+				// aLeadingStr += String::CreateFromAscii( "_" );
 			}
 			else
 			{
 				aLeadingStr = aFileObj.getName( INetURLObject::LAST_SEGMENT, true, INetURLObject::DECODE_WITH_CHARSET );
-				 aLeadingStr += String::CreateFromAscii( "_" );
+				// aLeadingStr += String::CreateFromAscii( "_" );
 			}
 		}
 
@@ -441,7 +441,7 @@
 					pExt = new String( aPDFExtension );
 					aFileObj.removeExtension();
 					aLeadingStr = aFileObj.getName( INetURLObject::LAST_SEGMENT, true, INetURLObject::DECODE_WITH_CHARSET );
-					aLeadingStr += String::CreateFromAscii( "_" );
+				//	aLeadingStr += String::CreateFromAscii( "_" );
 				}
 				else
 				{
