Index: unotools/inc/unotools/tempfile.hxx
===================================================================
RCS file: /cvs/util/unotools/inc/unotools/tempfile.hxx,v
retrieving revision 1.4
diff -u -r1.4 tempfile.hxx
--- unotools/inc/unotools/tempfile.hxx	2 Nov 2000 10:22:45 -0000	1.4
+++ unotools/inc/unotools/tempfile.hxx	12 Oct 2004 05:27:42 -0000
@@ -107,7 +107,7 @@
                     The extension string may be f.e. ".txt" or "", if no extension string is given, ".tmp" is used
                     */
                     TempFile( const String& rLeadingChars, const String* pExtension=NULL, const String* pParent=NULL,
-                                sal_Bool bDirectory=sal_False );
+                                sal_Bool bDirectory=sal_False, sal_Bool bEmail=sal_False );
 
                     /**
                     TempFile will be removed from disk in dtor if EnableKillingTempFile was called before.
Index: unotools/source/ucbhelper/tempfile.cxx
===================================================================
RCS file: /cvs/util/unotools/source/ucbhelper/tempfile.cxx,v
retrieving revision 1.14.72.1
diff -u -r1.14.72.1 tempfile.cxx
--- unotools/source/ucbhelper/tempfile.cxx	1 Sep 2004 09:28:55 -0000	1.14.72.1
+++ unotools/source/ucbhelper/tempfile.cxx	12 Oct 2004 05:28:20 -0000
@@ -134,7 +134,10 @@
         return sal_True;
 
     // try to create the directory
-    nError = osl::Directory::create( aPath );
+    nError = osl::Directory::createWithMode( aPath,
+											 osl_File_Attribute_OwnWrite |
+											 osl_File_Attribute_OwnRead |
+											 osl_File_Attribute_OwnExe );
 	sal_Bool  bSuccess = ( nError == osl::File::E_None || nError == osl::FileBase::E_EXIST );
     if( !bSuccess )
 	{
@@ -148,7 +151,10 @@
 			if ( bSuccess )
             {
                 // Parent directory exists, retry creation of directory
-				nError = osl::Directory::create( aPath );
+				nError = osl::Directory::createWithMode( aPath,
+														 osl_File_Attribute_OwnWrite |
+														 osl_File_Attribute_OwnRead |
+														 osl_File_Attribute_OwnExe );
 				bSuccess =( nError == osl::File::E_None || nError == osl::FileBase::E_EXIST );
 			}
 		}
@@ -231,7 +237,10 @@
 
         if ( bDir )
         {
-            FileBase::RC err = Directory::create( aTmp );
+			FileBase::RC err = Directory::createWithMode( aTmp,
+														  osl_File_Attribute_OwnWrite |
+														  osl_File_Attribute_OwnRead |
+														  osl_File_Attribute_OwnExe );
             if (  err == FileBase::E_None )
             {
                 // !bKeep: only for creating a name, not a file or directory
@@ -301,7 +310,7 @@
     CreateTempName_Impl( pImp->aName, sal_True, bDirectory );
 }
 
-TempFile::TempFile( const String& rLeadingChars, const String* pExtension, const String* pParent, sal_Bool bDirectory )
+TempFile::TempFile( const String& rLeadingChars, const String* pExtension, const String* pParent, sal_Bool bDirectory ,sal_Bool bEmail)
     : pImp( new TempFile_Impl )
     , bKillingFileEnabled( sal_False )
 {
@@ -315,14 +324,18 @@
     for ( sal_Int32 i=0;; i++ )
     {
         String aTmp( aName );
-        aTmp += String::CreateFromInt32( i );
+	if (!bEmail)
+        	aTmp += String::CreateFromInt32( i );
         if ( pExtension )
             aTmp += *pExtension;
         else
             aTmp += String::CreateFromAscii( ".tmp" );
         if ( bDirectory )
         {
-            FileBase::RC err = Directory::create( aTmp );
+			FileBase::RC err = osl::Directory::createWithMode( aTmp,
+															   osl_File_Attribute_OwnWrite |
+															   osl_File_Attribute_OwnRead |
+															   osl_File_Attribute_OwnExe );
             if ( err == FileBase::E_None )
             {
                 pImp->aName = aTmp;
@@ -426,7 +439,10 @@
 
     // try to create the directory
     sal_Bool bRet = sal_False;
-    osl::FileBase::RC err = osl::Directory::create( aUnqPath );
+    osl::FileBase::RC err = osl::Directory::createWithMode( aUnqPath,
+															osl_File_Attribute_OwnWrite |
+															osl_File_Attribute_OwnRead |
+															osl_File_Attribute_OwnExe );
     if ( err != FileBase::E_None && err != FileBase::E_EXIST )
         // perhaps parent(s) don't exist
         bRet = ensuredir( aUnqPath );
Index: sfx2/source/dialog/mailmodel.cxx
===================================================================
RCS file: /cvs/framework/sfx2/source/dialog/mailmodel.cxx,v
retrieving revision 1.21
diff -u -r1.21 mailmodel.cxx
--- sfx2/source/dialog/mailmodel.cxx	17 Apr 2003 16:09:52 -0000	1.21
+++ sfx2/source/dialog/mailmodel.cxx	12 Oct 2004 05:29:07 -0000
@@ -341,12 +341,12 @@
 				pExt = new String( String::CreateFromAscii( "." ) + aFileObj.getExtension() );
 				aFileObj.removeExtension();
 				aLeadingStr = aFileObj.getName( INetURLObject::LAST_SEGMENT, true, INetURLObject::DECODE_WITH_CHARSET );
-				aLeadingStr += String::CreateFromAscii( "_" );
+				// aLeadingStr += String::CreateFromAscii( "_" );
 			}
 			else
 			{
 				aLeadingStr = aFileObj.getName( INetURLObject::LAST_SEGMENT, true, INetURLObject::DECODE_WITH_CHARSET );
-				aLeadingStr += String::CreateFromAscii( "_" );
+				// aLeadingStr += String::CreateFromAscii( "_" );
 			}
 		}
 
@@ -357,8 +357,7 @@
 			pExt->Erase( 0, 1 );
 		}
 
-        ::utl::TempFile aTempFile( aLeadingStr, pExt );
-
+        ::utl::TempFile aTempFile( aLeadingStr, pExt,NULL,sal_False,sal_True );
         delete pExt;
 
         rFileName = aTempFile.GetURL();
@@ -366,7 +365,6 @@
 		SfxStringItem aFileName( SID_FILE_NAME, rFileName );
 		SfxBoolItem aPicklist( SID_PICKLIST, FALSE );
 		SfxBoolItem aSaveTo( SID_SAVETO, TRUE );
-
 		SfxStringItem* pFilterName = NULL;
 		if ( pFilter && bHasFilter )
             pFilterName = new SfxStringItem( SID_FILTER_NAME, pFilter->GetFilterName() );
@@ -443,7 +441,7 @@
 					pExt = new String( aPDFExtension );
 					aFileObj.removeExtension();
 					aLeadingStr = aFileObj.getName( INetURLObject::LAST_SEGMENT, true, INetURLObject::DECODE_WITH_CHARSET );
-					aLeadingStr += String::CreateFromAscii( "_" );
+				//	aLeadingStr += String::CreateFromAscii( "_" );
 				}
 				else
 				{
