--- officecfg/registry/data/org/openoffice/Office/TypeDetection.xcu	27 Apr 2004 16:32:24 -0000	1.44.8.5
+++ officecfg/registry/data/org/openoffice/Office/TypeDetection.xcu	8 Nov 2004 11:56:22 -0000
@@ -232,7 +232,7 @@
         <value xml:lang="en-US">Microsoft Excel 4.0</value>
       </prop>
       <prop oor:name="Data">
-        <value>0,,,,xls;xlw,0,</value>
+        <value>0,application/vnd.ms-excel,,,xls;xlw,0,</value>
       </prop>
     </node>
     <node oor:name="calc_MS_Excel_40_VorlageTemplate" oor:op="replace">
@@ -240,7 +240,7 @@
         <value xml:lang="en-US">MS Excel 4.0 Template</value>
       </prop>
       <prop oor:name="Data">
-        <value>0,,,,xlt,0,</value>
+        <value>0,application/vnd.ms-excel,,,xlt,0,</value>
       </prop>
     </node>
     <node oor:name="calc_MS_Excel_5095" oor:op="replace">
@@ -248,7 +248,7 @@
         <value xml:lang="en-US">Microsoft Excel 5.0</value>
       </prop>
       <prop oor:name="Data">
-        <value>0,,Biff5,,xls;xlw,0,</value>
+        <value>0,application/vnd.ms-excel,Biff5,,xls;xlw,0,</value>
       </prop>
     </node>
     <node oor:name="calc_MS_Excel_5095_VorlageTemplate" oor:op="replace">
@@ -256,7 +256,7 @@
         <value xml:lang="en-US">MS Excel 5.0 Template</value>
       </prop>
       <prop oor:name="Data">
-        <value>0,,Biff5,,xlt,0,</value>
+        <value>0,application/vnd.ms-excel,Biff5,,xlt,0,</value>
       </prop>
     </node>
     <node oor:name="calc_MS_Excel_95" oor:op="replace">
@@ -264,7 +264,7 @@
         <value xml:lang="en-US">Microsoft Excel 95</value>
       </prop>
       <prop oor:name="Data">
-        <value>0,,Biff5,,xls;xlw,0,</value>
+        <value>0,application/vnd.ms-excel,Biff5,,xls;xlw,0,</value>
       </prop>
     </node>
     <node oor:name="calc_MS_Excel_95_VorlageTemplate" oor:op="replace">
@@ -272,7 +272,7 @@
         <value xml:lang="en-US">MS Excel 95 Template</value>
       </prop>
       <prop oor:name="Data">
-        <value>0,,Biff5,,xlt,0,</value>
+        <value>0,application/vnd.ms-excel,Biff5,,xlt,0,</value>
       </prop>
     </node>
     <node oor:name="calc_MS_Excel_97" oor:op="replace">
@@ -280,7 +280,7 @@
         <value xml:lang="en-US">Microsoft Excel 97/2000/XP</value>
       </prop>
       <prop oor:name="Data">
-        <value>1,,Biff8,,xls;xlw,0,</value>
+        <value>1,application/vnd.ms-excel,Biff8,,xls;xlw,0,</value>
       </prop>
     </node>
     <node oor:name="calc_MS_Excel_97_VorlageTemplate" oor:op="replace">
@@ -288,7 +288,7 @@
         <value xml:lang="en-US">MS Excel 97/2000 Template</value>
       </prop>
       <prop oor:name="Data">
-        <value>1,,Biff8,,xlt,0,</value>
+        <value>1,application/vnd.ms-excel,Biff8,,xlt,0,</value>
       </prop>
     </node>
     <node oor:name="calc_SYLK" oor:op="replace">
@@ -1112,7 +1112,7 @@
         <value xml:lang="en-US">Microsoft MacWord 3.0</value>
       </prop>
       <prop oor:name="Data">
-        <value>0,,,,doc,20002,</value>
+        <value>0,application/vnd.ms-word,,,doc,20002,</value>
       </prop>
     </node>
     <node oor:name="writer_MS_MacWord_40_W4W" oor:op="replace">
@@ -1120,7 +1120,7 @@
         <value xml:lang="en-US">Microsoft MacWord 4.0</value>
       </prop>
       <prop oor:name="Data">
-        <value>0,,,,doc,20002,</value>
+        <value>0,application/vnd.ms-word,,,doc,20002,</value>
       </prop>
     </node>
     <node oor:name="writer_MS_MacWord_5x_W4W" oor:op="replace">
@@ -1128,7 +1128,7 @@
         <value xml:lang="en-US">Microsoft MacWord 5.x</value>
       </prop>
       <prop oor:name="Data">
-        <value>0,,,,doc,20002,</value>
+        <value>0,application/vnd.ms-word,,,doc,20002,</value>
       </prop>
     </node>
     <node oor:name="writer_MS_WinWord_1x_W4W" oor:op="replace">
@@ -1136,7 +1136,7 @@
         <value xml:lang="en-US">Microsoft WinWord 1.x</value>
       </prop>
       <prop oor:name="Data">
-        <value>0,,,,doc,20002,</value>
+        <value>0,application/vnd.ms-word,,,doc,20002,</value>
       </prop>
     </node>
     <node oor:name="writer_MS_WinWord_2x_W4W" oor:op="replace">
@@ -1144,7 +1144,7 @@
         <value xml:lang="en-US">Microsoft WinWord 2.x</value>
       </prop>
       <prop oor:name="Data">
-        <value>0,,,,doc,20002,</value>
+        <value>0,application/vnd.ms-word,,,doc,20002,</value>
       </prop>
     </node>
     <node oor:name="writer_MS_WinWord_5" oor:op="replace">
@@ -1152,7 +1152,7 @@
         <value xml:lang="en-US">Microsoft WinWord 5</value>
       </prop>
       <prop oor:name="Data">
-        <value>0,application/msword,MSWordDoc,,doc,20002,</value>
+        <value>0,application/vnd.ms-word,MSWordDoc,,doc,20002,</value>
       </prop>
     </node>
     <node oor:name="writer_MS_WinWord_60" oor:op="replace">
@@ -1160,7 +1160,7 @@
         <value xml:lang="en-US">Microsoft Word 6.0</value>
       </prop>
       <prop oor:name="Data">
-        <value>0,application/msword,MSWordDoc,,doc,20002,</value>
+        <value>0,application/vnd.ms-word,MSWordDoc,,doc,20002,</value>
       </prop>
     </node>
     <node oor:name="writer_MS_Word_3x_W4W" oor:op="replace">
@@ -1168,7 +1168,7 @@
         <value xml:lang="en-US">Microsoft Word 3.x</value>
       </prop>
       <prop oor:name="Data">
-        <value>0,,,,txt,20002,</value>
+        <value>0,application/vnd.ms-word,,,txt,20002,</value>
       </prop>
     </node>
     <node oor:name="writer_MS_Word_4x_W4W" oor:op="replace">
@@ -1176,7 +1176,7 @@
         <value xml:lang="en-US">Microsoft Word 4.x</value>
       </prop>
       <prop oor:name="Data">
-        <value>0,,,,txt,20002,</value>
+        <value>0,application/vnd.ms-word,,,txt,20002,</value>
       </prop>
     </node>
     <node oor:name="writer_MS_Word_5x_W4W" oor:op="replace">
@@ -1184,7 +1184,7 @@
         <value xml:lang="en-US">Microsoft Word 5.x</value>
       </prop>
       <prop oor:name="Data">
-        <value>0,,,,txt,20002,</value>
+        <value>0,application/vnd.ms-word,,,txt,20002,</value>
       </prop>
     </node>
     <node oor:name="writer_MS_Word_6x_W4W" oor:op="replace">
@@ -1192,7 +1192,7 @@
         <value xml:lang="en-US">Microsoft Word 6.x</value>
       </prop>
       <prop oor:name="Data">
-        <value>0,,,,txt,20002,</value>
+        <value>0,application/vnd.ms-word,,,txt,20002,</value>
       </prop>
     </node>
     <node oor:name="writer_MS_Word_95" oor:op="replace">
@@ -1200,7 +1200,7 @@
         <value xml:lang="en-US">Microsoft Word 95</value>
       </prop>
       <prop oor:name="Data">
-        <value>0,application/msword,MSWordDoc,,doc,20002,</value>
+        <value>0,application/vnd.ms-word,MSWordDoc,,doc,20002,</value>
       </prop>
     </node>
     <node oor:name="writer_MS_Word_95_Vorlage" oor:op="replace">
@@ -1208,7 +1208,7 @@
         <value xml:lang="en-US">MS Word 95 Template</value>
       </prop>
       <prop oor:name="Data">
-        <value>0,application/msword,MSWordDoc,,dot,20002,</value>
+        <value>0,application/vnd.ms-word,MSWordDoc,,dot,20002,</value>
       </prop>
     </node>
     <node oor:name="writer_MS_Word_97" oor:op="replace">
@@ -1216,7 +1216,7 @@
         <value xml:lang="en-US">Microsoft Word 97/2000/XP</value>
       </prop>
       <prop oor:name="Data">
-        <value>1,application/msword,MSWordDoc,,doc,20002,</value>
+        <value>1,application/vnd.ms-word,MSWordDoc,,doc,20002,</value>
       </prop>
     </node>
     <node oor:name="writer_MS_Word_97_Vorlage" oor:op="replace">
@@ -1224,7 +1224,7 @@
         <value xml:lang="en-US">MS Word 97/2000 Template</value>
       </prop>
       <prop oor:name="Data">
-        <value>1,application/msword,MSWordDoc,,dot,20002,</value>
+        <value>1,application/vnd.ms-word,MSWordDoc,,dot,20002,</value>
       </prop>
     </node>
     <node oor:name="writer_MS_Works_20_DOS_W4W" oor:op="replace">
