--- sc/source/filter/inc/formel.hxx.bak	2004-12-15 14:46:17.734822112 +0530
+++ sc/source/filter/inc/formel.hxx	2004-12-15 16:12:19.045183672 +0530
@@ -87,6 +87,9 @@
 #include <compiler.hxx>
 #endif
 
+#ifndef __QPRO_HXX
+#include <qpro.hxx>
+#endif
 
 // ----- forwards --------------------------------------------------------
 
@@ -282,6 +285,24 @@
 
 
 
+class QProConverterBase : public ConverterBase
+{
+protected:
+       BiffReader&                      aIn;
+       sal_uInt16                       nLength;
+                                                                                                                             
+                                         QProConverterBase( BiffReader& rStr, sal_uInt16 qBuffer );
+       virtual                          ~QProConverterBase();
+                                                                                                                             
+public:
+       void                             Reset( ScAddress aEingPos );
+       void                             Reset();
+       virtual ConvErr          Convert( const ScTokenArray*& pErg, sal_uInt16 nLength,
+                                                                        const FORMULA_TYPE eFT = FT_CellFormula )=0;
+};
+
+
+
 class LotusConverterBase : public ConverterBase
 {
 protected:
