--- fpicker/source/unx/kde/kdefilepicker.cxx
+++ fpicker/source/unx/kde/kdefilepicker.cxx
@@ -229,6 +229,8 @@
             case CommandEvent::GetDirectory:
                 {
                     QString qDirectory = baseURL().url();
+                    if ( qDirectory.startsWith( "file:/" ) && qDirectory.mid( 6, 1 ) != "/" )
+                        qDirectory.replace( "file:/", "file:///" );
                     sendCommand( "currentDirectory " + escapeString( qDirectory ) );
                 }
                 break;
@@ -251,7 +253,10 @@
                         for ( KURL::List::const_iterator it = qList.begin(); it != qList.end(); ++it )
                         {
                             qString.append( " " );
-                            appendEscaped( qString, addExtension( (*it).url() ) );
+                            QString qUrlStr = (*it).url();
+                            if ( qUrlStr.startsWith( "file:/" ) && qUrlStr.mid( 6, 1 ) != "/" )
+                                qUrlStr.replace( "file:/", "file:///" );
+                            appendEscaped( qString, addExtension( qUrlStr ) );
                         }
                     }
                     else
@@ -261,7 +266,10 @@
                         for ( KFileItemListIterator it( *pItems ); it.current(); ++it )
                         {
                             qString.append( " " );
-                            appendEscaped( qString, addExtension( (*it)->url().url() ) );
+                            QString qUrlStr = (*it)->url().url();
+                            if ( qUrlStr.startsWith( "file:/" ) && qUrlStr.mid( 6, 1 ) != "/" )
+                                qUrlStr.replace( "file:/", "file:///" );
+                            appendEscaped( qString, addExtension( qUrlStr ) );
                         }
                     }
 
