--- sc/source/filter/excel/frmbase.cxx.bak	2004-12-14 17:42:15.000000000 +0530
+++ sc/source/filter/excel/frmbase.cxx	2004-12-15 15:03:48.292113128 +0530
@@ -312,3 +312,24 @@
 	aEingPos = aNewEingPos;
 }
 
+QProConverterBase::QProConverterBase( BiffReader &rStr, sal_uInt16 nNewBuffer ) :
+        ConverterBase( nNewBuffer ),
+        aIn( rStr )
+{
+}
+                                                                                                                             
+QProConverterBase::~QProConverterBase()
+{
+}
+                                                                                                                             
+void QProConverterBase::Reset( ScAddress aNewPos )
+{
+  ConverterBase::Reset();
+  aEingPos = aNewPos;
+}
+                                                                                                                             
+void QProConverterBase::Reset()
+{
+        ConverterBase::Reset();
+        aEingPos.Set( 0, 0, 0 );
+}
