Index: registry/schema/org/openoffice/Office/Common.xcs
===================================================================
RCS file: /cvs/util/officecfg/registry/schema/org/openoffice/Office/Common.xcs,v
retrieving revision 1.84.8.5
Index: officecfg/registry/schema/org/openoffice/Office/Common.xcs
===================================================================
--- officecfg/registry/schema/org/openoffice/Office/Common.xcs.orig	2004-09-12 12:14:15.192985792 +0200
+++ officecfg/registry/schema/org/openoffice/Office/Common.xcs	2004-09-12 12:18:47.003664288 +0200
@@ -36038,50 +36038,55 @@
 					</enumeration>
 					<enumeration oor:value="1">
 						<info>
-							<desc xml:lang="en-US">32x32 pixel icons</desc>
-							<desc xml:lang="el">32x32 pixel icons</desc>
-							<desc xml:lang="nl">32x32 pixel iconen</desc>
-							<desc xml:lang="fr">32x32 pixel icons</desc>
-							<desc xml:lang="fi">32x32 pixel icons</desc>
-							<desc xml:lang="cs">32x32 pixel icons</desc>
-							<desc xml:lang="da">32x32 pixel icons</desc>
-							<desc xml:lang="zh-CN">32x32 像素图标</desc>
-							<desc xml:lang="zh-TW">32x32 pixel icons</desc>
-							<desc xml:lang="ar">32x32 pixel icons</desc>
-							<desc xml:lang="he">32x32 pixel icons</desc>
-							<desc xml:lang="ca">32x32 pixel icons</desc>
-							<desc xml:lang="hi-IN">32x32 pixel icons</desc>
-							<desc xml:lang="pt">32x32 pixel icons</desc>
-							<desc xml:lang="ru">32x32 pixel icons</desc>
-							<desc xml:lang="es">32x32 pixel icons</desc>
-							<desc xml:lang="it">32x32 pixel icons</desc>
-							<desc xml:lang="sk">32x32 pixel icons</desc>
-							<desc xml:lang="sv">32x32 pixel icons</desc>
-							<desc xml:lang="pl">32x32 pixel icons</desc>
-							<desc xml:lang="pt-BR">32x32 pixel icons</desc>
-							<desc xml:lang="ja">32×32 ピクセルアイコン</desc>
-							<desc xml:lang="ko">32x32 pixel icons</desc>
-							<desc xml:lang="tr">32x32 piksel simgeler</desc>
-							<desc xml:lang="th">32x32 pixel icons</desc>
-							<desc xml:lang="x-comment">32x32 pixel icons</desc>
-							<desc xml:lang="hu">32x32 pixel icons</desc>
-							<desc xml:lang="sl">32x32 pixel icons</desc>
-							<desc xml:lang="ns">32x32 pixel icons</desc>
-							<desc xml:lang="af">32x32 pixel icons</desc>
-							<desc xml:lang="zu">32x32 pixel icons</desc>
-							<desc xml:lang="nb">32x32 pixel icons</desc>
-							<desc xml:lang="et">32x32 pixel icons</desc>
-							<desc xml:lang="cy">32x32 pixel icons</desc>
-							<desc xml:lang="tn">32x32 pixel icons</desc>
-							<desc xml:lang="nn">32x32 pixel icons</desc>
-							<desc xml:lang="eu">32x32 pixel icons</desc>
-							<desc xml:lang="lt">32x32 taškų paveikslai</desc>
-							<desc xml:lang="kn-IN">32x32 pixel icons</desc>
-							<desc xml:lang="gl">32x32 pixel icons</desc>
+							<desc xml:lang="en-US">24x24 pixel icons</desc>
+							<desc xml:lang="el">24x24 pixel icons</desc>
+							<desc xml:lang="nl">24x24 pixel iconen</desc>
+							<desc xml:lang="fr">24x24 pixel icons</desc>
+							<desc xml:lang="fi">24x24 pixel icons</desc>
+							<desc xml:lang="cs">24x24 pixel icons</desc>
+							<desc xml:lang="da">24x24 pixel icons</desc>
+							<desc xml:lang="zh-CN">24x24 像素图标</desc>
+							<desc xml:lang="zh-TW">24x24 pixel icons</desc>
+							<desc xml:lang="ar">24x24 pixel icons</desc>
+							<desc xml:lang="he">24x24 pixel icons</desc>
+							<desc xml:lang="ca">24x24 pixel icons</desc>
+							<desc xml:lang="hi-IN">24x24 pixel icons</desc>
+							<desc xml:lang="pt">24x24 pixel icons</desc>
+							<desc xml:lang="ru">24x24 pixel icons</desc>
+							<desc xml:lang="es">24x24 pixel icons</desc>
+							<desc xml:lang="it">24x24 pixel icons</desc>
+							<desc xml:lang="sk">24x24 pixel icons</desc>
+							<desc xml:lang="sv">24x24 pixel icons</desc>
+							<desc xml:lang="pl">24x24 pixel icons</desc>
+							<desc xml:lang="pt-BR">24x24 pixel icons</desc>
+							<desc xml:lang="ja">24×24 ピクセルアイコン</desc>
+							<desc xml:lang="ko">24x24 pixel icons</desc>
+							<desc xml:lang="tr">24x24 piksel simgeler</desc>
+							<desc xml:lang="th">24x24 pixel icons</desc>
+							<desc xml:lang="x-comment">24x24 pixel icons</desc>
+							<desc xml:lang="hu">24x24 pixel icons</desc>
+							<desc xml:lang="sl">24x24 pixel icons</desc>
+							<desc xml:lang="ns">24x24 pixel icons</desc>
+							<desc xml:lang="af">24x24 pixel icons</desc>
+							<desc xml:lang="zu">24x24 pixel icons</desc>
+							<desc xml:lang="nb">24x24 pixel icons</desc>
+							<desc xml:lang="et">24x24 pixel icons</desc>
+							<desc xml:lang="cy">24x24 pixel icons</desc>
+							<desc xml:lang="tn">24x24 pixel icons</desc>
+							<desc xml:lang="nn">24x24 pixel icons</desc>
+							<desc xml:lang="eu">24x24 pixel icons</desc>
+							<desc xml:lang="lt">24x24 taškų paveikslai</desc>
+							<desc xml:lang="kn-IN">24x24 pixel icons</desc>
+							<desc xml:lang="gl">24x24 pixel icons</desc>
+						</info>
+					</enumeration>
+					<enumeration oor:value="2">
+						<info>
+							<desc xml:lang="en-US">automatic icon sizing</desc>
 						</info>
 					</enumeration>
 				</constraints>
-				<value>0</value>
+				<value>1</value>
 			</prop>
 			<prop oor:name="ToolboxStyle" oor:type="xs:short">
 				<!-- UIHints: Tools  Options General View -->
