Index: registry/data/org/openoffice/VCL.xcu
===================================================================
RCS file: /cvs/util/officecfg/registry/data/org/openoffice/VCL.xcu,v
retrieving revision 1.20.14.7
diff -u -u -r1.20.14.7 VCL.xcu
--- officecfg/registry/data/org/openoffice/VCL.xcu	27 May 2004 13:21:14 -0000	1.20.14.7
+++ officecfg/registry/data/org/openoffice/VCL.xcu	7 Nov 2004 14:21:56 -0000
@@ -409,7 +409,7 @@
     </node>
     <node oor:name="he" oor:op="replace">
      <prop oor:name="UI_SANS" oor:op="replace" oor:type="xs:string">
-        <value>Tahoma;Raanana;Lucidasans;Lucida Sans;Supplement;Andale Sans UI;Arial Unicode MS;Lucida Sans Unicode;Interface User;WarpSans;Geneva;MS Sans Serif;Helv;Dialog;Albany;Lucida;Helvetica;Charcoal;Chicago;Arial;Helmet;Interface System;Sans Serif</value>
+        <value>Ellinia CLM;Tahoma;Raanana;Lucidasans;Lucida Sans;Supplement;Andale Sans UI;Arial Unicode MS;Lucida Sans Unicode;Interface User;WarpSans;Geneva;MS Sans Serif;Helv;Dialog;Albany;Lucida;Helvetica;Charcoal;Chicago;Arial;Helmet;Interface System;Sans Serif</value>
      </prop>
       <prop oor:name="SANS_UNICODE" oor:type="xs:string" oor:op="replace">
         <value>Tahoma;Raanana;Lucidasans;Lucida Sans;Arial Unicode MS;Lucida Sans Unicode;clearlyU;Albany;Arial;Helvetica;Lucida;Geneva;Helmet;SansSerif</value>
