Index: calc/profile_calc.lng
===================================================================
RCS file: /cvs/installation/scp/source/calc/Attic/profile_calc.lng,v
retrieving revision 1.49.24.5
Index: scp/source/calc/profile_calc.lng
===================================================================
--- scp/source/calc/profile_calc.lng.orig	2004-09-13 22:31:50.000000000 +0200
+++ scp/source/calc/profile_calc.lng	2004-09-14 15:49:37.746332016 +0200
@@ -678,7 +678,7 @@
 82 = "%PRODUCTNAME %PRODUCTVERSION μ¤ν”„λ λ“μ‹νΈ"
 86 = "%PRODUCTNAME %PRODUCTVERSION ε·¥δ½θ΅¨"
 88 = "%PRODUCTNAME %PRODUCTVERSION ε·¥δ½θ΅¨"
-03 = "StarOffice 6.0 Folha de cΓ΅lculo"
+03 = "StarOffice 6.0 Folha de cαlculo"
 07 = "έλεκςπξννΰÿ ςΰαλθφΰ %PRODUCTNAME %PRODUCTVERSION"
 30 = "Υπολογιστικό φύλλο %PRODUCTNAME %PRODUCTVERSION"
 31 = "%PRODUCTNAME %PRODUCTVERSION Werkblad"
Index: scp/source/impress/profile_impress.lng
===================================================================
--- scp/source/impress/profile_impress.lng.orig	2004-09-13 22:32:00.000000000 +0200
+++ scp/source/impress/profile_impress.lng	2004-09-14 15:49:37.749331560 +0200
@@ -724,7 +724,7 @@
 82 = "%PRODUCTNAME %PRODUCTVERSION ν”„λ μ  ν…μ΄μ…"
 86 = "%PRODUCTNAME %PRODUCTVERSION ζΌ”η¤Ίζ–‡η¨Ώ"
 88 = "%PRODUCTNAME %PRODUCTVERSION η°΅ε ±"
-03 = "StarOffice 6.0 ApresentaΓ§Γ£o"
+03 = "StarOffice 6.0 Apresentaηγo"
 07 = "Οπεηενςΰφθÿ %PRODUCTNAME %PRODUCTVERSION"
 30 = "Παρουσίαση %PRODUCTNAME %PRODUCTVERSION"
 31 = "%PRODUCTNAME %PRODUCTVERSION Presentatie"
@@ -767,7 +767,7 @@
 34 = "~Presentaciσn"
 55 = "A~presentaηγo"
 48 = "~Prezentacja"
-03 = "A~presentaΓ§Γ£o"
+03 = "A~presentaηγo"
 82 = "ν”„λ μ  ν…μ΄μ…(~P)"
 33 = "~Prιsentation"
 96 = "ΪΡΦ Κ~ήΟνγν"
@@ -809,7 +809,7 @@
 34 = "~Presentaciσn..."
 55 = "A~presentaηγo..."
 48 = "~Prezentacja..."
-03 = "A~presentaΓ§Γ£o..."
+03 = "A~presentaηγo..."
 82 = "ν”„λ μ  ν…μ΄μ…(~P)..."
 33 = "~Prιsentation..."
 96 = "~ΪΡΦ ΚήΟνγμ..."
Index: scp/source/math/profile_math.lng
===================================================================
--- scp/source/math/profile_math.lng.orig	2004-09-13 22:32:04.000000000 +0200
+++ scp/source/math/profile_math.lng	2004-09-14 15:49:37.750331408 +0200
@@ -90,7 +90,7 @@
 34 = "~Fσrmula"
 55 = "~Fσrmula"
 48 = "F~ormu³a"
-03 = "FΓ³~rmula"
+03 = "Fσ~rmula"
 82 = "μμ‹(~O)"
 33 = "F~ormule"
 96 = "ΥνΫΙ"
Index: scp/source/office/profile.lng
===================================================================
--- scp/source/office/profile.lng.orig	2004-09-13 22:32:06.000000000 +0200
+++ scp/source/office/profile.lng	2004-09-14 15:49:37.756330496 +0200
@@ -1225,7 +1225,7 @@
 34 = "~Tarjetas de visita"
 55 = "Cartυes de ~Visita"
 48 = "~Wizytσwki"
-03 = "CartΓµes de ~visita"
+03 = "Cartυes de ~visita"
 82 = "λª…ν•¨(~U)"
 33 = "Cartes de ~visite"
 96 = "ί~ΡΚ ΗαΪγα"
@@ -1477,7 +1477,7 @@
 34 = "Pαgina ~Web..."
 55 = "Pαgina da ~Web..."
 48 = "Strona ~WWW..."
-03 = "PΓ΅gina ~Web..."
+03 = "Pαgina ~Web..."
 82 = "μ›Ή νμ΄μ§€(~W)..."
 33 = "Page ~Web..."
 96 = "~ΥέΝΙ ζνΘ..."
@@ -1519,7 +1519,7 @@
 34 = "Formulario..."
 55 = "~Formulαrio..."
 48 = "Fo~rmularz..."
-03 = "F~ormulΓ΅rio..."
+03 = "F~ormulαrio..."
 82 = "μ–‘μ‹(~R)..."
 33 = "Form~ulaire..."
 96 = "δγ~ζΠΜ..."
@@ -1725,7 +1725,7 @@
 01 = "Address Data Source..."
 44 = "Address Data Source..."
 49 = "Adress-Datenquelle..."
-03 = "Fonte de dados de endereΓ§os..."
+03 = "Fonte de dados de endereηos..."
 07 = "Θρςξχνθκθ δΰννϋυ ΰδπερξβ..."
 30 = "Προέλευση δεδομένων για διευθύνσεις..."
 31 = "Adresgegevensbron..."
