--- ./tools/source/generic/l2txtenc.cxx.indic	2004-11-25 14:50:34.000000000 -0500
+++ ./tools/source/generic/l2txtenc.cxx	2004-11-25 14:53:58.000000000 -0500
@@ -96,6 +96,8 @@
 #define CATALAN					37
 #define THAI					66
 #define HINDI					91
+#define TAMIL					92
+#define GUJARATI				94
 #define EXTERN					99
 
 rtl_TextEncoding Langcode2TextEncoding( USHORT nLang )
@@ -137,6 +139,8 @@
 		case NORTHERNSOTHO: return RTL_TEXTENCODING_UTF8;
 		case AFRIKAANS: return RTL_TEXTENCODING_UTF8;
 		case ZULU: return RTL_TEXTENCODING_UTF8;
+		case TAMIL: return RTL_TEXTENCODING_UTF8;
+		case GUJARATI: return RTL_TEXTENCODING_UTF8;
 	}
 	return RTL_TEXTENCODING_MS_1252;
 }
--- ./tools/source/rc/resmgr.cxx.indic	2004-11-25 14:50:46.000000000 -0500
+++ ./tools/source/rc/resmgr.cxx	2004-11-25 14:53:58.000000000 -0500
@@ -1268,6 +1268,10 @@
 				return "66";
             case LANGUAGE_HINDI:
                 return "91";
+            case LANGUAGE_TAMIL:
+                return "92";
+            case LANGUAGE_GUJARATI:
+                return "94";
 
             case LANGUAGE_ARABIC:
             case LANGUAGE_ARABIC_IRAQ:
@@ -1446,7 +1450,9 @@
         LANGUAGE_HEBREW,
 	LANGUAGE_NORTHERNSOTHO,
 	LANGUAGE_AFRIKAANS,
-	LANGUAGE_ZULU
+	LANGUAGE_ZULU,
+        LANGUAGE_TAMIL,
+        LANGUAGE_GUJARATI
     };
 
     for( size_t i = 0; i < sizeof( aLanguages )/sizeof( aLanguages[0] ); ++i )
--- ./solenv/inc/postset.mk.indic	2004-11-25 16:16:07.957481328 -0500
+++ ./solenv/inc/postset.mk	2004-11-25 16:20:44.931374896 -0500
@@ -518,6 +518,32 @@
 RCLANGFLAGS_28+= -d ZULU
 iso_28=zu
 
+.IF "$(RES_TAMIL)"!="" || "$(give_me_all_languages)"!=""
+alllangext+=92
+.ENDIF
+completelangext+=92
+tamil$(LANG_GUI)*=$(default$(LANG_GUI))
+lang_92=tamil
+longlang_92=tamil
+langext_92=92
+rsclang_92=-lgTAMIL $(UTF8)
+rescharset_92=$(tamil$(LANG_GUI))
+RCLANGFLAGS_92+= -d TAMIL
+iso_92=ta
+
+.IF "$(RES_GUJARATI)"!="" || "$(give_me_all_languages)"!=""
+alllangext+=94
+.ENDIF
+completelangext+=94
+gujarati$(LANG_GUI)*=$(default$(LANG_GUI))
+lang_94=gujarati
+longlang_94=gujarati
+langext_94=94
+rsclang_94=-lgGUJARATI $(UTF8)
+rescharset_94=$(gujarati$(LANG_GUI))
+RCLANGFLAGS_94+= -d GUJARATI
+iso_94=gu
+
 .IF "$(L10N_framework)"!=""
 .IF "$(native_lang_list)"!=""
 helplist=$(uniq $(alllangext) $(native_lang_list))
--- ./solenv/inc/lang.mk.indic	2004-11-25 16:22:31.007248912 -0500
+++ ./solenv/inc/lang.mk	2004-11-25 16:22:55.972453624 -0500
@@ -96,6 +96,8 @@
 nsotho$(LANG_GUI)=-CHARSET_UTF8
 afrik$(LANG_GUI)=-CHARSET_UTF8
 zulu$(LANG_GUI)=-CHARSET_UTF8
+tamil$(LANG_GUI)=-CHARSET_UTF8
+gujarati$(LANG_GUI)=-CHARSET_UTF8
 
 ###################################################
 #
--- ./transex3/inc/export.hxx.indic	2004-11-25 14:49:10.000000000 -0500
+++ ./transex3/inc/export.hxx	2004-11-25 14:53:58.000000000 -0500
@@ -223,11 +223,21 @@
 #define NORWEGIAN_NYNORSK_ISO			"nn"
 #define NORWEGIAN_NYNORSK_INDEX                 37
 
+#define TAMIL					92
+#define TAMIL_ISO				"ta-IN"
+#define TAMIL_INDEX				38
+
+#define GUJARATI				94
+#define GUJARATI_ISO			"ga-IN"
+#define GUJARATI_INDEX			39
+
+
+
 // special language for l10n framework
 #define EXTERN					99
-#define EXTERN_INDEX				38
+#define EXTERN_INDEX				40
 
-#define LANGUAGES					39
+#define LANGUAGES					41
 
 #define NO_TRANSLATE_ISO		"x-no-translate"
 
@@ -244,6 +254,7 @@
 								( Export::LangId[ n ] == 88 ) || ( Export::LangId[ n ] == 90 ) || ( Export::LangId[ n ] == 96 ) || 	\
 								( Export::LangId[ n ] == 42 ) || ( Export::LangId[ n ] == 43 ) || ( Export::LangId[ n ] == 97 ) || 	\
 								( Export::LangId[ n ] == 66 ) || ( Export::LangId[ n ] == 91 ) || ( Export::LangId[ n ] == 00 ) ||  \
+								( Export::LangId[ n ] == 92 ) || ( Export::LangId[ n ] == 94 ) ||  \
 								( Export::LangId[ n ] == 99 ) || ( Export::LangId[ n ] == 77 ) || ( Export::LangId[ n ] == 36 )))
 
 #define GERMAN_LIST_LINE_INDEX 	LANGUAGES
--- ./transex3/source/export2.cxx.indic	2004-11-25 14:50:01.000000000 -0500
+++ ./transex3/source/export2.cxx	2004-11-25 14:53:58.000000000 -0500
@@ -154,6 +154,8 @@
     AFRIKAANS,
     ZULU,
     NORWEGIAN_NYNORSK,
+	TAMIL,
+	GUJARATI,
 	EXTERN
 };
 
@@ -257,6 +259,10 @@
 		return HINDI;
 	else if ( sLang == ByteString( ESTONIAN_ISO ).ToUpperAscii())
 		return ESTONIAN;
+	else if ( sLang == ByteString( TAMIL_ISO ).ToUpperAscii())
+		return TAMIL;
+	else if ( sLang == ByteString( GUJARATI_ISO ).ToUpperAscii())
+		return GUJARATI;
 	else if ( sLang == ByteString( sIsoCode99 ).ToUpperAscii())
 		return EXTERN;
 
@@ -305,6 +311,8 @@
         case NORTHERNSOTHO_INDEX: return NORTHERNSOTHO_ISO;
         case AFRIKAANS_INDEX: return AFRIKAANS_ISO;
         case ZULU_INDEX: return ZULU_ISO;
+		case TAMIL_INDEX: return TAMIL_ISO;
+		case GUJARATI_INDEX: return GUJARATI_ISO;
         case EXTERN_INDEX: return sIsoCode99;
 	}
 	return "";
@@ -439,6 +447,8 @@
     "afrikaans",
     "zulu",
     "norwegian_nynorsk",
+	"tamil",
+	"gujarati",
 	"extern"
 };
 
--- ./transex3/source/export.cxx.indic	2004-11-25 14:49:45.000000000 -0500
+++ ./transex3/source/export.cxx	2004-11-25 14:53:58.000000000 -0500
@@ -585,7 +585,7 @@
 				// cur. line has macro line end
 				ByteString sTmpLine( sToken );
 				sTmpLine.EraseAllChars( '\t' ); sTmpLine.EraseAllChars( ' ' );
-				if( sTmpLine.Len() < 0 ){
+				if( sTmpLine.Len() > 0 ){
 					if ( sTmpLine.GetChar(( USHORT )( sTmpLine.Len() - 1 )) != '\\' )
 						bNextMustBeDefineEOL = TRUE;
 				}
@@ -1277,6 +1277,10 @@
 		return THAI_INDEX;
 	else if ( sLang == "HINDI" )
 		return HINDI_INDEX;
+	else if ( sLang == "TAMIL" )
+		return TAMIL_INDEX;
+	else if ( sLang == "GUJARATI" )
+		return GUJARATI_INDEX;
 	else if ( sLang == "EXTERN" )
 		return EXTERN_INDEX;
 
@@ -2416,7 +2420,8 @@
 									if ( sLine.GetChar( nEnd ) == '\"' )
 										bFound = TRUE;
 								}
-								nEnd --;
+								if (sLine.Search (" \"\\\"\" ;") == STRING_NOTFOUND)
+									nEnd --;
 								ByteString sPostFix( sLine.Copy( ++nEnd ));
 								sLine.Erase( nStart );
 
--- ./transex3/source/merge.cxx.indic	2004-11-25 14:50:17.000000000 -0500
+++ ./transex3/source/merge.cxx	2004-11-25 14:53:58.000000000 -0500
@@ -183,7 +183,8 @@
 		aInputStream.ReadLine( sLine );
 		sLine = sLine.Convert( RTL_TEXTENCODING_MS_1252, aCharSet );
 
-		if ( sLine.GetTokenCount( '\t' ) == 15 ) {
+		int tcnt = sLine.GetTokenCount( '\t' );
+		if ( tcnt >= 13 && tcnt <= 16 ) {
 			if ( nFileFormat != FFORMAT_NEW ) {
 				nFileFormat = FFORMAT_NEW;
 				fprintf( stdout, "File detection: Version 2.0 detected!\n" );
@@ -428,6 +429,8 @@
         case NORTHERNSOTHO: return NORTHERNSOTHO_INDEX;
         case AFRIKAANS: return AFRIKAANS_INDEX;
         case ZULU: return ZULU_INDEX;
+		case TAMIL: return TAMIL_INDEX;
+		case GUJARATI: return GUJARATI_INDEX;
         case EXTERN: return EXTERN_INDEX;
 	}
 	return 0xFFFF;
--- ./transex3/source/srclex.l.indic	2004-11-25 14:51:04.000000000 -0500
+++ ./transex3/source/srclex.l	2004-11-25 14:54:10.000000000 -0500
@@ -129,6 +129,12 @@
 	WorkOnTokenSet( LEVELDOWN, yytext );
 }
 
+"}"[ \t]*	{
+/* LEVELDOWN alternate */
+	YYWarning ("Missing ; at end of block (should be '};')");
+	WorkOnTokenSet( LEVELDOWN, yytext );
+}
+
 [a-zA-Z0-9_]+[ \t]*"="[ \t]*"MAP_APPFONT"[ \t]*"(".+")".*	{
 /* APPFONTMAPPING  Typ = MAP_APPFONT( ... ) */
 	WorkOnTokenSet( APPFONTMAPPING, yytext );
--- ./readlicense_oo/prj/d.lst.indic	2004-11-27 15:49:19.000000000 -0500
+++ ./readlicense_oo/prj/d.lst	2004-11-27 16:07:19.000000000 -0500
@@ -71,6 +71,10 @@
 mkdir: %_DEST%\bin%_EXT%\90\osl
 mkdir: %_DEST%\bin%_EXT%\91
 mkdir: %_DEST%\bin%_EXT%\91\osl
+mkdir: %_DEST%\bin%_EXT%\92
+mkdir: %_DEST%\bin%_EXT%\92\osl
+mkdir: %_DEST%\bin%_EXT%\94
+mkdir: %_DEST%\bin%_EXT%\94\osl
 mkdir: %_DEST%\bin%_EXT%\96
 mkdir: %_DEST%\bin%_EXT%\96\osl
 mkdir: %_DEST%\bin%_EXT%\97
@@ -205,6 +209,18 @@
 ..\%__SRC%\misc\license\%GUI%\LICENSE %_DEST%\bin%_EXT%\90\osl\LICENSE90
 ..\%__SRC%\misc\license\WNT\license.txt %_DEST%\bin%_EXT%\90\osl\license90.txt
 ..\%__SRC%\misc\license\%GUI%\license.html %_DEST%\bin%_EXT%\90\osl\LICENSE90.html
+..\%__SRC%\misc\license\%GUI%\* %_DEST%\bin%_EXT%\91\osl\*
+..\%__SRC%\misc\license\%GUI%\LICENSE %_DEST%\bin%_EXT%\91\osl\LICENSE91
+..\%__SRC%\misc\license\WNT\license.txt %_DEST%\bin%_EXT%\91\osl\license91.txt
+..\%__SRC%\misc\license\%GUI%\license.html %_DEST%\bin%_EXT%\91\osl\LICENSE91.html
+..\%__SRC%\misc\license\%GUI%\* %_DEST%\bin%_EXT%\92\osl\*
+..\%__SRC%\misc\license\%GUI%\LICENSE %_DEST%\bin%_EXT%\92\osl\LICENSE92
+..\%__SRC%\misc\license\WNT\license.txt %_DEST%\bin%_EXT%\92\osl\license92.txt
+..\%__SRC%\misc\license\%GUI%\license.html %_DEST%\bin%_EXT%\92\osl\LICENSE92.html
+..\%__SRC%\misc\license\%GUI%\* %_DEST%\bin%_EXT%\94\osl\*
+..\%__SRC%\misc\license\%GUI%\LICENSE %_DEST%\bin%_EXT%\94\osl\LICENSE94
+..\%__SRC%\misc\license\WNT\license.txt %_DEST%\bin%_EXT%\94\osl\license94.txt
+..\%__SRC%\misc\license\%GUI%\license.html %_DEST%\bin%_EXT%\94\osl\LICENSE94.html
 ..\%__SRC%\misc\license\%GUI%\* %_DEST%\bin%_EXT%\96\osl\*
 ..\%__SRC%\misc\license\%GUI%\LICENSE %_DEST%\bin%_EXT%\96\osl\LICENSE96
 ..\%__SRC%\misc\license\WNT\license.txt %_DEST%\bin%_EXT%\96\osl\license96.txt
@@ -317,6 +333,12 @@
 ..\%__SRC%\misc\91\* %_DEST%\bin%_EXT%\91\osl\*
 ..\%__SRC%\misc\91\readme91.txt %_DEST%\bin%_EXT%\91\osl\README91
 ..\%__SRC%\misc\91\readme91.html %_DEST%\bin%_EXT%\91\osl\README91.html
+..\%__SRC%\misc\92\* %_DEST%\bin%_EXT%\92\osl\*
+..\%__SRC%\misc\92\readme92.txt %_DEST%\bin%_EXT%\92\osl\README92
+..\%__SRC%\misc\92\readme92.html %_DEST%\bin%_EXT%\92\osl\README92.html
+..\%__SRC%\misc\94\* %_DEST%\bin%_EXT%\94\osl\*
+..\%__SRC%\misc\94\readme94.txt %_DEST%\bin%_EXT%\94\osl\README94
+..\%__SRC%\misc\94\readme94.html %_DEST%\bin%_EXT%\94\osl\README94.html
 ..\%__SRC%\misc\96\* %_DEST%\bin%_EXT%\96\osl\*
 ..\%__SRC%\misc\96\readme96.txt %_DEST%\bin%_EXT%\96\osl\README96
 ..\%__SRC%\misc\96\readme96.html %_DEST%\bin%_EXT%\96\osl\README96.html
--- ./scp/inc/gcfg.inc.indic	2004-11-27 15:13:34.000000000 -0500
+++ ./scp/inc/gcfg.inc	2004-11-27 15:40:16.000000000 -0500
@@ -264,6 +264,8 @@
     ident (88)      = resid; \
     ident (90)      = resid; \
     ident (91)      = resid; \
+	ident (92)		= resid; \
+	ident (94)		= resid; \
     ident (96)      = resid; \
     ident (97)      = resid; \
     ident (99)      = resid
@@ -303,6 +305,8 @@
     Name (88)       = RESFILENAME(name,88);     \
     Name (90)       = RESFILENAME(name,90);     \
     Name (91)       = RESFILENAME(name,91);     \
+    Name (92)       = RESFILENAME(name,92);     \
+    Name (94)       = RESFILENAME(name,94);     \
     Name (96)       = RESFILENAME(name,96);     \
 	Name (97) 		= RESFILENAME(name,97);		\
     Name (99)       = RESFILENAME(name,99)
@@ -342,6 +346,8 @@
     Name (88)       = SHORTRESFILENAME(name,88);     \
     Name (90)       = SHORTRESFILENAME(name,90);     \
     Name (91)       = SHORTRESFILENAME(name,91);     \
+    Name (92)       = SHORTRESFILENAME(name,92);     \
+    Name (94)       = SHORTRESFILENAME(name,94);     \
     Name (96)       = SHORTRESFILENAME(name,96);     \
     Name (97)       = SHORTRESFILENAME(name,97);     \
     Name (99)       = SHORTRESFILENAME(name,99)
@@ -381,6 +387,8 @@
     key (88)       = READMETXTFILENAME(name,88,ext);     \
     key (90)       = READMETXTFILENAME(name,90,ext);     \
     key (91)       = READMETXTFILENAME(name,91,ext);     \
+    key (92)       = READMETXTFILENAME(name,92,ext);     \
+    key (94)       = READMETXTFILENAME(name,94,ext);     \
     key (96)       = READMETXTFILENAME(name,96,ext);     \
     key (97)       = READMETXTFILENAME(name,97,ext);     \
     key (99)       = READMETXTFILENAME(name,99,ext)
@@ -420,6 +428,8 @@
     key (88)       = READMEFILENAME(name,88);     \
     key (90)       = READMEFILENAME(name,90);     \
     key (91)       = READMEFILENAME(name,91);     \
+    key (92)       = READMEFILENAME(name,92);     \
+    key (94)       = READMEFILENAME(name,94);     \
     key (96)       = READMEFILENAME(name,96);     \
     key (97)       = READMEFILENAME(name,97);     \
     key (99)       = READMEFILENAME(name,99)
@@ -459,6 +469,8 @@
     Name (88)       = EXTRAFILENAME(name,88);     \
     Name (90)       = EXTRAFILENAME(name,90);     \
     Name (91)       = EXTRAFILENAME(name,91);     \
+    Name (92)       = EXTRAFILENAME(name,92);     \
+    Name (94)       = EXTRAFILENAME(name,94);     \
     Name (96)       = EXTRAFILENAME(name,96);     \
     Name (97)       = EXTRAFILENAME(name,97);     \
     Name (99)       = EXTRAFILENAME(name,99)
@@ -498,6 +510,8 @@
     Name (88)       = STRING(name);     \
     Name (90)       = STRING(name);     \
     Name (91)       = STRING(name);     \
+    Name (92)       = STRING(name);     \
+    Name (94)       = STRING(name);     \
     Name (96)       = STRING(name);     \
     Name (97)       = STRING(name);     \
     Name (99)       = STRING(name)
@@ -537,6 +551,8 @@
     Name (88)       = HELPFILENAME(name,88);     \
     Name (90)       = HELPFILENAME(name,90);     \
     Name (91)       = HELPFILENAME(name,91);     \
+    Name (92)       = HELPFILENAME(name,92);     \
+    Name (94)       = HELPFILENAME(name,94);     \
     Name (96)       = HELPFILENAME(name,96);     \
     Name (97)       = HELPFILENAME(name,97);     \
     Name (99)       = HELPFILENAME(name,99)
@@ -576,6 +592,8 @@
     Name (88)       = HELPFILENAME(name,88x);     \
     Name (90)       = HELPFILENAME(name,90x);     \
     Name (91)       = HELPFILENAME(name,91x);     \
+    Name (92)       = HELPFILENAME(name,92x);     \
+    Name (94)       = HELPFILENAME(name,94x);     \
     Name (96)       = HELPFILENAME(name,96x);     \
     Name (97)       = HELPFILENAME(name,97x);     \
     Name (99)       = HELPFILENAME(name,99x)
@@ -615,6 +633,8 @@
     Name (88)       = HELPFILENAME(name,endung);     \
     Name (90)       = HELPFILENAME(name,endung);     \
     Name (91)       = HELPFILENAME(name,endung);     \
+    Name (92)       = HELPFILENAME(name,endung);     \
+    Name (94)       = HELPFILENAME(name,endung);     \
     Name (96)       = HELPFILENAME(name,endung);     \
     Name (97)       = HELPFILENAME(name,endung);     \
     Name (99)       = HELPFILENAME(name,endung)
@@ -654,6 +674,8 @@
     Dir (88)       = GID_DIR_HELP_88;     \
     Dir (90)       = GID_DIR_HELP_90;     \
     Dir (91)       = GID_DIR_HELP_91;     \
+    Dir (92)       = GID_DIR_HELP_92;     \
+    Dir (94)       = GID_DIR_HELP_94;     \
     Dir (96)       = GID_DIR_HELP_96;     \
     Dir (97)       = GID_DIR_HELP_97;     \
     Dir (99)       = GID_DIR_HELP_99
@@ -693,6 +715,8 @@
     DosName (88)       = STRING(name);     \
     DosName (90)       = STRING(name);     \
     DosName (91)       = STRING(name);     \
+    DosName (92)       = STRING(name);     \
+    DosName (94)       = STRING(name);     \
     DosName (96)       = STRING(name);     \
     DosName (97)       = STRING(name);     \
     DosName (99)       = STRING(name)
@@ -732,6 +756,8 @@
     DosName (88)       = "chinese_traditional";    \
     DosName (90)       = "turkish";     \
     DosName (91)       = "hindi";     \
+    DosName (92)       = "tamil";     \
+    DosName (94)       = "gujarati";     \
     DosName (96)       = "arabic";     \
     DosName (97)       = "hebrew";     \
     DosName (99)       = STRING(ISO_CODE)
@@ -771,6 +797,8 @@
     DosName (88)       = "zh-TW"; \
     DosName (90)       = "tr";  \
     DosName (91)       = "hi-IN";  \
+    DosName (92)       = "ta-IN";  \
+    DosName (94)       = "gu-IN";  \
     DosName (96)       = "ar";  \
     DosName (97)       = "he";  \
     DosName (99)       = STRING(ISO_CODE)
@@ -810,6 +838,8 @@
     DosName (88)       = "zh-TW"; \
     DosName (90)       = "tr";  \
     DosName (91)       = "hi-IN";  \
+    DosName (92)       = "ta-IN";  \
+    DosName (94)       = "gu-IN";  \
     DosName (96)       = "ar";  \
     DosName (97)       = "he";  \
     DosName (99)       = STRING(ISO_CODE)
@@ -849,6 +879,8 @@
         Section (88)              = STRING(ident##-88); \
         Section (90)              = STRING(ident##-90); \
         Section (91)              = STRING(ident##-91); \
+        Section (92)              = STRING(ident##-92); \
+        Section (94)              = STRING(ident##-94); \
         Section (96)              = STRING(ident##-96); \
         Section (97)              = STRING(ident##-97); \
         Section (99)              = STRING(ident##-99)
@@ -891,6 +923,8 @@
         Key (88)              = STRING(ident##-88-C); \
         Key (90)              = STRING(ident##-90-C); \
         Key (91)              = STRING(ident##-91-C); \
+        Key (92)              = STRING(ident##-92-C); \
+        Key (94)              = STRING(ident##-94-C); \
         Key (96)              = STRING(ident##-96-C); \
         Key (97)              = STRING(ident##-97); \
         Key (99)              = STRING(ident##-99)
@@ -948,7 +982,9 @@
         Key (86)              = STRING(2052); \
         Key (88)              = STRING(1028); \
         Key (90)              = STRING(1055); \
-        Key (91)              = STRING(91); \
+        Key (91)              = STRING(1081); \
+        Key (92)              = STRING(1097); \
+        Key (94)              = STRING(1095); \
         Key (96)              = STRING(1055); \
         Key (97)              = STRING(1037); \
         Key (99)              = STRING(ISO_CODE)
@@ -988,6 +1024,8 @@
         Key (88)              = STRING(zh-TW); \
         Key (90)              = STRING(tr); \
         Key (91)              = STRING(hi-IN); \
+        Key (92)              = STRING(ta-IN); \
+        Key (94)              = STRING(gu-IN); \
         Key (96)              = STRING(ar); \
         Key (97)              = STRING(he); \
         Key (99)              = STRING(ISO_CODE)
@@ -1025,7 +1063,9 @@
         Value (86)              = STRING(2052); \
         Value (88)              = STRING(1028); \
         Value (90)              = STRING(1055); \
-        Value (91)              = STRING(91); \
+        Value (91)              = STRING(1081); \
+        Value (92)              = STRING(1097); \
+        Value (94)              = STRING(1095); \
         Value (96)              = STRING(1025); \
         Value (97)              = STRING(1037); \
         Value (99)              = STRING(ISO_CODE)
@@ -1065,6 +1105,8 @@
         Value (88)              = STRING(zh-TW); \
         Value (90)              = STRING(tr); \
         Value (91)              = STRING(hi-IN); \
+        Value (92)              = STRING(ta-IN); \
+        Value (94)              = STRING(gu-IN); \
         Value (96)              = STRING(ar); \
         Value (97)              = STRING(he); \
         Value (99)              = STRING(ISO_CODE)
@@ -1104,6 +1146,8 @@
         Value (88)              = STRING(en-US); \
         Value (90)              = STRING(tr); \
         Value (91)              = STRING(hi-IN); \
+        Value (92)              = STRING(ta-IN); \
+        Value (94)              = STRING(gu-IN); \
         Value (96)              = STRING(ar); \
         Value (97)              = STRING(he); \
         Value (99)              = STRING(ISO_CODE)
@@ -1143,6 +1187,8 @@
         Value (88)              = STRING(zh-TW); \
         Value (90)              = STRING(tr); \
         Value (91)              = STRING(hi-IN); \
+        Value (92)              = STRING(ta-IN); \
+        Value (94)              = STRING(gu-IN); \
         Value (96)              = STRING(ar); \
         Value (97)              = STRING(he); \
         Value (99)              = STRING(ISO_CODE)
@@ -1182,6 +1228,8 @@
         Value (88)              = STRING(88); \
         Value (90)              = STRING(90); \
         Value (91)              = STRING(91); \
+        Value (92)              = STRING(92); \
+        Value (94)              = STRING(94); \
         Value (96)              = STRING(96); \
         Value (97)              = STRING(97); \
         Value (99)              = STRING(99)
--- ./scp/source/global/setupzip.scp.indic	2004-11-27 15:14:12.000000000 -0500
+++ ./scp/source/global/setupzip.scp	2004-11-27 16:13:24.000000000 -0500
@@ -653,6 +653,8 @@
         Name (88)       = "fontunxpsprint.zip";
         Name (90)       = "fontunxpsprint.zip";
         Name (91)       = "fontunxpsprint.zip";
+        Name (92)       = "fontunxpsprint.zip";
+        Name (94)       = "fontunxpsprint.zip";
         Name (96)       = "fontunxpsprint.zip";
         Name (97)       = "fontunxpsprint.zip";
         Name (99)       = "fontunxpsprint.zip";
--- ./i18npool/source/breakiterator/breakiteratorImpl.cxx.orig	2004-12-09 01:57:22.922849304 -0500
+++ ./i18npool/source/breakiterator/breakiteratorImpl.cxx	2004-12-08 22:51:27.000000000 -0500
@@ -403,6 +403,10 @@
     { UnicodeScript_kHebrew, 			ScriptType::COMPLEX },		// 10,
     { UnicodeScript_kArabic, 			ScriptType::COMPLEX },		// 11,
     { UnicodeScript_kDevanagari,		ScriptType::COMPLEX },		// 14,
+    { UnicodeScript_kBengali,			ScriptType::COMPLEX },		// 15,
+    { UnicodeScript_kGurmukhi,		ScriptType::COMPLEX },		// 16,
+    { UnicodeScript_kGujarati,		ScriptType::COMPLEX },		// 17,
+    { UnicodeScript_kTamil,			ScriptType::COMPLEX },		// 19,
     { UnicodeScript_kThai,			ScriptType::COMPLEX },		// 24,
 
     { UnicodeScript_kTibetan,			ScriptType::LATIN },		// 26,
