--- scp/source/office/profile.lng.ita	2004-09-25 10:54:15.143495568 +0200
+++ scp/source/office/profile.lng	2004-09-25 10:55:01.387437949 +0200
@@ -1858,7 +1858,7 @@
 46 = "FontOOo Wizard..."
 45 = "Installer skrifttyper fra internettet..."
 49 = "Schriftarten aus dem Internet installieren..."
-39 = "FontOOo Wizard..."
+39 = "Installa font da Internet..."
 35 = "FontOOo Wizard..."
 37 = "Instal·la tipus de lletra des d'Internet..."
 66 = "FontOOo Wizard..."
