--- ./config_office/set_soenv.in.indic	2005-01-06 09:58:28.764700768 -0500
+++ ./config_office/set_soenv.in	2005-01-06 09:58:50.442129887 -0500
@@ -50,7 +50,8 @@
 #
 # Platform independent constant values.
 my @LANGUAGES = ("AFRIK", "ARAB", "BASQUE", "CAT", "CHINSIM", "CHINTRAD", "CZECH", "DAN", "DTCH", "ENUS", "ESTONIAN", "FINN", "FREN", "GAL", "GREEK", "HEBREW", "HINDI", "HUNG", "ITAL",
-"JAPN", "KANNADA", "KOREAN", "LITHUANIAN", "NORBOK", "NORNYN", "NSOTHO", "POL", "PORT", "PORTBR", "RUSS", "SLOVAK", "SLOVENIAN", "SPAN", "SWED", "THAI", "TSWANA", "TURK", "WELSH", "ZULU" ); 
+"JAPN", "KANNADA", "KOREAN", "LITHUANIAN", "NORBOK", "NORNYN", "NSOTHO", "POL", "PORT", "PORTBR", "RUSS", "SLOVAK", "SLOVENIAN", "SPAN", "SWED", "THAI", "TSWANA", "TURK", "WELSH", "ZULU",
+"TAMIL", "GUJARATI" ); 
 my @DICTIONARIES = ("DADK", "DEDE", "ENGB", "ENUS", "ITIT", "RURU" );
 my ( $CALL_CDECL, $COMMON_OUTDIR, $NO_SRS_PATCH, $PRODUCT, $PROFULLSWITCH, $BIG_GOODIES, $BMP_WRITES_FLAG, 
      $common_build, $MK_UNROLL, $NO_REC_RES, $PROEXT, $SO3, $SOLAR_JAVA, 
--- ./transex3/inc/export.hxx.indic	2005-01-04 16:14:53.403273610 -0500
+++ ./transex3/inc/export.hxx	2005-01-04 16:16:26.379237203 -0500
@@ -242,11 +242,20 @@
 #define GALICIAN_ISO				"gl"
 #define GALICIAN_INDEX				42
 
+#define TAMIL					92
+#define TAMIL_ISO				"ta-IN"
+#define TAMIL_INDEX				43
+
+#define GUJARATI				94
+#define GUJARATI_ISO			"ga-IN"
+#define GUJARATI_INDEX			44
+
+
 // special language for l10n framework
 #define EXTERN					99
-#define EXTERN_INDEX				43
+#define EXTERN_INDEX				45
 
-#define LANGUAGES					44
+#define LANGUAGES					46
 
 #define NO_TRANSLATE_ISO		"x-no-translate"
 
@@ -265,6 +274,7 @@
 								( Export::LangId[ n ] == 42 ) || ( Export::LangId[ n ] == 43 ) || ( Export::LangId[ n ] == 97 ) || 	\
 								( Export::LangId[ n ] == 66 ) || ( Export::LangId[ n ] == 91 ) || ( Export::LangId[ n ] == 00 ) ||  \
 								( Export::LangId[ n ] == 99 ) || ( Export::LangId[ n ] == 77 ) || ( Export::LangId[ n ] == 36 ) ||	\
+								( Export::LangId[ n ] == 92 ) || ( Export::LangId[ n ] == 94 ) ||  \
 								( Export::LangId[ n ] == 93 )))
 
 #define GERMAN_LIST_LINE_INDEX 	LANGUAGES
--- ./transex3/source/export2.cxx.indic	2005-01-04 16:16:38.116464596 -0500
+++ ./transex3/source/export2.cxx	2005-01-04 16:17:39.983118807 -0500
@@ -158,6 +158,8 @@
 	LITHUANIAN,
 	KANNADA,
 	GALICIAN,
+	TAMIL,
+	GUJARATI,
 	EXTERN
 };
 
@@ -271,6 +273,10 @@
 		return KANNADA;
 	else if ( sLang == ByteString( GALICIAN_ISO ).ToUpperAscii())
 		return GALICIAN;
+	else if ( sLang == ByteString( TAMIL_ISO ).ToUpperAscii())
+		return TAMIL;
+	else if ( sLang == ByteString( GUJARATI_ISO ).ToUpperAscii())
+		return GUJARATI;
 	else if ( sLang == ByteString( sIsoCode99 ).ToUpperAscii())
 		return EXTERN;
 
@@ -324,6 +330,8 @@
 		case LITHUANIAN_INDEX: return LITHUANIAN_ISO;
 		case KANNADA_INDEX: return KANNADA_ISO;
 		case GALICIAN_INDEX: return GALICIAN_ISO;
+		case TAMIL_INDEX: return TAMIL_ISO;
+		case GUJARATI_INDEX: return GUJARATI_ISO;
 		case EXTERN_INDEX: return sIsoCode99;
 	}
 	return "";
@@ -463,6 +471,8 @@
 	"lithuanian",
 	"kannada",
 	"galician",
+	"tamil",
+	"gujarati",
 	"extern"
 };
 
--- ./transex3/source/export.cxx.indic	2005-01-04 16:17:53.418088741 -0500
+++ ./transex3/source/export.cxx	2005-01-04 16:18:40.374991931 -0500
@@ -1286,6 +1286,10 @@
 		return LITHUANIAN_INDEX; 
 	else if ( sLang == "KANNADA" )
 		return KANNADA_INDEX; 
+	else if ( sLang == "TAMIL" )
+		return TAMIL_INDEX;
+	else if ( sLang == "GUJARATI" )
+		return GUJARATI_INDEX;
 	else if ( sLang == "EXTERN" )
 		return EXTERN_INDEX;
 
--- ./transex3/source/merge.cxx.indic	2005-01-04 16:19:44.545289901 -0500
+++ ./transex3/source/merge.cxx	2005-01-04 16:20:00.996801887 -0500
@@ -432,6 +432,8 @@
 		case LITHUANIAN: return LITHUANIAN_INDEX;
 		case KANNADA: return KANNADA_INDEX;
 		case GALICIAN: return GALICIAN_INDEX;
+		case TAMIL: return TAMIL_INDEX;
+		case GUJARATI: return GUJARATI_INDEX;
 		case EXTERN: return EXTERN_INDEX;
 	}
 	return 0xFFFF;
--- ./scp/inc/gcfg.inc.indic	2005-01-04 16:22:02.398426293 -0500
+++ ./scp/inc/gcfg.inc	2005-01-04 16:27:59.327611142 -0500
@@ -268,7 +268,9 @@
     ident (88)      = resid; \
     ident (90)      = resid; \
     ident (91)      = resid; \
+    ident (92)      = resid; \
     ident (93)      = resid; \
+    ident (94)      = resid; \
     ident (96)      = resid; \
     ident (97)      = resid; \
     ident (99)      = resid
@@ -312,7 +314,9 @@
     Name (88)       = RESFILENAME(name,88);     \
     Name (90)       = RESFILENAME(name,90);     \
     Name (91)       = RESFILENAME(name,91);     \
+    Name (92)       = RESFILENAME(name,92);     \
     Name (93)       = RESFILENAME(name,93);     \
+    Name (94)       = RESFILENAME(name,94);     \
     Name (96)       = RESFILENAME(name,96);     \
 	Name (97) 		= RESFILENAME(name,97);		\
     Name (99)       = RESFILENAME(name,99)
@@ -356,7 +360,9 @@
     Name (88)       = SHORTRESFILENAME(name,88);     \
     Name (90)       = SHORTRESFILENAME(name,90);     \
     Name (91)       = SHORTRESFILENAME(name,91);     \
+    Name (92)       = SHORTRESFILENAME(name,92);     \
     Name (93)       = SHORTRESFILENAME(name,93);     \
+    Name (94)       = SHORTRESFILENAME(name,94);     \
     Name (96)       = SHORTRESFILENAME(name,96);     \
     Name (97)       = SHORTRESFILENAME(name,97);     \
     Name (99)       = SHORTRESFILENAME(name,99)
@@ -400,7 +406,9 @@
     key (88)       = READMETXTFILENAME(name,88,ext);     \
     key (90)       = READMETXTFILENAME(name,90,ext);     \
     key (91)       = READMETXTFILENAME(name,91,ext);     \
+    key (92)       = READMETXTFILENAME(name,92,ext);     \
     key (93)       = READMETXTFILENAME(name,93,ext);     \
+    key (94)       = READMETXTFILENAME(name,94,ext);     \
     key (96)       = READMETXTFILENAME(name,96,ext);     \
     key (97)       = READMETXTFILENAME(name,97,ext);     \
     key (99)       = READMETXTFILENAME(name,99,ext)
@@ -444,7 +452,9 @@
     key (88)       = READMEFILENAME(name,88);     \
     key (90)       = READMEFILENAME(name,90);     \
     key (91)       = READMEFILENAME(name,91);     \
+    key (92)       = READMEFILENAME(name,92);     \
     key (93)       = READMEFILENAME(name,93);     \
+    key (94)       = READMEFILENAME(name,94);     \
     key (96)       = READMEFILENAME(name,96);     \
     key (97)       = READMEFILENAME(name,97);     \
     key (99)       = READMEFILENAME(name,99)
@@ -488,7 +498,9 @@
     Name (88)       = EXTRAFILENAME(name,88);     \
     Name (90)       = EXTRAFILENAME(name,90);     \
     Name (91)       = EXTRAFILENAME(name,91);     \
+    Name (92)       = EXTRAFILENAME(name,92);     \
     Name (93)       = EXTRAFILENAME(name,93);     \
+    Name (94)       = EXTRAFILENAME(name,94);     \
     Name (96)       = EXTRAFILENAME(name,96);     \
     Name (97)       = EXTRAFILENAME(name,97);     \
     Name (99)       = EXTRAFILENAME(name,99)
@@ -532,7 +544,9 @@
     Name (88)       = STRING(name);     \
     Name (90)       = STRING(name);     \
     Name (91)       = STRING(name);     \
+    Name (92)       = STRING(name);     \
     Name (93)       = STRING(name);     \
+    Name (94)       = STRING(name);     \
     Name (96)       = STRING(name);     \
     Name (97)       = STRING(name);     \
     Name (99)       = STRING(name)
@@ -576,7 +590,9 @@
     Name (88)       = HELPFILENAME(name,88);     \
     Name (90)       = HELPFILENAME(name,90);     \
     Name (91)       = HELPFILENAME(name,91);     \
+    Name (92)       = HELPFILENAME(name,92);     \
     Name (93)       = HELPFILENAME(name,93);     \
+    Name (94)       = HELPFILENAME(name,94);     \
     Name (96)       = HELPFILENAME(name,96);     \
     Name (97)       = HELPFILENAME(name,97);     \
     Name (99)       = HELPFILENAME(name,99)
@@ -620,7 +636,9 @@
     Name (88)       = HELPFILENAME(name,88x);     \
     Name (90)       = HELPFILENAME(name,90x);     \
     Name (91)       = HELPFILENAME(name,91x);     \
+    Name (92)       = HELPFILENAME(name,92x);     \
     Name (93)       = HELPFILENAME(name,93x);     \
+    Name (94)       = HELPFILENAME(name,94x);     \
     Name (96)       = HELPFILENAME(name,96x);     \
     Name (97)       = HELPFILENAME(name,97x);     \
     Name (99)       = HELPFILENAME(name,99x)
@@ -664,7 +682,9 @@
     Name (88)       = HELPFILENAME(name,endung);     \
     Name (90)       = HELPFILENAME(name,endung);     \
     Name (91)       = HELPFILENAME(name,endung);     \
+    Name (92)       = HELPFILENAME(name,endung);     \
     Name (93)       = HELPFILENAME(name,endung);     \
+    Name (94)       = HELPFILENAME(name,endung);     \
     Name (96)       = HELPFILENAME(name,endung);     \
     Name (97)       = HELPFILENAME(name,endung);     \
     Name (99)       = HELPFILENAME(name,endung)
@@ -708,7 +728,9 @@
     Dir (88)       = GID_DIR_HELP_88;     \
     Dir (90)       = GID_DIR_HELP_90;     \
     Dir (91)       = GID_DIR_HELP_91;     \
+    Dir (92)       = GID_DIR_HELP_92;     \
     Dir (93)       = GID_DIR_HELP_93;     \
+    Dir (94)       = GID_DIR_HELP_94;     \
     Dir (96)       = GID_DIR_HELP_96;     \
     Dir (97)       = GID_DIR_HELP_97;     \
     Dir (99)       = GID_DIR_HELP_99
@@ -752,7 +774,9 @@
     DosName (88)       = STRING(name);     \
     DosName (90)       = STRING(name);     \
     DosName (91)       = STRING(name);     \
+    DosName (92)       = STRING(name);     \
     DosName (93)       = STRING(name);     \
+    DosName (94)       = STRING(name);     \
     DosName (96)       = STRING(name);     \
     DosName (97)       = STRING(name);     \
     DosName (99)       = STRING(name)
@@ -796,7 +820,9 @@
     DosName (88)       = "chinese_traditional";    \
     DosName (90)       = "turkish";     \
     DosName (91)       = "hindi";     \
+    DosName (92)       = "tamil";     \
     DosName (93)       = "kannada";     \
+    DosName (94)       = "gujarati";     \
     DosName (96)       = "arabic";     \
     DosName (97)       = "hebrew";     \
     DosName (99)       = STRING(ISO_CODE)
@@ -840,7 +866,9 @@
     DosName (88)       = "zh-TW"; \
     DosName (90)       = "tr";  \
     DosName (91)       = "hi-IN";  \
+    DosName (92)       = "ta-IN";  \
     DosName (93)       = "kn-IN";  \
+    DosName (94)       = "gu-IN";  \
     DosName (96)       = "ar";  \
     DosName (97)       = "he";  \
     DosName (99)       = STRING(ISO_CODE)
@@ -884,7 +912,9 @@
     DosName (88)       = "zh-TW"; \
     DosName (90)       = "tr";  \
     DosName (91)       = "hi-IN";  \
+    DosName (92)       = "ta-IN";  \
     DosName (93)       = "kn-IN";  \
+    DosName (94)       = "gu-IN";  \
     DosName (96)       = "ar";  \
     DosName (97)       = "he";  \
     DosName (99)       = STRING(ISO_CODE)
@@ -928,7 +958,9 @@
         Section (88)              = STRING(ident##-88); \
         Section (90)              = STRING(ident##-90); \
         Section (91)              = STRING(ident##-91); \
+        Section (92)              = STRING(ident##-92); \
         Section (93)              = STRING(ident##-93); \
+        Section (94)              = STRING(ident##-94); \
         Section (96)              = STRING(ident##-96); \
         Section (97)              = STRING(ident##-97); \
         Section (99)              = STRING(ident##-99)
@@ -975,7 +1007,9 @@
         Key (88)              = STRING(ident##-88-C); \
         Key (90)              = STRING(ident##-90-C); \
         Key (91)              = STRING(ident##-91-C); \
+        Key (92)              = STRING(ident##-92-C); \
         Key (93)              = STRING(ident##-93-C); \
+        Key (94)              = STRING(ident##-94-C); \
         Key (96)              = STRING(ident##-96-C); \
         Key (97)              = STRING(ident##-97); \
         Key (99)              = STRING(ident##-99)
@@ -1037,8 +1071,10 @@
         Key (86)              = STRING(2052); \
         Key (88)              = STRING(1028); \
         Key (90)              = STRING(1055); \
-        Key (91)              = STRING(91); \
+        Key (91)              = STRING(1081); \
+        Key (92)              = STRING(1097); \
         Key (93)              = STRING(1099); \
+        Key (94)              = STRING(1095); \
         Key (96)              = STRING(1055); \
         Key (97)              = STRING(1037); \
         Key (99)              = STRING(ISO_CODE)
@@ -1082,7 +1118,9 @@
         Key (88)              = STRING(zh-TW); \
         Key (90)              = STRING(tr); \
         Key (91)              = STRING(hi-IN); \
+        Key (92)              = STRING(ta-IN); \
         Key (93)              = STRING(kn-IN); \
+        Key (94)              = STRING(gu-IN); \
         Key (96)              = STRING(ar); \
         Key (97)              = STRING(he); \
         Key (99)              = STRING(ISO_CODE)
@@ -1124,8 +1162,10 @@
         Value (86)              = STRING(2052); \
         Value (88)              = STRING(1028); \
         Value (90)              = STRING(1055); \
-        Value (91)              = STRING(91); \
+        Value (91)              = STRING(1081); \
+        Value (92)              = STRING(1097); \
         Value (93)              = STRING(1099); \
+        Value (94)              = STRING(1095); \
         Value (96)              = STRING(1025); \
         Value (97)              = STRING(1037); \
         Value (99)              = STRING(ISO_CODE)
@@ -1169,7 +1209,9 @@
         Value (88)              = STRING(zh-TW); \
         Value (90)              = STRING(tr); \
         Value (91)              = STRING(hi-IN); \
+        Value (92)              = STRING(ta-IN); \
         Value (93)              = STRING(kn-IN); \
+        Value (94)              = STRING(gu-IN); \
         Value (96)              = STRING(ar); \
         Value (97)              = STRING(he); \
         Value (99)              = STRING(ISO_CODE)
@@ -1213,7 +1255,9 @@
         Value (88)              = STRING(en-US); \
         Value (90)              = STRING(tr); \
         Value (91)              = STRING(hi-IN); \
+        Value (92)              = STRING(ta-IN); \
         Value (93)              = STRING(kn-IN); \
+        Value (94)              = STRING(gu-IN); \
         Value (96)              = STRING(ar); \
         Value (97)              = STRING(he); \
         Value (99)              = STRING(ISO_CODE)
@@ -1257,7 +1301,9 @@
         Value (88)              = STRING(zh-TW); \
         Value (90)              = STRING(tr); \
         Value (91)              = STRING(hi-IN); \
+        Value (92)              = STRING(ta-IN); \
         Value (93)              = STRING(kn-IN); \
+        Value (94)              = STRING(gu-IN); \
         Value (96)              = STRING(ar); \
         Value (97)              = STRING(he); \
         Value (99)              = STRING(ISO_CODE)
@@ -1301,7 +1347,9 @@
         Value (88)              = STRING(88); \
         Value (90)              = STRING(90); \
         Value (91)              = STRING(91); \
+        Value (92)              = STRING(92); \
         Value (93)              = STRING(93); \
+        Value (94)              = STRING(94); \
         Value (96)              = STRING(96); \
         Value (97)              = STRING(97); \
         Value (99)              = STRING(99)
--- ./scp/source/global/setupzip.scp.indic	2005-01-04 16:28:21.049273456 -0500
+++ ./scp/source/global/setupzip.scp	2005-01-04 16:28:44.401684974 -0500
@@ -656,6 +656,9 @@
         Name (88)       = "fontunxpsprint.zip";
         Name (90)       = "fontunxpsprint.zip";
         Name (91)       = "fontunxpsprint.zip";
+        Name (92)       = "fontunxpsprint.zip";
+        Name (93)       = "fontunxpsprint.zip";
+        Name (94)       = "fontunxpsprint.zip";
         Name (96)       = "fontunxpsprint.zip";
         Name (97)       = "fontunxpsprint.zip";
         Name (99)       = "fontunxpsprint.zip";
--- ./solenv/inc/postset.mk.indic	2005-01-04 16:13:24.512684427 -0500
+++ ./solenv/inc/postset.mk	2005-01-04 16:14:00.492257284 -0500
@@ -373,6 +373,32 @@
 RCLANGFLAGS_91+= -d HINDI
 iso_91=hi-IN
 
+.IF "$(RES_TAMIL)"!="" || "$(give_me_all_languages)"!=""
+alllangext+=92
+.ENDIF
+completelangext+=92
+tamil$(LANG_GUI)*=$(default$(LANG_GUI))
+lang_92=tamil
+longlang_92=tamil
+langext_92=92
+rsclang_92=-lgTAMIL $(UTF8)
+rescharset_92=$(tamil$(LANG_GUI))
+RCLANGFLAGS_92+= -d TAMIL
+iso_92=ta
+
+.IF "$(RES_GUJARATI)"!="" || "$(give_me_all_languages)"!=""
+alllangext+=94
+.ENDIF
+completelangext+=94
+gujarati$(LANG_GUI)*=$(default$(LANG_GUI))
+lang_94=gujarati
+longlang_94=gujarati
+langext_94=94
+rsclang_94=-lgGUJARATI $(UTF8)
+rescharset_94=$(gujarati$(LANG_GUI))
+RCLANGFLAGS_94+= -d GUJARATI
+iso_94=gu
+
 .IF "$(RES_PORTBR)"!="" || "$(give_me_all_languages)"!=""
 alllangext+=55
 .ENDIF 
--- ./solenv/inc/lang.mk.indic	2005-01-04 16:14:14.717111217 -0500
+++ ./solenv/inc/lang.mk	2005-01-04 16:14:29.787837292 -0500
@@ -93,6 +93,8 @@
 korean$(LANG_GUI)=-CHARSET_UTF8
 thai$(LANG_GUI)=-CHARSET_UTF8
 hindi$(LANG_GUI)=-CHARSET_UTF8
+tamil$(LANG_GUI)=-CHARSET_UTF8
+gujarati$(LANG_GUI)=-CHARSET_UTF8
 extern$(LANG_GUI)=-CHARSET_UTF8
 nsotho$(LANG_GUI)=-CHARSET_UTF8
 afrik$(LANG_GUI)=-CHARSET_UTF8
--- ./tools/source/generic/l2txtenc.cxx.indic	2005-01-04 16:09:30.628927549 -0500
+++ ./tools/source/generic/l2txtenc.cxx	2005-01-04 16:10:10.017996577 -0500
@@ -101,6 +101,8 @@
 #define THAI					66
 #define HINDI					91
 #define KANNADA					93
+#define TAMIL					92
+#define GUJARATI					94
 #define EXTERN					99
 
 rtl_TextEncoding Langcode2TextEncoding( USHORT nLang )
@@ -139,6 +141,8 @@
 		case CATALAN: return RTL_TEXTENCODING_MS_1252;
 		case THAI: return RTL_TEXTENCODING_UTF8;
 		case HINDI: return RTL_TEXTENCODING_UTF8;
+		case TAMIL: return RTL_TEXTENCODING_UTF8;
+		case GUJARATI: return RTL_TEXTENCODING_UTF8;
 		case EXTERN: return RTL_TEXTENCODING_UTF8;
 		case NORTHERNSOTHO: return RTL_TEXTENCODING_UTF8;
 		case AFRIKAANS: return RTL_TEXTENCODING_UTF8;
--- ./tools/source/rc/resmgr.cxx.indic	2005-01-04 16:10:42.368124128 -0500
+++ ./tools/source/rc/resmgr.cxx	2005-01-04 16:12:49.034034598 -0500
@@ -1268,6 +1268,10 @@
 				return "66";
             case LANGUAGE_HINDI:
                 return "91";
+            case LANGUAGE_TAMIL:
+                return "92";
+            case LANGUAGE_GUJARATI:
+                return "94";
 
             case LANGUAGE_ARABIC:
             case LANGUAGE_ARABIC_IRAQ:
@@ -1458,6 +1462,8 @@
         LANGUAGE_KOREAN_JOHAB,
         LANGUAGE_THAI,
         LANGUAGE_HINDI,
+        LANGUAGE_TAMIL,
+        LANGUAGE_GUJARATI,
         LANGUAGE_HEBREW,
 	LANGUAGE_NORTHERNSOTHO,
 	LANGUAGE_AFRIKAANS,
--- ./i18npool/source/breakiterator/breakiteratorImpl.cxx.indic	2005-01-04 16:29:19.069357296 -0500
+++ ./i18npool/source/breakiterator/breakiteratorImpl.cxx	2005-01-04 16:29:53.395081670 -0500
@@ -403,6 +403,10 @@
     { UnicodeScript_kHebrew, 			ScriptType::COMPLEX },		// 10,
     { UnicodeScript_kArabic, 			ScriptType::COMPLEX },		// 11,
     { UnicodeScript_kDevanagari,		ScriptType::COMPLEX },		// 14,
+    { UnicodeScript_kBengali,			ScriptType::COMPLEX },		// 15,
+    { UnicodeScript_kGurmukhi,		ScriptType::COMPLEX },		// 16,
+    { UnicodeScript_kGujarati,		ScriptType::COMPLEX },		// 17,
+    { UnicodeScript_kTamil,			ScriptType::COMPLEX },		// 19,
     { UnicodeScript_kThai,			ScriptType::COMPLEX },		// 24,
 
     { UnicodeScript_kTibetan,			ScriptType::LATIN },		// 26,
--- ./readlicense_oo/prj/d.lst.indic	2005-01-04 16:19:34.493809880 -0500
+++ ./readlicense_oo/prj/d.lst	2005-01-04 16:21:27.906653715 -0500
@@ -79,8 +79,12 @@
 mkdir: %_DEST%\bin%_EXT%\90\osl
 mkdir: %_DEST%\bin%_EXT%\91
 mkdir: %_DEST%\bin%_EXT%\91\osl
+mkdir: %_DEST%\bin%_EXT%\92
+mkdir: %_DEST%\bin%_EXT%\92\osl
 mkdir: %_DEST%\bin%_EXT%\93
 mkdir: %_DEST%\bin%_EXT%\93\osl
+mkdir: %_DEST%\bin%_EXT%\94
+mkdir: %_DEST%\bin%_EXT%\94\osl
 mkdir: %_DEST%\bin%_EXT%\96
 mkdir: %_DEST%\bin%_EXT%\96\osl
 mkdir: %_DEST%\bin%_EXT%\97
@@ -231,10 +235,22 @@
 ..\%__SRC%\misc\license\%GUI%\LICENSE %_DEST%\bin%_EXT%\90\osl\LICENSE90
 ..\%__SRC%\misc\license\WNT\license.txt %_DEST%\bin%_EXT%\90\osl\license90.txt
 ..\%__SRC%\misc\license\%GUI%\license.html %_DEST%\bin%_EXT%\90\osl\LICENSE90.html
+..\%__SRC%\misc\license\%GUI%\* %_DEST%\bin%_EXT%\91\osl\*
+..\%__SRC%\misc\license\%GUI%\LICENSE %_DEST%\bin%_EXT%\91\osl\LICENSE91
+..\%__SRC%\misc\license\WNT\license.txt %_DEST%\bin%_EXT%\91\osl\license91.txt
+..\%__SRC%\misc\license\%GUI%\license.html %_DEST%\bin%_EXT%\91\osl\LICENSE91.html
+..\%__SRC%\misc\license\%GUI%\* %_DEST%\bin%_EXT%\92\osl\*
+..\%__SRC%\misc\license\%GUI%\LICENSE %_DEST%\bin%_EXT%\92\osl\LICENSE92
+..\%__SRC%\misc\license\WNT\license.txt %_DEST%\bin%_EXT%\92\osl\license92.txt
+..\%__SRC%\misc\license\%GUI%\license.html %_DEST%\bin%_EXT%\92\osl\LICENSE92.html
 ..\%__SRC%\misc\license\%GUI%\* %_DEST%\bin%_EXT%\93\osl\*
 ..\%__SRC%\misc\license\%GUI%\LICENSE %_DEST%\bin%_EXT%\93\osl\LICENSE93
 ..\%__SRC%\misc\license\WNT\license.txt %_DEST%\bin%_EXT%\93\osl\license93.txt
 ..\%__SRC%\misc\license\%GUI%\license.html %_DEST%\bin%_EXT%\93\osl\LICENSE93.html
+..\%__SRC%\misc\license\%GUI%\* %_DEST%\bin%_EXT%\94\osl\*
+..\%__SRC%\misc\license\%GUI%\LICENSE %_DEST%\bin%_EXT%\94\osl\LICENSE94
+..\%__SRC%\misc\license\WNT\license.txt %_DEST%\bin%_EXT%\94\osl\license94.txt
+..\%__SRC%\misc\license\%GUI%\license.html %_DEST%\bin%_EXT%\94\osl\LICENSE94.html
 ..\%__SRC%\misc\license\%GUI%\* %_DEST%\bin%_EXT%\96\osl\*
 ..\%__SRC%\misc\license\%GUI%\LICENSE %_DEST%\bin%_EXT%\96\osl\LICENSE96
 ..\%__SRC%\misc\license\WNT\license.txt %_DEST%\bin%_EXT%\96\osl\license96.txt
@@ -359,9 +375,15 @@
 ..\%__SRC%\misc\91\* %_DEST%\bin%_EXT%\91\osl\*
 ..\%__SRC%\misc\91\readme91.txt %_DEST%\bin%_EXT%\91\osl\README91
 ..\%__SRC%\misc\91\readme91.html %_DEST%\bin%_EXT%\91\osl\README91.html
+..\%__SRC%\misc\92\* %_DEST%\bin%_EXT%\92\osl\*
+..\%__SRC%\misc\92\readme92.txt %_DEST%\bin%_EXT%\92\osl\README92
+..\%__SRC%\misc\92\readme92.html %_DEST%\bin%_EXT%\92\osl\README92.html
 ..\%__SRC%\misc\93\* %_DEST%\bin%_EXT%\93\osl\*
 ..\%__SRC%\misc\93\readme93.txt %_DEST%\bin%_EXT%\93\osl\README93
 ..\%__SRC%\misc\93\readme93.html %_DEST%\bin%_EXT%\93\osl\README93.html
+..\%__SRC%\misc\94\* %_DEST%\bin%_EXT%\94\osl\*
+..\%__SRC%\misc\94\readme94.txt %_DEST%\bin%_EXT%\94\osl\README94
+..\%__SRC%\misc\94\readme94.html %_DEST%\bin%_EXT%\94\osl\README94.html
 ..\%__SRC%\misc\96\* %_DEST%\bin%_EXT%\96\osl\*
 ..\%__SRC%\misc\96\readme96.txt %_DEST%\bin%_EXT%\96\osl\README96
 ..\%__SRC%\misc\96\readme96.html %_DEST%\bin%_EXT%\96\osl\README96.html
