--- officecfg/registry/data/org/openoffice/VCL.xcu.indic	2004-10-19 13:56:32.755677016 -0400
+++ officecfg/registry/data/org/openoffice/VCL.xcu	2004-10-19 14:38:10.975889816 -0400
@@ -383,7 +383,106 @@
         <value>Andale Sans UI;Arial Unicode MS;Lucida Sans Unicode;Tahoma;Luxi Sans;Interface User;WarpSans;Geneva;Tahoma;MS Sans Serif;Helv;Dialog;Albany AMT;Albany;Lucida;Helvetica;Charcoal;Chicago;Arial;Nimbus Sans L;Helmet;Interface System;Sans Serif</value>
       </prop>
     </node>
-
+    <node oor:name="hi-IN" oor:op="replace">
+     <prop oor:name="UI_SANS" oor:op="replace" oor:type="xs:string">
+      <value>Lohit Hindi;Tahoma;Lucidasans;Lucida Sans;Supplement;Andale Sans UI;Arial Unicode MS;Lucida Sans Unicode;clearlyU;Interface User;WarpSans;Geneva;MS Sans Serif;Helv;Dialog;Albany AMT;Albany;Lucida;Helvetica;Charcoal;Chicago;Arial;Nimbus Sans L;Helmet;Interface System;Sans Serif</value>
+     </prop>
+     <prop oor:name="CTL_DISPLAY" oor:op="replace" oor:type="xs:string">
+      <value>Lohit Hindi;Tahoma;Lucidasans;Lucida Sans;Arial Unicode MS</value>
+     </prop>
+     <prop oor:name="CTL_HEADING" oor:op="replace" oor:type="xs:string">
+      <value>Lohit Hindi;Tahoma;Lucidasans;Lucida Sans;Arial Unicode MS</value>
+     </prop>
+     <prop oor:name="CTL_PRESENTATION" oor:op="replace" oor:type="xs:string">
+      <value>Lohit Hindi;Tahoma;Lucidasans;Lucida Sans;Arial Unicode MS</value>
+     </prop>
+     <prop oor:name="CTL_SPREADSHEET" oor:op="replace" oor:type="xs:string">
+      <value>Lohit Hindi;Tahoma;Lucidasans;Lucida Sans;Arial Unicode MS</value>
+     </prop>
+     <prop oor:name="CTL_TEXT" oor:op="replace" oor:type="xs:string">
+      <value>Lohit Hindi;Tahoma;Lucidasans;Lucida Sans;Arial Unicode MS</value>
+     </prop>
+    </node>
+    <node oor:name="bn-IN" oor:op="replace">
+     <prop oor:name="UI_SANS" oor:op="replace" oor:type="xs:string">
+      <value>Lohit Bengali;Tahoma;Lucidasans;Lucida Sans;Supplement;Andale Sans UI;Arial Unicode MS;Lucida Sans Unicode;clearlyU;Interface User;WarpSans;Geneva;MS Sans Serif;Helv;Dialog;Albany AMT;Albany;Lucida;Helvetica;Charcoal;Chicago;Arial;Nimbus Sans L;Helmet;Interface System;Sans Serif</value>
+     </prop>
+     <prop oor:name="CTL_DISPLAY" oor:op="replace" oor:type="xs:string">
+      <value>Lohit Bengali;Tahoma;Lucidasans;Lucida Sans;Arial Unicode MS</value>
+     </prop>
+     <prop oor:name="CTL_HEADING" oor:op="replace" oor:type="xs:string">
+      <value>Lohit Bengali;Tahoma;Lucidasans;Lucida Sans;Arial Unicode MS</value>
+     </prop>
+     <prop oor:name="CTL_PRESENTATION" oor:op="replace" oor:type="xs:string">
+      <value>Lohit Bengali;Tahoma;Lucidasans;Lucida Sans;Arial Unicode MS</value>
+     </prop>
+     <prop oor:name="CTL_SPREADSHEET" oor:op="replace" oor:type="xs:string">
+      <value>Lohit Bengali;Tahoma;Lucidasans;Lucida Sans;Arial Unicode MS</value>
+     </prop>
+     <prop oor:name="CTL_TEXT" oor:op="replace" oor:type="xs:string">
+      <value>Lohit Bengali;Tahoma;Lucidasans;Lucida Sans;Arial Unicode MS</value>
+     </prop>
+    </node>
+    <node oor:name="gu-IN" oor:op="replace">
+     <prop oor:name="UI_SANS" oor:op="replace" oor:type="xs:string">
+      <value>Lohit Gujarati;Tahoma;Lucidasans;Lucida Sans;Supplement;Andale Sans UI;Arial Unicode MS;Lucida Sans Unicode;clearlyU;Interface User;WarpSans;Geneva;MS Sans Serif;Helv;Dialog;Albany AMT;Albany;Lucida;Helvetica;Charcoal;Chicago;Arial;Nimbus Sans L;Helmet;Interface System;Sans Serif</value>
+     </prop>
+     <prop oor:name="CTL_DISPLAY" oor:op="replace" oor:type="xs:string">
+      <value>Lohit Gujarati;Tahoma;Lucidasans;Lucida Sans;Arial Unicode MS</value>
+     </prop>
+     <prop oor:name="CTL_HEADING" oor:op="replace" oor:type="xs:string">
+      <value>Lohit Gujarati;Tahoma;Lucidasans;Lucida Sans;Arial Unicode MS</value>
+     </prop>
+     <prop oor:name="CTL_PRESENTATION" oor:op="replace" oor:type="xs:string">
+      <value>Lohit Gujarati;Tahoma;Lucidasans;Lucida Sans;Arial Unicode MS</value>
+     </prop>
+     <prop oor:name="CTL_SPREADSHEET" oor:op="replace" oor:type="xs:string">
+      <value>Lohit Gujarati;Tahoma;Lucidasans;Lucida Sans;Arial Unicode MS</value>
+     </prop>
+     <prop oor:name="CTL_TEXT" oor:op="replace" oor:type="xs:string">
+      <value>Lohit Gujarati;Tahoma;Lucidasans;Lucida Sans;Arial Unicode MS</value>
+     </prop>
+    </node>
+    <node oor:name="pa-IN" oor:op="replace">
+     <prop oor:name="UI_SANS" oor:op="replace" oor:type="xs:string">
+      <value>Lohit Punjabi;Tahoma;Lucidasans;Lucida Sans;Supplement;Andale Sans UI;Arial Unicode MS;Lucida Sans Unicode;clearlyU;Interface User;WarpSans;Geneva;MS Sans Serif;Helv;Dialog;Albany AMT;Albany;Lucida;Helvetica;Charcoal;Chicago;Arial;Nimbus Sans L;Helmet;Interface System;Sans Serif</value>
+     </prop>
+     <prop oor:name="CTL_DISPLAY" oor:op="replace" oor:type="xs:string">
+      <value>Lohit Punjabi;Tahoma;Lucidasans;Lucida Sans;Arial Unicode MS</value>
+     </prop>
+     <prop oor:name="CTL_HEADING" oor:op="replace" oor:type="xs:string">
+      <value>Lohit Punjabi;Tahoma;Lucidasans;Lucida Sans;Arial Unicode MS</value>
+     </prop>
+     <prop oor:name="CTL_PRESENTATION" oor:op="replace" oor:type="xs:string">
+      <value>Lohit Punjabi;Tahoma;Lucidasans;Lucida Sans;Arial Unicode MS</value>
+     </prop>
+     <prop oor:name="CTL_SPREADSHEET" oor:op="replace" oor:type="xs:string">
+      <value>Lohit Punjabi;Tahoma;Lucidasans;Lucida Sans;Arial Unicode MS</value>
+     </prop>
+     <prop oor:name="CTL_TEXT" oor:op="replace" oor:type="xs:string">
+      <value>Lohit Punjabi;Tahoma;Lucidasans;Lucida Sans;Arial Unicode MS</value>
+     </prop>
+    </node>
+    <node oor:name="ta-IN" oor:op="replace">
+     <prop oor:name="UI_SANS" oor:op="replace" oor:type="xs:string">
+      <value>Lohit Tamil;Tahoma;Lucidasans;Lucida Sans;Supplement;Andale Sans UI;Arial Unicode MS;Lucida Sans Unicode;clearlyU;Interface User;WarpSans;Geneva;MS Sans Serif;Helv;Dialog;Albany AMT;Albany;Lucida;Helvetica;Charcoal;Chicago;Arial;Nimbus Sans L;Helmet;Interface System;Sans Serif</value>
+     </prop>
+     <prop oor:name="CTL_DISPLAY" oor:op="replace" oor:type="xs:string">
+      <value>Lohit Tamil;Tahoma;Lucidasans;Lucida Sans;Arial Unicode MS</value>
+     </prop>
+     <prop oor:name="CTL_HEADING" oor:op="replace" oor:type="xs:string">
+      <value>Lohit Tamil;Tahoma;Lucidasans;Lucida Sans;Arial Unicode MS</value>
+     </prop>
+     <prop oor:name="CTL_PRESENTATION" oor:op="replace" oor:type="xs:string">
+      <value>Lohit Tamil;Tahoma;Lucidasans;Lucida Sans;Arial Unicode MS</value>
+     </prop>
+     <prop oor:name="CTL_SPREADSHEET" oor:op="replace" oor:type="xs:string">
+      <value>Lohit Tamil;Tahoma;Lucidasans;Lucida Sans;Arial Unicode MS</value>
+     </prop>
+     <prop oor:name="CTL_TEXT" oor:op="replace" oor:type="xs:string">
+      <value>Lohit Tamil;Tahoma;Lucidasans;Lucida Sans;Arial Unicode MS</value>
+     </prop>
+    </node>
     <node oor:name="ar" oor:op="replace">
      <prop oor:name="UI_SANS" oor:op="replace" oor:type="xs:string">
         <value>Tahoma;Traditional Arabic;Simplified Arabic;Lucidasans;Lucida Sans;Supplement;Andale Sans UI;Arial Unicode MS;Lucida Sans Unicode;clearlyU;Interface User;WarpSans;Geneva;MS Sans Serif;Helv;Dialog;Albany AMT;Albany;Lucida;Helvetica;Charcoal;Chicago;Arial;Nimbus Sans L;Helmet;Interface System;Sans Serif</value>
