Index: i18npool/source/localedata/localedata.cxx
===================================================================
RCS file: /cvs/l10n/i18npool/source/localedata/localedata.cxx,v
retrieving revision 1.22.18.1
diff -u -p -r1.22.18.1 localedata.cxx
--- i18npool/source/localedata/localedata.cxx	9 Jan 2004 17:39:21 -0000	1.22.18.1
+++ i18npool/source/localedata/localedata.cxx	8 Dec 2004 15:24:03 -0000
@@ -188,6 +188,7 @@ static const struct {
 	{ "ta_IN",  lcl_DATA_OTHERS, "ta" },
 	{ "te_IN",  lcl_DATA_OTHERS, "te" },
 	{ "gu_IN",  lcl_DATA_OTHERS, "gu" },
+	{ "bn_IN",  lcl_DATA_OTHERS, "bn" },
 	{ "mr_IN",  lcl_DATA_OTHERS, "mr" },
 	{ "pa_IN",  lcl_DATA_OTHERS, "pa" },
 	{ "th_TH",  lcl_DATA_OTHERS, "th" },
Index: i18npool/source/localedata/data/localedata_others.map
===================================================================
RCS file: /cvs/l10n/i18npool/source/localedata/data/localedata_others.map,v
retrieving revision 1.2
diff -u -p -r1.2 localedata_others.map
--- i18npool/source/localedata/data/localedata_others.map	15 Apr 2003 14:27:24 -0000	1.2
+++ i18npool/source/localedata/data/localedata_others.map	8 Dec 2004 15:24:03 -0000
@@ -16,6 +16,7 @@ getAllCalendars_ja_JP;
 getAllCalendars_kn_IN;
 getAllCalendars_ko_KR;
 getAllCalendars_mr_IN;
+getAllCalendars_bn_IN;
 getAllCalendars_pa_IN;
 getAllCalendars_ta_IN;
 getAllCalendars_te_IN;
@@ -39,6 +40,7 @@ getAllCurrencies_ja_JP;
 getAllCurrencies_kn_IN;
 getAllCurrencies_ko_KR;
 getAllCurrencies_mr_IN;
+getAllCurrencies_bn_IN;
 getAllCurrencies_pa_IN;
 getAllCurrencies_ta_IN;
 getAllCurrencies_te_IN;
@@ -62,6 +64,7 @@ getAllFormats_ja_JP;
 getAllFormats_kn_IN;
 getAllFormats_ko_KR;
 getAllFormats_mr_IN;
+getAllFormats_bn_IN;
 getAllFormats_pa_IN;
 getAllFormats_ta_IN;
 getAllFormats_te_IN;
@@ -84,6 +87,7 @@ getCollationOptions_id_ID;
 getCollationOptions_ja_JP;
 getCollationOptions_kn_IN;
 getCollationOptions_ko_KR;
+getCollationOptions_bn_IN;
 getCollationOptions_mr_IN;
 getCollationOptions_pa_IN;
 getCollationOptions_ta_IN;
@@ -107,6 +111,7 @@ getCollatorImplementation_id_ID;
 getCollatorImplementation_ja_JP;
 getCollatorImplementation_kn_IN;
 getCollatorImplementation_ko_KR;
+getCollatorImplementation_bn_IN;
 getCollatorImplementation_mr_IN;
 getCollatorImplementation_pa_IN;
 getCollatorImplementation_ta_IN;
@@ -130,6 +135,7 @@ getContinuousNumberingLevels_id_ID;
 getContinuousNumberingLevels_ja_JP;
 getContinuousNumberingLevels_kn_IN;
 getContinuousNumberingLevels_ko_KR;
+getContinuousNumberingLevels_bn_IN;
 getContinuousNumberingLevels_mr_IN;
 getContinuousNumberingLevels_pa_IN;
 getContinuousNumberingLevels_ta_IN;
@@ -153,6 +159,7 @@ getForbiddenCharacters_id_ID;
 getForbiddenCharacters_ja_JP;
 getForbiddenCharacters_kn_IN;
 getForbiddenCharacters_ko_KR;
+getForbiddenCharacters_bn_IN;
 getForbiddenCharacters_mr_IN;
 getForbiddenCharacters_pa_IN;
 getForbiddenCharacters_ta_IN;
@@ -176,6 +183,7 @@ getLCInfo_id_ID;
 getLCInfo_ja_JP;
 getLCInfo_kn_IN;
 getLCInfo_ko_KR;
+getLCInfo_bn_IN;
 getLCInfo_mr_IN;
 getLCInfo_pa_IN;
 getLCInfo_ta_IN;
@@ -199,6 +207,7 @@ getLocaleItem_id_ID;
 getLocaleItem_ja_JP;
 getLocaleItem_kn_IN;
 getLocaleItem_ko_KR;
+getLocaleItem_bn_IN;
 getLocaleItem_mr_IN;
 getLocaleItem_pa_IN;
 getLocaleItem_ta_IN;
@@ -222,6 +231,7 @@ getOutlineNumberingLevels_id_ID;
 getOutlineNumberingLevels_ja_JP;
 getOutlineNumberingLevels_kn_IN;
 getOutlineNumberingLevels_ko_KR;
+getOutlineNumberingLevels_bn_IN;
 getOutlineNumberingLevels_mr_IN;
 getOutlineNumberingLevels_pa_IN;
 getOutlineNumberingLevels_ta_IN;
@@ -245,6 +255,7 @@ getReservedWords_id_ID;
 getReservedWords_ja_JP;
 getReservedWords_kn_IN;
 getReservedWords_ko_KR;
+getReservedWords_bn_IN;
 getReservedWords_mr_IN;
 getReservedWords_pa_IN;
 getReservedWords_ta_IN;
@@ -268,6 +279,7 @@ getSearchOptions_id_ID;
 getSearchOptions_ja_JP;
 getSearchOptions_kn_IN;
 getSearchOptions_ko_KR;
+getSearchOptions_bn_IN;
 getSearchOptions_mr_IN;
 getSearchOptions_pa_IN;
 getSearchOptions_ta_IN;
@@ -291,6 +303,7 @@ getTransliterations_id_ID;
 getTransliterations_ja_JP;
 getTransliterations_kn_IN;
 getTransliterations_ko_KR;
+getTransliterations_bn_IN;
 getTransliterations_mr_IN;
 getTransliterations_pa_IN;
 getTransliterations_ta_IN;
Index: i18npool/source/localedata/data/makefile.mk
===================================================================
RCS file: /cvs/l10n/i18npool/source/localedata/data/makefile.mk,v
retrieving revision 1.21.16.1
diff -u -p -r1.21.16.1 makefile.mk
--- i18npool/source/localedata/data/makefile.mk	9 Jan 2004 17:39:41 -0000	1.21.16.1
+++ i18npool/source/localedata/data/makefile.mk	8 Dec 2004 15:24:03 -0000
@@ -156,6 +156,7 @@ MY_MISC_CXXFILES = \
     $(MISC)$/localedata_lt_LT.cxx  \
     $(MISC)$/localedata_lv_LV.cxx  \
     $(MISC)$/localedata_mr_IN.cxx  \
+    $(MISC)$/localedata_bn_IN.cxx  \
     $(MISC)$/localedata_nb_NO.cxx  \
     $(MISC)$/localedata_nl_BE.cxx  \
     $(MISC)$/localedata_nl_NL.cxx  \
@@ -319,6 +320,7 @@ SHL4OBJS= \
 	$(SLO)$/localedata_kn_IN.obj \
 	$(SLO)$/localedata_ko_KR.obj \
 	$(SLO)$/localedata_mr_IN.obj \
+	$(SLO)$/localedata_bn_IN.obj \
 	$(SLO)$/localedata_pa_IN.obj \
 	$(SLO)$/localedata_ta_IN.obj \
 	$(SLO)$/localedata_te_IN.obj \
--- /dev/null	2004-12-07 08:05:45.878783080 +0000
+++ i18npool/source/localedata/data/bn_IN.xml	2002-09-04 20:19:06.000000000 +0100
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE Locale SYSTEM 'locale.dtd'>
+<Locale version="1.2">
+<LC_INFO>
+<Language>
+<LangID>bn</LangID>
+<DefaultName>Bengali</DefaultName>
+</Language>
+<Country>
+<CountryID>IN</CountryID>
+<DefaultName>India</DefaultName>
+</Country>
+</LC_INFO>
+<LC_CTYPE ref="hi_IN"/>
+<LC_FORMAT ref="hi_IN" replaceTo="[$रु.-44e]"/>
+<LC_COLLATION ref="en_US"/>
+<LC_SEARCH ref="en_US"/>
+<LC_CALENDAR ref="hi_IN"/>
+<LC_CURRENCY ref="hi_IN"/>
+<LC_TRANSLITERATION ref="en_US"/>
+<LC_MISC ref="hi_IN"/>
+<LC_NumberingLevel ref="hi_IN"/>
+<LC_OutLineNumberingLevel ref="en_US"/>
+</Locale>
Index: svtools/source/config/languageoptions.cxx
===================================================================
RCS file: /cvs/util/svtools/source/config/languageoptions.cxx,v
retrieving revision 1.5
diff -u -p -r1.5 languageoptions.cxx
--- svtools/source/config/languageoptions.cxx	22 May 2003 09:03:44 -0000	1.5
+++ svtools/source/config/languageoptions.cxx	8 Dec 2004 15:18:56 -0000
@@ -270,6 +270,7 @@ sal_uInt16 SvtLanguageOptions::GetScript
 		case LANGUAGE_HEBREW:
 		case LANGUAGE_MARATHI:
 		case LANGUAGE_PUNJABI:
+	    case LANGUAGE_BENGALI:
 	    case LANGUAGE_GUJARATI:
 	    case LANGUAGE_HINDI:
 	    case LANGUAGE_KANNADA:
Index: tools/source/intntl/isolang.cxx
===================================================================
RCS file: /cvs/util/tools/source/intntl/isolang.cxx,v
retrieving revision 1.12.50.2
diff -u -p -r1.12.50.2 isolang.cxx
--- tools/source/intntl/isolang.cxx	3 May 2004 17:05:38 -0000	1.12.50.2
+++ tools/source/intntl/isolang.cxx	8 Dec 2004 15:15:40 -0000
@@ -229,7 +229,7 @@ static IsoLangEntry const aImplIsoLangEn
     { LANGUAGE_SERBIAN,                     "sr", ""   },   // SERBIAN is only LID, MS-LCID not defined (was dupe of CROATIAN)
     { LANGUAGE_ARMENIAN,                    "hy", "AM" },
     { LANGUAGE_AZERI,                       "az", ""   },
-    { LANGUAGE_BENGALI,                     "bn", "BD" },
+    { LANGUAGE_BENGALI,                     "bn", "IN" },
     { LANGUAGE_KAZAK,                       "kk", "KZ" },
     { LANGUAGE_URDU,                        "ur", "IN" },
     { LANGUAGE_HINDI,                       "hi", "IN" },
