Index: unx/source/gdi/salogl.cxx
===================================================================
RCS file: /cvs/gsl/vcl/unx/source/gdi/salogl.cxx,v
retrieving revision 1.7
diff -u -r1.7 salogl.cxx
--- vcl/unx/source/gdi/salogl.cxx	15 Apr 2003 16:10:05 -0000	1.7
+++ vcl/unx/source/gdi/salogl.cxx	5 Jan 2005 15:34:37 -0000
@@ -90,7 +90,7 @@
 #ifdef MACOSX
 #define OGL_LIBNAME "libGL.dylib"
 #else
-#define OGL_LIBNAME "libGL.so"
+#define OGL_LIBNAME "libGL.so.1"
 #endif
 
 // ----------
