--- offmgr/source/offapp/intro/ooo.src.mdksplash	2004-04-27 18:19:18.000000000 +0200
+++ offmgr/source/offapp/intro/ooo.src	2004-09-01 19:00:08.793566108 +0200
@@ -77,18 +77,18 @@
 #include "intro.hrc"
 
 #ifdef BUILD_SPECIAL
-#define OOO_VENDOR "Sun Microsystems Inc."
+#define OOO_VENDOR "Mandrakesoft S.A."
 #define OOO_LICENSE "LGPL"
-#define OOO_INTRO "openintro_sun.bmp"
+#define OOO_INTRO "openintro_mandrakelinux.bmp"
 #else
-//#define OOO_VENDOR "my company"
+#define OOO_VENDOR "Mandrakesoft S.A."
 #define OOO_LICENSE "LGPL"
-#define OOO_INTRO "openintro.bmp"
+#define OOO_INTRO "openintro_mandrakelinux.bmp"
 #endif
 
 Bitmap RID_DEFAULT_ABOUT_BMP_LOGO
 {
-	File = "openabout.bmp" ;
+	File = "openabout_mandrakelinux.bmp" ;
 };
 
 Bitmap RID_DEFAULTINTRO
--- desktop/source/splash/splash.cxx.orig	2004-09-01 19:17:07.307088887 +0200
+++ desktop/source/splash/splash.cxx	2004-09-01 19:16:02.745484965 +0200
@@ -254,6 +254,7 @@
 		if (length < 0) length = 0;
 		const Color cBlue(COL_BLUE);
 		const Color cGray(COL_LIGHTGRAY);
+		const Color cWhite(COL_WHITE);
 
 		// border
 		SetFillColor();
@@ -262,7 +263,7 @@
 			_tly+_barheight));	
 
 		// progress bar
-		SetFillColor(cBlue);
+		SetFillColor(cWhite);
 		SetLineColor();
 		DrawRect(Rectangle(_tlx+_barspace, _tly+_barspace, 
 			_tlx+_barspace+length, _tly+_barheight-_barspace));		
