Index: Common.xcu
===================================================================
RCS file: /cvs/util/officecfg/registry/data/org/openoffice/Office/Common.xcu,v
retrieving revision 1.13.34.2
diff -u -r1.13.34.2 Common.xcu
--- officecfg/registry/data/org/openoffice/Office/Common.xcu	6 Jul 2004 10:50:06 -0000	1.13.34.2
+++ officecfg/registry/data/org/openoffice/Office/Common.xcu	21 Dec 2004 15:40:02 -0000
@@ -437,4 +437,9 @@
       </node>
     </node>
   </node>
+  <node oor:name="Misc">
+    <prop oor:name="UseSystemFileDialog" oor:type="xs:boolean">
+      <value>false</value>
+    </prop>
+  </node> 
 </oor:component-data>
