Index: settings/settings.mk
===================================================================
RCS file: /cvs/api/odk/settings/settings.mk,v
retrieving revision 1.6.22.3
diff -u -u -r1.6.22.3 settings.mk
--- odk/settings/settings.mk	30 Mar 2004 14:31:46 -0000	1.6.22.3
+++ odk/settings/settings.mk	27 Nov 2004 21:09:09 -0000
@@ -212,6 +212,21 @@
 JAVA_PROC_TYPE=ppc
 endif
 
+ifeq "$(PROCTYPE)" "arm"
+PACKAGE_LIB_DIR=linux_arm.plt
+JAVA_PROC_TYPE=arm
+endif
+
+ifeq "$(PROCTYPE)" "sparc"
+PACKAGE_LIB_DIR=linux_sparc.plt
+JAVA_PROC_TYPE=sparc
+endif
+
+ifeq "$(PROCTYOE)" "s390"
+PPACKAGE_LIB_DIR=linux_s390.plt
+JAVA_PROC_TYPE=s390
+endif
+
 OS=LINUX
 PS=/
 CC=gcc
@@ -265,7 +280,7 @@
 CC_FLAGS=-c -O -fpic -fno-rtti
 endif
 
-ifeq "$(PROCTYPE)" "ppc"
+ifneq "$(PROCTYPE)" "i386" 
 CC_FLAGS+=-fPIC
 endif
 
@@ -282,7 +297,7 @@
 
 LIBRARY_LINK_FLAGS=-shared '-Wl,-rpath,$$ORIGIN'
 
-ifeq "$(PROCTYPE)" "ppc"
+ifneq "$(PROCTYPE)" "i386"
 LIBRARY_LINK_FLAGS+=-fPIC
 endif
 
