--- scp/source/office/profile.scp.report	2004-11-05 00:34:03.817000040 +0000
+++ scp/source/office/profile.scp	2004-11-05 00:41:47.927318603 +0000
@@ -1171,42 +1171,6 @@
         Styles           = (CFG_STRING, CREATE);
 End
 
-ConfigurationItem GID_CONFIGURATIONITEM_COMMON_MENUS_WIZARD_M9_URL
-    #ifdef WEB_PRODUCT
-        ModuleID         = GID_MODULE_PRG_WRT;
-    #else
-        ModuleID         = GID_MODULE_PRG_WRT_BIN;
-    #endif
-        Path             = "org.openoffice.Office.Common/Menus/Wizard/*['m9']";
-        Key              = "URL";
-        Value            = "service:com.sun.star.wizards.report.CallReportWizard?start";
-        Styles           = (CFG_STRING, CREATE);
-End
-
-ConfigurationItem GID_CONFIGURATIONITEM_COMMON_MENUS_WIZARD_M9_TARGETNAME
-    #ifdef WEB_PRODUCT
-        ModuleID         = GID_MODULE_PRG_WRT;
-    #else
-        ModuleID         = GID_MODULE_PRG_WRT_BIN;
-    #endif
-        Path             = "org.openoffice.Office.Common/Menus/Wizard/*['m9']";
-        Key              = "TargetName";
-        Value            = "_self";
-        Styles           = (CFG_STRING, CREATE);
-End
-
-ConfigurationItem GID_CONFIGURATIONITEM_COMMON_MENUS_WIZARD_M9_IMAGEIDENTIFIER
-    #ifdef WEB_PRODUCT
-        ModuleID         = GID_MODULE_PRG_WRT;
-    #else
-        ModuleID         = GID_MODULE_PRG_WRT_BIN;
-    #endif
-        Path             = "org.openoffice.Office.Common/Menus/Wizard/*['m9']";
-        Key              = "ImageIdentifier";
-        Value            = "private:image/3216";
-        Styles           = (CFG_STRING, CREATE);
-End
-
 ConfigurationItem GID_CONFIGURATIONITEM_COMMON_MENUS_WIZARD_M10_URL
         ModuleID         = GID_MODULE_SO_ROOT;
         Path             = "org.openoffice.Office.Common/Menus/Wizard/*['m10']";
@@ -3464,43 +3428,6 @@
 	Styles		 = (CFG_STRING);
 End
 
-#ifdef FAT_PRODUCT
-
-StarRegistryItem GID_STARREGISTRYITEM_SERVICES_REPORTWIZARD
-    ModuleID    = GID_MODULE_SO_ROOT;
-    RegistryID  = GID_STARREGISTRY_SERVICES_RDB;
-    Key         = "/SERVICES/com.sun.star.wizards.report.CallReportWizard";
-    SeqValue    = "com.sun.star.wizards.report.CallReportWizard$ReportWizardImplementation";
-    Styles      = (NETWORK);
-End
-
-StarRegistryItem GID_STARREGISTRYITEM_IMPLEMENTATIONS_SERVICES_REPORTWIZARD
-    ModuleID    = GID_MODULE_SO_ROOT;
-    RegistryID  = GID_STARREGISTRY_SERVICES_RDB;
-    Key         = "/IMPLEMENTATIONS/com.sun.star.wizards.report.CallReportWizard$ReportWizardImplementation/UNO/SERVICES/com.sun.star.wizards.report.CallReportWizard";
-    Value       = "";
-    Styles      = (NETWORK);
-End
-
-StarRegistryItem GID_STARREGISTRYITEM_IMPLEMENTATIONS_ACTIVATOR_REPORTWIZARD
-    ModuleID    = GID_MODULE_SO_ROOT;
-    RegistryID  = GID_STARREGISTRY_SERVICES_RDB;
-    Key         = "/IMPLEMENTATIONS/com.sun.star.wizards.report.CallReportWizard$ReportWizardImplementation/UNO/ACTIVATOR";
-    Value       = "com.sun.star.loader.Java2";
-    Styles      = (NETWORK);
-End
-
-StarRegistryItem GID_STARREGISTRYITEM_IMPLEMENTATIONS_LOCATION_REPORTWIZARD
-    ModuleID    = GID_MODULE_SO_ROOT;
-    RegistryID  = GID_STARREGISTRY_SERVICES_RDB;
-    Key         = "/IMPLEMENTATIONS/com.sun.star.wizards.report.CallReportWizard$ReportWizardImplementation/UNO/LOCATION";
-    Value       = "vnd.sun.star.expand:$SYSBINDIR/classes/report.jar";
-    Styles      = (NETWORK);
-End
-
-#endif
-
-
 // HACK: write accessibility bridge service info
 
 #if ! ( defined (WEB_PRODUCT) || defined (SRV_PRODUCT) )
