--- padmin/source/helper.cxx.lang-fix	2004-11-08 17:00:46.335593232 -0500
+++ padmin/source/helper.cxx	2004-11-08 17:02:21.028197768 -0500
@@ -133,27 +133,10 @@
 	static ResMgr* pPaResMgr = NULL;
 	if( ! pPaResMgr )
 	{
-		LanguageType nLang = LANGUAGE_SYSTEM;
-       
-        utl::OConfigurationNode aNode =
-            utl::OConfigurationTreeRoot::tryCreateWithServiceFactory(
-                    vcl::unohelper::GetMultiServiceFactory(),
-                    OUString::createFromAscii( "org.openoffice.Setup/L10N" ) );
-        if ( aNode.isValid() )
-        {
-            rtl::OUString aLoc;
-            Any aValue = aNode.getNodeValue( OUString::createFromAscii( "ooLocale" ) );
-            if( aValue >>= aLoc )
-            {
-                LanguageType nTmpLang = ConvertIsoStringToLanguage( aLoc );
-                if( nTmpLang != LANGUAGE_DONTKNOW )
-                    nLang = nTmpLang;
-            }
-        }
-		pPaResMgr = ResMgr::SearchCreateResMgr( "spa" MAKE_NUMSTR(SUPD), nLang );
+		LanguageType nLang;
 		AllSettings aSettings = Application::GetSettings();
-        aSettings.SetUILanguage( nLang );
-        Application::SetSettings( aSettings );
+		nLang = aSettings.GetUILanguage();
+		pPaResMgr = ResMgr::SearchCreateResMgr( "spa" MAKE_NUMSTR(SUPD), nLang );
 	}
 	return ResId( nId, pPaResMgr );
 }
