Index: unxlngppc.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/unxlngppc.mk,v
retrieving revision 1.11.50.1
diff -u -u -r1.11.50.1 unxlngppc.mk
--- solenv/inc/unxlngppc.mk	9 Jan 2004 18:59:14 -0000	1.11.50.1
+++ solenv/inc/unxlngppc.mk	16 Aug 2005 21:28:40 -0000
@@ -142,7 +142,7 @@
 DYNAMIC = -Wl,-Bdynamic 
 
 # name of linker
-LINK*=gcc 
+LINK*=$(CC) 
 # default linker flags
 LINKFLAGS= 
 
Index: unxlngppc4.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/unxlngppc4.mk,v
retrieving revision 1.5.26.1
diff -u -u -r1.5.26.1 unxlngppc4.mk
--- solenv/inc/unxlngppc4.mk	9 Jan 2004 18:59:15 -0000	1.5.26.1
+++ solenv/inc/unxlngppc4.mk	16 Aug 2005 21:28:40 -0000
@@ -152,7 +152,7 @@
 DYNAMIC		= -Wl,-Bdynamic
 
 # name of linker
-LINK=gcc
+LINK=$(CC)
 # default linker flags
 LINKFLAGS=
 
