--- sc/source/ui/docshell/docsh.cxx	2004-12-16 13:27:05.842137808 +0530
+++ sc/source/ui/docshell/docsh.cxx	2004-12-16 13:29:38.516927712 +0530
@@ -185,6 +185,7 @@
 static const sal_Char __FAR_DATA pFilterXML[]		= "StarOffice XML (Calc)";
 static const sal_Char __FAR_DATA pFilterAscii[]		= "Text - txt - csv (StarCalc)";
 static const sal_Char __FAR_DATA pFilterLotus[]		= "Lotus";
+static const sal_Char __FAR_DATA pFilterQPro[]		= "QPro";
 static const sal_Char __FAR_DATA pFilterExcel4[]	= "MS Excel 4.0";
 static const sal_Char __FAR_DATA pFilterEx4Temp[]	= "MS Excel 4.0 Vorlage/Template";
 static const sal_Char __FAR_DATA pFilterExcel5[]	= "MS Excel 5.0/95";
@@ -977,6 +978,22 @@
             // #93255# update of row height done inside of Excel filter to speed up chart import
 //            bSetRowHeights = TRUE;      //  #75357# optimal row heights must be updated
 		}
+		else if (aFltName.EqualsAscii(pFilterQPro))
+		{
+			ScColumn::bDoubleAlloc = TRUE;
+			FltError eError = ScImportQuattroPro( rMedium, &aDocument );
+			ScColumn::bDoubleAlloc = FALSE;
+			if (eError != eERR_OK)
+			{
+			    if (!GetError())
+				    SetError( eError );
+
+				if( ( eError & ERRCODE_WARNING_MASK ) == ERRCODE_WARNING_MASK )
+					bRet = TRUE;
+			}
+			else
+				bRet = TRUE;
+		}
 		else if (aFltName.EqualsAscii(pFilterAscii))
 		{
 			SfxItemSet*	 pSet = rMedium.GetItemSet();
--- sc/source/ui/app/sclib.cxx	2004-12-16 13:26:32.229247744 +0530
+++ sc/source/ui/app/sclib.cxx	2004-12-16 13:29:38.527926040 +0530
@@ -103,6 +103,7 @@
 static const sal_Char __FAR_DATA pFilterXML[]		= "StarOffice XML (Calc)";
 static const sal_Char __FAR_DATA pFilterAscii[]		= "Text - txt - csv (StarCalc)";
 static const sal_Char __FAR_DATA pFilterLotus[]		= "Lotus";
+static const sal_Char __FAR_DATA pFilterQPro[]		= "QPro";
 static const sal_Char __FAR_DATA pFilterExcel4[]	= "MS Excel 4.0";
 static const sal_Char __FAR_DATA pFilterEx4Temp[]	= "MS Excel 4.0 Vorlage/Template";
 static const sal_Char __FAR_DATA pFilterExcel5[]	= "MS Excel 5.0/95";
@@ -557,6 +558,13 @@
 			  0x0000, 0x0000,
 			  M_ENDE };
 
+		const UINT16 pQPro[] =
+				{ 0x0000, 0x0000, 0x0002, 0x0000,
+				  M_ALT(4), 0x0001, 0x0002, // WB1, WB2
+				  0x0006, 0x0007,           // QPro 6/7 (?)
+				  0x0010,
+				  M_ENDE };
+
 		const UINT16 pDIF1[] =			// DIF mit CR-LF
 			{
 			'T', 'A', 'B', 'L', 'E',
@@ -604,6 +612,7 @@
 			pDIF1,
 			pDIF2,
 			pSylk,
+			pQPro,
 			pLotus97,
 			pLotus2
 			};
@@ -620,6 +629,7 @@
 			pFilterDif,
 			pFilterDif,
 			pFilterSylk,
+			pFilterQPro,
 			pFilterLotus,
 			pFilterLotus
 			};
--- officecfg/registry/data/org/openoffice/Office/TypeDetection.xcu	2004-12-16 13:26:44.706350936 +0530
+++ officecfg/registry/data/org/openoffice/Office/TypeDetection.xcu	2004-12-16 13:29:38.560921024 +0530
@@ -261,6 +261,14 @@
         <value>1,,,,wk1;wks;123,0,</value>
       </prop>
     </node>
+    <node oor:name="calc_QPro" oor:op="replace">
+      <prop oor:name="UIName">
+        <value xml:lang="en-US">Quattro Pro</value>
+      </prop>
+      <prop oor:name="Data">
+        <value>1,,,,wb1;wb2,0,</value>
+      </prop>
+    </node>
     <node oor:name="calc_MS_Excel_40" oor:op="replace">
       <prop oor:name="UIName">
         <value xml:lang="en-US">Microsoft Excel 4.0</value>
@@ -4810,6 +4818,17 @@
         <value>0,calc_Lotus,com.sun.star.sheet.SpreadsheetDocument,,268435649,,0,,com.sun.star.comp.Calc.FilterOptionsDialog</value>
       </prop>
     </node>
+    <node oor:name="QPro" oor:op="replace">
+      <prop oor:name="UIName">
+        <value xml:lang="en-US">Quattro Pro</value>
+      </prop>
+      <prop oor:name="Data">
+        <value>0,calc_QPro,com.sun.star.sheet.SpreadsheetDocument,,268435649,,0,,</value>
+      </prop>
+      <prop oor:name="Installed" oor:type="xs:boolean">
+        <value>true</value>
+      </prop>
+    </node>
     <node oor:name="MS Excel 4.0" oor:op="replace">
       <prop oor:name="Installed">
         <value>false</value>
--- sc/inc/filter.hxx	2004-12-16 13:27:16.969446200 +0530
+++ sc/inc/filter.hxx	2004-12-16 13:29:38.562920720 +0530
@@ -122,6 +122,8 @@
 
 FltError ScImportLotus123( SfxMedium&, ScDocument*, CharSet eSrc = RTL_TEXTENCODING_DONTKNOW );
 
+FltError ScImportQuattroPro( SfxMedium&, ScDocument* );
+
 FltError ScImportExcel( SvStream&, ScDocument* );
 
 FltError ScImportExcel( SfxMedium&, ScDocument* );
