--- pyuno/source/module/makefile.mk
+++ pyuno/source/module/makefile.mk
@@ -139,7 +146,7 @@ $(DLLDEST)$/%.py: %.py
 .IF "$(GUI)" == "UNX"
 $(PYUNO_MODULE) : $(SLO)$/pyuno_dlopenwrapper.obj
 .IF "$(OS)" == "LINUX"
-	ld -shared -ldl -o $@ $(SLO)$/pyuno_dlopenwrapper.o
+	ld -rpath \''$$ORIGIN'\' -shared -ldl -o $@ $(SLO)$/pyuno_dlopenwrapper.o
 .ELIF "$(OS)" == "SOLARIS"
 	ld -G -ldl -o $@ $(SLO)$/pyuno_dlopenwrapper.o
 .ELIF "$(OS)" == "FREEBSD"
