--- desktop/scripts/soffice.sh
+++ desktop/scripts/soffice.sh
@@ -137,36 +137,15 @@ case $sd_platform in
     export LD_LIBRARYN32_PATH
     ;;
 
+  Linux*)
+    #-rpath $ORIGIN is sufficient
+    ;;
   *)
     LD_LIBRARY_PATH="$sd_progsub":"$sd_prog":$LD_LIBRARY_PATH
     export LD_LIBRARY_PATH
     ;;
 esac
 
-# extend the ld_library_path for java: javaldx checks the sofficerc for us
-if [ -x "$sd_prog/javaldx" ] ; then
-    java_ld_library_path=`"$sd_prog/javaldx"`
-    if [ "$java_ld_library_path" != "" ] ; then
-        case $sd_platform in
-            AIX)
-                LIBPATH=${java_ld_library_path}:${LIBPATH}
-                ;;
-            Darwin)
-                DYLD_LIBRARY_PATH=${java_ld_library_path}:${DYLD_LIBRARY_PATH}
-                ;;
-            HP-UX)
-                SHLIB_PATH=${java_ld_library_path}:${SHLIB_PATH}
-                ;;
-            IRIX*)
-                LD_LIBRARYN32_PATH=${java_ld_library_path}:${LD_LIBRARYN32_PATH}
-                ;;
-            *)
-                LD_LIBRARY_PATH=${java_ld_library_path}:${LD_LIBRARY_PATH}
-                ;;
-        esac
-    fi
-fi
-
 # set java environment variables
 THREADS_TYPE=native_threads
 
@@ -232,9 +211,12 @@ sd_pagein_args="${sd_pagein_args:+${sd_p
 "${sd_prog}"/pagein -L"${sd_prog}" ${sd_pagein_args}
 
 # set path so that other apps can be started from soffice just by name
-PATH="$sd_prog":$PATH
+if [ $PATH ] ; then
+    PATH="$sd_prog":$PATH
+else
+    PATH="$sd_prog"
+fi
 export PATH
 
 # execute soffice binary
 exec "$sd_prog/$sd_binary" "$@"
-
