--- python/makefile.mk
+++ python/makefile.mk
@@ -200,9 +200,13 @@ $(PYTHONCORESHL) : makefile.mk $(PACKAGE
 .IF "$(OS)" == "MACOSX"
 	cp $(MISC)$/build$/$(TARFILE_NAME)$/$(DLLPRE)python$(PYMAJOR).$(PYMINOR)$(DLLPOST) $(OUT)$/lib
 .ELSE
+.IF "$(OS)" == "LINUX"
+	ld -shared -o $@ -rpath \''$$ORIGIN'\' --whole-archive $(MISC)$/build$/$(TARFILE_NAME)$/libpython$(PYMAJOR).$(PYMINOR).a --no-whole-archive -soname libpython.so.$(PYMAJOR)  -lm -ldl -lutil -lc -lpthread
+.ELSE
 	echo "$(OS)"
 	ld -shared -o $@ --whole-archive $(MISC)$/build$/$(TARFILE_NAME)$/libpython$(PYMAJOR).$(PYMINOR).a --no-whole-archive -soname libpython.so.$(PYMAJOR)  -lm -ldl -lutil -lc -lpthread
 .ENDIF # MACOSX
+.ENDIF # LINUX
 .ENDIF # IRIX
 .ENDIF # NETBSD
 .ENDIF # FREEBSD
