--- solenv/inc/unxlngppc.mk	9 Jan 2004 18:59:14 -0000	1.11.50.1
+++ solenv/inc/unxlngppc.mk	3 Nov 2004 17:58:20 -0000
@@ -144,14 +144,15 @@ DYNAMIC = -Wl,-Bdynamic 
 # name of linker
 LINK*=gcc 
 # default linker flags
-LINKFLAGS= 
+LINKFLAGS=-Wl,-rpath,\''$$ORIGIN'\'
+
 
 # linker flags for linking applications
 LINKFLAGSAPPGUI= -Wl,-export-dynamic 
 LINKFLAGSAPPCUI= -Wl,-export-dynamic 
 # linker flags for linking shared libraries
-LINKFLAGSSHLGUI= -shared 
-LINKFLAGSSHLCUI= -shared 
+LINKFLAGSSHLGUI= -shared
+LINKFLAGSSHLCUI= -shared
 
 LINKFLAGSTACK= 
 LINKFLAGSPROF= 
--- solenv/inc/unxlngppc4.mk	9 Jan 2004 18:59:15 -0000	1.5.26.1
+++ solenv/inc/unxlngppc4.mk	3 Nov 2004 17:58:20 -0000
@@ -154,7 +154,7 @@ DYNAMIC		= -Wl,-Bdynamic
 # name of linker
 LINK=gcc
 # default linker flags
-LINKFLAGS=
+LINKFLAGS=-Wl,-rpath,\''$$ORIGIN'\'
 
 # linker flags for linking applications
 LINKFLAGSAPPGUI= -Wl,-export-dynamic -Wl,--noinhibit-exec
--- solenv/inc/unxlngs.mk	30 Mar 2004 13:17:38 -0000	1.2.50.2
+++ solenv/inc/unxlngs.mk	3 Nov 2004 17:58:20 -0000
@@ -148,7 +148,7 @@ DYNAMIC		= -Wl,-Bdynamic
 LINK*=$(CC)
 
 # default linker flags
-# LINKFLAGSRUNPATH*=-Wl,-rpath\''$$ORIGIN'\'
+LINKFLAGSRUNPATH*=-Wl,-rpath\''$$ORIGIN'\'
 LINKFLAGS=-z combreloc $(LINKFLAGSRUNPATH)
 
 # linker flags for linking applications
