Index: unxlngs3904.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/unxlngs3904.mk,v
retrieving revision 1.4.26.1
diff -u -u -r1.4.26.1 unxlngs3904.mk
--- solenv/inc/unxlngs3904.mk	9 Jan 2004 18:59:15 -0000	1.4.26.1
+++ solenv/inc/unxlngs3904.mk	16 Aug 2005 21:32:03 -0000
@@ -153,7 +153,7 @@
 DYNAMIC		= -Wl,-Bdynamic
 
 # name of linker
-LINK*=g++-3.2
+LINK*=$(CC)
 # default linker flags
 LINKFLAGS=
 
