--- sax/source/expatwrap/xml2utf.cxx.old	2006-06-29 14:55:13.811764614 +0200
+++ sax/source/expatwrap/xml2utf.cxx	2006-06-29 15:54:51.806276945 +0200
@@ -196,7 +196,7 @@ void XMLFile2UTFConverter::removeEncodin
 			}
 			else
 			{
-				int nStop  = str.indexOf( "\"" , nStart +1);
+				nStop  = str.indexOf( "\"" , nStart +1);
 			}
 
 			if( nStart >= 0 && nStop >= 0 && nStart+1 < nStop )
@@ -292,7 +292,7 @@ sal_Bool XMLFile2UTFConverter::scanForEn
 			}
 			else
 			{
-				int nStop  = str.indexOf( "\"" , nStart +1);
+				nStop  = str.indexOf( "\"" , nStart +1);
 			}
 			if( nStart >= 0 && nStop >= 0 && nStart+1 < nStop )
 			{

