--- sc/source/ui/inc/viewdata.hxx	2004-10-19 11:08:50.000000000 +0530
+++ sc/source/ui/inc/viewdata.hxx	2004-10-19 11:31:25.222005216 +0530
@@ -54,7 +54,7 @@
  *
  *  All Rights Reserved.
  *
- *  Contributor(s): _______________________________________
+ *  Contributor(s): Kohei Yoshida__________________________
  *
  *
  ************************************************************************/
@@ -218,6 +218,8 @@
 	BYTE				nFillMode;					// Modus
 	BOOL				bPagebreak;					// Seitenumbruch-Vorschaumodus
 
+	BOOL				bSelCtrlMouseClick;
+
 	DECL_LINK (EmptyEditHdl, EditStatus*);
 	DECL_LINK (EditEngineHdl, EditStatus*);
 
@@ -307,6 +309,8 @@
 
 	void			SetZoom( const Fraction& rNewX, const Fraction& rNewY );
 
+	void			EnableSelCtrlMouseClick( BOOL bTmp ) { bSelCtrlMouseClick = bTmp; }
+
 	const Fraction&	GetZoomX() const		{ return bPagebreak ? aPageZoomX : aZoomX; }
 	const Fraction&	GetZoomY() const		{ return bPagebreak ? aPageZoomY : aZoomY; }
 
@@ -464,6 +468,8 @@
 	const Size&		GetScenButSize() const				{ return aScenButSize; }
 	void			SetScenButSize(const Size& rNew)	{ aScenButSize = rNew; }
 
+	BOOL			IsSelCtrlMouseClickEnabled() { return bSelCtrlMouseClick; }
+
 	static inline long ToPixel( USHORT nTwips, double nFactor );
 };
 
--- sc/source/ui/view/select.cxx	2004-10-19 11:08:52.000000000 +0530
+++ sc/source/ui/view/select.cxx	2004-10-19 11:31:25.252000656 +0530
@@ -54,7 +54,7 @@
  *
  *  All Rights Reserved.
  *
- *  Contributor(s): _______________________________________
+ *  Contributor(s): Kohei Yoshida__________________________
  *
  *
  ************************************************************************/
@@ -598,6 +598,28 @@
 				aAnchorPos.Put( nPosX, nPosY, nTab );
 				bStarted = TRUE;
 			}
+			// #i3875# *Hack* When a new cell is Ctrl-clicked with no pre-selected cells, 
+			// it highlights that new cell as well as the old cell where the cursor is 
+			// positioned prior to the click.  A selection mode via Shift-F8 should also 
+			// follow the same behavior.
+			else if ( pViewData->IsSelCtrlMouseClickEnabled() )
+			{
+				USHORT nOldX = pViewData->GetCurX();
+				USHORT nOldY = pViewData->GetCurY();
+				
+				pView->InitBlockMode( nOldX, nOldY, nTab, TRUE );
+				pView->MarkCursor( (USHORT) nOldX, (USHORT) nOldY, nTab );
+				
+				if ( nOldX != nPosX || nOldY != nPosY )
+				{
+					pView->DoneBlockMode( TRUE );
+					pView->InitBlockMode( nPosX, nPosY, nTab, TRUE );
+					pView->MarkCursor( (USHORT) nPosX, (USHORT) nPosY, nTab );
+					aAnchorPos.Put( nPosX, nPosY, nTab );
+				}
+				
+				bStarted = TRUE;
+			}
 		}
 
 		pView->SetCursor( (USHORT) nPosX, (USHORT) nPosY );
--- sc/source/ui/view/tabview3.cxx	2004-10-19 11:08:52.000000000 +0530
+++ sc/source/ui/view/tabview3.cxx	2004-10-19 11:31:25.273997312 +0530
@@ -54,7 +54,7 @@
  *
  *  All Rights Reserved.
  *
- *  Contributor(s): _______________________________________
+ *  Contributor(s): Kohei Yoshida__________________________
  *
  *
  ************************************************************************/
@@ -917,12 +917,20 @@
 BOOL ScTabView::SelMouseButtonDown( const MouseEvent& rMEvt )
 {
 	BOOL bRet = FALSE;
+	
+	// #i3875# *Hack*
+	BOOL bMod1Locked = aViewData.GetViewShell()->GetLockedModifiers() & KEY_MOD1 ? TRUE : FALSE;
+	aViewData.EnableSelCtrlMouseClick( rMEvt.IsMod1() || bMod1Locked ); 
+	
 	if ( pSelEngine )
 	{
 		bMoveIsShift = rMEvt.IsShift();
 		bRet = pSelEngine->SelMouseButtonDown( rMEvt );
 		bMoveIsShift = FALSE;
 	}
+	
+	aViewData.EnableSelCtrlMouseClick( FALSE ); // #i3875# *Hack*
+	
 	return bRet;
 }
 
--- sc/source/ui/view/viewdata.cxx	2004-10-19 11:08:52.000000000 +0530
+++ sc/source/ui/view/viewdata.cxx	2004-10-19 11:31:25.300993208 +0530
@@ -54,7 +54,7 @@
  *
  *  All Rights Reserved.
  *
- *  Contributor(s): _______________________________________
+ *  Contributor(s): Kohei Yoshida__________________________
  *
  *
  ************************************************************************/
@@ -327,7 +327,8 @@
 		bDelMarkValid( FALSE ),
 		bActive		( TRUE ),					//! wie initialisieren?
 		bPagebreak	( FALSE ),
-		pSpellingView ( NULL )
+		pSpellingView ( NULL ),
+		bSelCtrlMouseClick( FALSE )
 {
 	SetGridMode		( TRUE );
 	SetSyntaxMode	( FALSE );
@@ -389,7 +390,8 @@
 		bDelMarkValid( FALSE ),
 		bActive		( TRUE ),								//! wie initialisieren?
 		bPagebreak	( rViewData.bPagebreak ),
-		pSpellingView ( rViewData.pSpellingView )
+		pSpellingView ( rViewData.pSpellingView ),
+		bSelCtrlMouseClick( rViewData.bSelCtrlMouseClick )
 {
 	SetGridMode		( rViewData.IsGridMode() );
 	SetSyntaxMode	( rViewData.IsSyntaxMode() );
