diff -u -r sc.prev/source/ui/dbgui/pvlaydlg.cxx sc/source/ui/dbgui/pvlaydlg.cxx
--- sc.prev/source/ui/dbgui/pvlaydlg.cxx	2005-02-09 17:02:34.078477544 +0530
+++ sc/source/ui/dbgui/pvlaydlg.cxx	2005-02-09 17:00:40.489745656 +0530
@@ -288,9 +288,21 @@
 
 	ScRange inRange;
 	String inString;
-        inRange = xDlgDPObject->GetSheetDesc()->aSourceRange;
-        inRange.Format( inString, STD_FORMAT, pDoc);
-        aEdInPos.SetText(inString);
+        if (xDlgDPObject->GetSheetDesc())
+        {
+                aEdInPos.Enable();
+                aRbInPos.Enable();
+                inRange = xDlgDPObject->GetSheetDesc()->aSourceRange;
+                inRange.Format( inString, STD_FORMAT, pDoc);
+                aEdInPos.SetText(inString);
+        }
+        else
+        {
+                /* Data is not reachable, so could be a remote database */
+                aEdInPos.Disable();
+                aRbInPos.Disable();
+ 
+        }
 	
     aSlider.SetPageSize( PAGE_SIZE );
     aSlider.SetVisibleSize( PAGE_SIZE );
