diff -u -r sc.prev/source/ui/dbgui/pvlaydlg.cxx sc/source/ui/dbgui/pvlaydlg.cxx
--- sc.prev/source/ui/dbgui/pvlaydlg.cxx	2005-02-09 09:29:56.097118264 +0530
+++ sc/source/ui/dbgui/pvlaydlg.cxx	2005-02-09 09:31:47.654159016 +0530
@@ -1380,7 +1380,7 @@
 {
 	if ( aBtnMore.GetState() )
 	{
-		bRefInputMode = TRUE;
+		bRefInputMode = FALSE;
 		//@BugID 54702 Enablen/Disablen nur noch in Basisklasse
 		//SFX_APPWINDOW->Enable();
 		aEdOutPos.Enable();
diff -u -r sc.prev/source/ui/inc/anyrefdg.hxx sc/source/ui/inc/anyrefdg.hxx
--- sc.prev/source/ui/inc/anyrefdg.hxx	2005-02-09 09:29:55.808162192 +0530
+++ sc/source/ui/inc/anyrefdg.hxx	2005-02-09 09:35:37.740180624 +0530
@@ -128,6 +128,7 @@
     Image               aImgRefDoneHC;  /// Stop reference input (high contrast)
     ScAnyRefDlg*        pAnyRefDlg;     // parent dialog
     ScRefEdit*          pRefEdit;       // zugeordnetes Edit-Control
+    BOOL		bState;
 
 protected:
     virtual void        Click();
diff -u -r sc.prev/source/ui/inc/pvlaydlg.hxx sc/source/ui/inc/pvlaydlg.hxx
--- sc.prev/source/ui/inc/pvlaydlg.hxx	2005-02-09 09:29:55.841157176 +0530
+++ sc/source/ui/inc/pvlaydlg.hxx	2005-02-09 09:36:07.703625488 +0530
@@ -133,6 +133,7 @@
 
     virtual void            SetReference( const ScRange& rRef, ScDocument* pDoc );
     virtual BOOL            IsRefInputMode() const { return bRefInputMode; }
+	    void            SetRefInputMode (BOOL bMode) { bRefInputMode = bMode; }
     virtual void            SetActive();
     virtual BOOL            Close();
 
diff -u -r sc.prev/source/ui/miscdlgs/anyrefdg.cxx sc/source/ui/miscdlgs/anyrefdg.cxx
--- sc.prev/source/ui/miscdlgs/anyrefdg.cxx	2005-02-09 09:29:56.236097136 +0530
+++ sc/source/ui/miscdlgs/anyrefdg.cxx	2005-02-09 09:59:40.354869856 +0530
@@ -95,6 +95,7 @@
 #include "cell.hxx"
 #include "global.hxx"
 #include "inputopt.hxx"
+#include "pvlaydlg.hxx"
 
 //----------------------------------------------------------------------------
 
@@ -286,6 +287,7 @@
     pAnyRefDlg( pParent ),
     pRefEdit( pEdit )
 {
+    bState = FALSE;
     SetStartImage();
 }
 
@@ -323,6 +325,9 @@
 
 void ScRefButton::Click()
 {
+    bState = !bState;
+    ScDPLayoutDlg *pDlg = (ScDPLayoutDlg *) this->pAnyRefDlg;
+    pDlg->SetRefInputMode (bState);
     if( pAnyRefDlg )
         pAnyRefDlg->ToggleCollapsed( pRefEdit, this );
 }
