diff -u -r ../OOO_1_1_3_bak/sc/idl/pivotsh.idl sc/idl/pivotsh.idl
--- ../OOO_1_1_3_bak/sc/idl/pivotsh.idl	2004-10-20 10:04:47.000000000 +0530
+++ sc/idl/pivotsh.idl	2004-11-17 10:31:29.000000000 +0530
@@ -64,6 +64,7 @@
 
 	SID_PIVOT_RECALC	[ ExecMethod = Execute; StateMethod = GetState; Export = FALSE; ]
 	SID_PIVOT_KILL		[ ExecMethod = Execute; StateMethod = GetState; Export = FALSE; ]
+	SID_PIVOT_AFMT		[ ExecMethod = Execute; StateMethod = GetState; Export = FALSE; ]
 }
 
 shell ScPivotShell
diff -u -r ../OOO_1_1_3_bak/sc/idl/scalc.idl sc/idl/scalc.idl
--- ../OOO_1_1_3_bak/sc/idl/scalc.idl	2004-10-20 10:04:47.000000000 +0530
+++ sc/idl/scalc.idl	2004-11-17 11:26:06.000000000 +0530
@@ -326,6 +326,63 @@
 ]
 
  //--------------------------------------------------------------------------
+SfxVoidItem AutoFormatPivotTable SID_PIVOT_AFMT
+( )
+[
+// ### ACHTUNG: Neuer ConfigName in Resource? "Datenpilottabelle l<94>schen" : "Datenpilottabelle löschen"
+        ConfigName = "Datenpilottabelle l<94>schen" ;
+// ### ACHTUNG: Neuer Helptext in Resource? "Datenpilottabelle l<94>schen" : "Datenpilottabelle löschen"
+        HelpText = "Datenpilottabelle l<94>schen" ;
+                                                                                                                             
+        /* flags: */
+        AutoUpdate = FALSE ,
+        Cachable = Cachable ,
+        FastCall = FALSE ,
+        HasCoreId = FALSE ,
+        HasDialog = FALSE ,
+        ReadOnlyDoc = TRUE ,
+        Toggle = FALSE ,
+        Synchron ;
+                                                                                                                             
+        /* plugin: */
+        Container = FALSE ,
+        PlugComm = FALSE ,
+                                                                                                                             
+        /* api: */
+        Export = TRUE ,
+        IsCollection = FALSE ,
+        RecordAbsolute = FALSE ,
+        RecordPerSet ;
+                                                                                                                             
+        /* config: */
+        AccelConfig = FALSE ,
+        MenuConfig = TRUE ,
+        StatusBarConfig = FALSE ,
+        ToolBoxConfig = FALSE ,
+        GroupId = GID_DATA ;
+        //idlpp ConfigName [ english_us ] = "Delete DataPilot Table" ;
+        //idlpp HelpText [ english_us ] = "Delete DataPilot table" ;
+        //idlpp ConfigName [ portuguese ] = "Eliminar tabela do Piloto de dados" ;
+        //idlpp HelpText [ portuguese ] = "Elimina a tabela do Piloto de dados" ;
+        //idlpp ConfigName [ dutch ] = "Datapiloottabel wissen" ;
+        //idlpp HelpText [ dutch ] = "Datapiloottabel wissen" ;
+        //idlpp ConfigName [ french ] = "Supprimer tableau analytique du DataPilote" ;
+        //idlpp HelpText [ french ] = "Supprime le tableau analytique du DataPilote" ;
+        //idlpp ConfigName [ spanish ] = "Eliminar tabla del Piloto de datos" ;
+        //idlpp HelpText [ spanish ] = "Eliminar la tabla del piloto de datos" ;
+        //idlpp ConfigName [ italian ] = "Elimina tabella DataPilot" ;
+        //idlpp HelpText [ italian ] = "Datenpilottabelle löschen" ;
+        //idlpp ConfigName [ danish ] = "DeletePivotTable" ;
+        //idlpp HelpText [ danish ] = "Datenpilottabelle löschen" ;
+        //idlpp ConfigName [ swedish ] = "Ta bort datapilottabell" ;
+        //idlpp HelpText [ swedish ] = "Ta bort datapilottabell" ;
+        //idlpp ConfigName [ portuguese_brazilian ] = "DeletePivotTable" ;
+        //idlpp HelpText [ portuguese_brazilian ] = "Datenpilottabelle löschen" ;
+]
+                                                                                                                             
+
+
+ //--------------------------------------------------------------------------
 SfxVoidItem DefineDBName SID_DEFINE_DBNAME
 ( )
 [
diff -u -r ../OOO_1_1_3_bak/sc/inc/dpobject.hxx sc/inc/dpobject.hxx
--- ../OOO_1_1_3_bak/sc/inc/dpobject.hxx	2004-10-20 10:04:48.000000000 +0530
+++ sc/inc/dpobject.hxx	2004-11-17 16:07:35.669577984 +0530
@@ -131,7 +131,7 @@
 	ScDPOutput*				pOutput;
 	BOOL					bSettingsChanged;
 	BOOL					bAlive;			// FALSE if only used to hold settings
-
+	USHORT					nAutoFormatIndex;
 
 	void				CreateObjects();
 	void				CreateOutput();
@@ -157,6 +157,9 @@
 	void				SetOutRange(const ScRange& rRange);
 	const ScRange&		GetOutRange() const		{ return aOutRange; }
 
+	void 			SetAutoFormatIndex (const USHORT nIndex);
+	const USHORT		GetAutoFormatIndex ()	{ if (this == NULL) return 65535; else return nAutoFormatIndex; }
+
 	void				SetSheetDesc(const ScSheetSourceDesc& rDesc);
 	void				SetImportDesc(const ScImportSourceDesc& rDesc);
 	void				SetServiceData(const ScDPServiceDesc& rDesc);
diff -u -r ../OOO_1_1_3_bak/sc/inc/sc.hrc sc/inc/sc.hrc
--- ../OOO_1_1_3_bak/sc/inc/sc.hrc	2004-10-20 10:04:48.000000000 +0530
+++ sc/inc/sc.hrc	2004-11-17 14:08:31.000000000 +0530
@@ -1575,6 +1575,7 @@
 
 #define RID_SCDLG_LINKAREA				(SC_DIALOGS_START + 132)
 
+#define SID_PIVOT_AFMT					(SC_DIALOGS_START + 133)
 #define SC_DIALOGS_END          (SC_DIALOGS_START + 150)
 
 #ifndef STD_MASKCOLOR
diff -u -r ../OOO_1_1_3_bak/sc/sdi/pivotsh.sdi sc/sdi/pivotsh.sdi
--- ../OOO_1_1_3_bak/sc/sdi/pivotsh.sdi	2004-10-20 10:04:54.000000000 +0530
+++ sc/sdi/pivotsh.sdi	2004-11-17 14:55:13.000000000 +0530
@@ -5,6 +5,7 @@
 	SID_PIVOT_RECALC	[ ExecMethod = Execute; StateMethod = GetState; Export = FALSE; ]
 	SID_PIVOT_KILL		[ ExecMethod = Execute; StateMethod = GetState; Export = FALSE; ]
         SID_DP_FILTER           [ ExecMethod = Execute; StateMethod = GetState; Export = FALSE; ]
+	SID_PIVOT_AFMT		[ ExecMethod = Execute; StateMethod = GetState; Export = FALSE; ]
 }
 
 shell ScPivotShell
diff -u -r ../OOO_1_1_3_bak/sc/sdi/scalc.sdi sc/sdi/scalc.sdi
--- ../OOO_1_1_3_bak/sc/sdi/scalc.sdi	2004-10-20 10:04:54.000000000 +0530
+++ sc/sdi/scalc.sdi	2004-11-17 14:59:48.000000000 +0530
@@ -1846,6 +1846,31 @@
 	GroupId = GID_DATA;
 ]
 
+SfxVoidItem AutoFormatPivotTable SID_PIVOT_AFMT
+()
+[
+        /* flags: */
+        AutoUpdate = FALSE,
+        Cachable = Cachable,
+        FastCall = FALSE,
+        HasCoreId = FALSE,
+        HasDialog = FALSE,
+        ReadOnlyDoc = TRUE,
+        Toggle = FALSE,
+        Container = FALSE,
+        RecordAbsolute = FALSE,
+        RecordPerSet;
+        Synchron;
+                                                                                                                             
+        /* config: */
+        AccelConfig = FALSE,
+        MenuConfig = TRUE,
+        StatusBarConfig = FALSE,
+        ToolBoxConfig = FALSE,
+        GroupId = GID_DATA;
+]
+
+
 //--------------------------------------------------------------------------
 SfxVoidItem DeletePrintArea SID_DELETE_PRINTAREA
 ()
diff -u -r ../OOO_1_1_3_bak/sc/source/core/data/dpobject.cxx sc/source/core/data/dpobject.cxx
--- ../OOO_1_1_3_bak/sc/source/core/data/dpobject.cxx	2004-10-20 10:05:04.000000000 +0530
+++ sc/source/core/data/dpobject.cxx	2004-10-28 14:59:46.000000000 +0530
@@ -171,6 +171,7 @@
 	pImpDesc( NULL ),
 	pServDesc( NULL ),
 	pOutput( NULL ),
+	nAutoFormatIndex(65535),
 	bSettingsChanged( FALSE )
 {
 }
@@ -182,6 +183,7 @@
 	aTableName( r.aTableName ),
 	aTableTag( r.aTableTag ),
 	aOutRange( r.aOutRange ),
+	nAutoFormatIndex(r.nAutoFormatIndex),
 	pSheetDesc( NULL ),
 	pImpDesc( NULL ),
 	pServDesc( NULL ),
@@ -226,6 +228,11 @@
 	InvalidateData();		// re-init source from SaveData
 }
 
+void ScDPObject::SetAutoFormatIndex(const USHORT nIndex)
+{
+	nAutoFormatIndex	= nIndex;
+}
+
 void ScDPObject::SetOutRange(const ScRange& rRange)
 {
 	aOutRange = rRange;
diff -u -r ../OOO_1_1_3_bak/sc/source/ui/docshell/dbdocfun.cxx sc/source/ui/docshell/dbdocfun.cxx
--- ../OOO_1_1_3_bak/sc/source/ui/docshell/dbdocfun.cxx	2004-10-20 10:05:03.000000000 +0530
+++ sc/source/ui/docshell/dbdocfun.cxx	2004-10-29 09:10:47.000000000 +0530
@@ -1357,6 +1357,37 @@
 	delete pNewUndoDoc;
 	delete pUndoDPObj;
 
+	if (const_cast<ScDPObject *>(pNewObj)->GetAutoFormatIndex() != 65535)
+	{
+		ScViewData* pViewData = rDocShell.GetViewData();
+  		ScDocument* pDoc = pViewData->GetDocument();
+		rDocShell.GetViewData()->GetViewShell()->GetDBData( TRUE, SC_DB_OLD );
+		const ScMarkData& rMark1 = pViewData->GetMarkData();
+		USHORT nStartCol;
+		USHORT nStartRow;
+		USHORT nStartTab;
+		USHORT nEndCol;
+		USHORT nEndRow;
+		USHORT nEndTab;
+
+		rDocShell.GetViewData()->GetSimpleArea( nStartCol,nStartRow,nStartTab,
+											  nEndCol,nEndRow,nEndTab );
+		ScRange Outrange = pNewObj->GetOutRange();
+		if (( (Outrange.aStart.Col() <= nStartCol) && (nStartCol <= Outrange.aEnd.Col() )&& (nStartTab == Outrange.aStart.Tab()))
+			&& ( (Outrange.aStart.Row() <= nStartRow) && (nStartRow <= Outrange.aEnd.Row() )) )
+			{
+		
+			if ( !rMark1.IsMarked() && !rMark1.IsMultiMarked() )
+				rDocShell.GetViewData()->GetViewShell()->MarkDataArea( TRUE );
+			//FIXME: Autoformat even when clicked from a non-dp cell.
+			pViewData->MoveNextRow();		
+			rDocShell.GetViewData()->GetViewShell()->AutoFormat( const_cast<ScDPObject *>(pNewObj)->GetAutoFormatIndex());
+			rDocShell.GetViewData()->GetViewShell()->AutoFormatPivotTable( const_cast<ScDPObject *>(pNewObj), const_cast<ScDPObject *>(pNewObj)->GetAutoFormatIndex() );		
+			pViewData->MovePrevRow();
+			}
+	}
+
+
 	if (bDone)
 		aModificator.SetDocumentModified();
 
diff -u -r ../OOO_1_1_3_bak/sc/source/ui/inc/dbfunc.hxx sc/source/ui/inc/dbfunc.hxx
--- ../OOO_1_1_3_bak/sc/source/ui/inc/dbfunc.hxx	2004-10-20 10:04:56.000000000 +0530
+++ sc/source/ui/inc/dbfunc.hxx	2004-10-28 15:27:28.000000000 +0530
@@ -112,7 +112,9 @@
 									const ScDPObject& rSource, BOOL bApi = FALSE );
 	void			DeletePivotTable();
 	void			RecalcPivotTable();
-
+	void			AutoFormatPivotTable(USHORT nIndex);
+	void 		AutoFormatPivotTable(ScDPObject*	pDPObj, USHORT nIndex);
+		
 	void			MakeOutline( BOOL bColumns, BOOL bRecord = TRUE );
 	void			RemoveOutline( BOOL bColumns, BOOL bRecord = TRUE );
 	void			RemoveAllOutlines( BOOL bRecord = TRUE );
diff -u -r ../OOO_1_1_3_bak/sc/source/ui/inc/viewdata.hxx sc/source/ui/inc/viewdata.hxx
--- ../OOO_1_1_3_bak/sc/source/ui/inc/viewdata.hxx	2004-10-20 10:04:55.000000000 +0530
+++ sc/source/ui/inc/viewdata.hxx	2004-10-28 16:32:32.000000000 +0530
@@ -321,6 +321,8 @@
 
 	BOOL			GetSimpleArea( USHORT& rStartCol, USHORT& rStartRow, USHORT& rStartTab,
 									USHORT& rEndCol, USHORT& rEndRow, USHORT& rEndTab );
+	void 			MoveNextRow();
+	void 			MovePrevRow();
 	BOOL			GetSimpleArea( ScRange& rRange );
 	void			GetMultiArea( ScRangeListRef& rRange );
 
diff -u -r ../OOO_1_1_3_bak/sc/source/ui/src/popup.src sc/source/ui/src/popup.src
--- ../OOO_1_1_3_bak/sc/source/ui/src/popup.src	2004-10-20 10:04:59.000000000 +0530
+++ sc/source/ui/src/popup.src	2004-11-17 15:36:31.310003720 +0530
@@ -1040,6 +1040,14 @@
 			Text[ kannada ] = "à²†à²°à²‚à²­...";
 		};
 		MenuItem
+                {
+                        Identifier = SID_PIVOT_AFMT ;
+                        HelpId = SID_PIVOT_AFMT ;
+                        Text = "A~ktualisieren" ;
+                        Text [ ENGLISH ] = "~AutoFormat..." ;
+			Text [ english_us ] = "~AutoFormat..." ; 
+		};
+		MenuItem
 		{
 			Identifier = SID_PIVOT_RECALC ;
 			HelpId = SID_PIVOT_RECALC ;
diff -u -r ../OOO_1_1_3_bak/sc/source/ui/view/dbfunc3.cxx sc/source/ui/view/dbfunc3.cxx
--- ../OOO_1_1_3_bak/sc/source/ui/view/dbfunc3.cxx	2004-10-20 10:05:00.000000000 +0530
+++ sc/source/ui/view/dbfunc3.cxx	2004-10-28 15:26:15.000000000 +0530
@@ -679,7 +679,29 @@
 	else
 		ErrorMessage(STR_PIVOT_NOTFOUND);
 }
+void ScDBFunc::AutoFormatPivotTable(USHORT nIndex)
+{
+	ScDocShell* pDocSh  = GetViewData()->GetDocShell();
+	ScDocument* pDoc	= GetViewData()->GetDocument();
+
+	//	old pivot not used any more
 
+	ScDPObject*	pDPObj	= pDoc->GetDPAtCursor( GetViewData()->GetCurX(),
+												  GetViewData()->GetCurY(),
+												  GetViewData()->GetTabNo() );
+	if ( pDPObj )
+	{
+		pDPObj->SetAutoFormatIndex(nIndex);
+	}
+	else
+		ErrorMessage(STR_PIVOT_NOTFOUND);
+
+}
+
+void ScDBFunc::AutoFormatPivotTable(ScDPObject*	pDPObj, USHORT nIndex)
+{
+	pDPObj->SetAutoFormatIndex(nIndex);
+}
 
 //
 //			DB-Operationen (Sortieren, Filtern, Teilergebnisse) wiederholen
diff -u -r ../OOO_1_1_3_bak/sc/source/ui/view/pivotsh.cxx sc/source/ui/view/pivotsh.cxx
--- ../OOO_1_1_3_bak/sc/source/ui/view/pivotsh.cxx	2004-10-20 10:05:00.000000000 +0530
+++ sc/source/ui/view/pivotsh.cxx	2004-11-17 10:29:37.000000000 +0530
@@ -87,6 +87,9 @@
 #include "dbdocfun.hxx"
 #include "uiitems.hxx"
 #include "pfiltdlg.hxx"
+#include "autoform.hxx"
+#include "autofmt.hxx"
+#include "editable.hxx"
 
 //------------------------------------------------------------------------
 
@@ -125,13 +128,56 @@
 {
 	switch ( rReq.GetSlot() )
 	{
-		case SID_PIVOT_RECALC:
-			pViewShell->RecalcPivotTable();
-			break;
+		case SID_PIVOT_AFMT:
+			{
+			ScViewData* pViewData = pViewShell->GetViewData();
+
+			pViewShell->GetDBData( TRUE, SC_DB_OLD );
+			const ScMarkData& rMark1 = pViewData->GetMarkData();
+			if ( !rMark1.IsMarked() && !rMark1.IsMultiMarked() )
+				pViewShell->MarkDataArea( TRUE );
+			
+			Window* pDlgParent = pViewShell->GetDialogParent();
+			
+			pViewData->MoveNextRow();
+				
+			ScGlobal::ClearAutoFormat();
+			ScAutoFormatData* pNewEntry = pViewShell->CreateAutoFormatData();
+			ScAutoFormatDlg*  pDlg		= new ScAutoFormatDlg(
+													pDlgParent,
+													ScGlobal::GetAutoFormat(),
+				                                                               pNewEntry,
+                            				                                   pViewShell->GetViewData()->GetDocument() );
+
+			if ( pDlg->Execute() == RET_OK )
+			{
+				ScEditableTester aTester( pViewShell );
+				if ( !aTester.IsEditable() )
+				{
+					pViewShell->ErrorMessage(aTester.GetMessageId());
+				}
+				else
+				{
+					/* AutoFormat and Store the index */
+					pViewShell->AutoFormat( pDlg->GetIndex() );
+					pViewShell->AutoFormatPivotTable( pDlg->GetIndex() );
+				}
+			}
+			delete pDlg;
+			delete pNewEntry;
+
 
+			//pViewShell->DeletePivotTable();
+			//pViewShell->RecalcPivotTable();
+			break;
+			}
 		case SID_PIVOT_KILL:
 			pViewShell->DeletePivotTable();
 			break;
+		case SID_PIVOT_RECALC:
+			//pViewShell->DeletePivotTable();
+			pViewShell->RecalcPivotTable();
+			break;
 
         case SID_DP_FILTER:
         {
diff -u -r ../OOO_1_1_3_bak/sc/source/ui/view/viewdata.cxx sc/source/ui/view/viewdata.cxx
--- ../OOO_1_1_3_bak/sc/source/ui/view/viewdata.cxx	2004-10-20 10:05:00.000000000 +0530
+++ sc/source/ui/view/viewdata.cxx	2004-10-28 16:32:16.000000000 +0530
@@ -645,6 +645,23 @@
 	aLogicMode.SetScaleY( GetZoomY() );
 }
 
+void ScViewData::MoveNextRow()
+{
+	ScRange aMarkRange;
+	aMarkData.GetMarkArea(aMarkRange);
+	aMarkRange.aStart.Set(aMarkRange.aStart.Col(), aMarkRange.aStart.Row() +1, aMarkRange.aStart.Tab());
+	aMarkData.SetMarkArea(aMarkRange);
+}
+
+void ScViewData::MovePrevRow()
+{
+	ScRange aMarkRange;
+	aMarkData.GetMarkArea(aMarkRange);
+	aMarkRange.aStart.Set(aMarkRange.aStart.Col(), aMarkRange.aStart.Row() -1, aMarkRange.aStart.Tab());
+	aMarkData.SetMarkArea(aMarkRange);
+}
+
+
 BOOL ScViewData::GetSimpleArea( USHORT& rStartCol, USHORT& rStartRow, USHORT& rStartTab,
 								USHORT& rEndCol, USHORT& rEndRow, USHORT& rEndTab )
 {
