--- /home/sragavan/cvs/bk/ooo-build/build/OOO_1_1_2/sc/source/core/data/table4.cxx	2003-03-26 23:34:00.000000000 +0530
+++ sc/source/core/data/table4.cxx	2004-10-14 11:12:38.050393488 +0530
@@ -1286,6 +1286,8 @@
 				{
 					for (rInner = nIMin; rInner <= nIMax; rInner++)
 					{
+						if (pDocument->RowFiltered( rInner, nTab))
+							continue;					
 						USHORT nInd = nActFormCnt;
 						FillFormula(nInd, bFirst, (ScFormulaCell*)pSrcCell,
 							nCol, nRow, (rInner == nIEnd) );
@@ -1297,6 +1299,8 @@
 				{
 					for (rInner = nIMin; rInner <= nIMax; rInner++)
 					{
+						if (pDocument->RowFiltered( rInner, nTab))
+							continue;
 						aCol[nCol].Insert(nRow, pSrcCell->Clone(pDocument));
 					}
 					nProgress += nIMax - nIMin + 1;
