--- /home/sragavan/cvs/bk/ooo-build/build/OOO_1_1_2/sc/source/ui/view/viewfun3.cxx	2003-10-31 21:40:15.000000000 +0530
+++ sc/source/ui/view/viewfun3.cxx	2004-10-14 13:29:36.754960480 +0530
@@ -442,8 +442,8 @@
 
 	if (pOwnClip)
 		PasteFromClip( IDF_ALL, pOwnClip->GetDocument(),
-						PASTE_NOFUNC, FALSE, FALSE, FALSE, INS_NONE, IDF_NONE,
-						TRUE );		// allow warning dialog
+						PASTE_NOFUNC, FALSE, FALSE, FALSE, INS_CELLSDOWN, IDF_NONE,
+						FALSE );		
 	else if (pDrawClip)
 		PasteDraw();
 	else
