--- /home/sragavan/cvs/bk/ooo-build/build/OOO_1_1_2/sc/source/ui/view/viewfun3.cxx	2003-10-31 21:40:15.000000000 +0530
+++ sc/source/ui/view/viewfun3.cxx	2004-10-13 14:23:56.248238672 +0530
@@ -721,15 +721,14 @@
 			}
 		}
 
-		if (nBlockAddX > nDestSizeX)
-			nMarkAddX = nBlockAddX - nDestSizeX;			// fuer Merge-Test
-		else
-			nEndCol = nStartCol + nDestSizeX;
+		/*
+		  * Therz no use in copying lesser data multiple times in a bigger marked area
+		  * So Im just reducing the size of the marked area.
+		  */
+		nEndCol = nStartCol + nDestSizeX;
 
-		if (nBlockAddY > nDestSizeY)
-			nMarkAddY = nBlockAddY - nDestSizeY;			// fuer Merge-Test
-		else
-			nEndRow = nStartRow + nDestSizeY;
+		nEndRow = nStartRow + nDestSizeY;
+		
 	}
 	else
 	{
