--- sc/source/ui/dbgui/fieldwnd.cxx
+++ sc/source/ui/dbgui/fieldwnd.cxx
@@ -638,8 +638,8 @@
 
 void ScDPFieldWindow::ClearFields()
 {
-    if( eType == TYPE_SELECT )
+    if( eType == TYPE_SELECT || eType == TYPE_COL || eType == TYPE_ROW || eType == TYPE_DATA)
 	{
         com::sun::star::uno::Reference < com::sun::star::accessibility::XAccessible > xTempAcc = xAccessible;
         if (!xTempAcc.is() && pAccessible)
             pAccessible = NULL;



--- sc/source/ui/dbgui/pvlaydlg.cxx
+++ sc/source/ui/dbgui/pvlaydlg.cxx
@@ -107,22 +107,22 @@
 
 
 //============================================================================
-//	struct FuncData
+//	struct ScDPFuncData
 
-struct FuncData
+struct ScDPFuncData
 {
 	short  nCol;
 	USHORT nFuncMask;
 
-	FuncData( short col, USHORT funcs = PIVOT_FUNC_SUM )
+	ScDPFuncData( short col, USHORT funcs = PIVOT_FUNC_SUM )
 		: nCol(col), nFuncMask(funcs) {}
-	FuncData( const FuncData& rCpy )
+	ScDPFuncData( const ScDPFuncData& rCpy )
 		: nCol(rCpy.nCol), nFuncMask(rCpy.nFuncMask) {}
 
-	FuncData& operator=( const FuncData& r )
+	ScDPFuncData& operator=( const ScDPFuncData& r )
 		{ nCol=r.nCol; nFuncMask=r.nFuncMask; return *this; }
 
-	BOOL operator==( const FuncData& r )
+	BOOL operator==( const ScDPFuncData& r )
 		{ return ( (nCol==r.nCol)&&(nFuncMask==r.nFuncMask) ); }
 };
 
@@ -413,7 +413,7 @@
 			if ( i <= nLast )
 			{
 				aWndSelect.AddField( *(aLabelDataArr[i]->pStrColName), i );
-				aSelectArr[i] = new FuncData( aLabelDataArr[i]->nCol,
+				aSelectArr[i] = new ScDPFuncData( aLabelDataArr[i]->nCol,
 											  aLabelDataArr[i]->nFuncMask );
 			}
 		}
@@ -427,7 +427,7 @@
 {
 	if ( pArr && (eType != TYPE_SELECT) )
 	{
-        FuncData**          pInitArr = NULL;
+        ScDPFuncData**          pInitArr = NULL;
         ScDPFieldWindow*    pInitWnd = NULL;
         BOOL                bDataArr = FALSE;
 
@@ -462,7 +462,7 @@
 
 				if ( nCol != PIVOT_DATA_FIELD )
 				{
-					pInitArr[j] = new FuncData( nCol, nMask );
+					pInitArr[j] = new ScDPFuncData( nCol, nMask );
 
 					if ( !bDataArr )
 					{
@@ -509,12 +509,12 @@
 
 void ScDPLayoutDlg::AddField( long nFromIndex, ScDPFieldType eToType, const Point& rAtPos )
 {
-    FuncData            fData( *(aSelectArr[nFromIndex]) );
+    ScDPFuncData            fData( *(aSelectArr[nFromIndex]) );
     long                nAt   = 0;
     ScDPFieldWindow*    toWnd = NULL;
     ScDPFieldWindow*    rmWnd = NULL;
-    FuncData**          toArr = NULL;
-    FuncData**          rmArr = NULL;
+    ScDPFuncData**          toArr = NULL;
+    ScDPFuncData**          rmArr = NULL;
     BOOL                bDataArr = FALSE;
 
 	switch ( eToType )
@@ -600,9 +600,9 @@
         ScDPFieldWindow*    fromWnd  = NULL;
         ScDPFieldWindow*    toWnd    = NULL;
         ScDPFieldWindow*    rmWnd    = NULL;
-        FuncData**          fromArr  = NULL;
-        FuncData**          toArr    = NULL;
-        FuncData**          rmArr    = NULL;
+        ScDPFuncData**          fromArr  = NULL;
+        ScDPFuncData**          toArr    = NULL;
+        ScDPFuncData**          rmArr    = NULL;
         long                nAt      = 0;
         BOOL                bDataArr = FALSE;
 
@@ -649,7 +649,7 @@
 
 		if ( fromArr && toArr && fromWnd && toWnd )
 		{
-            FuncData fData( *(fromArr[nFromIndex]) );
+            ScDPFuncData fData( *(fromArr[nFromIndex]) );
 
 			if ( Contains( fromArr, fData.nCol, nAt ) )
 			{
@@ -703,7 +703,7 @@
 	else // -> eFromType == eToType
 	{
         ScDPFieldWindow*    theWnd  = NULL;
-        FuncData**          theArr   = NULL;
+        ScDPFuncData**          theArr   = NULL;
         long                nAt      = 0;
         long                nToIndex = 0;
         Point               aToPos;
@@ -728,7 +728,7 @@
 				break;
 		}
 
-        FuncData fData( *(theArr[nFromIndex]) );
+        ScDPFuncData fData( *(theArr[nFromIndex]) );
 
 		if ( Contains( theArr, fData.nCol, nAt ) )
 		{
@@ -775,7 +775,7 @@
 
 void ScDPLayoutDlg::RemoveField( ScDPFieldType eFromType, long nIndex )
 {
-    FuncData** pArr = NULL;
+    ScDPFuncData** pArr = NULL;
     switch( eFromType )
     {
         case TYPE_COL:  pArr = aColArr;     break;
@@ -886,7 +886,7 @@
 
 void ScDPLayoutDlg::NotifyDoubleClick( ScDPFieldType eType, long nFieldIndex )
 {
-	FuncData** pArr = NULL;
+	ScDPFuncData** pArr = NULL;
 	switch ( eType )
 	{
 		case TYPE_COL:		pArr = aColArr;		break;
@@ -1010,7 +1010,7 @@
 
 //----------------------------------------------------------------------------
 
-BOOL ScDPLayoutDlg::Contains( FuncData** pArr, short nCol, long& nAt )
+BOOL ScDPLayoutDlg::Contains( ScDPFuncData** pArr, short nCol, long& nAt )
 {
 	if ( !pArr )
 		return FALSE;
@@ -1032,7 +1032,7 @@
 
 //----------------------------------------------------------------------------
 
-void ScDPLayoutDlg::Remove( FuncData** pArr, long nAt )
+void ScDPLayoutDlg::Remove( ScDPFuncData** pArr, long nAt )
 {
 	if ( !pArr || (nAt>MAX_FIELDS-1) )
 		return;
@@ -1054,14 +1054,14 @@
 
 //----------------------------------------------------------------------------
 
-void ScDPLayoutDlg::Insert( FuncData** pArr, const FuncData& rFData, long nAt )
+void ScDPLayoutDlg::Insert( ScDPFuncData** pArr, const ScDPFuncData& rFData, long nAt )
 {
 	if ( !pArr || (nAt>MAX_FIELDS-1) )
 		return;
 
 	if ( pArr[nAt] == NULL )
 	{
-		pArr[nAt] = new FuncData( rFData );
+		pArr[nAt] = new ScDPFuncData( rFData );
 	}
 	else
 	{
@@ -1069,7 +1069,7 @@
 		{
             for ( long i=MAX_FIELDS-1; i>nAt; i-- )
 				pArr[i] = pArr[i-1];
-			pArr[nAt] = new FuncData( rFData );
+			pArr[nAt] = new ScDPFuncData( rFData );
 		}
 	}
 }
@@ -1482,26 +1482,26 @@
 	pDlgDPObject->FillOldParam( thePivotData, FALSE );
 	pDlgDPObject->FillLabelData( thePivotData, bShowAll, MAX_LABELS );
 
+	aWndSelect.ClearFields();
+	aWndData.ClearFields();
+	aWndRow.ClearFields();
+	aWndCol.ClearFields();
 	
 	USHORT i;
- 	for ( i=0; i<MAX_LABELS; i++ )
-		aSelectArr[i] = NULL;
-
 	for ( i=0; i<MAX_FIELDS; i++ )
 	{
-		aColArr[i]  = NULL;
-		aRowArr[i]  = NULL;
-		aDataArr[i] = NULL;
+		RemoveField (TYPE_COL, i);
+		RemoveField (TYPE_ROW, i);
+		RemoveField (TYPE_DATA, i);
 	}
 
+ 	for ( i=0; i<MAX_LABELS; i++ )
+		aSelectArr[i] = NULL;
 	for ( i=0; i<nLabelCount; i++ )
-	{
 		delete aLabelDataArr[i];		
-	}	
 	delete [] aLabelDataArr;
-	
+	aLabelDataArr = NULL;
 	nLabelCount = 0;
-	aWndSelect.ClearFields();
 	
 	InitWndSelect( thePivotData.ppLabelArr, thePivotData.nLabels );
 	InitWnd( thePivotData.aColArr,  thePivotData.nColCount,  TYPE_COL );
@@ -1601,9 +1606,9 @@
 		pData = aLabelDataArr[nOffset+i];
 		aWndSelect.AddField( *(pData->pStrColName), i );
 		if ( aSelectArr[i] )
-			*(aSelectArr[i]) = FuncData( pData->nCol, pData->nFuncMask );
+			*(aSelectArr[i]) = ScDPFuncData( pData->nCol, pData->nFuncMask );
 		else
-			aSelectArr[i] = new FuncData( pData->nCol, pData->nFuncMask );
+			aSelectArr[i] = new ScDPFuncData( pData->nCol, pData->nFuncMask );
 	}
 	for ( ; i<MAX_LABELS; i++ )
 		DELETEZ( aSelectArr[i] );



--- sc/source/ui/inc/pvlaydlg.hxx
+++ sc/source/ui/inc/pvlaydlg.hxx
@@ -114,7 +114,7 @@
 class ScViewData;
 class ScDocument;
 class ScRangeData;
-struct FuncData;
+struct ScDPFuncData;
 class ScDPObject;
 
 //============================================================================
@@ -200,10 +200,10 @@
     ScDPFieldType           eLastActiveType;        /// Type of last active area.
     long                    nOffset;                /// Offset of first field in TYPE_SELECT area.
 
-    FuncData*               aSelectArr[MAX_LABELS]; // (nCol, nFuncMask)
-    FuncData*               aColArr   [MAX_FIELDS]; // (nCol, nFuncMask)
-    FuncData*               aRowArr   [MAX_FIELDS]; // (nCol, nFuncMask)
-    FuncData*               aDataArr  [MAX_FIELDS]; // (nCol, nFuncMask)
+    ScDPFuncData*               aSelectArr[MAX_LABELS]; // (nCol, nFuncMask)
+    ScDPFuncData*               aColArr   [MAX_FIELDS]; // (nCol, nFuncMask)
+    ScDPFuncData*               aRowArr   [MAX_FIELDS]; // (nCol, nFuncMask)
+    ScDPFuncData*               aDataArr  [MAX_FIELDS]; // (nCol, nFuncMask)
 
     BOOL                    bShowAll[MAX_LABELS];
 
@@ -225,9 +225,9 @@
     LabelData*              GetLabelData    ( short nCol, long* pPos = NULL );
     String                  GetLabelString  ( short nCol );
     String                  GetFuncString   ( USHORT& rFuncMask, BOOL bIsValue = TRUE );
-    BOOL                    Contains        ( FuncData** pArr, short nCol, long& nAt );
-    void                    Remove          ( FuncData** pArr, long nAt );
-    void                    Insert          ( FuncData** pArr, const FuncData& rFData, long nAt );
+    BOOL                    Contains        ( ScDPFuncData** pArr, short nCol, long& nAt );
+    void                    Remove          ( ScDPFuncData** pArr, long nAt );
+    void                    Insert          ( ScDPFuncData** pArr, const ScDPFuncData& rFData, long nAt );
 
     void                    AddField        ( long nFromIndex,
                                               ScDPFieldType eToType, const Point& rAtPos );
