Index: sfx2/source/doc/docinf.cxx
===================================================================
RCS file: /cvs/framework/sfx2/source/doc/docinf.cxx,v
retrieving revision 1.24
diff -u -p -r1.24 docinf.cxx
--- sfx2/source/doc/docinf.cxx	26 May 2003 08:29:06 -0000	1.24
+++ sfx2/source/doc/docinf.cxx	1 Nov 2004 11:23:17 -0000
@@ -924,7 +924,7 @@ ULONG SfxDocumentInfo::LoadPropertySet( 
 {
 	SvStorageStreamRef aStrPropSet = pStorage->OpenStream(
 		String::CreateFromAscii( pPropSlot ), STREAM_STD_READ );
-	if ( !aStrPropSet.Is() )
+	if ( !aStrPropSet.Is() || aStrPropSet->GetError() != SVSTREAM_OK)
 		return ERRCODE_IO_ACCESSDENIED;
 	aStrPropSet->SetBufferSize( STREAM_BUFFER_SIZE );
 	SfxPS_Impl* pPS = new SfxPS_Impl;
