--- svtools/source/config/ctloptions.cxx.default-on	2004-12-09 11:15:05.895615360 -0500
+++ svtools/source/config/ctloptions.cxx	2004-12-09 11:40:31.287720512 -0500
@@ -165,7 +165,7 @@
     utl::ConfigItem( ASCII_STR("Office.Common/I18N/CTL") ),
 
 	m_bIsLoaded				( sal_False ),
-    m_bCTLFontEnabled		( sal_False ),
+    m_bCTLFontEnabled		( sal_True ),
     m_bCTLSequenceChecking	( sal_False ),
 	m_eCTLCursorMovement	( SvtCTLOptions::MOVEMENT_LOGICAL ),
     m_eCTLTextNumerals      ( SvtCTLOptions::NUMERALS_ARABIC ),
