Index: svx/source/items/textitem.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/items/textitem.cxx,v
retrieving revision 1.52
diff -u -p -r1.52 textitem.cxx
--- svx/source/items/textitem.cxx	17 Apr 2003 09:57:58 -0000	1.52
+++ svx/source/items/textitem.cxx	9 Dec 2004 11:43:56 -0000
@@ -3704,24 +3704,21 @@ void GetDefaultFonts( SvxFontItem& rLati
 {
 	const USHORT nItemCnt = 3;
 
-	static struct
+	USHORT aOutTypeArr[ nItemCnt ] =
 	{
-		USHORT nFontType;
-		USHORT nLanguage;
-	}
-	aOutTypeArr[ nItemCnt ] =
-	{
-		{  DEFAULTFONT_LATIN_TEXT, LANGUAGE_ENGLISH_US },
-		{  DEFAULTFONT_CJK_TEXT, LANGUAGE_ENGLISH_US },
-		{  DEFAULTFONT_CTL_TEXT, LANGUAGE_ARABIC_SAUDI_ARABIA }
+		DEFAULTFONT_LATIN_TEXT,
+		DEFAULTFONT_CJK_TEXT,
+		DEFAULTFONT_CTL_TEXT
 	};
 
+	LanguageType eLanguage = Application::GetSettings().GetLanguage(); 
+
 	SvxFontItem* aItemArr[ nItemCnt ] = { &rLatin, &rAsian, &rComplex };
 
 	for ( USHORT n = 0; n < nItemCnt; ++n )
 	{
-		Font aFont( OutputDevice::GetDefaultFont( aOutTypeArr[ n ].nFontType,
-												  aOutTypeArr[ n ].nLanguage,
+		Font aFont( OutputDevice::GetDefaultFont( aOutTypeArr[ n ],
+												  eLanguage,
 												  DEFAULTFONT_FLAGS_ONLYONE, 0 ) );
 		SvxFontItem* pItem = aItemArr[ n ];
 		pItem->GetFamily() = aFont.GetFamily();
Index: svx/source/svdraw/svdmodel.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/svdraw/svdmodel.cxx,v
retrieving revision 1.48
diff -u -p -r1.48 svdmodel.cxx
--- svx/source/svdraw/svdmodel.cxx	24 Apr 2003 16:42:29 -0000	1.48
+++ svx/source/svdraw/svdmodel.cxx	9 Dec 2004 11:44:19 -0000
@@ -935,19 +935,20 @@ void SdrModel::SetTextDefaults() const
 void ImpGetDefaultFontsLanguage( SvxFontItem& rLatin, SvxFontItem& rAsian, SvxFontItem& rComplex)
 {
 	const USHORT nItemCnt = 3;
-	static struct {
-		USHORT nFntType, nLanguage;
-	}  aOutTypeArr[ nItemCnt ] = {
-		{  DEFAULTFONT_LATIN_TEXT, LANGUAGE_ENGLISH_US },
-		{  DEFAULTFONT_CJK_TEXT, LANGUAGE_ENGLISH_US },
-		{  DEFAULTFONT_CTL_TEXT, LANGUAGE_ARABIC_SAUDI_ARABIA }
+	USHORT aOutTypeArr[ nItemCnt ] =
+	{
+		DEFAULTFONT_LATIN_TEXT,
+		DEFAULTFONT_CJK_TEXT,
+		DEFAULTFONT_CTL_TEXT
 	};
+
+	LanguageType eLanguage = Application::GetSettings().GetLanguage(); 
 	SvxFontItem* aItemArr[ nItemCnt ] = { &rLatin, &rAsian, &rComplex };
 
 	for( USHORT n = 0; n < nItemCnt; ++n )
 	{
 		Font aFnt( OutputDevice::GetDefaultFont(
-			aOutTypeArr[ n ].nFntType, aOutTypeArr[ n ].nLanguage,
+			aOutTypeArr[ n ], eLanguage,
 			DEFAULTFONT_FLAGS_ONLYONE, 0 ));
 		SvxFontItem* pI = aItemArr[ n ];
 		pI->GetFamily() = aFnt.GetFamily();
