Index: source/core/layout/fly.cxx
===================================================================
RCS file: /cvs/sw/sw/source/core/layout/fly.cxx,v
retrieving revision 1.46.52.3
diff -u -p -r1.46.52.3 fly.cxx
--- sw/source/core/layout/fly.cxx	28 Apr 2004 10:29:47 -0000	1.46.52.3
+++ sw/source/core/layout/fly.cxx	5 Nov 2004 11:15:08 -0000
@@ -1695,8 +1695,9 @@ void SwFlyFrm::MakeFlyPos()
 			const long nBottom = GetAnchor()->Frm().Top() + aRelPos.Y() + Frm().Height();
 			if ( nAnchorBottom < nBottom )
 			{
-				((SwPageFrm*)GetAnchor())->FindBodyCont()->
-                                    Grow( nBottom - nAnchorBottom PHEIGHT );
+				if (SwPageFrm *pOne = ((SwPageFrm*)GetAnchor()))
+						if (SwLayoutFrm *pTwo = pOne->FindBodyCont())
+							pTwo->Grow( nBottom - nAnchorBottom PHEIGHT );
 			}
 		}
 
