Index: vcl/source/gdi/pdfwriter_impl.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/gdi/pdfwriter_impl.cxx,v
retrieving revision 1.55.10.6
diff -u -p -u -r1.55.10.6 pdfwriter_impl.cxx
--- vcl/source/gdi/pdfwriter_impl.cxx	13 Feb 2004 13:20:46 -0000	1.55.10.6
+++ vcl/source/gdi/pdfwriter_impl.cxx	9 Nov 2004 12:01:48 -0000
@@ -2585,9 +2585,9 @@ void PDFWriterImpl::registerGlyphs(
                     }
                 }
                 else
-                    cChar = 0;
+                    pEncoding = 0;
             }
-            else
+            if( !pEncoding )
             {
                 if( cChar & 0xff00 )
                 {
