Index: extensions/prj/build.lst
===================================================================
RCS file: /cvs/util/extensions/prj/build.lst,v
retrieving revision 1.21
diff -u -p -u -r1.21 build.lst
--- extensions/prj/build.lst	15 Apr 2003 13:51:01 -0000	1.21
+++ extensions/prj/build.lst	30 Aug 2005 15:11:29 -0000
@@ -1,4 +1,4 @@
-ex	extensions	:	offmgr rdbmaker svx sane twain offuh np_sdk NULL
+ex	extensions	:	offmgr rdbmaker svx SANE:sane twain offuh np_sdk NULL
 ex	extensions					usr1	-	all	ex_mkout NULL
 ex	extensions\inc					get		-	all	ex_inc NULL
 ex	extensions\inc\ucbhelper			get		-	all	ex_incucbh NULL
