? delzip
? legacy/delzip
Index: calc.desktop
===================================================================
RCS file: /cvs/gsl/sysui/desktop/gnome/calc.desktop,v
retrieving revision 1.2.30.2
diff -u -u -r1.2.30.2 calc.desktop
--- sysui/desktop/gnome/calc.desktop	9 Jan 2004 19:02:26 -0000	1.2.30.2
+++ sysui/desktop/gnome/calc.desktop	14 Oct 2004 07:22:58 -0000
@@ -8,3 +8,4 @@
 Type=Application
 Name=<productname> Calc
 Comment=<productname> Spreadsheet
+MimeType=application/vnd.sun.xml.calc;application/vnd.sun.xml.calc.template;application/vnd.sun.xml.math;application/vnd.ms-excel;application/vnd.stardivision.calc;application/vnd.stardivision.chart;application/vnd.stardivision.math;text/x-comma-separated-values
Index: draw.desktop
===================================================================
RCS file: /cvs/gsl/sysui/desktop/gnome/draw.desktop,v
retrieving revision 1.2.30.2
diff -u -u -r1.2.30.2 draw.desktop
--- sysui/desktop/gnome/draw.desktop	9 Jan 2004 19:02:27 -0000	1.2.30.2
+++ sysui/desktop/gnome/draw.desktop	14 Oct 2004 07:22:58 -0000
@@ -8,3 +8,4 @@
 Type=Application
 Name=<productname> Draw
 Comment=<productname> Drawing
+MimeType=application/vnd.sun.xml.draw;application/vnd.sun.xml.draw.template;application/vnd.stardivision.draw
Index: impress.desktop
===================================================================
RCS file: /cvs/gsl/sysui/desktop/gnome/impress.desktop,v
retrieving revision 1.2.30.3
diff -u -u -r1.2.30.3 impress.desktop
--- sysui/desktop/gnome/impress.desktop	27 Aug 2004 10:27:44 -0000	1.2.30.3
+++ sysui/desktop/gnome/impress.desktop	14 Oct 2004 07:22:58 -0000
@@ -8,3 +8,4 @@
 Exec=<progpath_utf8>/program/simpress %U
 Name=<productname> Impress
 Comment=<productname> Presentation
+MimeType=application/vnd.sun.xml.impress;application/vnd.sun.xml.impress.template;application/vnd.ms-powerpoint;application/vnd.stardivision.impress
Index: writer.desktop
===================================================================
RCS file: /cvs/gsl/sysui/desktop/gnome/writer.desktop,v
retrieving revision 1.2.30.2
diff -u -u -r1.2.30.2 writer.desktop
--- sysui/desktop/gnome/writer.desktop	9 Jan 2004 19:02:29 -0000	1.2.30.2
+++ sysui/desktop/gnome/writer.desktop	14 Oct 2004 07:22:58 -0000
@@ -8,3 +8,4 @@
 Type=Application
 Name=<productname> Writer
 Comment=<productname> Text Document
+MimeType=application/vnd.sun.xml.writer;application/vnd.sun.xml.writer.global;application/vnd.sun.xml.writer.template;application/vnd.ms-word;application/x-msword;application/x-doc;application/x-mswrite;application/vnd.stardivision.writer;application/rtf;text/rtf;text/richtext;text/plain
--- sysui/gnome/desktop/math.desktop	9 Jan 2004 19:02:27 -0000	1.2.30.2
+++ sysui/gnome/desktop/math.desktop	24 Nov 2004 22:18:33 -0000
@@ -8,3 +8,4 @@
 Type=Application
 Name=<productname> Math
 Comment=<productname> Formula
+MimeType=application/vnd.sun.xml.writer.math;application/vnd.stardivision.math
