Index: impress.desktop
===================================================================
RCS file: /cvs/gsl/sysui/desktop/gnome/impress.desktop,v
retrieving revision 1.2.30.2
diff -u -u -r1.2.30.2 impress.desktop
--- sysui/desktop/gnome/impress.desktop	9 Jan 2004 19:02:27 -0000	1.2.30.2
+++ sysui/desktop/gnome/impress.desktop	8 Nov 2004 13:27:53 -0000
@@ -7,4 +7,4 @@
 Type=Application
 Exec=<progpath_utf8>/program/simpress %U
 Name=<productname> Impress
-Commment=<productname> Presentation
+Comment=<productname> Presentation
