Index: offmgr/source/offapp/dialog/treeopt.src
===================================================================
RCS file: /cvs/framework/offmgr/source/offapp/dialog/Attic/treeopt.src,v
retrieving revision 1.132.10.4
diff -u -p -r1.132.10.4 treeopt.src
--- offmgr/source/offapp/dialog/treeopt.src	27 Apr 2004 16:17:20 -0000	1.132.10.4
+++ offmgr/source/offapp/dialog/treeopt.src	9 Nov 2004 10:07:12 -0000
@@ -902,16 +902,16 @@ Resource RID_OFADLG_OPTIONS_TREE_PAGES
 			< "%PRODUCTNAME" ; 0; > ;
 			< "Dados de utilizador" ;  RID_SFXPAGE_GENERAL; > ;
 			< "Geral" ;  OFA_TP_MISC; > ;
-			< "Ver" ; OFA_TP_MEMORY; > ;
-			< "Imprimir" ; OFA_TP_VIEW ; > ;
-			< "Programas auxiliares" ; RID_SFXPAGE_PRINTOPTIONS; > ;
-			< "Atalhos" ; OFA_TP_HELPERPROG; > ;
-			< "Cores" ;  RID_SFXPAGE_PATH; > ;
-			< "Substituição de caracteres" ;  RID_SVXPAGE_COLOR; > ;
-			< "Segurança" ; RID_SVX_FONT_SUBSTITUTION ; > ;
+			< "Memória" ; OFA_TP_MEMORY; > ;
+			< "Ver" ; OFA_TP_VIEW; > ;
+			< "Imprimir" ; RID_SFXPAGE_PRINTOPTIONS; > ;
+			< "Programas auxiliares" ; OFA_TP_HELPERPROG; > ;
+			< "Atalhos" ; RID_SFXPAGE_PATH; > ;
+			< "Cores" ;  RID_SVXPAGE_COLOR; > ;
+			< "Substituição de caracteres" ;  RID_SVX_FONT_SUBSTITUTION; > ;
 			< "Segurança" ; RID_SVXPAGE_INET_SCRIPTING ; > ;
-			< "Java" ;        RID_SVXPAGE_COLORCONFIG        ; > ;
-			< "Outros" ; RID_SVXPAGE_ACCESSIBILITYCONFIG ; > ;
+			< "Aparência" ; RID_SVXPAGE_COLORCONFIG ; > ;
+			< "Acessibilidade" ; RID_SVXPAGE_ACCESSIBILITYCONFIG ; > ;
 			< "Netzwerkidentität" ; RID_SVXPAGE_SSO ; > ;
 		};
 		ItemList [ russian ] =
