--- ./transex3/source/export.cxx.indic	2004-11-25 14:49:45.000000000 -0500
+++ ./transex3/source/export.cxx	2004-11-25 14:53:58.000000000 -0500
@@ -585,7 +585,7 @@
 				// cur. line has macro line end
 				ByteString sTmpLine( sToken );
 				sTmpLine.EraseAllChars( '\t' ); sTmpLine.EraseAllChars( ' ' );
-				if( sTmpLine.Len() < 0 ){
+				if( sTmpLine.Len() > 0 ){
 					if ( sTmpLine.GetChar(( USHORT )( sTmpLine.Len() - 1 )) != '\\' )
 						bNextMustBeDefineEOL = TRUE;
 				}
@@ -2416,7 +2420,8 @@
 									if ( sLine.GetChar( nEnd ) == '\"' )
 										bFound = TRUE;
 								}
-								nEnd --;
+								if (sLine.Search (" \"\\\"\" ;") == STRING_NOTFOUND)
+									nEnd --;
 								ByteString sPostFix( sLine.Copy( ++nEnd ));
 								sLine.Erase( nStart );
 
--- ./transex3/source/merge.cxx.indic	2004-11-25 14:50:17.000000000 -0500
+++ ./transex3/source/merge.cxx	2004-11-25 14:53:58.000000000 -0500
@@ -183,7 +183,8 @@
 		aInputStream.ReadLine( sLine );
 		sLine = sLine.Convert( RTL_TEXTENCODING_MS_1252, aCharSet );
 
-		if ( sLine.GetTokenCount( '\t' ) == 15 ) {
+		int tcnt = sLine.GetTokenCount( '\t' );
+		if ( tcnt >= 13 && tcnt <= 16 ) {
 			if ( nFileFormat != FFORMAT_NEW ) {
 				nFileFormat = FFORMAT_NEW;
 				fprintf( stdout, "File detection: Version 2.0 detected!\n" );
--- ./transex3/source/srclex.l.indic	2004-11-25 14:51:04.000000000 -0500
+++ ./transex3/source/srclex.l	2004-11-25 14:54:10.000000000 -0500
@@ -129,6 +129,12 @@
 	WorkOnTokenSet( LEVELDOWN, yytext );
 }
 
+"}"[ \t]*	{
+/* LEVELDOWN alternate */
+	YYWarning ("Missing ; at end of block (should be '};')");
+	WorkOnTokenSet( LEVELDOWN, yytext );
+}
+
 [a-zA-Z0-9_]+[ \t]*"="[ \t]*"MAP_APPFONT"[ \t]*"(".+")".*	{
 /* APPFONTMAPPING  Typ = MAP_APPFONT( ... ) */
 	WorkOnTokenSet( APPFONTMAPPING, yytext );
