--- unotools/source/ucbhelper/makefile.mk	16 Dec 2003 11:43:22 -0000	1.13.4.1
+++ unotools/source/ucbhelper/makefile.mk	6 Nov 2004 00:37:31 -0000
@@ -76,11 +76,13 @@ ENABLE_EXCEPTIONS=TRUE
 
 .INCLUDE : settings.mk
 
-.IF "$(OS)$(CPU)" == "SOLARISI"
+#.IF "$(OS)$(CPU)" == "SOLARISI"
 # introduced because of bug 114346, pure virtual function call.
 # only under solaris intel in nondebug version, compiler version
 NOOPTFILES=$(SLO)$/ucblockbytes.obj
-.ENDIF
+#.ENDIF
+#Actually I (cmc) can see this pure virtual call as well in linux ix86 fedora
+#1.1.2-10, i.e. #rh110934#
 
 # --- Files -------------------------------------
 
