Index: basic/source/comp/scanner.cxx
===================================================================
RCS file: /cvs/script/basic/source/comp/scanner.cxx,v
retrieving revision 1.8
diff -u -p -u -r1.8 scanner.cxx
--- basic/source/comp/scanner.cxx	23 Apr 2003 16:57:09 -0000	1.8
+++ basic/source/comp/scanner.cxx	11 Mar 2004 15:10:40 -0000
@@ -184,7 +184,7 @@ static SbxDataType GetSuffixType( sal_Un
 BOOL SbiScanner::NextSym()
 {
 	// Fuer den EOLN-Fall merken
-	USHORT nOldLine = nLine;
+	UINT32 nOldLine = nLine;
 	USHORT nOldCol1 = nCol1;
 	USHORT nOldCol2 = nCol2;
 	sal_Unicode buf[ BUF_SIZE ], *p = buf;
--- basic/source/inc/scanner.hxx	2005-01-17 14:08:28.867361749 -0800
+++ basic/source/inc/scanner.hxx	2005-01-17 14:08:52.439391875 -0800
@@ -95,7 +95,7 @@
 	short  nErrors;						// Anzahl Fehler
 	short  nColLock;					// Lock-Zaehler fuer Col1
 	INT32  nBufPos;						// aktuelle Buffer-Pos
-	USHORT nLine;						// aktuelle Zeile
+	UINT32 nLine;						// aktuelle Zeile
 	USHORT nCol1, nCol2;				// aktuelle 1. und 2. Spalte
 	BOOL   bSymbol;						// TRUE: Symbol gescannt
 	BOOL   bNumber;						// TRUE: Zahl gescannt
