Index: sc/source/filter/excel/xelink.cxx
===================================================================
RCS file: /cvs/sc/sc/source/filter/excel/xelink.cxx,v
retrieving revision 1.4
diff -u -r1.4 xelink.cxx
--- sc/source/filter/excel/xelink.cxx	27 May 2003 10:37:28 -0000	1.4
+++ sc/source/filter/excel/xelink.cxx	8 Sep 2004 09:55:46 -0000
@@ -88,6 +88,7 @@
 #endif
 
 #include "root.hxx"
+#include "excrecds.hxx"
 
 const sal_Unicode DDE_DELIM         = '|';
 
@@ -845,6 +846,13 @@
     rnXti = InsertXti( nSupbook, EXC_TAB_EXTERNAL, EXC_TAB_EXTERNAL );
 }
 
+
+void
+XclExpLinkManager::InsertMacro(sal_uInt16& rnFnIdx, const String& rName )
+{
+	rnFnIdx = mpRD->pNameList->GetMacroIdx( *mpRD, rName );
+}
+
 bool XclExpLinkManager::InsertDde(
         sal_uInt16& rnXti, sal_uInt16& rnExtName,
         const String& rApplic, const String& rTopic, const String& rItem )
Index: sc/source/filter/inc/xelink.hxx
===================================================================
RCS file: /cvs/sc/sc/source/filter/inc/xelink.hxx,v
retrieving revision 1.4
diff -u -r1.4 xelink.hxx
--- sc/source/filter/inc/xelink.hxx	27 May 2003 10:37:38 -0000	1.4
+++ sc/source/filter/inc/xelink.hxx	8 Sep 2004 09:55:47 -0000
@@ -581,6 +581,10 @@
     void                        InsertAddIn(
                                     sal_uInt16& rnXti, sal_uInt16& rnExtName,
                                     const String& rName );
+    /** Finds or inserts a NAME record for a macro function name.
+        @param rnXti  Returns the index of the name for this macro */
+    void                        InsertMacro(
+                                    sal_uInt16& rnFnIdx, const String& rName );
     /** Finds or inserts an EXTERNNAME record for DDE links.
         @param rnXti  Returns the index of the XTI structure which contains the DDE link.
         @param rnExtName  Returns the 1-based EXTERNNAME record index. */
--- sc/source/filter/excel/exccomp.cxx.bak	2005-01-17 15:13:40.281789882 -0800
+++ sc/source/filter/excel/exccomp.cxx	2005-01-17 15:13:53.583550657 -0800
@@ -1371,10 +1371,6 @@
         // for ptgNameX (extern name): [0] = EXTERNSHEET index, [1] = EXTERNNAME index
         UINT16 pExtNameData[ 2 ] = { 0, 0xFFFF };
 
-		if( pCur->GetOpCode() == ocExternal )
-            pExcRoot->pER->GetLinkManager().InsertAddIn(
-                pExtNameData[ 0 ], pExtNameData[ 1 ],
-                pExcRoot->pER->GetXclAddInName( pCur->GetExternal() ) );
 		UINT16 nFunctionOp = pCur->GetOpCode();
 		if( nFunctionOp  == ocExternal )
 		{
