--- vcl/unx/gtk/window/gtkframe.cxx	2005-02-07 11:51:33.685360165 +0530
+++ vcl/unx/gtk/window/gtkframe.cxx	2005-02-07 12:05:14.805240868 +0530
@@ -138,6 +138,8 @@
             case GDK_KP_Right:
             case GDK_Right:			nCode =	KEY_RIGHT;		break;
             case GDK_KP_Home:
+            case GDK_KP_Begin:
+	    case GDK_Begin:
             case GDK_Home:			nCode = KEY_HOME;		break;
             case GDK_KP_End:
             case GDK_End:			nCode = KEY_END;		break;
@@ -154,22 +156,32 @@
             case GDK_BackSpace:		nCode = KEY_BACKSPACE;	break;
             case GDK_KP_Space:
             case GDK_space:			nCode = KEY_SPACE;		break;
+	    case GDK_KP_Insert:
             case GDK_Insert:		nCode = KEY_INSERT;		break;
+	    case GDK_KP_Delete:
             case GDK_Delete:		nCode = KEY_DELETE;		break;
+	    case GDK_plus:
             case GDK_KP_Add:		nCode = KEY_ADD;		break;
+	    case GDK_minus:
             case GDK_KP_Subtract:	nCode = KEY_SUBTRACT;	break;
+	    case GDK_asterisk:
             case GDK_KP_Multiply:	nCode = KEY_MULTIPLY;	break;
+	    case GDK_slash:
             case GDK_KP_Divide:		nCode = KEY_DIVIDE;		break;
             case GDK_period:
             case GDK_decimalpoint:	nCode = KEY_POINT;		break;
             case GDK_comma:			nCode = KEY_COMMA;		break;
             case GDK_less:			nCode = KEY_LESS;		break;
             case GDK_greater:		nCode = KEY_GREATER;	break;
+	    case GDK_KP_Equal:
             case GDK_equal:			nCode = KEY_EQUAL;		break;
             case GDK_Find:			nCode = KEY_FIND;		break;
             case GDK_Menu:			nCode = KEY_MENU;		break;
             case GDK_Help:			nCode = KEY_HELP;		break;
             case GDK_Undo:			nCode = KEY_UNDO;		break;
+            case GDK_Redo:                      nCode = KEY_REPEAT;             break;
+            case GDK_KP_Decimal:
+            case GDK_KP_Separator:     nCode = KEY_DECIMAL;    break;
         }
     }
     return nCode;
