--- stlport/win32_custom.sh.bak	2004-10-21 17:13:18.218750000 +0530
+++ stlport/win32_custom.sh	2004-10-21 15:03:01.718750000 +0530
@@ -3,6 +3,8 @@
 #rem - path to unpacked tarball
 #rem - relative path back to module root
 
+/usr/bin/dos2unix.exe ./wntmsci9.pro/misc/build/STLport-4.0/src/vc_common.mak
+
 cd $1
 
 patch -i $2/dos_lineends.patch -b -p2
--- solenv/inc/tg_ext.mk.bak	2004-10-28 14:40:56.671875000 +0530
+++ solenv/inc/tg_ext.mk	2004-10-28 14:47:41.531250000 +0530
@@ -176,6 +176,9 @@
 	@+-$(MKDIR) $(PACKAGE_DIR)
 	+cd $(PACKAGE_DIR) && ( $(shell +$(TYPE) $(PRJ)$/$(ROUT)$/misc$/$(TARFILE_NAME).unpack)) && $(TOUCH) $(UNTAR_FLAG_FILE)
 	@+echo make writeable...
+.IF "$(TARFILE_NAME)"=="curl-7.9.8"
+	@+/usr/bin/dos2unix.exe $(PACKAGE_DIR)/curl-7.9.8/lib/Makefile.vc6
+.ENDIF
 .IF "$(GUI)"=="UNX" || "$(USE_SHELL)"!="4nt"
 	@+cd $(PACKAGE_DIR) && chmod -R +rw * && $(TOUCH) $(UNTAR_FLAG_FILE)
 	@+cd $(PACKAGE_DIR) && find . -type d -exec chmod a+x {{}} \;
--- berkeleydb/makefile.mk	2004-10-28 17:24:48.000000000 +0530
+++ berkeleydb/makefile.mk	2005-02-10 08:42:15.890625000 +0530
@@ -105,7 +105,7 @@ OUT2CLASS=$(BUILD_DIR)$/db.jar
 EXT_USE_STLPORT=TRUE
 
 BUILD_DIR=build_win32
-.IF "$(COMEX)"=="8"
+.IF "$(COMEX)"=="8" || "$(COMEX)"=="10"
 CONFIGURE_DIR=build_win32
 CONFIGURE_ACTION=wdevenv Berkeley_DB Release
 .IF "$(USE_SHELL)"!="4nt"
@@ -113,9 +113,9 @@ BUILD_ACTION_SEP=;
 .ELSE # "$(USE_SHELL)"!="4nt"
 BUILD_ACTION_SEP=^
 .ENDIF # "$(USE_SHELL)"!="4nt"
-BUILD_ACTION=devenv Berkeley_DB.sln /build Release /project db_buildall /useenv $(BUILD_ACTION_SEP) devenv Berkeley_DB.sln /build Release /project db_java /useenv
+BUILD_ACTION=devenv Berkeley_DB.sln /build Release /project build_all /useenv $(BUILD_ACTION_SEP) devenv Berkeley_DB.sln /build Release /project db_java /useenv
 .ELSE
-BUILD_ACTION=msdev Berkeley_DB.dsw /useenv /MAKE "db_buildall - RELEASE" /MAKE "db_java - RELEASE"
+BUILD_ACTION=msdev Berkeley_DB.dsw /useenv /MAKE "build_all - RELEASE" /MAKE "db_java - RELEASE"
 .ENDIF
 
 OUT2LIB= \
