Index: sc/util/makefile.mk
===================================================================
RCS file: /cvs/sc/sc/util/makefile.mk,v
retrieving revision 1.33
diff -u -p -u -r1.33 makefile.mk
--- sc/util/makefile.mk	27 Jan 2006 15:52:35 -0000	1.33
+++ sc/util/makefile.mk	19 Apr 2006 13:08:14 -0000
@@ -182,6 +182,7 @@ LIB4FILES=	\
 	$(SLB)$/html.lib \
 	$(SLB)$/lotus.lib \
 	$(SLB)$/qpro.lib\
+	$(SLB)$/ecma.lib\
 	$(SLB)$/rtf.lib \
 	$(SLB)$/xml.lib \
 	$(SLB)$/accessibility.lib \


Index: sc/prj/build.lst
===================================================================
RCS file: /cvs/sc/sc/prj/build.lst,v
retrieving revision 1.11
diff -u -p -u -r1.11 build.lst
--- sc/prj/build.lst	27 Jan 2006 15:49:40 -0000	1.11
+++ sc/prj/build.lst	19 Apr 2006 13:08:47 -0000
@@ -39,6 +40,7 @@ sc	sc\source\filter\pch					get		-	all	s
 sc	sc\source\filter\excel					nmake	-	all	sc_excel sc_sdi NULL
 sc	sc\source\filter\lotus					nmake	-	all	sc_lotus sc_sdi NULL
 sc      sc\source\filter\qpro                                  nmake   -       all     sc_qpro sc_sdi NULL
+sc      sc\source\filter\ecma                                  nmake   -       all     sc_ecma sc_sdi NULL
 sc	sc\source\filter\starcalc				nmake	-	all	sc_scalc sc_sdi NULL
 sc	sc\source\filter\dif					nmake	-	all	sc_dif sc_sdi NULL
 sc	sc\source\filter\rtf					nmake	-	all	sc_rtf sc_sdi NULL
@@ -48,4 +50,4 @@ sc	sc\source\filter\xcl97					nmake	-	al
 sc	sc\addin\datefunc						nmake	-	all	sc_addfu sc_sdi NULL
 sc	sc\addin\rot13							nmake	-	all	sc_adrot sc_sdi NULL
 sc	sc\addin\util							nmake	-	all	sc_adutil sc_addfu sc_adrot sc_sdi NULL
-sc      sc\util                                                                 nmake   -       all     sc_util sc_addfu sc_adrot sc_adutil sc_app sc_attr sc_cctrl sc_cosrc sc_data sc_dbgui sc_dif sc_docsh sc_drfnc sc_excel sc_form sc_html sc_lotus sc_qpro sc_misc sc_name sc_nvipi sc_opt sc_page sc_rtf sc_scalc sc_style sc_tool sc_uisrc sc_undo sc_unobj sc_view sc_xcl97 sc_xml sc_acc sc_ftools sc_vba NULL
+sc      sc\util                                                                 nmake   -       all     sc_util sc_addfu sc_adrot sc_adutil sc_app sc_attr sc_cctrl sc_cosrc sc_data sc_dbgui sc_dif sc_docsh sc_drfnc sc_excel sc_form sc_html sc_lotus sc_ecma sc_qpro sc_misc sc_name sc_nvipi sc_opt sc_page sc_rtf sc_scalc sc_style sc_tool sc_uisrc sc_undo sc_unobj sc_view sc_xcl97 sc_xml sc_acc sc_ftools sc_vba NULL
