--- /dev/null	2006-04-07 20:25:18.000000000 +0100
+++ sc/source/filter/ecma/makefile.mk	2006-04-19 12:14:46.000000000 +0100
@@ -0,0 +1,64 @@
+#*************************************************************************
+#
+#   OpenOffice.org - a multi-platform office productivity suite
+#
+#   $RCSfile$
+#
+#   $Revision$
+#
+#   last change: $Author$ $Date$
+#
+#   The Contents of this file are made available subject to
+#   the terms of GNU Lesser General Public License Version 2.1.
+#
+#
+#     GNU Lesser General Public License Version 2.1
+#     =============================================
+#     Copyright 2005 by Sun Microsystems, Inc.
+#     901 San Antonio Road, Palo Alto, CA 94303, USA
+#
+#     This library is free software; you can redistribute it and/or
+#     modify it under the terms of the GNU Lesser General Public
+#     License version 2.1, as published by the Free Software Foundation.
+#
+#     This library is distributed in the hope that it will be useful,
+#     but WITHOUT ANY WARRANTY; without even the implied warranty of
+#     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#     Lesser General Public License for more details.
+#
+#     You should have received a copy of the GNU Lesser General Public
+#     License along with this library; if not, write to the Free Software
+#     Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+#     MA  02111-1307  USA
+#
+#*************************************************************************
+
+PRJ=..$/..$/..
+
+PRJNAME=sc
+TARGET=ecma
+
+PROJECTPCH4DLL=TRUE
+PROJECTPCH=filt_pch
+PROJECTPCHSOURCE=..\pch\filt_pch
+
+AUTOSEG=true
+
+
+# --- Settings -----------------------------------------------------
+
+.INCLUDE :  scpre.mk
+.INCLUDE :  settings.mk
+.INCLUDE :  sc.mk
+.INCLUDE :  $(PRJ)$/util$/makefile.pmk
+
+# --- Files --------------------------------------------------------
+
+CXXFILES = ecma.cxx
+
+SLOFILES = $(SLO)$/ecma.obj
+
+# --- Tagets -------------------------------------------------------
+
+.INCLUDE :  target.mk
+
--- /dev/null	2006-04-07 20:25:18.000000000 +0100
+++ sc/source/filter/ecma/ecma.cxx	2006-04-19 12:29:16.000000000 +0100
@@ -0,0 +1,67 @@
+#include <sal/config.h>
+#include <stdio.h>
+#include <sfx2/docfile.hxx>
+
+#include "global.hxx"
+#include "scerrors.hxx"
+#include "docpool.hxx" 
+#include "patattr.hxx"
+#include "filter.hxx"
+#include "document.hxx"
+#include "cell.hxx"
+#include "biff.hxx"
+#include <tools/stream.hxx>
+#include <com/sun/star/embed/ElementModes.hpp>
+#include <com/sun/star/container/XNameAccess.hpp>
+#include <comphelper/processfactory.hxx>
+#include <unotools/streamwrap.hxx>
+
+using namespace com::sun::star;
+
+static void
+recursiveDump (uno::Reference < embed::XStorage > xStorage, int nDepth = 0)
+{
+	uno::Reference< container::XNameAccess > xName ( xStorage, uno::UNO_QUERY );
+	if (!xName.is())
+		return;
+	uno::Sequence<rtl::OUString> aElements = xName->getElementNames();
+	for (int i = 0; i < aElements.getLength(); i++)
+	{
+		bool isDir = xStorage->isStorageElement(aElements[i]);
+		for (int j = 0; j < nDepth; j++)
+				fprintf (stderr, "\t");
+		fprintf (stderr, "Item '%s' (%s)\n",
+				 rtl::OUStringToOString (aElements[i], RTL_TEXTENCODING_UTF8).getStr(),
+				 isDir ? "storage" : "stream");
+		if (isDir)
+		{
+			uno::Reference < embed::XStorage > xSubStor;
+			xSubStor = xStorage->openStorageElement (aElements[i],
+													 embed::ElementModes::READ);
+			recursiveDump (xSubStor, nDepth+1);
+		}
+	}
+}
+
+FltError ScImportECMA( SfxMedium &rMedium, ScDocument *pDoc )
+{
+	fprintf (stderr, "ScImportECMA\n");
+
+	uno::Reference < embed::XStorage > xStorage;
+
+    FltError   eRet = eERR_OK;
+//    ScECMAReader aReader( rMedium, pDoc );
+//    eRet = aReader.import( pDoc );
+
+	xStorage = rMedium.GetStorage();
+	if ( !xStorage.is() )
+	{
+		fprintf (stderr, "No storage!\n");
+		return eERR_OPEN;
+	}
+
+	// Dump the contents:
+	recursiveDump (xStorage);
+
+    return eRet;
+}
