--- /dev/null	2006-04-07 20:25:18.000000000 +0100
+++ filter/source/config/fragments/filters/OfficeOpen.xcu	2006-04-19 13:57:17.000000000 +0100
@@ -0,0 +1,13 @@
+	<node oor:name="Office Open XML" oor:op="replace">
+		<prop oor:name="Flags"><value>IMPORT EXPORT TEMPLATE</value></prop>
+		<prop oor:name="UIComponent"/>
+		<prop oor:name="FilterService"/>
+		<prop oor:name="UserData"><value></value></prop>
+		<prop oor:name="FileFormatVersion"><value>6800</value></prop>
+		<prop oor:name="Type"><value>calc_OfficeOpen</value></prop>
+		<prop oor:name="TemplateName"/>
+		<prop oor:name="DocumentService"><value>com.sun.star.sheet.SpreadsheetDocument</value></prop>
+		<prop oor:name="UIName">
+			<value xml:lang="x-default">ECMA Office Open</value>
+		</prop>
+	</node>
--- /dev/null	2006-04-07 20:25:18.000000000 +0100
+++ filter/source/config/fragments/types/calc_OfficeOpen.xcu	2006-04-19 13:55:29.000000000 +0100
@@ -0,0 +1,12 @@
+	<node oor:name="calc_OfficeOpen" oor:op="replace">
+		<prop oor:name="DetectService"><value>com.sun.star.comp.calc.FormatDetector</value></prop>
+		<prop oor:name="URLPattern"/>
+		<prop oor:name="Extensions"><value>xlsx</value></prop>
+		<prop oor:name="MediaType"><value>application/vnd.ms-excel-FIXME</value></prop>
+		<prop oor:name="Preferred"><value>false</value></prop>
+		<prop oor:name="PreferredFilter"><value>Office Open XML</value></prop>
+		<prop oor:name="UIName">
+			<value>ECMA Office Open</value>
+		</prop>
+		<prop oor:name="ClipboardFormat"><value>OfficeOpen</value></prop>
+	</node>
Index: sc/source/ui/docshell/docsh.cxx
===================================================================
RCS file: /cvs/sc/sc/source/ui/docshell/docsh.cxx,v
retrieving revision 1.83
diff -u -p -u -r1.83 docsh.cxx
--- sc/source/ui/docshell/docsh.cxx	27 Jan 2006 15:51:57 -0000	1.83
+++ sc/source/ui/docshell/docsh.cxx	20 Apr 2006 10:51:09 -0000
@@ -185,6 +185,7 @@ static const sal_Char __FAR_DATA pFilter
 static const sal_Char __FAR_DATA pFilterHtml[]		= "HTML (StarCalc)";
 static const sal_Char __FAR_DATA pFilterHtmlWebQ[]	= "calc_HTML_WebQuery";
 static const sal_Char __FAR_DATA pFilterRtf[]		= "Rich Text Format (StarCalc)";
+static const sal_Char __FAR_DATA pFilterOOXml[]		= "ooxml";
 
 //----------------------------------------------------------------------
 
@@ -886,6 +888,14 @@ BOOL __EXPORT ScDocShell::ConvertFrom( S
 			bSetColWidths = TRUE;
 			bSetRowHeights = TRUE;
 		}
+		else if (aFltName.EqualsAscii(pFilterOOXml))
+		{
+			fprintf (stderr, "Load OO.o XML\n");
+            ScColumn::bDoubleAlloc = TRUE;
+            FltError eError = ScImportECMA( rMedium, &aDocument);
+            ScColumn::bDoubleAlloc = FALSE;
+			bRet = TRUE;
+		}
 		else if ( aFltName.EqualsAscii(pFilterExcel4) || aFltName.EqualsAscii(pFilterExcel5) ||
 				   aFltName.EqualsAscii(pFilterExcel95) || aFltName.EqualsAscii(pFilterExcel97) ||
 				   aFltName.EqualsAscii(pFilterEx4Temp) || aFltName.EqualsAscii(pFilterEx5Temp) ||
