From a885df255a3b35a03b18c0e841cbfd6456db8b29 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:59:14 +0200
Subject: [PATCH 324/768] default-evo2-ab.diff

---
 extensions/source/abpilot/abspilot.cxx |    7 +++----
 1 files changed, 3 insertions(+), 4 deletions(-)

diff --git extensions/source/abpilot/abspilot.cxx extensions/source/abpilot/abspilot.cxx
index f186394..7849a14 100644
--- extensions/source/abpilot/abspilot.cxx
+++ extensions/source/abpilot/abspilot.cxx
@@ -115,13 +115,12 @@ namespace abp
         m_pCancel->SetClickHdl( LINK( this, OAddessBookSourcePilot, OnCancelClicked) );
 
         // some initial settings
+#ifdef UNX
 #ifdef MACOSX
         m_aSettings.eType = AST_MACAB;
-#elif WITH_MOZILLA
-#ifdef UNX
-        m_aSettings.eType = AST_MORK;
 #else
-        m_aSettings.eType = AST_OE;
+// FIXME: if KDE use KAB instead
+        m_aSettings.eType = AST_EVOLUTION;
 #endif
 #else
         m_aSettings.eType = AST_OTHER;
-- 
1.7.0.1

