From 0f233c8c2034ba74b464a8e4b3f538c7ebbda28c Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 16:59:14 +0200
Subject: [PATCH 361/878] default-evo2-ab.diff

---
 extensions/source/abpilot/abspilot.cxx |    7 +++----
 1 files changed, 3 insertions(+), 4 deletions(-)

diff --git a/extensions/source/abpilot/abspilot.cxx b/extensions/source/abpilot/abspilot.cxx
index efb24e5..cd88cfe 100644
--- a/extensions/source/abpilot/abspilot.cxx
+++ b/extensions/source/abpilot/abspilot.cxx
@@ -119,13 +119,12 @@ namespace abp
         m_pCancel->SetClickHdl( LINK( this, OAddessBookSourcePilot, OnCancelClicked) );
 
         // some initial settings
+#ifdef UNX
 #ifdef MACOSX
         m_aSettings.eType = AST_MACAB;
-#elif WITH_MOZILLA
-#ifdef UNX
-        m_aSettings.eType = AST_MORK;
 #else
-        m_aSettings.eType = AST_OE;
+// FIXME: if KDE use KAB instead
+        m_aSettings.eType = AST_EVOLUTION;
 #endif
 #else
         m_aSettings.eType = AST_OTHER;
-- 
1.7.0.1

