---
 extensions/source/abpilot/abspilot.cxx |    7 +++----
 1 files changed, 3 insertions(+), 4 deletions(-)

diff --git extensions/source/abpilot/abspilot.cxx extensions/source/abpilot/abspilot.cxx
index 01fb844..508f79f 100644
--- extensions/source/abpilot/abspilot.cxx
+++ extensions/source/abpilot/abspilot.cxx
@@ -115,13 +115,12 @@ namespace abp
         m_pCancel->SetClickHdl( LINK( this, OAddessBookSourcePilot, OnCancelClicked) );
 
         // some initial settings
+#ifdef UNX
 #ifdef MACOSX
         m_aSettings.eType = AST_MACAB;
-#elif WITH_MOZILLA
-#ifdef UNX
-        m_aSettings.eType = AST_MORK;
 #else
-        m_aSettings.eType = AST_OE;
+// FIXME: if KDE use KAB instead
+        m_aSettings.eType = AST_EVOLUTION;
 #endif
 #else
         m_aSettings.eType = AST_OTHER;
-- 
1.7.0.1

