diff -urd psprint/source/printer/cupsmgr.cxx psprint-fixppdcrash/source/printer/cupsmgr.cxx
--- psprint/source/printer/cupsmgr.cxx	2004-11-09 16:37:51.000000000 +0000
+++ psprint-fixppdcrash/source/printer/cupsmgr.cxx	2004-11-24 21:42:59.000000000 +0000
@@ -601,14 +601,20 @@
                     // remember the default context for later use
                     PPDContext& rContext = m_aDefaultContexts[ aPrinter ];
                     rContext.setParser( pNewParser );
-                    for( int i = 0; i < pPPD->num_groups; i++ )
-                        updatePrinterContextInfo( pPPD->groups + i, rContext );
+					if(pPPD)
+					{
+						for( int i = 0; i < pPPD->num_groups; i++ )
+							updatePrinterContextInfo( pPPD->groups + i, rContext );
+					}
                     
                     rInfo.m_pParser = pNewParser;
                     rInfo.m_aContext = rContext;
                     
-                    // clean up the mess
-                    m_pCUPSWrapper->ppdClose( pPPD );
+					if(pPPD)
+					{
+						// clean up the mess
+						m_pCUPSWrapper->ppdClose( pPPD );
+					}
                 }
 #if OSL_DEBUG_LEVEL > 1
                 else
