--- svtools/prj/d.lst	2005-02-02 19:31:35.000000000 +0530
+++ svtools/prj/d.lst	2005-02-24 20:53:25.000000000 +0530
@@ -259,6 +259,7 @@ hedabu: ..\inc\custritm.hxx %_DEST%\inc%
 hedabu: ..\inc\ehdl.hxx %_DEST%\inc%_EXT%\svtools\ehdl.hxx
 hedabu: ..\inc\svtools.hrc %_DEST%\inc%_EXT%\svtools\svtools.hrc
 hedabu: ..\inc\svpatch.hrc %_DEST%\inc%_EXT%\svtools\svpatch.hrc
+hedabu: ..\source\dialogs\filedlg2.hrc %_DEST%\inc%_EXT%\svtools\filedlg2.hrc
 
 hedabu: ..\source\xmlfilt\attrlist.hxx %_DEST%\inc%_EXT%\svtools\attrlist.hxx
 hedabu: ..\source\xmlfilt\i18nmap.hxx %_DEST%\inc%_EXT%\svtools\i18nmap.hxx
