--- psprint/source/fontmanager/fontmanager.cxx	2004-10-13 09:22:36.000000000 +0100
+++ psprint-patched/source/fontmanager/fontmanager.cxx	2004-11-25 18:10:05.000000000 +0000
@@ -829,10 +829,10 @@
 
         if( m_aEncoding == RTL_TEXTENCODING_DONTKNOW )
             m_aEncoding = nAdobeEncoding == 1 ?
-                RTL_TEXTENCODING_ADOBE_STANDARD : RTL_TEXTENCODING_SYMBOL;
+                RTL_TEXTENCODING_MS_1252 : RTL_TEXTENCODING_SYMBOL;
     }
     else if( m_aEncoding == RTL_TEXTENCODING_DONTKNOW )
-        m_aEncoding = RTL_TEXTENCODING_ADOBE_STANDARD;
+        m_aEncoding = RTL_TEXTENCODING_MS_1252;
 
     // try to parse the font name and decide wether it might be a
     // japanese font. Who invented this PITA ?
@@ -1669,10 +1669,7 @@
     const char* pEnc = rtl_getBestUnixCharsetFromTextEncoding( pFont->m_aEncoding );
     if( ! pEnc )
     {
-        if( pFont->m_aEncoding == RTL_TEXTENCODING_ADOBE_STANDARD )
-            pEnc = "adobe-standard";
-        else
-            pEnc = "iso8859-1";
+        pEnc = "iso8859-1";
     }
     aXLFD .append( pEnc );
 
