--- vcl/unx/gtk/window/gtkframe.cxx.menubar	2004-10-19 15:35:49.000000000 -0400
+++ vcl/unx/gtk/window/gtkframe.cxx	2004-10-24 01:19:50.439496320 -0400
@@ -1294,19 +1294,27 @@
     aStyleSet.SetHighlightTextColor( aHighlightTextColor );

     // menu colors
+    GtkWidget* pMenuBar = gtk_menu_bar_new();
+    GtkWidget* pMenuBarMenuItem = gtk_menu_item_new_with_label( "blah" );
     GtkWidget* pMenu = gtk_menu_new();
     GtkWidget* pMenuItem = gtk_menu_item_new_with_label( "blah" );
+    gtk_menu_shell_append( GTK_MENU_SHELL( pMenuBar ), pMenuBarMenuItem );
+    gtk_menu_item_set_submenu( GTK_MENU_ITEM( pMenuBarMenuItem ), pMenu );
     gtk_menu_shell_append( GTK_MENU_SHELL( pMenu ), pMenuItem );

+    gtk_widget_ensure_style( pMenuBar );
+    gtk_widget_ensure_style( pMenuBarMenuItem );
     gtk_widget_ensure_style( pMenu );
     gtk_widget_ensure_style( pMenuItem );
+    GtkStyle* pMenuBarStyle = gtk_rc_get_style( pMenuBar );
     GtkStyle* pMenuStyle = gtk_rc_get_style( pMenu );
     GtkStyle* pMenuItemStyle = gtk_rc_get_style( pMenuItem );
     GtkStyle* pMenuTextStyle = gtk_rc_get_style( gtk_bin_get_child( GTK_BIN( pMenuItem )));

+    aBackColor = getColor ( pMenuBarStyle->bg[GTK_STATE_NORMAL] );
+    aStyleSet.SetMenuBarColor( aBackColor );
     aBackColor = getColor( pMenuStyle->bg[GTK_STATE_NORMAL] );
     aStyleSet.SetMenuColor( aBackColor );
-    aStyleSet.SetMenuBarColor( aBackColor );
     aTextColor = getColor( pMenuTextStyle->text[GTK_STATE_NORMAL] );
     aStyleSet.SetMenuTextColor( aTextColor );
     aHighlightColor = getColor( pMenuItemStyle->bg[ GTK_STATE_SELECTED ] );
