--- sc/source/ui/vba/vbarange.cxx	2008-07-29 23:21:38.000000000 +0100
+++ sc/source/ui/vba/vbarange.cxx	2008-07-30 11:54:23.000000000 +0100
@@ -2995,10 +2995,10 @@ void updateTableSortField( const uno::Re
 
 		// make sure that upper left poing of key range is within the
 		// parent range
-		if (  colRowKeyAddress.StartColumn >= parentRangeAddress.StartColumn &&
-			colRowKeyAddress.StartColumn <= parentRangeAddress.EndColumn  &&
+		if (  ( !bIsSortColumn && colRowKeyAddress.StartColumn >= parentRangeAddress.StartColumn &&
+			colRowKeyAddress.StartColumn <= parentRangeAddress.EndColumn ) || ( bIsSortColumn && 
 			colRowKeyAddress.StartRow >= parentRangeAddress.StartRow &&
-			colRowKeyAddress.StartRow <= parentRangeAddress.EndRow  )
+			colRowKeyAddress.StartRow <= parentRangeAddress.EndRow  ) )
 		{
 			//determine col/row index
 			if ( bIsSortColumn )
