--- sc/source/filter/xcl97/xcl97esc.cxx.old	2010-07-30 16:25:42.000000000 +0200
+++ sc/source/filter/xcl97/xcl97esc.cxx	2010-07-30 19:54:00.000000000 +0200
@@ -256,7 +256,8 @@ EscherExHostAppData* XclEscherEx::StartS
         {
             // #107540# ignore permanent note shapes
             // #i12190# do not ignore callouts (do not filter by object type ID)
-            pCurrXclObj = new XclObjAny( mrObjMgr );   // just a metafile
+            // this line will be solved better by cws-vbasupportdev300.diff
+            pCurrXclObj = new XclObjAny( mrObjMgr, rxShape );   // just a metafile
         }
     }
     if ( pCurrXclObj )
