diff --git toolkit/inc/toolkit/controls/dialogcontrol.hxx toolkit/inc/toolkit/controls/dialogcontrol.hxx
index 9ae8442..13ccc1a 100644
--- toolkit/inc/toolkit/controls/dialogcontrol.hxx
+++ toolkit/inc/toolkit/controls/dialogcontrol.hxx
@@ -233,6 +233,8 @@ typedef ::cppu::ImplHelper2	< ::com::sun::star::container::XContainerListener
 class UnoDialogContainerControl : public UnoControlContainer, public UnoDialogContainerControl_IBase
 {
 protected:
+    bool                                                                        mbSizeModified;
+    bool                                                                        mbPosModified;
     ::com::sun::star::uno::Reference< ::com::sun::star::awt::XTabController >	mxTabController;
     void		ImplInsertControl( ::com::sun::star::uno::Reference< ::com::sun::star::awt::XControlModel >& rxModel, const ::rtl::OUString& rName );
     void		ImplRemoveControl( ::com::sun::star::uno::Reference< ::com::sun::star::awt::XControlModel >& rxModel );
@@ -283,8 +285,6 @@ private:
     ::com::sun::star::uno::Reference< ::com::sun::star::awt::XMenuBar >			mxMenuBar;
     TopWindowListenerMultiplexer	                                            maTopWindowListeners;
     bool                                                                        mbWindowListener;
-    bool                                                                        mbSizeModified;
-    bool                                                                        mbPosModified;
     ::com::sun::star::uno::Reference< ::com::sun::star::graphic::XGraphicObject > mxGrfObj;
     ::com::sun::star::uno::Reference< ::com::sun::star::util::XModifyListener > mxListener;
  
diff --git toolkit/source/controls/dialogcontrol.cxx toolkit/source/controls/dialogcontrol.cxx
index e9f24e6..92d510d 100644
--- toolkit/source/controls/dialogcontrol.cxx
+++ toolkit/source/controls/dialogcontrol.cxx
@@ -2181,7 +2181,9 @@ uno::Reference< beans::XPropertySetInfo > UnoFrameModel::getPropertySetInfo(  )
 //	----------------------------------------------------
 //	class DialogContainerControl
 //	----------------------------------------------------
-UnoDialogContainerControl::UnoDialogContainerControl()
+UnoDialogContainerControl::UnoDialogContainerControl() :
+    mbSizeModified(false),
+    mbPosModified(false)
 {
 	maComponentInfos.nWidth = 280;
 	maComponentInfos.nHeight = 400;
@@ -2254,7 +2256,6 @@ void UnoDialogContainerControl::ImplSetPosSize( Reference< XControl >& rxCtrl )
     xP->getPropertyValue( ::rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( "PositionY" ) ) ) >>= nY;
     xP->getPropertyValue( ::rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( "Width" ) ) ) >>= nWidth;
     xP->getPropertyValue( ::rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( "Height" ) ) ) >>= nHeight;
-
 	MapMode aMode( MAP_APPFONT );
 	sal_Bool bVBAForm = sal_False;
 	Reference< XPropertySet > xDlgModelProps( getModel(), UNO_QUERY );
@@ -2484,9 +2485,12 @@ void UnoDialogContainerControl::ImplModelPropertiesChanged( const Sequence< Prop
             {
                 if ( bOwnModel )
                 {
+                    if ( !mbPosModified && !mbSizeModified )
+                    {
                         // Don't set new pos/size if we get new values from window listener
                         Reference< XControl > xThis( (XAggregation*)(::cppu::OWeakAggObject*)this, UNO_QUERY );
                         ImplSetPosSize( xThis );
+                    }
                 }
                 else
                 {
@@ -2554,9 +2558,7 @@ void SAL_CALL UnoDialogContainerControl::changesOccurred( const ChangesEvent& )
 
 UnoDialogControl::UnoDialogControl() :
     maTopWindowListeners( *this ),
-    mbWindowListener(false),
-    mbSizeModified(false),
-    mbPosModified(false)
+    mbWindowListener(false)
 {
     maComponentInfos.nWidth = 300;
     maComponentInfos.nHeight = 450;
