diff --git sc/source/core/tool/interpr4.cxx sc/source/core/tool/interpr4.cxx
index 35177aa..ca045f0 100644
--- sc/source/core/tool/interpr4.cxx
+++ sc/source/core/tool/interpr4.cxx
@@ -50,6 +50,7 @@
 
 #include <com/sun/star/table/XCellRange.hpp>
 #include <com/sun/star/sheet/XSheetCellRange.hpp>
+#include <com/sun/star/script/vba/XVBACompatibility.hpp>
 #include <comphelper/processfactory.hxx>
 
 #include "interpre.hxx"
@@ -2706,8 +2707,14 @@ uno::Any lcl_getSheetModule( const uno::Reference<table::XCellRange>& xCellRange
     if ( pBasMgr && pBasMgr->GetName().Len() )
     {
         String sProj = String( RTL_CONSTASCII_USTRINGPARAM( "Standard" ) );
-        if ( pDok->GetDocumentShell()->GetBasicManager()->GetName().Len() )
-            sProj = pDok->GetDocumentShell()->GetBasicManager()->GetName();
+        try
+        {
+            uno::Reference< beans::XPropertySet > xProps( pDok->GetDocumentShell()->GetModel(), uno::UNO_QUERY_THROW );
+            uno::Reference< script::vba::XVBACompatibility > xVBAMode( xProps->getPropertyValue( rtl::OUString( RTL_CONSTASCII_USTRINGPARAM("BasicLibraries") ) ), uno::UNO_QUERY_THROW );
+            sProj = xVBAMode->getProjectName();
+        }
+        catch( uno::Exception& ){}
+
         StarBASIC* pBasic = pDok->GetDocumentShell()->GetBasicManager()->GetLib( sProj );
         if ( pBasic )
         {
diff --git sc/source/ui/docshell/macromgr.cxx sc/source/ui/docshell/macromgr.cxx
index 97623f3..5eb823a 100644
--- sc/source/ui/docshell/macromgr.cxx
+++ sc/source/ui/docshell/macromgr.cxx
@@ -41,6 +41,7 @@
 #include "sfx2/objsh.hxx"
 #include "cell.hxx"
 #include <com/sun/star/container/XContainer.hpp>
+#include <com/sun/star/script/vba/XVBACompatibility.hpp>
 
 #include <list>
 
@@ -154,8 +155,16 @@ void ScMacroManager::InitUserFuncData()
 
     Reference< container::XContainer > xModuleContainer;
     SfxObjectShell* pShell = mpDoc->GetDocumentShell();
-    if ( pShell && pShell->GetBasicManager()->GetName().Len() > 0 )
-        sProjectName = pShell->GetBasicManager()->GetName();
+    if ( pShell )
+    { 
+        try
+        {
+            uno::Reference< beans::XPropertySet > xProps( pShell->GetModel(), uno::UNO_QUERY_THROW );
+            uno::Reference< script::vba::XVBACompatibility > xVBAMode( xProps->getPropertyValue( rtl::OUString( RTL_CONSTASCII_USTRINGPARAM("BasicLibraries") ) ), uno::UNO_QUERY_THROW );
+            sProjectName = xVBAMode->getProjectName(); 
+        }
+        catch( uno::Exception& ) {}
+    }
     try
     { 
         Reference< script::XLibraryContainer > xLibraries( pShell->GetBasicContainer(), uno::UNO_QUERY_THROW );
diff --git sw/source/core/unocore/unocoll.cxx sw/source/core/unocore/unocoll.cxx
index 07cb0f6..ae19ad0 100644
--- sw/source/core/unocore/unocoll.cxx
+++ sw/source/core/unocore/unocoll.cxx
@@ -86,6 +86,7 @@
 #include <com/sun/star/script/ModuleType.hpp>
 #include <com/sun/star/script/ScriptEventDescriptor.hpp>
 #include <com/sun/star/script/vba/XVBAModuleInfo.hpp>
+#include <com/sun/star/script/vba/XVBACompatibility.hpp>
 #include <vbahelper/vbaaccesshelper.hxx>
 #include <basic/basmgr.hxx>
 #include <comphelper/processfactory.hxx>
@@ -115,8 +116,9 @@ public:
                 uno::Reference< beans::XPropertySet > xProps( mpDocShell->GetModel(), uno::UNO_QUERY_THROW );
                 uno::Reference< container::XNameAccess > xLibContainer( xProps->getPropertyValue( rtl::OUString( RTL_CONSTASCII_USTRINGPARAM("BasicLibraries") ) ), uno::UNO_QUERY_THROW );
 		rtl::OUString sProjectName( rtl::OUString( RTL_CONSTASCII_USTRINGPARAM("Standard") ) );
-                if ( mpDocShell->GetBasicManager()->GetName().Len() )
-                    sProjectName =  mpDocShell->GetBasicManager()->GetName();
+                uno::Reference< script::vba::XVBACompatibility > xVBAMode( xLibContainer, uno::UNO_QUERY );
+                if ( xVBAMode.is() ) 
+                    sProjectName =  xVBAMode->getProjectName();
     
                 uno::Reference< container::XNameAccess > xLib( xLibContainer->getByName( sProjectName ), uno::UNO_QUERY_THROW );
                 uno::Sequence< rtl::OUString > sModuleNames = xLib->getElementNames();
diff --git vbahelper/source/vbahelper/vbahelper.cxx vbahelper/source/vbahelper/vbahelper.cxx
index c02569d..e9c6f23 100644
--- vbahelper/source/vbahelper/vbahelper.cxx
+++ vbahelper/source/vbahelper/vbahelper.cxx
@@ -33,6 +33,7 @@
 #include <com/sun/star/frame/XModel2.hpp>
 #include <com/sun/star/frame/XNotifyingDispatch.hpp>
 #include <com/sun/star/script/XDefaultProperty.hpp>
+#include <com/sun/star/script/vba/XVBACompatibility.hpp>
 #include <com/sun/star/uno/XComponentContext.hpp>
 #include <com/sun/star/lang/XMultiComponentFactory.hpp>
 #include <com/sun/star/lang/XUnoTunnel.hpp>
@@ -1271,6 +1272,13 @@ UserFormGeometryHelper::UserFormGeometryHelper( const uno::Reference< uno::XComp
             if ( pShell )
             {
                 rtl::OUString sProj( RTL_CONSTASCII_USTRINGPARAM("Standard") );
+                try
+                {
+                    uno::Reference< beans::XPropertySet > xProps( pShell->GetModel(), uno::UNO_QUERY_THROW );
+                    uno::Reference< script::vba::XVBACompatibility > xVBAMode( xProps->getPropertyValue( rtl::OUString( RTL_CONSTASCII_USTRINGPARAM("BasicLibraries") ) ), uno::UNO_QUERY_THROW );
+                    sProj = xVBAMode->getProjectName();
+                }
+                catch( uno::Exception& ){}
                 BasicManager* pBasMgr = pShell->GetBasicManager();
                 if ( pBasMgr && pBasMgr->GetName().Len() )
                     sProj = pShell->GetBasicManager()->GetName();
