diff --git oox/source/ole/vbamodule.cxx oox/source/ole/vbamodule.cxx
index 2ce7c3e..c49ac1e 100755
--- oox/source/ole/vbamodule.cxx
+++ oox/source/ole/vbamodule.cxx
@@ -332,7 +332,11 @@ void VbaModule::importSourceCode( StorageBase& rVbaStrg,
     // insert the module into the passed Basic library
     try
     {
-        rxBasicLib->insertByName( maName, Any( aSourceCode.makeStringAndClear() ) );
+        // sheet modules are 'pre-created'
+        if ( rxBasicLib->hasByName( maName ) )
+            rxBasicLib->replaceByName( maName, Any( aSourceCode.makeStringAndClear() ) );
+        else
+            rxBasicLib->insertByName( maName, Any( aSourceCode.makeStringAndClear() ) );
     }
     catch( Exception& )
     {
