diff --git sc/source/filter/excel/xiescher.cxx sc/source/filter/excel/xiescher.cxx
index 35336f7..e708d07 100644
--- sc/source/filter/excel/xiescher.cxx
+++ sc/source/filter/excel/xiescher.cxx
@@ -539,8 +539,9 @@ void XclImpDrawObjBase::PreProcessSdrObject( XclImpDffConverter& rDffConv, SdrOb
     }
 
     // macro and hyperlink
-    if( mbSimpleMacro && ((maMacroName.Len() > 0) ||
- (maHyperlink.Len() > 0)) )
+    // removed oracle/sun check for mbSimpleMacro ( no idea what its for )
+    if( (maMacroName.Len() > 0 ) ||
+ (maHyperlink.Len() > 0) )
     {
         if( ScMacroInfo* pInfo = ScDrawLayer::GetMacroInfo( &rSdrObj, TRUE ) )
         {
diff --git sc/source/filter/excel/xltools.cxx sc/source/filter/excel/xltools.cxx
index 51146d5..e703159 100644
--- sc/source/filter/excel/xltools.cxx
+++ sc/source/filter/excel/xltools.cxx
@@ -695,7 +695,7 @@ void XclTools::SkipSubStream( XclImpStream& rStrm )
 
 // Basic macro names ----------------------------------------------------------
 
-const OUString XclTools::maSbMacroPrefix( RTL_CONSTASCII_USTRINGPARAM( "vnd.sun.star.script:Standard." ) );
+const OUString XclTools::maSbMacroPrefix( RTL_CONSTASCII_USTRINGPARAM( "vnd.sun.star.script:" ) );
 const OUString XclTools::maSbMacroSuffix( RTL_CONSTASCII_USTRINGPARAM( "?language=Basic&location=document" ) );
 
 OUString XclTools::GetSbMacroUrl( const String& rMacroName, SfxObjectShell* pDocShell )
@@ -727,7 +727,10 @@ String XclTools::GetXclMacroName( const OUString& rSbMacroUrl )
     sal_Int32 nMacroNameLen = nSbMacroUrlLen - maSbMacroPrefix.getLength() - maSbMacroSuffix.getLength();
     if( (nMacroNameLen > 0) && rSbMacroUrl.matchIgnoreAsciiCase( maSbMacroPrefix, 0 ) &&
             rSbMacroUrl.matchIgnoreAsciiCase( maSbMacroSuffix, nSbMacroUrlLen - maSbMacroSuffix.getLength() ) )
-        return rSbMacroUrl.copy( maSbMacroPrefix.getLength(), nMacroNameLen );
+    {
+        sal_Int32 nPrjDot = rSbMacroUrl.indexOf( '.', maSbMacroPrefix.getLength() ) + 1;
+        return rSbMacroUrl.copy( nPrjDot, nSbMacroUrlLen - nPrjDot - maSbMacroSuffix.getLength() );
+    }
     return String::EmptyString();
 }
 
