diff --git basic/source/comp/exprtree.cxx basic/source/comp/exprtree.cxx
index 1c9ad8f..e12496c 100644
--- basic/source/comp/exprtree.cxx
+++ basic/source/comp/exprtree.cxx
@@ -574,7 +574,11 @@ SbiExprNode* SbiExpression::Unary()
             eTok = NEG;
         case NOT:
             pParser->Next();
-            pNd = new SbiExprNode( pParser, Unary(), eTok, NULL );
+            // process something like "Do While Not "foo"="" "
+            if( pParser->IsVBASupportOn() )
+                pNd = new SbiExprNode( pParser, Like(), eTok, NULL );
+            else    
+                pNd = new SbiExprNode( pParser, Unary(), eTok, NULL );
             break;
         case PLUS:
             pParser->Next();
