diff --git sw/source/ui/vba/makefile.mk sw/source/ui/vba/makefile.mk
index 741ad0c..47525f3 100644
--- sw/source/ui/vba/makefile.mk
+++ sw/source/ui/vba/makefile.mk
@@ -46,7 +46,7 @@ dummy:
 .ENDIF
  
 INCPRE=$(INCCOM)$/$(TARGET)
-CDEFS+=-DVBA_OOBUILD_HACK
+CDEFS+=-DVBA_OOBUILD_HACK -DFIELDPATCH
 # ------------------------------------------------------------------
 
 SLOFILES= \
@@ -119,7 +119,10 @@ SLOFILES= \
 				$(SLO)$/vbarevisions.obj \
 				$(SLO)$/vbaframe.obj \
 				$(SLO)$/vbaframes.obj \
+				$(SLO)$/vbaformfield.obj \
+				$(SLO)$/vbaformfields.obj \
 				$(SLO)$/vbaeventshelper.obj \
+				$(SLO)$/vbacheckbox.obj \
 
 # --- Targets ------------------------------------------------------
 
