--- vbahelper/source/msforms/vbamultipage.cxx	2010-02-05 10:43:43.000000000 +0000
+++ ../../build-good/ooo320-m8/vbahelper/source/msforms//vbamultipage.cxx	2010-02-05 10:20:38.000000000 +0000
@@ -92,8 +92,11 @@ void SAL_CALL
 ScVbaMultiPage::setValue( const sal_Int32 _value ) throw (::com::sun::star::uno::RuntimeException)
 {
     // Openoffice 1 based tab index
-   sal_Int32 nVal = _value + 1;
+    sal_Int32 nVal = _value + 1;
+    sal_Int32 nOldVal = getValue();
     m_xProps->setPropertyValue( SVALUE, uno::makeAny( nVal ) );
+    if ( nVal != nOldVal )
+        fireChangeEvent();  
 }
 
 
