diff --git scripting/source/vbaevents/makefile.mk scripting/source/vbaevents/makefile.mk
index 96f4e3d..3055ba7 100644
--- scripting/source/vbaevents/makefile.mk
+++ scripting/source/vbaevents/makefile.mk
@@ -1,6 +1,6 @@
 PRJ=..$/..
 
-PRJNAME=vbaevents
+PRJNAME=scripting
 TARGET=vbaevents
 .IF "$(ENABLE_VBA)"!="YES"
 dummy:
@@ -19,14 +19,6 @@ DLLPRE =
 
 # ------------------------------------------------------------------
 
-ALLTAR : \
-        $(MISC)$/$(TARGET).don \
-        $(SLOTARGET)
-
-$(MISC)$/$(TARGET).don : $(SOLARBINDIR)$/oovbaapi.rdb
-        +$(CPPUMAKER) -O$(OUT)$/inc -BUCR $(SOLARBINDIR)$/oovbaapi.rdb -X$(SOLARBINDIR)$/types.rdb && echo > $@
-        echo $@
-# ------------------------------------------------------------------
 
 #.INCLUDE :  ..$/cppumaker.mk
 
@@ -57,3 +49,14 @@ SHL1LIBS=$(SLB)$/$(TARGET).lib
 # --- Targets ------------------------------------------------------
 
 .INCLUDE :	target.mk
+
+# ------------------------------------------------------------------
+
+ALLTAR : \
+        $(MISC)$/$(TARGET).don \
+
+$(SLOFILES) : $(MISC)$/$(TARGET).don
+
+$(MISC)$/$(TARGET).don : $(SOLARBINDIR)$/oovbaapi.rdb
+        +$(CPPUMAKER) -O$(INCCOM)$/$(TARGET) -BUCR $(SOLARBINDIR)$/oovbaapi.rdb -X$(SOLARBINDIR)$/types.rdb && echo > $@
+        echo $@
