diff --git basic/source/runtime/methods1.cxx basic/source/runtime/methods1.cxx
index 9d3b51a..e429217 100644
--- basic/source/runtime/methods1.cxx
+++ basic/source/runtime/methods1.cxx
@@ -532,11 +532,18 @@ RTLFUNC(DoEvents)
     (void)pBasic;
     (void)bWrite;
     (void)rPar;
+// don't undstand what upstream are up to
+// we already process application events etc. in between
+// basic runtime pcode ( on a timed basis )
+#if 0
     Timer aTimer;
     aTimer.SetTimeout( 1 );
     aTimer.Start();
     while ( aTimer.IsActive() )
         Application::Yield();
+#endif
+    // always return 0
+    rPar.Get(0)->PutInteger( 0 );
 }
 
 RTLFUNC(GetGUIVersion)
diff --git basic/source/runtime/stdobj.cxx basic/source/runtime/stdobj.cxx
index 38b995d..89de3b0 100644
--- basic/source/runtime/stdobj.cxx
+++ basic/source/runtime/stdobj.cxx
@@ -260,7 +260,7 @@ static Methods aMethods[] = {
 { "Dir",            SbxSTRING,    2 | _FUNCTION, RTLNAME(Dir),0             },
   { "FileSpec",     SbxSTRING,        _OPT, NULL,0 },
   { "attrmask",     SbxINTEGER,       _OPT, NULL,0 },
-{ "DoEvents",		SbxEMPTY,     _FUNCTION, RTLNAME(DoEvents),0			},
+{ "DoEvents",		SbxINTEGER,     _FUNCTION, RTLNAME(DoEvents),0			},
 { "DumpAllObjects", SbxEMPTY,     2 | _SUB, RTLNAME(DumpAllObjects),0       },
   { "FileSpec",     SbxSTRING, 0,NULL,0 },
   { "DumpAll",      SbxINTEGER,		  _OPT, NULL,0 },
