diff --git toolkit/source/awt/vclxwindows.cxx toolkit/source/awt/vclxwindows.cxx
index dc32c06..7b2f801 100644
--- toolkit/source/awt/vclxwindows.cxx
+++ toolkit/source/awt/vclxwindows.cxx
@@ -1181,16 +1181,6 @@ void VCLXRadioButton::setProperty( const ::rtl::OUString& PropertyName, const ::
                         pButton->Check( b );
                     else
                         pButton->SetState( b );
-					// If VBA - need to be able to test
-					// simulate click event
-					if ( maActionListeners.getLength() )
-					{
-						::com::sun::star::awt::ActionEvent aEvent;
-						aEvent.Source = (::cppu::OWeakObject*)this;
-						aEvent.ActionCommand = maActionCommand;
-						maActionListeners.actionPerformed( aEvent );
-					}
-					
                 }
             }
             break;
@@ -1953,11 +1943,8 @@ void VCLXListBox::setProperty( const ::rtl::OUString& PropertyName, const ::com:
                 ::com::sun::star::uno::Sequence< ::rtl::OUString> aItems;
                 if ( Value >>= aItems )
                 {
-					sal_Int16 nElem  = pListBox->GetEntryCount();
                     pListBox->Clear();
                     addItems( aItems, 0 );
-					if ( aItems.getLength() == 0 && nElem && maItemListeners.getLength() )
-						ImplCallItemListeners();
                 }
             }
             break;
@@ -1970,11 +1957,7 @@ void VCLXListBox::setProperty( const ::rtl::OUString& PropertyName, const ::com:
                         pListBox->SelectEntryPos( --n, sal_False );
 
                     if ( aItems.getLength() )
-                    {			
                         selectItemsPos( aItems, sal_True );
-                       if ( maItemListeners.getLength() )
-                           ImplCallItemListeners();
-                    }
                     else
                         pListBox->SetNoSelection();
 
