diff --git basic/source/runtime/runtime.cxx basic/source/runtime/runtime.cxx
index 69c75b3..93a8afd 100644
--- basic/source/runtime/runtime.cxx
+++ basic/source/runtime/runtime.cxx
@@ -294,7 +294,6 @@ SbiInstance::SbiInstance( StarBASIC* p )
 	nErl	 = 0;
 	bReschedule = TRUE;
 	bCompatibility = FALSE;
-	SbxErrObject::getUnoErrObject()->Clear();
 }
 
 SbiInstance::~SbiInstance()
diff --git basic/source/runtime/step0.cxx basic/source/runtime/step0.cxx
index fa66194..c43931d 100644
--- basic/source/runtime/step0.cxx
+++ basic/source/runtime/step0.cxx
@@ -1132,6 +1132,7 @@ void SbiRuntime::StepNOERROR()
 	pInst->nErr = 0L;
 	pInst->nErl = 0;
 	nError = 0L;
+	SbxErrObject::getUnoErrObject()->Clear();
 	bError = FALSE;
 }
 
@@ -1140,6 +1141,9 @@ void SbiRuntime::StepNOERROR()
 void SbiRuntime::StepLEAVE()
 {
 	bRun = FALSE;
+        // If VBA and we are leaving an ErrorHandler then clear the error ( it's been processed )
+	if ( bInError && pError )
+	    SbxErrObject::getUnoErrObject()->Clear();
 }
 
 void SbiRuntime::StepCHANNEL()	  	// TOS = Kanalnummer
diff --git basic/source/runtime/step1.cxx basic/source/runtime/step1.cxx
index 57d7f02..d6dc7d0 100644
--- basic/source/runtime/step1.cxx
+++ basic/source/runtime/step1.cxx
@@ -38,6 +38,7 @@
 #include "iosys.hxx"
 #include "image.hxx"
 #include "sbunoobj.hxx"
+#include "errobject.hxx"
 
 bool checkUnoObjectType( SbUnoObject* refVal, 
 	const String& aClass );
@@ -374,6 +375,7 @@ void SbiRuntime::StepERRHDL( UINT32 nOp1 )
 	pInst->nErr = 0;
 	pInst->nErl = 0;
 	nError = 0;
+	SbxErrObject::getUnoErrObject()->Clear();
 }
 
 // Resume nach Fehlern (+0=statement, 1=next or Label)
@@ -394,6 +396,8 @@ void SbiRuntime::StepRESUME( UINT32 nOp1 )
 	}
 	else
 		pCode = pErrStmnt;
+	if ( pError ) // current in error handler ( and got a Resume Next statment )
+		SbxErrObject::getUnoErrObject()->Clear();
 
 	if( nOp1 > 1 )
 		StepJUMP( nOp1 );
