From ad4c937c45b8c44db06a5804f0b88edcedd02a98 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:00:14 +0200
Subject: [PATCH 401/878] vba-multipage-fireapichange.diff

---
 vbahelper/source/msforms/vbamultipage.cxx |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/vbahelper/source/msforms/vbamultipage.cxx b/vbahelper/source/msforms/vbamultipage.cxx
index b8fa10b..69084f0 100644
--- a/vbahelper/source/msforms/vbamultipage.cxx
+++ b/vbahelper/source/msforms/vbamultipage.cxx
@@ -92,8 +92,11 @@ void SAL_CALL
 ScVbaMultiPage::setValue( const sal_Int32 _value ) throw (::com::sun::star::uno::RuntimeException)
 {
     // Openoffice 1 based tab index
-   sal_Int32 nVal = _value + 1;
+    sal_Int32 nVal = _value + 1;
+    sal_Int32 nOldVal = getValue();
     m_xProps->setPropertyValue( SVALUE, uno::makeAny( nVal ) );
+    if ( nVal != nOldVal )
+        fireChangeEvent();
 }
 
 
-- 
1.7.0.1

