diff --git sw/source/core/swg/SwXMLTextBlocks1.cxx sw/source/core/swg/SwXMLTextBlocks1.cxx
index 14a0497..41e468f 100644
--- sw/source/core/swg/SwXMLTextBlocks1.cxx
+++ sw/source/core/swg/SwXMLTextBlocks1.cxx
@@ -40,6 +40,7 @@
 #include <com/sun/star/xml/sax/InputSource.hpp>
 #include <com/sun/star/io/XActiveDataSource.hpp>
 #include <com/sun/star/xml/sax/XParser.hpp>
+#include <com/sun/star/document/XStorageBasedDocument.hpp>
 #include <doc.hxx>
 #ifndef _DOCSH_HXX
 #include <docsh.hxx>
@@ -81,6 +82,21 @@ ULONG SwXMLTextBlocks::GetDoc( USHORT nIdx )
             ReadXML->SetBlockMode( sal_True );
             aReader.Read( *ReadXML );
             ReadXML->SetBlockMode( sal_False );
+            // Ole objects fails to display when inserted into document 
+            // because the ObjectReplacement folder ( and contents are missing )
+            rtl::OUString sObjReplacements( RTL_CONSTASCII_USTRINGPARAM( "ObjectReplacements" ) );
+            if ( xRoot->hasByName( sObjReplacements ) )
+            {
+                uno::Reference< document::XStorageBasedDocument > xDocStor( pDoc->GetDocShell()->GetModel(), uno::UNO_QUERY_THROW );
+                uno::Reference< embed::XStorage > xStr( xDocStor->getDocumentStorage() );
+                if ( xStr.is() )
+                {
+                    xRoot->copyElementTo( sObjReplacements, xStr, sObjReplacements );
+                    uno::Reference< embed::XTransactedObject > xTrans( xStr, uno::UNO_QUERY );
+                    if ( xTrans.is() )
+                        xTrans->commit();
+                }
+            }
         }
         catch( uno::Exception& )
 		{
