diff --git sw/source/filter/ww8/ww8toolbar.cxx sw/source/filter/ww8/ww8toolbar.cxx
index 42f22d1..44c8253 100644
--- sw/source/filter/ww8/ww8toolbar.cxx
+++ sw/source/filter/ww8/ww8toolbar.cxx
@@ -105,6 +105,7 @@ bool CTBWrapper::Read( SvStream* pS )
     Tcg255SubStruct::Read( pS );
     *pS >> reserved2 >> reserved3 >> reserved4 >> reserved5;
     *pS >> cbTBD >> cCust >> cbDTBC;
+    long nExpectedPos =  pS->Tell() + cbDTBC;
     if ( cbDTBC )
     {
         // cbDTBC is the size in bytes of the TBC array
@@ -125,6 +126,16 @@ bool CTBWrapper::Read( SvStream* pS )
             bytesToRead = cbDTBC - ( pS->Tell() - nStart ); 
         } while ( bytesToRead > 0 );
     }
+    if ( pS->Tell() != nExpectedPos )
+    {
+        // Strange error condition, shouldn't happen ( but does in at least
+        // one test document ) In the case where it happens the TBC & 
+        // TBCHeader records seem blank??? ( and incorrect )
+        OSL_ENSURE( pS->Tell() == nExpectedPos, "### Error: Expected pos not equal to actual pos after reading rtbdc");
+        OSL_TRACE("\tPos now is 0x%x should be 0x%x", pS->Tell(), nExpectedPos );
+        // seek to correct position after rtbdc
+        pS->Seek( nExpectedPos );
+    }
     if ( cCust )
     {
         for ( sal_Int32 index = 0; index < cCust; ++index )
