diff --git sc/source/ui/vba/vbawsfunction.cxx sc/source/ui/vba/vbawsfunction.cxx
index d0e7ee3..ed3a749 100644
--- sc/source/ui/vba/vbawsfunction.cxx
+++ sc/source/ui/vba/vbawsfunction.cxx
@@ -81,6 +81,16 @@ ScVbaWSFunction::invoke(const rtl::OUString& FunctionName, const uno::Sequence<
 			aArrayTemp[i] = myRange->getCellRange();
 			continue;
 		}
+                else if (  aArray[ i ].getValueType().getTypeClass() == uno::TypeClass_BOOLEAN )
+		{
+			sal_Bool bValue( sal_False );
+			aArray[ i ] >>= bValue;
+			if ( bValue )
+				aArrayTemp[ i ] <<= double( 1.0 ); 
+			else
+				aArrayTemp[ i ] <<= double( 0.0 ); 
+                        
+		}
         else if ( aArray[ i ].getValueType().getTypeClass() == uno::TypeClass_SEQUENCE )
         {
             // the sheet.FunctionAccess service doesn't deal with Sequences, only Sequences of Sequence

