--- oovbaapi/org/openoffice/excel/XWorksheet.idl.orig	2008-05-15 10:55:34.000000000 +0800
+++ oovbaapi/org/openoffice/excel/XWorksheet.idl	2008-05-15 10:55:40.000000000 +0800
@@ -56,6 +56,7 @@ module org {  module openoffice {  modul
 interface XComments;
 interface XRange;
 interface XOutline;
+interface XPageSetup;
 interface XWorksheet
 {
 	interface ::org::openoffice::vba::XHelperInterface;
@@ -88,6 +89,7 @@ interface XWorksheet
 	any PivotTables([in] any Index);
 	any Comments([in] any Index);
 	XOutline Outline();
+	XPageSetup PageSetup();
 	any OLEObjects([in] any Index);
 	void ShowDataForm();
 	any Shapes([in] any Index);
--- oovbaapi/org/openoffice/excel/makefile.mk.orig	2008-05-15 10:55:34.000000000 +0800
+++ oovbaapi/org/openoffice/excel/makefile.mk	2008-05-15 10:55:40.000000000 +0800
@@ -95,6 +95,7 @@ IDLFILES= XApplication.idl\
 	XNames.idl \
 	Workbook.idl \
 	Worksheet.idl \
+  	XPageSetup.idl \
 
 
 # ------------------------------------------------------------------
--- /dev/null	2007-09-22 03:12:45.000000000 +0800
+++ oovbaapi/org/openoffice/excel/XPageSetup.idl	2008-05-15 10:55:40.000000000 +0800
@@ -0,0 +1,64 @@
+/*************************************************************************
+ *
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ * 
+ * Copyright 2008 by Sun Microsystems, Inc.
+ *
+ * OpenOffice.org - a multi-platform office productivity suite
+ *
+ * $RCSfile: XPageSetup.idl,v $
+ * $Revision: 1.0 $
+ *
+ * This file is part of OpenOffice.org.
+ *
+ * OpenOffice.org is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License version 3
+ * only, as published by the Free Software Foundation.
+ *
+ * OpenOffice.org is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License version 3 for more details
+ * (a copy is included in the LICENSE file that accompanied this code).
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * version 3 along with OpenOffice.org.  If not, see
+ * <http://www.openoffice.org/license.html>
+ * for a copy of the LGPLv3 License.
+ *
+ ************************************************************************/
+#ifndef __org_openoffice_excel_XPageSetup_idl__
+#define __org_openoffice_excel_XPageSetup_idl__
+
+#ifndef __com_sun_star_uno_XInterface_idl__
+#include <com/sun/star/uno/XInterface.idl>
+#endif
+
+#ifndef __org_openoffice_vba_XHelperInterface_idl__
+#include <org/openoffice/vba/XHelperInterface.idl>
+#endif
+
+//=============================================================================
+
+module org {  module openoffice {  module excel { 
+
+//=============================================================================
+
+
+interface XPageSetup
+{
+
+	interface ::org::openoffice::vba::XHelperInterface;
+
+	//Attributes
+    [attribute] string PrintArea;
+
+};
+
+//=============================================================================
+
+}; }; };
+
+#endif
+
+
--- sc/source/ui/vba/vbaworksheet.hxx.orig	2008-05-15 10:55:34.000000000 +0800
+++ sc/source/ui/vba/vbaworksheet.hxx	2008-05-15 10:55:40.000000000 +0800
@@ -47,6 +47,7 @@
 #include <com/sun/star/uno/XComponentContext.hpp>
 #include <com/sun/star/frame/XModel.hpp>
 #include <org/openoffice/excel/XOutline.hpp>
+#include <org/openoffice/excel/XPageSetup.hpp>
 #include <org/openoffice/excel/XChartObjects.hpp>
 
 #include "vbahelperinterface.hxx"
@@ -94,6 +95,7 @@ public:
 	virtual css::uno::Reference< oo::excel::XRange > SAL_CALL getUsedRange() throw (css::uno::RuntimeException) ;
 	virtual css::uno::Any SAL_CALL ChartObjects( const css::uno::Any& Index ) throw (css::uno::RuntimeException);
 	virtual css::uno::Reference< oo::excel::XOutline > SAL_CALL Outline( ) throw (css::uno::RuntimeException);
+	virtual css::uno::Reference< oo::excel::XPageSetup > SAL_CALL PageSetup( ) throw (css::uno::RuntimeException);
 	virtual css::uno::Reference< oo::excel::XWorksheet > SAL_CALL getNext() throw (css::uno::RuntimeException);
 	virtual css::uno::Reference< oo::excel::XWorksheet > SAL_CALL getPrevious() throw (css::uno::RuntimeException);
 
--- sc/source/ui/vba/vbaworksheet.cxx.orig	2008-05-15 10:55:34.000000000 +0800
+++ sc/source/ui/vba/vbaworksheet.cxx	2008-05-15 20:44:11.000000000 +0800
@@ -89,6 +89,7 @@
 #include "vbaoleobject.hxx"
 #include "vbaoleobjects.hxx"
 #include "vbashapes.hxx"
+#include "vbapagesetup.hxx"
 
 #define STANDARDWIDTH 2267
 #define STANDARDHEIGHT 427
@@ -271,6 +272,13 @@ ScVbaWorksheet::Outline( ) throw (uno::R
 	return new ScVbaOutline( this, mxContext, xOutline);
 }
 
+uno::Reference< excel::XPageSetup >
+ScVbaWorksheet::PageSetup( ) throw (uno::RuntimeException)
+{
+	uno::Reference<sheet::XPrintAreas> xPrintAreas(getSheet(),uno::UNO_QUERY_THROW);
+	return new ScVbaPageSetup( this, mxContext, xPrintAreas, getModel());
+}
+
 sal_Int32 
 ScVbaWorksheet::getStandardWidth() throw (uno::RuntimeException)
 {
--- sc/source/ui/vba/makefile.mk.orig	2008-05-15 10:55:34.000000000 +0800
+++ sc/source/ui/vba/makefile.mk	2008-05-15 10:55:40.000000000 +0800
@@ -121,6 +121,7 @@ SLOFILES= \
                 $(SLO)$/vbacommandbarcontrols.obj \
                 $(SLO)$/vbacommandbar.obj \
                 $(SLO)$/vbacommandbars.obj \
+		        $(SLO)$/vbapagesetup.obj \
 		$(SLO)$/service.obj \
  
 
--- /dev/null	2007-09-22 03:12:45.000000000 +0800
+++ sc/source/ui/vba/vbapagesetup.hxx	2008-05-15 11:13:19.000000000 +0800
@@ -0,0 +1,60 @@
+/*************************************************************************
+ *
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ * 
+ * Copyright 2008 by Sun Microsystems, Inc.
+ *
+ * OpenOffice.org - a multi-platform office productivity suite
+ *
+ * $RCSfile: vbapagesetup.hxx,v $
+ * $Revision: 1.4 $
+ *
+ * This file is part of OpenOffice.org.
+ *
+ * OpenOffice.org is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License version 3
+ * only, as published by the Free Software Foundation.
+ *
+ * OpenOffice.org is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License version 3 for more details
+ * (a copy is included in the LICENSE file that accompanied this code).
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * version 3 along with OpenOffice.org.  If not, see
+ * <http://www.openoffice.org/license.html>
+ * for a copy of the LGPLv3 License.
+ *
+ ************************************************************************/
+#ifndef SC_VBA_PAGESETUP_HXX
+#define SC_VBA_PAGESETUP_HXX
+
+#include <com/sun/star/sheet/XPrintAreas.hpp>
+#include <cppuhelper/implbase1.hxx>
+#include <org/openoffice/excel/XPageSetup.hpp>
+#include <com/sun/star/uno/XComponentContext.hpp>
+#include "vbahelperinterface.hxx"
+
+typedef InheritedHelperInterfaceImpl1< oo::excel::XPageSetup > ScVbaPageSetup_BASE;
+
+class ScVbaPageSetup :  public ScVbaPageSetup_BASE 
+{
+	css::uno::Reference< css::sheet::XPrintAreas > mxPrintAreas;
+	css::uno::Reference< css::frame::XModel > mxModel;
+public:
+	ScVbaPageSetup( const css::uno::Reference< oo::vba::XHelperInterface >& xParent, 
+					const css::uno::Reference< css::uno::XComponentContext >& xContext, 
+					const css::uno::Reference< css::sheet::XPrintAreas >& xPrintareas, 
+					const css::uno::Reference< css::frame::XModel >& xModel) throw (css::uno::RuntimeException);
+	virtual ~ScVbaPageSetup(){}
+    
+    // Attribute
+	virtual rtl::OUString SAL_CALL getPrintArea() throw (css::uno::RuntimeException);
+	virtual void SAL_CALL setPrintArea( const rtl::OUString& rAreas ) throw (css::uno::RuntimeException);
+
+	// XHelperInterface
+	virtual rtl::OUString& getServiceImplName();
+	virtual css::uno::Sequence<rtl::OUString> getServiceNames();
+};
+#endif
--- /dev/null	2007-09-22 03:12:45.000000000 +0800
+++ sc/source/ui/vba/vbapagesetup.cxx	2008-05-15 20:44:11.000000000 +0800
@@ -0,0 +1,117 @@
+/*************************************************************************
+ *
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ * 
+ * Copyright 2008 by Sun Microsystems, Inc.
+ *
+ * OpenOffice.org - a multi-platform office productivity suite
+ *
+ * $RCSfile: vbapagesetup.cxx,v $
+ * $Revision: 1.4 $
+ *
+ * This file is part of OpenOffice.org.
+ *
+ * OpenOffice.org is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License version 3
+ * only, as published by the Free Software Foundation.
+ *
+ * OpenOffice.org is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License version 3 for more details
+ * (a copy is included in the LICENSE file that accompanied this code).
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * version 3 along with OpenOffice.org.  If not, see
+ * <http://www.openoffice.org/license.html>
+ * for a copy of the LGPLv3 License.
+ *
+ ************************************************************************/
+#include "vbapagesetup.hxx"
+#include "cellsuno.hxx"
+#include "convuno.hxx"
+#include "rangelst.hxx"
+using namespace ::com::sun::star;
+using namespace ::org::openoffice;
+
+bool getScRangeListForAddress( const rtl::OUString& sName, ScDocShell* pDocSh, ScRange& refRange, ScRangeList& aCellRanges, ScAddress::Convention aConv = ScAddress::CONV_XL_A1 ) throw ( uno::RuntimeException );
+
+ScVbaPageSetup::ScVbaPageSetup(const uno::Reference< vba::XHelperInterface >& xParent, 
+				const uno::Reference< uno::XComponentContext >& xContext,
+				const uno::Reference< sheet::XPrintAreas >& xPrintareas,
+				const uno::Reference< frame::XModel >& xModel) throw (uno::RuntimeException):
+	   	ScVbaPageSetup_BASE( xParent, xContext ),
+		mxPrintAreas( xPrintareas ), mxModel( xModel )
+{
+}		
+
+rtl::OUString SAL_CALL ScVbaPageSetup::getPrintArea() throw (css::uno::RuntimeException)
+{
+    String aPrintArea;
+    uno::Sequence< table::CellRangeAddress > aSeq = mxPrintAreas->getPrintAreas();
+    sal_Int32 nCount = aSeq.getLength();
+    if( nCount )
+    {
+        ScAddress::Details aDetails( ScAddress::CONV_XL_A1, 0, 0 );
+        USHORT nFlags = SCA_VALID;
+        nFlags |= ( SCA_TAB_ABSOLUTE | SCA_COL_ABSOLUTE | SCA_ROW_ABSOLUTE | SCA_TAB2_ABSOLUTE | SCA_COL2_ABSOLUTE | SCA_ROW2_ABSOLUTE );
+		ScRangeList aRangeList;
+		for( sal_Int32 i=0; i<nCount; i++ )
+		{
+			ScRange aRange;
+			ScUnoConversion::FillScRange( aRange, aSeq[i] );
+			aRangeList.Append( aRange );
+		}
+		ScDocument* pDoc = getDocShell( mxModel )->GetDocument();
+		aRangeList.Format( aPrintArea, nFlags, pDoc, ScAddress::CONV_XL_A1, ','  );
+    }
+    
+    return aPrintArea;
+}
+
+void SAL_CALL ScVbaPageSetup::setPrintArea( const rtl::OUString& rAreas ) throw (css::uno::RuntimeException)
+{
+        if( rAreas.getLength() == 0 ||
+            rAreas.equalsIgnoreAsciiCase ( rtl::OUString::createFromAscii("FALSE") ) )
+        {
+            // print the whole sheet
+            uno::Sequence< table::CellRangeAddress > aSeq;
+            mxPrintAreas->setPrintAreas( aSeq );
+        }
+        else
+        {
+			ScRangeList aCellRanges;
+			ScRange aRange;
+			if( getScRangeListForAddress( rAreas, getDocShell( mxModel ) , aRange, aCellRanges ) ) 
+			{
+				uno::Sequence< table::CellRangeAddress > aSeq( aCellRanges.Count() );
+				USHORT i=0;
+				for( ScRange* pRange = aCellRanges.First(); pRange; pRange = aCellRanges.Next() )
+				{
+					table::CellRangeAddress aRangeAddress;
+					ScUnoConversion::FillApiRange( aRangeAddress, *pRange );
+					aSeq[ i++ ] = aRangeAddress;
+				}
+				mxPrintAreas->setPrintAreas( aSeq );
+			}
+        }
+}
+
+rtl::OUString& 
+ScVbaPageSetup::getServiceImplName()
+{
+	static rtl::OUString sImplName( RTL_CONSTASCII_USTRINGPARAM("ScVbaPageSetup") );
+	return sImplName;
+}
+
+uno::Sequence< rtl::OUString > 
+ScVbaPageSetup::getServiceNames()
+{
+	static uno::Sequence< rtl::OUString > aServiceNames;
+	if ( aServiceNames.getLength() == 0 )
+	{
+		aServiceNames.realloc( 1 );
+		aServiceNames[ 0 ] = rtl::OUString( RTL_CONSTASCII_USTRINGPARAM("org.openoffice.excel.PageSetup" ) );
+	}
+	return aServiceNames;
+}
