From b83705241c40b3c92027e0b8de34a01af3f93461 Mon Sep 17 00:00:00 2001
From: Jan Holesovsky <kendy@suse.cz>
Date: Fri, 14 May 2010 17:00:27 +0200
Subject: [PATCH 409/878] vba-fixerror-nonvba.diff

---
 basic/source/runtime/stdobj.cxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/basic/source/runtime/stdobj.cxx b/basic/source/runtime/stdobj.cxx
index 0fc7f41..0ccd56a 100644
--- a/basic/source/runtime/stdobj.cxx
+++ b/basic/source/runtime/stdobj.cxx
@@ -770,7 +770,7 @@ SbxVariable* SbiStdObject::Find( const String& rName, SbxClassType t )
     // #TODO #FIXME hack for substituting ooo-basic Err with vba-ish
     // ErrObject object
     static String sErr( RTL_CONSTASCII_USTRINGPARAM("Err") );
-    if (  rName.EqualsIgnoreCaseAscii( sErr ) )
+    if (  SbiRuntime::isVBAEnabled() && rName.EqualsIgnoreCaseAscii( sErr ) )
         return SbxErrObject::getErrObject();
     // Bereits eingetragen?
     SbxVariable* pVar = SbxObject::Find( rName, t );
-- 
1.7.0.1

