diff --git scripting/source/vbaevents/makefile.mk scripting/source/vbaevents/makefile.mk
index 96f4e3d..d485c73 100644
--- scripting/source/vbaevents/makefile.mk
+++ scripting/source/vbaevents/makefile.mk
@@ -19,14 +19,6 @@ DLLPRE =
 
 # ------------------------------------------------------------------
 
-ALLTAR : \
-        $(MISC)$/$(TARGET).don \
-        $(SLOTARGET)
-
-$(MISC)$/$(TARGET).don : $(SOLARBINDIR)$/oovbaapi.rdb
-        +$(CPPUMAKER) -O$(OUT)$/inc -BUCR $(SOLARBINDIR)$/oovbaapi.rdb -X$(SOLARBINDIR)$/types.rdb && echo > $@
-        echo $@
-# ------------------------------------------------------------------
 
 #.INCLUDE :  ..$/cppumaker.mk
 
@@ -57,3 +49,13 @@ SHL1LIBS=$(SLB)$/$(TARGET).lib
 # --- Targets ------------------------------------------------------
 
 .INCLUDE :	target.mk
+
+# ------------------------------------------------------------------
+
+ALLTAR : \
+        $(MISC)$/$(TARGET).don \
+        $(SLOTARGET)
+
+$(MISC)$/$(TARGET).don : $(SOLARBINDIR)$/oovbaapi.rdb
+        +$(CPPUMAKER) -O$(OUT)$/inc -BUCR $(SOLARBINDIR)$/oovbaapi.rdb -X$(SOLARBINDIR)$/types.rdb && echo > $@
+        echo $@
