diff --git sc/source/core/data/documen2.cxx sc/source/core/data/documen2.cxx
index b75279e..5c7a375 100644
--- sc/source/core/data/documen2.cxx
+++ sc/source/core/data/documen2.cxx
@@ -1463,6 +1462,7 @@ BOOL ScDocument::CopyTab( SCTAB nOldPos, SCTAB nNewPos, const ScMarkData* pOnlyM
 		if (nNewPos == nMaxTableNumber)
 		{
 			pTab[nMaxTableNumber] = new ScTable(this, nMaxTableNumber, aName);
+                        pTab[nMaxTableNumber]->SetCodeName( aName );
 			++nMaxTableNumber;
 		}
 		else
@@ -1497,8 +1497,9 @@ BOOL ScDocument::CopyTab( SCTAB nOldPos, SCTAB nNewPos, const ScMarkData* pOnlyM
 					pTab[i] = pTab[i - 1];
 				if (nNewPos <= nOldPos)
 					nOldPos++;
-				pTab[nNewPos] = new ScTable(this, nNewPos, aName);
-				++nMaxTableNumber;
+					pTab[nNewPos] = new ScTable(this, nNewPos, aName);
+                                        pTab[nNewPos]->SetCodeName( aName );
+					++nMaxTableNumber;
 				bValid = TRUE;
 				for (i = 0; i <= MAXTAB; i++)
 					if (pTab[i] && i != nOldPos && i != nNewPos)
diff --git sc/source/ui/docshell/docsh5.cxx sc/source/ui/docshell/docsh5.cxx
index 8e38e2a..9a26cd2 100644
--- sc/source/ui/docshell/docsh5.cxx
+++ sc/source/ui/docshell/docsh5.cxx
@@ -843,11 +843,16 @@ BOOL ScDocShell::MoveTable( SCTAB nSrcTab, SCTAB nDestTab, BOOL bCopy, BOOL bRec
 			
 			StarBASIC* pStarBASIC = GetBasic();
 			BOOL bVbaEnabled = pStarBASIC ? pStarBASIC->isVBAEnabled() : FALSE;
+			SCTAB nTabToUse = nDestTab; 
+
+			if ( nDestTab == SC_TAB_APPEND )
+				nTabToUse = aDocument.GetMaxTableNumber() - 1;
+
 			if ( bVbaEnabled )
 			{
 				String sCodeName;
 				String sSource;
-				aDocument.GetCodeName( nDestTab, sCodeName );
+				aDocument.GetCodeName( nTabToUse, sCodeName );
 				com::sun::star::uno::Reference< com::sun::star::script::XLibraryContainer > xLibContainer = GetBasicContainer();
 				com::sun::star::uno::Reference< com::sun::star::container::XNameContainer > xLib;
     				if( xLibContainer.is() )
@@ -865,7 +870,7 @@ BOOL ScDocShell::MoveTable( SCTAB nSrcTab, SCTAB nDestTab, BOOL bCopy, BOOL bRec
 					xLib->getByName( sModName ) >>= sModuleInfo;
 					sSource = sModuleInfo.ModuleSource;
 				} 
-				lcl_InsertModule( *this, nDestTab, sCodeName, sSource );
+				lcl_InsertModule( *this, nTabToUse, sCodeName, sSource );
 			}
 		}
 
