diff --git basic/source/runtime/step2.cxx basic/source/runtime/step2.cxx
index 68bef06..d50d38b 100755
--- basic/source/runtime/step2.cxx
+++ basic/source/runtime/step2.cxx
@@ -489,7 +489,7 @@ SbxVariable* SbiRuntime::CheckArray( SbxVariable* pElem )
             pPar->Put( NULL, 0 );
     }
     // Index-Access bei UnoObjekten beruecksichtigen
-	else if( pElem->GetType() == SbxOBJECT && !pElem->ISA(SbxMethod) && !pElem->ISA(SbxProperty) )
+    else if( pElem->GetType() == SbxOBJECT && !pElem->ISA(SbxMethod) && ( !bVBAEnabled || ( bVBAEnabled && !pElem->ISA(SbxProperty) ) ) )
     {
         pPar = pElem->GetParameters();
         if ( pPar )
