diff --git oovbaapi/ooo/vba/XCommandBarControl.idl oovbaapi/ooo/vba/XCommandBarControl.idl
index 167bce8..b10750e 100644
--- oovbaapi/ooo/vba/XCommandBarControl.idl
+++ oovbaapi/ooo/vba/XCommandBarControl.idl
@@ -55,6 +55,7 @@ interface XCommandBarControl
     [attribute] string OnAction;
     [attribute] boolean Visible;
     [attribute, readonly] long Type;
+    [attribute] boolean Enabled;
     
     void Delete() raises ( com::sun::star::script::BasicErrorException );
     any Controls( [in] any Index ) raises ( com::sun::star::script::BasicErrorException );
diff --git vbahelper/source/vbahelper/vbacommandbar.cxx vbahelper/source/vbahelper/vbacommandbar.cxx
index 43abe41..fcb3c62 100644
--- vbahelper/source/vbahelper/vbacommandbar.cxx
+++ vbahelper/source/vbahelper/vbacommandbar.cxx
@@ -151,16 +151,14 @@ ScVbaCommandBar::setVisible( ::sal_Bool _visible ) throw (uno::RuntimeException)
 ::sal_Bool SAL_CALL 
 ScVbaCommandBar::getEnabled() throw (uno::RuntimeException)
 {
-    // #FIXME: seems tha main menu is alwayse enabled.
-    // for the toolbar, need to investigate
-    return sal_True;
+    // emulated by Visiable
+    return getVisible();
 }
 
 void SAL_CALL
-ScVbaCommandBar::setEnabled( sal_Bool/* _enabled */ ) throw (uno::RuntimeException)
+ScVbaCommandBar::setEnabled( sal_Bool _enabled ) throw (uno::RuntimeException)
 {
-    // #FIXME: seems tha main menu is alwayse enabled.
-    // for the toolbar, need to investigate
+    setVisible( _enabled );
 }
 
 void SAL_CALL 
diff --git vbahelper/source/vbahelper/vbacommandbarcontrol.cxx vbahelper/source/vbahelper/vbacommandbarcontrol.cxx
index dd20256..111123c 100644
--- vbahelper/source/vbahelper/vbacommandbarcontrol.cxx
+++ vbahelper/source/vbahelper/vbacommandbarcontrol.cxx
@@ -118,6 +118,19 @@ ScVbaCommandBarControl::setVisible( ::sal_Bool _visible ) throw (uno::RuntimeExc
     }
 }
 
+::sal_Bool SAL_CALL 
+ScVbaCommandBarControl::getEnabled() throw (uno::RuntimeException)
+{
+    // emulated by Visiable
+    return getVisible();
+}
+
+void SAL_CALL
+ScVbaCommandBarControl::setEnabled( sal_Bool _enabled ) throw (uno::RuntimeException)
+{
+    setVisible( _enabled );
+}
+
 void SAL_CALL 
 ScVbaCommandBarControl::Delete(  ) throw (script::BasicErrorException, uno::RuntimeException)
 {
diff --git vbahelper/source/vbahelper/vbacommandbarcontrol.hxx vbahelper/source/vbahelper/vbacommandbarcontrol.hxx
index f40ad82..e125317 100644
--- vbahelper/source/vbahelper/vbacommandbarcontrol.hxx
+++ vbahelper/source/vbahelper/vbacommandbarcontrol.hxx
@@ -71,6 +71,8 @@ public:
     virtual void SAL_CALL setOnAction( const ::rtl::OUString& _onaction ) throw (css::uno::RuntimeException);
     virtual ::sal_Bool SAL_CALL getVisible() throw (css::uno::RuntimeException);
     virtual void SAL_CALL setVisible( ::sal_Bool _visible ) throw (css::uno::RuntimeException);
+    virtual ::sal_Bool SAL_CALL getEnabled() throw (css::uno::RuntimeException);
+    virtual void SAL_CALL setEnabled( ::sal_Bool _enabled ) throw (css::uno::RuntimeException);
     virtual sal_Int32 SAL_CALL getType() throw (css::uno::RuntimeException)
     {
         return ov::office::MsoControlType::msoControlButton;
