diff --git basic/source/runtime/stdobj.cxx basic/source/runtime/stdobj.cxx
index 4c8c807..9341037 100644
--- basic/source/runtime/stdobj.cxx
+++ basic/source/runtime/stdobj.cxx
@@ -773,7 +773,7 @@ SbxVariable* SbiStdObject::Find( const String& rName, SbxClassType t )
 	// #TODO #FIXME hack for substituting ooo-basic Err with vba-ish
 	// ErrObject object
 	static String sErr( RTL_CONSTASCII_USTRINGPARAM("Err") );
-	if (  rName.EqualsIgnoreCaseAscii( sErr ) ) 
+    if (  SbiRuntime::isVBAEnabled() && rName.EqualsIgnoreCaseAscii( sErr ) ) 
 		return SbxErrObject::getErrObject();
     // Bereits eingetragen?
     SbxVariable* pVar = SbxObject::Find( rName, t );
