Index: sc/source/ui/vba/makefile.mk
===================================================================
RCS file: /cvs/sc/sc/source/ui/vba/makefile.mk,v
retrieving revision 1.5
retrieving revision 1.5.42.1
diff -u -p -u -p -b -w -B -r1.5 -r1.5.42.1
--- sc/source/ui/vba/makefile.mk	7 Dec 2007 10:42:39 -0000	1.5
+++ sc/source/ui/vba/makefile.mk	18 Jan 2008 16:51:37 -0000	1.5.42.1
@@ -115,6 +115,7 @@ SLOFILES= \
                 $(SLO)$/vbaformatconditions.obj \
                 $(SLO)$/vbastyle.obj \
                 $(SLO)$/vbastyles.obj \
+                $(SLO)$/vbaassistant.obj \
 		$(SLO)$/service.obj \
  
 
Index: sc/source/ui/vba/vbaapplication.cxx
===================================================================
RCS file: /cvs/sc/sc/source/ui/vba/vbaapplication.cxx,v
retrieving revision 1.5
retrieving revision 1.5.42.1
diff -u -p -u -p -b -w -B -r1.5 -r1.5.42.1
--- sc/source/ui/vba/vbaapplication.cxx	7 Dec 2007 10:43:04 -0000	1.5
+++ sc/source/ui/vba/vbaapplication.cxx	18 Jan 2008 16:51:37 -0000	1.5.42.1
@@ -62,6 +62,7 @@
 #include "gridwin.hxx"
 #include "vbanames.hxx"
 #include "vbashape.hxx"
+#include "vbaassistant.hxx"
 #include "sc.hrc"
 
 #include <osl/file.hxx>
@@ -127,6 +128,12 @@ ScVbaApplication::getThisWorkbook() thro
 	return getActiveWorkbook();
 }
 
+uno::Reference< vba::XAssistant > SAL_CALL
+ScVbaApplication::getAssistant() throw (css::uno::RuntimeException)
+{
+    return uno::Reference< vba::XAssistant >( new ScVbaAssistant( this, mxContext ) );
+}
+
 uno::Any SAL_CALL
 ScVbaApplication::getSelection() throw (uno::RuntimeException)
 {
Index: sc/source/ui/vba/vbaapplication.hxx
===================================================================
RCS file: /cvs/sc/sc/source/ui/vba/vbaapplication.hxx,v
retrieving revision 1.3
retrieving revision 1.3.42.1
diff -u -p -u -p -b -w -B -r1.3 -r1.3.42.1
--- sc/source/ui/vba/vbaapplication.hxx	7 Dec 2007 10:43:17 -0000	1.3
+++ sc/source/ui/vba/vbaapplication.hxx	18 Jan 2008 16:51:37 -0000	1.3.42.1
@@ -80,6 +80,7 @@ public:
 	virtual ::sal_Bool SAL_CALL getDisplayFormulaBar() throw ( css::uno::RuntimeException );
 	virtual void SAL_CALL setDisplayFormulaBar( ::sal_Bool _displayformulabar ) throw ( css::uno::RuntimeException );
 
+    virtual css::uno::Reference< oo::vba::XAssistant > SAL_CALL getAssistant() throw (css::uno::RuntimeException);
 	virtual css::uno::Reference< oo::excel::XWorkbook > SAL_CALL getThisWorkbook() throw (css::uno::RuntimeException);
 	virtual css::uno::Any SAL_CALL Workbooks( const css::uno::Any& aIndex ) throw (css::uno::RuntimeException);
 	virtual css::uno::Any SAL_CALL Worksheets( const css::uno::Any& aIndex ) throw (css::uno::RuntimeException);
Index: sc/source/ui/vba/vbaassistant.cxx
===================================================================
RCS file: sc/source/ui/vba/vbaassistant.cxx
diff -N sc/source/ui/vba/vbaassistant.cxx
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ sc/source/ui/vba/vbaassistant.cxx	18 Jan 2008 19:27:41 -0000	1.1.2.1
@@ -0,0 +1,140 @@
+/*************************************************************************
+ *
+ *  OpenOffice.org - a multi-platform office productivity suite
+ *
+ *  $RCSfile$
+ *
+ *  $Revision$
+ *
+ *  last change: $Author$ $Date$
+ *
+ *  The Contents of this file are made available subject to
+ *  the terms of GNU Lesser General Public License Version 2.1.
+ *
+ *
+ *    GNU Lesser General Public License Version 2.1
+ *    =============================================
+ *    Copyright 2005 by Sun Microsystems, Inc.
+ *    901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License version 2.1, as published by the Free Software Foundation.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with this library; if not, write to the Free Software
+ *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *    MA  02111-1307  USA
+ *
+ ************************************************************************/
+#include <sfx2/app.hxx>
+#include <svtools/helpopt.hxx>
+
+#include <org/openoffice/office/MsoAnimationType.hpp>
+
+#include"vbaassistant.hxx"
+
+
+using namespace com::sun::star;
+using namespace org::openoffice;
+
+using namespace org::openoffice::office::MsoAnimationType;
+
+ScVbaAssistant::ScVbaAssistant( const uno::Reference< vba::XHelperInterface > xParent, const uno::Reference< uno::XComponentContext > xContext ): ScVbaAssistantImpl_BASE( xParent, xContext )
+{
+    m_bIsVisible = sal_False;
+    m_nPointsLeft = 795;
+    m_nPointsTop = 248;
+    m_sName = rtl::OUString::createFromAscii( "Clippit" );
+    m_nAnimation = msoAnimationIdle;
+}
+
+ScVbaAssistant::~ScVbaAssistant()
+{
+}
+
+sal_Bool SAL_CALL ScVbaAssistant::getVisible() throw (uno::RuntimeException)
+{
+    return m_bIsVisible;
+}
+
+void SAL_CALL ScVbaAssistant::setVisible( sal_Bool bVisible ) throw (uno::RuntimeException)
+{
+    m_bIsVisible = bVisible;
+}
+
+sal_Bool SAL_CALL ScVbaAssistant::getOn() throw (uno::RuntimeException)
+{
+    if( SvtHelpOptions().IsHelpAgentAutoStartMode() )
+        return sal_True;
+    else
+        return sal_False;
+}
+
+void SAL_CALL ScVbaAssistant::setOn( sal_Bool bOn ) throw (uno::RuntimeException)
+{
+    SvtHelpOptions().SetHelpAgentAutoStartMode( bOn );
+    setVisible( bOn );
+}
+
+
+::sal_Int32 SAL_CALL 
+ScVbaAssistant::getTop() throw (css::uno::RuntimeException)
+{
+    return m_nPointsTop;
+}
+void SAL_CALL 
+ScVbaAssistant::setTop( ::sal_Int32 _top ) throw (css::uno::RuntimeException)
+{
+    m_nPointsTop = _top;
+}
+::sal_Int32 SAL_CALL 
+ScVbaAssistant::getLeft() throw (css::uno::RuntimeException)
+{
+    return m_nPointsLeft;
+}
+void SAL_CALL 
+ScVbaAssistant::setLeft( ::sal_Int32 _left ) throw (css::uno::RuntimeException)
+{
+    m_nPointsLeft = _left;
+}
+::sal_Int32 SAL_CALL 
+ScVbaAssistant::getAnimation() throw (css::uno::RuntimeException)
+{
+    return m_nAnimation;
+}
+void SAL_CALL 
+ScVbaAssistant::setAnimation( ::sal_Int32 _animation ) throw (css::uno::RuntimeException)
+{
+    m_nAnimation = _animation;
+}
+
+::rtl::OUString SAL_CALL 
+ScVbaAssistant::Name(  ) throw (css::script::BasicErrorException, css::uno::RuntimeException)
+{
+    return m_sName;
+}
+
+rtl::OUString& 
+ScVbaAssistant::getServiceImplName()
+{
+	static rtl::OUString sImplName( RTL_CONSTASCII_USTRINGPARAM("ScVbaAssistant") );
+	return sImplName;
+}
+
+uno::Sequence< rtl::OUString > 
+ScVbaAssistant::getServiceNames()
+{
+	static uno::Sequence< rtl::OUString > aServiceNames;
+	if ( aServiceNames.getLength() == 0 )
+	{
+		aServiceNames.realloc( 1 );
+		aServiceNames[ 0 ] = rtl::OUString( RTL_CONSTASCII_USTRINGPARAM("org.openoffice.vba.Assistant" ) );
+	}
+	return aServiceNames;
+}
Index: sc/source/ui/vba/vbaassistant.hxx
===================================================================
RCS file: sc/source/ui/vba/vbaassistant.hxx
diff -N sc/source/ui/vba/vbaassistant.hxx
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ sc/source/ui/vba/vbaassistant.hxx	18 Jan 2008 19:27:42 -0000	1.1.2.1
@@ -0,0 +1,78 @@
+/*************************************************************************
+ *
+ *  OpenOffice.org - a multi-platform office productivity suite
+ *
+ *  $RCSfile$
+ *
+ *  $Revision$
+ *
+ *  last change: $Author$ $Date$
+ *
+ *  The Contents of this file are made available subject to
+ *  the terms of GNU Lesser General Public License Version 2.1.
+ *
+ *
+ *    GNU Lesser General Public License Version 2.1
+ *    =============================================
+ *    Copyright 2005 by Sun Microsystems, Inc.
+ *    901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License version 2.1, as published by the Free Software Foundation.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with this library; if not, write to the Free Software
+ *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *    MA  02111-1307  USA
+ *
+ ************************************************************************/
+#ifndef SC_VBA_ASSISTANT_HXX
+#define SC_VBA_ASSISTANT_HXX
+
+#include <cppuhelper/implbase1.hxx>
+#include <org/openoffice/vba/XAssistant.hpp>
+
+#include <sfx2/sfxhelp.hxx>
+
+#include "vbahelper.hxx"
+#include "vbahelperinterface.hxx"
+
+typedef ::cppu::WeakImplHelper1< oo::vba::XAssistant > Assistant;
+typedef InheritedHelperInterfaceImpl< Assistant > ScVbaAssistantImpl_BASE;
+
+class ScVbaAssistant : public ScVbaAssistantImpl_BASE
+{
+private:
+    sal_Bool        m_bIsVisible;
+    sal_Int32       m_nPointsLeft;
+    sal_Int32       m_nPointsTop;
+    rtl::OUString   m_sName;
+    sal_Int32       m_nAnimation;
+public:
+    ScVbaAssistant( const css::uno::Reference< oo::vba::XHelperInterface > xParent, const css::uno::Reference< css::uno::XComponentContext > xContext );
+    virtual ~ScVbaAssistant();
+    // XAssistant 
+    virtual sal_Bool SAL_CALL getOn() throw (css::uno::RuntimeException);
+    virtual void SAL_CALL setOn( sal_Bool _on ) throw (css::uno::RuntimeException);
+    virtual sal_Bool SAL_CALL getVisible() throw (css::uno::RuntimeException);
+    virtual void SAL_CALL setVisible( sal_Bool _visible ) throw (css::uno::RuntimeException);
+    virtual ::sal_Int32 SAL_CALL getTop() throw (css::uno::RuntimeException);
+    virtual void SAL_CALL setTop( ::sal_Int32 _top ) throw (css::uno::RuntimeException);
+    virtual ::sal_Int32 SAL_CALL getLeft() throw (css::uno::RuntimeException);
+    virtual void SAL_CALL setLeft( ::sal_Int32 _left ) throw (css::uno::RuntimeException);
+    virtual ::sal_Int32 SAL_CALL getAnimation() throw (css::uno::RuntimeException);
+    virtual void SAL_CALL setAnimation( ::sal_Int32 _animation ) throw (css::uno::RuntimeException);
+
+    virtual ::rtl::OUString SAL_CALL Name(  ) throw (css::script::BasicErrorException, css::uno::RuntimeException);
+	// XHelperInterface
+	virtual rtl::OUString& getServiceImplName();
+	virtual css::uno::Sequence<rtl::OUString> getServiceNames();
+};
+
+#endif//SC_VBA_ASSISTANT_HXX
Index: oovbaapi/org/openoffice/excel/XApplication.idl
===================================================================
RCS file: /cvs/api/oovbaapi/org/openoffice/excel/XApplication.idl,v
retrieving revision 1.3
retrieving revision 1.3.4.1
diff -u -p -u -p -b -w -B -r1.3 -r1.3.4.1
--- oovbaapi/org/openoffice/excel/XApplication.idl	7 Dec 2007 11:21:12 -0000	1.3
+++ oovbaapi/org/openoffice/excel/XApplication.idl	18 Jan 2008 16:51:20 -0000	1.3.4.1
@@ -43,6 +43,10 @@
 #include <org/openoffice/vba/XHelperInterface.idl>
 #endif
 
+#ifndef __org_openoffice_vba_XAssistant_idl__
+#include <org/openoffice/vba/XAssistant.idl>
+#endif
+
 module org {  module openoffice {  module excel { 
 
 interface XRange;
@@ -63,6 +67,7 @@ interface XApplication
 	[attribute, readonly] XRange ActiveCell;
 	[attribute, readonly] XWindow ActiveWindow;
 	[attribute, readonly] XWorksheet ActiveSheet;
+    [attribute, readonly] org::openoffice::vba::XAssistant Assistant;
 	[attribute] long Calculation;
 	//#TODO #FIXME this is more of a placeholder, will return
 	// the value of activeworkbook, in xl 'ThisWorkbook' should return the
Index: oovbaapi/org/openoffice/vba/XAssistant.idl
===================================================================
RCS file: oovbaapi/org/openoffice/vba/XAssistant.idl
diff -N oovbaapi/org/openoffice/vba/XAssistant.idl
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ oovbaapi/org/openoffice/vba/XAssistant.idl	18 Jan 2008 19:28:17 -0000	1.1.2.1
@@ -0,0 +1,65 @@
+/*************************************************************************
+ *
+ *  OpenOffice.org - a multi-platform office productivity suite
+ *
+ *  $RCSfile$
+ *
+ *  $Revision$
+ *
+ *  last change: $Author$ $Date$
+ *
+ *  The Contents of this file are made available subject to
+ *  the terms of GNU Lesser General Public License Version 2.1.
+ *
+ *
+ *    GNU Lesser General Public License Version 2.1
+ *    =============================================
+ *    Copyright 2005 by Sun Microsystems, Inc.
+ *    901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License version 2.1, as published by the Free Software Foundation.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with this library; if not, write to the Free Software
+ *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *    MA  02111-1307  USA
+ *
+ ************************************************************************/
+#ifndef __org_openoffice_vba_XAssistant_idl__
+#define __org_openoffice_vba_XAssistant_idl__
+
+#ifndef __com_sun_star_uno_XInterface_idl__
+#include <com/sun/star/uno/XInterface.idl>
+#endif
+
+#ifndef __org_openoffice_vba_XHelperInterface_idl__
+#include <org/openoffice/vba/XHelperInterface.idl>
+#endif
+//=============================================================================
+
+module org {  module openoffice {  module vba { 
+//=============================================================================
+
+interface XAssistant
+{
+	interface ::org::openoffice::vba::XHelperInterface;
+
+    [attribute] boolean On;
+    [attribute] boolean Visible;
+    [attribute] long Top;
+    [attribute] long Left;
+    [attribute] long Animation;
+    
+    string Name() raises ( com::sun::star::script::BasicErrorException );
+};
+
+}; }; };
+
+#endif
Index: oovbaapi/org/openoffice/vba/makefile.mk
===================================================================
RCS file: /cvs/api/oovbaapi/org/openoffice/vba/makefile.mk,v
retrieving revision 1.3
retrieving revision 1.3.4.1
diff -u -p -u -p -b -w -B -r1.3 -r1.3.4.1
--- oovbaapi/org/openoffice/vba/makefile.mk	7 Dec 2007 12:18:21 -0000	1.3
+++ oovbaapi/org/openoffice/vba/makefile.mk	18 Jan 2008 16:51:21 -0000	1.3.4.1
@@ -56,6 +56,7 @@ IDLFILES=\
 	XVBAToOOEventDescGen.idl\
 	XPropValue.idl\
 	XHelperInterface.idl\
+	XAssistant.idl\
 	Globals.idl\
 
 # ------------------------------------------------------------------
