diff --git sc/source/ui/vba/vbaworksheet.cxx sc/source/ui/vba/vbaworksheet.cxx
index 3a38bd3..5584cab 100644
--- sc/source/ui/vba/vbaworksheet.cxx
+++ sc/source/ui/vba/vbaworksheet.cxx
@@ -77,9 +77,7 @@
 
 #include "cellsuno.hxx"
 #include "drwlayer.hxx"
-#ifdef CWSSCSHEETPROTECTION
 #include "tabprotection.hxx"
-#endif
 #include "scextopt.hxx"
 #include "vbaoutline.hxx"
 #include "vbarange.hxx"
@@ -411,7 +409,6 @@ ScVbaWorksheet::getProtectContents()throw (uno::RuntimeException)
 sal_Bool 
 ScVbaWorksheet::getProtectDrawingObjects() throw (uno::RuntimeException) 
 {
-#ifdef CWSSCSHEETPROTECTION
 	SCTAB nTab = 0;
 	rtl::OUString aSheetName = getName();
 	uno::Reference <sheet::XSpreadsheetDocument> xSpreadDoc( getModel(), uno::UNO_QUERY_THROW );
@@ -422,11 +419,8 @@ ScVbaWorksheet::getProtectDrawingObjects() throw (uno::RuntimeException)
 		ScDocument* pDoc = excel::getDocShell( xModel )->GetDocument();
 		ScTableProtection* pProtect = pDoc->GetTabProtection(nTab);
 		if ( pProtect )
-			return pProtect->getOption( ScTableProtection::OBJECTS ); 
+			return pProtect->isOptionEnabled( ScTableProtection::OBJECTS ); 
         }
-#else
-	return getProtectContents(); // fallback
-#endif 
     return false;
 }
 
