--- sc/source/filter/excel/xlocx.cxx.orig	2008-07-11 16:17:23.000000000 +0800
+++ sc/source/filter/excel/xlocx.cxx	2008-07-11 16:19:20.000000000 +0800
@@ -363,7 +363,13 @@ void XclImpOcxConverter::RegisterOleMacr
     if ( rxModel.is() )
     {
         // Need to search basic for macro's named e.g. ControlButton1_Click
-        OUString sSearchString = rOcxCtrlObj.GetObjectName();
+        String sObjectName = rOcxCtrlObj.GetObjectName();
+        const XclImpRoot& rRoot = rOcxCtrlObj;
+        XclImpObjectManager& rObjectManager = rRoot.GetObjectManager();
+        String sOverRide( rObjectManager.GetOleNameOverride( rOcxCtrlObj.GetObjId() ) );
+        if ( sOverRide.Len() > 0 )
+            sObjectName = sOverRide;
+        OUString sSearchString = sObjectName;
         sSearchString += OUString::createFromAscii("_");
         ScfPropertySet aPropSet( rxModel );
  
@@ -413,8 +419,8 @@ void XclImpOcxConverter::RegisterOleMacr
                             if ( nSearchIndex == 0  )
                             {
                                 Sequence< OUString > sHandlerExtns(1);
-                                sHandlerExtns[ 0 ] = sMeth.copy( rOcxCtrlObj.GetObjectName().Len() );
-                                OUString sMacroName = sMeth.copy( 0, rOcxCtrlObj.GetObjectName().Len() );
+                                sHandlerExtns[ 0 ] = sMeth.copy( sObjectName.Len() );
+                                OUString sMacroName = sMeth.copy( 0, sObjectName.Len() );
                                 Sequence< ScriptEventDescriptor > aDescs( xEvtDescGen->getEventDescriptions( xCntrl, sLibModName, sHandlerExtns ) );
                                 if ( aDescs.getLength() )
                                 {
