diff --git basic/source/runtime/step2.cxx basic/source/runtime/step2.cxx
index 7975b65..ec9469f 100644
--- basic/source/runtime/step2.cxx
+++ basic/source/runtime/step2.cxx
@@ -381,7 +381,8 @@ void SbiRuntime::SetupArgs( SbxVariable* p, UINT32 nOp1 )
 				bool bError_ = true;
 
 				SbUnoMethod* pUnoMethod = PTR_CAST(SbUnoMethod,p);
-				if( pUnoMethod )
+                SbUnoProperty* pUnoProperty = PTR_CAST(SbUnoProperty,p);
+                if( pUnoMethod || pUnoProperty )
 				{
 					SbUnoObject* pParentUnoObj = PTR_CAST( SbUnoObject,p->GetParent() );
 					if( pParentUnoObj )
diff --git extensions/source/ole/oleobjw.cxx extensions/source/ole/oleobjw.cxx
index 32f0ffd..7707ecf 100644
--- extensions/source/ole/oleobjw.cxx
+++ extensions/source/ole/oleobjw.cxx
@@ -1531,6 +1531,9 @@ Any  IUnknownWrapper_Impl::invokeWithDispIdComTlb(FuncDesc& aFuncDesc,
         arDispidNamedArgs.reset(new DISPID[nSizeAr]);   
         HRESULT hr = getTypeInfo()->GetIDsOfNames(arNames, nSizeAr,
                                                   arDispidNamedArgs.get()); 
+        if ( hr == E_NOTIMPL )
+            hr = m_spDispatch->GetIDsOfNames(IID_NULL, arNames, nSizeAr, LOCALE_USER_DEFAULT, arDispidNamedArgs.get() );
+
         if (hr == S_OK)
         {
             // In a "property put" operation, the property value is a named param with the
