diff --git sc/source/ui/docshell/docsh5.cxx sc/source/ui/docshell/docsh5.cxx
index 8e38e2a..9a26cd2 100644
--- sc/source/ui/docshell/docsh5.cxx
+++ sc/source/ui/docshell/docsh5.cxx
@@ -843,11 +843,16 @@ BOOL ScDocShell::MoveTable( SCTAB nSrcTab, SCTAB nDestTab, BOOL bCopy, BOOL bRec
 			
 			StarBASIC* pStarBASIC = GetBasic();
 			BOOL bVbaEnabled = pStarBASIC ? pStarBASIC->isVBAEnabled() : FALSE;
+			SCTAB nTabToUse = nDestTab; 
+
+			if ( nDestTab == SC_TAB_APPEND )
+				nTabToUse = aDocument.GetMaxTableNumber() - 1;
+
 			if ( bVbaEnabled )
 			{
 				String sCodeName;
 				String sSource;
-				aDocument.GetCodeName( nDestTab, sCodeName );
+				aDocument.GetCodeName( nTabToUse, sCodeName );
 				com::sun::star::uno::Reference< com::sun::star::script::XLibraryContainer > xLibContainer = GetBasicContainer();
 				com::sun::star::uno::Reference< com::sun::star::container::XNameContainer > xLib;
     				if( xLibContainer.is() )
@@ -865,7 +870,7 @@ BOOL ScDocShell::MoveTable( SCTAB nSrcTab, SCTAB nDestTab, BOOL bCopy, BOOL bRec
 					xLib->getByName( sModName ) >>= sModuleInfo;
 					sSource = sModuleInfo.ModuleSource;
 				} 
-				lcl_InsertModule( *this, nDestTab, sCodeName, sSource );
+				lcl_InsertModule( *this, nTabToUse, sCodeName, sSource );
 			}
 		}
 
