diff --git sw/source/filter/ww8/ww8par.cxx sw/source/filter/ww8/ww8par.cxx
index a46a293..09a35ae 100644
--- sw/source/filter/ww8/ww8par.cxx
+++ sw/source/filter/ww8/ww8par.cxx
@@ -3919,9 +3919,13 @@ bool WW8Customizations::Import( SwDocShell* pShell )
     Tcg aTCG;
     long nCur = mpTableStream->Tell();
     mpTableStream->Seek( mWw8Fib.fcCmds ); // point at tgc record
-    if (!aTCG.Read( mpTableStream ) )
-        OSL_TRACE("** Read of Customization data failed!!!! ");
+    bool bReadResult = aTCG.Read( mpTableStream );
     mpTableStream->Seek( nCur ); // return to previous position, is that necessary?
+    if ( !bReadResult )
+    {
+        OSL_TRACE("** Read of Customization data failed!!!! ");
+        return false;
+    }
 #if DEBUG
     aTCG.Print( stderr );
 #endif
diff --git sw/source/filter/ww8/ww8toolbar.cxx sw/source/filter/ww8/ww8toolbar.cxx
index 42f22d1..f20a7b9 100644
--- sw/source/filter/ww8/ww8toolbar.cxx
+++ sw/source/filter/ww8/ww8toolbar.cxx
@@ -105,6 +105,7 @@ bool CTBWrapper::Read( SvStream* pS )
     Tcg255SubStruct::Read( pS );
     *pS >> reserved2 >> reserved3 >> reserved4 >> reserved5;
     *pS >> cbTBD >> cCust >> cbDTBC;
+    long nExpectedPos =  pS->Tell() + cbDTBC;
     if ( cbDTBC )
     {
         // cbDTBC is the size in bytes of the TBC array
@@ -125,6 +126,16 @@ bool CTBWrapper::Read( SvStream* pS )
             bytesToRead = cbDTBC - ( pS->Tell() - nStart ); 
         } while ( bytesToRead > 0 );
     }
+    if ( pS->Tell() != nExpectedPos )
+    {
+        // Strange error condition, shouldn't happen ( but does in at least
+        // one test document ) In the case where it happens the TBC & 
+        // TBCHeader records seem blank??? ( and incorrect )
+        OSL_ENSURE( pS->Tell() == nExpectedPos, "### Error: Expected pos not equal to actual pos after reading rtbdc");
+        OSL_TRACE("\tPos now is 0x%x should be 0x%x", pS->Tell(), nExpectedPos );
+        // seek to correct position after rtbdc
+        pS->Seek( nExpectedPos );
+    }
     if ( cCust )
     {
         for ( sal_Int32 index = 0; index < cCust; ++index )
@@ -773,12 +784,15 @@ void Tcg::Print( FILE* fp )
     Indent a(true);
     indent_printf(fp, "[ 0x%x ] Tcg - dump %d\n", nOffSet, nTcgVer);
     indent_printf(fp,"  nTcgVer %d\n", nTcgVer);
-    tcg->Print( fp );
+    if ( tcg.get() )
+        tcg->Print( fp );
 }
 
 bool Tcg::ImportCustomToolBar( SfxObjectShell& rDocSh )
 {
-    return tcg->ImportCustomToolBar( rDocSh );
+    if ( tcg.get() )
+        return tcg->ImportCustomToolBar( rDocSh );
+    return false;
 }
 
 Tcg255::Tcg255()
