diff --git sc/source/ui/vba/vbarange.cxx sc/source/ui/vba/vbarange.cxx
index 8724906..0d0b86a 100644
--- sc/source/ui/vba/vbarange.cxx
+++ sc/source/ui/vba/vbarange.cxx
@@ -1969,6 +1969,19 @@ ScVbaRange::Select() throw (uno::Runtime
 			xSelection->select( uno::makeAny( mxRanges ) );
 		else
 			xSelection->select( uno::makeAny( mxRange ) );
+		// set focus on document e.g.
+		// ThisComponent.CurrentController.Frame.getContainerWindow.SetFocus
+		try
+		{
+			uno::Reference< frame::XController > xController( getCurrentDocument()->getCurrentController(), uno::UNO_QUERY_THROW );
+			uno::Reference< frame::XFrame > xFrame( xController->getFrame(), uno::UNO_QUERY_THROW );
+			uno::Reference< awt::XWindow > xWin( xFrame->getContainerWindow(), uno::UNO_QUERY_THROW );
+			xWin->setFocus();
+		}
+		catch( uno::Exception& )
+		{
+		}		
+
 	}
 }
 
