diff --git vbahelper/inc/vbahelper/vbahelper.hxx vbahelper/inc/vbahelper/vbahelper.hxx
index 4cc1082..ab07aed 100644
--- vbahelper/inc/vbahelper/vbahelper.hxx
+++ vbahelper/inc/vbahelper/vbahelper.hxx
@@ -36,6 +36,7 @@
 #include <com/sun/star/awt/XControl.hpp>
 #include <com/sun/star/awt/XDevice.hpp>
 #include <com/sun/star/frame/XDispatchResultListener.hpp>
+#include <com/sun/star/awt/XUnitConversion.hpp>
 #include <com/sun/star/frame/DispatchResultEvent.hpp>
 #include <com/sun/star/frame/DispatchResultState.hpp>
 #include <com/sun/star/lang/EventObject.hpp>
@@ -48,7 +49,6 @@
 #include <sfx2/docfilt.hxx>
 #include <sfx2/docfile.hxx>
 #include <vcl/pointr.hxx>
-#include <oox/helper/graphichelper.hxx>
 #include <memory>
 #include "vbaaccesshelper.hxx"
 
@@ -195,7 +195,6 @@ class VBAHELPER_DLLPUBLIC UserFormGeomet
 {
     css::uno::Reference< css::awt::XUnitConversion > mxControlUnits;
     css::uno::Reference< css::beans::XPropertySet > mxModel;
-    std::auto_ptr< oox::GraphicHelper > mpGraphicHelper;
 
     sal_Int32 ConvertLogicToPixel( sal_Int32 nValue, sal_Bool bIsPoint, sal_Bool bIsX, sal_Int16 nSourceUnit );
     sal_Int32 ConvertPixelToLogic( sal_Int32 nValue, sal_Bool bIsPoint, sal_Bool bIsX, sal_Int16 nTargetUnit );
diff --git vbahelper/prj/build.lst vbahelper/prj/build.lst
index ded9ee2..20001db 100644
--- vbahelper/prj/build.lst
+++ vbahelper/prj/build.lst
@@ -1,4 +1,4 @@
-vba     vbahelper       :       oovbaapi oox basic sfx2 svx filter cppuhelper vcl comphelper svtools tools sal NULL
+vba     vbahelper       :       oovbaapi basic sfx2 svx filter cppuhelper vcl comphelper svtools tools sal NULL
 vba	vbahelper									usr1	-	all	  vba_mkout NULL
 #vba	vbahelper\inc								nmake	-	all	vba_inc NULL
 vba	vbahelper\source\vbahelper                        nmake	-	all	vba_vbahelper NULL
--- vbahelper/util/makefile.mk.old	2010-08-03 15:12:54.000000000 +0200
+++ vbahelper/util/makefile.mk	2010-08-03 17:44:26.000000000 +0200
@@ -61,7 +61,6 @@ SHL1STDLIBS=       \
         $(VCLLIB) \
         $(SVTOOLLIB)    \
         $(MSFILTERLIB)    \
-		$(OOXLIB)    \
         $(TKLIB)
 
 SHL1DEPN=
