--- sc/source/ui/vba/vbarange.cxx	2008-07-01 10:43:33.000000000 +0100
+++ sc/source/ui/vba/vbarange.cxx	2008-07-07 13:35:08.000000000 +0100
@@ -223,18 +223,21 @@ uno::Reference< excel::XRange > lcl_make
 		ScUnoConversion::FillScRange( refRange, sAddresses[ index ] );
 		aCellRanges.Append( refRange );
 	}
-	// Single range
-	if ( aCellRanges.First() == aCellRanges.Last() )
-	{
-		uno::Reference< table::XCellRange > xTmpRange( new ScCellRangeObj( pDoc, *aCellRanges.First() ) );
-		// #FIXME need proper (WorkSheet) parent
-		xRange = new ScVbaRange( xParent, xContext, xTmpRange );
-	}
-	else
+	if ( nLen )
 	{
-		uno::Reference< sheet::XSheetCellRangeContainer > xRanges( new ScCellRangesObj( pDoc, aCellRanges ) );
-		// #FIXME need proper (WorkSheet) parent
-		xRange = new ScVbaRange( xParent, xContext, xRanges );
+		// Single range
+		if ( aCellRanges.First() == aCellRanges.Last() )
+		{
+			uno::Reference< table::XCellRange > xTmpRange( new ScCellRangeObj( pDoc, *aCellRanges.First() ) );
+			// #FIXME need proper (WorkSheet) parent
+			xRange = new ScVbaRange( xParent, xContext, xTmpRange );
+		}
+		else
+		{
+			uno::Reference< sheet::XSheetCellRangeContainer > xRanges( new ScCellRangesObj( pDoc, aCellRanges ) );
+			// #FIXME need proper (WorkSheet) parent
+			xRange = new ScVbaRange( xParent, xContext, xRanges );
+		}
 	}
 	return xRange;
 }
