--- unixODBC/inc/sqltypes.h.orig	2008-05-25 15:07:18.000000000 +0200
+++ unixODBC/inc/sqltypes.h	2008-05-25 15:08:05.000000000 +0200
@@ -31,7 +31,7 @@
 #endif
 
 #ifndef SIZEOF_LONG
-# if defined(__alpha__) || defined(__x86_64__) || defined(__ia64__)
+# if defined(__alpha__) || defined(__x86_64__) || defined(__ia64__) || defined(__ppc64__)
 # define SIZEOF_LONG        8
 #else
 # define SIZEOF_LONG        4
