--- config_office/configure.in-old	2008-01-28 10:21:21.000000000 +0000
+++ config_office/configure.in	2008-01-28 10:22:05.000000000 +0000
@@ -2734,6 +2734,7 @@
 
 AWTLIB=
 if test "$SOLAR_JAVA" != ""; then
+   AC_LANG_PUSH([C])
    AC_MSG_CHECKING([for jawt lib name])
    if test "$JDK" = "gcj"; then 
       save_CFLAGS=$CFLAGS
@@ -2774,6 +2775,7 @@
    if test -z "$AWTLIB"; then
       AWTLIB=-ljawt
    fi
+   AC_LANG_POP([C])
    AC_MSG_RESULT([$AWTLIB])
    AC_SUBST(AWTLIB)
 fi
