diff -rup canvas/source/cairo-orig/cairo_cairo.cxx canvas/source/cairo/cairo_cairo.cxx
--- canvas/source/cairo-orig/cairo_cairo.cxx	2008-04-28 18:00:30.000000000 +0200
+++ canvas/source/cairo/cairo_cairo.cxx	2008-04-28 18:10:30.000000000 +0200
@@ -353,7 +353,7 @@ namespace cairo
 
 			return new Surface( mpSysData, mpDisplay, (long) hPixmap, pFormat,
 								cairo_xlib_surface_create_with_xrender_format( (Display*) mpDisplay, hPixmap,
-																			   DefaultScreenOfDisplay( (Display *) mpDisplay ),
+																			   (Screen*) cairoHelperGetScreen( mpSysData ),
 																			   pFormat, width, height ) );
 		} else
 			return new Surface( mpSysData, mpDisplay, 0, NULL, cairo_surface_create_similar( mpSurface, aContent, width, height ) );
diff -rup canvas/source/cairo-orig/cairo_helper.cxx canvas/source/cairo/cairo_helper.cxx
--- canvas/source/cairo-orig/cairo_helper.cxx	2008-04-28 18:00:30.000000000 +0200
+++ canvas/source/cairo/cairo_helper.cxx	2008-04-28 18:11:31.000000000 +0200
@@ -57,13 +57,19 @@ namespace cairo {
 #endif
 }
 
+#include "cairo_helper.hxx"
+
 using namespace ::cairo;
 
 //#define OOO_CANVAS_GLITZ 1
 
-int cairoHelperGetDefaultScreen( void* display )
+void* cairoHelperGetScreen( const void* pSysData )
 {
-    return DefaultScreen( ( (Display *) display ) );
+    XWindowAttributes aWA;
+
+    XGetWindowAttributes( (::cairo::Display*) cairoHelperGetDisplay( pSysData ), cairoHelperGetWindow( pSysData ), &aWA );
+
+    return aWA.screen;
 }
 
 void*
@@ -74,7 +80,7 @@ cairoHelperGetGlitzDrawable( const Syste
     Display* display = (Display*) pSysData->pDisplay;
     ::cairo::Window window = pSysData->aWindow;
     VisualID vid = XVisualIDFromVisual( (Visual*) pSysData->pVisual );
-    int screen = DefaultScreen( display );
+    int screen = cairoHelperGetScreen( pSysData );
 
     glitz_drawable_t *pDrawable = NULL;
     glitz_format_t* pFormat = NULL;
diff -rup canvas/source/cairo-orig/cairo_helper.hxx canvas/source/cairo/cairo_helper.hxx
--- canvas/source/cairo-orig/cairo_helper.hxx	2008-04-28 18:00:30.000000000 +0200
+++ canvas/source/cairo/cairo_helper.hxx	2008-04-28 18:10:10.000000000 +0200
@@ -1,4 +1,4 @@
-int   cairoHelperGetDefaultScreen( void* display );
+void* cairoHelperGetScreen( const void* pSysData );
 void* cairoHelperGetGlitzDrawable( const void* pSysData, int width, int height );
 void* cairoHelperGetGlitzSurface( const void* pSysData, void *drawable, int x, int y, int width, int height );
 void* cairoHelperGetSurface( const void* pSysData, int x, int y, int width, int height );
