--- config_office/configure.in	2007-03-22 16:54:38.000000000 +0100
+++ config_office/configure.in	2007-03-29 15:24:14.000000000 +0200
@@ -1548,6 +1616,10 @@ if test "$_os" = "WINNT"; then
 		if test -e "$with_mspdb_path/mspdb71.dll"; then
 			MSPDB_PATH="$with_mspdb_path"
 		fi
+		dnl  .NET 2005 Compiler
+		if test -e "$with_mspdb_path/mspdb80.dll"; then
+			MSPDB_PATH="$with_mspdb_path"
+		fi
 		dnl .NET case
 		if test -z "$MSPDB_PATH" -a -e "$with_cl_home/../Common7/IDE/mspdb71.dll"; then
 			MSPDB_PATH="$with_cl_home/../Common7/IDE"
--- config_office/oowintool	12 Dec 2007 13:24:57 -0000	1.4
+++ config_office/oowintool	14 Jul 2008 22:01:44 -0000
@@ -101,6 +101,16 @@
         $key = reg_find_key ('HKEY_LOCAL_MACHINE/SOFTWARE/Microsoft/MicrosoftSDK/InstalledSDKs/*/Install Dir');
         $value = reg_get_value ($key);
     }
+    if (!defined $value)
+    {
+        $key = reg_find_key ('HKEY_LOCAL_MACHINE/SOFTWARE/Microsoft/Microsoft SDKs/Windows/CurrentInstallFolder');
+        $value = reg_get_value ($key);
+    }
+    if (!defined $value)
+    {
+        my $dir = cygpath (find_msvc()->{'product_dir'}, 'w', $output_format);
+	$value = `/bin/find "$dir" -iname platformsdk | head -n 1`;
+    }
     defined $value || die "psdk not found";
 
     print cygpath ($value, 'w', $output_format);
@@ -145,10 +155,38 @@
     'dll_path' => 'Visual Studio .NET Enterprise Architect 2003 - English', # testme ...
     'dll_suffix' => '71'
 );
+my %msvs_express_2005 = (
+    'ver' => '8.0',
+    'key' => 'Microsoft/VCExpress/8.0/Setup/VS/ProductDir',
+    'instmsi_path' => '../SDK/v2.0/BootStrapper/Packages/InstMSI',
+    'dll_path' => '../SDK/v2.0/Bin',
+    'dll_suffix' => '80'
+);
+my %msvc_express_2005 = (
+    'ver' => '8.0',
+    'key' => 'Microsoft/VCExpress/8.0/Setup/VC/ProductDir',
+    'instmsi_path' => '../SDK/v2.0/BootStrapper/Packages/InstMSI',
+    'dll_path' => '../SDK/v2.0/Bin',
+    'dll_suffix' => '80'
+);
+my %msvs_2005 = (
+    'ver' => '8.0',
+    'key' => 'Microsoft/VisualStudio/8.0/Setup/VS/ProductDir',
+    'instmsi_path' => 'SDK/v2.0/BootStrapper/Packages/InstMSI',
+    'dll_path' => 'Visual Studio .NET Professional 2005 - English',
+    'dll_suffix' => '80'
+);
+my %msvc_2005 = (
+    'ver' => '8.0',
+    'key' => 'Microsoft/VisualStudio/8.0/Setup/VC/ProductDir',
+    'instmsi_path' => '../SDK/v2.0/BootStrapper/Packages/InstMSI',
+    'dll_path' => '../SDK/v2.0/Bin',
+    'dll_suffix' => '80'
+);
 
 sub find_msvs()
 {
-    my @ms_versions = ( \%msvs_net_2003_ea, \%msvs_net_2003, \%msvs_net_2002, \%msvc6 );
+    my @ms_versions = ( \%msvs_2005, \%msvs_express_2005, \%msvs_net_2003_ea, \%msvs_net_2003, \%msvs_net_2002, \%msvc6 );
 
     for $ver (@ms_versions)
     {
@@ -163,7 +201,7 @@
 
 sub find_msvc()
 {
-    my @ms_versions = ( \%msvc_net_2003, \%msvc_net_2002, \%msvc6 );
+    my @ms_versions = ( \%msvc_2005, \%msvc_express_2005, \%msvc_net_2003, \%msvc_net_2002, \%msvc6 );
 
     for $ver (@ms_versions)
     {
@@ -196,14 +234,16 @@
 
 sub print_csc_compiler_dir()
 {
-    my $dir = reg_get_value ("HKEY_LOCAL_MACHINE/SOFTWARE/Microsoft/.NETFramework/InstallRoot");
-    $dir .= "v1.1.4322"; # lame-but ...
-    print cygpath ($dir, 'w', $output_format);
+    my $dir = cygpath (reg_get_value ("HKEY_LOCAL_MACHINE/SOFTWARE/Microsoft/.NETFramework/InstallRoot"), 'w', $output_format);
+    my $csc_exe = `/bin/find "$dir" -iname csc.exe | head -n 1`;
+    print `dirname $csc_exe`;
 }
 
 sub print_dotnetsdk_dir()
 {
-    my $dir = reg_get_value ("HKEY_LOCAL_MACHINE/SOFTWARE/Microsoft/.NETFramework/sdkInstallRootv1.1");
+    my $dir = 
+          reg_get_value ("HKEY_LOCAL_MACHINE/SOFTWARE/Microsoft/.NETFramework/sdkInstallRootv1.1") ||
+          reg_get_value ("HKEY_LOCAL_MACHINE/SOFTWARE/Microsoft/.NETFramework/sdkInstallRootv2.0");
     print cygpath ($dir, 'w', $output_format);
 }
 
@@ -211,7 +251,8 @@
 {
     my $dir =
 	  reg_get_value ("HKEY_LOCAL_MACHINE/SOFTWARE/JavaSoft/Java\ Development\ Kit/1.5/JavaHome") ||
-	  reg_get_value ("HKEY_LOCAL_MACHINE/SOFTWARE/JavaSoft/Java\ Development\ Kit/1.4/JavaHome");
+	  reg_get_value ("HKEY_LOCAL_MACHINE/SOFTWARE/JavaSoft/Java\ Development\ Kit/1.4/JavaHome") ||
+	  reg_get_value ("HKEY_LOCAL_MACHINE/SOFTWARE/JavaSoft/Java\ Development\ Kit/1.3/JavaHome");
     print cygpath($dir, 'w', $output_format); 
 }
 
