--- config_office/configure.in	7 Mar 2008 16:56:35 -0000	1.252
+++ config_office/configure.in	25 Mar 2008 15:51:13 -0000	1.252.6.6
@@ -4443,9 +4476,20 @@ are installed or use --with-psdk-home .]
          -o ! -x "$PSDK_HOME/bin/msitran.exe" ; then
         AC_MSG_ERROR([Some (all) files of the Windows Installer SDK are missing, please install.])
     fi
-    AC_MSG_RESULT([PSDK files found ($PSDK_HOME)])
+    AC_MSG_RESULT([SDK files found ...)])
+dnl Check if this is the right SDK.
+    if echo $PSDK_HOME | grep "Windows/v6.1" >/dev/null 2>/dev/null; then
+        AC_MSG_RESULT([Found Windows SDK 6.1 ($PSDK_HOME)])
+        WINDOWS_VISTA_PSDK=TRUE
+    elif echo $PSDK_HOME | grep "Windows/v6.0" >/dev/null 2>/dev/null; then
+        AC_MSG_RESULT([Found Windows SDK 6.0 ($PSDK_HOME)])
+        WINDOWS_VISTA_PSDK=TRUE
+    else
+        AC_MSG_RESULT([Found Legacy Windows Platform SDK ($PSDK_HOME)])
+    fi
 fi
 AC_SUBST(PSDK_HOME)
+AC_SUBST(WINDOWS_VISTA_PSDK)
 
 dnl =========================================
 dnl Check for the Microsoft DirectX SDK.
--- config_office/set_soenv.in	7 Mar 2008 16:56:50 -0000	1.178
+++ config_office/set_soenv.in	23 Mar 2008 13:49:34 -0000	1.178.10.2
@@ -1784,6 +1786,7 @@ ToFile( "Variable values.",  $empty,    
 if ( $platform =~ m/cygwin/ )
 {  ToFile( "CYGWIN_PATH",       $CYGWIN_PATH,       "e" );
    ToFile( "PSDK_HOME",         $PSDK_HOME,         "e" );
+   ToFile( "WINDOWS_VISTA_PSDK","@WINDOWS_VISTA_PSDK@","e" );
    ToFile( "DIRECTXSDK_LIB",    $DIRECTXSDK_LIB,    "e" );
    ToFile( "USE_DIRECTX5",      $USE_DIRECTX5,      "e" );
    ToFile( "ATL_LIB",           $ATL_LIB,         "e" );
