--- sfx2/source/doc/docfile.cxx	25 Feb 2005 09:44:12 -0000	1.161
+++ sfx2/source/doc/docfile.cxx	15 Mar 2005 16:18:30 -0000	1.161.12.1
@@ -1429,7 +1429,7 @@ sal_Bool SfxMedium::TransactedTransferFo
        eError = ERRCODE_IO_GENERAL;
     }
 
-    if( !eError )
+    if( !eError || (eError & ERRCODE_WARNING_MASK) )
 	{
 		Close();
 		::ucb::Content aTempCont;
@@ -1530,7 +1530,7 @@ void SfxMedium::Transfer_Impl()
 			OSL_ENSURE( sal_False, "The medium name is not convertable!\n" );
 	}
 
-	if ( aNameURL.Len() && ( !eError || eError & ERRCODE_WARNING_MASK ) )
+    if ( aNameURL.Len() && ( !eError || (eError & ERRCODE_WARNING_MASK) ) )
     {
         Reference < ::com::sun::star::ucb::XCommandEnvironment > xEnv;
 		Reference< XOutputStream > rOutStream;
@@ -1724,7 +1724,7 @@ void SfxMedium::Transfer_Impl()
                 eError = ERRCODE_IO_GENERAL;
             }
 
-			if (!eError)
+            if ( !eError || (eError & ERRCODE_WARNING_MASK) )
             {
                 // free resources, otherwise the transfer may fail
                 Close();
@@ -1781,7 +1781,7 @@ void SfxMedium::Transfer_Impl()
 	    	}
 		}
 
-		if ( !eError && !pImp->pTempFile )
+        if ( ( !eError || (eError & ERRCODE_WARNING_MASK) ) && !pImp->pTempFile )
 		{
 			// without a TempFile the physical and logical name should be the same after successful transfer
   			::utl::LocalFileHelper::ConvertURLToPhysicalName( GetURLObject().GetMainURL( INetURLObject::NO_DECODE ),
