Index: sw/source/ui/shells/textsh1.cxx
===================================================================
RCS file: /cvs/sw/sw/source/ui/shells/textsh1.cxx,v
retrieving revision 1.58
retrieving revision 1.58.60.1
diff -u -p -u -p -r1.58 -r1.58.60.1
--- sw/source/ui/shells/textsh1.cxx	3 Jul 2007 14:18:26 -0000	1.58
+++ sw/source/ui/shells/textsh1.cxx	9 Oct 2007 13:14:22 -0000	1.58.60.1
@@ -911,7 +911,7 @@ void SwTextShell::Execute(SfxRequest &rR
                 pSet = (SfxItemSet*) pArgs;
 
             }
-            else if ( pDlg->Execute() == RET_OK )
+            else if ( NULL != pDlg && pDlg->Execute() == RET_OK )
             {
                 // Defaults evtl umsetzen
                 pSet = (SfxItemSet*)pDlg->GetOutputItemSet();
