Index: vcl/source/gdi/virdev.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/gdi/virdev.cxx,v
retrieving revision 1.30
retrieving revision 1.30.60.1
diff -u -p -u -p -r1.30 -r1.30.60.1
--- vcl/source/gdi/virdev.cxx	24 Jul 2007 10:15:08 -0000	1.30
+++ vcl/source/gdi/virdev.cxx	25 Sep 2007 09:52:17 -0000	1.30.60.1
@@ -119,7 +119,6 @@ void VirtualDevice::ImplInitVirDev( cons
 	mnOutWidth		= nDX;
 	mnOutHeight 	= nDY;
 	mbScreenComp	= TRUE;
-	mbScreenComp	= FALSE;
     mnAlphaDepth	= -1;
 
     // #i59315# init vdev size from system object, when passed a
