Index: svtools/source/svhtml/htmlout.cxx
===================================================================
RCS file: /cvs/util/svtools/source/svhtml/htmlout.cxx,v
retrieving revision 1.22
retrieving revision 1.22.80.1
diff -u -p -u -p -r1.22 -r1.22.80.1
--- svtools/source/svhtml/htmlout.cxx	18 Jul 2007 08:56:49 -0000	1.22
+++ svtools/source/svhtml/htmlout.cxx	17 Oct 2007 15:05:30 -0000	1.22.80.1
@@ -717,7 +717,10 @@ SvStream& HTMLOutFuncs::Out_ImageMap( Sv
 					Out_String( rStream, rTarget, eDestEnc, pNonConvertableChars ) << '\"';
 				}
 
-				const String& rDesc = pObj->GetAltText();
+				String rDesc( pObj->GetAltText() );
+				if( rDesc.Len() == 0 )
+					rDesc = pObj->GetDesc();
+
 				if( rDesc.Len() )
 				{
 					((sOut = ' ') += sHTML_O_alt) += "=\"";
Index: xmloff/source/draw/XMLImageMapContext.cxx
===================================================================
RCS file: /cvs/xml/xmloff/source/draw/XMLImageMapContext.cxx,v
retrieving revision 1.13
retrieving revision 1.13.60.1
diff -u -p -u -p -r1.13 -r1.13.60.1
--- xmloff/source/draw/XMLImageMapContext.cxx	27 Jun 2007 14:59:11 -0000	1.13
+++ xmloff/source/draw/XMLImageMapContext.cxx	17 Oct 2007 15:03:27 -0000	1.13.60.1
@@ -388,7 +388,7 @@ void XMLImageMapObjectContext::Prepare(
 	rPropertySet->setPropertyValue( sURL, Any( sUrl ) );
 	rPropertySet->setPropertyValue( sTitle, Any( sTitleBuffer.makeStringAndClear() ) );
 	rPropertySet->setPropertyValue( sDescription, Any( sDescriptionBuffer.makeStringAndClear() ) );
-	rPropertySet->setPropertyValue( sTarget, Any( sTarget ) );
+	rPropertySet->setPropertyValue( sTarget, Any( sTargt ) );
 	rPropertySet->setPropertyValue( sIsActive, Any( bIsActive ) );
 	rPropertySet->setPropertyValue( sName, Any( sNam ) );
 }
Index: sd/source/ui/dlg/headerfooterdlg.cxx
===================================================================
RCS file: /cvs/graphics/sd/source/ui/dlg/headerfooterdlg.cxx,v
retrieving revision 1.15
retrieving revision 1.15.114.1
diff -u -p -u -p -r1.15 -r1.15.114.1
--- sd/source/ui/dlg/headerfooterdlg.cxx	10 May 2007 15:27:11 -0000	1.15
+++ sd/source/ui/dlg/headerfooterdlg.cxx	17 Sep 2007 13:40:49 -0000	1.15.114.1
@@ -791,7 +791,9 @@ void HeaderFooterTabPage::GetOrSetDateTi
 
 			EditEngine* pEdit = const_cast< EditEngine* >(&pOutl->GetEditEngine());
 
-			pOutl->SetText( *pObj->GetOutlinerParaObject() );
+			OutlinerParaObject* pOPO = pObj->GetOutlinerParaObject();
+			if( pOPO )
+				pOutl->SetText( *pOPO );
 
 			EFieldInfo aFieldInfo;
 			aFieldInfo.pFieldItem = NULL;
Index: sd/source/ui/view/drtxtob1.cxx
===================================================================
RCS file: /cvs/graphics/sd/source/ui/view/drtxtob1.cxx,v
retrieving revision 1.29
retrieving revision 1.29.62.1
diff -u -p -u -p -r1.29 -r1.29.62.1
--- sd/source/ui/view/drtxtob1.cxx	18 Jul 2007 09:42:55 -0000	1.29
+++ sd/source/ui/view/drtxtob1.cxx	17 Oct 2007 13:35:49 -0000	1.29.62.1
@@ -362,7 +362,7 @@ void TextObjectBar::Execute( SfxRequest 
             aAttr.Put( SvxWritingModeItem(
                 nSlot == SID_TEXTDIRECTION_LEFT_TO_RIGHT ?
                     com::sun::star::text::WritingMode_LR_TB : com::sun::star::text::WritingMode_TB_RL,
-                    EE_PARA_WRITINGDIR ) );
+                    SDRATTR_TEXTDIRECTION ) );
 			rReq.Done( aAttr );
 			mpView->SetAttributes( aAttr );
 			Invalidate();
Index: svx/source/dialog/tpcolor.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/dialog/tpcolor.cxx,v
retrieving revision 1.29
retrieving revision 1.29.114.1
diff -u -p -u -p -r1.29 -r1.29.114.1
--- svx/source/dialog/tpcolor.cxx	27 Jun 2007 17:48:39 -0000	1.29
+++ svx/source/dialog/tpcolor.cxx	18 Sep 2007 13:20:48 -0000	1.29.114.1
@@ -513,7 +513,7 @@ IMPL_LINK( SvxColorTabPage, ClickAddHdl_
 		pWindow = pWindow->GetParent();
 	}
 
-	ResMgr rMgr = DIALOG_MGR();
+	ResMgr& rMgr = DIALOG_MGR();
 	String aDesc( ResId( RID_SVXSTR_DESC_COLOR, rMgr ) );
 	String aName( aEdtName.GetText() );
 	XColorEntry* pEntry;
