Index: reportdesign/java/com/sun/star/report/makefile.mk
===================================================================
RCS file: /cvs/dba/reportdesign/java/com/sun/star/report/makefile.mk,v
retrieving revision 1.4
retrieving revision 1.4.56.1
diff -u -u -r1.4 -r1.4.56.1
--- reportdesign/java/com/sun/star/report/makefile.mk	3 Aug 2007 09:48:29 -0000	1.4
+++ reportdesign/java/com/sun/star/report/makefile.mk	2 Dec 2007 12:34:15 -0000	1.4.56.1
@@ -42,7 +42,13 @@
 .INCLUDE: settings.mk
 #----- compile .java files -----------------------------------------
 
-JARFILES        = ridl.jar unoil.jar jurt.jar juh.jar jut.jar java_uno.jar jcommon-1.0.10.jar
+JARFILES        = ridl.jar unoil.jar jurt.jar juh.jar jut.jar java_uno.jar
+.IF "$(SYSTEM_JFREEREPORT)" == "YES"
+XCLASSPATH!:==$(XCLASSPATH)$(PATH_SEPERATOR)$(JCOMMON_JAR)
+.ELSE
+JARFILES += jcommon-1.0.10.jar
+.ENDIF
+
 JAVAFILES       :=	DataRow.java\
 					DataSource.java\
 					DataSourceException.java\
Index: reportdesign/java/com/sun/star/report/pentaho/makefile.mk
===================================================================
RCS file: /cvs/dba/reportdesign/java/com/sun/star/report/pentaho/makefile.mk,v
retrieving revision 1.4
retrieving revision 1.4.4.1
diff -u -u -r1.4 -r1.4.4.1
--- reportdesign/java/com/sun/star/report/pentaho/makefile.mk	6 Nov 2007 15:42:30 -0000	1.4
+++ reportdesign/java/com/sun/star/report/pentaho/makefile.mk	2 Dec 2007 12:34:15 -0000	1.4.4.1
@@ -44,7 +44,13 @@
 
 #----- compile .java files -----------------------------------------
 
-JARFILES        = ridl.jar unoil.jar jurt.jar juh.jar jut.jar java_uno.jar jcommon-1.0.10.jar libxml-0.9.5.jar jfreereport-0.9.0-05.jar libloader-0.3.4.jar sac.jar liblayout-0.2.6.jar jcommon-serializer-0.1.0.jar libfonts-0.2.6.jar libformula-0.1.8.jar
+JARFILES        = ridl.jar unoil.jar jurt.jar juh.jar jut.jar java_uno.jar
+.IF "$(SYSTEM_JFREEREPORT)" == "YES"
+XCLASSPATH!:=$(XCLASSPATH)$(PATH_SEPERATOR)$(JCOMMON_JAR)$(PATH_SEPERATOR)$(LIBXML_JAR)$(PATH_SEPERATOR)$(JFREEREPORT_JAR)$(PATH_SEPERATOR)$(LIBLOADER_JAR)$(PATH_SEPERATOR)$(SAC_JAR)$(PATH_SEPERATOR)$(LIBLAYOUT_JAR)$(PATH_SEPERATOR)$(JCOMMON_SERIALIZER_JAR)$(PATH_SEPERATOR)$(LIBFONTS_JAR)$(PATH_SEPERATOR)$(LIBFORMULA_JAR)
+.ELSE
+JARFILES += jcommon-1.0.10.jar libxml-0.9.5.jar jfreereport-0.9.0-05.jar libloader-0.3.4.jar sac.jar liblayout-0.2.6.jar jcommon-serializer-0.1.0.jar libfonts-0.2.6.jar libformula-0.1.8.jar
+.ENDIF
+
 JAVAFILES       := $(shell @$(FIND) . -name "*.java")
 # JAVACLASSFILES	= $(foreach,i,$(JAVAFILES) $(CLASSDIR)$/$(PACKAGE)$/$(i:s/.java/.class/))
 
Index: reportdesign/java/com/sun/star/report/util/makefile.mk
===================================================================
RCS file: /cvs/dba/reportdesign/java/com/sun/star/report/util/makefile.mk,v
retrieving revision 1.4
retrieving revision 1.4.56.1
diff -u -u -r1.4 -r1.4.56.1
--- reportdesign/java/com/sun/star/report/util/makefile.mk	3 Aug 2007 09:52:41 -0000	1.4
+++ reportdesign/java/com/sun/star/report/util/makefile.mk	2 Dec 2007 12:34:15 -0000	1.4.56.1
@@ -42,7 +42,12 @@
 .INCLUDE: settings.mk
 #----- compile .java files -----------------------------------------
 
+.IF "$(SYSTEM_JFREEREPORT)" == "YES"
+CLASSPATH!:=$(CLASSPATH)$(PATH_SEPERATOR)$(JCOMMON_JAR)$(PATH_SEPERATOR)$(LIBXML_JAR)$(PATH_SEPERATOR)$(JFREEREPORT_JAR)
+.ELSE
 JARFILES        = jcommon-1.0.10.jar libxml-0.9.5.jar jfreereport-0.9.0-05.jar
+.ENDIF
+
 JAVAFILES       = DefaultJobProperties.java DefaultParameterMap.java DefaultReportJobDefinition.java FileRepository.java ManifestWriter.java
 # --- Targets ------------------------------------------------------
 .INCLUDE :  target.mk
Index: reportdesign/prj/build.lst
===================================================================
RCS file: /cvs/dba/reportdesign/prj/build.lst,v
retrieving revision 1.5
retrieving revision 1.5.4.2
diff -u -u -r1.5 -r1.5.4.2
--- reportdesign/prj/build.lst	27 Nov 2007 11:57:56 -0000	1.5
+++ reportdesign/prj/build.lst	3 Dec 2007 15:39:29 -0000	1.5.4.2
@@ -1,4 +1,4 @@
-rd  reportdesign    :   boost jfreereport comphelper dbaccess NULL
+rd  reportdesign    :   boost JFREEREPORT:jfreereport comphelper dbaccess readlicense_oo NULL
 rd  reportdesign                                                        usr1    -   all rd_mkout NULL
 rd  reportdesign\inc                                                    nmake   -   all rd_inc NULL
 rd  reportdesign\java							nmake   -   all rd_java NULL
Index: reportdesign/util/makefile.mk
===================================================================
RCS file: /cvs/dba/reportdesign/util/makefile.mk,v
retrieving revision 1.9
retrieving revision 1.9.4.2
diff -u -u -r1.9 -r1.9.4.2
--- reportdesign/util/makefile.mk	27 Nov 2007 11:59:51 -0000	1.9
+++ reportdesign/util/makefile.mk	3 Dec 2007 15:39:29 -0000	1.9.4.2
@@ -233,7 +233,7 @@
 LICLINES:=$(foreach,i,$(TXTFILES)  <license-text xlink:href="registration/$(i:f)" lang="$(subst,.txt, $(subst,license_, $(i:f)))" license-id="$(subst,.txt, $(subst,license_, $(i:f)))" />)
 .ENDIF  # "$(GUI)"!="WNT"
 
-REPRORTJARFILES := \
+REPORTJARFILES := \
 	$(ZIP1DIR)$/jcommon-1.0.10.jar										\
 	$(ZIP1DIR)$/sac.jar													\
 	$(ZIP1DIR)$/libxml-0.9.5.jar										\
@@ -243,7 +243,7 @@
 .INCLUDE : target.mk
 
 .IF "$(ZIP1TARGETN)"!=""
-$(ZIP1TARGETN) :  $(TXTFILES) $(XMLFILES) $(HTMLFILES) $(REPRORTJARFILES)
+$(ZIP1TARGETN) :  $(TXTFILES) $(XMLFILES) $(HTMLFILES) $(REPORTJARFILES)
 .ENDIF # "$(ZIP1TARGETN)"!=""
 
 $(ZIP1DIR)$/description.xml : pre.xml post.xml
@@ -286,9 +285,15 @@
 	echo $(XCU_TMP)
     $(COPY) $< $@
 
+.IF "$(SYSTEM_JFREEREPORT)" == "YES"
+$(ZIP1DIR)$/%.jar : $(JFREEREPORT_JAR:d:d)$/%.jar
+	@@-$(MKDIRHIER) $(@:d)
+    $(COPY) $< $@
+.ELSE
 $(ZIP1DIR)$/%.jar : $(SOLARBINDIR)$/%.jar
 	@@-$(MKDIRHIER) $(@:d)
     $(COPY) $< $@
+.ENDIF
 
 $(ZIP1DIR)$/%.jar : $(CLASSDIR)$/%.jar
 	@@-$(MKDIRHIER) $(@:d)
