Index: vcl/source/gdi/pdfwriter_impl.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/gdi/pdfwriter_impl.cxx,v
retrieving revision 1.114
retrieving revision 1.114.72.1
diff -u -p -u -p -r1.114 -r1.114.72.1
--- vcl/source/gdi/pdfwriter_impl.cxx	2 Aug 2007 15:00:04 -0000	1.114
+++ vcl/source/gdi/pdfwriter_impl.cxx	13 Oct 2007 08:26:07 -0000	1.114.72.1
@@ -5496,13 +5496,13 @@ void PDFWriterImpl::registerGlyphs( int 
             }
 
             pGlyphWidths[ i ] = 0;
-            pMappedGlyphs[ i ] = (sal_Int8)pUnicodes[i];
+            pMappedGlyphs[ i ] = sal::static_int_cast<sal_Int8>( pGlyphs[i] );
             pMappedFontObjects[ i ] = nFontID;
             const ImplPdfBuiltinFontData* pFD = GetPdfFontData( pCurrentFont );
             if( pFD )
             {
                 const BuiltinFont* pBuiltinFont = pFD->GetBuiltinFont();
-                pGlyphWidths[i] = pBuiltinFont->m_aWidths[ pUnicodes[i] & 0x00ff ];
+                pGlyphWidths[i] = pBuiltinFont->m_aWidths[ pGlyphs[i] & 0x00ff ];
             }
         }
         else if( pCurrentFont->mbSubsettable )
