Index: solenv/bin/modules/installer/windows/registry.pm
===================================================================
RCS file: /cvs/tools/solenv/bin/modules/installer/windows/registry.pm,v
retrieving revision 1.11
retrieving revision 1.11.44.1
diff -u -p -r1.11 -r1.11.44.1
--- solenv/bin/modules/installer/windows/registry.pm	26 Jul 2007 08:48:44 -0000	1.11
+++ solenv/bin/modules/installer/windows/registry.pm	12 Oct 2007 09:53:51 -0000	1.11.44.1
@@ -228,7 +228,7 @@ sub add_userregs_to_registry_table
 
 		$registry{'Registry'} = $onefile->{'userregkeypath'};
 		$registry{'Root'} = "1";  # always HKCU
-		$registry{'Key'} = "Software/$allvariables->{'MANUFACTURER'}/$allvariables->{'PRODUCTNAME'} $allvariables->{'PRODUCTVERSION'}/";
+		$registry{'Key'} = "Software\\$allvariables->{'MANUFACTURER'}\\$allvariables->{'PRODUCTNAME'} $allvariables->{'PRODUCTVERSION'}\\";
 		if ( $onefile->{'needs_user_registry_key'} ) { $registry{'Key'} = $registry{'Key'} . "StartMenu"; }
 		else { $registry{'Key'} = $registry{'Key'} . "ShellNew"; }
 		$registry{'Name'} = $onefile->{'Name'};
