Index: xmloff/source/text/txtflde.cxx
===================================================================
RCS file: /cvs/xml/xmloff/source/text/txtflde.cxx,v
retrieving revision 1.77
retrieving revision 1.77.30.1
diff -u -p -u -p -r1.77 -r1.77.30.1
--- xmloff/source/text/txtflde.cxx	3 Aug 2007 12:55:10 -0000	1.77
+++ xmloff/source/text/txtflde.cxx	9 Oct 2007 14:03:16 -0000	1.77.30.1
@@ -1236,12 +1236,14 @@ void XMLTextFieldExport::ExportFieldHelp
 					  GetStringProperty(sPropertyContent, rPropSet));
 		sal_Bool bCmd = GetBoolProperty(sPropertyIsShowFormula, rPropSet);
 		ProcessDisplay(sal_True, bCmd);
+        // #i81766# for older versions export of the value-type
+        sal_Bool bExportValueType = !bCmd && ( GetExport().getExportFlags() & EXPORT_SAVEBACKWARDCOMPATIBLE );
 		// show style, unless name will be shown
 		ProcessValueAndType(IsStringField(nToken, rPropSet),
 							GetIntProperty(sPropertyNumberFormat, rPropSet),
 							sEmpty, sEmpty, 0.0, // values not used
 							sal_False, 
-                            sal_False, //#148500# value_type not exported anymore
+                            bExportValueType,
                             !bCmd,
                             ! GetOptionalBoolProperty(
                                  sPropertyIsFixedLanguage,
Index: sw/source/core/docnode/ndtbl.cxx
===================================================================
RCS file: /cvs/sw/sw/source/core/docnode/ndtbl.cxx,v
retrieving revision 1.47
retrieving revision 1.47.88.1
diff -u -p -u -p -r1.47 -r1.47.88.1
--- sw/source/core/docnode/ndtbl.cxx	31 Jul 2007 17:40:51 -0000	1.47
+++ sw/source/core/docnode/ndtbl.cxx	25 Sep 2007 08:52:53 -0000	1.47.88.1
@@ -3558,6 +3558,7 @@ SwTableNode* SwNodes::SplitTable( const 
 		SwNodeIndex aIdx( *pBox->GetSttNd() );
 		new SwEndNode( aIdx, *pTNd );
 		pNewTblNd = new SwTableNode( aIdx );
+        pNewTblNd->GetTable().SetTableModel( rTbl.IsNewModel() );
 
 		pOldTblEndNd->pStartOfSection = pNewTblNd;
 		pNewTblNd->pEndOfSection = pOldTblEndNd;
Index: sw/source/core/edit/edtab.cxx
===================================================================
RCS file: /cvs/sw/sw/source/core/edit/edtab.cxx,v
retrieving revision 1.17
retrieving revision 1.17.206.1
diff -u -p -u -p -r1.17 -r1.17.206.1
--- sw/source/core/edit/edtab.cxx	22 May 2007 16:27:06 -0000	1.17
+++ sw/source/core/edit/edtab.cxx	25 Sep 2007 08:50:43 -0000	1.17.206.1
@@ -498,11 +498,13 @@ BOOL SwEditShell::CanMergeTable( BOOL bW
 	const SwTableNode* pTblNd = pCrsr->GetNode()->FindTableNode();
 	if( pTblNd && !pTblNd->GetTable().ISA( SwDDETable ))
 	{
+        bool bNew = pTblNd->GetTable().IsNewModel();
 		const SwNodes& rNds = GetDoc()->GetNodes();
 		if( pChkNxtPrv )
 		{
             const SwTableNode* pChkNd = rNds[ pTblNd->GetIndex() - 1 ]->FindTableNode();
 			if( pChkNd && !pChkNd->GetTable().ISA( SwDDETable ) &&
+                bNew == pChkNd->GetTable().IsNewModel() &&
                 // --> FME 2004-09-17 #117418# Consider table in table case
                 pChkNd->EndOfSectionIndex() == pTblNd->GetIndex() - 1 )
                 // <--
@@ -510,7 +512,8 @@ BOOL SwEditShell::CanMergeTable( BOOL bW
 			else
 			{
 				pChkNd = rNds[ pTblNd->EndOfSectionIndex() + 1 ]->GetTableNode();
-				if( pChkNd && !pChkNd->GetTable().ISA( SwDDETable ) )
+				if( pChkNd && !pChkNd->GetTable().ISA( SwDDETable ) &&
+                    bNew == pChkNd->GetTable().IsNewModel() )
 					*pChkNxtPrv = FALSE, bRet = TRUE;		// mit Next ist moeglich
 			}
 		}
@@ -529,7 +532,8 @@ BOOL SwEditShell::CanMergeTable( BOOL bW
             else
                 pTmpTblNd = rNds[ pTblNd->EndOfSectionIndex() + 1 ]->GetTableNode();
 
-            bRet = pTmpTblNd && !pTmpTblNd->GetTable().ISA( SwDDETable );
+            bRet = pTmpTblNd && !pTmpTblNd->GetTable().ISA( SwDDETable ) &&
+                   bNew == pTmpTblNd->GetTable().IsNewModel();
 		}
 	}
 	return bRet;
Index: sw/source/filter/ww8/wrtww8.cxx
===================================================================
RCS file: /cvs/sw/sw/source/filter/ww8/wrtww8.cxx,v
retrieving revision 1.83
retrieving revision 1.83.74.1
diff -u -p -u -p -r1.83 -r1.83.74.1
--- sw/source/filter/ww8/wrtww8.cxx	17 Jul 2007 13:09:18 -0000	1.83
+++ sw/source/filter/ww8/wrtww8.cxx	8 Oct 2007 12:08:56 -0000	1.83.74.1
@@ -2215,6 +2215,14 @@ ULONG SwWW8Writer::StoreDoc()
 
     PrepareStorage();
 
+    // --> OD 2007-10-08 #i81405#
+    // Collect anchored objects before changing the redline mode.
+    if (bWrtWW8)
+        maFrames = GetAllFrames(*pDoc, bWriteAll ? 0 : pOrigPam);
+    else
+        maFrames = GetNonDrawingFrames(*pDoc, bWriteAll ? 0 : pOrigPam);
+    // <--
+
     USHORT nRedlineMode = pDoc->GetRedlineMode();
     if (pDoc->GetRedlineTbl().Count())
     {
@@ -2332,13 +2340,6 @@ ULONG SwWW8Writer::StoreDoc()
     pDop->fRMView = 0 != ( IDocumentRedlineAccess::REDLINE_SHOW_DELETE & nRedlineMode );
     pDop->fRMPrint = pDop->fRMView;
 
-    // Tabelle fuer die freifliegenden Rahmen erzeugen, aber nur wenn
-    // das gesamte Dokument geschrieben wird
-    if (bWrtWW8)
-        maFrames = GetAllFrames(*pDoc, bWriteAll ? 0 : pOrigPam);
-    else
-        maFrames = GetNonDrawingFrames(*pDoc, bWriteAll ? 0 : pOrigPam);
-
     CollectOutlineBookmarks(*pDoc);
 
     // set AutoHyphenation flag if found in default para style
