Index: scp2/source/ooo/profileitem_ooo.scp
===================================================================
RCS file: /cvs/installation/scp2/source/ooo/profileitem_ooo.scp,v
retrieving revision 1.49.12.1
retrieving revision 1.49.12.1.4.1
diff -u -p -u -p -r1.49.12.1 -r1.49.12.1.4.1
--- scp2/source/ooo/profileitem_ooo.scp	24 Aug 2007 12:29:56 -0000	1.49.12.1
+++ scp2/source/ooo/profileitem_ooo.scp	26 Sep 2007 12:07:21 -0000	1.49.12.1.4.1
@@ -789,7 +789,7 @@ ProfileItem gid_Profileitem_Uno_Ure_Inte
     Section = "Bootstrap";
     Order = 11;
     Key = "URE_INTERNAL_JAVA_CLASSPATH";
-    Value = "$ORIGIN/classes/unoil.jar ${PKG_UserUnoFile:UNO_JAVA_CLASSPATH} ${PKG_SharedUnoFile:UNO_JAVA_CLASSPATH}";
+    Value = "$ORIGIN/classes/unoil.jar $ORIGIN/classes/ScriptFramework.jar ${PKG_UserUnoFile:UNO_JAVA_CLASSPATH} ${PKG_SharedUnoFile:UNO_JAVA_CLASSPATH}";
 End
 
 
