Index: scp2/source/base/file_base.scp
===================================================================
RCS file: /cvs/installation/scp2/source/base/file_base.scp,v
retrieving revision 1.22
retrieving revision 1.22.4.1
diff -u -u -r1.22 -r1.22.4.1
--- scp2/source/base/file_base.scp	21 Nov 2007 16:25:37 -0000	1.22
+++ scp2/source/base/file_base.scp	2 Dec 2007 12:34:16 -0000	1.22.4.1
@@ -64,6 +64,8 @@
 STD_RES_FILE_PATCH( gid_File_Res_Sdbcl, sdbcl)
 STD_RES_FILE_PATCH( gid_File_Res_Sdberr, sdberr)
 
+#ifdef ENABLE_REPORTDESIGN
+
 STD_UNO_LIB_FILE_PATCH( gid_File_Lib_Rpt, rpt )
 STD_RES_FILE_PATCH( gid_File_Res_Rpt, rpt)
 
@@ -72,6 +74,8 @@
 
 STD_UNO_LIB_FILE_PATCH( gid_File_Lib_Rptxml, rptxml )
 
+#endif
+
 #if defined UNX && ! defined MACOSX
 
 File gid_File_Share_Xdg_Base
Index: scp2/source/base/makefile.mk
===================================================================
RCS file: /cvs/installation/scp2/source/base/makefile.mk,v
retrieving revision 1.5
retrieving revision 1.5.820.2
diff -u -u -r1.5 -r1.5.820.2
--- scp2/source/base/makefile.mk	15 Nov 2005 12:54:19 -0000	1.5
+++ scp2/source/base/makefile.mk	4 Dec 2007 20:59:44 -0000	1.5.820.2
@@ -45,6 +45,10 @@
 
 .INCLUDE :	settings.mk
 
+.IF "$(ENABLE_REPORTDESIGN)" == "YES"
+SCPDEFS += -DENABLE_REPORTDESIGN
+.ENDIF
+
 SCP_PRODUCT_TYPE=osl
 
 PARFILES= \
