Index: source/ui/app/localize.sdf
===================================================================
RCS file: /cvs/graphics/sd/source/ui/app/localize.sdf,v
retrieving revision 1.30.40.1
retrieving revision 1.30.40.1.16.1
diff -u -u -r1.30.40.1 -r1.30.40.1.16.1
--- sd/source/ui/app/localize.sdf	2 Feb 2007 10:34:36 -0000	1.30.40.1
+++ sd/source/ui/app/localize.sdf	15 May 2007 10:26:47 -0000	1.30.40.1.16.1
@@ -22030,6 +22030,83 @@
 sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE	SID_INSERTPAGE		0	zh-CN	插入幻灯片(~I)				2002-02-02 02:02:02
 sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE	SID_INSERTPAGE		0	zh-TW	插入投影片(~I)				2002-02-02 02:02:02
 sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE	SID_INSERTPAGE		0	zu	~Faka Islaydi				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	af	~Voeg skyfie in!				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	ar	~Insert Slide				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	as	~Insert Slide				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	be-BY	Уставіць слайд				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	bg	~Вмъкване на кадър				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	bn	~Insert Slide				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	br	~Enlakaat un dreyonenn				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	bs	~Ubaci slajd				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	ca	~Insereix una diapositiva				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	cs	Vložit snímek				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	cy	Mewnosod Tryloyw				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	da	~Indsæt dias				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	el	Ε~ισαγωγή διαφάνειας				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	en-GB	~Insert Slide				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	en-ZA	~Insert Slide				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	eo	~Insert Slide				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	es	~Insertar diapositiva				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	et	Lisa slaid				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	eu	~Insert Slide				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	fa	~درج اسلاید				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	fi	~Lisää dia				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	fr	~Insérer une diapo				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	ga	~Ionsáigh Sleamhnán				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	gl-ES	~Insert Slide				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	gu-IN	સ્લાઈડ ઉમેરો (~I)				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	he	הכנסת שקף				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	hi-IN	जोड़ें स्लाइड (~I)				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	hr	~Umetni slajd				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	hu	~Dia beszúrása				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	it	~Inserisci diapositiva				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	ja	スライドの挿入(~I)				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	ka	~სლაიდების ჩასმა				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	km	បញ្ចូល​​ស្លាយ 				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	kn	~Insert Slide				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	ko	슬라이드 삽입(~I)				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	ku	Slaytan ~Lê Zêde Bike				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	lt	Įterpti ~lapą				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	mk	~Вметни слајд				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	ml	~Insert Slide				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	mr	~Insert Slide				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	nb	Sett ~inn lysbilde				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	ne	स्लाईड घुसाउनुहोस्				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	nl	Dia ~invoegen				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	nn	Set ~inn lysbilete				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	nr	~Faka Isilayidi				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	ns	~Tsenya Selaete				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	or	~Insert Slide				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	pa-IN	ਸਲਾਈਡ ਸ਼ਾਮਲ(~i)				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	pl	~Wstaw slajd				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	pt	~Inserir diapositivo				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	pt-BR	Inserir página				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	ru	Вставить слайд				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	rw	Kongeramo Igice				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	sh-YU	U~metni slajd				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	sk	Vlož~iť snímok				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	sl	~Vstavi prosojnico				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	sr-CS	У~метни слајд				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	ss	~Faka Bomsheleletane				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	st	~Kenya Selaete				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	sv	Infoga sida				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	sw-TZ	~Insert Slide				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	ta	~Insert Slide				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	te	~Insert Slide				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	tg	Иловакунии слайд				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	th	แ~ทรกภาพนิ่ง				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	ti	~Insert Slide				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	tn	~Insert Slide				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	tr	Slayt ~Ekle				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	ts	~Nghenisa Slayidi				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	uk	~Insert Slide				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	ur	~Insert Slide				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	ve	~Longelani Siḽaidi				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	vi	Chèn ảnh ch~iếu				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	xh	~Faka Islayidi				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	zh-CN	插入幻灯片(~I)				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	zh-TW	插入投影片(~I)				2002-02-02 02:02:02
+sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_INSERTPAGE_LAYOUT_MENU	SID_INSERTPAGE		0	zu	~Faka Islaydi				2002-02-02 02:02:02
 sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_TP_APPLY_TO_SELECTED_SLIDES	SID_TP_APPLY_TO_SELECTED_SLIDES		0	af	Pas op ~geselekteerde skyfies toe				2002-02-02 02:02:02
 sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_TP_APPLY_TO_SELECTED_SLIDES	SID_TP_APPLY_TO_SELECTED_SLIDES		0	as-IN	নিৰ্বাচিত শ্লাইডবোৰলৈ প্ৰয়োগ কৰক				2002-02-02 02:02:02
 sd	source\ui\app\popup.src	0	menuitem	RID_TASKPANE_LAYOUTMENU_POPUP	SID_TP_APPLY_TO_SELECTED_SLIDES	SID_TP_APPLY_TO_SELECTED_SLIDES		13691	be-BY	Ужыць над азначанымі слайдамі				2002-02-02 02:02:02
