Index: config_office/configure.in
===================================================================
RCS file: /cvs/tools/config_office/configure.in,v
retrieving revision 1.232
retrieving revision 1.221.4.23
diff -u -u -r1.232 -r1.221.4.23
--- config_office/configure.in	26 Nov 2007 18:03:39 -0000	1.232
+++ config_office/configure.in	4 Dec 2007 23:46:26 -0000	1.221.4.23
@@ -3858,15 +3923,15 @@
    SYSTEM_ICU=YES
    AC_MSG_CHECKING([for unicode/rbbi.h])
    AC_TRY_CPP(unicode/rbbi.h, AC_MSG_RESULT([checked.]), AC_MSG_ERROR([icu headers not found.]))
-   AC_PATH_PROG(SYSTEM_GENBRK, genbrk)
+   AC_PATH_PROG(SYSTEM_GENBRK, genbrk, [], [$PATH:/usr/sbin:/sbin])
    if test -z "$SYSTEM_GENBRK"; then
       AC_MSG_ERROR([\"genbrk\" not found in \$PATH, install the icu development tool \"genbrk"\])
    fi
-   AC_PATH_PROG(SYSTEM_GENCCODE, genccode)
+   AC_PATH_PROG(SYSTEM_GENCCODE, genccode, [], [$PATH:/usr/sbin:/sbin])
    if test -z "$SYSTEM_GENCCODE"; then
       AC_MSG_ERROR([\"genccode\" not found in \$PATH, install the icu development tool \"genccode"\])
    fi
-   AC_PATH_PROG(SYSTEM_GENCMN, gencmn)
+   AC_PATH_PROG(SYSTEM_GENCMN, gencmn, [], [$PATH:/usr/sbin:/sbin])
    if test -z "$SYSTEM_GENCMN"; then
       AC_MSG_ERROR([\"gencmn\" not found in \$PATH, install the icu development tool \"gencmn"\])
    fi
@@ -3875,13 +3940,14 @@
 #include <unicode/uversion.h>
 
 int main(int argc, char **argv) {
-      if(U_ICU_VERSION_MAJOR_NUM == 3 && U_ICU_VERSION_MINOR_NUM == 6) return 0;
+      if(U_ICU_VERSION_MAJOR_NUM == 3 && (U_ICU_VERSION_MINOR_NUM == 6 ||
+      	U_ICU_VERSION_MINOR_NUM == 8)) return 0;
       else return 1;
 }
       ], [AC_MSG_RESULT(OK)], [AC_MSG_RESULT([not suitable, only 3.6 supported currently])])
      AC_MSG_CHECKING(whether setBreakType is public)
-     save_LDFLAGS="$LDFLAGS"
-     LDFLAGS="$LDFLAGS -licuuc"
+     save_LIBS="$LIBS"
+     LIBS="$LIBS -licuuc"
 	   AC_TRY_LINK([#include <unicode/rbbi.h>], [
 	   RuleBasedBreakIterator *rbi = NULL;
 	   rbi = new RuleBasedBreakIterator();
@@ -3892,7 +3958,7 @@
       else
         AC_MSG_RESULT(yes)
       fi
-      LDFLAGS="$save_LDFLAGS"
+      LIBS="$save_LIBS"
 else
     AC_MSG_RESULT([internal])
     SYSTEM_ICU=NO
