Index: vcl/source/window/dialog.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/window/dialog.cxx,v
retrieving revision 1.40
retrieving revision 1.40.48.1
diff -u -p -u -p -r1.40 -r1.40.48.1
--- vcl/source/window/dialog.cxx	24 Jul 2007 10:18:51 -0000	1.40
+++ vcl/source/window/dialog.cxx	11 Sep 2007 15:18:33 -0000	1.40.48.1
@@ -508,7 +508,7 @@ long Dialog::Notify( NotifyEvent& rNEvt 
             // make sure the dialog is still modal
             // changing focus between application frames may
             // have re-enabled input for our parent
-            if( mbInExecute )
+            if( mbInExecute && mbModalMode )
             {
                 // do not change modal counter (pSVData->maAppData.mnModalDialog)
                 SetModalInputMode( FALSE );
