Index: svx/source/dialog/srchdlg.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/dialog/srchdlg.cxx,v
retrieving revision 1.44
retrieving revision 1.44.128.1
diff -u -p -u -p -r1.44 -r1.44.128.1
--- svx/source/dialog/srchdlg.cxx	6 Jul 2007 12:15:06 -0000	1.44
+++ svx/source/dialog/srchdlg.cxx	9 Oct 2007 13:14:58 -0000	1.44.128.1
@@ -93,6 +93,7 @@
 #ifndef _COM_SUN_STAR_BEANS_PROPERTYVALUE_HPP_
 #include <com/sun/star/beans/PropertyValue.hpp>
 #endif
+#include <com/sun/star/frame/XModuleManager.hpp>
 #ifndef _COMPHELPER_PROCESSFACTORY_HXX_
 #include <comphelper/processfactory.hxx>
 #endif
@@ -820,12 +821,30 @@ void SvxSearchDialog::CalculateDelta_Imp
 {
     DBG_ASSERT( pSearchItem, "no search item" );
 
+    bool bDrawApp = false;
+    bool bCalcApp = false;
+    const uno::Reference< frame::XFrame > xFrame = rBindings.GetActiveFrame();
+    uno::Reference< frame::XModuleManager > xModuleManager( 
+        ::comphelper::getProcessServiceFactory()->createInstance(
+            DEFINE_CONST_UNICODE("com.sun.star.frame.ModuleManager") ), uno::UNO_QUERY );
+    if ( xModuleManager.is() )
+    {
+        try
+        {
+            ::rtl::OUString aModuleIdentifier = xModuleManager->identify( xFrame );
+            bCalcApp = aModuleIdentifier.equalsAscii( "com.sun.star.sheet.SpreadsheetDocument" );
+            bDrawApp = aModuleIdentifier.equalsAscii( "com.sun.star.drawing.DrawingDocument" );
+        }
+        catch ( uno::Exception& )
+        {
+        }
+    }
+    
     if ( pImpl->bDeltaCalculated )
         return;
     else
         pImpl->bDeltaCalculated = TRUE;
 
-    bool bDrawApp = ( pSearchItem->GetAppFlag() == SVX_SEARCHAPP_DRAW );
     pMoreBtn->AddWindow( &aOptionsFL );
     if ( !bDrawApp )
         pMoreBtn->AddWindow( &aLayoutBtn );
@@ -889,7 +908,7 @@ void SvxSearchDialog::CalculateDelta_Imp
         }
     }
 
-    if ( pSearchItem->GetAppFlag() == SVX_SEARCHAPP_CALC )
+    if ( bCalcApp )
     {
         Window* pWins[] =
         {
