Index: wizards/com/sun/star/wizards/db/TypeInspector.java
===================================================================
RCS file: /cvs/installation/wizards/com/sun/star/wizards/db/TypeInspector.java,v
retrieving revision 1.7
retrieving revision 1.7.16.1
diff -u -p -u -p -r1.7 -r1.7.16.1
--- wizards/com/sun/star/wizards/db/TypeInspector.java	10 May 2007 10:53:42 -0000	1.7
+++ wizards/com/sun/star/wizards/db/TypeInspector.java	19 Sep 2007 08:33:57 -0000	1.7.16.1
@@ -382,8 +382,7 @@ public class TypeInspector{
                     if (this.bisAutoIncrementableInfos[i] )
                         return new TypeInfo(nDataType, this.sDataTypeNames[i], true);
                 }
-                else
-                    startindex = i+1;
+                startindex = i+1;
             }
         }
         // As Autoincrementation is not supported for any numeric datatype we take the first available numeric Type;
