Index: lingucomponent/source/hyphenator/altlinuxhyph/hyphen/makefile.mk
===================================================================
RCS file: /cvs/whiteboard/lingucomponent/source/hyphenator/altlinuxhyph/hyphen/makefile.mk,v
retrieving revision 1.15
retrieving revision 1.15.16.2
diff -u -u -r1.15 -r1.15.16.2
--- lingucomponent/source/hyphenator/altlinuxhyph/hyphen/makefile.mk	20 Sep 2007 14:31:18 -0000	1.15
+++ lingucomponent/source/hyphenator/altlinuxhyph/hyphen/makefile.mk	4 Dec 2007 20:59:44 -0000	1.15.16.2
@@ -41,7 +41,11 @@
 USE_DEFFILE=TRUE
 
 .IF "$(GUI)"=="UNX"
+.IF "$(SYSTEM_HYPH)" == "YES"
+HNJLIB=$(HYPHEN_LIB)
+.ELSE
 HNJLIB=-lhnj
+.ENDIF
 .ELSE
 HNJLIB=libhnj.lib
 .ENDIF
