Index: sw/source/ui/app/docsh2.cxx
===================================================================
RCS file: /cvs/sw/sw/source/ui/app/docsh2.cxx,v
retrieving revision 1.96
retrieving revision 1.96.68.1
diff -u -p -u -p -r1.96 -r1.96.68.1
--- sw/source/ui/app/docsh2.cxx	17 Jul 2007 13:10:23 -0000	1.96
+++ sw/source/ui/app/docsh2.cxx	26 Oct 2007 10:44:56 -0000	1.96.68.1
@@ -1626,21 +1626,24 @@ void SwDocShell::FillClass( SvGlobalName
 void SwDocShell::SetModified( BOOL bSet )
 {
     SfxObjectShell::SetModified( bSet );
-	if( IsEnableSetModified() && !pDoc->IsInCallModified() )
-	{
-		EnableSetModified( FALSE );
+	if( IsEnableSetModified() )
+    {
+        if ( !pDoc->IsInCallModified() )
+        {
+            EnableSetModified( FALSE );
 
-		if( bSet )
-		{
-			BOOL bOld = pDoc->IsModified();
-			pDoc->SetModified();
-			if( !bOld )
-				pDoc->SetUndoNoResetModified();
-		}
-		else
-			pDoc->ResetModified();
+            if( bSet )
+            {
+                BOOL bOld = pDoc->IsModified();
+                pDoc->SetModified();
+                if( !bOld )
+                    pDoc->SetUndoNoResetModified();
+            }
+            else
+                pDoc->ResetModified();
 
-		EnableSetModified( TRUE );
+            EnableSetModified( TRUE );
+        }
 
 		UpdateChildWindows();
 		Broadcast(SfxSimpleHint(SFX_HINT_DOCCHANGED));
