--- jvmfwk/plugins/sunmajor/pluginlib/gnujre.cxx.old	2006-03-23 19:51:26.000000000 +0100
+++ jvmfwk/plugins/sunmajor/pluginlib/gnujre.cxx	2006-03-27 16:24:26.000000000 +0200
@@ -219,5 +219,11 @@
     return true;
 }
 
+/* OOo has to be restarted to preload libgcj */
+bool GnuInfo::needsRestart() const
+{
+    return true;
+}
+
 
 }
--- jvmfwk/plugins/sunmajor/pluginlib/gnujre.hxx.old	2005-09-07 21:29:54.000000000 +0200
+++ jvmfwk/plugins/sunmajor/pluginlib/gnujre.hxx	2006-03-27 16:25:23.000000000 +0200
@@ -53,6 +53,9 @@
 
     virtual bool initialize(
         std::vector<std::pair<rtl::OUString, rtl::OUString> > props);
+
+    /* OOo has to be restarted to preload libgcj */
+    virtual bool needsRestart() const;
 };
 
 }
--- desktop/scripts/soffice.sh.old	2006-02-10 11:53:40.000000000 +0100
+++ desktop/scripts/soffice.sh	2006-03-27 16:29:53.000000000 +0200
@@ -183,6 +183,36 @@
                 ;;
         esac
     fi
+    
+    # preload libgcj if gcj/gij JRE is selected and enabled
+    sd_user=`sed -n -e "s|^UserInstallation=\(.*\)$|\1|p" $sd_cwd/bootstraprc`
+    sd_user=${sd_user/\$SYSUSERCONFIG/$HOME}
+    sd_user=${sd_user/file\:\/\//}
+    #echo "UserDir = $sd_user"
+    
+    sd_hw_platform=`uname -i`
+    case $sd_hw_platform in
+	i386|x86_64) sd_user_java_conf=$sd_user/user/config/javasettings_Linux_x86.xml
+	    ;;
+	ppc*)        sd_user_java_conf=$sd_user/user/config/javasettings_Linux_PowerPC.xml
+	    ;;
+	*) sd_user_java_conf=
+    esac
+    #echo "UserDirJavaConf = $sd_user_java_conf"
+
+    if test -n "$sd_user_java_conf" -a -r "$sd_user_java_conf" ; then
+	if grep -q '<enabled.*true' $sd_user_java_conf &&
+	   grep -q '<vendor.*Free Software Foundation, Inc.' $sd_user_java_conf &&
+	   grep -q '<version.*1.4.2' $sd_user_java_conf &&
+	   test -L /usr/lib/libgcj.so.7 ; then
+	    sd_libgcj=/usr/\$LIB/`readlink /usr/lib/libgcj.so.7`
+	    if test -e "${sd_libgcj/\$LIB/lib}" ; then
+		#echo "Preloading $sd_libgcj ..."
+		LD_PRELOAD="$sd_libgcj $LD_PRELOAD"
+		export LD_PRELOAD
+	    fi
+	fi
+    fi
 fi
 
 # misc. environment variables
