--- solenv/bin/modules/installer/packagelist.pm
+++ solenv/bin/modules/installer/packagelist.pm
@@ -75,6 +75,8 @@
 	
 	@allpackages = ();
 	
+	my $moduleshash = get_module_hash($moduleslist);
+	
 	for ( my $i = 0; $i <= $#{$packagelist}; $i++ )
 	{
 		my $onepackage = ${$packagelist}[$i];
@@ -91,7 +93,8 @@
 
 		push(@allmodules, $onegid);
 		
-		get_children($moduleslist, $onegid, \@allmodules);
+		# get_children($moduleslist, $onegid, \@allmodules);
+		get_children_with_hash($moduleshash, $onegid, \@allmodules);
 		
 		$onepackage->{'allmodules'} = \@allmodules;
 
@@ -101,6 +104,49 @@
 	return \@allpackages;
 }
 
+###################################################
+# Creating a hash, that contains the module gids
+# as keys and the parentids as values
+###################################################
+
+sub get_module_hash
+{
+	my ($moduleslist) = @_;
+	
+	my %modulehash = ();
+	
+	for ( my $i = 0; $i <= $#{$moduleslist}; $i++ )
+	{
+		my $gid = ${$moduleslist}[$i]->{'gid'};
+		# Containing only modules with parent. Root modules can be ignored.
+		if ( ${$moduleslist}[$i]->{'ParentID'} ) { $modulehash{$gid} = ${$moduleslist}[$i]->{'ParentID'}; }
+	}
+	
+	return \%modulehash;
+}
+
+########################################################
+# Recursively defined procedure to order 
+# modules and directories
+########################################################
+
+sub get_children_with_hash
+{
+	my ($modulehash, $parentgid, $newitemorder) = @_;
+
+	foreach my $gid ( keys %{$modulehash} )
+	{
+		my $parent = $modulehash->{$gid};
+
+		if ( $parent eq $parentgid )
+		{
+			push(@{$newitemorder}, $gid);
+			my $parent = $gid;
+			get_children_with_hash($modulehash, $parent, $newitemorder);	# recursive!
+		}		
+	}
+}
+
 ########################################################
 # Recursively defined procedure to order 
 # modules and directories
