Index: htmlnum.cxx
===================================================================
RCS file: /cvs/sw/sw/source/filter/html/htmlnum.cxx,v
retrieving revision 1.27
retrieving revision 1.27.8.1
diff -u -u -r1.27 -r1.27.8.1
--- sw/source/filter/html/htmlnum.cxx	13 Jun 2008 10:41:03 -0000	1.27
+++ sw/source/filter/html/htmlnum.cxx	26 Jun 2008 13:39:00 -0000	1.27.8.1
@@ -359,6 +359,18 @@
 			if( !aPropInfo.bRightMargin )
 				aItemSet.ClearItem( RES_LR_SPACE );
 
+            // --> OD 2008-06-26 #i89812#
+            // Perform change to list style before calling <DoPositioning(..)>,
+            // because <DoPositioning(..)> may open a new context and thus may
+            // clear the <SwHTMLNumRuleInfo> instance hold by local variable <rInfo>.
+            if( bChangeNumFmt )
+            {
+                rInfo.GetNumRule()->Set( nLevel, aNumFmt );
+                pDoc->ChgNumRuleFmts( *rInfo.GetNumRule() );
+                bChangeNumFmt = sal_False;
+            }
+            // <--
+
 			DoPositioning( aItemSet, aPropInfo, pCntxt );
 
 			InsertAttrs( aItemSet, aPropInfo, pCntxt );
