Index: config_office/set_soenv.in
===================================================================
RCS file: /cvs/tools/config_office/set_soenv.in,v
retrieving revision 1.160
diff -u -r1.160 set_soenv.in
--- config_office/set_soenv.in	2 Nov 2007 15:23:26 -0000	1.160
+++ config_office/set_soenv.in	9 Nov 2007 12:58:11 -0000
@@ -83,7 +83,8 @@
      $LD_LIBRARY_PATH, $PATH, $SOLAREXTRAINC, $SOLAREXTRALIB, $SOLARLIB, 
      $SOLARINC, $LOCALINI, $PATHEXTRA, $FRAMEWORKSHOME, $COMEX, $PERL, 
      $COMP_ENV, $ILIB, $JAVAHOME, $PSDK_HOME, $DIRECTXSDK_LIB, $USE_NEW_SDK, $FRAME_HOME, 
-     $ATL_LIB, $ATL_INCLUDE, $MFC_LIB, $MFC_INCLUDE, $NO_HIDS, $TEMP, $COMMON_BUILD_TOOLS, $WIN_GREP, $WIN_FIND, $WIN_LS,
+     $USE_DIRECTX5, $ATL_LIB, $ATL_INCLUDE, $MFC_LIB, $MFC_INCLUDE, $NO_HIDS, $TEMP,
+     $COMMON_BUILD_TOOLS, $WIN_GREP, $WIN_FIND, $WIN_LS,
      $WIN_GNUCOPY, $WIN_TOUCH, $STLPORT4, $MOZILLA_VERSION, $MOZILLA_TOOLKIT, $PREBUILD_MOZAB,
 	 $PROEXT,
      $SYSTEM_PYTHON, $PYTHONPATH, $PYTHONHOME, $SYSTEM_MOZILLA, $EPM_FLAGS);
@@ -617,6 +618,7 @@
       $CSC_PATH       = PosixFormat('@CSC_PATH@');
       $PSDK_HOME      = PosixFormat('@PSDK_HOME@');
       $DIRECTXSDK_LIB = PosixFormat('@DIRECTXSDK_LIB@');
+      $USE_DIRECTX5   = "";
       $FRAME_HOME     = PosixFormat('@FRAME_HOME@');
       $USE_NEW_SDK    = "TRUE";
       $CPUNAME        = "INTEL";
@@ -1134,6 +1136,13 @@
                         $ps.'$COMPATH'.$LIB;
       if ( '@ENABLE_DIRECTX@' ne "" ) {
           $ILIB   .= $ps.$DIRECTXSDK_LIB;
+          if ( '@COMEX@' eq "10" ) {
+              # .NET2003 provides and compiles against DirectX 5 libraries and is
+              # hereby still Win9x compatible.
+              $ILIB .= $ps.'$COMPATH'.$ds.'PlatformSDK'.$LIB;
+              $USE_DIRECTX5 = "YES";
+          }
+
       }
    }
    elsif ( $COM eq "GCC" )
@@ -1452,6 +1461,11 @@
                          $I.'$COMPATH'.$ds."include";
       if ( '@ENABLE_DIRECTX@' ne "" ) {
           $SOLARINC   .= $I.PosixFormat('@DIRECTXSDK_HOME@').$INCLUDE;
+          if ( '@COMEX@' eq "10" ) {
+              # .NET2003 provides and compiles against DirectX 5 libraries and is
+              # hereby still Win9x compatible.
+              $SOLARINC .= $I.'$COMPATH'.$ds.'PlatformSDK'.$ds."include";
+          }
       }
    }
    elsif ( $COM eq "GCC" )
@@ -1754,6 +1768,7 @@
 {  ToFile( "CYGWIN_PATH",       $CYGWIN_PATH,       "e" );
    ToFile( "PSDK_HOME",         $PSDK_HOME,         "e" );
    ToFile( "DIRECTXSDK_LIB",    $DIRECTXSDK_LIB,    "e" );
+   ToFile( "USE_DIRECTX5",      $USE_DIRECTX5,      "e" );
    ToFile( "ATL_LIB",           $ATL_LIB,         "e" );
    if ( $MINGW ne "yes" )
    {  ToFile( "ATL_INCLUDE",       $ATL_INCLUDE,       "e" );
