--- javaunohelper/prj/build.lst.old	2006-12-14 15:59:01.000000000 +0100
+++ javaunohelper/prj/build.lst	2007-05-17 19:09:24.000000000 +0200
@@ -7,4 +7,4 @@
 jh	javaunohelper\com\sun\star\lib\uno\adapter	nmake	-	all	jh_adapter jh_libhelp jh_inc NULL
 jh	javaunohelper\com\sun\star\lib\uno\helper	nmake	-	all	jh_libhelp jh_inc NULL
 jh	javaunohelper\source						nmake	-	all	jh_source jh_inc NULL
-jh	javaunohelper\util							nmake	-	all	jh_util jh_helper jh_source jh_adapter jh_libhelp NULL
+jh	javaunohelper\util							nmake	-	all	jh_util jh_helper jh_comp jh_source jh_adapter jh_libhelp NULL
