--- jvmfwk/plugins/sunmajor/javaenvsetup/javaldx.cxx.ark	2005-12-21 14:57:59.000000000 +0000
+++ jvmfwk/plugins/sunmajor/javaenvsetup/javaldx.cxx	2005-12-21 14:58:36.000000000 +0000
@@ -128,11 +128,13 @@
     rtl::OUString sVendor3(RTL_CONSTASCII_USTRINGPARAM("Blackdown Java-Linux Team"));
     rtl::OUString sVendor4(RTL_CONSTASCII_USTRINGPARAM("Apple Computer, Inc."));
     rtl::OUString sVendor5(RTL_CONSTASCII_USTRINGPARAM("BEA Systems, Inc."));
+    rtl::OUString sVendor6(RTL_CONSTASCII_USTRINGPARAM("Free Software Foundation, Inc."));
     if ( ! (sVendor1.equals(pInfo->sVendor) == sal_True
             || sVendor2.equals(pInfo->sVendor) == sal_True
             || sVendor3.equals(pInfo->sVendor) == sal_True
             || sVendor4.equals(pInfo->sVendor) == sal_True
-            || sVendor5.equals(pInfo->sVendor) == sal_True))
+            || sVendor5.equals(pInfo->sVendor) == sal_True
+	    || sVendor6.equals(pInfo->sVendor) == sal_True))
         return 0;
     
     rtl::OString sPaths = getLD_LIBRARY_PATH(pInfo->arVendorData);
--- jvmfwk/plugins/sunmajor/pluginlib/gnujre.cxx.ark	2005-12-21 17:09:49.000000000 +0000
+++ jvmfwk/plugins/sunmajor/pluginlib/gnujre.cxx	2005-12-21 17:10:16.000000000 +0000
@@ -57,6 +57,8 @@
         "bin/gij",
 	"gij-4.0",
 	"bin/gij-4.0"
+	"gij-4.1",
+	"bin/gij-4.1"
     };
     *size = sizeof (ar) / sizeof (char*);
     return ar;
@@ -65,7 +67,8 @@
 char const* const* GnuInfo::getRuntimePaths(int * size)
 {
     static char const* ar[]= {
-          "/lib/libgcj.so.6"
+          "/lib/libgcj.so.7"
+        , "/lib/libgcj.so.6"
 #if 0   //unreliable
 	, "/lib/libgcj.so.5"
         , "/lib/libgcj.so.4"
