--- jvmfwk/plugins/sunmajor/pluginlib/gnujre.cxx.old	2006-11-20 14:42:50.000000000 +0100
+++ jvmfwk/plugins/sunmajor/pluginlib/gnujre.cxx	2006-11-20 19:55:15.000000000 +0100
@@ -58,7 +58,16 @@
 	"gij-4.1",
 	"bin/gij-4.1",
 	"gij-4.2",
-	"bin/gij-4.2",
+	"bin/gij-4.2"
+#if defined(__i386__)
+	,
+	"gij32",
+	"bin/gij32",
+	"gij32-4.1",
+	"bin/gij32-4.1",
+	"gij32-4.2",
+	"bin/gij32-4.2"
+#endif
     };
     *size = sizeof (ar) / sizeof (char*);
     return ar;
