--- svtools/inc/svtools/miscopt.hxx.orig	2008-05-08 08:58:57.000000000 +0200
+++ svtools/inc/svtools/miscopt.hxx	2008-05-08 09:12:01.000000000 +0200
@@ -160,6 +160,11 @@
         sal_Bool    IsModifyByPrinting() const;
         void        SetModifyByPrinting(sal_Bool bSet );
 
+        sal_Bool    ShowLinkWarningDialog() const;
+        void        SetShowLinkWarningDialog( sal_Bool bSet );
+        sal_Bool    IsShowLinkWarningDialogReadOnly() const;
+ 
+
     //-------------------------------------------------------------------------------------------------------------
 	//	private methods
 	//-------------------------------------------------------------------------------------------------------------
--- svtools/source/config/miscopt.cxx.orig	2008-05-08 08:58:57.000000000 +0200
+++ svtools/source/config/miscopt.cxx	2008-05-08 09:27:56.000000000 +0200
@@ -114,8 +114,10 @@
 #define PROPERTYHANDLE_SYMBOLSTYLE			4
 #define PROPERTYNAME_DISABLEUICUSTOMIZATION	ASCII_STR("DisableUICustomization")
 #define PROPERTYHANDLE_DISABLEUICUSTOMIZATION			5
+#define PROPERTYNAME_SHOWLINKWARNINGDIALOG	ASCII_STR("ShowLinkWarningDialog")
+#define PROPERTYHANDLE_SHOWLINKWARNINGDIALOG 6
 
-#define PROPERTYCOUNT						6
+#define PROPERTYCOUNT						7
 
 #define VCL_TOOLBOX_STYLE_FLAT				((USHORT)0x0004) // from <vcl/toolbox.hxx>
 
@@ -143,6 +145,8 @@
     sal_Int16   m_nToolboxStyle;
     sal_Bool    m_bIsToolboxStyleRO;
     sal_Bool    m_bDisableUICustomization;
+    sal_Bool    m_bShowLinkWarningDialog;
+    sal_Bool    m_bIsShowLinkWarningDialogRO;
 
     //-------------------------------------------------------------------------------------------------------------
 	//	public methods
@@ -253,6 +257,15 @@
         inline sal_Bool IsGetToolboxStyleReadOnly()
         { return m_bIsToolboxStyleRO; }
 
+        inline sal_Bool ShowLinkWarningDialog() const
+        { return m_bShowLinkWarningDialog; }
+
+        void SetShowLinkWarningDialog( sal_Bool bSet )
+        {  m_bShowLinkWarningDialog = bSet; SetModified(); }
+
+        sal_Bool IsShowLinkWarningDialogReadOnly() const
+        { return m_bIsShowLinkWarningDialogRO; }
+
         void AddListener( const Link& rLink );
         void RemoveListener( const Link& rLink );
         void CallListeners();
@@ -332,6 +345,8 @@
     , m_bIsSymbolsStyleRO( sal_False )
     , m_nToolboxStyle( 1 )
     , m_bIsToolboxStyleRO( sal_False )
+    , m_bShowLinkWarningDialog( sal_True )
+    , m_bIsShowLinkWarningDialogRO( sal_False )
 
 {
 	// Use our static list of configuration keys to get his values.
@@ -405,6 +420,16 @@
                 break;
             }
 	    
+            case PROPERTYHANDLE_SHOWLINKWARNINGDIALOG :
+            {
+                if( !(seqValues[nProperty] >>= m_bShowLinkWarningDialog) )
+                {
+                    DBG_ERROR("Wrong type of \"Misc\\ShowLinkWarningDialog\"!" );
+                }
+                m_bIsShowLinkWarningDialogRO = seqRO[nProperty];
+                break;
+            }
+
             case PROPERTYHANDLE_DISABLEUICUSTOMIZATION : 
             {
                 if( !(seqValues[nProperty] >>= m_bDisableUICustomization) )
@@ -495,6 +520,13 @@
                                                             }
                                                         }
                                                     break;
+            case PROPERTYHANDLE_SHOWLINKWARNINGDIALOG     :   {
+                                                            if( !(seqValues[nProperty] >>= m_bShowLinkWarningDialog) )
+                                                            {
+                                                                DBG_ERROR("Wrong type of \"Misc\\ShowLinkWarningDialog\"!" );
+                                                            }
+                                                        }
+                                                    break;
             case PROPERTYHANDLE_SYMBOLSTYLE         :   {
                                                             ::rtl::OUString aSymbolsStyle;
                                                             if( seqValues[nProperty] >>= aSymbolsStyle )
@@ -656,6 +688,13 @@
                 seqValues[nProperty] <<= m_bDisableUICustomization;
                 break;
             }
+
+            case PROPERTYHANDLE_SHOWLINKWARNINGDIALOG :
+            {
+                if ( !m_bIsShowLinkWarningDialogRO )
+                    seqValues[nProperty] <<= m_bShowLinkWarningDialog;
+                break;
+            }
         }
 	}
 	// Set properties in configuration.
@@ -675,7 +714,8 @@
         PROPERTYNAME_TOOLBOXSTYLE,
         PROPERTYNAME_USESYSTEMFILEDIALOG,
         PROPERTYNAME_SYMBOLSTYLE,
-        PROPERTYNAME_DISABLEUICUSTOMIZATION
+        PROPERTYNAME_DISABLEUICUSTOMIZATION,
+        PROPERTYNAME_SHOWLINKWARNINGDIALOG
 	};
 
 	// Initialize return sequence with these list ...
@@ -841,6 +881,21 @@
     return m_pDataContainer->IsGetToolboxStyleReadOnly();
 }
 
+sal_Bool SvtMiscOptions::ShowLinkWarningDialog() const
+{
+    return m_pDataContainer->ShowLinkWarningDialog();
+}
+
+void SvtMiscOptions::SetShowLinkWarningDialog( sal_Bool bSet )
+{
+    m_pDataContainer->SetShowLinkWarningDialog( bSet );
+}
+
+sal_Bool SvtMiscOptions::IsShowLinkWarningDialogReadOnly() const
+{
+    return m_pDataContainer->IsShowLinkWarningDialogReadOnly();
+}
+
 //*****************************************************************************************************************
 //	private method
 //*****************************************************************************************************************
