Index: desktop/menus/math.desktop
===================================================================
--- sysui/desktop/menus/math.desktop	(Revision 261918)
+++ sysui/desktop/menus/math.desktop	(Arbeitskopie)
@@ -7,4 +7,5 @@
 Exec=openoffice -math %U
 MimeType=application/vnd.oasis.opendocument.formula;application/vnd.sun.xml.math;application/vnd.stardivision.math;
 Name=%PRODUCTNAME Math
+GenericName=Formula Editor
 Comment=Create and edit scientific formulas and equations by using Math.
Index: desktop/menus/draw.desktop
===================================================================
--- sysui/desktop/menus/draw.desktop	(Revision 261918)
+++ sysui/desktop/menus/draw.desktop	(Arbeitskopie)
@@ -7,4 +7,5 @@
 Exec=openoffice -draw %U
 MimeType=application/vnd.oasis.opendocument.graphics;application/vnd.oasis.opendocument.graphics-template;application/vnd.sun.xml.draw;application/vnd.sun.xml.draw.template;application/vnd.stardivision.draw;
 Name=%PRODUCTNAME Draw
+GenericName=Drawing Program
 Comment=Create and edit drawings, flow charts, and logos by using Draw.
Index: desktop/share/launcher_genericname.ulf
===================================================================
--- sysui/desktop/share/launcher_genericname.ulf	(Revision 261918)
+++ sysui/desktop/share/launcher_genericname.ulf	(Arbeitskopie)
@@ -10,3 +10,8 @@
 [base]
 en-US = "Database Development"
 
+[math]
+en-US = "Formula Editor"
+
+[draw]
+en-US = "Drawing Program"
