Index: desktop/share/localize.sdf
===================================================================
--- sysui/desktop/share/localize.sdf	(Revision 261918)
+++ sysui/desktop/share/localize.sdf	(Arbeitskopie)
@@ -3102,6 +3102,8 @@
 sysui	desktop\share\launcher_genericname.ulf	0	LngText	writer				0	vi	Xử lý từ				2002-02-02 02:02:02
 sysui	desktop\share\launcher_genericname.ulf	0	LngText	writer				0	zh-CN	字处理器				2002-02-02 02:02:02
 sysui	desktop\share\launcher_genericname.ulf	0	LngText	writer				0	zh-TW	文書處理器				2002-02-02 02:02:02
+sysui	desktop\share\launcher_genericname.ulf	0	LngText	math				0	de	Formeleditor				2002-02-02 02:02:02
+sysui	desktop\share\launcher_genericname.ulf	0	LngText	draw				0	de	Zeichenprogramm				2002-02-02 02:02:02
 sysui	desktop\share\launcher_name.ulf	0	LngText	printeradmin				0	af	%PRODUCTNAME-drukkeradministrasie				2002-02-02 02:02:02
 sysui	desktop\share\launcher_name.ulf	0	LngText	printeradmin				0	ar	إدارة طابعات %PRODUCTNAME				2002-02-02 02:02:02
 sysui	desktop\share\launcher_name.ulf	0	LngText	printeradmin				0	as-IN	%PRODUCTNAME মুদ্ৰক প্ৰশাসন				2002-02-02 02:02:02
