--- config_office/configure.in.mozff	2006-12-03 15:47:12.000000000 +0100
+++ config_office/configure.in	2006-12-03 15:59:06.000000000 +0100
@@ -3364,7 +3364,7 @@
     SYSTEM_MOZILLA=YES
     AC_MSG_CHECKING([which Mozilla flavour to use])
     if test -n "$with_firefox" && test "$with_firefox" != "no"; then
-      MOZ_FLAVOUR=firefox
+      MOZ_FLAVOUR=mozilla-firefox
     elif test -n "$with_xulrunner" && test "$with_xulrunner" != "no"; then
       MOZ_FLAVOUR=xulrunner
     else
