Index: source/config/fragments/fcfg_writer.mk
===================================================================
RCS file: /cvs/framework/filter/source/config/fragments/fcfg_writer.mk,v
retrieving revision 1.14.20.1
diff -u -r1.14.20.1 fcfg_writer.mk
--- filter/source/config/fragments/fcfg_writer.mk	15 Aug 2008 10:33:45 -0000	1.14.20.1
+++ filter/source/config/fragments/fcfg_writer.mk	3 Sep 2008 14:16:49 -0000
@@ -33,9 +33,13 @@
     writer8 \
     writer_MS_Word_2003_XML \
     writer_MS_Word_2007_XML \
-    MediaWiki_File \
+    MediaWiki_File
+
+.IF "$(WITH_WRITER2LATEX)" != "NO"
+T4_WRITER += \
 	LaTeX \
 	BibTeX
+.ENDIF
 
 # -----------------------------------------------
 # count = 39
@@ -71,9 +75,13 @@
     writer_MediaWiki_File \
     writer_web_MediaWiki_File \
     MS_Word_2003_XML \
-    MS_Word_2007_XML \
+    MS_Word_2007_XML
+
+.IF "$(WITH_WRITER2LATEX)" != "NO"
+F4_WRITER += \
 	LaTeX_Writer \
 	BibTeX_Writer
+.ENDIF
 
 # -----------------------------------------------
 # count = 14
