--- officecfg/registry/schema/org/openoffice/Office/Common.xcs.old	2006-10-18 22:30:58.000000000 +0200
+++ officecfg/registry/schema/org/openoffice/Office/Common.xcs	2006-10-19 19:58:54.000000000 +0200
@@ -1382,7 +1382,7 @@
 						<author>MBA</author>
 						<desc>Specifies the templates originate from these folders and sub-folders.</desc>
 					</info>
-					<value oor:separator=":">$(insturl)/share/template/$(vlang):$(userurl)/template</value>
+					<value oor:separator=":">$(insturl)/share/template/$(vlang):$(insturl)/share/template/common:$(userurl)/template</value>
 				</prop>
 				<prop oor:name="UIConfig" oor:type="oor:string-list">
 					<info>
