Index: source/gdi/sallayout.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/gdi/sallayout.cxx,v
retrieving revision 1.78
diff -u -r1.78 sallayout.cxx
--- vcl/source/gdi/sallayout.cxx	3 May 2006 16:35:28 -0000	1.78
+++ vcl/source/gdi/sallayout.cxx	1 Aug 2006 15:25:19 -0000
@@ -264,7 +264,7 @@
     if( (0x0001 <= cChar) && (cChar <= 0x001F) )
         return true;
     // formatting characters
-    if( (0x200B <= cChar) && (cChar <= 0x200F) )
+    if( (0x200E <= cChar) && (cChar <= 0x200F) )
         return true;
     if( (0x2028 <= cChar) && (cChar <= 0x202E) )
         return true;
--- vcl/source/glyphs/gcach_layout.cxx	2006-08-08 10:18:47.000000000 +0100
+++ vcl/source/glyphs/gcach_layout.cxx	2006-08-08 10:20:49.000000000 +0100
@@ -388,6 +388,11 @@
 
 // -----------------------------------------------------------------------
 
+static bool lcl_CharIsJoiner(sal_Unicode cChar)
+{
+	return ((cChar == 0x200C) || (cChar == 0x200D));
+}
+
 bool IcuLayoutEngine::operator()( ServerFontLayout& rLayout, ImplLayoutArgs& rArgs )
 {
     LEUnicode* pIcuChars;
@@ -499,7 +504,13 @@
             if( !nGlyphIndex )
             {
                 if( nCharPos >= 0 )
+		{
                     rArgs.NeedFallback( nCharPos, bRightToLeft );
+		    if ( (nCharPos > 0) && lcl_CharIsJoiner(rArgs.mpStr[nCharPos-1]) )
+                    	rArgs.NeedFallback( nCharPos-1, bRightToLeft );
+		    else if ( (nCharPos + 1 < nRawRunGlyphCount) && lcl_CharIsJoiner(rArgs.mpStr[nCharPos+1]) )
+                    	rArgs.NeedFallback( nCharPos+1, bRightToLeft );
+		}
                 if( SAL_LAYOUT_FOR_FALLBACK & rArgs.mnFlags )
                     continue;
             }
