--- sd/source/ui/func/fupage.cxx	26 Nov 2007 17:01:36 -0000	1.29
+++ sd/source/ui/func/fupage.cxx	26 Apr 2008 14:31:03 -0000
@@ -342,7 +342,7 @@
 				pTempSet.reset( new SfxItemSet( mpDoc->GetPool(), XATTR_FILL_FIRST, XATTR_FILL_LAST, 0) );
 
 				pTempSet->Put( XFillStyleItem( XFILL_BITMAP ) );
-				pTempSet->Put( XFillBitmapItem( String(), XOBitmap(aGraphic) ) );
+				pTempSet->Put( XFillBitmapItem( String(RTL_CONSTASCII_USTRINGPARAM("background")), XOBitmap(aGraphic) ) );
 				pTempSet->Put( XFillBmpStretchItem( TRUE ));
 				pTempSet->Put( XFillBmpTileItem( FALSE ));
 			}
