--- vcl/source/glyphs/gcach_ftyp.cxx~	2008-06-16 21:00:08.000000000 +0200
+++ vcl/source/glyphs/gcach_ftyp.cxx	2008-06-16 21:01:50.000000000 +0200
@@ -2235,7 +2235,7 @@
     int nGlyphFlags;
     SplitGlyphFlags( *this, nGlyphIndex, nGlyphFlags );
 
-    FT_Int nLoadFlags = FT_LOAD_DEFAULT;
+    FT_Int nLoadFlags = FT_LOAD_DEFAULT | FT_LOAD_IGNORE_TRANSFORM;
 
 #ifdef FT_LOAD_TARGET_LIGHT
     // enable "light hinting" if available
--- vcl/unx/source/gdi/salgdi3.cxx~	2008-06-16 20:59:00.000000000 +0200
+++ vcl/unx/source/gdi/salgdi3.cxx	2008-06-16 21:02:59.000000000 +0200
@@ -998,7 +998,7 @@
     Display* pDisplay = GetXDisplay();
 
     cairo_surface_t *surface = rCairo.xlib_surface_create_with_xrender_format (pDisplay,
-        hDrawable_, DefaultScreenOfDisplay(pDisplay), pVisualFormat, 1, 1);
+         hDrawable_, DefaultScreenOfDisplay(pDisplay), pVisualFormat, SAL_MAX_INT16, SAL_MAX_INT16);
 
     /*
      * It might be ideal to cache surface and cairo context between calls and
