Index: psprint/source/printer/printerinfomanager.cxx
===================================================================
RCS file: /cvs/gsl/psprint/source/printer/printerinfomanager.cxx,v
retrieving revision 1.27
diff -u -r1.27 printerinfomanager.cxx
--- psprint/source/printer/printerinfomanager.cxx	9 Nov 2004 16:38:07 -0000	1.27
+++ psprint/source/printer/printerinfomanager.cxx	18 Mar 2005 11:52:20 -0000
@@ -122,22 +122,31 @@
 
 // -----------------------------------------------------------------
 
+static PrinterInfoManager* pManager = NULL;
+static bool                bInitialized = false;
+
+namespace psp {
+extern void CUPSManagerKick ();
+void CUPSManagerKick()
+{
+	if( !pManager )
+			pManager = CUPSManager::tryLoadCUPS();
+}
+} // end namespace psp
+
 PrinterInfoManager& PrinterInfoManager::get()
 {
-    static PrinterInfoManager* pManager = NULL;
-
     if( ! pManager )
     {
         pManager = CUPSManager::tryLoadCUPS();
         if( ! pManager )
             pManager = new PrinterInfoManager();
-
-        if( pManager )
-            pManager->initialize();
-#if OSL_DEBUG_LEVEL > 1
-        fprintf( stderr, "PrinterInfoManager::get create Manager of type %d\n", pManager->getType() );
-#endif
-    }
+	}
+	if( pManager && !bInitialized )
+	{
+		bInitialized = true;
+		pManager->initialize();
+	}
 
     return *pManager;
 }


Index: vcl/unx/source/app/saldata.cxx
===================================================================
RCS file: /cvs/gsl/vcl/unx/source/app/saldata.cxx,v
retrieving revision 1.39
diff -u -r1.39 saldata.cxx
--- vcl/unx/source/app/saldata.cxx	9 Nov 2004 16:47:56 -0000	1.39
+++ vcl/unx/source/app/saldata.cxx	18 Mar 2005 11:52:22 -0000
@@ -351,6 +351,10 @@
 // -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
 #include <pthread.h>
 
+namespace psp {
+extern void CUPSManagerKick ();
+}
+
 SalData::SalData()
 {
 	bNoExceptions_	= !!getenv( "SAL_NOSEGV" );
@@ -361,6 +365,9 @@
     m_pPlugin		= NULL;
 
 	hMainThread_	= pthread_self();
+
+	// could this be the ugliest hack ever ?
+	::psp::CUPSManagerKick ();
 }
 
 SalData::~SalData()
