Index: source/filter/ww8/ww8scan.cxx
===================================================================
RCS file: /cvs/sw/sw/source/filter/ww8/ww8scan.cxx,v
retrieving revision 1.132
diff -u -r1.132 ww8scan.cxx
--- sw/source/filter/ww8/ww8scan.cxx	25 May 2007 13:03:35 -0000	1.132
+++ sw/source/filter/ww8/ww8scan.cxx	12 Oct 2007 10:06:35 -0000
@@ -4349,13 +4349,15 @@
         pBkm = &aD[1];
         pEdn = &aD[2];
         pFtn = &aD[3];
-        pAnd = &aD[4];
 
-        pPcd = ( pBase->pPLCFx_PCD ) ? &aD[5] : 0;
+        pPcd = ( pBase->pPLCFx_PCD ) ? &aD[4] : 0;
         //pPcdA index == pPcd index + 1
-        pPcdA = ( pBase->pPLCFx_PCDAttrs ) ? &aD[6] : 0;
+        pPcdA = ( pBase->pPLCFx_PCDAttrs ) ? &aD[5] : 0;
+
+        pChp = &aD[6];
+
+        pAnd = &aD[7];
 
-        pChp = &aD[7];
         pPap = &aD[8];
         pSep = &aD[9];
 
