Index: svx/source/msfilter/msoleexp.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/msfilter/msoleexp.cxx,v
retrieving revision 1.20
diff -u -p -r1.20 msoleexp.cxx
--- svx/source/msfilter/msoleexp.cxx	27 Jun 2007 18:35:44 -0000	1.20
+++ svx/source/msfilter/msoleexp.cxx	14 Nov 2007 12:19:05 -0000
@@ -272,7 +272,12 @@ void SvxMSExportOLEObjects::ExportOLEObj
         aSeq[1].Name = ::rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( "FilterName" ) );
         aSeq[1].Value <<= ::rtl::OUString( pExpFilter->GetName() );
         uno::Reference < frame::XStorable > xStor( rObj->getComponent(), uno::UNO_QUERY );
-        xStor->storeToURL( ::rtl::OUString::createFromAscii( "private:stream" ), aSeq );
+        try
+        {
+            xStor->storeToURL( ::rtl::OUString::createFromAscii( "private:stream" ), aSeq );
+        }
+        catch( uno::Exception& ) {} // #TODO really handle exceptions - interactionalhandler etc. ? 
+        
         SotStorageRef xOLEStor = new SotStorage( pStream, TRUE );
         xOLEStor->CopyTo( &rDestStg );
         rDestStg.Commit();
