--- sc/inc/address.hxx.original	2008-05-29 20:39:04.000000000 +0200
+++ sc/inc/address.hxx	2008-06-03 20:44:01.000000000 +0200
@@ -98,8 +98,8 @@
 
 // The maximum values. Defines are needed for preprocessor checks in
 // bcaslot.cxx, otherwise type safe constants are preferred.
-#define MAXROWCOUNT_DEFINE 65536
-#define MAXCOLCOUNT_DEFINE 256
+#define MAXROWCOUNT_DEFINE 1048576
+#define MAXCOLCOUNT_DEFINE 16384
 
 // Count values
 const SCROW       MAXROWCOUNT    = MAXROWCOUNT_DEFINE;
