--- sc/source/ui/view/hdrcont.cxx.orig	2008-06-07 17:19:14.000000000 +0200
+++ sc/source/ui/view/hdrcont.cxx	2008-06-07 17:28:08.000000000 +0200
@@ -109,7 +109,7 @@
 
 	nWidth = nSmallWidth = aSize.Width();
 	nBigWidth = LogicToPixel( Size( GetTextWidth(
-		String::CreateFromAscii(RTL_CONSTASCII_STRINGPARAM("88888")) ), 0 ) ).Width() + 4;
+		String::CreateFromAscii(RTL_CONSTASCII_STRINGPARAM("8888888")) ), 0 ) ).Width() + 4;
 
 	SetBackground();	// sonst Probleme auf OS/2 !?!?!
 }
