diff -ur scp2.orig/prj/build.lst scp2/prj/build.lst
--- scp2.orig/prj/build.lst	2006-10-13 14:12:18.000000000 +0200
+++ scp2/prj/build.lst	2006-10-13 15:48:08.000000000 +0200
@@ -23,5 +23,5 @@
 cp    scp2\source\xsltfilter      nmake    -    all    cp_xsltfilter cp_langmacros NULL
 cp    scp2\source\winexplorerext  nmake    -    all    cp_winexplorerext cp_langmacros NULL
 cp    scp2\source\ure             nmake    -    all    cp_ure cp_langmacros NULL
-cp    scp2\source\onlineupdate    nmake    -    all    cp_update NULL
+cp    scp2\source\onlineupdate    nmake    -    all    cp_update cp_langmacros NULL
 cp    scp2\util                   nmake    -    all    cp_util cp_activex cp_calc cp_canvas cp_crashrep cp_draw cp_gnome cp_graphicfilter cp_impress cp_javafilter cp_kde cp_lingu cp_math cp_ooo cp_python cp_quickstart cp_testtool cp_writer cp_base cp_xsltfilter cp_winexplorerext cp_ure cp_sdkoo cp_update NULL
Only in scp2.orig/prj: .build.lst.swp
