--- sd/source/ui/animations/CustomAnimationCreateDialog.cxx	12 Mar 2008 11:33:03 -0000	1.12
+++ sd/source/ui/animations/CustomAnimationCreateDialog.cxx	17 Mar 2008 12:47:29 -0000	1.11.150.2
@@ -253,6 +253,8 @@ public:
 	bool getIsPreview() const;
 	void setIsPreview( bool bIsPreview );
 
+	bool select( const OUString& rsPresetId );
+
 private:
 	DECL_LINK( implSelectHdl, Control* );
 	DECL_LINK( implDoubleClickHdl, Control* );
@@ -545,13 +547,33 @@ void CustomAnimationCreateTabPage::setIs
 	mpCBXPReview->Check( bIsPreview ? TRUE : FALSE );
 }
 
+bool CustomAnimationCreateTabPage::select( const OUString& rsPresetId )
+{
+	USHORT nPos = mpLBEffects->GetEntryCount();
+	while( nPos-- )
+	{
+		void* pEntryData = mpLBEffects->GetEntryData( nPos );
+		if( pEntryData )
+		{
+			CustomAnimationPresetPtr& pPtr = *static_cast< CustomAnimationPresetPtr* >(pEntryData);
+			if( pPtr.get() && pPtr->getPresetId() == rsPresetId )
+			{
+				mpLBEffects->SelectEntryPos( nPos );
+				return true;
+			}
+		}
+	}
+
+	return false;
+}
+
 // --------------------------------------------------------------------
 
-CustomAnimationCreateDialog::CustomAnimationCreateDialog( Window* pParent, CustomAnimationPane* pPane, const std::vector< ::com::sun::star::uno::Any >& rTargets, bool bHasText  )
-:	TabDialog( pParent, SdResId( DLG_CUSTOMANIMATION_CREATE ) ),
-	mpPane( pPane ), 
-	mrTargets( rTargets ),
-	mfDuration( 2.0f )
+CustomAnimationCreateDialog::CustomAnimationCreateDialog( Window* pParent, CustomAnimationPane* pPane, const std::vector< ::com::sun::star::uno::Any >& rTargets, bool bHasText, const ::rtl::OUString& rsPresetId, double fDuration  )
+:	TabDialog( pParent, SdResId( DLG_CUSTOMANIMATION_CREATE ) )
+,	mpPane( pPane )
+,	mrTargets( rTargets )
+,	mfDuration( fDuration )
 {
 	mpTabControl = new TabControl( this, SdResId( 1 ) );
 	mpOKButton = new OKButton(this, SdResId( 1 ) ) ;
@@ -584,6 +606,19 @@ CustomAnimationCreateDialog::CustomAnima
 	mpTabControl->SetDeactivatePageHdl( LINK( this, CustomAnimationCreateDialog, implDeactivatePagekHdl ) );
 
     setPosition();
+
+	// select current preset if available
+	if( rsPresetId.getLength() != 0 )
+	{
+		for( sal_uInt16 i = ENTRANCE; i <= MOTIONPATH; i++ )
+		{
+			if( mpTabPages[i]->select( rsPresetId ) )
+			{
+				mpTabControl->SetCurPageId( RID_TP_CUSTOMANIMATION_ENTRANCE + i );
+				break;
+			}
+		}
+	}
 }
 
 CustomAnimationCreateDialog::~CustomAnimationCreateDialog()
--- sd/source/ui/animations/CustomAnimationCreateDialog.hxx	1 Aug 2007 11:08:21 -0000	1.7
+++ sd/source/ui/animations/CustomAnimationCreateDialog.hxx	12 Mar 2008 12:25:14 -0000	1.7.150.1
@@ -62,7 +62,7 @@ class CustomAnimationCreateDialog : publ
 {
 	friend class CustomAnimationCreateTabPage;
 public:
-	CustomAnimationCreateDialog( ::Window* pParent, CustomAnimationPane* pPane, const std::vector< ::com::sun::star::uno::Any >& rTargets, bool bHasText );
+	CustomAnimationCreateDialog( ::Window* pParent, CustomAnimationPane* pPane, const std::vector< ::com::sun::star::uno::Any >& rTargets, bool bHasText, const ::rtl::OUString& rsPresetId, double fDuration );
 	~CustomAnimationCreateDialog();
 
 	PathKind getCreatePathKind() const;
--- sd/source/ui/animations/CustomAnimationDialog.cxx	7 Mar 2008 16:25:36 -0000	1.20
+++ sd/source/ui/animations/CustomAnimationDialog.cxx	17 Mar 2008 12:47:26 -0000	1.19.76.2
@@ -2356,6 +2356,8 @@ private:
 	CheckBox	maCBXReverse;
 
 	const STLPropertySet* mpSet;
+
+	bool mbHasVisibleShapes;
 };
 
 CustomAnimationTextAnimTabPage::CustomAnimationTextAnimTabPage(Window* pParent, const ResId& rResId, const STLPropertySet* pSet)
@@ -2366,7 +2368,8 @@ CustomAnimationTextAnimTabPage::CustomAn
 	maMFGroupAuto( this, SdResId( MF_GROUP_AUTO ) ),
 	maCBXAnimateForm( this, SdResId( CBX_ANIMATE_FORM ) ),
 	maCBXReverse( this, SdResId( CBX_REVERSE ) ),
-	mpSet( pSet )
+	mpSet( pSet ),
+	mbHasVisibleShapes(true)
 {
 	FreeResource();
 
@@ -2379,6 +2382,9 @@ CustomAnimationTextAnimTabPage::CustomAn
 			maLBGroupText.SelectEntryPos( (USHORT)(nTextGrouping + 1) );
 	}
 
+	if( pSet->getPropertyState( nHandleHasVisibleShape ) != STLPropertyState_AMBIGUOUS )
+		pSet->getPropertyValue( nHandleHasVisibleShape ) >>= mbHasVisibleShapes;
+
 	if( pSet->getPropertyState( nHandleTextGroupingAuto ) != STLPropertyState_AMBIGUOUS )
 	{
 		double fTextGroupingAuto = 0.0;
@@ -2491,6 +2497,16 @@ void CustomAnimationTextAnimTabPage::upd
 	maCBXGroupAuto.Enable( nPos > 1 );
 	maMFGroupAuto.Enable( nPos > 1 );
 	maCBXReverse.Enable( nPos > 0 );
+
+	if( !mbHasVisibleShapes && nPos > 0 )
+	{
+		maCBXAnimateForm.Check(FALSE);
+		maCBXAnimateForm.Enable(FALSE);
+	}
+	else
+	{
+		maCBXAnimateForm.Enable(TRUE);
+	}
 }
 
 IMPL_LINK( CustomAnimationTextAnimTabPage, implSelectHdl, Control*, EMPTYARG )
@@ -2597,6 +2613,7 @@ STLPropertySet* CustomAnimationDialog::c
 	pSet->setPropertyDefaultValue( nHandleTrigger, aEmpty );
 
 	pSet->setPropertyDefaultValue( nHandleHasText, makeAny( sal_False ) );
+	pSet->setPropertyDefaultValue( nHandleHasVisibleShape, makeAny( sal_False ) );
 	pSet->setPropertyDefaultValue( nHandleTextGrouping, makeAny( (sal_Int32)-1 ) );
 	pSet->setPropertyDefaultValue( nHandleAnimateForm, makeAny( sal_True ) );
 	pSet->setPropertyDefaultValue( nHandleTextGroupingAuto, makeAny( (double)-1.0 ) );
--- sd/source/ui/animations/CustomAnimationDialog.hxx	12 Dec 2006 16:51:07 -0000	1.6
+++ sd/source/ui/animations/CustomAnimationDialog.hxx	13 Mar 2008 15:03:48 -0000	1.6.282.1
@@ -98,6 +98,8 @@ const sal_Int32 nHandleSoundEndAfterSlid
 
 const sal_Int32 nHandleCommand = 31;
 
+const sal_Int32 nHandleHasVisibleShape = 32;
+
 const sal_Int32 nPropertyTypeNone = 0;
 const sal_Int32 nPropertyTypeDirection = 1;
 const sal_Int32 nPropertyTypeSpokes = 2;
--- sd/source/ui/animations/CustomAnimationList.cxx	6 Jul 2007 13:11:11 -0000	1.12
+++ sd/source/ui/animations/CustomAnimationList.cxx	12 Mar 2008 13:14:10 -0000	1.12.168.1
@@ -569,14 +569,14 @@ void CustomAnimationList::KeyInput( cons
 	const int nKeyCode = rKEvt.GetKeyCode().GetCode();
 	switch( nKeyCode )
 	{
-		case KEY_DELETE:	mpController->onContextMenu( CM_REMOVE );	break;
-		case KEY_INSERT:	mpController->onContextMenu( CM_CREATE ); break;
+		case KEY_DELETE:	mpController->onContextMenu( CM_REMOVE ); return;
+		case KEY_INSERT:	mpController->onContextMenu( CM_CREATE ); return;
 		case KEY_SPACE:		
 			{
 				const Point aPos;
 				const CommandEvent aCEvt( aPos, COMMAND_CONTEXTMENU ); 
 				Command( aCEvt );
-				break;
+				return;
 			}
 			
 	}
--- sd/source/ui/animations/CustomAnimationPane.cxx	17 Aug 2007 15:33:28 -0000	1.29
+++ sd/source/ui/animations/CustomAnimationPane.cxx	17 Mar 2008 15:01:58 -0000	1.29.118.3
@@ -80,6 +80,8 @@
 #ifndef _COM_SUN_STAR_AWT_XWINDOW_HPP_
 #include <com/sun/star/awt/XWindow.hpp>
 #endif
+#include <com/sun/star/drawing/LineStyle.hpp>
+#include <com/sun/star/drawing/FillStyle.hpp>
 
 #ifndef _COMPHELPER_PROCESSFACTORY_HXX_
 #include <comphelper/processfactory.hxx>
@@ -201,23 +203,14 @@ using namespace ::com::sun::star::presen
 using namespace ::com::sun::star::text;
 
 using ::rtl::OUString;
-using ::com::sun::star::uno::UNO_QUERY;
-using ::com::sun::star::uno::UNO_QUERY_THROW;
-using ::com::sun::star::uno::Any;
-using ::com::sun::star::uno::makeAny;
-using ::com::sun::star::uno::Sequence;
-using ::com::sun::star::uno::Reference;
-using ::com::sun::star::uno::Exception;
+using namespace ::com::sun::star::uno;
+using namespace ::com::sun::star::drawing;
 using ::com::sun::star::view::XSelectionSupplier;
 using ::com::sun::star::view::XSelectionChangeListener;
 using ::com::sun::star::frame::XController;
 using ::com::sun::star::frame::XModel;
 using ::com::sun::star::beans::XPropertySet;
 using ::com::sun::star::beans::XPropertyChangeListener;
-using ::com::sun::star::drawing::XDrawView;
-using ::com::sun::star::drawing::XShape;
-using ::com::sun::star::drawing::XShapes;
-using ::com::sun::star::drawing::XDrawPage;
 using ::com::sun::star::container::XIndexAccess;
 using ::com::sun::star::container::XEnumerationAccess;
 using ::com::sun::star::container::XEnumeration;
@@ -422,6 +415,12 @@ void CustomAnimationPane::StateChanged( 
 		updateMotionPathTags();
 }
 
+void CustomAnimationPane::KeyInput( const KeyEvent& rKEvt )
+{
+	if( mpCustomAnimationList )
+		mpCustomAnimationList->KeyInput( rKEvt );
+}
+
 void CustomAnimationPane::addListener()
 {
 	Link aLink( LINK(this,CustomAnimationPane,EventMultiplexerListener) );
@@ -1362,6 +1361,36 @@ bool CustomAnimationPane::setProperty1Va
 	return bEffectChanged;
 }
 
+static sal_Bool hasVisibleShape( const Reference< XShape >& xShape )
+{
+	try
+	{
+		const OUString sShapeType( xShape->getShapeType() );
+
+		if( sShapeType.equalsAsciiL( RTL_CONSTASCII_STRINGPARAM("com.sun.star.presentation.TitleTextShape") ) ||
+			sShapeType.equalsAsciiL( RTL_CONSTASCII_STRINGPARAM("com.sun.star.presentation.OutlinerShape") ) ||
+			sShapeType.equalsAsciiL( RTL_CONSTASCII_STRINGPARAM("com.sun.star.presentation.SubtitleShape") ) ||
+			sShapeType.equalsAsciiL( RTL_CONSTASCII_STRINGPARAM("com.sun.star.drawing.TextShape") ) )
+		{
+			const OUString sFillStyle( RTL_CONSTASCII_USTRINGPARAM("FillStyle" ) );
+			const OUString sLineStyle( RTL_CONSTASCII_USTRINGPARAM("LineStyle" ) );
+			Reference< XPropertySet > xSet( xShape, UNO_QUERY_THROW );
+
+			FillStyle eFillStyle;
+			xSet->getPropertyValue( sFillStyle ) >>= eFillStyle;
+
+			::com::sun::star::drawing::LineStyle eLineStyle;
+			xSet->getPropertyValue( sLineStyle ) >>= eLineStyle;
+
+			return eFillStyle != FillStyle_NONE || eLineStyle != ::com::sun::star::drawing::LineStyle_NONE;
+		}
+	}
+	catch( Exception& e )
+	{
+		(void)e;
+	}
+	return sal_True;
+}
 
 STLPropertySet* CustomAnimationPane::createSelectionSet()
 {
@@ -1413,6 +1442,8 @@ STLPropertySet* CustomAnimationPane::cre
 		addValue( pSet, nHandlePresetId, makeAny( pEffect->getPresetId() ) );
 
 		addValue( pSet, nHandleHasText, makeAny( (sal_Bool)pEffect->hasText() ) );
+		
+		addValue( pSet, nHandleHasVisibleShape, Any( hasVisibleShape( pEffect->getTargetShape() ) ) );
 
 		Any aSoundSource;
 		if( pEffect->getAudio().is() )
@@ -1926,6 +1957,9 @@ void CustomAnimationPane::onChange( bool
 
 	// first create vector of targets for dialog preview
 	std::vector< Any > aTargets;
+	OUString sPresetId;
+	double fDuration = 2.0f;
+
 	if( bCreate )
 	{
 		// gather shapes from the selection
@@ -1992,15 +2026,22 @@ void CustomAnimationPane::onChange( bool
 		{
 			if( !bHasText || !(*aIter)->hasText() )
 				bHasText = false;
+
+			if( sPresetId.getLength() == 0 )
+			{
+				sPresetId = (*aIter)->getPresetId();
+				fDuration = (*aIter)->getDuration();
+			}
+
 			aTargets.push_back( (*aIter++)->getTarget() );
 		}
 	}
 
-	CustomAnimationCreateDialog* pDlg = new CustomAnimationCreateDialog( this, this, aTargets, bHasText );
+	CustomAnimationCreateDialog* pDlg = new CustomAnimationCreateDialog( this, this, aTargets, bHasText, sPresetId, fDuration );
 	if( pDlg->Execute() )
 	{
 		addUndo();
-		double fDuration = pDlg->getSelectedDuration();
+		fDuration = pDlg->getSelectedDuration();
 		CustomAnimationPresetPtr pDescriptor = pDlg->getSelectedPreset();
 		if( pDescriptor.get() )
 		{
--- sd/source/ui/animations/CustomAnimationPane.hxx	6 Jul 2007 13:11:36 -0000	1.6
+++ sd/source/ui/animations/CustomAnimationPane.hxx	12 Mar 2008 13:14:10 -0000	1.6.168.1
@@ -111,6 +111,7 @@ public:
 	// Control
 	virtual void Resize();
 	virtual void StateChanged( StateChangedType nStateChange );
+	virtual void KeyInput( const KeyEvent& rKEvt );
 
 	// ICustomAnimationListController
 	virtual void onSelect();
