Index: scripts/soffice.sh
===================================================================
RCS file: /cvs/framework/desktop/scripts/soffice.sh,v
retrieving revision 1.28.48.1
diff -u -r1.28.48.1 soffice.sh
--- desktop/scripts/soffice.sh	2 Jan 2008 14:47:34 -0000	1.28.48.1
+++ desktop/scripts/soffice.sh	23 Sep 2008 10:47:17 -0000
@@ -247,6 +247,18 @@
 fi
 export PATH
 
+# Fix for bug #mdv41228 #i89207
+# Remove this line:
+# <library:library library:name="WebWizard"
+xlink:href="file:///usr/lib/openoffice/share/basic/WebWizard/dialog.xlb/"
+xlink:type="simple" library:link="true" library:readonly="true"/>
+# from files below.
+# Since 2007.0 WebWizard does not exist anymore.
+dialog_xlc=~/.openoffice.org2/user/basic/dialog.xlc
+script_xlc=~/.openoffice.org2/user/basic/script.xlc
+if [[ -f $dialog_xlc && -f $script_xlc ]]; then
+    sed -i "s/.*library:name=\"WebWizard\".*openoffice.*//" $dialog_xlc $script_xlc
+fi
 
 # execute soffice binary
 "$sd_prog/$sd_binary" "$@" &
