--- soltools/HIDCompiler/hidclex.l	10 Apr 2008 17:56:30 -0000	1.7
+++ soltools/HIDCompiler/hidclex.l	5 May 2008 03:49:47 -0000
@@ -454,40 +454,43 @@
 }
 
 
 /* forward */
 int eat_comment();
 int eat_cpp_comment();
 
 /*===================================================*/
 //
 // '+' im identifier wg basic\source\classes\sb.src
 //
 // '<' im identifier wg sc subtdlg.src
 //
 // '&' im identifier wg scerror.src so2..nocode.src svxerr.src scwarngs.src
 
 
 //string		(\"[^"]*\") alter einfacher string ohne "
 
 %}
 
+/* i89012 */
+%option never-interactive
+
 simple		([^\n\"]*)
 %p 7000
 string		\"{simple}((((\\\\)*(\\\"))?){simple})*\"
 
 %x MACRO_STATE
 
 %x EXCLUDE_STATE
 resfilelist	    ([Ff][Ii][Ll][Ee][Ll][Ii][Ss][Tt])
 resstringlist	([Ss][Tt][Rr][Ii][Nn][Gg][Ll][Ii][Ss][Tt])
 resstring   	([Ss][Tt][Rr][Ii][Nn][Gg])
 
 identifier ([a-z_A-Z]+[a-z_A-Z<+&0-9]*)
 number		(([0-9]+)|(0x[0-9a-fA-F]+))
 residentifier ([Ii][Dd][Ee][Nn][Tt][Ii][Ff][Ii][Ee][Rr])
 w			([ \t\n]*)
 wspecial	([\\ \t\n]*)
 texttag		(([Tt][Ii][Tt][Ll][Ee])|([Tt][Ee][Xx][Tt])|([Mm][Ee][Ss][Ss][Aa][Gg][Ee]))
 qhelptag	(([Qq][Uu][Ii][Cc][Kk])?([Hh][Ee][Ll][Pp][Tt][Ee][Xx][Tt]))
 langtag		([a-zA-Z_]+)
 helptag		([Hh][Ee][Ll][Pp][Ii][Dd])
