diff -rup svx-bak/source/svdraw/svdedxv.cxx svx/source/svdraw/svdedxv.cxx
--- svx-bak/source/svdraw/svdedxv.cxx	2007-10-15 16:10:35.000000000 +0200
+++ svx/source/svdraw/svdedxv.cxx	2007-10-15 16:14:17.000000000 +0200
@@ -487,8 +487,9 @@ OutlinerView* SdrObjEditView::ImpMakeOut
         pOutlView->SetAnchorMode((EVAnchorMode)(pText->GetOutlinerViewAnchorMode()));
 		pTextEditOutliner->SetFixedCellHeight(((const SdrTextFixedCellHeightItem&)pText->GetMergedItem(SDRATTR_TEXT_USEFIXEDCELLHEIGHT)).GetValue());
     }
-    pOutlView->SetOutputArea(aTextEditArea);
+    // do update before setting output area so that aTextEditArea can be recalculated
     pTextEditOutliner->SetUpdateMode(TRUE);
+    pOutlView->SetOutputArea(aTextEditArea);
     ImpPaintOutlinerView(*pOutlView,NULL,TRUE);
     return pOutlView;
 }
