--- sw/source/filter/ww8/ww8par.cxx.save	2008-06-21 01:35:56.000000000 +0200
+++ sw/source/filter/ww8/ww8par.cxx	2008-06-21 01:36:04.000000000 +0200
@@ -3602,7 +3602,12 @@
 
         bool bInsertSection = aIter != aStart ? aIter->IsContinous() : false;
         bool bInsertPageDesc = !bInsertSection;
-        bool bProtected = !bUseEnhFields && SectionIsProtected(*aIter); // do we really  need this ?? I guess I have a different logic in editshell which disales this...
+        bool bProtected = SectionIsProtected(*aIter); // do we really  need this ?? I guess I have a different logic in editshell which disales this...
+	if (bUseEnhFields && mrReader.pWDop->fProtEnabled && aIter->IsNotProtected()) {
+	    // here we have the special case that the whole document is protected, with the execption of this section.
+	    // I want to address this when I do the section rework, so for the moment we disable the overall protection then...
+	    mrReader.rDoc.set(IDocumentSettingAccess::PROTECT_FORM, false );
+	}
 
         if (bInsertPageDesc)
         {
