--- svtools/source/filter.vcl/wmf/enhwmf.cxx.old	2006-10-11 11:24:33.000000000 +0200
+++ svtools/source/filter.vcl/wmf/enhwmf.cxx	2007-02-26 16:38:31.000000000 +0100
@@ -176,8 +176,8 @@
 static float GetSwapFloat( SvStream& rSt )
 {
 	float	fTmp;
-	sal_Int8* pPtr = (sal_Int8*)&fTmp + 3;
-	rSt >> *pPtr-- >> *pPtr-- >> *pPtr-- >> *pPtr;	// Little Endian <-> Big Endian switch
+	sal_Int8* pPtr = (sal_Int8*)&fTmp;
+	rSt >> pPtr[3] >> pPtr[2] >> pPtr[1] >> pPtr[0];	// Little Endian <-> Big Endian switch
 	return fTmp;
 }
 #endif
