--- scp2/util/makefile.mk	2006-03-23 14:12:18.593750000 +0200
+++ scp2/util/makefile.mk.new	2006-03-23 14:12:35.937500000 +0200
@@ -126,8 +126,6 @@ SCP1FILES +=                           \
 
 .IF "$(NETTOOLKIT)"==""
 SCP1FILES +=                           \
-             module_activex.par        \
-             file_activex.par          \
              module_winexplorerext.par \
              file_winexplorerext.par   \
              registryitem_winexplorerext.par
@@ -233,8 +231,6 @@ SCP2FILES +=                           \
 
 .IF "$(NETTOOLKIT)"==""
 SCP2FILES +=                           \
-             module_activex.par        \
-             file_activex.par          \
              module_winexplorerext.par \
              file_winexplorerext.par   \
              registryitem_winexplorerext.par
