--- wizards/source/formwizard/dbwizres.src.old	2005-01-13 20:57:39.000000000 +0100
+++ wizards/source/formwizard/dbwizres.src	2005-03-15 17:07:00.000000000 +0100
@@ -5688,7 +5688,7 @@
 String RID_DB_TABLE_WIZARD_START + 35
 {
     Text [ de ] = "Herzlichen Glückwunsch! Sie haben alle Informationen zum Erstellen Ihrer Tabelle eingegeben.";
-    Text [ en-US] = "Congratulations. You have entered all the information needed to create your table";
+    Text [ en-US] = "Congratulations. You have entered all the information needed to create your table.";
 };
 
 
--- wizards/com/sun/star/wizards/table/Finalizer.java.old	2004-10-27 15:38:12.000000000 +0200
+++ wizards/com/sun/star/wizards/table/Finalizer.java	2005-03-15 16:56:52.000000000 +0100
@@ -115,23 +115,23 @@
 		
 		CurUnoDialog.insertLabel("lblcongratulations", 
 		  new String[] {"Height", "Label", "MultiLine", "PositionX", "PositionY", "Step", "TabIndex", "Width"},
-		  new Object[] { UIConsts.INTEGERS[8], sCongratulations, Boolean.TRUE, new Integer(97), new Integer(62),IFINALSTEP, new Short(curtabindex++), new Integer(154)}
+		  new Object[] { new Integer(20), sCongratulations, Boolean.TRUE, new Integer(97), new Integer(62),IFINALSTEP, new Short(curtabindex++), new Integer(185)}
 		);
 		CurUnoDialog.insertLabel("lblProceed", 
 		  new String[] {"Height", "Label", "PositionX", "PositionY", "Step", "TabIndex", "Width"},
-		  new Object[] { UIConsts.INTEGERS[8], slblProceed, new Integer(97), new Integer(82),IFINALSTEP, new Short(curtabindex++), new Integer(154)}
+		  new Object[] { UIConsts.INTEGERS[8], slblProceed, new Integer(97), new Integer(82),IFINALSTEP, new Short(curtabindex++), new Integer(170)}
 		);
 		optWorkWithTable = CurUnoDialog.insertRadioButton("optWorkWithTable", null,
 		  new String[] {"Height", "HelpURL", "Label", "PositionX", "PositionY", "State", "Step", "TabIndex", "Width"},
-		  new Object[] { UIConsts.INTEGERS[8], "HID:41242", sWorkWithTable, new Integer(101), new Integer(97), new Short((short)1),IFINALSTEP, new Short(curtabindex++), new Integer(107)}
+		  new Object[] { UIConsts.INTEGERS[8], "HID:41242", sWorkWithTable, new Integer(101), new Integer(97), new Short((short)1),IFINALSTEP, new Short(curtabindex++), new Integer(160)}
 		);
 		optModifyTable = CurUnoDialog.insertRadioButton("optModifyTable", null,
 		  new String[] {"Height", "HelpURL", "Label", "PositionX", "PositionY", "Step", "TabIndex", "Width"},
-		  new Object[] { UIConsts.INTEGERS[8], "HID:41241", sModifyTable, new Integer(101), new Integer(109),IFINALSTEP, new Short(curtabindex++), new Integer(107)}
+		  new Object[] { UIConsts.INTEGERS[8], "HID:41241", sModifyTable, new Integer(101), new Integer(109),IFINALSTEP, new Short(curtabindex++), new Integer(160)}
 		);
 		optStartFormWizard = CurUnoDialog.insertRadioButton("optStartFormWizard", null,
 		  new String[] {"Height", "HelpURL", "Label", "PositionX", "PositionY", "Step", "TabIndex", "Width"},
-		  new Object[] { UIConsts.INTEGERS[8], "HID:41243", sStartFormWizard, new Integer(101), new Integer(121),IFINALSTEP, new Short(curtabindex++), new Integer(107)}
+		  new Object[] { UIConsts.INTEGERS[8], "HID:41243", sStartFormWizard, new Integer(101), new Integer(121),IFINALSTEP, new Short(curtabindex++), new Integer(160)}
 		);
 	}
 
