Index: xmloff/source/text/txtvfldi.cxx
===================================================================
RCS file: /cvs/xml/xmloff/source/text/txtvfldi.cxx,v
retrieving revision 1.33
retrieving revision 1.33.78.1
diff -c -r1.33 -r1.33.78.1
*** xmloff/source/text/txtvfldi.cxx	3 Aug 2007 12:56:03 -0000	1.33
--- xmloff/source/text/txtvfldi.cxx	5 Dec 2007 11:13:46 -0000	1.33.78.1
***************
*** 638,645 ****
  								 nPrfx, rLocalName, 
  								 // name, description, style
  								 sal_True, sal_False, sal_False,
                                   sal_False, sal_False,
- 								 sal_True, sal_False, sal_False,
  								 sal_False /*???*/, sal_True, sal_False,
  								 sal_False)
  {
--- 638,645 ----
  								 nPrfx, rLocalName, 
  								 // name, description, style
  								 sal_True, sal_False, sal_False,
+                                  sal_True, sal_False, sal_False,
                                   sal_False, sal_False,
  								 sal_False /*???*/, sal_True, sal_False,
  								 sal_False)
  {
