--- scp2/source/ooo/profileitem_ooo.scp.nopre	2005-12-30 13:44:58.000000000 +0100
+++ scp2/source/ooo/profileitem_ooo.scp	2005-12-30 13:44:58.000000000 +0100
@@ -186,7 +186,7 @@
   #ifdef WNT
     Value = "$SYSUSERCONFIG/%ONEWORDPRODUCTNAME%USERDIRPRODUCTVERSION";
   #else
-    Value = "$SYSUSERCONFIG/.%LCONEWORDPRODUCTNAME%USERDIRPRODUCTVERSION";
+    Value = "$SYSUSERCONFIG/.ooo-2.0";
   #endif
     StandaloneValue = "$ORIGIN/..";
 End
