diff --git a/solenv/bin/build.pl b/solenv/bin/build.pl
index c5cb123..6fddbf5 100755
--- solenv/bin/build.pl
+++ solenv/bin/build.pl
@@ -188,6 +188,10 @@
                             # the server considered as an error/client crash 
     my %lost_client_jobs = (); # hash containing lost jobs
     my %job_jobdir = (); # hash containing job-dir pairs
+
+    $unit_tests = 0;  # use tests.lst in addition to build.lst & build with 'dmake check' there - for unit testing
+    %unit_tests_dirs; # directories that should be built with 'dmake check'
+
 ### main ###
 
     get_options();
@@ -494,7 +498,11 @@ sub dmake_dir {
         };
         RemoveFromDependencies($BuildDir, \%LocalDepsHash) if (!$child);
         return if ($cmd_file || $show);
-        $error_code = run_job($dmake, $BuildDir);
+        my $dmake_command = $dmake;
+        if ($unit_tests && exists $unit_tests_dirs{$BuildDir} && $unit_tests_dirs{$BuildDir}) {
+            $dmake_command .= " check";
+        }
+        $error_code = run_job($dmake_command, $BuildDir);
 #        chdir $BuildDir;
 #        getcwd();
 #        if ($html) {
@@ -503,14 +511,14 @@ sub dmake_dir {
 #            if (!-d $log_dir) {
 #                 system("$perl $mkout");
 #            };
-#            $error_code = system ("$dmake > $log_file 2>&1");
-#           if (!$grab_output && -f $log_file) {
+#            $error_code = system ("$dmake $dmake_check > $log_file 2>&1");
+#            if (!$grab_output && -f $log_file) {
 #                system("cat $log_file");
 #            };
 #        } else {
-#            $error_code = system ("$dmake");
+#            $error_code = system ("$dmake $dmake_check");
 #        };
-        
+
         html_store_job_info(\%LocalDepsHash, $BuildDir, $error_code) if (!$child);
     };
     if ($error_code && $ignore) {
@@ -553,8 +561,27 @@ sub store_build_list_content {
     } else {
         if (open (BUILD_LST, $build_list_path)) {
             my @build_lst = <BUILD_LST>;
-            $build_lists_hash{$module} = \@build_lst;
             close BUILD_LST;
+
+            # add the content of tests.lst to @build_lst if we are issued
+            # with 'check'
+            if ($unit_tests) {
+                (my $tests_path = $build_list_path ) =~ s/build\.lst$/tests.lst/;
+                if ($tests_path ne $build_list_path && open (TESTS_LST, $tests_path)) {
+                    while (<TESTS_LST>) {
+                        s/#.*//;
+                        next if (!/\snmake\s/);
+
+                        push @build_lst, $_;
+                        if (/^\s*\S+\s+(\S+)/ ) {
+                            $unit_tests_dirs{CorrectPath($StandDir . $1)} = 1;
+                        }
+                    }
+                    close TESTS_LST;
+                }
+            }
+
+            $build_lists_hash{$module} = \@build_lst;
             return;
         }
         $dead_parents{$module}++;
@@ -1278,6 +1305,7 @@ sub get_options {
             push (@dmake_args, get_dmake_args()) if (!$custom_job);
             next;
         };
+        $arg =~ /^check$/       and $unit_tests = 1 and next; # it's a dmake target, but we strip it here to add it later
         push (@dmake_args, $arg);
     };
     if (!$html) {
@@ -1335,6 +1363,10 @@ sub get_dmake_args {
     my @job_args = (); 
     while ($arg = shift @ARGV) {
         next if ($arg =~ /^--$/);
+        if ($arg =~ /^check$/) {
+            $unit_tests = 1;
+            next; # it's a dmake target, but we strip it here to add it later
+        }
         push (@job_args, $arg);
     };
     return @job_args;
diff --git a/solenv/inc/_cppunit.mk b/solenv/inc/_cppunit.mk
index 7aab2b0..40f64ba 100644
--- solenv/inc/_cppunit.mk
+++ solenv/inc/_cppunit.mk
@@ -15,8 +15,8 @@ TEST1LIB=$(SHL1TARGETN)
 .ENDIF
 
 .IF "$(SHL1TARGET)"!=""
-test : test1
-test1: ALLTAR
+check : check1
+check1: ALLTAR
         @echo ----------------------------------------------------------
         @echo - start unit test \#1 on library $(TEST1LIB)
         @echo ----------------------------------------------------------
@@ -44,8 +44,8 @@ TEST2LIB=$(SHL2TARGETN)
 .ENDIF
 
 .IF "$(SHL2TARGET)"!=""
-test : test2
-test2: ALLTAR
+check : check2
+check2: ALLTAR
         @echo ----------------------------------------------------------
         @echo - start unit test \#2 on library $(TEST2LIB)
         @echo ----------------------------------------------------------
@@ -73,8 +73,8 @@ TEST3LIB=$(SHL3TARGETN)
 .ENDIF
 
 .IF "$(SHL3TARGET)"!=""
-test : test3
-test3: ALLTAR
+check : check3
+check3: ALLTAR
         @echo ----------------------------------------------------------
         @echo - start unit test \#3 on library $(TEST3LIB)
         @echo ----------------------------------------------------------
@@ -102,8 +102,8 @@ TEST4LIB=$(SHL4TARGETN)
 .ENDIF
 
 .IF "$(SHL4TARGET)"!=""
-test : test4
-test4: ALLTAR
+check : check4
+check4: ALLTAR
         @echo ----------------------------------------------------------
         @echo - start unit test \#4 on library $(TEST4LIB)
         @echo ----------------------------------------------------------
@@ -131,8 +131,8 @@ TEST5LIB=$(SHL5TARGETN)
 .ENDIF
 
 .IF "$(SHL5TARGET)"!=""
-test : test5
-test5: ALLTAR
+check : check5
+check5: ALLTAR
         @echo ----------------------------------------------------------
         @echo - start unit test \#5 on library $(TEST5LIB)
         @echo ----------------------------------------------------------
@@ -160,8 +160,8 @@ TEST6LIB=$(SHL6TARGETN)
 .ENDIF
 
 .IF "$(SHL6TARGET)"!=""
-test : test6
-test6: ALLTAR
+check : check6
+check6: ALLTAR
         @echo ----------------------------------------------------------
         @echo - start unit test \#6 on library $(TEST6LIB)
         @echo ----------------------------------------------------------
@@ -189,8 +189,8 @@ TEST7LIB=$(SHL7TARGETN)
 .ENDIF
 
 .IF "$(SHL7TARGET)"!=""
-test : test7
-test7: ALLTAR
+check : check7
+check7: ALLTAR
         @echo ----------------------------------------------------------
         @echo - start unit test \#7 on library $(TEST7LIB)
         @echo ----------------------------------------------------------
@@ -218,8 +218,8 @@ TEST8LIB=$(SHL8TARGETN)
 .ENDIF
 
 .IF "$(SHL8TARGET)"!=""
-test : test8
-test8: ALLTAR
+check : check8
+check8: ALLTAR
         @echo ----------------------------------------------------------
         @echo - start unit test \#8 on library $(TEST8LIB)
         @echo ----------------------------------------------------------
@@ -247,8 +247,8 @@ TEST9LIB=$(SHL9TARGETN)
 .ENDIF
 
 .IF "$(SHL9TARGET)"!=""
-test : test9
-test9: ALLTAR
+check : check9
+check9: ALLTAR
         @echo ----------------------------------------------------------
         @echo - start unit test \#9 on library $(TEST9LIB)
         @echo ----------------------------------------------------------
@@ -276,8 +276,8 @@ TEST10LIB=$(SHL10TARGETN)
 .ENDIF
 
 .IF "$(SHL10TARGET)"!=""
-test : test10
-test10: ALLTAR
+check : check10
+check10: ALLTAR
         @echo ----------------------------------------------------------
         @echo - start unit test \#10 on library $(TEST10LIB)
         @echo ----------------------------------------------------------
diff --git a/solenv/inc/cppunit.mk b/solenv/inc/cppunit.mk
index 3567463..89b5df5 100644
--- solenv/inc/cppunit.mk
+++ solenv/inc/cppunit.mk
@@ -69,8 +69,8 @@ TEST$(TNR)LIB=$(SHL$(TNR)TARGETN)
 .ENDIF
 
 .IF "$(SHL$(TNR)TARGET)"!=""
-test : test$(TNR)
-test$(TNR): ALLTAR
+check : check$(TNR)
+check$(TNR): ALLTAR
         @echo ----------------------------------------------------------
         @echo - start unit test \#$(TNR) on library $(TEST$(TNR)LIB)
         @echo ----------------------------------------------------------
diff --git a/solenv/inc/target.mk b/solenv/inc/target.mk
index 0264ac9..231ec2e 100644
--- solenv/inc/target.mk
+++ solenv/inc/target.mk
@@ -2109,3 +2109,10 @@ $(SUBDIRS) .PHONY :
 # if the previous block was a rule or a target, "\#" isn't recognized
 # as an escaped "#". if it was an assignment, escaping works...
 some_unique_variable_name:=1
+
+# ---------
+# - CHECK -
+# ---------
+# can be redefined in the makefiles to issue the real tests
+
+check: ALLTAR
